<!doctype html>
<html class="">
	<head>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no" />
		<meta name="description" content="We’re on a journey to advance and democratize artificial intelligence through open source and open science." />
		<meta property="fb:app_id" content="1321688464574422" />
		<meta name="twitter:card" content="summary_large_image" />
		<meta name="twitter:site" content="@huggingface" />
		<meta name="twitter:image" content="https://cdn-thumbnails.huggingface.co/social-thumbnails/datasets/griffin/ChemSum.png" />
		<meta property="og:title" content="griffin/ChemSum · Datasets at Hugging Face" />
		<meta property="og:type" content="website" />
		<meta property="og:url" content="https://huggingface.co/datasets/griffin/ChemSum" />
		<meta property="og:image" content="https://cdn-thumbnails.huggingface.co/social-thumbnails/datasets/griffin/ChemSum.png" />

		<link rel="stylesheet" href="/front/build/kube-68d7aa0/style.css" />

		<link rel="preconnect" href="https://fonts.gstatic.com" />
		<link
			href="https://fonts.googleapis.com/css2?family=Source+Sans+Pro:ital,wght@0,200;0,300;0,400;0,600;0,700;0,900;1,200;1,300;1,400;1,600;1,700;1,900&display=swap"
			rel="stylesheet"
		/>
		<link
			href="https://fonts.googleapis.com/css2?family=IBM+Plex+Mono:wght@400;600;700&display=swap"
			rel="stylesheet"
		/>

		<link
			rel="preload"
			href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.12.0/katex.min.css"
			as="style"
			onload="this.onload=null;this.rel='stylesheet'"
		/>
		<noscript>
			<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.12.0/katex.min.css" />
		</noscript>

		<script>const guestTheme = document.cookie.match(/theme=(\w+)/)?.[1]; document.documentElement.classList.toggle('dark', guestTheme === 'dark' || ( (!guestTheme || guestTheme === 'system') && window.matchMedia('(prefers-color-scheme: dark)').matches));</script>
<link rel="canonical" href="https://huggingface.co/datasets/griffin/ChemSum"> <script type="application/ld+json">{
  "@context": {
    "@language": "en",
    "@vocab": "https:\/\/schema.org\/",
    "citeAs": "cr:citeAs",
    "column": "cr:column",
    "conformsTo": "dct:conformsTo",
    "cr": "http:\/\/mlcommons.org\/croissant\/",
    "data": {
      "@id": "cr:data",
      "@type": "@json"
    },
    "dataBiases": "cr:dataBiases",
    "dataCollection": "cr:dataCollection",
    "dataType": {
      "@id": "cr:dataType",
      "@type": "@vocab"
    },
    "dct": "http:\/\/purl.org\/dc\/terms\/",
    "extract": "cr:extract",
    "field": "cr:field",
    "fileProperty": "cr:fileProperty",
    "fileObject": "cr:fileObject",
    "fileSet": "cr:fileSet",
    "format": "cr:format",
    "includes": "cr:includes",
    "isLiveDataset": "cr:isLiveDataset",
    "jsonPath": "cr:jsonPath",
    "key": "cr:key",
    "md5": "cr:md5",
    "parentField": "cr:parentField",
    "path": "cr:path",
    "personalSensitiveInformation": "cr:personalSensitiveInformation",
    "recordSet": "cr:recordSet",
    "references": "cr:references",
    "regex": "cr:regex",
    "repeated": "cr:repeated",
    "replace": "cr:replace",
    "sc": "https:\/\/schema.org\/",
    "separator": "cr:separator",
    "source": "cr:source",
    "subField": "cr:subField",
    "transform": "cr:transform"
  },
  "@type": "sc:Dataset",
  "distribution": [
    {
      "@type": "cr:FileObject",
      "@id": "repo",
      "name": "repo",
      "description": "The Hugging Face git repository.",
      "contentUrl": "https:\/\/huggingface.co\/datasets\/griffin\/ChemSum\/tree\/refs%2Fconvert%2Fparquet",
      "encodingFormat": "git+https",
      "sha256": "https:\/\/github.com\/mlcommons\/croissant\/issues\/80"
    },
    {
      "@type": "cr:FileSet",
      "@id": "parquet-files-for-config-default",
      "name": "parquet-files-for-config-default",
      "description": "The underlying Parquet files as converted by Hugging Face (see: https:\/\/huggingface.co\/docs\/dataset-viewer\/parquet).",
      "containedIn": {
        "@id": "repo"
      },
      "encodingFormat": "application\/x-parquet",
      "includes": "default\/*\/*.parquet"
    }
  ],
  "recordSet": [
    {
      "@type": "cr:RecordSet",
      "dataType": "cr:Split",
      "key": {
        "@id": "default_splits\/split_name"
      },
      "@id": "default_splits",
      "name": "default_splits",
      "description": "Splits for the default config.",
      "field": [
        {
          "@type": "cr:Field",
          "@id": "default_splits\/split_name",
          "name": "split_name",
          "description": "The name of the split.",
          "dataType": "sc:Text"
        }
      ],
      "data": [
        {
          "default_splits\/split_name": "train"
        },
        {
          "default_splits\/split_name": "validation"
        },
        {
          "default_splits\/split_name": "test"
        }
      ]
    },
    {
      "@type": "cr:RecordSet",
      "@id": "default",
      "name": "default",
      "description": "griffin\/ChemSum - 'default' subset\n\nAdditional information:\n- 3 splits: train, validation, test",
      "field": [
        {
          "@type": "cr:Field",
          "@id": "default\/split",
          "name": "default\/split",
          "description": "Split to which the example belongs to.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-default"
            },
            "extract": {
              "fileProperty": "fullpath"
            },
            "transform": {
              "regex": "default\/(?:partial-)?(train|validation|test)\/.+parquet$"
            }
          },
          "references": {
            "field": {
              "@id": "default_splits\/split_name"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "default\/title",
          "name": "default\/title",
          "description": "Column 'title' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-default"
            },
            "extract": {
              "column": "title"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "default\/abstract",
          "name": "default\/abstract",
          "description": "Column 'abstract' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-default"
            },
            "extract": {
              "column": "abstract"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "default\/uuid",
          "name": "default\/uuid",
          "description": "Column 'uuid' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-default"
            },
            "extract": {
              "column": "uuid"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "default\/source_toks",
          "name": "default\/source_toks",
          "description": "Column 'source_toks' from the Hugging Face parquet file.",
          "dataType": "sc:Integer",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-default"
            },
            "extract": {
              "column": "source_toks"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "default\/target_toks",
          "name": "default\/target_toks",
          "description": "Column 'target_toks' from the Hugging Face parquet file.",
          "dataType": "sc:Integer",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-default"
            },
            "extract": {
              "column": "target_toks"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "default\/compression",
          "name": "default\/compression",
          "description": "Column 'compression' from the Hugging Face parquet file.",
          "dataType": "sc:Float",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-default"
            },
            "extract": {
              "column": "compression"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "default\/headers",
          "name": "default\/headers",
          "description": "Column 'headers' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-default"
            },
            "extract": {
              "column": "headers"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "default\/sections",
          "name": "default\/sections",
          "description": "Column 'sections' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-default"
            },
            "extract": {
              "column": "sections"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "default\/article_source",
          "name": "default\/article_source",
          "description": "Column 'article_source' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-default"
            },
            "extract": {
              "column": "article_source"
            }
          }
        }
      ]
    }
  ],
  "conformsTo": "http:\/\/mlcommons.org\/croissant\/1.0",
  "name": "ChemSum",
  "description": "\n\t\n\t\t\n\t\n\t\n\t\tDataset Card for ChemSum\n\t\n\n\n\t\n\t\t\n\t\n\t\n\t\tChemSum Description\n\t\n\n\n\nPaper: What are the Desired Characteristics of Calibration Sets? Identifying Correlates on Long Form Scientific Summarization \nJournal: ACL 2023\nPoint of Contact: griffin.adams@columbia.edu\nRepository: https:\/\/github.com\/griff4692\/calibrating-summaries\n\n\n\t\n\t\t\n\t\n\t\n\t\tChemSum Summary\n\t\n\nWe introduce a dataset with a pure chemistry focus by compiling a list of chemistry academic journals with Open-Access articles. For… See the full description on the dataset page: https:\/\/huggingface.co\/datasets\/griffin\/ChemSum.",
  "alternateName": [
    "griffin\/ChemSum",
    "Generating Abstracts of Academic Chemistry Papers"
  ],
  "creator": {
    "@type": "Person",
    "name": "Griffin Adams",
    "url": "https:\/\/huggingface.co\/griffin"
  },
  "keywords": [
    "summarization",
    "English",
    "100K - 1M",
    "parquet",
    "Tabular",
    "Text",
    "Datasets",
    "Dask",
    "Croissant",
    "Polars",
    "arxiv:2305.07615",
    "🇺🇸 Region: US",
    "chemistry",
    "biology",
    "medical"
  ],
  "url": "https:\/\/huggingface.co\/datasets\/griffin\/ChemSum"
}</script> 

		<title>griffin/ChemSum · Datasets at Hugging Face</title>

		<script
			defer
			data-domain="huggingface.co"
			event-loggedIn="false"
			src="/js/script.pageview-props.js"
		></script>
		<script>
			window.plausible =
				window.plausible ||
				function () {
					(window.plausible.q = window.plausible.q || []).push(arguments);
				};
		</script>
		<script>
			window.hubConfig = {"features":{"signupDisabled":false},"sshGitUrl":"git@hf.co","moonHttpUrl":"https:\/\/huggingface.co","captchaApiKey":"bd5f2066-93dc-4bdd-a64b-a24646ca3859","captchaDisabledOnSignup":true,"datasetViewerPublicUrl":"https:\/\/datasets-server.huggingface.co","stripePublicKey":"pk_live_x2tdjFXBCvXo2FFmMybezpeM00J6gPCAAc","environment":"production","userAgent":"HuggingFace (production)","spacesIframeDomain":"hf.space","spacesApiUrl":"https:\/\/api.hf.space","docSearchKey":"ece5e02e57300e17d152c08056145326e90c4bff3dd07d7d1ae40cf1c8d39cb6","logoDev":{"apiUrl":"https:\/\/img.logo.dev\/","apiKey":"pk_UHS2HZOeRnaSOdDp7jbd5w"}};
		</script>
		<script type="text/javascript" src="https://de5282c3ca0c.edge.sdk.awswaf.com/de5282c3ca0c/526cf06acb0d/challenge.js" defer></script>
	</head>
	<body class="flex flex-col min-h-dvh bg-white dark:bg-gray-950 text-black DatasetPage">
		<div class="flex min-h-dvh flex-col"><div class="SVELTE_HYDRATER contents" data-target="SystemThemeMonitor" data-props="{&quot;isLoggedIn&quot;:false}"></div>

	<div class="SVELTE_HYDRATER contents" data-target="MainHeader" data-props="{&quot;classNames&quot;:&quot;&quot;,&quot;isWide&quot;:false,&quot;isZh&quot;:false,&quot;isPro&quot;:false}"><header class="border-b border-gray-100 "><div class="w-full px-4 container flex h-16 items-center"><div class="flex flex-1 items-center"><a class="mr-5 flex flex-none items-center lg:mr-6" href="/"><img alt="Hugging Face's logo" class="w-7 md:mr-2" src="/front/assets/huggingface_logo-noborder.svg">
				<span class="hidden whitespace-nowrap text-lg font-bold md:block">Hugging Face</span></a>
			<div class="relative flex-1 lg:max-w-sm mr-2 sm:mr-4 md:mr-3 xl:mr-6"><input autocomplete="off" class="w-full dark:bg-gray-950 pl-8 form-input-alt h-9 pr-3 focus:shadow-xl " name="" placeholder="Search models, datasets, users..."   spellcheck="false" type="text" value="">
	<svg class="absolute left-2.5 text-gray-400 top-1/2 transform -translate-y-1/2" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M30 28.59L22.45 21A11 11 0 1 0 21 22.45L28.59 30zM5 14a9 9 0 1 1 9 9a9 9 0 0 1-9-9z" fill="currentColor"></path></svg>
	</div>
			<div class="flex flex-none items-center justify-center p-0.5 place-self-stretch lg:hidden"><button class="relative z-40 flex h-6 w-8 items-center justify-center" type="button"><svg width="1em" height="1em" viewBox="0 0 10 10" class="text-xl" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" preserveAspectRatio="xMidYMid meet" fill="currentColor"><path fill-rule="evenodd" clip-rule="evenodd" d="M1.65039 2.9999C1.65039 2.8066 1.80709 2.6499 2.00039 2.6499H8.00039C8.19369 2.6499 8.35039 2.8066 8.35039 2.9999C8.35039 3.1932 8.19369 3.3499 8.00039 3.3499H2.00039C1.80709 3.3499 1.65039 3.1932 1.65039 2.9999ZM1.65039 4.9999C1.65039 4.8066 1.80709 4.6499 2.00039 4.6499H8.00039C8.19369 4.6499 8.35039 4.8066 8.35039 4.9999C8.35039 5.1932 8.19369 5.3499 8.00039 5.3499H2.00039C1.80709 5.3499 1.65039 5.1932 1.65039 4.9999ZM2.00039 6.6499C1.80709 6.6499 1.65039 6.8066 1.65039 6.9999C1.65039 7.1932 1.80709 7.3499 2.00039 7.3499H8.00039C8.19369 7.3499 8.35039 7.1932 8.35039 6.9999C8.35039 6.8066 8.19369 6.6499 8.00039 6.6499H2.00039Z"></path></svg>
		</button>

	</div></div>
		<nav aria-label="Main" class="ml-auto hidden lg:block"><ul class="flex items-center space-x-1.5 2xl:space-x-2"><li class="hover:text-indigo-700"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/models"><svg class="mr-1.5 text-gray-400 group-hover:text-indigo-500" style="" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 24 24"><path class="uim-quaternary" d="M20.23 7.24L12 12L3.77 7.24a1.98 1.98 0 0 1 .7-.71L11 2.76c.62-.35 1.38-.35 2 0l6.53 3.77c.29.173.531.418.7.71z" opacity=".25" fill="currentColor"></path><path class="uim-tertiary" d="M12 12v9.5a2.09 2.09 0 0 1-.91-.21L4.5 17.48a2.003 2.003 0 0 1-1-1.73v-7.5a2.06 2.06 0 0 1 .27-1.01L12 12z" opacity=".5" fill="currentColor"></path><path class="uim-primary" d="M20.5 8.25v7.5a2.003 2.003 0 0 1-1 1.73l-6.62 3.82c-.275.13-.576.198-.88.2V12l8.23-4.76c.175.308.268.656.27 1.01z" fill="currentColor"></path></svg>
					Models</a>
			</li><li class="hover:text-red-700"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/datasets"><svg class="mr-1.5 text-gray-400 group-hover:text-red-500" style="" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 25 25"><ellipse cx="12.5" cy="5" fill="currentColor" fill-opacity="0.25" rx="7.5" ry="2"></ellipse><path d="M12.5 15C16.6421 15 20 14.1046 20 13V20C20 21.1046 16.6421 22 12.5 22C8.35786 22 5 21.1046 5 20V13C5 14.1046 8.35786 15 12.5 15Z" fill="currentColor" opacity="0.5"></path><path d="M12.5 7C16.6421 7 20 6.10457 20 5V11.5C20 12.6046 16.6421 13.5 12.5 13.5C8.35786 13.5 5 12.6046 5 11.5V5C5 6.10457 8.35786 7 12.5 7Z" fill="currentColor" opacity="0.5"></path><path d="M5.23628 12C5.08204 12.1598 5 12.8273 5 13C5 14.1046 8.35786 15 12.5 15C16.6421 15 20 14.1046 20 13C20 12.8273 19.918 12.1598 19.7637 12C18.9311 12.8626 15.9947 13.5 12.5 13.5C9.0053 13.5 6.06886 12.8626 5.23628 12Z" fill="currentColor"></path></svg>
					Datasets</a>
			</li><li class="hover:text-blue-700"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/spaces"><svg class="mr-1.5 text-gray-400 group-hover:text-blue-500" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" viewBox="0 0 25 25"><path opacity=".5" d="M6.016 14.674v4.31h4.31v-4.31h-4.31ZM14.674 14.674v4.31h4.31v-4.31h-4.31ZM6.016 6.016v4.31h4.31v-4.31h-4.31Z" fill="currentColor"></path><path opacity=".75" fill-rule="evenodd" clip-rule="evenodd" d="M3 4.914C3 3.857 3.857 3 4.914 3h6.514c.884 0 1.628.6 1.848 1.414a5.171 5.171 0 0 1 7.31 7.31c.815.22 1.414.964 1.414 1.848v6.514A1.914 1.914 0 0 1 20.086 22H4.914A1.914 1.914 0 0 1 3 20.086V4.914Zm3.016 1.102v4.31h4.31v-4.31h-4.31Zm0 12.968v-4.31h4.31v4.31h-4.31Zm8.658 0v-4.31h4.31v4.31h-4.31Zm0-10.813a2.155 2.155 0 1 1 4.31 0 2.155 2.155 0 0 1-4.31 0Z" fill="currentColor"></path><path opacity=".25" d="M16.829 6.016a2.155 2.155 0 1 0 0 4.31 2.155 2.155 0 0 0 0-4.31Z" fill="currentColor"></path></svg>
					Spaces</a>
			</li><li class="hover:text-yellow-700 max-xl:hidden"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/posts"><svg class="mr-1.5 text-gray-400 group-hover:text-yellow-500 text-yellow-500!" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" viewBox="0 0 12 12" preserveAspectRatio="xMidYMid meet"><path fill="currentColor" fill-rule="evenodd" d="M3.73 2.4A4.25 4.25 0 1 1 6 10.26H2.17l-.13-.02a.43.43 0 0 1-.3-.43l.01-.06a.43.43 0 0 1 .12-.22l.84-.84A4.26 4.26 0 0 1 3.73 2.4Z" clip-rule="evenodd"></path></svg>
					Posts</a>
			</li><li class="hover:text-yellow-700"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/docs"><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" class="mr-1.5 text-gray-400 group-hover:text-yellow-500" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 16 16"><path d="m2.28 3.7-.3.16a.67.67 0 0 0-.34.58v8.73l.01.04.02.07.01.04.03.06.02.04.02.03.04.06.05.05.04.04.06.04.06.04.08.04.08.02h.05l.07.02h.11l.04-.01.07-.02.03-.01.07-.03.22-.12a5.33 5.33 0 0 1 5.15.1.67.67 0 0 0 .66 0 5.33 5.33 0 0 1 5.33 0 .67.67 0 0 0 1-.58V4.36a.67.67 0 0 0-.34-.5l-.3-.17v7.78a.63.63 0 0 1-.87.59 4.9 4.9 0 0 0-4.35.35l-.65.39a.29.29 0 0 1-.15.04.29.29 0 0 1-.16-.04l-.65-.4a4.9 4.9 0 0 0-4.34-.34.63.63 0 0 1-.87-.59V3.7Z" fill="currentColor" class="dark:opacity-40"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M8 3.1a5.99 5.99 0 0 0-5.3-.43.66.66 0 0 0-.42.62v8.18c0 .45.46.76.87.59a4.9 4.9 0 0 1 4.34.35l.65.39c.05.03.1.04.16.04.05 0 .1-.01.15-.04l.65-.4a4.9 4.9 0 0 1 4.35-.34.63.63 0 0 0 .86-.59V3.3a.67.67 0 0 0-.41-.62 5.99 5.99 0 0 0-5.3.43l-.3.17L8 3.1Zm.73 1.87a.43.43 0 1 0-.86 0v5.48a.43.43 0 0 0 .86 0V4.97Z" fill="currentColor" class="opacity-40 dark:opacity-100"></path><path d="M8.73 4.97a.43.43 0 1 0-.86 0v5.48a.43.43 0 1 0 .86 0V4.96Z" fill="currentColor" class="dark:opacity-40"></path></svg>
					Docs</a>
			</li><li class="hover:text-black dark:hover:text-white"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/enterprise"><svg class="mr-1.5 text-gray-400 group-hover:text-black dark:group-hover:text-white" xmlns="http://www.w3.org/2000/svg" fill="none" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 33 27"><path fill="currentColor" fill-rule="evenodd" d="M13.5.7a8.7 8.7 0 0 0-7.7 5.7L1 20.6c-1 3.1.9 5.7 4.1 5.7h15c3.3 0 6.8-2.6 7.8-5.7l4.6-14.2c1-3.1-.8-5.7-4-5.7h-15Zm1.1 5.7L9.8 20.3h9.8l1-3.1h-5.8l.8-2.5h4.8l1.1-3h-4.8l.8-2.3H23l1-3h-9.5Z" clip-rule="evenodd"></path></svg>
					Enterprise</a>
			</li>

		<li><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/pricing">Pricing
			</a></li>

		<li><div class="relative group">
	<button class="px-2 py-0.5 hover:text-gray-500 dark:hover:text-gray-600 flex items-center " type="button">
		<svg class=" text-gray-500 w-5 group-hover:text-gray-400 dark:text-gray-300 dark:group-hover:text-gray-100" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" viewBox="0 0 32 18" preserveAspectRatio="xMidYMid meet"><path fill-rule="evenodd" clip-rule="evenodd" d="M14.4504 3.30221C14.4504 2.836 14.8284 2.45807 15.2946 2.45807H28.4933C28.9595 2.45807 29.3374 2.836 29.3374 3.30221C29.3374 3.76842 28.9595 4.14635 28.4933 4.14635H15.2946C14.8284 4.14635 14.4504 3.76842 14.4504 3.30221Z" fill="currentColor"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M14.4504 9.00002C14.4504 8.53382 14.8284 8.15588 15.2946 8.15588H28.4933C28.9595 8.15588 29.3374 8.53382 29.3374 9.00002C29.3374 9.46623 28.9595 9.84417 28.4933 9.84417H15.2946C14.8284 9.84417 14.4504 9.46623 14.4504 9.00002Z" fill="currentColor"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M14.4504 14.6978C14.4504 14.2316 14.8284 13.8537 15.2946 13.8537H28.4933C28.9595 13.8537 29.3374 14.2316 29.3374 14.6978C29.3374 15.164 28.9595 15.542 28.4933 15.542H15.2946C14.8284 15.542 14.4504 15.164 14.4504 14.6978Z" fill="currentColor"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M1.94549 6.87377C2.27514 6.54411 2.80962 6.54411 3.13928 6.87377L6.23458 9.96907L9.32988 6.87377C9.65954 6.54411 10.194 6.54411 10.5237 6.87377C10.8533 7.20343 10.8533 7.73791 10.5237 8.06756L6.23458 12.3567L1.94549 8.06756C1.61583 7.73791 1.61583 7.20343 1.94549 6.87377Z" fill="currentColor"></path></svg>
			
		</button>
	
	
	</div></li>
		<li><hr class="h-5 w-0.5 border-none bg-gray-100 dark:bg-gray-800"></li>
		<li><a class="block cursor-pointer whitespace-nowrap px-2 py-0.5 hover:text-gray-500 dark:text-gray-300 dark:hover:text-gray-100" href="/login">Log In
				</a></li>
			<li><a class="whitespace-nowrap rounded-full border border-transparent bg-gray-900 px-3 py-1 leading-none text-white hover:border-black hover:bg-white hover:text-black" href="/join">Sign Up
					</a></li></ul></nav></div></header></div>
	
	
	
	<div class="SVELTE_HYDRATER contents" data-target="SSOBanner" data-props="{}"></div>
	
	

	<main class="flex flex-1 flex-col">
	<div class="SVELTE_HYDRATER contents" data-target="DatasetHeader" data-props="{&quot;activeTab&quot;:&quot;datasetCard&quot;,&quot;author&quot;:{&quot;_id&quot;:&quot;611693254ef9fdfbf45dc4d6&quot;,&quot;avatarUrl&quot;:&quot;https://cdn-avatars.huggingface.co/v1/production/uploads/1628869402886-noauth.jpeg&quot;,&quot;fullname&quot;:&quot;Griffin Adams&quot;,&quot;name&quot;:&quot;griffin&quot;,&quot;type&quot;:&quot;user&quot;,&quot;isPro&quot;:false,&quot;isHf&quot;:false,&quot;isHfAdmin&quot;:false,&quot;isMod&quot;:false,&quot;followerCount&quot;:26},&quot;canReadRepoSettings&quot;:false,&quot;dataset&quot;:{&quot;author&quot;:&quot;griffin&quot;,&quot;cardData&quot;:{&quot;task_categories&quot;:[&quot;summarization&quot;],&quot;language&quot;:[&quot;en&quot;],&quot;tags&quot;:[&quot;chemistry&quot;,&quot;biology&quot;,&quot;medical&quot;],&quot;pretty_name&quot;:&quot;Generating Abstracts of Academic Chemistry Papers&quot;,&quot;size_categories&quot;:[&quot;100K<n<1M&quot;]},&quot;cardExists&quot;:true,&quot;createdAt&quot;:&quot;2023-05-10T02:05:05.000Z&quot;,&quot;description&quot;:&quot;\n\t\n\t\t\n\t\n\t\n\t\tDataset Card for ChemSum\n\t\n\n\n\t\n\t\t\n\t\n\t\n\t\tChemSum Description\n\t\n\n\n\nPaper: What are the Desired Characteristics of Calibration Sets? Identifying Correlates on Long Form Scientific Summarization \nJournal: ACL 2023\nPoint of Contact: griffin.adams@columbia.edu\nRepository: https://github.com/griff4692/calibrating-summaries\n\n\n\t\n\t\t\n\t\n\t\n\t\tChemSum Summary\n\t\n\nWe introduce a dataset with a pure chemistry focus by compiling a list of chemistry academic journals with Open-Access articles. For… See the full description on the dataset page: https://huggingface.co/datasets/griffin/ChemSum.&quot;,&quot;downloads&quot;:725,&quot;downloadsAllTime&quot;:2958,&quot;id&quot;:&quot;griffin/ChemSum&quot;,&quot;isLikedByUser&quot;:false,&quot;lastModified&quot;:&quot;2024-01-20T12:38:53.000Z&quot;,&quot;likes&quot;:14,&quot;datasetsServerInfo&quot;:{&quot;viewer&quot;:&quot;viewer&quot;,&quot;numRows&quot;:118956,&quot;libraries&quot;:[&quot;datasets&quot;,&quot;dask&quot;,&quot;mlcroissant&quot;,&quot;polars&quot;],&quot;formats&quot;:[&quot;parquet&quot;],&quot;modalities&quot;:[&quot;tabular&quot;,&quot;text&quot;]},&quot;discussionsDisabled&quot;:false,&quot;repoType&quot;:&quot;dataset&quot;,&quot;private&quot;:false,&quot;gated&quot;:false,&quot;tags&quot;:[&quot;task_categories:summarization&quot;,&quot;language:en&quot;,&quot;size_categories:100K<n<1M&quot;,&quot;format:parquet&quot;,&quot;modality:tabular&quot;,&quot;modality:text&quot;,&quot;library:datasets&quot;,&quot;library:dask&quot;,&quot;library:mlcroissant&quot;,&quot;library:polars&quot;,&quot;arxiv:2305.07615&quot;,&quot;region:us&quot;,&quot;chemistry&quot;,&quot;biology&quot;,&quot;medical&quot;],&quot;tag_objs&quot;:[{&quot;id&quot;:&quot;task_categories:summarization&quot;,&quot;label&quot;:&quot;summarization&quot;,&quot;type&quot;:&quot;task_categories&quot;,&quot;subType&quot;:&quot;nlp&quot;},{&quot;id&quot;:&quot;language:en&quot;,&quot;label&quot;:&quot;English&quot;,&quot;type&quot;:&quot;language&quot;},{&quot;id&quot;:&quot;size_categories:100K<n<1M&quot;,&quot;label&quot;:&quot;100K - 1M&quot;,&quot;type&quot;:&quot;size_categories&quot;},{&quot;id&quot;:&quot;format:parquet&quot;,&quot;label&quot;:&quot;parquet&quot;,&quot;type&quot;:&quot;format&quot;},{&quot;id&quot;:&quot;modality:tabular&quot;,&quot;label&quot;:&quot;Tabular&quot;,&quot;type&quot;:&quot;modality&quot;},{&quot;id&quot;:&quot;modality:text&quot;,&quot;label&quot;:&quot;Text&quot;,&quot;type&quot;:&quot;modality&quot;},{&quot;id&quot;:&quot;library:datasets&quot;,&quot;label&quot;:&quot;Datasets&quot;,&quot;type&quot;:&quot;library&quot;},{&quot;id&quot;:&quot;library:dask&quot;,&quot;label&quot;:&quot;Dask&quot;,&quot;type&quot;:&quot;library&quot;},{&quot;id&quot;:&quot;library:mlcroissant&quot;,&quot;label&quot;:&quot;Croissant&quot;,&quot;type&quot;:&quot;library&quot;},{&quot;id&quot;:&quot;library:polars&quot;,&quot;label&quot;:&quot;Polars&quot;,&quot;type&quot;:&quot;library&quot;},{&quot;id&quot;:&quot;arxiv:2305.07615&quot;,&quot;label&quot;:&quot;arxiv:2305.07615&quot;,&quot;type&quot;:&quot;arxiv&quot;,&quot;extra&quot;:{&quot;paperTitle&quot;:&quot;What are the Desired Characteristics of Calibration Sets? Identifying\n  Correlates on Long Form Scientific Summarization&quot;}},{&quot;type&quot;:&quot;region&quot;,&quot;label&quot;:&quot;🇺🇸 Region: US&quot;,&quot;id&quot;:&quot;region:us&quot;},{&quot;id&quot;:&quot;chemistry&quot;,&quot;label&quot;:&quot;chemistry&quot;,&quot;type&quot;:&quot;other&quot;},{&quot;id&quot;:&quot;biology&quot;,&quot;label&quot;:&quot;biology&quot;,&quot;type&quot;:&quot;other&quot;},{&quot;id&quot;:&quot;medical&quot;,&quot;label&quot;:&quot;medical&quot;,&quot;type&quot;:&quot;other&quot;}],&quot;hasBlockedOids&quot;:false,&quot;region&quot;:&quot;us&quot;,&quot;xetEnabled&quot;:false},&quot;discussionsStats&quot;:{&quot;closed&quot;:0,&quot;open&quot;:1,&quot;total&quot;:1}}"><header class="bg-linear-to-t border-b border-gray-100 pt-6 sm:pt-9 from-gray-50-to-white via-white dark:via-gray-950"><div class="container relative "><h1 class="flex flex-wrap items-center max-md:leading-tight mb-3 text-lg max-sm:gap-y-1.5 md:text-xl"><a href="/datasets" class="group flex items-center"><svg class="sm:mr-1.5 -mr-1 text-gray-400" style="" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 25 25"><ellipse cx="12.5" cy="5" fill="currentColor" fill-opacity="0.25" rx="7.5" ry="2"></ellipse><path d="M12.5 15C16.6421 15 20 14.1046 20 13V20C20 21.1046 16.6421 22 12.5 22C8.35786 22 5 21.1046 5 20V13C5 14.1046 8.35786 15 12.5 15Z" fill="currentColor" opacity="0.5"></path><path d="M12.5 7C16.6421 7 20 6.10457 20 5V11.5C20 12.6046 16.6421 13.5 12.5 13.5C8.35786 13.5 5 12.6046 5 11.5V5C5 6.10457 8.35786 7 12.5 7Z" fill="currentColor" opacity="0.5"></path><path d="M5.23628 12C5.08204 12.1598 5 12.8273 5 13C5 14.1046 8.35786 15 12.5 15C16.6421 15 20 14.1046 20 13C20 12.8273 19.918 12.1598 19.7637 12C18.9311 12.8626 15.9947 13.5 12.5 13.5C9.0053 13.5 6.06886 12.8626 5.23628 12Z" fill="currentColor"></path></svg>
					<span class="mr-2.5 font-semibold text-gray-400 group-hover:text-gray-500 max-sm:hidden">Datasets:</span></a>
				<hr class="mx-1.5 h-2 translate-y-px rounded-sm border-r dark:border-gray-600 sm:hidden">
			<div class="group flex flex-none items-center"><div class="relative mr-1 flex items-center">

			

<span class="inline-block "><span class="contents"><a href="/griffin" class="text-gray-400 hover:text-blue-600"><img alt="" class="w-3.5 h-3.5 rounded-full  flex-none" src="https://cdn-avatars.huggingface.co/v1/production/uploads/1628869402886-noauth.jpeg" crossorigin="anonymous"></a></span>
	</span></div>
		

<span class="inline-block "><span class="contents"><a href="/griffin" class="text-gray-400 hover:text-blue-600">griffin</a></span>
	</span>
		<div class="mx-0.5 text-gray-300">/</div></div>

<div class="max-w-full "><a class="break-words font-mono font-semibold hover:text-blue-600 " href="/datasets/griffin/ChemSum">ChemSum</a>
	<button class="relative text-sm mr-4 focus:outline-hidden inline-flex cursor-pointer items-center text-sm  mx-0.5   text-gray-600 " title="Copy dataset name to clipboard" type="button"><svg class="" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" fill="currentColor" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M28,10V28H10V10H28m0-2H10a2,2,0,0,0-2,2V28a2,2,0,0,0,2,2H28a2,2,0,0,0,2-2V10a2,2,0,0,0-2-2Z" transform="translate(0)"></path><path d="M4,18H2V4A2,2,0,0,1,4,2H18V4H4Z" transform="translate(0)"></path><rect fill="none" width="32" height="32"></rect></svg>
	
	</button></div>
			<div class="inline-flex items-center overflow-hidden whitespace-nowrap rounded-md border bg-white text-sm leading-none text-gray-500  mr-2"><button class="relative flex items-center overflow-hidden from-red-50 to-transparent dark:from-red-900 px-1.5 py-1 hover:bg-linear-to-t focus:outline-hidden"  title="Like"><svg class="left-1.5 absolute" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32" fill="currentColor"><path d="M22.45,6a5.47,5.47,0,0,1,3.91,1.64,5.7,5.7,0,0,1,0,8L16,26.13,5.64,15.64a5.7,5.7,0,0,1,0-8,5.48,5.48,0,0,1,7.82,0L16,10.24l2.53-2.58A5.44,5.44,0,0,1,22.45,6m0-2a7.47,7.47,0,0,0-5.34,2.24L16,7.36,14.89,6.24a7.49,7.49,0,0,0-10.68,0,7.72,7.72,0,0,0,0,10.82L16,29,27.79,17.06a7.72,7.72,0,0,0,0-10.82A7.49,7.49,0,0,0,22.45,4Z"></path></svg>

		
		<span class="ml-4 pl-0.5 ">like</span></button>
	<button class="focus:outline-hidden flex items-center border-l px-1.5 py-1 text-gray-400 hover:bg-gray-50 focus:bg-gray-100 dark:hover:bg-gray-900 dark:focus:bg-gray-800" title="See users who liked this repository">14</button></div>




			
			
	</h1>
		<div class="mb-3 flex flex-wrap md:mb-4"><div class="mr-1 flex flex-wrap items-center"><span class="mb-1 mr-1 p-1 text-sm leading-tight text-gray-400 md:mb-1.5">Tasks:
	</span>
	<a class="mb-1 mr-1 md:mb-1.5 md:mr-1.5 rounded-lg" href="/datasets?task_categories=task_categories%3Asummarization"><div class="tag tag-white   "><div class="tag-ico -ml-2 tag-ico-indigo"><svg class="" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" fill="currentColor" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 18 19"><path d="M15.4988 8.79309L12.1819 5.47621C12.0188 5.25871 11.7469 5.14996 11.475 5.14996H7.12501C6.52688 5.14996 6.03751 5.63934 6.03751 6.23746V16.025C6.03751 16.6231 6.52688 17.1125 7.12501 17.1125H14.7375C15.3356 17.1125 15.825 16.6231 15.825 16.025V9.55434C15.825 9.28246 15.7163 9.01059 15.4988 8.79309V8.79309ZM11.475 6.23746L14.6831 9.49996H11.475V6.23746ZM7.12501 16.025V6.23746H10.3875V9.49996C10.3875 10.0981 10.8769 10.5875 11.475 10.5875H14.7375V16.025H7.12501Z"></path><path d="M3.8625 10.5875H2.775V2.97498C2.775 2.37686 3.26438 1.88748 3.8625 1.88748H11.475V2.97498H3.8625V10.5875Z"></path></svg></div>

	

	<span>Summarization</span>
	

	</div></a>

	</div><div class="mr-1 flex flex-wrap items-center"><span class="mb-1 mr-1 p-1 text-sm leading-tight text-gray-400 md:mb-1.5">Modalities:
	</span>
	<a class="mb-1 mr-1 md:mb-1.5 md:mr-1.5 rounded-lg" href="/datasets?modality=modality%3Atabular"><div class="tag tag-white   ">
		<svg class="" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path fill-rule="evenodd" clip-rule="evenodd" d="M4.619 4.619C2.667 6.572 2.667 9.715 2.667 16s0 9.43 1.952 11.38C6.572 29.333 9.715 29.333 16 29.333s9.43 0 11.38-1.953c1.953-1.952 1.953-5.095 1.953-11.38 0-6.285 0-9.427-1.953-11.381C25.428 2.667 22.285 2.667 16 2.667c-6.285 0-9.427 0-11.381 1.952ZM16.615 10a1 1 0 1 1 0 2h-1.23a1 1 0 1 1 0-2h1.23Zm0 5a1 1 0 1 1 0 2h-1.23a1 1 0 1 1 0-2h1.23Zm3.154 1a1 1 0 0 0 1 1H22a1 1 0 1 0 0-2h-1.23a1 1 0 0 0-1 1Zm-3.154 4a1 1 0 1 1 0 2h-1.23a1 1 0 1 1 0-2h1.23Zm3.154 1a1 1 0 0 0 1 1H22a1 1 0 1 0 0-2h-1.23a1 1 0 0 0-1 1Zm0-10a1 1 0 0 0 1 1H22a1 1 0 1 0 0-2h-1.23a1 1 0 0 0-1 1Zm-8.538-1H10a1 1 0 1 0 0 2h1.23a1 1 0 1 0 0-2Zm0 5H10a1 1 0 1 0 0 2h1.23a1 1 0 1 0 0-2Zm0 5H10a1 1 0 1 0 0 2h1.23a1 1 0 1 0 0-2Z" fill="currentColor"></path></svg>

	

	<span>Tabular</span>
	

	</div></a><a class="mb-1 mr-1 md:mb-1.5 md:mr-1.5 rounded-lg" href="/datasets?modality=modality%3Atext"><div class="tag tag-white   ">
		<svg class="text-red-700 dark:text-red-600" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path fill-rule="evenodd" clip-rule="evenodd" d="M4.619 4.619C2.667 6.573 2.667 9.715 2.667 16s0 9.428 1.952 11.38C6.573 29.333 9.715 29.333 16 29.333s9.428 0 11.38-1.953c1.953-1.95 1.953-5.095 1.953-11.38s0-9.428-1.953-11.381C25.43 2.667 22.285 2.667 16 2.667s-9.428 0-11.381 1.952m8.65 3.714c-.573 0-1.109 0-1.546.066-.495.073-1.003.248-1.41.7-.392.436-.53.956-.59 1.452-.056.464-.056 1.04-.056 1.689V13a1 1 0 1 0 2 0v-.704c0-.724.001-1.176.041-1.505q.015-.15.061-.294a.2.2 0 0 1 .031-.061q0-.003.016-.01a.8.8 0 0 1 .203-.05c.272-.04.654-.043 1.314-.043H15v11.334h-2.333a1 1 0 1 0 0 2H20a1 1 0 0 0 0-2h-3V10.333h1.667c.66 0 1.042.003 1.314.043.123.019.18.04.203.05l.015.009a.2.2 0 0 1 .032.061c.018.05.042.14.061.295.04.329.041.781.041 1.506V13a1 1 0 1 0 2 0v-.76c0-.65 0-1.225-.056-1.69-.06-.495-.198-1.015-.59-1.453-.407-.45-.915-.625-1.41-.698-.437-.067-.973-.067-1.546-.066z" fill="currentColor"></path></svg>

	

	<span>Text</span>
	

	</div></a>

	</div><div class="mr-1 flex flex-wrap items-center"><span class="mb-1 mr-1 p-1 text-sm leading-tight text-gray-400 md:mb-1.5">Formats:
	</span>
	<a class="mb-1 mr-1 md:mb-1.5 md:mr-1.5 rounded-lg" href="/datasets?format=format%3Aparquet"><div class="tag tag-white   ">
		<svg class="" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 34 34"><path fill-rule="evenodd" clip-rule="evenodd" d="m17.97 18.44-3.98-2.3-7.1 3.78 3.98 2.3 7.1-3.78Zm15.1-1.4-3.99-2.3-16.22 8.63 3.98 2.3 16.22-8.63Zm-5.98-3.45-3.97-2.3-7.1 3.78 3.98 2.3 7.1-3.78Zm-9.94-5.74 3.98 2.3-11.16 5.93L6 13.78l11.16-5.93Zm-13.19 7 3.98 2.3-3.04 1.62-3.98-2.3 3.04-1.61Z" fill="currentColor"></path></svg>

	

	<span>parquet</span>
	

	</div></a>

	</div><div class="mr-1 flex flex-wrap items-center"><span class="mb-1 mr-1 p-1 text-sm leading-tight text-gray-400 md:mb-1.5">Languages:
	</span>
	<a class="mb-1 mr-1 md:mb-1.5 md:mr-1.5 rounded-lg" href="/datasets?language=language%3Aen"><div class="tag tag-white   ">
		<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" class="text-green-600/80" preserveAspectRatio="xMidYMid meet" width="1em" height="1em" viewBox="0 0 10 10"><path fill-rule="evenodd" clip-rule="evenodd" d="M0.625 5C0.625 6.16032 1.08594 7.27312 1.90641 8.09359C2.72688 8.91406 3.83968 9.375 5 9.375C6.16032 9.375 7.27312 8.91406 8.09359 8.09359C8.91406 7.27312 9.375 6.16032 9.375 5C9.375 3.83968 8.91406 2.72688 8.09359 1.90641C7.27312 1.08594 6.16032 0.625 5 0.625C3.83968 0.625 2.72688 1.08594 1.90641 1.90641C1.08594 2.72688 0.625 3.83968 0.625 5ZM7.64365 7.48027C7.61734 7.50832 7.59054 7.53598 7.56326 7.56326C7.13828 7.98824 6.61864 8.2968 6.0539 8.46842C6.29802 8.11949 6.49498 7.64804 6.63475 7.09483C7.00845 7.18834 7.35014 7.3187 7.64365 7.48027ZM8.10076 6.87776C8.37677 6.42196 8.55005 5.90894 8.60556 5.37499H6.86808C6.85542 5.71597 6.82551 6.04557 6.77971 6.35841C7.25309 6.47355 7.68808 6.6414 8.062 6.85549C8.07497 6.86283 8.08789 6.87025 8.10076 6.87776ZM6.03795 6.22536C6.07708 5.95737 6.1044 5.67232 6.11705 5.37499H3.88295C3.89666 5.69742 3.92764 6.00542 3.9722 6.29287C4.37075 6.21726 4.79213 6.17749 5.224 6.17749C5.50054 6.17749 5.77294 6.19376 6.03795 6.22536ZM4.1261 7.02673C4.34894 7.84835 4.68681 8.375 5 8.375C5.32122 8.375 5.66839 7.82101 5.8908 6.963C5.67389 6.93928 5.45082 6.92699 5.224 6.92699C4.84316 6.92699 4.47332 6.96176 4.1261 7.02673ZM3.39783 7.21853C3.53498 7.71842 3.72038 8.14579 3.9461 8.46842C3.42141 8.30898 2.93566 8.03132 2.52857 7.65192C2.77253 7.48017 3.06711 7.33382 3.39783 7.21853ZM3.23916 6.48077C3.18263 6.13193 3.14625 5.76074 3.13192 5.37499H1.39444C1.4585 5.99112 1.67936 6.57938 2.03393 7.08403C2.3706 6.83531 2.78055 6.63162 3.23916 6.48077ZM1.39444 4.62499H3.13192C3.14615 4.24204 3.18211 3.87344 3.23794 3.52681C2.77814 3.37545 2.36731 3.17096 2.03024 2.92123C1.67783 3.42469 1.45828 4.011 1.39444 4.62499ZM2.5237 2.35262C2.76812 2.52552 3.06373 2.67281 3.39584 2.78875C3.53318 2.28573 3.71928 1.85578 3.9461 1.53158C3.41932 1.69166 2.93178 1.97089 2.5237 2.35262ZM3.97101 3.71489C3.92709 4.00012 3.89654 4.30547 3.88295 4.62499H6.11705C6.10453 4.33057 6.07761 4.04818 6.03909 3.78248C5.77372 3.81417 5.50093 3.83049 5.224 3.83049C4.79169 3.83049 4.3699 3.79065 3.97101 3.71489ZM5.8928 3.04476C5.67527 3.06863 5.45151 3.08099 5.224 3.08099C4.84241 3.08099 4.47186 3.04609 4.12405 2.98086C4.34686 2.1549 4.68584 1.625 5 1.625C5.32218 1.625 5.67048 2.18233 5.8928 3.04476ZM6.78083 3.6493C6.826 3.95984 6.85552 4.28682 6.86808 4.62499H8.60556C8.55029 4.09337 8.37827 3.58251 8.10436 3.1282C8.0903 3.1364 8.07618 3.14449 8.062 3.15249C7.68838 3.36641 7.25378 3.53417 6.78083 3.6493ZM7.64858 2.52499C7.35446 2.68754 7.0117 2.81868 6.63664 2.91268C6.49676 2.35623 6.29913 1.88209 6.0539 1.53158C6.61864 1.7032 7.13828 2.01176 7.56326 2.43674C7.59224 2.46572 7.62068 2.49514 7.64858 2.52499Z" fill="currentColor"></path></svg>

	

	<span>English</span>
	

	</div></a>

	</div><div class="mr-1 flex flex-wrap items-center"><span class="mb-1 mr-1 p-1 text-sm leading-tight text-gray-400 md:mb-1.5">Size:
	</span>
	<a class="mb-1 mr-1 md:mb-1.5 md:mr-1.5 rounded-lg" href="/datasets?size_categories=size_categories%3A100K%3Cn%3C1M"><div class="tag tag-white   ">

	

	<span>100K - 1M</span>
	

	</div></a>

	</div><div class="mr-1 flex flex-wrap items-center"><span class="mb-1 mr-1 p-1 text-sm leading-tight text-gray-400 md:mb-1.5">ArXiv:
	</span>
	

	<div class="relative inline-block ">
	<button class="group mr-1 mb-1 md:mr-1.5 md:mb-1.5  rounded-full rounded-br-none " type="button">
		<div class="tag tag-white rounded-full  relative rounded-br-none pr-2.5">
		<svg class="-mr-1 text-gray-400" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" viewBox="0 0 12 12" preserveAspectRatio="xMidYMid meet" fill="none"><path fill="currentColor" fill-rule="evenodd" d="M8.007 1.814a1.176 1.176 0 0 0-.732-.266H3.088c-.64 0-1.153.512-1.153 1.152v6.803c0 .64.513 1.152 1.153 1.152h5.54c.632 0 1.144-.511 1.144-1.152V3.816c0-.338-.137-.658-.412-.887L8.007 1.814Zm-1.875 1.81c0 .695.55 1.253 1.244 1.253h.983a.567.567 0 0 1 .553.585v4.041c0 .165-.119.302-.283.302h-5.55c-.156 0-.275-.137-.275-.302V2.7a.284.284 0 0 1 .284-.301h2.468a.574.574 0 0 1 .434.19.567.567 0 0 1 .142.395v.64Z" clip-rule="evenodd"></path><path fill="currentColor" fill-opacity=".2" fill-rule="evenodd" d="M6.132 3.624c0 .695.55 1.253 1.244 1.253h.97a.567.567 0 0 1 .566.585v4.041c0 .165-.119.302-.283.302h-5.55c-.156 0-.275-.137-.275-.302V2.7a.284.284 0 0 1 .284-.301h2.468a.567.567 0 0 1 .576.585v.64Z" clip-rule="evenodd"></path></svg>

	

	<span class="-mr-2 text-gray-400">arxiv:</span>
		<span>2305.07615</span>
	

	<div class="border-br-gray-200 absolute bottom-0.5 right-0.5 h-1 w-1 border-[3px] border-l-transparent border-t-transparent border-b-gray-200 border-r-gray-200 dark:border-b-gray-700 dark:border-r-gray-700"></div></div>
		
		</button>
	
	
	</div>

	</div><div class="mr-1 flex flex-wrap items-center"><span class="mb-1 mr-1 p-1 text-sm leading-tight text-gray-400 md:mb-1.5">Tags:
	</span>
	<a class="mb-1 mr-1 md:mb-1.5 md:mr-1.5 rounded-lg" href="/datasets?other=chemistry"><div class="tag tag-white   ">

	

	<span>chemistry</span>
	

	</div></a><a class="mb-1 mr-1 md:mb-1.5 md:mr-1.5 rounded-lg" href="/datasets?other=biology"><div class="tag tag-white   ">

	

	<span>biology</span>
	

	</div></a><a class="mb-1 mr-1 md:mb-1.5 md:mr-1.5 rounded-lg" href="/datasets?other=medical"><div class="tag tag-white   ">

	

	<span>medical</span>
	

	</div></a>

	</div><div class="mr-1 flex flex-wrap items-center"><span class="mb-1 mr-1 p-1 text-sm leading-tight text-gray-400 md:mb-1.5">Libraries:
	</span>
	<a class="mb-1 mr-1 md:mb-1.5 md:mr-1.5 rounded-lg" href="/datasets?library=library%3Adatasets"><div class="tag tag-white   "><svg class="text-black inline-block text-sm" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" preserveAspectRatio="xMidYMid meet" width="1em" height="1em" viewBox="0 0 95 88"><path fill="#fff" d="M94.25 70.08a8.28 8.28 0 0 1-.43 6.46 10.57 10.57 0 0 1-3 3.6 25.18 25.18 0 0 1-5.7 3.2 65.74 65.74 0 0 1-7.56 2.65 46.67 46.67 0 0 1-11.42 1.68c-5.42.05-10.09-1.23-13.4-4.5a40.4 40.4 0 0 1-10.14.03c-3.34 3.25-7.99 4.52-13.39 4.47a46.82 46.82 0 0 1-11.43-1.68 66.37 66.37 0 0 1-7.55-2.65c-2.28-.98-4.17-2-5.68-3.2a10.5 10.5 0 0 1-3.02-3.6c-.99-2-1.18-4.3-.42-6.46a8.54 8.54 0 0 1-.33-5.63c.25-.95.66-1.83 1.18-2.61a8.67 8.67 0 0 1 2.1-8.47 8.23 8.23 0 0 1 2.82-2.07 41.75 41.75 0 1 1 81.3-.12 8.27 8.27 0 0 1 3.11 2.19 8.7 8.7 0 0 1 2.1 8.47c.52.78.93 1.66 1.18 2.61a8.61 8.61 0 0 1-.32 5.63Z"></path><path fill="#FFD21E" d="M47.21 76.5a34.75 34.75 0 1 0 0-69.5 34.75 34.75 0 0 0 0 69.5Z"></path><path fill="#FF9D0B" d="M81.96 41.75a34.75 34.75 0 1 0-69.5 0 34.75 34.75 0 0 0 69.5 0Zm-73.5 0a38.75 38.75 0 1 1 77.5 0 38.75 38.75 0 0 1-77.5 0Z"></path><path fill="#3A3B45" d="M58.5 32.3c1.28.44 1.78 3.06 3.07 2.38a5 5 0 1 0-6.76-2.07c.61 1.15 2.55-.72 3.7-.32ZM34.95 32.3c-1.28.44-1.79 3.06-3.07 2.38a5 5 0 1 1 6.76-2.07c-.61 1.15-2.56-.72-3.7-.32Z"></path><path fill="#FF323D" d="M46.96 56.29c9.83 0 13-8.76 13-13.26 0-2.34-1.57-1.6-4.09-.36-2.33 1.15-5.46 2.74-8.9 2.74-7.19 0-13-6.88-13-2.38s3.16 13.26 13 13.26Z"></path><path fill="#3A3B45" fill-rule="evenodd" d="M39.43 54a8.7 8.7 0 0 1 5.3-4.49c.4-.12.81.57 1.24 1.28.4.68.82 1.37 1.24 1.37.45 0 .9-.68 1.33-1.35.45-.7.89-1.38 1.32-1.25a8.61 8.61 0 0 1 5 4.17c3.73-2.94 5.1-7.74 5.1-10.7 0-2.34-1.57-1.6-4.09-.36l-.14.07c-2.31 1.15-5.39 2.67-8.77 2.67s-6.45-1.52-8.77-2.67c-2.6-1.29-4.23-2.1-4.23.29 0 3.05 1.46 8.06 5.47 10.97Z" clip-rule="evenodd"></path><path fill="#FF9D0B" d="M70.71 37a3.25 3.25 0 1 0 0-6.5 3.25 3.25 0 0 0 0 6.5ZM24.21 37a3.25 3.25 0 1 0 0-6.5 3.25 3.25 0 0 0 0 6.5ZM17.52 48c-1.62 0-3.06.66-4.07 1.87a5.97 5.97 0 0 0-1.33 3.76 7.1 7.1 0 0 0-1.94-.3c-1.55 0-2.95.59-3.94 1.66a5.8 5.8 0 0 0-.8 7 5.3 5.3 0 0 0-1.79 2.82c-.24.9-.48 2.8.8 4.74a5.22 5.22 0 0 0-.37 5.02c1.02 2.32 3.57 4.14 8.52 6.1 3.07 1.22 5.89 2 5.91 2.01a44.33 44.33 0 0 0 10.93 1.6c5.86 0 10.05-1.8 12.46-5.34 3.88-5.69 3.33-10.9-1.7-15.92-2.77-2.78-4.62-6.87-5-7.77-.78-2.66-2.84-5.62-6.25-5.62a5.7 5.7 0 0 0-4.6 2.46c-1-1.26-1.98-2.25-2.86-2.82A7.4 7.4 0 0 0 17.52 48Zm0 4c.51 0 1.14.22 1.82.65 2.14 1.36 6.25 8.43 7.76 11.18.5.92 1.37 1.31 2.14 1.31 1.55 0 2.75-1.53.15-3.48-3.92-2.93-2.55-7.72-.68-8.01.08-.02.17-.02.24-.02 1.7 0 2.45 2.93 2.45 2.93s2.2 5.52 5.98 9.3c3.77 3.77 3.97 6.8 1.22 10.83-1.88 2.75-5.47 3.58-9.16 3.58-3.81 0-7.73-.9-9.92-1.46-.11-.03-13.45-3.8-11.76-7 .28-.54.75-.76 1.34-.76 2.38 0 6.7 3.54 8.57 3.54.41 0 .7-.17.83-.6.79-2.85-12.06-4.05-10.98-8.17.2-.73.71-1.02 1.44-1.02 3.14 0 10.2 5.53 11.68 5.53.11 0 .2-.03.24-.1.74-1.2.33-2.04-4.9-5.2-5.21-3.16-8.88-5.06-6.8-7.33.24-.26.58-.38 1-.38 3.17 0 10.66 6.82 10.66 6.82s2.02 2.1 3.25 2.1c.28 0 .52-.1.68-.38.86-1.46-8.06-8.22-8.56-11.01-.34-1.9.24-2.85 1.31-2.85Z"></path><path fill="#FFD21E" d="M38.6 76.69c2.75-4.04 2.55-7.07-1.22-10.84-3.78-3.77-5.98-9.3-5.98-9.3s-.82-3.2-2.69-2.9c-1.87.3-3.24 5.08.68 8.01 3.91 2.93-.78 4.92-2.29 2.17-1.5-2.75-5.62-9.82-7.76-11.18-2.13-1.35-3.63-.6-3.13 2.2.5 2.79 9.43 9.55 8.56 11-.87 1.47-3.93-1.71-3.93-1.71s-9.57-8.71-11.66-6.44c-2.08 2.27 1.59 4.17 6.8 7.33 5.23 3.16 5.64 4 4.9 5.2-.75 1.2-12.28-8.53-13.36-4.4-1.08 4.11 11.77 5.3 10.98 8.15-.8 2.85-9.06-5.38-10.74-2.18-1.7 3.21 11.65 6.98 11.76 7.01 4.3 1.12 15.25 3.49 19.08-2.12Z"></path><path fill="#FF9D0B" d="M77.4 48c1.62 0 3.07.66 4.07 1.87a5.97 5.97 0 0 1 1.33 3.76 7.1 7.1 0 0 1 1.95-.3c1.55 0 2.95.59 3.94 1.66a5.8 5.8 0 0 1 .8 7 5.3 5.3 0 0 1 1.78 2.82c.24.9.48 2.8-.8 4.74a5.22 5.22 0 0 1 .37 5.02c-1.02 2.32-3.57 4.14-8.51 6.1-3.08 1.22-5.9 2-5.92 2.01a44.33 44.33 0 0 1-10.93 1.6c-5.86 0-10.05-1.8-12.46-5.34-3.88-5.69-3.33-10.9 1.7-15.92 2.78-2.78 4.63-6.87 5.01-7.77.78-2.66 2.83-5.62 6.24-5.62a5.7 5.7 0 0 1 4.6 2.46c1-1.26 1.98-2.25 2.87-2.82A7.4 7.4 0 0 1 77.4 48Zm0 4c-.51 0-1.13.22-1.82.65-2.13 1.36-6.25 8.43-7.76 11.18a2.43 2.43 0 0 1-2.14 1.31c-1.54 0-2.75-1.53-.14-3.48 3.91-2.93 2.54-7.72.67-8.01a1.54 1.54 0 0 0-.24-.02c-1.7 0-2.45 2.93-2.45 2.93s-2.2 5.52-5.97 9.3c-3.78 3.77-3.98 6.8-1.22 10.83 1.87 2.75 5.47 3.58 9.15 3.58 3.82 0 7.73-.9 9.93-1.46.1-.03 13.45-3.8 11.76-7-.29-.54-.75-.76-1.34-.76-2.38 0-6.71 3.54-8.57 3.54-.42 0-.71-.17-.83-.6-.8-2.85 12.05-4.05 10.97-8.17-.19-.73-.7-1.02-1.44-1.02-3.14 0-10.2 5.53-11.68 5.53-.1 0-.19-.03-.23-.1-.74-1.2-.34-2.04 4.88-5.2 5.23-3.16 8.9-5.06 6.8-7.33-.23-.26-.57-.38-.98-.38-3.18 0-10.67 6.82-10.67 6.82s-2.02 2.1-3.24 2.1a.74.74 0 0 1-.68-.38c-.87-1.46 8.05-8.22 8.55-11.01.34-1.9-.24-2.85-1.31-2.85Z"></path><path fill="#FFD21E" d="M56.33 76.69c-2.75-4.04-2.56-7.07 1.22-10.84 3.77-3.77 5.97-9.3 5.97-9.3s.82-3.2 2.7-2.9c1.86.3 3.23 5.08-.68 8.01-3.92 2.93.78 4.92 2.28 2.17 1.51-2.75 5.63-9.82 7.76-11.18 2.13-1.35 3.64-.6 3.13 2.2-.5 2.79-9.42 9.55-8.55 11 .86 1.47 3.92-1.71 3.92-1.71s9.58-8.71 11.66-6.44c2.08 2.27-1.58 4.17-6.8 7.33-5.23 3.16-5.63 4-4.9 5.2.75 1.2 12.28-8.53 13.36-4.4 1.08 4.11-11.76 5.3-10.97 8.15.8 2.85 9.05-5.38 10.74-2.18 1.69 3.21-11.65 6.98-11.76 7.01-4.31 1.12-15.26 3.49-19.08-2.12Z"></path></svg>

	

	<span>Datasets</span>
	

	</div></a><a class="mb-1 mr-1 md:mb-1.5 md:mr-1.5 rounded-lg" href="/datasets?library=library%3Adask"><div class="tag tag-white   "><svg class="text-black inline-block text-sm" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" fill="currentColor" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 12 12"><path d="m3.368 3.694 2.965-1.71A.095.095 0 0 0 6.38 1.9V.875a.49.49 0 0 0-.183-.394.468.468 0 0 0-.523-.034L1.526 2.84a.471.471 0 0 0-.235.408l-.002 5.405c0 .151.062.302.183.394.157.118.358.13.524.035l.878-.507a.095.095 0 0 0 .048-.082V4.465a.89.89 0 0 1 .446-.771Z" fill="#FFC11E"></path><path d="M10.475 2.919a.47.47 0 0 0-.47 0L5.856 5.312a.473.473 0 0 0-.236.408l-.002 5.425c0 .17.088.323.236.408a.466.466 0 0 0 .471 0l4.147-2.393a.473.473 0 0 0 .236-.408l.002-5.425a.467.467 0 0 0-.236-.408Z" fill="#EF1161"></path><path d="m5.647 4.949 2.737-1.58a.095.095 0 0 0 .047-.082V2.093a.49.49 0 0 0-.183-.394.468.468 0 0 0-.523-.035l-1.135.655-3.013 1.738a.471.471 0 0 0-.236.408v4.083L3.34 9.87c0 .152.062.302.183.394.157.118.358.13.524.035l1.106-.639a.094.094 0 0 0 .047-.082l.001-3.859a.89.89 0 0 1 .446-.77Z" fill="#FC6E6B"></path></svg>

	

	<span>Dask</span>
	

	</div></a><div class="relative inline-block ">
	<button class="group mr-1 mb-1 md:mr-1.5 md:mb-1.5  rounded-lg rounded-br-none " type="button">
		<div class="tag tag-white   relative rounded-br-none pr-2.5"><svg class="text-black inline-block text-sm" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" fill="none" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M22.2812 12.2656L25.9532 15.7187C26.8932 16.6587 28.0913 17.3931 29.0313 16.4531C29.8594 15.7812 29.9332 14.1 29.9532 13.5C29.9532 11.5625 28.9219 8.375 27.25 6.71875C25.5604 5.04493 23.3782 3.91692 22.7032 3.78125L22.2812 12.2656Z" fill="#F5AB6A"></path><path d="M22.2812 12.2656L25.9532 15.7187C26.8932 16.6587 28.0913 17.3931 29.0313 16.4531C29.8594 15.7812 29.9332 14.1 29.9532 13.5C29.9532 11.5625 28.9219 8.375 27.25 6.71875C25.5604 5.04493 23.3782 3.91692 22.7032 3.78125L22.2812 12.2656Z" fill="url(#paint0_radial_18_31665)"></path><g filter="url(#filter0_f_18_31665)"><path d="M22.2849 12.1817L23.4375 13.2656L24.4375 4.70312C23.5121 4.1242 23.0198 3.96369 22.6563 3.89062L22.2849 12.1817Z" fill="url(#paint1_linear_18_31665)"></path></g><path d="M22.7969 3.05741L21.8437 2.65269C19.6421 1.96765 17.2344 1.81208 14.6719 2.23236C12.1094 2.65264 11.5156 3.2442 11.5156 3.2442C10.896 3.3674 10.8671 3.88898 11.1718 4.45363C13.0797 6.72576 15.176 13.4043 18.0469 14.2506C18.89 14.4662 21.3791 14.4776 22.2019 14.1593L22.3238 14.108C22.6692 13.9745 23.1672 13.2814 23.1875 12.9118L23.4219 4.03043C23.4523 3.59924 23.1484 3.25204 22.7969 3.05741Z" fill="url(#paint2_radial_18_31665)"></path><path d="M22.7969 3.05741L21.8437 2.65269C19.6421 1.96765 17.2344 1.81208 14.6719 2.23236C12.1094 2.65264 11.5156 3.2442 11.5156 3.2442C10.896 3.3674 10.8671 3.88898 11.1718 4.45363C13.0797 6.72576 15.176 13.4043 18.0469 14.2506C18.89 14.4662 21.3791 14.4776 22.2019 14.1593L22.3238 14.108C22.6692 13.9745 23.1672 13.2814 23.1875 12.9118L23.4219 4.03043C23.4523 3.59924 23.1484 3.25204 22.7969 3.05741Z" fill="url(#paint3_radial_18_31665)"></path><path d="M22.7969 3.05741L21.8437 2.65269C19.6421 1.96765 17.2344 1.81208 14.6719 2.23236C12.1094 2.65264 11.5156 3.2442 11.5156 3.2442C10.896 3.3674 10.8671 3.88898 11.1718 4.45363C13.0797 6.72576 15.176 13.4043 18.0469 14.2506C18.89 14.4662 21.3791 14.4776 22.2019 14.1593L22.3238 14.108C22.6692 13.9745 23.1672 13.2814 23.1875 12.9118L23.4219 4.03043C23.4523 3.59924 23.1484 3.25204 22.7969 3.05741Z" fill="url(#paint4_radial_18_31665)"></path><path d="M22.7969 3.05741L21.8437 2.65269C19.6421 1.96765 17.2344 1.81208 14.6719 2.23236C12.1094 2.65264 11.5156 3.2442 11.5156 3.2442C10.896 3.3674 10.8671 3.88898 11.1718 4.45363C13.0797 6.72576 15.176 13.4043 18.0469 14.2506C18.89 14.4662 21.3791 14.4776 22.2019 14.1593L22.3238 14.108C22.6692 13.9745 23.1672 13.2814 23.1875 12.9118L23.4219 4.03043C23.4523 3.59924 23.1484 3.25204 22.7969 3.05741Z" fill="url(#paint5_radial_18_31665)"></path><path d="M22.7969 3.05741L21.8437 2.65269C19.6421 1.96765 17.2344 1.81208 14.6719 2.23236C12.1094 2.65264 11.5156 3.2442 11.5156 3.2442C10.896 3.3674 10.8671 3.88898 11.1718 4.45363C13.0797 6.72576 15.176 13.4043 18.0469 14.2506C18.89 14.4662 21.3791 14.4776 22.2019 14.1593L22.3238 14.108C22.6692 13.9745 23.1672 13.2814 23.1875 12.9118L23.4219 4.03043C23.4523 3.59924 23.1484 3.25204 22.7969 3.05741Z" fill="url(#paint6_linear_18_31665)"></path><g filter="url(#filter1_f_18_31665)"><path d="M13.1016 2.72656C11.862 3.06924 11.5298 3.40016 11.5298 3.40016C10.9102 3.52335 10.936 4.11525 11.2408 4.6799C13.1487 6.95202 15.1361 13.2496 18.007 14.0958C18.2707 14.1633 18.6953 14.2107 19.1797 14.2344L13.1016 2.72656Z" fill="url(#paint7_linear_18_31665)"></path></g><path d="M12.2187 22.7187L15.7656 26.2031C16.7332 27.1171 17.3334 28.2487 16.4219 29.2188C15.7749 30.0687 14.2241 29.9933 13.625 30.0313C11.6883 30.0891 9.09014 29.5622 6.84373 27.5313C5.07737 25.9343 4.09321 23.688 3.93751 23.0156L12.2187 22.7187Z" fill="url(#paint8_radial_18_31665)"></path><path d="M12.2187 22.7187L15.7656 26.2031C16.7332 27.1171 17.3334 28.2487 16.4219 29.2188C15.7749 30.0687 14.2241 29.9933 13.625 30.0313C11.6883 30.0891 9.09014 29.5622 6.84373 27.5313C5.07737 25.9343 4.09321 23.688 3.93751 23.0156L12.2187 22.7187Z" fill="url(#paint9_radial_18_31665)"></path><g filter="url(#filter2_f_18_31665)"><path d="M12.0523 22.7916L13.2187 23.9375L4.81018 24.5721C4.4328 23.8671 4.20835 23.2768 4.14062 22.9844L12.0523 22.7916Z" fill="url(#paint10_linear_18_31665)"></path><path d="M12.0523 22.7916L13.2187 23.9375L4.81018 24.5721C4.4328 23.8671 4.20835 23.2768 4.14062 22.9844L12.0523 22.7916Z" fill="url(#paint11_radial_18_31665)"></path></g><path d="M2.99219 22.6484C2.07068 20.538 1.78913 17.4452 2.21096 15.0703C2.63279 12.6953 3.20759 11.951 3.20759 11.951C3.31231 11.3264 3.83281 11.2818 4.40626 11.5703C6.73409 13.4144 13.3119 15.2264 14.2432 18.0781C14.5947 19.3034 14.6279 21.3125 14.1641 22.5156C14.0409 22.8657 13.5 23.3516 13.0625 23.4141C12.625 23.4765 9.47656 23.3516 8.61719 23.3516C7.75781 23.3516 4.64844 23.6719 4.14062 23.6172C3.63281 23.5625 3.23437 23.2031 2.99219 22.6484Z" fill="#EC9F6A"></path><path d="M2.99219 22.6484C2.07068 20.538 1.78913 17.4452 2.21096 15.0703C2.63279 12.6953 3.20759 11.951 3.20759 11.951C3.31231 11.3264 3.83281 11.2818 4.40626 11.5703C6.73409 13.4144 13.3119 15.2264 14.2432 18.0781C14.5947 19.3034 14.6279 21.3125 14.1641 22.5156C14.0409 22.8657 13.5 23.3516 13.0625 23.4141C12.625 23.4765 9.47656 23.3516 8.61719 23.3516C7.75781 23.3516 4.64844 23.6719 4.14062 23.6172C3.63281 23.5625 3.23437 23.2031 2.99219 22.6484Z" fill="url(#paint12_radial_18_31665)"></path><path d="M2.99219 22.6484C2.07068 20.538 1.78913 17.4452 2.21096 15.0703C2.63279 12.6953 3.20759 11.951 3.20759 11.951C3.31231 11.3264 3.83281 11.2818 4.40626 11.5703C6.73409 13.4144 13.3119 15.2264 14.2432 18.0781C14.5947 19.3034 14.6279 21.3125 14.1641 22.5156C14.0409 22.8657 13.5 23.3516 13.0625 23.4141C12.625 23.4765 9.47656 23.3516 8.61719 23.3516C7.75781 23.3516 4.64844 23.6719 4.14062 23.6172C3.63281 23.5625 3.23437 23.2031 2.99219 22.6484Z" fill="url(#paint13_linear_18_31665)"></path><path d="M2.99219 22.6484C2.07068 20.538 1.78913 17.4452 2.21096 15.0703C2.63279 12.6953 3.20759 11.951 3.20759 11.951C3.31231 11.3264 3.83281 11.2818 4.40626 11.5703C6.73409 13.4144 13.3119 15.2264 14.2432 18.0781C14.5947 19.3034 14.6279 21.3125 14.1641 22.5156C14.0409 22.8657 13.5 23.3516 13.0625 23.4141C12.625 23.4765 9.47656 23.3516 8.61719 23.3516C7.75781 23.3516 4.64844 23.6719 4.14062 23.6172C3.63281 23.5625 3.23437 23.2031 2.99219 22.6484Z" fill="url(#paint14_radial_18_31665)"></path><path d="M2.99219 22.6484C2.07068 20.538 1.78913 17.4452 2.21096 15.0703C2.63279 12.6953 3.20759 11.951 3.20759 11.951C3.31231 11.3264 3.83281 11.2818 4.40626 11.5703C6.73409 13.4144 13.3119 15.2264 14.2432 18.0781C14.5947 19.3034 14.6279 21.3125 14.1641 22.5156C14.0409 22.8657 13.5 23.3516 13.0625 23.4141C12.625 23.4765 9.47656 23.3516 8.61719 23.3516C7.75781 23.3516 4.64844 23.6719 4.14062 23.6172C3.63281 23.5625 3.23437 23.2031 2.99219 22.6484Z" fill="url(#paint15_radial_18_31665)"></path><path d="M2.99219 22.6484C2.07068 20.538 1.78913 17.4452 2.21096 15.0703C2.63279 12.6953 3.20759 11.951 3.20759 11.951C3.31231 11.3264 3.83281 11.2818 4.40626 11.5703C6.73409 13.4144 13.3119 15.2264 14.2432 18.0781C14.5947 19.3034 14.6279 21.3125 14.1641 22.5156C14.0409 22.8657 13.5 23.3516 13.0625 23.4141C12.625 23.4765 9.47656 23.3516 8.61719 23.3516C7.75781 23.3516 4.64844 23.6719 4.14062 23.6172C3.63281 23.5625 3.23437 23.2031 2.99219 22.6484Z" fill="url(#paint16_radial_18_31665)"></path><g filter="url(#filter3_f_18_31665)"><path d="M2.70313 13.6719C3.04135 12.4711 3.36224 12.0555 3.36224 12.0555C3.46697 11.4309 3.98746 11.3864 4.56092 11.6749C6.88874 13.5189 13.0809 15.1104 14.0121 17.9622C14.1731 18.5231 14.2766 19.0394 14.3287 19.5128L2.70313 13.6719Z" fill="url(#paint17_linear_18_31665)"></path><path d="M2.70313 13.6719C3.04135 12.4711 3.36224 12.0555 3.36224 12.0555C3.46697 11.4309 3.98746 11.3864 4.56092 11.6749C6.88874 13.5189 13.0809 15.1104 14.0121 17.9622C14.1731 18.5231 14.2766 19.0394 14.3287 19.5128L2.70313 13.6719Z" fill="url(#paint18_linear_18_31665)"></path></g><path d="M9.83184 2.82184C6.62184 4.07184 4.07184 6.62184 2.82184 9.83184C2.48184 10.7118 2.82184 11.7018 3.62184 12.1918L14.0418 18.5418C14.6618 18.9218 15.4418 18.9218 16.0618 18.5418C17.0718 17.9218 17.9318 17.0718 18.5418 16.0618C18.9218 15.4418 18.9218 14.6618 18.5418 14.0418L12.1918 3.62184C11.7018 2.82184 10.7118 2.48184 9.83184 2.82184Z" fill="#D79453"></path><path d="M9.83184 2.82184C6.62184 4.07184 4.07184 6.62184 2.82184 9.83184C2.48184 10.7118 2.82184 11.7018 3.62184 12.1918L14.0418 18.5418C14.6618 18.9218 15.4418 18.9218 16.0618 18.5418C17.0718 17.9218 17.9318 17.0718 18.5418 16.0618C18.9218 15.4418 18.9218 14.6618 18.5418 14.0418L12.1918 3.62184C11.7018 2.82184 10.7118 2.48184 9.83184 2.82184Z" fill="url(#paint19_radial_18_31665)"></path><path d="M9.83184 2.82184C6.62184 4.07184 4.07184 6.62184 2.82184 9.83184C2.48184 10.7118 2.82184 11.7018 3.62184 12.1918L14.0418 18.5418C14.6618 18.9218 15.4418 18.9218 16.0618 18.5418C17.0718 17.9218 17.9318 17.0718 18.5418 16.0618C18.9218 15.4418 18.9218 14.6618 18.5418 14.0418L12.1918 3.62184C11.7018 2.82184 10.7118 2.48184 9.83184 2.82184Z" fill="url(#paint20_radial_18_31665)"></path><path d="M9.83184 2.82184C6.62184 4.07184 4.07184 6.62184 2.82184 9.83184C2.48184 10.7118 2.82184 11.7018 3.62184 12.1918L14.0418 18.5418C14.6618 18.9218 15.4418 18.9218 16.0618 18.5418C17.0718 17.9218 17.9318 17.0718 18.5418 16.0618C18.9218 15.4418 18.9218 14.6618 18.5418 14.0418L12.1918 3.62184C11.7018 2.82184 10.7118 2.48184 9.83184 2.82184Z" fill="url(#paint21_radial_18_31665)"></path><path d="M9.83184 2.82184C6.62184 4.07184 4.07184 6.62184 2.82184 9.83184C2.48184 10.7118 2.82184 11.7018 3.62184 12.1918L14.0418 18.5418C14.6618 18.9218 15.4418 18.9218 16.0618 18.5418C17.0718 17.9218 17.9318 17.0718 18.5418 16.0618C18.9218 15.4418 18.9218 14.6618 18.5418 14.0418L12.1918 3.62184C11.7018 2.82184 10.7118 2.48184 9.83184 2.82184Z" fill="url(#paint22_linear_18_31665)"></path><path d="M9.83184 2.82184C6.62184 4.07184 4.07184 6.62184 2.82184 9.83184C2.48184 10.7118 2.82184 11.7018 3.62184 12.1918L14.0418 18.5418C14.6618 18.9218 15.4418 18.9218 16.0618 18.5418C17.0718 17.9218 17.9318 17.0718 18.5418 16.0618C18.9218 15.4418 18.9218 14.6618 18.5418 14.0418L12.1918 3.62184C11.7018 2.82184 10.7118 2.48184 9.83184 2.82184Z" fill="url(#paint23_radial_18_31665)"></path><path d="M9.83184 2.82184C6.62184 4.07184 4.07184 6.62184 2.82184 9.83184C2.48184 10.7118 2.82184 11.7018 3.62184 12.1918L14.0418 18.5418C14.6618 18.9218 15.4418 18.9218 16.0618 18.5418C17.0718 17.9218 17.9318 17.0718 18.5418 16.0618C18.9218 15.4418 18.9218 14.6618 18.5418 14.0418L12.1918 3.62184C11.7018 2.82184 10.7118 2.48184 9.83184 2.82184Z" fill="url(#paint24_radial_18_31665)"></path><defs><filter id="filter0_f_18_31665" x="22.0349" y="3.64062" width="2.65265" height="9.875" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB"><feFlood flood-opacity="0" result="BackgroundImageFix"></feFlood><feBlend mode="normal" in="SourceGraphic" in2="BackgroundImageFix" result="shape"></feBlend><feGaussianBlur stdDeviation="0.125" result="effect1_foregroundBlur_18_31665"></feGaussianBlur></filter><filter id="filter1_f_18_31665" x="10.7815" y="2.47656" width="8.64819" height="12.0078" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB"><feFlood flood-opacity="0" result="BackgroundImageFix"></feFlood><feBlend mode="normal" in="SourceGraphic" in2="BackgroundImageFix" result="shape"></feBlend><feGaussianBlur stdDeviation="0.125" result="effect1_foregroundBlur_18_31665"></feGaussianBlur></filter><filter id="filter2_f_18_31665" x="3.89062" y="22.5416" width="9.57812" height="2.2804" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB"><feFlood flood-opacity="0" result="BackgroundImageFix"></feFlood><feBlend mode="normal" in="SourceGraphic" in2="BackgroundImageFix" result="shape"></feBlend><feGaussianBlur stdDeviation="0.125" result="effect1_foregroundBlur_18_31665"></feGaussianBlur></filter><filter id="filter3_f_18_31665" x="2.45312" y="11.2538" width="12.1255" height="8.50903" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB"><feFlood flood-opacity="0" result="BackgroundImageFix"></feFlood><feBlend mode="normal" in="SourceGraphic" in2="BackgroundImageFix" result="shape"></feBlend><feGaussianBlur stdDeviation="0.125" result="effect1_foregroundBlur_18_31665"></feGaussianBlur></filter><radialGradient id="paint0_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(22.8125 12.9375) rotate(42.7741) scale(12.5164 7.08839)"><stop offset="0.0937591" stop-color="#C05159"></stop><stop offset="0.553697" stop-color="#F6AC6A"></stop><stop offset="0.832916" stop-color="#FFD186"></stop><stop offset="0.916927" stop-color="#FFDC87"></stop></radialGradient><linearGradient id="paint1_linear_18_31665" x1="24.7344" y1="4.67187" x2="20.8594" y2="12.8906" gradientUnits="userSpaceOnUse"><stop stop-color="#EBD67C"></stop><stop offset="0.0655686" stop-color="#FFFFA6"></stop><stop offset="0.530552" stop-color="#F8C281"></stop><stop offset="0.937338" stop-color="#E99E6B"></stop></linearGradient><radialGradient id="paint2_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(22.9009 13.1847) rotate(-127.648) scale(14.3438 11.7966)"><stop stop-color="#FFBE66"></stop><stop offset="1" stop-color="#E2AE5B"></stop></radialGradient><radialGradient id="paint3_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(18 11.4375) rotate(53.9726) scale(11.9013 4.84018)"><stop stop-color="#D67C63"></stop><stop offset="1" stop-color="#D97D67" stop-opacity="0"></stop></radialGradient><radialGradient id="paint4_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(23 4.1875) rotate(45.7639) scale(3.31486 5.75622)"><stop stop-color="#FFE4A6"></stop><stop offset="0.711285" stop-color="#F8B76F"></stop><stop offset="1" stop-color="#F9B870" stop-opacity="0"></stop></radialGradient><radialGradient id="paint5_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(22.875 12.4375) rotate(88.9391) scale(3.37558 1.29066)"><stop stop-color="#FFBC67"></stop><stop offset="1" stop-color="#FFBC67" stop-opacity="0"></stop></radialGradient><linearGradient id="paint6_linear_18_31665" x1="20.375" y1="15.6875" x2="20.125" y2="12.7813" gradientUnits="userSpaceOnUse"><stop offset="0.461609" stop-color="#B45077"></stop><stop offset="0.855389" stop-color="#B75077" stop-opacity="0"></stop></linearGradient><linearGradient id="paint7_linear_18_31665" x1="12.9375" y1="2.57056" x2="18.5625" y2="14.3891" gradientUnits="userSpaceOnUse"><stop stop-color="#DDC173"></stop><stop offset="0.485173" stop-color="#D59F65"></stop><stop offset="1" stop-color="#E49966"></stop></linearGradient><radialGradient id="paint8_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(13.5625 23.5) rotate(109.113) scale(6.68078 10.2578)"><stop offset="0.165756" stop-color="#FFBF7E"></stop><stop offset="0.827674" stop-color="#DF8C6D"></stop><stop offset="1" stop-color="#B05A66"></stop></radialGradient><radialGradient id="paint9_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(15.1875 26) rotate(41.0652) scale(8.37243 2.03649)"><stop stop-color="#FFD483"></stop><stop offset="1" stop-color="#FFD688" stop-opacity="0"></stop></radialGradient><linearGradient id="paint10_linear_18_31665" x1="3.96063" y1="23.794" x2="13.3748" y2="23.5143" gradientUnits="userSpaceOnUse"><stop stop-color="#A8716F"></stop><stop offset="0.103615" stop-color="#B37173"></stop><stop offset="0.225484" stop-color="#DB9F84"></stop><stop offset="0.799889" stop-color="#F1BB8A"></stop><stop offset="1" stop-color="#FFD780"></stop></linearGradient><radialGradient id="paint11_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(11.4219 23.1719) rotate(-178.616) scale(3.23532 0.569081)"><stop offset="0.621498" stop-color="#AF5A3E"></stop><stop offset="1" stop-color="#B35445" stop-opacity="0"></stop></radialGradient><radialGradient id="paint12_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(13.625 19.125) rotate(-171.737) scale(15.2205 15.0767)"><stop offset="0.138435" stop-color="#FFB974"></stop><stop offset="0.403618" stop-color="#F2A56D"></stop><stop offset="0.925938" stop-color="#A16948"></stop></radialGradient><linearGradient id="paint13_linear_18_31665" x1="8.22184" y1="13.125" x2="6.81191" y2="15.4996" gradientUnits="userSpaceOnUse"><stop offset="0.610751" stop-color="#984847"></stop><stop offset="0.850075" stop-color="#9A4947" stop-opacity="0"></stop></linearGradient><radialGradient id="paint14_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(7.25 23.7461) scale(11.25 5.68361)"><stop stop-color="#C66364"></stop><stop offset="1" stop-color="#D4766B" stop-opacity="0"></stop></radialGradient><radialGradient id="paint15_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(7.875 23.5313) scale(10.0937 1.29657)"><stop stop-color="#B64B4B"></stop><stop offset="1" stop-color="#C56158" stop-opacity="0"></stop></radialGradient><radialGradient id="paint16_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(11.4375 19.875) rotate(-46.8882) scale(4.02385 7.51767)"><stop stop-color="#FFC083"></stop><stop offset="0.620218" stop-color="#FFBD7D" stop-opacity="0"></stop></radialGradient><linearGradient id="paint17_linear_18_31665" x1="2.8125" y1="13.0312" x2="14.5582" y2="18.9404" gradientUnits="userSpaceOnUse"><stop stop-color="#B89367"></stop><stop offset="1" stop-color="#C5835E"></stop></linearGradient><linearGradient id="paint18_linear_18_31665" x1="8.21875" y1="14.6406" x2="7.59349" y2="15.6717" gradientUnits="userSpaceOnUse"><stop offset="0.351552" stop-color="#A74746"></stop><stop offset="0.845198" stop-color="#A04346" stop-opacity="0"></stop></linearGradient><radialGradient id="paint19_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(15.5625 14.5625) rotate(140.244) scale(18.3733 13.7403)"><stop stop-color="#FDAE69"></stop><stop offset="0.729021" stop-color="#CE8C4F"></stop><stop offset="0.921546" stop-color="#AD7B45"></stop><stop offset="1" stop-color="#8B6B4A"></stop></radialGradient><radialGradient id="paint20_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(15.0625 7) rotate(65.3152) scale(11.0745 3.16547)"><stop offset="0.233237" stop-color="#FFD47C"></stop><stop offset="0.853648" stop-color="#FFD98B" stop-opacity="0"></stop></radialGradient><radialGradient id="paint21_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(15.3125 8.875) rotate(100.886) scale(6.6191 5.57808)"><stop offset="0.128419" stop-color="#FFD88C"></stop><stop offset="0.924134" stop-color="#FFBE7B" stop-opacity="0"></stop></radialGradient><linearGradient id="paint22_linear_18_31665" x1="7.25" y1="15.1875" x2="10.7588" y2="10.3142" gradientUnits="userSpaceOnUse"><stop offset="0.142353" stop-color="#C15F4D"></stop><stop offset="1" stop-color="#D58366" stop-opacity="0"></stop></linearGradient><radialGradient id="paint23_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(8.15625 15.7813) rotate(28.5422) scale(12.5574 1.96589)"><stop offset="0.149989" stop-color="#E4745D"></stop><stop offset="0.453292" stop-color="#C8604C"></stop><stop offset="0.632597" stop-color="#C0605F"></stop><stop offset="1" stop-color="#C0605F" stop-opacity="0"></stop></radialGradient><radialGradient id="paint24_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(1.40625 2.69067) rotate(46.0943) scale(22.3963)"><stop offset="0.935802" stop-color="#C17C61" stop-opacity="0"></stop><stop offset="0.982109" stop-color="#C17C61"></stop></radialGradient></defs></svg>

	

	<span>Croissant</span>
	

	<div class="border-br-gray-200 absolute bottom-0.5 right-0.5 h-1 w-1 border-[3px] border-l-transparent border-t-transparent border-b-gray-200 border-r-gray-200 dark:border-b-gray-700 dark:border-r-gray-700"></div></div>
		
		</button>
	
	
	</div>

	<button class="tag tag-ghost px-1! -ml-0.5 mb-1 md:mb-1.5" type="button">+ 1</button></div></div>

		<div class="flex flex-col-reverse lg:flex-row lg:items-center lg:justify-between"><div class="-mb-px flex h-12 items-center overflow-x-auto overflow-y-hidden ">
	<a class="tab-alternate active" href="/datasets/griffin/ChemSum"><svg class="mr-1.5 text-gray-400 flex-none" style="" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 24 24"><path class="uim-quaternary" d="M20.23 7.24L12 12L3.77 7.24a1.98 1.98 0 0 1 .7-.71L11 2.76c.62-.35 1.38-.35 2 0l6.53 3.77c.29.173.531.418.7.71z" opacity=".25" fill="currentColor"></path><path class="uim-tertiary" d="M12 12v9.5a2.09 2.09 0 0 1-.91-.21L4.5 17.48a2.003 2.003 0 0 1-1-1.73v-7.5a2.06 2.06 0 0 1 .27-1.01L12 12z" opacity=".5" fill="currentColor"></path><path class="uim-primary" d="M20.5 8.25v7.5a2.003 2.003 0 0 1-1 1.73l-6.62 3.82c-.275.13-.576.198-.88.2V12l8.23-4.76c.175.308.268.656.27 1.01z" fill="currentColor"></path></svg>
	Dataset card
	

	
		</a><a class="tab-alternate" href="/datasets/griffin/ChemSum/viewer/"><svg class="mr-1.5 text-gray-400 flex-none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 12 12"><path fill="currentColor" d="M2.5 2h7a1 1 0 0 1 1 1v6a1 1 0 0 1-1 1h-7a1 1 0 0 1-1-1V3a1 1 0 0 1 1-1Zm0 2v2h3V4h-3Zm4 0v2h3V4h-3Zm-4 3v2h3V7h-3Zm4 0v2h3V7h-3Z"></path></svg>
	Data Studio
	

	
		</a><a class="tab-alternate" href="/datasets/griffin/ChemSum/tree/main"><svg class="mr-1.5 text-gray-400 flex-none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 24 24"><path class="uim-tertiary" d="M21 19h-8a1 1 0 0 1 0-2h8a1 1 0 0 1 0 2zm0-4h-8a1 1 0 0 1 0-2h8a1 1 0 0 1 0 2zm0-8h-8a1 1 0 0 1 0-2h8a1 1 0 0 1 0 2zm0 4h-8a1 1 0 0 1 0-2h8a1 1 0 0 1 0 2z" opacity=".5" fill="currentColor"></path><path class="uim-primary" d="M9 19a1 1 0 0 1-1-1V6a1 1 0 0 1 2 0v12a1 1 0 0 1-1 1zm-6-4.333a1 1 0 0 1-.64-1.769L3.438 12l-1.078-.898a1 1 0 0 1 1.28-1.538l2 1.667a1 1 0 0 1 0 1.538l-2 1.667a.999.999 0 0 1-.64.231z" fill="currentColor"></path></svg>
	<span class="xl:hidden">Files</span>
		<span class="hidden xl:inline">Files and versions</span>
	

	
		</a><a class="tab-alternate" href="/datasets/griffin/ChemSum/discussions"><svg class="mr-1.5 text-gray-400 flex-none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M20.6081 3C21.7684 3 22.8053 3.49196 23.5284 4.38415C23.9756 4.93678 24.4428 5.82749 24.4808 7.16133C24.9674 7.01707 25.4353 6.93643 25.8725 6.93643C26.9833 6.93643 27.9865 7.37587 28.696 8.17411C29.6075 9.19872 30.0124 10.4579 29.8361 11.7177C29.7523 12.3177 29.5581 12.8555 29.2678 13.3534C29.8798 13.8646 30.3306 14.5763 30.5485 15.4322C30.719 16.1032 30.8939 17.5006 29.9808 18.9403C30.0389 19.0342 30.0934 19.1319 30.1442 19.2318C30.6932 20.3074 30.7283 21.5229 30.2439 22.6548C29.5093 24.3704 27.6841 25.7219 24.1397 27.1727C21.9347 28.0753 19.9174 28.6523 19.8994 28.6575C16.9842 29.4379 14.3477 29.8345 12.0653 29.8345C7.87017 29.8345 4.8668 28.508 3.13831 25.8921C0.356375 21.6797 0.754104 17.8269 4.35369 14.1131C6.34591 12.058 7.67023 9.02782 7.94613 8.36275C8.50224 6.39343 9.97271 4.20438 12.4172 4.20438H12.4179C12.6236 4.20438 12.8314 4.2214 13.0364 4.25468C14.107 4.42854 15.0428 5.06476 15.7115 6.02205C16.4331 5.09583 17.134 4.359 17.7682 3.94323C18.7242 3.31737 19.6794 3 20.6081 3ZM20.6081 5.95917C20.2427 5.95917 19.7963 6.1197 19.3039 6.44225C17.7754 7.44319 14.8258 12.6772 13.7458 14.7131C13.3839 15.3952 12.7655 15.6837 12.2086 15.6837C11.1036 15.6837 10.2408 14.5497 12.1076 13.1085C14.9146 10.9402 13.9299 7.39584 12.5898 7.1776C12.5311 7.16799 12.4731 7.16355 12.4172 7.16355C11.1989 7.16355 10.6615 9.33114 10.6615 9.33114C10.6615 9.33114 9.0863 13.4148 6.38031 16.206C3.67434 18.998 3.5346 21.2388 5.50675 24.2246C6.85185 26.2606 9.42666 26.8753 12.0653 26.8753C14.8021 26.8753 17.6077 26.2139 19.1799 25.793C19.2574 25.7723 28.8193 22.984 27.6081 20.6107C27.4046 20.212 27.0693 20.0522 26.6471 20.0522C24.9416 20.0522 21.8393 22.6726 20.5057 22.6726C20.2076 22.6726 19.9976 22.5416 19.9116 22.222C19.3433 20.1173 28.552 19.2325 27.7758 16.1839C27.639 15.6445 27.2677 15.4256 26.746 15.4263C24.4923 15.4263 19.4358 19.5181 18.3759 19.5181C18.2949 19.5181 18.2368 19.4937 18.2053 19.4419C17.6743 18.557 17.9653 17.9394 21.7082 15.6009C25.4511 13.2617 28.0783 11.8545 26.5841 10.1752C26.4121 9.98141 26.1684 9.8956 25.8725 9.8956C23.6001 9.89634 18.2311 14.9403 18.2311 14.9403C18.2311 14.9403 16.7821 16.496 15.9057 16.496C15.7043 16.496 15.533 16.4139 15.4169 16.2112C14.7956 15.1296 21.1879 10.1286 21.5484 8.06535C21.7928 6.66715 21.3771 5.95917 20.6081 5.95917Z" fill="#FF9D00"></path><path d="M5.50686 24.2246C3.53472 21.2387 3.67446 18.9979 6.38043 16.206C9.08641 13.4147 10.6615 9.33111 10.6615 9.33111C10.6615 9.33111 11.2499 6.95933 12.59 7.17757C13.93 7.39581 14.9139 10.9401 12.1069 13.1084C9.29997 15.276 12.6659 16.7489 13.7459 14.713C14.8258 12.6772 17.7747 7.44316 19.304 6.44221C20.8326 5.44128 21.9089 6.00204 21.5484 8.06532C21.188 10.1286 14.795 15.1295 15.4171 16.2118C16.0391 17.2934 18.2312 14.9402 18.2312 14.9402C18.2312 14.9402 25.0907 8.49588 26.5842 10.1752C28.0776 11.8545 25.4512 13.2616 21.7082 15.6008C17.9646 17.9393 17.6744 18.557 18.2054 19.4418C18.7372 20.3266 26.9998 13.1351 27.7759 16.1838C28.5513 19.2324 19.3434 20.1173 19.9117 22.2219C20.48 24.3274 26.3979 18.2382 27.6082 20.6107C28.8193 22.9839 19.2574 25.7722 19.18 25.7929C16.0914 26.62 8.24723 28.3726 5.50686 24.2246Z" fill="#FFD21E"></path></svg>
	Community
	<div class="ml-1.5 flex h-4 min-w-[1rem] items-center justify-center rounded px-1 text-xs leading-none shadow-sm bg-black text-white dark:bg-gray-800 dark:text-gray-200">1</div>

	
		</a></div>
	
			</div></div></header>
</div>
	
<div class="container relative flex flex-col md:grid md:space-y-0 w-full md:grid-cols-12 md:flex-1 md:grid-rows-full space-y-4 md:gap-6 ">
		<section class="pt-6 border-gray-100 md:col-span-8 pb-24 relative break-words copiable-code-container">
				<div class="SVELTE_HYDRATER contents" data-target="UnsafeBanner" data-props="{&quot;classNames&quot;:&quot;mb-4&quot;,&quot;repoId&quot;:&quot;griffin/ChemSum&quot;,&quot;repoType&quot;:&quot;dataset&quot;,&quot;minLevel&quot;:&quot;unsafe&quot;}"></div>
					<div class="SVELTE_HYDRATER contents" data-target="DatasetViewer" data-props="{&quot;data&quot;:{&quot;kind&quot;:&quot;DatasetAndSampleData&quot;,&quot;datasetInfo&quot;:[{&quot;isValid&quot;:true,&quot;href&quot;:&quot;/datasets/griffin/ChemSum/tree/refs%2Fconvert%2Fparquet/&quot;,&quot;label&quot;:&quot;Size of the auto-converted Parquet files:&quot;,&quot;value&quot;:&quot;1.9 GB&quot;},{&quot;isValid&quot;:true,&quot;href&quot;:&quot;&quot;,&quot;label&quot;:&quot;Number of rows:&quot;,&quot;value&quot;:&quot;118,956&quot;}],&quot;partial&quot;:false,&quot;configsData&quot;:{&quot;configInfos&quot;:[{&quot;name&quot;:&quot;default&quot;,&quot;status&quot;:&quot;ok&quot;,&quot;numRows&quot;:118956}],&quot;selectedConfig&quot;:&quot;default&quot;,&quot;hasSelectedConfigParquet&quot;:true},&quot;splitsData&quot;:{&quot;splitInfos&quot;:[{&quot;name&quot;:&quot;train&quot;,&quot;numRows&quot;:115956},{&quot;name&quot;:&quot;validation&quot;,&quot;numRows&quot;:1000},{&quot;name&quot;:&quot;test&quot;,&quot;numRows&quot;:2000}],&quot;selectedSplit&quot;:&quot;train&quot;},&quot;sampleData&quot;:{&quot;dataset&quot;:&quot;griffin/ChemSum&quot;,&quot;config&quot;:&quot;default&quot;,&quot;split&quot;:&quot;train&quot;,&quot;capabilities&quot;:{&quot;rows&quot;:true,&quot;search&quot;:true,&quot;filter&quot;:true,&quot;statistics&quot;:true},&quot;navigation&quot;:{&quot;p&quot;:0},&quot;jwt&quot;:&quot;eyJhbGciOiJFZERTQSJ9.eyJyZWFkIjp0cnVlLCJwZXJtaXNzaW9ucyI6eyJyZXBvLmNvbnRlbnQucmVhZCI6dHJ1ZX0sImlhdCI6MTc0MjkyMzEwMSwic3ViIjoiL2RhdGFzZXRzL2dyaWZmaW4vQ2hlbVN1bSIsImV4cCI6MTc0MjkyNjcwMSwiaXNzIjoiaHR0cHM6Ly9odWdnaW5nZmFjZS5jbyJ9.oLUPfE4h-9UDlgAxT6uVqCWV4yWWKcm4EwZAooNrHrciAGQ-wfxQWM9VVqX2jG96jTGbWW5D6OcmPvvTgFpPAA&quot;,&quot;sampleData&quot;:{&quot;columns&quot;:[{&quot;name&quot;:&quot;title&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;title&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:0,&quot;max&quot;:1218,&quot;mean&quot;:106.43075,&quot;median&quot;:103,&quot;std&quot;:36.07426,&quot;histogram&quot;:{&quot;hist&quot;:[80905,34771,260,12,1,3,3,0,0,1],&quot;bin_edges&quot;:[0,122,244,366,488,610,732,854,976,1098,1218]}}}},{&quot;name&quot;:&quot;abstract&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;abstract&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:239,&quot;max&quot;:18177,&quot;mean&quot;:1344.4474,&quot;median&quot;:1245,&quot;std&quot;:648.88405,&quot;histogram&quot;:{&quot;hist&quot;:[106085,8549,1286,29,5,1,0,0,0,1],&quot;bin_edges&quot;:[239,2033,3827,5621,7415,9209,11003,12797,14591,16385,18177]}}}},{&quot;name&quot;:&quot;uuid&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;uuid&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:0,&quot;max&quot;:100,&quot;mean&quot;:89.29988,&quot;median&quot;:100,&quot;std&quot;:16.0604,&quot;histogram&quot;:{&quot;hist&quot;:[3,67,463,1556,3701,7053,10425,13440,14676,64572],&quot;bin_edges&quot;:[0,11,22,33,44,55,66,77,88,99,100]}}}},{&quot;name&quot;:&quot;source_toks&quot;,&quot;align&quot;:&quot;right&quot;,&quot;type&quot;:&quot;int64&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;source_toks&quot;,&quot;column_type&quot;:&quot;int&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:510,&quot;max&quot;:158923,&quot;mean&quot;:4352.48135,&quot;median&quot;:3927.5,&quot;std&quot;:2616.79439,&quot;histogram&quot;:{&quot;hist&quot;:[115484,418,41,6,5,1,0,0,0,1],&quot;bin_edges&quot;:[510,16352,32194,48036,63878,79720,95562,111404,127246,143088,158923]}}}},{&quot;name&quot;:&quot;target_toks&quot;,&quot;align&quot;:&quot;right&quot;,&quot;type&quot;:&quot;int64&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;target_toks&quot;,&quot;column_type&quot;:&quot;int&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:50,&quot;max&quot;:2234,&quot;mean&quot;:189.53765,&quot;median&quot;:175,&quot;std&quot;:95.97534,&quot;histogram&quot;:{&quot;hist&quot;:[103082,9981,2794,73,22,2,0,1,0,1],&quot;bin_edges&quot;:[50,269,488,707,926,1145,1364,1583,1802,2021,2234]}}}},{&quot;name&quot;:&quot;compression&quot;,&quot;align&quot;:&quot;right&quot;,&quot;type&quot;:&quot;float64&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;compression&quot;,&quot;column_type&quot;:&quot;float&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:1.12109,&quot;max&quot;:845.33511,&quot;mean&quot;:26.65884,&quot;median&quot;:22.03884,&quot;std&quot;:20.26429,&quot;histogram&quot;:{&quot;hist&quot;:[113708,2031,170,26,16,1,1,1,1,1],&quot;bin_edges&quot;:[1.12109,85.54249,169.96389,254.3853,338.8067,423.2281,507.6495,592.0709,676.49231,760.91371,845.33511]}}}},{&quot;name&quot;:&quot;headers&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;headers&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:0,&quot;max&quot;:35531,&quot;mean&quot;:526.08096,&quot;median&quot;:365,&quot;std&quot;:773.15047,&quot;histogram&quot;:{&quot;hist&quot;:[114710,980,208,43,9,2,0,2,0,2],&quot;bin_edges&quot;:[0,3554,7108,10662,14216,17770,21324,24878,28432,31986,35531]}}}},{&quot;name&quot;:&quot;sections&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;sections&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:3324,&quot;max&quot;:584955,&quot;mean&quot;:28849.62609,&quot;median&quot;:26042.5,&quot;std&quot;:17068.34678,&quot;histogram&quot;:{&quot;hist&quot;:[112037,3595,228,51,20,11,5,5,1,3],&quot;bin_edges&quot;:[3324,61488,119652,177816,235980,294144,352308,410472,468636,526800,584955]}}}},{&quot;name&quot;:&quot;article_source&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;article_source&quot;,&quot;column_type&quot;:&quot;string_label&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;no_label_count&quot;:0,&quot;no_label_proportion&quot;:0,&quot;n_unique&quot;:9,&quot;frequencies&quot;:{&quot;Beilstein&quot;:1789,&quot;PubMed Author Manuscript&quot;:56230,&quot;Chemistry Open&quot;:390,&quot;Nature Communications Chemistry&quot;:552,&quot;Chem Cell&quot;:525,&quot;Royal Society of Chemistry (RSC)&quot;:9074,&quot;PubMed Open Access&quot;:28825,&quot;Scientific Reports - Nature&quot;:6666,&quot;ChemRxiv&quot;:11905}}}}],&quot;rows&quot;:[{&quot;rowIdx&quot;:0,&quot;cells&quot;:{&quot;title&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Solvent-dependent conductance decay constants in single cluster junctions&quot;},&quot;abstract&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Single-molecule conductance measurements have focused primarily on organic molecular systems. Here, we carry out scanning tunneling microscope-based break-junction measurements on a series of metal chalcogenide Co 6 Se 8 clusters capped with conducting ligands of varying lengths. We compare these measurements with those of individual free ligands and find that the conductance of these clusters and the free ligands have different decay constants with increasing ligand length. We also show, through measurements in two different solvents, 1-bromonaphthalene and 1,2,4-trichlorobenzene, that the conductance decay of the clusters depends on the solvent environment. We discuss several mechanisms to explain our observations.&quot;},&quot;uuid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;solvent-dependent_conductance_decay_constants_in_single_cluster_junctions&quot;},&quot;source_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:2400,&quot;string&quot;:&quot;2,400&quot;},&quot;target_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:99,&quot;string&quot;:&quot;99&quot;},&quot;compression&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:24.242424242424242,&quot;string&quot;:&quot;24.242424&quot;},&quot;headers&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Introduction<!>Results and discussion<!>Conclusions&quot;},&quot;sections&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;<p>Controlling charge transport through molecular junctions is critical to the realization of nanoscale electronic devices. 1,2 While numerous organic molecules have been studied as connecting wires for single-molecule junction studies, [3][4][5][6][7][8][9][10] very little is known about the effect of metal complexes in these types of junctions. [11][12][13][14] We recently reported that we could incorporate metal chalcogenide molecular clusters in single-molecule electrical circuits. 15 In this study, in order to determine how transport through such systems depends on molecular length, we connect these same clusters to conducting ligands of varying lengths. We have found that the inclusion of the cluster in the molecular circuit reduces the effect of ligand length on conductance decay with apparent molecular size. Moreover, we have found that the decay constant is impacted greatly by changing the solvent from 1,2,4-trichlorobenzene (TCB) to 1bromonaphthalene (BrN). Specically, the decay constant of the cluster is 0.04 ÅÀ1 in BrN, while it is 0.12 ÅÀ1 in TCB. We consider two possible mechanisms to explain these remarkable observations. Our work demonstrates, for the rst time, a molecular system where the tunneling decay constant can be modied by altering the environment around the molecule.</p><!><p>The single cluster circuits that we have designed, assembled, and studied consist of an atomically dened Co 6 Se 8 molecular cluster 16,17 (Fig. 1a) wired between nanoscale electrodes. The wiring is formed from bifunctional, conjugated ligands (Fig. 1b) that bind specically and directionally to the electrode and to the cluster. We employ an atomically dened segment of polyacetylene 18 that has an arylphosphine group on one terminus that coordinates to a cobalt atom on the clusters and an arylthiomethyl group on the other terminus that attaches to the Au electrode. 19,20 The mono-, di-, and triene ligands are L1, L2, and L3 and the corresponding clusters are 1, 2, and 3, respectively. Fig. 1c shows the molecular structure of 1 as determined by single crystal X-ray diffraction (SCXRD). 15 We measured the conductance of both the individual molecular clusters (1-3) and the free conducting ligands (L1-L3) using a scanning tunneling microscope-based break-junction (STM-BJ) technique. 21 In this technique, an Au STM tip and substrate are repeatedly brought into and out of contact to form and break Au-Au point contacts in solutions of the target compounds. During this process, a bias voltage is applied across the junction while current is measured in order to determine conductance (G ¼ I/V) of the junction. The measurements are repeated thousands of times, and the data is analyzed to reveal statistically signicant results. The data is processed by compiling thousands of individual conductance traces into one-dimensional, logarithmically-binned conductance histograms. 22 We further generate two-dimensional (2D) histograms of the conductance versus displacement by aligning each conductance trace aer the point contact ruptures (at a conductance of 0.5 G 0 ) and overlaying all conductance traces.</p><p>In order to characterize transport through the molecular clusters and the free ligands, we measured the conductance of 1-3 and L1-L3 in two different solvents, BrN and TCB. These solvents were chosen taking into consideration the solubility of both the ligand and the cluster systems as well as for their varied affinity to gold electrodes. 23 Fig. 2a and b contain the onedimensional histograms for the measurements in BrN, and Fig. 2d and e show the same for the measurements in TCB. The 2D histograms for 1 and L1 in each solvent are insets in the respective gures. The 2D histograms show a signicant difference in length of the molecular feature for 1 and for L1. Moreover, the cluster junction lengths measured from the 2D histograms correlate with the molecular lengths of the cluster with the ligands fully extended (measured in BrN: 9 Å and 21 Å, and expected from SCXRD: 13 Å and 32 Å, for L1 and 1 respectively). Despite the additional complexity of the cluster system, we conclude that we are indeed probing transport through Au-ligand-cluster-ligand-Au junctions based on this large difference in the observed lengths. Furthermore, the histograms in Fig. 2a and d show shoulders, with an increasing prominence for the longer-ligand systems. Comparing the conductance of these shoulders with the ligand conductance in Fig. 2b and e, we attribute these shoulders to free ligands, that is, ligands that have detached from the clusters.</p><p>We t the peaks of all conductance histograms for both solvents with a Gaussian function and plot the peak conductance values versus the number of C]C units or \&quot;enes\&quot; in each molecule (Fig. 2c and f). In both solvents, and for both free ligand and cluster, we observe that the conductance decreases exponentially with increasing molecular length following the relationship G $ e Àbn , where n is the number of \&quot;ene\&quot; units in the backbone and b is the decay constant. We report the decay constant per Angstrom using a length of 2.48 Å per \&quot;ene\&quot; unit. The decay constant for the free ligand series is essentially independent of the solvent (b ¼ 0.15 ÅÀ1 in TCB and 0.17 ÅÀ1 BrN).</p><p>The unexpected result is the factor of 3 difference in the decay constants of the cluster series in different solvents as can be seen comparing Fig. 2c and f. In TCB, the b of the cluster system is 0.12 ÅÀ1 , and in BrN it is 0.04 ÅÀ1 . We note that the difference between the decay constant of the ligand and that of the cluster is greater in BrN than in TCB. Furthermore, the absolute values of the conductance of the cluster series are signicantly higher when measured in BrN than in TCB, with the conductance of 3 being almost an order of magnitude higher in BrN compared to TCB. Such a solvent-induced effect on the conductance has been observed in other systems, and this has been attributed to the solvent's ability to modulate the electrode work function. 23,24 These ndings are summarized: regardless of solvent the effect of C]C chain-length on conductance is less pronounced in 1-3 than in L1-L3. Furthermore, the conductance and the decay of L1-L3 are essentially insensitive to the choice of solvent, while the solvent signicantly inuences those of 1-3.</p><p>To understand these results we consider several possible mechanisms of charge transport through these junctions. Charge transport can occur via a coherent off-resonance process through an orbital on the ligand-cluster-ligand assembly that is coupled to both electrodes. In that situation, the conductance depends on at least two related factors: (1) the energy of this conducting orbital relative to the metal E F , and (2) the coupling between this orbital and both electrodes. 25 As the length of the molecule increases, the HOMO-LUMO gap narrows, and if conductance were just related to energy level alignment, one would naively expect conductance to actually increase. However, transport through the junction is also related to how well the conducting orbital overlaps with the leads, and since the orbital is more delocalized over a longer conjugated molecule, this overlap decreases with increasing length. The conductance thus typically decays exponentially with increasing molecular length. Specically, as the conjugated backbone gets longer, the molecular orbital is delocalized over a longer molecule, and since the orbital is normalized, a smaller fraction of its amplitude resides on the sulfur atoms; therefore the coupling between the molecule and the electrodes decreases.</p><p>If we assume that the conducting orbitals of the cluster and of the ligand for a given length are similar in both character and energy, we can develop a simple tight-binding model of the molecular junctions. The objective of this model is not to reproduce the experimental data but to examine and illustrate how the additional electronic structure of the cluster impacts transport through the system. Our tight-binding model is schematically presented in the insets of Fig. 3a and b for the ligand and the cluster respectively. For the conducting ligands, we assign a single energy level, 3, for each unit, and allow nearest neighbors to be coupled by d. The terminal units are coupled to the Au electrodes using an imaginary self-energy, iG/2. We apply a similar model for the cluster, adding an additional energy level, E 0 , between two ligands and coupling this site to its nearest neighbor ligand states with s. We compute the transmission functions for these model systems using a Green's function approach (see the ESI for a detailed description †). 25,26 Sample computed transmission functions are shown in Fig. 3a and b using the same values for 3, d and G for the ligand and the cluster series. The transmission functions display resonances at energy values corresponding to the molecular orbitals of the system where the probability of an electron being transmitted through the system is unity. The transmission function for L1 contains one resonance at energy 3, while longer ligands have resonances equal to the number of sites in the corresponding model. As the length of the molecule increases, the frontier resonance moves closer to E F but also narrows, which is a consequence of the frontier orbital being delocalized over a longer molecular backbone. Upon comparing the transmission functions for the ligands with those of the clusters, we see that the clusters contain resonances that are closer to E F than their ligand counterparts, but with narrower full widths at half max (i.e. they are more poorly coupled to the leads). This observation leads to a lower transmission at E F ; more importantly, it also leads to a conductance that is more sensitive to the exact location of the E F .</p><p>In Fig. 3c, we show the conductances that are determined from the tight-binding model for each molecule versus the number of ligand levels in the molecule (using the same model parameter values for both series). From the t to these values, it is clear that the predicted decay constants are essentially the same for the ligand and cluster series. We use one set of E 0 and s values to calculate the representative transmission/conductance functions shown in Fig. 3a and b. Regardless of what value is assigned to E 0 and s, we nd that this model predicts very similar decay constants for the two systems (Fig. 3d).</p><p>Our tight-binding calculations suggest that the addition of a cluster level E 0 between two ligands cannot explain the observed change in b. In other words, the ligand and cluster series should have the same b values, unless the energy alignment of the cluster resonance is altered relative to the electrode Fermi level in this model. We have three sets of observations that are consistent with a change in E F : (1) b of the cluster in BrN is signicantly lower than in TCB, (2) b values measured in both solvents are almost the same for the ligand series, and (3) b of the cluster is lower than that of the ligand in both solvents. The steeper transmission curves of the cluster series in Fig. 3 indicate that the resonance energies are closer to E F . Within this coherent transport model, we can see that a small change in E F will result in a large shi in b for the cluster relative to the ligand. For instance, changing E F by À0.5 eV shis the b value to 0.1 ÅÀ1 for the clusters while a similar change in E F for the ligand changes b to 0.3 ÅÀ1 . These results, when viewed in light of the known ability of solvent-binding to produce changes in E F , 23 point to BrN shiing E F closer to resonance relative to TCB. This effect is compounded by the sensitivity of the metal. The free ligand and the cluster have very different characteristics (e.g., size, steric hindrance, redox behaviour, dielectric constant polarizability and binding ability) that will result in different shis in E F .</p><p>We also consider a hopping mechanism for charge transfer, a process generally mediated by an activation-controlled reaction (e.g., a thermally induced conformational change or an electron transfer reaction). 27,28 We rst rule out the possibility that such a conformational change can occur within the ligand. 3 We also refute the process involving a direct through-space charge transfer from the electrode to an unoccupied molecular level on the cluster through a resonant transfer process. 14 In this picture, the cluster does not have to be chemically attached to the electrodes to form a conducting junction and the charge transfer efficiency depends on the core-electrode spacing. We discount this mechanism based on a previously published study in which we demonstrated that our clusters form molecular junctions by bonding their terminal thiomethyl groups to the Au electrodes. 15 By varying the substitution pattern or removing the aurophilic functionality, we can modulate or completely shut down the conductivity of these molecular junctions, suggesting that there is an orbital pathway for the transport of charge in these cluster systems. These ndings refute the idea of direct through-space charge transfer mediated by an orbital localized on the core.</p><p>We are le with a hopping mechanism in which the charge tunnels from the source electrode across the ligand to the cluster core and then transfers to the drain electrode through a second coherent tunneling process. Such a transport process requires that the cluster can reversibly change its oxidation state with each charge transfer. Since the applied bias in these measurements is not small ($0.5 V) and the cluster core Co 6 Se 8 is redox active, it is plausible that such a hopping process is at play. In this case, the activation energy arises from the charge transfer process reorganization energy, which can be strongly inuenced by the solvent. This mechanism is consistent with our observation that b changes with solvent. Within our experimental constraints, it is therefore difficult to conclusively establish which process (off-resonance tunneling or hopping) is at work in our single cluster junction system.</p><!><p>In summary, we measured charge transport through molecular clusters with ligands of different lengths and showed that the conductance decay depends on the solvent used for these measurements. Our results illustrate a novel effect that allows the environment to alter the conductance decay constants. This study opens up the possibility to carry out conductance measurements in which clusters can be controllably gated by changing the environment. 20 While the conducting ligands alone are limited to a one-dimensional system, the threedimensional architecture of the metal chalcogenide cluster allows us to envision novel electronic devices where a molecular cluster is contacted by electrodes at multiple locations.</p>&quot;},&quot;article_source&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Royal Society of Chemistry (RSC)&quot;}}},{&quot;rowIdx&quot;:1,&quot;cells&quot;:{&quot;title&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Characterization of Frequency-chirped Dynamic Nuclear Polarization in Rotating Solids&quot;},&quot;abstract&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Continuous wave (CW) dynamic nuclear polarization (DNP) is used with magic angle spinning (MAS) to enhance the typically poor sensitivity of nuclear magnetic resonance (NMR) by orders of magnitude. In a recent publication we show that further enhancement is obtained by using a frequency-agile gyrotron to chirp incident microwave frequency through the electron resonance frequency during DNP transfer. Here we characterize the effect of chirped MAS DNP by investigating the sweep time, sweep width, center-frequency, and electron Rabi frequency of the chirps. We show the advantages of chirped DNP with a tritylnitroxide biradical, and a lack of improvement with chirped DNP using AMUPol, a nitroxide biradical. Frequency-chirped DNP on a model system of urea in a cryoprotecting matrix yields an enhancement of 142, 21% greater than that obtained with CW DNP. We then go beyond this model system and apply chirped DNP to intact human cells. In human Jurkat cells, frequency-chirped DNP improves enhancement by 24% over CW DNP. The characterization of the chirped DNP effect reveals instrument limitations on sweep time and sweep width, promising even greater increases in sensitivity with further technology development. These improvements in gyrotron technology, frequency-agile methods, and incell applications are expected to play a significant role in the advancement of MAS DNP.&quot;},&quot;uuid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;characterization_of_frequency-chirped_dynamic_nuclear_polarization_in_rotating_solids&quot;},&quot;source_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:2758,&quot;string&quot;:&quot;2,758&quot;},&quot;target_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:208,&quot;string&quot;:&quot;208&quot;},&quot;compression&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:13.259615384615385,&quot;string&quot;:&quot;13.259615&quot;},&quot;headers&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Introduction<!>NMR Experiments<!>7<!>Sample Preparation<!>Results and Discussion<!>Frequency-chirped DNP on a Model System<!>Frequency-chirped DNP in Intact Jurkat Cells<!>Power Dependence of CW and Frequency-chirped DNP<!>Characterization of Frequency-chirped DNP<!>Conclusion&quot;},&quot;sections&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;<p>Dynamic nuclear polarization (DNP) is commonly used to improve the inherent insensitivity of nuclear magnetic resonance (NMR) spectroscopy [1][2][3][4][5][6][7][8][9][10][11][12][13]. Typically, only continuous wave (CW) microwave methods have been employed with magic angle spinning (MAS) DNP. The solid effect and the cross effect are the primary DNP mechanisms used in moderate magnetic field strengths of 5-14 Tesla (T) [14][15][16][17][18]. While CW approaches can significantly increase NMR sensitivity, they have limitations. Except in certain model systems [6,19,20], the solid effect and cross effect are inefficient at room temperature due to short longitudinal electron relaxation times. To perform CW DNP, samples are commonly cooled to <120 K, which adds complexity not only to the instrumentation, but also often leads to a loss of spectral resolution [14,21]. Arrested molecular motion at these temperatures can cause substantial line broadening in most samples [3,[21][22][23]. The cross effect and solid effect also exhibit worse performance at higher magnetic field, with cross effect efficiency decreasing as 1/B0 and that of solid effect as 1/B0 2 [15,24,25]. Therefore new mechanisms will be required for efficient DNP at magnetic fields of 28 T and higher.</p><p>Frequency-chirped DNP techniques, such as the frequency-swept integrated solid effect (FS-ISE) [15,26], nuclear orientation via electron spin locking (NOVEL) [27,28], and timeoptimized pulsed (TOP) DNP [29] show promise to perform well both at high magnetic field and room temperature. For instance, ISE yields DNP enhancements of ~150 at room temperature and is predicted to be unaffected by the strength of the external magnetic field [15]. However, these experiments have been performed without MAS and at magnetic fields <3 T [15,27,29], primarily due to the difficulty of implementing MAS with the microwave resonators required to generate considerable electron nutation frequencies. Frequency-swept DNP at higher magnetic fields has also been shown to improve DNP performance [30,31], but has only recently been implemented with MAS [32,33]. MAS improves the sensitivity and resolution of solid-state NMR [34][35][36][37][38] by partially averaging anisotropic interactions of the magnetic resonance Hamiltonian, and is a crucial aspect of applying DNP to systems of interest.</p><p>Here we characterize the behavior of frequency-chirped DNP experiments performed with MAS, expanding on our recent work [32]. We optimize frequency chirps from a custom-built frequency-agile high-power gyrotron [39] to produce large gains in intensity beyond those obtained with CW DNP. In addition to measuring its performance on a model system, we conduct optimized chirped experiments on intact human Jurkat cells to demonstrate frequency-chirped DNP in a biologically complex environment.</p><!><p>MAS DNP NMR experiments were performed using a custom-built DNP spectrometer at a magnetic field of 7.1584 T [41]. 13 C and 1 H Larmor frequencies were 75.4937 MHz and 300.1790 MHz, respectively. A CPMAS, rotor synchronized, Hahn echo sequence with TPPM decoupling [42] was used for all experiments (Fig. 1a). The initial magnetization of 1 H and 13 C spins was destroyed using a saturation train. 1 H and 13 C pulses were performed with nutation frequencies of 77 kHz and 100 kHz, respectively. The Hartmann-Hahn matching condition (γB1) for 1 H and 13 C was 30 kHz. Frequency chirps were applied over the DNP polarization period (τpol), and CW microwaves were employed over the rest of the experiment. The spinning frequency was 4.5 kHz for all experiments, and the sample temperature was 90 K. Typical polarization times (τpol) for optimized spectra were 5-times the T1 of the sample in the absence of microwaves, in order to remove contamination of the data by differences in the nuclear T1 and the T1DNP.</p><p>Microwaves were generated using a frequency-agile gyrotron, whose output frequency was adjusted by varying the electron acceleration potential at the electron gun anode. An arbitrary waveform generator (AWG) integrated into the NMR spectrometer (Redstone, Tecmag Inc. Houston, TX) was used to generate a waveform, which ramped the output frequency of the gyrotron in a linear fashion through 197.670 GHz, the frequency of maximum DNP enhancement of the TEMTriPol-1 radical [39]. The frequency chirps were a triangular waveform, which was repeated over the entire polarization period. For frequency chirp optimization the incident microwave power, the center DNP microwave frequency, and the sweep width and sweep time of the individual chirps were varied. The center frequency of the sweeps was varied by changing the voltage at the gyrotron anode with the AWG amplified by a high-voltage amplifier (TREK, Inc.</p><p>Lockport, NY). The sweep width corresponded to the frequency range of one sweep/chirp (either up or down) in MHz, and sweep time was the time to complete a sweep/chirp. Microwave power was attenuated from full power by inserting copper foil with slits cut in it into a gap in the waveguide to partially pass the microwave beam. The optimal power of 7 W incident on the sample was used for most experiments, which provided an estimated electron Rabi frequency of 0.43 MHz [43].</p><!><p>The 13 C carbonyl resonance was fit using DMfit [44] to determine resulting enhancement increases.</p><p>For all optimization spectra, the magnitude of the Hahn echo was used to calculate the percent increase in intensity. All experiments were repeated four times to acquire adequate error values for the measurements.</p><!><p>Experiments were performed on 4 M [U-13 C, 15 N] urea mixed with 5 mM TEMTriPol-1 or 5 mM AMUPol in a cryoprotecting matrix consisting of 60% d8 glycerol, 30% D2O, and 10% H2O by volume. Intact Jurkat cells (ATCC, Manassas, VA) were cultured in [U- 13 C, 98%; U-15 N, 98%] BioExpress-6000 mammalian cell growth medium (Cambridge Isotope Laboratories, Tewksbury, MA) at a concentration of 3 × 10 6 cells/mL in a six-well plate at 37°C and 5% CO2 for 48 hr. 3.6 × 10 7 cells were collected, spun at 170 g for 5 min, washed with 1× phosphate-buffered saline (PBS), and spun again at 170 g for 5 min to remove extracellular NMR labels (g = 9.8 m/s 2 ). 40 µL of 20 mM TEMTriPol-1 in 1×PBS with 10% DMSO was added to a cell pellet containing 36 million Jurkat cells. This suspension was centrifuged directly into the 3.2 mm zirconia rotor at 800 g for 30 s and immediately frozen in liquid nitrogen as detailed in our previous work [4].</p><!><p>Frequency-chirped DNP refers to a change in the microwave frequency or intensity throughout the course of an experiment. The frequency-chirped DNP pulse sequence is shown in Fig. 1A.</p><p>Frequency chirps (represented by the rainbow gradient) are applied over the DNP polarization period and the resulting NMR signal is detected through a cross polarization (CP) Hahn echo sequence. We emphasize that microwave frequency chirps result in better manipulation of the electron spin polarization, yet the active DNP mechanism is still the cross effect. Selection of appropriate radicals for frequency-chirped DNP is crucial due to drastic differences in electron spin g-anisotropy and relaxation properties. In our previous demonstrations of electron decoupling using chirped microwave pulses with MAS, we employed trityl rather than nitroxide radicals [3,22]. Those successes led us to explore the use of trityl-nitroxide biradicals, with the rational that the narrow trityl resonance would be easier to manipulate and the tethered nitroxide would provide greater DNP enhancements through the cross effect mechanism. TEMTriPol-1 is such a biradical, consisting of a Finland trityl radical covalently linked to a 4-amino TEMPO radical, which is used for cross effect DNP [13,45]. TEMTriPol-1 improves cross effect efficiency at high magnetic fields. Where other biradicals, such as AMUPol, depolarize nuclear spins at 100 K in the absence of microwave irradiation, TEMTriPol-1 preserves nuclear polarization [5,46].</p><!><p>CW DNP CPMAS experiments were performed at various microwave frequencies to record a 1 H DNP enhancement profile with TEMTriPol-1 [40]. The enhancement profile shows the trityl resonance frequency as the optimal frequency for CW DNP enhancement. This will be the target for the center of the frequency chirps. In a 7.1584 T magnetic field, the microwave frequency for maximum CW DNP enhancement was 197.670 GHz (Fig. 1B).</p><p>Experiments were performed to determine the effect of frequency-chirped microwave pulses during the polarization period of MAS DNP (Fig. 2). For comparison, cross effect DNP experiments were performed with CW microwave irradiation. CW DNP experiments on a model system of urea with TEMTriPol-1 resulted in an enhancement of 118 (Fig. 2, red). Enhancements herein are defined as the NMR signal intensity recorded with DNP compared to that without DNP [46]. For frequency-chirped DNP experiments, the microwave frequency was linearly chirped with a triangular waveform over 197.670 GHz, with a 28 µs sweep time and a 120 MHz sweep width. These optimized chirps yielded a 21% increase over CW DNP and an enhancement of 142 (Fig. 2, blue). Polarization times of 53 s (5×T1DNP, Fig. S1) were used to ensure that >99% of the polarization had built up for both experiments, allowing for direct comparison of the CW and frequency-chirped experiments. To determine the necessity of a narrow-line radical, such as trityl, for frequency-chirped DNP, experiments were performed on a sample containing the nitroxide-nitroxide biradical, AMUPol.</p><p>The frequency chirps were centered at 197.674 GHz (maximum with 1 H-enhancement for AMUPol) the previously optimized sweep time of 28 µs and sweep width of 120 MHz were used.</p><p>Frequency chirps over the polarization period resulted in a decrease in signal intensity of 3% compared to CW DNP (Fig. 3). These frequency chirps do not yield the same improved electron spin control over the nitroxide biradical, AMUPol, as they do over TEMTriPol-1. This implies that a narrow-line radical is required for implementation of frequency-chirped DNP.</p><!><p>The performance of frequency-chirped DNP was then examined within intact human Jurkat cells (Fig. 4). Frequency chirps improved the NMR signal by 24%, yielding an enhancement of 6 (versus 4.8 for CW DNP). These results display the application of frequency-chirped DNP to more complex samples of biological interest.</p><!><p>To determine the dependence of CW and frequency-chirped enhancement on microwave power, CPMAS experiments were performed with varying microwave attenuation on the TEMTriPol-1/urea sample (Fig. 5). For frequency-chirped DNP the optimized triangle waveform (28 µs sweep time and 120 MHz sweep width) was repeated over a polarization time of 20 s. 35 W of microwave power incident on the sample (Rabi frequency of 0.95 MHz) produced a 123% increase in signal with frequency-chirped DNP compared to CW, yielding enhancements of 17 and 8, respectively (Fig. 5a, b). We note that such high microwave powers place the cross effect in the oversaturated regime, leading to less overall enhancement. 7 W of microwave power resulted in the highest sensitivity and an improvement of 25% with frequency-chirped DNP compared to CW. Higher microwave power yielded greater improvements with frequency-chirped DNP over CW DNP, but the overall signal intensity obtained was suboptimal due to saturation of the cross effect [47].</p><!><p>The effects of sweep time, sweep width, and center frequency on the improvement with frequencychirped DNP over CW irradiation are shown in Fig. 6. For this dependence the polarization time was 20 s; the sweep width was held constant at 80 MHz, the incident microwave power at 7 W, and the center frequency at 197.670 GHz. Shorter sweep times increased the sensitivity to a greater extent than longer sweep times, with the greatest improvement over CW (15%) occurring with a 20 μs sweep time (Fig. 6a). Sweep times below 20 μs were not achievable with the current microwave frequency agility circuit, as the frequency output waveform became distorted. A sweep time of 150 µs resulted in only a 1% improvement in signal intensity over CW. We suspect that at longer sweep times electron spin saturation is lost through relaxation mechanisms.</p><p>The dependence of frequency-chirped DNP sensitivity on the sweep width of the frequency chirps is shown in Fig. 6b. For this dependence the polarization time was 20 s; the sweep time was held constant at 28 μs, the incident microwave power at 7 W, and the center frequency at 197.670 GHz.</p><p>The improvement from the frequency chirps increased as the sweep width increased. A 120 MHz sweep width resulted in an improvement of 21%, while the signal intensity decreased by 1% with a sweep width of 10 MHz. Due to instrument limitations, sweep widths greater than 120 MHz could not be attained. This width is roughly that of the base of the trityl lineshape in the enhancement profile (Fig. 1b). We previously reported a similar optimal sweep width in electron decoupling experiments involving the Finland trityl radical [3]. Larger sweep widths provide microwave irradiation that is resonant with a greater number of trityl electron spins, enabling better electron spin control and improving the efficiency of frequency-chirped DNP.</p><p>During characterization it is important to consider multiple points on the enhancement profile. Fig. 6d provides a clear picture of the effect of frequency chirping, whereas Fig. 6c shows the potential for misinformation. The choice of irradiation frequency can lead to suspiciously high improvements due to difference in positive and negative enhancement regions between CW and frequency-chirped DNP. The CW enhancement profile shows maximum positive and negative enhancements at 197.670 GHz and 197.850 GHz, respectively (Fig. 6d). Frequency chirping at microwave frequencies lower than 197.750 GHz (positive enhancement), yielded greater enhancements than CW (Fig. 6d). However, at frequencies greater than 197.750 GHz (negative enhancement), the frequency-chirped DNP provided lower signal intensity than CW DNP. Note that at this point we have simply demonstrated the methodology of performing frequencychirped DNP experiments with TEMTriPol-1. To compare the sensitivity of the experiments with TEMTriPol-1 and AMUPol, we can divide the signal-to-noise from each experiment by the square root of the polarization time for the respective experiments. In doing so, we obtain a sensitivity of 79 with AMUPol (Fig. 3) and 73 with TEMTriPol-1 (Fig. 2). Thus, while the sensitivity of the experiments performed on each radical are similar at this stage, advances in instrumentation that enable greater sweep times and sweep widths will make frequency-chirped DNP experiments with TEMTriPol-1 more sensitive than AMUPol, and thus more feasible for sensitivity-demanding, multidimensional experiments.</p><!><p>To date, frequency-chirped DNP experiments, such as FS-ISE, NOVEL, and TOP DNP, have been largely restricted to static samples due to the difficulties of housing microwave resonators with the instrumentation required for magic angle spinning (MAS). Here, we have characterized the optimal experimental conditions for frequency-chirped MAS DNP. At a magnetic field of 7 T and with 7 W of microwave power, frequency-chirped microwaves over the polarization period improved DNP enhancements by 21%. Greater microwave powers resulted in up to 123% improvements with frequency-chirped DNP, but saturation of the cross effect resulted in less overall signal intensity.</p><p>These optimized frequency-chirped experiments were applied to a more biologically complex sample: intact Jurkat cells. This resulted in an improvement in signal intensity of 24% over CW DNP. Characterization of the parameters of frequency-chirped DNP revealed areas for further improvements to elicit even greater sensitivity. More powerful gyrotrons with larger frequency bandwidths, and gating mechanisms for chirps can be developed to increase sweep widths and shorten the sweep times, thus improving electron spin control. To take full advantage of frequencychirped DNP at high power and high electron Rabi frequencies, duty cycling of the microwaves can be implemented to reduce dielectric heating [29]. We expect optimization of the waveform, with respect to both intensity and phase, to result in improved frequency-chirped DNP MAS performance. Future studies could analyze the effect of the spinning frequency on the enhancement achieved by frequency chirped DNP over CW DNP. Both the solid effect and cross effect are driven by interactions between the spin system, the microwave field, and the spinning rotor.</p><p>Understanding these effects will prove crucial in the future development of DNP, as MAS frequencies and magnetic fields are pushed to ever higher values.</p><p>New radicals composed of tethered broad and narrow line radicals are currently being investigated with useful electronic properties such as long longitudinal relaxation times. Longer relaxation times will afford even more electron spin control with frequency-chirped DNP. Although the precise mechanism governing the improvement in sensitivity will require further investigation, it is possible that it is governed by an adiabatic process. As such, future experiments could focus on maintaining a constant sweep rate by simultaneously varying the sweep time and sweep width in an inverse manner. This could prove important, as adiabatic processes often show a remarkable resilience to microwave inhomogeneities and frequency offsets arising from difference in molecular orientation and conformations in a solid sample. These techniques can be paired with other advances in instrumentation such as higher power microwave sources and microwave lenses for improved microwave intensity and high frequency MAS for 1 H detected spectra in future experiments. These could allow for the implementation of pulsed DNP mechanisms such as electron-nuclear cross polarization at high magnetic fields in the foreseeable future.</p>&quot;},&quot;article_source&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ChemRxiv&quot;}}},{&quot;rowIdx&quot;:2,&quot;cells&quot;:{&quot;title&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Engineering forward genetics into cultured cancer cells for chemical target identification&quot;},&quot;abstract&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Summary Target identification for biologically active small molecules remains a major barrier for drug discovery. Cancer cells exhibiting defective DNA mismatch repair (dMMR) have been used as a forward genetics system to uncover compound targets. However, this approach has been limited by the dearth of cancer cell lines that harbor naturally arising dMMR. Here, we establish a platform for forward genetic screening using CRISPR-Cas9 to engineer dMMR into mammalian cells. We demonstrate the utility of this approach to identify mechanisms of drug action in mouse and human cancer cell lines using in vitro selections against three cellular toxins. In each screen, compound-resistant alleles emerged in drug-resistant clones, supporting the notion that engineered dMMR enables forward genetic screening in mammalian cells.&quot;},&quot;uuid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;engineering_forward_genetics_into_cultured_cancer_cells_for_chemical_target_identification&quot;},&quot;source_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:3370,&quot;string&quot;:&quot;3,370&quot;},&quot;target_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:120,&quot;string&quot;:&quot;120&quot;},&quot;compression&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:28.083333333333332,&quot;string&quot;:&quot;28.083333&quot;},&quot;headers&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Introduction<!>MSH2 deletion in cancer cell lines induces MSI and hypermutation<!>Engineered dMMR enables forward genetic screening in mammalian cancer cells<!>Discussion<!>LEAD CONTACT AND MATERIALS AVAILABILITY<!>EXPERIMENTAL MODEL AND SUBJECT DETAILS<!>sgRNAs, surveyor, CRISPR KO of MSH2, MLH1.<!>Western-blotting<!>Compounds<!>Dose response curves<!>Selection of resistant clones.<!>Crystal Violet experiment<!>Whole Exome Sequencing Analysis<!>Identification of recurrently mutated genes in MLN4924-resistant clones<!>QUANTIFICATION ANS STATISTICAL ANALYSIS<!>DATA AND CODE AVAILABILITY<!>&quot;},&quot;sections&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;<p>The identification of therapeutic targets in cancer can be divided into two complementary approaches. Target-based approaches use tumor sequencing or laboratory-based genetic studies to identify cancer driver genes followed by screening for small molecules that impair the protein products of cancer driver genes. Phenotypic high-throughput small molecule screens (HTS) first identify drug-like chemicals that selectively impair the growth of cancer cells. The latter approach has been limited by the technical challenge of identifying the direct protein targets of small molecules exhibiting anti-cancer effects. One strategy to identify chemical targets is through the identification of compound resistant alleles that impair compound-target interaction. Cancer cells with dMMR exhibit mutation rates increased as much as 50-750-fold compared to cells with intact MMR (Glaab and Tindall, 1997). As a result, these cells are predisposed to develop resistance through the acquisition of compound resistant alleles. Following selection, these compound resistant alleles can be identified by transcriptome or whole exome sequencing of multiple independent drug-resistant clones (Han et al., 2017, Han et al., 2016, Wacker et al., 2012). However, it is not clear whether this approach can be applied to other cancer cell lines, particularly those established from cancers without MMR deficiency or those harboring low mutation frequencies, such as pediatric malignancies. To date, a single cancer cell line, HCT116, derived from a human colorectal cancer harboring a naturally-arising mutation in the MMR protein MLH1, has been used successfully for these studies. Here, we sought to determine if somatic deletion of the MMR protein MSH2 using CRISPR-Cas9 could be used to expand the repertoire of dMMR cancer cell lines for use in forward genetic screens.</p><!><p>We sought to develop cancer cell lines with engineered loss of MSH2 in order to establish forward genetics screening in other cancer types, including Ewing sarcoma (EWS) and small cell lung cancer (SCLC). Ewing sarcoma is a pediatric malignancy without approved targeted therapies, and the overall mutational burden in these tumors is extremely low, similar to other pediatric malignancies (Pishas and Lessnick, 2016, Yu et al., 2017, Tirode et al., 2014, Brohl et al., 2014, Crompton et al., 2014). We used CRISPR-Cas9 to generate MSH2-null Ewing sarcoma A673 cells (Figure 1A, Methods). We examined multiple microsatellite regions by PCR and capillary electrophoresis for evidence of microsatellite instability (MSI), a hallmark of dMMR cells, in two independently generated MSH2-null A673 lines (A673-M1, A673-M8). MSI was observed in three out of five loci analyzed in both A673-M1 and A673-M8 clones compared to the parental, MMR-proficient, A673 cell line (Fig. 1B,C)(Boland et al., 1998).</p><p>We next tested whether MMR deletion would facilitate MSI in tumor cell lines derived from genetically-engineered mouse cancer models (GEMMs), which also exhibit very low mutation frequencies compared to many human malignancies (McFadden et al., 2014, McFadden et al., 2016). Msh2 was deleted in a cell line generated from a small cell lung cancer (SCLC) GEMM initiated by loss of the p53, retinoblastoma (Rb), and p130 tumor suppressors (319-N1 cell line) (Figure S1A)(Schaffer et al., 2010). Msh2-null murine SCLC (mSCLC) cells also exhibited evidence of MSI, with two of three microsatellite loci exhibiting instability (Figure S1B, C)(Bacher et al., 2005).</p><p>Following observation of MSI in MMR-edited human and murine cells, we sought direct evidence of an increased mutation frequency using whole exome sequencing (WES). Two MSH2-null A673 clones (M1 and M8) and three independent MSH2-wild-type parental A673 clones (CL1, CL2, and CL3) were subjected to WES. Because a reference germline genome was not available for A673 cells, we used the parental A673 cell line as the germline reference (Methods). We observed an increased frequency of somatic mutations in M1 (n=221) and M8 (n=198) clones, compared to MSH2-wild-type parental clones (n= 77, 74, 64) (Table S1).</p><p>We performed WES on multiple mSCLC tumor cell lines generated from independent tumors isolated from the same mouse in order to accurately compare mutation frequencies between MMR-proficient and dMMR cell lines (Figure S2A). Common variants between two independent primary tumor cell lines (319-T1, T2) represented germline variants, whereas variants unique to individual cell lines represented somatic mutations. The Msh2-wild type cell line 319-N1 exhibited 25 high-confidence somatic mutations (Table S2). In contrast, 319-N1-Cl31 cells with engineered Msh2 loss exhibited 352 somatic mutations. Therefore, elevated somatic mutation frequencies were observed following MMR editing in both human and murine cancer cell lines. (Figure 1D, Figure S2B,C).</p><!><p>After establishing the dMMR phenotype of CRISPR-edited MSH2-null cell lines, we determined whether CRISPR-mediated dMMR facilitated the emergence of compound-resistant clones. We performed drug selections using MSH2-wild type A673 cells (parental A673), MSH2-null A673 cells (A673-M1 and A673-M8), and HCT116 cells. Selections were performed against three cellular toxins: bortezomib (an inhibitor of the subunit β5 of the proteasome, PSMB5), MLN4924 (a NEDD8-activating enzyme (NAE) inhibitor), and CD437 (a DNA polymerase alpha (POLA1) inhibitor)(Soucy et al., 2009, Han et al., 2016, Lu and Wang, 2013, Chen et al., 2011). Following compound selection at lethal doses, resistant colonies growing in the presence of compound were visualized by crystal violet staining. Consistent with the notion that CRISPR-mediated MSH2 loss enabled acquisition and emergence of compound resistant clones, colonies were observed following selection only on the MSH2-null A673 and HCT116 plates (Figure 1E). No colonies were observed on the parental MSH2-wild type A673 plates.</p><p>We next performed additional selections using bortezomib, CD437, and MLN4924. We selected MSH2-null A673 cells and Msh2-null mSCLC cells using bortezomib at three concentrations close to the lethal dose, as determined by one week of compound exposure (EC1001wk) for bortezomib (see methods). Following 2 weeks of selection, compound-resistant colonies emerged and were expanded from both A673-M1 (4 clones) and Msh2-null mSCLC cells (11 clones).</p><p>Bortezomib-resistant alleles have been reported within exon 2 of PSMB5, which encodes a binding pocket for the drug (Lu and Wang, 2013, Chen et al., 2011). We therefore amplified and sequenced exon 2 in bortezomib-resistant MSH2-null A673 and Msh2-null mSCLC clones. These regions were also amplified and sequenced from the parental A673 and mSCLC cell lines to ensure these mutations did not exist prior to MMR impairment. We identified PSMB5 mutations in MMR-deficient A673 (4 out of 4 clones harbored mutations) and mSCLC cell lines (11 out of 11 clones harbored PSMB5 exon 2 mutations), including mutations previously reported to mediate bortezomib resistance (Fig. 2B, D, E) (Lu and Wang, 2013, Wacker et al., 2012). All mutations identified in PSMB5 mapped to the bortezomib binding pocket (Huber et al., 2016) (Fig. 2F). We confirmed in vitro resistance to bortezomib in all 4 clones harboring putative compound-resistant alleles from A673-M1 by cell viability assay (CellTiter Glo, Promega) following 72 hours of drug exposure (Figure 2A, B). We also tested 7 out of the 11 mSCLC clones that harbored Psmb5 mutations and confirmed in vitro resistance to bortezomib (Figure 2C, D). We confirmed that all clones harboring PSMB5 mutations exhibited bortezomib resistance (2.36 to 13.84-fold increase in EC50), whereas resistance to etoposide was not observed (Figure 2A–D; Figure S3A, B).</p><p>To further confirm that CRISPR-dMMR cells exhibited the capacity to acquire compound resistant alleles to different classes of cellular toxins, we performed additional selections using MSH2-null A673 cells and Msh2-null mSCLC cells against the DNA polymerase-alpha inhibitor CD437. After 2 weeks of selection, compound-resistant colonies emerged and were expanded from both A673-M1 (4 clones) and Msh2-null mSCLC cells (17 clones). For CD437-selected clones, cDNA flanking exons 19 to 25 was amplified and sequenced after clonal expansion. We identified POLA1 mutations in the MMR-deficient A673 (4 out of 4 clones harbored mutations) and mSCLC lines (11 out of 17 clones harbored mutations) (Figure 3B, D, E) (Han et al., 2016). None of these mutations were detected in the parental, MMR proficient, cell lines. Mutations impacted five amino acids that clustered within the POLA1 structure (Coloma et al., 2016) (Figure 3F).</p><p>We confirmed CD437-resistance in all 4 clones from A673-M1 cell line and 11 out of 17 mSCLC clones harboring putative compound-resistant alleles to CD437 (Figure 3A, B). The six mSCLC resistant clones that did not harbor POLA1 mutations exhibited resistance to both CD437 and etoposide, which suggested a generalized mechanism of acquired resistance drove expansion of these clones during compound selection (Figure S3C, D). The ten mSCLC clones harboring POLA1 mutations all exhibited CD437 resistance (5.31 to 15.05-fold increase in EC50), whereas no difference in sensitivity to etoposide was observed (Figure 3C, D; Figure S3E, F).</p><p>The selections performed with bortezomib and CD437 demonstrate that engineered MSH2 loss in enables the emergence of compound resistant alleles during drug selections in A673 and mSCLC cells. We finally tested whether prospective identification of compound targets could be accomplished using exome sequencing of compound resistant clones. Therefore, we performed selections using MLN4924 in the MSH2-null clone, A673-M1, at three concentrations for MLN4924 (see Methods). Following two weeks of selection, six colonies emerged and were expanded. To confirm in vitro resistance, we determined the EC50 for both parental A673-M1 cells and six resistant clones (Figure 4A,B). To identify potential clones exhibiting general resistance, including increased expression of drug efflux pumps, we assessed sensitivity to the topoisomerase II-inhibitor, etoposide (Figure S3F). We validated that all MLN clones exhibited resistance to MLN4924 (21.75 to 135.03-fold increase in EC50), while no difference in etoposide resistance was observed (Figure 4A,B; Figure S3F).</p><p>We next determined if the target of MLN4924, NAE subunit encoded by UBA3, could be identified by WES of MLN4924-resistant clones. Indeed, UBA3 was identified as the single gene mutated in 6/6 MSH2-null clones (Figure 4C). This gene encodes the NAE subunit targeted by MLN4924, and three of the mutations observed, A171T, E204K, and Y352H, were previously reported as MLN4924-resistant alleles (Figure 4D) (Michael et al., 2012, Xu et al., 2014). We compared somatic mutations in MLN4924-resistant A673 clones to establish whether the MLN4924-resistant clones arose independently (see Methods). A673-MLN-D and A673-MLN-H shared a majority of mutations (783/965), establishing that these two clones arose from the same founder cell. However, no other clones shared more than 6 somatic mutations, suggesting that the other clones arose independently, including A673-MLN-A and A673-MLN-C that shared the A171T mutation in UBA3.</p><!><p>We establish that induced MMR deficiency using CRISPR-Cas9 methods is sufficient to induce MSI, hypermutation, and facilitate the emergence of compound resistant alleles in established human and murine cancer cell lines derived from diverse cancer lineages. This approach offers the potential to significantly expand the use of forward genetics to identify the mechanisms of action of compounds with anticancer activity. In particular, we demonstrate that this strategy can be employed in cancers with low mutation rates such as pediatric malignancies. Therefore, cell lines with engineered MMR deficiency represent an experimental tool to facilitate the identification of mechanisms of action of selective cancer toxins identified by HTS campaigns, and to model genetic mechanisms of acquired resistance to anti-cancer therapies in current use. However, we recognize limitations of the current study. First, following identification of candidate compound-resistant alleles (i.e., recurrent mutations in compound-resistant clones), additional biochemical studies are necessary to establish the direct molecular target. Second, the forward genetics approach requires that the small molecule target a protein essential for viability of the cancer cell. In addition, we cannot from our current data establish a frequency of mutation necessary to facilitate the emergence of compound-resistant clones. Hypermutation due to endogenous defects in DNA repair might also be more broadly applied to other phenotypic genetic screens, including in vivo screens in GEMMs. Therefore, cancer cell lines with induced MMR deficiency and hypermutation represent a tool with wide potential application in cancer genetics and drug discovery.</p><!><p>Further information and requests for resources and reagent should directed to and will be fulfilled by the Lead Contact, David McFadden (david.mcfadden@utsouthwestern.edu)</p><!><p>Ewing sarcoma A673 cell line were cultured at 37°C and 5% CO2 in RPMI (R8758, Sigma-Aldrich) supplemented with 10% FBS (#35-150-CV, Corning), 2 mM L-glutamine (G7513, Sigma-Aldrich), and penicillin streptomycin (P0781, Sigma-Aldrich). Cells were expanded using Trypsin (T4049, Sigma-Aldrich) every 3-4 days. A673 cell lines are derived from a female subject and were authenticated by STR profiling.</p><p>The Trp53fl/fl; Rb1fl/fl; Rbl2fl/fl; RosaLSL-Tomato/+ mouse model of small cell lung cancer mice has been previously described (Schaffer et al., 2010). 319-T1 and 319-T2 were established from primary tumors in the lung, and 319-N1 was established from a lymph node metastasis, all developed in a PRP female mouse. 319-N1 as well as clones derived from those cell lines were cultured with DMEM (D6429, Sigma-Aldrich) supplemented with 5% FBS (#35-150-CV, Corning), 2 mM L-glutamine (G7513, Sigma-Aldrich), and penicillin streptomycin (P0781, Sigma-Aldrich). Cells were expanded using Trypsin (T4049, Sigma-Aldrich) diluted in PBS (2:1 ratio) every 3-4 days.</p><p>All animal experiments were approved by the UTSW IACUC 2018-102383 (D.G.M., P.I.).</p><!><p>Single-guide RNA (sgRNA) targeting human MSH2 and murine Msh2 were designed by \&quot;sgRNA Designer: CRISPR KO\&quot; (https://portals.broadinstitute.org/gpp/public/analysis-tools/sgrna-design) from the Broad Institute(Doench et al., 2016). sgRNAs were cloned into LentiCRISPR V2 plasmid (Addgene plasmid #52961) as previously described(Ran et al., 2013), and validated by T7 endonuclease assay (T7 endonuclease I from NEB, Cat. #M0302). sgRNA sequences used were as follows: human MSH2 5'-TGAGAGGCTGCTTAATCCAC-3'; murine Msh2 5'-GGTTAATACCCT GATACAGT-3'. For the generation of lentiviral vectors, 293T/17 cell were transfected with LentiCRISPR V2, psPAX2 (Addgene plasmid #12260), and pMD2.G (Addgene plasmid #12259) in a ratio (4:3:1) using TransIT®-LT1Transfection reagent (MIR 2304, Mirus Bio) as described by manufacturer. Mouse SCLC and Ewing sarcoma cell lines were plated at 106 cells in a 10 cm dish and cultured overnight. The next day, cells were transduced with lentiviruses (MOI<0.5 determined by visual assessment) using 8ug/mL of polybrene transfection agent (TR-1003-G, EMD Millipore). Cells were selected with 2 mg/ml of Puromycin (P8833, Sigma-Aldrich) for 72 hours. Then surviving clones were picked and expanded for validation by western-blotting.</p><!><p>Western-blotting for Ewing sarcoma and mouse SCLC protein samples was performed using standard methods. Odyssey Nitrocellulose membrane (LICOR, #926-31092) were used for protein transference and then blocked using Odyssey® Blocking Buffer in PBS (LICOR, #927-40000) for 1h at RT. Primary antibodies were incubated for 1h at RT diluted 1:1,000 in Odyssey® Blocking Buffer : PBS-Tween (0.1%). Antibodies used were anti-Msh2 [D24B5] XP rabbit mAb (#2017, Cell Signaling Technology), and anti-β-actin (8H10D10) mouse mAb (#3700, Cell Signaling Technologies). Membrane was washed with PBS-Tween (0.1%) three times for 5 minutes each wash. Secondary antibodies were incubated for 30 min at RT using IRDye 800CW donkey anti-rabbit IgG (H+L) (#926-32213, LI-COR), and IRDye 680RD donkey anti-mouse IgG (H+L) (#926-68072, LI-COR), at dilution 1:10,000. Visualization was performed with Odyssey CLx Imaging System (LI-COR).</p><!><p>Bortezomib was purchased from Selleck Chemicals (#S1013). Etoposide was purchased from Sigma-Aldrich (#E1383-100MG). CD437 was purchased from Sigma-Aldrich (#C5865). MLN4924 was purchased from ApexBio (#B1036). Compounds were diluted using DMSO (Sigma-Aldrich, D650-100ML) and aliquoted at 10 mM and aliquots were exposed to a maximum of three freeze-thaw cycles.</p><!><p>Mouse SCLC cell lines were plated in 96-well plates, 6,600 cells per well in 200 μL of media. Ewing sarcoma cell lines were plated in 96-well plates, 10,000 cells per well in 200 μL of media. After overnight incubation, compounds were dispensed using a D300e Digtal Dispenser (TECAN). Cell viability assay was assessed after 72 hours using CellTiter-Glo luminescent cell viability assay (Promega, #G7571). The CellTiter-Glo reagent was diluted by adding PBS-Triton-X (1%) (1:1 ratio). EC1001wk determination for bortezomib was performed in a 12-well plate seeding 25,000 cells per well. After 24h, Bortezomib was dispensed using TECAN D300e setting up a minimum concentration of EC5072h and a maximum concentration of EC10072h. Media was changed every 3-4 days to refresh compounds. Cell viability was determined visually following seven days.</p><!><p>10 cm plates for each MMR deficient cell lines (1 million cells per plate) were treated with bortezomib or CD437 at EC1001wk ÷ 1.5, EC1001wk, and EC1001wk × 1.5 concentrations. Media with bortezomib or CD437 was replenished every 3 – 4 days over the course of 2 weeks. Surviving clones were expanded. Exon 2 of PSMB5 from both human and mouse cell lines was amplified and sequenced using primers specified in Table S3.</p><p>Amplification and sequence of exon 19 to 25 of POLA1 from both human and mouse cell lines was amplified and sequenced using primers specified in Table S3.</p><!><p>10 cm plates for each MMR deficient cell lines (3 million cells per plate) were treated with botezomib, MLN4924, and CD437 at EC1001wk ÷ 1.5, EC1001wk ÷ 1.25, EC1001wk, and EC1001wk × 1.25, EC1001wk × 1.5 concentrations. Media with either bortezomib, MLN4924, or CD437 was replenished every 3 – 4 days over the course of 2 weeks followed by growth in media without compound for 1 week.</p><p>Staining solution was prepared with 1% (weight/volume ratio) crystal violet from Sigma-Aldrich (#C6158-50G) in 10% ethanol.</p><!><p>Whole-exome sequencing of cell line samples was performed by BGI Genomics using SureSelect Human All Exon V5 (Ewing sarcoma samples) and SureSelect Mouse All Exon V1 (mSCLC) and BGISEQ-500. The analysis workflow was based on Genome Analysis Toolkit (GATK, v3.8-0) best practices(McKenna et al., 2010, DePristo et al., 2011). The qualities of sequencing reads were evaluated using NGS QC Toolkit (v2.3.3)(Patel and Jain, 2012) and the extracted high-quality reads were mapped to human and mouse reference genome (UCSC hg19 and Ensembl 91) using Burrows-Wheeler Aligner (BWA, v0.7.15a)(Li and Durbin, 2009). Picard (v2.12.0) (https://broadinstitute.github.io/picard.) was used to remove PCR duplicates and GATK was used to recalibrate base qualities. For murine SCLC cell lines, calling variants and joint genotyping together were performed using HaplotypeCaller and the variant calls were filtered by applying the following criteria: QD (Variant Confidence/Quality by Depth) < 2, FS (Phred-scaled p-value using Fisher's exact test to detect strand bias) > 60, MQ (RMS Mapping Quality) < 40, DP (Approximate read depth) < 10, GQ (Genotype Quality) < 20, maximum VAF (variant allele fraction) < 0.2. For each murine SCLC cell lines, the somatic mutations in 319-N1 were defined by the VAF > 0.15 and VAF < 0.05 for 319-T1 and 319-T2. Somatic mutations in 319-N1-Cl31 were defined by the VAF > 0.15 and VAF < 0.05 for the other cell lines. For human cell lines, MuTect2(Cibulskis et al., 2013) was used to identify somatic mutations in clones A673-M1 and A673-M8 comparing to the A673 parental cell line. Somatic mutations for each clones (A673-M1 and A673-M8) were defined by VAF > 0.15 and VAF < 0.05 for the other human cell lines.</p><!><p>We defined acquired somatic mutations for each A673-M1 MLN-resistant clone by VAF > 0.01 and VAF < 0.01 for the parental A673-M1 cell line. Non-coding mutations were excluded.</p><!><p>Data were analyzed using Prism 8.0 by GraphPad. Dose response curves were fitting in Figure 2A, 2C, 3A, 3C and 4A, to calculate IC50. Hill coefficients and standard error were done using Log [inhibitor] vs Normalized response. Quantitative data are presented as mean.</p><!><p>The variants were annotated using a custom Perl script (https://github.com/jiwoongbio/Annomen) with mouse transcripts, proteins, and variations (Ensembl 91 for mouse, RefSeq and dbSNP build 150 for human). The variant allele frequencies were calculated using a custom Perl script and SAMtools (v1.4) (Goncearenco et al., 2017, Li et al., 2009) (all analysis scripts are available at https://github.com/jiwoongbio/Annomen).</p><p>The sequencing datasets exposed in this study have been deposited in SRA under accession code PRJNA543281.</p><!><p>Table S1. WES_A673 Ewing_Related to Figures 1 and 4. This table shows whole exome sequencing data for the A673 cell lines described in this manuscript.</p><p>Tab 1. Raw data. Whole exome sequencing data for A673 cell line, A673 MMR-proficient and MMR-deficient cell lines, and MLN4924-resistant clones.</p><p>Tab 2. Somatic mutations in A673-Cl1 (A673 VAF < 0.05; A673-Cl1 > 0.15).</p><p>Tab 3. Somatic mutations in A673-Cl2 (A673 VAF < 0.05; A673-Cl2 > 0.15).</p><p>Tab 4. Somatic mutations in A673-Cl4 (A673 VAF < 0.05; A673-Cl4 > 0.15).</p><p>Tab 5. Somatic mutations in A673-M1 (A673 VAF < 0.05; A673-M1 > 0.15).</p><p>Tab 6. Somatic mutations in A673-M8 (A673 VAF < 0.05; A673-M8 > 0.15).</p><p>Table S2. WES_319 mSCLC_Related to Figures 1, S1 and S2. This table shows whole exome sequencing data for the murine small cell lung cancer cell lines described in this manuscript.</p><p>Tab 1. Raw data. Whole exome sequencing data for 319-T1, 319-T2, 319-N1, and MMR-deficient 319-N1-Cl31 cell line.</p><p>Tab 2. Somatic mutations in 319-N1 (319-T1 VAF < 0.05; 319-T2 VAF < 0.05; 319-N1 > 0.15).</p><p>Tab 3. Somatic mutations in 319-N1-Cl31 (319-T1 VAF < 0.05; 319-T2 VAF < 0.05; 319-N1-Cl31 > 0.15).</p><p>Table S3. Oligos PCR and sequencing_Related to STAR Methods. This table shows the oligos used for amplification and sequencing of exon 2 from PSMB5 and exons 19 to 25 from POLA1.</p>&quot;},&quot;article_source&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;PubMed Author Manuscript&quot;}}},{&quot;rowIdx&quot;:3,&quot;cells&quot;:{&quot;title&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Efficient oxidation of oleanolic acid derivatives using magnesium bis(monoperoxyphthalate) hexahydrate (MMPP): A convenient 2-step procedure towards 12-oxo-28-carboxylic acid derivatives&quot;},&quot;abstract&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;A new, straightforward and high yielding procedure to convert oleanolic acid derivatives into the corresponding δ-hydroxy-γ-lactones, by using the convenient oxidizing agent magnesium bis(monoperoxyphthalate) hexahydrate (MMPP) in refluxing acetonitrile, is reported. In addition, a two-step procedure for the preparation of oleanolic 12-oxo-28-carboxylic acid derivatives directly from Δ12-oleananes, without the need for an intermediary work-up, and keeping the same reaction solvent in both steps, is described as applied to the synthesis of 3,12-dioxoolean-28-oic acid.&quot;},&quot;uuid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;efficient_oxidation_of_oleanolic_acid_derivatives_using_magnesium_bis(monoperoxyphthalate)_hexahydra&quot;},&quot;source_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:1408,&quot;string&quot;:&quot;1,408&quot;},&quot;target_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:73,&quot;string&quot;:&quot;73&quot;},&quot;compression&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:19.28767123287671,&quot;string&quot;:&quot;19.287671&quot;},&quot;headers&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Findings<!><!>Findings<!><!>Findings<!><!>Findings<!><!>Findings<!>Supporting Information<!>&quot;},&quot;sections&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;<p>The molecular diversity that arises from research into natural products represents a valuable tool for driving drug discovery and development [1–2]. In this context, pentacyclic triterpenoids are currently regarded as important scaffolds for new drug development [3]. The chemistry of oleanane-type triterpenoids has been investigated with particular interest and many relevant biological and pharmacological activities of these derivatives have been reported in the literature, among which are antitumor, antiviral, anti-inflammatory, hepatoprotective, gastroprotective, antimicrobial, antidiabetic, and hemolytic properties, as well as many others [3–5]. Functionalized γ-lactones are important building blocks of bioactive natural products [6–7]. The δ-hydroxy-γ-lactone motif is part of such bioactive natural products as (±)-muricatacin [8–9] or the three hydroxylactones found in the mushroom Mycoleptodonoides aitchisonii [10]. Terpenoid δ-hydroxy-γ-spirolactones have been found to act as significant feeding deterrents to the lesser mealworm Alphitobius diaperinus [11]. In particular, oleanane-type triterpenoids bearing a γ-lactone function, either isolated from natural sources or obtained by semisynthesis, have shown interesting biological activities [12–14]. From the synthetic point of view, the oxidative 28,13β-lactonization allows the preparation of 12α-hydroxyoleananes with a protected 28-carboxyl acid function. In fact, 12α-hydroxy-3-oxooleanan-28,13β-olide (2) is a key intermediate in the synthesis of S-0139, an endothelin A receptor antagonist [15]. Moreover, as part of our ongoing work on pentacyclic triterpenoid chemistry [16–17], we recently demonstrated that oleanolic δ-hydroxy-γ-lactones can be efficiently converted into the corresponding 12-oxo-28-carboxylic acid derivatives by bismuth(III) triflate catalysis [18]. This new approach not only avoids an inconvenient multistep synthesis by means of a protection/deprotection strategy [19–20] but also results in chemical modification of ring C, a strategy known to increase the anti-inflammatory and cytotoxic activities of oleanolic acid (OA) derivatives [19,21–22].</p><p>Oleanolic δ-hydroxy-γ-lactones can be obtained from Δ12-oleananes by oxidative 28,13β-lactonization. This reaction was performed under photochemical irradiation [23–24], but weak selectivity and low isolated yields were observed. Alternatively, oxidation reagents such as H2O2 in acetic acid [25–26], the inorganic salt mixture KMnO4/CuSO4 [27], ozone [15,28–29] and m-chloroperoxybenzoic acid (mCPBA) [30–31] have also been reported. Magnesium bis(monoperoxyphthalate) hexahydrate (MMPP) is commercially available, inexpensive and relatively stable [32–34] and has been used in the oxidation of various functional groups [35–42]. This oxidant is non-shock-sensitive and non-deflagrating [43]. Moreover, its use greatly simplifies the isolation of the reaction products, because it may simply be filtered off from the reaction crude, which is then worked up as usual.</p><p>In this letter, we report the use of MMPP for the efficient and high-yielding oxidation of OA derivatives to afford the corresponding δ-hydroxy-γ-lactones. Moreover, we have set up a protocol that allows the convenient sequential two-step preparation of 3,12-dioxoolean-28-oic acid directly from 3-oxooleanolic acid, without the need of an intermediary work-up, and keeping the same reaction solvent in both steps.</p><p>We found that the reaction of 3-oxooleanolic acid 1 with 2.0 equiv of MMPP, in refluxing acetonitrile, afforded the corresponding δ-hydroxy-γ-lactone 2 in 85% yield after 5 hours (Table 1, entry 1). These new reaction conditions were successfully extended to OA 3 and other 3β-substituted OA derivatives 5, 7 and 9 (Table 1, entries 2–5).</p><!><p>Reaction of OA derivatives with MMPP to afford oleanolic δ-hydroxy-γ-lactones directly.a</p><p>aReactions were performed in acetonitrile, under reflux; bAnalytical data for compounds 2 [29], 4 [28], 6 [27], 8 [18] and 10 [18] are in accordance with the literature; cIsolated yield.</p><!><p>The substrates were dissolved in acetonitrile under reflux, and MMPP (2.0–3.0 equiv) was added to the solution under strong magnetic stirring. After completion of the reaction, the magnesium salts were easily filtered off after evaporation of the acetonitrile and suspension of the resulting white solid in ethyl acetate. We found that 2.0 equiv of MMPP were sufficient to effectively convert 3-oxooleanolic acid 1, OA 3, and 3β-acetoxyoleanolic acid 5 into the corresponding δ-hydroxy-γ-lactones 2, 4 and 6, in 84 to 88% yield (Table 1, entries 1–3). Substrates 7 and 9, bearing a trifluoroacetoxy and a methoxy group at C3, respectively, required higher amounts of the reagent and longer reaction times (Table 1, entries 4 and 5). The formation of the oleanolic δ-hydroxy-γ-lactones 2, 4, 6, 8 and 10, may be explained by epoxidation of the parent Δ12-oleanane compound, followed by nucleophilic attack of the 28-carboxyl group at C13 from the β-face, with ring-opening of the 12α,13α-epoxide intermediate [15,44].</p><p>Quite recently, we demonstrated that the 28,13β-lactonization of 3-oxooleanolic acid 1 promoted by Bi(OTf)3·xH2O affords a 3-oxo-18α-olean-28,13β-olide product, with inversion of configuration at the C18-stereocenter, as demonstrated by X-ray crystallography [45–46]. In order to assign the orientation of the 18-H of the 12α-hydroxy-γ-lactones obtained in this work, 2D NMR data were collected for compounds 2, 6 and 8, and X-ray data were gathered for compound 4. Combining 1D and 2D-NMR spectroscopy, we were able to determine the chemical shift of 18-H (2.02 ppm) for compound 2. This value is much lower than the one of the parent substrate 1 (2.84 ppm), which may be explained by magnetic anisotropy induced by the 28,13β-lactone moiety. It is also interesting to note that a long-distance coupling between the 18-H and 12β-H (3.90 ppm) signals was found in the COSY spectrum of 2. Correlation between these two signals was also observed in the NOESY experiment and, therefore, the β-configuration was assigned at the C18-stereocenter. The same NMR pattern was present for compounds 6 and 8. Unequivocal evidence of the molecular structure of compound 4 was obtained by single-crystal X-ray crystallography, and the ORTEP diagram with the corresponding atomic numbering scheme is depicted in Figure 1.</p><!><p>ORTEP diagram of compound 4 (50% probability level, H atoms of arbitrary sizes). The asymmetric unit also contains a molecule of CH3CN.</p><!><p>In the past few years, bismuth(III) salts have emerged as convenient reagents for the development of new chemical processes under more \&quot;ecofriendly\&quot; reaction conditions, which avoid the use of large amounts of toxic and corrosive materials [47–51]. Bearing in mind the solubility properties of MMPP and that both the oxidative 28,13β-lactonization and the bismuth(III) triflate-catalyzed direct opening of δ-hydroxy-γ-lactones are performed in acetonitrile, we designed a protocol to perform the synthesis of oleanolic 12-oxo-28-carboxylic acid derivatives directly from Δ12-oleananes, without the need for an intermediary work-up, and keeping the same reaction solvent in both steps (Scheme 1).</p><!><p>Sequential 2-step synthesis of 3,12-dioxoolean-28-oic acid (11) directly from 3-oxooleanolic acid (1).</p><!><p>Thus, after the formation of the 12α-hydroxy-28,13β-olide compound 2 by MMPP oxidation, a filtration step allowed the elimination of insoluble magnesium salts, taking advantage of their low solubility in acetonitrile. Then, a catalytic amount of bismuth(III) triflate (5 mol %) was added to the resulting filtrate, and the expected 12-oxo-28-carboxylic acid 11 was obtained, in 85% yield, after the typical work-up procedure [18]. The formation of compound 11 from the 12α-hydroxy-28,13β-olide 2 is likely to occur due to the in situ generation of a Brønsted acid species from bismuth(III) triflate, which promotes ring opening of the 28,13β-olide group, creating a tertiary carbocation at C-13. Then, a concerted stereoselective 1,2-migration of the 12β-H to the 13β-position with the rearrangement of the 12α-hydroxy group affords the final 12-oxo-28-carboxylic acid structure [18]. The molecular structure of compound 11, determined by single-crystal X-ray crystallography, is shown in Figure 2.</p><!><p>ORTEP diagram of compound 11 (50% probability level, H atoms of arbitrary sizes).</p><!><p>In conclusion, we have found a new straightforward procedure to convert OA derivatives into δ-hydroxy-γ-lactones, in very high yields, using the convenient oxidizing agent MMPP. This procedure has considerable advantages over the previously reported oxidation methods, because no other positions of the molecule are oxidized concomitantly, it avoids the use of halogenated solvents, and allows easy recovery of the reaction products. Combination of this oxidative 28,13β-lactonization process with the ability of bismuth(III) triflate to catalyze the opening of the resulting δ-hydroxy-γ-lactone with subsequent generation of the carbonyl group, allowed us to set up a sequential two step strategy for the preparation of 3,12-dioxoolean-28-oic acid (11) directly from 3-oxooleanolic acid 1, that avoids an intermediary work-up and conveniently uses the same reaction solvent in both steps. Thus, the procedure reported herein greatly simplifies the obtainment of oleanolic δ-hydroxy-γ-lactones, which are versatile intermediates for organic synthesis, and in addition can provide very easy access to the corresponding oleanolic 12-oxo-28-carboxylic acids.</p><!><p>The Supporting Information contains the typical procedure for the MMPP oxidative 28,13β-lactonization and preparation of compounds 2, 4, 6, 8 and 10. Moreover, the procedure for the sequential two step synthesis of 3,12-dioxoolean-28-oic acid (11) is described and the 1D and 2D NMR spectra of compounds 2, 4, 6, 8, 10 and 1D NMR spectra of compound 11 are shown.</p><!><p>Experimental and analytical data.</p>&quot;},&quot;article_source&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;PubMed Open Access&quot;}}},{&quot;rowIdx&quot;:4,&quot;cells&quot;:{&quot;title&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Spectral, thermal, antimicrobial studies for silver(I) complexes of pyrazolone derivatives&quot;},&quot;abstract&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;BackgroundSynthesize new complexes of Ag(I) to enhance efficacy or stability and also, pharmacological activities on the operation of pyrazolone's biological properties.ResultsEfficient and high yielding pathways starting from the versatile and readily available 3-methyl-1-phenyl-5-pyrazolone by Knoevenagel condensation of a sequence of 4-arylidene-3-methyl-1-phenyl-5-pyrazolone derivatives (2a-c) have been formed by the reaction of various substituted aromatic aldehydes Used as ligands to synthesize Ag(I) chelates. Synthesized compounds and their complexes have been characterized by elemental analysis, magnetic and spectroscopic methods (IR, 13C, 1HNMR, mass) and thermal analysis. The spectrophotometric determinations suggest distorted octaedral geometry for all complexes. Both ligands and their metal complexes have also been tested for their antibacterial and antifungal efficacy.ConclusionsNewly synthesized compounds have shown potent antimicrobial activity. The results showed that the complex 's high activity was higher than its free ligands, and that Ag(I)-L3 had the highest activity.&quot;},&quot;uuid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;spectral,_thermal,_antimicrobial_studies_for_silver(i)_complexes_of_pyrazolone_derivatives&quot;},&quot;source_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:1891,&quot;string&quot;:&quot;1,891&quot;},&quot;target_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:138,&quot;string&quot;:&quot;138&quot;},&quot;compression&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:13.702898550724637,&quot;string&quot;:&quot;13.702899&quot;},&quot;headers&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Introduction<!><!>Infrared spectra<!><!>Thermal studies<!><!>Antimicrobial studies<!><!>Conclusion<!>Chemistry<!>Common 3-methyl-1-phenyl-5-pyrazolone synthesis technique (1)<!>Specific method for preparing derivatives of 4-arylidene-3-methyl-1-phenyl-5-pyrazolone (2a-c)<!>4-(4-dimethylamino benzylidene)-3-methyl-1-phenyl-1H-pyrazol-5(4H)-one (2a) L1<!>4-(4-Thiophene)-3-methyl-1-phenyl-1Hpyrazol-5(4H)-one (2b) L2<!>4-(4-methoxy benzylidene)-3-methyl-1-phenyl-1Hpyrazol-5(4H)-one (2c) L3<!>Synthesis of the complexes<!>[Ag(C19H19N3O)2(H2O)2]NO3 (AgC38H42N7O7) complex<!>[Ag(C15H12N2OS)2(H2O)2]NO3.H2O (AgC30H30N5O8S2) complex<!>[Ag(C18H16N2O2)2(H2O)2]NO3 (AgC36H36N5O9) complex<!><!>Supplementary information&quot;},&quot;sections&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;<p>Pyrazolone chemistry began in 1883 when Ludwig Knorr first reacted to phenyl hydrazine with aceto-acetate ester. As pyrazolones were discovered as binding components for azo dyes in the late 1800s, they rapidly increased in importance. Today, pyrazolon is still an significant trade precursor to dyes and pharmaceuticals. Pyrazolone is a biologically important scaffold associated with different pharmacological activities such as antimicrobials [1–5], anti-inflammatory [6], analgesic [7], antidepressant [8], anticonvulsant [9], antidiabetic [10], antihyperlipidemic [11, 12], antiviral [13, 14], anti-tuberculosis [15, 16], antioxidant [17, 18] and anticancer [19, 20]. For several years, the preparation of pyrazolone and its derivatives has attracted significant attention from organic and medicinal chemists, as they belong to a class of compounds with promising results in medicinal chemistry. The heterocycles condensed to the pyrazole ring are an important source of bioactive molecules [21, 22]. Compounds containing both pyrazole and other essential heterocyclic active structural units usually demonstrate more remarkable biological activity. A number of condensed pyrazole derivatives have been reported as four-fold antibacterial agents against Gram-positive and Gram-negative bacteria compared to general pyrazole compounds [23, 24]. A digit of antimicrobial active silver(I) complexes have the capacity to disrupt microbial transpiration as well as block tyrosinase synthesis and are extremely cytotoxic to cancer cells [24]. Massive attention in silver ions (Ag(I)) as a broad spectrum antimicrobial has upped the size and importance of in vitro biocompatibility research [25]. Silver ions are toxic to many bacteria, viruses, algae and fungi. Silver-based medicines have been widely used for this task for decades [26]. The objective of this study is to display the synthesis and characterization of three Ag(I) pyrazolone complexes in an attempt to verify the mode of coordination and the biological properties of the final complexes.</p><!><p>Synthesis of 4-arylidene-3-methyl-1-phenyl-5-pyrazolone derivatives</p><p>The coordinationn mode of Ag (I) with three ligand</p><!><p>KBr disks registered mid-infrared spectra of L1, L2, L3 and their metal complexes. As expected, with changes in band intensities and wave numbers, the absorption bands characteristic of L1, L2, L3 acting as a monodentate unit are observed in the complexes. The proposed structures of the complexes must be considered prior to determining the assignments of the infrared spectra. Here, Ag(I) ion interacts with these monodentate ligands forming monomeric structure complexes in which the Ag(I) ion is four coordinated (Scheme 2) [27–30].</p><!><p>Infrared spectra for a L1, b [Ag(L1)2(H2O)2]NO3, c L2, d [Ag(L2)2(H2O)2]NO3.H2O, e L3 and f [Ag(L3)2(H2O)2]NO3</p><p>Infrared frequencies (cm−1)a and tentative assignmentsb for (A) L1, (B) [Ag(L1)2(H2O)2]NO3, (C), L2 (D) [Ag(L2)2(H2O)2]NO3.H2O, (E) L3 and (F) [Ag(L3)2(H2O)2]NO3</p><p>1550s</p><p>1400m</p><p>1523m</p><p>1410s</p><p>1527m</p><p>1381</p><p>1508s</p><p>1427vw</p><p>1520s</p><p>1415</p><p>ν(C = N)</p><p>ν(C = C)</p><p>1319s</p><p>–</p><p>1319s</p><p>1188s</p><p>1300s</p><p>-</p><p>1311w</p><p>1165m</p><p>1311sh</p><p>-</p><p>1311m</p><p>1172s</p><p>δb(-CH2),</p><p>ν(NO3−1)</p><p>1122s</p><p>1018w</p><p>–</p><p>1122s</p><p>1018w</p><p>–</p><p>1130m</p><p>–</p><p>1056w</p><p>1104vw</p><p>–</p><p>1099sh</p><p>1110w</p><p>–</p><p>–</p><p>1130m</p><p>–</p><p>–</p><p>ν(C–C),</p><p>ν(C-N)</p><p>ν(C = S)</p><p>954w</p><p>943vw</p><p>995w</p><p>995m</p><p>991s</p><p>921w</p><p>941vw</p><p>910vw</p><p>988w</p><p>938 s</p><p>985w</p><p>965sh</p><p>as = strong, w = weak, sh = shoulder, v = very, br = broad, bν = stretching and δ = bending</p><p>Electronic absorption spectra for L1, [Ag(L1)2(H2O)2]NO3, L2, [Ag(L2)2(H2O)2]NO3.H2O, L3 and [Ag(L3)2(H2O)2]NO3</p><p>1H NMR spectra for a L1, b [Ag(L1)2(H2O)2]NO3, c L2, d [Ag(L2)2(H2O)2]NO3.H2O, e L3 and f [Ag(L3)2(H2O)2]NO3</p><p>Thermogravimetric data of L1, L2,L3 and their metal complexes</p><!><p>The ligand (L2) degradates at 273, 475 °C. This stage is followed by a complete loss of weight of 86.70 percent, close to 86.56 percent of the estimated value (Additional file 1: Fig. S1c). Equivalent to 6C2H2 + SO + N2 loss and 31.93 kJ mol−1 (endothermic) activation energy. Decomposition of the residual value occurs at 771 °C and the real weight loss from this stage is 13.30 percent, similar to the estimated value of 13.43 percent corresponding to 3C. The [Ag(L2)2(H2O)2]NO3.H2O complex decomposes at two levels of decay (Additional file 1: Fig. S1d), the first phase occurs at 99 °C and is followed by a weight loss of 2.08 per cent relating to the removal of H2O, activation energy of 79.28 kJ mol−1. The second step of decomposition occurs at temperature is 203, 528 and is accompanied by a weight loss of 75.90%; corresponding to the value of 10C2H2 + 4HCN + 2H2O + NO2 + SO + SO2 theoretically, close to the calculated value 76.404%. The Residue value decomposition occurs at maximum 881 °C and the actual weight loss from this step is 23.35%, corresponding to Ag + 6C, close to the calculated value 23.596%.</p><p>The thermal decay of L3 happens in two phases of degradation (Additional file 1: Fig. S1e), the first step arises at 291 °C and is followed by a weight loss of 70.55 percent leading to a loss of 8C2H2 similar to the measured value of 71.23 per cent with activation energy of 35.31 kJ mol−1. The second step occurs at 518 οC and is accompanied by a weight loss of 28.604%; corresponding to the value of 2CO + N2 theoretically, close to the calculated value 28.67%. The [Ag(L3)2(H2O)2]NO3 degradation takes place in two stages (Additional file 1: Fig. S1f), the first occurs at 244 οC and is accompained by a weight loss of 51.071% corresponding to loss of 14C2H2 + 2H2O close to the calculated value 50.60% with an activation energy 15.31 kJ mol−1. The second one begins at 543 οC and is followed by a weight loss of 30.17%; corresponding to C2H2 + CO + 2HCN + 3NO2 theoretically, close to the calculated value 31.25%. The Residue remains at 677 °C and the actual weight loss is 17.76%, equal to Ag + 3C, close to the calculated value 18.15%.</p><!><p>Thermal behavior and kinetic parameters determined using the Coats–Redfern (CR) and Horowitz–Metzger (HM) operated for L1, L2, L3 and their complexes</p><p>CR</p><p>HM</p><p>95.656</p><p>116.813</p><p>69.368 × 103</p><p>30.233 × 103</p><p>− 432.844</p><p>− 425.910</p><p>88.863</p><p>110.020</p><p>442.497</p><p>457.988</p><p>0.970</p><p>0.984</p><p>0.187</p><p>0.065</p><p>CR</p><p>HM</p><p>34.379</p><p>38.825</p><p>3.390 × 102</p><p>2.621 × 103</p><p>− 393.344</p><p>− 410.348</p><p>30.537</p><p>34.983</p><p>212.262</p><p>224.564</p><p>0.984</p><p>0.970</p><p>0.116</p><p>0.087</p><p>CR</p><p>HM</p><p>31.931</p><p>40.983</p><p>37.451</p><p>672.979</p><p>− 373.640</p><p>− 397.656</p><p>27.391</p><p>36.443</p><p>231.398</p><p>253.563</p><p>0.943</p><p>0.945</p><p>0.207</p><p>0.112</p><p>CR</p><p>HM</p><p>79.284</p><p>93.856</p><p>6.618 × 103</p><p>113.45 × 103</p><p>− 413.475</p><p>− 437.099</p><p>72.624</p><p>87.196</p><p>403.817</p><p>437.312</p><p>0.945</p><p>0.940</p><p>0.220</p><p>0.116</p><p>CR</p><p>HM</p><p>35.317</p><p>48.603</p><p>1.185 × 102</p><p>2.847 × 103</p><p>− 382.948</p><p>− 409.379</p><p>30.627</p><p>43.913</p><p>246.610</p><p>274.803</p><p>0.960</p><p>0.952</p><p>0.182</p><p>0.106</p><p>CR</p><p>HM</p><p>15.316</p><p>22.370</p><p>0.758</p><p>0.119 × 102</p><p>− 342.002</p><p>− 364.917</p><p>11.183</p><p>18.237</p><p>181.158</p><p>199.601</p><p>0.981</p><p>0.985</p><p>0.089</p><p>0.054</p><p>aCorrelation coefficients of the Arrhenius plots and bStandard deviation</p><p>Mass spectra diagrams for (A) L1, (B) [Ag(L1)2(H2O)2]NO3, (C), L2 (D) [Ag(L2)2(H2O)2]NO3.H2O, (E) L3 and (F) [Ag(L3)2(H2O)2]NO3</p><p>Fragmentation pattern of L1</p><p>Fragmentation pattern of L2</p><p>Fragmentation pattern of L3</p><p>The inhibitation diameters zone values (mm) for L1, L2, L3 and its complexes</p><p>Statistical significance PNS – P not significant, P > 0.05; P+1 – P significant, P < 0.05; P+2 – P highly significant, P < 0.01; P+3 – P very highly significant, P > 0.001; Student's t-test (Paired)</p><p>Statistical representation for biological activity of L1, L2, L3 and its metal complexes</p><!><p>Normal antibiotic efficacy of antimicrobials (AMC, CTX, NS, FU). The AMC mixture give the effective against E. coli, Coliform, S. aureus and NS high inhibitory activity on A. niger. Other antibiotics have shown no action on other microorganisms. Eventually, the bacterial strains showed a varied response to the three free ligands and their complex antimicrobial activity, but the results indicated that the high activity of ligand complexes was better than their free ligands. The two fungal strains are more resistant to synthesis ligands and their complexes than bacterial strains [42–46].</p><!><p>(A) Of One-way ANOVA: E. coli vs MIC Compounds. (B) Of One-way ANOVA: Coliform versus MIC Compounds. (C) Of One-way ANOVA: S. aureus vs MIC Compounds. (D) Of One-way ANOVA: Salm. typhi vs MIC Compounds. (E) Of One-way ANOVA: A. niger vs MIC Compounds. (E) Of One-way ANOVA: A. niger vs MIC Compounds. (F) Of One-way ANOVA: P.expansum vs MIC Compounds</p><p>Means that do not share a letter are significantly different</p><p>Fisher 95% Simultaneous Confidence Intervals</p><p>MIC for the most sensitive organisms</p><!><p>Development and characterisation of three novel complexes of some replaced pyrazole derivatives as ligands (4-(4-dimethylamino benzylidene)-3-methyl-1-phenyl-1H-pyrazol-5(4H)-one (2a) L1, 4-(4-Thiophene)-3-methyl-1-phenyl-1Hpyrazol-5(4H)-one (2b) L2, 4-(4-methoxy benzylidene)-3-methyl-1-phenyl-1Hpyrazol-5(4H)-one (2c) L3) with Ag(I) was achieved using physicochemical and spectroscopic methods.. In the resulting complexes, L1, L2, and L3 were bound by the nitrogen atom to the metal ion via ν(C = N). For the three ligands and their complexes, thermogravimetric kinetic parameters and their differential were evaluated using the Coats-Redfern and Horowitz-Metzger equations. Metal complexes exhibited higher inhibition against all tested microorganisms and pathogenic bacteria and fungi and were the most susceptible pathogens with a minimum inhibitory concentration ( MIC).</p><!><p>Analytical grade reagents, commercially available from multiple suppliers and used without further purification, were all the chemicals used in the complex preparation. Synthesized compounds and their complexes have been characterized by elemental analysis, magnetic and spectroscopic methods (IR, 13C, 1HNMR, mass) and thermal analysis using the known apparatuses [42].</p><!><p>Pure ethyl acetoacetate (0.05 mol, 6.2 mL) was mixed with pure phenyl hydrazine (0.05 mol, 5 mL), 0.5 mL of acetic acid was added, according to knowm method [42]. Methyl phenyl pyrazolone was obtained as colorless crystals, 127 °C melting point and 83.6 percent yield [27].</p><!><p>The oil bath heated a mixture of 1-aryl-3-methyl-5-pyrazolone (0.01 mol, 1.74 g) and replaced aromatic aldehydes (0.012 mol) at 150–160 °C for 2-4hrs. TLC has tracked the progress of the reaction using ethyl acetate: hexane (9:1) as solvent. The mixture was cooled, triturated and washed off with ether (20 mL). The colored residue was recrystallized from ethanol to provide the corresponding 4-arylidene-3-methyl-1-phenyl-5-pyrazolone (2a-c) as colored products, respectively [28].</p><p>4-(4-dimethylamino benzylidene)-3-methyl-1-phenyl-1H-pyrazol-5(4H)-one (2a) L1.</p><p>4-(4-Thiophene)-3-methyl-1-phenyl-1Hpyrazol-5(4H)-one (2b) L2.</p><p>4-(4-methoxy benzylidene)-3-methyl-1-phenyl-1Hpyrazol-5(4H)-one (2c) L3.</p><!><p>Brick Red, mp = 170 °C, yield 83% IR (KBr, v, cm−1): 3444 (OH), 1670 (C = O), and 1550 cm−1. 1H NMR (DMSO-d6, 300 MHz): δ = 2.28 (s, 3H, CH3), 3.03 (s, 6H, -N (CH3)2), 7.14 (S, 1H, = CH-Ar), 9.66 (d, 3H, Ar–H),. Anal. Calcd for C19H19N3O (305.19): C, 74.40; H, 6.22; N 13.76; Found C, 74.23; H, 6.13; N, 13.35%.</p><!><p>Orange, mp = 125 °C, yield 74% IR (KBr, v, cm−1): 3448 (OH), 1681 (C = O), 1496 cm−1 (C = N) and 1056 cm−1(C = S). 1H NMR (DMSO-d6, 300 MHz): δ = 2.30 (s, 3H, CH3), 7.39 (S, 1H, = CH-Ar), 8.25 (d, 3H, Ar–H). Anal. Calcd for C15H12N2OS (268): C, 67.16; H, 4.47; N 10.44; S, 11.94; Found C, 67.00; H, 4.32; N, 10.21; S, 11.65%.</p><!><p>Orange, mp = 122 °C, yield 82% IR (KBr, v, cm−1): 3444 (OH), 1678 (C = O), 1508 cm−1 (C = N) and. 1H NMR (DMSO-d6, 300 MHz): δ = 1.91 (s, 3H, CH3), 3.69 (s, 3H, -OCH3), 7.20 (S, 1H, = CH-Ar), 8.71 (d, 3H, Ar–H).Anal. Calcd for C18H16N2O2 (292): C, 73.97; H, 5.47; N 9.58; Found C, 73.78; H, 5.13; N, 9.34%.</p><!><p>The brown solid complex [Ag(L1)2(H2O)2]NO3 was prepared by adding 0.5 mmol (0.085 g) of AgNO3 in 20 ml of acetone to a stirred suspended solution 1 mmol (0.305 g) of L1 in 50 ml acetone. The reaction mixture was refluxed for 6 h, the precipitate was drained off, washed several times with acetone and dried under vacuum over anhydrous CaCl2. Dark brown [Ag(L2)2(H2O)2]NO3.H2O, [Ag(L3)2(H2O)2]NO3 solid complexes were prepared in the same manner as mentioned above.</p><!><p>Brown; Yield: 85%; m.p.: 160 οC; M.Wt: 816.65; Elemental analysis for AgC38H42N7O7: found, C, 55.31; H, 4.99; N, 12.00; Ag, 13.14; Calcd, C 55.89; H, 5.18; N, 12.01; Ag, 13.21; Λm = 115.75 S cm2 mol−1; IR (KBr, v, cm−1): 3450 m,br (OH), 1666 m (C = O), 1523vw cm−1(C = N) and 813w and 837w (M–N). 1H NMR (DMSO-d6, 300 MHz): δ = 2.49 (s, 3H, CH3), 3.46 (s, 2H, H2O), 2.27–2.33 (s, 6H, -N (CH3)2), 9.67 (S, 1H, = CH-Ar), 7.14–7.97 (m, 4H, Ar–H).</p><!><p>Dark brown; Yield: 74%; m.p.: 125 οC; M.Wt: 760.59; Elemental analysis for AgC30H30N5O8S2: found, C, 47.22; H, 3.91; N, 9.15; Ag, 14.13; Calcd, C, 47.37; H, 3.98; N, 9.21; Ag, 14.18; Λm = 135.50 S cm2 mol−1; IR (KBr, v, cm−1): 3444 m, br (OH), 1685 m (C = O), 1527vw cm−1 (C = N), 1099 m cm−1(C = S), 748w and 792w (M–N). 1H NMR (DMSO-d6, 300 MHz): δ = 2.49 (s, 3H, CH3), 3.37 (s, 2H, H2O), 8.64 (S, 1H, = CH-Ar), 7.20–7.94 (d, 3H, Ar–H).</p><!><p>Dark brown; Yield: 90%; m.p.: 150 οC; M.Wt: 790.57; Elemental analysis for AgC36H36N5O9: found, C, 54.47; H, 4.11; N, 8.80; Ag, 13.60; Calcd, C, 54.69; H, 4.59; N, 8.86; Ag, 13.64; Λm = 114.52 S cm2 mol−1; IR (KBr, v, cm−1): 3444 (OH), 1678 (C = O), 1520 cm−1 (C = N), 759w and 779w (M–N). 1H NMR (DMSO-d6, 300 MHz): δ = 2.33 (s, 3H, CH3), 3.31 (s, 3H, -OCH3), 8.42 (S, 1H, = CH-Ar), 7.18–7.46 (d, 3H, Ar–H).</p><!><p>Additional file 1: Table S1. UV-Vis. spectral data of the free ligand L1, L2, L3 and their Ag(I)-complexes. Table S2. Selected 1H NMR data of L1, L2, L3 and its diamagnetic complexes. Fig. S1. TGA and DTG diagrams for a L1, b [Ag(L1)2(H2O)2]NO3, c, L2\nd [Ag(L2)2(H2O)2]NO3.H2O, e L3 and f [Ag(L3)2(H2O)2]NO3. Fig. S2. The diagrams of kinetic parameters of L1, [Ag(L1)2(H2O)2]NO3, L2, [Ag(L2)2(H2O)2]NO3.H2O, L3 and [Ag(L3)2(H2O)2]NO3using Coats-Redfern (CR) and Horowitz-Metzger (HM) equations. Scheme S1. Fragmentation pattern of [Ag(L1)2(H2O)2]NO3. Scheme S2. Fragmentation pattern of [Ag(L2)2(H2O)2]NO3.H2O. Scheme S3. Fragmentation pattern of [Ag(L3)2(H2O)2]NO3</p><p>Ethanol</p><p>Nuclear magnetic resonance</p><p>Infrared radiation</p><p>Dimethyl sulfoxide</p><p>Minimum inhibation concentrations</p><p>Publisher's Note</p><p>Springer Nature remains neutral with regard to jurisdictional claims in published maps and institutional affiliations.</p><!><p>Supplementary information accompanies this paper at 10.1186/s13065-020-00723-0.</p>&quot;},&quot;article_source&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;PubMed Open Access&quot;}}},{&quot;rowIdx&quot;:5,&quot;cells&quot;:{&quot;title&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Evaluation of an Artificial Neural Network Retention Index Model for Chemical Structure Identification in Nontargeted Metabolomics&quot;},&quot;abstract&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Liquid chromatography coupled with electrospray ionization tandem mass spectrometry (LC-ESI-MS/MS) is a major analytical technique used for nontargeted identification of metabolites in biological fluids. Typically, in LC-ESI-MS/MS based database assisted structure elucidation pipelines, the exact mass of an unknown compound is used to mine a chemical structure database to acquire an initial set of possible candidates. Subsequent matching of the collision induced dissociation (CID) spectrum of the unknown to the CID spectra of candidate structures facilitates identification. However, this approach often fails because of the large numbers of potential candidates (i.e., false positives) for which CID spectra are not available. To overcome this problem, CID fragmentation predication programs have been developed, but these also have limited success if large numbers of isomers with similar CID spectra are present in the candidate set. In this study, we investigated the use of a retention index (RI) predictive model as an orthogonal method to help improve identification rates. The model was used to eliminate candidate structures whose predicted RI values differed significantly from the experimentally determined RI value of the unknown compound. We tested this approach using a set of ninety-one endogenous metabolites and four in silico CID fragmentation algorithms: CFM-ID, CSI:FingerID, Mass Frontier, and MetFrag. Candidate sets obtained from PubChem and the Human Metabolite Database (HMDB) were ranked with and without RI filtering followed by in silico spectral matching. Upon RI filtering, 12 of the ninety-one metabolites were eliminated from their respective candidate sets, i.e., were scored incorrectly as negatives. For the remaining seventy-nine compounds, we show that RI filtering eliminated an average of 58% from PubChem candidate sets. This resulted in an approximately 2-fold improvement in average rankings when using CFM-ID, Mass Frontier, and MetFrag. In addition, RI filtering slightly increased the occurrence of number one rankings for all 4 fragmentation algorithms. However, RI filtering did not significantly improve average rankings when HMDB was used as the candidate database, nor did it significantly improve average rankings when using CSI:FingerID. Overall, we show that the current RI model incorrectly eliminated more true positives (12) than were expected (4\\xe2\\x80\\x935) on the basis of the filtering method. However, it slightly improved the number of correct first place rankings and improved overall average rankings when using CFM-ID, Mass Frontier, and MetFrag.&quot;},&quot;uuid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;evaluation_of_an_artificial_neural_network_retention_index_model_for_chemical_structure_identificati&quot;},&quot;source_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:5610,&quot;string&quot;:&quot;5,610&quot;},&quot;target_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:376,&quot;string&quot;:&quot;376&quot;},&quot;compression&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:14.920212765957446,&quot;string&quot;:&quot;14.920213&quot;},&quot;headers&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;<!>Reagents and Chemicals.<!>Sample Preparation.<!>UPLC-ESI-MS/MS.<!>Retention Index Measurements.<!>Compound Set Diversity.<!>Data Analysis.<!>Candidate Structures.<!>Preprocessing of Candidates.<!>In Silico RI Prediction and Filtering.<!>In Silico Predictive Fragmenters.<!>Reduction in Candidate Set Sizes Upon RI Filtering.<!>Predictive Fragmenters Performance with and without RI Filtering.<!>Fragmentation Tree Filter.<!>Evaluation of RI Model Filter Windows.<!>CONCLUSIONS&quot;},&quot;sections&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;<p>The composition and concentrations of small molecule metabolites (20–1000 Da) in living organisms frequently change over time and represent the biochemical phenotype of an individual. These changes can be due to multiple factors including diet, time of day, environmental exposures, disease, drug exposures, genetic manipulation, gender, and age.1,2 Nontargeted metabolomics is the unbiased quantification and identification of these metabolites in biological samples.3</p><p>In nontargeted metabolomics, researchers often utilize liquid chromatography coupled with electrospray ionization mass spectrometry (LC-ESI-MS) as the major analytical technique to separate and accurately measure the precursor masses of thousands of metabolites present in biological samples.4 However, to elucidate the chemical structure of these compounds, tandem mass spectrometry (LC-ESI-MS/MS) is often used.5 In LC-ESI-MS/MS, an experimental collision induced dissociation (CID) spectrum of an isolated precursor ion is used as a fingerprint in matching against a collection of reference CID spectra of known compounds in spectral libraries (e.g., MassBank, Metlin).6–8 Unfortunately, this approach often fails due to the dependency of CID spectral profiles on experimental conditions and lack of coverage of chemical space that pertains to endogenous human metabolites within existing spectral libraries.6,9–11</p><p>To overcome this limitation, computational fragmentation software (predictive fragmenters) has been developed with the aim of predicting experimental tandem mass spectral (MS/MS) profiles and the chemical structure of ensuing predictive fragments10,12 Working principles of these predictive fragmenters are described elsewhere;13 thus, only a brief overview is given here.</p><p>Commercial predictive fragmenters such as ACD/MS fragmenter (Advanced Chemistry Development Laboratories, www.acdlabs.com) and MassFroniter (Thermo Scientific, www.thermoscientific.com) rely on general ionization fragmentation and rearrangement rules along with fragmentation schemes collected from the literature to predict the chemical structures of energy induced fragment ions generated from precusor ions of specific structural composition (https://tools.thermofisher.com).14 These tools are extremely helpful in aiding manual spectral interpretation, which can be laborintensive.15 However, cost, lack of automated candidate retrieval, and ranking protocols limit their use in high-throughput metabolomics pipelines.16,17 In contrast, predictive fragmenters such as MetFrag and MAGMa attempt to explain ion peaks in an experimental CID spectrum by systematic dissociation of all the bonds in a given molecule. In other words, these predictive fragmenters compute all possible fragments of a molecule and then compare the mass of these fragments with the m/z values of fragments in an experimental CID spectrum. Compounds are ranked by assigning a score which is a function of ion peak intensity and the number of peaks explained in the experimental CID spectrum, bond dissociation energies, and neutral losses to account for rearrangements.10,18,19 Free availability, automated workflows, and faster processing times makes these programs popular among metabolomic researchers.20</p><p>Machine learning (ML) is one of the most rapidly growing areas in computer science.21 ML involves the development of computer algorithms that learn from example data or past experience to solve or predict the outcome of an unfamiliar problem.22 Predictive fragmenters such as CFM-ID and CSI:FingerID have been developed on the basis of ML paradigms.20,23 CFM-ID23 utilizes a stochastic, generative Markov model trained using the CID mass spectral profiles of approximately 3500 metabolites randomly chosen from the Metlin database.23 This method allows for the prediction of CID spectral profiles and can also be used to rank candidates based solely on the similarity between predicted and experimental spectra.23 CSI:FingerID is based on fragmentation trees and kernel based support vector machines trained to predict molecular structural features from CID spectra.20,24,25 The predicted set of molecular features (or a fingerprint) is used to rank candidates based on maximum likelihood considerations and Platt probabilities to refine the fingerprint similarity scoring.20 It is important to note that training data used in ML methods have a significant influence on identification quality. Generally, ML based predictive fragmenters outperform other competing predictive fragmenters but have longer processing times.13,20,26</p><p>Upon completion of an ESI-LC-MS/MS analysis, measured experimental features such as monoisotopic mass (MIM), retention time, and CID spectra are often utilized by researchers to elucidate the chemical structure of an unknown compound (peak) of interest. In database assisted structure elucidation pipelines, the first step is the acquisition of candidate structures for the unknown by matching the measured MIM to compounds in an all-purpose chemical database such as PubChem or ChemSpider or specialized biological databases such as the Human Metabolite Database (HMDB) or Kyoto Encyclopedia of Genes and Genomes (KEGG).13,27,28 According to the critical assessment of small molecule identification (CASMI) contests,26,29 the use of specialized databases improves the chances of correctly identifying the unknown structure. However, a major disadvantage of this approach is the inherent incompleteness of such databases. If the unknown compound is not contained in the database, it cannot be correctly identified.30 Thus, it can be argued that there are advantages in mining large chemical databases such as PubChem (which currently contains more than 90 million compounds) or ChemSpider (which currently contains more than 59 million compounds) because of their much larger size and the inclusion of more diverse compound classes. Nonetheless, mining such databases may also result in large candidate sets.31,32 For example, searching for 5-hydroxytryptophan (MIM = 220.0848 Da, ±5 ppm) using the MetFrag web interface (http://msbi.ipb-halle.de/MetFragBeta/) yields 8223 candidates from PubChem and 3777 from Chemspider. The same search resulted in 3 candidates from HMDB and 4 candidates from KEGG.</p><p>Generating the candidate list from large databases increases the likelihood that the unknown will be included in the candidate list but also dramatically increases the number of false positives. To address this problem, we have developed a software package called MolFind,33 which relies on a set of orthogonal experimental features acquired from LC-ESI-MS/MS experiments. These include Retention Index (RI), Drift Index (DI), and Ecom50 (collision gas normalized energy required to fragment 50% of precursor ions). For each experimental feature, MolFind eliminates false positives by comparing the experimental value of the unknown to a value predicted for each candidate compound using a computational model. A candiate compound is excluded when a predicted value for the candidate is substantially different from the experimental value of the unknown.33 In previous studies, when both RI and Ecom50 filters were applied concurrently with MetFrag, ranking of the correct compound improved from 142 to 102 on average (over 35 sets).33 More importantly, it was suggested that enhancements in the accuracy of the RI and Ecom50 models could lead to removal of 87.2% of candidates on average, attaining a potential increase in an average MetFrag ranking of 15.5.33</p><p>The robustness of modeling approaches in predicting experimental features is heavily dependent on the training set. The models used in our earlier study30 were far from optimal, as the Ecom50 model and the RI models were trained using 54 and 400 compounds, respectively, having 99.5% confidence intervals of 2.1 eV and 114 RI units (RIU), respectively.30 Recently, we have improved our RI model by training with a diverse set of synthetic chemicals (1955 in total) covering diverse chemical classes representative of endogenous human metabolites.34 Confirmed endogenous human metabolites were deliberately excluded from the model data set in the previous study. A total of 202 confirmed metabolites were set aside as an independent validation set to test if a model based exclusively on relatively simple synthetic compounds could be used to make predictions for more complex metabolites.34 For the model used in this present study, the 202 independent validation compounds were reintroduced and the model was rebuilt using the same descriptors and learning protocol as the previous study resulting in a model based on 2157 compounds. This was done to take advantage of the available data on confirmed metabolites and expand the model applicability domain.</p><p>As an extension to our previous work, the present study investigates the improvement in identification quality by enrichment of candidate sets using the expanded 2157 compoound RI model in conjunction with four different fragmentation algorithms: CFM-ID, CSI-FingerID, Mass Frontier, and MetFrag. Candidate compounds were taken from both a large (PubChem) and small (HMDB) database.</p><!><p>Acetonitrile (HPLC, gradient grade) and methanol (HPLC grade) were purchased from Sigma-Aldrich (St Louis, MO, USA). Water (18.2 MΩ·cm) used for the UPLC mobile phase and sample preparation. Reagent grade water was generated on a Burnstead Nanopure Diamond system (Thermo Scientific, Ward Hill, MA, USA). Heptafluorobutyric acid (HPLC grade) was purchased from Thermo Fisher Scientific Chemicals Inc. (Ward Hill, MA, USA). n-Propionamide, n-butanamide, and n-hexanamide were ordered from Aldrich (St Louis, MO, USA). n-Pentanamide was ordered from MP Biochemicals, LLC (Solon, OH, USA). A series of n-C7–C14 amides were synthesized as described in Supporting Information SI–1. The 91 test compounds and the controls used in the study were purchased from various sources, and the vendor information is summarized in Table S2. HPLC grade formic acid (98–100%) was purchased from EMD Millipore Corporation (Billerica, MA, USA).</p><!><p>Two different approaches were followed for the sample preparation. As the chemicals ordered from IROA technologies were contained in plates of polypropylene wells containing 5 μg of each chemical, 100 μL of solvent (0.1% formic acid in water, 0.05% formic acid in water/methanol (1:1) (v/v), or methanol) based on the XLOGP3 (taken from PubChem) value was added to each well. The plates were covered with sealing tape to prevent evaporation. Dissolution was achieved by shaking wells on an Innova 2100 platform shaker (New Brunswick, CT, USA) for 45 min. Finally, the dissolved chemicals were transferred to 2 mL HPLC vials with micro volume glass inserts (Thermo Fisher Scientific, Ward Hill, MA, USA), sealed with Teflon septum caps, and used directly for UPLC analysis. Stock solutions of all other chemicals were prepared at 1–10 μmol/mL concentrations in the appropriate solvent based on the analyte's XLOGP3 value as described above. The prepared stock solutions were further diluted at appropriate concentrations and used for the UPLC analysis.</p><!><p>Retention index values were measured on a Zorbax, SB-C18, 2.1 mm × 150 mm, 1.8 μm column (Agilent Technologies, Santa Clara, CA, USA) using an Acquity UPLC liquid chromatographic system (Waters, Milford, MA, USA). Solvent A was 0.766 mM heptafluoroacetic acid (HFBA) in water, and solvent B was 0.766 mM HFBA in 10% water/acetonitrile (v/v). Compounds were eluted from the column using a solvent program consisting of a 4 min isocratic hold of 2% solvent B followed by a 20 min linear gradient to 100% solvent B and a 5 min isocratic hold at a flow rate of 388 μL/min. The RI model used in this study was trained and validated using retention data for compounds analyzed on an Agilent 1100 capillary HPLC system (Agilent Technologies, Santa Clara, CA, USA). Thus, the protocols used to transfer and validate the LC method to the Acquity UPLC system are summerized in Supporting Information SI–2. The outlet of the UPLC system was connected to the electrospray (ESI) ionization source of a Synapt G2-Si mass spectrometer (Waters, Milford, MA, USA) operating in the positive ion mode. A solution of leucine enkephalin (556.2771 Da, 400 pg/uL) in 0.1% (v/v) formic acid–methanol/water (1:1) was infused as the lock mass reference compound at a flow rate of 5 μL/min. The retention times of the test compounds were measured in duplicate using detection parameters described in Supporting Information SI–2. The mass range of the detector was set to 20–1000 Da. CID was carried out with nitrogen as the collision gas, and the collision energy was varied from 0 to 30 eV (30–60 eV if required) in incremental steps of 2 eV at a scan rate of 12.5 scans/s.</p><!><p>RI values were measured on the basis of a method developed by Hall et al.35 At the beginning and the end of each run, 1 μL of a homologues series of n-C3–C14 amides was injected on the described UPLC system. The average retention times (RT) of the individual n-amides were used as the calibration reference for calculating RI values for the test compounds. The RI of each n-amide was defined as 100 times the number of carbon atoms. Owing to the linear relationships between RI vs log of RT in the isocratic part and RT vs RI in the gradient part of the solvent program, RI values of compounds eluted during isocratic and gradient parts were calculated by the following equations, respectively\n(1)RIisocratic=(logTx−logTz)100/(logTz+1−logTz)+100z\n(2)RIgradient=(Tx−Tz)100/(Tz+1−Tz)+100z\nwhere Tx is the retention time of the analyte; Tz is the retention time of the n-amides eluting just before the analyte; Tz+1 is the retention time of the n-amides eluting just after the analyte.</p><!><p>A set of ninety-one endogenous metabolites, not included in the training set of the RI model, were selected for the study. The set consisted of a variety of chemical classes including alkaloids, amines, benzene and substituted derivatives, benzenoids, carbohydrates and conjugates, carboxylic acids, fatty acyls, flavin nucleotides, imidazopyrimidines, indoles, lipids and lipid like molecules, morphinans, organic acids, organic carbonic acids, organic nitrogen compounds, organic phosphonic acids, organoheterocyclic compounds, pteridines and derivatives, purine nucleosides, pyridines, quinolines, sphingolipids, steroids, stilbenes, and tetrahydroisoquinolines. Four of the compounds had an overall charge of +1 while the rest were neutral. The XLOGP3 values (predicted octanol–water partition coefficient) of the compound set varied from −5.0 to 8.5 (Figure 1) with a standard deviation of 3.4. The MIMs of the compounds varied from 105 to 785 Da with a standard deviation of 99 Da. Structural details for each test compound can be found in Table S3.</p><!><p>Retention and mass spectral data were processed using Masslynx version 4.1 (Waters, Milford, MA, USA). The experimental survival yields of precursor ions at each collision energy were calculated using eq 3 on the basis of a code written in Python version 3 (http://www.python.org).</p><!><p>Candidate structures were downloaded from PubChem and HMDB by matching the experimental MIM within a relative mass error of ±15 ppm.36 MIMs were calculated by averaging scans with counts higher than or equal to 25% of the respective precursor ion peak apex in the CID spectrum acquired at an energy which resulted in a precursor survival yield closest to 20%.36 The relative mass error (15 ppm) was calculated at the 3-sigma limit when comparing actual to experimental MIMs. PubChem was queried using an in-house program written in python version 3 using underlying cheminformatics functions of Rdkit (http://rdkit.org/)37 and power user gateway (PUG) service (https://pubchem.ncbi.nlm.nih.gov/pug/). Downloaded candidate sets of compounds were saved in 91 separate structure-data (SD) files. A snapshot of HMDB database was downloaded as an SD file, and http://www.hmdb.ca/downloads was used to acquire candidate compound structures. Resulting candidate sets were saved in 91 separate SD files.</p><!><p>Compounds in candidate sets that contained33 salts, had disconnections, contained heavy isotopes, had an overall charge (this filter was not applied to the candidate sets of the quaternary ammonium ions), contained only carbon and hydrogen, or were duplicate stereoisomers were removed prior to processing. Compounds with elements other than CHNOPS were also removed.</p><!><p>In silico RI prediction of the remaining candidates was carried out using topological descriptors calculated by winMolconn (version 2.1)38 and newly trained parameters. The learning process used to develop the RI model has been described previously.34 Briefly, a 4 × 10 × 10 artificial neural network ensemble model was built on RI data for 2157 compounds (1955 commercial synthetic compounds and 202 confirmed human endogenous metabolites) measured according to the protocol described above. The model was trained according to the learning method described in the previous publication34 with RPROP40 back propagation on a network architecture of 47 input neurons and one hidden layer of 23 neurons.</p><p>The significant difference between the model used for this study and the previous publication is that confirmed human endogenous metabolites were deliberately excluded from the model data set in the previous publication. This was done in part to determine if a model based exclusively on relatively simple synthetic compounds could be used to make reasonable predictions for more complex human metabolites. To facilitate this, 202 confirmed human metabolites were set aside as an independent validation set. For this study, the independent validation set was reintroduced to the model data set to. After the additional data was added, the model was rebuilt using the same descriptors and learning method as the previous study. The metabolite data was reintroduced because the goal of this present study was to test RI model performance in the context of the MolFind algorithm, not to test if a model based on synthetic compounds could be used to predict the RI of metabolites. The addition of the confirmed human metabolites makes use of all available data so as to achieve the maximal applicability domain.</p><p>RI predictions were made for compounds in the PubChem and HMDB candidate lists corresponding to each of the 91 unknowns. Candidates with predicted RI values that deviated more than a threshold value were eliminated. The threshold windows were chosen on the basis of an algorithm that utilized the experimental RI value of the \&quot;unknown\&quot; and the similarity of each candidate to model data. Similarity was evaluated using a partial molecular fingerprint encoded in a bit key. The filter windows can be found in Table 4. The bit key similarity approach and algorithm for derivation of the filter windows are discussed in Supporting Information SI–4. The resulting RI-filtered candidate sets were saved separately.</p><!><p>CFM-ID, CSI:FingerID, Mass Frontier, and MetFrag were used without modifications to rank candidate sets resulting from pre- and RI filtering. MAGMa19 was not used in the study as it does not support processing compounds with an overall charge of +1 (adduct type [M+]). Relative and absolute mass errors of 15.0 ppm and 0.01 Da were used to annotate fragments against peaks in respective experimental CID spectra with relative ion intensities closest to 20% survival yield (eq 3).</p><p>For single energy CFM-ID (version 2.0), the pretrained model params_se_cfm, having parameter file param_output.-log, was used (https://sourceforge.net/p/cfm-id/wiki/Home/). The experimental CID spectrum was repeated in all three energies (low, medium, high) such that CFM-ID assigned an average Jaccard score (default method) in ranking candidates.23</p><p>CSI:FingerID (version 3.5) was run with instrument mode set to \&quot;qtof\&quot; and setting all the other parameters to defaults. Candidates with molecular formulas having tree scores lower than the cutoff of <75% of the highest scoring tree were eliminated. The remaining candidates were processed with FingerID.</p><p>Mass Frontier (version 7.0.5.9) was run in batch mode using general fragmentation rules. Parameter settings were as follows : ionization method = protonation, maximum number of reaction steps = 5, reaction limit = 10 000, and mass range = 20–1000 Da. As Mass Frontier lacked functionality in automated candidate ranking, a python program was written to calculate number of peaks matched in the CID spectrum by Mass Frontier generated fragments (within either 15 ppm or 0.01 Da). Candidates were ranked considering the number of peaks matched in the experimental CID spectrum.</p><p>The command line version of MetFrag (version 2.3.1) was downloaded and used http://c-ruttkies.github.io/MetFrag/. None of the pre- or postprocessing filters (MetFragPreProcessingCandidateFilter and MetFragPostProcessingCandidateFilter) were applied. Default values for all other parameters were used.27</p><p>It is important to note that the objective of the study was to investigate the performance of the RI model in eliminating false positives contained in candidate sets and the subsequent improvement in predictive fragmenter performance. Our intent was not to evaluate or benchmark the 4 predictive fragmenters that we used (as done in previous CASMI contests).26 Researchers are encouraged to use this study as a guide to test the RI model with any predictive fragmenter of their choice.</p><!><p>The aim of this study was to determine whether we could improve identification rankings by eliminating false positives using an RI filter prior to CID spectral matching. CFM-ID, CSI:FingerID, Mass Frontier, and MetFrag were used for spectral matching using data sets downloaded from PubChem and HMDB for 91 endogenous metabolites treated as \&quot;unknowns\&quot;. As with all approaches such as this, we face the problem of false negatives; i.e., the RI model will incorrectly eliminate a certain percentage of correct candidates. In this case, 12 of the ninety-one compounds evaluated were eliminated from their respective candidate sets after RI filtering. A sensitivity (true positive rate) of 95% was anticipated given the method used to set the filter ranges. This would equate to 4–5 expected false negative predictions. The lower sensitivity observed (79/91 = 87%) was unexpected as the new RI model was anticipated to outperform its predecessor which also had a sensitivity of approximately 87%.33 The model comes close to the expected sensitivity for compounds with RI values <900 (92% sensitivity, with six false negatives, Figure 2).</p><p>Nevertheless, for the remaining 79 compounds, the RI filter eliminated 58% of false positives from candidate sets acquired from the PubChem database (Figure 3). This was an improvement over the previous model,33 which eliminated approximately 21% of compounds. Likewise, approximately 32% of compounds were removed from candidate sets acquired from HMDB (Figure 3). There were only 22 of the 79 \&quot;unknowns\&quot; where the final number of compounds remaining in the PubChem set after RI filtering was <175. A complete list of these results is given in Tables S4, S5, and S6.</p><!><p>The performance of each predictive fragmenter was evaluated using the method described by Allen et al.23 CSI:FingerID scored 53 correct number one ranks (the known compound ranked as the best match after predictive fragmenter evaluation) when ranked using candidate sets acquired from PubChem database. After the RI filter, the number of correct number one ranks increased to 56. Mass Frontier scored 13 correct number one ranks which improved to 16 after RI filtering. Both CFM-ID and MetFrag had three correct number one ranked sets which increased to seven and six, respectively, after RI filtering. CSI:FingerID and Mass Frontier both scored 66 correct number one ranks with candidate sets acquired from the HMDB database. Upon applications of the RI filter, the number of correct number one rankings increased to 68 and 71, respectively. CFM-ID achieved 62 and 64 correct number one ranks from the HMDB database, while for MetFrag the number of correct number one ranks were 54 and 59 before and after the RI filter. Although these improvements in number one rankings are small, when taken together, they are statistically significant (paired t test; p < 0.001). Thus, our results suggest that RI filtering slightly improved a predictive fragmenter's ability to rank an unknown as the correct number one ranked candidate. Interestingly, 37 of the 53 compounds that were correctly ranked number one by CSI:FingerID were compounds included in the algorithm's original training set.20 This likely causes a significant performance skew in favor of CSI:FingerID and is consistent with the results of the 2016 CASMI contest where ML methods were shown to perform better when unknowns were included in their training sets.26 In contrast, for CFM-ID, only 18 of the 79 compounds studied were in its training set.</p><p>In addition to an evaluation of our RI model based on correct number one rankings, we also compared the results based on overall average rankings. The fragmentation tree filter used in CSI:FingerID eliminated glycocholic acid from its candidate set.20 Thus, Table 1 summarizes overall average rankings of the remaining 78 candidate sets. As seen (Table 1), following RI filtering, the performance of CFM-ID, Mass Frontier, and MetFrag was improved by approximately 2-fold (1.7-, 1.8-, and 1.9-fold, respectively; p < 0.05) when candidate sets acquired from PubChem database were used. This is consistent with the approximate 2-fold reduction in candidates following RI filtering (Figure 3a). There was no significant improvement in the performance of CSI:FingerID upon RI filtering using candidate sets from PubChem, and the RI filter provided no significant improvement for any of the fragmenters when using candidate sets from HMDB. It is important to note that the SD values in Table 1 are relatively large suggesting that the average ranking results are markedly dependent on the structural characteristics of the compounds in each candidate list.</p><p>In our previous work,33 we used multiple orthologous experimental features that can be measured by LC-MS (such as RI) in order to eliminate false positive from candidate sets. The RI model used in this study was based on quantitative structure property relationships (QSPR) using a set of 2D structural descriptors.30 Matching the experimental and predicted RI values (considering the error of the model) enabled us to eliminate a large percentage of interfering false positives (Table 2). To better understand how RI filtering improves predictive fragmenter ranking, let us consider dihydrocapsaicin as an example of where the method performed well (Table 2) using MetFrag and candidates taken from PubChem. Dihydrocapsaicin was ranked as the fourth best match by MetFrag (Table 2). Moreover, for compounds a and b, MetFrag successfully annotated five peaks in the experimental CID spectrum of dihydrocapsaicin, while only detecting four peaks for compounds c and d. Thus, for this example, CID peak matching alone did not result in the correct ranking of dihydrocapsaicin (d) using MetFrag. However, the predicted RI values of compounds a, b, and c were sufficiently different from the experimental value so that they were excluded from the candidate set, permitting the ranking of d to become number one.</p><!><p>CSI:FingerID, developed by Dührkop et al., uses fragmentation trees (FT) to eliminate false positives contained in candidate sets prior to FingerID ranking.20 The theoretical and implementation details of FT are given elsewhere; however, a brief description is provided here.24 The first step in the CSI:FingerID algorithm is to generate all possible candidate molecular formulas of the \&quot;unknown\&quot; (based on precursor mass) and use them to generate fragmentation trees. Each tree is assigned a score based on how well each explains the experimental CID spectrum. Next, candidate molecular formulas having tree scores less than 75% of the top scored tree are excluded. Finally, the remaining formulas are used to eliminate false positives (i.e., candidate structures having molecular formulas other than the >75% scored trees) from the candidate set (e.g., acquired from PubChem). Therefore, we investigated the effect of tree filtering on the performance of CFM-ID, Mass Frontier, and MetFrag (Table 3), in addition to CSI:FingerID. The FT filter eliminated on average approximately 28% (1710 compounds) of false positives in candidate sets acquired from PubChem. It eliminated glycocholic acid from its respective candidate set (false negative). As shown in Table 3, the average performance of CFM-ID, MetFrag, and Mass Frontier improved following FT filtering. However, overall average performance was lower when compared to the RI filter (Table 1). Similar to the RI filter, applying the FT filter on HMDB candidate sets resulted in no increased performance for any of the predictive fragmenters studied.</p><!><p>As mentioned, 12 of the ninety-one compounds were eliminated from their respective candidate sets upon RI filter enrichment. We identify this as a shortcoming of this technique in that only 4–5 false eliminations were anticipated on the basis of the method used to set the filter window. The magnitude of the filter window for each candidate was based on the structural similarity of each candidate to the RI model data and the experimental RI of the \&quot;unknown\&quot;. A compound was defined as \&quot;similar\&quot; (high bit key (HBK)) when three or more compounds in the model data had the same bit key value as the compound being predicted (bit key match). This means that at least three compounds in the model data have the same combination of heteroatomic structure features as the compound being predicted. Compounds with two or fewer data set compounds with a matching bit key were considered \&quot;not similar\&quot; (low bit key (LBK)). The bit key similarity metric is described in detail in Supporting Information SI–4. A filter window of ±2SE (validation standard error) was used when a candidate was similar to model data, and a window of ±3SE was used when a candidate was not similar. Further, an SE value of 48.5 RIU was used when the compound's RI value was 300–850 (data range where the model performed best), and a larger SE value of 62 RIU was used when the compound's RI was 851–1400 (data range where the model preformed least well). This method (termed the \&quot;bit key and range\&quot; approach) and the metrics used to assess similarity are discussed in Supporting Information SI–4. This method was suggested by the independent validation results of a previous study.34</p><p>According to the similarity metrics, only four of the 12 eliminated compounds had low structural similaritiy to the model data (Table S6). This raises the question as to whether using the described approach to set the filter window gives optimal results. In order to evaluate this, additional statistics were calculated using three other approaches for setting the filter window: SE (standard error), bit key, and range. The SE method used a filter window of ±2SE based on the validation SE of the entire data set (52.4 RIU) and used the 95% cutoff values where the \&quot;unknown\&quot; RI was outside the RI range of the reference standards (<RI 300 or >RI 1400). The bit key method used a filter window of ±2SE (52.4 RIU) when a candidate was similar to the model's data and a filter window of ±3SE (78.6 RI Units) when a candidate was not similar. For \&quot;unknown\&quot; RI values <300 or >1400, the bit key method used the 95% cutoff when the candidate was similar and the 98.5% cutoff when the candidate was not similar. The range method used a filter window of ±2SE (48.5 RIU) when the \&quot;unknown\&quot; RI was 300–850 and a filter window of ±2SE (62.0 RIU) when the \&quot;unknown\&quot; RI was 851–1400. The range method used the 95% cutoff values where the \&quot;unknown\&quot; RI was <300 or >1400. Table 4 gives the filter window values used for all 4 approaches along with the sensitivity and specificity for each.</p><p>The highest sensitivity is found for the combined bit key and range approach at 87%. In contrast, the SE and range approaches have sensitivities of only 80%. As expected, the use of wider filter windows when screening some compounds results in a loss of specificity for the bit key and range approach. Both the SE and range approaches removed more false positives. The differences, however, are small at 1–3%, and since the elimination of an unknown from the candidate list results in failure, it is likely necessary to accept the reduced specificity in favor of greater sensitivity.</p><p>An analysis of Figure 2 indicates that, by far, sensitivity is worse where the RI of the unknown falls between 950 and 1120 RIU. Sensitivity is only 38% with false negative classifications for 5 out of the 8 compounds in this range. All 5 false negative compounds are significantly under-predicted. It may be of interest that 4 out the 5 compounds are lipid like with a hydrocarbon tail of at least 15 carbons. All 4 of these lipids are also predicted to have a charge of +1 at the pH of the mobile phase (pH 2.5). It is unclear if these structural characteristics are responsible for the poor prediction. The sample size in this RI range is small, but the poor performance of the mode in this range suggests that this approach would not be recommended in cases where the RI of the unknown falls in the range of 900–1150.</p><p>An analysis of Figure 2 also indicates that there are several close misses in the \&quot;unknowns\&quot; that were eliminated from their respective candidate lists. Two \&quot;unknowns\&quot; are within 20 RIU of being retained and 3 more are within 45. Boundary effects are anticipated when cutoff values are used and cannot be entirely avoided. Even so, given the failure of the method when the unknown is eliminated from its candidate list, some consideration of larger filter windows should be made. Larger filter windows will likely reduce the specifcity of the RI filter, but the results of this study suggest that the reduction could be small. An increase in the width of the filter range by 1/2SE increases the sensitivity to 90% while decreasing the specificity to 53%. Ultimately, the study showed that the RI model improved CID spectral rankings, but the overall level of improvement is modest. Further increases in model accuracy could be made by increasing the number and diversity of the training set to account for the current 3:1 skew toward the lower RI range, including additional relevant descriptors (such as 3D structural information) and QSAR-feature engineering. However, it is important to note that inclusion of 3D descriptors was avoided in the current RI model, as obtaining an accurate global minimum of flexible compounds (e.g., lipids) in a computationally efficient way remains problematic in the field of computational chemistry.39 Our results would suggest that, as candidate lists increase in size, the number of structurally similar compounds also likely increases. Thus, distinguishing among these extremely similar structures, either by CID prediction algorithms or by RI models, becomes increasingly difficult. Clearly, additional orthologous structure identification methods would help in these situations. We are in the process of upgrading our existing Ecom50 model which we hope to use as an additional filter to further enrich acquired candidate sets. From such enrichment, we foresee a potential increase in predictive fragmenter performance due to further reduction in candidate sizes.33 Additionally, the inclusion of an Ecom50 filter would allow us to use broader RI filter windows enabling us to overcome some of the limitations shown here. In addition, future improvements in the sensitivity of NMR and/or IR spectroscopy methods would dramatically enhance our ability to identify unknown compounds in nontargeted metabolomics applications.</p><!><p>We show that RI based enrichment of large candidate sets prior to in silico predictive fragmenter ranking slightly improved the performance of three of the four predictive fragmenters studied. We have developed an approach termed \&quot;bit key and range\&quot; to define the filter windows, but in spite of this, the model eliminated 12 of the 91 \&quot;unknowns\&quot; from their respective candidate sets where only 4–5 eliminations were expected. In the remaining PubChem candidate sets, approximately 58% of the false positives were eliminated on average and overall average rankings were improved 2-fold. Further improvements in the RI model (e.g., 3D descriptors, additional training compounds) would increase its reliability in LC-ESI-MS/MS based database assisted structure elucidation pipelines.</p>&quot;},&quot;article_source&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;PubMed Author Manuscript&quot;}}},{&quot;rowIdx&quot;:6,&quot;cells&quot;:{&quot;title&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Direct Quantification of Cannabinoids and Cannabinoid Glucuronides in Whole Blood by Liquid Chromatography Tandem Mass Spectrometry&quot;},&quot;abstract&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;The first method for quantifying cannabinoids and cannabinoid glucuronides in whole blood by liquid chromatography-tandem mass spectrometry (LC-MS/MS) was developed and validated. Solid-phase extraction followed protein precipitation with acetonitrile. HPLC separation was achieved in 16 min via gradient elution. Electrospray ionization was utilized for cannabinoid detection; both positive (\\xce\\x949-tetrahydrocannabinol [THC], cannabinol [CBN]) and negative (11-hydroxy-THC [11-OH-THC], 11-nor-9-carboxy-THC [THCCOOH], cannabidiol [CBD], THC-glucuronide and THCCOOH glucuronide) polarity were employed with multiple reaction monitoring. Calibration by linear regression analysis utilized deuterium-labeled internal standards and a 1/x2 weighting factor, yielding R2 values > 0.997 for all analytes. Linearity ranged from 0.5\\xe2\\x80\\x9350 \\xce\\xbcg/L (THC-glucuronide), 1.0\\xe2\\x80\\x93100 \\xce\\xbcg/L (THC, 11-OH-THC, THCCOOH, CBD and CBN) and 5.0\\xe2\\x80\\x93250 \\xce\\xbcg/L (THCCOOH-glucuronide). Imprecision was < 10.5% CV, recovery was > 50.5% and bias within \\xc2\\xb1 13.1% of target for all analytes at three concentrations across the linear range. No carryover, endogenous or exogenous interferences were observed. This new analytical method should be useful for quantifying cannabinoids in whole blood and further investigating cannabinoid glucuronides as markers of recent cannabis intake.&quot;},&quot;uuid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;direct_quantification_of_cannabinoids_and_cannabinoid_glucuronides_in_whole_blood_by_liquid_chromato&quot;},&quot;source_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:4081,&quot;string&quot;:&quot;4,081&quot;},&quot;target_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:169,&quot;string&quot;:&quot;169&quot;},&quot;compression&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:24.14792899408284,&quot;string&quot;:&quot;24.147929&quot;},&quot;headers&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Introduction<!>Clinical Samples<!>Instrumentation<!>Reagents<!>Preparation of Standard Solutions<!>Sample Preparation<!>Solid Phase Extraction<!>Liquid Chromatography<!>Mass Spectrometry<!>Data Analysis<!>Validation<!>Results and Discussion<!>Calibration and Validation<!>Application of Method<!>Conclusions&quot;},&quot;sections&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;<p>Cannabis use substantially impacts public safety, as many individuals drive or operate complex equipment soon after self-administration. The National Highway Traffic Safety Administration (NHTSA) reported that in 2007, 8.6% of nighttime drivers tested positive for cannabinoids in blood and/or oral fluid, a rate almost 4 times higher than the percentage of drunk drivers with a blood alcohol concentration ≥0.8 g/L [1]. While finding cannabinoids in blood or oral fluid does not necessarily imply impairment, windows of drug detection in these matrices are often short for occasional or moderate smokers [2–4], increasing impairment probability.</p><p>Δ9-tetrahydrocannabinol (THC) is the primary psychoactive component in cannabis and is metabolized via cytochrome P450 (CYP) 2C9 and 2C19 isozymes to several phase I metabolites, most prominently 11-hydroxy-THC (11-OH-THC) and 11-nor-9-carboxy-THC (THCCOOH) [5–6]. THC and its phase I metabolites also undergo UDP-glucuronosyltransferase-catalyzed phase II metabolism to form cannabinoid glucuronides in vivo [7–9], facilitating excretion. Currently, little is known about cannabinoid glucuronide pharmacological activity or detection windows following cannabis intake, although others hypothesized that these glucuronides could serve as markers of recent cannabis intake due to a shorter half-life in vivo [10–11]. Detection and quantification of these metabolites may provide scientific data permitting researchers, physicians and law enforcement personnel to document recent cannabis intake.</p><p>Analysis of glucuronides by gas chromatography-mass spectrometry (GC-MS) is difficult as chemical derivatization requirements and volatility issues preclude direct detection and quantification. Therefore, analytical procedures for cannabinoids in urine [12–14], blood [15], meconium [16–17] and oral fluid [18] typically include expensive and time-consuming alkaline and/or enzymatic glucuronide hydrolysis to liberate cannabinoids prior to extraction and GC-MS analysis. However, these hydrolyses introduce multiple confounding issues, including, but not limited to poor chromatography [15] and variable hydrolysis efficiencies of the ether- and ester-linked glucuronide species [15, 19–22].</p><p>To circumvent hydrolysis and facilitate direct quantification of phase II cannabinoid metabolites, sensitive liquid chromatography-tandem mass spectrometry (LC-MS/MS) methods are required. Yet few LC-MS/MS methods are available for cannabinoids in whole blood [23–25], likely resulting from higher limits of quantification (LOQ) than typically achieved by GC-MS. Additionally, to date, these published methods do not included glucuronide metabolites. Furthermore, development of glucuronide analytical methods is hampered by a lack of commercially available native and isotopically-labeled cannabinoid glucuronide standards.</p><p>To this end, we developed and validated the first sensitive and specific LC-MS/MS method for simultaneous detection of free and glucuronidated cannabinoids in human whole blood. This method is unique in that THC, THCCOOH and their glucuronides, 11-OH-THC, cannabidiol (CBD) and cannabinol (CBN) are simultaneously extracted and quantified in 16 min. While whole blood is frequently the specimen collected in driving under the influence of drugs (DUID) cases and other investigations, to our knowledge no studies directly investigated whole blood pharmacokinetics following smoked cannabis. Therefore, we will utilize this method to investigate in vitro cannabinoid stability, evaluate cannabinoid glucuronides as markers of recent cannabis intake and determine whole blood cannabinoid pharmacokinetics in order to provide a scientific database for researchers, clinicians and forensic toxicologists interpreting whole blood cannabinoid concentrations.</p><!><p>A healthy cannabis smoker provided written informed consent to participate in a study investigating cannabinoid pharmacokinetics, in vitro cannabinoid stability and novel markers of cannabis intake following a single smoked cannabis dose. The Institutional Review Board of the National Institute on Drug Abuse, National Institutes of Health approved this protocol. Cannabis cigarettes contained 6.8% THC (w/w) or approximately 56 mg THC and were smoked ad libitum over a 10 min period following an overnight stay on a secure residential unit. Whole blood was collected with sodium heparin 0.5 h prior to, 0.25 h after and 1.0 h after the start of cannabis smoking. Blood was transferred to polypropylene storage tubes and stored refrigerated until analysis within 24 h.</p><!><p>All experiments were performed on an AB Sciex 3200 Qtrap triple quadrupole mass spectrometer with a TurboV ESI source (AB Sciex, Foster City, CA). The mass spectrometer was interfaced with a Shimadzu UFLCxr system consisting of two LC-20ADXR pumps, a SIL-20ACXR autosampler, and a CTO-20AC column oven (Shimadzu Corporation, Columbia, MD). Evaporation under nitrogen was completed using a TurboVap LV evaporator from Zymark (Hopkinton, MA).</p><!><p>Standards and deuterated internal standards were purchased from Cerilliant (Round Rock, TX) except for THC-glucuronide that was from ElSohly Laboratories, Inc (Oxford, MS). Ammonium acetate, formic acid and acetonitrile (ACN) were obtained from Sigma-Aldrich (St. Louis, MO). Methanol was from Fisher Scientific (Fair Lawn, NJ). Ammonium hydroxide and glacial acetic acid were from Mallinckrodt Baker (Phillipsburg, NJ). Water was purified in house by an ELGA Purelab Ultra Analytic purifier (Siemens Water Technologies, Lowell, MA). All solvents were HPLC grade or better. 200-mg, 6-mL Bond Elut Plexa solid-phase extraction (SPE) cartridges were utilized for preparing samples (Agilent Technologies, Culver City, CA). Blank human whole blood was evaluated for absence of cannabinoids prior to use.</p><!><p>Individual stock solutions of 1.0 g/L THC, 11-OH-THC, THCCOOH, CBD and CBN, 100 mg/L THCCOOH-glucuronide and 10 mg/L THC-glucuronide were diluted with methanol to prepare calibration solutions. 11-OH-THC-glucuronide and di-glucuronide metabolites are not commercially available. A stock solution containing 10 mg/L of analytes other than THC-glucuronide was prepared in methanol and stored at −20°C. Dilutions of the stock solution (adding in THC-glucuronide) created calibrators at 0.5, 1.0, 2.0, 5.0, 10, 20, 50, 100, and 250μg/L when fortifying 25 μL of standard solution into 500 μL of blank human whole blood.</p><p>Quality control samples were prepared in methanol from different vials than utilized for preparing standards. Low-, medium-, and high-quality control samples were prepared across the linear dynamic range of the assay. Whole blood low-, medium-, and high-quality control target concentrations were: THC-glucuronide 1.5, 4.5, and 45 μg/L; 11-OH-THC, CBD, CBN, THC and THCCOOH 2.5, 7.5, and 75 μg/L; and THCCOOH-glucuronide 7.5, 22.5, and 225 μg/L, respectively. All quality control solutions were stored at −20°C.</p><p>Stock internal standard solution was prepared by diluting 100 mg/L solutions of THC-d3, 11-OH-THC-d3, THCCOOH-d9 and CBD-d3 1:10 with methanol and storing at −20°C. A 1:50 dilution of internal standard stock solution was prepared in methanol and 25 μL of the diluted solution was added to each 500 μL whole blood sample, providing a final internal standard concentration of 10 μg/L. Deuterated CBN, THCCOOH-glucuronide, and THC-glucuronide are not currently commercially available; THC-d3 was utilized for CBN quantification and THCCOOH-d9 for quantification of both glucuronides.</p><!><p>Blank blood (0.5 mL) was pipetted into a 10-mL conical polypropylene tube (Sarstedt, Newton, NC). 25 μL of internal standard and either 25 μL of standard or quality control solution were added. 25 μL blank methanol was added to authentic specimens. Ice-cold ACN (1.5 mL) was added drop-wise while vortexing. Tubes were capped and centrifuged (4000g, 4°C) for 5 min. Supernatants were decanted into clean tubes, 4.5 mL 0.2% NH4OH in de-ionized water (v/v) was added, and samples mixed immediately prior to SPE loading.</p><!><p>Extraction columns were conditioned with 2 mL methanol and 2 mL de-ionized water. Samples were decanted onto conditioned columns and loaded by gravity. Columns were washed with 2 mL 79:20:1 de-ionized water:acetonitrile:glacial acetic acid (v/v/v) and then dried under full vacuum (≥ 30 kPa) for 20 sec. Analytes were eluted with two separate 1.5 mL aliquots of 1% glacial acetic acid in ACN (v/v) under gravity. Vacuum was briefly applied after both aliquots were collected. Eluents were collected in a 10-mL conical polypropylene tube and dried under nitrogen at 42°C in a Zymark TurboVap evaporator. Samples were reconstituted in 150 μL of initial mobile phase (70:30 A:B), vortexed for 15 sec and transferred to 250 μL pulled-point glass inserts in autosampler vials.</p><!><p>Chromatographic separation was performed with an Ultra Biphenyl column (100 × 2.1 mm, 5μm) fitted with an Ultra II Biphenyl guard cartridge (10 × 2.0 mm) (Restek Corp, Malvern PA). The autosampler temperature was 4°C and column oven 40°C throughout analysis. The injection volume was 25 μL. Gradient elution was performed with (A) 10 mM ammonium acetate in water adjusted to pH 6.15 (± 0.05) with formic acid and (B) 15% methanol in acetonitrile (v/v) at a flow rate of 400 μL/min. The initial gradient conditions were 30% B, hold for 30 sec, then increase to 90% B at 6.0 min. 90% B was maintained for 7.5 min, at which time the column was re-equilibrated to 30% B over 0.75 min and held for 1.75 min. HPLC eluent was diverted to waste for the first 2.5 min and the final 9 min of analysis.</p><!><p>Mass spectrometric data were acquired with electrospray ionization (ESI). THC-glucuronide, THCCOOH-glucuronide, THCCOOH, 11-OH-THC and CBD were acquired in negative ionization mode while THC and CBN were acquired in positive ionization mode. MS/MS parameter settings (Table 1, compound-specific optimization) were optimized via direct infusion of individual analytes (500 μg/L in initial mobile phase) at 10 μL/min. Optimized source parameters were as follows: Gas (1) 0.31 MPa, Gas (2) 0.48 MPa, Curtain Gas 0.17 MPa, Source Temperature 650°C. Three acquisition periods were employed, with dwell times of 150 ms for each MS/MS transition in the first, 100 ms for the second and 150 ms for the final period. Unit resolution was used for all experiments.</p><!><p>Linear regression with 1/x2 weighting was employed for all analytes. Peak area ratios of target analytes and their respective internal standards were calculated for each concentration. Analyst Version 1.5 (AB Sciex, Foster City, CA) was utilized for all data collection and processing; statistical calculations were completed with GraphPad Prism 5 for Windows (GraphPad Software, La Jolla, CA).</p><!><p>Specificity, sensitivity, linearity, intra- and inter-batch imprecision, bias, extraction efficiency, matrix effect, carryover, dilution integrity, endogenous and exogenous interferences and analyte stability were investigated to evaluate method integrity. Specificity was based on relative retention time, precursor mass, and fragment ion. Retention times for QC and authentic specimens were required to be within ± 0.2 min of the mean calibrator retention time. Transition peak area ratios for QC and authentic specimens were required to be within ± 20% of the mean peak area ratios for calibrators of each respective analyte.</p><p>Sensitivity was evaluated by determining limits of detection (LOD) and (LOQ). A series of decreasing concentrations of drug-fortified whole blood was analyzed to empirically determine LOD and LOQ. LOD was determined as the concentration with a signal-to-noise ratio of at least 3, transition peak area ratios within 20% of the mean calibrator ratio and acceptable chromatographic retention time and peak shape. LOQ was the lowest concentration with a signal-to-noise ratio of at least 10, acceptable bias and imprecision (within at least 20% of target concentration and relative standard deviation within at least 20%, n = 6), transition peak area ratios within 20% of the mean calibrator ratio and acceptable chromatographic retention time and peak shape.</p><p>Linearity of the method was investigated by calculation of the regression line by the method of least squares and expressed by the squared correlation coefficient (R2). A 1/x2 weighting factor was applied to compensate for heteroscedasticity as evaluated through residuals analysis. Linearity of each analyte was determined with at least five concentration levels, not including the blank matrix, on 4 separate days.</p><p>Imprecision and bias were evaluated at three QC concentrations spanning the dynamic linear range. Intra-batch imprecision (% CV) was evaluated by six determinations per concentration in 1 day. Inter-batch imprecision (% CV) was evaluated for two replicates per concentration on 10 days (n total = 20). One-way ANOVA was employed to evaluate inter-batch repeatability as detailed by Peters and Maurer [26]; p< 0.05 indicated significance. Bias was determined comparing the mean measured concentration of six analyses to the target value and was expressed as the percent of target concentration.</p><p>Extraction efficiency (%) and matrix effect (%) for each analyte also were determined at low, medium, and high control concentrations according to the design proposed by Matuszewski et al [27]. For determination of extraction efficiency, quality control standard solution was added prior to or following SPE. Extraction efficiency, %, was expressed as the mean analyte area of samples with control solution added before SPE (n = 6) divided by the mean analyte area of samples with control solution added after SPE (n = 6). Matrix effect was investigated by comparing analyte peak areas of extracted blank samples that were fortified after SPE versus analyte peak areas of neat samples prepared in initial mobile phase (30:70 A:B) at equivalent concentrations. Matrix effect was computed by dividing the analyte areas of blank samples fortified after SPE by areas of neat samples, expressed as percent.</p><p>Carryover was determined by injecting a negative specimen containing internal standard after a specimen containing two times the upper LOQ. As high concentrations are sometimes observed in blood following cannabis smoking, dilution integrity (1:5 and 1:10) was assessed with three blank blood specimens fortified with high QC solution. Specimens were combined with additional blank whole blood at 1:5 and 1:10 ratios to yield a 500 μL sample. Internal standard was added and specimens were processed as normal.</p><p>Interference from endogenous whole blood compounds was assessed by fortifying aliquots from ten blank whole blood pools with low QC solution and evaluating calculated concentrations. Interferences from over 80 illicit and common therapeutic drugs, metabolites and related compounds were evaluated by adding potential interferents into whole blood aliquots fortified with low QC solution. A compound did not interfere if the low QC quantified within 20% of target and had stable retention times and correct transition ratios. All interferences (Table 2) were tested at 1000 μg/L except for the cannabinoids that were tested at 250 μg/L.</p><p>Hydrolysis of glucuronides during sample processing was evaluated with blank whole blood fortified to 50 μg/L THC-glucuronide and 250 μg/L THCCOOH-glucuronide. Quantifying THC and THCCOOH formed in these hydrolysis controls allowed the calculation of percent hydrolysis for glucuronide metabolites. THC-glucuronide and THCCOOH-glucuronide standards also were investigated individually for presence of THC and THCCOOH, respectively. Individual neat standards were evaporated, reconstituted in mobile phase and quantified against a neat calibration curve to quantify any free cannabinoids present.</p><p>Analyte stability in whole blood (n= 5) was evaluated at three QC concentrations under three conditions: 16 h at room temperature (RT), 72 h at 4 °C and three freeze-cycles at −20 °C (23 h freeze, 1 h thaw at RT). Stability of extracted whole blood samples while in the 4°C autosampler was evaluated over 24 h. Extracted low, medium, and high QC samples (n = 3 at each level) were analyzed immediately after extraction along with calibration standards. Another set of three low, medium, and high QC samples were analyzed 24 h after extraction and subsequent storage in autosampler vials at 4°C. All samples were quantified from the initial calibration curve.</p><!><p>Cannabinoids are the most commonly abused illicit drugs, and cannabinoid medications are utilized for an increasing number of indications, documenting the need for accurate, sensitive and robust cannabinoid quantification. Numerous analytical methods are available to quantify cannabinoids in human whole blood, with and without conjugate hydrolysis [28–32]. However, these methods are limited to parent THC, phase I metabolites and other minor cannabinoids, and fail to consider implications of phase II metabolites. Specifically, factors such as poor hydrolysis efficiency [19,15] and glucuronide instability [33] can introduce unnecessary (and potentially substantial) error into quantitative determinations. Direct identification and quantification of glucuronides negates these issues and can yield novel insight into glucuronide pharmacokinetics and glucuronide in vitro stability while possibly providing an opportunity to utilize cannabinoid glucuronides as markers of recent cannabis intake. The present method sensitively and specifically quantifies these glucuronides directly in addition to typical cannabinoids of interest, including minor cannabinoids CBD and CBN (Figure 1). Thus, this first analytical method for directly analyzing free and glucuronidated cannabinoids in the same whole blood specimen is a significant advancement in the detection and quantification of this important class of compounds.</p><!><p>The method was validated according to the criteria described in the Experimental Section. Table 3 details LOD, LOQ and calibration results for each analyte. LOQs were determined empirically through analysis of decreasing concentrations of drug-fortified whole blood and were 1 μg/L for THC, 11-OH-THC, THCCOOH, CBD and CBN with a 0.5 mL whole blood specimen, exceeding cutoff criteria proposed by Farrell et al [34] and meeting the 1 μg/L THC cutoff typically employed for DUID testing [35]. To extend the dynamic linear range for THCCOOH-glucuronide and minimize the number of re-extractions that might be required due to high THCCOOH-glucuronide concentrations, a 250 μg/L calibrator was included for this analyte. However, extending the linear range required increasing the LOQ from 2.0 to 5.0 μg/L to meet a priori specifications for calibration curve linearity. Linear ranges and R2 values (1/x2 weighting) were acceptable (R2 > 0.990) for all analytes. Linear ranges were THC-glucuronide 0.5–50 μg/L, THCCOOH-glucuronide 5.0–250 μg/L and THC, 11-OH-THC, THCCOOH, CBD and CBN 1.0–100 μg/L (Table 3); these ranges should prove useful for clinical and forensic casework. Calibrators for THC, 11-OH-THC, THCCOOH, CBD and CBN quantified within ± 15% (± 20% for LOQ and glucuronides) when quantified against the entire calibration curve. We expect that our ongoing clinical studies will help establish the utility of glucuronide metabolites for establishing recency of use, generating wider interest in cannabinoid glucuronide testing. Additional interest might prompt proper deuterated internal standards synthesis, allowing more stringent criteria (± 15%) to be applied to all analytes at concentrations > LOQ.</p><p>Deuterium-labeled analogues are not currently commercially available for THCCOOH-glucuronide, THC-glucuronide and CBN. The decision to implement THC-d3 and THCCOOH-d9 for CBN and glucuronides, respectively, was based on similarities in extraction efficiency and matrix effects. This choice was not ideal as differences in efficiencies were present and these can vary depending on the matrix pool; nevertheless, a priori specifications for sensitivity and linearity were met. Other glucuronide metabolites, including morphine-3-glucuronide-d3, buprenorphine-glucuronide and mefenamic acyl-β-D-glucuronide-d3 were investigated as potential internal standards. However, these were either not extracted efficiently (buprenorphine-glucuronide) or not well-retained on our chromatographic system (morphine-3-glucuronide-d3 and mefenamic acyl-β-D-glucuronide-d3). While we attempted to minimize matrix effects through sample preparation including solid phase extraction, some matrix effect remained. The matrix effects for glucuronides and their respective internal standards were not identical, but our approach is the best available at this time. Furthermore, we investigated matrix effect in 10 different whole blood pools demonstrating that low QC quantification remained within ± 20% in all 10 whole blood pools. Despite these efforts, differential matrix effect cannot be excluded, and glucuronide quantification could be affected. It should be noted that deuterated glucuronide analogues are recommended should they become available, as improvements in imprecision, bias and reliability could be realized.</p><p>Bias and imprecision were evaluated at three concentrations across the linear dynamic range of each analyte (Table 4). Intra-batch imprecision (% CV) was less than 7.9% for all analytes at all concentrations (n= 6); inter-batch imprecision (% CV) was less than 10.4% (n= 20). Bias, calculated as the percent of target concentrations at low, mid and high QC concentrations for each analyte, ranged from 93.8% to 113.1% of target concentrations (n= 6). One-way ANOVA yielded statistically significant differences in inter-batch repeatability for several analytes; however, differences were less than 10.4% CV and considered clinically insignificant.</p><p>Extraction efficiency for native and deuterium-labeled analytes ranged from 50.5% to 93.9% (Table 5). Table 5 also displays ion suppression/enhancement produced by matrix effect; positive values indicate ion enhancement and negative values indicate ion suppression. While substantial matrix effects were observed for 11-OH-THC, similar results were obtained for the corresponding deuterated analogue and quantification was not adversely affected.</p><p>Development of an effective sample cleanup that removed matrix interferences while maintaining high extraction efficiency proved to be the greatest challenge during method development. The extraction procedure (reversed-phase polymeric SPE), gentle wash step (20% acetonitrile in water) and polar elution solvent (acetonitrile) yielded high concentrations of phospholipids in extracts as evidenced through a positive precursor ion scan of m/z 184 as detailed by Xia and Jemal [36]. Extending the 90% acetonitrile hold to 7.5 min during the chromatographic gradient provided effective column washing and removal of phospholipids; forgoing this wash yielded substantial increases in ion suppression for subsequent injections. In addition to high phospholipid concentrations, rapid increases in system backpressure were observed during initial method development with a smaller HPLC column particle size (3 μm) and methanolic mobile phase. Backpressure increases were mitigated through replacement of methanol with acetonitrile, increasing column particle size to 5 μm and more frequent replacement of the guard column. Thus, slight decreases in resolution and cost-efficiency were offset by increased column life and a more reliable method.</p><p>Carryover in a negative specimen following a specimen containing twice the upper limit of quantification was assessed. No carryover was observed for any analyte; ion transition ratios were not within 20% of calibrators and any signal present was less than LODs. Common therapeutic and illicit drugs and metabolites at concentrations of 1000 μg/L (cannabinoids 250μg/L) did not interfere with analytes of interest at the low QC concentration. Additionally, ten pools of whole blood were tested for potential endogenous interferences; none were observed in any pool for any analyte. Dilution integrity was maintained up to 10 times dilution with blank whole blood and all analytes quantified within 20% of the theoretical high QC concentration.</p><p>Quantification of THCCOOH and THC formed in glucuronide control samples during extraction was conducted (n= 6 each). Mean (SD) percentages of THCCOOH-glucuronide and THC-glucuronide hydrolysis were 0.6 ± 0.05% and 3.7 ± 0.35%, respectively. However, these are both likely artifacts as neat THCCOOH-glucuronide and THC-glucuronide calibrators were determined to contain 0.5 ± 0.1% THCCOOH and 3.2 ± 0.2% THC, respectively (n = 5 each). While the ester-linked THCCOOH-glucuronide was reported as relatively labile [33], we observed minimal hydrolysis of THCCOOH-glucuronide during extraction. The THC impurity has a minor effect on THC quantification that is less than the analytical error for the method and a low LOQ of 1 μg/L was achieved. To confirm a lack of substantial effect on THC quantifications, samples fortified with only THC at the LOQ (1 μg/L) were quantified against the entire calibration curve containing all analytes. Acceptable quantifications were obtained (± 20%) for these samples, confirming minimal bias resulting from the THC impurity present in the THC-glucuronide standard.</p><p>Stability at 4 °C on the autosampler for 24 h was determined for extracted specimens. All analytes at all concentrations (low, mid and high QC) were stable under these conditions, with mean concentrations differing from samples injected immediately (n= 3) by less than −8.3% (Table 6). For fortified whole blood samples, THCCOOH, 11-OH-THC and both glucuronides were stable under all other conditions tested (three freeze-thaw cycles, 72 h at 4 °C and 16 h at RT). However, losses up to 35.7% were observed for THC after 72 h at 4 °C. Additionally, CBD, CBN and THC demonstrated relative instability under three freeze-thaw cycles, 72 h at 4 °C and 16 h at RT. It should be noted that these losses were observed in fortified samples; losses in authentic specimens may not reflect these findings due to differences in protein binding [15].</p><!><p>Whole blood was collected from a clinical research participant prior to and after smoking a single cannabis cigarette ad libitum. Baseline concentrations were less than LOQ for all cannabinoids except THCCOOH and THCCOOH-glucuronide. 15 and 60 min after the start of smoking, blood was collected and concentrations determined by this new analytical method (Table 7). THC-glucuronide quantified at 0.6 μg/L in the first specimen, demonstrating the necessity for the low LOQ that this method achieved. It should be noted that specimens were analyzed within 24 h of collection, minimizing any potential losses due to analyte degradation. Concentrations suggest THC-glucuronide may serve as possible marker of recent cannabis intake, given that it is detectable following cannabis smoking, albeit at a low concentration. Further research is required to assess detection windows for THC-glucuronide or other minor cannabinoids, such as CBD or CBN, following smoked cannabis.</p><!><p>This method is the first robust, sensitive and specific LC-MS/MS technique for direct detection and quantification of several cannabinoids and two cannabinoid glucuronides in human whole blood, yielding a comprehensive cannabinoid whole blood profile following cannabis intake. The rapid and simple extraction and 16 min analysis are beneficial; however, care should be taken to prevent buildup of phospholipids and other matrix components, leading to increased HPLC backpressure and loss of resolution. This method is utilized for several controlled cannabinoid administration studies and will provide whole blood pharmacokinetic and cannabinoid stability data useful to clinicians and forensic toxicologists interpreting whole blood cannabinoid concentrations often obtained during DUID cases and other investigations. This new analytical method for cannabinoids in whole blood offers advantages in sensitivity and spectrum of cannabinoid analytes included over existing LC-MS/MS and GC-MS assays, and when applied to controlled cannabinoid administration studies, may improve our ability to interpret cannabinoid whole blood results.</p>&quot;},&quot;article_source&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;PubMed Author Manuscript&quot;}}},{&quot;rowIdx&quot;:7,&quot;cells&quot;:{&quot;title&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;A relatively low level of ribosome depurination by mutant forms of ricin toxin A chain can trigger protein synthesis inhibition, cell signaling and apoptosis in mammalian cells&quot;},&quot;abstract&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;The A chain of the plant toxin ricin (RTA) is an N-glycosidase that inhibits protein synthesis by removing a specific adenine from the 28S rRNA. RTA also induces ribotoxic stress, which activates stress-induced cell signaling cascades and apoptosis. However, the mechanistic relationship between depurination, protein synthesis inhibition and apoptosis remains an open question. We previously identified two RTA mutants that suggested partial independence of these processes in a yeast model. The goals of this study were to establish an endogenous RTA expression system in mammalian cells and utilize RTA mutants to examine the relationship between depurination, protein synthesis inhibition, cell signaling and apoptosis in mammalian cells. The non-transformed epithelial cell line MAC-T was transiently transfected with plasmid vectors encoding precursor (pre) or mature forms of wild-type (WT) RTA or mutants. PreRTA was glycosylated indicating that the native signal peptide targeted RTA to the ER in mammalian cells. Mature RTA was not glycosylated and thus served as a control to detect changes in catalytic activity. Both pre- and mature WT RTA induced ribosome depurination, protein synthesis inhibition, activation of cell signaling and apoptosis. Analysis of RTA mutants showed for the first time that depurination can be reduced by 40% in mammalian cells with minimal effects on inhibition of protein synthesis, activation of cell signaling and apoptosis. We further show that protein synthesis inhibition by RTA correlates more linearly with apoptosis than ribosome depurination.&quot;},&quot;uuid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;a_relatively_low_level_of_ribosome_depurination_by_mutant_forms_of_ricin_toxin_a_chain_can_trigger_p&quot;},&quot;source_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:4477,&quot;string&quot;:&quot;4,477&quot;},&quot;target_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:232,&quot;string&quot;:&quot;232&quot;},&quot;compression&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:19.29741379310345,&quot;string&quot;:&quot;19.297414&quot;},&quot;headers&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;1. Introduction<!>2.1. Reagents<!>2.2 Mutant RTA plasmid construction<!>2.3. Cell culture<!>2.4. Transfection<!>2.5. Western immunoblotting<!>2.6. rRNA depurination assays<!>2.7. Protein synthesis inhibition assay<!>2.8. Caspase 3/7 assay<!>2.9. Nucleosome accumulation assay<!>2.10. Statistical analysis<!>3.1. The native signal peptide targets RTA to the ER in mammalian cells<!>3.2. Ribosome depurination and protein synthesis inhibition are reduced relative to WT RTA in cells transfected with RTA mutant constructs<!>3.3. Activation of apoptosis corresponds with protein synthesis inhibition in cells transfected with RTA and RTA mutant constructs<!>4.1. Development of a mammalian expression system to study biological activity of pre- and mature WT RTA and RTA mutants<!>4.2. A substantial reduction in depurination is necessary to prevent inhibition of translation by RTA in mammalian cells<!>4.3. Activation of apoptosis by RTA mutants corresponds better with the extent of protein synthesis inhibition than depurination in mammalian cells<!>Supplemental Figure 1<!>Supplemental Figure 2&quot;},&quot;sections&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;<p>The plant toxin ricin is produced by the castor bean plant Ricinus communis and belongs to a family of ribosome-inactivating proteins (RIPs). Its severe toxicity and wide availability has led to its use as an agent of bioterrorism (Rainey and Young, 2004, Audi et al., 2005). In addition, ricin has been investigated as the active moiety of immunotoxins selectively targeted to cancer cells (Castelletti et al., 2004, Schindler et al., 2011, Zhou et al., 2010). Ricin is composed of two subunits which are encoded by a single gene. The catalytically active A subunit (RTA) depurinates a specific adenine in the α-sarcin/ricin loop (SRL) of the 28S rRNA, resulting in protein synthesis inhibition (Sandvig and van Deurs, 2005, Watson and Spooner, 2006). The B subunit (RTB) binds to cell surface receptors through galactose and N-acetyl galactosamine moieties. After internalization, ricin is transported from early endosomes to the endoplasmic reticulum (ER) via the trans-Golgi network (Spooner and Lord, 2012). While active research is underway to develop effective vaccines (Marsden et al., 2004, Smallshaw et al., 2007, Smallshaw and Vitetta, 2012, Vitetta et al., 2006), monoclonal antibodies (Dai et al., 2011) and small molecule inhibitors (Pang et al., 2011, Pruet et al., 2011, Stechmann et al., 2010, Wahome et al., 2010) to prevent or treat ricin toxicity, there are currently no approved antidotesor therapeutics available.</p><p>The ability to develop effective antidotes against ricin or to use it as the active component of an immunotoxin hinges on a thorough understanding of its biological actions in mammalian cells. In addition to its inhibitory effect on protein synthesis, ricin also induces apoptosis in multiple cell types in vitro and in vivo (Tesh, 2012). Ricin triggers the intrinsic apoptotic pathway as evidenced by release of cytochrome c from mitochondria and subsequent activation of caspase 3, caspase 9 and PARP (Hu et al., 2001, Jetzt et al., 2009, Rao et al., 2005). However, the role of ribosome depurination and protein synthesis inhibition in the apoptotic response remains unclear. The finding that protein synthesis inhibitors that act on the 28S rRNA (i.e. anisomycin and ricin) activate caspase 3 while protein synthesis inhibitors with a different mechanism of action (i.e. diphtheria toxin and cycloheximide) do not suggests that the mode of protein synthesis inhibition may influence the induction of apoptosis (Kageyama et al., 2002). In addition to its inhibitory effect on protein synthesis, ricin also activates the signaling cascades JNK and p38 (Iordanov et al., 1997, Jetzt et al., 2009, Korcheva et al., 2007). The ability to activate these pathways requires a ribosome that is translationally active, indicating that the ribosome actively senses damage to the 28S rRNA. This has been termed the ribotoxic stress response (Iordanov et al., 1997). We have shown that inhibiting the JNK pathway attenuates the ability of RTA to induce apoptosis in MAC-T cells (Jetzt et al., 2009), while the p38 pathway has been shown to play a role in the proinflammatory cytokine response that is observed with ricin toxicity (Higuchi et al., 2003, Korcheva et al., 2007, Lindauer et al., 2010). Although activation of the ribotoxic stress response by ricin clearly triggers signaling cascades involved in apoptosis, the precise role of this response as it relates to protein synthesis inhibition has not been established.</p><p>We previously conducted chemical mutagenesis of the precursor form of RTA (preRTA) which contains a 35-residue leader peptide and isolated mutants based on their inability to induce cell death. Two mutants were identified (P95L/E145K and S215F) that depurinate ribosomes and inhibit protein synthesis similar to WT RTA at 6 h post induction. However, these mutants failed to induce nuclear fragmentation and reactive oxygen species (ROS) generation, which are apoptotic-like characteristics in yeast (Li et al., 2007). These data provide support for the concept that the level of depurination and protein synthesis inhibition may not correspond with cell death. To investigate these relationships in mammalian cells, WT RTA and RTA mutants that caused different levels of depurination in yeast were expressed in MAC-T cells. RTA mutants included the two mentioned above (i.e. P95L/E145K and S215F), G212E, which has very low enzymatic activity and is not toxic in yeast and RTA active site mutants E177K and E177Q. Since the preRTA gene containing the leader sequence would target RTA to the ER, the mature RTA gene lacking the leader sequence was also expressed to examine direct effects of the mutations on catalytic activity in the absence of ER trafficking.</p><!><p>Insulin, gentamicin, D-(+)-glucose, RTA purified from Ricinus communis and phenol red-free Dulbecco's modified Eagle's medium (DMEM) with low glucose were purchased from Sigma-Aldrich (St. Louis, MO). DMEM containing 4.5 g/l D-glucose (i.e. DMEM-H) and penicillin/streptomycin were obtained from Invitrogen (Carlsbad, CA). Fetal bovine serum (FBS) was purchased from Atlanta Biologicals (Lawrenceville, GA). Endoglycosidase H was obtained from New England Biolabs (Ipswich, MA). Recombinant RTA with N-terminal histidine tag expressed in E. coli (NR-853) was obtained through NIAID NIH Biodefense and Emerging Infections (BEI) Research Resources Repository (Manassas, VA). Anti-RTA antibody was produced in rabbits (Covance Research Products; Denver, PA). Antibodies against JNK, p38 and phospho-p38 were purchased from Cell Signaling Technology (Danvers, MA) and phospho-JNK antibody was obtained from Santa Cruz Biotechnology (Santa Cruz, CA). Donkey anti-rabbit and horse anti-mouse horseradish peroxidase-linked secondary antibodies were purchased from GE Healthcare (Piscataway, NJ) and Vector Laboratories (Burlingame, CA), respectively. Peroxidase activity was detected by Pierce ECL Western Blotting Substrate (Thermo Scientific, Rockford, IL) or ECL Prime (GE Healthcare).</p><!><p>The coding sequence of Ricinus communis preRTA containing the 35-residue leader sequence (Piatak et al., 1988) was converted to an optimized codon usage for Bos taurus (Fig. S1) and synthesized. Mature RTA lacking the leader sequence was then constructed from preRTA by PCR cloning (Genscript; Piscataway, NJ). Genes were subcloned into pCAGGS mammalian expression vector (Niwa et al., 1991). Site-directed mutagenesis was performed using the QuikChange Lightning Site-Directed Mutagenesis Kit (Stratagene; La Jolla, CA). The locations of individual mutations are shown in Figure 1. Mutagenesis was confirmed by sequencing. Constructs for transfection were prepared using an EndoFree Plasmid Maxi Kit (Qiagen; Valencia, CA).</p><!><p>The bovine mammary epithelial cell line MAC-T (Huynh et al., 1991) was maintained as previously described (Fleming et al., 2005). For all experiments MAC-T cells were plated in phenol-red free DMEM containing 4.5 g/l D-glucose (DMEM-H), 10% FBS, 20 U/ml penicillin, 20 μg/ml streptomycin, and 50 μg/ml gentamicin (complete media). The human epithelial cell line HEK293T/17 was obtained from ATCC (Manassas, VA). HEK293T/17 cells were maintained and plated for experiments in complete media with phenol red and without gentamicin. All cells were cultured at 37°C in a humidified environment with 5% CO2.</p><!><p>MAC-T cells were plated in complete media at 3.5 × 104 cells/cm2. The next day subconfluent cells were transfected with endotoxin-free plasmid DNA and SuperFect (Qiagen) combined in a 1:5 ratio for 60 × 15 mm dishes and in a 1:10 ratio for 96 well plates. The transfection mixture was prepared in DMEM-H with no additives, vortexed for 10 sec, and incubated at RT for 10 min. Spent media was removed from cells and replaced with fresh complete media and the transfection mixture. After 3 h media was removed and replaced with fresh complete media. HEK293T/17 cells were plated at 5 × 104/cm2. The following day, subconfluent cells were transfected with endotoxin-free plasmid DNA and GeneJuice (EMD Chemicals Inc.; San Diego, CA) according to the manufacturer's protocol. Transfection efficiencies for both cell lines were monitored using pEGFP (Clontech, Mountain View, CA).</p><!><p>MAC-T or HEK293T/17 cells were plated in 60 × 15 mm dishes at 3.5 × 104 cells/cm2 or 5 × 104 cells/cm2, respectively, and transfected the next day as described above. After incubation in serum-containing media for the indicated times cells were washed twice with cold PBS and total cell lysates were collected by scraping into cold cell lysis buffer as previously described (Grill et al., 2002). Cells were then incubated on ice at 4°C for 30 min and spun at 1000 × g for 5 min at 4°C. The supernatant was removed and passed ten times through an 18 gauge needle. The cell lysates were aliquoted and stored at −80°C until use. Protein concentration was determined using the Bio-Rad Protein Assay (Bio-Rad; Hercules, CA). To determine glycosylation status of expressed RTA, cell lysates were denatured at 100°C for 10 min and incubated with Endoglycosidase H (25 U/mg) at 37°C for 1 h. Proteins were separated by SDS-PAGE and transferred to 0.45 μm PVDF (Millipore, Billerica, MA) or 0.2 μm nitrocellulose (Bio-Rad).</p><!><p>MAC-T cells were plated in 60 × 15 mm dishes at 3.5 × 104 cells/cm2 and transfected the following day as described above. After transfection cells were scraped into 1 ml Trizol (Invitrogen) and stored at −80°C. Total RNA was purified using RNeasy columns (Qiagen). rRNA depurination was then analyzed by dual primer extension analysis as described previously (Jetzt et al., 2009).</p><p>rRNA depurination was also analyzed by quantitative reverse transcription PCR (qRT-PCR) as previously described (Melchior and Tolleson, 2010) with minor modifications. Briefly, total RNA (2 μg) was reverse transcribed into cDNA using the High Capacity cDNA Reverse Transcription (RT) kit (Applied Biosystems, Carlsbad, CA). RT reactions (20 μl total volume) were incubated at 25 C for 10 min, 37°C for 2 h, and 85°C for 5 min, and stored at −20°C. Quantitative PCR was performed in a StepOnePlus Real-Time PCR System (Applied Biosystems). Each RT reaction was diluted 1:500 and assayed in triplicate. Total reaction volume (20 μl) contained 5 μl of cDNA, 0.5 μl of each primer, 10 μl Power SYBR Green PCR Master Mix (Applied Biosystems) and 4 μl nuclease-free water. Primers (Sigma) that recognize bovine 28S rRNA and the depurination (dep) fragment were designed as described by Melchior and Tollensen (2010). Primers were 28S-F, 5′-GATGTTGGCTCTTCCTATCATTGT-3′; 28S-R, 5′-CCAGCTCACATGCCCTATTAGTT-3′; dep-F, 5′-TGCCATGGTACCTGCTCAGCA-3′; dep-R, 5′-TCTGAACCTGTGGTTCCACA-3′. Final primer concentrations were 0.25 μM except for dep-R which was 0.75 μM. Cycling conditions were 95°C for 10 min followed by 35 cycles of 95°C for 15 sec and 60°C for 1 min. Melting curves were generated for each sample. Each primer set was validated by constructing standard curves using serial dilutions (1:50 to 1:500,000) of an RT reaction derived from RNA collected from MAC-T cells treated for 6 h with RTA (1 μg/ml). The calculated amplification efficiencies were 95.8% ± 1.3 for the 28S primer pair and 95.4% ± 0.2 for the depurination primer pair (mean ± SD for three independent curves). Treatment effects were determined as fold-change using the ΔΔCT method. The average CT value for reactions containing 28S (endogenous reference) primers was subtracted from the average CT value for reactions containing depurination primers for that same sample (ΔCT ). The same calculation was preformed for a calibrator sample (vector control). The ΔΔCT was then determined by subtracting the ΔCT of a calibrator sample (vector control) from the ΔCT of each test sample.</p><!><p>MAC-T cells were plated in black 96-well plates at 3.5 × 104 cells/cm2 and transfected the following day as described above. Cells were co-transfected with equal amounts of RTA mutant and pEGFP-N1 reporter plasmid. Twenty-one hours after the start of transfection the EGFP fluorescent signal was measured in a plate reader (BioTek) with excitation filter 485/20 and emission filter 530/25. Fluorescence measured in cells co-transfected with GFP and empty vector was considered 100%.</p><!><p>Cells were plated in black 96-well plates at 3.5 × 104 cells/cm2 and transfected the following day as described above. Caspase 3/7 activation was determined using the SensoLyte Homogeneous AMC Caspase 3/7 Assay Kit from AnaSpec (San Jose, CA) as described by the manufacturer.</p><!><p>Cells were plated in 96-well plates at 3.5 × 104 cells /cm2 and transfected the following day as described above. Nucleosome accumulation was determined using the Cell Death Detection ELISA kit (Roche Applied Science, Indianapolis, IN) as described by the manufacturer.</p><!><p>Data were analyzed by one-way ANOVA with Dunnett's Multiple Comparison Test. PreRTA mutants were compared to WT preRTA while mature RTA mutants were compared to WT mature RTA. Differences were considered significant for P < 0.05. Analyses were performed using GraphPad Prism (5.02).</p><!><p>The Ricinus communis gene encoding RTA was modified to utilize preferred codons for the bovine in order to optimize expression of pre- and mature RTA in the MAC-T cell line. The preRTA contained the native 35-residue N-terminal leader peptide followed by the 267-residue mature RTA, while mature RTA did not contain the leader peptide (Fig. 1A). A comparison of the two sequences is shown in Fig. S1. Point mutations were then introduced to create RTA mutants that we had previously shown to decrease cytotoxicity in yeast (Li et al., 2007) (Fig. 1A, B). MAC-T cells were transiently transfected with either pre- or mature forms of WT and mutant RTA. As shown in Fig. 2A, the mature WT protein and all seven mature RTA mutants were detectable by western blotting with an RTA antibody although the mature double mutant P95L/E145K was very faint. All preRTA forms were easily detectable with the exception of E145K and P95L/E145K. The latter two mutants were only visible with much longer exposure times. To determine if higher expression of E145K and the double mutant could be obtained by transfecting a different cell line, HEK293T/17 cells were transfected with the codon-optimized RTA-expressing vectors. As shown in Fig. S2, the pattern of expression for pre- and mature WT, E177K, E145K and the double mutant was similar in HEK293T/17 cells to that observed in MAC-T cells. Pre- and mature E177K were more abundantly expressed than the other RTA forms in both MAC-T and HEK293T/17 cells.</p><p>RTA is N-glycosylated on asparagine residues 10 and 236 (Rutenber et al., 1991) in the ER (Rapak et al., 1997). The first 26 residues of the 35-residue N-terminal leader peptide target RTA to the ER in yeast (Yan et al., 2012) and in plants (Halling et al., 1985). PreRTA ran as a doublet of approximately 30 and 32 kDa which corresponded with the molecular weight of the doublet observed for glycosylated RTA purified from Ricinus communis. Mature RTA ran as a single band at approximately 30 kDa similar to non-glycosylated recombinant RTA (Fig. 2B). To determine if the differences in size were due to glycosylation, lysates collected from cells transfected with pre- or mature WT and E177K were treated with Endoglycosidase H (Endo H) which cleaves the N-linked mannose groups of oligosaccharides (Fig. 2C). The doublet observed in cells expressing the pre-form of RTA or E177K was reduced to one band after Endo H treatment. The single band observed in cells transfected with mature RTA and E177K did not change in size with Endo H treatment. These data demonstrate that the signal sequence of native RTA from the castor bean plant successfully targets the precursor form of RTA to the ER in MAC-T cells. In contrast, expression of the mature form resulted in nonglycosylated RTA, indicating that it is not translocated into the ER.</p><!><p>To determine if endogenously expressed RTA and RTA mutants depurinate rRNA in transfected MAC-T cells, total RNA was collected 19 h after transfection and a dual primer extension assay was performed (Fig. 3A). Since the dual primer extension assay is relatively semi-quantitative we also established a qRT-PCR assay to determine ribosome depurination (Fig. 3B). As shown in Table S1, the results obtained with dual primer extension were matched closely by qRT-PCR . In general, the levels of depurination in cells transfected with preRTA constructs were similar to those in cells transfected with the mature RTA counterpart. All mutants tested depurinated MAC-T rRNA significantly less than their WT control with the exception of mature P95L. The reduction in depurination was the most marked with the active site mutants E177Q and E177K which depurinated less than 40% and 20%, respectively, relative to their WT controls. Ribosome depurination was 60 to 70% of that observed for WT RTA for E145K, P95L/E145K, S215F and G212E.</p><p>A GFP transfection assay was used to determine if protein synthesis inhibition corresponded with changes in ribosome depurination. Overall, the pattern of ribosome depurination observed with the mutated RTA proteins was reflected in the degree of protein synthesis inhibition (Fig. 4). Fluorescence was almost nondetectable in cells transfected with pre- or mature WT RTA relative to cells transfected with plasmid vector alone. Similar results were obtained with P95L, which depurinated ribosomes similarly to WT RTA based on qRT-PCR results. Pre- and mature E177K did not inhibit protein synthesis which corresponded with the very slight degree of depurination observed. Protein synthesis levels observed with pre- and mature E177Q were intermediate between E177K and the other mutants. Surprisingly, a greater level of inhibition of protein synthesis was observed with mature E177Q (60% inhibition of vector controls) compared to preE177Q (40% inhibition of vector controls) even though they showed similar levels of depurination at 19 h after transfection. Protein synthesis was inhibited significantly less relative to WT RTA for preP95L/E145K and for pre- and mature S215F and G212E, however, this still represented an 80 to 90% inhibition of protein synthesis relative to vector controls. Interestingly, the mature RTA mutants tended to have a greater effect on protein synthesis inhibition than their preRTA counterparts. Specifically, preE177Q, P95L/E145K, S215F and G212E inhibited protein synthesis less than their corresponding mature forms and less than WT preRTA.</p><!><p>To investigate if apoptosis was induced in the transfected cells, activation of caspase 3/7 was investigated in MAC-T cells 19 h after transfection (Fig. 5). Caspase activity was induced similarly by pre- and mature WT RTA (2.7 ± 0.2 and 2.8 ± 0.2-fold, respectively; mean ± SE of 9 experiments) relative to vector controls. The active site mutant E177K elicited negligible caspase activation which corresponded with the lack of ribosome depurination and protein synthesis inhibition. In contrast the pre- and mature forms of E145K, G212E and P95L/E145K as well as mature S215F activated caspase activity to the same degree as their respective WT RTA controls. PreS215F activated caspase 3/7 to a lesser extent than WT preRTA, which corresponded with decreases in depurination and protein synthesis inhibition. The E177Q mutant also showed a difference between the pre- and mature forms in terms of caspase activation, with mature E177Q eliciting a greater increase in caspase 3/7 activity relative to preE177Q. Interestingly, both the pre- and mature forms of P95L activated caspase to a greater extent relative to their WT counterparts.</p><p>As a second indicator of apoptosis, a nucleosome accumulation assay was conducted (Fig. 6). Wild-type pre- and mature RTA each increased nucleosome accumulation two-fold over cells transfected with vector alone. While there was more variability with this assay, the results mirrored overall those of the caspase assay. Mature E177Q had more activity than pre E177Q while neither form of E177K showed any activity. The other mutants also showed activity that was similar to what was observed with the caspase assay.</p><p>RTA has been shown to activate JNK and p38 signaling pathways in MAC-T cells (Jetzt et al., 2009), therefore the ability of the different mutants to activate these pathways was examined (Fig. 7). Endogenous expression of both pre- and mature RTA activated JNK and p38 approximately 2-fold relative to vector alone. However, neither JNK nor p38 was activated by expression of pre- or mature E177K. Mutants E145K, G212E, P95L/E145K and S215F activated JNK and p38 signaling similarly to WT RTA controls. This corresponded with full caspase activation with the exception of preS215F, which exhibited decreases in caspase activation. Interestingly, the ability of the active site mutant E177Q to activate JNK and p38 appeared intermediate between E177K and the other mutants.</p><!><p>This is the first report where catalytically active WT RTA and RTA mutants have been expressed in mammalian cells to study the relationship between depurination, protein synthesis inhibition, cell signaling and apoptosis. By optimizing codon usage for the bovine, we were able to increase expression such that we could detect RTA protein by immunoblot analysis in 25 to 50 μg total cell lysates from either bovine or human cells using a specific RTA antibody. Redmann and coworkers (Redmann et al., 2011) recently reported the expression of preRTA and two preRTA mutants in mammalian cells. In their study an N-terminal murine MHC class I heavy chain H2-Kb signal peptide was used to target an enzymatically attenuated RTA variant to the ER membrane. Also, RTA trafficking was the main endpoint studied in that report and the enzymatically attenuated RTA variant was expressed with an HA epitope tag for detection by immunoprecipitation. Using the native signal peptide of ricin, we successfully targeted RTA to the ER membrane in mammalian cells as shown by production of glycosylated protein. The mature form of RTA was not glycosylated, indicating that it was not translocated into the ER. The mature form served as a control to detect changes in catalytic activity, since changes in depurination by mature RTA would be due to catalytic activity and not to trafficking. At the time point measured, levels of depurination were similar between pre- and mature forms of RTA, indicating that preRTA successfully trafficked from the ER to the ribosome.</p><p>E177 has been identified as an invariant amino acid across the RIP family. It lies within the active site of RTA and is known to be a key catalytic residue (Monzingo and Robertus, 1992). In the present study, conversion of E177 to lysine (E177K) produced mutants that exhibited virtually no detectable depurination activity and failed to inhibit protein synthesis, induce apoptosis or activate signaling of the JNK and p38 cascades. This agrees with results obtained with other systems, e.g., this mutation led to total inactivation of the enzyme in an in vitro translation assay (Chaddock and Roberts, 1993) and allowed growth in yeast (Allen et al., 2005, Li et al., 2007). Pre- and mature E177K were expressed at the highest level relative to all other constructs. Interestingly, in a study where amino acids were systematically deleted to determine their role in RTA activity, an inverse correlation was observed between expression in E. coli and enzymatic activity in vitro (Morris and Wool, 1992). This suggests that the very high level of E177K expression may be related to complete lack of biological activity of the mutant protein. These results are also consistent with previous studies in yeast (Li et al., 2007) and suggest that greater enzymatic activity of RTA will lead to higher translation inhibition and reduced protein accumulation.</p><!><p>In yeast, we found that S215F and P95L/E145K induced depurination and inhibited protein synthesis similarly to WT preRTA but did not induce nuclear fragmentation and ROS generation, hallmarks of apoptosis. In addition, G212E had low biological activity and did not affect any of these endpoints (Li et al., 2007). Expression of pre- and mature forms of each of these three mutants reduced depurination levels to 60% of WT RTA control levels in mammalian cells at 19 h after transfection. This lower level of depurination was still sufficient to inhibit protein synthesis by 80 to 90% relative to the vector control. Interestingly, the pre-forms of G212E, S215F, and P95L/E145K did tend to have higher levels of protein synthesis compared to the mature forms or WT preRTA when expressed in mammalian cells, although protein synthesis was still only 15 to 20% of vector control levels. However, they produced full caspase activation, nucleosome accumulation and JNK/p38 signaling. These results indicate that depurination can be reduced by as much as 40% in mammalian cells with minimal effects on protein synthesis inhibition and activation of stress-activated signaling cascades and apoptosis. A substantial reduction in depurination as was observed with pre E177Q may be necessary to prevent protein synthesis inhibition by RTA, possibly due to the high sensitivity of mammalian ribosomes to RTA.</p><!><p>Since E177K exhibited virtually no biological activity, we converted Glu177 to glutamine (E177Q) with the goal of producing an active site mutant that retained some biological activity. This resulted in the expression of pre- and mature RTA proteins that exhibited depurination activity that was approximately 35% of that observed with WT RTA. This agrees with previous studies (Ready et al., 1991) demonstrating that the E177Q mutation decreases enzymatic activity at least 170-fold relative to WT RTA in vitro but less than that of the E177K mutation. Interestingly, while ribosome depurination was similar between the pre- and mature forms of E177Q, the degree of protein synthesis inhibition as well as the induction of apoptosis differed. For the pre form of E177Q, protein synthesis was reduced 32% relative to vector control. However, apoptosis was not induced. In contrast, the mature form of E177Q inhibited protein synthesis 60% relative to vector contols which corresponded with full caspase activation and nucleosome accumulation. The reason for the difference in the degree of protein synthesis inhibition when depurination levels were similar is unknown at this time, but may be related to differences in the rate of depurination (Yan et al., 2012). These results indicate that there may be a threshold level of protein synthesis inhibition that correlates with activation of apoptosis in mammalian cells. Interestingly, the difference in the activation of apoptosis did not appear to correlate with differences in activation of JNK or p38, since JNK activation was statistically less for mature E177Q compared to the pre form and there was no difference between the two for p38 activation. Recently it was reported that ricin mediates IL-1β release from bone-marrow derived macrophages through a scaffolding complex termed the NALP3 inflammasome, which facilitates cleavage of pro-IL-1β to active IL-1β by caspase-1. Using inhibitors for proteosome degradation and the JNK and p38 pathways it was concluded that ricin-mediated translation inhibition caused the disappearance of labile proteins that normally suppress inflammasome formation independent of JNK and p38 kinase activation (Lindauer et al., 2010). Therefore protein synthesis inhibition itself may mediate RTA-induced apoptosis through mechanisms that are independent of stress kinases.</p><p>In summary, we have successfully expressed pre- and mature forms of RTA in mammalian cells that retain biological activity. This was achieved by optimizing the codon usage of RTA for bovine ribosomes. We present evidence that a relatively low level of depurination by RTA can trigger protein synthesis inhibition, apoptosis and stress-activated signaling, indicating that a substantial reduction in depurination is necessary to prevent protein synthesis inhibition by RTA in mammalian cells. Our results show that protein synthesis inhibition correlates more linearly with apoptosis than the level of depurination. Further studies are warranted to identify the specific link between protein synthesis inhibition and apoptosis in RTA-treated cells.</p><!><p>Codon optimized RTA sequence. The original Ricinus communis preRTA sequence is shown on the bottom. The sequence optimized for codon usage by Bos taurus is shown on the top. Nucleotides 7–111 represent the signal sequence.</p><!><p>Expression of RTA and RTA mutants in HEK293T/17 cells. Total cell lysates (50 μg for vector, WT and double; 25 μg for E177K) were collected from HEK 293T/17 cells 21 h after transfection. Membranes were immunoblotted with anti-RTA antibody then stripped and reprobed with HSP60 antibody. V= vector; WT = wild-type; P= pre; M = mature; double = P95L/E145K. Blots are representative of 2 to 3 experiments.</p>&quot;},&quot;article_source&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;PubMed Author Manuscript&quot;}}},{&quot;rowIdx&quot;:8,&quot;cells&quot;:{&quot;title&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Synthesis, Structural Characterization, and Antibacterial Activity of Novel Erbium(III) Complex Con&quot;},&quot;abstract&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;The novel 3D edta-linked heterometallic complex [Sb2Er(edta)2(H2O)4]NO3·4H2O (H4edta = ethylenedia&quot;},&quot;uuid&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;synthesis,_structural_characterization,_and_antibacterial_activity_of_novel_erbium(iii)_complex_con&quot;},&quot;source_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:2533,&quot;string&quot;:&quot;2,533&quot;},&quot;target_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:208,&quot;string&quot;:&quot;208&quot;},&quot;compression&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:12.177884615384615,&quot;string&quot;:&quot;12.177885&quot;},&quot;headers&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;1. Introduction<!>2.1. Materials and Physical Measurements<!>2.2. Synthesis of [Sb2Er(edta)2(H2O)4]&quot;},&quot;sections&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;<p>Much attention is currently focused on the rational design and controlled synthesis of metal-org&quot;},&quot;article_source&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;PubMed Open Access&quot;}}},{&quot;rowIdx&quot;:9,&quot;cells&quot;:{&quot;title&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Physico-chemical properties and catalytic activity of the sol-gel prepared Ce-ion doped LaMnO3 pero&quot;},&quot;abstract&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Ce-doped LaMno 3 perovskite ceramics (La 1−x Ce x Mno 3 ) were synthesized by sol-gel based copre&quot;},&quot;uuid&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;physico-chemical_properties_and_catalytic_activity_of_the_sol-gel_prepared_ce-ion_doped_lamno3_pero&quot;},&quot;source_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:5446,&quot;string&quot;:&quot;5,446&quot;},&quot;target_toks&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:213,&quot;string&quot;:&quot;213&quot;},&quot;compression&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:25.568075117370892,&quot;string&quot;:&quot;25.568075&quot;},&quot;headers&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;<!>Catalyst characterization.<!>Specific activity<!>Results and Discussion<!>Redox properties (tpR/&quot;},&quot;sections&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;<p>Presently, perovskite-based materials are gaining immense popularity in the field of material sc&quot;},&quot;article_source&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Scientific Reports - Nature&quot;}}}],&quot;truncated&quot;:true},&quot;paginationData&quot;:{&quot;pageIndex&quot;:0,&quot;numItemsPerPage&quot;:100,&quot;numTotalItems&quot;:115956,&quot;offset&quot;:0,&quot;length&quot;:100}},&quot;jwt&quot;:&quot;eyJhbGciOiJFZERTQSJ9.eyJyZWFkIjp0cnVlLCJwZXJtaXNzaW9ucyI6eyJyZXBvLmNvbnRlbnQucmVhZCI6dHJ1ZX0sImlhdCI6MTc0MjkyMzEwMSwic3ViIjoiL2RhdGFzZXRzL2dyaWZmaW4vQ2hlbVN1bSIsImV4cCI6MTc0MjkyNjcwMSwiaXNzIjoiaHR0cHM6Ly9odWdnaW5nZmFjZS5jbyJ9.oLUPfE4h-9UDlgAxT6uVqCWV4yWWKcm4EwZAooNrHrciAGQ-wfxQWM9VVqX2jG96jTGbWW5D6OcmPvvTgFpPAA&quot;,&quot;displayUrls&quot;:true},&quot;dataset&quot;:&quot;griffin/ChemSum&quot;,&quot;isGated&quot;:false,&quot;isPrivate&quot;:false,&quot;hasParquetFormat&quot;:true,&quot;author&quot;:{&quot;_id&quot;:&quot;611693254ef9fdfbf45dc4d6&quot;,&quot;avatarUrl&quot;:&quot;https://cdn-avatars.huggingface.co/v1/production/uploads/1628869402886-noauth.jpeg&quot;,&quot;fullname&quot;:&quot;Griffin Adams&quot;,&quot;name&quot;:&quot;griffin&quot;,&quot;type&quot;:&quot;user&quot;,&quot;isPro&quot;:false,&quot;isHf&quot;:false,&quot;isHfAdmin&quot;:false,&quot;isMod&quot;:false,&quot;followerCount&quot;:26},&quot;compact&quot;:true}"><div class="flex flex-col overflow-hidden shadow-xs mx-auto mb-10 rounded-lg border pt-2  px-2.5"><div class="mb-2 flex flex-wrap items-center gap-2"><div class="mr-auto flex items-center"><svg class="mr-1 flex-none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 12 12"><path fill="currentColor" d="M2.5 2h7a1 1 0 0 1 1 1v6a1 1 0 0 1-1 1h-7a1 1 0 0 1-1-1V3a1 1 0 0 1 1-1Zm0 2v2h3V4h-3Zm4 0v2h3V4h-3Zm-4 3v2h3V7h-3Zm4 0v2h3V7h-3Z"></path></svg>
	<div class="whitespace-nowrap font-semibold">Dataset Viewer</div>
	</div>
				<a href="/datasets/griffin/ChemSum/tree/refs%2Fconvert%2Fparquet/default" class="group mr-1 text-xs text-gray-400 max-sm:hidden"><svg class="text-[.6rem] mr-1 inline -translate-y-px" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path fill="currentColor" d="M12 10H6.78A11 11 0 0 1 27 16h2A13 13 0 0 0 6 7.68V4H4v8h8zm8 12h5.22A11 11 0 0 1 5 16H3a13 13 0 0 0 23 8.32V28h2v-8h-8z"></path></svg>
						<span class="underline decoration-gray-300 group-hover:decoration-gray-400 dark:decoration-gray-500 dark:group-hover:decoration-gray-300">Auto-converted</span> to Parquet
					</a>
				<button class="btn shadow-xs flex cursor-pointer items-center rounded-sm border px-1 py-0.5 text-xs font-normal text-gray-700 hover:text-gray-800 hover:shadow-inner dark:hover:text-gray-200"><svg class="mr-1.5" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32" style="transform: rotate(360deg);"><path d="M31 16l-7 7l-1.41-1.41L28.17 16l-5.58-5.59L24 9l7 7z" fill="currentColor"></path><path d="M1 16l7-7l1.41 1.41L3.83 16l5.58 5.59L8 23l-7-7z" fill="currentColor"></path><path d="M12.419 25.484L17.639 6l1.932.518L14.35 26z" fill="currentColor"></path></svg>API</button>
					<button class="btn shadow-xs flex cursor-pointer items-center rounded-sm border px-1 py-0.5 text-xs font-normal text-gray-700 hover:text-gray-800 hover:shadow-inner dark:hover:text-gray-200"><svg class="mr-1.5" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" fill="currentColor" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 12 12"><path d="M9.80603 2.86737H3.56107C3.37704 2.86737 3.20055 2.94048 3.07042 3.0706C2.94029 3.20073 2.86719 3.37723 2.86719 3.56126V9.80622C2.86719 9.99025 2.94029 10.1667 3.07042 10.2969C3.20055 10.427 3.37704 10.5001 3.56107 10.5001H9.80603C9.99006 10.5001 10.1666 10.427 10.2967 10.2969C10.4268 10.1667 10.4999 9.99025 10.4999 9.80622V3.56126C10.4999 3.37723 10.4268 3.20073 10.2967 3.0706C10.1666 2.94048 9.99006 2.86737 9.80603 2.86737Z" fill="currentColor" fill-opacity="0.3"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M2.40942 1.66191C2.05175 1.66191 1.7618 1.95186 1.7618 2.30953V6.76191H1.43799V2.30953C1.43799 1.77303 1.87291 1.3381 2.40942 1.3381H6.45704V1.66191H2.40942Z" fill="currentColor"></path></svg>Embed</button>

					<button class="bg-linear-to-b shadow-xs flex items-center gap-1.5 rounded-full border from-white to-red-100/90 px-2 py-0.5 text-xs font-medium text-[#2D3648] transition-shadow hover:shadow-inner dark:from-gray-900 dark:to-red-800/30 dark:text-gray-100 dark:hover:shadow-inner dark:hover:shadow-red-800/30" ><svg class="h-3.5 w-3.5 text-red-500" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 12 12"><path fill="currentColor" d="M2.5 2h7a1 1 0 0 1 1 1v6a1 1 0 0 1-1 1h-7a1 1 0 0 1-1-1V3a1 1 0 0 1 1-1Zm0 2v2h3V4h-3Zm4 0v2h3V4h-3Zm-4 3v2h3V7h-3Zm4 0v2h3V7h-3Z"></path></svg>
						<span>Data Studio</span></button></div>
		<div class="flex flex-1 flex-col overflow-hidden -mx-2.5"><div class="flex flex-1 flex-col overflow-hidden"><div class="flex min-h-0 flex-1"><div class="flex flex-1 flex-col overflow-hidden"><div class="md:-mx-2.5 flex min-w-0 flex-wrap border-t"><div class="flex min-w-0 flex-1 flex-wrap"><div class="grid flex-1 grid-cols-1 overflow-hidden text-sm md:grid-cols-2 md:place-content-center sm:mx-2.5"><label class="relative block flex-1 px-3 py-2 hover:bg-gray-50 dark:border-gray-850 dark:hover:bg-gray-950 md:border-r md:border-r-0 hidden" title="default"><span class="text-gray-500">Subset (1)</span>
			<div class="flex items-center whitespace-nowrap"><span class="truncate">default</span>
				<span class="mx-2 text-gray-500">·</span>
					<span class="text-gray-500">119k rows</span>
				<svg class="ml-auto min-w-6 pl-2" width="1em" height="1em" viewBox="0 0 12 7" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M1 1L6 6L11 1" stroke="currentColor"></path></svg></div>
			<select class="absolute inset-0 z-10 w-full cursor-pointer border-0 bg-white text-base opacity-0"><optgroup label="Subset (1)"><option value="default" selected>default (119k rows)</option></optgroup></select></label>
		<label class="relative block flex-1 px-3 py-2 hover:bg-gray-50 dark:border-gray-850 dark:hover:bg-gray-900 md:border-r md:border-r" title="train"><div class="text-gray-500">Split (3)</div>
				<div class="flex items-center overflow-hidden whitespace-nowrap"><span class="truncate">train</span>
					<span class="mx-2 text-gray-500">·</span>
						<span class="text-gray-500">116k rows</span>
					<svg class="ml-auto min-w-6 pl-2" width="1em" height="1em" viewBox="0 0 12 7" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M1 1L6 6L11 1" stroke="currentColor"></path></svg></div>
				<select class="absolute inset-0 z-10 w-full cursor-pointer border-0 bg-white text-base opacity-0"><optgroup label="Split (3)"><option value="train" selected>train (116k rows)</option><option value="validation" >validation (1k rows)</option><option value="test" >test (2k rows)</option></optgroup></select></label></div></div>
								</div>

							<div class="flex min-h-0 flex-1 flex-col ">
	<div class="bg-linear-to-r text-smd relative flex items-center dark:border-gray-900 dark:bg-gray-950 false border-t [&amp;:has(:focus)]:from-gray-50 [&amp;:has(:focus)]:to-transparent [&amp;:has(:focus)]:to-20% dark:[&amp;:has(:focus)]:from-gray-900"><form class="flex-1"><svg class="absolute left-3 top-1/2 transform -translate-y-1/2 pointer-events-none text-gray-400" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M30 28.59L22.45 21A11 11 0 1 0 21 22.45L28.59 30zM5 14a9 9 0 1 1 9 9a9 9 0 0 1-9-9z" fill="currentColor"></path></svg>
				<input disabled class="outline-hidden h-9 w-full border-none bg-transparent px-1 pl-9 pr-3 placeholder:text-gray-400 " placeholder="Search this dataset" dir="auto"></form>
			<div class="flex items-center gap-2 px-2 py-1"><button type="button" class="hover:bg-yellow-200/70 flex items-center gap-1 rounded-md border border-yellow-200 bg-yellow-100 pl-0.5 pr-1 text-[.8rem] leading-normal text-gray-700 dark:border-orange-500/25 dark:bg-orange-500/20 dark:text-gray-300 dark:hover:brightness-110 hidden"><div class="rounded-sm bg-yellow-300 px-1 font-mono text-[.7rem] font-bold text-black dark:bg-yellow-700 dark:text-gray-200">SQL
	</div>
	Console
</button></div></div>


<div class="flex flex-1 flex-col overflow-hidden min-h-64 border-t">
		

<div class="max-h-96 relative overflow-auto"><table class="w-full table-auto rounded-lg font-mono text-xs text-gray-900"><thead class="shadow-xs sticky left-0 right-0 top-0 z-1 bg-white align-top"><tr class="space-y-54 h-full min-w-fit divide-x border-b text-left"><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">title
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="0" width="11.2" height="30" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="13.2" y="14.82583276682529" width="11.2" height="15.17416723317471" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="26.4" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="39.599999999999994" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="52.8" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="66" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="79.19999999999999" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="92.39999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="105.6" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="118.8" y="25" width="11.2" height="5" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="12.2" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="25.4" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="38.599999999999994" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="51.8" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="65" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="78.19999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="91.39999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="104.6" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="117.8" y="0" width="13.2" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">0</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">1.22k</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">abstract
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="0" width="11.2" height="30" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="13.2" y="23.904755620493" width="11.2" height="6.095244379506999" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="26.4" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="39.599999999999994" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="52.8" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="66" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="79.19999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="92.39999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="105.6" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="118.8" y="25" width="11.2" height="5" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="12.2" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="25.4" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="38.599999999999994" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="51.8" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="65" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="78.19999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="91.39999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="104.6" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="117.8" y="0" width="13.2" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">239</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">18.2k</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">uuid
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="13.2" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="26.4" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="39.599999999999994" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="52.8" y="24.509787524004214" width="11.2" height="5.490212475995787" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="66" y="23.160100353094222" width="11.2" height="6.83989964690578" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="79.19999999999999" y="21.80236015610481" width="11.2" height="8.197639843895187" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="92.39999999999999" y="20.58836647463297" width="11.2" height="9.411633525367032" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="105.6" y="20.090689462925106" width="11.2" height="9.909310537074894" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="118.8" y="0" width="11.2" height="30" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="12.2" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="25.4" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="38.599999999999994" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="51.8" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="65" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="78.19999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="91.39999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="104.6" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="117.8" y="0" width="13.2" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">0</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">100</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">source_toks
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>int64</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="0" y="0" width="11.2" height="30" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="13.2" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="26.4" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="39.599999999999994" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="52.8" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="66" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="79.19999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="92.39999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="105.6" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="118.8" y="25" width="11.2" height="5" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="12.2" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="25.4" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="38.599999999999994" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="51.8" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="65" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="78.19999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="91.39999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="104.6" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="117.8" y="0" width="13.2" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">510</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">159k</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">target_toks
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>int64</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="0" y="0" width="11.2" height="30" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="13.2" y="23.482528472478222" width="11.2" height="6.517471527521779" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="26.4" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="39.599999999999994" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="52.8" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="66" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="79.19999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="92.39999999999999" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="105.6" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="118.8" y="25" width="11.2" height="5" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="12.2" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="25.4" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="38.599999999999994" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="51.8" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="65" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="78.19999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="91.39999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="104.6" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="117.8" y="0" width="13.2" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">50</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">2.23k</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">compression
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>float64</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="0" y="0" width="11.2" height="30" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="13.2" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="26.4" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="39.599999999999994" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="52.8" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="66" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="79.19999999999999" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="92.39999999999999" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="105.6" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" rx="2" x="118.8" y="25" width="11.2" height="5" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="12.2" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="25.4" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="38.599999999999994" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="51.8" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="65" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="78.19999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="91.39999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="104.6" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="117.8" y="0" width="13.2" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">1.12</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">845</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">headers
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="0" width="11.2" height="30" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="13.2" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="26.4" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="39.599999999999994" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="52.8" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="66" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="79.19999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="92.39999999999999" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="105.6" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="118.8" y="25" width="11.2" height="5" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="12.2" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="25.4" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="38.599999999999994" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="51.8" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="65" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="78.19999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="91.39999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="104.6" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="117.8" y="0" width="13.2" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">0</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">35.5k</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">sections
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="0" width="11.2" height="30" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="13.2" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="26.4" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="39.599999999999994" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="52.8" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="66" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="79.19999999999999" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="92.39999999999999" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="105.6" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="118.8" y="25" width="11.2" height="5" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="12.2" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="25.4" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="38.599999999999994" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="51.8" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="65" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="78.19999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="91.39999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="104.6" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="117.8" y="0" width="13.2" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">3.32k</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">585k</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">article_source
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">classes</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><defs><clipPath id="rounded-bar"><rect x="0" y="0" width="130" height="8" rx="4"></rect></clipPath><pattern id="hatching" patternUnits="userSpaceOnUse" patternTransform="rotate(-45)" height="1" width="5"><line y1="0" class="stroke-gray-400 dark:stroke-gray-500/80" stroke-width="3" y2="1" x1="2" x2="2"></line></pattern><pattern id="hatching-faded" patternUnits="userSpaceOnUse" patternTransform="rotate(-45)" height="1" width="5"><line y1="0" class="stroke-gray-100 dark:stroke-gray-500/20" stroke-width="3" y2="1" x1="2" x2="2"></line></pattern></defs><g height="8" style="transform: translateY(20px)" clip-path="url(#rounded-bar)"><g style="transform: scaleX(1.0153846153846153) translateX(-1px)"><g><rect class="fill-indigo-500 dark:fill-indigo-600/80" x="1" y="0" width="61.040291144917035" height="8" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" x="64.04029114491703" y="0" width="30.316137155472767" height="8" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" x="96.35642830038981" y="0" width="11.346872951809306" height="8" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" x="109.70330125219911" y="0" width="8.172996653903205" height="8" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" x="119.87629790610231" y="0" width="5.473351961088689" height="8" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" x="127.349649867191" y="0" width="0.005674566214771026" height="8" fill-opacity="1"></rect></g></g></g><g style="transform: scaleX(1.0153846153846153) translateX(-1px)"><g><rect class="fill-white cursor-pointer" x="0" y="0" width="63.040291144917035" height="28" fill-opacity="0"></rect><rect class="fill-white cursor-pointer" x="63.040291144917035" y="0" width="32.31613715547277" height="28" fill-opacity="0"></rect><rect class="fill-white cursor-pointer" x="95.35642830038981" y="0" width="13.346872951809306" height="28" fill-opacity="0"></rect><rect class="fill-white cursor-pointer" x="108.70330125219911" y="0" width="10.172996653903205" height="28" fill-opacity="0"></rect><rect class="fill-white cursor-pointer" x="118.87629790610231" y="0" width="7.473351961088689" height="28" fill-opacity="0"></rect><rect class="fill-white cursor-pointer" x="126.349649867191" y="0" width="2.005674566214771" height="28" fill-opacity="0"></rect><rect class="fill-white cursor-pointer" x="128.35532443340577" y="0" width="0.6188554279209355" height="28" fill-opacity="0"></rect><rect class="fill-white cursor-pointer" x="128.97417986132672" y="0" width="0.588585325468281" height="28" fill-opacity="0"></rect><rect class="fill-white cursor-pointer" x="129.562765186795" y="0" width="0.43723481320500884" height="28" fill-opacity="0"></rect></g></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 max-w-full overflow-hidden text-ellipsis whitespace-nowrap">9
				values</div></div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th></tr></thead>
			<tbody class="h-16 overflow-scroll"><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="0"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Solvent-dependent conductance decay constants in single cluster junctions</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Single-molecule conductance measurements have focused primarily on organic molecular systems. Here, we carry out scanning tunneling microscope-based break-junction measurements on a series of metal chalcogenide Co 6 Se 8 clusters capped with conducting ligands of varying lengths. We compare these measurements with those of individual free ligands and find that the conductance of these clusters and the free ligands have different decay constants with increasing ligand length. We also show, through measurements in two different solvents, 1-bromonaphthalene and 1,2,4-trichlorobenzene, that the conductance decay of the clusters depends on the solvent environment. We discuss several mechanisms to explain our observations.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">solvent-dependent_conductance_decay_constants_in_single_cluster_junctions</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">2,400</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">99</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">24.242424</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Introduction&lt;!>Results and discussion&lt;!>Conclusions</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">&lt;p>Controlling charge transport through molecular junctions is critical to the realization of nanoscale electronic devices. 1,2 While numerous organic molecules have been studied as connecting wires for single-molecule junction studies, [3][4][5][6][7][8][9][10] very little is known about the effect of metal complexes in these types of junctions. [11][12][13][14] We recently reported that we could incorporate metal chalcogenide molecular clusters in single-molecule electrical circuits. 15 In this study, in order to determine how transport through such systems depends on molecular length, we connect these same clusters to conducting ligands of varying lengths. We have found that the inclusion of the cluster in the molecular circuit reduces the effect of ligand length on conductance decay with apparent molecular size. Moreover, we have found that the decay constant is impacted greatly by changing the solvent from 1,2,4-trichlorobenzene (TCB) to 1bromonaphthalene (BrN). Specically, the decay constant of the cluster is 0.04 ÅÀ1 in BrN, while it is 0.12 ÅÀ1 in TCB. We consider two possible mechanisms to explain these remarkable observations. Our work demonstrates, for the rst time, a molecular system where the tunneling decay constant can be modied by altering the environment around the molecule.&lt;/p>&lt;!>&lt;p>The single cluster circuits that we have designed, assembled, and studied consist of an atomically dened Co 6 Se 8 molecular cluster 16,17 (Fig. 1a) wired between nanoscale electrodes. The wiring is formed from bifunctional, conjugated ligands (Fig. 1b) that bind specically and directionally to the electrode and to the cluster. We employ an atomically dened segment of polyacetylene 18 that has an arylphosphine group on one terminus that coordinates to a cobalt atom on the clusters and an arylthiomethyl group on the other terminus that attaches to the Au electrode. 19,20 The mono-, di-, and triene ligands are L1, L2, and L3 and the corresponding clusters are 1, 2, and 3, respectively. Fig. 1c shows the molecular structure of 1 as determined by single crystal X-ray diffraction (SCXRD). 15 We measured the conductance of both the individual molecular clusters (1-3) and the free conducting ligands (L1-L3) using a scanning tunneling microscope-based break-junction (STM-BJ) technique. 21 In this technique, an Au STM tip and substrate are repeatedly brought into and out of contact to form and break Au-Au point contacts in solutions of the target compounds. During this process, a bias voltage is applied across the junction while current is measured in order to determine conductance (G ¼ I/V) of the junction. The measurements are repeated thousands of times, and the data is analyzed to reveal statistically signicant results. The data is processed by compiling thousands of individual conductance traces into one-dimensional, logarithmically-binned conductance histograms. 22 We further generate two-dimensional (2D) histograms of the conductance versus displacement by aligning each conductance trace aer the point contact ruptures (at a conductance of 0.5 G 0 ) and overlaying all conductance traces.&lt;/p>&lt;p>In order to characterize transport through the molecular clusters and the free ligands, we measured the conductance of 1-3 and L1-L3 in two different solvents, BrN and TCB. These solvents were chosen taking into consideration the solubility of both the ligand and the cluster systems as well as for their varied affinity to gold electrodes. 23 Fig. 2a and b contain the onedimensional histograms for the measurements in BrN, and Fig. 2d and e show the same for the measurements in TCB. The 2D histograms for 1 and L1 in each solvent are insets in the respective gures. The 2D histograms show a signicant difference in length of the molecular feature for 1 and for L1. Moreover, the cluster junction lengths measured from the 2D histograms correlate with the molecular lengths of the cluster with the ligands fully extended (measured in BrN: 9 Å and 21 Å, and expected from SCXRD: 13 Å and 32 Å, for L1 and 1 respectively). Despite the additional complexity of the cluster system, we conclude that we are indeed probing transport through Au-ligand-cluster-ligand-Au junctions based on this large difference in the observed lengths. Furthermore, the histograms in Fig. 2a and d show shoulders, with an increasing prominence for the longer-ligand systems. Comparing the conductance of these shoulders with the ligand conductance in Fig. 2b and e, we attribute these shoulders to free ligands, that is, ligands that have detached from the clusters.&lt;/p>&lt;p>We t the peaks of all conductance histograms for both solvents with a Gaussian function and plot the peak conductance values versus the number of C]C units or "enes" in each molecule (Fig. 2c and f). In both solvents, and for both free ligand and cluster, we observe that the conductance decreases exponentially with increasing molecular length following the relationship G $ e Àbn , where n is the number of "ene" units in the backbone and b is the decay constant. We report the decay constant per Angstrom using a length of 2.48 Å per "ene" unit. The decay constant for the free ligand series is essentially independent of the solvent (b ¼ 0.15 ÅÀ1 in TCB and 0.17 ÅÀ1 BrN).&lt;/p>&lt;p>The unexpected result is the factor of 3 difference in the decay constants of the cluster series in different solvents as can be seen comparing Fig. 2c and f. In TCB, the b of the cluster system is 0.12 ÅÀ1 , and in BrN it is 0.04 ÅÀ1 . We note that the difference between the decay constant of the ligand and that of the cluster is greater in BrN than in TCB. Furthermore, the absolute values of the conductance of the cluster series are signicantly higher when measured in BrN than in TCB, with the conductance of 3 being almost an order of magnitude higher in BrN compared to TCB. Such a solvent-induced effect on the conductance has been observed in other systems, and this has been attributed to the solvent's ability to modulate the electrode work function. 23,24 These ndings are summarized: regardless of solvent the effect of C]C chain-length on conductance is less pronounced in 1-3 than in L1-L3. Furthermore, the conductance and the decay of L1-L3 are essentially insensitive to the choice of solvent, while the solvent signicantly inuences those of 1-3.&lt;/p>&lt;p>To understand these results we consider several possible mechanisms of charge transport through these junctions. Charge transport can occur via a coherent off-resonance process through an orbital on the ligand-cluster-ligand assembly that is coupled to both electrodes. In that situation, the conductance depends on at least two related factors: (1) the energy of this conducting orbital relative to the metal E F , and (2) the coupling between this orbital and both electrodes. 25 As the length of the molecule increases, the HOMO-LUMO gap narrows, and if conductance were just related to energy level alignment, one would naively expect conductance to actually increase. However, transport through the junction is also related to how well the conducting orbital overlaps with the leads, and since the orbital is more delocalized over a longer conjugated molecule, this overlap decreases with increasing length. The conductance thus typically decays exponentially with increasing molecular length. Specically, as the conjugated backbone gets longer, the molecular orbital is delocalized over a longer molecule, and since the orbital is normalized, a smaller fraction of its amplitude resides on the sulfur atoms; therefore the coupling between the molecule and the electrodes decreases.&lt;/p>&lt;p>If we assume that the conducting orbitals of the cluster and of the ligand for a given length are similar in both character and energy, we can develop a simple tight-binding model of the molecular junctions. The objective of this model is not to reproduce the experimental data but to examine and illustrate how the additional electronic structure of the cluster impacts transport through the system. Our tight-binding model is schematically presented in the insets of Fig. 3a and b for the ligand and the cluster respectively. For the conducting ligands, we assign a single energy level, 3, for each unit, and allow nearest neighbors to be coupled by d. The terminal units are coupled to the Au electrodes using an imaginary self-energy, iG/2. We apply a similar model for the cluster, adding an additional energy level, E 0 , between two ligands and coupling this site to its nearest neighbor ligand states with s. We compute the transmission functions for these model systems using a Green's function approach (see the ESI for a detailed description †). 25,26 Sample computed transmission functions are shown in Fig. 3a and b using the same values for 3, d and G for the ligand and the cluster series. The transmission functions display resonances at energy values corresponding to the molecular orbitals of the system where the probability of an electron being transmitted through the system is unity. The transmission function for L1 contains one resonance at energy 3, while longer ligands have resonances equal to the number of sites in the corresponding model. As the length of the molecule increases, the frontier resonance moves closer to E F but also narrows, which is a consequence of the frontier orbital being delocalized over a longer molecular backbone. Upon comparing the transmission functions for the ligands with those of the clusters, we see that the clusters contain resonances that are closer to E F than their ligand counterparts, but with narrower full widths at half max (i.e. they are more poorly coupled to the leads). This observation leads to a lower transmission at E F ; more importantly, it also leads to a conductance that is more sensitive to the exact location of the E F .&lt;/p>&lt;p>In Fig. 3c, we show the conductances that are determined from the tight-binding model for each molecule versus the number of ligand levels in the molecule (using the same model parameter values for both series). From the t to these values, it is clear that the predicted decay constants are essentially the same for the ligand and cluster series. We use one set of E 0 and s values to calculate the representative transmission/conductance functions shown in Fig. 3a and b. Regardless of what value is assigned to E 0 and s, we nd that this model predicts very similar decay constants for the two systems (Fig. 3d).&lt;/p>&lt;p>Our tight-binding calculations suggest that the addition of a cluster level E 0 between two ligands cannot explain the observed change in b. In other words, the ligand and cluster series should have the same b values, unless the energy alignment of the cluster resonance is altered relative to the electrode Fermi level in this model. We have three sets of observations that are consistent with a change in E F : (1) b of the cluster in BrN is signicantly lower than in TCB, (2) b values measured in both solvents are almost the same for the ligand series, and (3) b of the cluster is lower than that of the ligand in both solvents. The steeper transmission curves of the cluster series in Fig. 3 indicate that the resonance energies are closer to E F . Within this coherent transport model, we can see that a small change in E F will result in a large shi in b for the cluster relative to the ligand. For instance, changing E F by À0.5 eV shis the b value to 0.1 ÅÀ1 for the clusters while a similar change in E F for the ligand changes b to 0.3 ÅÀ1 . These results, when viewed in light of the known ability of solvent-binding to produce changes in E F , 23 point to BrN shiing E F closer to resonance relative to TCB. This effect is compounded by the sensitivity of the metal. The free ligand and the cluster have very different characteristics (e.g., size, steric hindrance, redox behaviour, dielectric constant polarizability and binding ability) that will result in different shis in E F .&lt;/p>&lt;p>We also consider a hopping mechanism for charge transfer, a process generally mediated by an activation-controlled reaction (e.g., a thermally induced conformational change or an electron transfer reaction). 27,28 We rst rule out the possibility that such a conformational change can occur within the ligand. 3 We also refute the process involving a direct through-space charge transfer from the electrode to an unoccupied molecular level on the cluster through a resonant transfer process. 14 In this picture, the cluster does not have to be chemically attached to the electrodes to form a conducting junction and the charge transfer efficiency depends on the core-electrode spacing. We discount this mechanism based on a previously published study in which we demonstrated that our clusters form molecular junctions by bonding their terminal thiomethyl groups to the Au electrodes. 15 By varying the substitution pattern or removing the aurophilic functionality, we can modulate or completely shut down the conductivity of these molecular junctions, suggesting that there is an orbital pathway for the transport of charge in these cluster systems. These ndings refute the idea of direct through-space charge transfer mediated by an orbital localized on the core.&lt;/p>&lt;p>We are le with a hopping mechanism in which the charge tunnels from the source electrode across the ligand to the cluster core and then transfers to the drain electrode through a second coherent tunneling process. Such a transport process requires that the cluster can reversibly change its oxidation state with each charge transfer. Since the applied bias in these measurements is not small ($0.5 V) and the cluster core Co 6 Se 8 is redox active, it is plausible that such a hopping process is at play. In this case, the activation energy arises from the charge transfer process reorganization energy, which can be strongly inuenced by the solvent. This mechanism is consistent with our observation that b changes with solvent. Within our experimental constraints, it is therefore difficult to conclusively establish which process (off-resonance tunneling or hopping) is at work in our single cluster junction system.&lt;/p>&lt;!>&lt;p>In summary, we measured charge transport through molecular clusters with ligands of different lengths and showed that the conductance decay depends on the solvent used for these measurements. Our results illustrate a novel effect that allows the environment to alter the conductance decay constants. This study opens up the possibility to carry out conductance measurements in which clusters can be controllably gated by changing the environment. 20 While the conducting ligands alone are limited to a one-dimensional system, the threedimensional architecture of the metal chalcogenide cluster allows us to envision novel electronic devices where a molecular cluster is contacted by electrodes at multiple locations.&lt;/p></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Royal Society of Chemistry (RSC)</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="1"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Characterization of Frequency-chirped Dynamic Nuclear Polarization in Rotating Solids</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Continuous wave (CW) dynamic nuclear polarization (DNP) is used with magic angle spinning (MAS) to enhance the typically poor sensitivity of nuclear magnetic resonance (NMR) by orders of magnitude. In a recent publication we show that further enhancement is obtained by using a frequency-agile gyrotron to chirp incident microwave frequency through the electron resonance frequency during DNP transfer. Here we characterize the effect of chirped MAS DNP by investigating the sweep time, sweep width, center-frequency, and electron Rabi frequency of the chirps. We show the advantages of chirped DNP with a tritylnitroxide biradical, and a lack of improvement with chirped DNP using AMUPol, a nitroxide biradical. Frequency-chirped DNP on a model system of urea in a cryoprotecting matrix yields an enhancement of 142, 21% greater than that obtained with CW DNP. We then go beyond this model system and apply chirped DNP to intact human cells. In human Jurkat cells, frequency-chirped DNP improves enhancement by 24% over CW DNP. The characterization of the chirped DNP effect reveals instrument limitations on sweep time and sweep width, promising even greater increases in sensitivity with further technology development. These improvements in gyrotron technology, frequency-agile methods, and incell applications are expected to play a significant role in the advancement of MAS DNP.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">characterization_of_frequency-chirped_dynamic_nuclear_polarization_in_rotating_solids</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">2,758</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">208</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">13.259615</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Introduction&lt;!>NMR Experiments&lt;!>7&lt;!>Sample Preparation&lt;!>Results and Discussion&lt;!>Frequency-chirped DNP on a Model System&lt;!>Frequency-chirped DNP in Intact Jurkat Cells&lt;!>Power Dependence of CW and Frequency-chirped DNP&lt;!>Characterization of Frequency-chirped DNP&lt;!>Conclusion</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">&lt;p>Dynamic nuclear polarization (DNP) is commonly used to improve the inherent insensitivity of nuclear magnetic resonance (NMR) spectroscopy [1][2][3][4][5][6][7][8][9][10][11][12][13]. Typically, only continuous wave (CW) microwave methods have been employed with magic angle spinning (MAS) DNP. The solid effect and the cross effect are the primary DNP mechanisms used in moderate magnetic field strengths of 5-14 Tesla (T) [14][15][16][17][18]. While CW approaches can significantly increase NMR sensitivity, they have limitations. Except in certain model systems [6,19,20], the solid effect and cross effect are inefficient at room temperature due to short longitudinal electron relaxation times. To perform CW DNP, samples are commonly cooled to &lt;120 K, which adds complexity not only to the instrumentation, but also often leads to a loss of spectral resolution [14,21]. Arrested molecular motion at these temperatures can cause substantial line broadening in most samples [3,[21][22][23]. The cross effect and solid effect also exhibit worse performance at higher magnetic field, with cross effect efficiency decreasing as 1/B0 and that of solid effect as 1/B0 2 [15,24,25]. Therefore new mechanisms will be required for efficient DNP at magnetic fields of 28 T and higher.&lt;/p>&lt;p>Frequency-chirped DNP techniques, such as the frequency-swept integrated solid effect (FS-ISE) [15,26], nuclear orientation via electron spin locking (NOVEL) [27,28], and timeoptimized pulsed (TOP) DNP [29] show promise to perform well both at high magnetic field and room temperature. For instance, ISE yields DNP enhancements of ~150 at room temperature and is predicted to be unaffected by the strength of the external magnetic field [15]. However, these experiments have been performed without MAS and at magnetic fields &lt;3 T [15,27,29], primarily due to the difficulty of implementing MAS with the microwave resonators required to generate considerable electron nutation frequencies. Frequency-swept DNP at higher magnetic fields has also been shown to improve DNP performance [30,31], but has only recently been implemented with MAS [32,33]. MAS improves the sensitivity and resolution of solid-state NMR [34][35][36][37][38] by partially averaging anisotropic interactions of the magnetic resonance Hamiltonian, and is a crucial aspect of applying DNP to systems of interest.&lt;/p>&lt;p>Here we characterize the behavior of frequency-chirped DNP experiments performed with MAS, expanding on our recent work [32]. We optimize frequency chirps from a custom-built frequency-agile high-power gyrotron [39] to produce large gains in intensity beyond those obtained with CW DNP. In addition to measuring its performance on a model system, we conduct optimized chirped experiments on intact human Jurkat cells to demonstrate frequency-chirped DNP in a biologically complex environment.&lt;/p>&lt;!>&lt;p>MAS DNP NMR experiments were performed using a custom-built DNP spectrometer at a magnetic field of 7.1584 T [41]. 13 C and 1 H Larmor frequencies were 75.4937 MHz and 300.1790 MHz, respectively. A CPMAS, rotor synchronized, Hahn echo sequence with TPPM decoupling [42] was used for all experiments (Fig. 1a). The initial magnetization of 1 H and 13 C spins was destroyed using a saturation train. 1 H and 13 C pulses were performed with nutation frequencies of 77 kHz and 100 kHz, respectively. The Hartmann-Hahn matching condition (γB1) for 1 H and 13 C was 30 kHz. Frequency chirps were applied over the DNP polarization period (τpol), and CW microwaves were employed over the rest of the experiment. The spinning frequency was 4.5 kHz for all experiments, and the sample temperature was 90 K. Typical polarization times (τpol) for optimized spectra were 5-times the T1 of the sample in the absence of microwaves, in order to remove contamination of the data by differences in the nuclear T1 and the T1DNP.&lt;/p>&lt;p>Microwaves were generated using a frequency-agile gyrotron, whose output frequency was adjusted by varying the electron acceleration potential at the electron gun anode. An arbitrary waveform generator (AWG) integrated into the NMR spectrometer (Redstone, Tecmag Inc. Houston, TX) was used to generate a waveform, which ramped the output frequency of the gyrotron in a linear fashion through 197.670 GHz, the frequency of maximum DNP enhancement of the TEMTriPol-1 radical [39]. The frequency chirps were a triangular waveform, which was repeated over the entire polarization period. For frequency chirp optimization the incident microwave power, the center DNP microwave frequency, and the sweep width and sweep time of the individual chirps were varied. The center frequency of the sweeps was varied by changing the voltage at the gyrotron anode with the AWG amplified by a high-voltage amplifier (TREK, Inc.&lt;/p>&lt;p>Lockport, NY). The sweep width corresponded to the frequency range of one sweep/chirp (either up or down) in MHz, and sweep time was the time to complete a sweep/chirp. Microwave power was attenuated from full power by inserting copper foil with slits cut in it into a gap in the waveguide to partially pass the microwave beam. The optimal power of 7 W incident on the sample was used for most experiments, which provided an estimated electron Rabi frequency of 0.43 MHz [43].&lt;/p>&lt;!>&lt;p>The 13 C carbonyl resonance was fit using DMfit [44] to determine resulting enhancement increases.&lt;/p>&lt;p>For all optimization spectra, the magnitude of the Hahn echo was used to calculate the percent increase in intensity. All experiments were repeated four times to acquire adequate error values for the measurements.&lt;/p>&lt;!>&lt;p>Experiments were performed on 4 M [U-13 C, 15 N] urea mixed with 5 mM TEMTriPol-1 or 5 mM AMUPol in a cryoprotecting matrix consisting of 60% d8 glycerol, 30% D2O, and 10% H2O by volume. Intact Jurkat cells (ATCC, Manassas, VA) were cultured in [U- 13 C, 98%; U-15 N, 98%] BioExpress-6000 mammalian cell growth medium (Cambridge Isotope Laboratories, Tewksbury, MA) at a concentration of 3 × 10 6 cells/mL in a six-well plate at 37°C and 5% CO2 for 48 hr. 3.6 × 10 7 cells were collected, spun at 170 g for 5 min, washed with 1× phosphate-buffered saline (PBS), and spun again at 170 g for 5 min to remove extracellular NMR labels (g = 9.8 m/s 2 ). 40 µL of 20 mM TEMTriPol-1 in 1×PBS with 10% DMSO was added to a cell pellet containing 36 million Jurkat cells. This suspension was centrifuged directly into the 3.2 mm zirconia rotor at 800 g for 30 s and immediately frozen in liquid nitrogen as detailed in our previous work [4].&lt;/p>&lt;!>&lt;p>Frequency-chirped DNP refers to a change in the microwave frequency or intensity throughout the course of an experiment. The frequency-chirped DNP pulse sequence is shown in Fig. 1A.&lt;/p>&lt;p>Frequency chirps (represented by the rainbow gradient) are applied over the DNP polarization period and the resulting NMR signal is detected through a cross polarization (CP) Hahn echo sequence. We emphasize that microwave frequency chirps result in better manipulation of the electron spin polarization, yet the active DNP mechanism is still the cross effect. Selection of appropriate radicals for frequency-chirped DNP is crucial due to drastic differences in electron spin g-anisotropy and relaxation properties. In our previous demonstrations of electron decoupling using chirped microwave pulses with MAS, we employed trityl rather than nitroxide radicals [3,22]. Those successes led us to explore the use of trityl-nitroxide biradicals, with the rational that the narrow trityl resonance would be easier to manipulate and the tethered nitroxide would provide greater DNP enhancements through the cross effect mechanism. TEMTriPol-1 is such a biradical, consisting of a Finland trityl radical covalently linked to a 4-amino TEMPO radical, which is used for cross effect DNP [13,45]. TEMTriPol-1 improves cross effect efficiency at high magnetic fields. Where other biradicals, such as AMUPol, depolarize nuclear spins at 100 K in the absence of microwave irradiation, TEMTriPol-1 preserves nuclear polarization [5,46].&lt;/p>&lt;!>&lt;p>CW DNP CPMAS experiments were performed at various microwave frequencies to record a 1 H DNP enhancement profile with TEMTriPol-1 [40]. The enhancement profile shows the trityl resonance frequency as the optimal frequency for CW DNP enhancement. This will be the target for the center of the frequency chirps. In a 7.1584 T magnetic field, the microwave frequency for maximum CW DNP enhancement was 197.670 GHz (Fig. 1B).&lt;/p>&lt;p>Experiments were performed to determine the effect of frequency-chirped microwave pulses during the polarization period of MAS DNP (Fig. 2). For comparison, cross effect DNP experiments were performed with CW microwave irradiation. CW DNP experiments on a model system of urea with TEMTriPol-1 resulted in an enhancement of 118 (Fig. 2, red). Enhancements herein are defined as the NMR signal intensity recorded with DNP compared to that without DNP [46]. For frequency-chirped DNP experiments, the microwave frequency was linearly chirped with a triangular waveform over 197.670 GHz, with a 28 µs sweep time and a 120 MHz sweep width. These optimized chirps yielded a 21% increase over CW DNP and an enhancement of 142 (Fig. 2, blue). Polarization times of 53 s (5×T1DNP, Fig. S1) were used to ensure that >99% of the polarization had built up for both experiments, allowing for direct comparison of the CW and frequency-chirped experiments. To determine the necessity of a narrow-line radical, such as trityl, for frequency-chirped DNP, experiments were performed on a sample containing the nitroxide-nitroxide biradical, AMUPol.&lt;/p>&lt;p>The frequency chirps were centered at 197.674 GHz (maximum with 1 H-enhancement for AMUPol) the previously optimized sweep time of 28 µs and sweep width of 120 MHz were used.&lt;/p>&lt;p>Frequency chirps over the polarization period resulted in a decrease in signal intensity of 3% compared to CW DNP (Fig. 3). These frequency chirps do not yield the same improved electron spin control over the nitroxide biradical, AMUPol, as they do over TEMTriPol-1. This implies that a narrow-line radical is required for implementation of frequency-chirped DNP.&lt;/p>&lt;!>&lt;p>The performance of frequency-chirped DNP was then examined within intact human Jurkat cells (Fig. 4). Frequency chirps improved the NMR signal by 24%, yielding an enhancement of 6 (versus 4.8 for CW DNP). These results display the application of frequency-chirped DNP to more complex samples of biological interest.&lt;/p>&lt;!>&lt;p>To determine the dependence of CW and frequency-chirped enhancement on microwave power, CPMAS experiments were performed with varying microwave attenuation on the TEMTriPol-1/urea sample (Fig. 5). For frequency-chirped DNP the optimized triangle waveform (28 µs sweep time and 120 MHz sweep width) was repeated over a polarization time of 20 s. 35 W of microwave power incident on the sample (Rabi frequency of 0.95 MHz) produced a 123% increase in signal with frequency-chirped DNP compared to CW, yielding enhancements of 17 and 8, respectively (Fig. 5a, b). We note that such high microwave powers place the cross effect in the oversaturated regime, leading to less overall enhancement. 7 W of microwave power resulted in the highest sensitivity and an improvement of 25% with frequency-chirped DNP compared to CW. Higher microwave power yielded greater improvements with frequency-chirped DNP over CW DNP, but the overall signal intensity obtained was suboptimal due to saturation of the cross effect [47].&lt;/p>&lt;!>&lt;p>The effects of sweep time, sweep width, and center frequency on the improvement with frequencychirped DNP over CW irradiation are shown in Fig. 6. For this dependence the polarization time was 20 s; the sweep width was held constant at 80 MHz, the incident microwave power at 7 W, and the center frequency at 197.670 GHz. Shorter sweep times increased the sensitivity to a greater extent than longer sweep times, with the greatest improvement over CW (15%) occurring with a 20 μs sweep time (Fig. 6a). Sweep times below 20 μs were not achievable with the current microwave frequency agility circuit, as the frequency output waveform became distorted. A sweep time of 150 µs resulted in only a 1% improvement in signal intensity over CW. We suspect that at longer sweep times electron spin saturation is lost through relaxation mechanisms.&lt;/p>&lt;p>The dependence of frequency-chirped DNP sensitivity on the sweep width of the frequency chirps is shown in Fig. 6b. For this dependence the polarization time was 20 s; the sweep time was held constant at 28 μs, the incident microwave power at 7 W, and the center frequency at 197.670 GHz.&lt;/p>&lt;p>The improvement from the frequency chirps increased as the sweep width increased. A 120 MHz sweep width resulted in an improvement of 21%, while the signal intensity decreased by 1% with a sweep width of 10 MHz. Due to instrument limitations, sweep widths greater than 120 MHz could not be attained. This width is roughly that of the base of the trityl lineshape in the enhancement profile (Fig. 1b). We previously reported a similar optimal sweep width in electron decoupling experiments involving the Finland trityl radical [3]. Larger sweep widths provide microwave irradiation that is resonant with a greater number of trityl electron spins, enabling better electron spin control and improving the efficiency of frequency-chirped DNP.&lt;/p>&lt;p>During characterization it is important to consider multiple points on the enhancement profile. Fig. 6d provides a clear picture of the effect of frequency chirping, whereas Fig. 6c shows the potential for misinformation. The choice of irradiation frequency can lead to suspiciously high improvements due to difference in positive and negative enhancement regions between CW and frequency-chirped DNP. The CW enhancement profile shows maximum positive and negative enhancements at 197.670 GHz and 197.850 GHz, respectively (Fig. 6d). Frequency chirping at microwave frequencies lower than 197.750 GHz (positive enhancement), yielded greater enhancements than CW (Fig. 6d). However, at frequencies greater than 197.750 GHz (negative enhancement), the frequency-chirped DNP provided lower signal intensity than CW DNP. Note that at this point we have simply demonstrated the methodology of performing frequencychirped DNP experiments with TEMTriPol-1. To compare the sensitivity of the experiments with TEMTriPol-1 and AMUPol, we can divide the signal-to-noise from each experiment by the square root of the polarization time for the respective experiments. In doing so, we obtain a sensitivity of 79 with AMUPol (Fig. 3) and 73 with TEMTriPol-1 (Fig. 2). Thus, while the sensitivity of the experiments performed on each radical are similar at this stage, advances in instrumentation that enable greater sweep times and sweep widths will make frequency-chirped DNP experiments with TEMTriPol-1 more sensitive than AMUPol, and thus more feasible for sensitivity-demanding, multidimensional experiments.&lt;/p>&lt;!>&lt;p>To date, frequency-chirped DNP experiments, such as FS-ISE, NOVEL, and TOP DNP, have been largely restricted to static samples due to the difficulties of housing microwave resonators with the instrumentation required for magic angle spinning (MAS). Here, we have characterized the optimal experimental conditions for frequency-chirped MAS DNP. At a magnetic field of 7 T and with 7 W of microwave power, frequency-chirped microwaves over the polarization period improved DNP enhancements by 21%. Greater microwave powers resulted in up to 123% improvements with frequency-chirped DNP, but saturation of the cross effect resulted in less overall signal intensity.&lt;/p>&lt;p>These optimized frequency-chirped experiments were applied to a more biologically complex sample: intact Jurkat cells. This resulted in an improvement in signal intensity of 24% over CW DNP. Characterization of the parameters of frequency-chirped DNP revealed areas for further improvements to elicit even greater sensitivity. More powerful gyrotrons with larger frequency bandwidths, and gating mechanisms for chirps can be developed to increase sweep widths and shorten the sweep times, thus improving electron spin control. To take full advantage of frequencychirped DNP at high power and high electron Rabi frequencies, duty cycling of the microwaves can be implemented to reduce dielectric heating [29]. We expect optimization of the waveform, with respect to both intensity and phase, to result in improved frequency-chirped DNP MAS performance. Future studies could analyze the effect of the spinning frequency on the enhancement achieved by frequency chirped DNP over CW DNP. Both the solid effect and cross effect are driven by interactions between the spin system, the microwave field, and the spinning rotor.&lt;/p>&lt;p>Understanding these effects will prove crucial in the future development of DNP, as MAS frequencies and magnetic fields are pushed to ever higher values.&lt;/p>&lt;p>New radicals composed of tethered broad and narrow line radicals are currently being investigated with useful electronic properties such as long longitudinal relaxation times. Longer relaxation times will afford even more electron spin control with frequency-chirped DNP. Although the precise mechanism governing the improvement in sensitivity will require further investigation, it is possible that it is governed by an adiabatic process. As such, future experiments could focus on maintaining a constant sweep rate by simultaneously varying the sweep time and sweep width in an inverse manner. This could prove important, as adiabatic processes often show a remarkable resilience to microwave inhomogeneities and frequency offsets arising from difference in molecular orientation and conformations in a solid sample. These techniques can be paired with other advances in instrumentation such as higher power microwave sources and microwave lenses for improved microwave intensity and high frequency MAS for 1 H detected spectra in future experiments. These could allow for the implementation of pulsed DNP mechanisms such as electron-nuclear cross polarization at high magnetic fields in the foreseeable future.&lt;/p></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ChemRxiv</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="2"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Engineering forward genetics into cultured cancer cells for chemical target identification</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Summary Target identification for biologically active small molecules remains a major barrier for drug discovery. Cancer cells exhibiting defective DNA mismatch repair (dMMR) have been used as a forward genetics system to uncover compound targets. However, this approach has been limited by the dearth of cancer cell lines that harbor naturally arising dMMR. Here, we establish a platform for forward genetic screening using CRISPR-Cas9 to engineer dMMR into mammalian cells. We demonstrate the utility of this approach to identify mechanisms of drug action in mouse and human cancer cell lines using in vitro selections against three cellular toxins. In each screen, compound-resistant alleles emerged in drug-resistant clones, supporting the notion that engineered dMMR enables forward genetic screening in mammalian cells.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">engineering_forward_genetics_into_cultured_cancer_cells_for_chemical_target_identification</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">3,370</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">120</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">28.083333</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Introduction&lt;!>MSH2 deletion in cancer cell lines induces MSI and hypermutation&lt;!>Engineered dMMR enables forward genetic screening in mammalian cancer cells&lt;!>Discussion&lt;!>LEAD CONTACT AND MATERIALS AVAILABILITY&lt;!>EXPERIMENTAL MODEL AND SUBJECT DETAILS&lt;!>sgRNAs, surveyor, CRISPR KO of MSH2, MLH1.&lt;!>Western-blotting&lt;!>Compounds&lt;!>Dose response curves&lt;!>Selection of resistant clones.&lt;!>Crystal Violet experiment&lt;!>Whole Exome Sequencing Analysis&lt;!>Identification of recurrently mutated genes in MLN4924-resistant clones&lt;!>QUANTIFICATION ANS STATISTICAL ANALYSIS&lt;!>DATA AND CODE AVAILABILITY&lt;!></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">&lt;p>The identification of therapeutic targets in cancer can be divided into two complementary approaches. Target-based approaches use tumor sequencing or laboratory-based genetic studies to identify cancer driver genes followed by screening for small molecules that impair the protein products of cancer driver genes. Phenotypic high-throughput small molecule screens (HTS) first identify drug-like chemicals that selectively impair the growth of cancer cells. The latter approach has been limited by the technical challenge of identifying the direct protein targets of small molecules exhibiting anti-cancer effects. One strategy to identify chemical targets is through the identification of compound resistant alleles that impair compound-target interaction. Cancer cells with dMMR exhibit mutation rates increased as much as 50-750-fold compared to cells with intact MMR (Glaab and Tindall, 1997). As a result, these cells are predisposed to develop resistance through the acquisition of compound resistant alleles. Following selection, these compound resistant alleles can be identified by transcriptome or whole exome sequencing of multiple independent drug-resistant clones (Han et al., 2017, Han et al., 2016, Wacker et al., 2012). However, it is not clear whether this approach can be applied to other cancer cell lines, particularly those established from cancers without MMR deficiency or those harboring low mutation frequencies, such as pediatric malignancies. To date, a single cancer cell line, HCT116, derived from a human colorectal cancer harboring a naturally-arising mutation in the MMR protein MLH1, has been used successfully for these studies. Here, we sought to determine if somatic deletion of the MMR protein MSH2 using CRISPR-Cas9 could be used to expand the repertoire of dMMR cancer cell lines for use in forward genetic screens.&lt;/p>&lt;!>&lt;p>We sought to develop cancer cell lines with engineered loss of MSH2 in order to establish forward genetics screening in other cancer types, including Ewing sarcoma (EWS) and small cell lung cancer (SCLC). Ewing sarcoma is a pediatric malignancy without approved targeted therapies, and the overall mutational burden in these tumors is extremely low, similar to other pediatric malignancies (Pishas and Lessnick, 2016, Yu et al., 2017, Tirode et al., 2014, Brohl et al., 2014, Crompton et al., 2014). We used CRISPR-Cas9 to generate MSH2-null Ewing sarcoma A673 cells (Figure 1A, Methods). We examined multiple microsatellite regions by PCR and capillary electrophoresis for evidence of microsatellite instability (MSI), a hallmark of dMMR cells, in two independently generated MSH2-null A673 lines (A673-M1, A673-M8). MSI was observed in three out of five loci analyzed in both A673-M1 and A673-M8 clones compared to the parental, MMR-proficient, A673 cell line (Fig. 1B,C)(Boland et al., 1998).&lt;/p>&lt;p>We next tested whether MMR deletion would facilitate MSI in tumor cell lines derived from genetically-engineered mouse cancer models (GEMMs), which also exhibit very low mutation frequencies compared to many human malignancies (McFadden et al., 2014, McFadden et al., 2016). Msh2 was deleted in a cell line generated from a small cell lung cancer (SCLC) GEMM initiated by loss of the p53, retinoblastoma (Rb), and p130 tumor suppressors (319-N1 cell line) (Figure S1A)(Schaffer et al., 2010). Msh2-null murine SCLC (mSCLC) cells also exhibited evidence of MSI, with two of three microsatellite loci exhibiting instability (Figure S1B, C)(Bacher et al., 2005).&lt;/p>&lt;p>Following observation of MSI in MMR-edited human and murine cells, we sought direct evidence of an increased mutation frequency using whole exome sequencing (WES). Two MSH2-null A673 clones (M1 and M8) and three independent MSH2-wild-type parental A673 clones (CL1, CL2, and CL3) were subjected to WES. Because a reference germline genome was not available for A673 cells, we used the parental A673 cell line as the germline reference (Methods). We observed an increased frequency of somatic mutations in M1 (n=221) and M8 (n=198) clones, compared to MSH2-wild-type parental clones (n= 77, 74, 64) (Table S1).&lt;/p>&lt;p>We performed WES on multiple mSCLC tumor cell lines generated from independent tumors isolated from the same mouse in order to accurately compare mutation frequencies between MMR-proficient and dMMR cell lines (Figure S2A). Common variants between two independent primary tumor cell lines (319-T1, T2) represented germline variants, whereas variants unique to individual cell lines represented somatic mutations. The Msh2-wild type cell line 319-N1 exhibited 25 high-confidence somatic mutations (Table S2). In contrast, 319-N1-Cl31 cells with engineered Msh2 loss exhibited 352 somatic mutations. Therefore, elevated somatic mutation frequencies were observed following MMR editing in both human and murine cancer cell lines. (Figure 1D, Figure S2B,C).&lt;/p>&lt;!>&lt;p>After establishing the dMMR phenotype of CRISPR-edited MSH2-null cell lines, we determined whether CRISPR-mediated dMMR facilitated the emergence of compound-resistant clones. We performed drug selections using MSH2-wild type A673 cells (parental A673), MSH2-null A673 cells (A673-M1 and A673-M8), and HCT116 cells. Selections were performed against three cellular toxins: bortezomib (an inhibitor of the subunit β5 of the proteasome, PSMB5), MLN4924 (a NEDD8-activating enzyme (NAE) inhibitor), and CD437 (a DNA polymerase alpha (POLA1) inhibitor)(Soucy et al., 2009, Han et al., 2016, Lu and Wang, 2013, Chen et al., 2011). Following compound selection at lethal doses, resistant colonies growing in the presence of compound were visualized by crystal violet staining. Consistent with the notion that CRISPR-mediated MSH2 loss enabled acquisition and emergence of compound resistant clones, colonies were observed following selection only on the MSH2-null A673 and HCT116 plates (Figure 1E). No colonies were observed on the parental MSH2-wild type A673 plates.&lt;/p>&lt;p>We next performed additional selections using bortezomib, CD437, and MLN4924. We selected MSH2-null A673 cells and Msh2-null mSCLC cells using bortezomib at three concentrations close to the lethal dose, as determined by one week of compound exposure (EC1001wk) for bortezomib (see methods). Following 2 weeks of selection, compound-resistant colonies emerged and were expanded from both A673-M1 (4 clones) and Msh2-null mSCLC cells (11 clones).&lt;/p>&lt;p>Bortezomib-resistant alleles have been reported within exon 2 of PSMB5, which encodes a binding pocket for the drug (Lu and Wang, 2013, Chen et al., 2011). We therefore amplified and sequenced exon 2 in bortezomib-resistant MSH2-null A673 and Msh2-null mSCLC clones. These regions were also amplified and sequenced from the parental A673 and mSCLC cell lines to ensure these mutations did not exist prior to MMR impairment. We identified PSMB5 mutations in MMR-deficient A673 (4 out of 4 clones harbored mutations) and mSCLC cell lines (11 out of 11 clones harbored PSMB5 exon 2 mutations), including mutations previously reported to mediate bortezomib resistance (Fig. 2B, D, E) (Lu and Wang, 2013, Wacker et al., 2012). All mutations identified in PSMB5 mapped to the bortezomib binding pocket (Huber et al., 2016) (Fig. 2F). We confirmed in vitro resistance to bortezomib in all 4 clones harboring putative compound-resistant alleles from A673-M1 by cell viability assay (CellTiter Glo, Promega) following 72 hours of drug exposure (Figure 2A, B). We also tested 7 out of the 11 mSCLC clones that harbored Psmb5 mutations and confirmed in vitro resistance to bortezomib (Figure 2C, D). We confirmed that all clones harboring PSMB5 mutations exhibited bortezomib resistance (2.36 to 13.84-fold increase in EC50), whereas resistance to etoposide was not observed (Figure 2A–D; Figure S3A, B).&lt;/p>&lt;p>To further confirm that CRISPR-dMMR cells exhibited the capacity to acquire compound resistant alleles to different classes of cellular toxins, we performed additional selections using MSH2-null A673 cells and Msh2-null mSCLC cells against the DNA polymerase-alpha inhibitor CD437. After 2 weeks of selection, compound-resistant colonies emerged and were expanded from both A673-M1 (4 clones) and Msh2-null mSCLC cells (17 clones). For CD437-selected clones, cDNA flanking exons 19 to 25 was amplified and sequenced after clonal expansion. We identified POLA1 mutations in the MMR-deficient A673 (4 out of 4 clones harbored mutations) and mSCLC lines (11 out of 17 clones harbored mutations) (Figure 3B, D, E) (Han et al., 2016). None of these mutations were detected in the parental, MMR proficient, cell lines. Mutations impacted five amino acids that clustered within the POLA1 structure (Coloma et al., 2016) (Figure 3F).&lt;/p>&lt;p>We confirmed CD437-resistance in all 4 clones from A673-M1 cell line and 11 out of 17 mSCLC clones harboring putative compound-resistant alleles to CD437 (Figure 3A, B). The six mSCLC resistant clones that did not harbor POLA1 mutations exhibited resistance to both CD437 and etoposide, which suggested a generalized mechanism of acquired resistance drove expansion of these clones during compound selection (Figure S3C, D). The ten mSCLC clones harboring POLA1 mutations all exhibited CD437 resistance (5.31 to 15.05-fold increase in EC50), whereas no difference in sensitivity to etoposide was observed (Figure 3C, D; Figure S3E, F).&lt;/p>&lt;p>The selections performed with bortezomib and CD437 demonstrate that engineered MSH2 loss in enables the emergence of compound resistant alleles during drug selections in A673 and mSCLC cells. We finally tested whether prospective identification of compound targets could be accomplished using exome sequencing of compound resistant clones. Therefore, we performed selections using MLN4924 in the MSH2-null clone, A673-M1, at three concentrations for MLN4924 (see Methods). Following two weeks of selection, six colonies emerged and were expanded. To confirm in vitro resistance, we determined the EC50 for both parental A673-M1 cells and six resistant clones (Figure 4A,B). To identify potential clones exhibiting general resistance, including increased expression of drug efflux pumps, we assessed sensitivity to the topoisomerase II-inhibitor, etoposide (Figure S3F). We validated that all MLN clones exhibited resistance to MLN4924 (21.75 to 135.03-fold increase in EC50), while no difference in etoposide resistance was observed (Figure 4A,B; Figure S3F).&lt;/p>&lt;p>We next determined if the target of MLN4924, NAE subunit encoded by UBA3, could be identified by WES of MLN4924-resistant clones. Indeed, UBA3 was identified as the single gene mutated in 6/6 MSH2-null clones (Figure 4C). This gene encodes the NAE subunit targeted by MLN4924, and three of the mutations observed, A171T, E204K, and Y352H, were previously reported as MLN4924-resistant alleles (Figure 4D) (Michael et al., 2012, Xu et al., 2014). We compared somatic mutations in MLN4924-resistant A673 clones to establish whether the MLN4924-resistant clones arose independently (see Methods). A673-MLN-D and A673-MLN-H shared a majority of mutations (783/965), establishing that these two clones arose from the same founder cell. However, no other clones shared more than 6 somatic mutations, suggesting that the other clones arose independently, including A673-MLN-A and A673-MLN-C that shared the A171T mutation in UBA3.&lt;/p>&lt;!>&lt;p>We establish that induced MMR deficiency using CRISPR-Cas9 methods is sufficient to induce MSI, hypermutation, and facilitate the emergence of compound resistant alleles in established human and murine cancer cell lines derived from diverse cancer lineages. This approach offers the potential to significantly expand the use of forward genetics to identify the mechanisms of action of compounds with anticancer activity. In particular, we demonstrate that this strategy can be employed in cancers with low mutation rates such as pediatric malignancies. Therefore, cell lines with engineered MMR deficiency represent an experimental tool to facilitate the identification of mechanisms of action of selective cancer toxins identified by HTS campaigns, and to model genetic mechanisms of acquired resistance to anti-cancer therapies in current use. However, we recognize limitations of the current study. First, following identification of candidate compound-resistant alleles (i.e., recurrent mutations in compound-resistant clones), additional biochemical studies are necessary to establish the direct molecular target. Second, the forward genetics approach requires that the small molecule target a protein essential for viability of the cancer cell. In addition, we cannot from our current data establish a frequency of mutation necessary to facilitate the emergence of compound-resistant clones. Hypermutation due to endogenous defects in DNA repair might also be more broadly applied to other phenotypic genetic screens, including in vivo screens in GEMMs. Therefore, cancer cell lines with induced MMR deficiency and hypermutation represent a tool with wide potential application in cancer genetics and drug discovery.&lt;/p>&lt;!>&lt;p>Further information and requests for resources and reagent should directed to and will be fulfilled by the Lead Contact, David McFadden (david.mcfadden@utsouthwestern.edu)&lt;/p>&lt;!>&lt;p>Ewing sarcoma A673 cell line were cultured at 37°C and 5% CO2 in RPMI (R8758, Sigma-Aldrich) supplemented with 10% FBS (#35-150-CV, Corning), 2 mM L-glutamine (G7513, Sigma-Aldrich), and penicillin streptomycin (P0781, Sigma-Aldrich). Cells were expanded using Trypsin (T4049, Sigma-Aldrich) every 3-4 days. A673 cell lines are derived from a female subject and were authenticated by STR profiling.&lt;/p>&lt;p>The Trp53fl/fl; Rb1fl/fl; Rbl2fl/fl; RosaLSL-Tomato/+ mouse model of small cell lung cancer mice has been previously described (Schaffer et al., 2010). 319-T1 and 319-T2 were established from primary tumors in the lung, and 319-N1 was established from a lymph node metastasis, all developed in a PRP female mouse. 319-N1 as well as clones derived from those cell lines were cultured with DMEM (D6429, Sigma-Aldrich) supplemented with 5% FBS (#35-150-CV, Corning), 2 mM L-glutamine (G7513, Sigma-Aldrich), and penicillin streptomycin (P0781, Sigma-Aldrich). Cells were expanded using Trypsin (T4049, Sigma-Aldrich) diluted in PBS (2:1 ratio) every 3-4 days.&lt;/p>&lt;p>All animal experiments were approved by the UTSW IACUC 2018-102383 (D.G.M., P.I.).&lt;/p>&lt;!>&lt;p>Single-guide RNA (sgRNA) targeting human MSH2 and murine Msh2 were designed by "sgRNA Designer: CRISPR KO" (https://portals.broadinstitute.org/gpp/public/analysis-tools/sgrna-design) from the Broad Institute(Doench et al., 2016). sgRNAs were cloned into LentiCRISPR V2 plasmid (Addgene plasmid #52961) as previously described(Ran et al., 2013), and validated by T7 endonuclease assay (T7 endonuclease I from NEB, Cat. #M0302). sgRNA sequences used were as follows: human MSH2 5'-TGAGAGGCTGCTTAATCCAC-3'; murine Msh2 5'-GGTTAATACCCT GATACAGT-3'. For the generation of lentiviral vectors, 293T/17 cell were transfected with LentiCRISPR V2, psPAX2 (Addgene plasmid #12260), and pMD2.G (Addgene plasmid #12259) in a ratio (4:3:1) using TransIT®-LT1Transfection reagent (MIR 2304, Mirus Bio) as described by manufacturer. Mouse SCLC and Ewing sarcoma cell lines were plated at 106 cells in a 10 cm dish and cultured overnight. The next day, cells were transduced with lentiviruses (MOI&lt;0.5 determined by visual assessment) using 8ug/mL of polybrene transfection agent (TR-1003-G, EMD Millipore). Cells were selected with 2 mg/ml of Puromycin (P8833, Sigma-Aldrich) for 72 hours. Then surviving clones were picked and expanded for validation by western-blotting.&lt;/p>&lt;!>&lt;p>Western-blotting for Ewing sarcoma and mouse SCLC protein samples was performed using standard methods. Odyssey Nitrocellulose membrane (LICOR, #926-31092) were used for protein transference and then blocked using Odyssey® Blocking Buffer in PBS (LICOR, #927-40000) for 1h at RT. Primary antibodies were incubated for 1h at RT diluted 1:1,000 in Odyssey® Blocking Buffer : PBS-Tween (0.1%). Antibodies used were anti-Msh2 [D24B5] XP rabbit mAb (#2017, Cell Signaling Technology), and anti-β-actin (8H10D10) mouse mAb (#3700, Cell Signaling Technologies). Membrane was washed with PBS-Tween (0.1%) three times for 5 minutes each wash. Secondary antibodies were incubated for 30 min at RT using IRDye 800CW donkey anti-rabbit IgG (H+L) (#926-32213, LI-COR), and IRDye 680RD donkey anti-mouse IgG (H+L) (#926-68072, LI-COR), at dilution 1:10,000. Visualization was performed with Odyssey CLx Imaging System (LI-COR).&lt;/p>&lt;!>&lt;p>Bortezomib was purchased from Selleck Chemicals (#S1013). Etoposide was purchased from Sigma-Aldrich (#E1383-100MG). CD437 was purchased from Sigma-Aldrich (#C5865). MLN4924 was purchased from ApexBio (#B1036). Compounds were diluted using DMSO (Sigma-Aldrich, D650-100ML) and aliquoted at 10 mM and aliquots were exposed to a maximum of three freeze-thaw cycles.&lt;/p>&lt;!>&lt;p>Mouse SCLC cell lines were plated in 96-well plates, 6,600 cells per well in 200 μL of media. Ewing sarcoma cell lines were plated in 96-well plates, 10,000 cells per well in 200 μL of media. After overnight incubation, compounds were dispensed using a D300e Digtal Dispenser (TECAN). Cell viability assay was assessed after 72 hours using CellTiter-Glo luminescent cell viability assay (Promega, #G7571). The CellTiter-Glo reagent was diluted by adding PBS-Triton-X (1%) (1:1 ratio). EC1001wk determination for bortezomib was performed in a 12-well plate seeding 25,000 cells per well. After 24h, Bortezomib was dispensed using TECAN D300e setting up a minimum concentration of EC5072h and a maximum concentration of EC10072h. Media was changed every 3-4 days to refresh compounds. Cell viability was determined visually following seven days.&lt;/p>&lt;!>&lt;p>10 cm plates for each MMR deficient cell lines (1 million cells per plate) were treated with bortezomib or CD437 at EC1001wk ÷ 1.5, EC1001wk, and EC1001wk × 1.5 concentrations. Media with bortezomib or CD437 was replenished every 3 – 4 days over the course of 2 weeks. Surviving clones were expanded. Exon 2 of PSMB5 from both human and mouse cell lines was amplified and sequenced using primers specified in Table S3.&lt;/p>&lt;p>Amplification and sequence of exon 19 to 25 of POLA1 from both human and mouse cell lines was amplified and sequenced using primers specified in Table S3.&lt;/p>&lt;!>&lt;p>10 cm plates for each MMR deficient cell lines (3 million cells per plate) were treated with botezomib, MLN4924, and CD437 at EC1001wk ÷ 1.5, EC1001wk ÷ 1.25, EC1001wk, and EC1001wk × 1.25, EC1001wk × 1.5 concentrations. Media with either bortezomib, MLN4924, or CD437 was replenished every 3 – 4 days over the course of 2 weeks followed by growth in media without compound for 1 week.&lt;/p>&lt;p>Staining solution was prepared with 1% (weight/volume ratio) crystal violet from Sigma-Aldrich (#C6158-50G) in 10% ethanol.&lt;/p>&lt;!>&lt;p>Whole-exome sequencing of cell line samples was performed by BGI Genomics using SureSelect Human All Exon V5 (Ewing sarcoma samples) and SureSelect Mouse All Exon V1 (mSCLC) and BGISEQ-500. The analysis workflow was based on Genome Analysis Toolkit (GATK, v3.8-0) best practices(McKenna et al., 2010, DePristo et al., 2011). The qualities of sequencing reads were evaluated using NGS QC Toolkit (v2.3.3)(Patel and Jain, 2012) and the extracted high-quality reads were mapped to human and mouse reference genome (UCSC hg19 and Ensembl 91) using Burrows-Wheeler Aligner (BWA, v0.7.15a)(Li and Durbin, 2009). Picard (v2.12.0) (https://broadinstitute.github.io/picard.) was used to remove PCR duplicates and GATK was used to recalibrate base qualities. For murine SCLC cell lines, calling variants and joint genotyping together were performed using HaplotypeCaller and the variant calls were filtered by applying the following criteria: QD (Variant Confidence/Quality by Depth) &lt; 2, FS (Phred-scaled p-value using Fisher's exact test to detect strand bias) > 60, MQ (RMS Mapping Quality) &lt; 40, DP (Approximate read depth) &lt; 10, GQ (Genotype Quality) &lt; 20, maximum VAF (variant allele fraction) &lt; 0.2. For each murine SCLC cell lines, the somatic mutations in 319-N1 were defined by the VAF > 0.15 and VAF &lt; 0.05 for 319-T1 and 319-T2. Somatic mutations in 319-N1-Cl31 were defined by the VAF > 0.15 and VAF &lt; 0.05 for the other cell lines. For human cell lines, MuTect2(Cibulskis et al., 2013) was used to identify somatic mutations in clones A673-M1 and A673-M8 comparing to the A673 parental cell line. Somatic mutations for each clones (A673-M1 and A673-M8) were defined by VAF > 0.15 and VAF &lt; 0.05 for the other human cell lines.&lt;/p>&lt;!>&lt;p>We defined acquired somatic mutations for each A673-M1 MLN-resistant clone by VAF > 0.01 and VAF &lt; 0.01 for the parental A673-M1 cell line. Non-coding mutations were excluded.&lt;/p>&lt;!>&lt;p>Data were analyzed using Prism 8.0 by GraphPad. Dose response curves were fitting in Figure 2A, 2C, 3A, 3C and 4A, to calculate IC50. Hill coefficients and standard error were done using Log [inhibitor] vs Normalized response. Quantitative data are presented as mean.&lt;/p>&lt;!>&lt;p>The variants were annotated using a custom Perl script (https://github.com/jiwoongbio/Annomen) with mouse transcripts, proteins, and variations (Ensembl 91 for mouse, RefSeq and dbSNP build 150 for human). The variant allele frequencies were calculated using a custom Perl script and SAMtools (v1.4) (Goncearenco et al., 2017, Li et al., 2009) (all analysis scripts are available at https://github.com/jiwoongbio/Annomen).&lt;/p>&lt;p>The sequencing datasets exposed in this study have been deposited in SRA under accession code PRJNA543281.&lt;/p>&lt;!>&lt;p>Table S1. WES_A673 Ewing_Related to Figures 1 and 4. This table shows whole exome sequencing data for the A673 cell lines described in this manuscript.&lt;/p>&lt;p>Tab 1. Raw data. Whole exome sequencing data for A673 cell line, A673 MMR-proficient and MMR-deficient cell lines, and MLN4924-resistant clones.&lt;/p>&lt;p>Tab 2. Somatic mutations in A673-Cl1 (A673 VAF &lt; 0.05; A673-Cl1 > 0.15).&lt;/p>&lt;p>Tab 3. Somatic mutations in A673-Cl2 (A673 VAF &lt; 0.05; A673-Cl2 > 0.15).&lt;/p>&lt;p>Tab 4. Somatic mutations in A673-Cl4 (A673 VAF &lt; 0.05; A673-Cl4 > 0.15).&lt;/p>&lt;p>Tab 5. Somatic mutations in A673-M1 (A673 VAF &lt; 0.05; A673-M1 > 0.15).&lt;/p>&lt;p>Tab 6. Somatic mutations in A673-M8 (A673 VAF &lt; 0.05; A673-M8 > 0.15).&lt;/p>&lt;p>Table S2. WES_319 mSCLC_Related to Figures 1, S1 and S2. This table shows whole exome sequencing data for the murine small cell lung cancer cell lines described in this manuscript.&lt;/p>&lt;p>Tab 1. Raw data. Whole exome sequencing data for 319-T1, 319-T2, 319-N1, and MMR-deficient 319-N1-Cl31 cell line.&lt;/p>&lt;p>Tab 2. Somatic mutations in 319-N1 (319-T1 VAF &lt; 0.05; 319-T2 VAF &lt; 0.05; 319-N1 > 0.15).&lt;/p>&lt;p>Tab 3. Somatic mutations in 319-N1-Cl31 (319-T1 VAF &lt; 0.05; 319-T2 VAF &lt; 0.05; 319-N1-Cl31 > 0.15).&lt;/p>&lt;p>Table S3. Oligos PCR and sequencing_Related to STAR Methods. This table shows the oligos used for amplification and sequencing of exon 2 from PSMB5 and exons 19 to 25 from POLA1.&lt;/p></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">PubMed Author Manuscript</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="3"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Efficient oxidation of oleanolic acid derivatives using magnesium bis(monoperoxyphthalate) hexahydrate (MMPP): A convenient 2-step procedure towards 12-oxo-28-carboxylic acid derivatives</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">A new, straightforward and high yielding procedure to convert oleanolic acid derivatives into the corresponding δ-hydroxy-γ-lactones, by using the convenient oxidizing agent magnesium bis(monoperoxyphthalate) hexahydrate (MMPP) in refluxing acetonitrile, is reported. In addition, a two-step procedure for the preparation of oleanolic 12-oxo-28-carboxylic acid derivatives directly from Δ12-oleananes, without the need for an intermediary work-up, and keeping the same reaction solvent in both steps, is described as applied to the synthesis of 3,12-dioxoolean-28-oic acid.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">efficient_oxidation_of_oleanolic_acid_derivatives_using_magnesium_bis(monoperoxyphthalate)_hexahydra</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">1,408</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">73</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">19.287671</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Findings&lt;!>&lt;!>Findings&lt;!>&lt;!>Findings&lt;!>&lt;!>Findings&lt;!>&lt;!>Findings&lt;!>Supporting Information&lt;!></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">&lt;p>The molecular diversity that arises from research into natural products represents a valuable tool for driving drug discovery and development [1–2]. In this context, pentacyclic triterpenoids are currently regarded as important scaffolds for new drug development [3]. The chemistry of oleanane-type triterpenoids has been investigated with particular interest and many relevant biological and pharmacological activities of these derivatives have been reported in the literature, among which are antitumor, antiviral, anti-inflammatory, hepatoprotective, gastroprotective, antimicrobial, antidiabetic, and hemolytic properties, as well as many others [3–5]. Functionalized γ-lactones are important building blocks of bioactive natural products [6–7]. The δ-hydroxy-γ-lactone motif is part of such bioactive natural products as (±)-muricatacin [8–9] or the three hydroxylactones found in the mushroom Mycoleptodonoides aitchisonii [10]. Terpenoid δ-hydroxy-γ-spirolactones have been found to act as significant feeding deterrents to the lesser mealworm Alphitobius diaperinus [11]. In particular, oleanane-type triterpenoids bearing a γ-lactone function, either isolated from natural sources or obtained by semisynthesis, have shown interesting biological activities [12–14]. From the synthetic point of view, the oxidative 28,13β-lactonization allows the preparation of 12α-hydroxyoleananes with a protected 28-carboxyl acid function. In fact, 12α-hydroxy-3-oxooleanan-28,13β-olide (2) is a key intermediate in the synthesis of S-0139, an endothelin A receptor antagonist [15]. Moreover, as part of our ongoing work on pentacyclic triterpenoid chemistry [16–17], we recently demonstrated that oleanolic δ-hydroxy-γ-lactones can be efficiently converted into the corresponding 12-oxo-28-carboxylic acid derivatives by bismuth(III) triflate catalysis [18]. This new approach not only avoids an inconvenient multistep synthesis by means of a protection/deprotection strategy [19–20] but also results in chemical modification of ring C, a strategy known to increase the anti-inflammatory and cytotoxic activities of oleanolic acid (OA) derivatives [19,21–22].&lt;/p>&lt;p>Oleanolic δ-hydroxy-γ-lactones can be obtained from Δ12-oleananes by oxidative 28,13β-lactonization. This reaction was performed under photochemical irradiation [23–24], but weak selectivity and low isolated yields were observed. Alternatively, oxidation reagents such as H2O2 in acetic acid [25–26], the inorganic salt mixture KMnO4/CuSO4 [27], ozone [15,28–29] and m-chloroperoxybenzoic acid (mCPBA) [30–31] have also been reported. Magnesium bis(monoperoxyphthalate) hexahydrate (MMPP) is commercially available, inexpensive and relatively stable [32–34] and has been used in the oxidation of various functional groups [35–42]. This oxidant is non-shock-sensitive and non-deflagrating [43]. Moreover, its use greatly simplifies the isolation of the reaction products, because it may simply be filtered off from the reaction crude, which is then worked up as usual.&lt;/p>&lt;p>In this letter, we report the use of MMPP for the efficient and high-yielding oxidation of OA derivatives to afford the corresponding δ-hydroxy-γ-lactones. Moreover, we have set up a protocol that allows the convenient sequential two-step preparation of 3,12-dioxoolean-28-oic acid directly from 3-oxooleanolic acid, without the need of an intermediary work-up, and keeping the same reaction solvent in both steps.&lt;/p>&lt;p>We found that the reaction of 3-oxooleanolic acid 1 with 2.0 equiv of MMPP, in refluxing acetonitrile, afforded the corresponding δ-hydroxy-γ-lactone 2 in 85% yield after 5 hours (Table 1, entry 1). These new reaction conditions were successfully extended to OA 3 and other 3β-substituted OA derivatives 5, 7 and 9 (Table 1, entries 2–5).&lt;/p>&lt;!>&lt;p>Reaction of OA derivatives with MMPP to afford oleanolic δ-hydroxy-γ-lactones directly.a&lt;/p>&lt;p>aReactions were performed in acetonitrile, under reflux; bAnalytical data for compounds 2 [29], 4 [28], 6 [27], 8 [18] and 10 [18] are in accordance with the literature; cIsolated yield.&lt;/p>&lt;!>&lt;p>The substrates were dissolved in acetonitrile under reflux, and MMPP (2.0–3.0 equiv) was added to the solution under strong magnetic stirring. After completion of the reaction, the magnesium salts were easily filtered off after evaporation of the acetonitrile and suspension of the resulting white solid in ethyl acetate. We found that 2.0 equiv of MMPP were sufficient to effectively convert 3-oxooleanolic acid 1, OA 3, and 3β-acetoxyoleanolic acid 5 into the corresponding δ-hydroxy-γ-lactones 2, 4 and 6, in 84 to 88% yield (Table 1, entries 1–3). Substrates 7 and 9, bearing a trifluoroacetoxy and a methoxy group at C3, respectively, required higher amounts of the reagent and longer reaction times (Table 1, entries 4 and 5). The formation of the oleanolic δ-hydroxy-γ-lactones 2, 4, 6, 8 and 10, may be explained by epoxidation of the parent Δ12-oleanane compound, followed by nucleophilic attack of the 28-carboxyl group at C13 from the β-face, with ring-opening of the 12α,13α-epoxide intermediate [15,44].&lt;/p>&lt;p>Quite recently, we demonstrated that the 28,13β-lactonization of 3-oxooleanolic acid 1 promoted by Bi(OTf)3·xH2O affords a 3-oxo-18α-olean-28,13β-olide product, with inversion of configuration at the C18-stereocenter, as demonstrated by X-ray crystallography [45–46]. In order to assign the orientation of the 18-H of the 12α-hydroxy-γ-lactones obtained in this work, 2D NMR data were collected for compounds 2, 6 and 8, and X-ray data were gathered for compound 4. Combining 1D and 2D-NMR spectroscopy, we were able to determine the chemical shift of 18-H (2.02 ppm) for compound 2. This value is much lower than the one of the parent substrate 1 (2.84 ppm), which may be explained by magnetic anisotropy induced by the 28,13β-lactone moiety. It is also interesting to note that a long-distance coupling between the 18-H and 12β-H (3.90 ppm) signals was found in the COSY spectrum of 2. Correlation between these two signals was also observed in the NOESY experiment and, therefore, the β-configuration was assigned at the C18-stereocenter. The same NMR pattern was present for compounds 6 and 8. Unequivocal evidence of the molecular structure of compound 4 was obtained by single-crystal X-ray crystallography, and the ORTEP diagram with the corresponding atomic numbering scheme is depicted in Figure 1.&lt;/p>&lt;!>&lt;p>ORTEP diagram of compound 4 (50% probability level, H atoms of arbitrary sizes). The asymmetric unit also contains a molecule of CH3CN.&lt;/p>&lt;!>&lt;p>In the past few years, bismuth(III) salts have emerged as convenient reagents for the development of new chemical processes under more "ecofriendly" reaction conditions, which avoid the use of large amounts of toxic and corrosive materials [47–51]. Bearing in mind the solubility properties of MMPP and that both the oxidative 28,13β-lactonization and the bismuth(III) triflate-catalyzed direct opening of δ-hydroxy-γ-lactones are performed in acetonitrile, we designed a protocol to perform the synthesis of oleanolic 12-oxo-28-carboxylic acid derivatives directly from Δ12-oleananes, without the need for an intermediary work-up, and keeping the same reaction solvent in both steps (Scheme 1).&lt;/p>&lt;!>&lt;p>Sequential 2-step synthesis of 3,12-dioxoolean-28-oic acid (11) directly from 3-oxooleanolic acid (1).&lt;/p>&lt;!>&lt;p>Thus, after the formation of the 12α-hydroxy-28,13β-olide compound 2 by MMPP oxidation, a filtration step allowed the elimination of insoluble magnesium salts, taking advantage of their low solubility in acetonitrile. Then, a catalytic amount of bismuth(III) triflate (5 mol %) was added to the resulting filtrate, and the expected 12-oxo-28-carboxylic acid 11 was obtained, in 85% yield, after the typical work-up procedure [18]. The formation of compound 11 from the 12α-hydroxy-28,13β-olide 2 is likely to occur due to the in situ generation of a Brønsted acid species from bismuth(III) triflate, which promotes ring opening of the 28,13β-olide group, creating a tertiary carbocation at C-13. Then, a concerted stereoselective 1,2-migration of the 12β-H to the 13β-position with the rearrangement of the 12α-hydroxy group affords the final 12-oxo-28-carboxylic acid structure [18]. The molecular structure of compound 11, determined by single-crystal X-ray crystallography, is shown in Figure 2.&lt;/p>&lt;!>&lt;p>ORTEP diagram of compound 11 (50% probability level, H atoms of arbitrary sizes).&lt;/p>&lt;!>&lt;p>In conclusion, we have found a new straightforward procedure to convert OA derivatives into δ-hydroxy-γ-lactones, in very high yields, using the convenient oxidizing agent MMPP. This procedure has considerable advantages over the previously reported oxidation methods, because no other positions of the molecule are oxidized concomitantly, it avoids the use of halogenated solvents, and allows easy recovery of the reaction products. Combination of this oxidative 28,13β-lactonization process with the ability of bismuth(III) triflate to catalyze the opening of the resulting δ-hydroxy-γ-lactone with subsequent generation of the carbonyl group, allowed us to set up a sequential two step strategy for the preparation of 3,12-dioxoolean-28-oic acid (11) directly from 3-oxooleanolic acid 1, that avoids an intermediary work-up and conveniently uses the same reaction solvent in both steps. Thus, the procedure reported herein greatly simplifies the obtainment of oleanolic δ-hydroxy-γ-lactones, which are versatile intermediates for organic synthesis, and in addition can provide very easy access to the corresponding oleanolic 12-oxo-28-carboxylic acids.&lt;/p>&lt;!>&lt;p>The Supporting Information contains the typical procedure for the MMPP oxidative 28,13β-lactonization and preparation of compounds 2, 4, 6, 8 and 10. Moreover, the procedure for the sequential two step synthesis of 3,12-dioxoolean-28-oic acid (11) is described and the 1D and 2D NMR spectra of compounds 2, 4, 6, 8, 10 and 1D NMR spectra of compound 11 are shown.&lt;/p>&lt;!>&lt;p>Experimental and analytical data.&lt;/p></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">PubMed Open Access</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="4"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Spectral, thermal, antimicrobial studies for silver(I) complexes of pyrazolone derivatives</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">BackgroundSynthesize new complexes of Ag(I) to enhance efficacy or stability and also, pharmacological activities on the operation of pyrazolone's biological properties.ResultsEfficient and high yielding pathways starting from the versatile and readily available 3-methyl-1-phenyl-5-pyrazolone by Knoevenagel condensation of a sequence of 4-arylidene-3-methyl-1-phenyl-5-pyrazolone derivatives (2a-c) have been formed by the reaction of various substituted aromatic aldehydes Used as ligands to synthesize Ag(I) chelates. Synthesized compounds and their complexes have been characterized by elemental analysis, magnetic and spectroscopic methods (IR, 13C, 1HNMR, mass) and thermal analysis. The spectrophotometric determinations suggest distorted octaedral geometry for all complexes. Both ligands and their metal complexes have also been tested for their antibacterial and antifungal efficacy.ConclusionsNewly synthesized compounds have shown potent antimicrobial activity. The results showed that the complex 's high activity was higher than its free ligands, and that Ag(I)-L3 had the highest activity.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">spectral,_thermal,_antimicrobial_studies_for_silver(i)_complexes_of_pyrazolone_derivatives</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">1,891</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">138</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">13.702899</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Introduction&lt;!>&lt;!>Infrared spectra&lt;!>&lt;!>Thermal studies&lt;!>&lt;!>Antimicrobial studies&lt;!>&lt;!>Conclusion&lt;!>Chemistry&lt;!>Common 3-methyl-1-phenyl-5-pyrazolone synthesis technique (1)&lt;!>Specific method for preparing derivatives of 4-arylidene-3-methyl-1-phenyl-5-pyrazolone (2a-c)&lt;!>4-(4-dimethylamino benzylidene)-3-methyl-1-phenyl-1H-pyrazol-5(4H)-one (2a) L1&lt;!>4-(4-Thiophene)-3-methyl-1-phenyl-1Hpyrazol-5(4H)-one (2b) L2&lt;!>4-(4-methoxy benzylidene)-3-methyl-1-phenyl-1Hpyrazol-5(4H)-one (2c) L3&lt;!>Synthesis of the complexes&lt;!>[Ag(C19H19N3O)2(H2O)2]NO3 (AgC38H42N7O7) complex&lt;!>[Ag(C15H12N2OS)2(H2O)2]NO3.H2O (AgC30H30N5O8S2) complex&lt;!>[Ag(C18H16N2O2)2(H2O)2]NO3 (AgC36H36N5O9) complex&lt;!>&lt;!>Supplementary information</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">&lt;p>Pyrazolone chemistry began in 1883 when Ludwig Knorr first reacted to phenyl hydrazine with aceto-acetate ester. As pyrazolones were discovered as binding components for azo dyes in the late 1800s, they rapidly increased in importance. Today, pyrazolon is still an significant trade precursor to dyes and pharmaceuticals. Pyrazolone is a biologically important scaffold associated with different pharmacological activities such as antimicrobials [1–5], anti-inflammatory [6], analgesic [7], antidepressant [8], anticonvulsant [9], antidiabetic [10], antihyperlipidemic [11, 12], antiviral [13, 14], anti-tuberculosis [15, 16], antioxidant [17, 18] and anticancer [19, 20]. For several years, the preparation of pyrazolone and its derivatives has attracted significant attention from organic and medicinal chemists, as they belong to a class of compounds with promising results in medicinal chemistry. The heterocycles condensed to the pyrazole ring are an important source of bioactive molecules [21, 22]. Compounds containing both pyrazole and other essential heterocyclic active structural units usually demonstrate more remarkable biological activity. A number of condensed pyrazole derivatives have been reported as four-fold antibacterial agents against Gram-positive and Gram-negative bacteria compared to general pyrazole compounds [23, 24]. A digit of antimicrobial active silver(I) complexes have the capacity to disrupt microbial transpiration as well as block tyrosinase synthesis and are extremely cytotoxic to cancer cells [24]. Massive attention in silver ions (Ag(I)) as a broad spectrum antimicrobial has upped the size and importance of in vitro biocompatibility research [25]. Silver ions are toxic to many bacteria, viruses, algae and fungi. Silver-based medicines have been widely used for this task for decades [26]. The objective of this study is to display the synthesis and characterization of three Ag(I) pyrazolone complexes in an attempt to verify the mode of coordination and the biological properties of the final complexes.&lt;/p>&lt;!>&lt;p>Synthesis of 4-arylidene-3-methyl-1-phenyl-5-pyrazolone derivatives&lt;/p>&lt;p>The coordinationn mode of Ag (I) with three ligand&lt;/p>&lt;!>&lt;p>KBr disks registered mid-infrared spectra of L1, L2, L3 and their metal complexes. As expected, with changes in band intensities and wave numbers, the absorption bands characteristic of L1, L2, L3 acting as a monodentate unit are observed in the complexes. The proposed structures of the complexes must be considered prior to determining the assignments of the infrared spectra. Here, Ag(I) ion interacts with these monodentate ligands forming monomeric structure complexes in which the Ag(I) ion is four coordinated (Scheme 2) [27–30].&lt;/p>&lt;!>&lt;p>Infrared spectra for a L1, b [Ag(L1)2(H2O)2]NO3, c L2, d [Ag(L2)2(H2O)2]NO3.H2O, e L3 and f [Ag(L3)2(H2O)2]NO3&lt;/p>&lt;p>Infrared frequencies (cm−1)a and tentative assignmentsb for (A) L1, (B) [Ag(L1)2(H2O)2]NO3, (C), L2 (D) [Ag(L2)2(H2O)2]NO3.H2O, (E) L3 and (F) [Ag(L3)2(H2O)2]NO3&lt;/p>&lt;p>1550s&lt;/p>&lt;p>1400m&lt;/p>&lt;p>1523m&lt;/p>&lt;p>1410s&lt;/p>&lt;p>1527m&lt;/p>&lt;p>1381&lt;/p>&lt;p>1508s&lt;/p>&lt;p>1427vw&lt;/p>&lt;p>1520s&lt;/p>&lt;p>1415&lt;/p>&lt;p>ν(C = N)&lt;/p>&lt;p>ν(C = C)&lt;/p>&lt;p>1319s&lt;/p>&lt;p>–&lt;/p>&lt;p>1319s&lt;/p>&lt;p>1188s&lt;/p>&lt;p>1300s&lt;/p>&lt;p>-&lt;/p>&lt;p>1311w&lt;/p>&lt;p>1165m&lt;/p>&lt;p>1311sh&lt;/p>&lt;p>-&lt;/p>&lt;p>1311m&lt;/p>&lt;p>1172s&lt;/p>&lt;p>δb(-CH2),&lt;/p>&lt;p>ν(NO3−1)&lt;/p>&lt;p>1122s&lt;/p>&lt;p>1018w&lt;/p>&lt;p>–&lt;/p>&lt;p>1122s&lt;/p>&lt;p>1018w&lt;/p>&lt;p>–&lt;/p>&lt;p>1130m&lt;/p>&lt;p>–&lt;/p>&lt;p>1056w&lt;/p>&lt;p>1104vw&lt;/p>&lt;p>–&lt;/p>&lt;p>1099sh&lt;/p>&lt;p>1110w&lt;/p>&lt;p>–&lt;/p>&lt;p>–&lt;/p>&lt;p>1130m&lt;/p>&lt;p>–&lt;/p>&lt;p>–&lt;/p>&lt;p>ν(C–C),&lt;/p>&lt;p>ν(C-N)&lt;/p>&lt;p>ν(C = S)&lt;/p>&lt;p>954w&lt;/p>&lt;p>943vw&lt;/p>&lt;p>995w&lt;/p>&lt;p>995m&lt;/p>&lt;p>991s&lt;/p>&lt;p>921w&lt;/p>&lt;p>941vw&lt;/p>&lt;p>910vw&lt;/p>&lt;p>988w&lt;/p>&lt;p>938 s&lt;/p>&lt;p>985w&lt;/p>&lt;p>965sh&lt;/p>&lt;p>as = strong, w = weak, sh = shoulder, v = very, br = broad, bν = stretching and δ = bending&lt;/p>&lt;p>Electronic absorption spectra for L1, [Ag(L1)2(H2O)2]NO3, L2, [Ag(L2)2(H2O)2]NO3.H2O, L3 and [Ag(L3)2(H2O)2]NO3&lt;/p>&lt;p>1H NMR spectra for a L1, b [Ag(L1)2(H2O)2]NO3, c L2, d [Ag(L2)2(H2O)2]NO3.H2O, e L3 and f [Ag(L3)2(H2O)2]NO3&lt;/p>&lt;p>Thermogravimetric data of L1, L2,L3 and their metal complexes&lt;/p>&lt;!>&lt;p>The ligand (L2) degradates at 273, 475 °C. This stage is followed by a complete loss of weight of 86.70 percent, close to 86.56 percent of the estimated value (Additional file 1: Fig. S1c). Equivalent to 6C2H2 + SO + N2 loss and 31.93 kJ mol−1 (endothermic) activation energy. Decomposition of the residual value occurs at 771 °C and the real weight loss from this stage is 13.30 percent, similar to the estimated value of 13.43 percent corresponding to 3C. The [Ag(L2)2(H2O)2]NO3.H2O complex decomposes at two levels of decay (Additional file 1: Fig. S1d), the first phase occurs at 99 °C and is followed by a weight loss of 2.08 per cent relating to the removal of H2O, activation energy of 79.28 kJ mol−1. The second step of decomposition occurs at temperature is 203, 528 and is accompanied by a weight loss of 75.90%; corresponding to the value of 10C2H2 + 4HCN + 2H2O + NO2 + SO + SO2 theoretically, close to the calculated value 76.404%. The Residue value decomposition occurs at maximum 881 °C and the actual weight loss from this step is 23.35%, corresponding to Ag + 6C, close to the calculated value 23.596%.&lt;/p>&lt;p>The thermal decay of L3 happens in two phases of degradation (Additional file 1: Fig. S1e), the first step arises at 291 °C and is followed by a weight loss of 70.55 percent leading to a loss of 8C2H2 similar to the measured value of 71.23 per cent with activation energy of 35.31 kJ mol−1. The second step occurs at 518 οC and is accompanied by a weight loss of 28.604%; corresponding to the value of 2CO + N2 theoretically, close to the calculated value 28.67%. The [Ag(L3)2(H2O)2]NO3 degradation takes place in two stages (Additional file 1: Fig. S1f), the first occurs at 244 οC and is accompained by a weight loss of 51.071% corresponding to loss of 14C2H2 + 2H2O close to the calculated value 50.60% with an activation energy 15.31 kJ mol−1. The second one begins at 543 οC and is followed by a weight loss of 30.17%; corresponding to C2H2 + CO + 2HCN + 3NO2 theoretically, close to the calculated value 31.25%. The Residue remains at 677 °C and the actual weight loss is 17.76%, equal to Ag + 3C, close to the calculated value 18.15%.&lt;/p>&lt;!>&lt;p>Thermal behavior and kinetic parameters determined using the Coats–Redfern (CR) and Horowitz–Metzger (HM) operated for L1, L2, L3 and their complexes&lt;/p>&lt;p>CR&lt;/p>&lt;p>HM&lt;/p>&lt;p>95.656&lt;/p>&lt;p>116.813&lt;/p>&lt;p>69.368 × 103&lt;/p>&lt;p>30.233 × 103&lt;/p>&lt;p>− 432.844&lt;/p>&lt;p>− 425.910&lt;/p>&lt;p>88.863&lt;/p>&lt;p>110.020&lt;/p>&lt;p>442.497&lt;/p>&lt;p>457.988&lt;/p>&lt;p>0.970&lt;/p>&lt;p>0.984&lt;/p>&lt;p>0.187&lt;/p>&lt;p>0.065&lt;/p>&lt;p>CR&lt;/p>&lt;p>HM&lt;/p>&lt;p>34.379&lt;/p>&lt;p>38.825&lt;/p>&lt;p>3.390 × 102&lt;/p>&lt;p>2.621 × 103&lt;/p>&lt;p>− 393.344&lt;/p>&lt;p>− 410.348&lt;/p>&lt;p>30.537&lt;/p>&lt;p>34.983&lt;/p>&lt;p>212.262&lt;/p>&lt;p>224.564&lt;/p>&lt;p>0.984&lt;/p>&lt;p>0.970&lt;/p>&lt;p>0.116&lt;/p>&lt;p>0.087&lt;/p>&lt;p>CR&lt;/p>&lt;p>HM&lt;/p>&lt;p>31.931&lt;/p>&lt;p>40.983&lt;/p>&lt;p>37.451&lt;/p>&lt;p>672.979&lt;/p>&lt;p>− 373.640&lt;/p>&lt;p>− 397.656&lt;/p>&lt;p>27.391&lt;/p>&lt;p>36.443&lt;/p>&lt;p>231.398&lt;/p>&lt;p>253.563&lt;/p>&lt;p>0.943&lt;/p>&lt;p>0.945&lt;/p>&lt;p>0.207&lt;/p>&lt;p>0.112&lt;/p>&lt;p>CR&lt;/p>&lt;p>HM&lt;/p>&lt;p>79.284&lt;/p>&lt;p>93.856&lt;/p>&lt;p>6.618 × 103&lt;/p>&lt;p>113.45 × 103&lt;/p>&lt;p>− 413.475&lt;/p>&lt;p>− 437.099&lt;/p>&lt;p>72.624&lt;/p>&lt;p>87.196&lt;/p>&lt;p>403.817&lt;/p>&lt;p>437.312&lt;/p>&lt;p>0.945&lt;/p>&lt;p>0.940&lt;/p>&lt;p>0.220&lt;/p>&lt;p>0.116&lt;/p>&lt;p>CR&lt;/p>&lt;p>HM&lt;/p>&lt;p>35.317&lt;/p>&lt;p>48.603&lt;/p>&lt;p>1.185 × 102&lt;/p>&lt;p>2.847 × 103&lt;/p>&lt;p>− 382.948&lt;/p>&lt;p>− 409.379&lt;/p>&lt;p>30.627&lt;/p>&lt;p>43.913&lt;/p>&lt;p>246.610&lt;/p>&lt;p>274.803&lt;/p>&lt;p>0.960&lt;/p>&lt;p>0.952&lt;/p>&lt;p>0.182&lt;/p>&lt;p>0.106&lt;/p>&lt;p>CR&lt;/p>&lt;p>HM&lt;/p>&lt;p>15.316&lt;/p>&lt;p>22.370&lt;/p>&lt;p>0.758&lt;/p>&lt;p>0.119 × 102&lt;/p>&lt;p>− 342.002&lt;/p>&lt;p>− 364.917&lt;/p>&lt;p>11.183&lt;/p>&lt;p>18.237&lt;/p>&lt;p>181.158&lt;/p>&lt;p>199.601&lt;/p>&lt;p>0.981&lt;/p>&lt;p>0.985&lt;/p>&lt;p>0.089&lt;/p>&lt;p>0.054&lt;/p>&lt;p>aCorrelation coefficients of the Arrhenius plots and bStandard deviation&lt;/p>&lt;p>Mass spectra diagrams for (A) L1, (B) [Ag(L1)2(H2O)2]NO3, (C), L2 (D) [Ag(L2)2(H2O)2]NO3.H2O, (E) L3 and (F) [Ag(L3)2(H2O)2]NO3&lt;/p>&lt;p>Fragmentation pattern of L1&lt;/p>&lt;p>Fragmentation pattern of L2&lt;/p>&lt;p>Fragmentation pattern of L3&lt;/p>&lt;p>The inhibitation diameters zone values (mm) for L1, L2, L3 and its complexes&lt;/p>&lt;p>Statistical significance PNS – P not significant, P > 0.05; P+1 – P significant, P &lt; 0.05; P+2 – P highly significant, P &lt; 0.01; P+3 – P very highly significant, P > 0.001; Student's t-test (Paired)&lt;/p>&lt;p>Statistical representation for biological activity of L1, L2, L3 and its metal complexes&lt;/p>&lt;!>&lt;p>Normal antibiotic efficacy of antimicrobials (AMC, CTX, NS, FU). The AMC mixture give the effective against E. coli, Coliform, S. aureus and NS high inhibitory activity on A. niger. Other antibiotics have shown no action on other microorganisms. Eventually, the bacterial strains showed a varied response to the three free ligands and their complex antimicrobial activity, but the results indicated that the high activity of ligand complexes was better than their free ligands. The two fungal strains are more resistant to synthesis ligands and their complexes than bacterial strains [42–46].&lt;/p>&lt;!>&lt;p>(A) Of One-way ANOVA: E. coli vs MIC Compounds. (B) Of One-way ANOVA: Coliform versus MIC Compounds. (C) Of One-way ANOVA: S. aureus vs MIC Compounds. (D) Of One-way ANOVA: Salm. typhi vs MIC Compounds. (E) Of One-way ANOVA: A. niger vs MIC Compounds. (E) Of One-way ANOVA: A. niger vs MIC Compounds. (F) Of One-way ANOVA: P.expansum vs MIC Compounds&lt;/p>&lt;p>Means that do not share a letter are significantly different&lt;/p>&lt;p>Fisher 95% Simultaneous Confidence Intervals&lt;/p>&lt;p>MIC for the most sensitive organisms&lt;/p>&lt;!>&lt;p>Development and characterisation of three novel complexes of some replaced pyrazole derivatives as ligands (4-(4-dimethylamino benzylidene)-3-methyl-1-phenyl-1H-pyrazol-5(4H)-one (2a) L1, 4-(4-Thiophene)-3-methyl-1-phenyl-1Hpyrazol-5(4H)-one (2b) L2, 4-(4-methoxy benzylidene)-3-methyl-1-phenyl-1Hpyrazol-5(4H)-one (2c) L3) with Ag(I) was achieved using physicochemical and spectroscopic methods.. In the resulting complexes, L1, L2, and L3 were bound by the nitrogen atom to the metal ion via ν(C = N). For the three ligands and their complexes, thermogravimetric kinetic parameters and their differential were evaluated using the Coats-Redfern and Horowitz-Metzger equations. Metal complexes exhibited higher inhibition against all tested microorganisms and pathogenic bacteria and fungi and were the most susceptible pathogens with a minimum inhibitory concentration ( MIC).&lt;/p>&lt;!>&lt;p>Analytical grade reagents, commercially available from multiple suppliers and used without further purification, were all the chemicals used in the complex preparation. Synthesized compounds and their complexes have been characterized by elemental analysis, magnetic and spectroscopic methods (IR, 13C, 1HNMR, mass) and thermal analysis using the known apparatuses [42].&lt;/p>&lt;!>&lt;p>Pure ethyl acetoacetate (0.05 mol, 6.2 mL) was mixed with pure phenyl hydrazine (0.05 mol, 5 mL), 0.5 mL of acetic acid was added, according to knowm method [42]. Methyl phenyl pyrazolone was obtained as colorless crystals, 127 °C melting point and 83.6 percent yield [27].&lt;/p>&lt;!>&lt;p>The oil bath heated a mixture of 1-aryl-3-methyl-5-pyrazolone (0.01 mol, 1.74 g) and replaced aromatic aldehydes (0.012 mol) at 150–160 °C for 2-4hrs. TLC has tracked the progress of the reaction using ethyl acetate: hexane (9:1) as solvent. The mixture was cooled, triturated and washed off with ether (20 mL). The colored residue was recrystallized from ethanol to provide the corresponding 4-arylidene-3-methyl-1-phenyl-5-pyrazolone (2a-c) as colored products, respectively [28].&lt;/p>&lt;p>4-(4-dimethylamino benzylidene)-3-methyl-1-phenyl-1H-pyrazol-5(4H)-one (2a) L1.&lt;/p>&lt;p>4-(4-Thiophene)-3-methyl-1-phenyl-1Hpyrazol-5(4H)-one (2b) L2.&lt;/p>&lt;p>4-(4-methoxy benzylidene)-3-methyl-1-phenyl-1Hpyrazol-5(4H)-one (2c) L3.&lt;/p>&lt;!>&lt;p>Brick Red, mp = 170 °C, yield 83% IR (KBr, v, cm−1): 3444 (OH), 1670 (C = O), and 1550 cm−1. 1H NMR (DMSO-d6, 300 MHz): δ = 2.28 (s, 3H, CH3), 3.03 (s, 6H, -N (CH3)2), 7.14 (S, 1H, = CH-Ar), 9.66 (d, 3H, Ar–H),. Anal. Calcd for C19H19N3O (305.19): C, 74.40; H, 6.22; N 13.76; Found C, 74.23; H, 6.13; N, 13.35%.&lt;/p>&lt;!>&lt;p>Orange, mp = 125 °C, yield 74% IR (KBr, v, cm−1): 3448 (OH), 1681 (C = O), 1496 cm−1 (C = N) and 1056 cm−1(C = S). 1H NMR (DMSO-d6, 300 MHz): δ = 2.30 (s, 3H, CH3), 7.39 (S, 1H, = CH-Ar), 8.25 (d, 3H, Ar–H). Anal. Calcd for C15H12N2OS (268): C, 67.16; H, 4.47; N 10.44; S, 11.94; Found C, 67.00; H, 4.32; N, 10.21; S, 11.65%.&lt;/p>&lt;!>&lt;p>Orange, mp = 122 °C, yield 82% IR (KBr, v, cm−1): 3444 (OH), 1678 (C = O), 1508 cm−1 (C = N) and. 1H NMR (DMSO-d6, 300 MHz): δ = 1.91 (s, 3H, CH3), 3.69 (s, 3H, -OCH3), 7.20 (S, 1H, = CH-Ar), 8.71 (d, 3H, Ar–H).Anal. Calcd for C18H16N2O2 (292): C, 73.97; H, 5.47; N 9.58; Found C, 73.78; H, 5.13; N, 9.34%.&lt;/p>&lt;!>&lt;p>The brown solid complex [Ag(L1)2(H2O)2]NO3 was prepared by adding 0.5 mmol (0.085 g) of AgNO3 in 20 ml of acetone to a stirred suspended solution 1 mmol (0.305 g) of L1 in 50 ml acetone. The reaction mixture was refluxed for 6 h, the precipitate was drained off, washed several times with acetone and dried under vacuum over anhydrous CaCl2. Dark brown [Ag(L2)2(H2O)2]NO3.H2O, [Ag(L3)2(H2O)2]NO3 solid complexes were prepared in the same manner as mentioned above.&lt;/p>&lt;!>&lt;p>Brown; Yield: 85%; m.p.: 160 οC; M.Wt: 816.65; Elemental analysis for AgC38H42N7O7: found, C, 55.31; H, 4.99; N, 12.00; Ag, 13.14; Calcd, C 55.89; H, 5.18; N, 12.01; Ag, 13.21; Λm = 115.75 S cm2 mol−1; IR (KBr, v, cm−1): 3450 m,br (OH), 1666 m (C = O), 1523vw cm−1(C = N) and 813w and 837w (M–N). 1H NMR (DMSO-d6, 300 MHz): δ = 2.49 (s, 3H, CH3), 3.46 (s, 2H, H2O), 2.27–2.33 (s, 6H, -N (CH3)2), 9.67 (S, 1H, = CH-Ar), 7.14–7.97 (m, 4H, Ar–H).&lt;/p>&lt;!>&lt;p>Dark brown; Yield: 74%; m.p.: 125 οC; M.Wt: 760.59; Elemental analysis for AgC30H30N5O8S2: found, C, 47.22; H, 3.91; N, 9.15; Ag, 14.13; Calcd, C, 47.37; H, 3.98; N, 9.21; Ag, 14.18; Λm = 135.50 S cm2 mol−1; IR (KBr, v, cm−1): 3444 m, br (OH), 1685 m (C = O), 1527vw cm−1 (C = N), 1099 m cm−1(C = S), 748w and 792w (M–N). 1H NMR (DMSO-d6, 300 MHz): δ = 2.49 (s, 3H, CH3), 3.37 (s, 2H, H2O), 8.64 (S, 1H, = CH-Ar), 7.20–7.94 (d, 3H, Ar–H).&lt;/p>&lt;!>&lt;p>Dark brown; Yield: 90%; m.p.: 150 οC; M.Wt: 790.57; Elemental analysis for AgC36H36N5O9: found, C, 54.47; H, 4.11; N, 8.80; Ag, 13.60; Calcd, C, 54.69; H, 4.59; N, 8.86; Ag, 13.64; Λm = 114.52 S cm2 mol−1; IR (KBr, v, cm−1): 3444 (OH), 1678 (C = O), 1520 cm−1 (C = N), 759w and 779w (M–N). 1H NMR (DMSO-d6, 300 MHz): δ = 2.33 (s, 3H, CH3), 3.31 (s, 3H, -OCH3), 8.42 (S, 1H, = CH-Ar), 7.18–7.46 (d, 3H, Ar–H).&lt;/p>&lt;!>&lt;p>Additional file 1: Table S1. UV-Vis. spectral data of the free ligand L1, L2, L3 and their Ag(I)-complexes. Table S2. Selected 1H NMR data of L1, L2, L3 and its diamagnetic complexes. Fig. S1. TGA and DTG diagrams for a L1, b [Ag(L1)2(H2O)2]NO3, c, L2
d [Ag(L2)2(H2O)2]NO3.H2O, e L3 and f [Ag(L3)2(H2O)2]NO3. Fig. S2. The diagrams of kinetic parameters of L1, [Ag(L1)2(H2O)2]NO3, L2, [Ag(L2)2(H2O)2]NO3.H2O, L3 and [Ag(L3)2(H2O)2]NO3using Coats-Redfern (CR) and Horowitz-Metzger (HM) equations. Scheme S1. Fragmentation pattern of [Ag(L1)2(H2O)2]NO3. Scheme S2. Fragmentation pattern of [Ag(L2)2(H2O)2]NO3.H2O. Scheme S3. Fragmentation pattern of [Ag(L3)2(H2O)2]NO3&lt;/p>&lt;p>Ethanol&lt;/p>&lt;p>Nuclear magnetic resonance&lt;/p>&lt;p>Infrared radiation&lt;/p>&lt;p>Dimethyl sulfoxide&lt;/p>&lt;p>Minimum inhibation concentrations&lt;/p>&lt;p>Publisher's Note&lt;/p>&lt;p>Springer Nature remains neutral with regard to jurisdictional claims in published maps and institutional affiliations.&lt;/p>&lt;!>&lt;p>Supplementary information accompanies this paper at 10.1186/s13065-020-00723-0.&lt;/p></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">PubMed Open Access</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="5"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Evaluation of an Artificial Neural Network Retention Index Model for Chemical Structure Identification in Nontargeted Metabolomics</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Liquid chromatography coupled with electrospray ionization tandem mass spectrometry (LC-ESI-MS/MS) is a major analytical technique used for nontargeted identification of metabolites in biological fluids. Typically, in LC-ESI-MS/MS based database assisted structure elucidation pipelines, the exact mass of an unknown compound is used to mine a chemical structure database to acquire an initial set of possible candidates. Subsequent matching of the collision induced dissociation (CID) spectrum of the unknown to the CID spectra of candidate structures facilitates identification. However, this approach often fails because of the large numbers of potential candidates (i.e., false positives) for which CID spectra are not available. To overcome this problem, CID fragmentation predication programs have been developed, but these also have limited success if large numbers of isomers with similar CID spectra are present in the candidate set. In this study, we investigated the use of a retention index (RI) predictive model as an orthogonal method to help improve identification rates. The model was used to eliminate candidate structures whose predicted RI values differed significantly from the experimentally determined RI value of the unknown compound. We tested this approach using a set of ninety-one endogenous metabolites and four in silico CID fragmentation algorithms: CFM-ID, CSI:FingerID, Mass Frontier, and MetFrag. Candidate sets obtained from PubChem and the Human Metabolite Database (HMDB) were ranked with and without RI filtering followed by in silico spectral matching. Upon RI filtering, 12 of the ninety-one metabolites were eliminated from their respective candidate sets, i.e., were scored incorrectly as negatives. For the remaining seventy-nine compounds, we show that RI filtering eliminated an average of 58% from PubChem candidate sets. This resulted in an approximately 2-fold improvement in average rankings when using CFM-ID, Mass Frontier, and MetFrag. In addition, RI filtering slightly increased the occurrence of number one rankings for all 4 fragmentation algorithms. However, RI filtering did not significantly improve average rankings when HMDB was used as the candidate database, nor did it significantly improve average rankings when using CSI:FingerID. Overall, we show that the current RI model incorrectly eliminated more true positives (12) than were expected (4\xe2\x80\x935) on the basis of the filtering method. However, it slightly improved the number of correct first place rankings and improved overall average rankings when using CFM-ID, Mass Frontier, and MetFrag.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">evaluation_of_an_artificial_neural_network_retention_index_model_for_chemical_structure_identificati</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">5,610</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">376</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">14.920213</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">&lt;!>Reagents and Chemicals.&lt;!>Sample Preparation.&lt;!>UPLC-ESI-MS/MS.&lt;!>Retention Index Measurements.&lt;!>Compound Set Diversity.&lt;!>Data Analysis.&lt;!>Candidate Structures.&lt;!>Preprocessing of Candidates.&lt;!>In Silico RI Prediction and Filtering.&lt;!>In Silico Predictive Fragmenters.&lt;!>Reduction in Candidate Set Sizes Upon RI Filtering.&lt;!>Predictive Fragmenters Performance with and without RI Filtering.&lt;!>Fragmentation Tree Filter.&lt;!>Evaluation of RI Model Filter Windows.&lt;!>CONCLUSIONS</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">&lt;p>The composition and concentrations of small molecule metabolites (20–1000 Da) in living organisms frequently change over time and represent the biochemical phenotype of an individual. These changes can be due to multiple factors including diet, time of day, environmental exposures, disease, drug exposures, genetic manipulation, gender, and age.1,2 Nontargeted metabolomics is the unbiased quantification and identification of these metabolites in biological samples.3&lt;/p>&lt;p>In nontargeted metabolomics, researchers often utilize liquid chromatography coupled with electrospray ionization mass spectrometry (LC-ESI-MS) as the major analytical technique to separate and accurately measure the precursor masses of thousands of metabolites present in biological samples.4 However, to elucidate the chemical structure of these compounds, tandem mass spectrometry (LC-ESI-MS/MS) is often used.5 In LC-ESI-MS/MS, an experimental collision induced dissociation (CID) spectrum of an isolated precursor ion is used as a fingerprint in matching against a collection of reference CID spectra of known compounds in spectral libraries (e.g., MassBank, Metlin).6–8 Unfortunately, this approach often fails due to the dependency of CID spectral profiles on experimental conditions and lack of coverage of chemical space that pertains to endogenous human metabolites within existing spectral libraries.6,9–11&lt;/p>&lt;p>To overcome this limitation, computational fragmentation software (predictive fragmenters) has been developed with the aim of predicting experimental tandem mass spectral (MS/MS) profiles and the chemical structure of ensuing predictive fragments10,12 Working principles of these predictive fragmenters are described elsewhere;13 thus, only a brief overview is given here.&lt;/p>&lt;p>Commercial predictive fragmenters such as ACD/MS fragmenter (Advanced Chemistry Development Laboratories, www.acdlabs.com) and MassFroniter (Thermo Scientific, www.thermoscientific.com) rely on general ionization fragmentation and rearrangement rules along with fragmentation schemes collected from the literature to predict the chemical structures of energy induced fragment ions generated from precusor ions of specific structural composition (https://tools.thermofisher.com).14 These tools are extremely helpful in aiding manual spectral interpretation, which can be laborintensive.15 However, cost, lack of automated candidate retrieval, and ranking protocols limit their use in high-throughput metabolomics pipelines.16,17 In contrast, predictive fragmenters such as MetFrag and MAGMa attempt to explain ion peaks in an experimental CID spectrum by systematic dissociation of all the bonds in a given molecule. In other words, these predictive fragmenters compute all possible fragments of a molecule and then compare the mass of these fragments with the m/z values of fragments in an experimental CID spectrum. Compounds are ranked by assigning a score which is a function of ion peak intensity and the number of peaks explained in the experimental CID spectrum, bond dissociation energies, and neutral losses to account for rearrangements.10,18,19 Free availability, automated workflows, and faster processing times makes these programs popular among metabolomic researchers.20&lt;/p>&lt;p>Machine learning (ML) is one of the most rapidly growing areas in computer science.21 ML involves the development of computer algorithms that learn from example data or past experience to solve or predict the outcome of an unfamiliar problem.22 Predictive fragmenters such as CFM-ID and CSI:FingerID have been developed on the basis of ML paradigms.20,23 CFM-ID23 utilizes a stochastic, generative Markov model trained using the CID mass spectral profiles of approximately 3500 metabolites randomly chosen from the Metlin database.23 This method allows for the prediction of CID spectral profiles and can also be used to rank candidates based solely on the similarity between predicted and experimental spectra.23 CSI:FingerID is based on fragmentation trees and kernel based support vector machines trained to predict molecular structural features from CID spectra.20,24,25 The predicted set of molecular features (or a fingerprint) is used to rank candidates based on maximum likelihood considerations and Platt probabilities to refine the fingerprint similarity scoring.20 It is important to note that training data used in ML methods have a significant influence on identification quality. Generally, ML based predictive fragmenters outperform other competing predictive fragmenters but have longer processing times.13,20,26&lt;/p>&lt;p>Upon completion of an ESI-LC-MS/MS analysis, measured experimental features such as monoisotopic mass (MIM), retention time, and CID spectra are often utilized by researchers to elucidate the chemical structure of an unknown compound (peak) of interest. In database assisted structure elucidation pipelines, the first step is the acquisition of candidate structures for the unknown by matching the measured MIM to compounds in an all-purpose chemical database such as PubChem or ChemSpider or specialized biological databases such as the Human Metabolite Database (HMDB) or Kyoto Encyclopedia of Genes and Genomes (KEGG).13,27,28 According to the critical assessment of small molecule identification (CASMI) contests,26,29 the use of specialized databases improves the chances of correctly identifying the unknown structure. However, a major disadvantage of this approach is the inherent incompleteness of such databases. If the unknown compound is not contained in the database, it cannot be correctly identified.30 Thus, it can be argued that there are advantages in mining large chemical databases such as PubChem (which currently contains more than 90 million compounds) or ChemSpider (which currently contains more than 59 million compounds) because of their much larger size and the inclusion of more diverse compound classes. Nonetheless, mining such databases may also result in large candidate sets.31,32 For example, searching for 5-hydroxytryptophan (MIM = 220.0848 Da, ±5 ppm) using the MetFrag web interface (http://msbi.ipb-halle.de/MetFragBeta/) yields 8223 candidates from PubChem and 3777 from Chemspider. The same search resulted in 3 candidates from HMDB and 4 candidates from KEGG.&lt;/p>&lt;p>Generating the candidate list from large databases increases the likelihood that the unknown will be included in the candidate list but also dramatically increases the number of false positives. To address this problem, we have developed a software package called MolFind,33 which relies on a set of orthogonal experimental features acquired from LC-ESI-MS/MS experiments. These include Retention Index (RI), Drift Index (DI), and Ecom50 (collision gas normalized energy required to fragment 50% of precursor ions). For each experimental feature, MolFind eliminates false positives by comparing the experimental value of the unknown to a value predicted for each candidate compound using a computational model. A candiate compound is excluded when a predicted value for the candidate is substantially different from the experimental value of the unknown.33 In previous studies, when both RI and Ecom50 filters were applied concurrently with MetFrag, ranking of the correct compound improved from 142 to 102 on average (over 35 sets).33 More importantly, it was suggested that enhancements in the accuracy of the RI and Ecom50 models could lead to removal of 87.2% of candidates on average, attaining a potential increase in an average MetFrag ranking of 15.5.33&lt;/p>&lt;p>The robustness of modeling approaches in predicting experimental features is heavily dependent on the training set. The models used in our earlier study30 were far from optimal, as the Ecom50 model and the RI models were trained using 54 and 400 compounds, respectively, having 99.5% confidence intervals of 2.1 eV and 114 RI units (RIU), respectively.30 Recently, we have improved our RI model by training with a diverse set of synthetic chemicals (1955 in total) covering diverse chemical classes representative of endogenous human metabolites.34 Confirmed endogenous human metabolites were deliberately excluded from the model data set in the previous study. A total of 202 confirmed metabolites were set aside as an independent validation set to test if a model based exclusively on relatively simple synthetic compounds could be used to make predictions for more complex metabolites.34 For the model used in this present study, the 202 independent validation compounds were reintroduced and the model was rebuilt using the same descriptors and learning protocol as the previous study resulting in a model based on 2157 compounds. This was done to take advantage of the available data on confirmed metabolites and expand the model applicability domain.&lt;/p>&lt;p>As an extension to our previous work, the present study investigates the improvement in identification quality by enrichment of candidate sets using the expanded 2157 compoound RI model in conjunction with four different fragmentation algorithms: CFM-ID, CSI-FingerID, Mass Frontier, and MetFrag. Candidate compounds were taken from both a large (PubChem) and small (HMDB) database.&lt;/p>&lt;!>&lt;p>Acetonitrile (HPLC, gradient grade) and methanol (HPLC grade) were purchased from Sigma-Aldrich (St Louis, MO, USA). Water (18.2 MΩ·cm) used for the UPLC mobile phase and sample preparation. Reagent grade water was generated on a Burnstead Nanopure Diamond system (Thermo Scientific, Ward Hill, MA, USA). Heptafluorobutyric acid (HPLC grade) was purchased from Thermo Fisher Scientific Chemicals Inc. (Ward Hill, MA, USA). n-Propionamide, n-butanamide, and n-hexanamide were ordered from Aldrich (St Louis, MO, USA). n-Pentanamide was ordered from MP Biochemicals, LLC (Solon, OH, USA). A series of n-C7–C14 amides were synthesized as described in Supporting Information SI–1. The 91 test compounds and the controls used in the study were purchased from various sources, and the vendor information is summarized in Table S2. HPLC grade formic acid (98–100%) was purchased from EMD Millipore Corporation (Billerica, MA, USA).&lt;/p>&lt;!>&lt;p>Two different approaches were followed for the sample preparation. As the chemicals ordered from IROA technologies were contained in plates of polypropylene wells containing 5 μg of each chemical, 100 μL of solvent (0.1% formic acid in water, 0.05% formic acid in water/methanol (1:1) (v/v), or methanol) based on the XLOGP3 (taken from PubChem) value was added to each well. The plates were covered with sealing tape to prevent evaporation. Dissolution was achieved by shaking wells on an Innova 2100 platform shaker (New Brunswick, CT, USA) for 45 min. Finally, the dissolved chemicals were transferred to 2 mL HPLC vials with micro volume glass inserts (Thermo Fisher Scientific, Ward Hill, MA, USA), sealed with Teflon septum caps, and used directly for UPLC analysis. Stock solutions of all other chemicals were prepared at 1–10 μmol/mL concentrations in the appropriate solvent based on the analyte's XLOGP3 value as described above. The prepared stock solutions were further diluted at appropriate concentrations and used for the UPLC analysis.&lt;/p>&lt;!>&lt;p>Retention index values were measured on a Zorbax, SB-C18, 2.1 mm × 150 mm, 1.8 μm column (Agilent Technologies, Santa Clara, CA, USA) using an Acquity UPLC liquid chromatographic system (Waters, Milford, MA, USA). Solvent A was 0.766 mM heptafluoroacetic acid (HFBA) in water, and solvent B was 0.766 mM HFBA in 10% water/acetonitrile (v/v). Compounds were eluted from the column using a solvent program consisting of a 4 min isocratic hold of 2% solvent B followed by a 20 min linear gradient to 100% solvent B and a 5 min isocratic hold at a flow rate of 388 μL/min. The RI model used in this study was trained and validated using retention data for compounds analyzed on an Agilent 1100 capillary HPLC system (Agilent Technologies, Santa Clara, CA, USA). Thus, the protocols used to transfer and validate the LC method to the Acquity UPLC system are summerized in Supporting Information SI–2. The outlet of the UPLC system was connected to the electrospray (ESI) ionization source of a Synapt G2-Si mass spectrometer (Waters, Milford, MA, USA) operating in the positive ion mode. A solution of leucine enkephalin (556.2771 Da, 400 pg/uL) in 0.1% (v/v) formic acid–methanol/water (1:1) was infused as the lock mass reference compound at a flow rate of 5 μL/min. The retention times of the test compounds were measured in duplicate using detection parameters described in Supporting Information SI–2. The mass range of the detector was set to 20–1000 Da. CID was carried out with nitrogen as the collision gas, and the collision energy was varied from 0 to 30 eV (30–60 eV if required) in incremental steps of 2 eV at a scan rate of 12.5 scans/s.&lt;/p>&lt;!>&lt;p>RI values were measured on the basis of a method developed by Hall et al.35 At the beginning and the end of each run, 1 μL of a homologues series of n-C3–C14 amides was injected on the described UPLC system. The average retention times (RT) of the individual n-amides were used as the calibration reference for calculating RI values for the test compounds. The RI of each n-amide was defined as 100 times the number of carbon atoms. Owing to the linear relationships between RI vs log of RT in the isocratic part and RT vs RI in the gradient part of the solvent program, RI values of compounds eluted during isocratic and gradient parts were calculated by the following equations, respectively
(1)RIisocratic=(logTx−logTz)100/(logTz+1−logTz)+100z
(2)RIgradient=(Tx−Tz)100/(Tz+1−Tz)+100z
where Tx is the retention time of the analyte; Tz is the retention time of the n-amides eluting just before the analyte; Tz+1 is the retention time of the n-amides eluting just after the analyte.&lt;/p>&lt;!>&lt;p>A set of ninety-one endogenous metabolites, not included in the training set of the RI model, were selected for the study. The set consisted of a variety of chemical classes including alkaloids, amines, benzene and substituted derivatives, benzenoids, carbohydrates and conjugates, carboxylic acids, fatty acyls, flavin nucleotides, imidazopyrimidines, indoles, lipids and lipid like molecules, morphinans, organic acids, organic carbonic acids, organic nitrogen compounds, organic phosphonic acids, organoheterocyclic compounds, pteridines and derivatives, purine nucleosides, pyridines, quinolines, sphingolipids, steroids, stilbenes, and tetrahydroisoquinolines. Four of the compounds had an overall charge of +1 while the rest were neutral. The XLOGP3 values (predicted octanol–water partition coefficient) of the compound set varied from −5.0 to 8.5 (Figure 1) with a standard deviation of 3.4. The MIMs of the compounds varied from 105 to 785 Da with a standard deviation of 99 Da. Structural details for each test compound can be found in Table S3.&lt;/p>&lt;!>&lt;p>Retention and mass spectral data were processed using Masslynx version 4.1 (Waters, Milford, MA, USA). The experimental survival yields of precursor ions at each collision energy were calculated using eq 3 on the basis of a code written in Python version 3 (http://www.python.org).&lt;/p>&lt;!>&lt;p>Candidate structures were downloaded from PubChem and HMDB by matching the experimental MIM within a relative mass error of ±15 ppm.36 MIMs were calculated by averaging scans with counts higher than or equal to 25% of the respective precursor ion peak apex in the CID spectrum acquired at an energy which resulted in a precursor survival yield closest to 20%.36 The relative mass error (15 ppm) was calculated at the 3-sigma limit when comparing actual to experimental MIMs. PubChem was queried using an in-house program written in python version 3 using underlying cheminformatics functions of Rdkit (http://rdkit.org/)37 and power user gateway (PUG) service (https://pubchem.ncbi.nlm.nih.gov/pug/). Downloaded candidate sets of compounds were saved in 91 separate structure-data (SD) files. A snapshot of HMDB database was downloaded as an SD file, and http://www.hmdb.ca/downloads was used to acquire candidate compound structures. Resulting candidate sets were saved in 91 separate SD files.&lt;/p>&lt;!>&lt;p>Compounds in candidate sets that contained33 salts, had disconnections, contained heavy isotopes, had an overall charge (this filter was not applied to the candidate sets of the quaternary ammonium ions), contained only carbon and hydrogen, or were duplicate stereoisomers were removed prior to processing. Compounds with elements other than CHNOPS were also removed.&lt;/p>&lt;!>&lt;p>In silico RI prediction of the remaining candidates was carried out using topological descriptors calculated by winMolconn (version 2.1)38 and newly trained parameters. The learning process used to develop the RI model has been described previously.34 Briefly, a 4 × 10 × 10 artificial neural network ensemble model was built on RI data for 2157 compounds (1955 commercial synthetic compounds and 202 confirmed human endogenous metabolites) measured according to the protocol described above. The model was trained according to the learning method described in the previous publication34 with RPROP40 back propagation on a network architecture of 47 input neurons and one hidden layer of 23 neurons.&lt;/p>&lt;p>The significant difference between the model used for this study and the previous publication is that confirmed human endogenous metabolites were deliberately excluded from the model data set in the previous publication. This was done in part to determine if a model based exclusively on relatively simple synthetic compounds could be used to make reasonable predictions for more complex human metabolites. To facilitate this, 202 confirmed human metabolites were set aside as an independent validation set. For this study, the independent validation set was reintroduced to the model data set to. After the additional data was added, the model was rebuilt using the same descriptors and learning method as the previous study. The metabolite data was reintroduced because the goal of this present study was to test RI model performance in the context of the MolFind algorithm, not to test if a model based on synthetic compounds could be used to predict the RI of metabolites. The addition of the confirmed human metabolites makes use of all available data so as to achieve the maximal applicability domain.&lt;/p>&lt;p>RI predictions were made for compounds in the PubChem and HMDB candidate lists corresponding to each of the 91 unknowns. Candidates with predicted RI values that deviated more than a threshold value were eliminated. The threshold windows were chosen on the basis of an algorithm that utilized the experimental RI value of the "unknown" and the similarity of each candidate to model data. Similarity was evaluated using a partial molecular fingerprint encoded in a bit key. The filter windows can be found in Table 4. The bit key similarity approach and algorithm for derivation of the filter windows are discussed in Supporting Information SI–4. The resulting RI-filtered candidate sets were saved separately.&lt;/p>&lt;!>&lt;p>CFM-ID, CSI:FingerID, Mass Frontier, and MetFrag were used without modifications to rank candidate sets resulting from pre- and RI filtering. MAGMa19 was not used in the study as it does not support processing compounds with an overall charge of +1 (adduct type [M+]). Relative and absolute mass errors of 15.0 ppm and 0.01 Da were used to annotate fragments against peaks in respective experimental CID spectra with relative ion intensities closest to 20% survival yield (eq 3).&lt;/p>&lt;p>For single energy CFM-ID (version 2.0), the pretrained model params_se_cfm, having parameter file param_output.-log, was used (https://sourceforge.net/p/cfm-id/wiki/Home/). The experimental CID spectrum was repeated in all three energies (low, medium, high) such that CFM-ID assigned an average Jaccard score (default method) in ranking candidates.23&lt;/p>&lt;p>CSI:FingerID (version 3.5) was run with instrument mode set to "qtof" and setting all the other parameters to defaults. Candidates with molecular formulas having tree scores lower than the cutoff of &lt;75% of the highest scoring tree were eliminated. The remaining candidates were processed with FingerID.&lt;/p>&lt;p>Mass Frontier (version 7.0.5.9) was run in batch mode using general fragmentation rules. Parameter settings were as follows : ionization method = protonation, maximum number of reaction steps = 5, reaction limit = 10 000, and mass range = 20–1000 Da. As Mass Frontier lacked functionality in automated candidate ranking, a python program was written to calculate number of peaks matched in the CID spectrum by Mass Frontier generated fragments (within either 15 ppm or 0.01 Da). Candidates were ranked considering the number of peaks matched in the experimental CID spectrum.&lt;/p>&lt;p>The command line version of MetFrag (version 2.3.1) was downloaded and used http://c-ruttkies.github.io/MetFrag/. None of the pre- or postprocessing filters (MetFragPreProcessingCandidateFilter and MetFragPostProcessingCandidateFilter) were applied. Default values for all other parameters were used.27&lt;/p>&lt;p>It is important to note that the objective of the study was to investigate the performance of the RI model in eliminating false positives contained in candidate sets and the subsequent improvement in predictive fragmenter performance. Our intent was not to evaluate or benchmark the 4 predictive fragmenters that we used (as done in previous CASMI contests).26 Researchers are encouraged to use this study as a guide to test the RI model with any predictive fragmenter of their choice.&lt;/p>&lt;!>&lt;p>The aim of this study was to determine whether we could improve identification rankings by eliminating false positives using an RI filter prior to CID spectral matching. CFM-ID, CSI:FingerID, Mass Frontier, and MetFrag were used for spectral matching using data sets downloaded from PubChem and HMDB for 91 endogenous metabolites treated as "unknowns". As with all approaches such as this, we face the problem of false negatives; i.e., the RI model will incorrectly eliminate a certain percentage of correct candidates. In this case, 12 of the ninety-one compounds evaluated were eliminated from their respective candidate sets after RI filtering. A sensitivity (true positive rate) of 95% was anticipated given the method used to set the filter ranges. This would equate to 4–5 expected false negative predictions. The lower sensitivity observed (79/91 = 87%) was unexpected as the new RI model was anticipated to outperform its predecessor which also had a sensitivity of approximately 87%.33 The model comes close to the expected sensitivity for compounds with RI values &lt;900 (92% sensitivity, with six false negatives, Figure 2).&lt;/p>&lt;p>Nevertheless, for the remaining 79 compounds, the RI filter eliminated 58% of false positives from candidate sets acquired from the PubChem database (Figure 3). This was an improvement over the previous model,33 which eliminated approximately 21% of compounds. Likewise, approximately 32% of compounds were removed from candidate sets acquired from HMDB (Figure 3). There were only 22 of the 79 "unknowns" where the final number of compounds remaining in the PubChem set after RI filtering was &lt;175. A complete list of these results is given in Tables S4, S5, and S6.&lt;/p>&lt;!>&lt;p>The performance of each predictive fragmenter was evaluated using the method described by Allen et al.23 CSI:FingerID scored 53 correct number one ranks (the known compound ranked as the best match after predictive fragmenter evaluation) when ranked using candidate sets acquired from PubChem database. After the RI filter, the number of correct number one ranks increased to 56. Mass Frontier scored 13 correct number one ranks which improved to 16 after RI filtering. Both CFM-ID and MetFrag had three correct number one ranked sets which increased to seven and six, respectively, after RI filtering. CSI:FingerID and Mass Frontier both scored 66 correct number one ranks with candidate sets acquired from the HMDB database. Upon applications of the RI filter, the number of correct number one rankings increased to 68 and 71, respectively. CFM-ID achieved 62 and 64 correct number one ranks from the HMDB database, while for MetFrag the number of correct number one ranks were 54 and 59 before and after the RI filter. Although these improvements in number one rankings are small, when taken together, they are statistically significant (paired t test; p &lt; 0.001). Thus, our results suggest that RI filtering slightly improved a predictive fragmenter's ability to rank an unknown as the correct number one ranked candidate. Interestingly, 37 of the 53 compounds that were correctly ranked number one by CSI:FingerID were compounds included in the algorithm's original training set.20 This likely causes a significant performance skew in favor of CSI:FingerID and is consistent with the results of the 2016 CASMI contest where ML methods were shown to perform better when unknowns were included in their training sets.26 In contrast, for CFM-ID, only 18 of the 79 compounds studied were in its training set.&lt;/p>&lt;p>In addition to an evaluation of our RI model based on correct number one rankings, we also compared the results based on overall average rankings. The fragmentation tree filter used in CSI:FingerID eliminated glycocholic acid from its candidate set.20 Thus, Table 1 summarizes overall average rankings of the remaining 78 candidate sets. As seen (Table 1), following RI filtering, the performance of CFM-ID, Mass Frontier, and MetFrag was improved by approximately 2-fold (1.7-, 1.8-, and 1.9-fold, respectively; p &lt; 0.05) when candidate sets acquired from PubChem database were used. This is consistent with the approximate 2-fold reduction in candidates following RI filtering (Figure 3a). There was no significant improvement in the performance of CSI:FingerID upon RI filtering using candidate sets from PubChem, and the RI filter provided no significant improvement for any of the fragmenters when using candidate sets from HMDB. It is important to note that the SD values in Table 1 are relatively large suggesting that the average ranking results are markedly dependent on the structural characteristics of the compounds in each candidate list.&lt;/p>&lt;p>In our previous work,33 we used multiple orthologous experimental features that can be measured by LC-MS (such as RI) in order to eliminate false positive from candidate sets. The RI model used in this study was based on quantitative structure property relationships (QSPR) using a set of 2D structural descriptors.30 Matching the experimental and predicted RI values (considering the error of the model) enabled us to eliminate a large percentage of interfering false positives (Table 2). To better understand how RI filtering improves predictive fragmenter ranking, let us consider dihydrocapsaicin as an example of where the method performed well (Table 2) using MetFrag and candidates taken from PubChem. Dihydrocapsaicin was ranked as the fourth best match by MetFrag (Table 2). Moreover, for compounds a and b, MetFrag successfully annotated five peaks in the experimental CID spectrum of dihydrocapsaicin, while only detecting four peaks for compounds c and d. Thus, for this example, CID peak matching alone did not result in the correct ranking of dihydrocapsaicin (d) using MetFrag. However, the predicted RI values of compounds a, b, and c were sufficiently different from the experimental value so that they were excluded from the candidate set, permitting the ranking of d to become number one.&lt;/p>&lt;!>&lt;p>CSI:FingerID, developed by Dührkop et al., uses fragmentation trees (FT) to eliminate false positives contained in candidate sets prior to FingerID ranking.20 The theoretical and implementation details of FT are given elsewhere; however, a brief description is provided here.24 The first step in the CSI:FingerID algorithm is to generate all possible candidate molecular formulas of the "unknown" (based on precursor mass) and use them to generate fragmentation trees. Each tree is assigned a score based on how well each explains the experimental CID spectrum. Next, candidate molecular formulas having tree scores less than 75% of the top scored tree are excluded. Finally, the remaining formulas are used to eliminate false positives (i.e., candidate structures having molecular formulas other than the >75% scored trees) from the candidate set (e.g., acquired from PubChem). Therefore, we investigated the effect of tree filtering on the performance of CFM-ID, Mass Frontier, and MetFrag (Table 3), in addition to CSI:FingerID. The FT filter eliminated on average approximately 28% (1710 compounds) of false positives in candidate sets acquired from PubChem. It eliminated glycocholic acid from its respective candidate set (false negative). As shown in Table 3, the average performance of CFM-ID, MetFrag, and Mass Frontier improved following FT filtering. However, overall average performance was lower when compared to the RI filter (Table 1). Similar to the RI filter, applying the FT filter on HMDB candidate sets resulted in no increased performance for any of the predictive fragmenters studied.&lt;/p>&lt;!>&lt;p>As mentioned, 12 of the ninety-one compounds were eliminated from their respective candidate sets upon RI filter enrichment. We identify this as a shortcoming of this technique in that only 4–5 false eliminations were anticipated on the basis of the method used to set the filter window. The magnitude of the filter window for each candidate was based on the structural similarity of each candidate to the RI model data and the experimental RI of the "unknown". A compound was defined as "similar" (high bit key (HBK)) when three or more compounds in the model data had the same bit key value as the compound being predicted (bit key match). This means that at least three compounds in the model data have the same combination of heteroatomic structure features as the compound being predicted. Compounds with two or fewer data set compounds with a matching bit key were considered "not similar" (low bit key (LBK)). The bit key similarity metric is described in detail in Supporting Information SI–4. A filter window of ±2SE (validation standard error) was used when a candidate was similar to model data, and a window of ±3SE was used when a candidate was not similar. Further, an SE value of 48.5 RIU was used when the compound's RI value was 300–850 (data range where the model performed best), and a larger SE value of 62 RIU was used when the compound's RI was 851–1400 (data range where the model preformed least well). This method (termed the "bit key and range" approach) and the metrics used to assess similarity are discussed in Supporting Information SI–4. This method was suggested by the independent validation results of a previous study.34&lt;/p>&lt;p>According to the similarity metrics, only four of the 12 eliminated compounds had low structural similaritiy to the model data (Table S6). This raises the question as to whether using the described approach to set the filter window gives optimal results. In order to evaluate this, additional statistics were calculated using three other approaches for setting the filter window: SE (standard error), bit key, and range. The SE method used a filter window of ±2SE based on the validation SE of the entire data set (52.4 RIU) and used the 95% cutoff values where the "unknown" RI was outside the RI range of the reference standards (&lt;RI 300 or >RI 1400). The bit key method used a filter window of ±2SE (52.4 RIU) when a candidate was similar to the model's data and a filter window of ±3SE (78.6 RI Units) when a candidate was not similar. For "unknown" RI values &lt;300 or >1400, the bit key method used the 95% cutoff when the candidate was similar and the 98.5% cutoff when the candidate was not similar. The range method used a filter window of ±2SE (48.5 RIU) when the "unknown" RI was 300–850 and a filter window of ±2SE (62.0 RIU) when the "unknown" RI was 851–1400. The range method used the 95% cutoff values where the "unknown" RI was &lt;300 or >1400. Table 4 gives the filter window values used for all 4 approaches along with the sensitivity and specificity for each.&lt;/p>&lt;p>The highest sensitivity is found for the combined bit key and range approach at 87%. In contrast, the SE and range approaches have sensitivities of only 80%. As expected, the use of wider filter windows when screening some compounds results in a loss of specificity for the bit key and range approach. Both the SE and range approaches removed more false positives. The differences, however, are small at 1–3%, and since the elimination of an unknown from the candidate list results in failure, it is likely necessary to accept the reduced specificity in favor of greater sensitivity.&lt;/p>&lt;p>An analysis of Figure 2 indicates that, by far, sensitivity is worse where the RI of the unknown falls between 950 and 1120 RIU. Sensitivity is only 38% with false negative classifications for 5 out of the 8 compounds in this range. All 5 false negative compounds are significantly under-predicted. It may be of interest that 4 out the 5 compounds are lipid like with a hydrocarbon tail of at least 15 carbons. All 4 of these lipids are also predicted to have a charge of +1 at the pH of the mobile phase (pH 2.5). It is unclear if these structural characteristics are responsible for the poor prediction. The sample size in this RI range is small, but the poor performance of the mode in this range suggests that this approach would not be recommended in cases where the RI of the unknown falls in the range of 900–1150.&lt;/p>&lt;p>An analysis of Figure 2 also indicates that there are several close misses in the "unknowns" that were eliminated from their respective candidate lists. Two "unknowns" are within 20 RIU of being retained and 3 more are within 45. Boundary effects are anticipated when cutoff values are used and cannot be entirely avoided. Even so, given the failure of the method when the unknown is eliminated from its candidate list, some consideration of larger filter windows should be made. Larger filter windows will likely reduce the specifcity of the RI filter, but the results of this study suggest that the reduction could be small. An increase in the width of the filter range by 1/2SE increases the sensitivity to 90% while decreasing the specificity to 53%. Ultimately, the study showed that the RI model improved CID spectral rankings, but the overall level of improvement is modest. Further increases in model accuracy could be made by increasing the number and diversity of the training set to account for the current 3:1 skew toward the lower RI range, including additional relevant descriptors (such as 3D structural information) and QSAR-feature engineering. However, it is important to note that inclusion of 3D descriptors was avoided in the current RI model, as obtaining an accurate global minimum of flexible compounds (e.g., lipids) in a computationally efficient way remains problematic in the field of computational chemistry.39 Our results would suggest that, as candidate lists increase in size, the number of structurally similar compounds also likely increases. Thus, distinguishing among these extremely similar structures, either by CID prediction algorithms or by RI models, becomes increasingly difficult. Clearly, additional orthologous structure identification methods would help in these situations. We are in the process of upgrading our existing Ecom50 model which we hope to use as an additional filter to further enrich acquired candidate sets. From such enrichment, we foresee a potential increase in predictive fragmenter performance due to further reduction in candidate sizes.33 Additionally, the inclusion of an Ecom50 filter would allow us to use broader RI filter windows enabling us to overcome some of the limitations shown here. In addition, future improvements in the sensitivity of NMR and/or IR spectroscopy methods would dramatically enhance our ability to identify unknown compounds in nontargeted metabolomics applications.&lt;/p>&lt;!>&lt;p>We show that RI based enrichment of large candidate sets prior to in silico predictive fragmenter ranking slightly improved the performance of three of the four predictive fragmenters studied. We have developed an approach termed "bit key and range" to define the filter windows, but in spite of this, the model eliminated 12 of the 91 "unknowns" from their respective candidate sets where only 4–5 eliminations were expected. In the remaining PubChem candidate sets, approximately 58% of the false positives were eliminated on average and overall average rankings were improved 2-fold. Further improvements in the RI model (e.g., 3D descriptors, additional training compounds) would increase its reliability in LC-ESI-MS/MS based database assisted structure elucidation pipelines.&lt;/p></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">PubMed Author Manuscript</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="6"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Direct Quantification of Cannabinoids and Cannabinoid Glucuronides in Whole Blood by Liquid Chromatography Tandem Mass Spectrometry</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">The first method for quantifying cannabinoids and cannabinoid glucuronides in whole blood by liquid chromatography-tandem mass spectrometry (LC-MS/MS) was developed and validated. Solid-phase extraction followed protein precipitation with acetonitrile. HPLC separation was achieved in 16 min via gradient elution. Electrospray ionization was utilized for cannabinoid detection; both positive (\xce\x949-tetrahydrocannabinol [THC], cannabinol [CBN]) and negative (11-hydroxy-THC [11-OH-THC], 11-nor-9-carboxy-THC [THCCOOH], cannabidiol [CBD], THC-glucuronide and THCCOOH glucuronide) polarity were employed with multiple reaction monitoring. Calibration by linear regression analysis utilized deuterium-labeled internal standards and a 1/x2 weighting factor, yielding R2 values > 0.997 for all analytes. Linearity ranged from 0.5\xe2\x80\x9350 \xce\xbcg/L (THC-glucuronide), 1.0\xe2\x80\x93100 \xce\xbcg/L (THC, 11-OH-THC, THCCOOH, CBD and CBN) and 5.0\xe2\x80\x93250 \xce\xbcg/L (THCCOOH-glucuronide). Imprecision was &lt; 10.5% CV, recovery was > 50.5% and bias within \xc2\xb1 13.1% of target for all analytes at three concentrations across the linear range. No carryover, endogenous or exogenous interferences were observed. This new analytical method should be useful for quantifying cannabinoids in whole blood and further investigating cannabinoid glucuronides as markers of recent cannabis intake.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">direct_quantification_of_cannabinoids_and_cannabinoid_glucuronides_in_whole_blood_by_liquid_chromato</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">4,081</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">169</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">24.147929</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Introduction&lt;!>Clinical Samples&lt;!>Instrumentation&lt;!>Reagents&lt;!>Preparation of Standard Solutions&lt;!>Sample Preparation&lt;!>Solid Phase Extraction&lt;!>Liquid Chromatography&lt;!>Mass Spectrometry&lt;!>Data Analysis&lt;!>Validation&lt;!>Results and Discussion&lt;!>Calibration and Validation&lt;!>Application of Method&lt;!>Conclusions</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">&lt;p>Cannabis use substantially impacts public safety, as many individuals drive or operate complex equipment soon after self-administration. The National Highway Traffic Safety Administration (NHTSA) reported that in 2007, 8.6% of nighttime drivers tested positive for cannabinoids in blood and/or oral fluid, a rate almost 4 times higher than the percentage of drunk drivers with a blood alcohol concentration ≥0.8 g/L [1]. While finding cannabinoids in blood or oral fluid does not necessarily imply impairment, windows of drug detection in these matrices are often short for occasional or moderate smokers [2–4], increasing impairment probability.&lt;/p>&lt;p>Δ9-tetrahydrocannabinol (THC) is the primary psychoactive component in cannabis and is metabolized via cytochrome P450 (CYP) 2C9 and 2C19 isozymes to several phase I metabolites, most prominently 11-hydroxy-THC (11-OH-THC) and 11-nor-9-carboxy-THC (THCCOOH) [5–6]. THC and its phase I metabolites also undergo UDP-glucuronosyltransferase-catalyzed phase II metabolism to form cannabinoid glucuronides in vivo [7–9], facilitating excretion. Currently, little is known about cannabinoid glucuronide pharmacological activity or detection windows following cannabis intake, although others hypothesized that these glucuronides could serve as markers of recent cannabis intake due to a shorter half-life in vivo [10–11]. Detection and quantification of these metabolites may provide scientific data permitting researchers, physicians and law enforcement personnel to document recent cannabis intake.&lt;/p>&lt;p>Analysis of glucuronides by gas chromatography-mass spectrometry (GC-MS) is difficult as chemical derivatization requirements and volatility issues preclude direct detection and quantification. Therefore, analytical procedures for cannabinoids in urine [12–14], blood [15], meconium [16–17] and oral fluid [18] typically include expensive and time-consuming alkaline and/or enzymatic glucuronide hydrolysis to liberate cannabinoids prior to extraction and GC-MS analysis. However, these hydrolyses introduce multiple confounding issues, including, but not limited to poor chromatography [15] and variable hydrolysis efficiencies of the ether- and ester-linked glucuronide species [15, 19–22].&lt;/p>&lt;p>To circumvent hydrolysis and facilitate direct quantification of phase II cannabinoid metabolites, sensitive liquid chromatography-tandem mass spectrometry (LC-MS/MS) methods are required. Yet few LC-MS/MS methods are available for cannabinoids in whole blood [23–25], likely resulting from higher limits of quantification (LOQ) than typically achieved by GC-MS. Additionally, to date, these published methods do not included glucuronide metabolites. Furthermore, development of glucuronide analytical methods is hampered by a lack of commercially available native and isotopically-labeled cannabinoid glucuronide standards.&lt;/p>&lt;p>To this end, we developed and validated the first sensitive and specific LC-MS/MS method for simultaneous detection of free and glucuronidated cannabinoids in human whole blood. This method is unique in that THC, THCCOOH and their glucuronides, 11-OH-THC, cannabidiol (CBD) and cannabinol (CBN) are simultaneously extracted and quantified in 16 min. While whole blood is frequently the specimen collected in driving under the influence of drugs (DUID) cases and other investigations, to our knowledge no studies directly investigated whole blood pharmacokinetics following smoked cannabis. Therefore, we will utilize this method to investigate in vitro cannabinoid stability, evaluate cannabinoid glucuronides as markers of recent cannabis intake and determine whole blood cannabinoid pharmacokinetics in order to provide a scientific database for researchers, clinicians and forensic toxicologists interpreting whole blood cannabinoid concentrations.&lt;/p>&lt;!>&lt;p>A healthy cannabis smoker provided written informed consent to participate in a study investigating cannabinoid pharmacokinetics, in vitro cannabinoid stability and novel markers of cannabis intake following a single smoked cannabis dose. The Institutional Review Board of the National Institute on Drug Abuse, National Institutes of Health approved this protocol. Cannabis cigarettes contained 6.8% THC (w/w) or approximately 56 mg THC and were smoked ad libitum over a 10 min period following an overnight stay on a secure residential unit. Whole blood was collected with sodium heparin 0.5 h prior to, 0.25 h after and 1.0 h after the start of cannabis smoking. Blood was transferred to polypropylene storage tubes and stored refrigerated until analysis within 24 h.&lt;/p>&lt;!>&lt;p>All experiments were performed on an AB Sciex 3200 Qtrap triple quadrupole mass spectrometer with a TurboV ESI source (AB Sciex, Foster City, CA). The mass spectrometer was interfaced with a Shimadzu UFLCxr system consisting of two LC-20ADXR pumps, a SIL-20ACXR autosampler, and a CTO-20AC column oven (Shimadzu Corporation, Columbia, MD). Evaporation under nitrogen was completed using a TurboVap LV evaporator from Zymark (Hopkinton, MA).&lt;/p>&lt;!>&lt;p>Standards and deuterated internal standards were purchased from Cerilliant (Round Rock, TX) except for THC-glucuronide that was from ElSohly Laboratories, Inc (Oxford, MS). Ammonium acetate, formic acid and acetonitrile (ACN) were obtained from Sigma-Aldrich (St. Louis, MO). Methanol was from Fisher Scientific (Fair Lawn, NJ). Ammonium hydroxide and glacial acetic acid were from Mallinckrodt Baker (Phillipsburg, NJ). Water was purified in house by an ELGA Purelab Ultra Analytic purifier (Siemens Water Technologies, Lowell, MA). All solvents were HPLC grade or better. 200-mg, 6-mL Bond Elut Plexa solid-phase extraction (SPE) cartridges were utilized for preparing samples (Agilent Technologies, Culver City, CA). Blank human whole blood was evaluated for absence of cannabinoids prior to use.&lt;/p>&lt;!>&lt;p>Individual stock solutions of 1.0 g/L THC, 11-OH-THC, THCCOOH, CBD and CBN, 100 mg/L THCCOOH-glucuronide and 10 mg/L THC-glucuronide were diluted with methanol to prepare calibration solutions. 11-OH-THC-glucuronide and di-glucuronide metabolites are not commercially available. A stock solution containing 10 mg/L of analytes other than THC-glucuronide was prepared in methanol and stored at −20°C. Dilutions of the stock solution (adding in THC-glucuronide) created calibrators at 0.5, 1.0, 2.0, 5.0, 10, 20, 50, 100, and 250μg/L when fortifying 25 μL of standard solution into 500 μL of blank human whole blood.&lt;/p>&lt;p>Quality control samples were prepared in methanol from different vials than utilized for preparing standards. Low-, medium-, and high-quality control samples were prepared across the linear dynamic range of the assay. Whole blood low-, medium-, and high-quality control target concentrations were: THC-glucuronide 1.5, 4.5, and 45 μg/L; 11-OH-THC, CBD, CBN, THC and THCCOOH 2.5, 7.5, and 75 μg/L; and THCCOOH-glucuronide 7.5, 22.5, and 225 μg/L, respectively. All quality control solutions were stored at −20°C.&lt;/p>&lt;p>Stock internal standard solution was prepared by diluting 100 mg/L solutions of THC-d3, 11-OH-THC-d3, THCCOOH-d9 and CBD-d3 1:10 with methanol and storing at −20°C. A 1:50 dilution of internal standard stock solution was prepared in methanol and 25 μL of the diluted solution was added to each 500 μL whole blood sample, providing a final internal standard concentration of 10 μg/L. Deuterated CBN, THCCOOH-glucuronide, and THC-glucuronide are not currently commercially available; THC-d3 was utilized for CBN quantification and THCCOOH-d9 for quantification of both glucuronides.&lt;/p>&lt;!>&lt;p>Blank blood (0.5 mL) was pipetted into a 10-mL conical polypropylene tube (Sarstedt, Newton, NC). 25 μL of internal standard and either 25 μL of standard or quality control solution were added. 25 μL blank methanol was added to authentic specimens. Ice-cold ACN (1.5 mL) was added drop-wise while vortexing. Tubes were capped and centrifuged (4000g, 4°C) for 5 min. Supernatants were decanted into clean tubes, 4.5 mL 0.2% NH4OH in de-ionized water (v/v) was added, and samples mixed immediately prior to SPE loading.&lt;/p>&lt;!>&lt;p>Extraction columns were conditioned with 2 mL methanol and 2 mL de-ionized water. Samples were decanted onto conditioned columns and loaded by gravity. Columns were washed with 2 mL 79:20:1 de-ionized water:acetonitrile:glacial acetic acid (v/v/v) and then dried under full vacuum (≥ 30 kPa) for 20 sec. Analytes were eluted with two separate 1.5 mL aliquots of 1% glacial acetic acid in ACN (v/v) under gravity. Vacuum was briefly applied after both aliquots were collected. Eluents were collected in a 10-mL conical polypropylene tube and dried under nitrogen at 42°C in a Zymark TurboVap evaporator. Samples were reconstituted in 150 μL of initial mobile phase (70:30 A:B), vortexed for 15 sec and transferred to 250 μL pulled-point glass inserts in autosampler vials.&lt;/p>&lt;!>&lt;p>Chromatographic separation was performed with an Ultra Biphenyl column (100 × 2.1 mm, 5μm) fitted with an Ultra II Biphenyl guard cartridge (10 × 2.0 mm) (Restek Corp, Malvern PA). The autosampler temperature was 4°C and column oven 40°C throughout analysis. The injection volume was 25 μL. Gradient elution was performed with (A) 10 mM ammonium acetate in water adjusted to pH 6.15 (± 0.05) with formic acid and (B) 15% methanol in acetonitrile (v/v) at a flow rate of 400 μL/min. The initial gradient conditions were 30% B, hold for 30 sec, then increase to 90% B at 6.0 min. 90% B was maintained for 7.5 min, at which time the column was re-equilibrated to 30% B over 0.75 min and held for 1.75 min. HPLC eluent was diverted to waste for the first 2.5 min and the final 9 min of analysis.&lt;/p>&lt;!>&lt;p>Mass spectrometric data were acquired with electrospray ionization (ESI). THC-glucuronide, THCCOOH-glucuronide, THCCOOH, 11-OH-THC and CBD were acquired in negative ionization mode while THC and CBN were acquired in positive ionization mode. MS/MS parameter settings (Table 1, compound-specific optimization) were optimized via direct infusion of individual analytes (500 μg/L in initial mobile phase) at 10 μL/min. Optimized source parameters were as follows: Gas (1) 0.31 MPa, Gas (2) 0.48 MPa, Curtain Gas 0.17 MPa, Source Temperature 650°C. Three acquisition periods were employed, with dwell times of 150 ms for each MS/MS transition in the first, 100 ms for the second and 150 ms for the final period. Unit resolution was used for all experiments.&lt;/p>&lt;!>&lt;p>Linear regression with 1/x2 weighting was employed for all analytes. Peak area ratios of target analytes and their respective internal standards were calculated for each concentration. Analyst Version 1.5 (AB Sciex, Foster City, CA) was utilized for all data collection and processing; statistical calculations were completed with GraphPad Prism 5 for Windows (GraphPad Software, La Jolla, CA).&lt;/p>&lt;!>&lt;p>Specificity, sensitivity, linearity, intra- and inter-batch imprecision, bias, extraction efficiency, matrix effect, carryover, dilution integrity, endogenous and exogenous interferences and analyte stability were investigated to evaluate method integrity. Specificity was based on relative retention time, precursor mass, and fragment ion. Retention times for QC and authentic specimens were required to be within ± 0.2 min of the mean calibrator retention time. Transition peak area ratios for QC and authentic specimens were required to be within ± 20% of the mean peak area ratios for calibrators of each respective analyte.&lt;/p>&lt;p>Sensitivity was evaluated by determining limits of detection (LOD) and (LOQ). A series of decreasing concentrations of drug-fortified whole blood was analyzed to empirically determine LOD and LOQ. LOD was determined as the concentration with a signal-to-noise ratio of at least 3, transition peak area ratios within 20% of the mean calibrator ratio and acceptable chromatographic retention time and peak shape. LOQ was the lowest concentration with a signal-to-noise ratio of at least 10, acceptable bias and imprecision (within at least 20% of target concentration and relative standard deviation within at least 20%, n = 6), transition peak area ratios within 20% of the mean calibrator ratio and acceptable chromatographic retention time and peak shape.&lt;/p>&lt;p>Linearity of the method was investigated by calculation of the regression line by the method of least squares and expressed by the squared correlation coefficient (R2). A 1/x2 weighting factor was applied to compensate for heteroscedasticity as evaluated through residuals analysis. Linearity of each analyte was determined with at least five concentration levels, not including the blank matrix, on 4 separate days.&lt;/p>&lt;p>Imprecision and bias were evaluated at three QC concentrations spanning the dynamic linear range. Intra-batch imprecision (% CV) was evaluated by six determinations per concentration in 1 day. Inter-batch imprecision (% CV) was evaluated for two replicates per concentration on 10 days (n total = 20). One-way ANOVA was employed to evaluate inter-batch repeatability as detailed by Peters and Maurer [26]; p&lt; 0.05 indicated significance. Bias was determined comparing the mean measured concentration of six analyses to the target value and was expressed as the percent of target concentration.&lt;/p>&lt;p>Extraction efficiency (%) and matrix effect (%) for each analyte also were determined at low, medium, and high control concentrations according to the design proposed by Matuszewski et al [27]. For determination of extraction efficiency, quality control standard solution was added prior to or following SPE. Extraction efficiency, %, was expressed as the mean analyte area of samples with control solution added before SPE (n = 6) divided by the mean analyte area of samples with control solution added after SPE (n = 6). Matrix effect was investigated by comparing analyte peak areas of extracted blank samples that were fortified after SPE versus analyte peak areas of neat samples prepared in initial mobile phase (30:70 A:B) at equivalent concentrations. Matrix effect was computed by dividing the analyte areas of blank samples fortified after SPE by areas of neat samples, expressed as percent.&lt;/p>&lt;p>Carryover was determined by injecting a negative specimen containing internal standard after a specimen containing two times the upper LOQ. As high concentrations are sometimes observed in blood following cannabis smoking, dilution integrity (1:5 and 1:10) was assessed with three blank blood specimens fortified with high QC solution. Specimens were combined with additional blank whole blood at 1:5 and 1:10 ratios to yield a 500 μL sample. Internal standard was added and specimens were processed as normal.&lt;/p>&lt;p>Interference from endogenous whole blood compounds was assessed by fortifying aliquots from ten blank whole blood pools with low QC solution and evaluating calculated concentrations. Interferences from over 80 illicit and common therapeutic drugs, metabolites and related compounds were evaluated by adding potential interferents into whole blood aliquots fortified with low QC solution. A compound did not interfere if the low QC quantified within 20% of target and had stable retention times and correct transition ratios. All interferences (Table 2) were tested at 1000 μg/L except for the cannabinoids that were tested at 250 μg/L.&lt;/p>&lt;p>Hydrolysis of glucuronides during sample processing was evaluated with blank whole blood fortified to 50 μg/L THC-glucuronide and 250 μg/L THCCOOH-glucuronide. Quantifying THC and THCCOOH formed in these hydrolysis controls allowed the calculation of percent hydrolysis for glucuronide metabolites. THC-glucuronide and THCCOOH-glucuronide standards also were investigated individually for presence of THC and THCCOOH, respectively. Individual neat standards were evaporated, reconstituted in mobile phase and quantified against a neat calibration curve to quantify any free cannabinoids present.&lt;/p>&lt;p>Analyte stability in whole blood (n= 5) was evaluated at three QC concentrations under three conditions: 16 h at room temperature (RT), 72 h at 4 °C and three freeze-cycles at −20 °C (23 h freeze, 1 h thaw at RT). Stability of extracted whole blood samples while in the 4°C autosampler was evaluated over 24 h. Extracted low, medium, and high QC samples (n = 3 at each level) were analyzed immediately after extraction along with calibration standards. Another set of three low, medium, and high QC samples were analyzed 24 h after extraction and subsequent storage in autosampler vials at 4°C. All samples were quantified from the initial calibration curve.&lt;/p>&lt;!>&lt;p>Cannabinoids are the most commonly abused illicit drugs, and cannabinoid medications are utilized for an increasing number of indications, documenting the need for accurate, sensitive and robust cannabinoid quantification. Numerous analytical methods are available to quantify cannabinoids in human whole blood, with and without conjugate hydrolysis [28–32]. However, these methods are limited to parent THC, phase I metabolites and other minor cannabinoids, and fail to consider implications of phase II metabolites. Specifically, factors such as poor hydrolysis efficiency [19,15] and glucuronide instability [33] can introduce unnecessary (and potentially substantial) error into quantitative determinations. Direct identification and quantification of glucuronides negates these issues and can yield novel insight into glucuronide pharmacokinetics and glucuronide in vitro stability while possibly providing an opportunity to utilize cannabinoid glucuronides as markers of recent cannabis intake. The present method sensitively and specifically quantifies these glucuronides directly in addition to typical cannabinoids of interest, including minor cannabinoids CBD and CBN (Figure 1). Thus, this first analytical method for directly analyzing free and glucuronidated cannabinoids in the same whole blood specimen is a significant advancement in the detection and quantification of this important class of compounds.&lt;/p>&lt;!>&lt;p>The method was validated according to the criteria described in the Experimental Section. Table 3 details LOD, LOQ and calibration results for each analyte. LOQs were determined empirically through analysis of decreasing concentrations of drug-fortified whole blood and were 1 μg/L for THC, 11-OH-THC, THCCOOH, CBD and CBN with a 0.5 mL whole blood specimen, exceeding cutoff criteria proposed by Farrell et al [34] and meeting the 1 μg/L THC cutoff typically employed for DUID testing [35]. To extend the dynamic linear range for THCCOOH-glucuronide and minimize the number of re-extractions that might be required due to high THCCOOH-glucuronide concentrations, a 250 μg/L calibrator was included for this analyte. However, extending the linear range required increasing the LOQ from 2.0 to 5.0 μg/L to meet a priori specifications for calibration curve linearity. Linear ranges and R2 values (1/x2 weighting) were acceptable (R2 > 0.990) for all analytes. Linear ranges were THC-glucuronide 0.5–50 μg/L, THCCOOH-glucuronide 5.0–250 μg/L and THC, 11-OH-THC, THCCOOH, CBD and CBN 1.0–100 μg/L (Table 3); these ranges should prove useful for clinical and forensic casework. Calibrators for THC, 11-OH-THC, THCCOOH, CBD and CBN quantified within ± 15% (± 20% for LOQ and glucuronides) when quantified against the entire calibration curve. We expect that our ongoing clinical studies will help establish the utility of glucuronide metabolites for establishing recency of use, generating wider interest in cannabinoid glucuronide testing. Additional interest might prompt proper deuterated internal standards synthesis, allowing more stringent criteria (± 15%) to be applied to all analytes at concentrations > LOQ.&lt;/p>&lt;p>Deuterium-labeled analogues are not currently commercially available for THCCOOH-glucuronide, THC-glucuronide and CBN. The decision to implement THC-d3 and THCCOOH-d9 for CBN and glucuronides, respectively, was based on similarities in extraction efficiency and matrix effects. This choice was not ideal as differences in efficiencies were present and these can vary depending on the matrix pool; nevertheless, a priori specifications for sensitivity and linearity were met. Other glucuronide metabolites, including morphine-3-glucuronide-d3, buprenorphine-glucuronide and mefenamic acyl-β-D-glucuronide-d3 were investigated as potential internal standards. However, these were either not extracted efficiently (buprenorphine-glucuronide) or not well-retained on our chromatographic system (morphine-3-glucuronide-d3 and mefenamic acyl-β-D-glucuronide-d3). While we attempted to minimize matrix effects through sample preparation including solid phase extraction, some matrix effect remained. The matrix effects for glucuronides and their respective internal standards were not identical, but our approach is the best available at this time. Furthermore, we investigated matrix effect in 10 different whole blood pools demonstrating that low QC quantification remained within ± 20% in all 10 whole blood pools. Despite these efforts, differential matrix effect cannot be excluded, and glucuronide quantification could be affected. It should be noted that deuterated glucuronide analogues are recommended should they become available, as improvements in imprecision, bias and reliability could be realized.&lt;/p>&lt;p>Bias and imprecision were evaluated at three concentrations across the linear dynamic range of each analyte (Table 4). Intra-batch imprecision (% CV) was less than 7.9% for all analytes at all concentrations (n= 6); inter-batch imprecision (% CV) was less than 10.4% (n= 20). Bias, calculated as the percent of target concentrations at low, mid and high QC concentrations for each analyte, ranged from 93.8% to 113.1% of target concentrations (n= 6). One-way ANOVA yielded statistically significant differences in inter-batch repeatability for several analytes; however, differences were less than 10.4% CV and considered clinically insignificant.&lt;/p>&lt;p>Extraction efficiency for native and deuterium-labeled analytes ranged from 50.5% to 93.9% (Table 5). Table 5 also displays ion suppression/enhancement produced by matrix effect; positive values indicate ion enhancement and negative values indicate ion suppression. While substantial matrix effects were observed for 11-OH-THC, similar results were obtained for the corresponding deuterated analogue and quantification was not adversely affected.&lt;/p>&lt;p>Development of an effective sample cleanup that removed matrix interferences while maintaining high extraction efficiency proved to be the greatest challenge during method development. The extraction procedure (reversed-phase polymeric SPE), gentle wash step (20% acetonitrile in water) and polar elution solvent (acetonitrile) yielded high concentrations of phospholipids in extracts as evidenced through a positive precursor ion scan of m/z 184 as detailed by Xia and Jemal [36]. Extending the 90% acetonitrile hold to 7.5 min during the chromatographic gradient provided effective column washing and removal of phospholipids; forgoing this wash yielded substantial increases in ion suppression for subsequent injections. In addition to high phospholipid concentrations, rapid increases in system backpressure were observed during initial method development with a smaller HPLC column particle size (3 μm) and methanolic mobile phase. Backpressure increases were mitigated through replacement of methanol with acetonitrile, increasing column particle size to 5 μm and more frequent replacement of the guard column. Thus, slight decreases in resolution and cost-efficiency were offset by increased column life and a more reliable method.&lt;/p>&lt;p>Carryover in a negative specimen following a specimen containing twice the upper limit of quantification was assessed. No carryover was observed for any analyte; ion transition ratios were not within 20% of calibrators and any signal present was less than LODs. Common therapeutic and illicit drugs and metabolites at concentrations of 1000 μg/L (cannabinoids 250μg/L) did not interfere with analytes of interest at the low QC concentration. Additionally, ten pools of whole blood were tested for potential endogenous interferences; none were observed in any pool for any analyte. Dilution integrity was maintained up to 10 times dilution with blank whole blood and all analytes quantified within 20% of the theoretical high QC concentration.&lt;/p>&lt;p>Quantification of THCCOOH and THC formed in glucuronide control samples during extraction was conducted (n= 6 each). Mean (SD) percentages of THCCOOH-glucuronide and THC-glucuronide hydrolysis were 0.6 ± 0.05% and 3.7 ± 0.35%, respectively. However, these are both likely artifacts as neat THCCOOH-glucuronide and THC-glucuronide calibrators were determined to contain 0.5 ± 0.1% THCCOOH and 3.2 ± 0.2% THC, respectively (n = 5 each). While the ester-linked THCCOOH-glucuronide was reported as relatively labile [33], we observed minimal hydrolysis of THCCOOH-glucuronide during extraction. The THC impurity has a minor effect on THC quantification that is less than the analytical error for the method and a low LOQ of 1 μg/L was achieved. To confirm a lack of substantial effect on THC quantifications, samples fortified with only THC at the LOQ (1 μg/L) were quantified against the entire calibration curve containing all analytes. Acceptable quantifications were obtained (± 20%) for these samples, confirming minimal bias resulting from the THC impurity present in the THC-glucuronide standard.&lt;/p>&lt;p>Stability at 4 °C on the autosampler for 24 h was determined for extracted specimens. All analytes at all concentrations (low, mid and high QC) were stable under these conditions, with mean concentrations differing from samples injected immediately (n= 3) by less than −8.3% (Table 6). For fortified whole blood samples, THCCOOH, 11-OH-THC and both glucuronides were stable under all other conditions tested (three freeze-thaw cycles, 72 h at 4 °C and 16 h at RT). However, losses up to 35.7% were observed for THC after 72 h at 4 °C. Additionally, CBD, CBN and THC demonstrated relative instability under three freeze-thaw cycles, 72 h at 4 °C and 16 h at RT. It should be noted that these losses were observed in fortified samples; losses in authentic specimens may not reflect these findings due to differences in protein binding [15].&lt;/p>&lt;!>&lt;p>Whole blood was collected from a clinical research participant prior to and after smoking a single cannabis cigarette ad libitum. Baseline concentrations were less than LOQ for all cannabinoids except THCCOOH and THCCOOH-glucuronide. 15 and 60 min after the start of smoking, blood was collected and concentrations determined by this new analytical method (Table 7). THC-glucuronide quantified at 0.6 μg/L in the first specimen, demonstrating the necessity for the low LOQ that this method achieved. It should be noted that specimens were analyzed within 24 h of collection, minimizing any potential losses due to analyte degradation. Concentrations suggest THC-glucuronide may serve as possible marker of recent cannabis intake, given that it is detectable following cannabis smoking, albeit at a low concentration. Further research is required to assess detection windows for THC-glucuronide or other minor cannabinoids, such as CBD or CBN, following smoked cannabis.&lt;/p>&lt;!>&lt;p>This method is the first robust, sensitive and specific LC-MS/MS technique for direct detection and quantification of several cannabinoids and two cannabinoid glucuronides in human whole blood, yielding a comprehensive cannabinoid whole blood profile following cannabis intake. The rapid and simple extraction and 16 min analysis are beneficial; however, care should be taken to prevent buildup of phospholipids and other matrix components, leading to increased HPLC backpressure and loss of resolution. This method is utilized for several controlled cannabinoid administration studies and will provide whole blood pharmacokinetic and cannabinoid stability data useful to clinicians and forensic toxicologists interpreting whole blood cannabinoid concentrations often obtained during DUID cases and other investigations. This new analytical method for cannabinoids in whole blood offers advantages in sensitivity and spectrum of cannabinoid analytes included over existing LC-MS/MS and GC-MS assays, and when applied to controlled cannabinoid administration studies, may improve our ability to interpret cannabinoid whole blood results.&lt;/p></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">PubMed Author Manuscript</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="7"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">A relatively low level of ribosome depurination by mutant forms of ricin toxin A chain can trigger protein synthesis inhibition, cell signaling and apoptosis in mammalian cells</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">The A chain of the plant toxin ricin (RTA) is an N-glycosidase that inhibits protein synthesis by removing a specific adenine from the 28S rRNA. RTA also induces ribotoxic stress, which activates stress-induced cell signaling cascades and apoptosis. However, the mechanistic relationship between depurination, protein synthesis inhibition and apoptosis remains an open question. We previously identified two RTA mutants that suggested partial independence of these processes in a yeast model. The goals of this study were to establish an endogenous RTA expression system in mammalian cells and utilize RTA mutants to examine the relationship between depurination, protein synthesis inhibition, cell signaling and apoptosis in mammalian cells. The non-transformed epithelial cell line MAC-T was transiently transfected with plasmid vectors encoding precursor (pre) or mature forms of wild-type (WT) RTA or mutants. PreRTA was glycosylated indicating that the native signal peptide targeted RTA to the ER in mammalian cells. Mature RTA was not glycosylated and thus served as a control to detect changes in catalytic activity. Both pre- and mature WT RTA induced ribosome depurination, protein synthesis inhibition, activation of cell signaling and apoptosis. Analysis of RTA mutants showed for the first time that depurination can be reduced by 40% in mammalian cells with minimal effects on inhibition of protein synthesis, activation of cell signaling and apoptosis. We further show that protein synthesis inhibition by RTA correlates more linearly with apoptosis than ribosome depurination.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">a_relatively_low_level_of_ribosome_depurination_by_mutant_forms_of_ricin_toxin_a_chain_can_trigger_p</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">4,477</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">232</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">19.297414</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">1. Introduction&lt;!>2.1. Reagents&lt;!>2.2 Mutant RTA plasmid construction&lt;!>2.3. Cell culture&lt;!>2.4. Transfection&lt;!>2.5. Western immunoblotting&lt;!>2.6. rRNA depurination assays&lt;!>2.7. Protein synthesis inhibition assay&lt;!>2.8. Caspase 3/7 assay&lt;!>2.9. Nucleosome accumulation assay&lt;!>2.10. Statistical analysis&lt;!>3.1. The native signal peptide targets RTA to the ER in mammalian cells&lt;!>3.2. Ribosome depurination and protein synthesis inhibition are reduced relative to WT RTA in cells transfected with RTA mutant constructs&lt;!>3.3. Activation of apoptosis corresponds with protein synthesis inhibition in cells transfected with RTA and RTA mutant constructs&lt;!>4.1. Development of a mammalian expression system to study biological activity of pre- and mature WT RTA and RTA mutants&lt;!>4.2. A substantial reduction in depurination is necessary to prevent inhibition of translation by RTA in mammalian cells&lt;!>4.3. Activation of apoptosis by RTA mutants corresponds better with the extent of protein synthesis inhibition than depurination in mammalian cells&lt;!>Supplemental Figure 1&lt;!>Supplemental Figure 2</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">&lt;p>The plant toxin ricin is produced by the castor bean plant Ricinus communis and belongs to a family of ribosome-inactivating proteins (RIPs). Its severe toxicity and wide availability has led to its use as an agent of bioterrorism (Rainey and Young, 2004, Audi et al., 2005). In addition, ricin has been investigated as the active moiety of immunotoxins selectively targeted to cancer cells (Castelletti et al., 2004, Schindler et al., 2011, Zhou et al., 2010). Ricin is composed of two subunits which are encoded by a single gene. The catalytically active A subunit (RTA) depurinates a specific adenine in the α-sarcin/ricin loop (SRL) of the 28S rRNA, resulting in protein synthesis inhibition (Sandvig and van Deurs, 2005, Watson and Spooner, 2006). The B subunit (RTB) binds to cell surface receptors through galactose and N-acetyl galactosamine moieties. After internalization, ricin is transported from early endosomes to the endoplasmic reticulum (ER) via the trans-Golgi network (Spooner and Lord, 2012). While active research is underway to develop effective vaccines (Marsden et al., 2004, Smallshaw et al., 2007, Smallshaw and Vitetta, 2012, Vitetta et al., 2006), monoclonal antibodies (Dai et al., 2011) and small molecule inhibitors (Pang et al., 2011, Pruet et al., 2011, Stechmann et al., 2010, Wahome et al., 2010) to prevent or treat ricin toxicity, there are currently no approved antidotesor therapeutics available.&lt;/p>&lt;p>The ability to develop effective antidotes against ricin or to use it as the active component of an immunotoxin hinges on a thorough understanding of its biological actions in mammalian cells. In addition to its inhibitory effect on protein synthesis, ricin also induces apoptosis in multiple cell types in vitro and in vivo (Tesh, 2012). Ricin triggers the intrinsic apoptotic pathway as evidenced by release of cytochrome c from mitochondria and subsequent activation of caspase 3, caspase 9 and PARP (Hu et al., 2001, Jetzt et al., 2009, Rao et al., 2005). However, the role of ribosome depurination and protein synthesis inhibition in the apoptotic response remains unclear. The finding that protein synthesis inhibitors that act on the 28S rRNA (i.e. anisomycin and ricin) activate caspase 3 while protein synthesis inhibitors with a different mechanism of action (i.e. diphtheria toxin and cycloheximide) do not suggests that the mode of protein synthesis inhibition may influence the induction of apoptosis (Kageyama et al., 2002). In addition to its inhibitory effect on protein synthesis, ricin also activates the signaling cascades JNK and p38 (Iordanov et al., 1997, Jetzt et al., 2009, Korcheva et al., 2007). The ability to activate these pathways requires a ribosome that is translationally active, indicating that the ribosome actively senses damage to the 28S rRNA. This has been termed the ribotoxic stress response (Iordanov et al., 1997). We have shown that inhibiting the JNK pathway attenuates the ability of RTA to induce apoptosis in MAC-T cells (Jetzt et al., 2009), while the p38 pathway has been shown to play a role in the proinflammatory cytokine response that is observed with ricin toxicity (Higuchi et al., 2003, Korcheva et al., 2007, Lindauer et al., 2010). Although activation of the ribotoxic stress response by ricin clearly triggers signaling cascades involved in apoptosis, the precise role of this response as it relates to protein synthesis inhibition has not been established.&lt;/p>&lt;p>We previously conducted chemical mutagenesis of the precursor form of RTA (preRTA) which contains a 35-residue leader peptide and isolated mutants based on their inability to induce cell death. Two mutants were identified (P95L/E145K and S215F) that depurinate ribosomes and inhibit protein synthesis similar to WT RTA at 6 h post induction. However, these mutants failed to induce nuclear fragmentation and reactive oxygen species (ROS) generation, which are apoptotic-like characteristics in yeast (Li et al., 2007). These data provide support for the concept that the level of depurination and protein synthesis inhibition may not correspond with cell death. To investigate these relationships in mammalian cells, WT RTA and RTA mutants that caused different levels of depurination in yeast were expressed in MAC-T cells. RTA mutants included the two mentioned above (i.e. P95L/E145K and S215F), G212E, which has very low enzymatic activity and is not toxic in yeast and RTA active site mutants E177K and E177Q. Since the preRTA gene containing the leader sequence would target RTA to the ER, the mature RTA gene lacking the leader sequence was also expressed to examine direct effects of the mutations on catalytic activity in the absence of ER trafficking.&lt;/p>&lt;!>&lt;p>Insulin, gentamicin, D-(+)-glucose, RTA purified from Ricinus communis and phenol red-free Dulbecco's modified Eagle's medium (DMEM) with low glucose were purchased from Sigma-Aldrich (St. Louis, MO). DMEM containing 4.5 g/l D-glucose (i.e. DMEM-H) and penicillin/streptomycin were obtained from Invitrogen (Carlsbad, CA). Fetal bovine serum (FBS) was purchased from Atlanta Biologicals (Lawrenceville, GA). Endoglycosidase H was obtained from New England Biolabs (Ipswich, MA). Recombinant RTA with N-terminal histidine tag expressed in E. coli (NR-853) was obtained through NIAID NIH Biodefense and Emerging Infections (BEI) Research Resources Repository (Manassas, VA). Anti-RTA antibody was produced in rabbits (Covance Research Products; Denver, PA). Antibodies against JNK, p38 and phospho-p38 were purchased from Cell Signaling Technology (Danvers, MA) and phospho-JNK antibody was obtained from Santa Cruz Biotechnology (Santa Cruz, CA). Donkey anti-rabbit and horse anti-mouse horseradish peroxidase-linked secondary antibodies were purchased from GE Healthcare (Piscataway, NJ) and Vector Laboratories (Burlingame, CA), respectively. Peroxidase activity was detected by Pierce ECL Western Blotting Substrate (Thermo Scientific, Rockford, IL) or ECL Prime (GE Healthcare).&lt;/p>&lt;!>&lt;p>The coding sequence of Ricinus communis preRTA containing the 35-residue leader sequence (Piatak et al., 1988) was converted to an optimized codon usage for Bos taurus (Fig. S1) and synthesized. Mature RTA lacking the leader sequence was then constructed from preRTA by PCR cloning (Genscript; Piscataway, NJ). Genes were subcloned into pCAGGS mammalian expression vector (Niwa et al., 1991). Site-directed mutagenesis was performed using the QuikChange Lightning Site-Directed Mutagenesis Kit (Stratagene; La Jolla, CA). The locations of individual mutations are shown in Figure 1. Mutagenesis was confirmed by sequencing. Constructs for transfection were prepared using an EndoFree Plasmid Maxi Kit (Qiagen; Valencia, CA).&lt;/p>&lt;!>&lt;p>The bovine mammary epithelial cell line MAC-T (Huynh et al., 1991) was maintained as previously described (Fleming et al., 2005). For all experiments MAC-T cells were plated in phenol-red free DMEM containing 4.5 g/l D-glucose (DMEM-H), 10% FBS, 20 U/ml penicillin, 20 μg/ml streptomycin, and 50 μg/ml gentamicin (complete media). The human epithelial cell line HEK293T/17 was obtained from ATCC (Manassas, VA). HEK293T/17 cells were maintained and plated for experiments in complete media with phenol red and without gentamicin. All cells were cultured at 37°C in a humidified environment with 5% CO2.&lt;/p>&lt;!>&lt;p>MAC-T cells were plated in complete media at 3.5 × 104 cells/cm2. The next day subconfluent cells were transfected with endotoxin-free plasmid DNA and SuperFect (Qiagen) combined in a 1:5 ratio for 60 × 15 mm dishes and in a 1:10 ratio for 96 well plates. The transfection mixture was prepared in DMEM-H with no additives, vortexed for 10 sec, and incubated at RT for 10 min. Spent media was removed from cells and replaced with fresh complete media and the transfection mixture. After 3 h media was removed and replaced with fresh complete media. HEK293T/17 cells were plated at 5 × 104/cm2. The following day, subconfluent cells were transfected with endotoxin-free plasmid DNA and GeneJuice (EMD Chemicals Inc.; San Diego, CA) according to the manufacturer's protocol. Transfection efficiencies for both cell lines were monitored using pEGFP (Clontech, Mountain View, CA).&lt;/p>&lt;!>&lt;p>MAC-T or HEK293T/17 cells were plated in 60 × 15 mm dishes at 3.5 × 104 cells/cm2 or 5 × 104 cells/cm2, respectively, and transfected the next day as described above. After incubation in serum-containing media for the indicated times cells were washed twice with cold PBS and total cell lysates were collected by scraping into cold cell lysis buffer as previously described (Grill et al., 2002). Cells were then incubated on ice at 4°C for 30 min and spun at 1000 × g for 5 min at 4°C. The supernatant was removed and passed ten times through an 18 gauge needle. The cell lysates were aliquoted and stored at −80°C until use. Protein concentration was determined using the Bio-Rad Protein Assay (Bio-Rad; Hercules, CA). To determine glycosylation status of expressed RTA, cell lysates were denatured at 100°C for 10 min and incubated with Endoglycosidase H (25 U/mg) at 37°C for 1 h. Proteins were separated by SDS-PAGE and transferred to 0.45 μm PVDF (Millipore, Billerica, MA) or 0.2 μm nitrocellulose (Bio-Rad).&lt;/p>&lt;!>&lt;p>MAC-T cells were plated in 60 × 15 mm dishes at 3.5 × 104 cells/cm2 and transfected the following day as described above. After transfection cells were scraped into 1 ml Trizol (Invitrogen) and stored at −80°C. Total RNA was purified using RNeasy columns (Qiagen). rRNA depurination was then analyzed by dual primer extension analysis as described previously (Jetzt et al., 2009).&lt;/p>&lt;p>rRNA depurination was also analyzed by quantitative reverse transcription PCR (qRT-PCR) as previously described (Melchior and Tolleson, 2010) with minor modifications. Briefly, total RNA (2 μg) was reverse transcribed into cDNA using the High Capacity cDNA Reverse Transcription (RT) kit (Applied Biosystems, Carlsbad, CA). RT reactions (20 μl total volume) were incubated at 25 C for 10 min, 37°C for 2 h, and 85°C for 5 min, and stored at −20°C. Quantitative PCR was performed in a StepOnePlus Real-Time PCR System (Applied Biosystems). Each RT reaction was diluted 1:500 and assayed in triplicate. Total reaction volume (20 μl) contained 5 μl of cDNA, 0.5 μl of each primer, 10 μl Power SYBR Green PCR Master Mix (Applied Biosystems) and 4 μl nuclease-free water. Primers (Sigma) that recognize bovine 28S rRNA and the depurination (dep) fragment were designed as described by Melchior and Tollensen (2010). Primers were 28S-F, 5′-GATGTTGGCTCTTCCTATCATTGT-3′; 28S-R, 5′-CCAGCTCACATGCCCTATTAGTT-3′; dep-F, 5′-TGCCATGGTACCTGCTCAGCA-3′; dep-R, 5′-TCTGAACCTGTGGTTCCACA-3′. Final primer concentrations were 0.25 μM except for dep-R which was 0.75 μM. Cycling conditions were 95°C for 10 min followed by 35 cycles of 95°C for 15 sec and 60°C for 1 min. Melting curves were generated for each sample. Each primer set was validated by constructing standard curves using serial dilutions (1:50 to 1:500,000) of an RT reaction derived from RNA collected from MAC-T cells treated for 6 h with RTA (1 μg/ml). The calculated amplification efficiencies were 95.8% ± 1.3 for the 28S primer pair and 95.4% ± 0.2 for the depurination primer pair (mean ± SD for three independent curves). Treatment effects were determined as fold-change using the ΔΔCT method. The average CT value for reactions containing 28S (endogenous reference) primers was subtracted from the average CT value for reactions containing depurination primers for that same sample (ΔCT ). The same calculation was preformed for a calibrator sample (vector control). The ΔΔCT was then determined by subtracting the ΔCT of a calibrator sample (vector control) from the ΔCT of each test sample.&lt;/p>&lt;!>&lt;p>MAC-T cells were plated in black 96-well plates at 3.5 × 104 cells/cm2 and transfected the following day as described above. Cells were co-transfected with equal amounts of RTA mutant and pEGFP-N1 reporter plasmid. Twenty-one hours after the start of transfection the EGFP fluorescent signal was measured in a plate reader (BioTek) with excitation filter 485/20 and emission filter 530/25. Fluorescence measured in cells co-transfected with GFP and empty vector was considered 100%.&lt;/p>&lt;!>&lt;p>Cells were plated in black 96-well plates at 3.5 × 104 cells/cm2 and transfected the following day as described above. Caspase 3/7 activation was determined using the SensoLyte Homogeneous AMC Caspase 3/7 Assay Kit from AnaSpec (San Jose, CA) as described by the manufacturer.&lt;/p>&lt;!>&lt;p>Cells were plated in 96-well plates at 3.5 × 104 cells /cm2 and transfected the following day as described above. Nucleosome accumulation was determined using the Cell Death Detection ELISA kit (Roche Applied Science, Indianapolis, IN) as described by the manufacturer.&lt;/p>&lt;!>&lt;p>Data were analyzed by one-way ANOVA with Dunnett's Multiple Comparison Test. PreRTA mutants were compared to WT preRTA while mature RTA mutants were compared to WT mature RTA. Differences were considered significant for P &lt; 0.05. Analyses were performed using GraphPad Prism (5.02).&lt;/p>&lt;!>&lt;p>The Ricinus communis gene encoding RTA was modified to utilize preferred codons for the bovine in order to optimize expression of pre- and mature RTA in the MAC-T cell line. The preRTA contained the native 35-residue N-terminal leader peptide followed by the 267-residue mature RTA, while mature RTA did not contain the leader peptide (Fig. 1A). A comparison of the two sequences is shown in Fig. S1. Point mutations were then introduced to create RTA mutants that we had previously shown to decrease cytotoxicity in yeast (Li et al., 2007) (Fig. 1A, B). MAC-T cells were transiently transfected with either pre- or mature forms of WT and mutant RTA. As shown in Fig. 2A, the mature WT protein and all seven mature RTA mutants were detectable by western blotting with an RTA antibody although the mature double mutant P95L/E145K was very faint. All preRTA forms were easily detectable with the exception of E145K and P95L/E145K. The latter two mutants were only visible with much longer exposure times. To determine if higher expression of E145K and the double mutant could be obtained by transfecting a different cell line, HEK293T/17 cells were transfected with the codon-optimized RTA-expressing vectors. As shown in Fig. S2, the pattern of expression for pre- and mature WT, E177K, E145K and the double mutant was similar in HEK293T/17 cells to that observed in MAC-T cells. Pre- and mature E177K were more abundantly expressed than the other RTA forms in both MAC-T and HEK293T/17 cells.&lt;/p>&lt;p>RTA is N-glycosylated on asparagine residues 10 and 236 (Rutenber et al., 1991) in the ER (Rapak et al., 1997). The first 26 residues of the 35-residue N-terminal leader peptide target RTA to the ER in yeast (Yan et al., 2012) and in plants (Halling et al., 1985). PreRTA ran as a doublet of approximately 30 and 32 kDa which corresponded with the molecular weight of the doublet observed for glycosylated RTA purified from Ricinus communis. Mature RTA ran as a single band at approximately 30 kDa similar to non-glycosylated recombinant RTA (Fig. 2B). To determine if the differences in size were due to glycosylation, lysates collected from cells transfected with pre- or mature WT and E177K were treated with Endoglycosidase H (Endo H) which cleaves the N-linked mannose groups of oligosaccharides (Fig. 2C). The doublet observed in cells expressing the pre-form of RTA or E177K was reduced to one band after Endo H treatment. The single band observed in cells transfected with mature RTA and E177K did not change in size with Endo H treatment. These data demonstrate that the signal sequence of native RTA from the castor bean plant successfully targets the precursor form of RTA to the ER in MAC-T cells. In contrast, expression of the mature form resulted in nonglycosylated RTA, indicating that it is not translocated into the ER.&lt;/p>&lt;!>&lt;p>To determine if endogenously expressed RTA and RTA mutants depurinate rRNA in transfected MAC-T cells, total RNA was collected 19 h after transfection and a dual primer extension assay was performed (Fig. 3A). Since the dual primer extension assay is relatively semi-quantitative we also established a qRT-PCR assay to determine ribosome depurination (Fig. 3B). As shown in Table S1, the results obtained with dual primer extension were matched closely by qRT-PCR . In general, the levels of depurination in cells transfected with preRTA constructs were similar to those in cells transfected with the mature RTA counterpart. All mutants tested depurinated MAC-T rRNA significantly less than their WT control with the exception of mature P95L. The reduction in depurination was the most marked with the active site mutants E177Q and E177K which depurinated less than 40% and 20%, respectively, relative to their WT controls. Ribosome depurination was 60 to 70% of that observed for WT RTA for E145K, P95L/E145K, S215F and G212E.&lt;/p>&lt;p>A GFP transfection assay was used to determine if protein synthesis inhibition corresponded with changes in ribosome depurination. Overall, the pattern of ribosome depurination observed with the mutated RTA proteins was reflected in the degree of protein synthesis inhibition (Fig. 4). Fluorescence was almost nondetectable in cells transfected with pre- or mature WT RTA relative to cells transfected with plasmid vector alone. Similar results were obtained with P95L, which depurinated ribosomes similarly to WT RTA based on qRT-PCR results. Pre- and mature E177K did not inhibit protein synthesis which corresponded with the very slight degree of depurination observed. Protein synthesis levels observed with pre- and mature E177Q were intermediate between E177K and the other mutants. Surprisingly, a greater level of inhibition of protein synthesis was observed with mature E177Q (60% inhibition of vector controls) compared to preE177Q (40% inhibition of vector controls) even though they showed similar levels of depurination at 19 h after transfection. Protein synthesis was inhibited significantly less relative to WT RTA for preP95L/E145K and for pre- and mature S215F and G212E, however, this still represented an 80 to 90% inhibition of protein synthesis relative to vector controls. Interestingly, the mature RTA mutants tended to have a greater effect on protein synthesis inhibition than their preRTA counterparts. Specifically, preE177Q, P95L/E145K, S215F and G212E inhibited protein synthesis less than their corresponding mature forms and less than WT preRTA.&lt;/p>&lt;!>&lt;p>To investigate if apoptosis was induced in the transfected cells, activation of caspase 3/7 was investigated in MAC-T cells 19 h after transfection (Fig. 5). Caspase activity was induced similarly by pre- and mature WT RTA (2.7 ± 0.2 and 2.8 ± 0.2-fold, respectively; mean ± SE of 9 experiments) relative to vector controls. The active site mutant E177K elicited negligible caspase activation which corresponded with the lack of ribosome depurination and protein synthesis inhibition. In contrast the pre- and mature forms of E145K, G212E and P95L/E145K as well as mature S215F activated caspase activity to the same degree as their respective WT RTA controls. PreS215F activated caspase 3/7 to a lesser extent than WT preRTA, which corresponded with decreases in depurination and protein synthesis inhibition. The E177Q mutant also showed a difference between the pre- and mature forms in terms of caspase activation, with mature E177Q eliciting a greater increase in caspase 3/7 activity relative to preE177Q. Interestingly, both the pre- and mature forms of P95L activated caspase to a greater extent relative to their WT counterparts.&lt;/p>&lt;p>As a second indicator of apoptosis, a nucleosome accumulation assay was conducted (Fig. 6). Wild-type pre- and mature RTA each increased nucleosome accumulation two-fold over cells transfected with vector alone. While there was more variability with this assay, the results mirrored overall those of the caspase assay. Mature E177Q had more activity than pre E177Q while neither form of E177K showed any activity. The other mutants also showed activity that was similar to what was observed with the caspase assay.&lt;/p>&lt;p>RTA has been shown to activate JNK and p38 signaling pathways in MAC-T cells (Jetzt et al., 2009), therefore the ability of the different mutants to activate these pathways was examined (Fig. 7). Endogenous expression of both pre- and mature RTA activated JNK and p38 approximately 2-fold relative to vector alone. However, neither JNK nor p38 was activated by expression of pre- or mature E177K. Mutants E145K, G212E, P95L/E145K and S215F activated JNK and p38 signaling similarly to WT RTA controls. This corresponded with full caspase activation with the exception of preS215F, which exhibited decreases in caspase activation. Interestingly, the ability of the active site mutant E177Q to activate JNK and p38 appeared intermediate between E177K and the other mutants.&lt;/p>&lt;!>&lt;p>This is the first report where catalytically active WT RTA and RTA mutants have been expressed in mammalian cells to study the relationship between depurination, protein synthesis inhibition, cell signaling and apoptosis. By optimizing codon usage for the bovine, we were able to increase expression such that we could detect RTA protein by immunoblot analysis in 25 to 50 μg total cell lysates from either bovine or human cells using a specific RTA antibody. Redmann and coworkers (Redmann et al., 2011) recently reported the expression of preRTA and two preRTA mutants in mammalian cells. In their study an N-terminal murine MHC class I heavy chain H2-Kb signal peptide was used to target an enzymatically attenuated RTA variant to the ER membrane. Also, RTA trafficking was the main endpoint studied in that report and the enzymatically attenuated RTA variant was expressed with an HA epitope tag for detection by immunoprecipitation. Using the native signal peptide of ricin, we successfully targeted RTA to the ER membrane in mammalian cells as shown by production of glycosylated protein. The mature form of RTA was not glycosylated, indicating that it was not translocated into the ER. The mature form served as a control to detect changes in catalytic activity, since changes in depurination by mature RTA would be due to catalytic activity and not to trafficking. At the time point measured, levels of depurination were similar between pre- and mature forms of RTA, indicating that preRTA successfully trafficked from the ER to the ribosome.&lt;/p>&lt;p>E177 has been identified as an invariant amino acid across the RIP family. It lies within the active site of RTA and is known to be a key catalytic residue (Monzingo and Robertus, 1992). In the present study, conversion of E177 to lysine (E177K) produced mutants that exhibited virtually no detectable depurination activity and failed to inhibit protein synthesis, induce apoptosis or activate signaling of the JNK and p38 cascades. This agrees with results obtained with other systems, e.g., this mutation led to total inactivation of the enzyme in an in vitro translation assay (Chaddock and Roberts, 1993) and allowed growth in yeast (Allen et al., 2005, Li et al., 2007). Pre- and mature E177K were expressed at the highest level relative to all other constructs. Interestingly, in a study where amino acids were systematically deleted to determine their role in RTA activity, an inverse correlation was observed between expression in E. coli and enzymatic activity in vitro (Morris and Wool, 1992). This suggests that the very high level of E177K expression may be related to complete lack of biological activity of the mutant protein. These results are also consistent with previous studies in yeast (Li et al., 2007) and suggest that greater enzymatic activity of RTA will lead to higher translation inhibition and reduced protein accumulation.&lt;/p>&lt;!>&lt;p>In yeast, we found that S215F and P95L/E145K induced depurination and inhibited protein synthesis similarly to WT preRTA but did not induce nuclear fragmentation and ROS generation, hallmarks of apoptosis. In addition, G212E had low biological activity and did not affect any of these endpoints (Li et al., 2007). Expression of pre- and mature forms of each of these three mutants reduced depurination levels to 60% of WT RTA control levels in mammalian cells at 19 h after transfection. This lower level of depurination was still sufficient to inhibit protein synthesis by 80 to 90% relative to the vector control. Interestingly, the pre-forms of G212E, S215F, and P95L/E145K did tend to have higher levels of protein synthesis compared to the mature forms or WT preRTA when expressed in mammalian cells, although protein synthesis was still only 15 to 20% of vector control levels. However, they produced full caspase activation, nucleosome accumulation and JNK/p38 signaling. These results indicate that depurination can be reduced by as much as 40% in mammalian cells with minimal effects on protein synthesis inhibition and activation of stress-activated signaling cascades and apoptosis. A substantial reduction in depurination as was observed with pre E177Q may be necessary to prevent protein synthesis inhibition by RTA, possibly due to the high sensitivity of mammalian ribosomes to RTA.&lt;/p>&lt;!>&lt;p>Since E177K exhibited virtually no biological activity, we converted Glu177 to glutamine (E177Q) with the goal of producing an active site mutant that retained some biological activity. This resulted in the expression of pre- and mature RTA proteins that exhibited depurination activity that was approximately 35% of that observed with WT RTA. This agrees with previous studies (Ready et al., 1991) demonstrating that the E177Q mutation decreases enzymatic activity at least 170-fold relative to WT RTA in vitro but less than that of the E177K mutation. Interestingly, while ribosome depurination was similar between the pre- and mature forms of E177Q, the degree of protein synthesis inhibition as well as the induction of apoptosis differed. For the pre form of E177Q, protein synthesis was reduced 32% relative to vector control. However, apoptosis was not induced. In contrast, the mature form of E177Q inhibited protein synthesis 60% relative to vector contols which corresponded with full caspase activation and nucleosome accumulation. The reason for the difference in the degree of protein synthesis inhibition when depurination levels were similar is unknown at this time, but may be related to differences in the rate of depurination (Yan et al., 2012). These results indicate that there may be a threshold level of protein synthesis inhibition that correlates with activation of apoptosis in mammalian cells. Interestingly, the difference in the activation of apoptosis did not appear to correlate with differences in activation of JNK or p38, since JNK activation was statistically less for mature E177Q compared to the pre form and there was no difference between the two for p38 activation. Recently it was reported that ricin mediates IL-1β release from bone-marrow derived macrophages through a scaffolding complex termed the NALP3 inflammasome, which facilitates cleavage of pro-IL-1β to active IL-1β by caspase-1. Using inhibitors for proteosome degradation and the JNK and p38 pathways it was concluded that ricin-mediated translation inhibition caused the disappearance of labile proteins that normally suppress inflammasome formation independent of JNK and p38 kinase activation (Lindauer et al., 2010). Therefore protein synthesis inhibition itself may mediate RTA-induced apoptosis through mechanisms that are independent of stress kinases.&lt;/p>&lt;p>In summary, we have successfully expressed pre- and mature forms of RTA in mammalian cells that retain biological activity. This was achieved by optimizing the codon usage of RTA for bovine ribosomes. We present evidence that a relatively low level of depurination by RTA can trigger protein synthesis inhibition, apoptosis and stress-activated signaling, indicating that a substantial reduction in depurination is necessary to prevent protein synthesis inhibition by RTA in mammalian cells. Our results show that protein synthesis inhibition correlates more linearly with apoptosis than the level of depurination. Further studies are warranted to identify the specific link between protein synthesis inhibition and apoptosis in RTA-treated cells.&lt;/p>&lt;!>&lt;p>Codon optimized RTA sequence. The original Ricinus communis preRTA sequence is shown on the bottom. The sequence optimized for codon usage by Bos taurus is shown on the top. Nucleotides 7–111 represent the signal sequence.&lt;/p>&lt;!>&lt;p>Expression of RTA and RTA mutants in HEK293T/17 cells. Total cell lysates (50 μg for vector, WT and double; 25 μg for E177K) were collected from HEK 293T/17 cells 21 h after transfection. Membranes were immunoblotted with anti-RTA antibody then stripped and reprobed with HSP60 antibody. V= vector; WT = wild-type; P= pre; M = mature; double = P95L/E145K. Blots are representative of 2 to 3 experiments.&lt;/p></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">PubMed Author Manuscript</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="8"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Synthesis, Structural Characterization, and Antibacterial Activity of Novel Erbium(III) Complex Con<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"The novel 3D edta-linked heterometallic complex [Sb2Er(edta)2(H2O)4]NO3·4H2O (H4edta = ethylenedia<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"synthesis,_structural_characterization,_and_antibacterial_activity_of_novel_erbium(iii)_complex_con<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">2,533</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">208</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">12.177885</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"1. Introduction&lt;!>2.1. Materials and Physical Measurements&lt;!>2.2. Synthesis of [Sb2Er(edta)2(H2O)4]<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"&lt;p>Much attention is currently focused on the rational design and controlled synthesis of metal-org<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">PubMed Open Access</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="9"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Physico-chemical properties and catalytic activity of the sol-gel prepared Ce-ion doped LaMnO3 pero<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Ce-doped LaMno 3 perovskite ceramics (La 1−x Ce x Mno 3 ) were synthesized by sol-gel based copre<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"physico-chemical_properties_and_catalytic_activity_of_the_sol-gel_prepared_ce-ion_doped_lamno3_pero<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">5,446</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">213</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">25.568075</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"&lt;!>Catalyst characterization.&lt;!>Specific activity&lt;!>Results and Discussion&lt;!>Redox properties (tpR/<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"&lt;p>Presently, perovskite-based materials are gaining immense popularity in the field of material sc<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Scientific Reports - Nature</span></div></div>
							</td>
					</tr></tbody></table>
		<div class="bg-linear-to-b sticky left-0 border-t border-dashed border-gray-300 from-gray-100 to-white py-3 text-center font-mono text-xs dark:border-gray-700 dark:from-gray-950 dark:to-gray-900">End of preview. <a href="/datasets/griffin/ChemSum/viewer/default/train" class="group"><span class="underline decoration-gray-300 group-hover:decoration-gray-400 dark:decoration-gray-500 dark:group-hover:decoration-gray-300">Expand</span>
						in <svg class="text-lg mr-0.5 inline -translate-y-px text-red-500" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 12 12"><path fill="currentColor" d="M2.5 2h7a1 1 0 0 1 1 1v6a1 1 0 0 1-1 1h-7a1 1 0 0 1-1-1V3a1 1 0 0 1 1-1Zm0 2v2h3V4h-3Zm4 0v2h3V4h-3Zm-4 3v2h3V7h-3Zm4 0v2h3V7h-3Z"></path></svg>Data Studio
					</a></div></div>




			<div class="bg-linear-to-b from-gray-100 to-white dark:from-gray-950 dark:to-gray-900 "><hr class="flex-none -translate-y-px border-t border-dashed border-gray-300 bg-white dark:border-gray-700 dark:bg-gray-950">
					<nav><ul class="flex select-none items-center justify-between space-x-2 text-gray-700 sm:justify-center py-1 text-center font-mono text-xs rounded-b-lg"><li><a class="flex items-center rounded-lg px-2.5 py-1 hover:bg-gray-50 dark:hover:bg-gray-800 pointer-events-none cursor-default text-gray-400 hover:text-gray-700" href=""><svg class="mr-1.5" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M10 16L20 6l1.4 1.4l-8.6 8.6l8.6 8.6L20 26z" fill="currentColor"></path></svg>
		Previous</a></li>
			<li class="hidden sm:block"><a class="rounded-lg px-2.5 py-1 bg-gray-50 font-semibold ring-1 ring-inset ring-gray-200 dark:bg-gray-900 dark:text-yellow-500 dark:ring-gray-900 hover:bg-gray-50 dark:hover:bg-gray-800" href="/datasets/griffin/ChemSum/viewer/default/train?p=0">1</a>
				</li><li class="hidden sm:block"><a class="rounded-lg px-2.5 py-1  hover:bg-gray-50 dark:hover:bg-gray-800" href="/datasets/griffin/ChemSum/viewer/default/train?p=1">2</a>
				</li><li class="hidden sm:block"><a class="rounded-lg px-2.5 py-1  hover:bg-gray-50 dark:hover:bg-gray-800" href="/datasets/griffin/ChemSum/viewer/default/train?p=2">3</a>
				</li><li class="hidden sm:block"><a class="rounded-lg px-2.5 py-1  pointer-events-none cursor-default" href="#">...</a>
				</li><li class="hidden sm:block"><a class="rounded-lg px-2.5 py-1  hover:bg-gray-50 dark:hover:bg-gray-800" href="/datasets/griffin/ChemSum/viewer/default/train?p=1159">1,160</a>
				</li>
			<li><a class="flex items-center rounded-lg px-2.5 py-1 hover:bg-gray-50 dark:hover:bg-gray-800 " href="/datasets/griffin/ChemSum/viewer/default/train?p=1">Next
		<svg class="ml-1.5 transform rotate-180" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M10 16L20 6l1.4 1.4l-8.6 8.6l8.6 8.6L20 26z" fill="currentColor"></path></svg></a></li></ul></nav></div></div>






</div></div></div></div></div></div></div>

				
					<div class="SVELTE_HYDRATER contents" data-target="RepoCodeCopy" data-props="{}"><div></div></div>
					

					<div class="SVELTE_HYDRATER contents" data-target="SideNavigation" data-props="{&quot;titleTree&quot;:[{&quot;id&quot;:&quot;chemsum-description&quot;,&quot;label&quot;:&quot;ChemSum Description&quot;,&quot;children&quot;:[{&quot;id&quot;:&quot;chemsum-summary&quot;,&quot;label&quot;:&quot;ChemSum Summary&quot;,&quot;children&quot;:[],&quot;isValid&quot;:true,&quot;title&quot;:&quot;ChemSum Summary&quot;},{&quot;id&quot;:&quot;languages&quot;,&quot;label&quot;:&quot;Languages&quot;,&quot;children&quot;:[],&quot;isValid&quot;:true,&quot;title&quot;:&quot;Languages&quot;}],&quot;isValid&quot;:true,&quot;title&quot;:&quot;ChemSum Description&quot;},{&quot;id&quot;:&quot;dataset-structure&quot;,&quot;label&quot;:&quot;Dataset Structure&quot;,&quot;children&quot;:[{&quot;id&quot;:&quot;data-fields&quot;,&quot;label&quot;:&quot;Data Fields&quot;,&quot;children&quot;:[],&quot;isValid&quot;:true,&quot;title&quot;:&quot;Data Fields&quot;},{&quot;id&quot;:&quot;data-splits&quot;,&quot;label&quot;:&quot;Data Splits&quot;,&quot;children&quot;:[],&quot;isValid&quot;:true,&quot;title&quot;:&quot;Data Splits&quot;},{&quot;id&quot;:&quot;citation-information&quot;,&quot;label&quot;:&quot;Citation Information&quot;,&quot;children&quot;:[],&quot;isValid&quot;:true,&quot;title&quot;:&quot;Citation Information&quot;}],&quot;isValid&quot;:true,&quot;title&quot;:&quot;Dataset Structure&quot;}],&quot;classNames&quot;:&quot;top-6&quot;}">

<div class="absolute -left-12 bottom-0 top-0 z-10 top-6"><div class="sticky top-4 flex"><div class="h-7 pt-[0.175rem]">
				<span class="peer" tabindex="0"><button class="select-none hover:cursor-pointer"><svg width="1em" height="1em" viewBox="0 0 10 10" class="text-lg opacity-80 hover:opacity-100" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" preserveAspectRatio="xMidYMid meet" fill="currentColor"><path fill-rule="evenodd" clip-rule="evenodd" d="M1.65039 2.9999C1.65039 2.8066 1.80709 2.6499 2.00039 2.6499H8.00039C8.19369 2.6499 8.35039 2.8066 8.35039 2.9999C8.35039 3.1932 8.19369 3.3499 8.00039 3.3499H2.00039C1.80709 3.3499 1.65039 3.1932 1.65039 2.9999ZM1.65039 4.9999C1.65039 4.8066 1.80709 4.6499 2.00039 4.6499H8.00039C8.19369 4.6499 8.35039 4.8066 8.35039 4.9999C8.35039 5.1932 8.19369 5.3499 8.00039 5.3499H2.00039C1.80709 5.3499 1.65039 5.1932 1.65039 4.9999ZM2.00039 6.6499C1.80709 6.6499 1.65039 6.8066 1.65039 6.9999C1.65039 7.1932 1.80709 7.3499 2.00039 7.3499H8.00039C8.19369 7.3499 8.35039 7.1932 8.35039 6.9999C8.35039 6.8066 8.19369 6.6499 8.00039 6.6499H2.00039Z"></path></svg></button></span>
				<div class="invisible w-0 -translate-x-24 -translate-y-6 overflow-hidden rounded-xl border bg-white transition-transform hover:visible hover:w-52 hover:translate-x-0 peer-focus-within:visible peer-focus-within:w-52 peer-focus-within:translate-x-0"><nav aria-label="Secondary" class="max-h-[550px] overflow-y-auto p-3"><ul><li class="mb-3 text-sm last:mb-0"><a class="mb-1 block break-words font-semibold text-gray-700 *:break-words hover:underline active:text-gray-900 dark:active:text-gray-200" href="#chemsum-description" title="ChemSum Description"><!-- HTML_TAG_START -->ChemSum Description<!-- HTML_TAG_END --></a>
									<ul class="pl-1"><li><a class="mb-0.5 block break-words hover:underline active:text-gray-700 dark:active:text-gray-300 text-gray-500" href="#chemsum-summary" title="ChemSum Summary"><!-- HTML_TAG_START -->ChemSum Summary<!-- HTML_TAG_END --></a>
												<ul class="pl-2"></ul>
											</li><li><a class="mb-0.5 block break-words hover:underline active:text-gray-700 dark:active:text-gray-300 text-gray-500" href="#languages" title="Languages"><!-- HTML_TAG_START -->Languages<!-- HTML_TAG_END --></a>
												<ul class="pl-2"></ul>
											</li></ul>
								</li><li class="mb-3 text-sm last:mb-0"><a class="mb-1 block break-words font-semibold text-gray-700 *:break-words hover:underline active:text-gray-900 dark:active:text-gray-200" href="#dataset-structure" title="Dataset Structure"><!-- HTML_TAG_START -->Dataset Structure<!-- HTML_TAG_END --></a>
									<ul class="pl-1"><li><a class="mb-0.5 block break-words hover:underline active:text-gray-700 dark:active:text-gray-300 text-gray-500" href="#data-fields" title="Data Fields"><!-- HTML_TAG_START -->Data Fields<!-- HTML_TAG_END --></a>
												<ul class="pl-2"></ul>
											</li><li><a class="mb-0.5 block break-words hover:underline active:text-gray-700 dark:active:text-gray-300 text-gray-500" href="#data-splits" title="Data Splits"><!-- HTML_TAG_START -->Data Splits<!-- HTML_TAG_END --></a>
												<ul class="pl-2"></ul>
											</li><li><a class="mb-0.5 block break-words hover:underline active:text-gray-700 dark:active:text-gray-300 text-gray-500" href="#citation-information" title="Citation Information"><!-- HTML_TAG_START -->Citation Information<!-- HTML_TAG_END --></a>
												<ul class="pl-2"></ul>
											</li></ul>
								</li></ul></nav></div></div></div></div></div>
					<div class="2xl:pr-6"><div class="prose pl-6 -ml-6 hf-sanitized hf-sanitized-G37dpiZG--yFMiB_GwEbk">
	<!-- HTML_TAG_START --><h1 class="relative group flex items-center">
	<a rel="nofollow" href="#dataset-card-for-chemsum" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full" id="dataset-card-for-chemsum">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		Dataset Card for ChemSum
	</span>
</h1>
<h2 class="relative group flex items-center">
	<a rel="nofollow" href="#chemsum-description" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full" id="chemsum-description">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		ChemSum Description
	</span>
</h2>

<ul>
<li><strong>Paper:</strong> <a rel="nofollow" href="https://arxiv.org/abs/2305.07615">What are the Desired Characteristics of Calibration Sets? Identifying Correlates on Long Form Scientific Summarization </a></li>
<li><strong>Journal:</strong> ACL 2023</li>
<li><strong>Point of Contact:</strong> <a rel="nofollow" href="mailto:griffin.adams@columbia.edu">griffin.adams@columbia.edu</a></li>
<li><strong>Repository:</strong> <a rel="nofollow" href="https://github.com/griff4692/calibrating-summaries">https://github.com/griff4692/calibrating-summaries</a></li>
</ul>
<h3 class="relative group flex items-center">
	<a rel="nofollow" href="#chemsum-summary" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full" id="chemsum-summary">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		ChemSum Summary
	</span>
</h3>
<p>We introduce a dataset with a pure chemistry focus by compiling a list of chemistry academic journals with Open-Access articles. For each journal, we downloaded full-text article PDFs from the Open-Access portion of the journal using available APIs, or scraping this content using <a rel="nofollow" href="https://www.selenium.dev/documentation/webdriver/">Selenium Chrome WebDriver</a>. 
Each PDF was processed with Grobid via a locally installed <a rel="nofollow" href="https://pypi.org/project/grobid-client-python/">client</a> to extract free-text paragraphs with sections. </p>
<p>The table below shows the journals from which Open Access articles were sourced, as well as the number of papers processed.</p>
<p>For all journals, we filtered for papers with the provided topic of Chemistry when papers from other disciplines were also available (e.g. PubMed).</p>
<div class="max-w-full overflow-auto">
	<table>
		<thead><tr>
<th>Source</th>
<th># of Articles</th>
</tr>

		</thead><tbody><tr>
<td>Beilstein</td>
<td>1,829</td>
</tr>
<tr>
<td>Chem Cell</td>
<td>546</td>
</tr>
<tr>
<td>ChemRxiv</td>
<td>12,231</td>
</tr>
<tr>
<td>Chemistry Open</td>
<td>398</td>
</tr>
<tr>
<td>Nature Communications Chemistry</td>
<td>572</td>
</tr>
<tr>
<td>PubMed Author Manuscript</td>
<td>57,680</td>
</tr>
<tr>
<td>PubMed Open Access</td>
<td>29,540</td>
</tr>
<tr>
<td>Royal Society of Chemistry (RSC)</td>
<td>9,334</td>
</tr>
<tr>
<td>Scientific Reports - Nature</td>
<td>6,826</td>
</tr>
</tbody>
	</table>
</div>


<h3 class="relative group flex items-center">
	<a rel="nofollow" href="#languages" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full" id="languages">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		Languages
	</span>
</h3>
<p>English</p>
<h2 class="relative group flex items-center">
	<a rel="nofollow" href="#dataset-structure" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full" id="dataset-structure">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		Dataset Structure
	</span>
</h2>


<h3 class="relative group flex items-center">
	<a rel="nofollow" href="#data-fields" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full" id="data-fields">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		Data Fields
	</span>
</h3>
<div class="max-w-full overflow-auto">
	<table>
		<thead><tr>
<th>Column</th>
<th>Description</th>
</tr>

		</thead><tbody><tr>
<td><code>uuid</code></td>
<td>Unique Identifier for the Example</td>
</tr>
<tr>
<td><code>title</code></td>
<td>Title of the Article</td>
</tr>
<tr>
<td><code>article_source</code></td>
<td>Open Source Journal (see above for list)</td>
</tr>
<tr>
<td><code>abstract</code></td>
<td>Abstract (summary reference)</td>
</tr>
<tr>
<td><code>sections</code></td>
<td>Full-text sections from the main body of paper (&lt;!&gt; indicates section boundaries)</td>
</tr>
<tr>
<td><code>headers</code></td>
<td>Corresponding section headers for <code>sections</code> field (&lt;!&gt; delimited)</td>
</tr>
<tr>
<td><code>source_toks</code></td>
<td>Aggregate number of tokens across <code>sections</code></td>
</tr>
<tr>
<td><code>target_toks</code></td>
<td>Number of tokens in the <code>abstract</code></td>
</tr>
<tr>
<td><code>compression</code></td>
<td>Ratio of <code>source_toks</code> to <code>target_toks</code></td>
</tr>
</tbody>
	</table>
</div>
<p>Please refer to <code>load_chemistry()</code> in <a rel="nofollow" href="https://github.com/griff4692/calibrating-summaries/blob/master/preprocess/preprocess.py">https://github.com/griff4692/calibrating-summaries/blob/master/preprocess/preprocess.py</a> for pre-processing as a summarization dataset. The inputs are <code>sections</code> and <code>headers</code> and the targets is the <code>abstract</code>.</p>
<h3 class="relative group flex items-center">
	<a rel="nofollow" href="#data-splits" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full" id="data-splits">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		Data Splits
	</span>
</h3>
<div class="max-w-full overflow-auto">
	<table>
		<thead><tr>
<th>Split</th>
<th>Count</th>
</tr>

		</thead><tbody><tr>
<td><code>train</code></td>
<td>115,956</td>
</tr>
<tr>
<td><code>validation</code></td>
<td>1,000</td>
</tr>
<tr>
<td><code>test</code></td>
<td>2,000</td>
</tr>
</tbody>
	</table>
</div>
<h3 class="relative group flex items-center">
	<a rel="nofollow" href="#citation-information" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full" id="citation-information">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		Citation Information
	</span>
</h3>
<pre><code>@inproceedings{adams-etal-2023-desired,
    title = "What are the Desired Characteristics of Calibration Sets? Identifying Correlates on Long Form Scientific Summarization",
    author = "Adams, Griffin  and
      Nguyen, Bichlien  and
      Smith, Jake  and
      Xia, Yingce  and
      Xie, Shufang  and
      Ostropolets, Anna  and
      Deb, Budhaditya  and
      Chen, Yuan-Jyue  and
      Naumann, Tristan  and
      Elhadad, No{\'e}mie",
    editor = "Rogers, Anna  and
      Boyd-Graber, Jordan  and
      Okazaki, Naoaki",
    booktitle = "Proceedings of the 61st Annual Meeting of the Association for Computational Linguistics (Volume 1: Long Papers)",
    month = jul,
    year = "2023",
    address = "Toronto, Canada",
    publisher = "Association for Computational Linguistics",
    url = "https://aclanthology.org/2023.acl-long.587",
    doi = "10.18653/v1/2023.acl-long.587",
    pages = "10520--10542",
    abstract = "Summarization models often generate text that is poorly calibrated to quality metrics because they are trained to maximize the likelihood of a single reference (MLE). To address this, recent work has added a calibration step, which exposes a model to its own ranked outputs to improve relevance or, in a separate line of work, contrasts positive and negative sets to improve faithfulness. While effective, much of this work has focused on \textit{how} to generate and optimize these sets. Less is known about \textit{why} one setup is more effective than another. In this work, we uncover the underlying characteristics of effective sets. For each training instance, we form a large, diverse pool of candidates and systematically vary the subsets used for calibration fine-tuning. Each selection strategy targets distinct aspects of the sets, such as lexical diversity or the size of the gap between positive and negatives. On three diverse scientific long-form summarization datasets (spanning biomedical, clinical, and chemical domains), we find, among others, that faithfulness calibration is optimal when the negative sets are extractive and more likely to be generated, whereas for relevance calibration, the metric margin between candidates should be maximized and surprise{--}the disagreement between model and metric defined candidate rankings{--}minimized.",
}
</code></pre>

<!-- HTML_TAG_END --></div>
</div></section>
			
			<section class="pt-6 border-gray-100 md:pb-24 md:pl-6 md:w-64 lg:w-80 xl:w-96 flex-none order-first md:order-none md:border-l pt-3! md:pt-6!"><dl class="flex items-baseline justify-between"><dt class="text-sm text-gray-500">Downloads last month</dt><div class="mx-4 flex-1 border-b border-dotted"></div><dd class="font-semibold">725</dd></dl>
				
				<div class="divider-column-vertical"></div>
				<div class="grid grid-cols-2 gap-x-2 md:flex md:flex-row md:flex-wrap"><div class="SVELTE_HYDRATER contents" data-target="DatasetAndModelActionsDropdown" data-props="{&quot;classNames&quot;:&quot;order-last&quot;,&quot;discussionsDisabled&quot;:false,&quot;repo&quot;:{&quot;type&quot;:&quot;dataset&quot;,&quot;name&quot;:&quot;griffin/ChemSum&quot;},&quot;canWrite&quot;:false,&quot;canDisable&quot;:false,&quot;repoIsPrivate&quot;:false,&quot;repoIsGated&quot;:false,&quot;repoIsDisabled&quot;:false,&quot;repoIsAdminFlaggedNFAA&quot;:false,&quot;repoHasBlockedOids&quot;:false}"><div class="order-last"><div class="relative ">
	<button class="btn px-1.5 py-1.5 " type="button">
		
			<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" class="p-0.5" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><circle cx="16" cy="7" r="3" fill="currentColor"></circle><circle cx="16" cy="16" r="3" fill="currentColor"></circle><circle cx="16" cy="25" r="3" fill="currentColor"></circle></svg>
		
		</button>
	
	
	</div></div>













</div>
					<div class="SVELTE_HYDRATER contents" data-target="DatasetLibrary" data-props="{&quot;classNames&quot;:&quot;md:w-full xl:w-auto xl:flex-none&quot;,&quot;libraries&quot;:[{&quot;language&quot;:&quot;python&quot;,&quot;library&quot;:&quot;datasets&quot;,&quot;function&quot;:&quot;load_dataset&quot;,&quot;loading_codes&quot;:[{&quot;config_name&quot;:&quot;default&quot;,&quot;arguments&quot;:{},&quot;code&quot;:&quot;from datasets import load_dataset\n\nds = load_dataset(\&quot;griffin/ChemSum\&quot;)&quot;}]},{&quot;language&quot;:&quot;python&quot;,&quot;library&quot;:&quot;dask&quot;,&quot;function&quot;:&quot;dd.read_parquet&quot;,&quot;loading_codes&quot;:[{&quot;config_name&quot;:&quot;default&quot;,&quot;arguments&quot;:{&quot;splits&quot;:{&quot;train&quot;:&quot;data/train-*-of-*.parquet&quot;,&quot;validation&quot;:&quot;data/validation-00000-of-00001.parquet&quot;,&quot;test&quot;:&quot;data/test-00000-of-00001.parquet&quot;}},&quot;code&quot;:&quot;import dask.dataframe as dd\n\nsplits = {'train': 'data/train-*-of-*.parquet', 'validation': 'data/validation-00000-of-00001.parquet', 'test': 'data/test-00000-of-00001.parquet'}\ndf = dd.read_parquet(\&quot;hf://datasets/griffin/ChemSum/\&quot; + splits[\&quot;train\&quot;])&quot;}]},{&quot;language&quot;:&quot;python&quot;,&quot;library&quot;:&quot;mlcroissant&quot;,&quot;function&quot;:&quot;Dataset&quot;,&quot;loading_codes&quot;:[{&quot;config_name&quot;:&quot;default&quot;,&quot;arguments&quot;:{&quot;record_set&quot;:&quot;default&quot;,&quot;partial&quot;:false},&quot;code&quot;:&quot;from mlcroissant import Dataset\n\nds = Dataset(jsonld=\&quot;https://huggingface.co/api/datasets/griffin/ChemSum/croissant\&quot;)\nrecords = ds.records(\&quot;default\&quot;)&quot;}]},{&quot;language&quot;:&quot;python&quot;,&quot;library&quot;:&quot;polars&quot;,&quot;function&quot;:&quot;pl.read_parquet&quot;,&quot;loading_codes&quot;:[{&quot;config_name&quot;:&quot;default&quot;,&quot;arguments&quot;:{&quot;splits&quot;:{&quot;train&quot;:&quot;data/train-*-of-*.parquet&quot;,&quot;validation&quot;:&quot;data/validation-00000-of-00001.parquet&quot;,&quot;test&quot;:&quot;data/test-00000-of-00001.parquet&quot;}},&quot;code&quot;:&quot;import polars as pl\n\nsplits = {'train': 'data/train-*-of-*.parquet', 'validation': 'data/validation-00000-of-00001.parquet', 'test': 'data/test-00000-of-00001.parquet'}\ndf = pl.read_parquet('hf://datasets/griffin/ChemSum/' + splits['train'])\n&quot;}]}]}"><div class="relative md:w-full xl:w-auto xl:flex-none">
	<button class="from-gray-800! to-black! max-xl:mb-2 text-white! gap-1! border-gray-800! dark:border-gray-900!  btn w-full cursor-pointer text-sm" type="button">
		<svg class="mr-1.5 mr-0.5! " xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32" style="transform: rotate(360deg);"><path d="M31 16l-7 7l-1.41-1.41L28.17 16l-5.58-5.59L24 9l7 7z" fill="currentColor"></path><path d="M1 16l7-7l1.41 1.41L3.83 16l5.58 5.59L8 23l-7-7z" fill="currentColor"></path><path d="M12.419 25.484L17.639 6l1.932.518L14.35 26z" fill="currentColor"></path></svg>
			Use this dataset
		<svg class="-mr-1 text-gray-500" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 24 24"><path d="M16.293 9.293L12 13.586L7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z" fill="currentColor"></path></svg></button>
	
	
	</div>

</div>
					
					</div>
				<div class="divider-column-vertical"></div>
					<div class="flex flex-col flex-wrap xl:flex-row"><a class="bg-linear-to-r dark:via-none group mb-1.5 flex max-w-full flex-col overflow-hidden rounded-lg border border-gray-100 from-white via-white to-white px-2 py-1 hover:from-gray-50 dark:from-gray-900 dark:to-gray-925 dark:hover:to-gray-900 md:mr-1.5 " href="/datasets/griffin/ChemSum/tree/refs%2Fconvert%2Fparquet/" rel="nofollow" target="_blank"><div class="truncate text-xs text-gray-400">Size of the auto-converted Parquet files:</div>
		<div class="truncate text-sm group-hover:underline">
			<!-- HTML_TAG_START -->1.9 GB<!-- HTML_TAG_END --></div></a><a class="bg-linear-to-r dark:via-none group mb-1.5 flex max-w-full flex-col overflow-hidden rounded-lg border border-gray-100 from-white via-white to-white px-2 py-1 hover:from-gray-50 dark:from-gray-900 dark:to-gray-925 dark:hover:to-gray-900 md:mr-1.5 pointer-events-none" rel="nofollow" target="_blank"><div class="truncate text-xs text-gray-400">Number of rows:</div>
		<div class="truncate text-sm ">
			<!-- HTML_TAG_START -->118,956<!-- HTML_TAG_END --></div></a></div>
				
				
				
				

				
				<div class="divider-column-vertical md:hidden"></div></section></div></main>

	<footer class="b-12 mb-2 flex border-t border-gray-100 md:h-14"><nav class="container relative flex flex-col justify-between space-y-2 py-6 text-gray-500 max-md:*:self-start md:flex-row md:items-center md:space-y-0 md:py-0 md:text-sm"><div class="SVELTE_HYDRATER contents" data-target="ThemeSwitcher" data-props="{&quot;theme&quot;:&quot;system&quot;,&quot;isLoggedIn&quot;:false,&quot;menuClassNames&quot;:&quot;md:-top-24&quot;,&quot;classNames&quot;:&quot;max-md:mb-5 max-md:*:self-start&quot;}">
<div class="relative inline-block max-md:mb-5 max-md:*:self-start">
	<button class="rounded-full border border-gray-100 pl-2 py-1 pr-2.5  flex items-center text-sm text-gray-500 bg-white hover:bg-purple-50 hover:border-purple-200 dark:hover:bg-gray-800 dark:hover:border-gray-950 dark:border-gray-800 " type="button">
		<svg class="mr-1.5 text-gray-500" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" fill="currentColor" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M29 25H3a1 1 0 1 0 0 2h26a1 1 0 1 0 0-2Z" fill="currentColor"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M6 22.5h20a2 2 0 0 0 2-2V7a2 2 0 0 0-2-2H6a2 2 0 0 0-2 2v13.5a2 2 0 0 0 2 2ZM7 7a1 1 0 0 0-1 1v11a1 1 0 0 0 1 1h18a1 1 0 0 0 1-1V8a1 1 0 0 0-1-1H7Z" fill="currentColor"></path><path d="M6 8a1 1 0 0 1 1-1h18a1 1 0 0 1 1 1v11a1 1 0 0 1-1 1H7a1 1 0 0 1-1-1V8Z" fill="currentColor" fill-opacity=".4"></path><path d="M29 25H3a1 1 0 1 0 0 2h26a1 1 0 1 0 0-2Z" fill="currentColor"></path></svg>
			System theme
		</button>
	
	
	</div></div>
		<div class="font-semibold text-black md:hidden">Company</div>
		<a class="hover:underline" href="/terms-of-service">TOS</a>
		<a class="hover:underline" href="/privacy">Privacy</a>
		<a class="hover:underline" href="/huggingface">About</a>
		<a class="hover:underline" href="https://apply.workable.com/huggingface/">Jobs</a>
		<a href="/" class="max-md:mb-4! max-md:mt-8! group flex-none max-md:order-last"><svg class="h-7 w-7 transition-transform group-hover:-translate-y-px" viewBox="0 0 95 88" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M47.2119 76.5C66.4037 76.5 81.9619 60.9419 81.9619 41.75C81.9619 22.5581 66.4037 7 47.2119 7C28.02 7 12.4619 22.5581 12.4619 41.75C12.4619 60.9419 28.02 76.5 47.2119 76.5Z" fill="#FFD21E"></path><path d="M81.9619 41.75C81.9619 22.5581 66.4037 7 47.2119 7C28.02 7 12.4619 22.5581 12.4619 41.75C12.4619 60.9419 28.02 76.5 47.2119 76.5C66.4037 76.5 81.9619 60.9419 81.9619 41.75ZM8.46185 41.75C8.46185 20.349 25.8108 3 47.2119 3C68.6129 3 85.9619 20.349 85.9619 41.75C85.9619 63.151 68.6129 80.5 47.2119 80.5C25.8108 80.5 8.46185 63.151 8.46185 41.75Z" fill="#FF9D0B"></path><path d="M58.5024 32.2915C59.7768 32.7415 60.2839 35.3615 61.5713 34.6769C64.0095 33.3805 64.9351 30.353 63.6387 27.9148C62.3423 25.4767 59.3148 24.5511 56.8766 25.8475C54.4384 27.1439 53.5128 30.1714 54.8092 32.6096C55.4211 33.7604 57.3632 31.8892 58.5024 32.2915Z" fill="#3A3B45"></path><path d="M34.9454 32.2915C33.671 32.7415 33.164 35.3615 31.8766 34.6769C29.4384 33.3805 28.5128 30.353 29.8092 27.9148C31.1056 25.4767 34.1331 24.5511 36.5713 25.8475C39.0095 27.1439 39.9351 30.1714 38.6387 32.6096C38.0268 33.7604 36.0846 31.8892 34.9454 32.2915Z" fill="#3A3B45"></path><path d="M46.9619 56.289C56.7903 56.289 59.9619 47.5261 59.9619 43.0262C59.9619 40.6875 58.3898 41.4236 55.8718 42.6702C53.5449 43.8222 50.4102 45.4101 46.9619 45.4101C39.7822 45.4101 33.9619 38.5263 33.9619 43.0262C33.9619 47.5261 37.1334 56.289 46.9619 56.289Z" fill="#3A3B45"></path><mask id="mask0" style="mask-type:alpha" maskUnits="userSpaceOnUse" x="33" y="41" width="27" height="16"><path d="M46.9619 56.289C56.7903 56.289 59.9619 47.5261 59.9619 43.0262C59.9619 40.6875 58.3898 41.4236 55.8718 42.6702C53.5449 43.8222 50.4102 45.4101 46.9619 45.4101C39.7822 45.4101 33.9619 38.5263 33.9619 43.0262C33.9619 47.5261 37.1334 56.289 46.9619 56.289Z" fill="white"></path></mask><g mask="url(#mask0)"><path d="M47.2119 66.5C52.0018 66.5 55.8848 62.617 55.8848 57.8271C55.8848 54.0962 53.5291 50.9156 50.224 49.6915C50.1023 49.6464 49.9794 49.604 49.8553 49.5643C49.0219 49.2979 48.1337 52.1623 47.2119 52.1623C46.3506 52.1623 45.5186 49.2797 44.7332 49.5135C41.151 50.5799 38.5389 53.8984 38.5389 57.8271C38.5389 62.617 42.4219 66.5 47.2119 66.5Z" fill="#F94040"></path></g><path d="M70.7119 37C72.5068 37 73.9619 35.5449 73.9619 33.75C73.9619 31.9551 72.5068 30.5 70.7119 30.5C68.9169 30.5 67.4619 31.9551 67.4619 33.75C67.4619 35.5449 68.9169 37 70.7119 37Z" fill="#FF9D0B"></path><path d="M24.2119 37C26.0068 37 27.4619 35.5449 27.4619 33.75C27.4619 31.9551 26.0068 30.5 24.2119 30.5C22.4169 30.5 20.9619 31.9551 20.9619 33.75C20.9619 35.5449 22.4169 37 24.2119 37Z" fill="#FF9D0B"></path><path class="origin-bottom-right transition-transform group-hover:-rotate-6" d="M17.5238 48C15.9048 48 14.4578 48.665 13.4488 49.871C12.8248 50.618 12.1728 51.822 12.1198 53.625C11.4408 53.43 10.7878 53.321 10.1778 53.321C8.6278 53.321 7.2278 53.915 6.2378 54.994C4.9658 56.379 4.4008 58.081 4.6468 59.784C4.7638 60.595 5.0348 61.322 5.4398 61.995C4.5858 62.686 3.9568 63.648 3.6528 64.805C3.4148 65.712 3.1708 67.601 4.4448 69.547C4.3638 69.674 4.2878 69.806 4.2168 69.941C3.4508 71.395 3.4018 73.038 4.0778 74.568C5.1028 76.887 7.6498 78.714 12.5958 80.675C15.6728 81.895 18.4878 82.675 18.5128 82.682C22.5808 83.737 26.2598 84.273 29.4448 84.273C35.2988 84.273 39.4898 82.48 41.9018 78.944C45.7838 73.25 45.2288 68.042 40.2058 63.022C37.4258 60.244 35.5778 56.148 35.1928 55.249C34.4168 52.587 32.3648 49.628 28.9538 49.628H28.9528C28.6658 49.628 28.3758 49.651 28.0898 49.696C26.5958 49.931 25.2898 50.791 24.3568 52.085C23.3498 50.833 22.3718 49.837 21.4868 49.275C20.1528 48.429 18.8198 48 17.5238 48ZM17.5238 52C18.0338 52 18.6568 52.217 19.3438 52.653C21.4768 54.006 25.5928 61.081 27.0998 63.833C27.6048 64.755 28.4678 65.145 29.2448 65.145C30.7868 65.145 31.9908 63.612 29.3858 61.664C25.4688 58.733 26.8428 53.942 28.7128 53.647C28.7948 53.634 28.8758 53.628 28.9538 53.628C30.6538 53.628 31.4038 56.558 31.4038 56.558C31.4038 56.558 33.6018 62.078 37.3778 65.851C41.1538 69.625 41.3488 72.654 38.5968 76.69C36.7198 79.442 33.1268 80.273 29.4448 80.273C25.6258 80.273 21.7108 79.379 19.5168 78.81C19.4088 78.782 6.0658 75.013 7.7558 71.805C8.0398 71.266 8.5078 71.05 9.0968 71.05C11.4768 71.05 15.8058 74.592 17.6668 74.592C18.0828 74.592 18.3758 74.415 18.4958 73.983C19.2888 71.138 6.4388 69.942 7.5218 65.821C7.7128 65.092 8.2308 64.796 8.9588 64.797C12.1038 64.797 19.1598 70.328 20.6388 70.328C20.7518 70.328 20.8328 70.295 20.8768 70.225C21.6178 69.029 21.2118 68.194 15.9888 65.033C10.7658 61.871 7.0998 59.969 9.1848 57.699C9.4248 57.437 9.7648 57.321 10.1778 57.321C13.3488 57.322 20.8408 64.14 20.8408 64.14C20.8408 64.14 22.8628 66.243 24.0858 66.243C24.3668 66.243 24.6058 66.132 24.7678 65.858C25.6348 64.396 16.7148 57.636 16.2118 54.847C15.8708 52.957 16.4508 52 17.5238 52Z" fill="#FF9D0B"></path><path class="origin-bottom-right transition-transform group-hover:-rotate-6" d="M38.5967 76.6898C41.3487 72.6538 41.1537 69.6248 37.3777 65.8508C33.6017 62.0778 31.4037 56.5578 31.4037 56.5578C31.4037 56.5578 30.5827 53.3518 28.7127 53.6468C26.8427 53.9418 25.4697 58.7328 29.3867 61.6638C33.3037 64.5938 28.6067 66.5848 27.0997 63.8328C25.5927 61.0808 21.4777 54.0058 19.3437 52.6528C17.2107 51.2998 15.7087 52.0578 16.2117 54.8468C16.7147 57.6358 25.6357 64.3958 24.7677 65.8588C23.8997 67.3208 20.8407 64.1398 20.8407 64.1398C20.8407 64.1398 11.2687 55.4288 9.18465 57.6988C7.10065 59.9688 10.7657 61.8708 15.9887 65.0328C21.2127 68.1938 21.6177 69.0288 20.8767 70.2248C20.1347 71.4208 8.60465 61.6998 7.52165 65.8208C6.43965 69.9418 19.2887 71.1378 18.4957 73.9828C17.7027 76.8288 9.44465 68.5978 7.75565 71.8048C6.06565 75.0128 19.4087 78.7818 19.5167 78.8098C23.8267 79.9278 34.7727 82.2968 38.5967 76.6898Z" fill="#FFD21E"></path><path class="origin-bottom-left transition-transform group-hover:rotate-6" d="M77.3999 48C79.0189 48 80.4659 48.665 81.4749 49.871C82.0989 50.618 82.7509 51.822 82.8039 53.625C83.4829 53.43 84.1359 53.321 84.7459 53.321C86.2959 53.321 87.6959 53.915 88.6859 54.994C89.9579 56.379 90.5229 58.081 90.2769 59.784C90.1599 60.595 89.8889 61.322 89.4839 61.995C90.3379 62.686 90.9669 63.648 91.2709 64.805C91.5089 65.712 91.7529 67.601 90.4789 69.547C90.5599 69.674 90.6359 69.806 90.7069 69.941C91.4729 71.395 91.5219 73.038 90.8459 74.568C89.8209 76.887 87.2739 78.714 82.3279 80.675C79.2509 81.895 76.4359 82.675 76.4109 82.682C72.3429 83.737 68.6639 84.273 65.4789 84.273C59.6249 84.273 55.4339 82.48 53.0219 78.944C49.1399 73.25 49.6949 68.042 54.7179 63.022C57.4979 60.244 59.3459 56.148 59.7309 55.249C60.5069 52.587 62.5589 49.628 65.9699 49.628H65.9709C66.2579 49.628 66.5479 49.651 66.8339 49.696C68.3279 49.931 69.6339 50.791 70.5669 52.085C71.5739 50.833 72.5519 49.837 73.4369 49.275C74.7709 48.429 76.1039 48 77.3999 48ZM77.3999 52C76.8899 52 76.2669 52.217 75.5799 52.653C73.4469 54.006 69.3309 61.081 67.8239 63.833C67.3189 64.755 66.4559 65.145 65.6789 65.145C64.1369 65.145 62.9329 63.612 65.5379 61.664C69.4549 58.733 68.0809 53.942 66.2109 53.647C66.1289 53.634 66.0479 53.628 65.9699 53.628C64.2699 53.628 63.5199 56.558 63.5199 56.558C63.5199 56.558 61.3219 62.078 57.5459 65.851C53.7699 69.625 53.5749 72.654 56.3269 76.69C58.2039 79.442 61.7969 80.273 65.4789 80.273C69.2979 80.273 73.2129 79.379 75.4069 78.81C75.5149 78.782 88.8579 75.013 87.1679 71.805C86.8839 71.266 86.4159 71.05 85.8269 71.05C83.4469 71.05 79.1179 74.592 77.2569 74.592C76.8409 74.592 76.5479 74.415 76.4279 73.983C75.6349 71.138 88.4849 69.942 87.4019 65.821C87.2109 65.092 86.6929 64.796 85.9649 64.797C82.8199 64.797 75.7639 70.328 74.2849 70.328C74.1719 70.328 74.0909 70.295 74.0469 70.225C73.3059 69.029 73.7119 68.194 78.9349 65.033C84.1579 61.871 87.8239 59.969 85.7389 57.699C85.4989 57.437 85.1589 57.321 84.7459 57.321C81.5749 57.322 74.0829 64.14 74.0829 64.14C74.0829 64.14 72.0609 66.243 70.8379 66.243C70.5569 66.243 70.3179 66.132 70.1559 65.858C69.2889 64.396 78.2089 57.636 78.7119 54.847C79.0529 52.957 78.4729 52 77.3999 52Z" fill="#FF9D0B"></path><path class="origin-bottom-left transition-transform group-hover:rotate-6" d="M56.3271 76.6898C53.5751 72.6538 53.7701 69.6248 57.5461 65.8508C61.3221 62.0778 63.5201 56.5578 63.5201 56.5578C63.5201 56.5578 64.3411 53.3518 66.2111 53.6468C68.0811 53.9418 69.4541 58.7328 65.5371 61.6638C61.6201 64.5938 66.3171 66.5848 67.8241 63.8328C69.3311 61.0808 73.4461 54.0058 75.5801 52.6528C77.7131 51.2998 79.2151 52.0578 78.7121 54.8468C78.2091 57.6358 69.2881 64.3958 70.1561 65.8588C71.0241 67.3208 74.0831 64.1398 74.0831 64.1398C74.0831 64.1398 83.6551 55.4288 85.7391 57.6988C87.8231 59.9688 84.1581 61.8708 78.9351 65.0328C73.7111 68.1938 73.3061 69.0288 74.0471 70.2248C74.7891 71.4208 86.3191 61.6998 87.4021 65.8208C88.4841 69.9418 75.6351 71.1378 76.4281 73.9828C77.2211 76.8288 85.4791 68.5978 87.1681 71.8048C88.8581 75.0128 75.5151 78.7818 75.4071 78.8098C71.0971 79.9278 60.1511 82.2968 56.3271 76.6898Z" fill="#FFD21E"></path></svg></a>
		<div class="max-md:mt-8! font-semibold text-black md:hidden">Website</div>

		<a class="hover:underline" href="/models">Models</a>
		<a class="hover:underline" href="/datasets">Datasets</a>
		<a class="hover:underline" href="/spaces">Spaces</a>
		<a class="hover:underline" href="/pricing">Pricing</a>
		<a class="hover:underline" href="/docs">Docs</a></nav></footer></div>

		<script>
			import("\/front\/build\/kube-68d7aa0\/index.js");
			window.moonSha = "kube-68d7aa0\/";
			window.__hf_deferred = {};
		</script>

		<!-- Stripe -->
		<script>
			if (["hf.co", "huggingface.co"].includes(window.location.hostname)) {
				const script = document.createElement("script");
				script.src = "https://js.stripe.com/v3/";
				script.async = true;
				document.head.appendChild(script);
			}
		</script>
	</body>
</html>
