<!doctype html>
<html class="">
	<head>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no" />
		<meta name="description" content="We’re on a journey to advance and democratize artificial intelligence through open source and open science." />
		<meta property="fb:app_id" content="1321688464574422" />
		<meta name="twitter:card" content="summary_large_image" />
		<meta name="twitter:site" content="@huggingface" />
		<meta name="twitter:image" content="https://cdn-thumbnails.huggingface.co/social-thumbnails/datasets/TomTBT/pmc_open_access_section.png" />
		<meta property="og:title" content="TomTBT/pmc_open_access_section · Datasets at Hugging Face" />
		<meta property="og:type" content="website" />
		<meta property="og:url" content="https://huggingface.co/datasets/TomTBT/pmc_open_access_section" />
		<meta property="og:image" content="https://cdn-thumbnails.huggingface.co/social-thumbnails/datasets/TomTBT/pmc_open_access_section.png" />

		<link rel="stylesheet" href="/front/build/kube-68d7aa0/style.css" />

		<link rel="preconnect" href="https://fonts.gstatic.com" />
		<link
			href="https://fonts.googleapis.com/css2?family=Source+Sans+Pro:ital,wght@0,200;0,300;0,400;0,600;0,700;0,900;1,200;1,300;1,400;1,600;1,700;1,900&display=swap"
			rel="stylesheet"
		/>
		<link
			href="https://fonts.googleapis.com/css2?family=IBM+Plex+Mono:wght@400;600;700&display=swap"
			rel="stylesheet"
		/>

		<link
			rel="preload"
			href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.12.0/katex.min.css"
			as="style"
			onload="this.onload=null;this.rel='stylesheet'"
		/>
		<noscript>
			<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.12.0/katex.min.css" />
		</noscript>

		<script>const guestTheme = document.cookie.match(/theme=(\w+)/)?.[1]; document.documentElement.classList.toggle('dark', guestTheme === 'dark' || ( (!guestTheme || guestTheme === 'system') && window.matchMedia('(prefers-color-scheme: dark)').matches));</script>
<link rel="canonical" href="https://huggingface.co/datasets/TomTBT/pmc_open_access_section"> <script type="application/ld+json">{
  "@context": {
    "@language": "en",
    "@vocab": "https:\/\/schema.org\/",
    "citeAs": "cr:citeAs",
    "column": "cr:column",
    "conformsTo": "dct:conformsTo",
    "cr": "http:\/\/mlcommons.org\/croissant\/",
    "data": {
      "@id": "cr:data",
      "@type": "@json"
    },
    "dataBiases": "cr:dataBiases",
    "dataCollection": "cr:dataCollection",
    "dataType": {
      "@id": "cr:dataType",
      "@type": "@vocab"
    },
    "dct": "http:\/\/purl.org\/dc\/terms\/",
    "extract": "cr:extract",
    "field": "cr:field",
    "fileProperty": "cr:fileProperty",
    "fileObject": "cr:fileObject",
    "fileSet": "cr:fileSet",
    "format": "cr:format",
    "includes": "cr:includes",
    "isLiveDataset": "cr:isLiveDataset",
    "jsonPath": "cr:jsonPath",
    "key": "cr:key",
    "md5": "cr:md5",
    "parentField": "cr:parentField",
    "path": "cr:path",
    "personalSensitiveInformation": "cr:personalSensitiveInformation",
    "recordSet": "cr:recordSet",
    "references": "cr:references",
    "regex": "cr:regex",
    "repeated": "cr:repeated",
    "replace": "cr:replace",
    "sc": "https:\/\/schema.org\/",
    "separator": "cr:separator",
    "source": "cr:source",
    "subField": "cr:subField",
    "transform": "cr:transform"
  },
  "@type": "sc:Dataset",
  "distribution": [
    {
      "@type": "cr:FileObject",
      "@id": "repo",
      "name": "repo",
      "description": "The Hugging Face git repository.",
      "contentUrl": "https:\/\/huggingface.co\/datasets\/TomTBT\/pmc_open_access_section\/tree\/refs%2Fconvert%2Fparquet",
      "encodingFormat": "git+https",
      "sha256": "https:\/\/github.com\/mlcommons\/croissant\/issues\/80"
    },
    {
      "@type": "cr:FileSet",
      "@id": "parquet-files-for-config-commercial",
      "name": "parquet-files-for-config-commercial",
      "description": "The underlying Parquet files as converted by Hugging Face (see: https:\/\/huggingface.co\/docs\/dataset-viewer\/parquet).",
      "containedIn": {
        "@id": "repo"
      },
      "encodingFormat": "application\/x-parquet",
      "includes": "commercial\/*\/*.parquet"
    },
    {
      "@type": "cr:FileSet",
      "@id": "parquet-files-for-config-non_commercial",
      "name": "parquet-files-for-config-non_commercial",
      "description": "The underlying Parquet files as converted by Hugging Face (see: https:\/\/huggingface.co\/docs\/dataset-viewer\/parquet).",
      "containedIn": {
        "@id": "repo"
      },
      "encodingFormat": "application\/x-parquet",
      "includes": "non_commercial\/*\/*.parquet"
    },
    {
      "@type": "cr:FileSet",
      "@id": "parquet-files-for-config-other",
      "name": "parquet-files-for-config-other",
      "description": "The underlying Parquet files as converted by Hugging Face (see: https:\/\/huggingface.co\/docs\/dataset-viewer\/parquet).",
      "containedIn": {
        "@id": "repo"
      },
      "encodingFormat": "application\/x-parquet",
      "includes": "other\/*\/*.parquet"
    }
  ],
  "recordSet": [
    {
      "@type": "cr:RecordSet",
      "dataType": "cr:Split",
      "key": {
        "@id": "commercial_splits\/split_name"
      },
      "@id": "commercial_splits",
      "name": "commercial_splits",
      "description": "Splits for the commercial config.",
      "field": [
        {
          "@type": "cr:Field",
          "@id": "commercial_splits\/split_name",
          "name": "split_name",
          "description": "The name of the split.",
          "dataType": "sc:Text"
        }
      ],
      "data": [
        {
          "commercial_splits\/split_name": "train"
        }
      ]
    },
    {
      "@type": "cr:RecordSet",
      "@id": "commercial",
      "name": "commercial",
      "description": "TomTBT\/pmc_open_access_section - 'commercial' subset (first 5GB)",
      "field": [
        {
          "@type": "cr:Field",
          "@id": "commercial\/split",
          "name": "commercial\/split",
          "description": "Split to which the example belongs to.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-commercial"
            },
            "extract": {
              "fileProperty": "fullpath"
            },
            "transform": {
              "regex": "commercial\/(?:partial-)?(train)\/.+parquet$"
            }
          },
          "references": {
            "field": {
              "@id": "commercial_splits\/split_name"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "commercial\/accession_id",
          "name": "commercial\/accession_id",
          "description": "Column 'accession_id' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-commercial"
            },
            "extract": {
              "column": "accession_id"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "commercial\/pmid",
          "name": "commercial\/pmid",
          "description": "Column 'pmid' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-commercial"
            },
            "extract": {
              "column": "pmid"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "commercial\/introduction",
          "name": "commercial\/introduction",
          "description": "Column 'introduction' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-commercial"
            },
            "extract": {
              "column": "introduction"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "commercial\/methods",
          "name": "commercial\/methods",
          "description": "Column 'methods' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-commercial"
            },
            "extract": {
              "column": "methods"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "commercial\/results",
          "name": "commercial\/results",
          "description": "Column 'results' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-commercial"
            },
            "extract": {
              "column": "results"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "commercial\/discussion",
          "name": "commercial\/discussion",
          "description": "Column 'discussion' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-commercial"
            },
            "extract": {
              "column": "discussion"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "commercial\/conclusion",
          "name": "commercial\/conclusion",
          "description": "Column 'conclusion' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-commercial"
            },
            "extract": {
              "column": "conclusion"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "commercial\/front",
          "name": "commercial\/front",
          "description": "Column 'front' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-commercial"
            },
            "extract": {
              "column": "front"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "commercial\/body",
          "name": "commercial\/body",
          "description": "Column 'body' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-commercial"
            },
            "extract": {
              "column": "body"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "commercial\/back",
          "name": "commercial\/back",
          "description": "Column 'back' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-commercial"
            },
            "extract": {
              "column": "back"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "commercial\/license",
          "name": "commercial\/license",
          "description": "Column 'license' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-commercial"
            },
            "extract": {
              "column": "license"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "commercial\/retracted",
          "name": "commercial\/retracted",
          "description": "Column 'retracted' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-commercial"
            },
            "extract": {
              "column": "retracted"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "commercial\/last_updated",
          "name": "commercial\/last_updated",
          "description": "Column 'last_updated' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-commercial"
            },
            "extract": {
              "column": "last_updated"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "commercial\/citation",
          "name": "commercial\/citation",
          "description": "Column 'citation' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-commercial"
            },
            "extract": {
              "column": "citation"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "commercial\/package_file",
          "name": "commercial\/package_file",
          "description": "Column 'package_file' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-commercial"
            },
            "extract": {
              "column": "package_file"
            }
          }
        }
      ]
    },
    {
      "@type": "cr:RecordSet",
      "dataType": "cr:Split",
      "key": {
        "@id": "non_commercial_splits\/split_name"
      },
      "@id": "non_commercial_splits",
      "name": "non_commercial_splits",
      "description": "Splits for the non_commercial config.",
      "field": [
        {
          "@type": "cr:Field",
          "@id": "non_commercial_splits\/split_name",
          "name": "split_name",
          "description": "The name of the split.",
          "dataType": "sc:Text"
        }
      ],
      "data": [
        {
          "non_commercial_splits\/split_name": "train"
        }
      ]
    },
    {
      "@type": "cr:RecordSet",
      "@id": "non_commercial",
      "name": "non_commercial",
      "description": "TomTBT\/pmc_open_access_section - 'non_commercial' subset (first 5GB)",
      "field": [
        {
          "@type": "cr:Field",
          "@id": "non_commercial\/split",
          "name": "non_commercial\/split",
          "description": "Split to which the example belongs to.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-non_commercial"
            },
            "extract": {
              "fileProperty": "fullpath"
            },
            "transform": {
              "regex": "non_commercial\/(?:partial-)?(train)\/.+parquet$"
            }
          },
          "references": {
            "field": {
              "@id": "non_commercial_splits\/split_name"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "non_commercial\/accession_id",
          "name": "non_commercial\/accession_id",
          "description": "Column 'accession_id' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-non_commercial"
            },
            "extract": {
              "column": "accession_id"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "non_commercial\/pmid",
          "name": "non_commercial\/pmid",
          "description": "Column 'pmid' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-non_commercial"
            },
            "extract": {
              "column": "pmid"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "non_commercial\/introduction",
          "name": "non_commercial\/introduction",
          "description": "Column 'introduction' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-non_commercial"
            },
            "extract": {
              "column": "introduction"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "non_commercial\/methods",
          "name": "non_commercial\/methods",
          "description": "Column 'methods' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-non_commercial"
            },
            "extract": {
              "column": "methods"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "non_commercial\/results",
          "name": "non_commercial\/results",
          "description": "Column 'results' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-non_commercial"
            },
            "extract": {
              "column": "results"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "non_commercial\/discussion",
          "name": "non_commercial\/discussion",
          "description": "Column 'discussion' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-non_commercial"
            },
            "extract": {
              "column": "discussion"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "non_commercial\/conclusion",
          "name": "non_commercial\/conclusion",
          "description": "Column 'conclusion' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-non_commercial"
            },
            "extract": {
              "column": "conclusion"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "non_commercial\/front",
          "name": "non_commercial\/front",
          "description": "Column 'front' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-non_commercial"
            },
            "extract": {
              "column": "front"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "non_commercial\/body",
          "name": "non_commercial\/body",
          "description": "Column 'body' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-non_commercial"
            },
            "extract": {
              "column": "body"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "non_commercial\/back",
          "name": "non_commercial\/back",
          "description": "Column 'back' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-non_commercial"
            },
            "extract": {
              "column": "back"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "non_commercial\/license",
          "name": "non_commercial\/license",
          "description": "Column 'license' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-non_commercial"
            },
            "extract": {
              "column": "license"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "non_commercial\/retracted",
          "name": "non_commercial\/retracted",
          "description": "Column 'retracted' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-non_commercial"
            },
            "extract": {
              "column": "retracted"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "non_commercial\/last_updated",
          "name": "non_commercial\/last_updated",
          "description": "Column 'last_updated' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-non_commercial"
            },
            "extract": {
              "column": "last_updated"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "non_commercial\/citation",
          "name": "non_commercial\/citation",
          "description": "Column 'citation' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-non_commercial"
            },
            "extract": {
              "column": "citation"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "non_commercial\/package_file",
          "name": "non_commercial\/package_file",
          "description": "Column 'package_file' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-non_commercial"
            },
            "extract": {
              "column": "package_file"
            }
          }
        }
      ]
    },
    {
      "@type": "cr:RecordSet",
      "dataType": "cr:Split",
      "key": {
        "@id": "other_splits\/split_name"
      },
      "@id": "other_splits",
      "name": "other_splits",
      "description": "Splits for the other config.",
      "field": [
        {
          "@type": "cr:Field",
          "@id": "other_splits\/split_name",
          "name": "split_name",
          "description": "The name of the split.",
          "dataType": "sc:Text"
        }
      ],
      "data": [
        {
          "other_splits\/split_name": "train"
        }
      ]
    },
    {
      "@type": "cr:RecordSet",
      "@id": "other",
      "name": "other",
      "description": "TomTBT\/pmc_open_access_section - 'other' subset (first 5GB)",
      "field": [
        {
          "@type": "cr:Field",
          "@id": "other\/split",
          "name": "other\/split",
          "description": "Split to which the example belongs to.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-other"
            },
            "extract": {
              "fileProperty": "fullpath"
            },
            "transform": {
              "regex": "other\/(?:partial-)?(train)\/.+parquet$"
            }
          },
          "references": {
            "field": {
              "@id": "other_splits\/split_name"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "other\/accession_id",
          "name": "other\/accession_id",
          "description": "Column 'accession_id' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-other"
            },
            "extract": {
              "column": "accession_id"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "other\/pmid",
          "name": "other\/pmid",
          "description": "Column 'pmid' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-other"
            },
            "extract": {
              "column": "pmid"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "other\/introduction",
          "name": "other\/introduction",
          "description": "Column 'introduction' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-other"
            },
            "extract": {
              "column": "introduction"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "other\/methods",
          "name": "other\/methods",
          "description": "Column 'methods' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-other"
            },
            "extract": {
              "column": "methods"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "other\/results",
          "name": "other\/results",
          "description": "Column 'results' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-other"
            },
            "extract": {
              "column": "results"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "other\/discussion",
          "name": "other\/discussion",
          "description": "Column 'discussion' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-other"
            },
            "extract": {
              "column": "discussion"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "other\/conclusion",
          "name": "other\/conclusion",
          "description": "Column 'conclusion' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-other"
            },
            "extract": {
              "column": "conclusion"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "other\/front",
          "name": "other\/front",
          "description": "Column 'front' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-other"
            },
            "extract": {
              "column": "front"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "other\/body",
          "name": "other\/body",
          "description": "Column 'body' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-other"
            },
            "extract": {
              "column": "body"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "other\/back",
          "name": "other\/back",
          "description": "Column 'back' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-other"
            },
            "extract": {
              "column": "back"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "other\/license",
          "name": "other\/license",
          "description": "Column 'license' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-other"
            },
            "extract": {
              "column": "license"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "other\/retracted",
          "name": "other\/retracted",
          "description": "Column 'retracted' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-other"
            },
            "extract": {
              "column": "retracted"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "other\/last_updated",
          "name": "other\/last_updated",
          "description": "Column 'last_updated' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-other"
            },
            "extract": {
              "column": "last_updated"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "other\/citation",
          "name": "other\/citation",
          "description": "Column 'citation' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-other"
            },
            "extract": {
              "column": "citation"
            }
          }
        },
        {
          "@type": "cr:Field",
          "@id": "other\/package_file",
          "name": "other\/package_file",
          "description": "Column 'package_file' from the Hugging Face parquet file.",
          "dataType": "sc:Text",
          "source": {
            "fileSet": {
              "@id": "parquet-files-for-config-other"
            },
            "extract": {
              "column": "package_file"
            }
          }
        }
      ]
    }
  ],
  "conformsTo": "http:\/\/mlcommons.org\/croissant\/1.0",
  "name": "pmc_open_access_section",
  "description": "TomTBT\/pmc_open_access_section dataset hosted on Hugging Face and contributed by the HF Datasets community",
  "alternateName": [
    "TomTBT\/pmc_open_access_section"
  ],
  "creator": {
    "@type": "Person",
    "name": "Tom Boissonnet",
    "url": "https:\/\/huggingface.co\/TomTBT"
  },
  "keywords": [
    "English",
    "cc-by-4.0",
    "1M - 10M",
    "arrow",
    "Text",
    "Datasets",
    "Croissant",
    "🇺🇸 Region: US"
  ],
  "license": "https:\/\/choosealicense.com\/licenses\/cc-by-4.0\/",
  "url": "https:\/\/huggingface.co\/datasets\/TomTBT\/pmc_open_access_section"
}</script> 

		<title>TomTBT/pmc_open_access_section · Datasets at Hugging Face</title>

		<script
			defer
			data-domain="huggingface.co"
			event-loggedIn="false"
			src="/js/script.pageview-props.js"
		></script>
		<script>
			window.plausible =
				window.plausible ||
				function () {
					(window.plausible.q = window.plausible.q || []).push(arguments);
				};
		</script>
		<script>
			window.hubConfig = {"features":{"signupDisabled":false},"sshGitUrl":"git@hf.co","moonHttpUrl":"https:\/\/huggingface.co","captchaApiKey":"bd5f2066-93dc-4bdd-a64b-a24646ca3859","captchaDisabledOnSignup":true,"datasetViewerPublicUrl":"https:\/\/datasets-server.huggingface.co","stripePublicKey":"pk_live_x2tdjFXBCvXo2FFmMybezpeM00J6gPCAAc","environment":"production","userAgent":"HuggingFace (production)","spacesIframeDomain":"hf.space","spacesApiUrl":"https:\/\/api.hf.space","docSearchKey":"ece5e02e57300e17d152c08056145326e90c4bff3dd07d7d1ae40cf1c8d39cb6","logoDev":{"apiUrl":"https:\/\/img.logo.dev\/","apiKey":"pk_UHS2HZOeRnaSOdDp7jbd5w"}};
		</script>
		<script type="text/javascript" src="https://de5282c3ca0c.edge.sdk.awswaf.com/de5282c3ca0c/526cf06acb0d/challenge.js" defer></script>
	</head>
	<body class="flex flex-col min-h-dvh bg-white dark:bg-gray-950 text-black DatasetPage">
		<div class="flex min-h-dvh flex-col"><div class="SVELTE_HYDRATER contents" data-target="SystemThemeMonitor" data-props="{&quot;isLoggedIn&quot;:false}"></div>

	<div class="SVELTE_HYDRATER contents" data-target="MainHeader" data-props="{&quot;classNames&quot;:&quot;&quot;,&quot;isWide&quot;:false,&quot;isZh&quot;:false,&quot;isPro&quot;:false}"><header class="border-b border-gray-100 "><div class="w-full px-4 container flex h-16 items-center"><div class="flex flex-1 items-center"><a class="mr-5 flex flex-none items-center lg:mr-6" href="/"><img alt="Hugging Face's logo" class="w-7 md:mr-2" src="/front/assets/huggingface_logo-noborder.svg">
				<span class="hidden whitespace-nowrap text-lg font-bold md:block">Hugging Face</span></a>
			<div class="relative flex-1 lg:max-w-sm mr-2 sm:mr-4 md:mr-3 xl:mr-6"><input autocomplete="off" class="w-full dark:bg-gray-950 pl-8 form-input-alt h-9 pr-3 focus:shadow-xl " name="" placeholder="Search models, datasets, users..."   spellcheck="false" type="text" value="">
	<svg class="absolute left-2.5 text-gray-400 top-1/2 transform -translate-y-1/2" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M30 28.59L22.45 21A11 11 0 1 0 21 22.45L28.59 30zM5 14a9 9 0 1 1 9 9a9 9 0 0 1-9-9z" fill="currentColor"></path></svg>
	</div>
			<div class="flex flex-none items-center justify-center p-0.5 place-self-stretch lg:hidden"><button class="relative z-40 flex h-6 w-8 items-center justify-center" type="button"><svg width="1em" height="1em" viewBox="0 0 10 10" class="text-xl" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" preserveAspectRatio="xMidYMid meet" fill="currentColor"><path fill-rule="evenodd" clip-rule="evenodd" d="M1.65039 2.9999C1.65039 2.8066 1.80709 2.6499 2.00039 2.6499H8.00039C8.19369 2.6499 8.35039 2.8066 8.35039 2.9999C8.35039 3.1932 8.19369 3.3499 8.00039 3.3499H2.00039C1.80709 3.3499 1.65039 3.1932 1.65039 2.9999ZM1.65039 4.9999C1.65039 4.8066 1.80709 4.6499 2.00039 4.6499H8.00039C8.19369 4.6499 8.35039 4.8066 8.35039 4.9999C8.35039 5.1932 8.19369 5.3499 8.00039 5.3499H2.00039C1.80709 5.3499 1.65039 5.1932 1.65039 4.9999ZM2.00039 6.6499C1.80709 6.6499 1.65039 6.8066 1.65039 6.9999C1.65039 7.1932 1.80709 7.3499 2.00039 7.3499H8.00039C8.19369 7.3499 8.35039 7.1932 8.35039 6.9999C8.35039 6.8066 8.19369 6.6499 8.00039 6.6499H2.00039Z"></path></svg>
		</button>

	</div></div>
		<nav aria-label="Main" class="ml-auto hidden lg:block"><ul class="flex items-center space-x-1.5 2xl:space-x-2"><li class="hover:text-indigo-700"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/models"><svg class="mr-1.5 text-gray-400 group-hover:text-indigo-500" style="" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 24 24"><path class="uim-quaternary" d="M20.23 7.24L12 12L3.77 7.24a1.98 1.98 0 0 1 .7-.71L11 2.76c.62-.35 1.38-.35 2 0l6.53 3.77c.29.173.531.418.7.71z" opacity=".25" fill="currentColor"></path><path class="uim-tertiary" d="M12 12v9.5a2.09 2.09 0 0 1-.91-.21L4.5 17.48a2.003 2.003 0 0 1-1-1.73v-7.5a2.06 2.06 0 0 1 .27-1.01L12 12z" opacity=".5" fill="currentColor"></path><path class="uim-primary" d="M20.5 8.25v7.5a2.003 2.003 0 0 1-1 1.73l-6.62 3.82c-.275.13-.576.198-.88.2V12l8.23-4.76c.175.308.268.656.27 1.01z" fill="currentColor"></path></svg>
					Models</a>
			</li><li class="hover:text-red-700"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/datasets"><svg class="mr-1.5 text-gray-400 group-hover:text-red-500" style="" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 25 25"><ellipse cx="12.5" cy="5" fill="currentColor" fill-opacity="0.25" rx="7.5" ry="2"></ellipse><path d="M12.5 15C16.6421 15 20 14.1046 20 13V20C20 21.1046 16.6421 22 12.5 22C8.35786 22 5 21.1046 5 20V13C5 14.1046 8.35786 15 12.5 15Z" fill="currentColor" opacity="0.5"></path><path d="M12.5 7C16.6421 7 20 6.10457 20 5V11.5C20 12.6046 16.6421 13.5 12.5 13.5C8.35786 13.5 5 12.6046 5 11.5V5C5 6.10457 8.35786 7 12.5 7Z" fill="currentColor" opacity="0.5"></path><path d="M5.23628 12C5.08204 12.1598 5 12.8273 5 13C5 14.1046 8.35786 15 12.5 15C16.6421 15 20 14.1046 20 13C20 12.8273 19.918 12.1598 19.7637 12C18.9311 12.8626 15.9947 13.5 12.5 13.5C9.0053 13.5 6.06886 12.8626 5.23628 12Z" fill="currentColor"></path></svg>
					Datasets</a>
			</li><li class="hover:text-blue-700"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/spaces"><svg class="mr-1.5 text-gray-400 group-hover:text-blue-500" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" viewBox="0 0 25 25"><path opacity=".5" d="M6.016 14.674v4.31h4.31v-4.31h-4.31ZM14.674 14.674v4.31h4.31v-4.31h-4.31ZM6.016 6.016v4.31h4.31v-4.31h-4.31Z" fill="currentColor"></path><path opacity=".75" fill-rule="evenodd" clip-rule="evenodd" d="M3 4.914C3 3.857 3.857 3 4.914 3h6.514c.884 0 1.628.6 1.848 1.414a5.171 5.171 0 0 1 7.31 7.31c.815.22 1.414.964 1.414 1.848v6.514A1.914 1.914 0 0 1 20.086 22H4.914A1.914 1.914 0 0 1 3 20.086V4.914Zm3.016 1.102v4.31h4.31v-4.31h-4.31Zm0 12.968v-4.31h4.31v4.31h-4.31Zm8.658 0v-4.31h4.31v4.31h-4.31Zm0-10.813a2.155 2.155 0 1 1 4.31 0 2.155 2.155 0 0 1-4.31 0Z" fill="currentColor"></path><path opacity=".25" d="M16.829 6.016a2.155 2.155 0 1 0 0 4.31 2.155 2.155 0 0 0 0-4.31Z" fill="currentColor"></path></svg>
					Spaces</a>
			</li><li class="hover:text-yellow-700 max-xl:hidden"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/posts"><svg class="mr-1.5 text-gray-400 group-hover:text-yellow-500 text-yellow-500!" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" viewBox="0 0 12 12" preserveAspectRatio="xMidYMid meet"><path fill="currentColor" fill-rule="evenodd" d="M3.73 2.4A4.25 4.25 0 1 1 6 10.26H2.17l-.13-.02a.43.43 0 0 1-.3-.43l.01-.06a.43.43 0 0 1 .12-.22l.84-.84A4.26 4.26 0 0 1 3.73 2.4Z" clip-rule="evenodd"></path></svg>
					Posts</a>
			</li><li class="hover:text-yellow-700"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/docs"><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" class="mr-1.5 text-gray-400 group-hover:text-yellow-500" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 16 16"><path d="m2.28 3.7-.3.16a.67.67 0 0 0-.34.58v8.73l.01.04.02.07.01.04.03.06.02.04.02.03.04.06.05.05.04.04.06.04.06.04.08.04.08.02h.05l.07.02h.11l.04-.01.07-.02.03-.01.07-.03.22-.12a5.33 5.33 0 0 1 5.15.1.67.67 0 0 0 .66 0 5.33 5.33 0 0 1 5.33 0 .67.67 0 0 0 1-.58V4.36a.67.67 0 0 0-.34-.5l-.3-.17v7.78a.63.63 0 0 1-.87.59 4.9 4.9 0 0 0-4.35.35l-.65.39a.29.29 0 0 1-.15.04.29.29 0 0 1-.16-.04l-.65-.4a4.9 4.9 0 0 0-4.34-.34.63.63 0 0 1-.87-.59V3.7Z" fill="currentColor" class="dark:opacity-40"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M8 3.1a5.99 5.99 0 0 0-5.3-.43.66.66 0 0 0-.42.62v8.18c0 .45.46.76.87.59a4.9 4.9 0 0 1 4.34.35l.65.39c.05.03.1.04.16.04.05 0 .1-.01.15-.04l.65-.4a4.9 4.9 0 0 1 4.35-.34.63.63 0 0 0 .86-.59V3.3a.67.67 0 0 0-.41-.62 5.99 5.99 0 0 0-5.3.43l-.3.17L8 3.1Zm.73 1.87a.43.43 0 1 0-.86 0v5.48a.43.43 0 0 0 .86 0V4.97Z" fill="currentColor" class="opacity-40 dark:opacity-100"></path><path d="M8.73 4.97a.43.43 0 1 0-.86 0v5.48a.43.43 0 1 0 .86 0V4.96Z" fill="currentColor" class="dark:opacity-40"></path></svg>
					Docs</a>
			</li><li class="hover:text-black dark:hover:text-white"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/enterprise"><svg class="mr-1.5 text-gray-400 group-hover:text-black dark:group-hover:text-white" xmlns="http://www.w3.org/2000/svg" fill="none" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 33 27"><path fill="currentColor" fill-rule="evenodd" d="M13.5.7a8.7 8.7 0 0 0-7.7 5.7L1 20.6c-1 3.1.9 5.7 4.1 5.7h15c3.3 0 6.8-2.6 7.8-5.7l4.6-14.2c1-3.1-.8-5.7-4-5.7h-15Zm1.1 5.7L9.8 20.3h9.8l1-3.1h-5.8l.8-2.5h4.8l1.1-3h-4.8l.8-2.3H23l1-3h-9.5Z" clip-rule="evenodd"></path></svg>
					Enterprise</a>
			</li>

		<li><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/pricing">Pricing
			</a></li>

		<li><div class="relative group">
	<button class="px-2 py-0.5 hover:text-gray-500 dark:hover:text-gray-600 flex items-center " type="button">
		<svg class=" text-gray-500 w-5 group-hover:text-gray-400 dark:text-gray-300 dark:group-hover:text-gray-100" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" viewBox="0 0 32 18" preserveAspectRatio="xMidYMid meet"><path fill-rule="evenodd" clip-rule="evenodd" d="M14.4504 3.30221C14.4504 2.836 14.8284 2.45807 15.2946 2.45807H28.4933C28.9595 2.45807 29.3374 2.836 29.3374 3.30221C29.3374 3.76842 28.9595 4.14635 28.4933 4.14635H15.2946C14.8284 4.14635 14.4504 3.76842 14.4504 3.30221Z" fill="currentColor"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M14.4504 9.00002C14.4504 8.53382 14.8284 8.15588 15.2946 8.15588H28.4933C28.9595 8.15588 29.3374 8.53382 29.3374 9.00002C29.3374 9.46623 28.9595 9.84417 28.4933 9.84417H15.2946C14.8284 9.84417 14.4504 9.46623 14.4504 9.00002Z" fill="currentColor"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M14.4504 14.6978C14.4504 14.2316 14.8284 13.8537 15.2946 13.8537H28.4933C28.9595 13.8537 29.3374 14.2316 29.3374 14.6978C29.3374 15.164 28.9595 15.542 28.4933 15.542H15.2946C14.8284 15.542 14.4504 15.164 14.4504 14.6978Z" fill="currentColor"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M1.94549 6.87377C2.27514 6.54411 2.80962 6.54411 3.13928 6.87377L6.23458 9.96907L9.32988 6.87377C9.65954 6.54411 10.194 6.54411 10.5237 6.87377C10.8533 7.20343 10.8533 7.73791 10.5237 8.06756L6.23458 12.3567L1.94549 8.06756C1.61583 7.73791 1.61583 7.20343 1.94549 6.87377Z" fill="currentColor"></path></svg>
			
		</button>
	
	
	</div></li>
		<li><hr class="h-5 w-0.5 border-none bg-gray-100 dark:bg-gray-800"></li>
		<li><a class="block cursor-pointer whitespace-nowrap px-2 py-0.5 hover:text-gray-500 dark:text-gray-300 dark:hover:text-gray-100" href="/login">Log In
				</a></li>
			<li><a class="whitespace-nowrap rounded-full border border-transparent bg-gray-900 px-3 py-1 leading-none text-white hover:border-black hover:bg-white hover:text-black" href="/join">Sign Up
					</a></li></ul></nav></div></header></div>
	
	
	
	<div class="SVELTE_HYDRATER contents" data-target="SSOBanner" data-props="{}"></div>
	
	

	<main class="flex flex-1 flex-col">
	<div class="SVELTE_HYDRATER contents" data-target="DatasetHeader" data-props="{&quot;activeTab&quot;:&quot;datasetCard&quot;,&quot;author&quot;:{&quot;_id&quot;:&quot;617900b59e61d7ff71b52b94&quot;,&quot;avatarUrl&quot;:&quot;/avatars/3435205ad008b625bb742afcdcd92764.svg&quot;,&quot;fullname&quot;:&quot;Tom Boissonnet&quot;,&quot;name&quot;:&quot;TomTBT&quot;,&quot;type&quot;:&quot;user&quot;,&quot;isPro&quot;:false,&quot;isHf&quot;:false,&quot;isHfAdmin&quot;:false,&quot;isMod&quot;:false,&quot;followerCount&quot;:1},&quot;canReadRepoSettings&quot;:false,&quot;dataset&quot;:{&quot;author&quot;:&quot;TomTBT&quot;,&quot;cardData&quot;:{&quot;configs&quot;:[{&quot;config_name&quot;:&quot;commercial&quot;,&quot;data_files&quot;:[{&quot;split&quot;:&quot;train&quot;,&quot;path&quot;:&quot;data/commercial/*.arrow&quot;}]},{&quot;config_name&quot;:&quot;non_commercial&quot;,&quot;data_files&quot;:[{&quot;split&quot;:&quot;train&quot;,&quot;path&quot;:&quot;data/non_commercial/*.arrow&quot;}]},{&quot;config_name&quot;:&quot;other&quot;,&quot;data_files&quot;:[{&quot;split&quot;:&quot;train&quot;,&quot;path&quot;:&quot;data/non_commercial/*.arrow&quot;}]}],&quot;license&quot;:&quot;cc-by-4.0&quot;,&quot;language&quot;:[&quot;en&quot;],&quot;size_categories&quot;:[&quot;1M<n<10M&quot;]},&quot;cardExists&quot;:true,&quot;createdAt&quot;:&quot;2022-11-26T00:20:40.000Z&quot;,&quot;downloads&quot;:566,&quot;downloadsAllTime&quot;:2107,&quot;id&quot;:&quot;TomTBT/pmc_open_access_section&quot;,&quot;isLikedByUser&quot;:false,&quot;lastModified&quot;:&quot;2024-09-15T10:13:36.000Z&quot;,&quot;likes&quot;:2,&quot;datasetsServerInfo&quot;:{&quot;viewer&quot;:&quot;viewer-partial&quot;,&quot;numRows&quot;:7219374,&quot;libraries&quot;:[&quot;datasets&quot;,&quot;mlcroissant&quot;],&quot;formats&quot;:[&quot;arrow&quot;],&quot;modalities&quot;:[&quot;text&quot;]},&quot;discussionsDisabled&quot;:false,&quot;repoType&quot;:&quot;dataset&quot;,&quot;private&quot;:false,&quot;gated&quot;:false,&quot;tags&quot;:[&quot;language:en&quot;,&quot;license:cc-by-4.0&quot;,&quot;size_categories:1M<n<10M&quot;,&quot;format:arrow&quot;,&quot;modality:text&quot;,&quot;library:datasets&quot;,&quot;library:mlcroissant&quot;,&quot;region:us&quot;],&quot;tag_objs&quot;:[{&quot;id&quot;:&quot;language:en&quot;,&quot;label&quot;:&quot;English&quot;,&quot;type&quot;:&quot;language&quot;},{&quot;id&quot;:&quot;license:cc-by-4.0&quot;,&quot;label&quot;:&quot;cc-by-4.0&quot;,&quot;type&quot;:&quot;license&quot;},{&quot;id&quot;:&quot;size_categories:1M<n<10M&quot;,&quot;label&quot;:&quot;1M - 10M&quot;,&quot;type&quot;:&quot;size_categories&quot;},{&quot;id&quot;:&quot;format:arrow&quot;,&quot;label&quot;:&quot;arrow&quot;,&quot;type&quot;:&quot;format&quot;},{&quot;id&quot;:&quot;modality:text&quot;,&quot;label&quot;:&quot;Text&quot;,&quot;type&quot;:&quot;modality&quot;},{&quot;id&quot;:&quot;library:datasets&quot;,&quot;label&quot;:&quot;Datasets&quot;,&quot;type&quot;:&quot;library&quot;},{&quot;id&quot;:&quot;library:mlcroissant&quot;,&quot;label&quot;:&quot;Croissant&quot;,&quot;type&quot;:&quot;library&quot;},{&quot;type&quot;:&quot;region&quot;,&quot;label&quot;:&quot;🇺🇸 Region: US&quot;,&quot;id&quot;:&quot;region:us&quot;}],&quot;hasBlockedOids&quot;:false,&quot;region&quot;:&quot;us&quot;,&quot;xetEnabled&quot;:false},&quot;discussionsStats&quot;:{&quot;closed&quot;:0,&quot;open&quot;:1,&quot;total&quot;:1}}"><header class="bg-linear-to-t border-b border-gray-100 pt-6 sm:pt-9 from-gray-50-to-white via-white dark:via-gray-950"><div class="container relative "><h1 class="flex flex-wrap items-center max-md:leading-tight mb-3 text-lg max-sm:gap-y-1.5 md:text-xl"><a href="/datasets" class="group flex items-center"><svg class="sm:mr-1.5 -mr-1 text-gray-400" style="" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 25 25"><ellipse cx="12.5" cy="5" fill="currentColor" fill-opacity="0.25" rx="7.5" ry="2"></ellipse><path d="M12.5 15C16.6421 15 20 14.1046 20 13V20C20 21.1046 16.6421 22 12.5 22C8.35786 22 5 21.1046 5 20V13C5 14.1046 8.35786 15 12.5 15Z" fill="currentColor" opacity="0.5"></path><path d="M12.5 7C16.6421 7 20 6.10457 20 5V11.5C20 12.6046 16.6421 13.5 12.5 13.5C8.35786 13.5 5 12.6046 5 11.5V5C5 6.10457 8.35786 7 12.5 7Z" fill="currentColor" opacity="0.5"></path><path d="M5.23628 12C5.08204 12.1598 5 12.8273 5 13C5 14.1046 8.35786 15 12.5 15C16.6421 15 20 14.1046 20 13C20 12.8273 19.918 12.1598 19.7637 12C18.9311 12.8626 15.9947 13.5 12.5 13.5C9.0053 13.5 6.06886 12.8626 5.23628 12Z" fill="currentColor"></path></svg>
					<span class="mr-2.5 font-semibold text-gray-400 group-hover:text-gray-500 max-sm:hidden">Datasets:</span></a>
				<hr class="mx-1.5 h-2 translate-y-px rounded-sm border-r dark:border-gray-600 sm:hidden">
			<div class="group flex flex-none items-center"><div class="relative mr-1 flex items-center">

			

<span class="inline-block "><span class="contents"><a href="/TomTBT" class="text-gray-400 hover:text-blue-600"><img alt="" class="w-3.5 h-3.5 rounded-full  flex-none" src="/avatars/3435205ad008b625bb742afcdcd92764.svg" crossorigin="anonymous"></a></span>
	</span></div>
		

<span class="inline-block "><span class="contents"><a href="/TomTBT" class="text-gray-400 hover:text-blue-600">TomTBT</a></span>
	</span>
		<div class="mx-0.5 text-gray-300">/</div></div>

<div class="max-w-full "><a class="break-words font-mono font-semibold hover:text-blue-600 " href="/datasets/TomTBT/pmc_open_access_section">pmc_open_access_section</a>
	<button class="relative text-sm mr-4 focus:outline-hidden inline-flex cursor-pointer items-center text-sm  mx-0.5   text-gray-600 " title="Copy dataset name to clipboard" type="button"><svg class="" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" fill="currentColor" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M28,10V28H10V10H28m0-2H10a2,2,0,0,0-2,2V28a2,2,0,0,0,2,2H28a2,2,0,0,0,2-2V10a2,2,0,0,0-2-2Z" transform="translate(0)"></path><path d="M4,18H2V4A2,2,0,0,1,4,2H18V4H4Z" transform="translate(0)"></path><rect fill="none" width="32" height="32"></rect></svg>
	
	</button></div>
			<div class="inline-flex items-center overflow-hidden whitespace-nowrap rounded-md border bg-white text-sm leading-none text-gray-500  mr-2"><button class="relative flex items-center overflow-hidden from-red-50 to-transparent dark:from-red-900 px-1.5 py-1 hover:bg-linear-to-t focus:outline-hidden"  title="Like"><svg class="left-1.5 absolute" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32" fill="currentColor"><path d="M22.45,6a5.47,5.47,0,0,1,3.91,1.64,5.7,5.7,0,0,1,0,8L16,26.13,5.64,15.64a5.7,5.7,0,0,1,0-8,5.48,5.48,0,0,1,7.82,0L16,10.24l2.53-2.58A5.44,5.44,0,0,1,22.45,6m0-2a7.47,7.47,0,0,0-5.34,2.24L16,7.36,14.89,6.24a7.49,7.49,0,0,0-10.68,0,7.72,7.72,0,0,0,0,10.82L16,29,27.79,17.06a7.72,7.72,0,0,0,0-10.82A7.49,7.49,0,0,0,22.45,4Z"></path></svg>

		
		<span class="ml-4 pl-0.5 ">like</span></button>
	<button class="focus:outline-hidden flex items-center border-l px-1.5 py-1 text-gray-400 hover:bg-gray-50 focus:bg-gray-100 dark:hover:bg-gray-900 dark:focus:bg-gray-800" title="See users who liked this repository">2</button></div>




			
			
	</h1>
		<div class="mb-3 flex flex-wrap md:mb-4"><div class="mr-1 flex flex-wrap items-center"><span class="mb-1 mr-1 p-1 text-sm leading-tight text-gray-400 md:mb-1.5">Modalities:
	</span>
	<a class="mb-1 mr-1 md:mb-1.5 md:mr-1.5 rounded-lg" href="/datasets?modality=modality%3Atext"><div class="tag tag-white   ">
		<svg class="text-red-700 dark:text-red-600" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path fill-rule="evenodd" clip-rule="evenodd" d="M4.619 4.619C2.667 6.573 2.667 9.715 2.667 16s0 9.428 1.952 11.38C6.573 29.333 9.715 29.333 16 29.333s9.428 0 11.38-1.953c1.953-1.95 1.953-5.095 1.953-11.38s0-9.428-1.953-11.381C25.43 2.667 22.285 2.667 16 2.667s-9.428 0-11.381 1.952m8.65 3.714c-.573 0-1.109 0-1.546.066-.495.073-1.003.248-1.41.7-.392.436-.53.956-.59 1.452-.056.464-.056 1.04-.056 1.689V13a1 1 0 1 0 2 0v-.704c0-.724.001-1.176.041-1.505q.015-.15.061-.294a.2.2 0 0 1 .031-.061q0-.003.016-.01a.8.8 0 0 1 .203-.05c.272-.04.654-.043 1.314-.043H15v11.334h-2.333a1 1 0 1 0 0 2H20a1 1 0 0 0 0-2h-3V10.333h1.667c.66 0 1.042.003 1.314.043.123.019.18.04.203.05l.015.009a.2.2 0 0 1 .032.061c.018.05.042.14.061.295.04.329.041.781.041 1.506V13a1 1 0 1 0 2 0v-.76c0-.65 0-1.225-.056-1.69-.06-.495-.198-1.015-.59-1.453-.407-.45-.915-.625-1.41-.698-.437-.067-.973-.067-1.546-.066z" fill="currentColor"></path></svg>

	

	<span>Text</span>
	

	</div></a>

	</div><div class="mr-1 flex flex-wrap items-center"><span class="mb-1 mr-1 p-1 text-sm leading-tight text-gray-400 md:mb-1.5">Formats:
	</span>
	<a class="mb-1 mr-1 md:mb-1.5 md:mr-1.5 rounded-lg" href="/datasets?format=format%3Aarrow"><div class="tag tag-white   ">
		<svg class="" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" viewBox="0 0 12 12" fill="none"><path fill-rule="evenodd" clip-rule="evenodd" d="M1.46 2.04 5.42 6 1.46 9.96V8.38L3.84 6 1.46 3.62V2.04Zm2.78 0L8.19 6 4.24 9.96V8.38L6.6 6 4.24 3.62V2.04ZM10.97 6 7.01 2.04v1.58L9.39 6 7 8.38v1.58L10.97 6Z" fill="currentColor"></path></svg>

	

	<span>arrow</span>
	

	</div></a>

	</div><div class="mr-1 flex flex-wrap items-center"><span class="mb-1 mr-1 p-1 text-sm leading-tight text-gray-400 md:mb-1.5">Languages:
	</span>
	<a class="mb-1 mr-1 md:mb-1.5 md:mr-1.5 rounded-lg" href="/datasets?language=language%3Aen"><div class="tag tag-white   ">
		<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" class="text-green-600/80" preserveAspectRatio="xMidYMid meet" width="1em" height="1em" viewBox="0 0 10 10"><path fill-rule="evenodd" clip-rule="evenodd" d="M0.625 5C0.625 6.16032 1.08594 7.27312 1.90641 8.09359C2.72688 8.91406 3.83968 9.375 5 9.375C6.16032 9.375 7.27312 8.91406 8.09359 8.09359C8.91406 7.27312 9.375 6.16032 9.375 5C9.375 3.83968 8.91406 2.72688 8.09359 1.90641C7.27312 1.08594 6.16032 0.625 5 0.625C3.83968 0.625 2.72688 1.08594 1.90641 1.90641C1.08594 2.72688 0.625 3.83968 0.625 5ZM7.64365 7.48027C7.61734 7.50832 7.59054 7.53598 7.56326 7.56326C7.13828 7.98824 6.61864 8.2968 6.0539 8.46842C6.29802 8.11949 6.49498 7.64804 6.63475 7.09483C7.00845 7.18834 7.35014 7.3187 7.64365 7.48027ZM8.10076 6.87776C8.37677 6.42196 8.55005 5.90894 8.60556 5.37499H6.86808C6.85542 5.71597 6.82551 6.04557 6.77971 6.35841C7.25309 6.47355 7.68808 6.6414 8.062 6.85549C8.07497 6.86283 8.08789 6.87025 8.10076 6.87776ZM6.03795 6.22536C6.07708 5.95737 6.1044 5.67232 6.11705 5.37499H3.88295C3.89666 5.69742 3.92764 6.00542 3.9722 6.29287C4.37075 6.21726 4.79213 6.17749 5.224 6.17749C5.50054 6.17749 5.77294 6.19376 6.03795 6.22536ZM4.1261 7.02673C4.34894 7.84835 4.68681 8.375 5 8.375C5.32122 8.375 5.66839 7.82101 5.8908 6.963C5.67389 6.93928 5.45082 6.92699 5.224 6.92699C4.84316 6.92699 4.47332 6.96176 4.1261 7.02673ZM3.39783 7.21853C3.53498 7.71842 3.72038 8.14579 3.9461 8.46842C3.42141 8.30898 2.93566 8.03132 2.52857 7.65192C2.77253 7.48017 3.06711 7.33382 3.39783 7.21853ZM3.23916 6.48077C3.18263 6.13193 3.14625 5.76074 3.13192 5.37499H1.39444C1.4585 5.99112 1.67936 6.57938 2.03393 7.08403C2.3706 6.83531 2.78055 6.63162 3.23916 6.48077ZM1.39444 4.62499H3.13192C3.14615 4.24204 3.18211 3.87344 3.23794 3.52681C2.77814 3.37545 2.36731 3.17096 2.03024 2.92123C1.67783 3.42469 1.45828 4.011 1.39444 4.62499ZM2.5237 2.35262C2.76812 2.52552 3.06373 2.67281 3.39584 2.78875C3.53318 2.28573 3.71928 1.85578 3.9461 1.53158C3.41932 1.69166 2.93178 1.97089 2.5237 2.35262ZM3.97101 3.71489C3.92709 4.00012 3.89654 4.30547 3.88295 4.62499H6.11705C6.10453 4.33057 6.07761 4.04818 6.03909 3.78248C5.77372 3.81417 5.50093 3.83049 5.224 3.83049C4.79169 3.83049 4.3699 3.79065 3.97101 3.71489ZM5.8928 3.04476C5.67527 3.06863 5.45151 3.08099 5.224 3.08099C4.84241 3.08099 4.47186 3.04609 4.12405 2.98086C4.34686 2.1549 4.68584 1.625 5 1.625C5.32218 1.625 5.67048 2.18233 5.8928 3.04476ZM6.78083 3.6493C6.826 3.95984 6.85552 4.28682 6.86808 4.62499H8.60556C8.55029 4.09337 8.37827 3.58251 8.10436 3.1282C8.0903 3.1364 8.07618 3.14449 8.062 3.15249C7.68838 3.36641 7.25378 3.53417 6.78083 3.6493ZM7.64858 2.52499C7.35446 2.68754 7.0117 2.81868 6.63664 2.91268C6.49676 2.35623 6.29913 1.88209 6.0539 1.53158C6.61864 1.7032 7.13828 2.01176 7.56326 2.43674C7.59224 2.46572 7.62068 2.49514 7.64858 2.52499Z" fill="currentColor"></path></svg>

	

	<span>English</span>
	

	</div></a>

	</div><div class="mr-1 flex flex-wrap items-center"><span class="mb-1 mr-1 p-1 text-sm leading-tight text-gray-400 md:mb-1.5">Size:
	</span>
	<a class="mb-1 mr-1 md:mb-1.5 md:mr-1.5 rounded-lg" href="/datasets?size_categories=size_categories%3A1M%3Cn%3C10M"><div class="tag tag-white   ">

	

	<span>1M - 10M</span>
	

	</div></a>

	</div><div class="mr-1 flex flex-wrap items-center"><span class="mb-1 mr-1 p-1 text-sm leading-tight text-gray-400 md:mb-1.5">Libraries:
	</span>
	<a class="mb-1 mr-1 md:mb-1.5 md:mr-1.5 rounded-lg" href="/datasets?library=library%3Adatasets"><div class="tag tag-white   "><svg class="text-black inline-block text-sm" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" preserveAspectRatio="xMidYMid meet" width="1em" height="1em" viewBox="0 0 95 88"><path fill="#fff" d="M94.25 70.08a8.28 8.28 0 0 1-.43 6.46 10.57 10.57 0 0 1-3 3.6 25.18 25.18 0 0 1-5.7 3.2 65.74 65.74 0 0 1-7.56 2.65 46.67 46.67 0 0 1-11.42 1.68c-5.42.05-10.09-1.23-13.4-4.5a40.4 40.4 0 0 1-10.14.03c-3.34 3.25-7.99 4.52-13.39 4.47a46.82 46.82 0 0 1-11.43-1.68 66.37 66.37 0 0 1-7.55-2.65c-2.28-.98-4.17-2-5.68-3.2a10.5 10.5 0 0 1-3.02-3.6c-.99-2-1.18-4.3-.42-6.46a8.54 8.54 0 0 1-.33-5.63c.25-.95.66-1.83 1.18-2.61a8.67 8.67 0 0 1 2.1-8.47 8.23 8.23 0 0 1 2.82-2.07 41.75 41.75 0 1 1 81.3-.12 8.27 8.27 0 0 1 3.11 2.19 8.7 8.7 0 0 1 2.1 8.47c.52.78.93 1.66 1.18 2.61a8.61 8.61 0 0 1-.32 5.63Z"></path><path fill="#FFD21E" d="M47.21 76.5a34.75 34.75 0 1 0 0-69.5 34.75 34.75 0 0 0 0 69.5Z"></path><path fill="#FF9D0B" d="M81.96 41.75a34.75 34.75 0 1 0-69.5 0 34.75 34.75 0 0 0 69.5 0Zm-73.5 0a38.75 38.75 0 1 1 77.5 0 38.75 38.75 0 0 1-77.5 0Z"></path><path fill="#3A3B45" d="M58.5 32.3c1.28.44 1.78 3.06 3.07 2.38a5 5 0 1 0-6.76-2.07c.61 1.15 2.55-.72 3.7-.32ZM34.95 32.3c-1.28.44-1.79 3.06-3.07 2.38a5 5 0 1 1 6.76-2.07c-.61 1.15-2.56-.72-3.7-.32Z"></path><path fill="#FF323D" d="M46.96 56.29c9.83 0 13-8.76 13-13.26 0-2.34-1.57-1.6-4.09-.36-2.33 1.15-5.46 2.74-8.9 2.74-7.19 0-13-6.88-13-2.38s3.16 13.26 13 13.26Z"></path><path fill="#3A3B45" fill-rule="evenodd" d="M39.43 54a8.7 8.7 0 0 1 5.3-4.49c.4-.12.81.57 1.24 1.28.4.68.82 1.37 1.24 1.37.45 0 .9-.68 1.33-1.35.45-.7.89-1.38 1.32-1.25a8.61 8.61 0 0 1 5 4.17c3.73-2.94 5.1-7.74 5.1-10.7 0-2.34-1.57-1.6-4.09-.36l-.14.07c-2.31 1.15-5.39 2.67-8.77 2.67s-6.45-1.52-8.77-2.67c-2.6-1.29-4.23-2.1-4.23.29 0 3.05 1.46 8.06 5.47 10.97Z" clip-rule="evenodd"></path><path fill="#FF9D0B" d="M70.71 37a3.25 3.25 0 1 0 0-6.5 3.25 3.25 0 0 0 0 6.5ZM24.21 37a3.25 3.25 0 1 0 0-6.5 3.25 3.25 0 0 0 0 6.5ZM17.52 48c-1.62 0-3.06.66-4.07 1.87a5.97 5.97 0 0 0-1.33 3.76 7.1 7.1 0 0 0-1.94-.3c-1.55 0-2.95.59-3.94 1.66a5.8 5.8 0 0 0-.8 7 5.3 5.3 0 0 0-1.79 2.82c-.24.9-.48 2.8.8 4.74a5.22 5.22 0 0 0-.37 5.02c1.02 2.32 3.57 4.14 8.52 6.1 3.07 1.22 5.89 2 5.91 2.01a44.33 44.33 0 0 0 10.93 1.6c5.86 0 10.05-1.8 12.46-5.34 3.88-5.69 3.33-10.9-1.7-15.92-2.77-2.78-4.62-6.87-5-7.77-.78-2.66-2.84-5.62-6.25-5.62a5.7 5.7 0 0 0-4.6 2.46c-1-1.26-1.98-2.25-2.86-2.82A7.4 7.4 0 0 0 17.52 48Zm0 4c.51 0 1.14.22 1.82.65 2.14 1.36 6.25 8.43 7.76 11.18.5.92 1.37 1.31 2.14 1.31 1.55 0 2.75-1.53.15-3.48-3.92-2.93-2.55-7.72-.68-8.01.08-.02.17-.02.24-.02 1.7 0 2.45 2.93 2.45 2.93s2.2 5.52 5.98 9.3c3.77 3.77 3.97 6.8 1.22 10.83-1.88 2.75-5.47 3.58-9.16 3.58-3.81 0-7.73-.9-9.92-1.46-.11-.03-13.45-3.8-11.76-7 .28-.54.75-.76 1.34-.76 2.38 0 6.7 3.54 8.57 3.54.41 0 .7-.17.83-.6.79-2.85-12.06-4.05-10.98-8.17.2-.73.71-1.02 1.44-1.02 3.14 0 10.2 5.53 11.68 5.53.11 0 .2-.03.24-.1.74-1.2.33-2.04-4.9-5.2-5.21-3.16-8.88-5.06-6.8-7.33.24-.26.58-.38 1-.38 3.17 0 10.66 6.82 10.66 6.82s2.02 2.1 3.25 2.1c.28 0 .52-.1.68-.38.86-1.46-8.06-8.22-8.56-11.01-.34-1.9.24-2.85 1.31-2.85Z"></path><path fill="#FFD21E" d="M38.6 76.69c2.75-4.04 2.55-7.07-1.22-10.84-3.78-3.77-5.98-9.3-5.98-9.3s-.82-3.2-2.69-2.9c-1.87.3-3.24 5.08.68 8.01 3.91 2.93-.78 4.92-2.29 2.17-1.5-2.75-5.62-9.82-7.76-11.18-2.13-1.35-3.63-.6-3.13 2.2.5 2.79 9.43 9.55 8.56 11-.87 1.47-3.93-1.71-3.93-1.71s-9.57-8.71-11.66-6.44c-2.08 2.27 1.59 4.17 6.8 7.33 5.23 3.16 5.64 4 4.9 5.2-.75 1.2-12.28-8.53-13.36-4.4-1.08 4.11 11.77 5.3 10.98 8.15-.8 2.85-9.06-5.38-10.74-2.18-1.7 3.21 11.65 6.98 11.76 7.01 4.3 1.12 15.25 3.49 19.08-2.12Z"></path><path fill="#FF9D0B" d="M77.4 48c1.62 0 3.07.66 4.07 1.87a5.97 5.97 0 0 1 1.33 3.76 7.1 7.1 0 0 1 1.95-.3c1.55 0 2.95.59 3.94 1.66a5.8 5.8 0 0 1 .8 7 5.3 5.3 0 0 1 1.78 2.82c.24.9.48 2.8-.8 4.74a5.22 5.22 0 0 1 .37 5.02c-1.02 2.32-3.57 4.14-8.51 6.1-3.08 1.22-5.9 2-5.92 2.01a44.33 44.33 0 0 1-10.93 1.6c-5.86 0-10.05-1.8-12.46-5.34-3.88-5.69-3.33-10.9 1.7-15.92 2.78-2.78 4.63-6.87 5.01-7.77.78-2.66 2.83-5.62 6.24-5.62a5.7 5.7 0 0 1 4.6 2.46c1-1.26 1.98-2.25 2.87-2.82A7.4 7.4 0 0 1 77.4 48Zm0 4c-.51 0-1.13.22-1.82.65-2.13 1.36-6.25 8.43-7.76 11.18a2.43 2.43 0 0 1-2.14 1.31c-1.54 0-2.75-1.53-.14-3.48 3.91-2.93 2.54-7.72.67-8.01a1.54 1.54 0 0 0-.24-.02c-1.7 0-2.45 2.93-2.45 2.93s-2.2 5.52-5.97 9.3c-3.78 3.77-3.98 6.8-1.22 10.83 1.87 2.75 5.47 3.58 9.15 3.58 3.82 0 7.73-.9 9.93-1.46.1-.03 13.45-3.8 11.76-7-.29-.54-.75-.76-1.34-.76-2.38 0-6.71 3.54-8.57 3.54-.42 0-.71-.17-.83-.6-.8-2.85 12.05-4.05 10.97-8.17-.19-.73-.7-1.02-1.44-1.02-3.14 0-10.2 5.53-11.68 5.53-.1 0-.19-.03-.23-.1-.74-1.2-.34-2.04 4.88-5.2 5.23-3.16 8.9-5.06 6.8-7.33-.23-.26-.57-.38-.98-.38-3.18 0-10.67 6.82-10.67 6.82s-2.02 2.1-3.24 2.1a.74.74 0 0 1-.68-.38c-.87-1.46 8.05-8.22 8.55-11.01.34-1.9-.24-2.85-1.31-2.85Z"></path><path fill="#FFD21E" d="M56.33 76.69c-2.75-4.04-2.56-7.07 1.22-10.84 3.77-3.77 5.97-9.3 5.97-9.3s.82-3.2 2.7-2.9c1.86.3 3.23 5.08-.68 8.01-3.92 2.93.78 4.92 2.28 2.17 1.51-2.75 5.63-9.82 7.76-11.18 2.13-1.35 3.64-.6 3.13 2.2-.5 2.79-9.42 9.55-8.55 11 .86 1.47 3.92-1.71 3.92-1.71s9.58-8.71 11.66-6.44c2.08 2.27-1.58 4.17-6.8 7.33-5.23 3.16-5.63 4-4.9 5.2.75 1.2 12.28-8.53 13.36-4.4 1.08 4.11-11.76 5.3-10.97 8.15.8 2.85 9.05-5.38 10.74-2.18 1.69 3.21-11.65 6.98-11.76 7.01-4.31 1.12-15.26 3.49-19.08-2.12Z"></path></svg>

	

	<span>Datasets</span>
	

	</div></a><div class="relative inline-block ">
	<button class="group mr-1 mb-1 md:mr-1.5 md:mb-1.5  rounded-lg rounded-br-none " type="button">
		<div class="tag tag-white   relative rounded-br-none pr-2.5"><svg class="text-black inline-block text-sm" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" fill="none" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M22.2812 12.2656L25.9532 15.7187C26.8932 16.6587 28.0913 17.3931 29.0313 16.4531C29.8594 15.7812 29.9332 14.1 29.9532 13.5C29.9532 11.5625 28.9219 8.375 27.25 6.71875C25.5604 5.04493 23.3782 3.91692 22.7032 3.78125L22.2812 12.2656Z" fill="#F5AB6A"></path><path d="M22.2812 12.2656L25.9532 15.7187C26.8932 16.6587 28.0913 17.3931 29.0313 16.4531C29.8594 15.7812 29.9332 14.1 29.9532 13.5C29.9532 11.5625 28.9219 8.375 27.25 6.71875C25.5604 5.04493 23.3782 3.91692 22.7032 3.78125L22.2812 12.2656Z" fill="url(#paint0_radial_18_31665)"></path><g filter="url(#filter0_f_18_31665)"><path d="M22.2849 12.1817L23.4375 13.2656L24.4375 4.70312C23.5121 4.1242 23.0198 3.96369 22.6563 3.89062L22.2849 12.1817Z" fill="url(#paint1_linear_18_31665)"></path></g><path d="M22.7969 3.05741L21.8437 2.65269C19.6421 1.96765 17.2344 1.81208 14.6719 2.23236C12.1094 2.65264 11.5156 3.2442 11.5156 3.2442C10.896 3.3674 10.8671 3.88898 11.1718 4.45363C13.0797 6.72576 15.176 13.4043 18.0469 14.2506C18.89 14.4662 21.3791 14.4776 22.2019 14.1593L22.3238 14.108C22.6692 13.9745 23.1672 13.2814 23.1875 12.9118L23.4219 4.03043C23.4523 3.59924 23.1484 3.25204 22.7969 3.05741Z" fill="url(#paint2_radial_18_31665)"></path><path d="M22.7969 3.05741L21.8437 2.65269C19.6421 1.96765 17.2344 1.81208 14.6719 2.23236C12.1094 2.65264 11.5156 3.2442 11.5156 3.2442C10.896 3.3674 10.8671 3.88898 11.1718 4.45363C13.0797 6.72576 15.176 13.4043 18.0469 14.2506C18.89 14.4662 21.3791 14.4776 22.2019 14.1593L22.3238 14.108C22.6692 13.9745 23.1672 13.2814 23.1875 12.9118L23.4219 4.03043C23.4523 3.59924 23.1484 3.25204 22.7969 3.05741Z" fill="url(#paint3_radial_18_31665)"></path><path d="M22.7969 3.05741L21.8437 2.65269C19.6421 1.96765 17.2344 1.81208 14.6719 2.23236C12.1094 2.65264 11.5156 3.2442 11.5156 3.2442C10.896 3.3674 10.8671 3.88898 11.1718 4.45363C13.0797 6.72576 15.176 13.4043 18.0469 14.2506C18.89 14.4662 21.3791 14.4776 22.2019 14.1593L22.3238 14.108C22.6692 13.9745 23.1672 13.2814 23.1875 12.9118L23.4219 4.03043C23.4523 3.59924 23.1484 3.25204 22.7969 3.05741Z" fill="url(#paint4_radial_18_31665)"></path><path d="M22.7969 3.05741L21.8437 2.65269C19.6421 1.96765 17.2344 1.81208 14.6719 2.23236C12.1094 2.65264 11.5156 3.2442 11.5156 3.2442C10.896 3.3674 10.8671 3.88898 11.1718 4.45363C13.0797 6.72576 15.176 13.4043 18.0469 14.2506C18.89 14.4662 21.3791 14.4776 22.2019 14.1593L22.3238 14.108C22.6692 13.9745 23.1672 13.2814 23.1875 12.9118L23.4219 4.03043C23.4523 3.59924 23.1484 3.25204 22.7969 3.05741Z" fill="url(#paint5_radial_18_31665)"></path><path d="M22.7969 3.05741L21.8437 2.65269C19.6421 1.96765 17.2344 1.81208 14.6719 2.23236C12.1094 2.65264 11.5156 3.2442 11.5156 3.2442C10.896 3.3674 10.8671 3.88898 11.1718 4.45363C13.0797 6.72576 15.176 13.4043 18.0469 14.2506C18.89 14.4662 21.3791 14.4776 22.2019 14.1593L22.3238 14.108C22.6692 13.9745 23.1672 13.2814 23.1875 12.9118L23.4219 4.03043C23.4523 3.59924 23.1484 3.25204 22.7969 3.05741Z" fill="url(#paint6_linear_18_31665)"></path><g filter="url(#filter1_f_18_31665)"><path d="M13.1016 2.72656C11.862 3.06924 11.5298 3.40016 11.5298 3.40016C10.9102 3.52335 10.936 4.11525 11.2408 4.6799C13.1487 6.95202 15.1361 13.2496 18.007 14.0958C18.2707 14.1633 18.6953 14.2107 19.1797 14.2344L13.1016 2.72656Z" fill="url(#paint7_linear_18_31665)"></path></g><path d="M12.2187 22.7187L15.7656 26.2031C16.7332 27.1171 17.3334 28.2487 16.4219 29.2188C15.7749 30.0687 14.2241 29.9933 13.625 30.0313C11.6883 30.0891 9.09014 29.5622 6.84373 27.5313C5.07737 25.9343 4.09321 23.688 3.93751 23.0156L12.2187 22.7187Z" fill="url(#paint8_radial_18_31665)"></path><path d="M12.2187 22.7187L15.7656 26.2031C16.7332 27.1171 17.3334 28.2487 16.4219 29.2188C15.7749 30.0687 14.2241 29.9933 13.625 30.0313C11.6883 30.0891 9.09014 29.5622 6.84373 27.5313C5.07737 25.9343 4.09321 23.688 3.93751 23.0156L12.2187 22.7187Z" fill="url(#paint9_radial_18_31665)"></path><g filter="url(#filter2_f_18_31665)"><path d="M12.0523 22.7916L13.2187 23.9375L4.81018 24.5721C4.4328 23.8671 4.20835 23.2768 4.14062 22.9844L12.0523 22.7916Z" fill="url(#paint10_linear_18_31665)"></path><path d="M12.0523 22.7916L13.2187 23.9375L4.81018 24.5721C4.4328 23.8671 4.20835 23.2768 4.14062 22.9844L12.0523 22.7916Z" fill="url(#paint11_radial_18_31665)"></path></g><path d="M2.99219 22.6484C2.07068 20.538 1.78913 17.4452 2.21096 15.0703C2.63279 12.6953 3.20759 11.951 3.20759 11.951C3.31231 11.3264 3.83281 11.2818 4.40626 11.5703C6.73409 13.4144 13.3119 15.2264 14.2432 18.0781C14.5947 19.3034 14.6279 21.3125 14.1641 22.5156C14.0409 22.8657 13.5 23.3516 13.0625 23.4141C12.625 23.4765 9.47656 23.3516 8.61719 23.3516C7.75781 23.3516 4.64844 23.6719 4.14062 23.6172C3.63281 23.5625 3.23437 23.2031 2.99219 22.6484Z" fill="#EC9F6A"></path><path d="M2.99219 22.6484C2.07068 20.538 1.78913 17.4452 2.21096 15.0703C2.63279 12.6953 3.20759 11.951 3.20759 11.951C3.31231 11.3264 3.83281 11.2818 4.40626 11.5703C6.73409 13.4144 13.3119 15.2264 14.2432 18.0781C14.5947 19.3034 14.6279 21.3125 14.1641 22.5156C14.0409 22.8657 13.5 23.3516 13.0625 23.4141C12.625 23.4765 9.47656 23.3516 8.61719 23.3516C7.75781 23.3516 4.64844 23.6719 4.14062 23.6172C3.63281 23.5625 3.23437 23.2031 2.99219 22.6484Z" fill="url(#paint12_radial_18_31665)"></path><path d="M2.99219 22.6484C2.07068 20.538 1.78913 17.4452 2.21096 15.0703C2.63279 12.6953 3.20759 11.951 3.20759 11.951C3.31231 11.3264 3.83281 11.2818 4.40626 11.5703C6.73409 13.4144 13.3119 15.2264 14.2432 18.0781C14.5947 19.3034 14.6279 21.3125 14.1641 22.5156C14.0409 22.8657 13.5 23.3516 13.0625 23.4141C12.625 23.4765 9.47656 23.3516 8.61719 23.3516C7.75781 23.3516 4.64844 23.6719 4.14062 23.6172C3.63281 23.5625 3.23437 23.2031 2.99219 22.6484Z" fill="url(#paint13_linear_18_31665)"></path><path d="M2.99219 22.6484C2.07068 20.538 1.78913 17.4452 2.21096 15.0703C2.63279 12.6953 3.20759 11.951 3.20759 11.951C3.31231 11.3264 3.83281 11.2818 4.40626 11.5703C6.73409 13.4144 13.3119 15.2264 14.2432 18.0781C14.5947 19.3034 14.6279 21.3125 14.1641 22.5156C14.0409 22.8657 13.5 23.3516 13.0625 23.4141C12.625 23.4765 9.47656 23.3516 8.61719 23.3516C7.75781 23.3516 4.64844 23.6719 4.14062 23.6172C3.63281 23.5625 3.23437 23.2031 2.99219 22.6484Z" fill="url(#paint14_radial_18_31665)"></path><path d="M2.99219 22.6484C2.07068 20.538 1.78913 17.4452 2.21096 15.0703C2.63279 12.6953 3.20759 11.951 3.20759 11.951C3.31231 11.3264 3.83281 11.2818 4.40626 11.5703C6.73409 13.4144 13.3119 15.2264 14.2432 18.0781C14.5947 19.3034 14.6279 21.3125 14.1641 22.5156C14.0409 22.8657 13.5 23.3516 13.0625 23.4141C12.625 23.4765 9.47656 23.3516 8.61719 23.3516C7.75781 23.3516 4.64844 23.6719 4.14062 23.6172C3.63281 23.5625 3.23437 23.2031 2.99219 22.6484Z" fill="url(#paint15_radial_18_31665)"></path><path d="M2.99219 22.6484C2.07068 20.538 1.78913 17.4452 2.21096 15.0703C2.63279 12.6953 3.20759 11.951 3.20759 11.951C3.31231 11.3264 3.83281 11.2818 4.40626 11.5703C6.73409 13.4144 13.3119 15.2264 14.2432 18.0781C14.5947 19.3034 14.6279 21.3125 14.1641 22.5156C14.0409 22.8657 13.5 23.3516 13.0625 23.4141C12.625 23.4765 9.47656 23.3516 8.61719 23.3516C7.75781 23.3516 4.64844 23.6719 4.14062 23.6172C3.63281 23.5625 3.23437 23.2031 2.99219 22.6484Z" fill="url(#paint16_radial_18_31665)"></path><g filter="url(#filter3_f_18_31665)"><path d="M2.70313 13.6719C3.04135 12.4711 3.36224 12.0555 3.36224 12.0555C3.46697 11.4309 3.98746 11.3864 4.56092 11.6749C6.88874 13.5189 13.0809 15.1104 14.0121 17.9622C14.1731 18.5231 14.2766 19.0394 14.3287 19.5128L2.70313 13.6719Z" fill="url(#paint17_linear_18_31665)"></path><path d="M2.70313 13.6719C3.04135 12.4711 3.36224 12.0555 3.36224 12.0555C3.46697 11.4309 3.98746 11.3864 4.56092 11.6749C6.88874 13.5189 13.0809 15.1104 14.0121 17.9622C14.1731 18.5231 14.2766 19.0394 14.3287 19.5128L2.70313 13.6719Z" fill="url(#paint18_linear_18_31665)"></path></g><path d="M9.83184 2.82184C6.62184 4.07184 4.07184 6.62184 2.82184 9.83184C2.48184 10.7118 2.82184 11.7018 3.62184 12.1918L14.0418 18.5418C14.6618 18.9218 15.4418 18.9218 16.0618 18.5418C17.0718 17.9218 17.9318 17.0718 18.5418 16.0618C18.9218 15.4418 18.9218 14.6618 18.5418 14.0418L12.1918 3.62184C11.7018 2.82184 10.7118 2.48184 9.83184 2.82184Z" fill="#D79453"></path><path d="M9.83184 2.82184C6.62184 4.07184 4.07184 6.62184 2.82184 9.83184C2.48184 10.7118 2.82184 11.7018 3.62184 12.1918L14.0418 18.5418C14.6618 18.9218 15.4418 18.9218 16.0618 18.5418C17.0718 17.9218 17.9318 17.0718 18.5418 16.0618C18.9218 15.4418 18.9218 14.6618 18.5418 14.0418L12.1918 3.62184C11.7018 2.82184 10.7118 2.48184 9.83184 2.82184Z" fill="url(#paint19_radial_18_31665)"></path><path d="M9.83184 2.82184C6.62184 4.07184 4.07184 6.62184 2.82184 9.83184C2.48184 10.7118 2.82184 11.7018 3.62184 12.1918L14.0418 18.5418C14.6618 18.9218 15.4418 18.9218 16.0618 18.5418C17.0718 17.9218 17.9318 17.0718 18.5418 16.0618C18.9218 15.4418 18.9218 14.6618 18.5418 14.0418L12.1918 3.62184C11.7018 2.82184 10.7118 2.48184 9.83184 2.82184Z" fill="url(#paint20_radial_18_31665)"></path><path d="M9.83184 2.82184C6.62184 4.07184 4.07184 6.62184 2.82184 9.83184C2.48184 10.7118 2.82184 11.7018 3.62184 12.1918L14.0418 18.5418C14.6618 18.9218 15.4418 18.9218 16.0618 18.5418C17.0718 17.9218 17.9318 17.0718 18.5418 16.0618C18.9218 15.4418 18.9218 14.6618 18.5418 14.0418L12.1918 3.62184C11.7018 2.82184 10.7118 2.48184 9.83184 2.82184Z" fill="url(#paint21_radial_18_31665)"></path><path d="M9.83184 2.82184C6.62184 4.07184 4.07184 6.62184 2.82184 9.83184C2.48184 10.7118 2.82184 11.7018 3.62184 12.1918L14.0418 18.5418C14.6618 18.9218 15.4418 18.9218 16.0618 18.5418C17.0718 17.9218 17.9318 17.0718 18.5418 16.0618C18.9218 15.4418 18.9218 14.6618 18.5418 14.0418L12.1918 3.62184C11.7018 2.82184 10.7118 2.48184 9.83184 2.82184Z" fill="url(#paint22_linear_18_31665)"></path><path d="M9.83184 2.82184C6.62184 4.07184 4.07184 6.62184 2.82184 9.83184C2.48184 10.7118 2.82184 11.7018 3.62184 12.1918L14.0418 18.5418C14.6618 18.9218 15.4418 18.9218 16.0618 18.5418C17.0718 17.9218 17.9318 17.0718 18.5418 16.0618C18.9218 15.4418 18.9218 14.6618 18.5418 14.0418L12.1918 3.62184C11.7018 2.82184 10.7118 2.48184 9.83184 2.82184Z" fill="url(#paint23_radial_18_31665)"></path><path d="M9.83184 2.82184C6.62184 4.07184 4.07184 6.62184 2.82184 9.83184C2.48184 10.7118 2.82184 11.7018 3.62184 12.1918L14.0418 18.5418C14.6618 18.9218 15.4418 18.9218 16.0618 18.5418C17.0718 17.9218 17.9318 17.0718 18.5418 16.0618C18.9218 15.4418 18.9218 14.6618 18.5418 14.0418L12.1918 3.62184C11.7018 2.82184 10.7118 2.48184 9.83184 2.82184Z" fill="url(#paint24_radial_18_31665)"></path><defs><filter id="filter0_f_18_31665" x="22.0349" y="3.64062" width="2.65265" height="9.875" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB"><feFlood flood-opacity="0" result="BackgroundImageFix"></feFlood><feBlend mode="normal" in="SourceGraphic" in2="BackgroundImageFix" result="shape"></feBlend><feGaussianBlur stdDeviation="0.125" result="effect1_foregroundBlur_18_31665"></feGaussianBlur></filter><filter id="filter1_f_18_31665" x="10.7815" y="2.47656" width="8.64819" height="12.0078" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB"><feFlood flood-opacity="0" result="BackgroundImageFix"></feFlood><feBlend mode="normal" in="SourceGraphic" in2="BackgroundImageFix" result="shape"></feBlend><feGaussianBlur stdDeviation="0.125" result="effect1_foregroundBlur_18_31665"></feGaussianBlur></filter><filter id="filter2_f_18_31665" x="3.89062" y="22.5416" width="9.57812" height="2.2804" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB"><feFlood flood-opacity="0" result="BackgroundImageFix"></feFlood><feBlend mode="normal" in="SourceGraphic" in2="BackgroundImageFix" result="shape"></feBlend><feGaussianBlur stdDeviation="0.125" result="effect1_foregroundBlur_18_31665"></feGaussianBlur></filter><filter id="filter3_f_18_31665" x="2.45312" y="11.2538" width="12.1255" height="8.50903" filterUnits="userSpaceOnUse" color-interpolation-filters="sRGB"><feFlood flood-opacity="0" result="BackgroundImageFix"></feFlood><feBlend mode="normal" in="SourceGraphic" in2="BackgroundImageFix" result="shape"></feBlend><feGaussianBlur stdDeviation="0.125" result="effect1_foregroundBlur_18_31665"></feGaussianBlur></filter><radialGradient id="paint0_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(22.8125 12.9375) rotate(42.7741) scale(12.5164 7.08839)"><stop offset="0.0937591" stop-color="#C05159"></stop><stop offset="0.553697" stop-color="#F6AC6A"></stop><stop offset="0.832916" stop-color="#FFD186"></stop><stop offset="0.916927" stop-color="#FFDC87"></stop></radialGradient><linearGradient id="paint1_linear_18_31665" x1="24.7344" y1="4.67187" x2="20.8594" y2="12.8906" gradientUnits="userSpaceOnUse"><stop stop-color="#EBD67C"></stop><stop offset="0.0655686" stop-color="#FFFFA6"></stop><stop offset="0.530552" stop-color="#F8C281"></stop><stop offset="0.937338" stop-color="#E99E6B"></stop></linearGradient><radialGradient id="paint2_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(22.9009 13.1847) rotate(-127.648) scale(14.3438 11.7966)"><stop stop-color="#FFBE66"></stop><stop offset="1" stop-color="#E2AE5B"></stop></radialGradient><radialGradient id="paint3_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(18 11.4375) rotate(53.9726) scale(11.9013 4.84018)"><stop stop-color="#D67C63"></stop><stop offset="1" stop-color="#D97D67" stop-opacity="0"></stop></radialGradient><radialGradient id="paint4_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(23 4.1875) rotate(45.7639) scale(3.31486 5.75622)"><stop stop-color="#FFE4A6"></stop><stop offset="0.711285" stop-color="#F8B76F"></stop><stop offset="1" stop-color="#F9B870" stop-opacity="0"></stop></radialGradient><radialGradient id="paint5_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(22.875 12.4375) rotate(88.9391) scale(3.37558 1.29066)"><stop stop-color="#FFBC67"></stop><stop offset="1" stop-color="#FFBC67" stop-opacity="0"></stop></radialGradient><linearGradient id="paint6_linear_18_31665" x1="20.375" y1="15.6875" x2="20.125" y2="12.7813" gradientUnits="userSpaceOnUse"><stop offset="0.461609" stop-color="#B45077"></stop><stop offset="0.855389" stop-color="#B75077" stop-opacity="0"></stop></linearGradient><linearGradient id="paint7_linear_18_31665" x1="12.9375" y1="2.57056" x2="18.5625" y2="14.3891" gradientUnits="userSpaceOnUse"><stop stop-color="#DDC173"></stop><stop offset="0.485173" stop-color="#D59F65"></stop><stop offset="1" stop-color="#E49966"></stop></linearGradient><radialGradient id="paint8_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(13.5625 23.5) rotate(109.113) scale(6.68078 10.2578)"><stop offset="0.165756" stop-color="#FFBF7E"></stop><stop offset="0.827674" stop-color="#DF8C6D"></stop><stop offset="1" stop-color="#B05A66"></stop></radialGradient><radialGradient id="paint9_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(15.1875 26) rotate(41.0652) scale(8.37243 2.03649)"><stop stop-color="#FFD483"></stop><stop offset="1" stop-color="#FFD688" stop-opacity="0"></stop></radialGradient><linearGradient id="paint10_linear_18_31665" x1="3.96063" y1="23.794" x2="13.3748" y2="23.5143" gradientUnits="userSpaceOnUse"><stop stop-color="#A8716F"></stop><stop offset="0.103615" stop-color="#B37173"></stop><stop offset="0.225484" stop-color="#DB9F84"></stop><stop offset="0.799889" stop-color="#F1BB8A"></stop><stop offset="1" stop-color="#FFD780"></stop></linearGradient><radialGradient id="paint11_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(11.4219 23.1719) rotate(-178.616) scale(3.23532 0.569081)"><stop offset="0.621498" stop-color="#AF5A3E"></stop><stop offset="1" stop-color="#B35445" stop-opacity="0"></stop></radialGradient><radialGradient id="paint12_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(13.625 19.125) rotate(-171.737) scale(15.2205 15.0767)"><stop offset="0.138435" stop-color="#FFB974"></stop><stop offset="0.403618" stop-color="#F2A56D"></stop><stop offset="0.925938" stop-color="#A16948"></stop></radialGradient><linearGradient id="paint13_linear_18_31665" x1="8.22184" y1="13.125" x2="6.81191" y2="15.4996" gradientUnits="userSpaceOnUse"><stop offset="0.610751" stop-color="#984847"></stop><stop offset="0.850075" stop-color="#9A4947" stop-opacity="0"></stop></linearGradient><radialGradient id="paint14_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(7.25 23.7461) scale(11.25 5.68361)"><stop stop-color="#C66364"></stop><stop offset="1" stop-color="#D4766B" stop-opacity="0"></stop></radialGradient><radialGradient id="paint15_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(7.875 23.5313) scale(10.0937 1.29657)"><stop stop-color="#B64B4B"></stop><stop offset="1" stop-color="#C56158" stop-opacity="0"></stop></radialGradient><radialGradient id="paint16_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(11.4375 19.875) rotate(-46.8882) scale(4.02385 7.51767)"><stop stop-color="#FFC083"></stop><stop offset="0.620218" stop-color="#FFBD7D" stop-opacity="0"></stop></radialGradient><linearGradient id="paint17_linear_18_31665" x1="2.8125" y1="13.0312" x2="14.5582" y2="18.9404" gradientUnits="userSpaceOnUse"><stop stop-color="#B89367"></stop><stop offset="1" stop-color="#C5835E"></stop></linearGradient><linearGradient id="paint18_linear_18_31665" x1="8.21875" y1="14.6406" x2="7.59349" y2="15.6717" gradientUnits="userSpaceOnUse"><stop offset="0.351552" stop-color="#A74746"></stop><stop offset="0.845198" stop-color="#A04346" stop-opacity="0"></stop></linearGradient><radialGradient id="paint19_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(15.5625 14.5625) rotate(140.244) scale(18.3733 13.7403)"><stop stop-color="#FDAE69"></stop><stop offset="0.729021" stop-color="#CE8C4F"></stop><stop offset="0.921546" stop-color="#AD7B45"></stop><stop offset="1" stop-color="#8B6B4A"></stop></radialGradient><radialGradient id="paint20_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(15.0625 7) rotate(65.3152) scale(11.0745 3.16547)"><stop offset="0.233237" stop-color="#FFD47C"></stop><stop offset="0.853648" stop-color="#FFD98B" stop-opacity="0"></stop></radialGradient><radialGradient id="paint21_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(15.3125 8.875) rotate(100.886) scale(6.6191 5.57808)"><stop offset="0.128419" stop-color="#FFD88C"></stop><stop offset="0.924134" stop-color="#FFBE7B" stop-opacity="0"></stop></radialGradient><linearGradient id="paint22_linear_18_31665" x1="7.25" y1="15.1875" x2="10.7588" y2="10.3142" gradientUnits="userSpaceOnUse"><stop offset="0.142353" stop-color="#C15F4D"></stop><stop offset="1" stop-color="#D58366" stop-opacity="0"></stop></linearGradient><radialGradient id="paint23_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(8.15625 15.7813) rotate(28.5422) scale(12.5574 1.96589)"><stop offset="0.149989" stop-color="#E4745D"></stop><stop offset="0.453292" stop-color="#C8604C"></stop><stop offset="0.632597" stop-color="#C0605F"></stop><stop offset="1" stop-color="#C0605F" stop-opacity="0"></stop></radialGradient><radialGradient id="paint24_radial_18_31665" cx="0" cy="0" r="1" gradientUnits="userSpaceOnUse" gradientTransform="translate(1.40625 2.69067) rotate(46.0943) scale(22.3963)"><stop offset="0.935802" stop-color="#C17C61" stop-opacity="0"></stop><stop offset="0.982109" stop-color="#C17C61"></stop></radialGradient></defs></svg>

	

	<span>Croissant</span>
	

	<div class="border-br-gray-200 absolute bottom-0.5 right-0.5 h-1 w-1 border-[3px] border-l-transparent border-t-transparent border-b-gray-200 border-r-gray-200 dark:border-b-gray-700 dark:border-r-gray-700"></div></div>
		
		</button>
	
	
	</div>

	</div><div class="mr-1 flex flex-wrap items-center"><span class="mb-1 mr-1 p-1 text-sm leading-tight text-gray-400 md:mb-1.5">License:
	</span>
	<div class="relative inline-block ">
	<button class="group mr-1 mb-1 md:mr-1.5 md:mb-1.5  rounded-full rounded-br-none " type="button">
		<div class="tag tag-white rounded-full  relative rounded-br-none pr-2.5">
		<svg class="text-xs text-gray-900" width="1em" height="1em" viewBox="0 0 10 10" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M1.46009 5.0945V6.88125C1.46009 7.25201 1.75937 7.55129 2.13012 7.55129C2.50087 7.55129 2.80016 7.25201 2.80016 6.88125V5.0945C2.80016 4.72375 2.50087 4.42446 2.13012 4.42446C1.75937 4.42446 1.46009 4.72375 1.46009 5.0945ZM4.14022 5.0945V6.88125C4.14022 7.25201 4.4395 7.55129 4.81026 7.55129C5.18101 7.55129 5.48029 7.25201 5.48029 6.88125V5.0945C5.48029 4.72375 5.18101 4.42446 4.81026 4.42446C4.4395 4.42446 4.14022 4.72375 4.14022 5.0945ZM1.23674 9.78473H8.38377C8.75452 9.78473 9.0538 9.48545 9.0538 9.1147C9.0538 8.74395 8.75452 8.44466 8.38377 8.44466H1.23674C0.865993 8.44466 0.566711 8.74395 0.566711 9.1147C0.566711 9.48545 0.865993 9.78473 1.23674 9.78473ZM6.82036 5.0945V6.88125C6.82036 7.25201 7.11964 7.55129 7.49039 7.55129C7.86114 7.55129 8.16042 7.25201 8.16042 6.88125V5.0945C8.16042 4.72375 7.86114 4.42446 7.49039 4.42446C7.11964 4.42446 6.82036 4.72375 6.82036 5.0945ZM4.39484 0.623142L0.865993 2.48137C0.682851 2.57517 0.566711 2.76725 0.566711 2.97273C0.566711 3.28094 0.816857 3.53109 1.12507 3.53109H8.49991C8.80365 3.53109 9.0538 3.28094 9.0538 2.97273C9.0538 2.76725 8.93766 2.57517 8.75452 2.48137L5.22568 0.623142C4.9666 0.484669 4.65391 0.484669 4.39484 0.623142V0.623142Z" fill="currentColor"></path></svg>

	

	<span>cc-by-4.0</span>
	

	<div class="border-br-gray-200 absolute bottom-0.5 right-0.5 h-1 w-1 border-[3px] border-l-transparent border-t-transparent border-b-gray-200 border-r-gray-200 dark:border-b-gray-700 dark:border-r-gray-700"></div></div>
		
		</button>
	
	
	</div>

	</div></div>

		<div class="flex flex-col-reverse lg:flex-row lg:items-center lg:justify-between"><div class="-mb-px flex h-12 items-center overflow-x-auto overflow-y-hidden ">
	<a class="tab-alternate active" href="/datasets/TomTBT/pmc_open_access_section"><svg class="mr-1.5 text-gray-400 flex-none" style="" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 24 24"><path class="uim-quaternary" d="M20.23 7.24L12 12L3.77 7.24a1.98 1.98 0 0 1 .7-.71L11 2.76c.62-.35 1.38-.35 2 0l6.53 3.77c.29.173.531.418.7.71z" opacity=".25" fill="currentColor"></path><path class="uim-tertiary" d="M12 12v9.5a2.09 2.09 0 0 1-.91-.21L4.5 17.48a2.003 2.003 0 0 1-1-1.73v-7.5a2.06 2.06 0 0 1 .27-1.01L12 12z" opacity=".5" fill="currentColor"></path><path class="uim-primary" d="M20.5 8.25v7.5a2.003 2.003 0 0 1-1 1.73l-6.62 3.82c-.275.13-.576.198-.88.2V12l8.23-4.76c.175.308.268.656.27 1.01z" fill="currentColor"></path></svg>
	Dataset card
	

	
		</a><a class="tab-alternate" href="/datasets/TomTBT/pmc_open_access_section/viewer/"><svg class="mr-1.5 text-gray-400 flex-none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 12 12"><path fill="currentColor" d="M2.5 2h7a1 1 0 0 1 1 1v6a1 1 0 0 1-1 1h-7a1 1 0 0 1-1-1V3a1 1 0 0 1 1-1Zm0 2v2h3V4h-3Zm4 0v2h3V4h-3Zm-4 3v2h3V7h-3Zm4 0v2h3V7h-3Z"></path></svg>
	Data Studio
	

	
		</a><a class="tab-alternate" href="/datasets/TomTBT/pmc_open_access_section/tree/main"><svg class="mr-1.5 text-gray-400 flex-none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 24 24"><path class="uim-tertiary" d="M21 19h-8a1 1 0 0 1 0-2h8a1 1 0 0 1 0 2zm0-4h-8a1 1 0 0 1 0-2h8a1 1 0 0 1 0 2zm0-8h-8a1 1 0 0 1 0-2h8a1 1 0 0 1 0 2zm0 4h-8a1 1 0 0 1 0-2h8a1 1 0 0 1 0 2z" opacity=".5" fill="currentColor"></path><path class="uim-primary" d="M9 19a1 1 0 0 1-1-1V6a1 1 0 0 1 2 0v12a1 1 0 0 1-1 1zm-6-4.333a1 1 0 0 1-.64-1.769L3.438 12l-1.078-.898a1 1 0 0 1 1.28-1.538l2 1.667a1 1 0 0 1 0 1.538l-2 1.667a.999.999 0 0 1-.64.231z" fill="currentColor"></path></svg>
	<span class="xl:hidden">Files</span>
		<span class="hidden xl:inline">Files and versions</span>
	

	
		</a><a class="tab-alternate" href="/datasets/TomTBT/pmc_open_access_section/discussions"><svg class="mr-1.5 text-gray-400 flex-none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M20.6081 3C21.7684 3 22.8053 3.49196 23.5284 4.38415C23.9756 4.93678 24.4428 5.82749 24.4808 7.16133C24.9674 7.01707 25.4353 6.93643 25.8725 6.93643C26.9833 6.93643 27.9865 7.37587 28.696 8.17411C29.6075 9.19872 30.0124 10.4579 29.8361 11.7177C29.7523 12.3177 29.5581 12.8555 29.2678 13.3534C29.8798 13.8646 30.3306 14.5763 30.5485 15.4322C30.719 16.1032 30.8939 17.5006 29.9808 18.9403C30.0389 19.0342 30.0934 19.1319 30.1442 19.2318C30.6932 20.3074 30.7283 21.5229 30.2439 22.6548C29.5093 24.3704 27.6841 25.7219 24.1397 27.1727C21.9347 28.0753 19.9174 28.6523 19.8994 28.6575C16.9842 29.4379 14.3477 29.8345 12.0653 29.8345C7.87017 29.8345 4.8668 28.508 3.13831 25.8921C0.356375 21.6797 0.754104 17.8269 4.35369 14.1131C6.34591 12.058 7.67023 9.02782 7.94613 8.36275C8.50224 6.39343 9.97271 4.20438 12.4172 4.20438H12.4179C12.6236 4.20438 12.8314 4.2214 13.0364 4.25468C14.107 4.42854 15.0428 5.06476 15.7115 6.02205C16.4331 5.09583 17.134 4.359 17.7682 3.94323C18.7242 3.31737 19.6794 3 20.6081 3ZM20.6081 5.95917C20.2427 5.95917 19.7963 6.1197 19.3039 6.44225C17.7754 7.44319 14.8258 12.6772 13.7458 14.7131C13.3839 15.3952 12.7655 15.6837 12.2086 15.6837C11.1036 15.6837 10.2408 14.5497 12.1076 13.1085C14.9146 10.9402 13.9299 7.39584 12.5898 7.1776C12.5311 7.16799 12.4731 7.16355 12.4172 7.16355C11.1989 7.16355 10.6615 9.33114 10.6615 9.33114C10.6615 9.33114 9.0863 13.4148 6.38031 16.206C3.67434 18.998 3.5346 21.2388 5.50675 24.2246C6.85185 26.2606 9.42666 26.8753 12.0653 26.8753C14.8021 26.8753 17.6077 26.2139 19.1799 25.793C19.2574 25.7723 28.8193 22.984 27.6081 20.6107C27.4046 20.212 27.0693 20.0522 26.6471 20.0522C24.9416 20.0522 21.8393 22.6726 20.5057 22.6726C20.2076 22.6726 19.9976 22.5416 19.9116 22.222C19.3433 20.1173 28.552 19.2325 27.7758 16.1839C27.639 15.6445 27.2677 15.4256 26.746 15.4263C24.4923 15.4263 19.4358 19.5181 18.3759 19.5181C18.2949 19.5181 18.2368 19.4937 18.2053 19.4419C17.6743 18.557 17.9653 17.9394 21.7082 15.6009C25.4511 13.2617 28.0783 11.8545 26.5841 10.1752C26.4121 9.98141 26.1684 9.8956 25.8725 9.8956C23.6001 9.89634 18.2311 14.9403 18.2311 14.9403C18.2311 14.9403 16.7821 16.496 15.9057 16.496C15.7043 16.496 15.533 16.4139 15.4169 16.2112C14.7956 15.1296 21.1879 10.1286 21.5484 8.06535C21.7928 6.66715 21.3771 5.95917 20.6081 5.95917Z" fill="#FF9D00"></path><path d="M5.50686 24.2246C3.53472 21.2387 3.67446 18.9979 6.38043 16.206C9.08641 13.4147 10.6615 9.33111 10.6615 9.33111C10.6615 9.33111 11.2499 6.95933 12.59 7.17757C13.93 7.39581 14.9139 10.9401 12.1069 13.1084C9.29997 15.276 12.6659 16.7489 13.7459 14.713C14.8258 12.6772 17.7747 7.44316 19.304 6.44221C20.8326 5.44128 21.9089 6.00204 21.5484 8.06532C21.188 10.1286 14.795 15.1295 15.4171 16.2118C16.0391 17.2934 18.2312 14.9402 18.2312 14.9402C18.2312 14.9402 25.0907 8.49588 26.5842 10.1752C28.0776 11.8545 25.4512 13.2616 21.7082 15.6008C17.9646 17.9393 17.6744 18.557 18.2054 19.4418C18.7372 20.3266 26.9998 13.1351 27.7759 16.1838C28.5513 19.2324 19.3434 20.1173 19.9117 22.2219C20.48 24.3274 26.3979 18.2382 27.6082 20.6107C28.8193 22.9839 19.2574 25.7722 19.18 25.7929C16.0914 26.62 8.24723 28.3726 5.50686 24.2246Z" fill="#FFD21E"></path></svg>
	Community
	<div class="ml-1.5 flex h-4 min-w-[1rem] items-center justify-center rounded px-1 text-xs leading-none shadow-sm bg-black text-white dark:bg-gray-800 dark:text-gray-200">1</div>

	
		</a></div>
	
			</div></div></header>
</div>
	
<div class="container relative flex flex-col md:grid md:space-y-0 w-full md:grid-cols-12 md:flex-1 md:grid-rows-full space-y-4 md:gap-6 ">
		<section class="pt-6 border-gray-100 md:col-span-8 pb-24 relative break-words copiable-code-container">
				<div class="SVELTE_HYDRATER contents" data-target="UnsafeBanner" data-props="{&quot;classNames&quot;:&quot;mb-4&quot;,&quot;repoId&quot;:&quot;TomTBT/pmc_open_access_section&quot;,&quot;repoType&quot;:&quot;dataset&quot;,&quot;minLevel&quot;:&quot;unsafe&quot;}"></div>
					<div class="SVELTE_HYDRATER contents" data-target="DatasetViewer" data-props="{&quot;data&quot;:{&quot;kind&quot;:&quot;DatasetAndSampleData&quot;,&quot;datasetInfo&quot;:[{&quot;isValid&quot;:true,&quot;href&quot;:&quot;/datasets/TomTBT/pmc_open_access_section/tree/refs%2Fconvert%2Fparquet/&quot;,&quot;label&quot;:&quot;Size of the auto-converted Parquet files (First 5GB per split):&quot;,&quot;value&quot;:&quot;7.86 GB&quot;},{&quot;isValid&quot;:true,&quot;href&quot;:&quot;&quot;,&quot;label&quot;:&quot;Number of rows (First 5GB per split):&quot;,&quot;value&quot;:&quot;720,000&quot;},{&quot;isValid&quot;:true,&quot;href&quot;:&quot;&quot;,&quot;label&quot;:&quot;Estimated number of rows:&quot;,&quot;value&quot;:&quot;7,219,374&quot;}],&quot;partial&quot;:true,&quot;configsData&quot;:{&quot;configInfos&quot;:[{&quot;name&quot;:&quot;commercial&quot;,&quot;status&quot;:&quot;ok&quot;,&quot;numRows&quot;:212000,&quot;estimatedNumRows&quot;:4308862},{&quot;name&quot;:&quot;non_commercial&quot;,&quot;status&quot;:&quot;ok&quot;,&quot;numRows&quot;:254000,&quot;estimatedNumRows&quot;:1455256},{&quot;name&quot;:&quot;other&quot;,&quot;status&quot;:&quot;ok&quot;,&quot;numRows&quot;:254000,&quot;estimatedNumRows&quot;:1455256}],&quot;selectedConfig&quot;:&quot;commercial&quot;,&quot;hasSelectedConfigParquet&quot;:true},&quot;splitsData&quot;:{&quot;splitInfos&quot;:[{&quot;name&quot;:&quot;train&quot;,&quot;numRows&quot;:212000,&quot;estimatedNumRows&quot;:4308862}],&quot;selectedSplit&quot;:&quot;train&quot;},&quot;sampleData&quot;:{&quot;dataset&quot;:&quot;TomTBT/pmc_open_access_section&quot;,&quot;config&quot;:&quot;commercial&quot;,&quot;split&quot;:&quot;train&quot;,&quot;capabilities&quot;:{&quot;rows&quot;:true,&quot;search&quot;:true,&quot;filter&quot;:true,&quot;statistics&quot;:true},&quot;navigation&quot;:{&quot;p&quot;:0},&quot;jwt&quot;:&quot;eyJhbGciOiJFZERTQSJ9.eyJyZWFkIjp0cnVlLCJwZXJtaXNzaW9ucyI6eyJyZXBvLmNvbnRlbnQucmVhZCI6dHJ1ZX0sImlhdCI6MTc0MjkyMzA1MSwic3ViIjoiL2RhdGFzZXRzL1RvbVRCVC9wbWNfb3Blbl9hY2Nlc3Nfc2VjdGlvbiIsImV4cCI6MTc0MjkyNjY1MSwiaXNzIjoiaHR0cHM6Ly9odWdnaW5nZmFjZS5jbyJ9.E_hva7Ymc1Dt_2W2flpcwgHbFVP0Hd__eqkzxhJ5YIAZjgrMa7a4jtjocFkUz9lcqbq6hkt2qX-24SzHbnLJCQ&quot;,&quot;sampleData&quot;:{&quot;columns&quot;:[{&quot;name&quot;:&quot;accession_id&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;accession_id&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:9,&quot;max&quot;:11,&quot;mean&quot;:10.23559,&quot;median&quot;:10,&quot;std&quot;:0.45638,&quot;histogram&quot;:{&quot;hist&quot;:[2989,156077,52934],&quot;bin_edges&quot;:[9,10,11,11]}}}},{&quot;name&quot;:&quot;pmid&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;pmid&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:1,&quot;max&quot;:8,&quot;mean&quot;:6.89344,&quot;median&quot;:8,&quot;std&quot;:2.49185,&quot;histogram&quot;:{&quot;hist&quot;:[31888,0,0,4,86,751,9599,169672],&quot;bin_edges&quot;:[1,2,3,4,5,6,7,8,8]}}}},{&quot;name&quot;:&quot;introduction&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;introduction&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:0,&quot;max&quot;:133892,&quot;mean&quot;:2773.22647,&quot;median&quot;:2389,&quot;std&quot;:3542.21362,&quot;histogram&quot;:{&quot;hist&quot;:[209874,1514,409,120,55,20,5,1,0,2],&quot;bin_edges&quot;:[0,13390,26780,40170,53560,66950,80340,93730,107120,120510,133892]}}}},{&quot;name&quot;:&quot;methods&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;methods&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:0,&quot;max&quot;:207902,&quot;mean&quot;:4369.09768,&quot;median&quot;:3410.5,&quot;std&quot;:5133.00816,&quot;histogram&quot;:{&quot;hist&quot;:[209907,1975,105,7,3,2,0,0,0,1],&quot;bin_edges&quot;:[0,20791,41582,62373,83164,103955,124746,145537,166328,187119,207902]}}}},{&quot;name&quot;:&quot;results&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;results&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:0,&quot;max&quot;:357299,&quot;mean&quot;:5167.48439,&quot;median&quot;:2882,&quot;std&quot;:7024.85558,&quot;histogram&quot;:{&quot;hist&quot;:[211158,798,35,3,3,1,0,0,1,1],&quot;bin_edges&quot;:[0,35730,71460,107190,142920,178650,214380,250110,285840,321570,357299]}}}},{&quot;name&quot;:&quot;discussion&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;discussion&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:0,&quot;max&quot;:357299,&quot;mean&quot;:5218.47567,&quot;median&quot;:4659,&quot;std&quot;:5802.93609,&quot;histogram&quot;:{&quot;hist&quot;:[211503,472,20,3,1,0,0,0,0,1],&quot;bin_edges&quot;:[0,35730,71460,107190,142920,178650,214380,250110,285840,321570,357299]}}}},{&quot;name&quot;:&quot;conclusion&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;conclusion&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:0,&quot;max&quot;:58253,&quot;mean&quot;:561.15002,&quot;median&quot;:0,&quot;std&quot;:1155.85125,&quot;histogram&quot;:{&quot;hist&quot;:[210110,1684,181,13,7,1,2,0,0,2],&quot;bin_edges&quot;:[0,5826,11652,17478,23304,29130,34956,40782,46608,52434,58253]}}}},{&quot;name&quot;:&quot;front&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;front&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:0,&quot;max&quot;:30888,&quot;mean&quot;:1479.57205,&quot;median&quot;:1566,&quot;std&quot;:927.16702,&quot;histogram&quot;:{&quot;hist&quot;:[206275,5206,457,60,1,0,0,0,0,1],&quot;bin_edges&quot;:[0,3089,6178,9267,12356,15445,18534,21623,24712,27801,30888]}}}},{&quot;name&quot;:&quot;body&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;body&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:0,&quot;max&quot;:572572,&quot;mean&quot;:3210.69802,&quot;median&quot;:487,&quot;std&quot;:8787.71678,&quot;histogram&quot;:{&quot;hist&quot;:[211199,737,44,10,5,1,1,0,0,3],&quot;bin_edges&quot;:[0,57258,114516,171774,229032,286290,343548,400806,458064,515322,572572]}}}},{&quot;name&quot;:&quot;back&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;back&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:0,&quot;max&quot;:126017,&quot;mean&quot;:564.58408,&quot;median&quot;:234,&quot;std&quot;:1179.12144,&quot;histogram&quot;:{&quot;hist&quot;:[211896,78,9,7,1,4,0,4,0,1],&quot;bin_edges&quot;:[0,12602,25204,37806,50408,63010,75612,88214,100816,113418,126017]}}}},{&quot;name&quot;:&quot;license&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;license&quot;,&quot;column_type&quot;:&quot;string_label&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;no_label_count&quot;:0,&quot;no_label_proportion&quot;:0,&quot;n_unique&quot;:4,&quot;frequencies&quot;:{&quot;CC BY&quot;:197420,&quot;CC0&quot;:14413,&quot;CC BY-SA&quot;:36,&quot;CC BY-ND&quot;:131}}}},{&quot;name&quot;:&quot;retracted&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;retracted&quot;,&quot;column_type&quot;:&quot;string_label&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;no_label_count&quot;:0,&quot;no_label_proportion&quot;:0,&quot;n_unique&quot;:2,&quot;frequencies&quot;:{&quot;no&quot;:211812,&quot;yes&quot;:188}}}},{&quot;name&quot;:&quot;last_updated&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;last_updated&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:19,&quot;max&quot;:19,&quot;mean&quot;:19,&quot;median&quot;:19,&quot;std&quot;:0,&quot;histogram&quot;:{&quot;hist&quot;:[212000],&quot;bin_edges&quot;:[19,19]}}}},{&quot;name&quot;:&quot;citation&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;citation&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:14,&quot;max&quot;:94,&quot;mean&quot;:39.03323,&quot;median&quot;:37,&quot;std&quot;:10.12198,&quot;histogram&quot;:{&quot;hist&quot;:[2018,30117,115531,44534,6767,1246,10629,1053,105],&quot;bin_edges&quot;:[14,23,32,41,50,59,68,77,86,94]}}}},{&quot;name&quot;:&quot;package_file&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;,&quot;statistics&quot;:{&quot;column_name&quot;:&quot;package_file&quot;,&quot;column_type&quot;:&quot;string_text&quot;,&quot;column_statistics&quot;:{&quot;nan_count&quot;:0,&quot;nan_proportion&quot;:0,&quot;min&quot;:0,&quot;max&quot;:35,&quot;mean&quot;:34.23443,&quot;median&quot;:34,&quot;std&quot;:0.49696,&quot;histogram&quot;:{&quot;hist&quot;:[7,0,0,0,0,0,0,0,211993],&quot;bin_edges&quot;:[0,4,8,12,16,20,24,28,32,35]}}}}],&quot;rows&quot;:[{&quot;rowIdx&quot;:0,&quot;cells&quot;:{&quot;accession_id&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;PMC6989633&quot;},&quot;pmid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;31996712&quot;},&quot;introduction&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Introduction\nThe composition of Australia’s theropod fauna is poorly understood in comparison to those of contemporaneous assemblages around the world, due primarily to the isolated and fragmentary mode of preservation in fossiliferous deposits. To date, the majority of documented theropod remains from Australia are from the ‘mid’-Cretaceous (Albian–Cenomanian) and pertain predominantly to megaraptorids 1 – 7 , an exclusively Gondwanan clade of theropods initially interpreted as a member of Allosauroidea 2 . However, recent hypotheses have suggested alternative positions for megaraptorids within Tyrannosauroidea 8 – 11  or close to the base of Coelurosauria 12 , 13 . Despite the preponderance of megaraptorids in ‘mid’-Cretaceous Australia, a diverse high palaeo-latitude (approximately 60 degrees south) theropod fauna has been hypothesised within the upper Barremian–lower Albian deposits on the south coast of Victoria, including megaraptorans 3 , 5 , 14 , ceratosaurs 15 , spinosaurids 16 , tyrannosauroids 3 , 17 , possible unenlagiine dromaeosaurids and indeterminate maniraptoriforms 3 .\nWhile members of Avetheropoda were undoubtedly present during the Cretaceous of Australia, the evidence for Ceratosauria in Australia is presently very limited, despite their abundance in the diverse Patagonian theropod fossil record 8 . The first suggested Australian ceratosaur came not from the better known Cretaceous sites in eastern Australia, but from the Middle Jurassic Colalura Sandstone of Western Australia.  Ozraptor subotaii  was described from a distal tibia characterised by a depressed and subdivided facet for the ascending process of the astragalus 18 . Examination of the tibial fragment failed to identify any convincing similarities with any theropod known at the time, and thus  Ozraptor  was referred to as an indeterminate theropod 17 . Subsequently, the description of abelisauroid remains from the Late Jurassic of Africa included tibiae that also had astragalar articular surfaces similar to that of  Ozraptor . On this basis, it was suggested that the Australian tibia represented a member of Abelisauroidea 19 . This interpretation was maintained in a reassessment of a theropod distal tibia from the Middle Jurassic of England 20 , which concluded that a depressed and subdivided facet for the astragalar ascending process was a synapomorphy of Abelisauroidea. However, this character was subsequently recognised in theropods outside of Abelisauroidea and therefore could not be considered as an abelisauroid synapomorphy 21 . As a consequence there was no convincing evidence to support abelisauroid affinities for  Ozraptor . The current consensus is that  Ozraptor  is too incomplete for referral to any theropod clade 21 , 22 .\nThere has also been suggestion that  Kakuru kujani , known from a partial tibia from the Aptian Marree Formation of South Australia 23  pertains to an abelisauroid based on the presence of a vertical median ridge on the distal tibia 24 . For the reasons stated above, this evidence is insufficient for referral of  Kakuru  to Abelisauroidea; subsequent revisions of this material concluded that  Kakuru  could only be referred to an indeterminate position within either Averostra or Tetanurae 25 , 26 .\nMore recently, a left astragalocalcaneum from the upper Barremian–lower Albian San Remo Member of the upper Strzelecki Group on the south coast of Victoria was described (Museum Victoria, Melbourne, Australia; NMV P221202, Fig.  1 ) and referred to Ceratosauria, based among other features on the co-ossification of the astragalus and calcaneum, a parallel-sided base of the ascending process of the astragalus, and a fossa at the base of the ascending process that is not associated with a transverse groove 15 . However, it was subsequently suggested that the evidence for referral of NMV P221202 to Ceratosauria was weak, and that it could only be considered as an indeterminate averostran at best 8 .\nHere, we present new evidence for the presence of ceratosaurian theropods from the Cenomanian Griman Creek Formation of Lightning Ridge, New South Wales. We also reappraise the evidence against the ceratosaurian interpretation of the specimen NMV P221202 8  with the objective of clarifying and elucidating its phylogenetic position.&quot;},&quot;methods&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Methods and Materials\nLRF 3050.AR and NMV P221202 were inserted into a recently published ceratosaurian phylogenetic matrix 39  (see Supplementary Dataset  S1 ) and analysed with equal weights parsimony in TNT 1.5 93 . A driven search strategy was implemented to calculate optimal trees, with each search using 100 replicates of random sectorial searches, each with 30 rounds of drifting, 5 rounds of tree fusing and 50 ratcheting cycles. The analysis was halted after two such successive searches returned shortest trees of the same length.&quot;},&quot;results&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;&quot;},&quot;discussion&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Discussion\nComparisons of LRF 3050.AR\nOpisthocoelous vertebral centra characterise the cervical series of many neotheropods. The posterior surfaces are typically moderately to strongly concave and the anterior surface may be generally flattened 49 , 50  or slightly convex as in ceratosaurians 51 – 54  and basal tetanurans 55 , or form a well-defined projection as in abelisaurids 30 , 56 , megalosauroids 57 – 59 , allosauroids 49 , 60 , 61 , megaraptorids 9 , 62  and alvarezsaurids 63 – 65 . In addition, opisthocoely continues into the anterior dorsal series in megalosauroids 57 , allosauroids 66 , 61 , megaraptorids 62 , and alvarezsaurids 63 . This differs from the condition in  Dilophosaurus wetherilli  and abelisauroids in which the anterior cervical centra are typically weakly opisthocoelous and transition along the series to amphicoelous in the most posterior cervicals and anterior dorsals 50 – 52 , 54 , 67 , 68 . All preserved mid-posterior cervical centra of  Elaphrosaurus  are amphicoelous 41 . Following these observations, the amphicoelous centrum and reduced inclination of the articular surfaces of LRF 3050.AR indicates a placement in the middle or posterior region of the neck. The distortion of the centrum, in particular the exaggerated offset of the articular surfaces resulting from taphonomic compression, precludes a more accurate placement of the centrum.\nAmong ceratosaurs, the dimensions of LRF 3050.AR are most similar to the anterior cervical series of the abelisaurid  Viavenator exxoni . However, as noted above, the anterior cervical series in  Viavenator  and other abelisaurids consists of opisthocoelous centra, contrary to the amphicoelous condition in LRF 3050.AR. Unfortunately, direct comparisons of the centrum proportions of LRF 3050.AR are complicated by the strong taphonomic dorsoventral compression of the specimen. However, when the anterior half of the cervical centra are excluded, the dimensions of LRF 3050.AR are more similar to the moderately elongate proportions of noasaurids 41 , 68 , 69  than the more robust and anteroposteriorly shortened centra in abelisaurids 51 , 52  or strongly elongate centra in  Elaphrosaurus 41 . The anterior and posterior articular surfaces are considerably wider mediolaterally than dorsoventrally tall (Table  1 ). This is similar to the proportions throughout the cervical series of  Masiakasaurus   knopfleri  and  Elaphrosaurus 41 , 67 , 68 , but may have been exaggerated by taphonomic distortion.\nThe preserved floor of the neural canal on the dorsal surface of LRF 3050.AR indicates that it was relatively wide mediolaterally relative to the width of the centrum and were likely wider than the thickness of the walls of the laterally bounding neural arch pedicels (Fig.  2 ). The neural canals in the cervicals of basal neotheropods and most ceratosaurs are narrower with respect to both the centrum and the neural arch pedicels 48 , 50 , 52 , 68 . In contrast, the neural canals of  Elaphrosaurus 41  and noasaurids 53 , 68 , 69  are considerably wider relative to the centrum and wider than the thickness of the walls of the neural arch pedicels, as seen in LRF 3050.AR.\nThe distinct posterior centrodiapophyseal lamina (pcdl) of LRF 3050.AR is remarkably similar to those of noasaurids (Fig.  3 ). In MACN-PV (Museo Argentino de Ciencias Naturales “Bernardino Rivadavia”, Buenos Aires, Argentina) 622, a cervical vertebra initially described as an oviraptorosaur 70 , 71  but which most likely pertains to  Noasaurus 53 , the pcdl narrows abruptly from the anteriorly placed diapophyses and contacts the centrum at approximately the anteroposterior midpoint (Fig.  3 ). A similar pcdl also appears to have been present in GSI (Geological Survey of India, Kolkata, India) K20/614, a cervical vertebra ascribed to the Indian noasaurid  Laevisuchus indicus 72 . The plesiomorphic condition of a posteriorly contacting pcdl is present in the middle cervicals of  Dilophosaurus 50 , abelisauroids 30 , 34 , 54  and also the recently described Brazilian noasaurid  Vespersaurus paranaensis 69 . Despite the loss of the posterior portion of the posterior centrodiapophyseal lamina, a medial attachment of the pcdl is most likely to have been present in LRF 3050.AR. A medially positioned pcdl also characterises the middle to posterior cervical series of other ceratosaurs, including  Elaphrosaurus 41 ,  Majungasaurus crenatissimus 52  and  Carnotaurus sastrei 51 .\nPerhaps the most distinguishing feature of LRF 3050.AR is the mediolaterally concave ventral surface of the centrum delimited by pronounced ventrolateral ridges. In most ceratosaurs, the ventral surface of the cervical centra is flattened or slightly convex, forming a distinct edge at the contact with the lateral surfaces 51 , 68 , 73 . Ventrolateral ridges on cervical centra such as those present in LRF 3050.AR have been reported only in the basal ceratosaurian  Elaphrosaurus  and the noasaurid  Noasaurus 41 , 53 . In  Elaphrosaurus , the sharp lateroventrally directed ridges are present only at the posterior part of the centrum 41 , which differs from the condition in LRF 3050.AR in which they are continuous with the parapophysis and extend along almost the entire length of the centrum. Similar ventrolateral ridges have also been reported in MACN-PV 622 53 . Ventrolateral ridges have been described in therizinosaurs and unenlagiine dromaeosaurids 74 – 76 ; however, they are developed only as comparatively weaker and rounded ridges that do not form the sharp edges that are seen in ceratosaurians. In addition, in unenlagiines the ventrolateral ridges transition into well-developed carotid processes at the anterior end of the centra 76 , 77 . This contrasts with the condition in LRF 3050.AR in which carotid processes are absent and the ridges remain sharply defined and contact the parapophyses at the anteroventral margins of the anterior articular surface.\nStatus of NMV P221202\nA ceratosaurian astragalocalcaneum (NMV P221202) was discovered from the upper Barremian–lower Aptian San Remo Member of the upper Strzelecki group in Victoria 15  (Fig.  4 ). NMV P221202 was compared to the only Australian theropod astragali known at the time, namely those of the megaraptorid  Australovenator wintonensis 1  and the Australian pygmy ‘ Allosaurus ’ 78 , now considered to also pertain to Megaraptoridae 2 , 8 . The Victorian astragalocalcaneum, NMV P221202, was found to differ from the two Australian megaraptorid astragali, most notably in the co-ossification of the astragalus and calcaneum, the absence of a horizontal vascular groove on the anterior surface of the astragalar body, and the lack of a crescentic groove on the posterior surface of the ascending process 15 . NMV P221202 was referred to Ceratosauria in a phylogenetic analysis, but possible ingroup relationships were not considered with confidence despite similarities with the astragalus of the Madagascan noasaurid  Masiakasaurus 15 .\nSubsequently, the assignment of NMV P221202 to Ceratosauria was questioned 8  on the basis of five observations: the presence of a distinct eminence on the medial surface of the ascending process and paired oval fossae at the base of the ascending process of the astragalus anteriorly (Fig.  4a ), both of which are present in alvarezsaurids 79 ; a vertical groove on the posterior surface of the ascending process and a lateral constriction of the tibial facet caused by a thickening of the ascending process laterally (Fig.  4c ), both of which are present in megaraptorids; and a prominent posterodorsal notch on the calcaneum for articulation of the tibia (Fig.  4b ), which they considered to be a tetanuran synapomorphy based on the results of a phylogenetic analysis of tetanurans 80 . Based on these observations, it was concluded that NMV P221202 could only be considered an indeterminate averostran 8 . The debate surrounding the affinities of NMV P221202 was commented on briefly in a review of the Victorian Cretaceous polar biota 81 , with no preference stated for either of the two hypotheses.\nHowever, a detailed consideration of these arguments as presented raises a number of problems. Firstly, as previously noted 8 , the ascending process of the astragalus in alvarezsaurids differs markedly from the condition present in NMV P221202. As is typical for coelurosaurs, the base of the ascending process in alvarezsaurids occupies almost the entire width of the astragalus 63 , 79 . Furthermore, in alvarezsaurids with the exception of  Patagonykus puertai , the medial surface of the ascending process is excavated by a deep notch, leaving only a low medial portion of the ascending process and a taller narrow lateral portion 63 , 65 , 82 – 84 . However, in NMV P221202 the ascending process is parallel-sided at the base, was likely subrectangular in its original form, and its base spans only the lateral two-thirds of the astragalus. In addition, contrary to previous remarks 8 , no medial eminence of the ascending process that resembles that of NMV P221202 is present in either  Patagonykus  or  Mononykus olecranus . In the former taxon, the medial edge of the ascending process is smoothly sinusoidal in anterior view with no noticeable eminences 79 , whereas the medial edges of the medially-notched ascending processes of  Mononykus  and other alvarezsaurids are straight or slightly concave, with no noticeable eminences 63 , 65 . Secondly, as noted in the original description of NMV P221202 15 , and contrary to previous observations 8 , there is no groove on the posterior surface of the ascending process similar to those that have been reported in megaraptorids. The lateral edge of the posterior surface of the base of the ascending process in NMV P221202 is slightly elevated with respect to the area immediately lateral to an abraded area of periosteum that may have given the appearance of a grooved surface. However, this is markedly different from the well-defined crescentic groove present on the posterior surface of the ascending process in megaraptorid astragali 1 , 14 , 78 . Thirdly, the lateral side of the tibial facet of the astragalus in the abelisaurid  Majungasaurus  is also constricted relative to the medial side 85 , indicating that this feature is not restricted to megaraptorids as previously asserted 8  and that abelisauroid affinities cannot be dismissed. Finally, tibial facets on the calcaneum have been observed in  Dilophosaurus ,  Majungasaurus ,  Elaphrosaurus ,  Ceratosaurus  and  Masiakasaurus 41 , 67 , 85 , 86 , indicating that this feature is diagnostic of Averostra, a more inclusive group than stated previously 8 .\nPhylogenetic analysis\nThe phylogenetic analysis including LRF 3050.AR and NMV P221202 (see Methods and Materials for details) returned 217 most parsimonious trees of 4293 steps (CI: 0.306, RI: 0.512). The strict consensus tree resolves both Australian specimens within Noasauridae (Fig.  5 ). The synapomorphies diagnosing Noasauridae include a spur on the medial surface of the ascending process of the astragalus (858:1), mediolaterally thin cervical epipophyses (1272:1), cervical postzygapophyses swept back posteriorly and surpassing the posterior end of the vertebral centra (1083:1), smooth medial surfaces of the anteromedial process of the maxilla (915:0), anteroposteriorly shortened palatal shelves of the maxilla (1310:1), paradental plates of the maxilla low and partially obscured by lamina of maxilla (972:1) and shaft of metatarsal II mediolaterally compressed (1208:1). The presence of ventrolateral ridges contacting the parapophyses on the cervical vertebrae (210:1) may represent an additional synapomorphy of Noasauridae. However, the distribution of this character is presently uncertain and so far has only been reported in MACN-PV 622 (cf.  Noasaurus ), in addition to LRF 3050.AR. The noasaurid with the most complete cervical series,  Masiakasaurus , has flattened ventral surfaces of the centra with no ventrolateral ridges 41 . When  Masiakasaurus  is coded as such for the aforementioned character, the presence of ventrolateral ridges does not optimise as a synapomorphy of Noasauridae. However, this may be an artifact of the long-standing lack of resolution among noasaurids due to their poor fossil record, and it remains plausible that ventrolateral ridges may represent a synapomorphy of a subclade within Noasauridae. However, more data is needed to thoroughly test this hypothesis.\nThe presence of a medial eminence on the ascending process is a synapomorphy that pertains directly to NMV P221202. Among theropods, this feature is shared only with  Masiakasaurus 68  and represents the strongest evidence in favour of noasaurid affinities for NMV P221202. Unfortunately, the lack of preserved ascending processes in the astragali of other noasaurid taxa precludes detailed comparisons.\nIf the results presented here are correct, then NMV P221202 and LRF 3050.AR represent novel reports of noasaurids from the late Barremian–early Aptian of Victoria and Cenomanian of New South Wales respectively. Under the taxonomic framework presented here, Noasauridae consists of at least six named taxa:  Laevisuchus ,  Noasaurus  and  Masiakasaurus  from the Maastrichtian of India, Argentina, and Madagascar respectively 44 , 67 , 87 ;  Velocisaurus , from the Santonian of Argentina 88 ;  Vespersaurus  from the Aptian–Campanian of Brazil 69  and  Afromimus tenerensis  from the Aptian–Albian of Niger, initially described as an ornithomimid 89  but recently reappraised as a probable noasaurid 90 .  Genusaurus sisteronis , from the Albian of France, has previously been considered as a noasaurid 22 , but subsequent analyses, including the one presented here, preferred a position within Abelisauridae.  Ligabueino andesi , from the Barremian–early Aptian of Argentina 91 , was also originally described as a noasaurid, but phylogenetic studies failed to identify any noasaurid synapomorphies in this taxon 22 , 68 . NMV P221202, which is identified by phylogenetic analysis as a noasaurid, therefore represents the oldest known representative of the clade in the world to date (Fig.  5 ). However, if the broader taxonomic scope of Noasauridae (i.e., inclusive of elaphrosaurines; see Taxonomic Framework) is favoured instead, then NMV P221202 would instead represent the oldest known noasaurine, with the oldest noasaurids represented by the Middle–Late Jurassic aged elaphrosaurines 33 , 41 , 92 . Regardless of their phylogenetic position, the newly described Australian noasaurids expands the known palaeogeographic range of the clade outside of South America, Madagascar and India. Presently, the poor fossil record of Noasauridae, and the corresponding lack of resolution among the known noasaurid taxa, precludes the formation of any novel palaeobiogeographic hypotheses including the newly discovered Australian record of noasaurid theropods. Future discoveries may reveal more detail about the evolution and palaeobiogeographic distribution of this enigmatic clade.&quot;},&quot;conclusion&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;&quot;},&quot;front&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;The diversity of Australia’s theropod fauna from the ‘mid’-Cretaceous (Albian–Cenomanian) is distinctly biased towards the medium-sized megaraptorids, despite the preponderance of abelisauroids in the younger but latitudinally equivalent Patagonian theropod fauna. Here, we present new evidence for the presence of ceratosaurian, and specifically abelisauroid, theropods from the Cenomanian Griman Creek Formation of Lightning Ridge, New South Wales. A partial cervical vertebra is described that bears a mediolaterally concave ventral surface of the centrum delimited by sharp ventrolateral ridges that contact the parapophyses. Among theropods, this feature has been reported only in a cervical vertebra attributed to the noasaurid  Noasaurus . We also reappraise evidence recently cited against the ceratosaurian interpretation of a recently described astragalocalcaneum from the upper Barremian–lower Aptian San Remo Member of the upper Strzelecki Group in Victoria. Inclusion of the Lightning Ridge cervical vertebra and Victorian astragalocalcaneum into a revised phylogenetic analysis focused on elucidating ceratosaurian affinities reveals support for placement of both specimens within Noasauridae, which among other characters is diagnosed by the presence of a medial eminence on the ascending process of the astragalus. The Lightning Ridge and Victorian specimens simultaneously represent the first noasaurids reported from Australia and the astragalocalcaneum is considered the earliest known example of a noasaurid in the world to date. The recognition of Australian noasaurids further indicates a more widespread Gondwanan distribution of the clade outside of South America, Madagascar and India consistent with the timing of the fragmentation of the supercontinent.\nSubject terms&quot;},&quot;body&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Taxonomic Framework\nThere are presently two hypotheses regarding the content of Noasauridae and the phylogeny of non-abelisaurid, non-ceratosaurid ceratosaurians. Abelisauroidea was originally considered to include Abelisauridae and Noasauridae, and all ceratosaurs more closely related to them than to  Ceratosaurus nasicornis 27 . The earliest phylogenetic analysis of ceratosaurs identified a monophyletic Abelisauroidea following this definition 28 , and which was subsequently expanded to include the African  Elaphrosaurus bambergi 29 . Subsequent phylogenetic studies expanded the taxonomic scope of Noasauridae to include small-bodied Late Cretaceous taxa from South America 21 , 30 – 32  and the Jurassic and Cretaceous of Africa 33 , to the exclusion of  Elaphrosaurus . This topology has been widely recovered in more recent analyses 21 , 34 – 39 . However,  Elaphrosaurus  has also been resolved within Noasauridae in other analyses 40 , most notably in the analysis accompanying the recent redescription of the holotype 41 . Under this hypothesis, the subclade Noasaurinae was coined to include ceratosaurs more closely related to  Noasaurus leali  than to  Elaphrosaurus ,  Ceratosaurus  and  Allosaurus fragilis , and Elaphrosaurinae was erected to include ceratosaurs more closely related to  Elaphrosaurus  than to  Noasaurus ,  Abelisaurus comahuensis ,  Ceratosaurus  and  Allosaurus 41 . The results of a revised phylogenetic analysis for  Limusaurus inextricabilis 42  were used to support a recently proposed phylogenetic framework for Ceratosauria 43  in which Noasaurinae and Elaphrosaurinae were recovered as subclades of Noasauridae. In line with the topology of our phylogenetic tree (see Phylogenetic Analysis), the following descriptions and discussions consider Noasauridae to have the same taxonomic content as Noasaurinae 41 , with members of Elaphrosaurinae representing ceratosaurs basal to Abelisauroidea (i.e., Noasauridae + Abelisauridae).\nSystematic Palaeontology\nTheropoda Marsh 1881\nNeotheropoda Bakker 1986\nAverostra Paul 2002\nCeratosauria Marsh 1884\nNoasauridae indet. Bonaparte and Powell 1980 44\nLRF 3050.AR\nLocality\nLRF (Australian Opal Centre, Lightning Ridge, New South Wales, Australia) 3050.AR was collected from an underground opal mine at the ‘Sheepyard’ opal field, approximately 40 km southwest of Lightning Ridge in central northern New South Wales (Fig.  1 ). The specimen derives from the Wallangulla Sandstone Member 45  of the Griman Creek Formation. Radiometric dates for the Wallangulla Sandstone Member at Lightning Ridge indicate a maximum depositional age of 100.2–96.6 Ma 46 . LRF 3050.AR was found within a monodominant bonebed of the iguanodontian  Fostoria dhimbangunmal 47 . Other faunal components from this accumulation include isolated unionid bivalves (LRF 3051), a testudine caudal vertebra (LRF 3053), a small ornithopod caudal centrum (LRF 3052), and a possible indeterminate theropod ulna (LRF 3054). A complete discussion of the geological setting, sedimentology, age and faunal diversity of the Griman Creek Formation is presented elsewhere 46 .\nDescription\nLRF 3050.AR has been taphonomically altered by erosion, breakage and through preparation. The centrum is markedly flattened dorsoventrally through taphonomic compaction, such that much of the left lateral surface is visible in ventral view. In addition, the dorsal portion of the centrum has been sheared off obliquely. Notwithstanding the dorsoventral compression, the centrum is hourglass-shaped in dorsal-ventral view; the narrowest point occurs approximately one-third of the length from the anterior articular surface (Fig.  2a,b ). In lateral view, the anterior and posterior articular surfaces are oriented obliquely relative to the long axis of the centrum (approximately 20 degrees from vertical; Fig.  2c,d ); however, this appearance is probably a result of the taphonomic compaction and not indicative of their original orientations. The ventral surface of the centrum is markedly concave in lateral view (Fig.  2c,d ). The centrum is slightly more than twice as long anteroposteriorly relative to the width of the posterior articular surface (Table  1 ). The centrum is amphicoelous. The central region of the anterior articular surface is flattened and surrounded laterally and ventrally by a concave rim (Fig.  2e ), whereas the centre of the posterior articular surface is concave and bordered ventrally by a convex rim (Fig.  2f ). The preserved portion of the anterior articular surface is elliptical in anterior view, wider mediolaterally than dorsoventrally tall (Fig.  2e ). Only the ventralmost portion of the left parapophysis is present on the ventrolateral edge of the centrum anteriorly, and which also projects ventrolaterally (Fig.  2a,c,e ). A region of exposed trabecular bone immediately dorsal to the preserved parapophysis indicates the likely size of its attachment to the centrum (Fig.  2c ). An anteroposteriorly oriented lamina is present anterodorsally, extending from the anterior articular surface to approximately one third of the length of the centrum and overhanging the right lateral surface (Fig.  2a,b ). The posterior edge of the lamina is broken, indicating that it likely continued further posteriorly. On the ventromedial surface of this lamina is the eroded remains of a smaller, vertically oriented lamina (Fig.  2d ). The position of this smaller lamina would have been dorsal to the parapophysis, and its vertical and lateral continuation indicates that it would have contacted the diapophysis ventrally. Therefore, this lamina is interpreted as a paradiapophyseal lamina (ppdl; following nomenclature for vertebral laminae of Wilson 48 ). Consequently, the portion of the larger lamina anterior to the ppdl is interpreted as the anterior centrodiapophyseal lamina (acdl), and the posterior portion is interpreted as the remains of the posterior centrodiapophyseal lamina (pcdl). The posterior articular surface is missing the dorsal portion due to erosion, similar to the anterior end, and is elliptical, having a greater mediolateral width than dorsoventral height (Fig.  2f ). A portion of the floor of the neural canal is preserved across the anterior half of the dorsal surface of the centrum (Fig.  2a ). Despite erosion to the dorsal surface of the centrum, the neural canal appears to have been mediolaterally wide, approximately half that of the centrum itself, and considerably wider than the neural arch pedicels as visible from their eroded bases (Fig.  2b ). The ventral surface of the centrum is concave mediolaterally and delimited by well-defined, subparallel ventrolateral ridges that extend as laminae from the parapophyses along nearly the entire length of the centrum, becoming less distinct posteriorly (Fig.  2b ). Two small (~3 mm long) lenticular foramina are present on the posterior half of the centrum (Fig.  2a ). Whether these foramina are pneumatic in origin cannot be determined.\nSupplementary information\n   &quot;},&quot;back&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Supplementary information\nis available for this paper at 10.1038/s41598-020-57667-7.\nAcknowledgements\nWe are indebted to Robert Foster who discovered the ‘Sheepyard’ specimens and Joanne Foster and Gregory Robert Foster who generously donated the specimens under the Australian Government’s Cultural Gifts program. We thank Jenni Brammall, Manager of the Australian Opal Centre, for allowing access to LRF 3050.AR and providing resources to facilitate their study while in Lightning Ridge, and Tim Ziegler of Museum Victoria for making NMV P221202 available for study. TNT is made freely available thanks to a subsidy from the Willi Hennig Society. We thank Stephen Poropat and two anonymous reviewers for their valuable comments that improved the quality of the manuscript. We acknowledge the Yuwaalaraay, Yuwaalayaay and Gamilaraay custodians of country in the Lightning Ridge district, and pay our respects to Elders past and present. This work was supported by an Australian Research Council Discovery Early Career Researcher Award (project ID: DE170101325) to P.R.B.\nAuthor contributions\nS.A.B. designed the research, performed the descriptive and comparative studies, analysed data, prepared figures and performed the phylogenetic analysis; E.S. and P.B. contributed specimen photographs and data; S.A.B., E.S. and P.B. wrote the paper.\nData availability\nAll data generated or analysed during this study are included in this published article (and its Supplementary Information).\nCompeting interests\nThe authors declare no competing interests.&quot;},&quot;license&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;CC BY&quot;},&quot;retracted&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;no&quot;},&quot;last_updated&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;2024-01-16 23:34:59&quot;},&quot;citation&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Sci Rep. 2020 Jan 29; 10:1428&quot;},&quot;package_file&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;oa_package/16/47/PMC6989633.tar.gz&quot;}}},{&quot;rowIdx&quot;:1,&quot;cells&quot;:{&quot;accession_id&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;PMC7168315&quot;},&quot;pmid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;32059467&quot;},&quot;introduction&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;1. Introduction\nBiofilms—aggregations of densely packed microbial cells embedded inside exopolysaccharide (EPS) matrix—are a major challenge in public health management. The EPS matrix provides a protective barrier to the biofilm, making them recalcitrant to antimicrobial agents and host defenses [ 1 ].  Staphylococcus aureus  is thought to be one of the major causes of nosocomial infections, globally. While the planktonic counterpart is limited to bacteremia and skin abscesses, more chronic infections such as cystic fibrosis, osteomyelitis, and endocarditis are associated with its biofilm mode of growth [ 2 , 3 ]. To eradicate this problem, many strategies were proposed, including (i) the inhibition of primary bacterial adhesion or attachment to the living or non-living surfaces; (ii) the disruption of biofilm architecture during maturation processes; and (iii) the inhibition of cell to cell communication—i.e., quorum sensing [ 4 , 5 ]. Chrysin (5, 7-dihydroxyflavone), a flavone constituent of the  Orocylumineicum vent  has already been documented for its anticancer, antioxidant and antibacterial properties [ 6 , 7 ]. In spite of its many biological activities, its low water solubility, poor biosorption in the intestinal lumen, low bioavailability and rapid metabolism in the body limits its therapeutic applications [ 8 ]. In this regard, a reduction in particle size may serve as a potential means for enhancing the solubility and dissolution of chrysin [ 9 ].\nNanocarriers have the ability to inhibit bacterial growth and biofilm formation, and are increasingly being used as an attractive tool to combat chronic infections [ 10 ]. They aid in increasing the efficacy of the drug by acting as a protective barrier against enzymatic hydrolysis, increase the biosorption efficacy of the drug in the intestinal lumen, increase solubility and also cause sustainable release [ 11 , 12 ]. In recent years, chitosan is widely being used as a nano-carrier due to its non-toxicity, biocompatibility, immunostimulating and mucoadhesive properties [ 13 ]. Chitosan is a cationic heteropolysaccharide composed of the β-(1,4) linked repeating unit of glucosamine (GlcN) and N-acetylglucosamine (GlcNAc), extracted by the partial alkaline N-deacetylation of chitin found in the exoskeleton of crustaceans [ 14 , 15 ]. The antimicrobial and anti-biofilm potential of chitosan and its nano-derivatives were reported against various microorganisms such as  Listeria monocytogenes ,  Bacillus cereus ,  Enterococcus faecalis,  etc. [ 1 , 16 ]. Chrysin-encapsulated chitosan nanoparticles (CCNPs), synthesized using the ionic gelation method, were characterized and evaluated for their anti-biofilm activity against  S. aureus .&quot;},&quot;methods&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;2. Materials and Methods\n2.1. Materials\nChitosan (75–85% deacetylated), sodium tripolyphosphate (TPP) and Chrysin were procured from Sigma-Aldrich. The test stain,  S. aureus  (MCC 2408) was purchased from Microbial Culture Collection (MCC), Pune, India.\n2.2. Synthesis of Chrysin-Loaded Chitosan Nanoparticles\nMedium molecular weight chitosan (0.2%,  w / v ) was mixed with an aqueous solution of acetic acid (0.1%,  v / v ) and incubated over-night at 60 °C with continuous agitation. A stock of 5 mg/mL of chrysin dissolved in DMSO was used to prepare the nanoparticle formulation. An aliquot of chrysin prepared in DMSO was added to the chitosan solution (pH 4.8). Subsequently, 40 mL of TPP solution (0.2%,  w / v ) was dispensed dropwise into the chitosan–chrysin solution and kept under continuous agitation at 1000×  g  for 30 min. The ratio of the chitosan-TPP was maintained at 5:1 [ 13 ] with the final chrysin concentration of 50 μg/mL. The nanoparticles formed were concentrated by centrifuging the suspension for 20 min at 12,000×  g , washed with MilliQ water to remove the unbound chrysin and dried at room temperature for further studies [ 17 ].\n2.3. Physical Characterization of Nanoparticles\nThe nanoparticles (NPs) were subjected to dynamic light scattering (DLS) to determine the mean hydrodynamic diameter (MHD) and polydispersity index (PDI). The FTIR spectrum was recorded in the range of 4000–500 cm −1 . A transmission electron microscope (TEM) was used to determine the morphology and size of the CCNPs [ 18 ].\n2.4. Determination of the Loading Efficiency and Drug Release of Chrysin-Loaded Chitosan NPs\nThe amount of chrysin loaded in the nanoparticles was determined using a UV-Vis spectrophotometer. After the collection of NPs from the reaction mixture, the absorbance of the supernatant was recorded at 348 nm and the concentration of unbound chrysin was estimated based on the standard curve of chrysin [ 13 , 19 ]. The CCNPs were dislodged in a medium constituting PBS and DMSO (co-solvent, 1%,  v / v ) and incubated with gentle agitation (100×  g ) at 37 °C. Two milliliters of the sample was retrieved at regular intervals, centrifuged at 10,000×  g  and the absorbance of the supernatant was recorded at 348 nm. The cumulative chrysin released in the medium was determined at the every 2 h interval, with reference to the standard curve of chrysin [ 20 ]. Chrysin release (%) = (Chrysin released in the supernatant/loaded chrysin concentration) × 100.\n2.5. Determination Sub-Minimum Inhibitory Concentration (Sub-MIC) of Chrysin-Loaded Chitosan NPs\nThe minimum inhibitory concentration (MIC) of CCNPs was determined using macro-broth dilution assay (Clinical and Laboratory Standards Institute Guidelines, 2006). Two-fold dilutions of the NPs were prepared in Muller-Hinton broth to achieve a final concentration ranging from 8 μg/mL to 1024 μg/mL. An overnight culture of  S. aureus  (100 μL) was added in each NPs suspension and incubated at 37 °C for 24 h. The test tubes were observed for visible signs of growth and the spectrophotometric readings were recorded at 600 nm [ 12 ].\n2.6. In Vitro Anti-Biofilm Assays of Chrysin-Loaded Chitosan NPs\nThe effect of sub-MIC of CCNPs of the on the biofilm formation of  S. aureus  was evaluated relative to chitosan NPs (CNPs) and Chrysin.\n2.6.1. Microtiter Plate (MTP) Assay for Biofilm Disruption and Inhibition\nMTP assay for biofilm disruption and inhibition was performed according to Mu et al. [ 1 ]. An over-night culture of  S. aureus  (100 μL) was transferred into the wells of 96-well flat-bottomed polystyrene plates. After incubation at 37 °C for 24 h, the wells were washed with 100 μL of 0.9% ( w / v ) NaCl to remove the unadhered cells. The biofilm formed was further incubated after adding 90 μL tryptone soy broth (TSB) supplemented with the sub-MIC concentration of CCNPs for another 24 h. The biofilm attached at the bottom of each well was fixed with 100 μL of absolute methanol for 15 min and subsequently treated with 100 μL of crystal violet (0.2%  w / v ). Control samples were maintained with  S. aureus  culture alone. Cultures of  S. aureus  treated with DMSO serve as a control. The dye attached to the biofilm was further solubilized in 150 μL of glacial acetic acid and the optical density was recorded at 595 nm.  S. aureus  culture (90 μL) grown in TSB was seeded into individual wells of microtiter plates in the presence of sub-MIC of CCNPs and incubated at 37 °C for 24 h. The planktonic cells were discarded and the MTP was stained with crystal violet (0.2%  w / v ). The inhibition of biofilm formation was determined by solubilizing the CV attached to the biofilm and measuring the optical density at 595 nm. Biofilm inhibition/disruption was quantified using the following formula:\n% Biofilm inhibition/disruption= ([OD 595  of control − OD 595  of test] = OD 595  of control) * 100\n2.6.2. Microscopic Examination of Biofilm\nReduction in the biofilms of  S. aureus  was observed using confocal laser scanning microscopy (CLSM).  S. aureus  biofilms was allowed to grow on glass coverslips (18 × 18 mm) placed in 12 well polystyrene plates containing TSB supplemented with CCNPs (sub-MIC) and incubated overnight at 37 °C. The glass coverslip was washed with sterile distilled water, stained and processed accordingly [ 21 ]. The biofilm formed was fixed using methanol and treated with crystal violet (0.2%  w / v ). The coverslip was subsequently washed, air dried and observed using a light microscope (40×). The biofilm formed on the coverslip was washed 0.01 M phosphate buffer saline (PBS), stained using acridine orange (0.2%  w / v ) for 1 min and observed using Confocal laser microscope at 20×. The 3D image was recorded and Z stacks were prepared to determine the effect of the CCNPs on the thickness of the biofilm [ 22 ].\n2.6.3. Exopolysaccharide (EPS) Quantification and Microbial Adhesion to Hydrocarbon (MATH) Assay\nProduction of EPS by  S. aureus  was quantified in presence and absence of CCNPs by total carbohydrate quantification method.  S. aureus  was grown in presence and absences of CCNPs were harvested by centrifugation (10,000×  g  for 2 min). The cell pellet was washed and suspended in 200 μL of sterile PBS to which an equal volume of 5% ( v / v ) phenol and 5× volume of concentrated sulfuric acid containing 0.2% ( w / v ) hydrazine sulphate was added. The tubes were incubated in dark for 1 h followed by centrifugation at 10,000×  g  for 10 min. The supernatant was aspirated and the optical density was measured at 490 nm [ 21 ]. A reduction in EPS production was quantified using the following formula:\nThe effect of CCNPs on the cell surface hydrophobicity of  S. aureus  was evaluated using MATH assay. The optical density of the treated CCNPs and the untreated cell suspension was recorded at 600 nm, after 24 h of incubation. The bacterial suspension was mixed with toluene (1 mL) and vortexed for 2 min. The optical density of the aqueous phase was measured at 600 nm. In both the assays, control samples were maintained with  S. aureus  culture only. Cultures of  S. aureus  treated with DMSO served as a negative control [ 21 ]. The percentage of inhibition in hydrophobicity is measured as follows; \n2.6.4. Growth Curve Analysis \nAn overnight culture of  S. aureus  was diluted with LB medium until the optical density of the cell suspension reaches 0.05 at 600 nm. The suspension was then supplemented with chrysin and NPs separately and incubated overnight at 37 °C at 100×  g . The cell suspension (1 mL) was withdrawn and the optical density was measured at 600 nm, at every 2 h interval [ 23 ]. \n2.7. Statistical Analysis\nAll the assays were repeated thrice, and the data are presented as mean ± standard error. Significance among treatments were investigated using one-way ANOVA and represented with a statistical significance of  p  ≤ 0.05. Significance in treatments of chrysin, chitosan and CCNPs are represented with asterisk sign. Non-significant groups are represented by NS.&quot;},&quot;results&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;3. Results\n3.1. Synthesis and Characterization of Chrysin-Loaded Chitosan NPs\nThe CCNPs were synthesized using the ionotropic gelation method, using TPP molecules as a linker. The ratio of chitosan and TPP used is one of the factors that influence the aggregation of nanoparticles. Chitosan/TPP in the ratio of 5:1 was found to be the best formulation for the synthesis of CCNPs. The mean hydrodynamic diameter of the synthesized CNPs and CCNPs were found to be ~299 nm and ~355 nm, respectively. CNP and CCNP nanoparticles showed an intermediate polydisperisty index of 0.434 and 0.487, respectively ( Figure 1 a,b). However, the CCNPs were found to be spherical with sizes ranging from 130–341 nm as indicated by the TEM micrograph ( Figure 1 c). On comparing the functional groups present in the CCNPs with their bulk counterparts, chitosan showed characteristic peaks at 3418 and 3238 cm −1 , which indicated the O-H stretching and N–H stretching vibration, the characteristic peak at 2908 cm −1  depicted the C–H stretch, 2342 cm −1  (C–N band stretching), 1610 cm −1  (amide II band), 1024 and 1051 cm −1  indicated that the CH 2  group and C–O stretch from glucosamine residue was observed. Chrysin showed characteristic bands at 2625 cm −1 , 2343 cm −1  indicating O–H stretching vibration and intramolecular H-bond ( Figure 1 d). The characteristic peaks of both chitosan (at 2342 cm −1 , 1051 cm −1 ) and chrysin (at 2625 cm −1  and 2343 cm −1 ) were observed in the CCNPs. A slight band shift was also observed at 3347 and 3200 cm −1  to the lower wave-number that indicated the presence of hydrogen bonding between O-H group of chrysin and O-H or -NH 2  group of chitosan [ 24 ]. The other peaks observed in the loaded nanoparticles were the P–O bending peak at 890 cm −1  and at 2625 cm −1 , which was broader compared to pure chrysin, indicating an increase in hydrogen bond interactions [ 13 ].\n3.2. Loading Efficiency and Release Kinetics of Chrysin-Loaded Chitosan NPs\nThe amount of chrysin loaded onto the CCNPs was found to be 80.86 ± 0.30%. The in vitro drug release profile of chrysin from the CCNPs was determined in a release medium constituting PBS and DMSO, at 37 °C. The pH of the media was set at 7.4, as the ionic strength of the media plays a vital part in the stability and drug release. The cumulative chrysin release as a function of time is depicted in  Figure 2 a. The drug release kinetics of the loaded NPs initially showed a burst release which was followed by a steady and sustainable release from the 8th h. The first burst release was observed in the first two hours with 36.33 ± 1.58% of chrysin release. The second burst was observed after the sixth hour with 80.11% ± 0.84% drug release. The graph takes the form of a plateau starting from the 10th h to 24th h. It was also revealed that about a total of 90.5% ± 0.50% of the chrysin was released from the NPs within 10 h.\n3.3. Minimum Inhibitory Concentration (MIC) and Sub-MIC of Chrysin-Loaded Chitosan NPs\nThe MIC value of the CCNPs was determined to be 1024 μg/mL for  S. aureus . At 768 μg/mL, the NPs that did not exert any effect on the growth of the test bacteria. Hence, 768 μg/mL was selected as a sub-MIC concentration and used in all the subsequent anti-biofilm assays.\n3.4. In Vitro Anti-Biofilm Activity of Chrysin-Loaded Chitosan NPs\n3.4.1. Crystal Violet Staining Assay for Biofilm Formation and Disruption\nThe CCNPs showed a reduction in biofilm formation as compared to its bulk counterparts. The biofilms were inhibited to 50.48 ± 2.42% and 54.1 ± 0.56 % on treatment with CNPs and chrysin, respectively ( Figure 2 b). However, the CCNPs inhibited the biofilm formation to 66.59 ± 3.09%. The treatment of the preformed biofilm with CCNPs also resulted in a reduction in the biofilm mass of 43.50 ± 1.29%, whereas a decrease in biofilm of 14.92 ± 2.17% and 20.94 ± 3.73% was observed in the presence of CNPs and chrysin, respectively ( Figure 2 c).\n3.4.2. Microscopic Examination of Biofilm\nLight microscopy and CLSM were used to observe the change in the biofilm architecture of  S. aureus  in presence and absence of CCNPs. The influence of CCNPs on the thickness of biofilm, overall structure and biofilm density was evident in the micrographs. However, a dense biofilm was visible in the light microscope images of the untreated samples ( Figure 3 ). A thick biofilm of 80 μm was observed in the control while the thickness was reduced to 20 μm on treatment with chrysin. However, a higher reduction in the thickness of the biofilm matrix to 16 μm was achieved in the presence of CCNPs ( Figure 3 ).\n3.4.3. Exopolysaccharide (EPS) Quantification and Microbial Adhesion to Hydrocarbon (MATH) Assay\nThe CCNPs showed better reduction in the synthesis of EPS compared to its bulk counterparts. On treatment with CCNPs, a reduction in EPS production of 38.03 ± 5.41% was observed ( Figure 4 a). Chrysin and CNPs were also able to restrict the production of EPS by 33.37 ± 4.84% and 26.54 ± 3.20%, respectively. Cell surface hydrophobicity (CSH) is another important factor in biofilm formation as it aids in the adherence of the cell to the substratum. The CCNPs reduced the CSH in  S. aureus  by 84.66 ± 2.84% as compared to its bulk counter parts ( Figure 4 b). CNPs and Chrysin showed approximately 61.28 ± 5.78% and 72.46 ± 4.21% decreases in cell surface hydrophobicity, respectively. \n3.4.4. Growth Curve Analysis \nThe growth pattern of the test organism when cultivated in the presence and absence of the CCNPs and CNPs is presented in  Figure 5  Though the cells exhibited retardation in growth on exposure to a sub-MIC of the NPs and Chrysin, there was no significant decrease in the cell density. Hence, it can be inferred that the CCNPs arrested the biofilm development in  S. aureus , indicating the potential application of CCNPs in the management of  S. aureus -related infections.&quot;},&quot;discussion&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;4. Discussion\nThe formation of biofilm is one of the major obstacles in the modern antibacterial therapy. The biofilm-forming ability of bacteria provides the pathogen with advantages by blocking the entry of antimicrobial agents, thus causing hindrance in the clearance of these pathogens by the host immune system. Drug nanonization—i.e., a reduction in the particle size of drugs to nano-size—enhances the intracellular uptake of nanoparticles thus, providing a way to overcome the problems associated with insoluble drugs. Moreover, an increase in the surface area of poorly soluble drugs also leads to a more pronounced increase in the therapeutic index by maximizing the action with lesser dose [ 25 ]. In the present study, the anti-biofilm activity of CCNPs was demonstrated against biofilm forming bacterium,  S. aureus  MCC 2408. Chrysin was encapsulated to chitosan using the ionotropic gelation method. The NPs were formed due to the electrostatic interaction between amine group of chitosan and polyphosphate ions of TPP [ 19 ]. The hydrodynamic size influences various properties such as loading efficiency, drug release kinetics and the stability of the NPs. Though smaller nanoparticles due to high surface area show greater encapsulation efficiency, however, it also tends to aggregate easily on storage [ 19 ]. Particles of a low polydispersity index (PDI) are homogenous in nature and provide maximum stability. High polydispersity index (PDI) indicates the heterogeneity of the nanoparticles in the mixture. The synthesized spherical nanoparticles showed intermediate polydispersity that aids in the stability of the CCNPs. Ilk et al. [ 19 ] reported the synthesis of kaempferol loaded chitosan/TPP nanoparticles with an average particle size of 192.27 nm. The FTIR spectra of CCNPs indicated the presence of a similar functional group as that of its bulk counterpart indicating the successful encapsulation of chrysin with chitosan and the formation of chrysin-loaded chitosan NPs.\nThe biological efficacy of a drug and its potential use in drug delivery is directly influenced by the loading efficiency and controlled release. The CCNPs demonstrated a high encapsulation efficiency with a sustainable release. The CCNPs showed a higher loading efficiency compared to the previously synthesized nanocomposites such as the BSA-loaded chitosan-TPP nanoparticle with a loading efficacy of 60% [ 13 ] and PEG-chrysin conjugates with a loading efficacy of 55.6% [ 26 ]. The high encapsulation efficiency of CCNPs may be attributed to the presence of hydrogen bond between the -OH group of chrysin and the -NH2 group of chitosan that help in better entrapment of chrysin into the CNPs. From these release kinetics, it can be interpreted that chrysin was not covalently bonded to the nanoparticle and was thus easily released when dislodged in the medium. A similar result was observed in kaempferol-loaded chitosan nanoparticles, where more than 85% of the drug release was attained within 4 h, and no significant quantity of the drug was released thereafter [ 19 ].\nBiofilm formation by  S. aureus  is associated with many nosocomial as well as chronic diseases associated with medical devices and surgical implants. It also leads to the emergence of the multi-drug resistant (MDR) strains viz. Methicillin-resistant  S. aureus  (MRSA) and Vancomycin-resistant  S. aureus  (VRSA) [ 3 ]. It was found that both CNPs and chrysin exhibited significant anti-biofilm activity relative to the untreated control. However, the anti-biofilm efficacy was comparatively enhanced when chrysin was encapsulated with chitosan. The ability to attach and establish biofilm on inert surfaces contributes to making  S. aureus  a major pathogen of chronic infections [ 10 ]. The data also suggested that the CCNPs showed better biofilm inhibition ability than disruption of preformed biofilm. Shi et al. [ 10 ] suggested that chitosan-coated iron oxide nanoparticles have the potential to effectively prevent bacterial colonization and control the biofilm formation by 53% in  S. aureus . The anti-biofilm efficacy of the NPs was also validated by light and CLSM micrographs which showed a reduction in thickness and density of the biofilm matrix in presence of CCNPs.\nThe EPS matrix plays an indispensable role in the initial cell attachment, the formation of biofilm architecture and in providing mechanical stability of the biofilm. The EPS produced by the biofilm-forming bacteria prevents the access of antimicrobial agents and antibiotics to the bacterial cell [ 18 ]. The CCNPs caused a considerable decrease in the EPS production and cell surface hydrophobicity in  S. aureus  which resulted in the decrease in bacteria accumulation and attachment to the substratum. Hence, it can be inferred that CCNPs have a profound effect on the early stages in biofilm formation, specifically in the adherence and colonization as compared to its bulk counterparts, chrysin and chitosan. \nFrom the growth curve analysis, it can be interpreted that at the sub-MIC level, the CCNPs exerted less bactericidal effect and selective pressure against  S. aureus . However, it had a profound effect on the  S. aureus  in the biofilm mode of growth. Chrysin and chitosan are found to be nontoxic in recommended concentrations. It was reported that the recommended daily concentration of this flavone is 0.5 to 3 g [ 27 ]. Likewise, chitosan nanoparticles are nontoxic at low concentrations and found to be toxic only at higher concentrations [ 28 ]. The nontoxic nature of these components may enable the application of CCNPs for biomedical applications. As these the outcome of the study suggested that the nanoformulation of chrysin exhibits enhanced synergistic anti-biofilm activity against  S. aureus  when compared to its bulk counterparts—chrysin and chitosan taken separately. Hence, CCNPs may be considered as a potential therapeutic agent for controlling biofilm formation in  S. aureus . The nanocomposites may be further exploited towards the development of anti-biofilm coatings.&quot;},&quot;conclusion&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;5. Conclusions\nThis study displayed an enhanced antibiofilm activity of chrysin against  S. aureus  when loaded on to chitosan-TPP nanoparticles, with a profound loading capacity. Chrysin-loaded chitosan nanoparticles were characterized to confirm the effective loading of the flavone on chitosan nanoparticles. Anti-biofilm activities of CCNPs were determined through biofilm inhibition, biofilm disruption, EPS reduction and hydrophobicity reduction assays. CCNPs synthesized could be used as a potential therapeutic agent for controlling biofilm formation in  S. aureus  in the future. The nanocomposites may be further exploited towards the development of anti-biofilm coatings.&quot;},&quot;front&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;The application of nanotechnology in medicine is gaining popularity due to its ability to increase the bioavailability and biosorption of numerous drugs. Chrysin, a flavone constituent of  Orocylumineicum vent  is well-reported for its biological properties. However, its therapeutic potential has not been fully exploited due to its poor solubility and bioavailability. In the present study, chrysin was encapsulated into chitosan nanoparticles using TPP as a linker. The nanoparticles were characterized and investigated for their anti-biofilm activity against  Staphylococcus aureus . At sub-Minimum Inhibitory Concentration, the nanoparticles exhibited enhanced anti-biofilm efficacy against  S. aureus  as compared to its bulk counterparts, chrysin and chitosan. The decrease in the cell surface hydrophobicity and exopolysaccharide production indicated the inhibitory effect of the nanoparticles on the initial stages of biofilm development. The growth curve analysis revealed that at a sub-MIC, the nanoparticles did not exert a bactericidal effect against  S. aureus . The findings indicated the anti-biofilm activity of the chrysin-loaded chitosan nanoparticles and their potential application in combating infections associated with  S. aureus .&quot;},&quot;body&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;&quot;},&quot;back&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Acknowledgments\nWe are grateful to the Central Instrumentation Facility (CIF), Pondicherry University for the DLS and FTIR analysis. We would like to thank the Sophisticated Test and Instrumentation Centre (STIC), Cochin for the TEM analysis. We are also thankful to Bharathidasan University, Tiruchirappalli for supporting us with the Confocal Laser Scanning Microscopy. The authors extend their appreciation to The Researchers supporting project number (RSP-2019/15) King Saud University, Riyadh, Saudi Arabia.\nAuthor Contributions\nConceptualization, B.S.; methodology, B.S., U.P., and A.S.; software, B.S., U.P., A.S., and R.P.; validation, B.S., and R.P.; formal analysis, B.S.; investigation, B.S., A.M.E., and A.H.B.; resources, B.S., A.S., and K.K.; data curation, B.S., U.P., R.P., K.K., and A.S.; writing—original draft preparation, B.S., U.P., R.P., A.S., and K.K.; writing—review and editing, B.S., R.P., A.S., and K.K.; visualization, B.S., A.M.E., and A.H.B.; supervision, B.S.; project administration, A.S.; funding acquisition, A.S. All authors have read and agreed to the published version of the manuscript.\nFunding\nThe APC was funded by The Researchers supporting project number (RSP-2019/15) King Saud University, Riyadh, Saudi Arabia. \nConflicts of Interest\nThe authors declare no conflict of interest.&quot;},&quot;license&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;CC BY&quot;},&quot;retracted&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;no&quot;},&quot;last_updated&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;2024-01-16 23:36:45&quot;},&quot;citation&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Pathogens. 2020 Feb 12; 9(2):115&quot;},&quot;package_file&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;oa_package/ba/c7/PMC7168315.tar.gz&quot;}}},{&quot;rowIdx&quot;:2,&quot;cells&quot;:{&quot;accession_id&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;PMC7408465&quot;},&quot;pmid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;32610539&quot;},&quot;introduction&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;1. Introduction\nFlibanserin (FLB) is a recently FDA-approved nonhormonal drug for the treatment of women with hypoactive sexual appetite disorder. FLB acts via decreasing the level of serotonin and increasing the levels of dopamine and norepinephrine for maintaining healthy sexual response [ 1 ]. FLB-treated women have demonstrated significant improvements in both the number of satisfying sexual events and the female sexual function index desire domain score compared placebo-treated ones. These findings proved the ability of the drug to enhance the women’s sexual desire. In addition, administration of FLB was linked with a significant reduction in the distress related with either sexual dysfunction or low sexual desire [ 2 , 3 , 4 , 5 ]. However, the major challenge for oral administration of FLB is the reduced bioavailability (~33%) that might be caused by the drug’s low solubility and its exposure to hepatic first-pass metabolism [ 6 , 7 ].\nRecently, intranasal drug administration has gained increasing interest. The nasal pathway is a noninvasive route for active pharmaceutical ingredient (API) administration with the aim of local, systemic, or central nervous system (CNS) action. The nasal cavity represents an ideal absorption surface for drug delivery due to the high vascularity of this area, in addition to the leaky epithelium that results from the low tightness of the intercellular nasal mucosal junctional complex. Furthermore, direct absorption of the molecules from the nasal cavity via the trigeminal and olfactory pathways provides direct entry into the brain and results in a favorable pharmacokinetic/pharmacodynamic profile for centrally acting drugs. Thus, the nasal route could offer an encouraging unconventional approach to enteral and systemic drug administration of CNS-targeting drugs [ 8 , 9 ].\nTransfersomes (TRFs), also called deformable or elastic liposomes, are flexible vesicular systems that involve a phospholipid (PL) and an edge activator. They are considered as a modified generation of liposomes and were firstly modified by Cevc and Blume [ 10 ] by adding edge activators. The edge activators are usually a single-chain surfactant which enhances the squeezing and penetration of the vesicles through the mucosal barrier through destabilization of the lipid bilayers. The commonly used edge activators include sodium deoxycholate, sodium cholate, Tween, and Span [ 11 , 12 , 13 ]. Intranasal administration of TRFs has been previously reported to enhance bioavailability of several drugs [ 14 , 15 , 16 ]. Moreover, TRFs have been effectively applied for enhancing brain distribution of centrally acting medicines [ 17 , 18 , 19 ].\nHydrogel-loaded nanoformulated drugs have drawn significant attention as promising nanoparticulate drug delivery systems that combine both hydrogel system properties (e.g., hydrophilicity and high water absorption affinity) and nanoparticulate properties (e.g., ultrasmall size) [ 20 , 21 , 22 , 23 , 24 , 25 , 26 , 27 ], can achieve high drug loading without chemical reactions, and are able to release integrated agents at the target site in a controlled behavior. A wide range of natural, naturally derived and synthetic hydrogels can be used for hydrogel-loaded nanoformulated drug preparation [ 26 , 27 , 28 ]. Hydrogels can be prepared from naturally derived protein or polysaccharide polymers [ 29 ]. The synthetic hydrogels have drawn great attention in the biomedical field [ 30 , 31 ]. The synthetic hydrogels are obtained through chemical and physical methods. Among the synthetic hydrogels, poly(2-isopropenyl-2-oxazoline) (PiPOx) is a biocompatible polymer synthesized using a simple protocol [ 30 ]. In addition, poly(vinyl alcohol) (PVA) and PVA/poly(ethylene glycol) (PEG) hybrid hydrogels were synthesized that showed improved mechanical strength when compared with PVA hydrogel [ 32 ].\nAmong natural and naturally derived hydrogels, the most frequently used are polysaccharides. Materials with polysaccharides can be divided into two groups, namely polyelectrolytes and non-polyelectrolytes. Additionally, polyelectrolytes may be classified according to their intrinsic charges, including cationic (chitosan), anionic (alginate, heparin, pectin, hyaluronic acid), and neutral (pullulan, dextran). Due to their desirable mucoadhesive properties, cellulose derivatives can significantly extend the residence time of drugs in the nasal cavity [ 33 ]. Furthermore, due to their high viscosity following hydration in the nasal cavity, celluloses can sustain the release of drugs. For these reasons, the use of cellulose as an absorption enhancer can lead to improved intranasal absorption and increased bioavailability [ 34 ]. Reports show that celluloses increase the intranasal bioavailability of both small hydrophobic and hydrophilic macromolecular drugs [ 35 ]. Hydroxypropyl methyl cellulose (HPMC) is a popular matrix material in controlled drug delivery systems, and HPMC matrices show a sustained release pattern by two mechanisms, i.e., diffusion and erosion of the gel layer [ 36 ]. The viscosity of the polymer affects the diffusion pathway. HPMC can be employed as a matrix for controlling the release of both hydrophilic and hydrophobic drugs [ 37 ]. HPMC-based gels showed good surface morphology with high drug loading efficiency. The viscosities of the preparations were found to be within a suitable range for nasal administration.\nTherefore, the main aim of this study was to acquire an optimized FLB-TRF-loaded HPMC-based hydrogel for an improved drug delivery to the brain via intranasal administration. Box–Behnken design was utilized for FLB TRF optimization. The effects of FLB-to-PL molar ratio, edge activator hydrophilic lipophilic balance (HLB), and pH of hydration medium on vesicle size were studied. The optimized TRFs with minimized vesicle size were prepared and fused into hydroxypropyl methyl cellulose based hydrogel. The prepared hydrogel was assessed for shape characteristics and ex vivo permeation. In addition, in vivo performance was evaluated after intranasal administration in Wistar rats.&quot;},&quot;methods&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;2. Materials and Methods\n2.1. Materials\nFlibanserin (FLB) was purchased from Qingdao Sigma Chemical Co., Ltd. (Qingdao, China); Phospholipon 90G (phosphatidyl choline from soy, at least 90% purity) was purchased from Lipoid GmbH (Frigenstr, Ludwigshafen, Germany); Span 65, Span 80, methanol, and chloroform were purchased from Sigma-Aldrich Co. (St. Louis, MO, USA).\n2.2. FLB TRF Preparation\nFLB TRFs were prepared by the hydration of the formed lipid film as previously described [ 38 ]. Briefly, specified amounts of FLB, PL, and edge activator (surfactant) were dissolved in methanol/chloroform mixture (1:1,  v / v ) and subjected to water bath sonication for 5 min. The amounts of FLB, PL, and surfactant were specified according to  Table 1 . Span 65 and span 80 were used in different ratios to achieve the required HLB value of the edge activator indicated in the design ( Table 1 ). The solution was then evaporated using a rotary evaporator at 45 °C. The formed film was kept in a vacuum oven overnight for complete removal of solvent residuals. Subsequently, the dried thin film was hydrated with 20 mL of buffer solution, according to the specified pH, for 3 h at 25 °C with gentle shaking.\n2.3. Box–Behnken Design for FLB TRF Preparations\nAccording to the previous screening results conducted in our laboratory, the optimization of FLB TRFs was carried out to achieve minimal size. FLB:PL molar ratio ( X 1 ), HLB ( X 2 ), and pH of hydration medium ( X 3 ) were the investigated factors, while vesicle size ( Y 1 ) was the studied response. The  X 1  ratios studied were 1:1, 1:3, and 1:5;  X 2  values were 2, 4, and 6; and  X 3  values were 5, 7, and 9. All other processing and formulation variables, including drug amount (10%  w / w ), were kept constant throughout the study. The experimental design using Design-Expert software (version 12; Stat-Ease, Inc., Minneapolis, MN, USA) yielded 17 formulations. The actual values of the independent variables of these runs and the observed responses are presented in  Table 1 . The measured responses were statistically analyzed by the analysis of variance (ANOVA) test. The polynomial equations representing the best fitting model for each variable was generated. Three-dimensional surface plots were plotted to illustrate the impact of the variables and interaction between them at  p  < 0.05. Afterwards, a numerical method following the desirability approach was utilized to predict the optimized FLB TRFs. The predicted formulation was then prepared and further evaluated. The measured responses were compared to the predicted ones, and the residual error was calculated to ensure the success of the optimization process.\n2.4. Vesicle Size Determination\nThe vesicle size of freshly prepared FLB TRF was measured using a Zetasizer Nano ZSP (Malvern Panalytical Ltd. Malvern, UK). The result is expressed as the mean of five determinations.\n2.5. Characterization of Optimized FLB TRFs\nFor investigation of vesicle size, polydispersity index (PDI), and zeta potential of the optimized FLB TRFs, the same method mentioned in  Section 2.4  using a Malvern size analyzer was employed. In addition, optimized FLB TRFs were subjected to transmission electron microscopy (TEM). A sample was placed on a copper grid and stained using phosphotungstic acid. After removing excess stain, the stained sample was dried and studied using a JEOL-JEM-1011 transmission electron microscope (JEOL, Tokyo, Japan).\n2.6. Preparation of Optimized FLB-TRF-Loaded Hydrogel\nOptimized FLB TRFs were incorporated into hydroxypropyl methyl cellulose (HPMC) based hydrogel. Briefly, specified amount of HPMC (0.1 g) was dispersed in distilled (10 mL) water to make a 1%  w / v  concentration. The gel was kept in the refrigerator overnight and then FLB TRFs were added with continuous stirring to obtain a drug concentration of 10 mg/g. Control hydrogels incorporating raw drug (10 mg/g gel) were prepared under the same conditions for comparison.\n2.7. Optimized FLB TRF Gel Ex Vivo Permeation Study\nFreshly excised goat nasal mucosa was utilized for ex vivo permeation studies. Mucosa were equilibrated in simulated nasal fluid (SNF) with pH 6.8 for 15 min. SNF was composed of sodium chloride (0.877%), calcium chloride (0.058%  w / v ), and potassium chloride (0.298%  w / v ) dissolved in deionized water [ 39 ]. Mucosa and gel samples were mounted between the two chambers and the donor chamber [ 40 ]. The area of the chamber of the utilized Franz automated diffusion cell (MicroettePlus; Hanson Research, Chatsworth, CA, USA) was 1.76 cm 2 . Gels loaded with optimized FLB TRF or raw FLB (0.1 g each 10 mg FLB/g gel) were utilized in this study. Seven milliliters of simulated nasal fluid (SNF) with pH 6.8 was used in the receiver chamber as the diffusion medium that was kept at 35 ± 0.5 °C with the agitation rate set at 400 rpm. At specified time intervals, 1.5 mL aliquots were withdrawn and replaced with fresh SNF.\n2.8. In Vivo Pharmacokinetic Assessment\nThe pharmacokinetic performance of the FLB-TRF-loaded hydrogel was investigated in Wistar rats, weighing 200–250 g each, and compared to control raw-FLB-loaded gel. The study protocol was approved by the Research Ethics Committee, Faculty of Pharmacy, King Abdulaziz University, Kingdom of Saudi Arabia, under approval number (PH-124-41). The committee ensures that animal use complies with the European Union Directive 2010/63/EU and the DHEW Publication NIH 80-23 Guiding Principles. The study included two animal groups (I and II), with all animals receiving FLB dose of 10 mg/kg intranasally. Group I received raw FLB gel, and group II received FLB-TRF-loaded hydrogel. Collection of blood samples was performed at specified time intervals. Six rats from each group were sacrificed at each time interval, and the whole brain was washed with saline after removal and then weighed. Brain tissues were homogenized with phosphate buffer (pH 7.4) at 5000 rpm for 3 min. Plasma and homogenized brain samples were stored at −80 °C prior to analysis [ 41 ].\nA volume of 200 μL of plasma sample, along with 200 μL of the brain homogenate, was transferred to a screw-capped test tube, mixed with 50 μL internal standard solution (valsartan, 625 ng/μL) and 1 mL acetonitrile, vortexed for 1 min, and then centrifuged at 5300 rpm for 8 min. An aliquot of the clear supernatant was transferred to a total recovery autosampler vial, and a volume of 7 μL was injected for LC-MS/MS-DAD analysis. The MS system was connected to an Agilent 1200 HPLC system equipped with an autosampler, a quaternary pump, and a column compartment (Palo Alto, CA, USA). The system was equipped with ChemStation software (Rev. B.01.03 SR2 (204)). The IT–MS was controlled using 6300 series trap control version 6.2 Build No. 62.24 (Bruker Daltonik GmbH), and the general MS adjustments were as follows: capillary voltage, 4200 V; nebulizer, 37 psi; drying gas,12 L/min; desolvation temperature, 330 °C; ion charge control (ICC) smart target, 200,000; and max accumulation time, 200 ms. The MS scan range was 50–550 m/z. For quantitative monitoring, single positive molar ion mode was applied at programed time segment, 0–4.0 min, m/z 391.2 [M+H]+ FLB; 4.0–10 min, m/z 436.3 [M+H]+ internal standard. Isocratic elution was conducted at a flow rate of 0.5 mL/min with a mobile system composed of 52% acetonitrile and 48% water containing 0.1% formic acid. FLB content in the assayed samples was quantified with reference to a calibration curve (range of 1–1000 ng/mL). The calibration curves for FLB were assessed using free-drug-plasma and free drug brain homogenate matrixes as a calibration matrix. The stock solutions of FLB and valsartan (InSt) were prepared separately by dissolving 10 mg of each in methanol to obtain a concentration of 0.1 mg/mL. A series of calibrator working solutions of FLB were prepared from its stock solutions by applying a serial dilution technique and using methanol as the diluting solvent. The calibration solutions were prepared by spiking the plasma-free drug with FLB solutions to give a concentration spanning the range of 1.0 to 1000.0 ng/mL of FLB and a fixed InSt concentration of 25 μg/mL. The calibrated solutions were extracted and analyzed by the developed method. The peak area ratios of FLB-to-InSt were found to be linear in the concentration range of 1.0 to 1000 ng/mL of FLB. Pharmacokinetic parameters including the maximum plasma concentration (C max ), time to maximum plasma concentration (T max ), and area under the plasma concentration–time curve (AUC 0–∞ ) were calculated using Kinetica software (Version 4; Thermo Fisher Scientific, Waltham, MA, USA). The parameters were analyzed for significance using SPSS software (Version 16; SPSS Inc., Chicago, IL, USA). Unpaired Student’s  t -test was performed on  C max  and AUC 0–∞ , while the nonparametric Mann–Whitney test was utilized for analysis of T max ; a level of significance of  p  < 0.05 was set for all investigated pharmacokinetic parameters. For histopathological evaluation, 12 rats were divided into four groups: untreated rats (gp1), rats treated with plain in situ gel without drug (gp2), rats that received FLB drug in the in situ gel (gp3), and rats treated with FLB-Nanostructured lipid carriers (FLB-NLCs, gp4). The same dosing procedure as previously described in the pharmacokinetics study was used. After 8 h, histopathologic analysis was conducted according to the method of Young [ 42 ]. In brief, the head was removed, and the brain and jaw were removed from the head along with any other listed tissues. The nasal cavity was initially fixed in a solution of 10% formalin and then decalcified in a solution of 10% EDTA. The tissue was then placed in 70% ethanol before being embedded in paraffin, sectioned, and stained with hematoxylin and eosin.\n2.9. Statistical Analysis\nFor the in vivo data, the software selected to perform the statistical analysis was GraphPad Prism (San Diego, CA, USA). One-way or two-way analysis of variance (ANOVA), followed by Tukey’s post hoc test, was used for multiple comparisons. Only values of  p  < 0.05 were considered statistically significant. Each set of experiments was performed at least in triplicate and is reported as means ± SD. For the in vitro Box–Behnken design data, the effects of factors on the response (vesicle size) were statistically analyzed by ANOVA using the Design-Expert software.&quot;},&quot;results&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;3. Results\n3.1. Polynomial Model Selection and Diagnostic Analysis\nThe observed vesicle size of the prepared TRFs best fitted to the quadratic model based on its highest correlation coefficient (R 2 ) is shown  Table 2 . There was a satisfactory agreement between the predicted and adjusted R 2 , indicating that the selected model was valid for analyzing the data. Moreover, an adequate precision value of greater than 4 indicates an adequate signal-to-noise ratio, implying the suitability of the quadratic model to navigate the design space. Diagnostic plots were generated to ensure the goodness of fit of the chosen model.  Figure 1 A, illustrating the residual vs. run plots, shows randomly scattered points, indicating that there is no lurking variable interfering with the vesicle size. Furthermore, the high linearity illustrated in the predicted versus actual values plot ( Figure 1 B) indicates that the observed vesicle size was analogous to the predicted one.\n3.2. Statistical Analysis for the Effect of Variables on Vesicle Size (Y)\nThe size of vesicles is a critical parameter that exhibits a significant impact on the drugs’ permeation via the biological membranes. FLB TRF showed size in the nanoscale range with mean size ranging from 88 ± 0.86 to 175 ± 2.43 nm ( Table 1 ). The relatively small standard deviation could indicate homogeneity of the TRF dispersions. The equation representing the selected sequential model was generated in terms of coded factors as follows:\nThe statistical analysis revealed that all the linear terms corresponding to the three investigated variables have a significant negative effect on FLB TRF size ( p  < 0.05). The quadratic terms corresponding to the surfactant HLB ( X 2 2 ) and hydration medium pH ( X 3 2 ), in addition to the interaction term  X 2 X 3  corresponding to the interaction between the two aforementioned variables, were also found to be significant at the same significance level.  Figure 2  illustrates the contour plots for the investigated variable effects on vesicle size.\n3.3. FLB TRF Optimization\nThe formation of the optimized FLB TRFs was accomplished using a numerical optimization technique with a minimized vesicular size. The optimized formulation was prepared at factor levels of 1:1.12 FLB:PL molar ratio, HLB value of 2.3, and hydration medium pH of 7.2. The observed and predicted values of the optimized FLB TRF formulation were in good agreement (with low error percentage), confirming the reliability of the optimization process ( Table 2 ).\n3.4. Charactarization of the Optimized FLB TRFs\nThe PDI of the optimized formulation was found to be 0.201 ± 0.012, while the zeta potential was equal to 8.12 ± 1.54 mV. TEM has been applied for assessing of the shape and lamellarity of the optimized FLB TRF at 25,000× magnification. As illustrated in  Figure 3 , the TRF showed vesicles with spherical shape. No aggregation was observed. In addition, the recorded size was within an acceptable agreement with that recorded using the dynamic light scattering technique of the particle size analyzer.\n3.5. Optimized FLB TRF Gel Ex Vivo Permeation\nEx vivo permeation through goat nasal mucosa was carried out to give an insight into the in vivo performance of the optimized FLB-TRF-loaded hydrogel.  Figure 4  illustrates the mean cumulative percent FLB permeated from the TRF-loaded hydrogel (test) compared to FLB-loaded hydrogel (control). The optimized FLB TRF hydrogel shows a significant increase in cumulative percent FLB permeated when compared to raw FLB gel ( p  < 0.05), with almost complete drug permeation after 4 h. The maximum amount of drug permeated within 4 h from optimized FLB TRF hydrogel was approximately 1.97-fold greater than that from raw FLB hydrogel.\n3.6. In Vivo Pharmacokinetics\nThe calibration curves of the concentrations of FLB spiked in plasma and brain homogenate show linear relationships with correlation coefficients of 0.9992 and 0.9984, respectively. The assay shows an adequate precision, with relative standard deviations (RSDs) of 8.1–10.9% and 10.1–12.9% for the intraday assay and the interday assay, respectively. The mean extraction recoveries were 94.8% ± 5.4% and 92.6% ± 7.6% for FLB-spiked plasma and brain samples, respectively. Mean FLB concentrations in plasma and brain following intranasal administration of optimized FLB-TRF-loaded hydrogels, compared to the control FLB-loaded hydrogels, are graphically represented in  Figure 5 . The computed pharmacokinetic parameters are compiled in  Table 3 .\nFrom the results of the histopathological evaluation to follow the impact of FLB TRFs on the nasal tissues ( Figure 6 A–D), no pathological signs of epithelial damage, hyperplasia, edema, or inflammatory infiltration can be see for the four investigated groups.&quot;},&quot;discussion&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;4. Discussion\nThe nanoscale size observed could contribute to enhancing the drug permeation via the nasal mucosa and facilitating passing through the blood–brain barrier. Analysis of variance (ANOVA) for the vesicle size affirmed that the quadratic model was significant ( p  < 0.0001). The positive sign of the coefficients of the linear terms  X 1  and  X 2  indicates that the vesicle size increases significantly with increasing drug:PL molar ratio and/or surfactant HLB. Contrarily, the negative sign of the linear term  X 3  indicates that the vesicle size decreases significantly with increasing hydration medium pH.\nThe increase of size with increasing drug:PL molar ratio could be credited to increasing the PL content of the vesicles. Similar results were reported for other vesicular systems. Dubey et al. [ 43 ] demonstrated increased vesicle size of ethosomes with increasing PL content. In another study, Ahmed and Badr-Eldin [ 44 ] reported an increase in avanafil invasome size with increasing PL content of the vesicles. Regarding the HLB of the surfactant, it was observed that a significant reduction of the vesicle size occurs as the HLB is decreased. This observation could be explained on the basis of the increased hydrophobicity of the surfactant with reduced HLB values. Increased surfactant hydrophobicity could lead to reduction of surface energy and low water uptake into the vesicle core, resulting in reduction of the vesicle size [ 38 , 45 , 46 , 47 ]. The boosted FLB permeation from optimized FLB TRF gel could be attributed to the synergistic advantages of TRFs and the nanosized system. The flexible and deformable structure of the TRF could impart the potential to pass easily through the mucosal barriers. Furthermore, the existence of surfactants which act as edge activators could contribute to the permeation-enhancing ability of TRF by disrupting the lipid bilayer of the membrane [ 46 ]. In addition, the nanoscale size of the vesicles results in a great surface area, thus increased contact area with the mucosal epithelium and successively improving the chance of drug permeation [ 38 ]. Nanovesicles have been reported to have the potential to enhance drug absorption through the nasal membrane barrier and to demonstrate a high efficacy in enhancing drug bioavailability [ 40 ]. However, mucociliary clearance can help to reduce the contact time of drug-loaded nanovesicles on the mucosal surface inside the nose. Thus, the application of hydrogel-specific properties is now considered to be a useful platform for the preparation of stabilized and smart nanoscopic vehicles for drug delivery purposes. In addition, the incorporation of transferosomes into the hydrogel network can offer remote-controlled applications and also improve characteristics such as mechanical strength [ 25 , 42 , 48 , 49 ]. The observed higher extent of absorption from optimized FLB TRF hydrogel compared to the raw FLB gel could be attributed to the drug’s improved solubility and permeability by loading on a hydrophobic carrier. Comparing the two intranasal hydrogels, the optimized FLB-TRF-loaded hydrogel shows significant increases in C max  and AUC ( p  < 0.05) for both plasma and brain compared to control, indicating higher bioavailability and enhanced brain delivery of the drug. This could be attributed to FLB movement from the nasal cavity along both the olfactory or trigeminal nerves to the parenchyma of the brain. FLB is delivered to the nerves in the cerebrum and pons and then disperses throughout the brain. The intracellular and extracellular pathways are the ways by which FLB brain dispersion occurs. For the intracellular mechanism, FLB is internalized by an olfactory neuron through endocytosis, trafficked within the cell to the neuron’s projection site, and then released by exocytosis. For the extracellular pathway, FLB crosses the nasal epithelium to the lamina propria and then is transported externally along the length of the neuronal axon that leads into the CNS, where FLB is distributed by fluid movement. The enhanced drug bioavailability could be ascribed to the improved permeation properties of TRFs owing to their flexible and ultra-deformable structure that enhances penetration across the mucosal barrier [ 50 ]. Furthermore, the elevated concentration of the drug in the brain highlights the capability of TRF to augment direct delivery of the drug to the brain through the nasal olfactory region and across the BBB. The nanoscale size of the vesicles might also yield a shielding effect for the drug, protecting it from fast excretion and metabolism and leading to improved CNS delivery [ 41 ].&quot;},&quot;conclusion&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;5. Conclusions\nTRF-loaded hydrogel has been investigated as a possible intranasal delivery system of FLB. Box–Behnken design was successfully applied for optimization of FLB TRFs with minimized vesicular size. The optimized FLB TRFs (1:1.12 drug:PL molar ratio, surfactant HLB of 2.3, and hydration medium pH of 7.2) were spherical, with a vesicle size of less than 100 nm. The optimized FLB-TRF-loaded hydrogel showed an enhanced ex vivo permeation profile through goat mucosa when compared to that of control FLB hydrogel. In vivo assessment in Wistar rats confirmed that the optimized hydrogel had higher bioavailability than the control and exhibited enhanced brain delivery. Based on these results, the proposed optimized FLB-TRF-loaded hydrogel could be considered a promising drug delivery system for nose-to-brain delivery of the drug.&quot;},&quot;front&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Flibanserin (FLB) is a nonhormonal medicine approved by the Food and Drug Administration (FDA) to treat the hypoactive sexual appetite disorder in females. However, the peroral administration of the medicine is greatly affected by its poor bioavailability as a result of its extensive first-pass effect and poor solubility. Aiming at circumventing these drawbacks, this work involves the formulation of optimized FLB transfersome (TRF) loaded intranasal hydrogel. Box–Behnken design was utilized for the improvement of FLB TRFs with decreased size. The FLB-to-phospholipid molar ratio, the edge activator hydrophilic lipophilic balance, and the pH of the hydration medium all exhibited significant effects on the TRF size. The optimized/developed TRFs were unilamellar in shape. Hydroxypropyl methyl cellulose based hydrogel filled with the optimized FLB TRFs exhibited an improved ex vivo permeation when compared with the control FLB-loaded hydrogel. In addition, the optimized TRF-loaded hydrogel exhibited higher bioavailability and enhanced brain delivery relative to the control hydrogel following intranasal administration in Wistar rats. The results foreshadow the possible potential application of the proposed intranasal optimized FLB-TRF-loaded hydrogel to increase the bioavailability and nose-to-brain delivery of the drug.&quot;},&quot;body&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;&quot;},&quot;back&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Author Contributions\nConceptualization, O.A.A.A. and U.A.F.; methodology, S.M.B.-E.; software, H.M.A.; validation, Z.A.A., H.Z.A., and A.K.K.; formal analysis, G.C.; investigation, F.C.; resources, A.A.; data curation, R.A.A.-G. (Raniyah A. Al-Ghamdi); writing—original draft preparation, U.A.F.; writing—review and editing, S.M.B.-E., G.C. and F.C.; visualization, R.A.A.-G. (Rawan A. Al-Ghamdi); supervision, Z.A.A.; project administration, N.A.A.; funding acquisition, N.A.A. All authors have read and agreed to the published version of the manuscript.\nFunding\nThis project was funded by the Deanship of Scientific Research (DSR) at King Abdulaziz University, Jeddah, under grant No. RG-13–166–41. The authors, therefore, acknowledge with thanks the DSR for technical and financial support\nConflicts of Interest\nThe authors declare no conflict of interest. The funders had no role in the design of the study; in the collection, analyses, or interpretation of data; in the writing of the manuscript, or in the decision to publish the results.&quot;},&quot;license&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;CC BY&quot;},&quot;retracted&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;no&quot;},&quot;last_updated&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;2024-01-16 23:35:06&quot;},&quot;citation&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Nanomaterials (Basel). 2020 Jun 29; 10(7):1270&quot;},&quot;package_file&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;oa_package/82/4e/PMC7408465.tar.gz&quot;}}},{&quot;rowIdx&quot;:3,&quot;cells&quot;:{&quot;accession_id&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;PMC7615524&quot;},&quot;pmid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;37727079&quot;},&quot;introduction&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Introduction\nRandomized controlled trials (RCTs) are considered the gold standard for assessing the causal effect of an exposure on an outcome, but are vulnerable to bias due to missingness in the outcome—or “dropout.” The impact of dropout depends on the missingness mechanism and the analysis model ( Dziura et al., 2013 ;  Little &amp; Rubin, 2020 ;  Rubin, 1976 ). Three missingness mechanisms can be distinguished: missing completely at random (MCAR), missing at random (MAR), and missing not at random (MNAR) ( Rubin, 1976 ). With MCAR, missingness is unrelated to any measured or unmeasured characteristics and the observed data are a representative subset of the full data. MAR means that the missingness can be explained by observed data and, with MNAR, missingness is a function of the unobserved data.\nTwo common methods of dealing with dropout are complete case analysis (CCA) and multiple imputation (MI). A CCA is the analysis model intended to be applied to the trial data at its outset, restricted only to individuals with observed outcomes. With MI, missing outcome values are repeatedly imputed conditional on the observed data, generating multiple complete datasets to which the analysis model is applied ( Buuren, 2018 ;  Little &amp; Rubin, 2020 ;  Rubin, 1987 ,  1996 ), with the resulting estimates subsequently pooled using Rubin’s rules ( Rubin, 1987 ). In practice, a CCA will be unbiased if dropout is MCAR or MAR, conditional on the analysis model covariates. MI will be unbiased if dropout is MCAR or MAR conditional on the analysis model and imputation model covariates, and if the imputation model is correctly specified. Generally, both will be biased when outcomes are MNAR ( Dziura et al., 2013 ;  Hughes et al., 2019 ;  Little &amp; Rubin, 2020 ). In this article, we consider the case of an RCT with an incomplete continuous outcome, where we assume that the outcome is generated from covariates and treatment according to a linear model. In such an RCT, a CCA will only be biased if the dropout is related the outcome, conditional on the model covariates ( Carpenter &amp; Smuk, 2021 ;  Hughes et al., 2019 ;  White &amp; Carlin, 2010 ). As RCT analyses typically adjust for a number of baseline covariates, in this article, we primarily focus on (the bias of) the treatment effect, when estimated conditional on some baseline covariate.\nIn the presence of dropout, observed data generally cannot be used to establish if outcomes are MNAR or MAR given the model covariates. Whether outcomes are MCAR can be partially tested using Little’s MCAR test, which compares the multivariate distribution of observed variables of patients with observed outcomes to those with unobserved outcomes ( Little, 1988 ). Little’s MCAR test and related tests, however, rely on strong parametric assumptions, with a conclusion that hinges on the specification of some arbitrary  P -value cutoff, which limits their practical value ( Li &amp; Stuart, 2019 ). Currently, there is no established statistical test for distinguishing between MAR and MNAR missingness, and consequently, no simple way to determine whether the treatment effect estimate is likely to be biased.\nCurrent guidance for assessing risk of bias due to dropout relies on checking if dropout is differential across trial arms, assessing the plausibility that dropout may be related to outcome (e.g., dropout due to lack of efficacy) ( Higgins et al., 2012 ;  Sterne et al., 2019 ), and comparing the baseline covariate distribution across trial arms in patients who are still observed at the end of follow-up ( Groenwold et al., 2014 ). While both differential dropout across trial arms and different baseline covariate distributions across trial arms in the observed data can be caused by MNAR dropout, these markers may also result from MAR dropout. The European Medical Agency (EMA) and the National Research Council (NRC), recommend using MAR-appropriate methods for the primary analysis, followed by sensitivity analyses that weaken this assumption ( Clinical Trials o. H. M. D. iNRCUP, 2010 ;  European Medicines Agency (EMA), 2020 ). These guidelines, however, are in practice implemented in a fraction of all trials, with on average only 6% (of  N  = 330 trials) describing the assumed missing data mechanism ( Hussain et al., 2017 ;  Rombach et al., 2015 ), 9% ( N  = 237) justifying their choice of main analysis ( Rombach et al., 2015 ), and 19% ( N  = 849) reporting some kind of sensitivity analysis ( Bell et al., 2014 ;  Hussain et al., 2017 ;  Rombach et al., 2015 ;  Wood et al., 2004 ;  Zhang et al., 2017 ), which rarely involves relaxing the primary analysis assumptions ( Bell et al., 2014 ), and only 9% ( N  = 200) discussing the risk of bias resulting from missing data ( Zhang et al., 2017 ). This discrepancy between recommended and implemented practice persists despite extensive literature on the subject and may be due to the relative complexity of such analyses.\nIn this paper, we propose using the differences between the observed variances of the outcome across the two arms of the trial to assess the risk of CCA estimator bias due to MNAR dropout. We show, using directed acyclic graphs (DAGs) and standard statistical theory, how MNAR may give rise to unequal outcome variances between the fully observed subjects in the two arms of the trial. We illustrate this method using individual-level data and summary-level data. Individual-level patient data were obtained from an RCT investigating the benefit of an acupuncture treatment policy for patients with chronic headaches (ISRCTN96537534) ( Vickers et al., 2004 ). The summary data application used published statistics from a cluster-randomized clinical trial, which investigated psychological outcomes following a nurse-led preventive psychological intervention for critically ill patients (POPPI, registration ISRCTN53448131).&quot;},&quot;methods&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Methods for Testing Differences in Variances\nVarious methods are available for testing and estimating the difference in variance between two groups, including Bartlett’s test ( Bartlett &amp; Fowler, 1937 ), Levene’s test ( Levene at al., 1960 ), the Brown–Forsythe test ( Brown &amp; Forsythe, 1974 ), the Breusch–Pagan test ( Breusch &amp; Pagan, 1979 ), and the studentized Breusch–Pagan test ( Koenker, 1981 ). In this paper, we employ the latter, as it has a straightforward implementation that allows for conditioning on additional covariates, and is also robust against nonnormally distributed errors. This is particularly relevant, as, in practice, outcomes are unlikely to be strictly normally distributed.\nThe studentized Breusch–Pagan estimate is obtained as follows. First, the outcome,  Y , is regressed on the treatment variable,  X , and optional additional covariates,  C , in an OLS regression: \nThe regression residuals are obtained and squared   and, in a second auxiliary OLS regression, regressed on the treatment variable: \nThe variance difference estimate is given by   and the test statistic is given by  nR 2 , with  n  the sample size and  R 2  the coefficient of determination, obtained from the second auxiliary regression.&quot;},&quot;results&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Results\nThe simulation results are shown in  Table 2 . We first consider the seven scenarios when there is no effect modification and the treatment effects are homogeneous (A–G). When dropout is MCAR (A), and when dropout is MAR conditional on treatment (B), some measured covariate (C), or both (D), the treatment effect estimates are unbiased, irrespective of conditioning on  Y b , with, on average, a zero outcome variance difference across trial arms in the observed data at baseline   and at follow-up   We observe the same for scenario F, where dropout is MNAR dependent on  U , but with  U  and  X  independent in the observed data. In scenario G, where dropout is MNAR dependent on  U  and  X , with  U  and  X  not independent in the observed data, the CCA treatment effect estimate is biased, and the mean outcome variance difference across trial arms at follow-up is nonzero. When dropout is MNAR dependent on  Y f  (E), we observe a biased treatment effect estimate and nonzero outcome variance differences at both baseline and at follow-up. Conditioning on  Y b  results in an attenuated bias estimate and a smaller variance difference at follow-up   When effect modification is present (EM), resulting in treatment effect heterogeneity, we observe a variance difference at follow-up, regardless of dropout mechanism. In contrast, we only observe a VD at baseline in scenario E, where dropout is MNAR dependent on outcome. In  Online Appendix C.2 , a companion table ( Table C.1 ) is provided for  Table 2 , with additional measures of simulation performance.\nIn summary, this simulation shows that a variance difference across trial arms, in the observed data, at baseline,   indicates outcome-dependent MNAR dropout (scenario E), which may result in a biased CCA treatment effect estimate. If   is zero but the variance difference across trial arms in the observed data at follow-up,   is nonzero, then one explanation is that dropout occurs according to the MNAR missingness mechanism in scenario G, which may result in a biased treatment effect estimate. Alternatively, this could be explained by effect modification together with a missingness mechanism that is MCAR (scenario A), MAR (scenarios B, C, and D), or MNAR (scenario F), which will result in an unbiased treatment effect estimate. If both   and   are zero, then the missingness mechanism is MCAR (scenario A), MAR (scenarios B, C, and D), or MNAR (scenario F) with no effect modification and the CCA treatment effect estimate will be unbiased. Adjusting for the outcome at baseline will result in an attenuation of the CCA estimator bias and a smaller   when dropout is MNAR as in scenarios E and G, irrespective of the presence of effect modification.&quot;},&quot;discussion&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Discussion\nIn this paper, we show that, in RCTs, the outcome variance difference across trial arms at baseline, in the set of study participants who did not drop out, is an indicator of outcome-dependent MNAR dropout, and consequently, a biased CCA treatment effect estimate. In contrast, when outcomes are MAR or MCAR, this baseline variance difference will be zero. We also show that the outcome variance difference across trial arms in the observed data at follow-up can be an indicator of both outcome-dependent MNAR dropout and nonoutcome dependent MNAR dropout, both of which may result in a biased treatment effect estimate. A variance difference across trial arms at follow-up, in the observed data, however, can only be meaningfully interpreted when the outcome variances can be expected to be equal across trial arms in the full data. This requires two assumptions: first, that there is no treatment effect heterogeneity; second, that the errors of the outcome are homoskedastic.\nTreatment effect heterogeneity can be thought of as nonrandom variability in treatment response that is attributable to patient characteristics. How plausible it is that heterogeneity is absent depends strongly on intervention type and study population. Efficacy trials, for example, typically have stricter inclusion criteria, resulting in more homogeneous study populations, and are less prone to large variations in treatment response. In contrast, pragmatic trials with broad eligibility criteria are more likely to have heterogeneous treatment effects ( Varadhan &amp; Seeger, 2013 ). As treatment effect heterogeneity affects the outcome variance of the intervention arm ( Mills et al., 2021 ), it is a second potential cause of observed outcome variance differences across trial arms. The presence of effect modification can be investigated by performing a stratified analysis, where the effect modifier serves as a stratification variable ( Corraini et al., 2017 ). Alternatively, if the observed difference in outcome variance between trial arms is solely the result of effect modification, then conditioning on the effect modifier and the interaction term between effect modifier and trial arm can be expected to remove all evidence of a variance difference.\nAn outcome variance difference across trial arms may also result from heteroskedastic outcome errors. Specifically, if the variance of the outcome errors is related to the outcome and if treatment has a causal effect on the outcome, this may cause a difference in outcome variances across trial arms at follow-up. For example, suppose an outcome such as body mass index (BMI) has greater day-to-day variability in people with higher BMI. Then, if the treatment lowers the mean BMI in the intervention arm, this will result in a comparatively smaller intervention arm variance. A simple way to investigate this is to consider the outcome variable measured at baseline, group its values into bins, and establish if the outcome error variance is different in bins with higher and lower mean values.\nWe additionally propose employing the (conditional) outcome variance difference across trial arms in the observed data as an MNAR bias assessment tool, and, indirectly, as a model building tool, which can be used to assess the added value of including variables for explaining the missingness mechanism. This method is easily implemented, using existing tests available in standard software, and has a straightforward interpretation of results. In  Section 7 , we demonstrated how outcome variance differences can be used to assess the risk of MNAR bias for various models, using both individual-level data from the acupuncture trial, and summary-level data from the POPPI trial.\nThe outcome variance difference across trial arms at baseline and at follow-up is suitable for assessing the risk of dropout bias for analysis models that are estimated with OLS linear regression, and assume that the outcome is continuous and given by a linear model. These methods cannot be used for noncontinuous outcomes, such as binary or time-to-event outcomes.\nA second limitation of our proposed method is its comparatively modest power, with the power to detect an outcome variance difference lower than the power to detect a difference in outcome means.  Brookes et al. (2004)  showed that if a trial is powered to detect a mean difference of a given size, then in order to detect an interaction effect of the same magnitude, the sample size needs to be approximately four times larger.  Mills et al. (2021)  found comparable numbers for the power to detect a variance difference.\nInstead of using our method as a strict significance test with a dichotomous conclusion, we recommend assessing the practical implications of the values inside the confidence interval ( Amrhein et al., 2019 ;  Andrade, 2019 ;  Wasserstein et al., 2019 ). For example, in the individual-level data application of  Section 7 , at follow-up, the outcome variances in the observed data were 188.2 and 289.4, in the intervention usual care arm, respectively, with a variance difference of −100.3 (95% CI: −222.0, 21.4). At baseline, in the observed data, we estimated a variance difference of −81.46 between trial arms, with a 95% CI of −183.21–20.30. These results are compatible with a large negative outcome variance difference as well as a small positive outcome variance difference. These large negative variance difference would raise concerns of MNAR dropout, with the large negative outcome variance difference at baseline specifically suggestive of outcome-dependent dropout.\nWhile a variance difference across trial arms in the observed data at baseline indicates outcome-dependent MNAR dropout, interpreting a variance difference at follow-up is less straightforward. For the latter, we suggest performing further analyses to identify the presence of heteroskedastic outcome errors and treatment effect heterogeneity, and using expert and contextual knowledge. If the presence of both heteroskedastic errors and treatment effect heterogeneity is judged to be unlikely, then the possibility of nonoutcome-dependent MNAR dropout should be investigated, for example, by conditioning on additional covariates in the analysis model or, if using MI, in the imputation model, and assessing the effect of this on the variance difference at follow-up. If the results suggest that dropout may be MNAR and that, consequently, the treatment effect estimate is at risk of bias, this motivates performing a sensitivity analysis to assess the robustness of the main analysis results under a plausible MNAR assumption. For example, we may observe a variance difference across trial arms in the observed data at baseline, suggesting outcome-dependent dropout, and also observe that the treatment effect estimate becomes smaller if we additionally condition on covariates that are correlated with the outcome. This would suggest outcome-dependent dropout that results in overestimation of the treatment effect, which may occur, for example, if, on average, more poor responders drop out. A natural subsequent step would involve performing a sensitivity analysis under the MNAR assumption of worse value dropout and investigating how strong this mechanism must be for the material conclusion to be affected.&quot;},&quot;conclusion&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;&quot;},&quot;front&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Randomized controlled trials (RCTs) are vulnerable to bias from missing data. When outcomes are missing not at random (MNAR), estimates from complete case analysis (CCA) and multiple imputation (MI) may be biased. There is no statistical test for distinguishing between outcomes missing at random (MAR) and MNAR. Current strategies rely on comparing dropout proportions and covariate distributions, and using auxiliary information to assess the likelihood of dropout being associated with the outcome. We propose using the observed variance difference across trial arms as a tool for assessing the risk of dropout being MNAR in RCTs with continuous outcomes. In an RCT, at randomization, the distributions of all covariates should be equal in the populations randomized to the intervention and control arms. Under the assumption of homogeneous treatment effects and homoskedastic outcome errors, the variance of the outcome will also be equal in the two populations over the course of follow-up. We show that under MAR dropout, the observed outcome variances, conditional on the variables included in the model, are equal across trial arms, whereas MNAR dropout may result in unequal variances. Consequently, unequal observed conditional trial arm variances are an indicator of MNAR dropout and possible bias of the estimated treatment effect. Heterogeneous treatment effects or heteroskedastic outcome errors are another potential cause of observing different outcome variances. We show that for longitudinal data, we can isolate the effect of MNAR outcome-dependent dropout by considering the variance difference at baseline in the same set of patients who are observed at final follow-up. We illustrate our method in simulation for CCA and MI, and in applications using individual-level data and summary data.&quot;},&quot;body&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Notation\nLet  U  be some unmeasured covariate, and  C  some measured covariate. Let  X  =  j  denote the randomized trial arms, with  j  = {0, 1}, and  X  = 0 denoting the comparator arm and  X  = 1 the intervention arm. We define the continuous outcome variable,  Y , as a linear function of  X, U , and  C  so that   with  α  some intercept,  β  the treatment effect,  γ  and  δ  the effects of  U  and  C  on  Y , respectively, and  ε Y  the mean-zero error term, with  ε Y  independent of  X, U , and  C , and with  U  and  C  additionally independent of  X .\nLet  μ 1  denote the mean of the outcome,  Y , when  X  = 1,  μ 0  the mean of  Y  when  X  = 0, with   and \nAs  C, U , and  ε Y  are independent of  X  so that, for example, E[ U | X  = 1] = E[ U | X  = 0], the mean difference across trial arms ( μ 1  −  μ 0 ) reduces to  β  and we can write: \nWe use “full data” to refer to all data that would have been observed on all trial participants, had there been no dropout. “Observed data” refer to all data for the study participants who did not drop out. We define a response indicator  R , with  R  = 1 when  Y  is observed, and  R  = 0 when  Y  is missing. Let  Y * denote the outcome in the observed data and  β * the treatment effect estimate in the observed data: \nThe bias,  B , of the CCA treatment effect estimate is given by the difference of the population treatment effect in the full data ( β ) and in the observed data ( β *): \nOptionally, the treatment effect in the observed data may be defined conditional on some observed covariate(s),  C , so that   and the bias,  B C , is given by   with   estimated in an ordinary least squares (OLS) regression of the observed outcome,  Y *, on  X  and  C .\nBy definition, for the linear model in  (1) , the population variance of  Y , for a given trial arm,  j , in the full data, is given by \nWith  U  and  C  independent of  ε Y , all covariance terms involving  ε Y  are 0, and  (8)  reduces to \nLet VD denote the outcome variance difference across trial arms in the full data:   With  Y  generated according to  (1) ,  U, C , and  ε Y  are independent of  X  so that, for example, var( U | X  = 1) = var( U | X  = 0), resulting in an expected outcome variance difference of 0 in the full data. The assumptions necessary for this to hold are discussed further in  Section 3 .\nIn the observed data, the outcome variance in a given trial arm,  j , is given by   and the variance difference across trial arms by   The exact form of  (11)  is determined by the relationship between the covariates, outcome, and  R , and is explored in  Section 3 . Again, the variance may be defined conditional on  C , with the variance difference in the full data then given by   and, in the observed data, by   We can estimate  (13)  using the studentized Breusch–Pagan estimator, detailed further in  Section 4 . In the next section ( Section 3 ), we show under which dropout mechanisms the variance difference across trial arms in the observed data can be expected to be different from 0.\nMar and Mnar Dropout and Outcome Variances Across Trial Arms in the Observed Data\nIn an RCT, patients are randomized to treatment, after which treatment is initiated and the patients are followed up over a period of time, during which dropout may occur. Randomization makes it plausible to assume that the trial arms have equal outcome variances prior to treatment initiation in the full data. Given two additional assumptions, we can expect this to hold after treatment initiation and throughout follow-up:\nAssumption A1\nThere is no treatment effect heterogeneity so that the treatment effect,  β , is the same for every individual.\nAssumption A2\nThe errors are homoskedastic so that the error term of the outcome  (1) ,  ε Y , does not depend on treatment or on  Y  itself.\nIf these two assumptions hold, then the trial arm population outcome variances can be expected to remain the same throughout follow-up in the full data. Then, it follows that if dropout is present and the trial arm outcome variances are different in the observed data, this must be due to dropout. Here, we use directed acyclic graphs (DAGs) to describe different MAR and MNAR dropout mechanisms, and show, using graphical model theory ( Mohan &amp; Pearl, 2021 ), that the trial arm variances, conditional on the model covariates, are the same when dropout is MAR, but may be different when dropout is MNAR. Additionally, we show, for an outcome,  Y  ( Equation (1) ), generated according to a linear model, that certain types of MNAR dropout do not result in a biased treatment effect estimate or different trial arm outcome variances. We define bias,  B C   (7)  with respect to a treatment effect,    (6) , estimated while adjusting for some observed baseline covariate,  C , which is a predictor of  Y  ( Equation (1) ). Analogously, we define the variance difference as the difference in trial arm outcome variances, conditional on  C , denoted as VD C   (12)  and    (13)  in the full and observed data, respectively. Let P( Y | X ,  C ) denote the density of  Y , conditional on  X  and some observed baseline covariate,  C , in the full data, and P( Y | X ,  C,R  = 1) the corresponding density in the observed data.\nProposition 1\nThe densities P( Y | X ,  C ) and P( Y | X ,  C, R ) will be identical only when dropout is MCAR or MAR, with R independent of Y given the variables included in the analysis model ( R  ⫫ Y|X, C). Any quantities derived from the densities, such as the mean difference and variance difference across X, will be also the same. If  assumptions A1  and  A2  are satisfied so that the variances of the outcome in the two trial arms are equal in the full data, then P( Y | X, C )=P( Y | X, C, R ) implies that the variances of the outcome in the two trial arms are also equal in the observed data.\nIn  Figure 1(a) , dropout is MCAR, with the response indicator,  R , unaffected by any observed or unobserved variables.  Figures 1(b)–(d)  depict MAR dropout mechanisms, with dropout dependent on treatment,  X , on some baseline covariate,  C , and on both  X  and  C , respectively. In all four scenarios,  R  is independent of  Y  given  C  and  X , and we can show that the density of  Y , conditional on  X  and  C , in the full data, is equal to the corresponding density of  Y  in the observed data: P( Y | X, C ) = P( Y | X, C,R  = 1) (proofs given in  Online Appendix A.1 ). This has the following implications. First, the outcome mean difference across trial arms conditional on  C ,    (6) , will be the same in the full and observed data so that the CCA treatment effect estimate is unbiased, with  B C  = 0. Second, if the outcome variances are equal in  X  = 1 and  X  = 0 in the full data, then the outcome variances in the observed data can also be expected to be equal across trial arms. The latter requires that  assumptions A1  and  A2  are satisfied so that that the treatment effects are homogeneous and the outcome errors homoskedastic.\nFigures 1(e)–(g)  depict MNAR dropout mechanisms, with dropout dependent on outcome,  Y , on some unobserved covariate,  U , and on treatment,  X , and  U  both. Then, for all three scenarios, we can show that  R  is not independent of  Y  given the covariates included in the analysis model, and consequently, that P( Y | X, C ) ≠ P( Y | X, C, R  = 1) (proofs given in  Online Appendix A.1 ). Different densities in the observed and full data imply that the outcome means and variances can be different also so that the CCA treatment effect estimate is biased and the outcome variance difference across trial arms nonzero. However, as we assume that  Y  ( Equation (1) ) is generated according to a linear model, an exception to this rule arises for MNAR dropout that occurs according to scenario of  Figure 1(f) .\nProposition 2\nLet the outcome, Y, be defined as in  (1) . If dropout depends on unmeasured covariate, U, and if(U ⫫  X )|R, then the outcome mean difference and variance difference across trial arms in the full data can be estimated from the observed data, without conditioning on U. If  assumptions A1  and  A2  are satisfied so that the variances of the outcome in the two trial arms are equal in the full data, this implies that the variances of the outcome in the two trial arms are also equal in the observed data.\nFor all scenarios in  Figure 1 , we assume that the measured and unmeasured covariates are independent of treatment ( C  ⫫  X  and  U  ⫫  X ), which can be expected to hold in a randomized trial setting. Under the additional assumption of homoskedastic errors ( assumption A2 ),  ε Y  ⫫  X . Then, the treatment effect estimate,  β , can be estimated by the unconditional mean difference across trial arms, as in  (2) . If we also assume that the treatment effects are homogeneous  (A1) , all the variance components in  (9)  can be expected to be equal across trial arms (e.g., var( U | X  = 0) = var( U | X  = 1)) so that outcome variance difference across trial arms in the full data, VD  (10) , is 0. If these independencies also hold in the observed data ( U  ⫫  X | R, C  ⫫  X | R , and  ε Y  ⫫  X | R ), then the unconditional treatment effect estimate in the observed data,  β *  (4) , is equal to  β   (2)  so that the bias of the CCA treatment effect estimate,  B  = 0  (5) , and the outcome variance difference across trial arms, in the observed data, VD*  (11) , is also 0.\nIn  Figure 1(f) , dropout depends on unmeasured covariate,  U . While this is an MNAR dropout mechanism, it results in an unbiased CCA treatment effect estimate and no outcome variance difference across trial arms, as  U  is independent of  X  in the observed data:  U  ⫫  X | R  (proof given in  Online Appendix A.2 ). While the same reasoning could be applied to  Figure 1(g) , where dropout is MNAR dependent on  X  and  U , this, however, would require the additional assumption that the effects of  X  and  U  on  R  are independent (e.g., sicker people drop out but equally so in both trial arms). For the purposes of this paper, we do not make this assumption, and allow  X  and  U  to interact (e.g., sicker people drop out but more so in the comparator arm). Then, in the observed data,  X  and  U  are no longer independent   Consequently,  β * will be biased ( B  ≠ 0), and VD* ≠ 0. Formulae for  β *,  B , and VD* are given in  Online Appendix A.2 (Equations (A.13), (A.14), and (A.16) , respectively). Formulae for the bias and outcome variance difference when conditioning on  C ,    (6)  and    (13)  are given in  Equations (A.15) and A.17) . Note that if  U  and  C  are related, this will additionally mean that also   Conditioning on  C  will result in attenuated estimates of the CCA estimator bias,  B C , and the outcome variance difference across trial arms in the observed data,   when compared to  B  and VD*, respectively. When  C  and  U  are independent, however, conditioning on  C  will leave both estimates unaffected.\nIn summary, when dropout is dependent only on some covariate that is either unobserved or excluded from the model, this will not, for a linear model of  Y , result in a biased CCA treatment effect estimate or in an outcome variance difference across trial arms in the observed data, even though such dropout is strictly speaking MNAR. When dropout depends on both some unmeasured covariate and  X , and they are not independent in the observed data, this may result in bias and a variance difference.  Table 1  provides an overview of when the seven dropout scenarios of  Figure 1  result in a biased CCA estimate and an outcome variance difference across trial arms in the observed data, for a linear regression of  Y  on  X  and  C .\nNote that the seven dropout scenarios of  Figure 1  and  Table 1  are illustrative settings and that we do not provide a comprehensive review of all possible settings. For example, dropout may simultaneously depend on treatment,  X  (scenario B), some observed covariate,  C , (scenario C), and on the outcome,  Y  (scenario E). If part of the dropout mechanism depends on  Y  (scenario E) or both  X  and  U  (scenario G), this is generally sufficient to cause a biased CCA estimate and an outcome variance difference across trial arms in the observed data. Under  assumptions A1  and  A2 , an outcome variance difference across trial arms serves as a marker of MNAR dropout that may result in bias. However, such MNAR dropout will not always result in an outcome variance difference, which may, because of several biases acting0. in different directions, be very small or 0. For example, dropout may depend on the outcome in such a way that the top quartile of the intervention arm and bottom quartile of the control arm drop out. Such a setting would result in a biased CCA estimate but no outcome variance difference across trial arms.\nMnar Dropout and Heterogeneous Treatment Effects in Longitudinal Data\nIn  Section 3 , we showed that outcome-dependent MNAR dropout and MNAR dropout dependent on  X  and  U , with  U  some unmeasured predictor of  Y , can result in a biased treatment effect estimate and an outcome variance difference across trial arms in the observed data, whereas MAR dropout will result in neither, subject to  assumptions A1  and  A2 . When  A1  and  A2  hold so that the treatment effects are homogeneous and the outcome errors are homoskedastic, the expected variance difference across trial arms in the full data is 0, which implies that a variance difference across trial arms in the observed data can be used as a marker of MNAR dropout and bias. Treatment effect heterogeneity and heteroskedastic outcome errors, however, will result in a nonzero variance difference across trial arms in both the full and observed data so that MNAR dropout is no longer the only potential cause of a variance difference. Treatment effect heterogeneity can be investigated by checking for the presence of an effect modifier, for example, by performing a stratified analysis. Heteroskedastic outcome errors can be investigated by exploring if the variability in the outcome at baseline is different for patients with lower and higher values. We elaborate on this in the applied example of  Section 7.1 .\nHere, we examine the implications of violating assumption  A1  through the introduction of effect modification, which will result in a nonzero expected outcome variance difference across trial arms in the full data and in the observed data (i.e., patients with outcomes at follow-up observed). We show that when this assumption is violated, for longitudinal data, where the outcome is measured in a time series, the presence of MNAR dropout can still be assessed by looking at the outcome variance difference across trial arms in the outcome measured at the baseline.\nOutcome variances across trial arms when  assumption A1  is violated\nLet  Y b  denote the outcome measured at baseline, prior to treatment initiation, which is a function of covariates  U  and  C :   with  ε b  the error term.  Y b  is unaffected by  X , and with  U  and  C  independent of  ε b , the baseline outcome variance for a given trial arm,  j , is   and the outcome variance difference across trial arms at baseline is 0:   Let  Y f  denote the outcome at follow-up, which is a function of the baseline outcome,  Y b , intervention,  X , covariates  U  and  C , and effect modifier,  S :   with  ε f  the error term, which is correlated with  ε b . In  (17) ,  S  modifies the effect of  X  on  Y f , with  ζ  the effect of  S  on the outcome at follow-up,  Y f , in the intervention arm, and the average treatment effect,  β av , given by \nFor simplicity, in  (17) , we do not specify a main effect of  S , and assume that the effect modification is limited to the intervention arm. More generally, an effect modifier can be expected to modify the outcome in both treatment arms. As for  (1) , we here assume that  X, U , and  C  are independent of  ε f , and that  U  and  C  are independent of  X ,,but allow  U  and  C  to be dependent. We make the same assumption for  S , but now assume, for simplicity, that  S  is independent of  U  and  C . The population variance of  Y f  in the full data, for the comparator arm, is then given by   and, for the intervention arm, by \nWith  U, C, ε b , and  ε f  independent of  X ,  (19)  and  (20)  result in a variance difference in the outcome at follow-up across trial arms in the full data of \nWhen  ζ  ≠ 0 in  (17)  so that  S  acts as an effect modifier,  assumption A1  is violated and the outcome variance difference across trial arms at follow-up  (21)  is nonzero. As  S  ⫫  C , the covariate-adjusted variance difference, VD f (C) =VD f . The derivations of  (19) – (21)  are given in full in  Online Appendix B.1 . Note that if we allow for a dependency between  S, U , and  C ,  (20)  and  (21)  will include additional covariance terms, with, for example, for  S  and  U , the term 2 ζ ( γ  +  γ b )cov( S ,  U | X  = 1) (see  Online Appendix B.1, Equations (B.7) and (B.8) ). Also note that while we omit a main effect of  S  in  (17) , including it will not affect the expected bias or variance difference, as  S  ⫫  X .\nIn the presence of heterogeneous treatment effects, the outcome variance difference across trial arms in the full data,  VD f   (21) , is nonzero. Then, the outcome variance difference across trial arms in the observed data,   will also be nonzero, irrespective of the dropout mechanism, and consequently, cannot be used to assess the risk of MNAR dropout. In contrast, treatment effect heterogeneity will not result in an outcome variance difference at baseline, in either the full data (VD b , 16) or the observed data   as  Y b  is not affected by treatment. Outcome-dependent dropout, however, may result in an outcome variance difference at baseline in the observed data, as the outcome errors at baseline,  ε b , and at follow-up,  ε f , are correlated. Consequently, unlike   can be used to assess the risk of MNAR dropout and, by extension, CCA estimator bias, when  assumption A1  is violated. Similarly, if  assumption A2  does not hold, for example, when the error term depends on the outcome, this will, given that a treatment effect is present, result in a variance difference across trial arms in the full data only in the outcome at follow-up and not at baseline (derivations are given in  Online Appendix B.2 ). In the simulation below, we explore the implications of violating  assumption A1  and show that, for longitudinal data, the outcome variance difference across trial arms at baseline, in the observed data, can be used to distinguish between treatment effect heterogeneity and outcome-dependent dropout.\nMethods\nWe performed a simulation study with the outcome at follow-up,  Y f , simulated according to  (17) , the outcome at baseline,  Y b , simulated according to  (14) , and with 1000 patients randomly assigned to each trial arm. The errors of  Y b  and  Y f ,  ε b  and  ε f , were drawn from a multivariate normal distribution, with variances of 1.5 and 2, respectively, and a correlation coefficient of 0.433. Dropout was simulated according to the seven mechanisms listed in  Table 1 , 2  and illustrated in  Figure 1 , under a logit model, with 28% overall dropout. Each scenario was simulated without effect modification ( ζ  = 0), with a true treatment effect  β  = 1 in the full data, and also with effect modification ( ζ  = 0.5), with the average treatment effect in the full data,  β av   (18) , also 1.\nThe outcome variance difference across trial arms in the observed data was calculated at final follow-up   and, in the same set of patients (i.e., patients with observed outcomes at follow-up), at baseline   CCA estimator bias and VD estimates were obtained conditional on observed baseline covariate,  C ,   and when additionally adjusting for   For each scenario, we obtained mean estimates of the CCA estimator bias,   and   with corresponding 95% confidence intervals (CIs). The 95% CIs were computed using the standard deviation (SD) of the relevant estimate across simulations. We simulated 1000 datasets of  N  = 2000, having verified, for each estimate, that the Monte Carlo SD (MCSD) and mean standard error were comparable, indicating that 1000 repetitions are sufficient. Additionally, we calculated the proportion of times the null was excluded from the confidence interval, as an indicator of how often a variance difference was correctly identified across simulations. A full description of the simulation framework, in accordance with ADEMP guidelines ( Morris et al., 2019 ), is given in  Online Appendix C.1 , where we describe, in detail, the simulation  a ims,  d ata-generating mechanisms,  m ethods, and  p erformance measures.\nUsing Conditional Trial Arm Outcome Variances to Evaluate Imputation Models\nIn this section, we consider the situation where there are measured covariates that are predictive of dropout and outcome, which are not included in the analysis model. In  Section 3 , we showed that for an MAR dropout mechanism, the outcome variances in the observed data are equal across trial arms, when conditioning on all variables that affect missingness. Additionally, we showed that this also holds when dropout is MNAR dependent on some unobserved variable, given that this variable is independent of treatment in the observed data. This property, in conjunction with the assumption of homogeneous treatment effects ( assumption A1 ) and homoskedastic outcome errors ( assumption A2 ), can be used to assess the plausibility of bias in a CCA analysis, by comparing the outcome variances across trial arms while conditioning on all analysis model variables. When data are missing, however, investigators may choose to use an MI approach, defining an imputation model that includes auxiliary variables that are not included in the main analysis model. In an MI model, assuming that dropout is MAR conditional on the imputation model variables and that the imputation model is correctly specified, we would expect the variance difference to be zero across the imputed datasets.\nIn this simulation study, we show that when dropout depends on some covariate,  C 2 , and  X , and  C 2  is excluded from the analysis model, the CCA treatment effect estimate is biased and there is an outcome variance difference across trial arms in the observed data. If  C 2  is included in the imputation model, however, fitting the same analysis model to the imputed datasets will result in an unbiased estimate of the treatment effect and no variance difference. Consequently, the outcome variance difference across trial arms in the imputed data can be used to assess the added value of including auxiliary variables in the imputation model.\nMethods\nWe performed a simulation study with the outcome,  Y , defined according to a linear model:  Y  =  α  +  βX  +  γ 1 C 1  +  γ 2 C 2  +  ε Y , with  C 1  and  C 2  two observed independent continuous covariates, and the remaining terms defined as in  (1)  ( Section 2 ). A total of  N  = 1000 and  N  = 10,000 patients were randomized to treatment, with a true treatment effect  β  = 1, and trial arm outcome variances of 8. Dropout was simulated according to the two dropout mechanisms shown as DAGs in  Figure 2 , defined in the same manner as in  Figure 1 , with, for example, in DAG 1,  Y  affected by treatment,  X , covariates  C 1  and  C 2 , and outcomes MAR conditional on  X  and  C 2 . Dropout was simulated under a logit mechanism, with 28% overall dropout.\nIn the observed data, we performed a CCA linear regression conditional on  C 1 . Missing outcomes were imputed conditional on  C 1 ,  C 2 ,  X , and  Y *, generating 10 complete datasets. Treatment effect estimates adjusted for  C 1  were obtained for each dataset and subsequently pooled using Rubin’s rules ( Rubin, 1996 ). The corresponding variance differences for both models were estimated conditional on  C 1 . The 95% CIs were computed using the SD of the relevant estimate across simulations. We simulated 1000 datasets for each scenario, having verified, for each estimate, that the Monte Carlo SD and mean standard error were comparable, indicating that 1000 repetitions are sufficient. A more detailed description of the simulation framework, in accordance with ADEMP guidelines ( Morris et al., 2019 ), is given in  Online Appendix D.1 .\nResults\nThe simulation results for  N  = 1000 are shown in  Table 3 . In scenario 1 ( Figure 2a ), dropout depends on  X  and  C 2 , and, consequently, is MNAR conditional on analysis model covariates  C 1  and  X , resulting in a biased CCA treatment effect estimate, with the corresponding outcome variance difference across trial arms nonzero. The dropout, however, is MAR conditional on  X  and  C 2 , and fitting the same analysis model, which regresses  Y  on  X  and  C 1 , to data imputed conditional on  Y*, X, C 1 , and  C 2 , results in an unbiased treatment effect estimate and a nearzero variance difference. In scenario 2 ( Figure 2b ), dropout is a function of  Y , in addition to  C 2  and  X , and consequently, MNAR conditional on the analysis model covariates and also MNAR conditional on the imputation model covariates. This results in a biased CCA estimate in the observed data and a biased treatment effect estimate in the imputed data, with, for both, nonzero outcome variance differences across trial arms. In  Online Appendix D.2 , a companion table is provided for  Table 3  ( Table D.1 ), which includes results for a sample size of  N  = 10,000 and various measures of simulation performance.\nBased on the outcome variance difference in the observed and imputed data, we would conclude, for scenario 1, that including variable  C 2  in the imputation model will result in a less biased estimate, while, for scenario 2, we would infer that the imputation model fails to address the dropout mechanism, suggesting that the data are MNAR given the variables in the imputation model.\nPreviously, in  Section 3 , we showed that if dropout depends on some covariate and  X , and the two are not independent in the observed data, including the covariate in the analysis model will result in an unbiased CCA estimate and no outcome variance difference across trial arms in the observed data. Here, we show, when using MI, that if this covariate is omitted from the analysis model but included in the imputation model, the resulting treatment effect estimate will be unbiased and the outcome variance difference will be zero. Consequently, the outcome variance difference across trial arms in the imputed data can be used to assess the added value of including variables in the imputation model for explaining the missingness mechanism.\nApplication\nAn application using individual-level data from the acupuncture trial\nWe now apply our method to individual-level data from an RCT, which compared the effect of two treatments on 401 patients suffering from chronic headaches ( Vickers et al., 2004 ;  Vickers, 2006 ). The primary outcome was the headache score at 12 months, with higher values indicating worse symptoms. Patients were randomly allocated to acupuncture intervention ( N  = 205) or usual care ( N  = 196). The trial found a beneficial effect of acupuncture treatment, with a mean difference in headache score of −4.6 (95% CI: −7.0, −2.2), adjusted for baseline headache score and minimization variables age, sex, headache type, number of years of headache disorder, and site (general practices in England and Wales). At 12 months, 21% and 29% of patients in the acupuncture and usual care arm, respectively, had dropped out. The investigators noted that while dropouts were generally comparable across the two arms, their baseline headache score was on average higher.\nExisting methods for assessing risk of bias due to dropout include checking if dropout is differential across trial arms ( Higgins et al., 2012 ;  Sterne et al., 2019 ), and if baseline covariate distributions are different across trial arms in patients who are still observed at the end of follow-up ( Groenwold et al., 2014 ). We assessed the relationship between trial arm and dropout by performing a logistic regression of the dropout indicator on treatment, which yielded an association of 0.38 (95% CI: −0.07, 0.84), with the CI just including the null. Note, however, that biased and unbiased treatment effect estimates can be obtained both when dropout is balanced and differential ( Bell et al., 2013 ). We compared the baseline covariate distributions across trial arms by performing linear regressions of each covariate included in the primary analysis model on treatment, using the subset of patients still observed at 12 months, with the covariates standardized to facilitate comparisons. The largest point estimate and narrowest confidence interval were observed for the headache score at baseline (0.14, 95% CI: −0.09, 0.37), though the latter also included the null ( Table 4 ). As for differential dropout, both MAR and MNAR dropout mechanisms can result in different baseline covariate distributions across trial arms, and consequently, neither method is a unique marker for the presence of MNAR dropout.\nUsing the outcome variance differences across trial arms at 12 months (VD 12 ) and at baseline (VD b ), we assessed the risk of bias due to MNAR dropout for an unadjusted CCA model (M1), a model adjusted for the minimization variables (M2), and a model adjusted for the minimization variables and baseline headache score (M3). VD 12  and VD b  were estimated using the studentized Breusch–Pagan test, for the subset of patients still observed at 12 months. Results are reported in  Table 5 .\nThe unadjusted model (M1), regressing headache score on treatment, showed a beneficial effect of acupuncture therapy (−6.1, 95% CI: −9.6, −2.6). At 12 months, the outcome variances in the acupuncture arm and usual care arm were 188.2 and 289.4, respectively, with a variance difference, VD 12 =−100.3 (95% CI: −222.0, 21.4). This result is compatible with a large negative variance difference (−222.0), with a smaller outcome variance in the acupuncture arm, and a small positive variance difference (21.4), with a smaller outcome variance in the usual care arm. At baseline, we estimated an outcome variance difference of VD b  = −81.5 (95% CI: −183.2, 20.3), which is once again compatible with a large negative variance difference and a small positive variance difference. A substantial outcome variance difference at baseline raises concerns of outcome-dependent MNAR dropout, whereas, at follow-up, an outcome variance difference may have multiple causes: MNAR dropout, heteroskedastic errors, or treatment effect heterogeneity resulting from effect modification. Adjusting for the five minimization variables (M2) did not affect the estimated treatment effect or VD 12 , whereas additionally adjusting for baseline headache score (M3) resulted in an attenuated treatment effect of −4.64 (95% CI: −7.08, −2.19) and a greatly reduced positive VD 12  of 21.23 (95% CI: −26.83, 69.30), with much tighter confidence bounds. In  Section 5.3 , we showed that when dropout is MNAR, conditioning on the outcome at baseline results in a smaller outcome variance difference at follow-up and attenuation of the CCA estimator bias.\nHowever, in the event that the outcome variance difference at follow-up is the result of treatment effect heterogeneity, conditioning on an effect modifier will also result in a decreased variance difference ( Mills et al., 2021 ). A simple way to check if a variable is an effect modifier is to perform a stratified analysis. We repeated the regression of M3 ( Table 5 ) in patients with baseline headache scores below the mean, and in patients with scores above the mean, yielding estimates of −2.92 (95% CI: −5.15, −0.69) and −6.66 (95% CI: −12.49, −0.83), respectively. The difference in treatment effect estimate in patients with lower and higher baseline scores suggests that the headache score at baseline may act as an effect modifier. For comparison, we performed an analogous analysis dividing the patients according to age, which showed no difference in treatment effect estimate between patients below mean age (−4.76; 95% CI: −8.89, −0.63) and above mean age (−4.57; 95% CI: −7.73, −1.40). This suggests that the baseline headache score may be acting as an effect modifier, which would imply that the observed outcome variance difference at 12 months may at least in part be the result of effect modification in the intervention arm.\nAn outcome variance difference at follow-up may also result from the presence of heteroskedastic outcome errors. This can be assessed by checking if variability in the outcome at baseline is different for patients with lower and higher values. We did this by ordering the baseline headache score values and dividing them into six bins. The variances across bins showed no evidence against homoskedastic outcome errors, with no corresponding increase or decrease in variance observed ( Table E.1, Online Appendix E ).\nIn summary, our results suggest that the CCA estimate, adjusted for minimization variables and baseline headache score may be biased due to outcome-dependent dropout, and that the true treatment effect may be more modest. The magnitude of this bias is, however, likely partly reduced by conditioning on the baseline headache score. The bias can be expected to be further attenuated when conditioning on additional variables that are predictors of the outcome, either by including them in the main analysis model or, if using an MI approach, in the imputation model. In the original trial publication ( Vickers et al., 2004 ), the authors additionally obtained the treatment effect estimate when imputing the 12-month dropouts using auxiliary variables that were highly correlated with the headache score at 12 months, including the headache score measured at a previous time point (3 months) and a post-hoc global assessment of headache severity. This yielded a smaller treatment effect estimate of −3.9 (95% CI: −6.4, −1.4), when compared to the CCA estimate adjusted for minimization variables and baseline headache score (−4.6; 95% CI: −7.1, −2.2). This attenuation on imputing the missing 12-month outcomes using variables highly correlated with the outcome further supports our conclusion that the CCA treatment effect estimate is likely affected by outcome-dependent dropout. This can be further investigated by performing sensitivity analyses under assumption of a dropout mechanisms that is MNAR dependent on the outcome.\nNote that observing a variance difference at baseline is sufficient to raise concern of outcome-dependent dropout. Further investigating the variance difference at follow-up, as we do here, is then not strictly necessary, though it may nevertheless be interesting for interpretation purposes. Investigating the possibility of heterogeneous treatment effects and heteroscedastic errors, however, becomes necessary when there is a non-outcome dependent MNAR dropout mechanism, which will only result in an outcome variance difference across trial arms at follow-up but not at baseline.\nAn application using summary-level data from the POPPI trial\nThe POPPI trial investigated whether a preventive, complex psychological intervention, initiated in the intensive care unit (ICU), would reduce the development of subsequent posttraumatic stress disorder (PTSD) symptoms at 6 months in ICU patients ( Wade et al., 2019 ). Symptom severity was quantified using the PTSD symptom scale-self-report (PSS-SR) questionnaire, with higher values indicating greater severity. Twenty-four ICUs were randomized to intervention or control, with intervention ICUs providing usual care during a baseline period and the preventive intervention during the intervention period, and control ICUs providing usual care throughout. At 6 months follow-up, 79.3% of patients had completed the PSS-SR questionnaire, with no difference across study arms. The trial found no beneficial effect of intervention, with a mean difference in PSS-SR score of −0.03 (95% CI: −2.58, 2.52), adjusted for age, sex, race/ethnicity, deprivation, preexisting anxiety/depression, planned admission following elective surgery, and the Intensive Care National Audit &amp; Research Centre (ICNARC) Physiology Score.\nUsing summary statistics from the published study, we performed a  t -test for the variance difference ( Mills et al., 2021 ) between trial arms at 6 months, to assess if the study’s reported null result may have been biased by MNAR dropout. Published estimates were means with 95% CIs, adjusted for the previously listed variables, which we used to calculate the outcome variances and corresponding variance differences). We found no evidence for a variance difference across trial arms at baseline (VID b , = 11.2; 95% CI: −22.7, 45.2), but a greater variance in the intervention arm at 6 months (VD 6  = 52.5; 95% CI: 18.8, 86.2).\nNo variance difference across trial arms in the outcome at baseline indicates that dropout is not outcome-dependent, whereas the variance difference at follow-up may be the result of treatment effect heterogeneity, heteroskedastic outcome errors, or MNAR dropout that does not depend on outcome, as in scenario of  Figure 1(g) , where dropout depends on some unobserved covariate,  U , and treatment. For nonoutcome-dependent MNAR dropout to result in bias and a variance difference, this requires  U  to interact with treatment in the dropout mechanism, which will result in differential dropout across trial arms. In the POPPI trial data, however, dropout was balanced at 6 months follow-up, suggesting that nonoutcome-dependent MNAR dropout is unlikely to be the cause of the observed outcome variance difference across trial arms at follow-up. An outcome variance difference at follow-up may also be the result of heteroskedastic outcome errors. This, however, requires that a treatment effect be present, whereas, here, a null treatment effect was estimated. In summary, our results suggest that there is no MNAR dropout in the POPPI trial at 6 months follow-up, but that there may be treatment effect heterogeneity, which, on average, results in a null treatment effect. Further investigation into potential effect modifiers would require access to individual-level data.\nSupplementary Material&quot;},&quot;back&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Acknowledgments\nWe would like to thank Dr Andrew Vickers and the acupuncture trial team for making the data from “Acupuncture for chronic headache in primary care: large, pragmatic, randomized trial” publicly available. AH, TP and KT were supported by the Integrative Epidemiology Unit, which receives funding from the UK Medical Research Council and the University of Bristol (MC_UU_00011/3). KHW is affiliated to the Integrative Cancer Epidemiology Programme (ICEP), and works within the Medical Research Council Integrative Epidemiology Unit.\nFunding information\nUniversity of Bristol; Medical Research Council, Grant/Award Number: MC_UU_00011/3\nData Availability Statement\nIndividual-level patient data from the acupuncture trial (ISRCTN96537534) are publicly available and can be found in the supplementary materials of  Vickers 2006 . This trial was approved by the South West Multicentre Research Ethics Committee and appropriate local ethics committees.&quot;},&quot;license&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;CC BY&quot;},&quot;retracted&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;no&quot;},&quot;last_updated&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;2024-01-16 23:47:20&quot;},&quot;citation&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Biom J. 2023 Dec 1; 65(8):e2200116&quot;},&quot;package_file&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;oa_package/98/16/PMC7615524.tar.gz&quot;}}},{&quot;rowIdx&quot;:4,&quot;cells&quot;:{&quot;accession_id&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;PMC7615526&quot;},&quot;pmid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;38160938&quot;},&quot;introduction&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Introduction\\nLysosomal acid lipase (LAL) hydrolyzes cholesteryl esters (CE) and triacylglycerols (&quot;},&quot;methods&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Materials and Methods\\nAnimals\\nMale young (8–12 or 12–17 weeks of age) and mature (40–55 wee&quot;},&quot;results&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Results\\nReduced cross-sectional area and SM mass in Lal −/− mice\\nTo study the consequences of&quot;},&quot;discussion&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Discussion\\nThe impact of LAL-D on the pathophysiology of SM is still unclear, although decreased m&quot;},&quot;conclusion&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Conclusion\\nTaken together, our data provide conclusive evidence that whole-body loss of LAL affect&quot;},&quot;front&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Objective\\nLysosomal acid lipase (LAL) is the only enzyme known to hydrolyze cholesteryl esters (CE&quot;},&quot;body&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Supplementary Material&quot;},&quot;back&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Acknowledgments\\nThis work was supported by the Austrian Science Fund FWF (SFB 10.55776/F73, DK-MCD&quot;},&quot;license&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;CC BY&quot;},&quot;retracted&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;no&quot;},&quot;last_updated&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;2024-01-16 23:47:21&quot;},&quot;citation&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Mol Metab. 2023 Dec 30;:101869&quot;},&quot;package_file&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;oa_package/db/ce/PMC7615526.tar.gz&quot;}}},{&quot;rowIdx&quot;:5,&quot;cells&quot;:{&quot;accession_id&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;PMC8550499&quot;},&quot;pmid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;34723203&quot;},&quot;introduction&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Introduction\\nAs a consequence of large-scale phenomena such as globalization, social integration, &quot;},&quot;methods&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;&quot;},&quot;results&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;&quot;},&quot;discussion&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;&quot;},&quot;conclusion&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Conclusions\\nThroughout this paper I tried to show that computational methods can play an important&quot;},&quot;front&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;In this paper I argue in favour of the adoption of an interdisciplinary approach based on computati&quot;},&quot;body&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Language policy and complexity\\nPublic policy can be defined as “ an intentional course of action&quot;},&quot;back&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Funding\\nOpen access funding provided by University of Geneva.\\nData availability\\nThe manuscript h&quot;},&quot;license&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;CC BY&quot;},&quot;retracted&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;no&quot;},&quot;last_updated&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;2024-01-16 23:35:01&quot;},&quot;citation&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;SN Soc Sci. 2021 Aug 2; 1(8):197&quot;},&quot;package_file&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;oa_package/eb/8e/PMC8550499.tar.gz&quot;}}},{&quot;rowIdx&quot;:6,&quot;cells&quot;:{&quot;accession_id&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;PMC9002620&quot;},&quot;pmid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;35406088&quot;},&quot;introduction&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;1. Introduction\\nHomocysteine (Hcy) is a sulfhydryl-containing amino acid that is produced when met&quot;},&quot;methods&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;2. Materials and Methods\\n2.1. Animals\\nMale Sprague Dawley rats (250 g, Envigo, Milan, Italy) were&quot;},&quot;results&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;3. Results\\n3.1. Effect of Cashew Nuts on Serum Hcy Levels after Methionine Administration\\nIn the &quot;},&quot;discussion&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;4. Discussion\\nHHcy is a methionine metabolism abnormality that can cause a variety of disorders in&quot;},&quot;conclusion&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;&quot;},&quot;front&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;These authors contributed equally to this work.\\nThese authors contributed equally to this work.\\nH&quot;},&quot;body&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;&quot;},&quot;back&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Acknowledgments\\nWe would like to acknowledge Salma Seetaroo from Ivorienne de Noix de Cajou S.A. o&quot;},&quot;license&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;CC BY&quot;},&quot;retracted&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;no&quot;},&quot;last_updated&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;2024-01-16 23:35:07&quot;},&quot;citation&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Nutrients. 2022 Apr 1; 14(7):1474&quot;},&quot;package_file&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;oa_package/5c/7a/PMC9002620.tar.gz&quot;}}},{&quot;rowIdx&quot;:7,&quot;cells&quot;:{&quot;accession_id&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;PMC9094045&quot;},&quot;pmid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;35573302&quot;},&quot;introduction&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Introduction\\nAcetylcholine is an important neurotransmitter for both the maintenance of internal h&quot;},&quot;methods&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;&quot;},&quot;results&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;&quot;},&quot;discussion&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;&quot;},&quot;conclusion&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Conclusion\\nIn conclusion, experimental models such as the Chrna9 KO mouse have allowed the develop&quot;},&quot;front&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Edited by: Victoria M. Bajo Lorenzana, University of Oxford, United Kingdom\\nReviewed by: Adrian Ro&quot;},&quot;body&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Auditory Efferent System\\nThe auditory efferent system is a neural network that originates in the a&quot;},&quot;back&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;&quot;},&quot;license&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;CC BY&quot;},&quot;retracted&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;no&quot;},&quot;last_updated&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;2024-01-16 23:36:46&quot;},&quot;citation&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Front Neurosci. 2022 Apr 27; 16:866161&quot;},&quot;package_file&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;oa_package/d6/a7/PMC9094045.tar.gz&quot;}}},{&quot;rowIdx&quot;:8,&quot;cells&quot;:{&quot;accession_id&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;PMC9117295&quot;},&quot;pmid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;35115661&quot;},&quot;introduction&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Introduction\\nGenetic variations at the  DLG2  gene locus are linked to multiple psychiatric disord&quot;},&quot;methods&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Materials and methods\\nAnimals and husbandry\\nAll procedures were carried out under local instituti&quot;},&quot;results&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Results\\nDlg2 + /−  heterozygous knockout rats were generated by CRISPR-Cas9 targeting of the  Dl&quot;},&quot;discussion&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Discussion\\nNMDAR currents are increased in the  Dlg2 + /−  heterozygous rat model. Additionally,&quot;},&quot;conclusion&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;&quot;},&quot;front&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Copy number variants indicating loss of function in the  DLG2  gene have been associated with marke&quot;},&quot;body&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Supplementary information\n     &quot;},&quot;back&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Supplementary information\\nThe online version contains supplementary material available at 10.1038/&quot;},&quot;license&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;CC BY&quot;},&quot;retracted&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;no&quot;},&quot;last_updated&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;2024-01-16 23:35:01&quot;},&quot;citation&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Neuropsychopharmacology. 2022 Jun 3; 47(7):1367-1378&quot;},&quot;package_file&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;oa_package/84/a0/PMC9117295.tar.gz&quot;}}},{&quot;rowIdx&quot;:9,&quot;cells&quot;:{&quot;accession_id&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;PMC9363242&quot;},&quot;pmid&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;35944931&quot;},&quot;introduction&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Background\\nDescription of the condition\\nHigh blood pressure is the leading cause of preventable d&quot;},&quot;methods&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Methods\\nCriteria for considering studies for this review\\nTypes of studies\\nThe Populations, Inter&quot;},&quot;results&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Results\\nDescription of studies\\nFor detailed information, see  Characteristics of included studies&quot;},&quot;discussion&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Discussion\\nSummary of main results\\nThis review examined the effects and safety of LSSS compared t&quot;},&quot;conclusion&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Authors' conclusions\n\n&quot;},&quot;front&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Abstract\\nBackground\\nElevated blood pressure, or hypertension, is the leading cause of preventable&quot;},&quot;body&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Summary of findings\\nObjectives\\nTo assess the effects and safety of replacing salt with LSSS to re&quot;},&quot;back&quot;:{&quot;kind&quot;:&quot;truncated&quot;,&quot;value&quot;:&quot;\&quot;Acknowledgements\\nThe World Health Organization (WHO) provided funding to Stellenbosch University t&quot;},&quot;license&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;CC BY&quot;},&quot;retracted&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;no&quot;},&quot;last_updated&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;2024-01-16 23:36:46&quot;},&quot;citation&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Cochrane Database Syst Rev. 2022 Aug 10; 2022(8):CD015207&quot;},&quot;package_file&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;oa_package/97/5a/PMC9363242.tar.gz&quot;}}}],&quot;truncated&quot;:true},&quot;paginationData&quot;:{&quot;pageIndex&quot;:0,&quot;numItemsPerPage&quot;:100,&quot;numTotalItems&quot;:212000,&quot;offset&quot;:0,&quot;length&quot;:100}},&quot;jwt&quot;:&quot;eyJhbGciOiJFZERTQSJ9.eyJyZWFkIjp0cnVlLCJwZXJtaXNzaW9ucyI6eyJyZXBvLmNvbnRlbnQucmVhZCI6dHJ1ZX0sImlhdCI6MTc0MjkyMzA1MSwic3ViIjoiL2RhdGFzZXRzL1RvbVRCVC9wbWNfb3Blbl9hY2Nlc3Nfc2VjdGlvbiIsImV4cCI6MTc0MjkyNjY1MSwiaXNzIjoiaHR0cHM6Ly9odWdnaW5nZmFjZS5jbyJ9.E_hva7Ymc1Dt_2W2flpcwgHbFVP0Hd__eqkzxhJ5YIAZjgrMa7a4jtjocFkUz9lcqbq6hkt2qX-24SzHbnLJCQ&quot;,&quot;displayUrls&quot;:true},&quot;dataset&quot;:&quot;TomTBT/pmc_open_access_section&quot;,&quot;isGated&quot;:false,&quot;isPrivate&quot;:false,&quot;hasParquetFormat&quot;:false,&quot;author&quot;:{&quot;_id&quot;:&quot;617900b59e61d7ff71b52b94&quot;,&quot;avatarUrl&quot;:&quot;/avatars/3435205ad008b625bb742afcdcd92764.svg&quot;,&quot;fullname&quot;:&quot;Tom Boissonnet&quot;,&quot;name&quot;:&quot;TomTBT&quot;,&quot;type&quot;:&quot;user&quot;,&quot;isPro&quot;:false,&quot;isHf&quot;:false,&quot;isHfAdmin&quot;:false,&quot;isMod&quot;:false,&quot;followerCount&quot;:1},&quot;compact&quot;:true}"><div class="flex flex-col overflow-hidden shadow-xs mx-auto mb-10 rounded-lg border pt-2  px-2.5"><div class="mb-2 flex flex-wrap items-center gap-2"><div class="mr-auto flex items-center"><svg class="mr-1 flex-none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 12 12"><path fill="currentColor" d="M2.5 2h7a1 1 0 0 1 1 1v6a1 1 0 0 1-1 1h-7a1 1 0 0 1-1-1V3a1 1 0 0 1 1-1Zm0 2v2h3V4h-3Zm4 0v2h3V4h-3Zm-4 3v2h3V7h-3Zm4 0v2h3V7h-3Z"></path></svg>
	<div class="whitespace-nowrap font-semibold">Dataset Viewer (First 5GB)</div>
	<div class="translate-y-px z-10"><svg class="mx-1.5 text-sm text-gray-400 hover:text-black dark:hover:text-white" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M17 22v-8h-4v2h2v6h-3v2h8v-2h-3z" fill="currentColor"></path><path d="M16 8a1.5 1.5 0 1 0 1.5 1.5A1.5 1.5 0 0 0 16 8z" fill="currentColor"></path><path d="M16 30a14 14 0 1 1 14-14a14 14 0 0 1-14 14zm0-26a12 12 0 1 0 12 12A12 12 0 0 0 16 4z" fill="currentColor"></path></svg>
	</div></div>
				<a href="/datasets/TomTBT/pmc_open_access_section/tree/refs%2Fconvert%2Fparquet/commercial" class="group mr-1 text-xs text-gray-400 max-sm:hidden"><svg class="text-[.6rem] mr-1 inline -translate-y-px" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path fill="currentColor" d="M12 10H6.78A11 11 0 0 1 27 16h2A13 13 0 0 0 6 7.68V4H4v8h8zm8 12h5.22A11 11 0 0 1 5 16H3a13 13 0 0 0 23 8.32V28h2v-8h-8z"></path></svg>
						<span class="underline decoration-gray-300 group-hover:decoration-gray-400 dark:decoration-gray-500 dark:group-hover:decoration-gray-300">Auto-converted</span> to Parquet
					</a>
				<button class="btn shadow-xs flex cursor-pointer items-center rounded-sm border px-1 py-0.5 text-xs font-normal text-gray-700 hover:text-gray-800 hover:shadow-inner dark:hover:text-gray-200"><svg class="mr-1.5" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32" style="transform: rotate(360deg);"><path d="M31 16l-7 7l-1.41-1.41L28.17 16l-5.58-5.59L24 9l7 7z" fill="currentColor"></path><path d="M1 16l7-7l1.41 1.41L3.83 16l5.58 5.59L8 23l-7-7z" fill="currentColor"></path><path d="M12.419 25.484L17.639 6l1.932.518L14.35 26z" fill="currentColor"></path></svg>API</button>
					<button class="btn shadow-xs flex cursor-pointer items-center rounded-sm border px-1 py-0.5 text-xs font-normal text-gray-700 hover:text-gray-800 hover:shadow-inner dark:hover:text-gray-200"><svg class="mr-1.5" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" fill="currentColor" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 12 12"><path d="M9.80603 2.86737H3.56107C3.37704 2.86737 3.20055 2.94048 3.07042 3.0706C2.94029 3.20073 2.86719 3.37723 2.86719 3.56126V9.80622C2.86719 9.99025 2.94029 10.1667 3.07042 10.2969C3.20055 10.427 3.37704 10.5001 3.56107 10.5001H9.80603C9.99006 10.5001 10.1666 10.427 10.2967 10.2969C10.4268 10.1667 10.4999 9.99025 10.4999 9.80622V3.56126C10.4999 3.37723 10.4268 3.20073 10.2967 3.0706C10.1666 2.94048 9.99006 2.86737 9.80603 2.86737Z" fill="currentColor" fill-opacity="0.3"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M2.40942 1.66191C2.05175 1.66191 1.7618 1.95186 1.7618 2.30953V6.76191H1.43799V2.30953C1.43799 1.77303 1.87291 1.3381 2.40942 1.3381H6.45704V1.66191H2.40942Z" fill="currentColor"></path></svg>Embed</button>

					<button class="bg-linear-to-b shadow-xs flex items-center gap-1.5 rounded-full border from-white to-red-100/90 px-2 py-0.5 text-xs font-medium text-[#2D3648] transition-shadow hover:shadow-inner dark:from-gray-900 dark:to-red-800/30 dark:text-gray-100 dark:hover:shadow-inner dark:hover:shadow-red-800/30" ><svg class="h-3.5 w-3.5 text-red-500" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 12 12"><path fill="currentColor" d="M2.5 2h7a1 1 0 0 1 1 1v6a1 1 0 0 1-1 1h-7a1 1 0 0 1-1-1V3a1 1 0 0 1 1-1Zm0 2v2h3V4h-3Zm4 0v2h3V4h-3Zm-4 3v2h3V7h-3Zm4 0v2h3V7h-3Z"></path></svg>
						<span>Data Studio</span></button></div>
		<div class="flex flex-1 flex-col overflow-hidden -mx-2.5"><div class="flex flex-1 flex-col overflow-hidden"><div class="flex min-h-0 flex-1"><div class="flex flex-1 flex-col overflow-hidden"><div class="md:-mx-2.5 flex min-w-0 flex-wrap border-t"><div class="flex min-w-0 flex-1 flex-wrap"><div class="grid flex-1 grid-cols-1 overflow-hidden text-sm md:grid-cols-2 md:place-content-center sm:mx-2.5"><label class="relative block flex-1 px-3 py-2 hover:bg-gray-50 dark:border-gray-850 dark:hover:bg-gray-950 md:border-r max-md:first:border-b md:border-r" title="commercial"><span class="text-gray-500">Subset (3)</span>
			<div class="flex items-center whitespace-nowrap"><span class="truncate">commercial</span>
				<span class="mx-2 text-gray-500">·</span>
					<span class="text-gray-500">~4.31M rows (showing the first 212k)</span>
				<svg class="ml-auto min-w-6 pl-2" width="1em" height="1em" viewBox="0 0 12 7" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M1 1L6 6L11 1" stroke="currentColor"></path></svg></div>
			<select class="absolute inset-0 z-10 w-full cursor-pointer border-0 bg-white text-base opacity-0"><optgroup label="Subset (3)"><option value="commercial" selected>commercial (~4.31M rows, showing the first 212k)</option><option value="non_commercial" >non_commercial (~1.46M rows, showing the first 254k)</option><option value="other" >other (~1.46M rows, showing the first 254k)</option></optgroup></select></label>
		<label class="relative block flex-1 px-3 py-2 hover:bg-gray-50 dark:border-gray-850 dark:hover:bg-gray-900 md:border-r md:border-r-0" title="train"><div class="text-gray-500">Split (1)</div>
				<div class="flex items-center overflow-hidden whitespace-nowrap"><span class="truncate">train</span>
					<span class="mx-2 text-gray-500">·</span>
						<span class="text-gray-500">~4.31M rows (showing the first 212k)</span>
					<svg class="ml-auto min-w-6 pl-2" width="1em" height="1em" viewBox="0 0 12 7" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M1 1L6 6L11 1" stroke="currentColor"></path></svg></div>
				<select class="absolute inset-0 z-10 w-full cursor-pointer border-0 bg-white text-base opacity-0"><optgroup label="Split (1)"><option value="train" selected>train (~4.31M rows, showing the first 212k)</option></optgroup></select></label></div></div>
								</div>

							<div class="flex min-h-0 flex-1 flex-col ">
	<div class="bg-linear-to-r text-smd relative flex items-center dark:border-gray-900 dark:bg-gray-950 false border-t [&amp;:has(:focus)]:from-gray-50 [&amp;:has(:focus)]:to-transparent [&amp;:has(:focus)]:to-20% dark:[&amp;:has(:focus)]:from-gray-900"><form class="flex-1"><svg class="absolute left-3 top-1/2 transform -translate-y-1/2 pointer-events-none text-gray-400" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M30 28.59L22.45 21A11 11 0 1 0 21 22.45L28.59 30zM5 14a9 9 0 1 1 9 9a9 9 0 0 1-9-9z" fill="currentColor"></path></svg>
				<input disabled class="outline-hidden h-9 w-full border-none bg-transparent px-1 pl-9 pr-3 placeholder:text-gray-400 " placeholder="Search this dataset" dir="auto"></form>
			<div class="flex items-center gap-2 px-2 py-1"><button type="button" class="hover:bg-yellow-200/70 flex items-center gap-1 rounded-md border border-yellow-200 bg-yellow-100 pl-0.5 pr-1 text-[.8rem] leading-normal text-gray-700 dark:border-orange-500/25 dark:bg-orange-500/20 dark:text-gray-300 dark:hover:brightness-110 hidden"><div class="rounded-sm bg-yellow-300 px-1 font-mono text-[.7rem] font-bold text-black dark:bg-yellow-700 dark:text-gray-200">SQL
	</div>
	Console
</button></div></div>


<div class="flex flex-1 flex-col overflow-hidden min-h-64 border-t">
		

<div class="max-h-96 relative overflow-auto"><table class="w-full table-auto rounded-lg font-mono text-xs text-gray-900"><thead class="shadow-xs sticky left-0 right-0 top-0 z-1 bg-white align-top"><tr class="space-y-54 h-full min-w-fit divide-x border-b text-left"><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">accession_id
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="25" width="42" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="44" y="0" width="42" height="30" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="88" y="17.182019131582486" width="42" height="12.817980868417512" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="44" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="43" y="0" width="44" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="87" y="0" width="44" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">9</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">11</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">pmid
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="21.11358385591023" width="14.5" height="8.886416144089774" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="16.5" y="26" width="14.5" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="33" y="26" width="14.5" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="49.5" y="25" width="14.5" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="66" y="25" width="14.5" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="82.5" y="25" width="14.5" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="99" y="24.529079636003583" width="14.5" height="5.470920363996417" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="115.5" y="0" width="14.5" height="30" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="16.5" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="15.5" y="0" width="16.5" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="32" y="0" width="16.5" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="48.5" y="0" width="16.5" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="65" y="0" width="16.5" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="81.5" y="0" width="16.5" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="98" y="0" width="16.5" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="114.5" y="0" width="16.5" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">1</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">8</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">introduction
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="0" width="11.2" height="30" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="13.2" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="26.4" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="39.599999999999994" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="52.8" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="66" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="79.19999999999999" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="92.39999999999999" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="105.6" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="118.8" y="25" width="11.2" height="5" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="12.2" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="25.4" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="38.599999999999994" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="51.8" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="65" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="78.19999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="91.39999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="104.6" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="117.8" y="0" width="13.2" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">0</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">134k</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">methods
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="0" width="11.2" height="30" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="13.2" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="26.4" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="39.599999999999994" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="52.8" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="66" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="79.19999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="92.39999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="105.6" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="118.8" y="25" width="11.2" height="5" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="12.2" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="25.4" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="38.599999999999994" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="51.8" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="65" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="78.19999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="91.39999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="104.6" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="117.8" y="0" width="13.2" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">0</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">208k</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">results
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="0" width="11.2" height="30" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="13.2" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="26.4" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="39.599999999999994" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="52.8" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="66" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="79.19999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="92.39999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="105.6" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="118.8" y="25" width="11.2" height="5" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="12.2" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="25.4" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="38.599999999999994" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="51.8" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="65" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="78.19999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="91.39999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="104.6" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="117.8" y="0" width="13.2" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">0</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">357k</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">discussion
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="0" width="11.2" height="30" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="13.2" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="26.4" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="39.599999999999994" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="52.8" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="66" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="79.19999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="92.39999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="105.6" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="118.8" y="25" width="11.2" height="5" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="12.2" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="25.4" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="38.599999999999994" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="51.8" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="65" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="78.19999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="91.39999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="104.6" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="117.8" y="0" width="13.2" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">0</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">357k</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">conclusion
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="0" width="11.2" height="30" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="13.2" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="26.4" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="39.599999999999994" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="52.8" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="66" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="79.19999999999999" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="92.39999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="105.6" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="118.8" y="25" width="11.2" height="5" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="12.2" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="25.4" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="38.599999999999994" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="51.8" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="65" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="78.19999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="91.39999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="104.6" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="117.8" y="0" width="13.2" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">0</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">58.3k</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">front
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="0" width="11.2" height="30" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="13.2" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="26.4" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="39.599999999999994" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="52.8" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="66" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="79.19999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="92.39999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="105.6" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="118.8" y="25" width="11.2" height="5" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="12.2" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="25.4" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="38.599999999999994" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="51.8" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="65" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="78.19999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="91.39999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="104.6" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="117.8" y="0" width="13.2" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">0</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">30.9k</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">body
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="0" width="11.2" height="30" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="13.2" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="26.4" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="39.599999999999994" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="52.8" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="66" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="79.19999999999999" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="92.39999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="105.6" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="118.8" y="25" width="11.2" height="5" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="12.2" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="25.4" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="38.599999999999994" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="51.8" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="65" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="78.19999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="91.39999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="104.6" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="117.8" y="0" width="13.2" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">0</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">573k</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">back
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="0" width="11.2" height="30" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="13.2" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="26.4" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="39.599999999999994" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="52.8" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="66" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="79.19999999999999" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="92.39999999999999" y="25" width="11.2" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="105.6" y="26" width="11.2" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="118.8" y="25" width="11.2" height="5" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="12.2" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="25.4" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="38.599999999999994" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="51.8" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="65" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="78.19999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="91.39999999999999" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="104.6" y="0" width="13.2" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="117.8" y="0" width="13.2" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">0</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">126k</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">license
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">classes</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><defs><clipPath id="rounded-bar"><rect x="0" y="0" width="130" height="8" rx="4"></rect></clipPath><pattern id="hatching" patternUnits="userSpaceOnUse" patternTransform="rotate(-45)" height="1" width="5"><line y1="0" class="stroke-gray-400 dark:stroke-gray-500/80" stroke-width="3" y2="1" x1="2" x2="2"></line></pattern><pattern id="hatching-faded" patternUnits="userSpaceOnUse" patternTransform="rotate(-45)" height="1" width="5"><line y1="0" class="stroke-gray-100 dark:stroke-gray-500/20" stroke-width="3" y2="1" x1="2" x2="2"></line></pattern></defs><g height="8" style="transform: translateY(20px)" clip-path="url(#rounded-bar)"><g style="transform: scaleX(1.0153846153846153) translateX(-1px)"><g><rect class="fill-indigo-500 dark:fill-indigo-600/80" x="1" y="0" width="119.05943396226415" height="8" fill-opacity="1"></rect><rect class="fill-indigo-500 dark:fill-indigo-600/80" x="122.05943396226415" y="0" width="6.838160377358491" height="8" fill-opacity="1"></rect></g></g></g><g style="transform: scaleX(1.0153846153846153) translateX(-1px)"><g><rect class="fill-white cursor-pointer" x="0" y="0" width="121.05943396226415" height="28" fill-opacity="0"></rect><rect class="fill-white cursor-pointer" x="121.05943396226415" y="0" width="8.838160377358491" height="28" fill-opacity="0"></rect><rect class="fill-white cursor-pointer" x="129.89759433962263" y="0" width="0.08033018867924528" height="28" fill-opacity="0"></rect><rect class="fill-white cursor-pointer" x="129.97792452830188" y="0" width="0.02207547169811321" height="28" fill-opacity="0"></rect></g></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 max-w-full overflow-hidden text-ellipsis whitespace-nowrap">4
				values</div></div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">retracted
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">classes</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><defs><clipPath id="rounded-bar"><rect x="0" y="0" width="130" height="8" rx="4"></rect></clipPath><pattern id="hatching" patternUnits="userSpaceOnUse" patternTransform="rotate(-45)" height="1" width="5"><line y1="0" class="stroke-gray-400 dark:stroke-gray-500/80" stroke-width="3" y2="1" x1="2" x2="2"></line></pattern><pattern id="hatching-faded" patternUnits="userSpaceOnUse" patternTransform="rotate(-45)" height="1" width="5"><line y1="0" class="stroke-gray-100 dark:stroke-gray-500/20" stroke-width="3" y2="1" x1="2" x2="2"></line></pattern></defs><g height="8" style="transform: translateY(20px)" clip-path="url(#rounded-bar)"><g style="transform: scaleX(1.0153846153846153) translateX(-1px)"><g><rect class="fill-indigo-500 dark:fill-indigo-600/80" x="1" y="0" width="127.88471698113207" height="8" fill-opacity="1"></rect></g></g></g><g style="transform: scaleX(1.0153846153846153) translateX(-1px)"><g><rect class="fill-white cursor-pointer" x="0" y="0" width="129.88471698113207" height="28" fill-opacity="0"></rect><rect class="fill-white cursor-pointer" x="129.88471698113207" y="0" width="0.11528301886792452" height="28" fill-opacity="0"></rect></g></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 max-w-full overflow-hidden text-ellipsis whitespace-nowrap">2
				values</div></div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">last_updated
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="0" width="130" height="30" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="132" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">19</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">19</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">citation
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="25" width="12.666666666666666" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="14.666666666666666" y="19.222234724879037" width="12.666666666666666" height="10.777765275120963" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="29.333333333333332" y="0" width="12.666666666666666" height="30" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="44" y="15.97772026555643" width="12.666666666666666" height="14.02227973444357" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="58.666666666666664" y="24.47710138404411" width="12.666666666666666" height="5.52289861595589" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="73.33333333333333" y="25" width="12.666666666666666" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="88" y="23.607966692922247" width="12.666666666666666" height="6.392033307077754" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="102.66666666666666" y="25" width="12.666666666666666" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="117.33333333333333" y="25" width="12.666666666666666" height="5" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="13.666666666666666" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="28.333333333333332" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="43" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="57.666666666666664" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="72.33333333333333" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="87" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="101.66666666666666" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="116.33333333333333" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">14</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">94</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">package_file
				<form class="flex flex-col"><button id="asc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="-rotate-180 transform text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button>
						<button id="desc" class="-mr-1 ml-2 h-[0.4rem] w-[0.8rem] transition ease-in-out"><svg class="text-gray-300 hover:text-gray-500" xmlns="http://www.w3.org/2000/svg" viewBox="0 64 256 128" fill="currentColor" aria-hidden="true"><path d="M213.65674,101.657l-80,79.99976a7.99945,7.99945,0,0,1-11.31348,0l-80-79.99976A8,8,0,0,1,48,88H208a8,8,0,0,1,5.65674,13.657Z"></path></svg></button></form></div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span><span class="italic text-gray-400 before:mx-1 before:content-['·']">lengths</span></div></div>

		<div><div class="" style="height: 40px; padding-top: 2px"><svg width="130" height="28"><g><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="0" y="25" width="12.666666666666666" height="5" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="14.666666666666666" y="26" width="12.666666666666666" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="29.333333333333332" y="26" width="12.666666666666666" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="44" y="26" width="12.666666666666666" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="58.666666666666664" y="26" width="12.666666666666666" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="73.33333333333333" y="26" width="12.666666666666666" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="88" y="26" width="12.666666666666666" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="102.66666666666666" y="26" width="12.666666666666666" height="4" fill-opacity="1"></rect><rect class="fill-gray-400 dark:fill-gray-500/80" rx="2" x="117.33333333333333" y="0" width="12.666666666666666" height="30" fill-opacity="1"></rect></g><rect class="fill-white dark:fill-gray-900" x="0" y="26" width="130" height="2" stroke-opacity="1"></rect><line class="stroke-gray-100 dark:stroke-gray-500/20" x1="0" y1="27.5" x2="130" y2="27.5" stroke-opacity="1"></line><g><rect class="fill-indigo-500 cursor-pointer" x="-1" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="13.666666666666666" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="28.333333333333332" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="43" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="57.666666666666664" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="72.33333333333333" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="87" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="101.66666666666666" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect><rect class="fill-indigo-500 cursor-pointer" x="116.33333333333333" y="0" width="14.666666666666666" height="30" fill-opacity="0"></rect></g></svg>
	<div class="relative font-light text-gray-400" style="height: 10px; width: 130px;"><div class="absolute left-0 overflow-hidden text-ellipsis whitespace-nowrap" style="max-width: 60px">0</div>
			<div class="absolute overflow-hidden text-ellipsis whitespace-nowrap" style="right: 0px; max-width: 60px">35</div>
			</div></div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th></tr></thead>
			<tbody class="h-16 overflow-scroll"><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="0"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">PMC6989633</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">31996712</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Introduction
The composition of Australia’s theropod fauna is poorly understood in comparison to those of contemporaneous assemblages around the world, due primarily to the isolated and fragmentary mode of preservation in fossiliferous deposits. To date, the majority of documented theropod remains from Australia are from the ‘mid’-Cretaceous (Albian–Cenomanian) and pertain predominantly to megaraptorids 1 – 7 , an exclusively Gondwanan clade of theropods initially interpreted as a member of Allosauroidea 2 . However, recent hypotheses have suggested alternative positions for megaraptorids within Tyrannosauroidea 8 – 11  or close to the base of Coelurosauria 12 , 13 . Despite the preponderance of megaraptorids in ‘mid’-Cretaceous Australia, a diverse high palaeo-latitude (approximately 60 degrees south) theropod fauna has been hypothesised within the upper Barremian–lower Albian deposits on the south coast of Victoria, including megaraptorans 3 , 5 , 14 , ceratosaurs 15 , spinosaurids 16 , tyrannosauroids 3 , 17 , possible unenlagiine dromaeosaurids and indeterminate maniraptoriforms 3 .
While members of Avetheropoda were undoubtedly present during the Cretaceous of Australia, the evidence for Ceratosauria in Australia is presently very limited, despite their abundance in the diverse Patagonian theropod fossil record 8 . The first suggested Australian ceratosaur came not from the better known Cretaceous sites in eastern Australia, but from the Middle Jurassic Colalura Sandstone of Western Australia.  Ozraptor subotaii  was described from a distal tibia characterised by a depressed and subdivided facet for the ascending process of the astragalus 18 . Examination of the tibial fragment failed to identify any convincing similarities with any theropod known at the time, and thus  Ozraptor  was referred to as an indeterminate theropod 17 . Subsequently, the description of abelisauroid remains from the Late Jurassic of Africa included tibiae that also had astragalar articular surfaces similar to that of  Ozraptor . On this basis, it was suggested that the Australian tibia represented a member of Abelisauroidea 19 . This interpretation was maintained in a reassessment of a theropod distal tibia from the Middle Jurassic of England 20 , which concluded that a depressed and subdivided facet for the astragalar ascending process was a synapomorphy of Abelisauroidea. However, this character was subsequently recognised in theropods outside of Abelisauroidea and therefore could not be considered as an abelisauroid synapomorphy 21 . As a consequence there was no convincing evidence to support abelisauroid affinities for  Ozraptor . The current consensus is that  Ozraptor  is too incomplete for referral to any theropod clade 21 , 22 .
There has also been suggestion that  Kakuru kujani , known from a partial tibia from the Aptian Marree Formation of South Australia 23  pertains to an abelisauroid based on the presence of a vertical median ridge on the distal tibia 24 . For the reasons stated above, this evidence is insufficient for referral of  Kakuru  to Abelisauroidea; subsequent revisions of this material concluded that  Kakuru  could only be referred to an indeterminate position within either Averostra or Tetanurae 25 , 26 .
More recently, a left astragalocalcaneum from the upper Barremian–lower Albian San Remo Member of the upper Strzelecki Group on the south coast of Victoria was described (Museum Victoria, Melbourne, Australia; NMV P221202, Fig.  1 ) and referred to Ceratosauria, based among other features on the co-ossification of the astragalus and calcaneum, a parallel-sided base of the ascending process of the astragalus, and a fossa at the base of the ascending process that is not associated with a transverse groove 15 . However, it was subsequently suggested that the evidence for referral of NMV P221202 to Ceratosauria was weak, and that it could only be considered as an indeterminate averostran at best 8 .
Here, we present new evidence for the presence of ceratosaurian theropods from the Cenomanian Griman Creek Formation of Lightning Ridge, New South Wales. We also reappraise the evidence against the ceratosaurian interpretation of the specimen NMV P221202 8  with the objective of clarifying and elucidating its phylogenetic position.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Methods and Materials
LRF 3050.AR and NMV P221202 were inserted into a recently published ceratosaurian phylogenetic matrix 39  (see Supplementary Dataset  S1 ) and analysed with equal weights parsimony in TNT 1.5 93 . A driven search strategy was implemented to calculate optimal trees, with each search using 100 replicates of random sectorial searches, each with 30 rounds of drifting, 5 rounds of tree fusing and 50 ratcheting cycles. The analysis was halted after two such successive searches returned shortest trees of the same length.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class=""></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Discussion
Comparisons of LRF 3050.AR
Opisthocoelous vertebral centra characterise the cervical series of many neotheropods. The posterior surfaces are typically moderately to strongly concave and the anterior surface may be generally flattened 49 , 50  or slightly convex as in ceratosaurians 51 – 54  and basal tetanurans 55 , or form a well-defined projection as in abelisaurids 30 , 56 , megalosauroids 57 – 59 , allosauroids 49 , 60 , 61 , megaraptorids 9 , 62  and alvarezsaurids 63 – 65 . In addition, opisthocoely continues into the anterior dorsal series in megalosauroids 57 , allosauroids 66 , 61 , megaraptorids 62 , and alvarezsaurids 63 . This differs from the condition in  Dilophosaurus wetherilli  and abelisauroids in which the anterior cervical centra are typically weakly opisthocoelous and transition along the series to amphicoelous in the most posterior cervicals and anterior dorsals 50 – 52 , 54 , 67 , 68 . All preserved mid-posterior cervical centra of  Elaphrosaurus  are amphicoelous 41 . Following these observations, the amphicoelous centrum and reduced inclination of the articular surfaces of LRF 3050.AR indicates a placement in the middle or posterior region of the neck. The distortion of the centrum, in particular the exaggerated offset of the articular surfaces resulting from taphonomic compression, precludes a more accurate placement of the centrum.
Among ceratosaurs, the dimensions of LRF 3050.AR are most similar to the anterior cervical series of the abelisaurid  Viavenator exxoni . However, as noted above, the anterior cervical series in  Viavenator  and other abelisaurids consists of opisthocoelous centra, contrary to the amphicoelous condition in LRF 3050.AR. Unfortunately, direct comparisons of the centrum proportions of LRF 3050.AR are complicated by the strong taphonomic dorsoventral compression of the specimen. However, when the anterior half of the cervical centra are excluded, the dimensions of LRF 3050.AR are more similar to the moderately elongate proportions of noasaurids 41 , 68 , 69  than the more robust and anteroposteriorly shortened centra in abelisaurids 51 , 52  or strongly elongate centra in  Elaphrosaurus 41 . The anterior and posterior articular surfaces are considerably wider mediolaterally than dorsoventrally tall (Table  1 ). This is similar to the proportions throughout the cervical series of  Masiakasaurus   knopfleri  and  Elaphrosaurus 41 , 67 , 68 , but may have been exaggerated by taphonomic distortion.
The preserved floor of the neural canal on the dorsal surface of LRF 3050.AR indicates that it was relatively wide mediolaterally relative to the width of the centrum and were likely wider than the thickness of the walls of the laterally bounding neural arch pedicels (Fig.  2 ). The neural canals in the cervicals of basal neotheropods and most ceratosaurs are narrower with respect to both the centrum and the neural arch pedicels 48 , 50 , 52 , 68 . In contrast, the neural canals of  Elaphrosaurus 41  and noasaurids 53 , 68 , 69  are considerably wider relative to the centrum and wider than the thickness of the walls of the neural arch pedicels, as seen in LRF 3050.AR.
The distinct posterior centrodiapophyseal lamina (pcdl) of LRF 3050.AR is remarkably similar to those of noasaurids (Fig.  3 ). In MACN-PV (Museo Argentino de Ciencias Naturales “Bernardino Rivadavia”, Buenos Aires, Argentina) 622, a cervical vertebra initially described as an oviraptorosaur 70 , 71  but which most likely pertains to  Noasaurus 53 , the pcdl narrows abruptly from the anteriorly placed diapophyses and contacts the centrum at approximately the anteroposterior midpoint (Fig.  3 ). A similar pcdl also appears to have been present in GSI (Geological Survey of India, Kolkata, India) K20/614, a cervical vertebra ascribed to the Indian noasaurid  Laevisuchus indicus 72 . The plesiomorphic condition of a posteriorly contacting pcdl is present in the middle cervicals of  Dilophosaurus 50 , abelisauroids 30 , 34 , 54  and also the recently described Brazilian noasaurid  Vespersaurus paranaensis 69 . Despite the loss of the posterior portion of the posterior centrodiapophyseal lamina, a medial attachment of the pcdl is most likely to have been present in LRF 3050.AR. A medially positioned pcdl also characterises the middle to posterior cervical series of other ceratosaurs, including  Elaphrosaurus 41 ,  Majungasaurus crenatissimus 52  and  Carnotaurus sastrei 51 .
Perhaps the most distinguishing feature of LRF 3050.AR is the mediolaterally concave ventral surface of the centrum delimited by pronounced ventrolateral ridges. In most ceratosaurs, the ventral surface of the cervical centra is flattened or slightly convex, forming a distinct edge at the contact with the lateral surfaces 51 , 68 , 73 . Ventrolateral ridges on cervical centra such as those present in LRF 3050.AR have been reported only in the basal ceratosaurian  Elaphrosaurus  and the noasaurid  Noasaurus 41 , 53 . In  Elaphrosaurus , the sharp lateroventrally directed ridges are present only at the posterior part of the centrum 41 , which differs from the condition in LRF 3050.AR in which they are continuous with the parapophysis and extend along almost the entire length of the centrum. Similar ventrolateral ridges have also been reported in MACN-PV 622 53 . Ventrolateral ridges have been described in therizinosaurs and unenlagiine dromaeosaurids 74 – 76 ; however, they are developed only as comparatively weaker and rounded ridges that do not form the sharp edges that are seen in ceratosaurians. In addition, in unenlagiines the ventrolateral ridges transition into well-developed carotid processes at the anterior end of the centra 76 , 77 . This contrasts with the condition in LRF 3050.AR in which carotid processes are absent and the ridges remain sharply defined and contact the parapophyses at the anteroventral margins of the anterior articular surface.
Status of NMV P221202
A ceratosaurian astragalocalcaneum (NMV P221202) was discovered from the upper Barremian–lower Aptian San Remo Member of the upper Strzelecki group in Victoria 15  (Fig.  4 ). NMV P221202 was compared to the only Australian theropod astragali known at the time, namely those of the megaraptorid  Australovenator wintonensis 1  and the Australian pygmy ‘ Allosaurus ’ 78 , now considered to also pertain to Megaraptoridae 2 , 8 . The Victorian astragalocalcaneum, NMV P221202, was found to differ from the two Australian megaraptorid astragali, most notably in the co-ossification of the astragalus and calcaneum, the absence of a horizontal vascular groove on the anterior surface of the astragalar body, and the lack of a crescentic groove on the posterior surface of the ascending process 15 . NMV P221202 was referred to Ceratosauria in a phylogenetic analysis, but possible ingroup relationships were not considered with confidence despite similarities with the astragalus of the Madagascan noasaurid  Masiakasaurus 15 .
Subsequently, the assignment of NMV P221202 to Ceratosauria was questioned 8  on the basis of five observations: the presence of a distinct eminence on the medial surface of the ascending process and paired oval fossae at the base of the ascending process of the astragalus anteriorly (Fig.  4a ), both of which are present in alvarezsaurids 79 ; a vertical groove on the posterior surface of the ascending process and a lateral constriction of the tibial facet caused by a thickening of the ascending process laterally (Fig.  4c ), both of which are present in megaraptorids; and a prominent posterodorsal notch on the calcaneum for articulation of the tibia (Fig.  4b ), which they considered to be a tetanuran synapomorphy based on the results of a phylogenetic analysis of tetanurans 80 . Based on these observations, it was concluded that NMV P221202 could only be considered an indeterminate averostran 8 . The debate surrounding the affinities of NMV P221202 was commented on briefly in a review of the Victorian Cretaceous polar biota 81 , with no preference stated for either of the two hypotheses.
However, a detailed consideration of these arguments as presented raises a number of problems. Firstly, as previously noted 8 , the ascending process of the astragalus in alvarezsaurids differs markedly from the condition present in NMV P221202. As is typical for coelurosaurs, the base of the ascending process in alvarezsaurids occupies almost the entire width of the astragalus 63 , 79 . Furthermore, in alvarezsaurids with the exception of  Patagonykus puertai , the medial surface of the ascending process is excavated by a deep notch, leaving only a low medial portion of the ascending process and a taller narrow lateral portion 63 , 65 , 82 – 84 . However, in NMV P221202 the ascending process is parallel-sided at the base, was likely subrectangular in its original form, and its base spans only the lateral two-thirds of the astragalus. In addition, contrary to previous remarks 8 , no medial eminence of the ascending process that resembles that of NMV P221202 is present in either  Patagonykus  or  Mononykus olecranus . In the former taxon, the medial edge of the ascending process is smoothly sinusoidal in anterior view with no noticeable eminences 79 , whereas the medial edges of the medially-notched ascending processes of  Mononykus  and other alvarezsaurids are straight or slightly concave, with no noticeable eminences 63 , 65 . Secondly, as noted in the original description of NMV P221202 15 , and contrary to previous observations 8 , there is no groove on the posterior surface of the ascending process similar to those that have been reported in megaraptorids. The lateral edge of the posterior surface of the base of the ascending process in NMV P221202 is slightly elevated with respect to the area immediately lateral to an abraded area of periosteum that may have given the appearance of a grooved surface. However, this is markedly different from the well-defined crescentic groove present on the posterior surface of the ascending process in megaraptorid astragali 1 , 14 , 78 . Thirdly, the lateral side of the tibial facet of the astragalus in the abelisaurid  Majungasaurus  is also constricted relative to the medial side 85 , indicating that this feature is not restricted to megaraptorids as previously asserted 8  and that abelisauroid affinities cannot be dismissed. Finally, tibial facets on the calcaneum have been observed in  Dilophosaurus ,  Majungasaurus ,  Elaphrosaurus ,  Ceratosaurus  and  Masiakasaurus 41 , 67 , 85 , 86 , indicating that this feature is diagnostic of Averostra, a more inclusive group than stated previously 8 .
Phylogenetic analysis
The phylogenetic analysis including LRF 3050.AR and NMV P221202 (see Methods and Materials for details) returned 217 most parsimonious trees of 4293 steps (CI: 0.306, RI: 0.512). The strict consensus tree resolves both Australian specimens within Noasauridae (Fig.  5 ). The synapomorphies diagnosing Noasauridae include a spur on the medial surface of the ascending process of the astragalus (858:1), mediolaterally thin cervical epipophyses (1272:1), cervical postzygapophyses swept back posteriorly and surpassing the posterior end of the vertebral centra (1083:1), smooth medial surfaces of the anteromedial process of the maxilla (915:0), anteroposteriorly shortened palatal shelves of the maxilla (1310:1), paradental plates of the maxilla low and partially obscured by lamina of maxilla (972:1) and shaft of metatarsal II mediolaterally compressed (1208:1). The presence of ventrolateral ridges contacting the parapophyses on the cervical vertebrae (210:1) may represent an additional synapomorphy of Noasauridae. However, the distribution of this character is presently uncertain and so far has only been reported in MACN-PV 622 (cf.  Noasaurus ), in addition to LRF 3050.AR. The noasaurid with the most complete cervical series,  Masiakasaurus , has flattened ventral surfaces of the centra with no ventrolateral ridges 41 . When  Masiakasaurus  is coded as such for the aforementioned character, the presence of ventrolateral ridges does not optimise as a synapomorphy of Noasauridae. However, this may be an artifact of the long-standing lack of resolution among noasaurids due to their poor fossil record, and it remains plausible that ventrolateral ridges may represent a synapomorphy of a subclade within Noasauridae. However, more data is needed to thoroughly test this hypothesis.
The presence of a medial eminence on the ascending process is a synapomorphy that pertains directly to NMV P221202. Among theropods, this feature is shared only with  Masiakasaurus 68  and represents the strongest evidence in favour of noasaurid affinities for NMV P221202. Unfortunately, the lack of preserved ascending processes in the astragali of other noasaurid taxa precludes detailed comparisons.
If the results presented here are correct, then NMV P221202 and LRF 3050.AR represent novel reports of noasaurids from the late Barremian–early Aptian of Victoria and Cenomanian of New South Wales respectively. Under the taxonomic framework presented here, Noasauridae consists of at least six named taxa:  Laevisuchus ,  Noasaurus  and  Masiakasaurus  from the Maastrichtian of India, Argentina, and Madagascar respectively 44 , 67 , 87 ;  Velocisaurus , from the Santonian of Argentina 88 ;  Vespersaurus  from the Aptian–Campanian of Brazil 69  and  Afromimus tenerensis  from the Aptian–Albian of Niger, initially described as an ornithomimid 89  but recently reappraised as a probable noasaurid 90 .  Genusaurus sisteronis , from the Albian of France, has previously been considered as a noasaurid 22 , but subsequent analyses, including the one presented here, preferred a position within Abelisauridae.  Ligabueino andesi , from the Barremian–early Aptian of Argentina 91 , was also originally described as a noasaurid, but phylogenetic studies failed to identify any noasaurid synapomorphies in this taxon 22 , 68 . NMV P221202, which is identified by phylogenetic analysis as a noasaurid, therefore represents the oldest known representative of the clade in the world to date (Fig.  5 ). However, if the broader taxonomic scope of Noasauridae (i.e., inclusive of elaphrosaurines; see Taxonomic Framework) is favoured instead, then NMV P221202 would instead represent the oldest known noasaurine, with the oldest noasaurids represented by the Middle–Late Jurassic aged elaphrosaurines 33 , 41 , 92 . Regardless of their phylogenetic position, the newly described Australian noasaurids expands the known palaeogeographic range of the clade outside of South America, Madagascar and India. Presently, the poor fossil record of Noasauridae, and the corresponding lack of resolution among the known noasaurid taxa, precludes the formation of any novel palaeobiogeographic hypotheses including the newly discovered Australian record of noasaurid theropods. Future discoveries may reveal more detail about the evolution and palaeobiogeographic distribution of this enigmatic clade.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class=""></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">The diversity of Australia’s theropod fauna from the ‘mid’-Cretaceous (Albian–Cenomanian) is distinctly biased towards the medium-sized megaraptorids, despite the preponderance of abelisauroids in the younger but latitudinally equivalent Patagonian theropod fauna. Here, we present new evidence for the presence of ceratosaurian, and specifically abelisauroid, theropods from the Cenomanian Griman Creek Formation of Lightning Ridge, New South Wales. A partial cervical vertebra is described that bears a mediolaterally concave ventral surface of the centrum delimited by sharp ventrolateral ridges that contact the parapophyses. Among theropods, this feature has been reported only in a cervical vertebra attributed to the noasaurid  Noasaurus . We also reappraise evidence recently cited against the ceratosaurian interpretation of a recently described astragalocalcaneum from the upper Barremian–lower Aptian San Remo Member of the upper Strzelecki Group in Victoria. Inclusion of the Lightning Ridge cervical vertebra and Victorian astragalocalcaneum into a revised phylogenetic analysis focused on elucidating ceratosaurian affinities reveals support for placement of both specimens within Noasauridae, which among other characters is diagnosed by the presence of a medial eminence on the ascending process of the astragalus. The Lightning Ridge and Victorian specimens simultaneously represent the first noasaurids reported from Australia and the astragalocalcaneum is considered the earliest known example of a noasaurid in the world to date. The recognition of Australian noasaurids further indicates a more widespread Gondwanan distribution of the clade outside of South America, Madagascar and India consistent with the timing of the fragmentation of the supercontinent.
Subject terms</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Taxonomic Framework
There are presently two hypotheses regarding the content of Noasauridae and the phylogeny of non-abelisaurid, non-ceratosaurid ceratosaurians. Abelisauroidea was originally considered to include Abelisauridae and Noasauridae, and all ceratosaurs more closely related to them than to  Ceratosaurus nasicornis 27 . The earliest phylogenetic analysis of ceratosaurs identified a monophyletic Abelisauroidea following this definition 28 , and which was subsequently expanded to include the African  Elaphrosaurus bambergi 29 . Subsequent phylogenetic studies expanded the taxonomic scope of Noasauridae to include small-bodied Late Cretaceous taxa from South America 21 , 30 – 32  and the Jurassic and Cretaceous of Africa 33 , to the exclusion of  Elaphrosaurus . This topology has been widely recovered in more recent analyses 21 , 34 – 39 . However,  Elaphrosaurus  has also been resolved within Noasauridae in other analyses 40 , most notably in the analysis accompanying the recent redescription of the holotype 41 . Under this hypothesis, the subclade Noasaurinae was coined to include ceratosaurs more closely related to  Noasaurus leali  than to  Elaphrosaurus ,  Ceratosaurus  and  Allosaurus fragilis , and Elaphrosaurinae was erected to include ceratosaurs more closely related to  Elaphrosaurus  than to  Noasaurus ,  Abelisaurus comahuensis ,  Ceratosaurus  and  Allosaurus 41 . The results of a revised phylogenetic analysis for  Limusaurus inextricabilis 42  were used to support a recently proposed phylogenetic framework for Ceratosauria 43  in which Noasaurinae and Elaphrosaurinae were recovered as subclades of Noasauridae. In line with the topology of our phylogenetic tree (see Phylogenetic Analysis), the following descriptions and discussions consider Noasauridae to have the same taxonomic content as Noasaurinae 41 , with members of Elaphrosaurinae representing ceratosaurs basal to Abelisauroidea (i.e., Noasauridae + Abelisauridae).
Systematic Palaeontology
Theropoda Marsh 1881
Neotheropoda Bakker 1986
Averostra Paul 2002
Ceratosauria Marsh 1884
Noasauridae indet. Bonaparte and Powell 1980 44
LRF 3050.AR
Locality
LRF (Australian Opal Centre, Lightning Ridge, New South Wales, Australia) 3050.AR was collected from an underground opal mine at the ‘Sheepyard’ opal field, approximately 40 km southwest of Lightning Ridge in central northern New South Wales (Fig.  1 ). The specimen derives from the Wallangulla Sandstone Member 45  of the Griman Creek Formation. Radiometric dates for the Wallangulla Sandstone Member at Lightning Ridge indicate a maximum depositional age of 100.2–96.6 Ma 46 . LRF 3050.AR was found within a monodominant bonebed of the iguanodontian  Fostoria dhimbangunmal 47 . Other faunal components from this accumulation include isolated unionid bivalves (LRF 3051), a testudine caudal vertebra (LRF 3053), a small ornithopod caudal centrum (LRF 3052), and a possible indeterminate theropod ulna (LRF 3054). A complete discussion of the geological setting, sedimentology, age and faunal diversity of the Griman Creek Formation is presented elsewhere 46 .
Description
LRF 3050.AR has been taphonomically altered by erosion, breakage and through preparation. The centrum is markedly flattened dorsoventrally through taphonomic compaction, such that much of the left lateral surface is visible in ventral view. In addition, the dorsal portion of the centrum has been sheared off obliquely. Notwithstanding the dorsoventral compression, the centrum is hourglass-shaped in dorsal-ventral view; the narrowest point occurs approximately one-third of the length from the anterior articular surface (Fig.  2a,b ). In lateral view, the anterior and posterior articular surfaces are oriented obliquely relative to the long axis of the centrum (approximately 20 degrees from vertical; Fig.  2c,d ); however, this appearance is probably a result of the taphonomic compaction and not indicative of their original orientations. The ventral surface of the centrum is markedly concave in lateral view (Fig.  2c,d ). The centrum is slightly more than twice as long anteroposteriorly relative to the width of the posterior articular surface (Table  1 ). The centrum is amphicoelous. The central region of the anterior articular surface is flattened and surrounded laterally and ventrally by a concave rim (Fig.  2e ), whereas the centre of the posterior articular surface is concave and bordered ventrally by a convex rim (Fig.  2f ). The preserved portion of the anterior articular surface is elliptical in anterior view, wider mediolaterally than dorsoventrally tall (Fig.  2e ). Only the ventralmost portion of the left parapophysis is present on the ventrolateral edge of the centrum anteriorly, and which also projects ventrolaterally (Fig.  2a,c,e ). A region of exposed trabecular bone immediately dorsal to the preserved parapophysis indicates the likely size of its attachment to the centrum (Fig.  2c ). An anteroposteriorly oriented lamina is present anterodorsally, extending from the anterior articular surface to approximately one third of the length of the centrum and overhanging the right lateral surface (Fig.  2a,b ). The posterior edge of the lamina is broken, indicating that it likely continued further posteriorly. On the ventromedial surface of this lamina is the eroded remains of a smaller, vertically oriented lamina (Fig.  2d ). The position of this smaller lamina would have been dorsal to the parapophysis, and its vertical and lateral continuation indicates that it would have contacted the diapophysis ventrally. Therefore, this lamina is interpreted as a paradiapophyseal lamina (ppdl; following nomenclature for vertebral laminae of Wilson 48 ). Consequently, the portion of the larger lamina anterior to the ppdl is interpreted as the anterior centrodiapophyseal lamina (acdl), and the posterior portion is interpreted as the remains of the posterior centrodiapophyseal lamina (pcdl). The posterior articular surface is missing the dorsal portion due to erosion, similar to the anterior end, and is elliptical, having a greater mediolateral width than dorsoventral height (Fig.  2f ). A portion of the floor of the neural canal is preserved across the anterior half of the dorsal surface of the centrum (Fig.  2a ). Despite erosion to the dorsal surface of the centrum, the neural canal appears to have been mediolaterally wide, approximately half that of the centrum itself, and considerably wider than the neural arch pedicels as visible from their eroded bases (Fig.  2b ). The ventral surface of the centrum is concave mediolaterally and delimited by well-defined, subparallel ventrolateral ridges that extend as laminae from the parapophyses along nearly the entire length of the centrum, becoming less distinct posteriorly (Fig.  2b ). Two small (~3 mm long) lenticular foramina are present on the posterior half of the centrum (Fig.  2a ). Whether these foramina are pneumatic in origin cannot be determined.
Supplementary information
   </span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Supplementary information
is available for this paper at 10.1038/s41598-020-57667-7.
Acknowledgements
We are indebted to Robert Foster who discovered the ‘Sheepyard’ specimens and Joanne Foster and Gregory Robert Foster who generously donated the specimens under the Australian Government’s Cultural Gifts program. We thank Jenni Brammall, Manager of the Australian Opal Centre, for allowing access to LRF 3050.AR and providing resources to facilitate their study while in Lightning Ridge, and Tim Ziegler of Museum Victoria for making NMV P221202 available for study. TNT is made freely available thanks to a subsidy from the Willi Hennig Society. We thank Stephen Poropat and two anonymous reviewers for their valuable comments that improved the quality of the manuscript. We acknowledge the Yuwaalaraay, Yuwaalayaay and Gamilaraay custodians of country in the Lightning Ridge district, and pay our respects to Elders past and present. This work was supported by an Australian Research Council Discovery Early Career Researcher Award (project ID: DE170101325) to P.R.B.
Author contributions
S.A.B. designed the research, performed the descriptive and comparative studies, analysed data, prepared figures and performed the phylogenetic analysis; E.S. and P.B. contributed specimen photographs and data; S.A.B., E.S. and P.B. wrote the paper.
Data availability
All data generated or analysed during this study are included in this published article (and its Supplementary Information).
Competing interests
The authors declare no competing interests.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">CC BY</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">no</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">2024-01-16 23:34:59</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Sci Rep. 2020 Jan 29; 10:1428</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">oa_package/16/47/PMC6989633.tar.gz</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="1"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">PMC7168315</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">32059467</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">1. Introduction
Biofilms—aggregations of densely packed microbial cells embedded inside exopolysaccharide (EPS) matrix—are a major challenge in public health management. The EPS matrix provides a protective barrier to the biofilm, making them recalcitrant to antimicrobial agents and host defenses [ 1 ].  Staphylococcus aureus  is thought to be one of the major causes of nosocomial infections, globally. While the planktonic counterpart is limited to bacteremia and skin abscesses, more chronic infections such as cystic fibrosis, osteomyelitis, and endocarditis are associated with its biofilm mode of growth [ 2 , 3 ]. To eradicate this problem, many strategies were proposed, including (i) the inhibition of primary bacterial adhesion or attachment to the living or non-living surfaces; (ii) the disruption of biofilm architecture during maturation processes; and (iii) the inhibition of cell to cell communication—i.e., quorum sensing [ 4 , 5 ]. Chrysin (5, 7-dihydroxyflavone), a flavone constituent of the  Orocylumineicum vent  has already been documented for its anticancer, antioxidant and antibacterial properties [ 6 , 7 ]. In spite of its many biological activities, its low water solubility, poor biosorption in the intestinal lumen, low bioavailability and rapid metabolism in the body limits its therapeutic applications [ 8 ]. In this regard, a reduction in particle size may serve as a potential means for enhancing the solubility and dissolution of chrysin [ 9 ].
Nanocarriers have the ability to inhibit bacterial growth and biofilm formation, and are increasingly being used as an attractive tool to combat chronic infections [ 10 ]. They aid in increasing the efficacy of the drug by acting as a protective barrier against enzymatic hydrolysis, increase the biosorption efficacy of the drug in the intestinal lumen, increase solubility and also cause sustainable release [ 11 , 12 ]. In recent years, chitosan is widely being used as a nano-carrier due to its non-toxicity, biocompatibility, immunostimulating and mucoadhesive properties [ 13 ]. Chitosan is a cationic heteropolysaccharide composed of the β-(1,4) linked repeating unit of glucosamine (GlcN) and N-acetylglucosamine (GlcNAc), extracted by the partial alkaline N-deacetylation of chitin found in the exoskeleton of crustaceans [ 14 , 15 ]. The antimicrobial and anti-biofilm potential of chitosan and its nano-derivatives were reported against various microorganisms such as  Listeria monocytogenes ,  Bacillus cereus ,  Enterococcus faecalis,  etc. [ 1 , 16 ]. Chrysin-encapsulated chitosan nanoparticles (CCNPs), synthesized using the ionic gelation method, were characterized and evaluated for their anti-biofilm activity against  S. aureus .</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">2. Materials and Methods
2.1. Materials
Chitosan (75–85% deacetylated), sodium tripolyphosphate (TPP) and Chrysin were procured from Sigma-Aldrich. The test stain,  S. aureus  (MCC 2408) was purchased from Microbial Culture Collection (MCC), Pune, India.
2.2. Synthesis of Chrysin-Loaded Chitosan Nanoparticles
Medium molecular weight chitosan (0.2%,  w / v ) was mixed with an aqueous solution of acetic acid (0.1%,  v / v ) and incubated over-night at 60 °C with continuous agitation. A stock of 5 mg/mL of chrysin dissolved in DMSO was used to prepare the nanoparticle formulation. An aliquot of chrysin prepared in DMSO was added to the chitosan solution (pH 4.8). Subsequently, 40 mL of TPP solution (0.2%,  w / v ) was dispensed dropwise into the chitosan–chrysin solution and kept under continuous agitation at 1000×  g  for 30 min. The ratio of the chitosan-TPP was maintained at 5:1 [ 13 ] with the final chrysin concentration of 50 μg/mL. The nanoparticles formed were concentrated by centrifuging the suspension for 20 min at 12,000×  g , washed with MilliQ water to remove the unbound chrysin and dried at room temperature for further studies [ 17 ].
2.3. Physical Characterization of Nanoparticles
The nanoparticles (NPs) were subjected to dynamic light scattering (DLS) to determine the mean hydrodynamic diameter (MHD) and polydispersity index (PDI). The FTIR spectrum was recorded in the range of 4000–500 cm −1 . A transmission electron microscope (TEM) was used to determine the morphology and size of the CCNPs [ 18 ].
2.4. Determination of the Loading Efficiency and Drug Release of Chrysin-Loaded Chitosan NPs
The amount of chrysin loaded in the nanoparticles was determined using a UV-Vis spectrophotometer. After the collection of NPs from the reaction mixture, the absorbance of the supernatant was recorded at 348 nm and the concentration of unbound chrysin was estimated based on the standard curve of chrysin [ 13 , 19 ]. The CCNPs were dislodged in a medium constituting PBS and DMSO (co-solvent, 1%,  v / v ) and incubated with gentle agitation (100×  g ) at 37 °C. Two milliliters of the sample was retrieved at regular intervals, centrifuged at 10,000×  g  and the absorbance of the supernatant was recorded at 348 nm. The cumulative chrysin released in the medium was determined at the every 2 h interval, with reference to the standard curve of chrysin [ 20 ]. Chrysin release (%) = (Chrysin released in the supernatant/loaded chrysin concentration) × 100.
2.5. Determination Sub-Minimum Inhibitory Concentration (Sub-MIC) of Chrysin-Loaded Chitosan NPs
The minimum inhibitory concentration (MIC) of CCNPs was determined using macro-broth dilution assay (Clinical and Laboratory Standards Institute Guidelines, 2006). Two-fold dilutions of the NPs were prepared in Muller-Hinton broth to achieve a final concentration ranging from 8 μg/mL to 1024 μg/mL. An overnight culture of  S. aureus  (100 μL) was added in each NPs suspension and incubated at 37 °C for 24 h. The test tubes were observed for visible signs of growth and the spectrophotometric readings were recorded at 600 nm [ 12 ].
2.6. In Vitro Anti-Biofilm Assays of Chrysin-Loaded Chitosan NPs
The effect of sub-MIC of CCNPs of the on the biofilm formation of  S. aureus  was evaluated relative to chitosan NPs (CNPs) and Chrysin.
2.6.1. Microtiter Plate (MTP) Assay for Biofilm Disruption and Inhibition
MTP assay for biofilm disruption and inhibition was performed according to Mu et al. [ 1 ]. An over-night culture of  S. aureus  (100 μL) was transferred into the wells of 96-well flat-bottomed polystyrene plates. After incubation at 37 °C for 24 h, the wells were washed with 100 μL of 0.9% ( w / v ) NaCl to remove the unadhered cells. The biofilm formed was further incubated after adding 90 μL tryptone soy broth (TSB) supplemented with the sub-MIC concentration of CCNPs for another 24 h. The biofilm attached at the bottom of each well was fixed with 100 μL of absolute methanol for 15 min and subsequently treated with 100 μL of crystal violet (0.2%  w / v ). Control samples were maintained with  S. aureus  culture alone. Cultures of  S. aureus  treated with DMSO serve as a control. The dye attached to the biofilm was further solubilized in 150 μL of glacial acetic acid and the optical density was recorded at 595 nm.  S. aureus  culture (90 μL) grown in TSB was seeded into individual wells of microtiter plates in the presence of sub-MIC of CCNPs and incubated at 37 °C for 24 h. The planktonic cells were discarded and the MTP was stained with crystal violet (0.2%  w / v ). The inhibition of biofilm formation was determined by solubilizing the CV attached to the biofilm and measuring the optical density at 595 nm. Biofilm inhibition/disruption was quantified using the following formula:
% Biofilm inhibition/disruption= ([OD 595  of control − OD 595  of test] = OD 595  of control) * 100
2.6.2. Microscopic Examination of Biofilm
Reduction in the biofilms of  S. aureus  was observed using confocal laser scanning microscopy (CLSM).  S. aureus  biofilms was allowed to grow on glass coverslips (18 × 18 mm) placed in 12 well polystyrene plates containing TSB supplemented with CCNPs (sub-MIC) and incubated overnight at 37 °C. The glass coverslip was washed with sterile distilled water, stained and processed accordingly [ 21 ]. The biofilm formed was fixed using methanol and treated with crystal violet (0.2%  w / v ). The coverslip was subsequently washed, air dried and observed using a light microscope (40×). The biofilm formed on the coverslip was washed 0.01 M phosphate buffer saline (PBS), stained using acridine orange (0.2%  w / v ) for 1 min and observed using Confocal laser microscope at 20×. The 3D image was recorded and Z stacks were prepared to determine the effect of the CCNPs on the thickness of the biofilm [ 22 ].
2.6.3. Exopolysaccharide (EPS) Quantification and Microbial Adhesion to Hydrocarbon (MATH) Assay
Production of EPS by  S. aureus  was quantified in presence and absence of CCNPs by total carbohydrate quantification method.  S. aureus  was grown in presence and absences of CCNPs were harvested by centrifugation (10,000×  g  for 2 min). The cell pellet was washed and suspended in 200 μL of sterile PBS to which an equal volume of 5% ( v / v ) phenol and 5× volume of concentrated sulfuric acid containing 0.2% ( w / v ) hydrazine sulphate was added. The tubes were incubated in dark for 1 h followed by centrifugation at 10,000×  g  for 10 min. The supernatant was aspirated and the optical density was measured at 490 nm [ 21 ]. A reduction in EPS production was quantified using the following formula:
The effect of CCNPs on the cell surface hydrophobicity of  S. aureus  was evaluated using MATH assay. The optical density of the treated CCNPs and the untreated cell suspension was recorded at 600 nm, after 24 h of incubation. The bacterial suspension was mixed with toluene (1 mL) and vortexed for 2 min. The optical density of the aqueous phase was measured at 600 nm. In both the assays, control samples were maintained with  S. aureus  culture only. Cultures of  S. aureus  treated with DMSO served as a negative control [ 21 ]. The percentage of inhibition in hydrophobicity is measured as follows; 
2.6.4. Growth Curve Analysis 
An overnight culture of  S. aureus  was diluted with LB medium until the optical density of the cell suspension reaches 0.05 at 600 nm. The suspension was then supplemented with chrysin and NPs separately and incubated overnight at 37 °C at 100×  g . The cell suspension (1 mL) was withdrawn and the optical density was measured at 600 nm, at every 2 h interval [ 23 ]. 
2.7. Statistical Analysis
All the assays were repeated thrice, and the data are presented as mean ± standard error. Significance among treatments were investigated using one-way ANOVA and represented with a statistical significance of  p  ≤ 0.05. Significance in treatments of chrysin, chitosan and CCNPs are represented with asterisk sign. Non-significant groups are represented by NS.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">3. Results
3.1. Synthesis and Characterization of Chrysin-Loaded Chitosan NPs
The CCNPs were synthesized using the ionotropic gelation method, using TPP molecules as a linker. The ratio of chitosan and TPP used is one of the factors that influence the aggregation of nanoparticles. Chitosan/TPP in the ratio of 5:1 was found to be the best formulation for the synthesis of CCNPs. The mean hydrodynamic diameter of the synthesized CNPs and CCNPs were found to be ~299 nm and ~355 nm, respectively. CNP and CCNP nanoparticles showed an intermediate polydisperisty index of 0.434 and 0.487, respectively ( Figure 1 a,b). However, the CCNPs were found to be spherical with sizes ranging from 130–341 nm as indicated by the TEM micrograph ( Figure 1 c). On comparing the functional groups present in the CCNPs with their bulk counterparts, chitosan showed characteristic peaks at 3418 and 3238 cm −1 , which indicated the O-H stretching and N–H stretching vibration, the characteristic peak at 2908 cm −1  depicted the C–H stretch, 2342 cm −1  (C–N band stretching), 1610 cm −1  (amide II band), 1024 and 1051 cm −1  indicated that the CH 2  group and C–O stretch from glucosamine residue was observed. Chrysin showed characteristic bands at 2625 cm −1 , 2343 cm −1  indicating O–H stretching vibration and intramolecular H-bond ( Figure 1 d). The characteristic peaks of both chitosan (at 2342 cm −1 , 1051 cm −1 ) and chrysin (at 2625 cm −1  and 2343 cm −1 ) were observed in the CCNPs. A slight band shift was also observed at 3347 and 3200 cm −1  to the lower wave-number that indicated the presence of hydrogen bonding between O-H group of chrysin and O-H or -NH 2  group of chitosan [ 24 ]. The other peaks observed in the loaded nanoparticles were the P–O bending peak at 890 cm −1  and at 2625 cm −1 , which was broader compared to pure chrysin, indicating an increase in hydrogen bond interactions [ 13 ].
3.2. Loading Efficiency and Release Kinetics of Chrysin-Loaded Chitosan NPs
The amount of chrysin loaded onto the CCNPs was found to be 80.86 ± 0.30%. The in vitro drug release profile of chrysin from the CCNPs was determined in a release medium constituting PBS and DMSO, at 37 °C. The pH of the media was set at 7.4, as the ionic strength of the media plays a vital part in the stability and drug release. The cumulative chrysin release as a function of time is depicted in  Figure 2 a. The drug release kinetics of the loaded NPs initially showed a burst release which was followed by a steady and sustainable release from the 8th h. The first burst release was observed in the first two hours with 36.33 ± 1.58% of chrysin release. The second burst was observed after the sixth hour with 80.11% ± 0.84% drug release. The graph takes the form of a plateau starting from the 10th h to 24th h. It was also revealed that about a total of 90.5% ± 0.50% of the chrysin was released from the NPs within 10 h.
3.3. Minimum Inhibitory Concentration (MIC) and Sub-MIC of Chrysin-Loaded Chitosan NPs
The MIC value of the CCNPs was determined to be 1024 μg/mL for  S. aureus . At 768 μg/mL, the NPs that did not exert any effect on the growth of the test bacteria. Hence, 768 μg/mL was selected as a sub-MIC concentration and used in all the subsequent anti-biofilm assays.
3.4. In Vitro Anti-Biofilm Activity of Chrysin-Loaded Chitosan NPs
3.4.1. Crystal Violet Staining Assay for Biofilm Formation and Disruption
The CCNPs showed a reduction in biofilm formation as compared to its bulk counterparts. The biofilms were inhibited to 50.48 ± 2.42% and 54.1 ± 0.56 % on treatment with CNPs and chrysin, respectively ( Figure 2 b). However, the CCNPs inhibited the biofilm formation to 66.59 ± 3.09%. The treatment of the preformed biofilm with CCNPs also resulted in a reduction in the biofilm mass of 43.50 ± 1.29%, whereas a decrease in biofilm of 14.92 ± 2.17% and 20.94 ± 3.73% was observed in the presence of CNPs and chrysin, respectively ( Figure 2 c).
3.4.2. Microscopic Examination of Biofilm
Light microscopy and CLSM were used to observe the change in the biofilm architecture of  S. aureus  in presence and absence of CCNPs. The influence of CCNPs on the thickness of biofilm, overall structure and biofilm density was evident in the micrographs. However, a dense biofilm was visible in the light microscope images of the untreated samples ( Figure 3 ). A thick biofilm of 80 μm was observed in the control while the thickness was reduced to 20 μm on treatment with chrysin. However, a higher reduction in the thickness of the biofilm matrix to 16 μm was achieved in the presence of CCNPs ( Figure 3 ).
3.4.3. Exopolysaccharide (EPS) Quantification and Microbial Adhesion to Hydrocarbon (MATH) Assay
The CCNPs showed better reduction in the synthesis of EPS compared to its bulk counterparts. On treatment with CCNPs, a reduction in EPS production of 38.03 ± 5.41% was observed ( Figure 4 a). Chrysin and CNPs were also able to restrict the production of EPS by 33.37 ± 4.84% and 26.54 ± 3.20%, respectively. Cell surface hydrophobicity (CSH) is another important factor in biofilm formation as it aids in the adherence of the cell to the substratum. The CCNPs reduced the CSH in  S. aureus  by 84.66 ± 2.84% as compared to its bulk counter parts ( Figure 4 b). CNPs and Chrysin showed approximately 61.28 ± 5.78% and 72.46 ± 4.21% decreases in cell surface hydrophobicity, respectively. 
3.4.4. Growth Curve Analysis 
The growth pattern of the test organism when cultivated in the presence and absence of the CCNPs and CNPs is presented in  Figure 5  Though the cells exhibited retardation in growth on exposure to a sub-MIC of the NPs and Chrysin, there was no significant decrease in the cell density. Hence, it can be inferred that the CCNPs arrested the biofilm development in  S. aureus , indicating the potential application of CCNPs in the management of  S. aureus -related infections.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">4. Discussion
The formation of biofilm is one of the major obstacles in the modern antibacterial therapy. The biofilm-forming ability of bacteria provides the pathogen with advantages by blocking the entry of antimicrobial agents, thus causing hindrance in the clearance of these pathogens by the host immune system. Drug nanonization—i.e., a reduction in the particle size of drugs to nano-size—enhances the intracellular uptake of nanoparticles thus, providing a way to overcome the problems associated with insoluble drugs. Moreover, an increase in the surface area of poorly soluble drugs also leads to a more pronounced increase in the therapeutic index by maximizing the action with lesser dose [ 25 ]. In the present study, the anti-biofilm activity of CCNPs was demonstrated against biofilm forming bacterium,  S. aureus  MCC 2408. Chrysin was encapsulated to chitosan using the ionotropic gelation method. The NPs were formed due to the electrostatic interaction between amine group of chitosan and polyphosphate ions of TPP [ 19 ]. The hydrodynamic size influences various properties such as loading efficiency, drug release kinetics and the stability of the NPs. Though smaller nanoparticles due to high surface area show greater encapsulation efficiency, however, it also tends to aggregate easily on storage [ 19 ]. Particles of a low polydispersity index (PDI) are homogenous in nature and provide maximum stability. High polydispersity index (PDI) indicates the heterogeneity of the nanoparticles in the mixture. The synthesized spherical nanoparticles showed intermediate polydispersity that aids in the stability of the CCNPs. Ilk et al. [ 19 ] reported the synthesis of kaempferol loaded chitosan/TPP nanoparticles with an average particle size of 192.27 nm. The FTIR spectra of CCNPs indicated the presence of a similar functional group as that of its bulk counterpart indicating the successful encapsulation of chrysin with chitosan and the formation of chrysin-loaded chitosan NPs.
The biological efficacy of a drug and its potential use in drug delivery is directly influenced by the loading efficiency and controlled release. The CCNPs demonstrated a high encapsulation efficiency with a sustainable release. The CCNPs showed a higher loading efficiency compared to the previously synthesized nanocomposites such as the BSA-loaded chitosan-TPP nanoparticle with a loading efficacy of 60% [ 13 ] and PEG-chrysin conjugates with a loading efficacy of 55.6% [ 26 ]. The high encapsulation efficiency of CCNPs may be attributed to the presence of hydrogen bond between the -OH group of chrysin and the -NH2 group of chitosan that help in better entrapment of chrysin into the CNPs. From these release kinetics, it can be interpreted that chrysin was not covalently bonded to the nanoparticle and was thus easily released when dislodged in the medium. A similar result was observed in kaempferol-loaded chitosan nanoparticles, where more than 85% of the drug release was attained within 4 h, and no significant quantity of the drug was released thereafter [ 19 ].
Biofilm formation by  S. aureus  is associated with many nosocomial as well as chronic diseases associated with medical devices and surgical implants. It also leads to the emergence of the multi-drug resistant (MDR) strains viz. Methicillin-resistant  S. aureus  (MRSA) and Vancomycin-resistant  S. aureus  (VRSA) [ 3 ]. It was found that both CNPs and chrysin exhibited significant anti-biofilm activity relative to the untreated control. However, the anti-biofilm efficacy was comparatively enhanced when chrysin was encapsulated with chitosan. The ability to attach and establish biofilm on inert surfaces contributes to making  S. aureus  a major pathogen of chronic infections [ 10 ]. The data also suggested that the CCNPs showed better biofilm inhibition ability than disruption of preformed biofilm. Shi et al. [ 10 ] suggested that chitosan-coated iron oxide nanoparticles have the potential to effectively prevent bacterial colonization and control the biofilm formation by 53% in  S. aureus . The anti-biofilm efficacy of the NPs was also validated by light and CLSM micrographs which showed a reduction in thickness and density of the biofilm matrix in presence of CCNPs.
The EPS matrix plays an indispensable role in the initial cell attachment, the formation of biofilm architecture and in providing mechanical stability of the biofilm. The EPS produced by the biofilm-forming bacteria prevents the access of antimicrobial agents and antibiotics to the bacterial cell [ 18 ]. The CCNPs caused a considerable decrease in the EPS production and cell surface hydrophobicity in  S. aureus  which resulted in the decrease in bacteria accumulation and attachment to the substratum. Hence, it can be inferred that CCNPs have a profound effect on the early stages in biofilm formation, specifically in the adherence and colonization as compared to its bulk counterparts, chrysin and chitosan. 
From the growth curve analysis, it can be interpreted that at the sub-MIC level, the CCNPs exerted less bactericidal effect and selective pressure against  S. aureus . However, it had a profound effect on the  S. aureus  in the biofilm mode of growth. Chrysin and chitosan are found to be nontoxic in recommended concentrations. It was reported that the recommended daily concentration of this flavone is 0.5 to 3 g [ 27 ]. Likewise, chitosan nanoparticles are nontoxic at low concentrations and found to be toxic only at higher concentrations [ 28 ]. The nontoxic nature of these components may enable the application of CCNPs for biomedical applications. As these the outcome of the study suggested that the nanoformulation of chrysin exhibits enhanced synergistic anti-biofilm activity against  S. aureus  when compared to its bulk counterparts—chrysin and chitosan taken separately. Hence, CCNPs may be considered as a potential therapeutic agent for controlling biofilm formation in  S. aureus . The nanocomposites may be further exploited towards the development of anti-biofilm coatings.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">5. Conclusions
This study displayed an enhanced antibiofilm activity of chrysin against  S. aureus  when loaded on to chitosan-TPP nanoparticles, with a profound loading capacity. Chrysin-loaded chitosan nanoparticles were characterized to confirm the effective loading of the flavone on chitosan nanoparticles. Anti-biofilm activities of CCNPs were determined through biofilm inhibition, biofilm disruption, EPS reduction and hydrophobicity reduction assays. CCNPs synthesized could be used as a potential therapeutic agent for controlling biofilm formation in  S. aureus  in the future. The nanocomposites may be further exploited towards the development of anti-biofilm coatings.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">The application of nanotechnology in medicine is gaining popularity due to its ability to increase the bioavailability and biosorption of numerous drugs. Chrysin, a flavone constituent of  Orocylumineicum vent  is well-reported for its biological properties. However, its therapeutic potential has not been fully exploited due to its poor solubility and bioavailability. In the present study, chrysin was encapsulated into chitosan nanoparticles using TPP as a linker. The nanoparticles were characterized and investigated for their anti-biofilm activity against  Staphylococcus aureus . At sub-Minimum Inhibitory Concentration, the nanoparticles exhibited enhanced anti-biofilm efficacy against  S. aureus  as compared to its bulk counterparts, chrysin and chitosan. The decrease in the cell surface hydrophobicity and exopolysaccharide production indicated the inhibitory effect of the nanoparticles on the initial stages of biofilm development. The growth curve analysis revealed that at a sub-MIC, the nanoparticles did not exert a bactericidal effect against  S. aureus . The findings indicated the anti-biofilm activity of the chrysin-loaded chitosan nanoparticles and their potential application in combating infections associated with  S. aureus .</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class=""></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Acknowledgments
We are grateful to the Central Instrumentation Facility (CIF), Pondicherry University for the DLS and FTIR analysis. We would like to thank the Sophisticated Test and Instrumentation Centre (STIC), Cochin for the TEM analysis. We are also thankful to Bharathidasan University, Tiruchirappalli for supporting us with the Confocal Laser Scanning Microscopy. The authors extend their appreciation to The Researchers supporting project number (RSP-2019/15) King Saud University, Riyadh, Saudi Arabia.
Author Contributions
Conceptualization, B.S.; methodology, B.S., U.P., and A.S.; software, B.S., U.P., A.S., and R.P.; validation, B.S., and R.P.; formal analysis, B.S.; investigation, B.S., A.M.E., and A.H.B.; resources, B.S., A.S., and K.K.; data curation, B.S., U.P., R.P., K.K., and A.S.; writing—original draft preparation, B.S., U.P., R.P., A.S., and K.K.; writing—review and editing, B.S., R.P., A.S., and K.K.; visualization, B.S., A.M.E., and A.H.B.; supervision, B.S.; project administration, A.S.; funding acquisition, A.S. All authors have read and agreed to the published version of the manuscript.
Funding
The APC was funded by The Researchers supporting project number (RSP-2019/15) King Saud University, Riyadh, Saudi Arabia. 
Conflicts of Interest
The authors declare no conflict of interest.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">CC BY</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">no</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">2024-01-16 23:36:45</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Pathogens. 2020 Feb 12; 9(2):115</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">oa_package/ba/c7/PMC7168315.tar.gz</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="2"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">PMC7408465</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">32610539</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">1. Introduction
Flibanserin (FLB) is a recently FDA-approved nonhormonal drug for the treatment of women with hypoactive sexual appetite disorder. FLB acts via decreasing the level of serotonin and increasing the levels of dopamine and norepinephrine for maintaining healthy sexual response [ 1 ]. FLB-treated women have demonstrated significant improvements in both the number of satisfying sexual events and the female sexual function index desire domain score compared placebo-treated ones. These findings proved the ability of the drug to enhance the women’s sexual desire. In addition, administration of FLB was linked with a significant reduction in the distress related with either sexual dysfunction or low sexual desire [ 2 , 3 , 4 , 5 ]. However, the major challenge for oral administration of FLB is the reduced bioavailability (~33%) that might be caused by the drug’s low solubility and its exposure to hepatic first-pass metabolism [ 6 , 7 ].
Recently, intranasal drug administration has gained increasing interest. The nasal pathway is a noninvasive route for active pharmaceutical ingredient (API) administration with the aim of local, systemic, or central nervous system (CNS) action. The nasal cavity represents an ideal absorption surface for drug delivery due to the high vascularity of this area, in addition to the leaky epithelium that results from the low tightness of the intercellular nasal mucosal junctional complex. Furthermore, direct absorption of the molecules from the nasal cavity via the trigeminal and olfactory pathways provides direct entry into the brain and results in a favorable pharmacokinetic/pharmacodynamic profile for centrally acting drugs. Thus, the nasal route could offer an encouraging unconventional approach to enteral and systemic drug administration of CNS-targeting drugs [ 8 , 9 ].
Transfersomes (TRFs), also called deformable or elastic liposomes, are flexible vesicular systems that involve a phospholipid (PL) and an edge activator. They are considered as a modified generation of liposomes and were firstly modified by Cevc and Blume [ 10 ] by adding edge activators. The edge activators are usually a single-chain surfactant which enhances the squeezing and penetration of the vesicles through the mucosal barrier through destabilization of the lipid bilayers. The commonly used edge activators include sodium deoxycholate, sodium cholate, Tween, and Span [ 11 , 12 , 13 ]. Intranasal administration of TRFs has been previously reported to enhance bioavailability of several drugs [ 14 , 15 , 16 ]. Moreover, TRFs have been effectively applied for enhancing brain distribution of centrally acting medicines [ 17 , 18 , 19 ].
Hydrogel-loaded nanoformulated drugs have drawn significant attention as promising nanoparticulate drug delivery systems that combine both hydrogel system properties (e.g., hydrophilicity and high water absorption affinity) and nanoparticulate properties (e.g., ultrasmall size) [ 20 , 21 , 22 , 23 , 24 , 25 , 26 , 27 ], can achieve high drug loading without chemical reactions, and are able to release integrated agents at the target site in a controlled behavior. A wide range of natural, naturally derived and synthetic hydrogels can be used for hydrogel-loaded nanoformulated drug preparation [ 26 , 27 , 28 ]. Hydrogels can be prepared from naturally derived protein or polysaccharide polymers [ 29 ]. The synthetic hydrogels have drawn great attention in the biomedical field [ 30 , 31 ]. The synthetic hydrogels are obtained through chemical and physical methods. Among the synthetic hydrogels, poly(2-isopropenyl-2-oxazoline) (PiPOx) is a biocompatible polymer synthesized using a simple protocol [ 30 ]. In addition, poly(vinyl alcohol) (PVA) and PVA/poly(ethylene glycol) (PEG) hybrid hydrogels were synthesized that showed improved mechanical strength when compared with PVA hydrogel [ 32 ].
Among natural and naturally derived hydrogels, the most frequently used are polysaccharides. Materials with polysaccharides can be divided into two groups, namely polyelectrolytes and non-polyelectrolytes. Additionally, polyelectrolytes may be classified according to their intrinsic charges, including cationic (chitosan), anionic (alginate, heparin, pectin, hyaluronic acid), and neutral (pullulan, dextran). Due to their desirable mucoadhesive properties, cellulose derivatives can significantly extend the residence time of drugs in the nasal cavity [ 33 ]. Furthermore, due to their high viscosity following hydration in the nasal cavity, celluloses can sustain the release of drugs. For these reasons, the use of cellulose as an absorption enhancer can lead to improved intranasal absorption and increased bioavailability [ 34 ]. Reports show that celluloses increase the intranasal bioavailability of both small hydrophobic and hydrophilic macromolecular drugs [ 35 ]. Hydroxypropyl methyl cellulose (HPMC) is a popular matrix material in controlled drug delivery systems, and HPMC matrices show a sustained release pattern by two mechanisms, i.e., diffusion and erosion of the gel layer [ 36 ]. The viscosity of the polymer affects the diffusion pathway. HPMC can be employed as a matrix for controlling the release of both hydrophilic and hydrophobic drugs [ 37 ]. HPMC-based gels showed good surface morphology with high drug loading efficiency. The viscosities of the preparations were found to be within a suitable range for nasal administration.
Therefore, the main aim of this study was to acquire an optimized FLB-TRF-loaded HPMC-based hydrogel for an improved drug delivery to the brain via intranasal administration. Box–Behnken design was utilized for FLB TRF optimization. The effects of FLB-to-PL molar ratio, edge activator hydrophilic lipophilic balance (HLB), and pH of hydration medium on vesicle size were studied. The optimized TRFs with minimized vesicle size were prepared and fused into hydroxypropyl methyl cellulose based hydrogel. The prepared hydrogel was assessed for shape characteristics and ex vivo permeation. In addition, in vivo performance was evaluated after intranasal administration in Wistar rats.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">2. Materials and Methods
2.1. Materials
Flibanserin (FLB) was purchased from Qingdao Sigma Chemical Co., Ltd. (Qingdao, China); Phospholipon 90G (phosphatidyl choline from soy, at least 90% purity) was purchased from Lipoid GmbH (Frigenstr, Ludwigshafen, Germany); Span 65, Span 80, methanol, and chloroform were purchased from Sigma-Aldrich Co. (St. Louis, MO, USA).
2.2. FLB TRF Preparation
FLB TRFs were prepared by the hydration of the formed lipid film as previously described [ 38 ]. Briefly, specified amounts of FLB, PL, and edge activator (surfactant) were dissolved in methanol/chloroform mixture (1:1,  v / v ) and subjected to water bath sonication for 5 min. The amounts of FLB, PL, and surfactant were specified according to  Table 1 . Span 65 and span 80 were used in different ratios to achieve the required HLB value of the edge activator indicated in the design ( Table 1 ). The solution was then evaporated using a rotary evaporator at 45 °C. The formed film was kept in a vacuum oven overnight for complete removal of solvent residuals. Subsequently, the dried thin film was hydrated with 20 mL of buffer solution, according to the specified pH, for 3 h at 25 °C with gentle shaking.
2.3. Box–Behnken Design for FLB TRF Preparations
According to the previous screening results conducted in our laboratory, the optimization of FLB TRFs was carried out to achieve minimal size. FLB:PL molar ratio ( X 1 ), HLB ( X 2 ), and pH of hydration medium ( X 3 ) were the investigated factors, while vesicle size ( Y 1 ) was the studied response. The  X 1  ratios studied were 1:1, 1:3, and 1:5;  X 2  values were 2, 4, and 6; and  X 3  values were 5, 7, and 9. All other processing and formulation variables, including drug amount (10%  w / w ), were kept constant throughout the study. The experimental design using Design-Expert software (version 12; Stat-Ease, Inc., Minneapolis, MN, USA) yielded 17 formulations. The actual values of the independent variables of these runs and the observed responses are presented in  Table 1 . The measured responses were statistically analyzed by the analysis of variance (ANOVA) test. The polynomial equations representing the best fitting model for each variable was generated. Three-dimensional surface plots were plotted to illustrate the impact of the variables and interaction between them at  p  &lt; 0.05. Afterwards, a numerical method following the desirability approach was utilized to predict the optimized FLB TRFs. The predicted formulation was then prepared and further evaluated. The measured responses were compared to the predicted ones, and the residual error was calculated to ensure the success of the optimization process.
2.4. Vesicle Size Determination
The vesicle size of freshly prepared FLB TRF was measured using a Zetasizer Nano ZSP (Malvern Panalytical Ltd. Malvern, UK). The result is expressed as the mean of five determinations.
2.5. Characterization of Optimized FLB TRFs
For investigation of vesicle size, polydispersity index (PDI), and zeta potential of the optimized FLB TRFs, the same method mentioned in  Section 2.4  using a Malvern size analyzer was employed. In addition, optimized FLB TRFs were subjected to transmission electron microscopy (TEM). A sample was placed on a copper grid and stained using phosphotungstic acid. After removing excess stain, the stained sample was dried and studied using a JEOL-JEM-1011 transmission electron microscope (JEOL, Tokyo, Japan).
2.6. Preparation of Optimized FLB-TRF-Loaded Hydrogel
Optimized FLB TRFs were incorporated into hydroxypropyl methyl cellulose (HPMC) based hydrogel. Briefly, specified amount of HPMC (0.1 g) was dispersed in distilled (10 mL) water to make a 1%  w / v  concentration. The gel was kept in the refrigerator overnight and then FLB TRFs were added with continuous stirring to obtain a drug concentration of 10 mg/g. Control hydrogels incorporating raw drug (10 mg/g gel) were prepared under the same conditions for comparison.
2.7. Optimized FLB TRF Gel Ex Vivo Permeation Study
Freshly excised goat nasal mucosa was utilized for ex vivo permeation studies. Mucosa were equilibrated in simulated nasal fluid (SNF) with pH 6.8 for 15 min. SNF was composed of sodium chloride (0.877%), calcium chloride (0.058%  w / v ), and potassium chloride (0.298%  w / v ) dissolved in deionized water [ 39 ]. Mucosa and gel samples were mounted between the two chambers and the donor chamber [ 40 ]. The area of the chamber of the utilized Franz automated diffusion cell (MicroettePlus; Hanson Research, Chatsworth, CA, USA) was 1.76 cm 2 . Gels loaded with optimized FLB TRF or raw FLB (0.1 g each 10 mg FLB/g gel) were utilized in this study. Seven milliliters of simulated nasal fluid (SNF) with pH 6.8 was used in the receiver chamber as the diffusion medium that was kept at 35 ± 0.5 °C with the agitation rate set at 400 rpm. At specified time intervals, 1.5 mL aliquots were withdrawn and replaced with fresh SNF.
2.8. In Vivo Pharmacokinetic Assessment
The pharmacokinetic performance of the FLB-TRF-loaded hydrogel was investigated in Wistar rats, weighing 200–250 g each, and compared to control raw-FLB-loaded gel. The study protocol was approved by the Research Ethics Committee, Faculty of Pharmacy, King Abdulaziz University, Kingdom of Saudi Arabia, under approval number (PH-124-41). The committee ensures that animal use complies with the European Union Directive 2010/63/EU and the DHEW Publication NIH 80-23 Guiding Principles. The study included two animal groups (I and II), with all animals receiving FLB dose of 10 mg/kg intranasally. Group I received raw FLB gel, and group II received FLB-TRF-loaded hydrogel. Collection of blood samples was performed at specified time intervals. Six rats from each group were sacrificed at each time interval, and the whole brain was washed with saline after removal and then weighed. Brain tissues were homogenized with phosphate buffer (pH 7.4) at 5000 rpm for 3 min. Plasma and homogenized brain samples were stored at −80 °C prior to analysis [ 41 ].
A volume of 200 μL of plasma sample, along with 200 μL of the brain homogenate, was transferred to a screw-capped test tube, mixed with 50 μL internal standard solution (valsartan, 625 ng/μL) and 1 mL acetonitrile, vortexed for 1 min, and then centrifuged at 5300 rpm for 8 min. An aliquot of the clear supernatant was transferred to a total recovery autosampler vial, and a volume of 7 μL was injected for LC-MS/MS-DAD analysis. The MS system was connected to an Agilent 1200 HPLC system equipped with an autosampler, a quaternary pump, and a column compartment (Palo Alto, CA, USA). The system was equipped with ChemStation software (Rev. B.01.03 SR2 (204)). The IT–MS was controlled using 6300 series trap control version 6.2 Build No. 62.24 (Bruker Daltonik GmbH), and the general MS adjustments were as follows: capillary voltage, 4200 V; nebulizer, 37 psi; drying gas,12 L/min; desolvation temperature, 330 °C; ion charge control (ICC) smart target, 200,000; and max accumulation time, 200 ms. The MS scan range was 50–550 m/z. For quantitative monitoring, single positive molar ion mode was applied at programed time segment, 0–4.0 min, m/z 391.2 [M+H]+ FLB; 4.0–10 min, m/z 436.3 [M+H]+ internal standard. Isocratic elution was conducted at a flow rate of 0.5 mL/min with a mobile system composed of 52% acetonitrile and 48% water containing 0.1% formic acid. FLB content in the assayed samples was quantified with reference to a calibration curve (range of 1–1000 ng/mL). The calibration curves for FLB were assessed using free-drug-plasma and free drug brain homogenate matrixes as a calibration matrix. The stock solutions of FLB and valsartan (InSt) were prepared separately by dissolving 10 mg of each in methanol to obtain a concentration of 0.1 mg/mL. A series of calibrator working solutions of FLB were prepared from its stock solutions by applying a serial dilution technique and using methanol as the diluting solvent. The calibration solutions were prepared by spiking the plasma-free drug with FLB solutions to give a concentration spanning the range of 1.0 to 1000.0 ng/mL of FLB and a fixed InSt concentration of 25 μg/mL. The calibrated solutions were extracted and analyzed by the developed method. The peak area ratios of FLB-to-InSt were found to be linear in the concentration range of 1.0 to 1000 ng/mL of FLB. Pharmacokinetic parameters including the maximum plasma concentration (C max ), time to maximum plasma concentration (T max ), and area under the plasma concentration–time curve (AUC 0–∞ ) were calculated using Kinetica software (Version 4; Thermo Fisher Scientific, Waltham, MA, USA). The parameters were analyzed for significance using SPSS software (Version 16; SPSS Inc., Chicago, IL, USA). Unpaired Student’s  t -test was performed on  C max  and AUC 0–∞ , while the nonparametric Mann–Whitney test was utilized for analysis of T max ; a level of significance of  p  &lt; 0.05 was set for all investigated pharmacokinetic parameters. For histopathological evaluation, 12 rats were divided into four groups: untreated rats (gp1), rats treated with plain in situ gel without drug (gp2), rats that received FLB drug in the in situ gel (gp3), and rats treated with FLB-Nanostructured lipid carriers (FLB-NLCs, gp4). The same dosing procedure as previously described in the pharmacokinetics study was used. After 8 h, histopathologic analysis was conducted according to the method of Young [ 42 ]. In brief, the head was removed, and the brain and jaw were removed from the head along with any other listed tissues. The nasal cavity was initially fixed in a solution of 10% formalin and then decalcified in a solution of 10% EDTA. The tissue was then placed in 70% ethanol before being embedded in paraffin, sectioned, and stained with hematoxylin and eosin.
2.9. Statistical Analysis
For the in vivo data, the software selected to perform the statistical analysis was GraphPad Prism (San Diego, CA, USA). One-way or two-way analysis of variance (ANOVA), followed by Tukey’s post hoc test, was used for multiple comparisons. Only values of  p  &lt; 0.05 were considered statistically significant. Each set of experiments was performed at least in triplicate and is reported as means ± SD. For the in vitro Box–Behnken design data, the effects of factors on the response (vesicle size) were statistically analyzed by ANOVA using the Design-Expert software.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">3. Results
3.1. Polynomial Model Selection and Diagnostic Analysis
The observed vesicle size of the prepared TRFs best fitted to the quadratic model based on its highest correlation coefficient (R 2 ) is shown  Table 2 . There was a satisfactory agreement between the predicted and adjusted R 2 , indicating that the selected model was valid for analyzing the data. Moreover, an adequate precision value of greater than 4 indicates an adequate signal-to-noise ratio, implying the suitability of the quadratic model to navigate the design space. Diagnostic plots were generated to ensure the goodness of fit of the chosen model.  Figure 1 A, illustrating the residual vs. run plots, shows randomly scattered points, indicating that there is no lurking variable interfering with the vesicle size. Furthermore, the high linearity illustrated in the predicted versus actual values plot ( Figure 1 B) indicates that the observed vesicle size was analogous to the predicted one.
3.2. Statistical Analysis for the Effect of Variables on Vesicle Size (Y)
The size of vesicles is a critical parameter that exhibits a significant impact on the drugs’ permeation via the biological membranes. FLB TRF showed size in the nanoscale range with mean size ranging from 88 ± 0.86 to 175 ± 2.43 nm ( Table 1 ). The relatively small standard deviation could indicate homogeneity of the TRF dispersions. The equation representing the selected sequential model was generated in terms of coded factors as follows:
The statistical analysis revealed that all the linear terms corresponding to the three investigated variables have a significant negative effect on FLB TRF size ( p  &lt; 0.05). The quadratic terms corresponding to the surfactant HLB ( X 2 2 ) and hydration medium pH ( X 3 2 ), in addition to the interaction term  X 2 X 3  corresponding to the interaction between the two aforementioned variables, were also found to be significant at the same significance level.  Figure 2  illustrates the contour plots for the investigated variable effects on vesicle size.
3.3. FLB TRF Optimization
The formation of the optimized FLB TRFs was accomplished using a numerical optimization technique with a minimized vesicular size. The optimized formulation was prepared at factor levels of 1:1.12 FLB:PL molar ratio, HLB value of 2.3, and hydration medium pH of 7.2. The observed and predicted values of the optimized FLB TRF formulation were in good agreement (with low error percentage), confirming the reliability of the optimization process ( Table 2 ).
3.4. Charactarization of the Optimized FLB TRFs
The PDI of the optimized formulation was found to be 0.201 ± 0.012, while the zeta potential was equal to 8.12 ± 1.54 mV. TEM has been applied for assessing of the shape and lamellarity of the optimized FLB TRF at 25,000× magnification. As illustrated in  Figure 3 , the TRF showed vesicles with spherical shape. No aggregation was observed. In addition, the recorded size was within an acceptable agreement with that recorded using the dynamic light scattering technique of the particle size analyzer.
3.5. Optimized FLB TRF Gel Ex Vivo Permeation
Ex vivo permeation through goat nasal mucosa was carried out to give an insight into the in vivo performance of the optimized FLB-TRF-loaded hydrogel.  Figure 4  illustrates the mean cumulative percent FLB permeated from the TRF-loaded hydrogel (test) compared to FLB-loaded hydrogel (control). The optimized FLB TRF hydrogel shows a significant increase in cumulative percent FLB permeated when compared to raw FLB gel ( p  &lt; 0.05), with almost complete drug permeation after 4 h. The maximum amount of drug permeated within 4 h from optimized FLB TRF hydrogel was approximately 1.97-fold greater than that from raw FLB hydrogel.
3.6. In Vivo Pharmacokinetics
The calibration curves of the concentrations of FLB spiked in plasma and brain homogenate show linear relationships with correlation coefficients of 0.9992 and 0.9984, respectively. The assay shows an adequate precision, with relative standard deviations (RSDs) of 8.1–10.9% and 10.1–12.9% for the intraday assay and the interday assay, respectively. The mean extraction recoveries were 94.8% ± 5.4% and 92.6% ± 7.6% for FLB-spiked plasma and brain samples, respectively. Mean FLB concentrations in plasma and brain following intranasal administration of optimized FLB-TRF-loaded hydrogels, compared to the control FLB-loaded hydrogels, are graphically represented in  Figure 5 . The computed pharmacokinetic parameters are compiled in  Table 3 .
From the results of the histopathological evaluation to follow the impact of FLB TRFs on the nasal tissues ( Figure 6 A–D), no pathological signs of epithelial damage, hyperplasia, edema, or inflammatory infiltration can be see for the four investigated groups.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">4. Discussion
The nanoscale size observed could contribute to enhancing the drug permeation via the nasal mucosa and facilitating passing through the blood–brain barrier. Analysis of variance (ANOVA) for the vesicle size affirmed that the quadratic model was significant ( p  &lt; 0.0001). The positive sign of the coefficients of the linear terms  X 1  and  X 2  indicates that the vesicle size increases significantly with increasing drug:PL molar ratio and/or surfactant HLB. Contrarily, the negative sign of the linear term  X 3  indicates that the vesicle size decreases significantly with increasing hydration medium pH.
The increase of size with increasing drug:PL molar ratio could be credited to increasing the PL content of the vesicles. Similar results were reported for other vesicular systems. Dubey et al. [ 43 ] demonstrated increased vesicle size of ethosomes with increasing PL content. In another study, Ahmed and Badr-Eldin [ 44 ] reported an increase in avanafil invasome size with increasing PL content of the vesicles. Regarding the HLB of the surfactant, it was observed that a significant reduction of the vesicle size occurs as the HLB is decreased. This observation could be explained on the basis of the increased hydrophobicity of the surfactant with reduced HLB values. Increased surfactant hydrophobicity could lead to reduction of surface energy and low water uptake into the vesicle core, resulting in reduction of the vesicle size [ 38 , 45 , 46 , 47 ]. The boosted FLB permeation from optimized FLB TRF gel could be attributed to the synergistic advantages of TRFs and the nanosized system. The flexible and deformable structure of the TRF could impart the potential to pass easily through the mucosal barriers. Furthermore, the existence of surfactants which act as edge activators could contribute to the permeation-enhancing ability of TRF by disrupting the lipid bilayer of the membrane [ 46 ]. In addition, the nanoscale size of the vesicles results in a great surface area, thus increased contact area with the mucosal epithelium and successively improving the chance of drug permeation [ 38 ]. Nanovesicles have been reported to have the potential to enhance drug absorption through the nasal membrane barrier and to demonstrate a high efficacy in enhancing drug bioavailability [ 40 ]. However, mucociliary clearance can help to reduce the contact time of drug-loaded nanovesicles on the mucosal surface inside the nose. Thus, the application of hydrogel-specific properties is now considered to be a useful platform for the preparation of stabilized and smart nanoscopic vehicles for drug delivery purposes. In addition, the incorporation of transferosomes into the hydrogel network can offer remote-controlled applications and also improve characteristics such as mechanical strength [ 25 , 42 , 48 , 49 ]. The observed higher extent of absorption from optimized FLB TRF hydrogel compared to the raw FLB gel could be attributed to the drug’s improved solubility and permeability by loading on a hydrophobic carrier. Comparing the two intranasal hydrogels, the optimized FLB-TRF-loaded hydrogel shows significant increases in C max  and AUC ( p  &lt; 0.05) for both plasma and brain compared to control, indicating higher bioavailability and enhanced brain delivery of the drug. This could be attributed to FLB movement from the nasal cavity along both the olfactory or trigeminal nerves to the parenchyma of the brain. FLB is delivered to the nerves in the cerebrum and pons and then disperses throughout the brain. The intracellular and extracellular pathways are the ways by which FLB brain dispersion occurs. For the intracellular mechanism, FLB is internalized by an olfactory neuron through endocytosis, trafficked within the cell to the neuron’s projection site, and then released by exocytosis. For the extracellular pathway, FLB crosses the nasal epithelium to the lamina propria and then is transported externally along the length of the neuronal axon that leads into the CNS, where FLB is distributed by fluid movement. The enhanced drug bioavailability could be ascribed to the improved permeation properties of TRFs owing to their flexible and ultra-deformable structure that enhances penetration across the mucosal barrier [ 50 ]. Furthermore, the elevated concentration of the drug in the brain highlights the capability of TRF to augment direct delivery of the drug to the brain through the nasal olfactory region and across the BBB. The nanoscale size of the vesicles might also yield a shielding effect for the drug, protecting it from fast excretion and metabolism and leading to improved CNS delivery [ 41 ].</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">5. Conclusions
TRF-loaded hydrogel has been investigated as a possible intranasal delivery system of FLB. Box–Behnken design was successfully applied for optimization of FLB TRFs with minimized vesicular size. The optimized FLB TRFs (1:1.12 drug:PL molar ratio, surfactant HLB of 2.3, and hydration medium pH of 7.2) were spherical, with a vesicle size of less than 100 nm. The optimized FLB-TRF-loaded hydrogel showed an enhanced ex vivo permeation profile through goat mucosa when compared to that of control FLB hydrogel. In vivo assessment in Wistar rats confirmed that the optimized hydrogel had higher bioavailability than the control and exhibited enhanced brain delivery. Based on these results, the proposed optimized FLB-TRF-loaded hydrogel could be considered a promising drug delivery system for nose-to-brain delivery of the drug.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Flibanserin (FLB) is a nonhormonal medicine approved by the Food and Drug Administration (FDA) to treat the hypoactive sexual appetite disorder in females. However, the peroral administration of the medicine is greatly affected by its poor bioavailability as a result of its extensive first-pass effect and poor solubility. Aiming at circumventing these drawbacks, this work involves the formulation of optimized FLB transfersome (TRF) loaded intranasal hydrogel. Box–Behnken design was utilized for the improvement of FLB TRFs with decreased size. The FLB-to-phospholipid molar ratio, the edge activator hydrophilic lipophilic balance, and the pH of the hydration medium all exhibited significant effects on the TRF size. The optimized/developed TRFs were unilamellar in shape. Hydroxypropyl methyl cellulose based hydrogel filled with the optimized FLB TRFs exhibited an improved ex vivo permeation when compared with the control FLB-loaded hydrogel. In addition, the optimized TRF-loaded hydrogel exhibited higher bioavailability and enhanced brain delivery relative to the control hydrogel following intranasal administration in Wistar rats. The results foreshadow the possible potential application of the proposed intranasal optimized FLB-TRF-loaded hydrogel to increase the bioavailability and nose-to-brain delivery of the drug.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class=""></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Author Contributions
Conceptualization, O.A.A.A. and U.A.F.; methodology, S.M.B.-E.; software, H.M.A.; validation, Z.A.A., H.Z.A., and A.K.K.; formal analysis, G.C.; investigation, F.C.; resources, A.A.; data curation, R.A.A.-G. (Raniyah A. Al-Ghamdi); writing—original draft preparation, U.A.F.; writing—review and editing, S.M.B.-E., G.C. and F.C.; visualization, R.A.A.-G. (Rawan A. Al-Ghamdi); supervision, Z.A.A.; project administration, N.A.A.; funding acquisition, N.A.A. All authors have read and agreed to the published version of the manuscript.
Funding
This project was funded by the Deanship of Scientific Research (DSR) at King Abdulaziz University, Jeddah, under grant No. RG-13–166–41. The authors, therefore, acknowledge with thanks the DSR for technical and financial support
Conflicts of Interest
The authors declare no conflict of interest. The funders had no role in the design of the study; in the collection, analyses, or interpretation of data; in the writing of the manuscript, or in the decision to publish the results.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">CC BY</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">no</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">2024-01-16 23:35:06</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Nanomaterials (Basel). 2020 Jun 29; 10(7):1270</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">oa_package/82/4e/PMC7408465.tar.gz</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="3"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">PMC7615524</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">37727079</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Introduction
Randomized controlled trials (RCTs) are considered the gold standard for assessing the causal effect of an exposure on an outcome, but are vulnerable to bias due to missingness in the outcome—or “dropout.” The impact of dropout depends on the missingness mechanism and the analysis model ( Dziura et al., 2013 ;  Little &amp; Rubin, 2020 ;  Rubin, 1976 ). Three missingness mechanisms can be distinguished: missing completely at random (MCAR), missing at random (MAR), and missing not at random (MNAR) ( Rubin, 1976 ). With MCAR, missingness is unrelated to any measured or unmeasured characteristics and the observed data are a representative subset of the full data. MAR means that the missingness can be explained by observed data and, with MNAR, missingness is a function of the unobserved data.
Two common methods of dealing with dropout are complete case analysis (CCA) and multiple imputation (MI). A CCA is the analysis model intended to be applied to the trial data at its outset, restricted only to individuals with observed outcomes. With MI, missing outcome values are repeatedly imputed conditional on the observed data, generating multiple complete datasets to which the analysis model is applied ( Buuren, 2018 ;  Little &amp; Rubin, 2020 ;  Rubin, 1987 ,  1996 ), with the resulting estimates subsequently pooled using Rubin’s rules ( Rubin, 1987 ). In practice, a CCA will be unbiased if dropout is MCAR or MAR, conditional on the analysis model covariates. MI will be unbiased if dropout is MCAR or MAR conditional on the analysis model and imputation model covariates, and if the imputation model is correctly specified. Generally, both will be biased when outcomes are MNAR ( Dziura et al., 2013 ;  Hughes et al., 2019 ;  Little &amp; Rubin, 2020 ). In this article, we consider the case of an RCT with an incomplete continuous outcome, where we assume that the outcome is generated from covariates and treatment according to a linear model. In such an RCT, a CCA will only be biased if the dropout is related the outcome, conditional on the model covariates ( Carpenter &amp; Smuk, 2021 ;  Hughes et al., 2019 ;  White &amp; Carlin, 2010 ). As RCT analyses typically adjust for a number of baseline covariates, in this article, we primarily focus on (the bias of) the treatment effect, when estimated conditional on some baseline covariate.
In the presence of dropout, observed data generally cannot be used to establish if outcomes are MNAR or MAR given the model covariates. Whether outcomes are MCAR can be partially tested using Little’s MCAR test, which compares the multivariate distribution of observed variables of patients with observed outcomes to those with unobserved outcomes ( Little, 1988 ). Little’s MCAR test and related tests, however, rely on strong parametric assumptions, with a conclusion that hinges on the specification of some arbitrary  P -value cutoff, which limits their practical value ( Li &amp; Stuart, 2019 ). Currently, there is no established statistical test for distinguishing between MAR and MNAR missingness, and consequently, no simple way to determine whether the treatment effect estimate is likely to be biased.
Current guidance for assessing risk of bias due to dropout relies on checking if dropout is differential across trial arms, assessing the plausibility that dropout may be related to outcome (e.g., dropout due to lack of efficacy) ( Higgins et al., 2012 ;  Sterne et al., 2019 ), and comparing the baseline covariate distribution across trial arms in patients who are still observed at the end of follow-up ( Groenwold et al., 2014 ). While both differential dropout across trial arms and different baseline covariate distributions across trial arms in the observed data can be caused by MNAR dropout, these markers may also result from MAR dropout. The European Medical Agency (EMA) and the National Research Council (NRC), recommend using MAR-appropriate methods for the primary analysis, followed by sensitivity analyses that weaken this assumption ( Clinical Trials o. H. M. D. iNRCUP, 2010 ;  European Medicines Agency (EMA), 2020 ). These guidelines, however, are in practice implemented in a fraction of all trials, with on average only 6% (of  N  = 330 trials) describing the assumed missing data mechanism ( Hussain et al., 2017 ;  Rombach et al., 2015 ), 9% ( N  = 237) justifying their choice of main analysis ( Rombach et al., 2015 ), and 19% ( N  = 849) reporting some kind of sensitivity analysis ( Bell et al., 2014 ;  Hussain et al., 2017 ;  Rombach et al., 2015 ;  Wood et al., 2004 ;  Zhang et al., 2017 ), which rarely involves relaxing the primary analysis assumptions ( Bell et al., 2014 ), and only 9% ( N  = 200) discussing the risk of bias resulting from missing data ( Zhang et al., 2017 ). This discrepancy between recommended and implemented practice persists despite extensive literature on the subject and may be due to the relative complexity of such analyses.
In this paper, we propose using the differences between the observed variances of the outcome across the two arms of the trial to assess the risk of CCA estimator bias due to MNAR dropout. We show, using directed acyclic graphs (DAGs) and standard statistical theory, how MNAR may give rise to unequal outcome variances between the fully observed subjects in the two arms of the trial. We illustrate this method using individual-level data and summary-level data. Individual-level patient data were obtained from an RCT investigating the benefit of an acupuncture treatment policy for patients with chronic headaches (ISRCTN96537534) ( Vickers et al., 2004 ). The summary data application used published statistics from a cluster-randomized clinical trial, which investigated psychological outcomes following a nurse-led preventive psychological intervention for critically ill patients (POPPI, registration ISRCTN53448131).</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Methods for Testing Differences in Variances
Various methods are available for testing and estimating the difference in variance between two groups, including Bartlett’s test ( Bartlett &amp; Fowler, 1937 ), Levene’s test ( Levene at al., 1960 ), the Brown–Forsythe test ( Brown &amp; Forsythe, 1974 ), the Breusch–Pagan test ( Breusch &amp; Pagan, 1979 ), and the studentized Breusch–Pagan test ( Koenker, 1981 ). In this paper, we employ the latter, as it has a straightforward implementation that allows for conditioning on additional covariates, and is also robust against nonnormally distributed errors. This is particularly relevant, as, in practice, outcomes are unlikely to be strictly normally distributed.
The studentized Breusch–Pagan estimate is obtained as follows. First, the outcome,  Y , is regressed on the treatment variable,  X , and optional additional covariates,  C , in an OLS regression: 
The regression residuals are obtained and squared   and, in a second auxiliary OLS regression, regressed on the treatment variable: 
The variance difference estimate is given by   and the test statistic is given by  nR 2 , with  n  the sample size and  R 2  the coefficient of determination, obtained from the second auxiliary regression.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Results
The simulation results are shown in  Table 2 . We first consider the seven scenarios when there is no effect modification and the treatment effects are homogeneous (A–G). When dropout is MCAR (A), and when dropout is MAR conditional on treatment (B), some measured covariate (C), or both (D), the treatment effect estimates are unbiased, irrespective of conditioning on  Y b , with, on average, a zero outcome variance difference across trial arms in the observed data at baseline   and at follow-up   We observe the same for scenario F, where dropout is MNAR dependent on  U , but with  U  and  X  independent in the observed data. In scenario G, where dropout is MNAR dependent on  U  and  X , with  U  and  X  not independent in the observed data, the CCA treatment effect estimate is biased, and the mean outcome variance difference across trial arms at follow-up is nonzero. When dropout is MNAR dependent on  Y f  (E), we observe a biased treatment effect estimate and nonzero outcome variance differences at both baseline and at follow-up. Conditioning on  Y b  results in an attenuated bias estimate and a smaller variance difference at follow-up   When effect modification is present (EM), resulting in treatment effect heterogeneity, we observe a variance difference at follow-up, regardless of dropout mechanism. In contrast, we only observe a VD at baseline in scenario E, where dropout is MNAR dependent on outcome. In  Online Appendix C.2 , a companion table ( Table C.1 ) is provided for  Table 2 , with additional measures of simulation performance.
In summary, this simulation shows that a variance difference across trial arms, in the observed data, at baseline,   indicates outcome-dependent MNAR dropout (scenario E), which may result in a biased CCA treatment effect estimate. If   is zero but the variance difference across trial arms in the observed data at follow-up,   is nonzero, then one explanation is that dropout occurs according to the MNAR missingness mechanism in scenario G, which may result in a biased treatment effect estimate. Alternatively, this could be explained by effect modification together with a missingness mechanism that is MCAR (scenario A), MAR (scenarios B, C, and D), or MNAR (scenario F), which will result in an unbiased treatment effect estimate. If both   and   are zero, then the missingness mechanism is MCAR (scenario A), MAR (scenarios B, C, and D), or MNAR (scenario F) with no effect modification and the CCA treatment effect estimate will be unbiased. Adjusting for the outcome at baseline will result in an attenuation of the CCA estimator bias and a smaller   when dropout is MNAR as in scenarios E and G, irrespective of the presence of effect modification.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Discussion
In this paper, we show that, in RCTs, the outcome variance difference across trial arms at baseline, in the set of study participants who did not drop out, is an indicator of outcome-dependent MNAR dropout, and consequently, a biased CCA treatment effect estimate. In contrast, when outcomes are MAR or MCAR, this baseline variance difference will be zero. We also show that the outcome variance difference across trial arms in the observed data at follow-up can be an indicator of both outcome-dependent MNAR dropout and nonoutcome dependent MNAR dropout, both of which may result in a biased treatment effect estimate. A variance difference across trial arms at follow-up, in the observed data, however, can only be meaningfully interpreted when the outcome variances can be expected to be equal across trial arms in the full data. This requires two assumptions: first, that there is no treatment effect heterogeneity; second, that the errors of the outcome are homoskedastic.
Treatment effect heterogeneity can be thought of as nonrandom variability in treatment response that is attributable to patient characteristics. How plausible it is that heterogeneity is absent depends strongly on intervention type and study population. Efficacy trials, for example, typically have stricter inclusion criteria, resulting in more homogeneous study populations, and are less prone to large variations in treatment response. In contrast, pragmatic trials with broad eligibility criteria are more likely to have heterogeneous treatment effects ( Varadhan &amp; Seeger, 2013 ). As treatment effect heterogeneity affects the outcome variance of the intervention arm ( Mills et al., 2021 ), it is a second potential cause of observed outcome variance differences across trial arms. The presence of effect modification can be investigated by performing a stratified analysis, where the effect modifier serves as a stratification variable ( Corraini et al., 2017 ). Alternatively, if the observed difference in outcome variance between trial arms is solely the result of effect modification, then conditioning on the effect modifier and the interaction term between effect modifier and trial arm can be expected to remove all evidence of a variance difference.
An outcome variance difference across trial arms may also result from heteroskedastic outcome errors. Specifically, if the variance of the outcome errors is related to the outcome and if treatment has a causal effect on the outcome, this may cause a difference in outcome variances across trial arms at follow-up. For example, suppose an outcome such as body mass index (BMI) has greater day-to-day variability in people with higher BMI. Then, if the treatment lowers the mean BMI in the intervention arm, this will result in a comparatively smaller intervention arm variance. A simple way to investigate this is to consider the outcome variable measured at baseline, group its values into bins, and establish if the outcome error variance is different in bins with higher and lower mean values.
We additionally propose employing the (conditional) outcome variance difference across trial arms in the observed data as an MNAR bias assessment tool, and, indirectly, as a model building tool, which can be used to assess the added value of including variables for explaining the missingness mechanism. This method is easily implemented, using existing tests available in standard software, and has a straightforward interpretation of results. In  Section 7 , we demonstrated how outcome variance differences can be used to assess the risk of MNAR bias for various models, using both individual-level data from the acupuncture trial, and summary-level data from the POPPI trial.
The outcome variance difference across trial arms at baseline and at follow-up is suitable for assessing the risk of dropout bias for analysis models that are estimated with OLS linear regression, and assume that the outcome is continuous and given by a linear model. These methods cannot be used for noncontinuous outcomes, such as binary or time-to-event outcomes.
A second limitation of our proposed method is its comparatively modest power, with the power to detect an outcome variance difference lower than the power to detect a difference in outcome means.  Brookes et al. (2004)  showed that if a trial is powered to detect a mean difference of a given size, then in order to detect an interaction effect of the same magnitude, the sample size needs to be approximately four times larger.  Mills et al. (2021)  found comparable numbers for the power to detect a variance difference.
Instead of using our method as a strict significance test with a dichotomous conclusion, we recommend assessing the practical implications of the values inside the confidence interval ( Amrhein et al., 2019 ;  Andrade, 2019 ;  Wasserstein et al., 2019 ). For example, in the individual-level data application of  Section 7 , at follow-up, the outcome variances in the observed data were 188.2 and 289.4, in the intervention usual care arm, respectively, with a variance difference of −100.3 (95% CI: −222.0, 21.4). At baseline, in the observed data, we estimated a variance difference of −81.46 between trial arms, with a 95% CI of −183.21–20.30. These results are compatible with a large negative outcome variance difference as well as a small positive outcome variance difference. These large negative variance difference would raise concerns of MNAR dropout, with the large negative outcome variance difference at baseline specifically suggestive of outcome-dependent dropout.
While a variance difference across trial arms in the observed data at baseline indicates outcome-dependent MNAR dropout, interpreting a variance difference at follow-up is less straightforward. For the latter, we suggest performing further analyses to identify the presence of heteroskedastic outcome errors and treatment effect heterogeneity, and using expert and contextual knowledge. If the presence of both heteroskedastic errors and treatment effect heterogeneity is judged to be unlikely, then the possibility of nonoutcome-dependent MNAR dropout should be investigated, for example, by conditioning on additional covariates in the analysis model or, if using MI, in the imputation model, and assessing the effect of this on the variance difference at follow-up. If the results suggest that dropout may be MNAR and that, consequently, the treatment effect estimate is at risk of bias, this motivates performing a sensitivity analysis to assess the robustness of the main analysis results under a plausible MNAR assumption. For example, we may observe a variance difference across trial arms in the observed data at baseline, suggesting outcome-dependent dropout, and also observe that the treatment effect estimate becomes smaller if we additionally condition on covariates that are correlated with the outcome. This would suggest outcome-dependent dropout that results in overestimation of the treatment effect, which may occur, for example, if, on average, more poor responders drop out. A natural subsequent step would involve performing a sensitivity analysis under the MNAR assumption of worse value dropout and investigating how strong this mechanism must be for the material conclusion to be affected.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class=""></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Randomized controlled trials (RCTs) are vulnerable to bias from missing data. When outcomes are missing not at random (MNAR), estimates from complete case analysis (CCA) and multiple imputation (MI) may be biased. There is no statistical test for distinguishing between outcomes missing at random (MAR) and MNAR. Current strategies rely on comparing dropout proportions and covariate distributions, and using auxiliary information to assess the likelihood of dropout being associated with the outcome. We propose using the observed variance difference across trial arms as a tool for assessing the risk of dropout being MNAR in RCTs with continuous outcomes. In an RCT, at randomization, the distributions of all covariates should be equal in the populations randomized to the intervention and control arms. Under the assumption of homogeneous treatment effects and homoskedastic outcome errors, the variance of the outcome will also be equal in the two populations over the course of follow-up. We show that under MAR dropout, the observed outcome variances, conditional on the variables included in the model, are equal across trial arms, whereas MNAR dropout may result in unequal variances. Consequently, unequal observed conditional trial arm variances are an indicator of MNAR dropout and possible bias of the estimated treatment effect. Heterogeneous treatment effects or heteroskedastic outcome errors are another potential cause of observing different outcome variances. We show that for longitudinal data, we can isolate the effect of MNAR outcome-dependent dropout by considering the variance difference at baseline in the same set of patients who are observed at final follow-up. We illustrate our method in simulation for CCA and MI, and in applications using individual-level data and summary data.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Notation
Let  U  be some unmeasured covariate, and  C  some measured covariate. Let  X  =  j  denote the randomized trial arms, with  j  = {0, 1}, and  X  = 0 denoting the comparator arm and  X  = 1 the intervention arm. We define the continuous outcome variable,  Y , as a linear function of  X, U , and  C  so that   with  α  some intercept,  β  the treatment effect,  γ  and  δ  the effects of  U  and  C  on  Y , respectively, and  ε Y  the mean-zero error term, with  ε Y  independent of  X, U , and  C , and with  U  and  C  additionally independent of  X .
Let  μ 1  denote the mean of the outcome,  Y , when  X  = 1,  μ 0  the mean of  Y  when  X  = 0, with   and 
As  C, U , and  ε Y  are independent of  X  so that, for example, E[ U | X  = 1] = E[ U | X  = 0], the mean difference across trial arms ( μ 1  −  μ 0 ) reduces to  β  and we can write: 
We use “full data” to refer to all data that would have been observed on all trial participants, had there been no dropout. “Observed data” refer to all data for the study participants who did not drop out. We define a response indicator  R , with  R  = 1 when  Y  is observed, and  R  = 0 when  Y  is missing. Let  Y * denote the outcome in the observed data and  β * the treatment effect estimate in the observed data: 
The bias,  B , of the CCA treatment effect estimate is given by the difference of the population treatment effect in the full data ( β ) and in the observed data ( β *): 
Optionally, the treatment effect in the observed data may be defined conditional on some observed covariate(s),  C , so that   and the bias,  B C , is given by   with   estimated in an ordinary least squares (OLS) regression of the observed outcome,  Y *, on  X  and  C .
By definition, for the linear model in  (1) , the population variance of  Y , for a given trial arm,  j , in the full data, is given by 
With  U  and  C  independent of  ε Y , all covariance terms involving  ε Y  are 0, and  (8)  reduces to 
Let VD denote the outcome variance difference across trial arms in the full data:   With  Y  generated according to  (1) ,  U, C , and  ε Y  are independent of  X  so that, for example, var( U | X  = 1) = var( U | X  = 0), resulting in an expected outcome variance difference of 0 in the full data. The assumptions necessary for this to hold are discussed further in  Section 3 .
In the observed data, the outcome variance in a given trial arm,  j , is given by   and the variance difference across trial arms by   The exact form of  (11)  is determined by the relationship between the covariates, outcome, and  R , and is explored in  Section 3 . Again, the variance may be defined conditional on  C , with the variance difference in the full data then given by   and, in the observed data, by   We can estimate  (13)  using the studentized Breusch–Pagan estimator, detailed further in  Section 4 . In the next section ( Section 3 ), we show under which dropout mechanisms the variance difference across trial arms in the observed data can be expected to be different from 0.
Mar and Mnar Dropout and Outcome Variances Across Trial Arms in the Observed Data
In an RCT, patients are randomized to treatment, after which treatment is initiated and the patients are followed up over a period of time, during which dropout may occur. Randomization makes it plausible to assume that the trial arms have equal outcome variances prior to treatment initiation in the full data. Given two additional assumptions, we can expect this to hold after treatment initiation and throughout follow-up:
Assumption A1
There is no treatment effect heterogeneity so that the treatment effect,  β , is the same for every individual.
Assumption A2
The errors are homoskedastic so that the error term of the outcome  (1) ,  ε Y , does not depend on treatment or on  Y  itself.
If these two assumptions hold, then the trial arm population outcome variances can be expected to remain the same throughout follow-up in the full data. Then, it follows that if dropout is present and the trial arm outcome variances are different in the observed data, this must be due to dropout. Here, we use directed acyclic graphs (DAGs) to describe different MAR and MNAR dropout mechanisms, and show, using graphical model theory ( Mohan &amp; Pearl, 2021 ), that the trial arm variances, conditional on the model covariates, are the same when dropout is MAR, but may be different when dropout is MNAR. Additionally, we show, for an outcome,  Y  ( Equation (1) ), generated according to a linear model, that certain types of MNAR dropout do not result in a biased treatment effect estimate or different trial arm outcome variances. We define bias,  B C   (7)  with respect to a treatment effect,    (6) , estimated while adjusting for some observed baseline covariate,  C , which is a predictor of  Y  ( Equation (1) ). Analogously, we define the variance difference as the difference in trial arm outcome variances, conditional on  C , denoted as VD C   (12)  and    (13)  in the full and observed data, respectively. Let P( Y | X ,  C ) denote the density of  Y , conditional on  X  and some observed baseline covariate,  C , in the full data, and P( Y | X ,  C,R  = 1) the corresponding density in the observed data.
Proposition 1
The densities P( Y | X ,  C ) and P( Y | X ,  C, R ) will be identical only when dropout is MCAR or MAR, with R independent of Y given the variables included in the analysis model ( R  ⫫ Y|X, C). Any quantities derived from the densities, such as the mean difference and variance difference across X, will be also the same. If  assumptions A1  and  A2  are satisfied so that the variances of the outcome in the two trial arms are equal in the full data, then P( Y | X, C )=P( Y | X, C, R ) implies that the variances of the outcome in the two trial arms are also equal in the observed data.
In  Figure 1(a) , dropout is MCAR, with the response indicator,  R , unaffected by any observed or unobserved variables.  Figures 1(b)–(d)  depict MAR dropout mechanisms, with dropout dependent on treatment,  X , on some baseline covariate,  C , and on both  X  and  C , respectively. In all four scenarios,  R  is independent of  Y  given  C  and  X , and we can show that the density of  Y , conditional on  X  and  C , in the full data, is equal to the corresponding density of  Y  in the observed data: P( Y | X, C ) = P( Y | X, C,R  = 1) (proofs given in  Online Appendix A.1 ). This has the following implications. First, the outcome mean difference across trial arms conditional on  C ,    (6) , will be the same in the full and observed data so that the CCA treatment effect estimate is unbiased, with  B C  = 0. Second, if the outcome variances are equal in  X  = 1 and  X  = 0 in the full data, then the outcome variances in the observed data can also be expected to be equal across trial arms. The latter requires that  assumptions A1  and  A2  are satisfied so that that the treatment effects are homogeneous and the outcome errors homoskedastic.
Figures 1(e)–(g)  depict MNAR dropout mechanisms, with dropout dependent on outcome,  Y , on some unobserved covariate,  U , and on treatment,  X , and  U  both. Then, for all three scenarios, we can show that  R  is not independent of  Y  given the covariates included in the analysis model, and consequently, that P( Y | X, C ) ≠ P( Y | X, C, R  = 1) (proofs given in  Online Appendix A.1 ). Different densities in the observed and full data imply that the outcome means and variances can be different also so that the CCA treatment effect estimate is biased and the outcome variance difference across trial arms nonzero. However, as we assume that  Y  ( Equation (1) ) is generated according to a linear model, an exception to this rule arises for MNAR dropout that occurs according to scenario of  Figure 1(f) .
Proposition 2
Let the outcome, Y, be defined as in  (1) . If dropout depends on unmeasured covariate, U, and if(U ⫫  X )|R, then the outcome mean difference and variance difference across trial arms in the full data can be estimated from the observed data, without conditioning on U. If  assumptions A1  and  A2  are satisfied so that the variances of the outcome in the two trial arms are equal in the full data, this implies that the variances of the outcome in the two trial arms are also equal in the observed data.
For all scenarios in  Figure 1 , we assume that the measured and unmeasured covariates are independent of treatment ( C  ⫫  X  and  U  ⫫  X ), which can be expected to hold in a randomized trial setting. Under the additional assumption of homoskedastic errors ( assumption A2 ),  ε Y  ⫫  X . Then, the treatment effect estimate,  β , can be estimated by the unconditional mean difference across trial arms, as in  (2) . If we also assume that the treatment effects are homogeneous  (A1) , all the variance components in  (9)  can be expected to be equal across trial arms (e.g., var( U | X  = 0) = var( U | X  = 1)) so that outcome variance difference across trial arms in the full data, VD  (10) , is 0. If these independencies also hold in the observed data ( U  ⫫  X | R, C  ⫫  X | R , and  ε Y  ⫫  X | R ), then the unconditional treatment effect estimate in the observed data,  β *  (4) , is equal to  β   (2)  so that the bias of the CCA treatment effect estimate,  B  = 0  (5) , and the outcome variance difference across trial arms, in the observed data, VD*  (11) , is also 0.
In  Figure 1(f) , dropout depends on unmeasured covariate,  U . While this is an MNAR dropout mechanism, it results in an unbiased CCA treatment effect estimate and no outcome variance difference across trial arms, as  U  is independent of  X  in the observed data:  U  ⫫  X | R  (proof given in  Online Appendix A.2 ). While the same reasoning could be applied to  Figure 1(g) , where dropout is MNAR dependent on  X  and  U , this, however, would require the additional assumption that the effects of  X  and  U  on  R  are independent (e.g., sicker people drop out but equally so in both trial arms). For the purposes of this paper, we do not make this assumption, and allow  X  and  U  to interact (e.g., sicker people drop out but more so in the comparator arm). Then, in the observed data,  X  and  U  are no longer independent   Consequently,  β * will be biased ( B  ≠ 0), and VD* ≠ 0. Formulae for  β *,  B , and VD* are given in  Online Appendix A.2 (Equations (A.13), (A.14), and (A.16) , respectively). Formulae for the bias and outcome variance difference when conditioning on  C ,    (6)  and    (13)  are given in  Equations (A.15) and A.17) . Note that if  U  and  C  are related, this will additionally mean that also   Conditioning on  C  will result in attenuated estimates of the CCA estimator bias,  B C , and the outcome variance difference across trial arms in the observed data,   when compared to  B  and VD*, respectively. When  C  and  U  are independent, however, conditioning on  C  will leave both estimates unaffected.
In summary, when dropout is dependent only on some covariate that is either unobserved or excluded from the model, this will not, for a linear model of  Y , result in a biased CCA treatment effect estimate or in an outcome variance difference across trial arms in the observed data, even though such dropout is strictly speaking MNAR. When dropout depends on both some unmeasured covariate and  X , and they are not independent in the observed data, this may result in bias and a variance difference.  Table 1  provides an overview of when the seven dropout scenarios of  Figure 1  result in a biased CCA estimate and an outcome variance difference across trial arms in the observed data, for a linear regression of  Y  on  X  and  C .
Note that the seven dropout scenarios of  Figure 1  and  Table 1  are illustrative settings and that we do not provide a comprehensive review of all possible settings. For example, dropout may simultaneously depend on treatment,  X  (scenario B), some observed covariate,  C , (scenario C), and on the outcome,  Y  (scenario E). If part of the dropout mechanism depends on  Y  (scenario E) or both  X  and  U  (scenario G), this is generally sufficient to cause a biased CCA estimate and an outcome variance difference across trial arms in the observed data. Under  assumptions A1  and  A2 , an outcome variance difference across trial arms serves as a marker of MNAR dropout that may result in bias. However, such MNAR dropout will not always result in an outcome variance difference, which may, because of several biases acting0. in different directions, be very small or 0. For example, dropout may depend on the outcome in such a way that the top quartile of the intervention arm and bottom quartile of the control arm drop out. Such a setting would result in a biased CCA estimate but no outcome variance difference across trial arms.
Mnar Dropout and Heterogeneous Treatment Effects in Longitudinal Data
In  Section 3 , we showed that outcome-dependent MNAR dropout and MNAR dropout dependent on  X  and  U , with  U  some unmeasured predictor of  Y , can result in a biased treatment effect estimate and an outcome variance difference across trial arms in the observed data, whereas MAR dropout will result in neither, subject to  assumptions A1  and  A2 . When  A1  and  A2  hold so that the treatment effects are homogeneous and the outcome errors are homoskedastic, the expected variance difference across trial arms in the full data is 0, which implies that a variance difference across trial arms in the observed data can be used as a marker of MNAR dropout and bias. Treatment effect heterogeneity and heteroskedastic outcome errors, however, will result in a nonzero variance difference across trial arms in both the full and observed data so that MNAR dropout is no longer the only potential cause of a variance difference. Treatment effect heterogeneity can be investigated by checking for the presence of an effect modifier, for example, by performing a stratified analysis. Heteroskedastic outcome errors can be investigated by exploring if the variability in the outcome at baseline is different for patients with lower and higher values. We elaborate on this in the applied example of  Section 7.1 .
Here, we examine the implications of violating assumption  A1  through the introduction of effect modification, which will result in a nonzero expected outcome variance difference across trial arms in the full data and in the observed data (i.e., patients with outcomes at follow-up observed). We show that when this assumption is violated, for longitudinal data, where the outcome is measured in a time series, the presence of MNAR dropout can still be assessed by looking at the outcome variance difference across trial arms in the outcome measured at the baseline.
Outcome variances across trial arms when  assumption A1  is violated
Let  Y b  denote the outcome measured at baseline, prior to treatment initiation, which is a function of covariates  U  and  C :   with  ε b  the error term.  Y b  is unaffected by  X , and with  U  and  C  independent of  ε b , the baseline outcome variance for a given trial arm,  j , is   and the outcome variance difference across trial arms at baseline is 0:   Let  Y f  denote the outcome at follow-up, which is a function of the baseline outcome,  Y b , intervention,  X , covariates  U  and  C , and effect modifier,  S :   with  ε f  the error term, which is correlated with  ε b . In  (17) ,  S  modifies the effect of  X  on  Y f , with  ζ  the effect of  S  on the outcome at follow-up,  Y f , in the intervention arm, and the average treatment effect,  β av , given by 
For simplicity, in  (17) , we do not specify a main effect of  S , and assume that the effect modification is limited to the intervention arm. More generally, an effect modifier can be expected to modify the outcome in both treatment arms. As for  (1) , we here assume that  X, U , and  C  are independent of  ε f , and that  U  and  C  are independent of  X ,,but allow  U  and  C  to be dependent. We make the same assumption for  S , but now assume, for simplicity, that  S  is independent of  U  and  C . The population variance of  Y f  in the full data, for the comparator arm, is then given by   and, for the intervention arm, by 
With  U, C, ε b , and  ε f  independent of  X ,  (19)  and  (20)  result in a variance difference in the outcome at follow-up across trial arms in the full data of 
When  ζ  ≠ 0 in  (17)  so that  S  acts as an effect modifier,  assumption A1  is violated and the outcome variance difference across trial arms at follow-up  (21)  is nonzero. As  S  ⫫  C , the covariate-adjusted variance difference, VD f (C) =VD f . The derivations of  (19) – (21)  are given in full in  Online Appendix B.1 . Note that if we allow for a dependency between  S, U , and  C ,  (20)  and  (21)  will include additional covariance terms, with, for example, for  S  and  U , the term 2 ζ ( γ  +  γ b )cov( S ,  U | X  = 1) (see  Online Appendix B.1, Equations (B.7) and (B.8) ). Also note that while we omit a main effect of  S  in  (17) , including it will not affect the expected bias or variance difference, as  S  ⫫  X .
In the presence of heterogeneous treatment effects, the outcome variance difference across trial arms in the full data,  VD f   (21) , is nonzero. Then, the outcome variance difference across trial arms in the observed data,   will also be nonzero, irrespective of the dropout mechanism, and consequently, cannot be used to assess the risk of MNAR dropout. In contrast, treatment effect heterogeneity will not result in an outcome variance difference at baseline, in either the full data (VD b , 16) or the observed data   as  Y b  is not affected by treatment. Outcome-dependent dropout, however, may result in an outcome variance difference at baseline in the observed data, as the outcome errors at baseline,  ε b , and at follow-up,  ε f , are correlated. Consequently, unlike   can be used to assess the risk of MNAR dropout and, by extension, CCA estimator bias, when  assumption A1  is violated. Similarly, if  assumption A2  does not hold, for example, when the error term depends on the outcome, this will, given that a treatment effect is present, result in a variance difference across trial arms in the full data only in the outcome at follow-up and not at baseline (derivations are given in  Online Appendix B.2 ). In the simulation below, we explore the implications of violating  assumption A1  and show that, for longitudinal data, the outcome variance difference across trial arms at baseline, in the observed data, can be used to distinguish between treatment effect heterogeneity and outcome-dependent dropout.
Methods
We performed a simulation study with the outcome at follow-up,  Y f , simulated according to  (17) , the outcome at baseline,  Y b , simulated according to  (14) , and with 1000 patients randomly assigned to each trial arm. The errors of  Y b  and  Y f ,  ε b  and  ε f , were drawn from a multivariate normal distribution, with variances of 1.5 and 2, respectively, and a correlation coefficient of 0.433. Dropout was simulated according to the seven mechanisms listed in  Table 1 , 2  and illustrated in  Figure 1 , under a logit model, with 28% overall dropout. Each scenario was simulated without effect modification ( ζ  = 0), with a true treatment effect  β  = 1 in the full data, and also with effect modification ( ζ  = 0.5), with the average treatment effect in the full data,  β av   (18) , also 1.
The outcome variance difference across trial arms in the observed data was calculated at final follow-up   and, in the same set of patients (i.e., patients with observed outcomes at follow-up), at baseline   CCA estimator bias and VD estimates were obtained conditional on observed baseline covariate,  C ,   and when additionally adjusting for   For each scenario, we obtained mean estimates of the CCA estimator bias,   and   with corresponding 95% confidence intervals (CIs). The 95% CIs were computed using the standard deviation (SD) of the relevant estimate across simulations. We simulated 1000 datasets of  N  = 2000, having verified, for each estimate, that the Monte Carlo SD (MCSD) and mean standard error were comparable, indicating that 1000 repetitions are sufficient. Additionally, we calculated the proportion of times the null was excluded from the confidence interval, as an indicator of how often a variance difference was correctly identified across simulations. A full description of the simulation framework, in accordance with ADEMP guidelines ( Morris et al., 2019 ), is given in  Online Appendix C.1 , where we describe, in detail, the simulation  a ims,  d ata-generating mechanisms,  m ethods, and  p erformance measures.
Using Conditional Trial Arm Outcome Variances to Evaluate Imputation Models
In this section, we consider the situation where there are measured covariates that are predictive of dropout and outcome, which are not included in the analysis model. In  Section 3 , we showed that for an MAR dropout mechanism, the outcome variances in the observed data are equal across trial arms, when conditioning on all variables that affect missingness. Additionally, we showed that this also holds when dropout is MNAR dependent on some unobserved variable, given that this variable is independent of treatment in the observed data. This property, in conjunction with the assumption of homogeneous treatment effects ( assumption A1 ) and homoskedastic outcome errors ( assumption A2 ), can be used to assess the plausibility of bias in a CCA analysis, by comparing the outcome variances across trial arms while conditioning on all analysis model variables. When data are missing, however, investigators may choose to use an MI approach, defining an imputation model that includes auxiliary variables that are not included in the main analysis model. In an MI model, assuming that dropout is MAR conditional on the imputation model variables and that the imputation model is correctly specified, we would expect the variance difference to be zero across the imputed datasets.
In this simulation study, we show that when dropout depends on some covariate,  C 2 , and  X , and  C 2  is excluded from the analysis model, the CCA treatment effect estimate is biased and there is an outcome variance difference across trial arms in the observed data. If  C 2  is included in the imputation model, however, fitting the same analysis model to the imputed datasets will result in an unbiased estimate of the treatment effect and no variance difference. Consequently, the outcome variance difference across trial arms in the imputed data can be used to assess the added value of including auxiliary variables in the imputation model.
Methods
We performed a simulation study with the outcome,  Y , defined according to a linear model:  Y  =  α  +  βX  +  γ 1 C 1  +  γ 2 C 2  +  ε Y , with  C 1  and  C 2  two observed independent continuous covariates, and the remaining terms defined as in  (1)  ( Section 2 ). A total of  N  = 1000 and  N  = 10,000 patients were randomized to treatment, with a true treatment effect  β  = 1, and trial arm outcome variances of 8. Dropout was simulated according to the two dropout mechanisms shown as DAGs in  Figure 2 , defined in the same manner as in  Figure 1 , with, for example, in DAG 1,  Y  affected by treatment,  X , covariates  C 1  and  C 2 , and outcomes MAR conditional on  X  and  C 2 . Dropout was simulated under a logit mechanism, with 28% overall dropout.
In the observed data, we performed a CCA linear regression conditional on  C 1 . Missing outcomes were imputed conditional on  C 1 ,  C 2 ,  X , and  Y *, generating 10 complete datasets. Treatment effect estimates adjusted for  C 1  were obtained for each dataset and subsequently pooled using Rubin’s rules ( Rubin, 1996 ). The corresponding variance differences for both models were estimated conditional on  C 1 . The 95% CIs were computed using the SD of the relevant estimate across simulations. We simulated 1000 datasets for each scenario, having verified, for each estimate, that the Monte Carlo SD and mean standard error were comparable, indicating that 1000 repetitions are sufficient. A more detailed description of the simulation framework, in accordance with ADEMP guidelines ( Morris et al., 2019 ), is given in  Online Appendix D.1 .
Results
The simulation results for  N  = 1000 are shown in  Table 3 . In scenario 1 ( Figure 2a ), dropout depends on  X  and  C 2 , and, consequently, is MNAR conditional on analysis model covariates  C 1  and  X , resulting in a biased CCA treatment effect estimate, with the corresponding outcome variance difference across trial arms nonzero. The dropout, however, is MAR conditional on  X  and  C 2 , and fitting the same analysis model, which regresses  Y  on  X  and  C 1 , to data imputed conditional on  Y*, X, C 1 , and  C 2 , results in an unbiased treatment effect estimate and a nearzero variance difference. In scenario 2 ( Figure 2b ), dropout is a function of  Y , in addition to  C 2  and  X , and consequently, MNAR conditional on the analysis model covariates and also MNAR conditional on the imputation model covariates. This results in a biased CCA estimate in the observed data and a biased treatment effect estimate in the imputed data, with, for both, nonzero outcome variance differences across trial arms. In  Online Appendix D.2 , a companion table is provided for  Table 3  ( Table D.1 ), which includes results for a sample size of  N  = 10,000 and various measures of simulation performance.
Based on the outcome variance difference in the observed and imputed data, we would conclude, for scenario 1, that including variable  C 2  in the imputation model will result in a less biased estimate, while, for scenario 2, we would infer that the imputation model fails to address the dropout mechanism, suggesting that the data are MNAR given the variables in the imputation model.
Previously, in  Section 3 , we showed that if dropout depends on some covariate and  X , and the two are not independent in the observed data, including the covariate in the analysis model will result in an unbiased CCA estimate and no outcome variance difference across trial arms in the observed data. Here, we show, when using MI, that if this covariate is omitted from the analysis model but included in the imputation model, the resulting treatment effect estimate will be unbiased and the outcome variance difference will be zero. Consequently, the outcome variance difference across trial arms in the imputed data can be used to assess the added value of including variables in the imputation model for explaining the missingness mechanism.
Application
An application using individual-level data from the acupuncture trial
We now apply our method to individual-level data from an RCT, which compared the effect of two treatments on 401 patients suffering from chronic headaches ( Vickers et al., 2004 ;  Vickers, 2006 ). The primary outcome was the headache score at 12 months, with higher values indicating worse symptoms. Patients were randomly allocated to acupuncture intervention ( N  = 205) or usual care ( N  = 196). The trial found a beneficial effect of acupuncture treatment, with a mean difference in headache score of −4.6 (95% CI: −7.0, −2.2), adjusted for baseline headache score and minimization variables age, sex, headache type, number of years of headache disorder, and site (general practices in England and Wales). At 12 months, 21% and 29% of patients in the acupuncture and usual care arm, respectively, had dropped out. The investigators noted that while dropouts were generally comparable across the two arms, their baseline headache score was on average higher.
Existing methods for assessing risk of bias due to dropout include checking if dropout is differential across trial arms ( Higgins et al., 2012 ;  Sterne et al., 2019 ), and if baseline covariate distributions are different across trial arms in patients who are still observed at the end of follow-up ( Groenwold et al., 2014 ). We assessed the relationship between trial arm and dropout by performing a logistic regression of the dropout indicator on treatment, which yielded an association of 0.38 (95% CI: −0.07, 0.84), with the CI just including the null. Note, however, that biased and unbiased treatment effect estimates can be obtained both when dropout is balanced and differential ( Bell et al., 2013 ). We compared the baseline covariate distributions across trial arms by performing linear regressions of each covariate included in the primary analysis model on treatment, using the subset of patients still observed at 12 months, with the covariates standardized to facilitate comparisons. The largest point estimate and narrowest confidence interval were observed for the headache score at baseline (0.14, 95% CI: −0.09, 0.37), though the latter also included the null ( Table 4 ). As for differential dropout, both MAR and MNAR dropout mechanisms can result in different baseline covariate distributions across trial arms, and consequently, neither method is a unique marker for the presence of MNAR dropout.
Using the outcome variance differences across trial arms at 12 months (VD 12 ) and at baseline (VD b ), we assessed the risk of bias due to MNAR dropout for an unadjusted CCA model (M1), a model adjusted for the minimization variables (M2), and a model adjusted for the minimization variables and baseline headache score (M3). VD 12  and VD b  were estimated using the studentized Breusch–Pagan test, for the subset of patients still observed at 12 months. Results are reported in  Table 5 .
The unadjusted model (M1), regressing headache score on treatment, showed a beneficial effect of acupuncture therapy (−6.1, 95% CI: −9.6, −2.6). At 12 months, the outcome variances in the acupuncture arm and usual care arm were 188.2 and 289.4, respectively, with a variance difference, VD 12 =−100.3 (95% CI: −222.0, 21.4). This result is compatible with a large negative variance difference (−222.0), with a smaller outcome variance in the acupuncture arm, and a small positive variance difference (21.4), with a smaller outcome variance in the usual care arm. At baseline, we estimated an outcome variance difference of VD b  = −81.5 (95% CI: −183.2, 20.3), which is once again compatible with a large negative variance difference and a small positive variance difference. A substantial outcome variance difference at baseline raises concerns of outcome-dependent MNAR dropout, whereas, at follow-up, an outcome variance difference may have multiple causes: MNAR dropout, heteroskedastic errors, or treatment effect heterogeneity resulting from effect modification. Adjusting for the five minimization variables (M2) did not affect the estimated treatment effect or VD 12 , whereas additionally adjusting for baseline headache score (M3) resulted in an attenuated treatment effect of −4.64 (95% CI: −7.08, −2.19) and a greatly reduced positive VD 12  of 21.23 (95% CI: −26.83, 69.30), with much tighter confidence bounds. In  Section 5.3 , we showed that when dropout is MNAR, conditioning on the outcome at baseline results in a smaller outcome variance difference at follow-up and attenuation of the CCA estimator bias.
However, in the event that the outcome variance difference at follow-up is the result of treatment effect heterogeneity, conditioning on an effect modifier will also result in a decreased variance difference ( Mills et al., 2021 ). A simple way to check if a variable is an effect modifier is to perform a stratified analysis. We repeated the regression of M3 ( Table 5 ) in patients with baseline headache scores below the mean, and in patients with scores above the mean, yielding estimates of −2.92 (95% CI: −5.15, −0.69) and −6.66 (95% CI: −12.49, −0.83), respectively. The difference in treatment effect estimate in patients with lower and higher baseline scores suggests that the headache score at baseline may act as an effect modifier. For comparison, we performed an analogous analysis dividing the patients according to age, which showed no difference in treatment effect estimate between patients below mean age (−4.76; 95% CI: −8.89, −0.63) and above mean age (−4.57; 95% CI: −7.73, −1.40). This suggests that the baseline headache score may be acting as an effect modifier, which would imply that the observed outcome variance difference at 12 months may at least in part be the result of effect modification in the intervention arm.
An outcome variance difference at follow-up may also result from the presence of heteroskedastic outcome errors. This can be assessed by checking if variability in the outcome at baseline is different for patients with lower and higher values. We did this by ordering the baseline headache score values and dividing them into six bins. The variances across bins showed no evidence against homoskedastic outcome errors, with no corresponding increase or decrease in variance observed ( Table E.1, Online Appendix E ).
In summary, our results suggest that the CCA estimate, adjusted for minimization variables and baseline headache score may be biased due to outcome-dependent dropout, and that the true treatment effect may be more modest. The magnitude of this bias is, however, likely partly reduced by conditioning on the baseline headache score. The bias can be expected to be further attenuated when conditioning on additional variables that are predictors of the outcome, either by including them in the main analysis model or, if using an MI approach, in the imputation model. In the original trial publication ( Vickers et al., 2004 ), the authors additionally obtained the treatment effect estimate when imputing the 12-month dropouts using auxiliary variables that were highly correlated with the headache score at 12 months, including the headache score measured at a previous time point (3 months) and a post-hoc global assessment of headache severity. This yielded a smaller treatment effect estimate of −3.9 (95% CI: −6.4, −1.4), when compared to the CCA estimate adjusted for minimization variables and baseline headache score (−4.6; 95% CI: −7.1, −2.2). This attenuation on imputing the missing 12-month outcomes using variables highly correlated with the outcome further supports our conclusion that the CCA treatment effect estimate is likely affected by outcome-dependent dropout. This can be further investigated by performing sensitivity analyses under assumption of a dropout mechanisms that is MNAR dependent on the outcome.
Note that observing a variance difference at baseline is sufficient to raise concern of outcome-dependent dropout. Further investigating the variance difference at follow-up, as we do here, is then not strictly necessary, though it may nevertheless be interesting for interpretation purposes. Investigating the possibility of heterogeneous treatment effects and heteroscedastic errors, however, becomes necessary when there is a non-outcome dependent MNAR dropout mechanism, which will only result in an outcome variance difference across trial arms at follow-up but not at baseline.
An application using summary-level data from the POPPI trial
The POPPI trial investigated whether a preventive, complex psychological intervention, initiated in the intensive care unit (ICU), would reduce the development of subsequent posttraumatic stress disorder (PTSD) symptoms at 6 months in ICU patients ( Wade et al., 2019 ). Symptom severity was quantified using the PTSD symptom scale-self-report (PSS-SR) questionnaire, with higher values indicating greater severity. Twenty-four ICUs were randomized to intervention or control, with intervention ICUs providing usual care during a baseline period and the preventive intervention during the intervention period, and control ICUs providing usual care throughout. At 6 months follow-up, 79.3% of patients had completed the PSS-SR questionnaire, with no difference across study arms. The trial found no beneficial effect of intervention, with a mean difference in PSS-SR score of −0.03 (95% CI: −2.58, 2.52), adjusted for age, sex, race/ethnicity, deprivation, preexisting anxiety/depression, planned admission following elective surgery, and the Intensive Care National Audit &amp; Research Centre (ICNARC) Physiology Score.
Using summary statistics from the published study, we performed a  t -test for the variance difference ( Mills et al., 2021 ) between trial arms at 6 months, to assess if the study’s reported null result may have been biased by MNAR dropout. Published estimates were means with 95% CIs, adjusted for the previously listed variables, which we used to calculate the outcome variances and corresponding variance differences). We found no evidence for a variance difference across trial arms at baseline (VID b , = 11.2; 95% CI: −22.7, 45.2), but a greater variance in the intervention arm at 6 months (VD 6  = 52.5; 95% CI: 18.8, 86.2).
No variance difference across trial arms in the outcome at baseline indicates that dropout is not outcome-dependent, whereas the variance difference at follow-up may be the result of treatment effect heterogeneity, heteroskedastic outcome errors, or MNAR dropout that does not depend on outcome, as in scenario of  Figure 1(g) , where dropout depends on some unobserved covariate,  U , and treatment. For nonoutcome-dependent MNAR dropout to result in bias and a variance difference, this requires  U  to interact with treatment in the dropout mechanism, which will result in differential dropout across trial arms. In the POPPI trial data, however, dropout was balanced at 6 months follow-up, suggesting that nonoutcome-dependent MNAR dropout is unlikely to be the cause of the observed outcome variance difference across trial arms at follow-up. An outcome variance difference at follow-up may also be the result of heteroskedastic outcome errors. This, however, requires that a treatment effect be present, whereas, here, a null treatment effect was estimated. In summary, our results suggest that there is no MNAR dropout in the POPPI trial at 6 months follow-up, but that there may be treatment effect heterogeneity, which, on average, results in a null treatment effect. Further investigation into potential effect modifiers would require access to individual-level data.
Supplementary Material</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Acknowledgments
We would like to thank Dr Andrew Vickers and the acupuncture trial team for making the data from “Acupuncture for chronic headache in primary care: large, pragmatic, randomized trial” publicly available. AH, TP and KT were supported by the Integrative Epidemiology Unit, which receives funding from the UK Medical Research Council and the University of Bristol (MC_UU_00011/3). KHW is affiliated to the Integrative Cancer Epidemiology Programme (ICEP), and works within the Medical Research Council Integrative Epidemiology Unit.
Funding information
University of Bristol; Medical Research Council, Grant/Award Number: MC_UU_00011/3
Data Availability Statement
Individual-level patient data from the acupuncture trial (ISRCTN96537534) are publicly available and can be found in the supplementary materials of  Vickers 2006 . This trial was approved by the South West Multicentre Research Ethics Committee and appropriate local ethics committees.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">CC BY</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">no</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">2024-01-16 23:47:20</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Biom J. 2023 Dec 1; 65(8):e2200116</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">oa_package/98/16/PMC7615524.tar.gz</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="4"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">PMC7615526</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">38160938</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Introduction\nLysosomal acid lipase (LAL) hydrolyzes cholesteryl esters (CE) and triacylglycerols (<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Materials and Methods\nAnimals\nMale young (8–12 or 12–17 weeks of age) and mature (40–55 wee<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Results\nReduced cross-sectional area and SM mass in Lal −/− mice\nTo study the consequences of<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Discussion\nThe impact of LAL-D on the pathophysiology of SM is still unclear, although decreased m<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Conclusion\nTaken together, our data provide conclusive evidence that whole-body loss of LAL affect<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Objective\nLysosomal acid lipase (LAL) is the only enzyme known to hydrolyze cholesteryl esters (CE<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Supplementary Material</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Acknowledgments\nThis work was supported by the Austrian Science Fund FWF (SFB 10.55776/F73, DK-MCD<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">CC BY</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">no</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">2024-01-16 23:47:21</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Mol Metab. 2023 Dec 30;:101869</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">oa_package/db/ce/PMC7615526.tar.gz</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="5"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">PMC8550499</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">34723203</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Introduction\nAs a consequence of large-scale phenomena such as globalization, social integration, <span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class=""></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class=""></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class=""></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Conclusions\nThroughout this paper I tried to show that computational methods can play an important<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"In this paper I argue in favour of the adoption of an interdisciplinary approach based on computati<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Language policy and complexity\nPublic policy can be defined as “ an intentional course of action<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Funding\nOpen access funding provided by University of Geneva.\nData availability\nThe manuscript h<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">CC BY</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">no</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">2024-01-16 23:35:01</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">SN Soc Sci. 2021 Aug 2; 1(8):197</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">oa_package/eb/8e/PMC8550499.tar.gz</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="6"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">PMC9002620</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">35406088</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"1. Introduction\nHomocysteine (Hcy) is a sulfhydryl-containing amino acid that is produced when met<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"2. Materials and Methods\n2.1. Animals\nMale Sprague Dawley rats (250 g, Envigo, Milan, Italy) were<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"3. Results\n3.1. Effect of Cashew Nuts on Serum Hcy Levels after Methionine Administration\nIn the <span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"4. Discussion\nHHcy is a methionine metabolism abnormality that can cause a variety of disorders in<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class=""></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"These authors contributed equally to this work.\nThese authors contributed equally to this work.\nH<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class=""></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Acknowledgments\nWe would like to acknowledge Salma Seetaroo from Ivorienne de Noix de Cajou S.A. o<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">CC BY</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">no</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">2024-01-16 23:35:07</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Nutrients. 2022 Apr 1; 14(7):1474</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">oa_package/5c/7a/PMC9002620.tar.gz</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="7"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">PMC9094045</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">35573302</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Introduction\nAcetylcholine is an important neurotransmitter for both the maintenance of internal h<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class=""></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class=""></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class=""></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Conclusion\nIn conclusion, experimental models such as the Chrna9 KO mouse have allowed the develop<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Edited by: Victoria M. Bajo Lorenzana, University of Oxford, United Kingdom\nReviewed by: Adrian Ro<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Auditory Efferent System\nThe auditory efferent system is a neural network that originates in the a<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class=""></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">CC BY</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">no</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">2024-01-16 23:36:46</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Front Neurosci. 2022 Apr 27; 16:866161</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">oa_package/d6/a7/PMC9094045.tar.gz</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="8"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">PMC9117295</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">35115661</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Introduction\nGenetic variations at the  DLG2  gene locus are linked to multiple psychiatric disord<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Materials and methods\nAnimals and husbandry\nAll procedures were carried out under local instituti<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Results\nDlg2 + /−  heterozygous knockout rats were generated by CRISPR-Cas9 targeting of the  Dl<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Discussion\nNMDAR currents are increased in the  Dlg2 + /−  heterozygous rat model. Additionally,<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class=""></span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Copy number variants indicating loss of function in the  DLG2  gene have been associated with marke<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Supplementary information
     </span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Supplementary information\nThe online version contains supplementary material available at 10.1038/<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">CC BY</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">no</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">2024-01-16 23:35:01</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Neuropsychopharmacology. 2022 Jun 3; 47(7):1367-1378</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">oa_package/84/a0/PMC9117295.tar.gz</span></div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="9"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">PMC9363242</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">35944931</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Background\nDescription of the condition\nHigh blood pressure is the leading cause of preventable d<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Methods\nCriteria for considering studies for this review\nTypes of studies\nThe Populations, Inter<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Results\nDescription of studies\nFor detailed information, see  Characteristics of included studies<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Discussion\nSummary of main results\nThis review examined the effects and safety of LSSS compared t<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Authors' conclusions

</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Abstract\nBackground\nElevated blood pressure, or hypertension, is the leading cause of preventable<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Summary of findings\nObjectives\nTo assess the effects and safety of replacing salt with LSSS to re<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto">"Acknowledgements\nThe World Health Organization (WHO) provided funding to Stellenbosch University t<span class="text-orange-500">(...TRUNCATED)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">CC BY</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">no</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">2024-01-16 23:36:46</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Cochrane Database Syst Rev. 2022 Aug 10; 2022(8):CD015207</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">oa_package/97/5a/PMC9363242.tar.gz</span></div></div>
							</td>
					</tr></tbody></table>
		<div class="bg-linear-to-b sticky left-0 border-t border-dashed border-gray-300 from-gray-100 to-white py-3 text-center font-mono text-xs dark:border-gray-700 dark:from-gray-950 dark:to-gray-900">End of preview. <a href="/datasets/TomTBT/pmc_open_access_section/viewer/commercial/train" class="group"><span class="underline decoration-gray-300 group-hover:decoration-gray-400 dark:decoration-gray-500 dark:group-hover:decoration-gray-300">Expand</span>
						in <svg class="text-lg mr-0.5 inline -translate-y-px text-red-500" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 12 12"><path fill="currentColor" d="M2.5 2h7a1 1 0 0 1 1 1v6a1 1 0 0 1-1 1h-7a1 1 0 0 1-1-1V3a1 1 0 0 1 1-1Zm0 2v2h3V4h-3Zm4 0v2h3V4h-3Zm-4 3v2h3V7h-3Zm4 0v2h3V7h-3Z"></path></svg>Data Studio
					</a></div></div>




			<div class="bg-linear-to-b from-gray-100 to-white dark:from-gray-950 dark:to-gray-900 "><hr class="flex-none -translate-y-px border-t border-dashed border-gray-300 bg-white dark:border-gray-700 dark:bg-gray-950">
					<nav><ul class="flex select-none items-center justify-between space-x-2 text-gray-700 sm:justify-center py-1 text-center font-mono text-xs rounded-b-lg"><li><a class="flex items-center rounded-lg px-2.5 py-1 hover:bg-gray-50 dark:hover:bg-gray-800 pointer-events-none cursor-default text-gray-400 hover:text-gray-700" href=""><svg class="mr-1.5" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M10 16L20 6l1.4 1.4l-8.6 8.6l8.6 8.6L20 26z" fill="currentColor"></path></svg>
		Previous</a></li>
			<li class="hidden sm:block"><a class="rounded-lg px-2.5 py-1 bg-gray-50 font-semibold ring-1 ring-inset ring-gray-200 dark:bg-gray-900 dark:text-yellow-500 dark:ring-gray-900 hover:bg-gray-50 dark:hover:bg-gray-800" href="/datasets/TomTBT/pmc_open_access_section/viewer/commercial/train?p=0">1</a>
				</li><li class="hidden sm:block"><a class="rounded-lg px-2.5 py-1  hover:bg-gray-50 dark:hover:bg-gray-800" href="/datasets/TomTBT/pmc_open_access_section/viewer/commercial/train?p=1">2</a>
				</li><li class="hidden sm:block"><a class="rounded-lg px-2.5 py-1  hover:bg-gray-50 dark:hover:bg-gray-800" href="/datasets/TomTBT/pmc_open_access_section/viewer/commercial/train?p=2">3</a>
				</li><li class="hidden sm:block"><a class="rounded-lg px-2.5 py-1  pointer-events-none cursor-default" href="#">...</a>
				</li><li class="hidden sm:block"><a class="rounded-lg px-2.5 py-1  hover:bg-gray-50 dark:hover:bg-gray-800" href="/datasets/TomTBT/pmc_open_access_section/viewer/commercial/train?p=2119">2,120</a>
				</li>
			<li><a class="flex items-center rounded-lg px-2.5 py-1 hover:bg-gray-50 dark:hover:bg-gray-800 " href="/datasets/TomTBT/pmc_open_access_section/viewer/commercial/train?p=1">Next
		<svg class="ml-1.5 transform rotate-180" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M10 16L20 6l1.4 1.4l-8.6 8.6l8.6 8.6L20 26z" fill="currentColor"></path></svg></a></li></ul></nav></div></div>






</div></div></div></div></div></div></div>

				
					<div class="SVELTE_HYDRATER contents" data-target="RepoCodeCopy" data-props="{}"><div></div></div>
					

					<div class="SVELTE_HYDRATER contents" data-target="SideNavigation" data-props="{&quot;titleTree&quot;:[],&quot;classNames&quot;:&quot;top-6&quot;}">

</div>
					<div class="mb-6 w-full rounded-lg border border-gray-100 bg-gray-50 px-3 py-2 leading-snug dark:bg-gray-900"><span class="text-smd text-gray-600">README.md exists but content is empty.
								</span></div></section>
			
			<section class="pt-6 border-gray-100 md:pb-24 md:pl-6 md:w-64 lg:w-80 xl:w-96 flex-none order-first md:order-none md:border-l pt-3! md:pt-6!"><dl class="flex items-baseline justify-between"><dt class="text-sm text-gray-500">Downloads last month</dt><div class="mx-4 flex-1 border-b border-dotted"></div><dd class="font-semibold">566</dd></dl>
				
				<div class="divider-column-vertical"></div>
				<div class="grid grid-cols-2 gap-x-2 md:flex md:flex-row md:flex-wrap"><div class="SVELTE_HYDRATER contents" data-target="DatasetAndModelActionsDropdown" data-props="{&quot;classNames&quot;:&quot;order-last&quot;,&quot;discussionsDisabled&quot;:false,&quot;repo&quot;:{&quot;type&quot;:&quot;dataset&quot;,&quot;name&quot;:&quot;TomTBT/pmc_open_access_section&quot;},&quot;canWrite&quot;:false,&quot;canDisable&quot;:false,&quot;repoIsPrivate&quot;:false,&quot;repoIsGated&quot;:false,&quot;repoIsDisabled&quot;:false,&quot;repoIsAdminFlaggedNFAA&quot;:false,&quot;repoHasBlockedOids&quot;:false}"><div class="order-last"><div class="relative ">
	<button class="btn px-1.5 py-1.5 " type="button">
		
			<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" class="p-0.5" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><circle cx="16" cy="7" r="3" fill="currentColor"></circle><circle cx="16" cy="16" r="3" fill="currentColor"></circle><circle cx="16" cy="25" r="3" fill="currentColor"></circle></svg>
		
		</button>
	
	
	</div></div>













</div>
					<div class="SVELTE_HYDRATER contents" data-target="DatasetLibrary" data-props="{&quot;classNames&quot;:&quot;md:w-full xl:w-auto xl:flex-none&quot;,&quot;libraries&quot;:[{&quot;language&quot;:&quot;python&quot;,&quot;library&quot;:&quot;datasets&quot;,&quot;function&quot;:&quot;load_dataset&quot;,&quot;loading_codes&quot;:[{&quot;config_name&quot;:&quot;commercial&quot;,&quot;arguments&quot;:{&quot;config_name&quot;:&quot;commercial&quot;},&quot;code&quot;:&quot;from datasets import load_dataset\n\nds = load_dataset(\&quot;TomTBT/pmc_open_access_section\&quot;, \&quot;commercial\&quot;)&quot;},{&quot;config_name&quot;:&quot;non_commercial&quot;,&quot;arguments&quot;:{&quot;config_name&quot;:&quot;non_commercial&quot;},&quot;code&quot;:&quot;from datasets import load_dataset\n\nds = load_dataset(\&quot;TomTBT/pmc_open_access_section\&quot;, \&quot;non_commercial\&quot;)&quot;},{&quot;config_name&quot;:&quot;other&quot;,&quot;arguments&quot;:{&quot;config_name&quot;:&quot;other&quot;},&quot;code&quot;:&quot;from datasets import load_dataset\n\nds = load_dataset(\&quot;TomTBT/pmc_open_access_section\&quot;, \&quot;other\&quot;)&quot;}]},{&quot;language&quot;:&quot;python&quot;,&quot;library&quot;:&quot;mlcroissant&quot;,&quot;function&quot;:&quot;Dataset&quot;,&quot;loading_codes&quot;:[{&quot;config_name&quot;:&quot;commercial&quot;,&quot;arguments&quot;:{&quot;record_set&quot;:&quot;commercial&quot;,&quot;partial&quot;:true},&quot;code&quot;:&quot;from mlcroissant import Dataset\n\n# The Croissant metadata exposes the first 5GB of this dataset\nds = Dataset(jsonld=\&quot;https://huggingface.co/api/datasets/TomTBT/pmc_open_access_section/croissant\&quot;)\nrecords = ds.records(\&quot;commercial\&quot;)&quot;},{&quot;config_name&quot;:&quot;non_commercial&quot;,&quot;arguments&quot;:{&quot;record_set&quot;:&quot;non_commercial&quot;,&quot;partial&quot;:true},&quot;code&quot;:&quot;from mlcroissant import Dataset\n\n# The Croissant metadata exposes the first 5GB of this dataset\nds = Dataset(jsonld=\&quot;https://huggingface.co/api/datasets/TomTBT/pmc_open_access_section/croissant\&quot;)\nrecords = ds.records(\&quot;non_commercial\&quot;)&quot;},{&quot;config_name&quot;:&quot;other&quot;,&quot;arguments&quot;:{&quot;record_set&quot;:&quot;other&quot;,&quot;partial&quot;:true},&quot;code&quot;:&quot;from mlcroissant import Dataset\n\n# The Croissant metadata exposes the first 5GB of this dataset\nds = Dataset(jsonld=\&quot;https://huggingface.co/api/datasets/TomTBT/pmc_open_access_section/croissant\&quot;)\nrecords = ds.records(\&quot;other\&quot;)&quot;}]}]}"><div class="relative md:w-full xl:w-auto xl:flex-none">
	<button class="from-gray-800! to-black! max-xl:mb-2 text-white! gap-1! border-gray-800! dark:border-gray-900!  btn w-full cursor-pointer text-sm" type="button">
		<svg class="mr-1.5 mr-0.5! " xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32" style="transform: rotate(360deg);"><path d="M31 16l-7 7l-1.41-1.41L28.17 16l-5.58-5.59L24 9l7 7z" fill="currentColor"></path><path d="M1 16l7-7l1.41 1.41L3.83 16l5.58 5.59L8 23l-7-7z" fill="currentColor"></path><path d="M12.419 25.484L17.639 6l1.932.518L14.35 26z" fill="currentColor"></path></svg>
			Use this dataset
		<svg class="-mr-1 text-gray-500" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 24 24"><path d="M16.293 9.293L12 13.586L7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z" fill="currentColor"></path></svg></button>
	
	
	</div>

</div>
					
					</div>
				<div class="divider-column-vertical"></div>
					<div class="flex flex-col flex-wrap xl:flex-row"><a class="bg-linear-to-r dark:via-none group mb-1.5 flex max-w-full flex-col overflow-hidden rounded-lg border border-gray-100 from-white via-white to-white px-2 py-1 hover:from-gray-50 dark:from-gray-900 dark:to-gray-925 dark:hover:to-gray-900 md:mr-1.5 " href="/datasets/TomTBT/pmc_open_access_section/tree/refs%2Fconvert%2Fparquet/" rel="nofollow" target="_blank"><div class="truncate text-xs text-gray-400">Size of the auto-converted Parquet files (First 5GB per split):</div>
		<div class="truncate text-sm group-hover:underline">
			<!-- HTML_TAG_START -->7.86 GB<!-- HTML_TAG_END --></div></a><a class="bg-linear-to-r dark:via-none group mb-1.5 flex max-w-full flex-col overflow-hidden rounded-lg border border-gray-100 from-white via-white to-white px-2 py-1 hover:from-gray-50 dark:from-gray-900 dark:to-gray-925 dark:hover:to-gray-900 md:mr-1.5 pointer-events-none" rel="nofollow" target="_blank"><div class="truncate text-xs text-gray-400">Number of rows (First 5GB per split):</div>
		<div class="truncate text-sm ">
			<!-- HTML_TAG_START -->720,000<!-- HTML_TAG_END --></div></a><a class="bg-linear-to-r dark:via-none group mb-1.5 flex max-w-full flex-col overflow-hidden rounded-lg border border-gray-100 from-white via-white to-white px-2 py-1 hover:from-gray-50 dark:from-gray-900 dark:to-gray-925 dark:hover:to-gray-900 md:mr-1.5 pointer-events-none" rel="nofollow" target="_blank"><div class="truncate text-xs text-gray-400">Estimated number of rows:</div>
		<div class="truncate text-sm ">
			<!-- HTML_TAG_START -->7,219,374<!-- HTML_TAG_END --></div></a></div>
				
				
				
				

				
				<div class="divider-column-vertical md:hidden"></div></section></div></main>

	<footer class="b-12 mb-2 flex border-t border-gray-100 md:h-14"><nav class="container relative flex flex-col justify-between space-y-2 py-6 text-gray-500 max-md:*:self-start md:flex-row md:items-center md:space-y-0 md:py-0 md:text-sm"><div class="SVELTE_HYDRATER contents" data-target="ThemeSwitcher" data-props="{&quot;theme&quot;:&quot;system&quot;,&quot;isLoggedIn&quot;:false,&quot;menuClassNames&quot;:&quot;md:-top-24&quot;,&quot;classNames&quot;:&quot;max-md:mb-5 max-md:*:self-start&quot;}">
<div class="relative inline-block max-md:mb-5 max-md:*:self-start">
	<button class="rounded-full border border-gray-100 pl-2 py-1 pr-2.5  flex items-center text-sm text-gray-500 bg-white hover:bg-purple-50 hover:border-purple-200 dark:hover:bg-gray-800 dark:hover:border-gray-950 dark:border-gray-800 " type="button">
		<svg class="mr-1.5 text-gray-500" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" fill="currentColor" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M29 25H3a1 1 0 1 0 0 2h26a1 1 0 1 0 0-2Z" fill="currentColor"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M6 22.5h20a2 2 0 0 0 2-2V7a2 2 0 0 0-2-2H6a2 2 0 0 0-2 2v13.5a2 2 0 0 0 2 2ZM7 7a1 1 0 0 0-1 1v11a1 1 0 0 0 1 1h18a1 1 0 0 0 1-1V8a1 1 0 0 0-1-1H7Z" fill="currentColor"></path><path d="M6 8a1 1 0 0 1 1-1h18a1 1 0 0 1 1 1v11a1 1 0 0 1-1 1H7a1 1 0 0 1-1-1V8Z" fill="currentColor" fill-opacity=".4"></path><path d="M29 25H3a1 1 0 1 0 0 2h26a1 1 0 1 0 0-2Z" fill="currentColor"></path></svg>
			System theme
		</button>
	
	
	</div></div>
		<div class="font-semibold text-black md:hidden">Company</div>
		<a class="hover:underline" href="/terms-of-service">TOS</a>
		<a class="hover:underline" href="/privacy">Privacy</a>
		<a class="hover:underline" href="/huggingface">About</a>
		<a class="hover:underline" href="https://apply.workable.com/huggingface/">Jobs</a>
		<a href="/" class="max-md:mb-4! max-md:mt-8! group flex-none max-md:order-last"><svg class="h-7 w-7 transition-transform group-hover:-translate-y-px" viewBox="0 0 95 88" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M47.2119 76.5C66.4037 76.5 81.9619 60.9419 81.9619 41.75C81.9619 22.5581 66.4037 7 47.2119 7C28.02 7 12.4619 22.5581 12.4619 41.75C12.4619 60.9419 28.02 76.5 47.2119 76.5Z" fill="#FFD21E"></path><path d="M81.9619 41.75C81.9619 22.5581 66.4037 7 47.2119 7C28.02 7 12.4619 22.5581 12.4619 41.75C12.4619 60.9419 28.02 76.5 47.2119 76.5C66.4037 76.5 81.9619 60.9419 81.9619 41.75ZM8.46185 41.75C8.46185 20.349 25.8108 3 47.2119 3C68.6129 3 85.9619 20.349 85.9619 41.75C85.9619 63.151 68.6129 80.5 47.2119 80.5C25.8108 80.5 8.46185 63.151 8.46185 41.75Z" fill="#FF9D0B"></path><path d="M58.5024 32.2915C59.7768 32.7415 60.2839 35.3615 61.5713 34.6769C64.0095 33.3805 64.9351 30.353 63.6387 27.9148C62.3423 25.4767 59.3148 24.5511 56.8766 25.8475C54.4384 27.1439 53.5128 30.1714 54.8092 32.6096C55.4211 33.7604 57.3632 31.8892 58.5024 32.2915Z" fill="#3A3B45"></path><path d="M34.9454 32.2915C33.671 32.7415 33.164 35.3615 31.8766 34.6769C29.4384 33.3805 28.5128 30.353 29.8092 27.9148C31.1056 25.4767 34.1331 24.5511 36.5713 25.8475C39.0095 27.1439 39.9351 30.1714 38.6387 32.6096C38.0268 33.7604 36.0846 31.8892 34.9454 32.2915Z" fill="#3A3B45"></path><path d="M46.9619 56.289C56.7903 56.289 59.9619 47.5261 59.9619 43.0262C59.9619 40.6875 58.3898 41.4236 55.8718 42.6702C53.5449 43.8222 50.4102 45.4101 46.9619 45.4101C39.7822 45.4101 33.9619 38.5263 33.9619 43.0262C33.9619 47.5261 37.1334 56.289 46.9619 56.289Z" fill="#3A3B45"></path><mask id="mask0" style="mask-type:alpha" maskUnits="userSpaceOnUse" x="33" y="41" width="27" height="16"><path d="M46.9619 56.289C56.7903 56.289 59.9619 47.5261 59.9619 43.0262C59.9619 40.6875 58.3898 41.4236 55.8718 42.6702C53.5449 43.8222 50.4102 45.4101 46.9619 45.4101C39.7822 45.4101 33.9619 38.5263 33.9619 43.0262C33.9619 47.5261 37.1334 56.289 46.9619 56.289Z" fill="white"></path></mask><g mask="url(#mask0)"><path d="M47.2119 66.5C52.0018 66.5 55.8848 62.617 55.8848 57.8271C55.8848 54.0962 53.5291 50.9156 50.224 49.6915C50.1023 49.6464 49.9794 49.604 49.8553 49.5643C49.0219 49.2979 48.1337 52.1623 47.2119 52.1623C46.3506 52.1623 45.5186 49.2797 44.7332 49.5135C41.151 50.5799 38.5389 53.8984 38.5389 57.8271C38.5389 62.617 42.4219 66.5 47.2119 66.5Z" fill="#F94040"></path></g><path d="M70.7119 37C72.5068 37 73.9619 35.5449 73.9619 33.75C73.9619 31.9551 72.5068 30.5 70.7119 30.5C68.9169 30.5 67.4619 31.9551 67.4619 33.75C67.4619 35.5449 68.9169 37 70.7119 37Z" fill="#FF9D0B"></path><path d="M24.2119 37C26.0068 37 27.4619 35.5449 27.4619 33.75C27.4619 31.9551 26.0068 30.5 24.2119 30.5C22.4169 30.5 20.9619 31.9551 20.9619 33.75C20.9619 35.5449 22.4169 37 24.2119 37Z" fill="#FF9D0B"></path><path class="origin-bottom-right transition-transform group-hover:-rotate-6" d="M17.5238 48C15.9048 48 14.4578 48.665 13.4488 49.871C12.8248 50.618 12.1728 51.822 12.1198 53.625C11.4408 53.43 10.7878 53.321 10.1778 53.321C8.6278 53.321 7.2278 53.915 6.2378 54.994C4.9658 56.379 4.4008 58.081 4.6468 59.784C4.7638 60.595 5.0348 61.322 5.4398 61.995C4.5858 62.686 3.9568 63.648 3.6528 64.805C3.4148 65.712 3.1708 67.601 4.4448 69.547C4.3638 69.674 4.2878 69.806 4.2168 69.941C3.4508 71.395 3.4018 73.038 4.0778 74.568C5.1028 76.887 7.6498 78.714 12.5958 80.675C15.6728 81.895 18.4878 82.675 18.5128 82.682C22.5808 83.737 26.2598 84.273 29.4448 84.273C35.2988 84.273 39.4898 82.48 41.9018 78.944C45.7838 73.25 45.2288 68.042 40.2058 63.022C37.4258 60.244 35.5778 56.148 35.1928 55.249C34.4168 52.587 32.3648 49.628 28.9538 49.628H28.9528C28.6658 49.628 28.3758 49.651 28.0898 49.696C26.5958 49.931 25.2898 50.791 24.3568 52.085C23.3498 50.833 22.3718 49.837 21.4868 49.275C20.1528 48.429 18.8198 48 17.5238 48ZM17.5238 52C18.0338 52 18.6568 52.217 19.3438 52.653C21.4768 54.006 25.5928 61.081 27.0998 63.833C27.6048 64.755 28.4678 65.145 29.2448 65.145C30.7868 65.145 31.9908 63.612 29.3858 61.664C25.4688 58.733 26.8428 53.942 28.7128 53.647C28.7948 53.634 28.8758 53.628 28.9538 53.628C30.6538 53.628 31.4038 56.558 31.4038 56.558C31.4038 56.558 33.6018 62.078 37.3778 65.851C41.1538 69.625 41.3488 72.654 38.5968 76.69C36.7198 79.442 33.1268 80.273 29.4448 80.273C25.6258 80.273 21.7108 79.379 19.5168 78.81C19.4088 78.782 6.0658 75.013 7.7558 71.805C8.0398 71.266 8.5078 71.05 9.0968 71.05C11.4768 71.05 15.8058 74.592 17.6668 74.592C18.0828 74.592 18.3758 74.415 18.4958 73.983C19.2888 71.138 6.4388 69.942 7.5218 65.821C7.7128 65.092 8.2308 64.796 8.9588 64.797C12.1038 64.797 19.1598 70.328 20.6388 70.328C20.7518 70.328 20.8328 70.295 20.8768 70.225C21.6178 69.029 21.2118 68.194 15.9888 65.033C10.7658 61.871 7.0998 59.969 9.1848 57.699C9.4248 57.437 9.7648 57.321 10.1778 57.321C13.3488 57.322 20.8408 64.14 20.8408 64.14C20.8408 64.14 22.8628 66.243 24.0858 66.243C24.3668 66.243 24.6058 66.132 24.7678 65.858C25.6348 64.396 16.7148 57.636 16.2118 54.847C15.8708 52.957 16.4508 52 17.5238 52Z" fill="#FF9D0B"></path><path class="origin-bottom-right transition-transform group-hover:-rotate-6" d="M38.5967 76.6898C41.3487 72.6538 41.1537 69.6248 37.3777 65.8508C33.6017 62.0778 31.4037 56.5578 31.4037 56.5578C31.4037 56.5578 30.5827 53.3518 28.7127 53.6468C26.8427 53.9418 25.4697 58.7328 29.3867 61.6638C33.3037 64.5938 28.6067 66.5848 27.0997 63.8328C25.5927 61.0808 21.4777 54.0058 19.3437 52.6528C17.2107 51.2998 15.7087 52.0578 16.2117 54.8468C16.7147 57.6358 25.6357 64.3958 24.7677 65.8588C23.8997 67.3208 20.8407 64.1398 20.8407 64.1398C20.8407 64.1398 11.2687 55.4288 9.18465 57.6988C7.10065 59.9688 10.7657 61.8708 15.9887 65.0328C21.2127 68.1938 21.6177 69.0288 20.8767 70.2248C20.1347 71.4208 8.60465 61.6998 7.52165 65.8208C6.43965 69.9418 19.2887 71.1378 18.4957 73.9828C17.7027 76.8288 9.44465 68.5978 7.75565 71.8048C6.06565 75.0128 19.4087 78.7818 19.5167 78.8098C23.8267 79.9278 34.7727 82.2968 38.5967 76.6898Z" fill="#FFD21E"></path><path class="origin-bottom-left transition-transform group-hover:rotate-6" d="M77.3999 48C79.0189 48 80.4659 48.665 81.4749 49.871C82.0989 50.618 82.7509 51.822 82.8039 53.625C83.4829 53.43 84.1359 53.321 84.7459 53.321C86.2959 53.321 87.6959 53.915 88.6859 54.994C89.9579 56.379 90.5229 58.081 90.2769 59.784C90.1599 60.595 89.8889 61.322 89.4839 61.995C90.3379 62.686 90.9669 63.648 91.2709 64.805C91.5089 65.712 91.7529 67.601 90.4789 69.547C90.5599 69.674 90.6359 69.806 90.7069 69.941C91.4729 71.395 91.5219 73.038 90.8459 74.568C89.8209 76.887 87.2739 78.714 82.3279 80.675C79.2509 81.895 76.4359 82.675 76.4109 82.682C72.3429 83.737 68.6639 84.273 65.4789 84.273C59.6249 84.273 55.4339 82.48 53.0219 78.944C49.1399 73.25 49.6949 68.042 54.7179 63.022C57.4979 60.244 59.3459 56.148 59.7309 55.249C60.5069 52.587 62.5589 49.628 65.9699 49.628H65.9709C66.2579 49.628 66.5479 49.651 66.8339 49.696C68.3279 49.931 69.6339 50.791 70.5669 52.085C71.5739 50.833 72.5519 49.837 73.4369 49.275C74.7709 48.429 76.1039 48 77.3999 48ZM77.3999 52C76.8899 52 76.2669 52.217 75.5799 52.653C73.4469 54.006 69.3309 61.081 67.8239 63.833C67.3189 64.755 66.4559 65.145 65.6789 65.145C64.1369 65.145 62.9329 63.612 65.5379 61.664C69.4549 58.733 68.0809 53.942 66.2109 53.647C66.1289 53.634 66.0479 53.628 65.9699 53.628C64.2699 53.628 63.5199 56.558 63.5199 56.558C63.5199 56.558 61.3219 62.078 57.5459 65.851C53.7699 69.625 53.5749 72.654 56.3269 76.69C58.2039 79.442 61.7969 80.273 65.4789 80.273C69.2979 80.273 73.2129 79.379 75.4069 78.81C75.5149 78.782 88.8579 75.013 87.1679 71.805C86.8839 71.266 86.4159 71.05 85.8269 71.05C83.4469 71.05 79.1179 74.592 77.2569 74.592C76.8409 74.592 76.5479 74.415 76.4279 73.983C75.6349 71.138 88.4849 69.942 87.4019 65.821C87.2109 65.092 86.6929 64.796 85.9649 64.797C82.8199 64.797 75.7639 70.328 74.2849 70.328C74.1719 70.328 74.0909 70.295 74.0469 70.225C73.3059 69.029 73.7119 68.194 78.9349 65.033C84.1579 61.871 87.8239 59.969 85.7389 57.699C85.4989 57.437 85.1589 57.321 84.7459 57.321C81.5749 57.322 74.0829 64.14 74.0829 64.14C74.0829 64.14 72.0609 66.243 70.8379 66.243C70.5569 66.243 70.3179 66.132 70.1559 65.858C69.2889 64.396 78.2089 57.636 78.7119 54.847C79.0529 52.957 78.4729 52 77.3999 52Z" fill="#FF9D0B"></path><path class="origin-bottom-left transition-transform group-hover:rotate-6" d="M56.3271 76.6898C53.5751 72.6538 53.7701 69.6248 57.5461 65.8508C61.3221 62.0778 63.5201 56.5578 63.5201 56.5578C63.5201 56.5578 64.3411 53.3518 66.2111 53.6468C68.0811 53.9418 69.4541 58.7328 65.5371 61.6638C61.6201 64.5938 66.3171 66.5848 67.8241 63.8328C69.3311 61.0808 73.4461 54.0058 75.5801 52.6528C77.7131 51.2998 79.2151 52.0578 78.7121 54.8468C78.2091 57.6358 69.2881 64.3958 70.1561 65.8588C71.0241 67.3208 74.0831 64.1398 74.0831 64.1398C74.0831 64.1398 83.6551 55.4288 85.7391 57.6988C87.8231 59.9688 84.1581 61.8708 78.9351 65.0328C73.7111 68.1938 73.3061 69.0288 74.0471 70.2248C74.7891 71.4208 86.3191 61.6998 87.4021 65.8208C88.4841 69.9418 75.6351 71.1378 76.4281 73.9828C77.2211 76.8288 85.4791 68.5978 87.1681 71.8048C88.8581 75.0128 75.5151 78.7818 75.4071 78.8098C71.0971 79.9278 60.1511 82.2968 56.3271 76.6898Z" fill="#FFD21E"></path></svg></a>
		<div class="max-md:mt-8! font-semibold text-black md:hidden">Website</div>

		<a class="hover:underline" href="/models">Models</a>
		<a class="hover:underline" href="/datasets">Datasets</a>
		<a class="hover:underline" href="/spaces">Spaces</a>
		<a class="hover:underline" href="/pricing">Pricing</a>
		<a class="hover:underline" href="/docs">Docs</a></nav></footer></div>

		<script>
			import("\/front\/build\/kube-68d7aa0\/index.js");
			window.moonSha = "kube-68d7aa0\/";
			window.__hf_deferred = {};
		</script>

		<!-- Stripe -->
		<script>
			if (["hf.co", "huggingface.co"].includes(window.location.hostname)) {
				const script = document.createElement("script");
				script.src = "https://js.stripe.com/v3/";
				script.async = true;
				document.head.appendChild(script);
			}
		</script>
	</body>
</html>
