<!doctype html>
<html class="">
	<head>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no" />
		<meta name="description" content="Weâ€™re on a journey to advance and democratize artificial intelligence through open source and open science." />
		<meta property="fb:app_id" content="1321688464574422" />
		<meta name="twitter:card" content="summary_large_image" />
		<meta name="twitter:site" content="@huggingface" />
		<meta name="twitter:image" content="https://cdn-thumbnails.huggingface.co/social-thumbnails/datasets/semeru/Text-Code-CodeSearchNet-Python.png" />
		<meta property="og:title" content="semeru/Text-Code-CodeSearchNet-Python Â· Datasets at Hugging Face" />
		<meta property="og:type" content="website" />
		<meta property="og:url" content="https://huggingface.co/datasets/semeru/Text-Code-CodeSearchNet-Python" />
		<meta property="og:image" content="https://cdn-thumbnails.huggingface.co/social-thumbnails/datasets/semeru/Text-Code-CodeSearchNet-Python.png" />

		<link rel="stylesheet" href="/front/build/kube-68d7aa0/style.css" />

		<link rel="preconnect" href="https://fonts.gstatic.com" />
		<link
			href="https://fonts.googleapis.com/css2?family=Source+Sans+Pro:ital,wght@0,200;0,300;0,400;0,600;0,700;0,900;1,200;1,300;1,400;1,600;1,700;1,900&display=swap"
			rel="stylesheet"
		/>
		<link
			href="https://fonts.googleapis.com/css2?family=IBM+Plex+Mono:wght@400;600;700&display=swap"
			rel="stylesheet"
		/>

		<link
			rel="preload"
			href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.12.0/katex.min.css"
			as="style"
			onload="this.onload=null;this.rel='stylesheet'"
		/>
		<noscript>
			<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/KaTeX/0.12.0/katex.min.css" />
		</noscript>

		<script>const guestTheme = document.cookie.match(/theme=(\w+)/)?.[1]; document.documentElement.classList.toggle('dark', guestTheme === 'dark' || ( (!guestTheme || guestTheme === 'system') && window.matchMedia('(prefers-color-scheme: dark)').matches));</script>
<link rel="canonical" href="https://huggingface.co/datasets/semeru/Text-Code-CodeSearchNet-Python"> <script type="application/ld+json">{
  "@context": {
    "@language": "en",
    "@vocab": "https:\/\/schema.org\/",
    "citeAs": "cr:citeAs",
    "column": "cr:column",
    "conformsTo": "dct:conformsTo",
    "cr": "http:\/\/mlcommons.org\/croissant\/",
    "data": {
      "@id": "cr:data",
      "@type": "@json"
    },
    "dataBiases": "cr:dataBiases",
    "dataCollection": "cr:dataCollection",
    "dataType": {
      "@id": "cr:dataType",
      "@type": "@vocab"
    },
    "dct": "http:\/\/purl.org\/dc\/terms\/",
    "extract": "cr:extract",
    "field": "cr:field",
    "fileProperty": "cr:fileProperty",
    "fileObject": "cr:fileObject",
    "fileSet": "cr:fileSet",
    "format": "cr:format",
    "includes": "cr:includes",
    "isLiveDataset": "cr:isLiveDataset",
    "jsonPath": "cr:jsonPath",
    "key": "cr:key",
    "md5": "cr:md5",
    "parentField": "cr:parentField",
    "path": "cr:path",
    "personalSensitiveInformation": "cr:personalSensitiveInformation",
    "recordSet": "cr:recordSet",
    "references": "cr:references",
    "regex": "cr:regex",
    "repeated": "cr:repeated",
    "replace": "cr:replace",
    "sc": "https:\/\/schema.org\/",
    "separator": "cr:separator",
    "source": "cr:source",
    "subField": "cr:subField",
    "transform": "cr:transform"
  },
  "@type": "sc:Dataset",
  "distribution": [
    {
      "@type": "cr:FileObject",
      "@id": "repo",
      "name": "repo",
      "description": "The Hugging Face git repository.",
      "contentUrl": "https:\/\/huggingface.co\/datasets\/semeru\/Text-Code-CodeSearchNet-Python\/tree\/refs%2Fconvert%2Fparquet",
      "encodingFormat": "git+https",
      "sha256": "https:\/\/github.com\/mlcommons\/croissant\/issues\/80"
    }
  ],
  "recordSet": [],
  "conformsTo": "http:\/\/mlcommons.org\/croissant\/1.0",
  "name": "Text-Code-CodeSearchNet-Python",
  "description": "\n\t\n\t\t\n\t\n\t\n\t\tDataset is imported from CodeXGLUE and pre-processed using their script.\n\t\n\n\n\t\n\t\t\n\t\n\t\n\t\tWhere to find in Semeru:\n\t\n\nThe dataset can be found at \/nfs\/semeru\/semeru_datasets\/code_xglue\/text-to-code\/codesearchnet\/python in Semeru\n\n\t\n\t\t\n\t\n\t\n\t\tCodeXGLUE -- Code Search (AdvTest)\n\t\n\n\n\t\n\t\t\n\t\n\t\n\t\tTask Definition\n\t\n\nGiven a natural language, the task is to search source code that matches the natural language. To test the generalization ability of a model,  function names and variables inâ€¦ See the full description on the dataset page: https:\/\/huggingface.co\/datasets\/semeru\/Text-Code-CodeSearchNet-Python.",
  "alternateName": [
    "semeru\/Text-Code-CodeSearchNet-Python"
  ],
  "creator": {
    "@type": "Organization",
    "name": "Semeru Lab",
    "url": "https:\/\/huggingface.co\/semeru"
  },
  "keywords": [
    "mit",
    "arxiv:1909.09436",
    "ðŸ‡ºðŸ‡¸ Region: US"
  ],
  "license": "https:\/\/choosealicense.com\/licenses\/mit\/",
  "url": "https:\/\/huggingface.co\/datasets\/semeru\/Text-Code-CodeSearchNet-Python"
}</script> 

		<title>semeru/Text-Code-CodeSearchNet-Python Â· Datasets at Hugging Face</title>

		<script
			defer
			data-domain="huggingface.co"
			event-loggedIn="false"
			src="/js/script.pageview-props.js"
		></script>
		<script>
			window.plausible =
				window.plausible ||
				function () {
					(window.plausible.q = window.plausible.q || []).push(arguments);
				};
		</script>
		<script>
			window.hubConfig = {"features":{"signupDisabled":false},"sshGitUrl":"git@hf.co","moonHttpUrl":"https:\/\/huggingface.co","captchaApiKey":"bd5f2066-93dc-4bdd-a64b-a24646ca3859","captchaDisabledOnSignup":true,"datasetViewerPublicUrl":"https:\/\/datasets-server.huggingface.co","stripePublicKey":"pk_live_x2tdjFXBCvXo2FFmMybezpeM00J6gPCAAc","environment":"production","userAgent":"HuggingFace (production)","spacesIframeDomain":"hf.space","spacesApiUrl":"https:\/\/api.hf.space","docSearchKey":"ece5e02e57300e17d152c08056145326e90c4bff3dd07d7d1ae40cf1c8d39cb6","logoDev":{"apiUrl":"https:\/\/img.logo.dev\/","apiKey":"pk_UHS2HZOeRnaSOdDp7jbd5w"}};
		</script>
		<script type="text/javascript" src="https://de5282c3ca0c.edge.sdk.awswaf.com/de5282c3ca0c/526cf06acb0d/challenge.js" defer></script>
	</head>
	<body class="flex flex-col min-h-dvh bg-white dark:bg-gray-950 text-black DatasetPage">
		<div class="flex min-h-dvh flex-col"><div class="SVELTE_HYDRATER contents" data-target="SystemThemeMonitor" data-props="{&quot;isLoggedIn&quot;:false}"></div>

	<div class="SVELTE_HYDRATER contents" data-target="MainHeader" data-props="{&quot;classNames&quot;:&quot;&quot;,&quot;isWide&quot;:false,&quot;isZh&quot;:false,&quot;isPro&quot;:false}"><header class="border-b border-gray-100 "><div class="w-full px-4 container flex h-16 items-center"><div class="flex flex-1 items-center"><a class="mr-5 flex flex-none items-center lg:mr-6" href="/"><img alt="Hugging Face's logo" class="w-7 md:mr-2" src="/front/assets/huggingface_logo-noborder.svg">
				<span class="hidden whitespace-nowrap text-lg font-bold md:block">Hugging Face</span></a>
			<div class="relative flex-1 lg:max-w-sm mr-2 sm:mr-4 md:mr-3 xl:mr-6"><input autocomplete="off" class="w-full dark:bg-gray-950 pl-8 form-input-alt h-9 pr-3 focus:shadow-xl " name="" placeholder="Search models, datasets, users..."   spellcheck="false" type="text" value="">
	<svg class="absolute left-2.5 text-gray-400 top-1/2 transform -translate-y-1/2" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M30 28.59L22.45 21A11 11 0 1 0 21 22.45L28.59 30zM5 14a9 9 0 1 1 9 9a9 9 0 0 1-9-9z" fill="currentColor"></path></svg>
	</div>
			<div class="flex flex-none items-center justify-center p-0.5 place-self-stretch lg:hidden"><button class="relative z-40 flex h-6 w-8 items-center justify-center" type="button"><svg width="1em" height="1em" viewBox="0 0 10 10" class="text-xl" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" preserveAspectRatio="xMidYMid meet" fill="currentColor"><path fill-rule="evenodd" clip-rule="evenodd" d="M1.65039 2.9999C1.65039 2.8066 1.80709 2.6499 2.00039 2.6499H8.00039C8.19369 2.6499 8.35039 2.8066 8.35039 2.9999C8.35039 3.1932 8.19369 3.3499 8.00039 3.3499H2.00039C1.80709 3.3499 1.65039 3.1932 1.65039 2.9999ZM1.65039 4.9999C1.65039 4.8066 1.80709 4.6499 2.00039 4.6499H8.00039C8.19369 4.6499 8.35039 4.8066 8.35039 4.9999C8.35039 5.1932 8.19369 5.3499 8.00039 5.3499H2.00039C1.80709 5.3499 1.65039 5.1932 1.65039 4.9999ZM2.00039 6.6499C1.80709 6.6499 1.65039 6.8066 1.65039 6.9999C1.65039 7.1932 1.80709 7.3499 2.00039 7.3499H8.00039C8.19369 7.3499 8.35039 7.1932 8.35039 6.9999C8.35039 6.8066 8.19369 6.6499 8.00039 6.6499H2.00039Z"></path></svg>
		</button>

	</div></div>
		<nav aria-label="Main" class="ml-auto hidden lg:block"><ul class="flex items-center space-x-1.5 2xl:space-x-2"><li class="hover:text-indigo-700"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/models"><svg class="mr-1.5 text-gray-400 group-hover:text-indigo-500" style="" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 24 24"><path class="uim-quaternary" d="M20.23 7.24L12 12L3.77 7.24a1.98 1.98 0 0 1 .7-.71L11 2.76c.62-.35 1.38-.35 2 0l6.53 3.77c.29.173.531.418.7.71z" opacity=".25" fill="currentColor"></path><path class="uim-tertiary" d="M12 12v9.5a2.09 2.09 0 0 1-.91-.21L4.5 17.48a2.003 2.003 0 0 1-1-1.73v-7.5a2.06 2.06 0 0 1 .27-1.01L12 12z" opacity=".5" fill="currentColor"></path><path class="uim-primary" d="M20.5 8.25v7.5a2.003 2.003 0 0 1-1 1.73l-6.62 3.82c-.275.13-.576.198-.88.2V12l8.23-4.76c.175.308.268.656.27 1.01z" fill="currentColor"></path></svg>
					Models</a>
			</li><li class="hover:text-red-700"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/datasets"><svg class="mr-1.5 text-gray-400 group-hover:text-red-500" style="" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 25 25"><ellipse cx="12.5" cy="5" fill="currentColor" fill-opacity="0.25" rx="7.5" ry="2"></ellipse><path d="M12.5 15C16.6421 15 20 14.1046 20 13V20C20 21.1046 16.6421 22 12.5 22C8.35786 22 5 21.1046 5 20V13C5 14.1046 8.35786 15 12.5 15Z" fill="currentColor" opacity="0.5"></path><path d="M12.5 7C16.6421 7 20 6.10457 20 5V11.5C20 12.6046 16.6421 13.5 12.5 13.5C8.35786 13.5 5 12.6046 5 11.5V5C5 6.10457 8.35786 7 12.5 7Z" fill="currentColor" opacity="0.5"></path><path d="M5.23628 12C5.08204 12.1598 5 12.8273 5 13C5 14.1046 8.35786 15 12.5 15C16.6421 15 20 14.1046 20 13C20 12.8273 19.918 12.1598 19.7637 12C18.9311 12.8626 15.9947 13.5 12.5 13.5C9.0053 13.5 6.06886 12.8626 5.23628 12Z" fill="currentColor"></path></svg>
					Datasets</a>
			</li><li class="hover:text-blue-700"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/spaces"><svg class="mr-1.5 text-gray-400 group-hover:text-blue-500" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" viewBox="0 0 25 25"><path opacity=".5" d="M6.016 14.674v4.31h4.31v-4.31h-4.31ZM14.674 14.674v4.31h4.31v-4.31h-4.31ZM6.016 6.016v4.31h4.31v-4.31h-4.31Z" fill="currentColor"></path><path opacity=".75" fill-rule="evenodd" clip-rule="evenodd" d="M3 4.914C3 3.857 3.857 3 4.914 3h6.514c.884 0 1.628.6 1.848 1.414a5.171 5.171 0 0 1 7.31 7.31c.815.22 1.414.964 1.414 1.848v6.514A1.914 1.914 0 0 1 20.086 22H4.914A1.914 1.914 0 0 1 3 20.086V4.914Zm3.016 1.102v4.31h4.31v-4.31h-4.31Zm0 12.968v-4.31h4.31v4.31h-4.31Zm8.658 0v-4.31h4.31v4.31h-4.31Zm0-10.813a2.155 2.155 0 1 1 4.31 0 2.155 2.155 0 0 1-4.31 0Z" fill="currentColor"></path><path opacity=".25" d="M16.829 6.016a2.155 2.155 0 1 0 0 4.31 2.155 2.155 0 0 0 0-4.31Z" fill="currentColor"></path></svg>
					Spaces</a>
			</li><li class="hover:text-yellow-700 max-xl:hidden"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/posts"><svg class="mr-1.5 text-gray-400 group-hover:text-yellow-500 text-yellow-500!" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" viewBox="0 0 12 12" preserveAspectRatio="xMidYMid meet"><path fill="currentColor" fill-rule="evenodd" d="M3.73 2.4A4.25 4.25 0 1 1 6 10.26H2.17l-.13-.02a.43.43 0 0 1-.3-.43l.01-.06a.43.43 0 0 1 .12-.22l.84-.84A4.26 4.26 0 0 1 3.73 2.4Z" clip-rule="evenodd"></path></svg>
					Posts</a>
			</li><li class="hover:text-yellow-700"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/docs"><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" class="mr-1.5 text-gray-400 group-hover:text-yellow-500" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 16 16"><path d="m2.28 3.7-.3.16a.67.67 0 0 0-.34.58v8.73l.01.04.02.07.01.04.03.06.02.04.02.03.04.06.05.05.04.04.06.04.06.04.08.04.08.02h.05l.07.02h.11l.04-.01.07-.02.03-.01.07-.03.22-.12a5.33 5.33 0 0 1 5.15.1.67.67 0 0 0 .66 0 5.33 5.33 0 0 1 5.33 0 .67.67 0 0 0 1-.58V4.36a.67.67 0 0 0-.34-.5l-.3-.17v7.78a.63.63 0 0 1-.87.59 4.9 4.9 0 0 0-4.35.35l-.65.39a.29.29 0 0 1-.15.04.29.29 0 0 1-.16-.04l-.65-.4a4.9 4.9 0 0 0-4.34-.34.63.63 0 0 1-.87-.59V3.7Z" fill="currentColor" class="dark:opacity-40"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M8 3.1a5.99 5.99 0 0 0-5.3-.43.66.66 0 0 0-.42.62v8.18c0 .45.46.76.87.59a4.9 4.9 0 0 1 4.34.35l.65.39c.05.03.1.04.16.04.05 0 .1-.01.15-.04l.65-.4a4.9 4.9 0 0 1 4.35-.34.63.63 0 0 0 .86-.59V3.3a.67.67 0 0 0-.41-.62 5.99 5.99 0 0 0-5.3.43l-.3.17L8 3.1Zm.73 1.87a.43.43 0 1 0-.86 0v5.48a.43.43 0 0 0 .86 0V4.97Z" fill="currentColor" class="opacity-40 dark:opacity-100"></path><path d="M8.73 4.97a.43.43 0 1 0-.86 0v5.48a.43.43 0 1 0 .86 0V4.96Z" fill="currentColor" class="dark:opacity-40"></path></svg>
					Docs</a>
			</li><li class="hover:text-black dark:hover:text-white"><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/enterprise"><svg class="mr-1.5 text-gray-400 group-hover:text-black dark:group-hover:text-white" xmlns="http://www.w3.org/2000/svg" fill="none" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 33 27"><path fill="currentColor" fill-rule="evenodd" d="M13.5.7a8.7 8.7 0 0 0-7.7 5.7L1 20.6c-1 3.1.9 5.7 4.1 5.7h15c3.3 0 6.8-2.6 7.8-5.7l4.6-14.2c1-3.1-.8-5.7-4-5.7h-15Zm1.1 5.7L9.8 20.3h9.8l1-3.1h-5.8l.8-2.5h4.8l1.1-3h-4.8l.8-2.3H23l1-3h-9.5Z" clip-rule="evenodd"></path></svg>
					Enterprise</a>
			</li>

		<li><a class="group flex items-center px-2 py-0.5 dark:text-gray-300 dark:hover:text-gray-100" href="/pricing">Pricing
			</a></li>

		<li><div class="relative group">
	<button class="px-2 py-0.5 hover:text-gray-500 dark:hover:text-gray-600 flex items-center " type="button">
		<svg class=" text-gray-500 w-5 group-hover:text-gray-400 dark:text-gray-300 dark:group-hover:text-gray-100" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" viewBox="0 0 32 18" preserveAspectRatio="xMidYMid meet"><path fill-rule="evenodd" clip-rule="evenodd" d="M14.4504 3.30221C14.4504 2.836 14.8284 2.45807 15.2946 2.45807H28.4933C28.9595 2.45807 29.3374 2.836 29.3374 3.30221C29.3374 3.76842 28.9595 4.14635 28.4933 4.14635H15.2946C14.8284 4.14635 14.4504 3.76842 14.4504 3.30221Z" fill="currentColor"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M14.4504 9.00002C14.4504 8.53382 14.8284 8.15588 15.2946 8.15588H28.4933C28.9595 8.15588 29.3374 8.53382 29.3374 9.00002C29.3374 9.46623 28.9595 9.84417 28.4933 9.84417H15.2946C14.8284 9.84417 14.4504 9.46623 14.4504 9.00002Z" fill="currentColor"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M14.4504 14.6978C14.4504 14.2316 14.8284 13.8537 15.2946 13.8537H28.4933C28.9595 13.8537 29.3374 14.2316 29.3374 14.6978C29.3374 15.164 28.9595 15.542 28.4933 15.542H15.2946C14.8284 15.542 14.4504 15.164 14.4504 14.6978Z" fill="currentColor"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M1.94549 6.87377C2.27514 6.54411 2.80962 6.54411 3.13928 6.87377L6.23458 9.96907L9.32988 6.87377C9.65954 6.54411 10.194 6.54411 10.5237 6.87377C10.8533 7.20343 10.8533 7.73791 10.5237 8.06756L6.23458 12.3567L1.94549 8.06756C1.61583 7.73791 1.61583 7.20343 1.94549 6.87377Z" fill="currentColor"></path></svg>
			
		</button>
	
	
	</div></li>
		<li><hr class="h-5 w-0.5 border-none bg-gray-100 dark:bg-gray-800"></li>
		<li><a class="block cursor-pointer whitespace-nowrap px-2 py-0.5 hover:text-gray-500 dark:text-gray-300 dark:hover:text-gray-100" href="/login">Log In
				</a></li>
			<li><a class="whitespace-nowrap rounded-full border border-transparent bg-gray-900 px-3 py-1 leading-none text-white hover:border-black hover:bg-white hover:text-black" href="/join">Sign Up
					</a></li></ul></nav></div></header></div>
	
	
	
	<div class="SVELTE_HYDRATER contents" data-target="SSOBanner" data-props="{}"></div>
	
	

	<main class="flex flex-1 flex-col">
	<div class="SVELTE_HYDRATER contents" data-target="DatasetHeader" data-props="{&quot;activeTab&quot;:&quot;datasetCard&quot;,&quot;author&quot;:{&quot;avatarUrl&quot;:&quot;https://cdn-avatars.huggingface.co/v1/production/uploads/1636726823137-5ebb6509a8e72729bee106a0.png&quot;,&quot;fullname&quot;:&quot;Semeru Lab&quot;,&quot;name&quot;:&quot;semeru&quot;,&quot;type&quot;:&quot;org&quot;,&quot;isHf&quot;:false,&quot;isHfAdmin&quot;:false,&quot;isMod&quot;:false,&quot;isEnterprise&quot;:false,&quot;followerCount&quot;:7},&quot;canReadRepoSettings&quot;:false,&quot;dataset&quot;:{&quot;author&quot;:&quot;semeru&quot;,&quot;cardData&quot;:{&quot;license&quot;:&quot;mit&quot;,&quot;Programminglanguage&quot;:&quot;Python&quot;,&quot;version&quot;:&quot;N/A&quot;,&quot;Date&quot;:&quot;Codesearchnet(Jun 2020 - paper release date)&quot;,&quot;Contaminated&quot;:&quot;Very Likely&quot;,&quot;Size&quot;:&quot;Standard Tokenizer (TreeSitter)&quot;},&quot;cardExists&quot;:true,&quot;createdAt&quot;:&quot;2023-03-03T17:03:14.000Z&quot;,&quot;description&quot;:&quot;\n\t\n\t\t\n\t\n\t\n\t\tDataset is imported from CodeXGLUE and pre-processed using their script.\n\t\n\n\n\t\n\t\t\n\t\n\t\n\t\tWhere to find in Semeru:\n\t\n\nThe dataset can be found at /nfs/semeru/semeru_datasets/code_xglue/text-to-code/codesearchnet/python in Semeru\n\n\t\n\t\t\n\t\n\t\n\t\tCodeXGLUE -- Code Search (AdvTest)\n\t\n\n\n\t\n\t\t\n\t\n\t\n\t\tTask Definition\n\t\n\nGiven a natural language, the task is to search source code that matches the natural language. To test the generalization ability of a model,  function names and variables inâ€¦ See the full description on the dataset page: https://huggingface.co/datasets/semeru/Text-Code-CodeSearchNet-Python.&quot;,&quot;downloads&quot;:56,&quot;downloadsAllTime&quot;:317,&quot;id&quot;:&quot;semeru/Text-Code-CodeSearchNet-Python&quot;,&quot;isLikedByUser&quot;:false,&quot;lastModified&quot;:&quot;2023-03-27T18:36:44.000Z&quot;,&quot;likes&quot;:2,&quot;datasetsServerInfo&quot;:{&quot;viewer&quot;:&quot;preview&quot;,&quot;numRows&quot;:0,&quot;libraries&quot;:[],&quot;formats&quot;:[],&quot;modalities&quot;:[]},&quot;discussionsDisabled&quot;:false,&quot;repoType&quot;:&quot;dataset&quot;,&quot;private&quot;:false,&quot;gated&quot;:false,&quot;tags&quot;:[&quot;license:mit&quot;,&quot;arxiv:1909.09436&quot;,&quot;region:us&quot;],&quot;tag_objs&quot;:[{&quot;id&quot;:&quot;license:mit&quot;,&quot;label&quot;:&quot;mit&quot;,&quot;type&quot;:&quot;license&quot;},{&quot;id&quot;:&quot;arxiv:1909.09436&quot;,&quot;label&quot;:&quot;arxiv:1909.09436&quot;,&quot;type&quot;:&quot;arxiv&quot;,&quot;extra&quot;:{&quot;paperTitle&quot;:&quot;CodeSearchNet Challenge: Evaluating the State of Semantic Code Search&quot;}},{&quot;type&quot;:&quot;region&quot;,&quot;label&quot;:&quot;ðŸ‡ºðŸ‡¸ Region: US&quot;,&quot;id&quot;:&quot;region:us&quot;}],&quot;hasBlockedOids&quot;:false,&quot;region&quot;:&quot;us&quot;,&quot;xetEnabled&quot;:false},&quot;discussionsStats&quot;:{&quot;closed&quot;:0,&quot;open&quot;:0,&quot;total&quot;:0}}"><header class="bg-linear-to-t border-b border-gray-100 pt-6 sm:pt-9 from-gray-50-to-white via-white dark:via-gray-950"><div class="container relative "><h1 class="flex flex-wrap items-center max-md:leading-tight mb-3 text-lg max-sm:gap-y-1.5 md:text-xl"><a href="/datasets" class="group flex items-center"><svg class="sm:mr-1.5 -mr-1 text-gray-400" style="" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 25 25"><ellipse cx="12.5" cy="5" fill="currentColor" fill-opacity="0.25" rx="7.5" ry="2"></ellipse><path d="M12.5 15C16.6421 15 20 14.1046 20 13V20C20 21.1046 16.6421 22 12.5 22C8.35786 22 5 21.1046 5 20V13C5 14.1046 8.35786 15 12.5 15Z" fill="currentColor" opacity="0.5"></path><path d="M12.5 7C16.6421 7 20 6.10457 20 5V11.5C20 12.6046 16.6421 13.5 12.5 13.5C8.35786 13.5 5 12.6046 5 11.5V5C5 6.10457 8.35786 7 12.5 7Z" fill="currentColor" opacity="0.5"></path><path d="M5.23628 12C5.08204 12.1598 5 12.8273 5 13C5 14.1046 8.35786 15 12.5 15C16.6421 15 20 14.1046 20 13C20 12.8273 19.918 12.1598 19.7637 12C18.9311 12.8626 15.9947 13.5 12.5 13.5C9.0053 13.5 6.06886 12.8626 5.23628 12Z" fill="currentColor"></path></svg>
					<span class="mr-2.5 font-semibold text-gray-400 group-hover:text-gray-500 max-sm:hidden">Datasets:</span></a>
				<hr class="mx-1.5 h-2 translate-y-px rounded-sm border-r dark:border-gray-600 sm:hidden">
			<div class="group flex flex-none items-center"><div class="relative mr-1 flex items-center">

			

<span class="inline-block "><span class="contents"><a href="/semeru" class="text-gray-400 hover:text-blue-600"><img alt="" class="w-3.5 h-3.5 rounded-sm  flex-none" src="https://cdn-avatars.huggingface.co/v1/production/uploads/1636726823137-5ebb6509a8e72729bee106a0.png" crossorigin="anonymous"></a></span>
	</span></div>
		

<span class="inline-block "><span class="contents"><a href="/semeru" class="text-gray-400 hover:text-blue-600">semeru</a></span>
	</span>
		<div class="mx-0.5 text-gray-300">/</div></div>

<div class="max-w-full "><a class="break-words font-mono font-semibold hover:text-blue-600 " href="/datasets/semeru/Text-Code-CodeSearchNet-Python">Text-Code-CodeSearchNet-Python</a>
	<button class="relative text-sm mr-4 focus:outline-hidden inline-flex cursor-pointer items-center text-sm  mx-0.5   text-gray-600 " title="Copy dataset name to clipboard" type="button"><svg class="" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" fill="currentColor" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M28,10V28H10V10H28m0-2H10a2,2,0,0,0-2,2V28a2,2,0,0,0,2,2H28a2,2,0,0,0,2-2V10a2,2,0,0,0-2-2Z" transform="translate(0)"></path><path d="M4,18H2V4A2,2,0,0,1,4,2H18V4H4Z" transform="translate(0)"></path><rect fill="none" width="32" height="32"></rect></svg>
	
	</button></div>
			<div class="inline-flex items-center overflow-hidden whitespace-nowrap rounded-md border bg-white text-sm leading-none text-gray-500  mr-2"><button class="relative flex items-center overflow-hidden from-red-50 to-transparent dark:from-red-900 px-1.5 py-1 hover:bg-linear-to-t focus:outline-hidden"  title="Like"><svg class="left-1.5 absolute" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32" fill="currentColor"><path d="M22.45,6a5.47,5.47,0,0,1,3.91,1.64,5.7,5.7,0,0,1,0,8L16,26.13,5.64,15.64a5.7,5.7,0,0,1,0-8,5.48,5.48,0,0,1,7.82,0L16,10.24l2.53-2.58A5.44,5.44,0,0,1,22.45,6m0-2a7.47,7.47,0,0,0-5.34,2.24L16,7.36,14.89,6.24a7.49,7.49,0,0,0-10.68,0,7.72,7.72,0,0,0,0,10.82L16,29,27.79,17.06a7.72,7.72,0,0,0,0-10.82A7.49,7.49,0,0,0,22.45,4Z"></path></svg>

		
		<span class="ml-4 pl-0.5 ">like</span></button>
	<button class="focus:outline-hidden flex items-center border-l px-1.5 py-1 text-gray-400 hover:bg-gray-50 focus:bg-gray-100 dark:hover:bg-gray-900 dark:focus:bg-gray-800" title="See users who liked this repository">2</button></div>




			<div class="relative flex items-center gap-1.5  "><div class="mr-2 inline-flex h-6 items-center overflow-hidden whitespace-nowrap rounded-md border text-sm text-gray-500"><button class="focus:outline-hidden relative flex h-full max-w-56 items-center gap-1.5 overflow-hidden px-1.5 hover:bg-gray-50 focus:bg-gray-100 dark:hover:bg-gray-900 dark:focus:bg-gray-800" type="button" ><div class="flex h-full flex-1 items-center justify-center ">Follow</div>
		<img alt="" class="rounded-xs size-3 flex-none" src="https://cdn-avatars.huggingface.co/v1/production/uploads/1636726823137-5ebb6509a8e72729bee106a0.png">
		<span class="truncate">Semeru Lab</span></button>
	<button class="focus:outline-hidden flex h-full items-center border-l pl-1.5 pr-1.5 text-gray-400 hover:bg-gray-50 focus:bg-gray-100 dark:hover:bg-gray-900 dark:focus:bg-gray-800" title="Show Semeru Lab's followers" type="button">7</button></div>

		</div>
			
	</h1>
		<div class="mb-3 flex flex-wrap md:mb-4"><div class="mr-1 flex flex-wrap items-center"><span class="mb-1 mr-1 p-1 text-sm leading-tight text-gray-400 md:mb-1.5">ArXiv:
	</span>
	

	<div class="relative inline-block ">
	<button class="group mr-1 mb-1 md:mr-1.5 md:mb-1.5  rounded-full rounded-br-none " type="button">
		<div class="tag tag-white rounded-full  relative rounded-br-none pr-2.5">
		<svg class="-mr-1 text-gray-400" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" viewBox="0 0 12 12" preserveAspectRatio="xMidYMid meet" fill="none"><path fill="currentColor" fill-rule="evenodd" d="M8.007 1.814a1.176 1.176 0 0 0-.732-.266H3.088c-.64 0-1.153.512-1.153 1.152v6.803c0 .64.513 1.152 1.153 1.152h5.54c.632 0 1.144-.511 1.144-1.152V3.816c0-.338-.137-.658-.412-.887L8.007 1.814Zm-1.875 1.81c0 .695.55 1.253 1.244 1.253h.983a.567.567 0 0 1 .553.585v4.041c0 .165-.119.302-.283.302h-5.55c-.156 0-.275-.137-.275-.302V2.7a.284.284 0 0 1 .284-.301h2.468a.574.574 0 0 1 .434.19.567.567 0 0 1 .142.395v.64Z" clip-rule="evenodd"></path><path fill="currentColor" fill-opacity=".2" fill-rule="evenodd" d="M6.132 3.624c0 .695.55 1.253 1.244 1.253h.97a.567.567 0 0 1 .566.585v4.041c0 .165-.119.302-.283.302h-5.55c-.156 0-.275-.137-.275-.302V2.7a.284.284 0 0 1 .284-.301h2.468a.567.567 0 0 1 .576.585v.64Z" clip-rule="evenodd"></path></svg>

	

	<span class="-mr-2 text-gray-400">arxiv:</span>
		<span>1909.09436</span>
	

	<div class="border-br-gray-200 absolute bottom-0.5 right-0.5 h-1 w-1 border-[3px] border-l-transparent border-t-transparent border-b-gray-200 border-r-gray-200 dark:border-b-gray-700 dark:border-r-gray-700"></div></div>
		
		</button>
	
	
	</div>

	</div><div class="mr-1 flex flex-wrap items-center"><span class="mb-1 mr-1 p-1 text-sm leading-tight text-gray-400 md:mb-1.5">License:
	</span>
	<div class="relative inline-block ">
	<button class="group mr-1 mb-1 md:mr-1.5 md:mb-1.5  rounded-full rounded-br-none " type="button">
		<div class="tag tag-white rounded-full  relative rounded-br-none pr-2.5">
		<svg class="text-xs text-gray-900" width="1em" height="1em" viewBox="0 0 10 10" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M1.46009 5.0945V6.88125C1.46009 7.25201 1.75937 7.55129 2.13012 7.55129C2.50087 7.55129 2.80016 7.25201 2.80016 6.88125V5.0945C2.80016 4.72375 2.50087 4.42446 2.13012 4.42446C1.75937 4.42446 1.46009 4.72375 1.46009 5.0945ZM4.14022 5.0945V6.88125C4.14022 7.25201 4.4395 7.55129 4.81026 7.55129C5.18101 7.55129 5.48029 7.25201 5.48029 6.88125V5.0945C5.48029 4.72375 5.18101 4.42446 4.81026 4.42446C4.4395 4.42446 4.14022 4.72375 4.14022 5.0945ZM1.23674 9.78473H8.38377C8.75452 9.78473 9.0538 9.48545 9.0538 9.1147C9.0538 8.74395 8.75452 8.44466 8.38377 8.44466H1.23674C0.865993 8.44466 0.566711 8.74395 0.566711 9.1147C0.566711 9.48545 0.865993 9.78473 1.23674 9.78473ZM6.82036 5.0945V6.88125C6.82036 7.25201 7.11964 7.55129 7.49039 7.55129C7.86114 7.55129 8.16042 7.25201 8.16042 6.88125V5.0945C8.16042 4.72375 7.86114 4.42446 7.49039 4.42446C7.11964 4.42446 6.82036 4.72375 6.82036 5.0945ZM4.39484 0.623142L0.865993 2.48137C0.682851 2.57517 0.566711 2.76725 0.566711 2.97273C0.566711 3.28094 0.816857 3.53109 1.12507 3.53109H8.49991C8.80365 3.53109 9.0538 3.28094 9.0538 2.97273C9.0538 2.76725 8.93766 2.57517 8.75452 2.48137L5.22568 0.623142C4.9666 0.484669 4.65391 0.484669 4.39484 0.623142V0.623142Z" fill="currentColor"></path></svg>

	

	<span>mit</span>
	

	<div class="border-br-gray-200 absolute bottom-0.5 right-0.5 h-1 w-1 border-[3px] border-l-transparent border-t-transparent border-b-gray-200 border-r-gray-200 dark:border-b-gray-700 dark:border-r-gray-700"></div></div>
		
		</button>
	
	
	</div>

	</div></div>

		<div class="flex flex-col-reverse lg:flex-row lg:items-center lg:justify-between"><div class="-mb-px flex h-12 items-center overflow-x-auto overflow-y-hidden ">
	<a class="tab-alternate active" href="/datasets/semeru/Text-Code-CodeSearchNet-Python"><svg class="mr-1.5 text-gray-400 flex-none" style="" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 24 24"><path class="uim-quaternary" d="M20.23 7.24L12 12L3.77 7.24a1.98 1.98 0 0 1 .7-.71L11 2.76c.62-.35 1.38-.35 2 0l6.53 3.77c.29.173.531.418.7.71z" opacity=".25" fill="currentColor"></path><path class="uim-tertiary" d="M12 12v9.5a2.09 2.09 0 0 1-.91-.21L4.5 17.48a2.003 2.003 0 0 1-1-1.73v-7.5a2.06 2.06 0 0 1 .27-1.01L12 12z" opacity=".5" fill="currentColor"></path><path class="uim-primary" d="M20.5 8.25v7.5a2.003 2.003 0 0 1-1 1.73l-6.62 3.82c-.275.13-.576.198-.88.2V12l8.23-4.76c.175.308.268.656.27 1.01z" fill="currentColor"></path></svg>
	Dataset card
	

	
		</a><a class="tab-alternate" href="/datasets/semeru/Text-Code-CodeSearchNet-Python/viewer/"><svg class="mr-1.5 text-gray-400 flex-none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 12 12"><path fill="currentColor" d="M2.5 2h7a1 1 0 0 1 1 1v6a1 1 0 0 1-1 1h-7a1 1 0 0 1-1-1V3a1 1 0 0 1 1-1Zm0 2v2h3V4h-3Zm4 0v2h3V4h-3Zm-4 3v2h3V7h-3Zm4 0v2h3V7h-3Z"></path></svg>
	Data Studio
	

	
		</a><a class="tab-alternate" href="/datasets/semeru/Text-Code-CodeSearchNet-Python/tree/main"><svg class="mr-1.5 text-gray-400 flex-none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 24 24"><path class="uim-tertiary" d="M21 19h-8a1 1 0 0 1 0-2h8a1 1 0 0 1 0 2zm0-4h-8a1 1 0 0 1 0-2h8a1 1 0 0 1 0 2zm0-8h-8a1 1 0 0 1 0-2h8a1 1 0 0 1 0 2zm0 4h-8a1 1 0 0 1 0-2h8a1 1 0 0 1 0 2z" opacity=".5" fill="currentColor"></path><path class="uim-primary" d="M9 19a1 1 0 0 1-1-1V6a1 1 0 0 1 2 0v12a1 1 0 0 1-1 1zm-6-4.333a1 1 0 0 1-.64-1.769L3.438 12l-1.078-.898a1 1 0 0 1 1.28-1.538l2 1.667a1 1 0 0 1 0 1.538l-2 1.667a.999.999 0 0 1-.64.231z" fill="currentColor"></path></svg>
	<span class="xl:hidden">Files</span>
		<span class="hidden xl:inline">Files and versions</span>
	

	
		</a><a class="tab-alternate" href="/datasets/semeru/Text-Code-CodeSearchNet-Python/discussions"><svg class="mr-1.5 text-gray-400 flex-none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M20.6081 3C21.7684 3 22.8053 3.49196 23.5284 4.38415C23.9756 4.93678 24.4428 5.82749 24.4808 7.16133C24.9674 7.01707 25.4353 6.93643 25.8725 6.93643C26.9833 6.93643 27.9865 7.37587 28.696 8.17411C29.6075 9.19872 30.0124 10.4579 29.8361 11.7177C29.7523 12.3177 29.5581 12.8555 29.2678 13.3534C29.8798 13.8646 30.3306 14.5763 30.5485 15.4322C30.719 16.1032 30.8939 17.5006 29.9808 18.9403C30.0389 19.0342 30.0934 19.1319 30.1442 19.2318C30.6932 20.3074 30.7283 21.5229 30.2439 22.6548C29.5093 24.3704 27.6841 25.7219 24.1397 27.1727C21.9347 28.0753 19.9174 28.6523 19.8994 28.6575C16.9842 29.4379 14.3477 29.8345 12.0653 29.8345C7.87017 29.8345 4.8668 28.508 3.13831 25.8921C0.356375 21.6797 0.754104 17.8269 4.35369 14.1131C6.34591 12.058 7.67023 9.02782 7.94613 8.36275C8.50224 6.39343 9.97271 4.20438 12.4172 4.20438H12.4179C12.6236 4.20438 12.8314 4.2214 13.0364 4.25468C14.107 4.42854 15.0428 5.06476 15.7115 6.02205C16.4331 5.09583 17.134 4.359 17.7682 3.94323C18.7242 3.31737 19.6794 3 20.6081 3ZM20.6081 5.95917C20.2427 5.95917 19.7963 6.1197 19.3039 6.44225C17.7754 7.44319 14.8258 12.6772 13.7458 14.7131C13.3839 15.3952 12.7655 15.6837 12.2086 15.6837C11.1036 15.6837 10.2408 14.5497 12.1076 13.1085C14.9146 10.9402 13.9299 7.39584 12.5898 7.1776C12.5311 7.16799 12.4731 7.16355 12.4172 7.16355C11.1989 7.16355 10.6615 9.33114 10.6615 9.33114C10.6615 9.33114 9.0863 13.4148 6.38031 16.206C3.67434 18.998 3.5346 21.2388 5.50675 24.2246C6.85185 26.2606 9.42666 26.8753 12.0653 26.8753C14.8021 26.8753 17.6077 26.2139 19.1799 25.793C19.2574 25.7723 28.8193 22.984 27.6081 20.6107C27.4046 20.212 27.0693 20.0522 26.6471 20.0522C24.9416 20.0522 21.8393 22.6726 20.5057 22.6726C20.2076 22.6726 19.9976 22.5416 19.9116 22.222C19.3433 20.1173 28.552 19.2325 27.7758 16.1839C27.639 15.6445 27.2677 15.4256 26.746 15.4263C24.4923 15.4263 19.4358 19.5181 18.3759 19.5181C18.2949 19.5181 18.2368 19.4937 18.2053 19.4419C17.6743 18.557 17.9653 17.9394 21.7082 15.6009C25.4511 13.2617 28.0783 11.8545 26.5841 10.1752C26.4121 9.98141 26.1684 9.8956 25.8725 9.8956C23.6001 9.89634 18.2311 14.9403 18.2311 14.9403C18.2311 14.9403 16.7821 16.496 15.9057 16.496C15.7043 16.496 15.533 16.4139 15.4169 16.2112C14.7956 15.1296 21.1879 10.1286 21.5484 8.06535C21.7928 6.66715 21.3771 5.95917 20.6081 5.95917Z" fill="#FF9D00"></path><path d="M5.50686 24.2246C3.53472 21.2387 3.67446 18.9979 6.38043 16.206C9.08641 13.4147 10.6615 9.33111 10.6615 9.33111C10.6615 9.33111 11.2499 6.95933 12.59 7.17757C13.93 7.39581 14.9139 10.9401 12.1069 13.1084C9.29997 15.276 12.6659 16.7489 13.7459 14.713C14.8258 12.6772 17.7747 7.44316 19.304 6.44221C20.8326 5.44128 21.9089 6.00204 21.5484 8.06532C21.188 10.1286 14.795 15.1295 15.4171 16.2118C16.0391 17.2934 18.2312 14.9402 18.2312 14.9402C18.2312 14.9402 25.0907 8.49588 26.5842 10.1752C28.0776 11.8545 25.4512 13.2616 21.7082 15.6008C17.9646 17.9393 17.6744 18.557 18.2054 19.4418C18.7372 20.3266 26.9998 13.1351 27.7759 16.1838C28.5513 19.2324 19.3434 20.1173 19.9117 22.2219C20.48 24.3274 26.3979 18.2382 27.6082 20.6107C28.8193 22.9839 19.2574 25.7722 19.18 25.7929C16.0914 26.62 8.24723 28.3726 5.50686 24.2246Z" fill="#FFD21E"></path></svg>
	Community
	

	
		</a></div>
	
			</div></div></header>


</div>
	
<div class="container relative flex flex-col md:grid md:space-y-0 w-full md:grid-cols-12 md:flex-1 md:grid-rows-full space-y-4 md:gap-6 ">
		<section class="pt-6 border-gray-100 md:col-span-8 pb-24 relative break-words copiable-code-container">
				<div class="SVELTE_HYDRATER contents" data-target="UnsafeBanner" data-props="{&quot;classNames&quot;:&quot;mb-4&quot;,&quot;repoId&quot;:&quot;semeru/Text-Code-CodeSearchNet-Python&quot;,&quot;repoType&quot;:&quot;dataset&quot;,&quot;minLevel&quot;:&quot;unsafe&quot;}"></div>
					<div class="SVELTE_HYDRATER contents" data-target="DatasetViewer" data-props="{&quot;data&quot;:{&quot;kind&quot;:&quot;DatasetAndSampleData&quot;,&quot;partial&quot;:false,&quot;configsData&quot;:{&quot;configInfos&quot;:[{&quot;name&quot;:&quot;default&quot;,&quot;status&quot;:&quot;ok&quot;}],&quot;selectedConfig&quot;:&quot;default&quot;,&quot;hasSelectedConfigParquet&quot;:false},&quot;splitsData&quot;:{&quot;splitInfos&quot;:[{&quot;name&quot;:&quot;train&quot;},{&quot;name&quot;:&quot;validation&quot;},{&quot;name&quot;:&quot;test&quot;}],&quot;selectedSplit&quot;:&quot;train&quot;},&quot;sampleData&quot;:{&quot;dataset&quot;:&quot;semeru/Text-Code-CodeSearchNet-Python&quot;,&quot;config&quot;:&quot;default&quot;,&quot;split&quot;:&quot;train&quot;,&quot;capabilities&quot;:{&quot;rows&quot;:false,&quot;search&quot;:false,&quot;filter&quot;:false,&quot;statistics&quot;:false},&quot;navigation&quot;:{&quot;p&quot;:0},&quot;jwt&quot;:&quot;eyJhbGciOiJFZERTQSJ9.eyJyZWFkIjp0cnVlLCJwZXJtaXNzaW9ucyI6eyJyZXBvLmNvbnRlbnQucmVhZCI6dHJ1ZX0sImlhdCI6MTc0MjkyMzA3NSwic3ViIjoiL2RhdGFzZXRzL3NlbWVydS9UZXh0LUNvZGUtQ29kZVNlYXJjaE5ldC1QeXRob24iLCJleHAiOjE3NDI5MjY2NzUsImlzcyI6Imh0dHBzOi8vaHVnZ2luZ2ZhY2UuY28ifQ.e1LqsISvqJrmZn0U8UPKAvn_sYv2f0nAmtwtwWtli0NhAjkvXHGnKzqp-hfhUZlScof814oiVH6_Y73tA3SNAQ&quot;,&quot;sampleData&quot;:{&quot;columns&quot;:[{&quot;name&quot;:&quot;repo&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;},{&quot;name&quot;:&quot;path&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;},{&quot;name&quot;:&quot;func_name&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;},{&quot;name&quot;:&quot;original_string&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;},{&quot;name&quot;:&quot;language&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;},{&quot;name&quot;:&quot;code&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;},{&quot;name&quot;:&quot;code_tokens&quot;,&quot;align&quot;:&quot;left&quot;,&quot;type&quot;:&quot;sequence&quot;},{&quot;name&quot;:&quot;docstring&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;},{&quot;name&quot;:&quot;docstring_tokens&quot;,&quot;align&quot;:&quot;left&quot;,&quot;type&quot;:&quot;sequence&quot;},{&quot;name&quot;:&quot;sha&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;},{&quot;name&quot;:&quot;url&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;},{&quot;name&quot;:&quot;partition&quot;,&quot;align&quot;:&quot;depends on text direction&quot;,&quot;type&quot;:&quot;string&quot;},{&quot;name&quot;:&quot;idx&quot;,&quot;align&quot;:&quot;right&quot;,&quot;type&quot;:&quot;int64&quot;}],&quot;rows&quot;:[{&quot;rowIdx&quot;:0,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;phylotoast/util.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;split_phylogeny&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def split_phylogeny(p, level=\&quot;s\&quot;):\n    \&quot;\&quot;\&quot;\n    Return either the full or truncated version of a QIIME-formatted taxonomy string.\n\n    :type p: str\n    :param p: A QIIME-formatted taxonomy string: k__Foo; p__Bar; ...\n\n    :type level: str\n    :param level: The different level of identification are kingdom (k), phylum (p),\n                  class (c),order (o), family (f), genus (g) and species (s). If level is\n                  not provided, the default level of identification is species.\n\n    :rtype: str\n    :return: A QIIME-formatted taxonomy string up to the classification given\n            by param level.\n    \&quot;\&quot;\&quot;\n    level = level+\&quot;__\&quot;\n    result = p.split(level)\n    return result[0]+level+result[1].split(\&quot;;\&quot;)[0]&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def split_phylogeny(p, level=\&quot;s\&quot;):\n    \&quot;\&quot;\&quot;\n    Return either the full or truncated version of a QIIME-formatted taxonomy string.\n\n    :type p: str\n    :param p: A QIIME-formatted taxonomy string: k__Foo; p__Bar; ...\n\n    :type level: str\n    :param level: The different level of identification are kingdom (k), phylum (p),\n                  class (c),order (o), family (f), genus (g) and species (s). If level is\n                  not provided, the default level of identification is species.\n\n    :rtype: str\n    :return: A QIIME-formatted taxonomy string up to the classification given\n            by param level.\n    \&quot;\&quot;\&quot;\n    level = level+\&quot;__\&quot;\n    result = p.split(level)\n    return result[0]+level+result[1].split(\&quot;;\&quot;)[0]&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;split_phylogeny&quot;,&quot;(&quot;,&quot;p&quot;,&quot;,&quot;,&quot;level&quot;,&quot;=&quot;,&quot;\&quot;s\&quot;&quot;,&quot;)&quot;,&quot;:&quot;,&quot;level&quot;,&quot;=&quot;,&quot;level&quot;,&quot;+&quot;,&quot;\&quot;__\&quot;&quot;,&quot;result&quot;,&quot;=&quot;,&quot;p&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;level&quot;,&quot;)&quot;,&quot;return&quot;,&quot;result&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;+&quot;,&quot;level&quot;,&quot;+&quot;,&quot;result&quot;,&quot;[&quot;,&quot;1&quot;,&quot;]&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;\&quot;;\&quot;&quot;,&quot;)&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;split_phylogeny\&quot;,\n  \&quot;(\&quot;,\n  \&quot;p\&quot;,\n  \&quot;,\&quot;,\n  \&quot;level\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;s\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;level\&quot;,\n  \&quot;=\&quot;,\n  \&quot;level\&quot;,\n  \&quot;+\&quot;,\n  \&quot;\\\&quot;__\\\&quot;\&quot;,\n  \&quot;result\&quot;,\n  \&quot;=\&quot;,\n  \&quot;p\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;level\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;result\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;+\&quot;,\n  \&quot;level\&quot;,\n  \&quot;+\&quot;,\n  \&quot;result\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;;\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Return either the full or truncated version of a QIIME-formatted taxonomy string.\n\n    :type p: str\n    :param p: A QIIME-formatted taxonomy string: k__Foo; p__Bar; ...\n\n    :type level: str\n    :param level: The different level of identification are kingdom (k), phylum (p),\n                  class (c),order (o), family (f), genus (g) and species (s). If level is\n                  not provided, the default level of identification is species.\n\n    :rtype: str\n    :return: A QIIME-formatted taxonomy string up to the classification given\n            by param level.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Return&quot;,&quot;either&quot;,&quot;the&quot;,&quot;full&quot;,&quot;or&quot;,&quot;truncated&quot;,&quot;version&quot;,&quot;of&quot;,&quot;a&quot;,&quot;QIIME&quot;,&quot;-&quot;,&quot;formatted&quot;,&quot;taxonomy&quot;,&quot;string&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Return\&quot;,\n  \&quot;either\&quot;,\n  \&quot;the\&quot;,\n  \&quot;full\&quot;,\n  \&quot;or\&quot;,\n  \&quot;truncated\&quot;,\n  \&quot;version\&quot;,\n  \&quot;of\&quot;,\n  \&quot;a\&quot;,\n  \&quot;QIIME\&quot;,\n  \&quot;-\&quot;,\n  \&quot;formatted\&quot;,\n  \&quot;taxonomy\&quot;,\n  \&quot;string\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/util.py#L159-L177&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:0,&quot;string&quot;:&quot;0&quot;}}},{&quot;rowIdx&quot;:1,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;phylotoast/util.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ensure_dir&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def ensure_dir(d):\n    \&quot;\&quot;\&quot;\n    Check to make sure the supplied directory path does not exist, if so, create it. The\n    method catches OSError exceptions and returns a descriptive message instead of\n    re-raising the error.\n\n    :type d: str\n    :param d: It is the full path to a directory.\n\n    :return: Does not return anything, but creates a directory path if it doesn't exist\n             already.\n    \&quot;\&quot;\&quot;\n    if not os.path.exists(d):\n        try:\n            os.makedirs(d)\n        except OSError as oe:\n            # should not happen with os.makedirs\n            # ENOENT: No such file or directory\n            if os.errno == errno.ENOENT:\n                msg = twdd(\&quot;\&quot;\&quot;One or more directories in the path ({}) do not exist. If\n                           you are specifying a new directory for output, please ensure\n                           all other directories in the path currently exist.\&quot;\&quot;\&quot;)\n                return msg.format(d)\n            else:\n                msg = twdd(\&quot;\&quot;\&quot;An error occurred trying to create the output directory\n                           ({}) with message: {}\&quot;\&quot;\&quot;)\n                return msg.format(d, oe.strerror)&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def ensure_dir(d):\n    \&quot;\&quot;\&quot;\n    Check to make sure the supplied directory path does not exist, if so, create it. The\n    method catches OSError exceptions and returns a descriptive message instead of\n    re-raising the error.\n\n    :type d: str\n    :param d: It is the full path to a directory.\n\n    :return: Does not return anything, but creates a directory path if it doesn't exist\n             already.\n    \&quot;\&quot;\&quot;\n    if not os.path.exists(d):\n        try:\n            os.makedirs(d)\n        except OSError as oe:\n            # should not happen with os.makedirs\n            # ENOENT: No such file or directory\n            if os.errno == errno.ENOENT:\n                msg = twdd(\&quot;\&quot;\&quot;One or more directories in the path ({}) do not exist. If\n                           you are specifying a new directory for output, please ensure\n                           all other directories in the path currently exist.\&quot;\&quot;\&quot;)\n                return msg.format(d)\n            else:\n                msg = twdd(\&quot;\&quot;\&quot;An error occurred trying to create the output directory\n                           ({}) with message: {}\&quot;\&quot;\&quot;)\n                return msg.format(d, oe.strerror)&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;ensure_dir&quot;,&quot;(&quot;,&quot;d&quot;,&quot;)&quot;,&quot;:&quot;,&quot;if&quot;,&quot;not&quot;,&quot;os&quot;,&quot;.&quot;,&quot;path&quot;,&quot;.&quot;,&quot;exists&quot;,&quot;(&quot;,&quot;d&quot;,&quot;)&quot;,&quot;:&quot;,&quot;try&quot;,&quot;:&quot;,&quot;os&quot;,&quot;.&quot;,&quot;makedirs&quot;,&quot;(&quot;,&quot;d&quot;,&quot;)&quot;,&quot;except&quot;,&quot;OSError&quot;,&quot;as&quot;,&quot;oe&quot;,&quot;:&quot;,&quot;# should not happen with os.makedirs&quot;,&quot;# ENOENT: No such file or directory&quot;,&quot;if&quot;,&quot;os&quot;,&quot;.&quot;,&quot;errno&quot;,&quot;==&quot;,&quot;errno&quot;,&quot;.&quot;,&quot;ENOENT&quot;,&quot;:&quot;,&quot;msg&quot;,&quot;=&quot;,&quot;twdd&quot;,&quot;(&quot;,&quot;\&quot;\&quot;\&quot;One or more directories in the path ({}) do not exist. If\n                           you are specifying a new directory for output, please ensure\n                           all other directories in the path currently exist.\&quot;\&quot;\&quot;&quot;,&quot;)&quot;,&quot;return&quot;,&quot;msg&quot;,&quot;.&quot;,&quot;format&quot;,&quot;(&quot;,&quot;d&quot;,&quot;)&quot;,&quot;else&quot;,&quot;:&quot;,&quot;msg&quot;,&quot;=&quot;,&quot;twdd&quot;,&quot;(&quot;,&quot;\&quot;\&quot;\&quot;An error occurred trying to create the output directory\n                           ({}) with message: {}\&quot;\&quot;\&quot;&quot;,&quot;)&quot;,&quot;return&quot;,&quot;msg&quot;,&quot;.&quot;,&quot;format&quot;,&quot;(&quot;,&quot;d&quot;,&quot;,&quot;,&quot;oe&quot;,&quot;.&quot;,&quot;strerror&quot;,&quot;)&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;ensure_dir\&quot;,\n  \&quot;(\&quot;,\n  \&quot;d\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;not\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;path\&quot;,\n  \&quot;.\&quot;,\n  \&quot;exists\&quot;,\n  \&quot;(\&quot;,\n  \&quot;d\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;try\&quot;,\n  \&quot;:\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;makedirs\&quot;,\n  \&quot;(\&quot;,\n  \&quot;d\&quot;,\n  \&quot;)\&quot;,\n  \&quot;except\&quot;,\n  \&quot;OSError\&quot;,\n  \&quot;as\&quot;,\n  \&quot;oe\&quot;,\n  \&quot;:\&quot;,\n  \&quot;# should not happen with os.makedirs\&quot;,\n  \&quot;# ENOENT: No such file or directory\&quot;,\n  \&quot;if\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;errno\&quot;,\n  \&quot;==\&quot;,\n  \&quot;errno\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ENOENT\&quot;,\n  \&quot;:\&quot;,\n  \&quot;msg\&quot;,\n  \&quot;=\&quot;,\n  \&quot;twdd\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;\\\&quot;\\\&quot;One or more directories in the path ({}) do not exist. If\\n                           you are specifying a new directory for output, please ensure\\n                           all other directories in the path currently exist.\\\&quot;\\\&quot;\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;msg\&quot;,\n  \&quot;.\&quot;,\n  \&quot;format\&quot;,\n  \&quot;(\&quot;,\n  \&quot;d\&quot;,\n  \&quot;)\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;msg\&quot;,\n  \&quot;=\&quot;,\n  \&quot;twdd\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;\\\&quot;\\\&quot;An error occurred trying to create the output directory\\n                           ({}) with message: {}\\\&quot;\\\&quot;\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;msg\&quot;,\n  \&quot;.\&quot;,\n  \&quot;format\&quot;,\n  \&quot;(\&quot;,\n  \&quot;d\&quot;,\n  \&quot;,\&quot;,\n  \&quot;oe\&quot;,\n  \&quot;.\&quot;,\n  \&quot;strerror\&quot;,\n  \&quot;)\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Check to make sure the supplied directory path does not exist, if so, create it. The\n    method catches OSError exceptions and returns a descriptive message instead of\n    re-raising the error.\n\n    :type d: str\n    :param d: It is the full path to a directory.\n\n    :return: Does not return anything, but creates a directory path if it doesn't exist\n             already.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Check&quot;,&quot;to&quot;,&quot;make&quot;,&quot;sure&quot;,&quot;the&quot;,&quot;supplied&quot;,&quot;directory&quot;,&quot;path&quot;,&quot;does&quot;,&quot;not&quot;,&quot;exist&quot;,&quot;if&quot;,&quot;so&quot;,&quot;create&quot;,&quot;it&quot;,&quot;.&quot;,&quot;The&quot;,&quot;method&quot;,&quot;catches&quot;,&quot;OSError&quot;,&quot;exceptions&quot;,&quot;and&quot;,&quot;returns&quot;,&quot;a&quot;,&quot;descriptive&quot;,&quot;message&quot;,&quot;instead&quot;,&quot;of&quot;,&quot;re&quot;,&quot;-&quot;,&quot;raising&quot;,&quot;the&quot;,&quot;error&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Check\&quot;,\n  \&quot;to\&quot;,\n  \&quot;make\&quot;,\n  \&quot;sure\&quot;,\n  \&quot;the\&quot;,\n  \&quot;supplied\&quot;,\n  \&quot;directory\&quot;,\n  \&quot;path\&quot;,\n  \&quot;does\&quot;,\n  \&quot;not\&quot;,\n  \&quot;exist\&quot;,\n  \&quot;if\&quot;,\n  \&quot;so\&quot;,\n  \&quot;create\&quot;,\n  \&quot;it\&quot;,\n  \&quot;.\&quot;,\n  \&quot;The\&quot;,\n  \&quot;method\&quot;,\n  \&quot;catches\&quot;,\n  \&quot;OSError\&quot;,\n  \&quot;exceptions\&quot;,\n  \&quot;and\&quot;,\n  \&quot;returns\&quot;,\n  \&quot;a\&quot;,\n  \&quot;descriptive\&quot;,\n  \&quot;message\&quot;,\n  \&quot;instead\&quot;,\n  \&quot;of\&quot;,\n  \&quot;re\&quot;,\n  \&quot;-\&quot;,\n  \&quot;raising\&quot;,\n  \&quot;the\&quot;,\n  \&quot;error\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/util.py#L180-L206&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:1,&quot;string&quot;:&quot;1&quot;}}},{&quot;rowIdx&quot;:2,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;phylotoast/util.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;file_handle&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def file_handle(fnh, mode=\&quot;rU\&quot;):\n    \&quot;\&quot;\&quot;\n    Takes either a file path or an open file handle, checks validity and returns an open\n    file handle or raises an appropriate Exception.\n\n    :type fnh: str\n    :param fnh: It is the full path to a file, or open file handle\n\n    :type mode: str\n    :param mode: The way in which this file will be used, for example to read or write or\n                 both. By default, file will be opened in rU mode.\n\n    :return: Returns an opened file for appropriate usage.\n    \&quot;\&quot;\&quot;\n    handle = None\n    if isinstance(fnh, file):\n        if fnh.closed:\n            raise ValueError(\&quot;Input file is closed.\&quot;)\n        handle = fnh\n    elif isinstance(fnh, str):\n        handle = open(fnh, mode)\n\n    return handle&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def file_handle(fnh, mode=\&quot;rU\&quot;):\n    \&quot;\&quot;\&quot;\n    Takes either a file path or an open file handle, checks validity and returns an open\n    file handle or raises an appropriate Exception.\n\n    :type fnh: str\n    :param fnh: It is the full path to a file, or open file handle\n\n    :type mode: str\n    :param mode: The way in which this file will be used, for example to read or write or\n                 both. By default, file will be opened in rU mode.\n\n    :return: Returns an opened file for appropriate usage.\n    \&quot;\&quot;\&quot;\n    handle = None\n    if isinstance(fnh, file):\n        if fnh.closed:\n            raise ValueError(\&quot;Input file is closed.\&quot;)\n        handle = fnh\n    elif isinstance(fnh, str):\n        handle = open(fnh, mode)\n\n    return handle&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;file_handle&quot;,&quot;(&quot;,&quot;fnh&quot;,&quot;,&quot;,&quot;mode&quot;,&quot;=&quot;,&quot;\&quot;rU\&quot;&quot;,&quot;)&quot;,&quot;:&quot;,&quot;handle&quot;,&quot;=&quot;,&quot;None&quot;,&quot;if&quot;,&quot;isinstance&quot;,&quot;(&quot;,&quot;fnh&quot;,&quot;,&quot;,&quot;file&quot;,&quot;)&quot;,&quot;:&quot;,&quot;if&quot;,&quot;fnh&quot;,&quot;.&quot;,&quot;closed&quot;,&quot;:&quot;,&quot;raise&quot;,&quot;ValueError&quot;,&quot;(&quot;,&quot;\&quot;Input file is closed.\&quot;&quot;,&quot;)&quot;,&quot;handle&quot;,&quot;=&quot;,&quot;fnh&quot;,&quot;elif&quot;,&quot;isinstance&quot;,&quot;(&quot;,&quot;fnh&quot;,&quot;,&quot;,&quot;str&quot;,&quot;)&quot;,&quot;:&quot;,&quot;handle&quot;,&quot;=&quot;,&quot;open&quot;,&quot;(&quot;,&quot;fnh&quot;,&quot;,&quot;,&quot;mode&quot;,&quot;)&quot;,&quot;return&quot;,&quot;handle&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;file_handle\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fnh\&quot;,\n  \&quot;,\&quot;,\n  \&quot;mode\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;rU\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;handle\&quot;,\n  \&quot;=\&quot;,\n  \&quot;None\&quot;,\n  \&quot;if\&quot;,\n  \&quot;isinstance\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fnh\&quot;,\n  \&quot;,\&quot;,\n  \&quot;file\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;fnh\&quot;,\n  \&quot;.\&quot;,\n  \&quot;closed\&quot;,\n  \&quot;:\&quot;,\n  \&quot;raise\&quot;,\n  \&quot;ValueError\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;Input file is closed.\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;handle\&quot;,\n  \&quot;=\&quot;,\n  \&quot;fnh\&quot;,\n  \&quot;elif\&quot;,\n  \&quot;isinstance\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fnh\&quot;,\n  \&quot;,\&quot;,\n  \&quot;str\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;handle\&quot;,\n  \&quot;=\&quot;,\n  \&quot;open\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fnh\&quot;,\n  \&quot;,\&quot;,\n  \&quot;mode\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;handle\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Takes either a file path or an open file handle, checks validity and returns an open\n    file handle or raises an appropriate Exception.\n\n    :type fnh: str\n    :param fnh: It is the full path to a file, or open file handle\n\n    :type mode: str\n    :param mode: The way in which this file will be used, for example to read or write or\n                 both. By default, file will be opened in rU mode.\n\n    :return: Returns an opened file for appropriate usage.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Takes&quot;,&quot;either&quot;,&quot;a&quot;,&quot;file&quot;,&quot;path&quot;,&quot;or&quot;,&quot;an&quot;,&quot;open&quot;,&quot;file&quot;,&quot;handle&quot;,&quot;checks&quot;,&quot;validity&quot;,&quot;and&quot;,&quot;returns&quot;,&quot;an&quot;,&quot;open&quot;,&quot;file&quot;,&quot;handle&quot;,&quot;or&quot;,&quot;raises&quot;,&quot;an&quot;,&quot;appropriate&quot;,&quot;Exception&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Takes\&quot;,\n  \&quot;either\&quot;,\n  \&quot;a\&quot;,\n  \&quot;file\&quot;,\n  \&quot;path\&quot;,\n  \&quot;or\&quot;,\n  \&quot;an\&quot;,\n  \&quot;open\&quot;,\n  \&quot;file\&quot;,\n  \&quot;handle\&quot;,\n  \&quot;checks\&quot;,\n  \&quot;validity\&quot;,\n  \&quot;and\&quot;,\n  \&quot;returns\&quot;,\n  \&quot;an\&quot;,\n  \&quot;open\&quot;,\n  \&quot;file\&quot;,\n  \&quot;handle\&quot;,\n  \&quot;or\&quot;,\n  \&quot;raises\&quot;,\n  \&quot;an\&quot;,\n  \&quot;appropriate\&quot;,\n  \&quot;Exception\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/util.py#L209-L231&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:2,&quot;string&quot;:&quot;2&quot;}}},{&quot;rowIdx&quot;:3,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;phylotoast/util.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;gather_categories&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def gather_categories(imap, header, categories=None):\n    \&quot;\&quot;\&quot;\n    Find the user specified categories in the map and create a dictionary to contain the\n    relevant data for each type within the categories. Multiple categories will have their\n    types combined such that each possible combination will have its own entry in the\n    dictionary.\n\n    :type imap: dict\n    :param imap: The input mapping file data keyed by SampleID\n    :type header: list\n    :param header: The header line from the input mapping file. This will be searched for\n                   the user-specified categories\n    :type categories: list\n    :param categories: The list of user-specified category column name from mapping file\n    :rtype: dict\n    :return: A sorted dictionary keyed on the combinations of all the types found within\n             the user-specified categories. Each entry will contain an empty DataCategory\n             namedtuple. If no categories are specified, a single entry with the key\n             'default' will be returned\n    \&quot;\&quot;\&quot;\n    # If no categories provided, return all SampleIDs\n    if categories is None:\n        return {\&quot;default\&quot;: DataCategory(set(imap.keys()), {})}\n\n    cat_ids = [header.index(cat)\n               for cat in categories if cat in header and \&quot;=\&quot; not in cat]\n\n    table = OrderedDict()\n    conditions = defaultdict(set)\n    for i, cat in enumerate(categories):\n        if \&quot;=\&quot; in cat and cat.split(\&quot;=\&quot;)[0] in header:\n            cat_name = header[header.index(cat.split(\&quot;=\&quot;)[0])]\n            conditions[cat_name].add(cat.split(\&quot;=\&quot;)[1])\n\n    # If invalid categories or conditions identified, return all SampleIDs\n    if not cat_ids and not conditions:\n        return {\&quot;default\&quot;: DataCategory(set(imap.keys()), {})}\n\n    #If only category column given, return column-wise SampleIDs\n    if cat_ids and not conditions:\n        for sid, row in imap.items():\n            cat_name = \&quot;_\&quot;.join([row[cid] for cid in cat_ids])\n            if cat_name not in table:\n                table[cat_name] = DataCategory(set(), {})\n            table[cat_name].sids.add(sid)\n        return table\n\n    # Collect all condition names\n    cond_ids = set()\n    for k in conditions:\n        try:\n            cond_ids.add(header.index(k))\n        except ValueError:\n            continue\n    idx_to_test = set(cat_ids).union(cond_ids)\n\n    # If column name and condition given, return overlapping SampleIDs of column and\n    # condition combinations\n    for sid, row in imap.items():\n        if all([row[header.index(c)] in conditions[c] for c in conditions]):\n            key = \&quot;_\&quot;.join([row[idx] for idx in idx_to_test])\n            try:\n                assert key in table.keys()\n            except AssertionError:\n                table[key] = DataCategory(set(), {})\n            table[key].sids.add(sid)\n    try:\n        assert len(table) > 0\n    except AssertionError:\n        return {\&quot;default\&quot;: DataCategory(set(imap.keys()), {})}\n    else:\n        return table&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def gather_categories(imap, header, categories=None):\n    \&quot;\&quot;\&quot;\n    Find the user specified categories in the map and create a dictionary to contain the\n    relevant data for each type within the categories. Multiple categories will have their\n    types combined such that each possible combination will have its own entry in the\n    dictionary.\n\n    :type imap: dict\n    :param imap: The input mapping file data keyed by SampleID\n    :type header: list\n    :param header: The header line from the input mapping file. This will be searched for\n                   the user-specified categories\n    :type categories: list\n    :param categories: The list of user-specified category column name from mapping file\n    :rtype: dict\n    :return: A sorted dictionary keyed on the combinations of all the types found within\n             the user-specified categories. Each entry will contain an empty DataCategory\n             namedtuple. If no categories are specified, a single entry with the key\n             'default' will be returned\n    \&quot;\&quot;\&quot;\n    # If no categories provided, return all SampleIDs\n    if categories is None:\n        return {\&quot;default\&quot;: DataCategory(set(imap.keys()), {})}\n\n    cat_ids = [header.index(cat)\n               for cat in categories if cat in header and \&quot;=\&quot; not in cat]\n\n    table = OrderedDict()\n    conditions = defaultdict(set)\n    for i, cat in enumerate(categories):\n        if \&quot;=\&quot; in cat and cat.split(\&quot;=\&quot;)[0] in header:\n            cat_name = header[header.index(cat.split(\&quot;=\&quot;)[0])]\n            conditions[cat_name].add(cat.split(\&quot;=\&quot;)[1])\n\n    # If invalid categories or conditions identified, return all SampleIDs\n    if not cat_ids and not conditions:\n        return {\&quot;default\&quot;: DataCategory(set(imap.keys()), {})}\n\n    #If only category column given, return column-wise SampleIDs\n    if cat_ids and not conditions:\n        for sid, row in imap.items():\n            cat_name = \&quot;_\&quot;.join([row[cid] for cid in cat_ids])\n            if cat_name not in table:\n                table[cat_name] = DataCategory(set(), {})\n            table[cat_name].sids.add(sid)\n        return table\n\n    # Collect all condition names\n    cond_ids = set()\n    for k in conditions:\n        try:\n            cond_ids.add(header.index(k))\n        except ValueError:\n            continue\n    idx_to_test = set(cat_ids).union(cond_ids)\n\n    # If column name and condition given, return overlapping SampleIDs of column and\n    # condition combinations\n    for sid, row in imap.items():\n        if all([row[header.index(c)] in conditions[c] for c in conditions]):\n            key = \&quot;_\&quot;.join([row[idx] for idx in idx_to_test])\n            try:\n                assert key in table.keys()\n            except AssertionError:\n                table[key] = DataCategory(set(), {})\n            table[key].sids.add(sid)\n    try:\n        assert len(table) > 0\n    except AssertionError:\n        return {\&quot;default\&quot;: DataCategory(set(imap.keys()), {})}\n    else:\n        return table&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;gather_categories&quot;,&quot;(&quot;,&quot;imap&quot;,&quot;,&quot;,&quot;header&quot;,&quot;,&quot;,&quot;categories&quot;,&quot;=&quot;,&quot;None&quot;,&quot;)&quot;,&quot;:&quot;,&quot;# If no categories provided, return all SampleIDs&quot;,&quot;if&quot;,&quot;categories&quot;,&quot;is&quot;,&quot;None&quot;,&quot;:&quot;,&quot;return&quot;,&quot;{&quot;,&quot;\&quot;default\&quot;&quot;,&quot;:&quot;,&quot;DataCategory&quot;,&quot;(&quot;,&quot;set&quot;,&quot;(&quot;,&quot;imap&quot;,&quot;.&quot;,&quot;keys&quot;,&quot;(&quot;,&quot;)&quot;,&quot;)&quot;,&quot;,&quot;,&quot;{&quot;,&quot;}&quot;,&quot;)&quot;,&quot;}&quot;,&quot;cat_ids&quot;,&quot;=&quot;,&quot;[&quot;,&quot;header&quot;,&quot;.&quot;,&quot;index&quot;,&quot;(&quot;,&quot;cat&quot;,&quot;)&quot;,&quot;for&quot;,&quot;cat&quot;,&quot;in&quot;,&quot;categories&quot;,&quot;if&quot;,&quot;cat&quot;,&quot;in&quot;,&quot;header&quot;,&quot;and&quot;,&quot;\&quot;=\&quot;&quot;,&quot;not&quot;,&quot;in&quot;,&quot;cat&quot;,&quot;]&quot;,&quot;table&quot;,&quot;=&quot;,&quot;OrderedDict&quot;,&quot;(&quot;,&quot;)&quot;,&quot;conditions&quot;,&quot;=&quot;,&quot;defaultdict&quot;,&quot;(&quot;,&quot;set&quot;,&quot;)&quot;,&quot;for&quot;,&quot;i&quot;,&quot;,&quot;,&quot;cat&quot;,&quot;in&quot;,&quot;enumerate&quot;,&quot;(&quot;,&quot;categories&quot;,&quot;)&quot;,&quot;:&quot;,&quot;if&quot;,&quot;\&quot;=\&quot;&quot;,&quot;in&quot;,&quot;cat&quot;,&quot;and&quot;,&quot;cat&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;\&quot;=\&quot;&quot;,&quot;)&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;in&quot;,&quot;header&quot;,&quot;:&quot;,&quot;cat_name&quot;,&quot;=&quot;,&quot;header&quot;,&quot;[&quot;,&quot;header&quot;,&quot;.&quot;,&quot;index&quot;,&quot;(&quot;,&quot;cat&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;\&quot;=\&quot;&quot;,&quot;)&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;)&quot;,&quot;]&quot;,&quot;conditions&quot;,&quot;[&quot;,&quot;cat_name&quot;,&quot;]&quot;,&quot;.&quot;,&quot;add&quot;,&quot;(&quot;,&quot;cat&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;\&quot;=\&quot;&quot;,&quot;)&quot;,&quot;[&quot;,&quot;1&quot;,&quot;]&quot;,&quot;)&quot;,&quot;# If invalid categories or conditions identified, return all SampleIDs&quot;,&quot;if&quot;,&quot;not&quot;,&quot;cat_ids&quot;,&quot;and&quot;,&quot;not&quot;,&quot;conditions&quot;,&quot;:&quot;,&quot;return&quot;,&quot;{&quot;,&quot;\&quot;default\&quot;&quot;,&quot;:&quot;,&quot;DataCategory&quot;,&quot;(&quot;,&quot;set&quot;,&quot;(&quot;,&quot;imap&quot;,&quot;.&quot;,&quot;keys&quot;,&quot;(&quot;,&quot;)&quot;,&quot;)&quot;,&quot;,&quot;,&quot;{&quot;,&quot;}&quot;,&quot;)&quot;,&quot;}&quot;,&quot;#If only category column given, return column-wise SampleIDs&quot;,&quot;if&quot;,&quot;cat_ids&quot;,&quot;and&quot;,&quot;not&quot;,&quot;conditions&quot;,&quot;:&quot;,&quot;for&quot;,&quot;sid&quot;,&quot;,&quot;,&quot;row&quot;,&quot;in&quot;,&quot;imap&quot;,&quot;.&quot;,&quot;items&quot;,&quot;(&quot;,&quot;)&quot;,&quot;:&quot;,&quot;cat_name&quot;,&quot;=&quot;,&quot;\&quot;_\&quot;&quot;,&quot;.&quot;,&quot;join&quot;,&quot;(&quot;,&quot;[&quot;,&quot;row&quot;,&quot;[&quot;,&quot;cid&quot;,&quot;]&quot;,&quot;for&quot;,&quot;cid&quot;,&quot;in&quot;,&quot;cat_ids&quot;,&quot;]&quot;,&quot;)&quot;,&quot;if&quot;,&quot;cat_name&quot;,&quot;not&quot;,&quot;in&quot;,&quot;table&quot;,&quot;:&quot;,&quot;table&quot;,&quot;[&quot;,&quot;cat_name&quot;,&quot;]&quot;,&quot;=&quot;,&quot;DataCategory&quot;,&quot;(&quot;,&quot;set&quot;,&quot;(&quot;,&quot;)&quot;,&quot;,&quot;,&quot;{&quot;,&quot;}&quot;,&quot;)&quot;,&quot;table&quot;,&quot;[&quot;,&quot;cat_name&quot;,&quot;]&quot;,&quot;.&quot;,&quot;sids&quot;,&quot;.&quot;,&quot;add&quot;,&quot;(&quot;,&quot;sid&quot;,&quot;)&quot;,&quot;return&quot;,&quot;table&quot;,&quot;# Collect all condition names&quot;,&quot;cond_ids&quot;,&quot;=&quot;,&quot;set&quot;,&quot;(&quot;,&quot;)&quot;,&quot;for&quot;,&quot;k&quot;,&quot;in&quot;,&quot;conditions&quot;,&quot;:&quot;,&quot;try&quot;,&quot;:&quot;,&quot;cond_ids&quot;,&quot;.&quot;,&quot;add&quot;,&quot;(&quot;,&quot;header&quot;,&quot;.&quot;,&quot;index&quot;,&quot;(&quot;,&quot;k&quot;,&quot;)&quot;,&quot;)&quot;,&quot;except&quot;,&quot;ValueError&quot;,&quot;:&quot;,&quot;continue&quot;,&quot;idx_to_test&quot;,&quot;=&quot;,&quot;set&quot;,&quot;(&quot;,&quot;cat_ids&quot;,&quot;)&quot;,&quot;.&quot;,&quot;union&quot;,&quot;(&quot;,&quot;cond_ids&quot;,&quot;)&quot;,&quot;# If column name and condition given, return overlapping SampleIDs of column and&quot;,&quot;# condition combinations&quot;,&quot;for&quot;,&quot;sid&quot;,&quot;,&quot;,&quot;row&quot;,&quot;in&quot;,&quot;imap&quot;,&quot;.&quot;,&quot;items&quot;,&quot;(&quot;,&quot;)&quot;,&quot;:&quot;,&quot;if&quot;,&quot;all&quot;,&quot;(&quot;,&quot;[&quot;,&quot;row&quot;,&quot;[&quot;,&quot;header&quot;,&quot;.&quot;,&quot;index&quot;,&quot;(&quot;,&quot;c&quot;,&quot;)&quot;,&quot;]&quot;,&quot;in&quot;,&quot;conditions&quot;,&quot;[&quot;,&quot;c&quot;,&quot;]&quot;,&quot;for&quot;,&quot;c&quot;,&quot;in&quot;,&quot;conditions&quot;,&quot;]&quot;,&quot;)&quot;,&quot;:&quot;,&quot;key&quot;,&quot;=&quot;,&quot;\&quot;_\&quot;&quot;,&quot;.&quot;,&quot;join&quot;,&quot;(&quot;,&quot;[&quot;,&quot;row&quot;,&quot;[&quot;,&quot;idx&quot;,&quot;]&quot;,&quot;for&quot;,&quot;idx&quot;,&quot;in&quot;,&quot;idx_to_test&quot;,&quot;]&quot;,&quot;)&quot;,&quot;try&quot;,&quot;:&quot;,&quot;assert&quot;,&quot;key&quot;,&quot;in&quot;,&quot;table&quot;,&quot;.&quot;,&quot;keys&quot;,&quot;(&quot;,&quot;)&quot;,&quot;except&quot;,&quot;AssertionError&quot;,&quot;:&quot;,&quot;table&quot;,&quot;[&quot;,&quot;key&quot;,&quot;]&quot;,&quot;=&quot;,&quot;DataCategory&quot;,&quot;(&quot;,&quot;set&quot;,&quot;(&quot;,&quot;)&quot;,&quot;,&quot;,&quot;{&quot;,&quot;}&quot;,&quot;)&quot;,&quot;table&quot;,&quot;[&quot;,&quot;key&quot;,&quot;]&quot;,&quot;.&quot;,&quot;sids&quot;,&quot;.&quot;,&quot;add&quot;,&quot;(&quot;,&quot;sid&quot;,&quot;)&quot;,&quot;try&quot;,&quot;:&quot;,&quot;assert&quot;,&quot;len&quot;,&quot;(&quot;,&quot;table&quot;,&quot;)&quot;,&quot;>&quot;,&quot;0&quot;,&quot;except&quot;,&quot;AssertionError&quot;,&quot;:&quot;,&quot;return&quot;,&quot;{&quot;,&quot;\&quot;default\&quot;&quot;,&quot;:&quot;,&quot;DataCategory&quot;,&quot;(&quot;,&quot;set&quot;,&quot;(&quot;,&quot;imap&quot;,&quot;.&quot;,&quot;keys&quot;,&quot;(&quot;,&quot;)&quot;,&quot;)&quot;,&quot;,&quot;,&quot;{&quot;,&quot;}&quot;,&quot;)&quot;,&quot;}&quot;,&quot;else&quot;,&quot;:&quot;,&quot;return&quot;,&quot;table&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;gather_categories\&quot;,\n  \&quot;(\&quot;,\n  \&quot;imap\&quot;,\n  \&quot;,\&quot;,\n  \&quot;header\&quot;,\n  \&quot;,\&quot;,\n  \&quot;categories\&quot;,\n  \&quot;=\&quot;,\n  \&quot;None\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;# If no categories provided, return all SampleIDs\&quot;,\n  \&quot;if\&quot;,\n  \&quot;categories\&quot;,\n  \&quot;is\&quot;,\n  \&quot;None\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;{\&quot;,\n  \&quot;\\\&quot;default\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;DataCategory\&quot;,\n  \&quot;(\&quot;,\n  \&quot;set\&quot;,\n  \&quot;(\&quot;,\n  \&quot;imap\&quot;,\n  \&quot;.\&quot;,\n  \&quot;keys\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;{\&quot;,\n  \&quot;}\&quot;,\n  \&quot;)\&quot;,\n  \&quot;}\&quot;,\n  \&quot;cat_ids\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;header\&quot;,\n  \&quot;.\&quot;,\n  \&quot;index\&quot;,\n  \&quot;(\&quot;,\n  \&quot;cat\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;cat\&quot;,\n  \&quot;in\&quot;,\n  \&quot;categories\&quot;,\n  \&quot;if\&quot;,\n  \&quot;cat\&quot;,\n  \&quot;in\&quot;,\n  \&quot;header\&quot;,\n  \&quot;and\&quot;,\n  \&quot;\\\&quot;=\\\&quot;\&quot;,\n  \&quot;not\&quot;,\n  \&quot;in\&quot;,\n  \&quot;cat\&quot;,\n  \&quot;]\&quot;,\n  \&quot;table\&quot;,\n  \&quot;=\&quot;,\n  \&quot;OrderedDict\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;conditions\&quot;,\n  \&quot;=\&quot;,\n  \&quot;defaultdict\&quot;,\n  \&quot;(\&quot;,\n  \&quot;set\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;,\&quot;,\n  \&quot;cat\&quot;,\n  \&quot;in\&quot;,\n  \&quot;enumerate\&quot;,\n  \&quot;(\&quot;,\n  \&quot;categories\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;\\\&quot;=\\\&quot;\&quot;,\n  \&quot;in\&quot;,\n  \&quot;cat\&quot;,\n  \&quot;and\&quot;,\n  \&quot;cat\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;=\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;in\&quot;,\n  \&quot;header\&quot;,\n  \&quot;:\&quot;,\n  \&quot;cat_name\&quot;,\n  \&quot;=\&quot;,\n  \&quot;header\&quot;,\n  \&quot;[\&quot;,\n  \&quot;header\&quot;,\n  \&quot;.\&quot;,\n  \&quot;index\&quot;,\n  \&quot;(\&quot;,\n  \&quot;cat\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;=\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;]\&quot;,\n  \&quot;conditions\&quot;,\n  \&quot;[\&quot;,\n  \&quot;cat_name\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;add\&quot;,\n  \&quot;(\&quot;,\n  \&quot;cat\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;=\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;# If invalid categories or conditions identified, return all SampleIDs\&quot;,\n  \&quot;if\&quot;,\n  \&quot;not\&quot;,\n  \&quot;cat_ids\&quot;,\n  \&quot;and\&quot;,\n  \&quot;not\&quot;,\n  \&quot;conditions\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;{\&quot;,\n  \&quot;\\\&quot;default\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;DataCategory\&quot;,\n  \&quot;(\&quot;,\n  \&quot;set\&quot;,\n  \&quot;(\&quot;,\n  \&quot;imap\&quot;,\n  \&quot;.\&quot;,\n  \&quot;keys\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;{\&quot;,\n  \&quot;}\&quot;,\n  \&quot;)\&quot;,\n  \&quot;}\&quot;,\n  \&quot;#If only category column given, return column-wise SampleIDs\&quot;,\n  \&quot;if\&quot;,\n  \&quot;cat_ids\&quot;,\n  \&quot;and\&quot;,\n  \&quot;not\&quot;,\n  \&quot;conditions\&quot;,\n  \&quot;:\&quot;,\n  \&quot;for\&quot;,\n  \&quot;sid\&quot;,\n  \&quot;,\&quot;,\n  \&quot;row\&quot;,\n  \&quot;in\&quot;,\n  \&quot;imap\&quot;,\n  \&quot;.\&quot;,\n  \&quot;items\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;cat_name\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;_\\\&quot;\&quot;,\n  \&quot;.\&quot;,\n  \&quot;join\&quot;,\n  \&quot;(\&quot;,\n  \&quot;[\&quot;,\n  \&quot;row\&quot;,\n  \&quot;[\&quot;,\n  \&quot;cid\&quot;,\n  \&quot;]\&quot;,\n  \&quot;for\&quot;,\n  \&quot;cid\&quot;,\n  \&quot;in\&quot;,\n  \&quot;cat_ids\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;cat_name\&quot;,\n  \&quot;not\&quot;,\n  \&quot;in\&quot;,\n  \&quot;table\&quot;,\n  \&quot;:\&quot;,\n  \&quot;table\&quot;,\n  \&quot;[\&quot;,\n  \&quot;cat_name\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;DataCategory\&quot;,\n  \&quot;(\&quot;,\n  \&quot;set\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;{\&quot;,\n  \&quot;}\&quot;,\n  \&quot;)\&quot;,\n  \&quot;table\&quot;,\n  \&quot;[\&quot;,\n  \&quot;cat_name\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;sids\&quot;,\n  \&quot;.\&quot;,\n  \&quot;add\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sid\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;table\&quot;,\n  \&quot;# Collect all condition names\&quot;,\n  \&quot;cond_ids\&quot;,\n  \&quot;=\&quot;,\n  \&quot;set\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;k\&quot;,\n  \&quot;in\&quot;,\n  \&quot;conditions\&quot;,\n  \&quot;:\&quot;,\n  \&quot;try\&quot;,\n  \&quot;:\&quot;,\n  \&quot;cond_ids\&quot;,\n  \&quot;.\&quot;,\n  \&quot;add\&quot;,\n  \&quot;(\&quot;,\n  \&quot;header\&quot;,\n  \&quot;.\&quot;,\n  \&quot;index\&quot;,\n  \&quot;(\&quot;,\n  \&quot;k\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;except\&quot;,\n  \&quot;ValueError\&quot;,\n  \&quot;:\&quot;,\n  \&quot;continue\&quot;,\n  \&quot;idx_to_test\&quot;,\n  \&quot;=\&quot;,\n  \&quot;set\&quot;,\n  \&quot;(\&quot;,\n  \&quot;cat_ids\&quot;,\n  \&quot;)\&quot;,\n  \&quot;.\&quot;,\n  \&quot;union\&quot;,\n  \&quot;(\&quot;,\n  \&quot;cond_ids\&quot;,\n  \&quot;)\&quot;,\n  \&quot;# If column name and condition given, return overlapping SampleIDs of column and\&quot;,\n  \&quot;# condition combinations\&quot;,\n  \&quot;for\&quot;,\n  \&quot;sid\&quot;,\n  \&quot;,\&quot;,\n  \&quot;row\&quot;,\n  \&quot;in\&quot;,\n  \&quot;imap\&quot;,\n  \&quot;.\&quot;,\n  \&quot;items\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;all\&quot;,\n  \&quot;(\&quot;,\n  \&quot;[\&quot;,\n  \&quot;row\&quot;,\n  \&quot;[\&quot;,\n  \&quot;header\&quot;,\n  \&quot;.\&quot;,\n  \&quot;index\&quot;,\n  \&quot;(\&quot;,\n  \&quot;c\&quot;,\n  \&quot;)\&quot;,\n  \&quot;]\&quot;,\n  \&quot;in\&quot;,\n  \&quot;conditions\&quot;,\n  \&quot;[\&quot;,\n  \&quot;c\&quot;,\n  \&quot;]\&quot;,\n  \&quot;for\&quot;,\n  \&quot;c\&quot;,\n  \&quot;in\&quot;,\n  \&quot;conditions\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;key\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;_\\\&quot;\&quot;,\n  \&quot;.\&quot;,\n  \&quot;join\&quot;,\n  \&quot;(\&quot;,\n  \&quot;[\&quot;,\n  \&quot;row\&quot;,\n  \&quot;[\&quot;,\n  \&quot;idx\&quot;,\n  \&quot;]\&quot;,\n  \&quot;for\&quot;,\n  \&quot;idx\&quot;,\n  \&quot;in\&quot;,\n  \&quot;idx_to_test\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;try\&quot;,\n  \&quot;:\&quot;,\n  \&quot;assert\&quot;,\n  \&quot;key\&quot;,\n  \&quot;in\&quot;,\n  \&quot;table\&quot;,\n  \&quot;.\&quot;,\n  \&quot;keys\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;except\&quot;,\n  \&quot;AssertionError\&quot;,\n  \&quot;:\&quot;,\n  \&quot;table\&quot;,\n  \&quot;[\&quot;,\n  \&quot;key\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;DataCategory\&quot;,\n  \&quot;(\&quot;,\n  \&quot;set\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;{\&quot;,\n  \&quot;}\&quot;,\n  \&quot;)\&quot;,\n  \&quot;table\&quot;,\n  \&quot;[\&quot;,\n  \&quot;key\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;sids\&quot;,\n  \&quot;.\&quot;,\n  \&quot;add\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sid\&quot;,\n  \&quot;)\&quot;,\n  \&quot;try\&quot;,\n  \&quot;:\&quot;,\n  \&quot;assert\&quot;,\n  \&quot;len\&quot;,\n  \&quot;(\&quot;,\n  \&quot;table\&quot;,\n  \&quot;)\&quot;,\n  \&quot;>\&quot;,\n  \&quot;0\&quot;,\n  \&quot;except\&quot;,\n  \&quot;AssertionError\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;{\&quot;,\n  \&quot;\\\&quot;default\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;DataCategory\&quot;,\n  \&quot;(\&quot;,\n  \&quot;set\&quot;,\n  \&quot;(\&quot;,\n  \&quot;imap\&quot;,\n  \&quot;.\&quot;,\n  \&quot;keys\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;{\&quot;,\n  \&quot;}\&quot;,\n  \&quot;)\&quot;,\n  \&quot;}\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;table\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Find the user specified categories in the map and create a dictionary to contain the\n    relevant data for each type within the categories. Multiple categories will have their\n    types combined such that each possible combination will have its own entry in the\n    dictionary.\n\n    :type imap: dict\n    :param imap: The input mapping file data keyed by SampleID\n    :type header: list\n    :param header: The header line from the input mapping file. This will be searched for\n                   the user-specified categories\n    :type categories: list\n    :param categories: The list of user-specified category column name from mapping file\n    :rtype: dict\n    :return: A sorted dictionary keyed on the combinations of all the types found within\n             the user-specified categories. Each entry will contain an empty DataCategory\n             namedtuple. If no categories are specified, a single entry with the key\n             'default' will be returned&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Find&quot;,&quot;the&quot;,&quot;user&quot;,&quot;specified&quot;,&quot;categories&quot;,&quot;in&quot;,&quot;the&quot;,&quot;map&quot;,&quot;and&quot;,&quot;create&quot;,&quot;a&quot;,&quot;dictionary&quot;,&quot;to&quot;,&quot;contain&quot;,&quot;the&quot;,&quot;relevant&quot;,&quot;data&quot;,&quot;for&quot;,&quot;each&quot;,&quot;type&quot;,&quot;within&quot;,&quot;the&quot;,&quot;categories&quot;,&quot;.&quot;,&quot;Multiple&quot;,&quot;categories&quot;,&quot;will&quot;,&quot;have&quot;,&quot;their&quot;,&quot;types&quot;,&quot;combined&quot;,&quot;such&quot;,&quot;that&quot;,&quot;each&quot;,&quot;possible&quot;,&quot;combination&quot;,&quot;will&quot;,&quot;have&quot;,&quot;its&quot;,&quot;own&quot;,&quot;entry&quot;,&quot;in&quot;,&quot;the&quot;,&quot;dictionary&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Find\&quot;,\n  \&quot;the\&quot;,\n  \&quot;user\&quot;,\n  \&quot;specified\&quot;,\n  \&quot;categories\&quot;,\n  \&quot;in\&quot;,\n  \&quot;the\&quot;,\n  \&quot;map\&quot;,\n  \&quot;and\&quot;,\n  \&quot;create\&quot;,\n  \&quot;a\&quot;,\n  \&quot;dictionary\&quot;,\n  \&quot;to\&quot;,\n  \&quot;contain\&quot;,\n  \&quot;the\&quot;,\n  \&quot;relevant\&quot;,\n  \&quot;data\&quot;,\n  \&quot;for\&quot;,\n  \&quot;each\&quot;,\n  \&quot;type\&quot;,\n  \&quot;within\&quot;,\n  \&quot;the\&quot;,\n  \&quot;categories\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Multiple\&quot;,\n  \&quot;categories\&quot;,\n  \&quot;will\&quot;,\n  \&quot;have\&quot;,\n  \&quot;their\&quot;,\n  \&quot;types\&quot;,\n  \&quot;combined\&quot;,\n  \&quot;such\&quot;,\n  \&quot;that\&quot;,\n  \&quot;each\&quot;,\n  \&quot;possible\&quot;,\n  \&quot;combination\&quot;,\n  \&quot;will\&quot;,\n  \&quot;have\&quot;,\n  \&quot;its\&quot;,\n  \&quot;own\&quot;,\n  \&quot;entry\&quot;,\n  \&quot;in\&quot;,\n  \&quot;the\&quot;,\n  \&quot;dictionary\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/util.py#L238-L309&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:3,&quot;string&quot;:&quot;3&quot;}}},{&quot;rowIdx&quot;:4,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;phylotoast/util.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;parse_unifrac&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def parse_unifrac(unifracFN):\n    \&quot;\&quot;\&quot;\n    Parses the unifrac results file into a dictionary\n\n    :type unifracFN: str\n    :param unifracFN: The path to the unifrac results file\n\n    :rtype: dict\n    :return: A dictionary with keys: 'pcd' (principle coordinates data) which is a\n             dictionary of the data keyed by sample ID, 'eigvals' (eigenvalues), and\n             'varexp' (variation explained)\n    \&quot;\&quot;\&quot;\n    with open(unifracFN, \&quot;rU\&quot;) as uF:\n        first = uF.next().split(\&quot;\\t\&quot;)\n        lines = [line.strip() for line in uF]\n\n    unifrac = {\&quot;pcd\&quot;: OrderedDict(), \&quot;eigvals\&quot;: [], \&quot;varexp\&quot;: []}\n    if first[0] == \&quot;pc vector number\&quot;:\n        return parse_unifrac_v1_8(unifrac, lines)\n    elif first[0] == \&quot;Eigvals\&quot;:\n        return parse_unifrac_v1_9(unifrac, lines)\n    else:\n        raise ValueError(\&quot;File format not supported/recognized. Please check input \&quot;\n                         \&quot;unifrac file.\&quot;)&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def parse_unifrac(unifracFN):\n    \&quot;\&quot;\&quot;\n    Parses the unifrac results file into a dictionary\n\n    :type unifracFN: str\n    :param unifracFN: The path to the unifrac results file\n\n    :rtype: dict\n    :return: A dictionary with keys: 'pcd' (principle coordinates data) which is a\n             dictionary of the data keyed by sample ID, 'eigvals' (eigenvalues), and\n             'varexp' (variation explained)\n    \&quot;\&quot;\&quot;\n    with open(unifracFN, \&quot;rU\&quot;) as uF:\n        first = uF.next().split(\&quot;\\t\&quot;)\n        lines = [line.strip() for line in uF]\n\n    unifrac = {\&quot;pcd\&quot;: OrderedDict(), \&quot;eigvals\&quot;: [], \&quot;varexp\&quot;: []}\n    if first[0] == \&quot;pc vector number\&quot;:\n        return parse_unifrac_v1_8(unifrac, lines)\n    elif first[0] == \&quot;Eigvals\&quot;:\n        return parse_unifrac_v1_9(unifrac, lines)\n    else:\n        raise ValueError(\&quot;File format not supported/recognized. Please check input \&quot;\n                         \&quot;unifrac file.\&quot;)&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;parse_unifrac&quot;,&quot;(&quot;,&quot;unifracFN&quot;,&quot;)&quot;,&quot;:&quot;,&quot;with&quot;,&quot;open&quot;,&quot;(&quot;,&quot;unifracFN&quot;,&quot;,&quot;,&quot;\&quot;rU\&quot;&quot;,&quot;)&quot;,&quot;as&quot;,&quot;uF&quot;,&quot;:&quot;,&quot;first&quot;,&quot;=&quot;,&quot;uF&quot;,&quot;.&quot;,&quot;next&quot;,&quot;(&quot;,&quot;)&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;\&quot;\\t\&quot;&quot;,&quot;)&quot;,&quot;lines&quot;,&quot;=&quot;,&quot;[&quot;,&quot;line&quot;,&quot;.&quot;,&quot;strip&quot;,&quot;(&quot;,&quot;)&quot;,&quot;for&quot;,&quot;line&quot;,&quot;in&quot;,&quot;uF&quot;,&quot;]&quot;,&quot;unifrac&quot;,&quot;=&quot;,&quot;{&quot;,&quot;\&quot;pcd\&quot;&quot;,&quot;:&quot;,&quot;OrderedDict&quot;,&quot;(&quot;,&quot;)&quot;,&quot;,&quot;,&quot;\&quot;eigvals\&quot;&quot;,&quot;:&quot;,&quot;[&quot;,&quot;]&quot;,&quot;,&quot;,&quot;\&quot;varexp\&quot;&quot;,&quot;:&quot;,&quot;[&quot;,&quot;]&quot;,&quot;}&quot;,&quot;if&quot;,&quot;first&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;==&quot;,&quot;\&quot;pc vector number\&quot;&quot;,&quot;:&quot;,&quot;return&quot;,&quot;parse_unifrac_v1_8&quot;,&quot;(&quot;,&quot;unifrac&quot;,&quot;,&quot;,&quot;lines&quot;,&quot;)&quot;,&quot;elif&quot;,&quot;first&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;==&quot;,&quot;\&quot;Eigvals\&quot;&quot;,&quot;:&quot;,&quot;return&quot;,&quot;parse_unifrac_v1_9&quot;,&quot;(&quot;,&quot;unifrac&quot;,&quot;,&quot;,&quot;lines&quot;,&quot;)&quot;,&quot;else&quot;,&quot;:&quot;,&quot;raise&quot;,&quot;ValueError&quot;,&quot;(&quot;,&quot;\&quot;File format not supported/recognized. Please check input \&quot;&quot;,&quot;\&quot;unifrac file.\&quot;&quot;,&quot;)&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;parse_unifrac\&quot;,\n  \&quot;(\&quot;,\n  \&quot;unifracFN\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;with\&quot;,\n  \&quot;open\&quot;,\n  \&quot;(\&quot;,\n  \&quot;unifracFN\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;rU\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;as\&quot;,\n  \&quot;uF\&quot;,\n  \&quot;:\&quot;,\n  \&quot;first\&quot;,\n  \&quot;=\&quot;,\n  \&quot;uF\&quot;,\n  \&quot;.\&quot;,\n  \&quot;next\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;\\\\t\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;lines\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;line\&quot;,\n  \&quot;.\&quot;,\n  \&quot;strip\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;line\&quot;,\n  \&quot;in\&quot;,\n  \&quot;uF\&quot;,\n  \&quot;]\&quot;,\n  \&quot;unifrac\&quot;,\n  \&quot;=\&quot;,\n  \&quot;{\&quot;,\n  \&quot;\\\&quot;pcd\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;OrderedDict\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;eigvals\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;[\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;varexp\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;[\&quot;,\n  \&quot;]\&quot;,\n  \&quot;}\&quot;,\n  \&quot;if\&quot;,\n  \&quot;first\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;==\&quot;,\n  \&quot;\\\&quot;pc vector number\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;parse_unifrac_v1_8\&quot;,\n  \&quot;(\&quot;,\n  \&quot;unifrac\&quot;,\n  \&quot;,\&quot;,\n  \&quot;lines\&quot;,\n  \&quot;)\&quot;,\n  \&quot;elif\&quot;,\n  \&quot;first\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;==\&quot;,\n  \&quot;\\\&quot;Eigvals\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;parse_unifrac_v1_9\&quot;,\n  \&quot;(\&quot;,\n  \&quot;unifrac\&quot;,\n  \&quot;,\&quot;,\n  \&quot;lines\&quot;,\n  \&quot;)\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;raise\&quot;,\n  \&quot;ValueError\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;File format not supported/recognized. Please check input \\\&quot;\&quot;,\n  \&quot;\\\&quot;unifrac file.\\\&quot;\&quot;,\n  \&quot;)\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Parses the unifrac results file into a dictionary\n\n    :type unifracFN: str\n    :param unifracFN: The path to the unifrac results file\n\n    :rtype: dict\n    :return: A dictionary with keys: 'pcd' (principle coordinates data) which is a\n             dictionary of the data keyed by sample ID, 'eigvals' (eigenvalues), and\n             'varexp' (variation explained)&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Parses&quot;,&quot;the&quot;,&quot;unifrac&quot;,&quot;results&quot;,&quot;file&quot;,&quot;into&quot;,&quot;a&quot;,&quot;dictionary&quot;],&quot;string&quot;:&quot;[\n  \&quot;Parses\&quot;,\n  \&quot;the\&quot;,\n  \&quot;unifrac\&quot;,\n  \&quot;results\&quot;,\n  \&quot;file\&quot;,\n  \&quot;into\&quot;,\n  \&quot;a\&quot;,\n  \&quot;dictionary\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/util.py#L311-L334&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:4,&quot;string&quot;:&quot;4&quot;}}},{&quot;rowIdx&quot;:5,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;phylotoast/util.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;parse_unifrac_v1_8&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def parse_unifrac_v1_8(unifrac, file_data):\n    \&quot;\&quot;\&quot;\n    Function to parse data from older version of unifrac file obtained from Qiime version\n    1.8 and earlier.\n\n    :type unifrac: dict\n    :param unifracFN: The path to the unifrac results file\n\n    :type file_data: list\n    :param file_data: Unifrac data lines after stripping whitespace characters.\n    \&quot;\&quot;\&quot;\n    for line in file_data:\n        if line == \&quot;\&quot;:\n            break\n        line = line.split(\&quot;\\t\&quot;)\n        unifrac[\&quot;pcd\&quot;][line[0]] = [float(e) for e in line[1:]]\n\n    unifrac[\&quot;eigvals\&quot;] = [float(entry) for entry in file_data[-2].split(\&quot;\\t\&quot;)[1:]]\n    unifrac[\&quot;varexp\&quot;] = [float(entry) for entry in file_data[-1].split(\&quot;\\t\&quot;)[1:]]\n    return unifrac&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def parse_unifrac_v1_8(unifrac, file_data):\n    \&quot;\&quot;\&quot;\n    Function to parse data from older version of unifrac file obtained from Qiime version\n    1.8 and earlier.\n\n    :type unifrac: dict\n    :param unifracFN: The path to the unifrac results file\n\n    :type file_data: list\n    :param file_data: Unifrac data lines after stripping whitespace characters.\n    \&quot;\&quot;\&quot;\n    for line in file_data:\n        if line == \&quot;\&quot;:\n            break\n        line = line.split(\&quot;\\t\&quot;)\n        unifrac[\&quot;pcd\&quot;][line[0]] = [float(e) for e in line[1:]]\n\n    unifrac[\&quot;eigvals\&quot;] = [float(entry) for entry in file_data[-2].split(\&quot;\\t\&quot;)[1:]]\n    unifrac[\&quot;varexp\&quot;] = [float(entry) for entry in file_data[-1].split(\&quot;\\t\&quot;)[1:]]\n    return unifrac&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;parse_unifrac_v1_8&quot;,&quot;(&quot;,&quot;unifrac&quot;,&quot;,&quot;,&quot;file_data&quot;,&quot;)&quot;,&quot;:&quot;,&quot;for&quot;,&quot;line&quot;,&quot;in&quot;,&quot;file_data&quot;,&quot;:&quot;,&quot;if&quot;,&quot;line&quot;,&quot;==&quot;,&quot;\&quot;\&quot;&quot;,&quot;:&quot;,&quot;break&quot;,&quot;line&quot;,&quot;=&quot;,&quot;line&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;\&quot;\\t\&quot;&quot;,&quot;)&quot;,&quot;unifrac&quot;,&quot;[&quot;,&quot;\&quot;pcd\&quot;&quot;,&quot;]&quot;,&quot;[&quot;,&quot;line&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;]&quot;,&quot;=&quot;,&quot;[&quot;,&quot;float&quot;,&quot;(&quot;,&quot;e&quot;,&quot;)&quot;,&quot;for&quot;,&quot;e&quot;,&quot;in&quot;,&quot;line&quot;,&quot;[&quot;,&quot;1&quot;,&quot;:&quot;,&quot;]&quot;,&quot;]&quot;,&quot;unifrac&quot;,&quot;[&quot;,&quot;\&quot;eigvals\&quot;&quot;,&quot;]&quot;,&quot;=&quot;,&quot;[&quot;,&quot;float&quot;,&quot;(&quot;,&quot;entry&quot;,&quot;)&quot;,&quot;for&quot;,&quot;entry&quot;,&quot;in&quot;,&quot;file_data&quot;,&quot;[&quot;,&quot;-&quot;,&quot;2&quot;,&quot;]&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;\&quot;\\t\&quot;&quot;,&quot;)&quot;,&quot;[&quot;,&quot;1&quot;,&quot;:&quot;,&quot;]&quot;,&quot;]&quot;,&quot;unifrac&quot;,&quot;[&quot;,&quot;\&quot;varexp\&quot;&quot;,&quot;]&quot;,&quot;=&quot;,&quot;[&quot;,&quot;float&quot;,&quot;(&quot;,&quot;entry&quot;,&quot;)&quot;,&quot;for&quot;,&quot;entry&quot;,&quot;in&quot;,&quot;file_data&quot;,&quot;[&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;\&quot;\\t\&quot;&quot;,&quot;)&quot;,&quot;[&quot;,&quot;1&quot;,&quot;:&quot;,&quot;]&quot;,&quot;]&quot;,&quot;return&quot;,&quot;unifrac&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;parse_unifrac_v1_8\&quot;,\n  \&quot;(\&quot;,\n  \&quot;unifrac\&quot;,\n  \&quot;,\&quot;,\n  \&quot;file_data\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;for\&quot;,\n  \&quot;line\&quot;,\n  \&quot;in\&quot;,\n  \&quot;file_data\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;line\&quot;,\n  \&quot;==\&quot;,\n  \&quot;\\\&quot;\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;break\&quot;,\n  \&quot;line\&quot;,\n  \&quot;=\&quot;,\n  \&quot;line\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;\\\\t\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;unifrac\&quot;,\n  \&quot;[\&quot;,\n  \&quot;\\\&quot;pcd\\\&quot;\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;float\&quot;,\n  \&quot;(\&quot;,\n  \&quot;e\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;e\&quot;,\n  \&quot;in\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;:\&quot;,\n  \&quot;]\&quot;,\n  \&quot;]\&quot;,\n  \&quot;unifrac\&quot;,\n  \&quot;[\&quot;,\n  \&quot;\\\&quot;eigvals\\\&quot;\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;float\&quot;,\n  \&quot;(\&quot;,\n  \&quot;entry\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;entry\&quot;,\n  \&quot;in\&quot;,\n  \&quot;file_data\&quot;,\n  \&quot;[\&quot;,\n  \&quot;-\&quot;,\n  \&quot;2\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;\\\\t\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;:\&quot;,\n  \&quot;]\&quot;,\n  \&quot;]\&quot;,\n  \&quot;unifrac\&quot;,\n  \&quot;[\&quot;,\n  \&quot;\\\&quot;varexp\\\&quot;\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;float\&quot;,\n  \&quot;(\&quot;,\n  \&quot;entry\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;entry\&quot;,\n  \&quot;in\&quot;,\n  \&quot;file_data\&quot;,\n  \&quot;[\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;\\\\t\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;:\&quot;,\n  \&quot;]\&quot;,\n  \&quot;]\&quot;,\n  \&quot;return\&quot;,\n  \&quot;unifrac\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Function to parse data from older version of unifrac file obtained from Qiime version\n    1.8 and earlier.\n\n    :type unifrac: dict\n    :param unifracFN: The path to the unifrac results file\n\n    :type file_data: list\n    :param file_data: Unifrac data lines after stripping whitespace characters.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Function&quot;,&quot;to&quot;,&quot;parse&quot;,&quot;data&quot;,&quot;from&quot;,&quot;older&quot;,&quot;version&quot;,&quot;of&quot;,&quot;unifrac&quot;,&quot;file&quot;,&quot;obtained&quot;,&quot;from&quot;,&quot;Qiime&quot;,&quot;version&quot;,&quot;1&quot;,&quot;.&quot;,&quot;8&quot;,&quot;and&quot;,&quot;earlier&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Function\&quot;,\n  \&quot;to\&quot;,\n  \&quot;parse\&quot;,\n  \&quot;data\&quot;,\n  \&quot;from\&quot;,\n  \&quot;older\&quot;,\n  \&quot;version\&quot;,\n  \&quot;of\&quot;,\n  \&quot;unifrac\&quot;,\n  \&quot;file\&quot;,\n  \&quot;obtained\&quot;,\n  \&quot;from\&quot;,\n  \&quot;Qiime\&quot;,\n  \&quot;version\&quot;,\n  \&quot;1\&quot;,\n  \&quot;.\&quot;,\n  \&quot;8\&quot;,\n  \&quot;and\&quot;,\n  \&quot;earlier\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/util.py#L337-L356&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:5,&quot;string&quot;:&quot;5&quot;}}},{&quot;rowIdx&quot;:6,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;phylotoast/util.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;parse_unifrac_v1_9&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def parse_unifrac_v1_9(unifrac, file_data):\n    \&quot;\&quot;\&quot;\n    Function to parse data from newer version of unifrac file obtained from Qiime version\n    1.9 and later.\n\n    :type unifracFN: str\n    :param unifracFN: The path to the unifrac results file\n\n    :type file_data: list\n    :param file_data: Unifrac data lines after stripping whitespace characters.\n    \&quot;\&quot;\&quot;\n    unifrac[\&quot;eigvals\&quot;] = [float(entry) for entry in file_data[0].split(\&quot;\\t\&quot;)]\n    unifrac[\&quot;varexp\&quot;] = [float(entry)*100 for entry in file_data[3].split(\&quot;\\t\&quot;)]\n\n    for line in file_data[8:]:\n        if line == \&quot;\&quot;:\n            break\n        line = line.split(\&quot;\\t\&quot;)\n        unifrac[\&quot;pcd\&quot;][line[0]] = [float(e) for e in line[1:]]\n    return unifrac&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def parse_unifrac_v1_9(unifrac, file_data):\n    \&quot;\&quot;\&quot;\n    Function to parse data from newer version of unifrac file obtained from Qiime version\n    1.9 and later.\n\n    :type unifracFN: str\n    :param unifracFN: The path to the unifrac results file\n\n    :type file_data: list\n    :param file_data: Unifrac data lines after stripping whitespace characters.\n    \&quot;\&quot;\&quot;\n    unifrac[\&quot;eigvals\&quot;] = [float(entry) for entry in file_data[0].split(\&quot;\\t\&quot;)]\n    unifrac[\&quot;varexp\&quot;] = [float(entry)*100 for entry in file_data[3].split(\&quot;\\t\&quot;)]\n\n    for line in file_data[8:]:\n        if line == \&quot;\&quot;:\n            break\n        line = line.split(\&quot;\\t\&quot;)\n        unifrac[\&quot;pcd\&quot;][line[0]] = [float(e) for e in line[1:]]\n    return unifrac&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;parse_unifrac_v1_9&quot;,&quot;(&quot;,&quot;unifrac&quot;,&quot;,&quot;,&quot;file_data&quot;,&quot;)&quot;,&quot;:&quot;,&quot;unifrac&quot;,&quot;[&quot;,&quot;\&quot;eigvals\&quot;&quot;,&quot;]&quot;,&quot;=&quot;,&quot;[&quot;,&quot;float&quot;,&quot;(&quot;,&quot;entry&quot;,&quot;)&quot;,&quot;for&quot;,&quot;entry&quot;,&quot;in&quot;,&quot;file_data&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;\&quot;\\t\&quot;&quot;,&quot;)&quot;,&quot;]&quot;,&quot;unifrac&quot;,&quot;[&quot;,&quot;\&quot;varexp\&quot;&quot;,&quot;]&quot;,&quot;=&quot;,&quot;[&quot;,&quot;float&quot;,&quot;(&quot;,&quot;entry&quot;,&quot;)&quot;,&quot;*&quot;,&quot;100&quot;,&quot;for&quot;,&quot;entry&quot;,&quot;in&quot;,&quot;file_data&quot;,&quot;[&quot;,&quot;3&quot;,&quot;]&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;\&quot;\\t\&quot;&quot;,&quot;)&quot;,&quot;]&quot;,&quot;for&quot;,&quot;line&quot;,&quot;in&quot;,&quot;file_data&quot;,&quot;[&quot;,&quot;8&quot;,&quot;:&quot;,&quot;]&quot;,&quot;:&quot;,&quot;if&quot;,&quot;line&quot;,&quot;==&quot;,&quot;\&quot;\&quot;&quot;,&quot;:&quot;,&quot;break&quot;,&quot;line&quot;,&quot;=&quot;,&quot;line&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;\&quot;\\t\&quot;&quot;,&quot;)&quot;,&quot;unifrac&quot;,&quot;[&quot;,&quot;\&quot;pcd\&quot;&quot;,&quot;]&quot;,&quot;[&quot;,&quot;line&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;]&quot;,&quot;=&quot;,&quot;[&quot;,&quot;float&quot;,&quot;(&quot;,&quot;e&quot;,&quot;)&quot;,&quot;for&quot;,&quot;e&quot;,&quot;in&quot;,&quot;line&quot;,&quot;[&quot;,&quot;1&quot;,&quot;:&quot;,&quot;]&quot;,&quot;]&quot;,&quot;return&quot;,&quot;unifrac&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;parse_unifrac_v1_9\&quot;,\n  \&quot;(\&quot;,\n  \&quot;unifrac\&quot;,\n  \&quot;,\&quot;,\n  \&quot;file_data\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;unifrac\&quot;,\n  \&quot;[\&quot;,\n  \&quot;\\\&quot;eigvals\\\&quot;\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;float\&quot;,\n  \&quot;(\&quot;,\n  \&quot;entry\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;entry\&quot;,\n  \&quot;in\&quot;,\n  \&quot;file_data\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;\\\\t\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;]\&quot;,\n  \&quot;unifrac\&quot;,\n  \&quot;[\&quot;,\n  \&quot;\\\&quot;varexp\\\&quot;\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;float\&quot;,\n  \&quot;(\&quot;,\n  \&quot;entry\&quot;,\n  \&quot;)\&quot;,\n  \&quot;*\&quot;,\n  \&quot;100\&quot;,\n  \&quot;for\&quot;,\n  \&quot;entry\&quot;,\n  \&quot;in\&quot;,\n  \&quot;file_data\&quot;,\n  \&quot;[\&quot;,\n  \&quot;3\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;\\\\t\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;]\&quot;,\n  \&quot;for\&quot;,\n  \&quot;line\&quot;,\n  \&quot;in\&quot;,\n  \&quot;file_data\&quot;,\n  \&quot;[\&quot;,\n  \&quot;8\&quot;,\n  \&quot;:\&quot;,\n  \&quot;]\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;line\&quot;,\n  \&quot;==\&quot;,\n  \&quot;\\\&quot;\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;break\&quot;,\n  \&quot;line\&quot;,\n  \&quot;=\&quot;,\n  \&quot;line\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;\\\\t\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;unifrac\&quot;,\n  \&quot;[\&quot;,\n  \&quot;\\\&quot;pcd\\\&quot;\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;float\&quot;,\n  \&quot;(\&quot;,\n  \&quot;e\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;e\&quot;,\n  \&quot;in\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;:\&quot;,\n  \&quot;]\&quot;,\n  \&quot;]\&quot;,\n  \&quot;return\&quot;,\n  \&quot;unifrac\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Function to parse data from newer version of unifrac file obtained from Qiime version\n    1.9 and later.\n\n    :type unifracFN: str\n    :param unifracFN: The path to the unifrac results file\n\n    :type file_data: list\n    :param file_data: Unifrac data lines after stripping whitespace characters.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Function&quot;,&quot;to&quot;,&quot;parse&quot;,&quot;data&quot;,&quot;from&quot;,&quot;newer&quot;,&quot;version&quot;,&quot;of&quot;,&quot;unifrac&quot;,&quot;file&quot;,&quot;obtained&quot;,&quot;from&quot;,&quot;Qiime&quot;,&quot;version&quot;,&quot;1&quot;,&quot;.&quot;,&quot;9&quot;,&quot;and&quot;,&quot;later&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Function\&quot;,\n  \&quot;to\&quot;,\n  \&quot;parse\&quot;,\n  \&quot;data\&quot;,\n  \&quot;from\&quot;,\n  \&quot;newer\&quot;,\n  \&quot;version\&quot;,\n  \&quot;of\&quot;,\n  \&quot;unifrac\&quot;,\n  \&quot;file\&quot;,\n  \&quot;obtained\&quot;,\n  \&quot;from\&quot;,\n  \&quot;Qiime\&quot;,\n  \&quot;version\&quot;,\n  \&quot;1\&quot;,\n  \&quot;.\&quot;,\n  \&quot;9\&quot;,\n  \&quot;and\&quot;,\n  \&quot;later\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/util.py#L359-L378&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:6,&quot;string&quot;:&quot;6&quot;}}},{&quot;rowIdx&quot;:7,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;phylotoast/util.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;color_mapping&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def color_mapping(sample_map, header, group_column, color_column=None):\n    \&quot;\&quot;\&quot;\n    Determine color-category mapping. If color_column was specified, then map the category\n    names to color values. Otherwise, use the palettable colors to automatically generate\n    a set of colors for the group values.\n\n    :type sample_map: dict\n    :param unifracFN: Map associating each line of the mapping file with the appropriate\n                      sample ID (each value of the map also contains the sample ID)\n\n    :type header: tuple\n    :param A tuple of header line for mapping file\n\n    :type group_column: str\n    :param group_column: String denoting the column name for sample groups.\n\n    :type color_column: str\n    :param color_column: String denoting the column name for sample colors.\n\n    :type return: dict\n    :param return: {SampleID: Color}\n    \&quot;\&quot;\&quot;\n    group_colors = OrderedDict()\n    group_gather = gather_categories(sample_map, header, [group_column])\n\n    if color_column is not None:\n        color_gather = gather_categories(sample_map, header, [color_column])\n        # match sample IDs between color_gather and group_gather\n        for group in group_gather:\n            for color in color_gather:\n                # allow incomplete assignment of colors, if group sids overlap at\n                # all with the color sids, consider it a match\n                if group_gather[group].sids.intersection(color_gather[color].sids):\n                    group_colors[group] = color\n    else:\n        bcolors = itertools.cycle(Set3_12.hex_colors)\n        for group in group_gather:\n            group_colors[group] = bcolors.next()\n\n    return group_colors&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def color_mapping(sample_map, header, group_column, color_column=None):\n    \&quot;\&quot;\&quot;\n    Determine color-category mapping. If color_column was specified, then map the category\n    names to color values. Otherwise, use the palettable colors to automatically generate\n    a set of colors for the group values.\n\n    :type sample_map: dict\n    :param unifracFN: Map associating each line of the mapping file with the appropriate\n                      sample ID (each value of the map also contains the sample ID)\n\n    :type header: tuple\n    :param A tuple of header line for mapping file\n\n    :type group_column: str\n    :param group_column: String denoting the column name for sample groups.\n\n    :type color_column: str\n    :param color_column: String denoting the column name for sample colors.\n\n    :type return: dict\n    :param return: {SampleID: Color}\n    \&quot;\&quot;\&quot;\n    group_colors = OrderedDict()\n    group_gather = gather_categories(sample_map, header, [group_column])\n\n    if color_column is not None:\n        color_gather = gather_categories(sample_map, header, [color_column])\n        # match sample IDs between color_gather and group_gather\n        for group in group_gather:\n            for color in color_gather:\n                # allow incomplete assignment of colors, if group sids overlap at\n                # all with the color sids, consider it a match\n                if group_gather[group].sids.intersection(color_gather[color].sids):\n                    group_colors[group] = color\n    else:\n        bcolors = itertools.cycle(Set3_12.hex_colors)\n        for group in group_gather:\n            group_colors[group] = bcolors.next()\n\n    return group_colors&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;color_mapping&quot;,&quot;(&quot;,&quot;sample_map&quot;,&quot;,&quot;,&quot;header&quot;,&quot;,&quot;,&quot;group_column&quot;,&quot;,&quot;,&quot;color_column&quot;,&quot;=&quot;,&quot;None&quot;,&quot;)&quot;,&quot;:&quot;,&quot;group_colors&quot;,&quot;=&quot;,&quot;OrderedDict&quot;,&quot;(&quot;,&quot;)&quot;,&quot;group_gather&quot;,&quot;=&quot;,&quot;gather_categories&quot;,&quot;(&quot;,&quot;sample_map&quot;,&quot;,&quot;,&quot;header&quot;,&quot;,&quot;,&quot;[&quot;,&quot;group_column&quot;,&quot;]&quot;,&quot;)&quot;,&quot;if&quot;,&quot;color_column&quot;,&quot;is&quot;,&quot;not&quot;,&quot;None&quot;,&quot;:&quot;,&quot;color_gather&quot;,&quot;=&quot;,&quot;gather_categories&quot;,&quot;(&quot;,&quot;sample_map&quot;,&quot;,&quot;,&quot;header&quot;,&quot;,&quot;,&quot;[&quot;,&quot;color_column&quot;,&quot;]&quot;,&quot;)&quot;,&quot;# match sample IDs between color_gather and group_gather&quot;,&quot;for&quot;,&quot;group&quot;,&quot;in&quot;,&quot;group_gather&quot;,&quot;:&quot;,&quot;for&quot;,&quot;color&quot;,&quot;in&quot;,&quot;color_gather&quot;,&quot;:&quot;,&quot;# allow incomplete assignment of colors, if group sids overlap at&quot;,&quot;# all with the color sids, consider it a match&quot;,&quot;if&quot;,&quot;group_gather&quot;,&quot;[&quot;,&quot;group&quot;,&quot;]&quot;,&quot;.&quot;,&quot;sids&quot;,&quot;.&quot;,&quot;intersection&quot;,&quot;(&quot;,&quot;color_gather&quot;,&quot;[&quot;,&quot;color&quot;,&quot;]&quot;,&quot;.&quot;,&quot;sids&quot;,&quot;)&quot;,&quot;:&quot;,&quot;group_colors&quot;,&quot;[&quot;,&quot;group&quot;,&quot;]&quot;,&quot;=&quot;,&quot;color&quot;,&quot;else&quot;,&quot;:&quot;,&quot;bcolors&quot;,&quot;=&quot;,&quot;itertools&quot;,&quot;.&quot;,&quot;cycle&quot;,&quot;(&quot;,&quot;Set3_12&quot;,&quot;.&quot;,&quot;hex_colors&quot;,&quot;)&quot;,&quot;for&quot;,&quot;group&quot;,&quot;in&quot;,&quot;group_gather&quot;,&quot;:&quot;,&quot;group_colors&quot;,&quot;[&quot;,&quot;group&quot;,&quot;]&quot;,&quot;=&quot;,&quot;bcolors&quot;,&quot;.&quot;,&quot;next&quot;,&quot;(&quot;,&quot;)&quot;,&quot;return&quot;,&quot;group_colors&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;color_mapping\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sample_map\&quot;,\n  \&quot;,\&quot;,\n  \&quot;header\&quot;,\n  \&quot;,\&quot;,\n  \&quot;group_column\&quot;,\n  \&quot;,\&quot;,\n  \&quot;color_column\&quot;,\n  \&quot;=\&quot;,\n  \&quot;None\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;group_colors\&quot;,\n  \&quot;=\&quot;,\n  \&quot;OrderedDict\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;group_gather\&quot;,\n  \&quot;=\&quot;,\n  \&quot;gather_categories\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sample_map\&quot;,\n  \&quot;,\&quot;,\n  \&quot;header\&quot;,\n  \&quot;,\&quot;,\n  \&quot;[\&quot;,\n  \&quot;group_column\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;color_column\&quot;,\n  \&quot;is\&quot;,\n  \&quot;not\&quot;,\n  \&quot;None\&quot;,\n  \&quot;:\&quot;,\n  \&quot;color_gather\&quot;,\n  \&quot;=\&quot;,\n  \&quot;gather_categories\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sample_map\&quot;,\n  \&quot;,\&quot;,\n  \&quot;header\&quot;,\n  \&quot;,\&quot;,\n  \&quot;[\&quot;,\n  \&quot;color_column\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;# match sample IDs between color_gather and group_gather\&quot;,\n  \&quot;for\&quot;,\n  \&quot;group\&quot;,\n  \&quot;in\&quot;,\n  \&quot;group_gather\&quot;,\n  \&quot;:\&quot;,\n  \&quot;for\&quot;,\n  \&quot;color\&quot;,\n  \&quot;in\&quot;,\n  \&quot;color_gather\&quot;,\n  \&quot;:\&quot;,\n  \&quot;# allow incomplete assignment of colors, if group sids overlap at\&quot;,\n  \&quot;# all with the color sids, consider it a match\&quot;,\n  \&quot;if\&quot;,\n  \&quot;group_gather\&quot;,\n  \&quot;[\&quot;,\n  \&quot;group\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;sids\&quot;,\n  \&quot;.\&quot;,\n  \&quot;intersection\&quot;,\n  \&quot;(\&quot;,\n  \&quot;color_gather\&quot;,\n  \&quot;[\&quot;,\n  \&quot;color\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;sids\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;group_colors\&quot;,\n  \&quot;[\&quot;,\n  \&quot;group\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;color\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;bcolors\&quot;,\n  \&quot;=\&quot;,\n  \&quot;itertools\&quot;,\n  \&quot;.\&quot;,\n  \&quot;cycle\&quot;,\n  \&quot;(\&quot;,\n  \&quot;Set3_12\&quot;,\n  \&quot;.\&quot;,\n  \&quot;hex_colors\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;group\&quot;,\n  \&quot;in\&quot;,\n  \&quot;group_gather\&quot;,\n  \&quot;:\&quot;,\n  \&quot;group_colors\&quot;,\n  \&quot;[\&quot;,\n  \&quot;group\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;bcolors\&quot;,\n  \&quot;.\&quot;,\n  \&quot;next\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;group_colors\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Determine color-category mapping. If color_column was specified, then map the category\n    names to color values. Otherwise, use the palettable colors to automatically generate\n    a set of colors for the group values.\n\n    :type sample_map: dict\n    :param unifracFN: Map associating each line of the mapping file with the appropriate\n                      sample ID (each value of the map also contains the sample ID)\n\n    :type header: tuple\n    :param A tuple of header line for mapping file\n\n    :type group_column: str\n    :param group_column: String denoting the column name for sample groups.\n\n    :type color_column: str\n    :param color_column: String denoting the column name for sample colors.\n\n    :type return: dict\n    :param return: {SampleID: Color}&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Determine&quot;,&quot;color&quot;,&quot;-&quot;,&quot;category&quot;,&quot;mapping&quot;,&quot;.&quot;,&quot;If&quot;,&quot;color_column&quot;,&quot;was&quot;,&quot;specified&quot;,&quot;then&quot;,&quot;map&quot;,&quot;the&quot;,&quot;category&quot;,&quot;names&quot;,&quot;to&quot;,&quot;color&quot;,&quot;values&quot;,&quot;.&quot;,&quot;Otherwise&quot;,&quot;use&quot;,&quot;the&quot;,&quot;palettable&quot;,&quot;colors&quot;,&quot;to&quot;,&quot;automatically&quot;,&quot;generate&quot;,&quot;a&quot;,&quot;set&quot;,&quot;of&quot;,&quot;colors&quot;,&quot;for&quot;,&quot;the&quot;,&quot;group&quot;,&quot;values&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Determine\&quot;,\n  \&quot;color\&quot;,\n  \&quot;-\&quot;,\n  \&quot;category\&quot;,\n  \&quot;mapping\&quot;,\n  \&quot;.\&quot;,\n  \&quot;If\&quot;,\n  \&quot;color_column\&quot;,\n  \&quot;was\&quot;,\n  \&quot;specified\&quot;,\n  \&quot;then\&quot;,\n  \&quot;map\&quot;,\n  \&quot;the\&quot;,\n  \&quot;category\&quot;,\n  \&quot;names\&quot;,\n  \&quot;to\&quot;,\n  \&quot;color\&quot;,\n  \&quot;values\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Otherwise\&quot;,\n  \&quot;use\&quot;,\n  \&quot;the\&quot;,\n  \&quot;palettable\&quot;,\n  \&quot;colors\&quot;,\n  \&quot;to\&quot;,\n  \&quot;automatically\&quot;,\n  \&quot;generate\&quot;,\n  \&quot;a\&quot;,\n  \&quot;set\&quot;,\n  \&quot;of\&quot;,\n  \&quot;colors\&quot;,\n  \&quot;for\&quot;,\n  \&quot;the\&quot;,\n  \&quot;group\&quot;,\n  \&quot;values\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/util.py#L380-L419&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:7,&quot;string&quot;:&quot;7&quot;}}},{&quot;rowIdx&quot;:8,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;christophertbrown/bioscripts&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ctbBio/shuffle_genome.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;rev_c&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def rev_c(read):\n    \&quot;\&quot;\&quot;\n    return reverse completment of read\n    \&quot;\&quot;\&quot;\n    rc = []\n    rc_nucs = {'A':'T', 'T':'A', 'G':'C', 'C':'G', 'N':'N'}\n    for base in read:\n        rc.extend(rc_nucs[base.upper()])\n    return rc[::-1]&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def rev_c(read):\n    \&quot;\&quot;\&quot;\n    return reverse completment of read\n    \&quot;\&quot;\&quot;\n    rc = []\n    rc_nucs = {'A':'T', 'T':'A', 'G':'C', 'C':'G', 'N':'N'}\n    for base in read:\n        rc.extend(rc_nucs[base.upper()])\n    return rc[::-1]&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;rev_c&quot;,&quot;(&quot;,&quot;read&quot;,&quot;)&quot;,&quot;:&quot;,&quot;rc&quot;,&quot;=&quot;,&quot;[&quot;,&quot;]&quot;,&quot;rc_nucs&quot;,&quot;=&quot;,&quot;{&quot;,&quot;'A'&quot;,&quot;:&quot;,&quot;'T'&quot;,&quot;,&quot;,&quot;'T'&quot;,&quot;:&quot;,&quot;'A'&quot;,&quot;,&quot;,&quot;'G'&quot;,&quot;:&quot;,&quot;'C'&quot;,&quot;,&quot;,&quot;'C'&quot;,&quot;:&quot;,&quot;'G'&quot;,&quot;,&quot;,&quot;'N'&quot;,&quot;:&quot;,&quot;'N'&quot;,&quot;}&quot;,&quot;for&quot;,&quot;base&quot;,&quot;in&quot;,&quot;read&quot;,&quot;:&quot;,&quot;rc&quot;,&quot;.&quot;,&quot;extend&quot;,&quot;(&quot;,&quot;rc_nucs&quot;,&quot;[&quot;,&quot;base&quot;,&quot;.&quot;,&quot;upper&quot;,&quot;(&quot;,&quot;)&quot;,&quot;]&quot;,&quot;)&quot;,&quot;return&quot;,&quot;rc&quot;,&quot;[&quot;,&quot;:&quot;,&quot;:&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;rev_c\&quot;,\n  \&quot;(\&quot;,\n  \&quot;read\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;rc\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;]\&quot;,\n  \&quot;rc_nucs\&quot;,\n  \&quot;=\&quot;,\n  \&quot;{\&quot;,\n  \&quot;'A'\&quot;,\n  \&quot;:\&quot;,\n  \&quot;'T'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'T'\&quot;,\n  \&quot;:\&quot;,\n  \&quot;'A'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'G'\&quot;,\n  \&quot;:\&quot;,\n  \&quot;'C'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'C'\&quot;,\n  \&quot;:\&quot;,\n  \&quot;'G'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'N'\&quot;,\n  \&quot;:\&quot;,\n  \&quot;'N'\&quot;,\n  \&quot;}\&quot;,\n  \&quot;for\&quot;,\n  \&quot;base\&quot;,\n  \&quot;in\&quot;,\n  \&quot;read\&quot;,\n  \&quot;:\&quot;,\n  \&quot;rc\&quot;,\n  \&quot;.\&quot;,\n  \&quot;extend\&quot;,\n  \&quot;(\&quot;,\n  \&quot;rc_nucs\&quot;,\n  \&quot;[\&quot;,\n  \&quot;base\&quot;,\n  \&quot;.\&quot;,\n  \&quot;upper\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;rc\&quot;,\n  \&quot;[\&quot;,\n  \&quot;:\&quot;,\n  \&quot;:\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;return reverse completment of read&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;return&quot;,&quot;reverse&quot;,&quot;completment&quot;,&quot;of&quot;,&quot;read&quot;],&quot;string&quot;:&quot;[\n  \&quot;return\&quot;,\n  \&quot;reverse\&quot;,\n  \&quot;completment\&quot;,\n  \&quot;of\&quot;,\n  \&quot;read\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;83b2566b3a5745437ec651cd6cafddd056846240&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/shuffle_genome.py#L27-L35&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:8,&quot;string&quot;:&quot;8&quot;}}},{&quot;rowIdx&quot;:9,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;christophertbrown/bioscripts&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ctbBio/shuffle_genome.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;shuffle_genome&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def shuffle_genome(genome, cat, fraction = float(100), plot = True, \\\n        alpha = 0.1, beta = 100000, \\\n        min_length = 1000, max_length = 200000):\n    \&quot;\&quot;\&quot;\n    randomly shuffle genome\n    \&quot;\&quot;\&quot;\n    header = '>randomized_%s' % (genome.name)\n    sequence = list(''.join([i[1] for i in parse_fasta(genome)]))\n    length = len(sequence)\n    shuffled = []\n    # break genome into pieces\n    while sequence is not False:\n        s = int(random.gammavariate(alpha, beta))\n        if s <= min_length or s >= max_length:\n            continue\n        if len(sequence) < s:\n            seq = sequence[0:]\n        else:\n            seq = sequence[0:s]\n        sequence = sequence[s:]\n#        if bool(random.getrandbits(1)) is True:\n#            seq = rev_c(seq)\n#            print('fragment length: %s reverse complement: True' % ('{:,}'.format(s)), file=sys.stderr)\n#        else:\n#            print('fragment length: %s reverse complement: False' % ('{:,}'.format(s)), file=sys.stderr)\n        shuffled.append(''.join(seq))\n        if sequence == []:\n            break\n    # shuffle pieces\n    random.shuffle(shuffled)\n    # subset fragments\n    if fraction == float(100):\n        subset = shuffled\n    else:\n        max_pieces = int(length * fraction/100)\n        subset, total = [], 0\n        for fragment in shuffled:\n            length = len(fragment)\n            if total + length <= max_pieces:\n                subset.append(fragment)\n                total += length\n            else:\n                diff = max_pieces - total\n                subset.append(fragment[0:diff])\n                break\n    # combine sequences, if requested\n    if cat is True:\n        yield [header, ''.join(subset)]\n    else:\n        for i, seq in enumerate(subset):\n            yield ['%s fragment:%s' % (header, i), seq]&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def shuffle_genome(genome, cat, fraction = float(100), plot = True, \\\n        alpha = 0.1, beta = 100000, \\\n        min_length = 1000, max_length = 200000):\n    \&quot;\&quot;\&quot;\n    randomly shuffle genome\n    \&quot;\&quot;\&quot;\n    header = '>randomized_%s' % (genome.name)\n    sequence = list(''.join([i[1] for i in parse_fasta(genome)]))\n    length = len(sequence)\n    shuffled = []\n    # break genome into pieces\n    while sequence is not False:\n        s = int(random.gammavariate(alpha, beta))\n        if s <= min_length or s >= max_length:\n            continue\n        if len(sequence) < s:\n            seq = sequence[0:]\n        else:\n            seq = sequence[0:s]\n        sequence = sequence[s:]\n#        if bool(random.getrandbits(1)) is True:\n#            seq = rev_c(seq)\n#            print('fragment length: %s reverse complement: True' % ('{:,}'.format(s)), file=sys.stderr)\n#        else:\n#            print('fragment length: %s reverse complement: False' % ('{:,}'.format(s)), file=sys.stderr)\n        shuffled.append(''.join(seq))\n        if sequence == []:\n            break\n    # shuffle pieces\n    random.shuffle(shuffled)\n    # subset fragments\n    if fraction == float(100):\n        subset = shuffled\n    else:\n        max_pieces = int(length * fraction/100)\n        subset, total = [], 0\n        for fragment in shuffled:\n            length = len(fragment)\n            if total + length <= max_pieces:\n                subset.append(fragment)\n                total += length\n            else:\n                diff = max_pieces - total\n                subset.append(fragment[0:diff])\n                break\n    # combine sequences, if requested\n    if cat is True:\n        yield [header, ''.join(subset)]\n    else:\n        for i, seq in enumerate(subset):\n            yield ['%s fragment:%s' % (header, i), seq]&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;shuffle_genome&quot;,&quot;(&quot;,&quot;genome&quot;,&quot;,&quot;,&quot;cat&quot;,&quot;,&quot;,&quot;fraction&quot;,&quot;=&quot;,&quot;float&quot;,&quot;(&quot;,&quot;100&quot;,&quot;)&quot;,&quot;,&quot;,&quot;plot&quot;,&quot;=&quot;,&quot;True&quot;,&quot;,&quot;,&quot;alpha&quot;,&quot;=&quot;,&quot;0.1&quot;,&quot;,&quot;,&quot;beta&quot;,&quot;=&quot;,&quot;100000&quot;,&quot;,&quot;,&quot;min_length&quot;,&quot;=&quot;,&quot;1000&quot;,&quot;,&quot;,&quot;max_length&quot;,&quot;=&quot;,&quot;200000&quot;,&quot;)&quot;,&quot;:&quot;,&quot;header&quot;,&quot;=&quot;,&quot;'>randomized_%s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;genome&quot;,&quot;.&quot;,&quot;name&quot;,&quot;)&quot;,&quot;sequence&quot;,&quot;=&quot;,&quot;list&quot;,&quot;(&quot;,&quot;''&quot;,&quot;.&quot;,&quot;join&quot;,&quot;(&quot;,&quot;[&quot;,&quot;i&quot;,&quot;[&quot;,&quot;1&quot;,&quot;]&quot;,&quot;for&quot;,&quot;i&quot;,&quot;in&quot;,&quot;parse_fasta&quot;,&quot;(&quot;,&quot;genome&quot;,&quot;)&quot;,&quot;]&quot;,&quot;)&quot;,&quot;)&quot;,&quot;length&quot;,&quot;=&quot;,&quot;len&quot;,&quot;(&quot;,&quot;sequence&quot;,&quot;)&quot;,&quot;shuffled&quot;,&quot;=&quot;,&quot;[&quot;,&quot;]&quot;,&quot;# break genome into pieces&quot;,&quot;while&quot;,&quot;sequence&quot;,&quot;is&quot;,&quot;not&quot;,&quot;False&quot;,&quot;:&quot;,&quot;s&quot;,&quot;=&quot;,&quot;int&quot;,&quot;(&quot;,&quot;random&quot;,&quot;.&quot;,&quot;gammavariate&quot;,&quot;(&quot;,&quot;alpha&quot;,&quot;,&quot;,&quot;beta&quot;,&quot;)&quot;,&quot;)&quot;,&quot;if&quot;,&quot;s&quot;,&quot;<=&quot;,&quot;min_length&quot;,&quot;or&quot;,&quot;s&quot;,&quot;>=&quot;,&quot;max_length&quot;,&quot;:&quot;,&quot;continue&quot;,&quot;if&quot;,&quot;len&quot;,&quot;(&quot;,&quot;sequence&quot;,&quot;)&quot;,&quot;<&quot;,&quot;s&quot;,&quot;:&quot;,&quot;seq&quot;,&quot;=&quot;,&quot;sequence&quot;,&quot;[&quot;,&quot;0&quot;,&quot;:&quot;,&quot;]&quot;,&quot;else&quot;,&quot;:&quot;,&quot;seq&quot;,&quot;=&quot;,&quot;sequence&quot;,&quot;[&quot;,&quot;0&quot;,&quot;:&quot;,&quot;s&quot;,&quot;]&quot;,&quot;sequence&quot;,&quot;=&quot;,&quot;sequence&quot;,&quot;[&quot;,&quot;s&quot;,&quot;:&quot;,&quot;]&quot;,&quot;#        if bool(random.getrandbits(1)) is True:&quot;,&quot;#            seq = rev_c(seq)&quot;,&quot;#            print('fragment length: %s reverse complement: True' % ('{:,}'.format(s)), file=sys.stderr)&quot;,&quot;#        else:&quot;,&quot;#            print('fragment length: %s reverse complement: False' % ('{:,}'.format(s)), file=sys.stderr)&quot;,&quot;shuffled&quot;,&quot;.&quot;,&quot;append&quot;,&quot;(&quot;,&quot;''&quot;,&quot;.&quot;,&quot;join&quot;,&quot;(&quot;,&quot;seq&quot;,&quot;)&quot;,&quot;)&quot;,&quot;if&quot;,&quot;sequence&quot;,&quot;==&quot;,&quot;[&quot;,&quot;]&quot;,&quot;:&quot;,&quot;break&quot;,&quot;# shuffle pieces&quot;,&quot;random&quot;,&quot;.&quot;,&quot;shuffle&quot;,&quot;(&quot;,&quot;shuffled&quot;,&quot;)&quot;,&quot;# subset fragments&quot;,&quot;if&quot;,&quot;fraction&quot;,&quot;==&quot;,&quot;float&quot;,&quot;(&quot;,&quot;100&quot;,&quot;)&quot;,&quot;:&quot;,&quot;subset&quot;,&quot;=&quot;,&quot;shuffled&quot;,&quot;else&quot;,&quot;:&quot;,&quot;max_pieces&quot;,&quot;=&quot;,&quot;int&quot;,&quot;(&quot;,&quot;length&quot;,&quot;*&quot;,&quot;fraction&quot;,&quot;/&quot;,&quot;100&quot;,&quot;)&quot;,&quot;subset&quot;,&quot;,&quot;,&quot;total&quot;,&quot;=&quot;,&quot;[&quot;,&quot;]&quot;,&quot;,&quot;,&quot;0&quot;,&quot;for&quot;,&quot;fragment&quot;,&quot;in&quot;,&quot;shuffled&quot;,&quot;:&quot;,&quot;length&quot;,&quot;=&quot;,&quot;len&quot;,&quot;(&quot;,&quot;fragment&quot;,&quot;)&quot;,&quot;if&quot;,&quot;total&quot;,&quot;+&quot;,&quot;length&quot;,&quot;<=&quot;,&quot;max_pieces&quot;,&quot;:&quot;,&quot;subset&quot;,&quot;.&quot;,&quot;append&quot;,&quot;(&quot;,&quot;fragment&quot;,&quot;)&quot;,&quot;total&quot;,&quot;+=&quot;,&quot;length&quot;,&quot;else&quot;,&quot;:&quot;,&quot;diff&quot;,&quot;=&quot;,&quot;max_pieces&quot;,&quot;-&quot;,&quot;total&quot;,&quot;subset&quot;,&quot;.&quot;,&quot;append&quot;,&quot;(&quot;,&quot;fragment&quot;,&quot;[&quot;,&quot;0&quot;,&quot;:&quot;,&quot;diff&quot;,&quot;]&quot;,&quot;)&quot;,&quot;break&quot;,&quot;# combine sequences, if requested&quot;,&quot;if&quot;,&quot;cat&quot;,&quot;is&quot;,&quot;True&quot;,&quot;:&quot;,&quot;yield&quot;,&quot;[&quot;,&quot;header&quot;,&quot;,&quot;,&quot;''&quot;,&quot;.&quot;,&quot;join&quot;,&quot;(&quot;,&quot;subset&quot;,&quot;)&quot;,&quot;]&quot;,&quot;else&quot;,&quot;:&quot;,&quot;for&quot;,&quot;i&quot;,&quot;,&quot;,&quot;seq&quot;,&quot;in&quot;,&quot;enumerate&quot;,&quot;(&quot;,&quot;subset&quot;,&quot;)&quot;,&quot;:&quot;,&quot;yield&quot;,&quot;[&quot;,&quot;'%s fragment:%s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;header&quot;,&quot;,&quot;,&quot;i&quot;,&quot;)&quot;,&quot;,&quot;,&quot;seq&quot;,&quot;]&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;shuffle_genome\&quot;,\n  \&quot;(\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;,\&quot;,\n  \&quot;cat\&quot;,\n  \&quot;,\&quot;,\n  \&quot;fraction\&quot;,\n  \&quot;=\&quot;,\n  \&quot;float\&quot;,\n  \&quot;(\&quot;,\n  \&quot;100\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;plot\&quot;,\n  \&quot;=\&quot;,\n  \&quot;True\&quot;,\n  \&quot;,\&quot;,\n  \&quot;alpha\&quot;,\n  \&quot;=\&quot;,\n  \&quot;0.1\&quot;,\n  \&quot;,\&quot;,\n  \&quot;beta\&quot;,\n  \&quot;=\&quot;,\n  \&quot;100000\&quot;,\n  \&quot;,\&quot;,\n  \&quot;min_length\&quot;,\n  \&quot;=\&quot;,\n  \&quot;1000\&quot;,\n  \&quot;,\&quot;,\n  \&quot;max_length\&quot;,\n  \&quot;=\&quot;,\n  \&quot;200000\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;header\&quot;,\n  \&quot;=\&quot;,\n  \&quot;'>randomized_%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;.\&quot;,\n  \&quot;name\&quot;,\n  \&quot;)\&quot;,\n  \&quot;sequence\&quot;,\n  \&quot;=\&quot;,\n  \&quot;list\&quot;,\n  \&quot;(\&quot;,\n  \&quot;''\&quot;,\n  \&quot;.\&quot;,\n  \&quot;join\&quot;,\n  \&quot;(\&quot;,\n  \&quot;[\&quot;,\n  \&quot;i\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;in\&quot;,\n  \&quot;parse_fasta\&quot;,\n  \&quot;(\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;)\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;length\&quot;,\n  \&quot;=\&quot;,\n  \&quot;len\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sequence\&quot;,\n  \&quot;)\&quot;,\n  \&quot;shuffled\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;]\&quot;,\n  \&quot;# break genome into pieces\&quot;,\n  \&quot;while\&quot;,\n  \&quot;sequence\&quot;,\n  \&quot;is\&quot;,\n  \&quot;not\&quot;,\n  \&quot;False\&quot;,\n  \&quot;:\&quot;,\n  \&quot;s\&quot;,\n  \&quot;=\&quot;,\n  \&quot;int\&quot;,\n  \&quot;(\&quot;,\n  \&quot;random\&quot;,\n  \&quot;.\&quot;,\n  \&quot;gammavariate\&quot;,\n  \&quot;(\&quot;,\n  \&quot;alpha\&quot;,\n  \&quot;,\&quot;,\n  \&quot;beta\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;s\&quot;,\n  \&quot;<=\&quot;,\n  \&quot;min_length\&quot;,\n  \&quot;or\&quot;,\n  \&quot;s\&quot;,\n  \&quot;>=\&quot;,\n  \&quot;max_length\&quot;,\n  \&quot;:\&quot;,\n  \&quot;continue\&quot;,\n  \&quot;if\&quot;,\n  \&quot;len\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sequence\&quot;,\n  \&quot;)\&quot;,\n  \&quot;<\&quot;,\n  \&quot;s\&quot;,\n  \&quot;:\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sequence\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;:\&quot;,\n  \&quot;]\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sequence\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;:\&quot;,\n  \&quot;s\&quot;,\n  \&quot;]\&quot;,\n  \&quot;sequence\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sequence\&quot;,\n  \&quot;[\&quot;,\n  \&quot;s\&quot;,\n  \&quot;:\&quot;,\n  \&quot;]\&quot;,\n  \&quot;#        if bool(random.getrandbits(1)) is True:\&quot;,\n  \&quot;#            seq = rev_c(seq)\&quot;,\n  \&quot;#            print('fragment length: %s reverse complement: True' % ('{:,}'.format(s)), file=sys.stderr)\&quot;,\n  \&quot;#        else:\&quot;,\n  \&quot;#            print('fragment length: %s reverse complement: False' % ('{:,}'.format(s)), file=sys.stderr)\&quot;,\n  \&quot;shuffled\&quot;,\n  \&quot;.\&quot;,\n  \&quot;append\&quot;,\n  \&quot;(\&quot;,\n  \&quot;''\&quot;,\n  \&quot;.\&quot;,\n  \&quot;join\&quot;,\n  \&quot;(\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;sequence\&quot;,\n  \&quot;==\&quot;,\n  \&quot;[\&quot;,\n  \&quot;]\&quot;,\n  \&quot;:\&quot;,\n  \&quot;break\&quot;,\n  \&quot;# shuffle pieces\&quot;,\n  \&quot;random\&quot;,\n  \&quot;.\&quot;,\n  \&quot;shuffle\&quot;,\n  \&quot;(\&quot;,\n  \&quot;shuffled\&quot;,\n  \&quot;)\&quot;,\n  \&quot;# subset fragments\&quot;,\n  \&quot;if\&quot;,\n  \&quot;fraction\&quot;,\n  \&quot;==\&quot;,\n  \&quot;float\&quot;,\n  \&quot;(\&quot;,\n  \&quot;100\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;subset\&quot;,\n  \&quot;=\&quot;,\n  \&quot;shuffled\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;max_pieces\&quot;,\n  \&quot;=\&quot;,\n  \&quot;int\&quot;,\n  \&quot;(\&quot;,\n  \&quot;length\&quot;,\n  \&quot;*\&quot;,\n  \&quot;fraction\&quot;,\n  \&quot;/\&quot;,\n  \&quot;100\&quot;,\n  \&quot;)\&quot;,\n  \&quot;subset\&quot;,\n  \&quot;,\&quot;,\n  \&quot;total\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;0\&quot;,\n  \&quot;for\&quot;,\n  \&quot;fragment\&quot;,\n  \&quot;in\&quot;,\n  \&quot;shuffled\&quot;,\n  \&quot;:\&quot;,\n  \&quot;length\&quot;,\n  \&quot;=\&quot;,\n  \&quot;len\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fragment\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;total\&quot;,\n  \&quot;+\&quot;,\n  \&quot;length\&quot;,\n  \&quot;<=\&quot;,\n  \&quot;max_pieces\&quot;,\n  \&quot;:\&quot;,\n  \&quot;subset\&quot;,\n  \&quot;.\&quot;,\n  \&quot;append\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fragment\&quot;,\n  \&quot;)\&quot;,\n  \&quot;total\&quot;,\n  \&quot;+=\&quot;,\n  \&quot;length\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;diff\&quot;,\n  \&quot;=\&quot;,\n  \&quot;max_pieces\&quot;,\n  \&quot;-\&quot;,\n  \&quot;total\&quot;,\n  \&quot;subset\&quot;,\n  \&quot;.\&quot;,\n  \&quot;append\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fragment\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;:\&quot;,\n  \&quot;diff\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;break\&quot;,\n  \&quot;# combine sequences, if requested\&quot;,\n  \&quot;if\&quot;,\n  \&quot;cat\&quot;,\n  \&quot;is\&quot;,\n  \&quot;True\&quot;,\n  \&quot;:\&quot;,\n  \&quot;yield\&quot;,\n  \&quot;[\&quot;,\n  \&quot;header\&quot;,\n  \&quot;,\&quot;,\n  \&quot;''\&quot;,\n  \&quot;.\&quot;,\n  \&quot;join\&quot;,\n  \&quot;(\&quot;,\n  \&quot;subset\&quot;,\n  \&quot;)\&quot;,\n  \&quot;]\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;,\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;in\&quot;,\n  \&quot;enumerate\&quot;,\n  \&quot;(\&quot;,\n  \&quot;subset\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;yield\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'%s fragment:%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;header\&quot;,\n  \&quot;,\&quot;,\n  \&quot;i\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;]\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;randomly shuffle genome&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;randomly&quot;,&quot;shuffle&quot;,&quot;genome&quot;],&quot;string&quot;:&quot;[\n  \&quot;randomly\&quot;,\n  \&quot;shuffle\&quot;,\n  \&quot;genome\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;83b2566b3a5745437ec651cd6cafddd056846240&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/shuffle_genome.py#L37-L87&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:9,&quot;string&quot;:&quot;9&quot;}}},{&quot;rowIdx&quot;:10,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;opengridcc/opengrid&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;opengrid/library/regression.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;MultiVarLinReg._prune&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _prune(self, fit, p_max):\n        \&quot;\&quot;\&quot;\n        If the fit contains statistically insignificant parameters, remove them.\n        Returns a pruned fit where all parameters have p-values of the t-statistic below p_max\n\n        Parameters\n        ----------\n        fit: fm.ols fit object\n            Can contain insignificant parameters\n        p_max : float\n            Maximum allowed probability of the t-statistic\n\n        Returns\n        -------\n        fit: fm.ols fit object\n            Won't contain any insignificant parameters\n\n        \&quot;\&quot;\&quot;\n\n        def remove_from_model_desc(x, model_desc):\n            \&quot;\&quot;\&quot;\n            Return a model_desc without x\n            \&quot;\&quot;\&quot;\n\n            rhs_termlist = []\n            for t in model_desc.rhs_termlist:\n                if not t.factors:\n                    # intercept, add anyway\n                    rhs_termlist.append(t)\n                elif not x == t.factors[0]._varname:\n                    # this is not the term with x\n                    rhs_termlist.append(t)\n\n            md = ModelDesc(model_desc.lhs_termlist, rhs_termlist)\n            return md\n\n        corrected_model_desc = ModelDesc(fit.model.formula.lhs_termlist[:], fit.model.formula.rhs_termlist[:])\n        pars_to_prune = fit.pvalues.where(fit.pvalues > p_max).dropna().index.tolist()\n        try:\n            pars_to_prune.remove('Intercept')\n        except:\n            pass\n        while pars_to_prune:\n            corrected_model_desc = remove_from_model_desc(pars_to_prune[0], corrected_model_desc)\n            fit = fm.ols(corrected_model_desc, data=self.df).fit()\n            pars_to_prune = fit.pvalues.where(fit.pvalues > p_max).dropna().index.tolist()\n            try:\n                pars_to_prune.remove('Intercept')\n            except:\n                pass\n        return fit&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _prune(self, fit, p_max):\n        \&quot;\&quot;\&quot;\n        If the fit contains statistically insignificant parameters, remove them.\n        Returns a pruned fit where all parameters have p-values of the t-statistic below p_max\n\n        Parameters\n        ----------\n        fit: fm.ols fit object\n            Can contain insignificant parameters\n        p_max : float\n            Maximum allowed probability of the t-statistic\n\n        Returns\n        -------\n        fit: fm.ols fit object\n            Won't contain any insignificant parameters\n\n        \&quot;\&quot;\&quot;\n\n        def remove_from_model_desc(x, model_desc):\n            \&quot;\&quot;\&quot;\n            Return a model_desc without x\n            \&quot;\&quot;\&quot;\n\n            rhs_termlist = []\n            for t in model_desc.rhs_termlist:\n                if not t.factors:\n                    # intercept, add anyway\n                    rhs_termlist.append(t)\n                elif not x == t.factors[0]._varname:\n                    # this is not the term with x\n                    rhs_termlist.append(t)\n\n            md = ModelDesc(model_desc.lhs_termlist, rhs_termlist)\n            return md\n\n        corrected_model_desc = ModelDesc(fit.model.formula.lhs_termlist[:], fit.model.formula.rhs_termlist[:])\n        pars_to_prune = fit.pvalues.where(fit.pvalues > p_max).dropna().index.tolist()\n        try:\n            pars_to_prune.remove('Intercept')\n        except:\n            pass\n        while pars_to_prune:\n            corrected_model_desc = remove_from_model_desc(pars_to_prune[0], corrected_model_desc)\n            fit = fm.ols(corrected_model_desc, data=self.df).fit()\n            pars_to_prune = fit.pvalues.where(fit.pvalues > p_max).dropna().index.tolist()\n            try:\n                pars_to_prune.remove('Intercept')\n            except:\n                pass\n        return fit&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;_prune&quot;,&quot;(&quot;,&quot;self&quot;,&quot;,&quot;,&quot;fit&quot;,&quot;,&quot;,&quot;p_max&quot;,&quot;)&quot;,&quot;:&quot;,&quot;def&quot;,&quot;remove_from_model_desc&quot;,&quot;(&quot;,&quot;x&quot;,&quot;,&quot;,&quot;model_desc&quot;,&quot;)&quot;,&quot;:&quot;,&quot;\&quot;\&quot;\&quot;\n            Return a model_desc without x\n            \&quot;\&quot;\&quot;&quot;,&quot;rhs_termlist&quot;,&quot;=&quot;,&quot;[&quot;,&quot;]&quot;,&quot;for&quot;,&quot;t&quot;,&quot;in&quot;,&quot;model_desc&quot;,&quot;.&quot;,&quot;rhs_termlist&quot;,&quot;:&quot;,&quot;if&quot;,&quot;not&quot;,&quot;t&quot;,&quot;.&quot;,&quot;factors&quot;,&quot;:&quot;,&quot;# intercept, add anyway&quot;,&quot;rhs_termlist&quot;,&quot;.&quot;,&quot;append&quot;,&quot;(&quot;,&quot;t&quot;,&quot;)&quot;,&quot;elif&quot;,&quot;not&quot;,&quot;x&quot;,&quot;==&quot;,&quot;t&quot;,&quot;.&quot;,&quot;factors&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;.&quot;,&quot;_varname&quot;,&quot;:&quot;,&quot;# this is not the term with x&quot;,&quot;rhs_termlist&quot;,&quot;.&quot;,&quot;append&quot;,&quot;(&quot;,&quot;t&quot;,&quot;)&quot;,&quot;md&quot;,&quot;=&quot;,&quot;ModelDesc&quot;,&quot;(&quot;,&quot;model_desc&quot;,&quot;.&quot;,&quot;lhs_termlist&quot;,&quot;,&quot;,&quot;rhs_termlist&quot;,&quot;)&quot;,&quot;return&quot;,&quot;md&quot;,&quot;corrected_model_desc&quot;,&quot;=&quot;,&quot;ModelDesc&quot;,&quot;(&quot;,&quot;fit&quot;,&quot;.&quot;,&quot;model&quot;,&quot;.&quot;,&quot;formula&quot;,&quot;.&quot;,&quot;lhs_termlist&quot;,&quot;[&quot;,&quot;:&quot;,&quot;]&quot;,&quot;,&quot;,&quot;fit&quot;,&quot;.&quot;,&quot;model&quot;,&quot;.&quot;,&quot;formula&quot;,&quot;.&quot;,&quot;rhs_termlist&quot;,&quot;[&quot;,&quot;:&quot;,&quot;]&quot;,&quot;)&quot;,&quot;pars_to_prune&quot;,&quot;=&quot;,&quot;fit&quot;,&quot;.&quot;,&quot;pvalues&quot;,&quot;.&quot;,&quot;where&quot;,&quot;(&quot;,&quot;fit&quot;,&quot;.&quot;,&quot;pvalues&quot;,&quot;>&quot;,&quot;p_max&quot;,&quot;)&quot;,&quot;.&quot;,&quot;dropna&quot;,&quot;(&quot;,&quot;)&quot;,&quot;.&quot;,&quot;index&quot;,&quot;.&quot;,&quot;tolist&quot;,&quot;(&quot;,&quot;)&quot;,&quot;try&quot;,&quot;:&quot;,&quot;pars_to_prune&quot;,&quot;.&quot;,&quot;remove&quot;,&quot;(&quot;,&quot;'Intercept'&quot;,&quot;)&quot;,&quot;except&quot;,&quot;:&quot;,&quot;pass&quot;,&quot;while&quot;,&quot;pars_to_prune&quot;,&quot;:&quot;,&quot;corrected_model_desc&quot;,&quot;=&quot;,&quot;remove_from_model_desc&quot;,&quot;(&quot;,&quot;pars_to_prune&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;,&quot;,&quot;corrected_model_desc&quot;,&quot;)&quot;,&quot;fit&quot;,&quot;=&quot;,&quot;fm&quot;,&quot;.&quot;,&quot;ols&quot;,&quot;(&quot;,&quot;corrected_model_desc&quot;,&quot;,&quot;,&quot;data&quot;,&quot;=&quot;,&quot;self&quot;,&quot;.&quot;,&quot;df&quot;,&quot;)&quot;,&quot;.&quot;,&quot;fit&quot;,&quot;(&quot;,&quot;)&quot;,&quot;pars_to_prune&quot;,&quot;=&quot;,&quot;fit&quot;,&quot;.&quot;,&quot;pvalues&quot;,&quot;.&quot;,&quot;where&quot;,&quot;(&quot;,&quot;fit&quot;,&quot;.&quot;,&quot;pvalues&quot;,&quot;>&quot;,&quot;p_max&quot;,&quot;)&quot;,&quot;.&quot;,&quot;dropna&quot;,&quot;(&quot;,&quot;)&quot;,&quot;.&quot;,&quot;index&quot;,&quot;.&quot;,&quot;tolist&quot;,&quot;(&quot;,&quot;)&quot;,&quot;try&quot;,&quot;:&quot;,&quot;pars_to_prune&quot;,&quot;.&quot;,&quot;remove&quot;,&quot;(&quot;,&quot;'Intercept'&quot;,&quot;)&quot;,&quot;except&quot;,&quot;:&quot;,&quot;pass&quot;,&quot;return&quot;,&quot;fit&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;_prune\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;,\&quot;,\n  \&quot;fit\&quot;,\n  \&quot;,\&quot;,\n  \&quot;p_max\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;def\&quot;,\n  \&quot;remove_from_model_desc\&quot;,\n  \&quot;(\&quot;,\n  \&quot;x\&quot;,\n  \&quot;,\&quot;,\n  \&quot;model_desc\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;\\\&quot;\\\&quot;\\\&quot;\\n            Return a model_desc without x\\n            \\\&quot;\\\&quot;\\\&quot;\&quot;,\n  \&quot;rhs_termlist\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;]\&quot;,\n  \&quot;for\&quot;,\n  \&quot;t\&quot;,\n  \&quot;in\&quot;,\n  \&quot;model_desc\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rhs_termlist\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;not\&quot;,\n  \&quot;t\&quot;,\n  \&quot;.\&quot;,\n  \&quot;factors\&quot;,\n  \&quot;:\&quot;,\n  \&quot;# intercept, add anyway\&quot;,\n  \&quot;rhs_termlist\&quot;,\n  \&quot;.\&quot;,\n  \&quot;append\&quot;,\n  \&quot;(\&quot;,\n  \&quot;t\&quot;,\n  \&quot;)\&quot;,\n  \&quot;elif\&quot;,\n  \&quot;not\&quot;,\n  \&quot;x\&quot;,\n  \&quot;==\&quot;,\n  \&quot;t\&quot;,\n  \&quot;.\&quot;,\n  \&quot;factors\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;_varname\&quot;,\n  \&quot;:\&quot;,\n  \&quot;# this is not the term with x\&quot;,\n  \&quot;rhs_termlist\&quot;,\n  \&quot;.\&quot;,\n  \&quot;append\&quot;,\n  \&quot;(\&quot;,\n  \&quot;t\&quot;,\n  \&quot;)\&quot;,\n  \&quot;md\&quot;,\n  \&quot;=\&quot;,\n  \&quot;ModelDesc\&quot;,\n  \&quot;(\&quot;,\n  \&quot;model_desc\&quot;,\n  \&quot;.\&quot;,\n  \&quot;lhs_termlist\&quot;,\n  \&quot;,\&quot;,\n  \&quot;rhs_termlist\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;md\&quot;,\n  \&quot;corrected_model_desc\&quot;,\n  \&quot;=\&quot;,\n  \&quot;ModelDesc\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fit\&quot;,\n  \&quot;.\&quot;,\n  \&quot;model\&quot;,\n  \&quot;.\&quot;,\n  \&quot;formula\&quot;,\n  \&quot;.\&quot;,\n  \&quot;lhs_termlist\&quot;,\n  \&quot;[\&quot;,\n  \&quot;:\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;fit\&quot;,\n  \&quot;.\&quot;,\n  \&quot;model\&quot;,\n  \&quot;.\&quot;,\n  \&quot;formula\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rhs_termlist\&quot;,\n  \&quot;[\&quot;,\n  \&quot;:\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;pars_to_prune\&quot;,\n  \&quot;=\&quot;,\n  \&quot;fit\&quot;,\n  \&quot;.\&quot;,\n  \&quot;pvalues\&quot;,\n  \&quot;.\&quot;,\n  \&quot;where\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fit\&quot;,\n  \&quot;.\&quot;,\n  \&quot;pvalues\&quot;,\n  \&quot;>\&quot;,\n  \&quot;p_max\&quot;,\n  \&quot;)\&quot;,\n  \&quot;.\&quot;,\n  \&quot;dropna\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;.\&quot;,\n  \&quot;index\&quot;,\n  \&quot;.\&quot;,\n  \&quot;tolist\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;try\&quot;,\n  \&quot;:\&quot;,\n  \&quot;pars_to_prune\&quot;,\n  \&quot;.\&quot;,\n  \&quot;remove\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'Intercept'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;except\&quot;,\n  \&quot;:\&quot;,\n  \&quot;pass\&quot;,\n  \&quot;while\&quot;,\n  \&quot;pars_to_prune\&quot;,\n  \&quot;:\&quot;,\n  \&quot;corrected_model_desc\&quot;,\n  \&quot;=\&quot;,\n  \&quot;remove_from_model_desc\&quot;,\n  \&quot;(\&quot;,\n  \&quot;pars_to_prune\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;corrected_model_desc\&quot;,\n  \&quot;)\&quot;,\n  \&quot;fit\&quot;,\n  \&quot;=\&quot;,\n  \&quot;fm\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ols\&quot;,\n  \&quot;(\&quot;,\n  \&quot;corrected_model_desc\&quot;,\n  \&quot;,\&quot;,\n  \&quot;data\&quot;,\n  \&quot;=\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;df\&quot;,\n  \&quot;)\&quot;,\n  \&quot;.\&quot;,\n  \&quot;fit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;pars_to_prune\&quot;,\n  \&quot;=\&quot;,\n  \&quot;fit\&quot;,\n  \&quot;.\&quot;,\n  \&quot;pvalues\&quot;,\n  \&quot;.\&quot;,\n  \&quot;where\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fit\&quot;,\n  \&quot;.\&quot;,\n  \&quot;pvalues\&quot;,\n  \&quot;>\&quot;,\n  \&quot;p_max\&quot;,\n  \&quot;)\&quot;,\n  \&quot;.\&quot;,\n  \&quot;dropna\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;.\&quot;,\n  \&quot;index\&quot;,\n  \&quot;.\&quot;,\n  \&quot;tolist\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;try\&quot;,\n  \&quot;:\&quot;,\n  \&quot;pars_to_prune\&quot;,\n  \&quot;.\&quot;,\n  \&quot;remove\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'Intercept'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;except\&quot;,\n  \&quot;:\&quot;,\n  \&quot;pass\&quot;,\n  \&quot;return\&quot;,\n  \&quot;fit\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;If the fit contains statistically insignificant parameters, remove them.\n        Returns a pruned fit where all parameters have p-values of the t-statistic below p_max\n\n        Parameters\n        ----------\n        fit: fm.ols fit object\n            Can contain insignificant parameters\n        p_max : float\n            Maximum allowed probability of the t-statistic\n\n        Returns\n        -------\n        fit: fm.ols fit object\n            Won't contain any insignificant parameters&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;If&quot;,&quot;the&quot;,&quot;fit&quot;,&quot;contains&quot;,&quot;statistically&quot;,&quot;insignificant&quot;,&quot;parameters&quot;,&quot;remove&quot;,&quot;them&quot;,&quot;.&quot;,&quot;Returns&quot;,&quot;a&quot;,&quot;pruned&quot;,&quot;fit&quot;,&quot;where&quot;,&quot;all&quot;,&quot;parameters&quot;,&quot;have&quot;,&quot;p&quot;,&quot;-&quot;,&quot;values&quot;,&quot;of&quot;,&quot;the&quot;,&quot;t&quot;,&quot;-&quot;,&quot;statistic&quot;,&quot;below&quot;,&quot;p_max&quot;],&quot;string&quot;:&quot;[\n  \&quot;If\&quot;,\n  \&quot;the\&quot;,\n  \&quot;fit\&quot;,\n  \&quot;contains\&quot;,\n  \&quot;statistically\&quot;,\n  \&quot;insignificant\&quot;,\n  \&quot;parameters\&quot;,\n  \&quot;remove\&quot;,\n  \&quot;them\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Returns\&quot;,\n  \&quot;a\&quot;,\n  \&quot;pruned\&quot;,\n  \&quot;fit\&quot;,\n  \&quot;where\&quot;,\n  \&quot;all\&quot;,\n  \&quot;parameters\&quot;,\n  \&quot;have\&quot;,\n  \&quot;p\&quot;,\n  \&quot;-\&quot;,\n  \&quot;values\&quot;,\n  \&quot;of\&quot;,\n  \&quot;the\&quot;,\n  \&quot;t\&quot;,\n  \&quot;-\&quot;,\n  \&quot;statistic\&quot;,\n  \&quot;below\&quot;,\n  \&quot;p_max\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;69b8da3c8fcea9300226c45ef0628cd6d4307651&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/opengridcc/opengrid/blob/69b8da3c8fcea9300226c45ef0628cd6d4307651/opengrid/library/regression.py#L222-L272&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:10,&quot;string&quot;:&quot;10&quot;}}},{&quot;rowIdx&quot;:11,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;opengridcc/opengrid&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;opengrid/library/regression.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;MultiVarLinReg.find_best_rsquared&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def find_best_rsquared(list_of_fits):\n        \&quot;\&quot;\&quot;Return the best fit, based on rsquared\&quot;\&quot;\&quot;\n        res = sorted(list_of_fits, key=lambda x: x.rsquared)\n        return res[-1]&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def find_best_rsquared(list_of_fits):\n        \&quot;\&quot;\&quot;Return the best fit, based on rsquared\&quot;\&quot;\&quot;\n        res = sorted(list_of_fits, key=lambda x: x.rsquared)\n        return res[-1]&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;find_best_rsquared&quot;,&quot;(&quot;,&quot;list_of_fits&quot;,&quot;)&quot;,&quot;:&quot;,&quot;res&quot;,&quot;=&quot;,&quot;sorted&quot;,&quot;(&quot;,&quot;list_of_fits&quot;,&quot;,&quot;,&quot;key&quot;,&quot;=&quot;,&quot;lambda&quot;,&quot;x&quot;,&quot;:&quot;,&quot;x&quot;,&quot;.&quot;,&quot;rsquared&quot;,&quot;)&quot;,&quot;return&quot;,&quot;res&quot;,&quot;[&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;find_best_rsquared\&quot;,\n  \&quot;(\&quot;,\n  \&quot;list_of_fits\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;res\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sorted\&quot;,\n  \&quot;(\&quot;,\n  \&quot;list_of_fits\&quot;,\n  \&quot;,\&quot;,\n  \&quot;key\&quot;,\n  \&quot;=\&quot;,\n  \&quot;lambda\&quot;,\n  \&quot;x\&quot;,\n  \&quot;:\&quot;,\n  \&quot;x\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rsquared\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;res\&quot;,\n  \&quot;[\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Return the best fit, based on rsquared&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Return&quot;,&quot;the&quot;,&quot;best&quot;,&quot;fit&quot;,&quot;based&quot;,&quot;on&quot;,&quot;rsquared&quot;],&quot;string&quot;:&quot;[\n  \&quot;Return\&quot;,\n  \&quot;the\&quot;,\n  \&quot;best\&quot;,\n  \&quot;fit\&quot;,\n  \&quot;based\&quot;,\n  \&quot;on\&quot;,\n  \&quot;rsquared\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;69b8da3c8fcea9300226c45ef0628cd6d4307651&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/opengridcc/opengrid/blob/69b8da3c8fcea9300226c45ef0628cd6d4307651/opengrid/library/regression.py#L275-L278&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:11,&quot;string&quot;:&quot;11&quot;}}},{&quot;rowIdx&quot;:12,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;opengridcc/opengrid&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;opengrid/library/regression.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;MultiVarLinReg._predict&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _predict(self, fit, df):\n        \&quot;\&quot;\&quot;\n        Return a df with predictions and confidence interval\n\n        Notes\n        -----\n        The df will contain the following columns:\n        - 'predicted': the model output\n        - 'interval_u', 'interval_l': upper and lower confidence bounds.\n\n        The result will depend on the following attributes of self:\n        confint : float (default=0.95)\n            Confidence level for two-sided hypothesis\n        allow_negative_predictions : bool (default=True)\n            If False, correct negative predictions to zero (typically for energy consumption predictions)\n\n        Parameters\n        ----------\n        fit : Statsmodels fit\n        df : pandas DataFrame or None (default)\n            If None, use self.df\n\n\n        Returns\n        -------\n        df_res : pandas DataFrame\n            Copy of df with additional columns 'predicted', 'interval_u' and 'interval_l'\n        \&quot;\&quot;\&quot;\n\n        # Add model results to data as column 'predictions'\n        df_res = df.copy()\n        if 'Intercept' in fit.model.exog_names:\n            df_res['Intercept'] = 1.0\n        df_res['predicted'] = fit.predict(df_res)\n        if not self.allow_negative_predictions:\n            df_res.loc[df_res['predicted'] < 0, 'predicted'] = 0\n\n        prstd, interval_l, interval_u = wls_prediction_std(fit,\n                                                           df_res[fit.model.exog_names],\n                                                           alpha=1 - self.confint)\n        df_res['interval_l'] = interval_l\n        df_res['interval_u'] = interval_u\n\n        if 'Intercept' in df_res:\n            df_res.drop(labels=['Intercept'], axis=1, inplace=True)\n\n        return df_res&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _predict(self, fit, df):\n        \&quot;\&quot;\&quot;\n        Return a df with predictions and confidence interval\n\n        Notes\n        -----\n        The df will contain the following columns:\n        - 'predicted': the model output\n        - 'interval_u', 'interval_l': upper and lower confidence bounds.\n\n        The result will depend on the following attributes of self:\n        confint : float (default=0.95)\n            Confidence level for two-sided hypothesis\n        allow_negative_predictions : bool (default=True)\n            If False, correct negative predictions to zero (typically for energy consumption predictions)\n\n        Parameters\n        ----------\n        fit : Statsmodels fit\n        df : pandas DataFrame or None (default)\n            If None, use self.df\n\n\n        Returns\n        -------\n        df_res : pandas DataFrame\n            Copy of df with additional columns 'predicted', 'interval_u' and 'interval_l'\n        \&quot;\&quot;\&quot;\n\n        # Add model results to data as column 'predictions'\n        df_res = df.copy()\n        if 'Intercept' in fit.model.exog_names:\n            df_res['Intercept'] = 1.0\n        df_res['predicted'] = fit.predict(df_res)\n        if not self.allow_negative_predictions:\n            df_res.loc[df_res['predicted'] < 0, 'predicted'] = 0\n\n        prstd, interval_l, interval_u = wls_prediction_std(fit,\n                                                           df_res[fit.model.exog_names],\n                                                           alpha=1 - self.confint)\n        df_res['interval_l'] = interval_l\n        df_res['interval_u'] = interval_u\n\n        if 'Intercept' in df_res:\n            df_res.drop(labels=['Intercept'], axis=1, inplace=True)\n\n        return df_res&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;_predict&quot;,&quot;(&quot;,&quot;self&quot;,&quot;,&quot;,&quot;fit&quot;,&quot;,&quot;,&quot;df&quot;,&quot;)&quot;,&quot;:&quot;,&quot;# Add model results to data as column 'predictions'&quot;,&quot;df_res&quot;,&quot;=&quot;,&quot;df&quot;,&quot;.&quot;,&quot;copy&quot;,&quot;(&quot;,&quot;)&quot;,&quot;if&quot;,&quot;'Intercept'&quot;,&quot;in&quot;,&quot;fit&quot;,&quot;.&quot;,&quot;model&quot;,&quot;.&quot;,&quot;exog_names&quot;,&quot;:&quot;,&quot;df_res&quot;,&quot;[&quot;,&quot;'Intercept'&quot;,&quot;]&quot;,&quot;=&quot;,&quot;1.0&quot;,&quot;df_res&quot;,&quot;[&quot;,&quot;'predicted'&quot;,&quot;]&quot;,&quot;=&quot;,&quot;fit&quot;,&quot;.&quot;,&quot;predict&quot;,&quot;(&quot;,&quot;df_res&quot;,&quot;)&quot;,&quot;if&quot;,&quot;not&quot;,&quot;self&quot;,&quot;.&quot;,&quot;allow_negative_predictions&quot;,&quot;:&quot;,&quot;df_res&quot;,&quot;.&quot;,&quot;loc&quot;,&quot;[&quot;,&quot;df_res&quot;,&quot;[&quot;,&quot;'predicted'&quot;,&quot;]&quot;,&quot;<&quot;,&quot;0&quot;,&quot;,&quot;,&quot;'predicted'&quot;,&quot;]&quot;,&quot;=&quot;,&quot;0&quot;,&quot;prstd&quot;,&quot;,&quot;,&quot;interval_l&quot;,&quot;,&quot;,&quot;interval_u&quot;,&quot;=&quot;,&quot;wls_prediction_std&quot;,&quot;(&quot;,&quot;fit&quot;,&quot;,&quot;,&quot;df_res&quot;,&quot;[&quot;,&quot;fit&quot;,&quot;.&quot;,&quot;model&quot;,&quot;.&quot;,&quot;exog_names&quot;,&quot;]&quot;,&quot;,&quot;,&quot;alpha&quot;,&quot;=&quot;,&quot;1&quot;,&quot;-&quot;,&quot;self&quot;,&quot;.&quot;,&quot;confint&quot;,&quot;)&quot;,&quot;df_res&quot;,&quot;[&quot;,&quot;'interval_l'&quot;,&quot;]&quot;,&quot;=&quot;,&quot;interval_l&quot;,&quot;df_res&quot;,&quot;[&quot;,&quot;'interval_u'&quot;,&quot;]&quot;,&quot;=&quot;,&quot;interval_u&quot;,&quot;if&quot;,&quot;'Intercept'&quot;,&quot;in&quot;,&quot;df_res&quot;,&quot;:&quot;,&quot;df_res&quot;,&quot;.&quot;,&quot;drop&quot;,&quot;(&quot;,&quot;labels&quot;,&quot;=&quot;,&quot;[&quot;,&quot;'Intercept'&quot;,&quot;]&quot;,&quot;,&quot;,&quot;axis&quot;,&quot;=&quot;,&quot;1&quot;,&quot;,&quot;,&quot;inplace&quot;,&quot;=&quot;,&quot;True&quot;,&quot;)&quot;,&quot;return&quot;,&quot;df_res&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;_predict\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;,\&quot;,\n  \&quot;fit\&quot;,\n  \&quot;,\&quot;,\n  \&quot;df\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;# Add model results to data as column 'predictions'\&quot;,\n  \&quot;df_res\&quot;,\n  \&quot;=\&quot;,\n  \&quot;df\&quot;,\n  \&quot;.\&quot;,\n  \&quot;copy\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;'Intercept'\&quot;,\n  \&quot;in\&quot;,\n  \&quot;fit\&quot;,\n  \&quot;.\&quot;,\n  \&quot;model\&quot;,\n  \&quot;.\&quot;,\n  \&quot;exog_names\&quot;,\n  \&quot;:\&quot;,\n  \&quot;df_res\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'Intercept'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;1.0\&quot;,\n  \&quot;df_res\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'predicted'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;fit\&quot;,\n  \&quot;.\&quot;,\n  \&quot;predict\&quot;,\n  \&quot;(\&quot;,\n  \&quot;df_res\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;not\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;allow_negative_predictions\&quot;,\n  \&quot;:\&quot;,\n  \&quot;df_res\&quot;,\n  \&quot;.\&quot;,\n  \&quot;loc\&quot;,\n  \&quot;[\&quot;,\n  \&quot;df_res\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'predicted'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;<\&quot;,\n  \&quot;0\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'predicted'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;0\&quot;,\n  \&quot;prstd\&quot;,\n  \&quot;,\&quot;,\n  \&quot;interval_l\&quot;,\n  \&quot;,\&quot;,\n  \&quot;interval_u\&quot;,\n  \&quot;=\&quot;,\n  \&quot;wls_prediction_std\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fit\&quot;,\n  \&quot;,\&quot;,\n  \&quot;df_res\&quot;,\n  \&quot;[\&quot;,\n  \&quot;fit\&quot;,\n  \&quot;.\&quot;,\n  \&quot;model\&quot;,\n  \&quot;.\&quot;,\n  \&quot;exog_names\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;alpha\&quot;,\n  \&quot;=\&quot;,\n  \&quot;1\&quot;,\n  \&quot;-\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;confint\&quot;,\n  \&quot;)\&quot;,\n  \&quot;df_res\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'interval_l'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;interval_l\&quot;,\n  \&quot;df_res\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'interval_u'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;interval_u\&quot;,\n  \&quot;if\&quot;,\n  \&quot;'Intercept'\&quot;,\n  \&quot;in\&quot;,\n  \&quot;df_res\&quot;,\n  \&quot;:\&quot;,\n  \&quot;df_res\&quot;,\n  \&quot;.\&quot;,\n  \&quot;drop\&quot;,\n  \&quot;(\&quot;,\n  \&quot;labels\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'Intercept'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;axis\&quot;,\n  \&quot;=\&quot;,\n  \&quot;1\&quot;,\n  \&quot;,\&quot;,\n  \&quot;inplace\&quot;,\n  \&quot;=\&quot;,\n  \&quot;True\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;df_res\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Return a df with predictions and confidence interval\n\n        Notes\n        -----\n        The df will contain the following columns:\n        - 'predicted': the model output\n        - 'interval_u', 'interval_l': upper and lower confidence bounds.\n\n        The result will depend on the following attributes of self:\n        confint : float (default=0.95)\n            Confidence level for two-sided hypothesis\n        allow_negative_predictions : bool (default=True)\n            If False, correct negative predictions to zero (typically for energy consumption predictions)\n\n        Parameters\n        ----------\n        fit : Statsmodels fit\n        df : pandas DataFrame or None (default)\n            If None, use self.df\n\n\n        Returns\n        -------\n        df_res : pandas DataFrame\n            Copy of df with additional columns 'predicted', 'interval_u' and 'interval_l'&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Return&quot;,&quot;a&quot;,&quot;df&quot;,&quot;with&quot;,&quot;predictions&quot;,&quot;and&quot;,&quot;confidence&quot;,&quot;interval&quot;],&quot;string&quot;:&quot;[\n  \&quot;Return\&quot;,\n  \&quot;a\&quot;,\n  \&quot;df\&quot;,\n  \&quot;with\&quot;,\n  \&quot;predictions\&quot;,\n  \&quot;and\&quot;,\n  \&quot;confidence\&quot;,\n  \&quot;interval\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;69b8da3c8fcea9300226c45ef0628cd6d4307651&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/opengridcc/opengrid/blob/69b8da3c8fcea9300226c45ef0628cd6d4307651/opengrid/library/regression.py#L292-L338&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:12,&quot;string&quot;:&quot;12&quot;}}},{&quot;rowIdx&quot;:13,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;phylotoast/biom_calc.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;relative_abundance&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def relative_abundance(biomf, sampleIDs=None):\n    \&quot;\&quot;\&quot;\n    Calculate the relative abundance of each OTUID in a Sample.\n\n    :type biomf: A BIOM file.\n    :param biomf: OTU table format.\n\n    :type sampleIDs: list\n    :param sampleIDs: A list of sample id's from BIOM format OTU table.\n\n    :rtype: dict\n    :return: Returns a keyed on SampleIDs, and the values are dictionaries keyed on\n             OTUID's and their values represent the relative abundance of that OTUID in\n             that SampleID.\n    \&quot;\&quot;\&quot;\n    if sampleIDs is None:\n        sampleIDs = biomf.ids()\n    else:\n        try:\n            for sid in sampleIDs:\n                assert sid in biomf.ids()\n        except AssertionError:\n            raise ValueError(\n                \&quot;\\nError while calculating relative abundances: The sampleIDs provided do\&quot;\n                \&quot; not match the sampleIDs in biom file. Please double check the sampleIDs\&quot;\n                \&quot; provided.\\n\&quot;)\n    otuIDs = biomf.ids(axis=\&quot;observation\&quot;)\n    norm_biomf = biomf.norm(inplace=False)\n\n    return {sample: {otuID: norm_biomf.get_value_by_ids(otuID, sample)\n                     for otuID in otuIDs} for sample in sampleIDs}&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def relative_abundance(biomf, sampleIDs=None):\n    \&quot;\&quot;\&quot;\n    Calculate the relative abundance of each OTUID in a Sample.\n\n    :type biomf: A BIOM file.\n    :param biomf: OTU table format.\n\n    :type sampleIDs: list\n    :param sampleIDs: A list of sample id's from BIOM format OTU table.\n\n    :rtype: dict\n    :return: Returns a keyed on SampleIDs, and the values are dictionaries keyed on\n             OTUID's and their values represent the relative abundance of that OTUID in\n             that SampleID.\n    \&quot;\&quot;\&quot;\n    if sampleIDs is None:\n        sampleIDs = biomf.ids()\n    else:\n        try:\n            for sid in sampleIDs:\n                assert sid in biomf.ids()\n        except AssertionError:\n            raise ValueError(\n                \&quot;\\nError while calculating relative abundances: The sampleIDs provided do\&quot;\n                \&quot; not match the sampleIDs in biom file. Please double check the sampleIDs\&quot;\n                \&quot; provided.\\n\&quot;)\n    otuIDs = biomf.ids(axis=\&quot;observation\&quot;)\n    norm_biomf = biomf.norm(inplace=False)\n\n    return {sample: {otuID: norm_biomf.get_value_by_ids(otuID, sample)\n                     for otuID in otuIDs} for sample in sampleIDs}&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;relative_abundance&quot;,&quot;(&quot;,&quot;biomf&quot;,&quot;,&quot;,&quot;sampleIDs&quot;,&quot;=&quot;,&quot;None&quot;,&quot;)&quot;,&quot;:&quot;,&quot;if&quot;,&quot;sampleIDs&quot;,&quot;is&quot;,&quot;None&quot;,&quot;:&quot;,&quot;sampleIDs&quot;,&quot;=&quot;,&quot;biomf&quot;,&quot;.&quot;,&quot;ids&quot;,&quot;(&quot;,&quot;)&quot;,&quot;else&quot;,&quot;:&quot;,&quot;try&quot;,&quot;:&quot;,&quot;for&quot;,&quot;sid&quot;,&quot;in&quot;,&quot;sampleIDs&quot;,&quot;:&quot;,&quot;assert&quot;,&quot;sid&quot;,&quot;in&quot;,&quot;biomf&quot;,&quot;.&quot;,&quot;ids&quot;,&quot;(&quot;,&quot;)&quot;,&quot;except&quot;,&quot;AssertionError&quot;,&quot;:&quot;,&quot;raise&quot;,&quot;ValueError&quot;,&quot;(&quot;,&quot;\&quot;\\nError while calculating relative abundances: The sampleIDs provided do\&quot;&quot;,&quot;\&quot; not match the sampleIDs in biom file. Please double check the sampleIDs\&quot;&quot;,&quot;\&quot; provided.\\n\&quot;&quot;,&quot;)&quot;,&quot;otuIDs&quot;,&quot;=&quot;,&quot;biomf&quot;,&quot;.&quot;,&quot;ids&quot;,&quot;(&quot;,&quot;axis&quot;,&quot;=&quot;,&quot;\&quot;observation\&quot;&quot;,&quot;)&quot;,&quot;norm_biomf&quot;,&quot;=&quot;,&quot;biomf&quot;,&quot;.&quot;,&quot;norm&quot;,&quot;(&quot;,&quot;inplace&quot;,&quot;=&quot;,&quot;False&quot;,&quot;)&quot;,&quot;return&quot;,&quot;{&quot;,&quot;sample&quot;,&quot;:&quot;,&quot;{&quot;,&quot;otuID&quot;,&quot;:&quot;,&quot;norm_biomf&quot;,&quot;.&quot;,&quot;get_value_by_ids&quot;,&quot;(&quot;,&quot;otuID&quot;,&quot;,&quot;,&quot;sample&quot;,&quot;)&quot;,&quot;for&quot;,&quot;otuID&quot;,&quot;in&quot;,&quot;otuIDs&quot;,&quot;}&quot;,&quot;for&quot;,&quot;sample&quot;,&quot;in&quot;,&quot;sampleIDs&quot;,&quot;}&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;relative_abundance\&quot;,\n  \&quot;(\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;,\&quot;,\n  \&quot;sampleIDs\&quot;,\n  \&quot;=\&quot;,\n  \&quot;None\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;sampleIDs\&quot;,\n  \&quot;is\&quot;,\n  \&quot;None\&quot;,\n  \&quot;:\&quot;,\n  \&quot;sampleIDs\&quot;,\n  \&quot;=\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ids\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;try\&quot;,\n  \&quot;:\&quot;,\n  \&quot;for\&quot;,\n  \&quot;sid\&quot;,\n  \&quot;in\&quot;,\n  \&quot;sampleIDs\&quot;,\n  \&quot;:\&quot;,\n  \&quot;assert\&quot;,\n  \&quot;sid\&quot;,\n  \&quot;in\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ids\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;except\&quot;,\n  \&quot;AssertionError\&quot;,\n  \&quot;:\&quot;,\n  \&quot;raise\&quot;,\n  \&quot;ValueError\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;\\\\nError while calculating relative abundances: The sampleIDs provided do\\\&quot;\&quot;,\n  \&quot;\\\&quot; not match the sampleIDs in biom file. Please double check the sampleIDs\\\&quot;\&quot;,\n  \&quot;\\\&quot; provided.\\\\n\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;otuIDs\&quot;,\n  \&quot;=\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ids\&quot;,\n  \&quot;(\&quot;,\n  \&quot;axis\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;observation\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;norm_biomf\&quot;,\n  \&quot;=\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;.\&quot;,\n  \&quot;norm\&quot;,\n  \&quot;(\&quot;,\n  \&quot;inplace\&quot;,\n  \&quot;=\&quot;,\n  \&quot;False\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;{\&quot;,\n  \&quot;sample\&quot;,\n  \&quot;:\&quot;,\n  \&quot;{\&quot;,\n  \&quot;otuID\&quot;,\n  \&quot;:\&quot;,\n  \&quot;norm_biomf\&quot;,\n  \&quot;.\&quot;,\n  \&quot;get_value_by_ids\&quot;,\n  \&quot;(\&quot;,\n  \&quot;otuID\&quot;,\n  \&quot;,\&quot;,\n  \&quot;sample\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;otuID\&quot;,\n  \&quot;in\&quot;,\n  \&quot;otuIDs\&quot;,\n  \&quot;}\&quot;,\n  \&quot;for\&quot;,\n  \&quot;sample\&quot;,\n  \&quot;in\&quot;,\n  \&quot;sampleIDs\&quot;,\n  \&quot;}\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Calculate the relative abundance of each OTUID in a Sample.\n\n    :type biomf: A BIOM file.\n    :param biomf: OTU table format.\n\n    :type sampleIDs: list\n    :param sampleIDs: A list of sample id's from BIOM format OTU table.\n\n    :rtype: dict\n    :return: Returns a keyed on SampleIDs, and the values are dictionaries keyed on\n             OTUID's and their values represent the relative abundance of that OTUID in\n             that SampleID.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Calculate&quot;,&quot;the&quot;,&quot;relative&quot;,&quot;abundance&quot;,&quot;of&quot;,&quot;each&quot;,&quot;OTUID&quot;,&quot;in&quot;,&quot;a&quot;,&quot;Sample&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Calculate\&quot;,\n  \&quot;the\&quot;,\n  \&quot;relative\&quot;,\n  \&quot;abundance\&quot;,\n  \&quot;of\&quot;,\n  \&quot;each\&quot;,\n  \&quot;OTUID\&quot;,\n  \&quot;in\&quot;,\n  \&quot;a\&quot;,\n  \&quot;Sample\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/biom_calc.py#L11-L41&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:13,&quot;string&quot;:&quot;13&quot;}}},{&quot;rowIdx&quot;:14,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;phylotoast/biom_calc.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;mean_otu_pct_abundance&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def mean_otu_pct_abundance(ra, otuIDs):\n    \&quot;\&quot;\&quot;\n    Calculate the mean OTU abundance percentage.\n\n    :type ra: Dict\n    :param ra: 'ra' refers to a dictionary keyed on SampleIDs, and the values are\n               dictionaries keyed on OTUID's and their values represent the relative\n               abundance of that OTUID in that SampleID. 'ra' is the output of\n               relative_abundance() function.\n\n    :type otuIDs: List\n    :param otuIDs: A list of OTUID's for which the percentage abundance needs to be\n                   measured.\n\n    :rtype: dict\n    :return: A dictionary of OTUID and their percent relative abundance as key/value pair.\n    \&quot;\&quot;\&quot;\n    sids = ra.keys()\n    otumeans = defaultdict(int)\n\n    for oid in otuIDs:\n        otumeans[oid] = sum([ra[sid][oid] for sid in sids\n                             if oid in ra[sid]]) / len(sids) * 100\n    return otumeans&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def mean_otu_pct_abundance(ra, otuIDs):\n    \&quot;\&quot;\&quot;\n    Calculate the mean OTU abundance percentage.\n\n    :type ra: Dict\n    :param ra: 'ra' refers to a dictionary keyed on SampleIDs, and the values are\n               dictionaries keyed on OTUID's and their values represent the relative\n               abundance of that OTUID in that SampleID. 'ra' is the output of\n               relative_abundance() function.\n\n    :type otuIDs: List\n    :param otuIDs: A list of OTUID's for which the percentage abundance needs to be\n                   measured.\n\n    :rtype: dict\n    :return: A dictionary of OTUID and their percent relative abundance as key/value pair.\n    \&quot;\&quot;\&quot;\n    sids = ra.keys()\n    otumeans = defaultdict(int)\n\n    for oid in otuIDs:\n        otumeans[oid] = sum([ra[sid][oid] for sid in sids\n                             if oid in ra[sid]]) / len(sids) * 100\n    return otumeans&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;mean_otu_pct_abundance&quot;,&quot;(&quot;,&quot;ra&quot;,&quot;,&quot;,&quot;otuIDs&quot;,&quot;)&quot;,&quot;:&quot;,&quot;sids&quot;,&quot;=&quot;,&quot;ra&quot;,&quot;.&quot;,&quot;keys&quot;,&quot;(&quot;,&quot;)&quot;,&quot;otumeans&quot;,&quot;=&quot;,&quot;defaultdict&quot;,&quot;(&quot;,&quot;int&quot;,&quot;)&quot;,&quot;for&quot;,&quot;oid&quot;,&quot;in&quot;,&quot;otuIDs&quot;,&quot;:&quot;,&quot;otumeans&quot;,&quot;[&quot;,&quot;oid&quot;,&quot;]&quot;,&quot;=&quot;,&quot;sum&quot;,&quot;(&quot;,&quot;[&quot;,&quot;ra&quot;,&quot;[&quot;,&quot;sid&quot;,&quot;]&quot;,&quot;[&quot;,&quot;oid&quot;,&quot;]&quot;,&quot;for&quot;,&quot;sid&quot;,&quot;in&quot;,&quot;sids&quot;,&quot;if&quot;,&quot;oid&quot;,&quot;in&quot;,&quot;ra&quot;,&quot;[&quot;,&quot;sid&quot;,&quot;]&quot;,&quot;]&quot;,&quot;)&quot;,&quot;/&quot;,&quot;len&quot;,&quot;(&quot;,&quot;sids&quot;,&quot;)&quot;,&quot;*&quot;,&quot;100&quot;,&quot;return&quot;,&quot;otumeans&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;mean_otu_pct_abundance\&quot;,\n  \&quot;(\&quot;,\n  \&quot;ra\&quot;,\n  \&quot;,\&quot;,\n  \&quot;otuIDs\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;sids\&quot;,\n  \&quot;=\&quot;,\n  \&quot;ra\&quot;,\n  \&quot;.\&quot;,\n  \&quot;keys\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;otumeans\&quot;,\n  \&quot;=\&quot;,\n  \&quot;defaultdict\&quot;,\n  \&quot;(\&quot;,\n  \&quot;int\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;oid\&quot;,\n  \&quot;in\&quot;,\n  \&quot;otuIDs\&quot;,\n  \&quot;:\&quot;,\n  \&quot;otumeans\&quot;,\n  \&quot;[\&quot;,\n  \&quot;oid\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sum\&quot;,\n  \&quot;(\&quot;,\n  \&quot;[\&quot;,\n  \&quot;ra\&quot;,\n  \&quot;[\&quot;,\n  \&quot;sid\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;oid\&quot;,\n  \&quot;]\&quot;,\n  \&quot;for\&quot;,\n  \&quot;sid\&quot;,\n  \&quot;in\&quot;,\n  \&quot;sids\&quot;,\n  \&quot;if\&quot;,\n  \&quot;oid\&quot;,\n  \&quot;in\&quot;,\n  \&quot;ra\&quot;,\n  \&quot;[\&quot;,\n  \&quot;sid\&quot;,\n  \&quot;]\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;/\&quot;,\n  \&quot;len\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sids\&quot;,\n  \&quot;)\&quot;,\n  \&quot;*\&quot;,\n  \&quot;100\&quot;,\n  \&quot;return\&quot;,\n  \&quot;otumeans\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Calculate the mean OTU abundance percentage.\n\n    :type ra: Dict\n    :param ra: 'ra' refers to a dictionary keyed on SampleIDs, and the values are\n               dictionaries keyed on OTUID's and their values represent the relative\n               abundance of that OTUID in that SampleID. 'ra' is the output of\n               relative_abundance() function.\n\n    :type otuIDs: List\n    :param otuIDs: A list of OTUID's for which the percentage abundance needs to be\n                   measured.\n\n    :rtype: dict\n    :return: A dictionary of OTUID and their percent relative abundance as key/value pair.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Calculate&quot;,&quot;the&quot;,&quot;mean&quot;,&quot;OTU&quot;,&quot;abundance&quot;,&quot;percentage&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Calculate\&quot;,\n  \&quot;the\&quot;,\n  \&quot;mean\&quot;,\n  \&quot;OTU\&quot;,\n  \&quot;abundance\&quot;,\n  \&quot;percentage\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/biom_calc.py#L44-L67&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:14,&quot;string&quot;:&quot;14&quot;}}},{&quot;rowIdx&quot;:15,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;phylotoast/biom_calc.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;MRA&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def MRA(biomf, sampleIDs=None, transform=None):\n    \&quot;\&quot;\&quot;\n    Calculate the mean relative abundance percentage.\n\n    :type biomf: A BIOM file.\n    :param biomf: OTU table format.\n\n    :type sampleIDs: list\n    :param sampleIDs: A list of sample id's from BIOM format OTU table.\n\n    :param transform: Mathematical function which is used to transform smax to another\n                      format. By default, the function has been set to None.\n\n    :rtype: dict\n    :return: A dictionary keyed on OTUID's and their mean relative abundance for a given\n             number of sampleIDs.\n    \&quot;\&quot;\&quot;\n    ra = relative_abundance(biomf, sampleIDs)\n    if transform is not None:\n        ra = {sample: {otuID: transform(abd) for otuID, abd in ra[sample].items()}\n              for sample in ra.keys()}\n    otuIDs = biomf.ids(axis=\&quot;observation\&quot;)\n    return mean_otu_pct_abundance(ra, otuIDs)&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def MRA(biomf, sampleIDs=None, transform=None):\n    \&quot;\&quot;\&quot;\n    Calculate the mean relative abundance percentage.\n\n    :type biomf: A BIOM file.\n    :param biomf: OTU table format.\n\n    :type sampleIDs: list\n    :param sampleIDs: A list of sample id's from BIOM format OTU table.\n\n    :param transform: Mathematical function which is used to transform smax to another\n                      format. By default, the function has been set to None.\n\n    :rtype: dict\n    :return: A dictionary keyed on OTUID's and their mean relative abundance for a given\n             number of sampleIDs.\n    \&quot;\&quot;\&quot;\n    ra = relative_abundance(biomf, sampleIDs)\n    if transform is not None:\n        ra = {sample: {otuID: transform(abd) for otuID, abd in ra[sample].items()}\n              for sample in ra.keys()}\n    otuIDs = biomf.ids(axis=\&quot;observation\&quot;)\n    return mean_otu_pct_abundance(ra, otuIDs)&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;MRA&quot;,&quot;(&quot;,&quot;biomf&quot;,&quot;,&quot;,&quot;sampleIDs&quot;,&quot;=&quot;,&quot;None&quot;,&quot;,&quot;,&quot;transform&quot;,&quot;=&quot;,&quot;None&quot;,&quot;)&quot;,&quot;:&quot;,&quot;ra&quot;,&quot;=&quot;,&quot;relative_abundance&quot;,&quot;(&quot;,&quot;biomf&quot;,&quot;,&quot;,&quot;sampleIDs&quot;,&quot;)&quot;,&quot;if&quot;,&quot;transform&quot;,&quot;is&quot;,&quot;not&quot;,&quot;None&quot;,&quot;:&quot;,&quot;ra&quot;,&quot;=&quot;,&quot;{&quot;,&quot;sample&quot;,&quot;:&quot;,&quot;{&quot;,&quot;otuID&quot;,&quot;:&quot;,&quot;transform&quot;,&quot;(&quot;,&quot;abd&quot;,&quot;)&quot;,&quot;for&quot;,&quot;otuID&quot;,&quot;,&quot;,&quot;abd&quot;,&quot;in&quot;,&quot;ra&quot;,&quot;[&quot;,&quot;sample&quot;,&quot;]&quot;,&quot;.&quot;,&quot;items&quot;,&quot;(&quot;,&quot;)&quot;,&quot;}&quot;,&quot;for&quot;,&quot;sample&quot;,&quot;in&quot;,&quot;ra&quot;,&quot;.&quot;,&quot;keys&quot;,&quot;(&quot;,&quot;)&quot;,&quot;}&quot;,&quot;otuIDs&quot;,&quot;=&quot;,&quot;biomf&quot;,&quot;.&quot;,&quot;ids&quot;,&quot;(&quot;,&quot;axis&quot;,&quot;=&quot;,&quot;\&quot;observation\&quot;&quot;,&quot;)&quot;,&quot;return&quot;,&quot;mean_otu_pct_abundance&quot;,&quot;(&quot;,&quot;ra&quot;,&quot;,&quot;,&quot;otuIDs&quot;,&quot;)&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;MRA\&quot;,\n  \&quot;(\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;,\&quot;,\n  \&quot;sampleIDs\&quot;,\n  \&quot;=\&quot;,\n  \&quot;None\&quot;,\n  \&quot;,\&quot;,\n  \&quot;transform\&quot;,\n  \&quot;=\&quot;,\n  \&quot;None\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;ra\&quot;,\n  \&quot;=\&quot;,\n  \&quot;relative_abundance\&quot;,\n  \&quot;(\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;,\&quot;,\n  \&quot;sampleIDs\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;transform\&quot;,\n  \&quot;is\&quot;,\n  \&quot;not\&quot;,\n  \&quot;None\&quot;,\n  \&quot;:\&quot;,\n  \&quot;ra\&quot;,\n  \&quot;=\&quot;,\n  \&quot;{\&quot;,\n  \&quot;sample\&quot;,\n  \&quot;:\&quot;,\n  \&quot;{\&quot;,\n  \&quot;otuID\&quot;,\n  \&quot;:\&quot;,\n  \&quot;transform\&quot;,\n  \&quot;(\&quot;,\n  \&quot;abd\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;otuID\&quot;,\n  \&quot;,\&quot;,\n  \&quot;abd\&quot;,\n  \&quot;in\&quot;,\n  \&quot;ra\&quot;,\n  \&quot;[\&quot;,\n  \&quot;sample\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;items\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;}\&quot;,\n  \&quot;for\&quot;,\n  \&quot;sample\&quot;,\n  \&quot;in\&quot;,\n  \&quot;ra\&quot;,\n  \&quot;.\&quot;,\n  \&quot;keys\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;}\&quot;,\n  \&quot;otuIDs\&quot;,\n  \&quot;=\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ids\&quot;,\n  \&quot;(\&quot;,\n  \&quot;axis\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;observation\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;mean_otu_pct_abundance\&quot;,\n  \&quot;(\&quot;,\n  \&quot;ra\&quot;,\n  \&quot;,\&quot;,\n  \&quot;otuIDs\&quot;,\n  \&quot;)\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Calculate the mean relative abundance percentage.\n\n    :type biomf: A BIOM file.\n    :param biomf: OTU table format.\n\n    :type sampleIDs: list\n    :param sampleIDs: A list of sample id's from BIOM format OTU table.\n\n    :param transform: Mathematical function which is used to transform smax to another\n                      format. By default, the function has been set to None.\n\n    :rtype: dict\n    :return: A dictionary keyed on OTUID's and their mean relative abundance for a given\n             number of sampleIDs.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Calculate&quot;,&quot;the&quot;,&quot;mean&quot;,&quot;relative&quot;,&quot;abundance&quot;,&quot;percentage&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Calculate\&quot;,\n  \&quot;the\&quot;,\n  \&quot;mean\&quot;,\n  \&quot;relative\&quot;,\n  \&quot;abundance\&quot;,\n  \&quot;percentage\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/biom_calc.py#L70-L92&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:15,&quot;string&quot;:&quot;15&quot;}}},{&quot;rowIdx&quot;:16,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;phylotoast/biom_calc.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;raw_abundance&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def raw_abundance(biomf, sampleIDs=None, sample_abd=True):\n    \&quot;\&quot;\&quot;\n    Calculate the total number of sequences in each OTU or SampleID.\n\n    :type biomf: A BIOM file.\n    :param biomf: OTU table format.\n\n    :type sampleIDs: List\n    :param sampleIDs: A list of column id's from BIOM format OTU table. By default, the\n                      list has been set to None.\n\n    :type sample_abd: Boolean\n    :param sample_abd: A boolean operator to provide output for OTUID's or SampleID's. By\n                       default, the output will be provided for SampleID's.\n\n    :rtype: dict\n    :return: Returns a dictionary keyed on either OTUID's or SampleIDs and their\n             respective abundance as values.\n    \&quot;\&quot;\&quot;\n    results = defaultdict(int)\n    if sampleIDs is None:\n        sampleIDs = biomf.ids()\n    else:\n        try:\n            for sid in sampleIDs:\n                assert sid in biomf.ids()\n        except AssertionError:\n            raise ValueError(\n                \&quot;\\nError while calculating raw total abundances: The sampleIDs provided \&quot;\n                \&quot;do not match the sampleIDs in biom file. Please double check the \&quot;\n                \&quot;sampleIDs provided.\\n\&quot;)\n    otuIDs = biomf.ids(axis=\&quot;observation\&quot;)\n\n    for sampleID in sampleIDs:\n        for otuID in otuIDs:\n            abd = biomf.get_value_by_ids(otuID, sampleID)\n            if sample_abd:\n                results[sampleID] += abd\n            else:\n                results[otuID] += abd\n    return results&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def raw_abundance(biomf, sampleIDs=None, sample_abd=True):\n    \&quot;\&quot;\&quot;\n    Calculate the total number of sequences in each OTU or SampleID.\n\n    :type biomf: A BIOM file.\n    :param biomf: OTU table format.\n\n    :type sampleIDs: List\n    :param sampleIDs: A list of column id's from BIOM format OTU table. By default, the\n                      list has been set to None.\n\n    :type sample_abd: Boolean\n    :param sample_abd: A boolean operator to provide output for OTUID's or SampleID's. By\n                       default, the output will be provided for SampleID's.\n\n    :rtype: dict\n    :return: Returns a dictionary keyed on either OTUID's or SampleIDs and their\n             respective abundance as values.\n    \&quot;\&quot;\&quot;\n    results = defaultdict(int)\n    if sampleIDs is None:\n        sampleIDs = biomf.ids()\n    else:\n        try:\n            for sid in sampleIDs:\n                assert sid in biomf.ids()\n        except AssertionError:\n            raise ValueError(\n                \&quot;\\nError while calculating raw total abundances: The sampleIDs provided \&quot;\n                \&quot;do not match the sampleIDs in biom file. Please double check the \&quot;\n                \&quot;sampleIDs provided.\\n\&quot;)\n    otuIDs = biomf.ids(axis=\&quot;observation\&quot;)\n\n    for sampleID in sampleIDs:\n        for otuID in otuIDs:\n            abd = biomf.get_value_by_ids(otuID, sampleID)\n            if sample_abd:\n                results[sampleID] += abd\n            else:\n                results[otuID] += abd\n    return results&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;raw_abundance&quot;,&quot;(&quot;,&quot;biomf&quot;,&quot;,&quot;,&quot;sampleIDs&quot;,&quot;=&quot;,&quot;None&quot;,&quot;,&quot;,&quot;sample_abd&quot;,&quot;=&quot;,&quot;True&quot;,&quot;)&quot;,&quot;:&quot;,&quot;results&quot;,&quot;=&quot;,&quot;defaultdict&quot;,&quot;(&quot;,&quot;int&quot;,&quot;)&quot;,&quot;if&quot;,&quot;sampleIDs&quot;,&quot;is&quot;,&quot;None&quot;,&quot;:&quot;,&quot;sampleIDs&quot;,&quot;=&quot;,&quot;biomf&quot;,&quot;.&quot;,&quot;ids&quot;,&quot;(&quot;,&quot;)&quot;,&quot;else&quot;,&quot;:&quot;,&quot;try&quot;,&quot;:&quot;,&quot;for&quot;,&quot;sid&quot;,&quot;in&quot;,&quot;sampleIDs&quot;,&quot;:&quot;,&quot;assert&quot;,&quot;sid&quot;,&quot;in&quot;,&quot;biomf&quot;,&quot;.&quot;,&quot;ids&quot;,&quot;(&quot;,&quot;)&quot;,&quot;except&quot;,&quot;AssertionError&quot;,&quot;:&quot;,&quot;raise&quot;,&quot;ValueError&quot;,&quot;(&quot;,&quot;\&quot;\\nError while calculating raw total abundances: The sampleIDs provided \&quot;&quot;,&quot;\&quot;do not match the sampleIDs in biom file. Please double check the \&quot;&quot;,&quot;\&quot;sampleIDs provided.\\n\&quot;&quot;,&quot;)&quot;,&quot;otuIDs&quot;,&quot;=&quot;,&quot;biomf&quot;,&quot;.&quot;,&quot;ids&quot;,&quot;(&quot;,&quot;axis&quot;,&quot;=&quot;,&quot;\&quot;observation\&quot;&quot;,&quot;)&quot;,&quot;for&quot;,&quot;sampleID&quot;,&quot;in&quot;,&quot;sampleIDs&quot;,&quot;:&quot;,&quot;for&quot;,&quot;otuID&quot;,&quot;in&quot;,&quot;otuIDs&quot;,&quot;:&quot;,&quot;abd&quot;,&quot;=&quot;,&quot;biomf&quot;,&quot;.&quot;,&quot;get_value_by_ids&quot;,&quot;(&quot;,&quot;otuID&quot;,&quot;,&quot;,&quot;sampleID&quot;,&quot;)&quot;,&quot;if&quot;,&quot;sample_abd&quot;,&quot;:&quot;,&quot;results&quot;,&quot;[&quot;,&quot;sampleID&quot;,&quot;]&quot;,&quot;+=&quot;,&quot;abd&quot;,&quot;else&quot;,&quot;:&quot;,&quot;results&quot;,&quot;[&quot;,&quot;otuID&quot;,&quot;]&quot;,&quot;+=&quot;,&quot;abd&quot;,&quot;return&quot;,&quot;results&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;raw_abundance\&quot;,\n  \&quot;(\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;,\&quot;,\n  \&quot;sampleIDs\&quot;,\n  \&quot;=\&quot;,\n  \&quot;None\&quot;,\n  \&quot;,\&quot;,\n  \&quot;sample_abd\&quot;,\n  \&quot;=\&quot;,\n  \&quot;True\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;results\&quot;,\n  \&quot;=\&quot;,\n  \&quot;defaultdict\&quot;,\n  \&quot;(\&quot;,\n  \&quot;int\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;sampleIDs\&quot;,\n  \&quot;is\&quot;,\n  \&quot;None\&quot;,\n  \&quot;:\&quot;,\n  \&quot;sampleIDs\&quot;,\n  \&quot;=\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ids\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;try\&quot;,\n  \&quot;:\&quot;,\n  \&quot;for\&quot;,\n  \&quot;sid\&quot;,\n  \&quot;in\&quot;,\n  \&quot;sampleIDs\&quot;,\n  \&quot;:\&quot;,\n  \&quot;assert\&quot;,\n  \&quot;sid\&quot;,\n  \&quot;in\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ids\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;except\&quot;,\n  \&quot;AssertionError\&quot;,\n  \&quot;:\&quot;,\n  \&quot;raise\&quot;,\n  \&quot;ValueError\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;\\\\nError while calculating raw total abundances: The sampleIDs provided \\\&quot;\&quot;,\n  \&quot;\\\&quot;do not match the sampleIDs in biom file. Please double check the \\\&quot;\&quot;,\n  \&quot;\\\&quot;sampleIDs provided.\\\\n\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;otuIDs\&quot;,\n  \&quot;=\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ids\&quot;,\n  \&quot;(\&quot;,\n  \&quot;axis\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;observation\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;sampleID\&quot;,\n  \&quot;in\&quot;,\n  \&quot;sampleIDs\&quot;,\n  \&quot;:\&quot;,\n  \&quot;for\&quot;,\n  \&quot;otuID\&quot;,\n  \&quot;in\&quot;,\n  \&quot;otuIDs\&quot;,\n  \&quot;:\&quot;,\n  \&quot;abd\&quot;,\n  \&quot;=\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;.\&quot;,\n  \&quot;get_value_by_ids\&quot;,\n  \&quot;(\&quot;,\n  \&quot;otuID\&quot;,\n  \&quot;,\&quot;,\n  \&quot;sampleID\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;sample_abd\&quot;,\n  \&quot;:\&quot;,\n  \&quot;results\&quot;,\n  \&quot;[\&quot;,\n  \&quot;sampleID\&quot;,\n  \&quot;]\&quot;,\n  \&quot;+=\&quot;,\n  \&quot;abd\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;results\&quot;,\n  \&quot;[\&quot;,\n  \&quot;otuID\&quot;,\n  \&quot;]\&quot;,\n  \&quot;+=\&quot;,\n  \&quot;abd\&quot;,\n  \&quot;return\&quot;,\n  \&quot;results\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Calculate the total number of sequences in each OTU or SampleID.\n\n    :type biomf: A BIOM file.\n    :param biomf: OTU table format.\n\n    :type sampleIDs: List\n    :param sampleIDs: A list of column id's from BIOM format OTU table. By default, the\n                      list has been set to None.\n\n    :type sample_abd: Boolean\n    :param sample_abd: A boolean operator to provide output for OTUID's or SampleID's. By\n                       default, the output will be provided for SampleID's.\n\n    :rtype: dict\n    :return: Returns a dictionary keyed on either OTUID's or SampleIDs and their\n             respective abundance as values.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Calculate&quot;,&quot;the&quot;,&quot;total&quot;,&quot;number&quot;,&quot;of&quot;,&quot;sequences&quot;,&quot;in&quot;,&quot;each&quot;,&quot;OTU&quot;,&quot;or&quot;,&quot;SampleID&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Calculate\&quot;,\n  \&quot;the\&quot;,\n  \&quot;total\&quot;,\n  \&quot;number\&quot;,\n  \&quot;of\&quot;,\n  \&quot;sequences\&quot;,\n  \&quot;in\&quot;,\n  \&quot;each\&quot;,\n  \&quot;OTU\&quot;,\n  \&quot;or\&quot;,\n  \&quot;SampleID\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/biom_calc.py#L95-L135&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:16,&quot;string&quot;:&quot;16&quot;}}},{&quot;rowIdx&quot;:17,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;phylotoast/biom_calc.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;transform_raw_abundance&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def transform_raw_abundance(biomf, fn=math.log10, sampleIDs=None, sample_abd=True):\n    \&quot;\&quot;\&quot;\n    Function to transform the total abundance calculation for each sample ID to another\n    format based on user given transformation function.\n\n    :type biomf: A BIOM file.\n    :param biomf: OTU table format.\n\n    :param fn: Mathematical function which is used to transform smax to another format.\n               By default, the function has been given as base 10 logarithm.\n\n    :rtype: dict\n    :return: Returns a dictionary similar to output of raw_abundance function but with\n             the abundance values modified by the mathematical operation. By default, the\n             operation performed on the abundances is base 10 logarithm.\n    \&quot;\&quot;\&quot;\n    totals = raw_abundance(biomf, sampleIDs, sample_abd)\n    return {sid: fn(abd) for sid, abd in totals.items()}&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def transform_raw_abundance(biomf, fn=math.log10, sampleIDs=None, sample_abd=True):\n    \&quot;\&quot;\&quot;\n    Function to transform the total abundance calculation for each sample ID to another\n    format based on user given transformation function.\n\n    :type biomf: A BIOM file.\n    :param biomf: OTU table format.\n\n    :param fn: Mathematical function which is used to transform smax to another format.\n               By default, the function has been given as base 10 logarithm.\n\n    :rtype: dict\n    :return: Returns a dictionary similar to output of raw_abundance function but with\n             the abundance values modified by the mathematical operation. By default, the\n             operation performed on the abundances is base 10 logarithm.\n    \&quot;\&quot;\&quot;\n    totals = raw_abundance(biomf, sampleIDs, sample_abd)\n    return {sid: fn(abd) for sid, abd in totals.items()}&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;transform_raw_abundance&quot;,&quot;(&quot;,&quot;biomf&quot;,&quot;,&quot;,&quot;fn&quot;,&quot;=&quot;,&quot;math&quot;,&quot;.&quot;,&quot;log10&quot;,&quot;,&quot;,&quot;sampleIDs&quot;,&quot;=&quot;,&quot;None&quot;,&quot;,&quot;,&quot;sample_abd&quot;,&quot;=&quot;,&quot;True&quot;,&quot;)&quot;,&quot;:&quot;,&quot;totals&quot;,&quot;=&quot;,&quot;raw_abundance&quot;,&quot;(&quot;,&quot;biomf&quot;,&quot;,&quot;,&quot;sampleIDs&quot;,&quot;,&quot;,&quot;sample_abd&quot;,&quot;)&quot;,&quot;return&quot;,&quot;{&quot;,&quot;sid&quot;,&quot;:&quot;,&quot;fn&quot;,&quot;(&quot;,&quot;abd&quot;,&quot;)&quot;,&quot;for&quot;,&quot;sid&quot;,&quot;,&quot;,&quot;abd&quot;,&quot;in&quot;,&quot;totals&quot;,&quot;.&quot;,&quot;items&quot;,&quot;(&quot;,&quot;)&quot;,&quot;}&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;transform_raw_abundance\&quot;,\n  \&quot;(\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;,\&quot;,\n  \&quot;fn\&quot;,\n  \&quot;=\&quot;,\n  \&quot;math\&quot;,\n  \&quot;.\&quot;,\n  \&quot;log10\&quot;,\n  \&quot;,\&quot;,\n  \&quot;sampleIDs\&quot;,\n  \&quot;=\&quot;,\n  \&quot;None\&quot;,\n  \&quot;,\&quot;,\n  \&quot;sample_abd\&quot;,\n  \&quot;=\&quot;,\n  \&quot;True\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;totals\&quot;,\n  \&quot;=\&quot;,\n  \&quot;raw_abundance\&quot;,\n  \&quot;(\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;,\&quot;,\n  \&quot;sampleIDs\&quot;,\n  \&quot;,\&quot;,\n  \&quot;sample_abd\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;{\&quot;,\n  \&quot;sid\&quot;,\n  \&quot;:\&quot;,\n  \&quot;fn\&quot;,\n  \&quot;(\&quot;,\n  \&quot;abd\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;sid\&quot;,\n  \&quot;,\&quot;,\n  \&quot;abd\&quot;,\n  \&quot;in\&quot;,\n  \&quot;totals\&quot;,\n  \&quot;.\&quot;,\n  \&quot;items\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;}\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Function to transform the total abundance calculation for each sample ID to another\n    format based on user given transformation function.\n\n    :type biomf: A BIOM file.\n    :param biomf: OTU table format.\n\n    :param fn: Mathematical function which is used to transform smax to another format.\n               By default, the function has been given as base 10 logarithm.\n\n    :rtype: dict\n    :return: Returns a dictionary similar to output of raw_abundance function but with\n             the abundance values modified by the mathematical operation. By default, the\n             operation performed on the abundances is base 10 logarithm.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Function&quot;,&quot;to&quot;,&quot;transform&quot;,&quot;the&quot;,&quot;total&quot;,&quot;abundance&quot;,&quot;calculation&quot;,&quot;for&quot;,&quot;each&quot;,&quot;sample&quot;,&quot;ID&quot;,&quot;to&quot;,&quot;another&quot;,&quot;format&quot;,&quot;based&quot;,&quot;on&quot;,&quot;user&quot;,&quot;given&quot;,&quot;transformation&quot;,&quot;function&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Function\&quot;,\n  \&quot;to\&quot;,\n  \&quot;transform\&quot;,\n  \&quot;the\&quot;,\n  \&quot;total\&quot;,\n  \&quot;abundance\&quot;,\n  \&quot;calculation\&quot;,\n  \&quot;for\&quot;,\n  \&quot;each\&quot;,\n  \&quot;sample\&quot;,\n  \&quot;ID\&quot;,\n  \&quot;to\&quot;,\n  \&quot;another\&quot;,\n  \&quot;format\&quot;,\n  \&quot;based\&quot;,\n  \&quot;on\&quot;,\n  \&quot;user\&quot;,\n  \&quot;given\&quot;,\n  \&quot;transformation\&quot;,\n  \&quot;function\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/biom_calc.py#L138-L155&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:17,&quot;string&quot;:&quot;17&quot;}}},{&quot;rowIdx&quot;:18,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;bin/diversity.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;print_MannWhitneyU&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def print_MannWhitneyU(div_calc):\n    \&quot;\&quot;\&quot;\n    Compute the Mann-Whitney U test for unequal group sample sizes.\n    \&quot;\&quot;\&quot;\n    try:\n        x = div_calc.values()[0].values()\n        y = div_calc.values()[1].values()\n    except:\n        return \&quot;Error setting up input arrays for Mann-Whitney U Test. Skipping \&quot;\\\n               \&quot;significance testing.\&quot;\n    T, p = stats.mannwhitneyu(x, y)\n    print \&quot;\\nMann-Whitney U test statistic:\&quot;, T\n    print \&quot;Two-tailed p-value: {}\&quot;.format(2 * p)&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def print_MannWhitneyU(div_calc):\n    \&quot;\&quot;\&quot;\n    Compute the Mann-Whitney U test for unequal group sample sizes.\n    \&quot;\&quot;\&quot;\n    try:\n        x = div_calc.values()[0].values()\n        y = div_calc.values()[1].values()\n    except:\n        return \&quot;Error setting up input arrays for Mann-Whitney U Test. Skipping \&quot;\\\n               \&quot;significance testing.\&quot;\n    T, p = stats.mannwhitneyu(x, y)\n    print \&quot;\\nMann-Whitney U test statistic:\&quot;, T\n    print \&quot;Two-tailed p-value: {}\&quot;.format(2 * p)&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;print_MannWhitneyU&quot;,&quot;(&quot;,&quot;div_calc&quot;,&quot;)&quot;,&quot;:&quot;,&quot;try&quot;,&quot;:&quot;,&quot;x&quot;,&quot;=&quot;,&quot;div_calc&quot;,&quot;.&quot;,&quot;values&quot;,&quot;(&quot;,&quot;)&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;.&quot;,&quot;values&quot;,&quot;(&quot;,&quot;)&quot;,&quot;y&quot;,&quot;=&quot;,&quot;div_calc&quot;,&quot;.&quot;,&quot;values&quot;,&quot;(&quot;,&quot;)&quot;,&quot;[&quot;,&quot;1&quot;,&quot;]&quot;,&quot;.&quot;,&quot;values&quot;,&quot;(&quot;,&quot;)&quot;,&quot;except&quot;,&quot;:&quot;,&quot;return&quot;,&quot;\&quot;Error setting up input arrays for Mann-Whitney U Test. Skipping \&quot;&quot;,&quot;\&quot;significance testing.\&quot;&quot;,&quot;T&quot;,&quot;,&quot;,&quot;p&quot;,&quot;=&quot;,&quot;stats&quot;,&quot;.&quot;,&quot;mannwhitneyu&quot;,&quot;(&quot;,&quot;x&quot;,&quot;,&quot;,&quot;y&quot;,&quot;)&quot;,&quot;print&quot;,&quot;\&quot;\\nMann-Whitney U test statistic:\&quot;&quot;,&quot;,&quot;,&quot;T&quot;,&quot;print&quot;,&quot;\&quot;Two-tailed p-value: {}\&quot;&quot;,&quot;.&quot;,&quot;format&quot;,&quot;(&quot;,&quot;2&quot;,&quot;*&quot;,&quot;p&quot;,&quot;)&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;print_MannWhitneyU\&quot;,\n  \&quot;(\&quot;,\n  \&quot;div_calc\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;try\&quot;,\n  \&quot;:\&quot;,\n  \&quot;x\&quot;,\n  \&quot;=\&quot;,\n  \&quot;div_calc\&quot;,\n  \&quot;.\&quot;,\n  \&quot;values\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;values\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;y\&quot;,\n  \&quot;=\&quot;,\n  \&quot;div_calc\&quot;,\n  \&quot;.\&quot;,\n  \&quot;values\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;values\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;except\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;\\\&quot;Error setting up input arrays for Mann-Whitney U Test. Skipping \\\&quot;\&quot;,\n  \&quot;\\\&quot;significance testing.\\\&quot;\&quot;,\n  \&quot;T\&quot;,\n  \&quot;,\&quot;,\n  \&quot;p\&quot;,\n  \&quot;=\&quot;,\n  \&quot;stats\&quot;,\n  \&quot;.\&quot;,\n  \&quot;mannwhitneyu\&quot;,\n  \&quot;(\&quot;,\n  \&quot;x\&quot;,\n  \&quot;,\&quot;,\n  \&quot;y\&quot;,\n  \&quot;)\&quot;,\n  \&quot;print\&quot;,\n  \&quot;\\\&quot;\\\\nMann-Whitney U test statistic:\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;T\&quot;,\n  \&quot;print\&quot;,\n  \&quot;\\\&quot;Two-tailed p-value: {}\\\&quot;\&quot;,\n  \&quot;.\&quot;,\n  \&quot;format\&quot;,\n  \&quot;(\&quot;,\n  \&quot;2\&quot;,\n  \&quot;*\&quot;,\n  \&quot;p\&quot;,\n  \&quot;)\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Compute the Mann-Whitney U test for unequal group sample sizes.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Compute&quot;,&quot;the&quot;,&quot;Mann&quot;,&quot;-&quot;,&quot;Whitney&quot;,&quot;U&quot;,&quot;test&quot;,&quot;for&quot;,&quot;unequal&quot;,&quot;group&quot;,&quot;sample&quot;,&quot;sizes&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Compute\&quot;,\n  \&quot;the\&quot;,\n  \&quot;Mann\&quot;,\n  \&quot;-\&quot;,\n  \&quot;Whitney\&quot;,\n  \&quot;U\&quot;,\n  \&quot;test\&quot;,\n  \&quot;for\&quot;,\n  \&quot;unequal\&quot;,\n  \&quot;group\&quot;,\n  \&quot;sample\&quot;,\n  \&quot;sizes\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/bin/diversity.py#L54-L66&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:18,&quot;string&quot;:&quot;18&quot;}}},{&quot;rowIdx&quot;:19,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;bin/diversity.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;print_KruskalWallisH&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def print_KruskalWallisH(div_calc):\n    \&quot;\&quot;\&quot;\n    Compute the Kruskal-Wallis H-test for independent samples. A typical rule is that\n    each group must have at least 5 measurements.\n    \&quot;\&quot;\&quot;\n    calc = defaultdict(list)\n    try:\n        for k1, v1 in div_calc.iteritems():\n            for k2, v2 in v1.iteritems():\n                calc[k1].append(v2)\n    except:\n        return \&quot;Error setting up input arrays for Kruskal-Wallis H-Test. Skipping \&quot;\\\n               \&quot;significance testing.\&quot;\n    h, p = stats.kruskal(*calc.values())\n    print \&quot;\\nKruskal-Wallis H-test statistic for {} groups: {}\&quot;.format(str(len(div_calc)), h)\n    print \&quot;p-value: {}\&quot;.format(p)&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def print_KruskalWallisH(div_calc):\n    \&quot;\&quot;\&quot;\n    Compute the Kruskal-Wallis H-test for independent samples. A typical rule is that\n    each group must have at least 5 measurements.\n    \&quot;\&quot;\&quot;\n    calc = defaultdict(list)\n    try:\n        for k1, v1 in div_calc.iteritems():\n            for k2, v2 in v1.iteritems():\n                calc[k1].append(v2)\n    except:\n        return \&quot;Error setting up input arrays for Kruskal-Wallis H-Test. Skipping \&quot;\\\n               \&quot;significance testing.\&quot;\n    h, p = stats.kruskal(*calc.values())\n    print \&quot;\\nKruskal-Wallis H-test statistic for {} groups: {}\&quot;.format(str(len(div_calc)), h)\n    print \&quot;p-value: {}\&quot;.format(p)&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;print_KruskalWallisH&quot;,&quot;(&quot;,&quot;div_calc&quot;,&quot;)&quot;,&quot;:&quot;,&quot;calc&quot;,&quot;=&quot;,&quot;defaultdict&quot;,&quot;(&quot;,&quot;list&quot;,&quot;)&quot;,&quot;try&quot;,&quot;:&quot;,&quot;for&quot;,&quot;k1&quot;,&quot;,&quot;,&quot;v1&quot;,&quot;in&quot;,&quot;div_calc&quot;,&quot;.&quot;,&quot;iteritems&quot;,&quot;(&quot;,&quot;)&quot;,&quot;:&quot;,&quot;for&quot;,&quot;k2&quot;,&quot;,&quot;,&quot;v2&quot;,&quot;in&quot;,&quot;v1&quot;,&quot;.&quot;,&quot;iteritems&quot;,&quot;(&quot;,&quot;)&quot;,&quot;:&quot;,&quot;calc&quot;,&quot;[&quot;,&quot;k1&quot;,&quot;]&quot;,&quot;.&quot;,&quot;append&quot;,&quot;(&quot;,&quot;v2&quot;,&quot;)&quot;,&quot;except&quot;,&quot;:&quot;,&quot;return&quot;,&quot;\&quot;Error setting up input arrays for Kruskal-Wallis H-Test. Skipping \&quot;&quot;,&quot;\&quot;significance testing.\&quot;&quot;,&quot;h&quot;,&quot;,&quot;,&quot;p&quot;,&quot;=&quot;,&quot;stats&quot;,&quot;.&quot;,&quot;kruskal&quot;,&quot;(&quot;,&quot;*&quot;,&quot;calc&quot;,&quot;.&quot;,&quot;values&quot;,&quot;(&quot;,&quot;)&quot;,&quot;)&quot;,&quot;print&quot;,&quot;\&quot;\\nKruskal-Wallis H-test statistic for {} groups: {}\&quot;&quot;,&quot;.&quot;,&quot;format&quot;,&quot;(&quot;,&quot;str&quot;,&quot;(&quot;,&quot;len&quot;,&quot;(&quot;,&quot;div_calc&quot;,&quot;)&quot;,&quot;)&quot;,&quot;,&quot;,&quot;h&quot;,&quot;)&quot;,&quot;print&quot;,&quot;\&quot;p-value: {}\&quot;&quot;,&quot;.&quot;,&quot;format&quot;,&quot;(&quot;,&quot;p&quot;,&quot;)&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;print_KruskalWallisH\&quot;,\n  \&quot;(\&quot;,\n  \&quot;div_calc\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;calc\&quot;,\n  \&quot;=\&quot;,\n  \&quot;defaultdict\&quot;,\n  \&quot;(\&quot;,\n  \&quot;list\&quot;,\n  \&quot;)\&quot;,\n  \&quot;try\&quot;,\n  \&quot;:\&quot;,\n  \&quot;for\&quot;,\n  \&quot;k1\&quot;,\n  \&quot;,\&quot;,\n  \&quot;v1\&quot;,\n  \&quot;in\&quot;,\n  \&quot;div_calc\&quot;,\n  \&quot;.\&quot;,\n  \&quot;iteritems\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;for\&quot;,\n  \&quot;k2\&quot;,\n  \&quot;,\&quot;,\n  \&quot;v2\&quot;,\n  \&quot;in\&quot;,\n  \&quot;v1\&quot;,\n  \&quot;.\&quot;,\n  \&quot;iteritems\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;calc\&quot;,\n  \&quot;[\&quot;,\n  \&quot;k1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;append\&quot;,\n  \&quot;(\&quot;,\n  \&quot;v2\&quot;,\n  \&quot;)\&quot;,\n  \&quot;except\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;\\\&quot;Error setting up input arrays for Kruskal-Wallis H-Test. Skipping \\\&quot;\&quot;,\n  \&quot;\\\&quot;significance testing.\\\&quot;\&quot;,\n  \&quot;h\&quot;,\n  \&quot;,\&quot;,\n  \&quot;p\&quot;,\n  \&quot;=\&quot;,\n  \&quot;stats\&quot;,\n  \&quot;.\&quot;,\n  \&quot;kruskal\&quot;,\n  \&quot;(\&quot;,\n  \&quot;*\&quot;,\n  \&quot;calc\&quot;,\n  \&quot;.\&quot;,\n  \&quot;values\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;print\&quot;,\n  \&quot;\\\&quot;\\\\nKruskal-Wallis H-test statistic for {} groups: {}\\\&quot;\&quot;,\n  \&quot;.\&quot;,\n  \&quot;format\&quot;,\n  \&quot;(\&quot;,\n  \&quot;str\&quot;,\n  \&quot;(\&quot;,\n  \&quot;len\&quot;,\n  \&quot;(\&quot;,\n  \&quot;div_calc\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;h\&quot;,\n  \&quot;)\&quot;,\n  \&quot;print\&quot;,\n  \&quot;\\\&quot;p-value: {}\\\&quot;\&quot;,\n  \&quot;.\&quot;,\n  \&quot;format\&quot;,\n  \&quot;(\&quot;,\n  \&quot;p\&quot;,\n  \&quot;)\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Compute the Kruskal-Wallis H-test for independent samples. A typical rule is that\n    each group must have at least 5 measurements.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Compute&quot;,&quot;the&quot;,&quot;Kruskal&quot;,&quot;-&quot;,&quot;Wallis&quot;,&quot;H&quot;,&quot;-&quot;,&quot;test&quot;,&quot;for&quot;,&quot;independent&quot;,&quot;samples&quot;,&quot;.&quot;,&quot;A&quot;,&quot;typical&quot;,&quot;rule&quot;,&quot;is&quot;,&quot;that&quot;,&quot;each&quot;,&quot;group&quot;,&quot;must&quot;,&quot;have&quot;,&quot;at&quot;,&quot;least&quot;,&quot;5&quot;,&quot;measurements&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Compute\&quot;,\n  \&quot;the\&quot;,\n  \&quot;Kruskal\&quot;,\n  \&quot;-\&quot;,\n  \&quot;Wallis\&quot;,\n  \&quot;H\&quot;,\n  \&quot;-\&quot;,\n  \&quot;test\&quot;,\n  \&quot;for\&quot;,\n  \&quot;independent\&quot;,\n  \&quot;samples\&quot;,\n  \&quot;.\&quot;,\n  \&quot;A\&quot;,\n  \&quot;typical\&quot;,\n  \&quot;rule\&quot;,\n  \&quot;is\&quot;,\n  \&quot;that\&quot;,\n  \&quot;each\&quot;,\n  \&quot;group\&quot;,\n  \&quot;must\&quot;,\n  \&quot;have\&quot;,\n  \&quot;at\&quot;,\n  \&quot;least\&quot;,\n  \&quot;5\&quot;,\n  \&quot;measurements\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/bin/diversity.py#L69-L84&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:19,&quot;string&quot;:&quot;19&quot;}}},{&quot;rowIdx&quot;:20,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;bin/diversity.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;handle_program_options&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def handle_program_options():\n    \&quot;\&quot;\&quot;Parses the given options passed in at the command line.\&quot;\&quot;\&quot;\n    parser = argparse.ArgumentParser(description=\&quot;Calculate the alpha diversity\\\n                                     of a set of samples using one or more \\\n                                     metrics and output a kernal density \\\n                                     estimator-smoothed histogram of the \\\n                                     results.\&quot;)\n    parser.add_argument(\&quot;-m\&quot;, \&quot;--map_file\&quot;,\n                        help=\&quot;QIIME mapping file.\&quot;)\n    parser.add_argument(\&quot;-i\&quot;, \&quot;--biom_fp\&quot;,\n                        help=\&quot;Path to the BIOM table\&quot;)\n    parser.add_argument(\&quot;-c\&quot;, \&quot;--category\&quot;,\n                        help=\&quot;Specific category from the mapping file.\&quot;)\n    parser.add_argument(\&quot;-d\&quot;, \&quot;--diversity\&quot;, default=[\&quot;shannon\&quot;], nargs=\&quot;+\&quot;,\n                        help=\&quot;The alpha diversity metric. Default \\\n                             value is 'shannon', which will calculate the Shannon\\\n                             entropy. Multiple metrics can be specified (space separated).\\\n                             The full list of metrics is available at:\\\n                             http://scikit-bio.org/docs/latest/generated/skbio.diversity.alpha.html.\\\n                             Beta diversity metrics will be supported in the future.\&quot;)\n    parser.add_argument(\&quot;--x_label\&quot;, default=[None], nargs=\&quot;+\&quot;,\n                        help=\&quot;The name of the diversity metric to be displayed on the\\\n                        plot as the X-axis label. If multiple metrics are specified,\\\n                        then multiple entries for the X-axis label should be given.\&quot;)\n    parser.add_argument(\&quot;--color_by\&quot;,\n                        help=\&quot;A column name in the mapping file containing\\\n                              hexadecimal (#FF0000) color values that will\\\n                              be used to color the groups. Each sample ID must\\\n                              have a color entry.\&quot;)\n    parser.add_argument(\&quot;--plot_title\&quot;, default=\&quot;\&quot;,\n                        help=\&quot;A descriptive title that will appear at the top \\\n                        of the output plot. Surround with quotes if there are\\\n                        spaces in the title.\&quot;)\n    parser.add_argument(\&quot;-o\&quot;, \&quot;--output_dir\&quot;, default=\&quot;.\&quot;,\n                        help=\&quot;The directory plots will be saved to.\&quot;)\n    parser.add_argument(\&quot;--image_type\&quot;, default=\&quot;png\&quot;,\n                        help=\&quot;The type of image to save: png, svg, pdf, eps, etc...\&quot;)\n    parser.add_argument(\&quot;--save_calculations\&quot;,\n                        help=\&quot;Path and name of text file to store the calculated \&quot;\n                        \&quot;diversity metrics.\&quot;)\n    parser.add_argument(\&quot;--suppress_stats\&quot;, action=\&quot;store_true\&quot;, help=\&quot;Do not display \&quot;\n                        \&quot;significance testing results which are shown by default.\&quot;)\n    parser.add_argument(\&quot;--show_available_metrics\&quot;, action=\&quot;store_true\&quot;,\n                        help=\&quot;Supply this parameter to see which alpha diversity metrics \&quot;\n                             \&quot; are available for usage. No calculations will be performed\&quot;\n                             \&quot; if this parameter is provided.\&quot;)\n    return parser.parse_args()&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def handle_program_options():\n    \&quot;\&quot;\&quot;Parses the given options passed in at the command line.\&quot;\&quot;\&quot;\n    parser = argparse.ArgumentParser(description=\&quot;Calculate the alpha diversity\\\n                                     of a set of samples using one or more \\\n                                     metrics and output a kernal density \\\n                                     estimator-smoothed histogram of the \\\n                                     results.\&quot;)\n    parser.add_argument(\&quot;-m\&quot;, \&quot;--map_file\&quot;,\n                        help=\&quot;QIIME mapping file.\&quot;)\n    parser.add_argument(\&quot;-i\&quot;, \&quot;--biom_fp\&quot;,\n                        help=\&quot;Path to the BIOM table\&quot;)\n    parser.add_argument(\&quot;-c\&quot;, \&quot;--category\&quot;,\n                        help=\&quot;Specific category from the mapping file.\&quot;)\n    parser.add_argument(\&quot;-d\&quot;, \&quot;--diversity\&quot;, default=[\&quot;shannon\&quot;], nargs=\&quot;+\&quot;,\n                        help=\&quot;The alpha diversity metric. Default \\\n                             value is 'shannon', which will calculate the Shannon\\\n                             entropy. Multiple metrics can be specified (space separated).\\\n                             The full list of metrics is available at:\\\n                             http://scikit-bio.org/docs/latest/generated/skbio.diversity.alpha.html.\\\n                             Beta diversity metrics will be supported in the future.\&quot;)\n    parser.add_argument(\&quot;--x_label\&quot;, default=[None], nargs=\&quot;+\&quot;,\n                        help=\&quot;The name of the diversity metric to be displayed on the\\\n                        plot as the X-axis label. If multiple metrics are specified,\\\n                        then multiple entries for the X-axis label should be given.\&quot;)\n    parser.add_argument(\&quot;--color_by\&quot;,\n                        help=\&quot;A column name in the mapping file containing\\\n                              hexadecimal (#FF0000) color values that will\\\n                              be used to color the groups. Each sample ID must\\\n                              have a color entry.\&quot;)\n    parser.add_argument(\&quot;--plot_title\&quot;, default=\&quot;\&quot;,\n                        help=\&quot;A descriptive title that will appear at the top \\\n                        of the output plot. Surround with quotes if there are\\\n                        spaces in the title.\&quot;)\n    parser.add_argument(\&quot;-o\&quot;, \&quot;--output_dir\&quot;, default=\&quot;.\&quot;,\n                        help=\&quot;The directory plots will be saved to.\&quot;)\n    parser.add_argument(\&quot;--image_type\&quot;, default=\&quot;png\&quot;,\n                        help=\&quot;The type of image to save: png, svg, pdf, eps, etc...\&quot;)\n    parser.add_argument(\&quot;--save_calculations\&quot;,\n                        help=\&quot;Path and name of text file to store the calculated \&quot;\n                        \&quot;diversity metrics.\&quot;)\n    parser.add_argument(\&quot;--suppress_stats\&quot;, action=\&quot;store_true\&quot;, help=\&quot;Do not display \&quot;\n                        \&quot;significance testing results which are shown by default.\&quot;)\n    parser.add_argument(\&quot;--show_available_metrics\&quot;, action=\&quot;store_true\&quot;,\n                        help=\&quot;Supply this parameter to see which alpha diversity metrics \&quot;\n                             \&quot; are available for usage. No calculations will be performed\&quot;\n                             \&quot; if this parameter is provided.\&quot;)\n    return parser.parse_args()&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;handle_program_options&quot;,&quot;(&quot;,&quot;)&quot;,&quot;:&quot;,&quot;parser&quot;,&quot;=&quot;,&quot;argparse&quot;,&quot;.&quot;,&quot;ArgumentParser&quot;,&quot;(&quot;,&quot;description&quot;,&quot;=&quot;,&quot;\&quot;Calculate the alpha diversity\\\n                                     of a set of samples using one or more \\\n                                     metrics and output a kernal density \\\n                                     estimator-smoothed histogram of the \\\n                                     results.\&quot;&quot;,&quot;)&quot;,&quot;parser&quot;,&quot;.&quot;,&quot;add_argument&quot;,&quot;(&quot;,&quot;\&quot;-m\&quot;&quot;,&quot;,&quot;,&quot;\&quot;--map_file\&quot;&quot;,&quot;,&quot;,&quot;help&quot;,&quot;=&quot;,&quot;\&quot;QIIME mapping file.\&quot;&quot;,&quot;)&quot;,&quot;parser&quot;,&quot;.&quot;,&quot;add_argument&quot;,&quot;(&quot;,&quot;\&quot;-i\&quot;&quot;,&quot;,&quot;,&quot;\&quot;--biom_fp\&quot;&quot;,&quot;,&quot;,&quot;help&quot;,&quot;=&quot;,&quot;\&quot;Path to the BIOM table\&quot;&quot;,&quot;)&quot;,&quot;parser&quot;,&quot;.&quot;,&quot;add_argument&quot;,&quot;(&quot;,&quot;\&quot;-c\&quot;&quot;,&quot;,&quot;,&quot;\&quot;--category\&quot;&quot;,&quot;,&quot;,&quot;help&quot;,&quot;=&quot;,&quot;\&quot;Specific category from the mapping file.\&quot;&quot;,&quot;)&quot;,&quot;parser&quot;,&quot;.&quot;,&quot;add_argument&quot;,&quot;(&quot;,&quot;\&quot;-d\&quot;&quot;,&quot;,&quot;,&quot;\&quot;--diversity\&quot;&quot;,&quot;,&quot;,&quot;default&quot;,&quot;=&quot;,&quot;[&quot;,&quot;\&quot;shannon\&quot;&quot;,&quot;]&quot;,&quot;,&quot;,&quot;nargs&quot;,&quot;=&quot;,&quot;\&quot;+\&quot;&quot;,&quot;,&quot;,&quot;help&quot;,&quot;=&quot;,&quot;\&quot;The alpha diversity metric. Default \\\n                             value is 'shannon', which will calculate the Shannon\\\n                             entropy. Multiple metrics can be specified (space separated).\\\n                             The full list of metrics is available at:\\\n                             http://scikit-bio.org/docs/latest/generated/skbio.diversity.alpha.html.\\\n                             Beta diversity metrics will be supported in the future.\&quot;&quot;,&quot;)&quot;,&quot;parser&quot;,&quot;.&quot;,&quot;add_argument&quot;,&quot;(&quot;,&quot;\&quot;--x_label\&quot;&quot;,&quot;,&quot;,&quot;default&quot;,&quot;=&quot;,&quot;[&quot;,&quot;None&quot;,&quot;]&quot;,&quot;,&quot;,&quot;nargs&quot;,&quot;=&quot;,&quot;\&quot;+\&quot;&quot;,&quot;,&quot;,&quot;help&quot;,&quot;=&quot;,&quot;\&quot;The name of the diversity metric to be displayed on the\\\n                        plot as the X-axis label. If multiple metrics are specified,\\\n                        then multiple entries for the X-axis label should be given.\&quot;&quot;,&quot;)&quot;,&quot;parser&quot;,&quot;.&quot;,&quot;add_argument&quot;,&quot;(&quot;,&quot;\&quot;--color_by\&quot;&quot;,&quot;,&quot;,&quot;help&quot;,&quot;=&quot;,&quot;\&quot;A column name in the mapping file containing\\\n                              hexadecimal (#FF0000) color values that will\\\n                              be used to color the groups. Each sample ID must\\\n                              have a color entry.\&quot;&quot;,&quot;)&quot;,&quot;parser&quot;,&quot;.&quot;,&quot;add_argument&quot;,&quot;(&quot;,&quot;\&quot;--plot_title\&quot;&quot;,&quot;,&quot;,&quot;default&quot;,&quot;=&quot;,&quot;\&quot;\&quot;&quot;,&quot;,&quot;,&quot;help&quot;,&quot;=&quot;,&quot;\&quot;A descriptive title that will appear at the top \\\n                        of the output plot. Surround with quotes if there are\\\n                        spaces in the title.\&quot;&quot;,&quot;)&quot;,&quot;parser&quot;,&quot;.&quot;,&quot;add_argument&quot;,&quot;(&quot;,&quot;\&quot;-o\&quot;&quot;,&quot;,&quot;,&quot;\&quot;--output_dir\&quot;&quot;,&quot;,&quot;,&quot;default&quot;,&quot;=&quot;,&quot;\&quot;.\&quot;&quot;,&quot;,&quot;,&quot;help&quot;,&quot;=&quot;,&quot;\&quot;The directory plots will be saved to.\&quot;&quot;,&quot;)&quot;,&quot;parser&quot;,&quot;.&quot;,&quot;add_argument&quot;,&quot;(&quot;,&quot;\&quot;--image_type\&quot;&quot;,&quot;,&quot;,&quot;default&quot;,&quot;=&quot;,&quot;\&quot;png\&quot;&quot;,&quot;,&quot;,&quot;help&quot;,&quot;=&quot;,&quot;\&quot;The type of image to save: png, svg, pdf, eps, etc...\&quot;&quot;,&quot;)&quot;,&quot;parser&quot;,&quot;.&quot;,&quot;add_argument&quot;,&quot;(&quot;,&quot;\&quot;--save_calculations\&quot;&quot;,&quot;,&quot;,&quot;help&quot;,&quot;=&quot;,&quot;\&quot;Path and name of text file to store the calculated \&quot;&quot;,&quot;\&quot;diversity metrics.\&quot;&quot;,&quot;)&quot;,&quot;parser&quot;,&quot;.&quot;,&quot;add_argument&quot;,&quot;(&quot;,&quot;\&quot;--suppress_stats\&quot;&quot;,&quot;,&quot;,&quot;action&quot;,&quot;=&quot;,&quot;\&quot;store_true\&quot;&quot;,&quot;,&quot;,&quot;help&quot;,&quot;=&quot;,&quot;\&quot;Do not display \&quot;&quot;,&quot;\&quot;significance testing results which are shown by default.\&quot;&quot;,&quot;)&quot;,&quot;parser&quot;,&quot;.&quot;,&quot;add_argument&quot;,&quot;(&quot;,&quot;\&quot;--show_available_metrics\&quot;&quot;,&quot;,&quot;,&quot;action&quot;,&quot;=&quot;,&quot;\&quot;store_true\&quot;&quot;,&quot;,&quot;,&quot;help&quot;,&quot;=&quot;,&quot;\&quot;Supply this parameter to see which alpha diversity metrics \&quot;&quot;,&quot;\&quot; are available for usage. No calculations will be performed\&quot;&quot;,&quot;\&quot; if this parameter is provided.\&quot;&quot;,&quot;)&quot;,&quot;return&quot;,&quot;parser&quot;,&quot;.&quot;,&quot;parse_args&quot;,&quot;(&quot;,&quot;)&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;handle_program_options\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;parser\&quot;,\n  \&quot;=\&quot;,\n  \&quot;argparse\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ArgumentParser\&quot;,\n  \&quot;(\&quot;,\n  \&quot;description\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;Calculate the alpha diversity\\\\\\n                                     of a set of samples using one or more \\\\\\n                                     metrics and output a kernal density \\\\\\n                                     estimator-smoothed histogram of the \\\\\\n                                     results.\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;parser\&quot;,\n  \&quot;.\&quot;,\n  \&quot;add_argument\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;-m\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;--map_file\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;help\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;QIIME mapping file.\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;parser\&quot;,\n  \&quot;.\&quot;,\n  \&quot;add_argument\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;-i\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;--biom_fp\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;help\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;Path to the BIOM table\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;parser\&quot;,\n  \&quot;.\&quot;,\n  \&quot;add_argument\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;-c\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;--category\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;help\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;Specific category from the mapping file.\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;parser\&quot;,\n  \&quot;.\&quot;,\n  \&quot;add_argument\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;-d\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;--diversity\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;default\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;\\\&quot;shannon\\\&quot;\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;nargs\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;+\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;help\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;The alpha diversity metric. Default \\\\\\n                             value is 'shannon', which will calculate the Shannon\\\\\\n                             entropy. Multiple metrics can be specified (space separated).\\\\\\n                             The full list of metrics is available at:\\\\\\n                             http://scikit-bio.org/docs/latest/generated/skbio.diversity.alpha.html.\\\\\\n                             Beta diversity metrics will be supported in the future.\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;parser\&quot;,\n  \&quot;.\&quot;,\n  \&quot;add_argument\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;--x_label\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;default\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;None\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;nargs\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;+\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;help\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;The name of the diversity metric to be displayed on the\\\\\\n                        plot as the X-axis label. If multiple metrics are specified,\\\\\\n                        then multiple entries for the X-axis label should be given.\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;parser\&quot;,\n  \&quot;.\&quot;,\n  \&quot;add_argument\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;--color_by\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;help\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;A column name in the mapping file containing\\\\\\n                              hexadecimal (#FF0000) color values that will\\\\\\n                              be used to color the groups. Each sample ID must\\\\\\n                              have a color entry.\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;parser\&quot;,\n  \&quot;.\&quot;,\n  \&quot;add_argument\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;--plot_title\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;default\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;help\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;A descriptive title that will appear at the top \\\\\\n                        of the output plot. Surround with quotes if there are\\\\\\n                        spaces in the title.\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;parser\&quot;,\n  \&quot;.\&quot;,\n  \&quot;add_argument\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;-o\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;--output_dir\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;default\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;.\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;help\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;The directory plots will be saved to.\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;parser\&quot;,\n  \&quot;.\&quot;,\n  \&quot;add_argument\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;--image_type\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;default\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;png\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;help\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;The type of image to save: png, svg, pdf, eps, etc...\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;parser\&quot;,\n  \&quot;.\&quot;,\n  \&quot;add_argument\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;--save_calculations\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;help\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;Path and name of text file to store the calculated \\\&quot;\&quot;,\n  \&quot;\\\&quot;diversity metrics.\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;parser\&quot;,\n  \&quot;.\&quot;,\n  \&quot;add_argument\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;--suppress_stats\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;action\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;store_true\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;help\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;Do not display \\\&quot;\&quot;,\n  \&quot;\\\&quot;significance testing results which are shown by default.\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;parser\&quot;,\n  \&quot;.\&quot;,\n  \&quot;add_argument\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;--show_available_metrics\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;action\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;store_true\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;help\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;Supply this parameter to see which alpha diversity metrics \\\&quot;\&quot;,\n  \&quot;\\\&quot; are available for usage. No calculations will be performed\\\&quot;\&quot;,\n  \&quot;\\\&quot; if this parameter is provided.\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;parser\&quot;,\n  \&quot;.\&quot;,\n  \&quot;parse_args\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Parses the given options passed in at the command line.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Parses&quot;,&quot;the&quot;,&quot;given&quot;,&quot;options&quot;,&quot;passed&quot;,&quot;in&quot;,&quot;at&quot;,&quot;the&quot;,&quot;command&quot;,&quot;line&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Parses\&quot;,\n  \&quot;the\&quot;,\n  \&quot;given\&quot;,\n  \&quot;options\&quot;,\n  \&quot;passed\&quot;,\n  \&quot;in\&quot;,\n  \&quot;at\&quot;,\n  \&quot;the\&quot;,\n  \&quot;command\&quot;,\n  \&quot;line\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/bin/diversity.py#L122-L168&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:20,&quot;string&quot;:&quot;20&quot;}}},{&quot;rowIdx&quot;:21,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;christophertbrown/bioscripts&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ctbBio/search.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;blastdb&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def blastdb(fasta, maxfile = 10000000):\n    \&quot;\&quot;\&quot;\n    make blast db\n    \&quot;\&quot;\&quot;\n    db = fasta.rsplit('.', 1)[0]\n    type = check_type(fasta)\n    if type == 'nucl':\n        type = ['nhr', type]\n    else:\n        type = ['phr', type]\n    if os.path.exists('%s.%s' % (db, type[0])) is False \\\n            and os.path.exists('%s.00.%s' % (db, type[0])) is False:\n        print('# ... making blastdb for: %s' % (fasta), file=sys.stderr)\n        os.system('makeblastdb \\\n                -in %s -out %s -dbtype %s -max_file_sz %s >> log.txt' \\\n                % (fasta, db, type[1], maxfile))\n    else:\n        print('# ... database found for: %s' % (fasta), file=sys.stderr)\n    return db&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def blastdb(fasta, maxfile = 10000000):\n    \&quot;\&quot;\&quot;\n    make blast db\n    \&quot;\&quot;\&quot;\n    db = fasta.rsplit('.', 1)[0]\n    type = check_type(fasta)\n    if type == 'nucl':\n        type = ['nhr', type]\n    else:\n        type = ['phr', type]\n    if os.path.exists('%s.%s' % (db, type[0])) is False \\\n            and os.path.exists('%s.00.%s' % (db, type[0])) is False:\n        print('# ... making blastdb for: %s' % (fasta), file=sys.stderr)\n        os.system('makeblastdb \\\n                -in %s -out %s -dbtype %s -max_file_sz %s >> log.txt' \\\n                % (fasta, db, type[1], maxfile))\n    else:\n        print('# ... database found for: %s' % (fasta), file=sys.stderr)\n    return db&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;blastdb&quot;,&quot;(&quot;,&quot;fasta&quot;,&quot;,&quot;,&quot;maxfile&quot;,&quot;=&quot;,&quot;10000000&quot;,&quot;)&quot;,&quot;:&quot;,&quot;db&quot;,&quot;=&quot;,&quot;fasta&quot;,&quot;.&quot;,&quot;rsplit&quot;,&quot;(&quot;,&quot;'.'&quot;,&quot;,&quot;,&quot;1&quot;,&quot;)&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;type&quot;,&quot;=&quot;,&quot;check_type&quot;,&quot;(&quot;,&quot;fasta&quot;,&quot;)&quot;,&quot;if&quot;,&quot;type&quot;,&quot;==&quot;,&quot;'nucl'&quot;,&quot;:&quot;,&quot;type&quot;,&quot;=&quot;,&quot;[&quot;,&quot;'nhr'&quot;,&quot;,&quot;,&quot;type&quot;,&quot;]&quot;,&quot;else&quot;,&quot;:&quot;,&quot;type&quot;,&quot;=&quot;,&quot;[&quot;,&quot;'phr'&quot;,&quot;,&quot;,&quot;type&quot;,&quot;]&quot;,&quot;if&quot;,&quot;os&quot;,&quot;.&quot;,&quot;path&quot;,&quot;.&quot;,&quot;exists&quot;,&quot;(&quot;,&quot;'%s.%s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;db&quot;,&quot;,&quot;,&quot;type&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;)&quot;,&quot;)&quot;,&quot;is&quot;,&quot;False&quot;,&quot;and&quot;,&quot;os&quot;,&quot;.&quot;,&quot;path&quot;,&quot;.&quot;,&quot;exists&quot;,&quot;(&quot;,&quot;'%s.00.%s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;db&quot;,&quot;,&quot;,&quot;type&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;)&quot;,&quot;)&quot;,&quot;is&quot;,&quot;False&quot;,&quot;:&quot;,&quot;print&quot;,&quot;(&quot;,&quot;'# ... making blastdb for: %s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;fasta&quot;,&quot;)&quot;,&quot;,&quot;,&quot;file&quot;,&quot;=&quot;,&quot;sys&quot;,&quot;.&quot;,&quot;stderr&quot;,&quot;)&quot;,&quot;os&quot;,&quot;.&quot;,&quot;system&quot;,&quot;(&quot;,&quot;'makeblastdb \\\n                -in %s -out %s -dbtype %s -max_file_sz %s >> log.txt'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;fasta&quot;,&quot;,&quot;,&quot;db&quot;,&quot;,&quot;,&quot;type&quot;,&quot;[&quot;,&quot;1&quot;,&quot;]&quot;,&quot;,&quot;,&quot;maxfile&quot;,&quot;)&quot;,&quot;)&quot;,&quot;else&quot;,&quot;:&quot;,&quot;print&quot;,&quot;(&quot;,&quot;'# ... database found for: %s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;fasta&quot;,&quot;)&quot;,&quot;,&quot;,&quot;file&quot;,&quot;=&quot;,&quot;sys&quot;,&quot;.&quot;,&quot;stderr&quot;,&quot;)&quot;,&quot;return&quot;,&quot;db&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;blastdb\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;,\&quot;,\n  \&quot;maxfile\&quot;,\n  \&quot;=\&quot;,\n  \&quot;10000000\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;db\&quot;,\n  \&quot;=\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rsplit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'.'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;type\&quot;,\n  \&quot;=\&quot;,\n  \&quot;check_type\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;type\&quot;,\n  \&quot;==\&quot;,\n  \&quot;'nucl'\&quot;,\n  \&quot;:\&quot;,\n  \&quot;type\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'nhr'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;type\&quot;,\n  \&quot;]\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;type\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'phr'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;type\&quot;,\n  \&quot;]\&quot;,\n  \&quot;if\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;path\&quot;,\n  \&quot;.\&quot;,\n  \&quot;exists\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'%s.%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;db\&quot;,\n  \&quot;,\&quot;,\n  \&quot;type\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;is\&quot;,\n  \&quot;False\&quot;,\n  \&quot;and\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;path\&quot;,\n  \&quot;.\&quot;,\n  \&quot;exists\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'%s.00.%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;db\&quot;,\n  \&quot;,\&quot;,\n  \&quot;type\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;is\&quot;,\n  \&quot;False\&quot;,\n  \&quot;:\&quot;,\n  \&quot;print\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'# ... making blastdb for: %s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;file\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sys\&quot;,\n  \&quot;.\&quot;,\n  \&quot;stderr\&quot;,\n  \&quot;)\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;system\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'makeblastdb \\\\\\n                -in %s -out %s -dbtype %s -max_file_sz %s >> log.txt'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;,\&quot;,\n  \&quot;db\&quot;,\n  \&quot;,\&quot;,\n  \&quot;type\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;maxfile\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;print\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'# ... database found for: %s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;file\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sys\&quot;,\n  \&quot;.\&quot;,\n  \&quot;stderr\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;db\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;make blast db&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;make&quot;,&quot;blast&quot;,&quot;db&quot;],&quot;string&quot;:&quot;[\n  \&quot;make\&quot;,\n  \&quot;blast\&quot;,\n  \&quot;db\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;83b2566b3a5745437ec651cd6cafddd056846240&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/search.py#L28-L46&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:21,&quot;string&quot;:&quot;21&quot;}}},{&quot;rowIdx&quot;:22,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;christophertbrown/bioscripts&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ctbBio/search.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;usearchdb&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def usearchdb(fasta, alignment = 'local', usearch_loc = 'usearch'):\n    \&quot;\&quot;\&quot;\n    make usearch db\n    \&quot;\&quot;\&quot;\n    if '.udb' in fasta:\n        print('# ... database found: %s' % (fasta), file=sys.stderr)\n        return fasta\n    type = check_type(fasta)\n    db = '%s.%s.udb' % (fasta.rsplit('.', 1)[0], type)\n    if os.path.exists(db) is False:\n        print('# ... making usearch db for: %s' % (fasta), file=sys.stderr)\n        if alignment == 'local':\n            os.system('%s -makeudb_ublast %s -output %s >> log.txt' % (usearch_loc, fasta, db))\n        elif alignment == 'global':\n            os.system('%s -makeudb_usearch %s -output %s >> log.txt' % (usearch_loc, fasta, db))\n    else:\n        print('# ... database found for: %s' % (fasta), file=sys.stderr)\n    return db&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def usearchdb(fasta, alignment = 'local', usearch_loc = 'usearch'):\n    \&quot;\&quot;\&quot;\n    make usearch db\n    \&quot;\&quot;\&quot;\n    if '.udb' in fasta:\n        print('# ... database found: %s' % (fasta), file=sys.stderr)\n        return fasta\n    type = check_type(fasta)\n    db = '%s.%s.udb' % (fasta.rsplit('.', 1)[0], type)\n    if os.path.exists(db) is False:\n        print('# ... making usearch db for: %s' % (fasta), file=sys.stderr)\n        if alignment == 'local':\n            os.system('%s -makeudb_ublast %s -output %s >> log.txt' % (usearch_loc, fasta, db))\n        elif alignment == 'global':\n            os.system('%s -makeudb_usearch %s -output %s >> log.txt' % (usearch_loc, fasta, db))\n    else:\n        print('# ... database found for: %s' % (fasta), file=sys.stderr)\n    return db&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;usearchdb&quot;,&quot;(&quot;,&quot;fasta&quot;,&quot;,&quot;,&quot;alignment&quot;,&quot;=&quot;,&quot;'local'&quot;,&quot;,&quot;,&quot;usearch_loc&quot;,&quot;=&quot;,&quot;'usearch'&quot;,&quot;)&quot;,&quot;:&quot;,&quot;if&quot;,&quot;'.udb'&quot;,&quot;in&quot;,&quot;fasta&quot;,&quot;:&quot;,&quot;print&quot;,&quot;(&quot;,&quot;'# ... database found: %s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;fasta&quot;,&quot;)&quot;,&quot;,&quot;,&quot;file&quot;,&quot;=&quot;,&quot;sys&quot;,&quot;.&quot;,&quot;stderr&quot;,&quot;)&quot;,&quot;return&quot;,&quot;fasta&quot;,&quot;type&quot;,&quot;=&quot;,&quot;check_type&quot;,&quot;(&quot;,&quot;fasta&quot;,&quot;)&quot;,&quot;db&quot;,&quot;=&quot;,&quot;'%s.%s.udb'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;fasta&quot;,&quot;.&quot;,&quot;rsplit&quot;,&quot;(&quot;,&quot;'.'&quot;,&quot;,&quot;,&quot;1&quot;,&quot;)&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;,&quot;,&quot;type&quot;,&quot;)&quot;,&quot;if&quot;,&quot;os&quot;,&quot;.&quot;,&quot;path&quot;,&quot;.&quot;,&quot;exists&quot;,&quot;(&quot;,&quot;db&quot;,&quot;)&quot;,&quot;is&quot;,&quot;False&quot;,&quot;:&quot;,&quot;print&quot;,&quot;(&quot;,&quot;'# ... making usearch db for: %s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;fasta&quot;,&quot;)&quot;,&quot;,&quot;,&quot;file&quot;,&quot;=&quot;,&quot;sys&quot;,&quot;.&quot;,&quot;stderr&quot;,&quot;)&quot;,&quot;if&quot;,&quot;alignment&quot;,&quot;==&quot;,&quot;'local'&quot;,&quot;:&quot;,&quot;os&quot;,&quot;.&quot;,&quot;system&quot;,&quot;(&quot;,&quot;'%s -makeudb_ublast %s -output %s >> log.txt'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;usearch_loc&quot;,&quot;,&quot;,&quot;fasta&quot;,&quot;,&quot;,&quot;db&quot;,&quot;)&quot;,&quot;)&quot;,&quot;elif&quot;,&quot;alignment&quot;,&quot;==&quot;,&quot;'global'&quot;,&quot;:&quot;,&quot;os&quot;,&quot;.&quot;,&quot;system&quot;,&quot;(&quot;,&quot;'%s -makeudb_usearch %s -output %s >> log.txt'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;usearch_loc&quot;,&quot;,&quot;,&quot;fasta&quot;,&quot;,&quot;,&quot;db&quot;,&quot;)&quot;,&quot;)&quot;,&quot;else&quot;,&quot;:&quot;,&quot;print&quot;,&quot;(&quot;,&quot;'# ... database found for: %s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;fasta&quot;,&quot;)&quot;,&quot;,&quot;,&quot;file&quot;,&quot;=&quot;,&quot;sys&quot;,&quot;.&quot;,&quot;stderr&quot;,&quot;)&quot;,&quot;return&quot;,&quot;db&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;usearchdb\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;,\&quot;,\n  \&quot;alignment\&quot;,\n  \&quot;=\&quot;,\n  \&quot;'local'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;usearch_loc\&quot;,\n  \&quot;=\&quot;,\n  \&quot;'usearch'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;'.udb'\&quot;,\n  \&quot;in\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;:\&quot;,\n  \&quot;print\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'# ... database found: %s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;file\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sys\&quot;,\n  \&quot;.\&quot;,\n  \&quot;stderr\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;type\&quot;,\n  \&quot;=\&quot;,\n  \&quot;check_type\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;)\&quot;,\n  \&quot;db\&quot;,\n  \&quot;=\&quot;,\n  \&quot;'%s.%s.udb'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rsplit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'.'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;type\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;path\&quot;,\n  \&quot;.\&quot;,\n  \&quot;exists\&quot;,\n  \&quot;(\&quot;,\n  \&quot;db\&quot;,\n  \&quot;)\&quot;,\n  \&quot;is\&quot;,\n  \&quot;False\&quot;,\n  \&quot;:\&quot;,\n  \&quot;print\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'# ... making usearch db for: %s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;file\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sys\&quot;,\n  \&quot;.\&quot;,\n  \&quot;stderr\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;alignment\&quot;,\n  \&quot;==\&quot;,\n  \&quot;'local'\&quot;,\n  \&quot;:\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;system\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'%s -makeudb_ublast %s -output %s >> log.txt'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;usearch_loc\&quot;,\n  \&quot;,\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;,\&quot;,\n  \&quot;db\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;elif\&quot;,\n  \&quot;alignment\&quot;,\n  \&quot;==\&quot;,\n  \&quot;'global'\&quot;,\n  \&quot;:\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;system\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'%s -makeudb_usearch %s -output %s >> log.txt'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;usearch_loc\&quot;,\n  \&quot;,\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;,\&quot;,\n  \&quot;db\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;print\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'# ... database found for: %s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;file\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sys\&quot;,\n  \&quot;.\&quot;,\n  \&quot;stderr\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;db\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;make usearch db&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;make&quot;,&quot;usearch&quot;,&quot;db&quot;],&quot;string&quot;:&quot;[\n  \&quot;make\&quot;,\n  \&quot;usearch\&quot;,\n  \&quot;db\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;83b2566b3a5745437ec651cd6cafddd056846240&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/search.py#L68-L85&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:22,&quot;string&quot;:&quot;22&quot;}}},{&quot;rowIdx&quot;:23,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;mkouhei/bootstrap-py&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;bootstrap_py/control.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;_pp&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _pp(dict_data):\n    \&quot;\&quot;\&quot;Pretty print.\&quot;\&quot;\&quot;\n    for key, val in dict_data.items():\n        # pylint: disable=superfluous-parens\n        print('{0:<11}: {1}'.format(key, val))&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _pp(dict_data):\n    \&quot;\&quot;\&quot;Pretty print.\&quot;\&quot;\&quot;\n    for key, val in dict_data.items():\n        # pylint: disable=superfluous-parens\n        print('{0:<11}: {1}'.format(key, val))&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;_pp&quot;,&quot;(&quot;,&quot;dict_data&quot;,&quot;)&quot;,&quot;:&quot;,&quot;for&quot;,&quot;key&quot;,&quot;,&quot;,&quot;val&quot;,&quot;in&quot;,&quot;dict_data&quot;,&quot;.&quot;,&quot;items&quot;,&quot;(&quot;,&quot;)&quot;,&quot;:&quot;,&quot;# pylint: disable=superfluous-parens&quot;,&quot;print&quot;,&quot;(&quot;,&quot;'{0:<11}: {1}'&quot;,&quot;.&quot;,&quot;format&quot;,&quot;(&quot;,&quot;key&quot;,&quot;,&quot;,&quot;val&quot;,&quot;)&quot;,&quot;)&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;_pp\&quot;,\n  \&quot;(\&quot;,\n  \&quot;dict_data\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;for\&quot;,\n  \&quot;key\&quot;,\n  \&quot;,\&quot;,\n  \&quot;val\&quot;,\n  \&quot;in\&quot;,\n  \&quot;dict_data\&quot;,\n  \&quot;.\&quot;,\n  \&quot;items\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;# pylint: disable=superfluous-parens\&quot;,\n  \&quot;print\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'{0:<11}: {1}'\&quot;,\n  \&quot;.\&quot;,\n  \&quot;format\&quot;,\n  \&quot;(\&quot;,\n  \&quot;key\&quot;,\n  \&quot;,\&quot;,\n  \&quot;val\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Pretty print.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Pretty&quot;,&quot;print&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Pretty\&quot;,\n  \&quot;print\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;95d56ed98ef409fd9f019dc352fd1c3711533275&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/mkouhei/bootstrap-py/blob/95d56ed98ef409fd9f019dc352fd1c3711533275/bootstrap_py/control.py#L11-L15&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:23,&quot;string&quot;:&quot;23&quot;}}},{&quot;rowIdx&quot;:24,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;mkouhei/bootstrap-py&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;bootstrap_py/control.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;print_licences&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def print_licences(params, metadata):\n    \&quot;\&quot;\&quot;Print licenses.\n\n    :param argparse.Namespace params: parameter\n    :param bootstrap_py.classifier.Classifiers metadata: package metadata\n    \&quot;\&quot;\&quot;\n    if hasattr(params, 'licenses'):\n        if params.licenses:\n            _pp(metadata.licenses_desc())\n        sys.exit(0)&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def print_licences(params, metadata):\n    \&quot;\&quot;\&quot;Print licenses.\n\n    :param argparse.Namespace params: parameter\n    :param bootstrap_py.classifier.Classifiers metadata: package metadata\n    \&quot;\&quot;\&quot;\n    if hasattr(params, 'licenses'):\n        if params.licenses:\n            _pp(metadata.licenses_desc())\n        sys.exit(0)&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;print_licences&quot;,&quot;(&quot;,&quot;params&quot;,&quot;,&quot;,&quot;metadata&quot;,&quot;)&quot;,&quot;:&quot;,&quot;if&quot;,&quot;hasattr&quot;,&quot;(&quot;,&quot;params&quot;,&quot;,&quot;,&quot;'licenses'&quot;,&quot;)&quot;,&quot;:&quot;,&quot;if&quot;,&quot;params&quot;,&quot;.&quot;,&quot;licenses&quot;,&quot;:&quot;,&quot;_pp&quot;,&quot;(&quot;,&quot;metadata&quot;,&quot;.&quot;,&quot;licenses_desc&quot;,&quot;(&quot;,&quot;)&quot;,&quot;)&quot;,&quot;sys&quot;,&quot;.&quot;,&quot;exit&quot;,&quot;(&quot;,&quot;0&quot;,&quot;)&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;print_licences\&quot;,\n  \&quot;(\&quot;,\n  \&quot;params\&quot;,\n  \&quot;,\&quot;,\n  \&quot;metadata\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;hasattr\&quot;,\n  \&quot;(\&quot;,\n  \&quot;params\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'licenses'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;params\&quot;,\n  \&quot;.\&quot;,\n  \&quot;licenses\&quot;,\n  \&quot;:\&quot;,\n  \&quot;_pp\&quot;,\n  \&quot;(\&quot;,\n  \&quot;metadata\&quot;,\n  \&quot;.\&quot;,\n  \&quot;licenses_desc\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;sys\&quot;,\n  \&quot;.\&quot;,\n  \&quot;exit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;0\&quot;,\n  \&quot;)\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Print licenses.\n\n    :param argparse.Namespace params: parameter\n    :param bootstrap_py.classifier.Classifiers metadata: package metadata&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Print&quot;,&quot;licenses&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Print\&quot;,\n  \&quot;licenses\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;95d56ed98ef409fd9f019dc352fd1c3711533275&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/mkouhei/bootstrap-py/blob/95d56ed98ef409fd9f019dc352fd1c3711533275/bootstrap_py/control.py#L27-L36&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:24,&quot;string&quot;:&quot;24&quot;}}},{&quot;rowIdx&quot;:25,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;mkouhei/bootstrap-py&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;bootstrap_py/control.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;check_repository_existence&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def check_repository_existence(params):\n    \&quot;\&quot;\&quot;Check repository existence.\n\n    :param argparse.Namespace params: parameters\n    \&quot;\&quot;\&quot;\n    repodir = os.path.join(params.outdir, params.name)\n    if os.path.isdir(repodir):\n        raise Conflict(\n            'Package repository \&quot;{0}\&quot; has already exists.'.format(repodir))&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def check_repository_existence(params):\n    \&quot;\&quot;\&quot;Check repository existence.\n\n    :param argparse.Namespace params: parameters\n    \&quot;\&quot;\&quot;\n    repodir = os.path.join(params.outdir, params.name)\n    if os.path.isdir(repodir):\n        raise Conflict(\n            'Package repository \&quot;{0}\&quot; has already exists.'.format(repodir))&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;check_repository_existence&quot;,&quot;(&quot;,&quot;params&quot;,&quot;)&quot;,&quot;:&quot;,&quot;repodir&quot;,&quot;=&quot;,&quot;os&quot;,&quot;.&quot;,&quot;path&quot;,&quot;.&quot;,&quot;join&quot;,&quot;(&quot;,&quot;params&quot;,&quot;.&quot;,&quot;outdir&quot;,&quot;,&quot;,&quot;params&quot;,&quot;.&quot;,&quot;name&quot;,&quot;)&quot;,&quot;if&quot;,&quot;os&quot;,&quot;.&quot;,&quot;path&quot;,&quot;.&quot;,&quot;isdir&quot;,&quot;(&quot;,&quot;repodir&quot;,&quot;)&quot;,&quot;:&quot;,&quot;raise&quot;,&quot;Conflict&quot;,&quot;(&quot;,&quot;'Package repository \&quot;{0}\&quot; has already exists.'&quot;,&quot;.&quot;,&quot;format&quot;,&quot;(&quot;,&quot;repodir&quot;,&quot;)&quot;,&quot;)&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;check_repository_existence\&quot;,\n  \&quot;(\&quot;,\n  \&quot;params\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;repodir\&quot;,\n  \&quot;=\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;path\&quot;,\n  \&quot;.\&quot;,\n  \&quot;join\&quot;,\n  \&quot;(\&quot;,\n  \&quot;params\&quot;,\n  \&quot;.\&quot;,\n  \&quot;outdir\&quot;,\n  \&quot;,\&quot;,\n  \&quot;params\&quot;,\n  \&quot;.\&quot;,\n  \&quot;name\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;path\&quot;,\n  \&quot;.\&quot;,\n  \&quot;isdir\&quot;,\n  \&quot;(\&quot;,\n  \&quot;repodir\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;raise\&quot;,\n  \&quot;Conflict\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'Package repository \\\&quot;{0}\\\&quot; has already exists.'\&quot;,\n  \&quot;.\&quot;,\n  \&quot;format\&quot;,\n  \&quot;(\&quot;,\n  \&quot;repodir\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Check repository existence.\n\n    :param argparse.Namespace params: parameters&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Check&quot;,&quot;repository&quot;,&quot;existence&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Check\&quot;,\n  \&quot;repository\&quot;,\n  \&quot;existence\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;95d56ed98ef409fd9f019dc352fd1c3711533275&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/mkouhei/bootstrap-py/blob/95d56ed98ef409fd9f019dc352fd1c3711533275/bootstrap_py/control.py#L39-L47&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:25,&quot;string&quot;:&quot;25&quot;}}},{&quot;rowIdx&quot;:26,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;mkouhei/bootstrap-py&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;bootstrap_py/control.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;generate_package&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def generate_package(params):\n    \&quot;\&quot;\&quot;Generate package repository.\n\n    :param argparse.Namespace params: parameters\n    \&quot;\&quot;\&quot;\n    pkg_data = package.PackageData(params)\n    pkg_tree = package.PackageTree(pkg_data)\n    pkg_tree.generate()\n    pkg_tree.move()\n    VCS(os.path.join(pkg_tree.outdir, pkg_tree.name), pkg_tree.pkg_data)&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def generate_package(params):\n    \&quot;\&quot;\&quot;Generate package repository.\n\n    :param argparse.Namespace params: parameters\n    \&quot;\&quot;\&quot;\n    pkg_data = package.PackageData(params)\n    pkg_tree = package.PackageTree(pkg_data)\n    pkg_tree.generate()\n    pkg_tree.move()\n    VCS(os.path.join(pkg_tree.outdir, pkg_tree.name), pkg_tree.pkg_data)&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;generate_package&quot;,&quot;(&quot;,&quot;params&quot;,&quot;)&quot;,&quot;:&quot;,&quot;pkg_data&quot;,&quot;=&quot;,&quot;package&quot;,&quot;.&quot;,&quot;PackageData&quot;,&quot;(&quot;,&quot;params&quot;,&quot;)&quot;,&quot;pkg_tree&quot;,&quot;=&quot;,&quot;package&quot;,&quot;.&quot;,&quot;PackageTree&quot;,&quot;(&quot;,&quot;pkg_data&quot;,&quot;)&quot;,&quot;pkg_tree&quot;,&quot;.&quot;,&quot;generate&quot;,&quot;(&quot;,&quot;)&quot;,&quot;pkg_tree&quot;,&quot;.&quot;,&quot;move&quot;,&quot;(&quot;,&quot;)&quot;,&quot;VCS&quot;,&quot;(&quot;,&quot;os&quot;,&quot;.&quot;,&quot;path&quot;,&quot;.&quot;,&quot;join&quot;,&quot;(&quot;,&quot;pkg_tree&quot;,&quot;.&quot;,&quot;outdir&quot;,&quot;,&quot;,&quot;pkg_tree&quot;,&quot;.&quot;,&quot;name&quot;,&quot;)&quot;,&quot;,&quot;,&quot;pkg_tree&quot;,&quot;.&quot;,&quot;pkg_data&quot;,&quot;)&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;generate_package\&quot;,\n  \&quot;(\&quot;,\n  \&quot;params\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;pkg_data\&quot;,\n  \&quot;=\&quot;,\n  \&quot;package\&quot;,\n  \&quot;.\&quot;,\n  \&quot;PackageData\&quot;,\n  \&quot;(\&quot;,\n  \&quot;params\&quot;,\n  \&quot;)\&quot;,\n  \&quot;pkg_tree\&quot;,\n  \&quot;=\&quot;,\n  \&quot;package\&quot;,\n  \&quot;.\&quot;,\n  \&quot;PackageTree\&quot;,\n  \&quot;(\&quot;,\n  \&quot;pkg_data\&quot;,\n  \&quot;)\&quot;,\n  \&quot;pkg_tree\&quot;,\n  \&quot;.\&quot;,\n  \&quot;generate\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;pkg_tree\&quot;,\n  \&quot;.\&quot;,\n  \&quot;move\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;VCS\&quot;,\n  \&quot;(\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;path\&quot;,\n  \&quot;.\&quot;,\n  \&quot;join\&quot;,\n  \&quot;(\&quot;,\n  \&quot;pkg_tree\&quot;,\n  \&quot;.\&quot;,\n  \&quot;outdir\&quot;,\n  \&quot;,\&quot;,\n  \&quot;pkg_tree\&quot;,\n  \&quot;.\&quot;,\n  \&quot;name\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;pkg_tree\&quot;,\n  \&quot;.\&quot;,\n  \&quot;pkg_data\&quot;,\n  \&quot;)\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Generate package repository.\n\n    :param argparse.Namespace params: parameters&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Generate&quot;,&quot;package&quot;,&quot;repository&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Generate\&quot;,\n  \&quot;package\&quot;,\n  \&quot;repository\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;95d56ed98ef409fd9f019dc352fd1c3711533275&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/mkouhei/bootstrap-py/blob/95d56ed98ef409fd9f019dc352fd1c3711533275/bootstrap_py/control.py#L59-L68&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:26,&quot;string&quot;:&quot;26&quot;}}},{&quot;rowIdx&quot;:27,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;christophertbrown/bioscripts&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ctbBio/sam2fastq.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;print_single&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def print_single(line, rev):\n    \&quot;\&quot;\&quot;\n    print single reads to stderr\n    \&quot;\&quot;\&quot;\n    if rev is True:\n        seq = rc(['', line[9]])[1]\n        qual = line[10][::-1]\n    else:\n        seq = line[9]\n        qual = line[10]\n    fq = ['@%s' % line[0], seq, '+%s' % line[0], qual]\n    print('\\n'.join(fq), file = sys.stderr)&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def print_single(line, rev):\n    \&quot;\&quot;\&quot;\n    print single reads to stderr\n    \&quot;\&quot;\&quot;\n    if rev is True:\n        seq = rc(['', line[9]])[1]\n        qual = line[10][::-1]\n    else:\n        seq = line[9]\n        qual = line[10]\n    fq = ['@%s' % line[0], seq, '+%s' % line[0], qual]\n    print('\\n'.join(fq), file = sys.stderr)&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;print_single&quot;,&quot;(&quot;,&quot;line&quot;,&quot;,&quot;,&quot;rev&quot;,&quot;)&quot;,&quot;:&quot;,&quot;if&quot;,&quot;rev&quot;,&quot;is&quot;,&quot;True&quot;,&quot;:&quot;,&quot;seq&quot;,&quot;=&quot;,&quot;rc&quot;,&quot;(&quot;,&quot;[&quot;,&quot;''&quot;,&quot;,&quot;,&quot;line&quot;,&quot;[&quot;,&quot;9&quot;,&quot;]&quot;,&quot;]&quot;,&quot;)&quot;,&quot;[&quot;,&quot;1&quot;,&quot;]&quot;,&quot;qual&quot;,&quot;=&quot;,&quot;line&quot;,&quot;[&quot;,&quot;10&quot;,&quot;]&quot;,&quot;[&quot;,&quot;:&quot;,&quot;:&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;,&quot;else&quot;,&quot;:&quot;,&quot;seq&quot;,&quot;=&quot;,&quot;line&quot;,&quot;[&quot;,&quot;9&quot;,&quot;]&quot;,&quot;qual&quot;,&quot;=&quot;,&quot;line&quot;,&quot;[&quot;,&quot;10&quot;,&quot;]&quot;,&quot;fq&quot;,&quot;=&quot;,&quot;[&quot;,&quot;'@%s'&quot;,&quot;%&quot;,&quot;line&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;,&quot;,&quot;seq&quot;,&quot;,&quot;,&quot;'+%s'&quot;,&quot;%&quot;,&quot;line&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;,&quot;,&quot;qual&quot;,&quot;]&quot;,&quot;print&quot;,&quot;(&quot;,&quot;'\\n'&quot;,&quot;.&quot;,&quot;join&quot;,&quot;(&quot;,&quot;fq&quot;,&quot;)&quot;,&quot;,&quot;,&quot;file&quot;,&quot;=&quot;,&quot;sys&quot;,&quot;.&quot;,&quot;stderr&quot;,&quot;)&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;print_single\&quot;,\n  \&quot;(\&quot;,\n  \&quot;line\&quot;,\n  \&quot;,\&quot;,\n  \&quot;rev\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;rev\&quot;,\n  \&quot;is\&quot;,\n  \&quot;True\&quot;,\n  \&quot;:\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;=\&quot;,\n  \&quot;rc\&quot;,\n  \&quot;(\&quot;,\n  \&quot;[\&quot;,\n  \&quot;''\&quot;,\n  \&quot;,\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;9\&quot;,\n  \&quot;]\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;qual\&quot;,\n  \&quot;=\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;10\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;:\&quot;,\n  \&quot;:\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;=\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;9\&quot;,\n  \&quot;]\&quot;,\n  \&quot;qual\&quot;,\n  \&quot;=\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;10\&quot;,\n  \&quot;]\&quot;,\n  \&quot;fq\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'@%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'+%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;qual\&quot;,\n  \&quot;]\&quot;,\n  \&quot;print\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'\\\\n'\&quot;,\n  \&quot;.\&quot;,\n  \&quot;join\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fq\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;file\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sys\&quot;,\n  \&quot;.\&quot;,\n  \&quot;stderr\&quot;,\n  \&quot;)\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;print single reads to stderr&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;print&quot;,&quot;single&quot;,&quot;reads&quot;,&quot;to&quot;,&quot;stderr&quot;],&quot;string&quot;:&quot;[\n  \&quot;print\&quot;,\n  \&quot;single\&quot;,\n  \&quot;reads\&quot;,\n  \&quot;to\&quot;,\n  \&quot;stderr\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;83b2566b3a5745437ec651cd6cafddd056846240&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/sam2fastq.py#L13-L24&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:27,&quot;string&quot;:&quot;27&quot;}}},{&quot;rowIdx&quot;:28,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;christophertbrown/bioscripts&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ctbBio/sam2fastq.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;sam2fastq&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def sam2fastq(sam, singles = False, force = False):\n    \&quot;\&quot;\&quot;\n    convert sam to fastq\n    \&quot;\&quot;\&quot;\n    L, R = None, None\n    for line in sam:\n        if line.startswith('@') is True:\n            continue\n        line = line.strip().split()\n        bit = [True if i == '1' else False \\\n                for i in bin(int(line[1])).split('b')[1][::-1]]\n        while len(bit) < 8:\n            bit.append(False)\n        pair, proper, na, nap, rev, mrev, left, right = bit\n        # make sure read is paired\n        if pair is False:\n            if singles is True:\n                print_single(line, rev)\n            continue\n        # check if sequence is reverse-complemented\n        if rev is True:\n            seq = rc(['', line[9]])[1]\n            qual = line[10][::-1]\n        else:\n            seq = line[9]\n            qual = line[10]\n        # check if read is forward or reverse, return when both have been found\n        if left is True:\n            if L is not None and force is False:\n                print('sam file is not sorted', file = sys.stderr)\n                print('\\te.g.: %s' % (line[0]), file = sys.stderr)\n                exit()\n            if L is not None:\n                L = None\n                continue\n            L = ['@%s' % line[0], seq, '+%s' % line[0], qual]\n            if R is not None:\n                yield L\n                yield R\n                L, R = None, None\n        if right is True:\n            if R is not None and force is False:\n                print('sam file is not sorted', file = sys.stderr)\n                print('\\te.g.: %s' % (line[0]), file = sys.stderr)\n                exit()\n            if R is not None:\n                R = None\n                continue\n            R = ['@%s' % line[0], seq, '+%s' % line[0], qual]\n            if L is not None:\n                yield L\n                yield R\n                L, R = None, None&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def sam2fastq(sam, singles = False, force = False):\n    \&quot;\&quot;\&quot;\n    convert sam to fastq\n    \&quot;\&quot;\&quot;\n    L, R = None, None\n    for line in sam:\n        if line.startswith('@') is True:\n            continue\n        line = line.strip().split()\n        bit = [True if i == '1' else False \\\n                for i in bin(int(line[1])).split('b')[1][::-1]]\n        while len(bit) < 8:\n            bit.append(False)\n        pair, proper, na, nap, rev, mrev, left, right = bit\n        # make sure read is paired\n        if pair is False:\n            if singles is True:\n                print_single(line, rev)\n            continue\n        # check if sequence is reverse-complemented\n        if rev is True:\n            seq = rc(['', line[9]])[1]\n            qual = line[10][::-1]\n        else:\n            seq = line[9]\n            qual = line[10]\n        # check if read is forward or reverse, return when both have been found\n        if left is True:\n            if L is not None and force is False:\n                print('sam file is not sorted', file = sys.stderr)\n                print('\\te.g.: %s' % (line[0]), file = sys.stderr)\n                exit()\n            if L is not None:\n                L = None\n                continue\n            L = ['@%s' % line[0], seq, '+%s' % line[0], qual]\n            if R is not None:\n                yield L\n                yield R\n                L, R = None, None\n        if right is True:\n            if R is not None and force is False:\n                print('sam file is not sorted', file = sys.stderr)\n                print('\\te.g.: %s' % (line[0]), file = sys.stderr)\n                exit()\n            if R is not None:\n                R = None\n                continue\n            R = ['@%s' % line[0], seq, '+%s' % line[0], qual]\n            if L is not None:\n                yield L\n                yield R\n                L, R = None, None&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;sam2fastq&quot;,&quot;(&quot;,&quot;sam&quot;,&quot;,&quot;,&quot;singles&quot;,&quot;=&quot;,&quot;False&quot;,&quot;,&quot;,&quot;force&quot;,&quot;=&quot;,&quot;False&quot;,&quot;)&quot;,&quot;:&quot;,&quot;L&quot;,&quot;,&quot;,&quot;R&quot;,&quot;=&quot;,&quot;None&quot;,&quot;,&quot;,&quot;None&quot;,&quot;for&quot;,&quot;line&quot;,&quot;in&quot;,&quot;sam&quot;,&quot;:&quot;,&quot;if&quot;,&quot;line&quot;,&quot;.&quot;,&quot;startswith&quot;,&quot;(&quot;,&quot;'@'&quot;,&quot;)&quot;,&quot;is&quot;,&quot;True&quot;,&quot;:&quot;,&quot;continue&quot;,&quot;line&quot;,&quot;=&quot;,&quot;line&quot;,&quot;.&quot;,&quot;strip&quot;,&quot;(&quot;,&quot;)&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;)&quot;,&quot;bit&quot;,&quot;=&quot;,&quot;[&quot;,&quot;True&quot;,&quot;if&quot;,&quot;i&quot;,&quot;==&quot;,&quot;'1'&quot;,&quot;else&quot;,&quot;False&quot;,&quot;for&quot;,&quot;i&quot;,&quot;in&quot;,&quot;bin&quot;,&quot;(&quot;,&quot;int&quot;,&quot;(&quot;,&quot;line&quot;,&quot;[&quot;,&quot;1&quot;,&quot;]&quot;,&quot;)&quot;,&quot;)&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;'b'&quot;,&quot;)&quot;,&quot;[&quot;,&quot;1&quot;,&quot;]&quot;,&quot;[&quot;,&quot;:&quot;,&quot;:&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;,&quot;]&quot;,&quot;while&quot;,&quot;len&quot;,&quot;(&quot;,&quot;bit&quot;,&quot;)&quot;,&quot;<&quot;,&quot;8&quot;,&quot;:&quot;,&quot;bit&quot;,&quot;.&quot;,&quot;append&quot;,&quot;(&quot;,&quot;False&quot;,&quot;)&quot;,&quot;pair&quot;,&quot;,&quot;,&quot;proper&quot;,&quot;,&quot;,&quot;na&quot;,&quot;,&quot;,&quot;nap&quot;,&quot;,&quot;,&quot;rev&quot;,&quot;,&quot;,&quot;mrev&quot;,&quot;,&quot;,&quot;left&quot;,&quot;,&quot;,&quot;right&quot;,&quot;=&quot;,&quot;bit&quot;,&quot;# make sure read is paired&quot;,&quot;if&quot;,&quot;pair&quot;,&quot;is&quot;,&quot;False&quot;,&quot;:&quot;,&quot;if&quot;,&quot;singles&quot;,&quot;is&quot;,&quot;True&quot;,&quot;:&quot;,&quot;print_single&quot;,&quot;(&quot;,&quot;line&quot;,&quot;,&quot;,&quot;rev&quot;,&quot;)&quot;,&quot;continue&quot;,&quot;# check if sequence is reverse-complemented&quot;,&quot;if&quot;,&quot;rev&quot;,&quot;is&quot;,&quot;True&quot;,&quot;:&quot;,&quot;seq&quot;,&quot;=&quot;,&quot;rc&quot;,&quot;(&quot;,&quot;[&quot;,&quot;''&quot;,&quot;,&quot;,&quot;line&quot;,&quot;[&quot;,&quot;9&quot;,&quot;]&quot;,&quot;]&quot;,&quot;)&quot;,&quot;[&quot;,&quot;1&quot;,&quot;]&quot;,&quot;qual&quot;,&quot;=&quot;,&quot;line&quot;,&quot;[&quot;,&quot;10&quot;,&quot;]&quot;,&quot;[&quot;,&quot;:&quot;,&quot;:&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;,&quot;else&quot;,&quot;:&quot;,&quot;seq&quot;,&quot;=&quot;,&quot;line&quot;,&quot;[&quot;,&quot;9&quot;,&quot;]&quot;,&quot;qual&quot;,&quot;=&quot;,&quot;line&quot;,&quot;[&quot;,&quot;10&quot;,&quot;]&quot;,&quot;# check if read is forward or reverse, return when both have been found&quot;,&quot;if&quot;,&quot;left&quot;,&quot;is&quot;,&quot;True&quot;,&quot;:&quot;,&quot;if&quot;,&quot;L&quot;,&quot;is&quot;,&quot;not&quot;,&quot;None&quot;,&quot;and&quot;,&quot;force&quot;,&quot;is&quot;,&quot;False&quot;,&quot;:&quot;,&quot;print&quot;,&quot;(&quot;,&quot;'sam file is not sorted'&quot;,&quot;,&quot;,&quot;file&quot;,&quot;=&quot;,&quot;sys&quot;,&quot;.&quot;,&quot;stderr&quot;,&quot;)&quot;,&quot;print&quot;,&quot;(&quot;,&quot;'\\te.g.: %s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;line&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;)&quot;,&quot;,&quot;,&quot;file&quot;,&quot;=&quot;,&quot;sys&quot;,&quot;.&quot;,&quot;stderr&quot;,&quot;)&quot;,&quot;exit&quot;,&quot;(&quot;,&quot;)&quot;,&quot;if&quot;,&quot;L&quot;,&quot;is&quot;,&quot;not&quot;,&quot;None&quot;,&quot;:&quot;,&quot;L&quot;,&quot;=&quot;,&quot;None&quot;,&quot;continue&quot;,&quot;L&quot;,&quot;=&quot;,&quot;[&quot;,&quot;'@%s'&quot;,&quot;%&quot;,&quot;line&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;,&quot;,&quot;seq&quot;,&quot;,&quot;,&quot;'+%s'&quot;,&quot;%&quot;,&quot;line&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;,&quot;,&quot;qual&quot;,&quot;]&quot;,&quot;if&quot;,&quot;R&quot;,&quot;is&quot;,&quot;not&quot;,&quot;None&quot;,&quot;:&quot;,&quot;yield&quot;,&quot;L&quot;,&quot;yield&quot;,&quot;R&quot;,&quot;L&quot;,&quot;,&quot;,&quot;R&quot;,&quot;=&quot;,&quot;None&quot;,&quot;,&quot;,&quot;None&quot;,&quot;if&quot;,&quot;right&quot;,&quot;is&quot;,&quot;True&quot;,&quot;:&quot;,&quot;if&quot;,&quot;R&quot;,&quot;is&quot;,&quot;not&quot;,&quot;None&quot;,&quot;and&quot;,&quot;force&quot;,&quot;is&quot;,&quot;False&quot;,&quot;:&quot;,&quot;print&quot;,&quot;(&quot;,&quot;'sam file is not sorted'&quot;,&quot;,&quot;,&quot;file&quot;,&quot;=&quot;,&quot;sys&quot;,&quot;.&quot;,&quot;stderr&quot;,&quot;)&quot;,&quot;print&quot;,&quot;(&quot;,&quot;'\\te.g.: %s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;line&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;)&quot;,&quot;,&quot;,&quot;file&quot;,&quot;=&quot;,&quot;sys&quot;,&quot;.&quot;,&quot;stderr&quot;,&quot;)&quot;,&quot;exit&quot;,&quot;(&quot;,&quot;)&quot;,&quot;if&quot;,&quot;R&quot;,&quot;is&quot;,&quot;not&quot;,&quot;None&quot;,&quot;:&quot;,&quot;R&quot;,&quot;=&quot;,&quot;None&quot;,&quot;continue&quot;,&quot;R&quot;,&quot;=&quot;,&quot;[&quot;,&quot;'@%s'&quot;,&quot;%&quot;,&quot;line&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;,&quot;,&quot;seq&quot;,&quot;,&quot;,&quot;'+%s'&quot;,&quot;%&quot;,&quot;line&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;,&quot;,&quot;qual&quot;,&quot;]&quot;,&quot;if&quot;,&quot;L&quot;,&quot;is&quot;,&quot;not&quot;,&quot;None&quot;,&quot;:&quot;,&quot;yield&quot;,&quot;L&quot;,&quot;yield&quot;,&quot;R&quot;,&quot;L&quot;,&quot;,&quot;,&quot;R&quot;,&quot;=&quot;,&quot;None&quot;,&quot;,&quot;,&quot;None&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;sam2fastq\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;,\&quot;,\n  \&quot;singles\&quot;,\n  \&quot;=\&quot;,\n  \&quot;False\&quot;,\n  \&quot;,\&quot;,\n  \&quot;force\&quot;,\n  \&quot;=\&quot;,\n  \&quot;False\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;L\&quot;,\n  \&quot;,\&quot;,\n  \&quot;R\&quot;,\n  \&quot;=\&quot;,\n  \&quot;None\&quot;,\n  \&quot;,\&quot;,\n  \&quot;None\&quot;,\n  \&quot;for\&quot;,\n  \&quot;line\&quot;,\n  \&quot;in\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;line\&quot;,\n  \&quot;.\&quot;,\n  \&quot;startswith\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'@'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;is\&quot;,\n  \&quot;True\&quot;,\n  \&quot;:\&quot;,\n  \&quot;continue\&quot;,\n  \&quot;line\&quot;,\n  \&quot;=\&quot;,\n  \&quot;line\&quot;,\n  \&quot;.\&quot;,\n  \&quot;strip\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;bit\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;True\&quot;,\n  \&quot;if\&quot;,\n  \&quot;i\&quot;,\n  \&quot;==\&quot;,\n  \&quot;'1'\&quot;,\n  \&quot;else\&quot;,\n  \&quot;False\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;in\&quot;,\n  \&quot;bin\&quot;,\n  \&quot;(\&quot;,\n  \&quot;int\&quot;,\n  \&quot;(\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'b'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;:\&quot;,\n  \&quot;:\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;]\&quot;,\n  \&quot;while\&quot;,\n  \&quot;len\&quot;,\n  \&quot;(\&quot;,\n  \&quot;bit\&quot;,\n  \&quot;)\&quot;,\n  \&quot;<\&quot;,\n  \&quot;8\&quot;,\n  \&quot;:\&quot;,\n  \&quot;bit\&quot;,\n  \&quot;.\&quot;,\n  \&quot;append\&quot;,\n  \&quot;(\&quot;,\n  \&quot;False\&quot;,\n  \&quot;)\&quot;,\n  \&quot;pair\&quot;,\n  \&quot;,\&quot;,\n  \&quot;proper\&quot;,\n  \&quot;,\&quot;,\n  \&quot;na\&quot;,\n  \&quot;,\&quot;,\n  \&quot;nap\&quot;,\n  \&quot;,\&quot;,\n  \&quot;rev\&quot;,\n  \&quot;,\&quot;,\n  \&quot;mrev\&quot;,\n  \&quot;,\&quot;,\n  \&quot;left\&quot;,\n  \&quot;,\&quot;,\n  \&quot;right\&quot;,\n  \&quot;=\&quot;,\n  \&quot;bit\&quot;,\n  \&quot;# make sure read is paired\&quot;,\n  \&quot;if\&quot;,\n  \&quot;pair\&quot;,\n  \&quot;is\&quot;,\n  \&quot;False\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;singles\&quot;,\n  \&quot;is\&quot;,\n  \&quot;True\&quot;,\n  \&quot;:\&quot;,\n  \&quot;print_single\&quot;,\n  \&quot;(\&quot;,\n  \&quot;line\&quot;,\n  \&quot;,\&quot;,\n  \&quot;rev\&quot;,\n  \&quot;)\&quot;,\n  \&quot;continue\&quot;,\n  \&quot;# check if sequence is reverse-complemented\&quot;,\n  \&quot;if\&quot;,\n  \&quot;rev\&quot;,\n  \&quot;is\&quot;,\n  \&quot;True\&quot;,\n  \&quot;:\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;=\&quot;,\n  \&quot;rc\&quot;,\n  \&quot;(\&quot;,\n  \&quot;[\&quot;,\n  \&quot;''\&quot;,\n  \&quot;,\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;9\&quot;,\n  \&quot;]\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;qual\&quot;,\n  \&quot;=\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;10\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;:\&quot;,\n  \&quot;:\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;=\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;9\&quot;,\n  \&quot;]\&quot;,\n  \&quot;qual\&quot;,\n  \&quot;=\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;10\&quot;,\n  \&quot;]\&quot;,\n  \&quot;# check if read is forward or reverse, return when both have been found\&quot;,\n  \&quot;if\&quot;,\n  \&quot;left\&quot;,\n  \&quot;is\&quot;,\n  \&quot;True\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;L\&quot;,\n  \&quot;is\&quot;,\n  \&quot;not\&quot;,\n  \&quot;None\&quot;,\n  \&quot;and\&quot;,\n  \&quot;force\&quot;,\n  \&quot;is\&quot;,\n  \&quot;False\&quot;,\n  \&quot;:\&quot;,\n  \&quot;print\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'sam file is not sorted'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;file\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sys\&quot;,\n  \&quot;.\&quot;,\n  \&quot;stderr\&quot;,\n  \&quot;)\&quot;,\n  \&quot;print\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'\\\\te.g.: %s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;file\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sys\&quot;,\n  \&quot;.\&quot;,\n  \&quot;stderr\&quot;,\n  \&quot;)\&quot;,\n  \&quot;exit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;L\&quot;,\n  \&quot;is\&quot;,\n  \&quot;not\&quot;,\n  \&quot;None\&quot;,\n  \&quot;:\&quot;,\n  \&quot;L\&quot;,\n  \&quot;=\&quot;,\n  \&quot;None\&quot;,\n  \&quot;continue\&quot;,\n  \&quot;L\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'@%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'+%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;qual\&quot;,\n  \&quot;]\&quot;,\n  \&quot;if\&quot;,\n  \&quot;R\&quot;,\n  \&quot;is\&quot;,\n  \&quot;not\&quot;,\n  \&quot;None\&quot;,\n  \&quot;:\&quot;,\n  \&quot;yield\&quot;,\n  \&quot;L\&quot;,\n  \&quot;yield\&quot;,\n  \&quot;R\&quot;,\n  \&quot;L\&quot;,\n  \&quot;,\&quot;,\n  \&quot;R\&quot;,\n  \&quot;=\&quot;,\n  \&quot;None\&quot;,\n  \&quot;,\&quot;,\n  \&quot;None\&quot;,\n  \&quot;if\&quot;,\n  \&quot;right\&quot;,\n  \&quot;is\&quot;,\n  \&quot;True\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;R\&quot;,\n  \&quot;is\&quot;,\n  \&quot;not\&quot;,\n  \&quot;None\&quot;,\n  \&quot;and\&quot;,\n  \&quot;force\&quot;,\n  \&quot;is\&quot;,\n  \&quot;False\&quot;,\n  \&quot;:\&quot;,\n  \&quot;print\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'sam file is not sorted'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;file\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sys\&quot;,\n  \&quot;.\&quot;,\n  \&quot;stderr\&quot;,\n  \&quot;)\&quot;,\n  \&quot;print\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'\\\\te.g.: %s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;file\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sys\&quot;,\n  \&quot;.\&quot;,\n  \&quot;stderr\&quot;,\n  \&quot;)\&quot;,\n  \&quot;exit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;R\&quot;,\n  \&quot;is\&quot;,\n  \&quot;not\&quot;,\n  \&quot;None\&quot;,\n  \&quot;:\&quot;,\n  \&quot;R\&quot;,\n  \&quot;=\&quot;,\n  \&quot;None\&quot;,\n  \&quot;continue\&quot;,\n  \&quot;R\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'@%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'+%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;qual\&quot;,\n  \&quot;]\&quot;,\n  \&quot;if\&quot;,\n  \&quot;L\&quot;,\n  \&quot;is\&quot;,\n  \&quot;not\&quot;,\n  \&quot;None\&quot;,\n  \&quot;:\&quot;,\n  \&quot;yield\&quot;,\n  \&quot;L\&quot;,\n  \&quot;yield\&quot;,\n  \&quot;R\&quot;,\n  \&quot;L\&quot;,\n  \&quot;,\&quot;,\n  \&quot;R\&quot;,\n  \&quot;=\&quot;,\n  \&quot;None\&quot;,\n  \&quot;,\&quot;,\n  \&quot;None\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;convert sam to fastq&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;convert&quot;,&quot;sam&quot;,&quot;to&quot;,&quot;fastq&quot;],&quot;string&quot;:&quot;[\n  \&quot;convert\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;to\&quot;,\n  \&quot;fastq\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;83b2566b3a5745437ec651cd6cafddd056846240&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/sam2fastq.py#L26-L78&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:28,&quot;string&quot;:&quot;28&quot;}}},{&quot;rowIdx&quot;:29,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;christophertbrown/bioscripts&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ctbBio/subset_sam.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;sort_sam&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def sort_sam(sam, sort):\n    \&quot;\&quot;\&quot;\n    sort sam file\n    \&quot;\&quot;\&quot;\n    tempdir = '%s/' % (os.path.abspath(sam).rsplit('/', 1)[0])\n    if sort is True:\n        mapping = '%s.sorted.sam' % (sam.rsplit('.', 1)[0])\n        if sam != '-':\n            if os.path.exists(mapping) is False:\n                os.system(\&quot;\\\n                    sort -k1 --buffer-size=%sG -T %s -o %s %s\\\n                    \&quot; % (sbuffer, tempdir, mapping, sam)) \n        else:\n            mapping = 'stdin-sam.sorted.sam'\n            p = Popen(\&quot;sort -k1 --buffer-size=%sG -T %s -o %s\&quot; \\\n                    % (sbuffer, tempdir, mapping), stdin = sys.stdin, shell = True) \n            p.communicate()\n        mapping = open(mapping)\n    else:\n        if sam == '-':\n            mapping = sys.stdin\n        else:\n            mapping = open(sam)\n    return mapping&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def sort_sam(sam, sort):\n    \&quot;\&quot;\&quot;\n    sort sam file\n    \&quot;\&quot;\&quot;\n    tempdir = '%s/' % (os.path.abspath(sam).rsplit('/', 1)[0])\n    if sort is True:\n        mapping = '%s.sorted.sam' % (sam.rsplit('.', 1)[0])\n        if sam != '-':\n            if os.path.exists(mapping) is False:\n                os.system(\&quot;\\\n                    sort -k1 --buffer-size=%sG -T %s -o %s %s\\\n                    \&quot; % (sbuffer, tempdir, mapping, sam)) \n        else:\n            mapping = 'stdin-sam.sorted.sam'\n            p = Popen(\&quot;sort -k1 --buffer-size=%sG -T %s -o %s\&quot; \\\n                    % (sbuffer, tempdir, mapping), stdin = sys.stdin, shell = True) \n            p.communicate()\n        mapping = open(mapping)\n    else:\n        if sam == '-':\n            mapping = sys.stdin\n        else:\n            mapping = open(sam)\n    return mapping&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;sort_sam&quot;,&quot;(&quot;,&quot;sam&quot;,&quot;,&quot;,&quot;sort&quot;,&quot;)&quot;,&quot;:&quot;,&quot;tempdir&quot;,&quot;=&quot;,&quot;'%s/'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;os&quot;,&quot;.&quot;,&quot;path&quot;,&quot;.&quot;,&quot;abspath&quot;,&quot;(&quot;,&quot;sam&quot;,&quot;)&quot;,&quot;.&quot;,&quot;rsplit&quot;,&quot;(&quot;,&quot;'/'&quot;,&quot;,&quot;,&quot;1&quot;,&quot;)&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;)&quot;,&quot;if&quot;,&quot;sort&quot;,&quot;is&quot;,&quot;True&quot;,&quot;:&quot;,&quot;mapping&quot;,&quot;=&quot;,&quot;'%s.sorted.sam'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;sam&quot;,&quot;.&quot;,&quot;rsplit&quot;,&quot;(&quot;,&quot;'.'&quot;,&quot;,&quot;,&quot;1&quot;,&quot;)&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;)&quot;,&quot;if&quot;,&quot;sam&quot;,&quot;!=&quot;,&quot;'-'&quot;,&quot;:&quot;,&quot;if&quot;,&quot;os&quot;,&quot;.&quot;,&quot;path&quot;,&quot;.&quot;,&quot;exists&quot;,&quot;(&quot;,&quot;mapping&quot;,&quot;)&quot;,&quot;is&quot;,&quot;False&quot;,&quot;:&quot;,&quot;os&quot;,&quot;.&quot;,&quot;system&quot;,&quot;(&quot;,&quot;\&quot;\\\n                    sort -k1 --buffer-size=%sG -T %s -o %s %s\\\n                    \&quot;&quot;,&quot;%&quot;,&quot;(&quot;,&quot;sbuffer&quot;,&quot;,&quot;,&quot;tempdir&quot;,&quot;,&quot;,&quot;mapping&quot;,&quot;,&quot;,&quot;sam&quot;,&quot;)&quot;,&quot;)&quot;,&quot;else&quot;,&quot;:&quot;,&quot;mapping&quot;,&quot;=&quot;,&quot;'stdin-sam.sorted.sam'&quot;,&quot;p&quot;,&quot;=&quot;,&quot;Popen&quot;,&quot;(&quot;,&quot;\&quot;sort -k1 --buffer-size=%sG -T %s -o %s\&quot;&quot;,&quot;%&quot;,&quot;(&quot;,&quot;sbuffer&quot;,&quot;,&quot;,&quot;tempdir&quot;,&quot;,&quot;,&quot;mapping&quot;,&quot;)&quot;,&quot;,&quot;,&quot;stdin&quot;,&quot;=&quot;,&quot;sys&quot;,&quot;.&quot;,&quot;stdin&quot;,&quot;,&quot;,&quot;shell&quot;,&quot;=&quot;,&quot;True&quot;,&quot;)&quot;,&quot;p&quot;,&quot;.&quot;,&quot;communicate&quot;,&quot;(&quot;,&quot;)&quot;,&quot;mapping&quot;,&quot;=&quot;,&quot;open&quot;,&quot;(&quot;,&quot;mapping&quot;,&quot;)&quot;,&quot;else&quot;,&quot;:&quot;,&quot;if&quot;,&quot;sam&quot;,&quot;==&quot;,&quot;'-'&quot;,&quot;:&quot;,&quot;mapping&quot;,&quot;=&quot;,&quot;sys&quot;,&quot;.&quot;,&quot;stdin&quot;,&quot;else&quot;,&quot;:&quot;,&quot;mapping&quot;,&quot;=&quot;,&quot;open&quot;,&quot;(&quot;,&quot;sam&quot;,&quot;)&quot;,&quot;return&quot;,&quot;mapping&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;sort_sam\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;,\&quot;,\n  \&quot;sort\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;tempdir\&quot;,\n  \&quot;=\&quot;,\n  \&quot;'%s/'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;path\&quot;,\n  \&quot;.\&quot;,\n  \&quot;abspath\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;)\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rsplit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'/'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;sort\&quot;,\n  \&quot;is\&quot;,\n  \&quot;True\&quot;,\n  \&quot;:\&quot;,\n  \&quot;mapping\&quot;,\n  \&quot;=\&quot;,\n  \&quot;'%s.sorted.sam'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rsplit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'.'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;!=\&quot;,\n  \&quot;'-'\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;path\&quot;,\n  \&quot;.\&quot;,\n  \&quot;exists\&quot;,\n  \&quot;(\&quot;,\n  \&quot;mapping\&quot;,\n  \&quot;)\&quot;,\n  \&quot;is\&quot;,\n  \&quot;False\&quot;,\n  \&quot;:\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;system\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;\\\\\\n                    sort -k1 --buffer-size=%sG -T %s -o %s %s\\\\\\n                    \\\&quot;\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sbuffer\&quot;,\n  \&quot;,\&quot;,\n  \&quot;tempdir\&quot;,\n  \&quot;,\&quot;,\n  \&quot;mapping\&quot;,\n  \&quot;,\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;mapping\&quot;,\n  \&quot;=\&quot;,\n  \&quot;'stdin-sam.sorted.sam'\&quot;,\n  \&quot;p\&quot;,\n  \&quot;=\&quot;,\n  \&quot;Popen\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;sort -k1 --buffer-size=%sG -T %s -o %s\\\&quot;\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sbuffer\&quot;,\n  \&quot;,\&quot;,\n  \&quot;tempdir\&quot;,\n  \&quot;,\&quot;,\n  \&quot;mapping\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;stdin\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sys\&quot;,\n  \&quot;.\&quot;,\n  \&quot;stdin\&quot;,\n  \&quot;,\&quot;,\n  \&quot;shell\&quot;,\n  \&quot;=\&quot;,\n  \&quot;True\&quot;,\n  \&quot;)\&quot;,\n  \&quot;p\&quot;,\n  \&quot;.\&quot;,\n  \&quot;communicate\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;mapping\&quot;,\n  \&quot;=\&quot;,\n  \&quot;open\&quot;,\n  \&quot;(\&quot;,\n  \&quot;mapping\&quot;,\n  \&quot;)\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;==\&quot;,\n  \&quot;'-'\&quot;,\n  \&quot;:\&quot;,\n  \&quot;mapping\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sys\&quot;,\n  \&quot;.\&quot;,\n  \&quot;stdin\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;mapping\&quot;,\n  \&quot;=\&quot;,\n  \&quot;open\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;mapping\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;sort sam file&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;sort&quot;,&quot;sam&quot;,&quot;file&quot;],&quot;string&quot;:&quot;[\n  \&quot;sort\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;file\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;83b2566b3a5745437ec651cd6cafddd056846240&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/subset_sam.py#L14-L37&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:29,&quot;string&quot;:&quot;29&quot;}}},{&quot;rowIdx&quot;:30,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;christophertbrown/bioscripts&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ctbBio/subset_sam.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;sub_sam&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def sub_sam(sam, percent, sort = True, sbuffer = False):\n    \&quot;\&quot;\&quot;\n    randomly subset sam file\n    \&quot;\&quot;\&quot;\n    mapping = sort_sam(sam, sort)\n    pool = [1 for i in range(0, percent)] + [0 for i in range(0, 100 - percent)]\n    c = cycle([1, 2])\n    for line in mapping:\n        line = line.strip().split()\n        if line[0].startswith('@'): # get the sam header\n            yield line\n            continue\n        if int(line[1]) <= 20: # is this from a single read?\n            if random.choice(pool) == 1:\n                yield line\n        else:\n            n = next(c)\n            if n == 1:\n                prev = line\n            if n == 2 and random.choice(pool) == 1:\n                yield prev\n                yield line&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def sub_sam(sam, percent, sort = True, sbuffer = False):\n    \&quot;\&quot;\&quot;\n    randomly subset sam file\n    \&quot;\&quot;\&quot;\n    mapping = sort_sam(sam, sort)\n    pool = [1 for i in range(0, percent)] + [0 for i in range(0, 100 - percent)]\n    c = cycle([1, 2])\n    for line in mapping:\n        line = line.strip().split()\n        if line[0].startswith('@'): # get the sam header\n            yield line\n            continue\n        if int(line[1]) <= 20: # is this from a single read?\n            if random.choice(pool) == 1:\n                yield line\n        else:\n            n = next(c)\n            if n == 1:\n                prev = line\n            if n == 2 and random.choice(pool) == 1:\n                yield prev\n                yield line&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;sub_sam&quot;,&quot;(&quot;,&quot;sam&quot;,&quot;,&quot;,&quot;percent&quot;,&quot;,&quot;,&quot;sort&quot;,&quot;=&quot;,&quot;True&quot;,&quot;,&quot;,&quot;sbuffer&quot;,&quot;=&quot;,&quot;False&quot;,&quot;)&quot;,&quot;:&quot;,&quot;mapping&quot;,&quot;=&quot;,&quot;sort_sam&quot;,&quot;(&quot;,&quot;sam&quot;,&quot;,&quot;,&quot;sort&quot;,&quot;)&quot;,&quot;pool&quot;,&quot;=&quot;,&quot;[&quot;,&quot;1&quot;,&quot;for&quot;,&quot;i&quot;,&quot;in&quot;,&quot;range&quot;,&quot;(&quot;,&quot;0&quot;,&quot;,&quot;,&quot;percent&quot;,&quot;)&quot;,&quot;]&quot;,&quot;+&quot;,&quot;[&quot;,&quot;0&quot;,&quot;for&quot;,&quot;i&quot;,&quot;in&quot;,&quot;range&quot;,&quot;(&quot;,&quot;0&quot;,&quot;,&quot;,&quot;100&quot;,&quot;-&quot;,&quot;percent&quot;,&quot;)&quot;,&quot;]&quot;,&quot;c&quot;,&quot;=&quot;,&quot;cycle&quot;,&quot;(&quot;,&quot;[&quot;,&quot;1&quot;,&quot;,&quot;,&quot;2&quot;,&quot;]&quot;,&quot;)&quot;,&quot;for&quot;,&quot;line&quot;,&quot;in&quot;,&quot;mapping&quot;,&quot;:&quot;,&quot;line&quot;,&quot;=&quot;,&quot;line&quot;,&quot;.&quot;,&quot;strip&quot;,&quot;(&quot;,&quot;)&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;)&quot;,&quot;if&quot;,&quot;line&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;.&quot;,&quot;startswith&quot;,&quot;(&quot;,&quot;'@'&quot;,&quot;)&quot;,&quot;:&quot;,&quot;# get the sam header&quot;,&quot;yield&quot;,&quot;line&quot;,&quot;continue&quot;,&quot;if&quot;,&quot;int&quot;,&quot;(&quot;,&quot;line&quot;,&quot;[&quot;,&quot;1&quot;,&quot;]&quot;,&quot;)&quot;,&quot;<=&quot;,&quot;20&quot;,&quot;:&quot;,&quot;# is this from a single read?&quot;,&quot;if&quot;,&quot;random&quot;,&quot;.&quot;,&quot;choice&quot;,&quot;(&quot;,&quot;pool&quot;,&quot;)&quot;,&quot;==&quot;,&quot;1&quot;,&quot;:&quot;,&quot;yield&quot;,&quot;line&quot;,&quot;else&quot;,&quot;:&quot;,&quot;n&quot;,&quot;=&quot;,&quot;next&quot;,&quot;(&quot;,&quot;c&quot;,&quot;)&quot;,&quot;if&quot;,&quot;n&quot;,&quot;==&quot;,&quot;1&quot;,&quot;:&quot;,&quot;prev&quot;,&quot;=&quot;,&quot;line&quot;,&quot;if&quot;,&quot;n&quot;,&quot;==&quot;,&quot;2&quot;,&quot;and&quot;,&quot;random&quot;,&quot;.&quot;,&quot;choice&quot;,&quot;(&quot;,&quot;pool&quot;,&quot;)&quot;,&quot;==&quot;,&quot;1&quot;,&quot;:&quot;,&quot;yield&quot;,&quot;prev&quot;,&quot;yield&quot;,&quot;line&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;sub_sam\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;,\&quot;,\n  \&quot;percent\&quot;,\n  \&quot;,\&quot;,\n  \&quot;sort\&quot;,\n  \&quot;=\&quot;,\n  \&quot;True\&quot;,\n  \&quot;,\&quot;,\n  \&quot;sbuffer\&quot;,\n  \&quot;=\&quot;,\n  \&quot;False\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;mapping\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sort_sam\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;,\&quot;,\n  \&quot;sort\&quot;,\n  \&quot;)\&quot;,\n  \&quot;pool\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;in\&quot;,\n  \&quot;range\&quot;,\n  \&quot;(\&quot;,\n  \&quot;0\&quot;,\n  \&quot;,\&quot;,\n  \&quot;percent\&quot;,\n  \&quot;)\&quot;,\n  \&quot;]\&quot;,\n  \&quot;+\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;in\&quot;,\n  \&quot;range\&quot;,\n  \&quot;(\&quot;,\n  \&quot;0\&quot;,\n  \&quot;,\&quot;,\n  \&quot;100\&quot;,\n  \&quot;-\&quot;,\n  \&quot;percent\&quot;,\n  \&quot;)\&quot;,\n  \&quot;]\&quot;,\n  \&quot;c\&quot;,\n  \&quot;=\&quot;,\n  \&quot;cycle\&quot;,\n  \&quot;(\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;,\&quot;,\n  \&quot;2\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;line\&quot;,\n  \&quot;in\&quot;,\n  \&quot;mapping\&quot;,\n  \&quot;:\&quot;,\n  \&quot;line\&quot;,\n  \&quot;=\&quot;,\n  \&quot;line\&quot;,\n  \&quot;.\&quot;,\n  \&quot;strip\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;startswith\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'@'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;# get the sam header\&quot;,\n  \&quot;yield\&quot;,\n  \&quot;line\&quot;,\n  \&quot;continue\&quot;,\n  \&quot;if\&quot;,\n  \&quot;int\&quot;,\n  \&quot;(\&quot;,\n  \&quot;line\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;<=\&quot;,\n  \&quot;20\&quot;,\n  \&quot;:\&quot;,\n  \&quot;# is this from a single read?\&quot;,\n  \&quot;if\&quot;,\n  \&quot;random\&quot;,\n  \&quot;.\&quot;,\n  \&quot;choice\&quot;,\n  \&quot;(\&quot;,\n  \&quot;pool\&quot;,\n  \&quot;)\&quot;,\n  \&quot;==\&quot;,\n  \&quot;1\&quot;,\n  \&quot;:\&quot;,\n  \&quot;yield\&quot;,\n  \&quot;line\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;n\&quot;,\n  \&quot;=\&quot;,\n  \&quot;next\&quot;,\n  \&quot;(\&quot;,\n  \&quot;c\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;n\&quot;,\n  \&quot;==\&quot;,\n  \&quot;1\&quot;,\n  \&quot;:\&quot;,\n  \&quot;prev\&quot;,\n  \&quot;=\&quot;,\n  \&quot;line\&quot;,\n  \&quot;if\&quot;,\n  \&quot;n\&quot;,\n  \&quot;==\&quot;,\n  \&quot;2\&quot;,\n  \&quot;and\&quot;,\n  \&quot;random\&quot;,\n  \&quot;.\&quot;,\n  \&quot;choice\&quot;,\n  \&quot;(\&quot;,\n  \&quot;pool\&quot;,\n  \&quot;)\&quot;,\n  \&quot;==\&quot;,\n  \&quot;1\&quot;,\n  \&quot;:\&quot;,\n  \&quot;yield\&quot;,\n  \&quot;prev\&quot;,\n  \&quot;yield\&quot;,\n  \&quot;line\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;randomly subset sam file&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;randomly&quot;,&quot;subset&quot;,&quot;sam&quot;,&quot;file&quot;],&quot;string&quot;:&quot;[\n  \&quot;randomly\&quot;,\n  \&quot;subset\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;file\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;83b2566b3a5745437ec651cd6cafddd056846240&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/subset_sam.py#L39-L60&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:30,&quot;string&quot;:&quot;30&quot;}}},{&quot;rowIdx&quot;:31,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;christophertbrown/bioscripts&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ctbBio/fastq2fasta.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;fq2fa&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def fq2fa(fq):\n    \&quot;\&quot;\&quot;\n    convert fq to fa\n    \&quot;\&quot;\&quot;\n    c = cycle([1, 2, 3, 4])\n    for line in fq:\n        n = next(c)\n        if n == 1:\n            seq = ['>%s' % (line.strip().split('@', 1)[1])]\n        if n == 2:\n            seq.append(line.strip())\n            yield seq&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def fq2fa(fq):\n    \&quot;\&quot;\&quot;\n    convert fq to fa\n    \&quot;\&quot;\&quot;\n    c = cycle([1, 2, 3, 4])\n    for line in fq:\n        n = next(c)\n        if n == 1:\n            seq = ['>%s' % (line.strip().split('@', 1)[1])]\n        if n == 2:\n            seq.append(line.strip())\n            yield seq&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;fq2fa&quot;,&quot;(&quot;,&quot;fq&quot;,&quot;)&quot;,&quot;:&quot;,&quot;c&quot;,&quot;=&quot;,&quot;cycle&quot;,&quot;(&quot;,&quot;[&quot;,&quot;1&quot;,&quot;,&quot;,&quot;2&quot;,&quot;,&quot;,&quot;3&quot;,&quot;,&quot;,&quot;4&quot;,&quot;]&quot;,&quot;)&quot;,&quot;for&quot;,&quot;line&quot;,&quot;in&quot;,&quot;fq&quot;,&quot;:&quot;,&quot;n&quot;,&quot;=&quot;,&quot;next&quot;,&quot;(&quot;,&quot;c&quot;,&quot;)&quot;,&quot;if&quot;,&quot;n&quot;,&quot;==&quot;,&quot;1&quot;,&quot;:&quot;,&quot;seq&quot;,&quot;=&quot;,&quot;[&quot;,&quot;'>%s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;line&quot;,&quot;.&quot;,&quot;strip&quot;,&quot;(&quot;,&quot;)&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;'@'&quot;,&quot;,&quot;,&quot;1&quot;,&quot;)&quot;,&quot;[&quot;,&quot;1&quot;,&quot;]&quot;,&quot;)&quot;,&quot;]&quot;,&quot;if&quot;,&quot;n&quot;,&quot;==&quot;,&quot;2&quot;,&quot;:&quot;,&quot;seq&quot;,&quot;.&quot;,&quot;append&quot;,&quot;(&quot;,&quot;line&quot;,&quot;.&quot;,&quot;strip&quot;,&quot;(&quot;,&quot;)&quot;,&quot;)&quot;,&quot;yield&quot;,&quot;seq&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;fq2fa\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fq\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;c\&quot;,\n  \&quot;=\&quot;,\n  \&quot;cycle\&quot;,\n  \&quot;(\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;,\&quot;,\n  \&quot;2\&quot;,\n  \&quot;,\&quot;,\n  \&quot;3\&quot;,\n  \&quot;,\&quot;,\n  \&quot;4\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;line\&quot;,\n  \&quot;in\&quot;,\n  \&quot;fq\&quot;,\n  \&quot;:\&quot;,\n  \&quot;n\&quot;,\n  \&quot;=\&quot;,\n  \&quot;next\&quot;,\n  \&quot;(\&quot;,\n  \&quot;c\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;n\&quot;,\n  \&quot;==\&quot;,\n  \&quot;1\&quot;,\n  \&quot;:\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'>%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;line\&quot;,\n  \&quot;.\&quot;,\n  \&quot;strip\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'@'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;]\&quot;,\n  \&quot;if\&quot;,\n  \&quot;n\&quot;,\n  \&quot;==\&quot;,\n  \&quot;2\&quot;,\n  \&quot;:\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;.\&quot;,\n  \&quot;append\&quot;,\n  \&quot;(\&quot;,\n  \&quot;line\&quot;,\n  \&quot;.\&quot;,\n  \&quot;strip\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;yield\&quot;,\n  \&quot;seq\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;convert fq to fa&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;convert&quot;,&quot;fq&quot;,&quot;to&quot;,&quot;fa&quot;],&quot;string&quot;:&quot;[\n  \&quot;convert\&quot;,\n  \&quot;fq\&quot;,\n  \&quot;to\&quot;,\n  \&quot;fa\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;83b2566b3a5745437ec651cd6cafddd056846240&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/fastq2fasta.py#L11-L22&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:31,&quot;string&quot;:&quot;31&quot;}}},{&quot;rowIdx&quot;:32,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;elbow-jason/Uno-deprecated&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;uno/decorators.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;change_return_type&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def change_return_type(f):\n    \&quot;\&quot;\&quot;\n    Converts the returned value of wrapped function to the type of the\n    first arg or to the type specified by a kwarg key return_type's value.\n    \&quot;\&quot;\&quot;\n    @wraps(f)\n    def wrapper(*args, **kwargs):\n        if kwargs.has_key('return_type'):\n            return_type = kwargs['return_type']\n            kwargs.pop('return_type')\n            return return_type(f(*args, **kwargs))\n        elif len(args) > 0:\n            return_type = type(args[0])\n            return return_type(f(*args, **kwargs))\n        else:\n            return f(*args, **kwargs)\n    return wrapper&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def change_return_type(f):\n    \&quot;\&quot;\&quot;\n    Converts the returned value of wrapped function to the type of the\n    first arg or to the type specified by a kwarg key return_type's value.\n    \&quot;\&quot;\&quot;\n    @wraps(f)\n    def wrapper(*args, **kwargs):\n        if kwargs.has_key('return_type'):\n            return_type = kwargs['return_type']\n            kwargs.pop('return_type')\n            return return_type(f(*args, **kwargs))\n        elif len(args) > 0:\n            return_type = type(args[0])\n            return return_type(f(*args, **kwargs))\n        else:\n            return f(*args, **kwargs)\n    return wrapper&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;change_return_type&quot;,&quot;(&quot;,&quot;f&quot;,&quot;)&quot;,&quot;:&quot;,&quot;@&quot;,&quot;wraps&quot;,&quot;(&quot;,&quot;f&quot;,&quot;)&quot;,&quot;def&quot;,&quot;wrapper&quot;,&quot;(&quot;,&quot;*&quot;,&quot;args&quot;,&quot;,&quot;,&quot;*&quot;,&quot;*&quot;,&quot;kwargs&quot;,&quot;)&quot;,&quot;:&quot;,&quot;if&quot;,&quot;kwargs&quot;,&quot;.&quot;,&quot;has_key&quot;,&quot;(&quot;,&quot;'return_type'&quot;,&quot;)&quot;,&quot;:&quot;,&quot;return_type&quot;,&quot;=&quot;,&quot;kwargs&quot;,&quot;[&quot;,&quot;'return_type'&quot;,&quot;]&quot;,&quot;kwargs&quot;,&quot;.&quot;,&quot;pop&quot;,&quot;(&quot;,&quot;'return_type'&quot;,&quot;)&quot;,&quot;return&quot;,&quot;return_type&quot;,&quot;(&quot;,&quot;f&quot;,&quot;(&quot;,&quot;*&quot;,&quot;args&quot;,&quot;,&quot;,&quot;*&quot;,&quot;*&quot;,&quot;kwargs&quot;,&quot;)&quot;,&quot;)&quot;,&quot;elif&quot;,&quot;len&quot;,&quot;(&quot;,&quot;args&quot;,&quot;)&quot;,&quot;>&quot;,&quot;0&quot;,&quot;:&quot;,&quot;return_type&quot;,&quot;=&quot;,&quot;type&quot;,&quot;(&quot;,&quot;args&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;)&quot;,&quot;return&quot;,&quot;return_type&quot;,&quot;(&quot;,&quot;f&quot;,&quot;(&quot;,&quot;*&quot;,&quot;args&quot;,&quot;,&quot;,&quot;*&quot;,&quot;*&quot;,&quot;kwargs&quot;,&quot;)&quot;,&quot;)&quot;,&quot;else&quot;,&quot;:&quot;,&quot;return&quot;,&quot;f&quot;,&quot;(&quot;,&quot;*&quot;,&quot;args&quot;,&quot;,&quot;,&quot;*&quot;,&quot;*&quot;,&quot;kwargs&quot;,&quot;)&quot;,&quot;return&quot;,&quot;wrapper&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;change_return_type\&quot;,\n  \&quot;(\&quot;,\n  \&quot;f\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;@\&quot;,\n  \&quot;wraps\&quot;,\n  \&quot;(\&quot;,\n  \&quot;f\&quot;,\n  \&quot;)\&quot;,\n  \&quot;def\&quot;,\n  \&quot;wrapper\&quot;,\n  \&quot;(\&quot;,\n  \&quot;*\&quot;,\n  \&quot;args\&quot;,\n  \&quot;,\&quot;,\n  \&quot;*\&quot;,\n  \&quot;*\&quot;,\n  \&quot;kwargs\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;kwargs\&quot;,\n  \&quot;.\&quot;,\n  \&quot;has_key\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'return_type'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return_type\&quot;,\n  \&quot;=\&quot;,\n  \&quot;kwargs\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'return_type'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;kwargs\&quot;,\n  \&quot;.\&quot;,\n  \&quot;pop\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'return_type'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;return_type\&quot;,\n  \&quot;(\&quot;,\n  \&quot;f\&quot;,\n  \&quot;(\&quot;,\n  \&quot;*\&quot;,\n  \&quot;args\&quot;,\n  \&quot;,\&quot;,\n  \&quot;*\&quot;,\n  \&quot;*\&quot;,\n  \&quot;kwargs\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;elif\&quot;,\n  \&quot;len\&quot;,\n  \&quot;(\&quot;,\n  \&quot;args\&quot;,\n  \&quot;)\&quot;,\n  \&quot;>\&quot;,\n  \&quot;0\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return_type\&quot;,\n  \&quot;=\&quot;,\n  \&quot;type\&quot;,\n  \&quot;(\&quot;,\n  \&quot;args\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;return_type\&quot;,\n  \&quot;(\&quot;,\n  \&quot;f\&quot;,\n  \&quot;(\&quot;,\n  \&quot;*\&quot;,\n  \&quot;args\&quot;,\n  \&quot;,\&quot;,\n  \&quot;*\&quot;,\n  \&quot;*\&quot;,\n  \&quot;kwargs\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;f\&quot;,\n  \&quot;(\&quot;,\n  \&quot;*\&quot;,\n  \&quot;args\&quot;,\n  \&quot;,\&quot;,\n  \&quot;*\&quot;,\n  \&quot;*\&quot;,\n  \&quot;kwargs\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;wrapper\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Converts the returned value of wrapped function to the type of the\n    first arg or to the type specified by a kwarg key return_type's value.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Converts&quot;,&quot;the&quot;,&quot;returned&quot;,&quot;value&quot;,&quot;of&quot;,&quot;wrapped&quot;,&quot;function&quot;,&quot;to&quot;,&quot;the&quot;,&quot;type&quot;,&quot;of&quot;,&quot;the&quot;,&quot;first&quot;,&quot;arg&quot;,&quot;or&quot;,&quot;to&quot;,&quot;the&quot;,&quot;type&quot;,&quot;specified&quot;,&quot;by&quot;,&quot;a&quot;,&quot;kwarg&quot;,&quot;key&quot;,&quot;return_type&quot;,&quot;s&quot;,&quot;value&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Converts\&quot;,\n  \&quot;the\&quot;,\n  \&quot;returned\&quot;,\n  \&quot;value\&quot;,\n  \&quot;of\&quot;,\n  \&quot;wrapped\&quot;,\n  \&quot;function\&quot;,\n  \&quot;to\&quot;,\n  \&quot;the\&quot;,\n  \&quot;type\&quot;,\n  \&quot;of\&quot;,\n  \&quot;the\&quot;,\n  \&quot;first\&quot;,\n  \&quot;arg\&quot;,\n  \&quot;or\&quot;,\n  \&quot;to\&quot;,\n  \&quot;the\&quot;,\n  \&quot;type\&quot;,\n  \&quot;specified\&quot;,\n  \&quot;by\&quot;,\n  \&quot;a\&quot;,\n  \&quot;kwarg\&quot;,\n  \&quot;key\&quot;,\n  \&quot;return_type\&quot;,\n  \&quot;s\&quot;,\n  \&quot;value\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;4ad07d7b84e5b6e3e2b2c89db69448906f24b4e4&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/elbow-jason/Uno-deprecated/blob/4ad07d7b84e5b6e3e2b2c89db69448906f24b4e4/uno/decorators.py#L11-L27&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:32,&quot;string&quot;:&quot;32&quot;}}},{&quot;rowIdx&quot;:33,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;elbow-jason/Uno-deprecated&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;uno/decorators.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;convert_args_to_sets&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def convert_args_to_sets(f):\n    \&quot;\&quot;\&quot;\n    Converts all args to 'set' type via self.setify function.\n    \&quot;\&quot;\&quot;\n    @wraps(f)\n    def wrapper(*args, **kwargs):\n        args = (setify(x) for x in args)\n        return f(*args, **kwargs)\n    return wrapper&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def convert_args_to_sets(f):\n    \&quot;\&quot;\&quot;\n    Converts all args to 'set' type via self.setify function.\n    \&quot;\&quot;\&quot;\n    @wraps(f)\n    def wrapper(*args, **kwargs):\n        args = (setify(x) for x in args)\n        return f(*args, **kwargs)\n    return wrapper&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;convert_args_to_sets&quot;,&quot;(&quot;,&quot;f&quot;,&quot;)&quot;,&quot;:&quot;,&quot;@&quot;,&quot;wraps&quot;,&quot;(&quot;,&quot;f&quot;,&quot;)&quot;,&quot;def&quot;,&quot;wrapper&quot;,&quot;(&quot;,&quot;*&quot;,&quot;args&quot;,&quot;,&quot;,&quot;*&quot;,&quot;*&quot;,&quot;kwargs&quot;,&quot;)&quot;,&quot;:&quot;,&quot;args&quot;,&quot;=&quot;,&quot;(&quot;,&quot;setify&quot;,&quot;(&quot;,&quot;x&quot;,&quot;)&quot;,&quot;for&quot;,&quot;x&quot;,&quot;in&quot;,&quot;args&quot;,&quot;)&quot;,&quot;return&quot;,&quot;f&quot;,&quot;(&quot;,&quot;*&quot;,&quot;args&quot;,&quot;,&quot;,&quot;*&quot;,&quot;*&quot;,&quot;kwargs&quot;,&quot;)&quot;,&quot;return&quot;,&quot;wrapper&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;convert_args_to_sets\&quot;,\n  \&quot;(\&quot;,\n  \&quot;f\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;@\&quot;,\n  \&quot;wraps\&quot;,\n  \&quot;(\&quot;,\n  \&quot;f\&quot;,\n  \&quot;)\&quot;,\n  \&quot;def\&quot;,\n  \&quot;wrapper\&quot;,\n  \&quot;(\&quot;,\n  \&quot;*\&quot;,\n  \&quot;args\&quot;,\n  \&quot;,\&quot;,\n  \&quot;*\&quot;,\n  \&quot;*\&quot;,\n  \&quot;kwargs\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;args\&quot;,\n  \&quot;=\&quot;,\n  \&quot;(\&quot;,\n  \&quot;setify\&quot;,\n  \&quot;(\&quot;,\n  \&quot;x\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;x\&quot;,\n  \&quot;in\&quot;,\n  \&quot;args\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;f\&quot;,\n  \&quot;(\&quot;,\n  \&quot;*\&quot;,\n  \&quot;args\&quot;,\n  \&quot;,\&quot;,\n  \&quot;*\&quot;,\n  \&quot;*\&quot;,\n  \&quot;kwargs\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;wrapper\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Converts all args to 'set' type via self.setify function.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Converts&quot;,&quot;all&quot;,&quot;args&quot;,&quot;to&quot;,&quot;set&quot;,&quot;type&quot;,&quot;via&quot;,&quot;self&quot;,&quot;.&quot;,&quot;setify&quot;,&quot;function&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Converts\&quot;,\n  \&quot;all\&quot;,\n  \&quot;args\&quot;,\n  \&quot;to\&quot;,\n  \&quot;set\&quot;,\n  \&quot;type\&quot;,\n  \&quot;via\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;setify\&quot;,\n  \&quot;function\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;4ad07d7b84e5b6e3e2b2c89db69448906f24b4e4&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/elbow-jason/Uno-deprecated/blob/4ad07d7b84e5b6e3e2b2c89db69448906f24b4e4/uno/decorators.py#L30-L38&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:33,&quot;string&quot;:&quot;33&quot;}}},{&quot;rowIdx&quot;:34,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;laymonage/kbbi-python&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;kbbi/kbbi.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;KBBI._init_entri&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _init_entri(self, laman):\n        \&quot;\&quot;\&quot;Membuat objek-objek entri dari laman yang diambil.\n\n        :param laman: Laman respons yang dikembalikan oleh KBBI daring.\n        :type laman: Response\n        \&quot;\&quot;\&quot;\n\n        sup = BeautifulSoup(laman.text, 'html.parser')\n        estr = ''\n        for label in sup.find('hr').next_siblings:\n            if label.name == 'hr':\n                self.entri.append(Entri(estr))\n                break\n            if label.name == 'h2':\n                if estr:\n                    self.entri.append(Entri(estr))\n                estr = ''\n            estr += str(label).strip()&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _init_entri(self, laman):\n        \&quot;\&quot;\&quot;Membuat objek-objek entri dari laman yang diambil.\n\n        :param laman: Laman respons yang dikembalikan oleh KBBI daring.\n        :type laman: Response\n        \&quot;\&quot;\&quot;\n\n        sup = BeautifulSoup(laman.text, 'html.parser')\n        estr = ''\n        for label in sup.find('hr').next_siblings:\n            if label.name == 'hr':\n                self.entri.append(Entri(estr))\n                break\n            if label.name == 'h2':\n                if estr:\n                    self.entri.append(Entri(estr))\n                estr = ''\n            estr += str(label).strip()&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;_init_entri&quot;,&quot;(&quot;,&quot;self&quot;,&quot;,&quot;,&quot;laman&quot;,&quot;)&quot;,&quot;:&quot;,&quot;sup&quot;,&quot;=&quot;,&quot;BeautifulSoup&quot;,&quot;(&quot;,&quot;laman&quot;,&quot;.&quot;,&quot;text&quot;,&quot;,&quot;,&quot;'html.parser'&quot;,&quot;)&quot;,&quot;estr&quot;,&quot;=&quot;,&quot;''&quot;,&quot;for&quot;,&quot;label&quot;,&quot;in&quot;,&quot;sup&quot;,&quot;.&quot;,&quot;find&quot;,&quot;(&quot;,&quot;'hr'&quot;,&quot;)&quot;,&quot;.&quot;,&quot;next_siblings&quot;,&quot;:&quot;,&quot;if&quot;,&quot;label&quot;,&quot;.&quot;,&quot;name&quot;,&quot;==&quot;,&quot;'hr'&quot;,&quot;:&quot;,&quot;self&quot;,&quot;.&quot;,&quot;entri&quot;,&quot;.&quot;,&quot;append&quot;,&quot;(&quot;,&quot;Entri&quot;,&quot;(&quot;,&quot;estr&quot;,&quot;)&quot;,&quot;)&quot;,&quot;break&quot;,&quot;if&quot;,&quot;label&quot;,&quot;.&quot;,&quot;name&quot;,&quot;==&quot;,&quot;'h2'&quot;,&quot;:&quot;,&quot;if&quot;,&quot;estr&quot;,&quot;:&quot;,&quot;self&quot;,&quot;.&quot;,&quot;entri&quot;,&quot;.&quot;,&quot;append&quot;,&quot;(&quot;,&quot;Entri&quot;,&quot;(&quot;,&quot;estr&quot;,&quot;)&quot;,&quot;)&quot;,&quot;estr&quot;,&quot;=&quot;,&quot;''&quot;,&quot;estr&quot;,&quot;+=&quot;,&quot;str&quot;,&quot;(&quot;,&quot;label&quot;,&quot;)&quot;,&quot;.&quot;,&quot;strip&quot;,&quot;(&quot;,&quot;)&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;_init_entri\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;,\&quot;,\n  \&quot;laman\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;sup\&quot;,\n  \&quot;=\&quot;,\n  \&quot;BeautifulSoup\&quot;,\n  \&quot;(\&quot;,\n  \&quot;laman\&quot;,\n  \&quot;.\&quot;,\n  \&quot;text\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'html.parser'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;estr\&quot;,\n  \&quot;=\&quot;,\n  \&quot;''\&quot;,\n  \&quot;for\&quot;,\n  \&quot;label\&quot;,\n  \&quot;in\&quot;,\n  \&quot;sup\&quot;,\n  \&quot;.\&quot;,\n  \&quot;find\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'hr'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;.\&quot;,\n  \&quot;next_siblings\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;label\&quot;,\n  \&quot;.\&quot;,\n  \&quot;name\&quot;,\n  \&quot;==\&quot;,\n  \&quot;'hr'\&quot;,\n  \&quot;:\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;entri\&quot;,\n  \&quot;.\&quot;,\n  \&quot;append\&quot;,\n  \&quot;(\&quot;,\n  \&quot;Entri\&quot;,\n  \&quot;(\&quot;,\n  \&quot;estr\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;break\&quot;,\n  \&quot;if\&quot;,\n  \&quot;label\&quot;,\n  \&quot;.\&quot;,\n  \&quot;name\&quot;,\n  \&quot;==\&quot;,\n  \&quot;'h2'\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;estr\&quot;,\n  \&quot;:\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;entri\&quot;,\n  \&quot;.\&quot;,\n  \&quot;append\&quot;,\n  \&quot;(\&quot;,\n  \&quot;Entri\&quot;,\n  \&quot;(\&quot;,\n  \&quot;estr\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;estr\&quot;,\n  \&quot;=\&quot;,\n  \&quot;''\&quot;,\n  \&quot;estr\&quot;,\n  \&quot;+=\&quot;,\n  \&quot;str\&quot;,\n  \&quot;(\&quot;,\n  \&quot;label\&quot;,\n  \&quot;)\&quot;,\n  \&quot;.\&quot;,\n  \&quot;strip\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Membuat objek-objek entri dari laman yang diambil.\n\n        :param laman: Laman respons yang dikembalikan oleh KBBI daring.\n        :type laman: Response&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Membuat&quot;,&quot;objek&quot;,&quot;-&quot;,&quot;objek&quot;,&quot;entri&quot;,&quot;dari&quot;,&quot;laman&quot;,&quot;yang&quot;,&quot;diambil&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Membuat\&quot;,\n  \&quot;objek\&quot;,\n  \&quot;-\&quot;,\n  \&quot;objek\&quot;,\n  \&quot;entri\&quot;,\n  \&quot;dari\&quot;,\n  \&quot;laman\&quot;,\n  \&quot;yang\&quot;,\n  \&quot;diambil\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;1a52ba8bcc6dc4c5c1215f9e00207aca264287d6&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L46-L63&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:34,&quot;string&quot;:&quot;34&quot;}}},{&quot;rowIdx&quot;:35,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;laymonage/kbbi-python&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;kbbi/kbbi.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Entri._init_kata_dasar&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _init_kata_dasar(self, dasar):\n        \&quot;\&quot;\&quot;Memproses kata dasar yang ada dalam nama entri.\n\n        :param dasar: ResultSet untuk label HTML dengan class=\&quot;rootword\&quot;\n        :type dasar: ResultSet\n        \&quot;\&quot;\&quot;\n\n        for tiap in dasar:\n            kata = tiap.find('a')\n            dasar_no = kata.find('sup')\n            kata = ambil_teks_dalam_label(kata)\n            self.kata_dasar.append(\n                kata + ' [{}]'.format(dasar_no.text.strip()) if dasar_no else kata\n            )&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _init_kata_dasar(self, dasar):\n        \&quot;\&quot;\&quot;Memproses kata dasar yang ada dalam nama entri.\n\n        :param dasar: ResultSet untuk label HTML dengan class=\&quot;rootword\&quot;\n        :type dasar: ResultSet\n        \&quot;\&quot;\&quot;\n\n        for tiap in dasar:\n            kata = tiap.find('a')\n            dasar_no = kata.find('sup')\n            kata = ambil_teks_dalam_label(kata)\n            self.kata_dasar.append(\n                kata + ' [{}]'.format(dasar_no.text.strip()) if dasar_no else kata\n            )&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;_init_kata_dasar&quot;,&quot;(&quot;,&quot;self&quot;,&quot;,&quot;,&quot;dasar&quot;,&quot;)&quot;,&quot;:&quot;,&quot;for&quot;,&quot;tiap&quot;,&quot;in&quot;,&quot;dasar&quot;,&quot;:&quot;,&quot;kata&quot;,&quot;=&quot;,&quot;tiap&quot;,&quot;.&quot;,&quot;find&quot;,&quot;(&quot;,&quot;'a'&quot;,&quot;)&quot;,&quot;dasar_no&quot;,&quot;=&quot;,&quot;kata&quot;,&quot;.&quot;,&quot;find&quot;,&quot;(&quot;,&quot;'sup'&quot;,&quot;)&quot;,&quot;kata&quot;,&quot;=&quot;,&quot;ambil_teks_dalam_label&quot;,&quot;(&quot;,&quot;kata&quot;,&quot;)&quot;,&quot;self&quot;,&quot;.&quot;,&quot;kata_dasar&quot;,&quot;.&quot;,&quot;append&quot;,&quot;(&quot;,&quot;kata&quot;,&quot;+&quot;,&quot;' [{}]'&quot;,&quot;.&quot;,&quot;format&quot;,&quot;(&quot;,&quot;dasar_no&quot;,&quot;.&quot;,&quot;text&quot;,&quot;.&quot;,&quot;strip&quot;,&quot;(&quot;,&quot;)&quot;,&quot;)&quot;,&quot;if&quot;,&quot;dasar_no&quot;,&quot;else&quot;,&quot;kata&quot;,&quot;)&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;_init_kata_dasar\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;,\&quot;,\n  \&quot;dasar\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;for\&quot;,\n  \&quot;tiap\&quot;,\n  \&quot;in\&quot;,\n  \&quot;dasar\&quot;,\n  \&quot;:\&quot;,\n  \&quot;kata\&quot;,\n  \&quot;=\&quot;,\n  \&quot;tiap\&quot;,\n  \&quot;.\&quot;,\n  \&quot;find\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'a'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;dasar_no\&quot;,\n  \&quot;=\&quot;,\n  \&quot;kata\&quot;,\n  \&quot;.\&quot;,\n  \&quot;find\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'sup'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;kata\&quot;,\n  \&quot;=\&quot;,\n  \&quot;ambil_teks_dalam_label\&quot;,\n  \&quot;(\&quot;,\n  \&quot;kata\&quot;,\n  \&quot;)\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;kata_dasar\&quot;,\n  \&quot;.\&quot;,\n  \&quot;append\&quot;,\n  \&quot;(\&quot;,\n  \&quot;kata\&quot;,\n  \&quot;+\&quot;,\n  \&quot;' [{}]'\&quot;,\n  \&quot;.\&quot;,\n  \&quot;format\&quot;,\n  \&quot;(\&quot;,\n  \&quot;dasar_no\&quot;,\n  \&quot;.\&quot;,\n  \&quot;text\&quot;,\n  \&quot;.\&quot;,\n  \&quot;strip\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;dasar_no\&quot;,\n  \&quot;else\&quot;,\n  \&quot;kata\&quot;,\n  \&quot;)\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Memproses kata dasar yang ada dalam nama entri.\n\n        :param dasar: ResultSet untuk label HTML dengan class=\&quot;rootword\&quot;\n        :type dasar: ResultSet&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Memproses&quot;,&quot;kata&quot;,&quot;dasar&quot;,&quot;yang&quot;,&quot;ada&quot;,&quot;dalam&quot;,&quot;nama&quot;,&quot;entri&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Memproses\&quot;,\n  \&quot;kata\&quot;,\n  \&quot;dasar\&quot;,\n  \&quot;yang\&quot;,\n  \&quot;ada\&quot;,\n  \&quot;dalam\&quot;,\n  \&quot;nama\&quot;,\n  \&quot;entri\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;1a52ba8bcc6dc4c5c1215f9e00207aca264287d6&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L126-L139&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:35,&quot;string&quot;:&quot;35&quot;}}},{&quot;rowIdx&quot;:36,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;laymonage/kbbi-python&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;kbbi/kbbi.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Entri.serialisasi&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def serialisasi(self):\n        \&quot;\&quot;\&quot;Mengembalikan hasil serialisasi objek Entri ini.\n\n        :returns: Dictionary hasil serialisasi\n        :rtype: dict\n        \&quot;\&quot;\&quot;\n\n        return {\n            \&quot;nama\&quot;: self.nama,\n            \&quot;nomor\&quot;: self.nomor,\n            \&quot;kata_dasar\&quot;: self.kata_dasar,\n            \&quot;pelafalan\&quot;: self.pelafalan,\n            \&quot;bentuk_tidak_baku\&quot;: self.bentuk_tidak_baku,\n            \&quot;varian\&quot;: self.varian,\n            \&quot;makna\&quot;: [makna.serialisasi() for makna in self.makna]\n        }&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def serialisasi(self):\n        \&quot;\&quot;\&quot;Mengembalikan hasil serialisasi objek Entri ini.\n\n        :returns: Dictionary hasil serialisasi\n        :rtype: dict\n        \&quot;\&quot;\&quot;\n\n        return {\n            \&quot;nama\&quot;: self.nama,\n            \&quot;nomor\&quot;: self.nomor,\n            \&quot;kata_dasar\&quot;: self.kata_dasar,\n            \&quot;pelafalan\&quot;: self.pelafalan,\n            \&quot;bentuk_tidak_baku\&quot;: self.bentuk_tidak_baku,\n            \&quot;varian\&quot;: self.varian,\n            \&quot;makna\&quot;: [makna.serialisasi() for makna in self.makna]\n        }&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;serialisasi&quot;,&quot;(&quot;,&quot;self&quot;,&quot;)&quot;,&quot;:&quot;,&quot;return&quot;,&quot;{&quot;,&quot;\&quot;nama\&quot;&quot;,&quot;:&quot;,&quot;self&quot;,&quot;.&quot;,&quot;nama&quot;,&quot;,&quot;,&quot;\&quot;nomor\&quot;&quot;,&quot;:&quot;,&quot;self&quot;,&quot;.&quot;,&quot;nomor&quot;,&quot;,&quot;,&quot;\&quot;kata_dasar\&quot;&quot;,&quot;:&quot;,&quot;self&quot;,&quot;.&quot;,&quot;kata_dasar&quot;,&quot;,&quot;,&quot;\&quot;pelafalan\&quot;&quot;,&quot;:&quot;,&quot;self&quot;,&quot;.&quot;,&quot;pelafalan&quot;,&quot;,&quot;,&quot;\&quot;bentuk_tidak_baku\&quot;&quot;,&quot;:&quot;,&quot;self&quot;,&quot;.&quot;,&quot;bentuk_tidak_baku&quot;,&quot;,&quot;,&quot;\&quot;varian\&quot;&quot;,&quot;:&quot;,&quot;self&quot;,&quot;.&quot;,&quot;varian&quot;,&quot;,&quot;,&quot;\&quot;makna\&quot;&quot;,&quot;:&quot;,&quot;[&quot;,&quot;makna&quot;,&quot;.&quot;,&quot;serialisasi&quot;,&quot;(&quot;,&quot;)&quot;,&quot;for&quot;,&quot;makna&quot;,&quot;in&quot;,&quot;self&quot;,&quot;.&quot;,&quot;makna&quot;,&quot;]&quot;,&quot;}&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;serialisasi\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;{\&quot;,\n  \&quot;\\\&quot;nama\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;nama\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;nomor\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;nomor\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;kata_dasar\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;kata_dasar\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;pelafalan\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;pelafalan\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;bentuk_tidak_baku\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;bentuk_tidak_baku\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;varian\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;varian\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;makna\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;[\&quot;,\n  \&quot;makna\&quot;,\n  \&quot;.\&quot;,\n  \&quot;serialisasi\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;makna\&quot;,\n  \&quot;in\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;makna\&quot;,\n  \&quot;]\&quot;,\n  \&quot;}\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Mengembalikan hasil serialisasi objek Entri ini.\n\n        :returns: Dictionary hasil serialisasi\n        :rtype: dict&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Mengembalikan&quot;,&quot;hasil&quot;,&quot;serialisasi&quot;,&quot;objek&quot;,&quot;Entri&quot;,&quot;ini&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Mengembalikan\&quot;,\n  \&quot;hasil\&quot;,\n  \&quot;serialisasi\&quot;,\n  \&quot;objek\&quot;,\n  \&quot;Entri\&quot;,\n  \&quot;ini\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;1a52ba8bcc6dc4c5c1215f9e00207aca264287d6&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L141-L156&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:36,&quot;string&quot;:&quot;36&quot;}}},{&quot;rowIdx&quot;:37,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;laymonage/kbbi-python&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;kbbi/kbbi.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Entri._makna&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _makna(self):\n        \&quot;\&quot;\&quot;Mengembalikan representasi string untuk semua makna entri ini.\n\n        :returns: String representasi makna-makna\n        :rtype: str\n        \&quot;\&quot;\&quot;\n\n        if len(self.makna) > 1:\n            return '\\n'.join(\n                str(i) + \&quot;. \&quot; + str(makna)\n                for i, makna in enumerate(self.makna, 1)\n            )\n        return str(self.makna[0])&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _makna(self):\n        \&quot;\&quot;\&quot;Mengembalikan representasi string untuk semua makna entri ini.\n\n        :returns: String representasi makna-makna\n        :rtype: str\n        \&quot;\&quot;\&quot;\n\n        if len(self.makna) > 1:\n            return '\\n'.join(\n                str(i) + \&quot;. \&quot; + str(makna)\n                for i, makna in enumerate(self.makna, 1)\n            )\n        return str(self.makna[0])&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;_makna&quot;,&quot;(&quot;,&quot;self&quot;,&quot;)&quot;,&quot;:&quot;,&quot;if&quot;,&quot;len&quot;,&quot;(&quot;,&quot;self&quot;,&quot;.&quot;,&quot;makna&quot;,&quot;)&quot;,&quot;>&quot;,&quot;1&quot;,&quot;:&quot;,&quot;return&quot;,&quot;'\\n'&quot;,&quot;.&quot;,&quot;join&quot;,&quot;(&quot;,&quot;str&quot;,&quot;(&quot;,&quot;i&quot;,&quot;)&quot;,&quot;+&quot;,&quot;\&quot;. \&quot;&quot;,&quot;+&quot;,&quot;str&quot;,&quot;(&quot;,&quot;makna&quot;,&quot;)&quot;,&quot;for&quot;,&quot;i&quot;,&quot;,&quot;,&quot;makna&quot;,&quot;in&quot;,&quot;enumerate&quot;,&quot;(&quot;,&quot;self&quot;,&quot;.&quot;,&quot;makna&quot;,&quot;,&quot;,&quot;1&quot;,&quot;)&quot;,&quot;)&quot;,&quot;return&quot;,&quot;str&quot;,&quot;(&quot;,&quot;self&quot;,&quot;.&quot;,&quot;makna&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;)&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;_makna\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;len\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;makna\&quot;,\n  \&quot;)\&quot;,\n  \&quot;>\&quot;,\n  \&quot;1\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;'\\\\n'\&quot;,\n  \&quot;.\&quot;,\n  \&quot;join\&quot;,\n  \&quot;(\&quot;,\n  \&quot;str\&quot;,\n  \&quot;(\&quot;,\n  \&quot;i\&quot;,\n  \&quot;)\&quot;,\n  \&quot;+\&quot;,\n  \&quot;\\\&quot;. \\\&quot;\&quot;,\n  \&quot;+\&quot;,\n  \&quot;str\&quot;,\n  \&quot;(\&quot;,\n  \&quot;makna\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;,\&quot;,\n  \&quot;makna\&quot;,\n  \&quot;in\&quot;,\n  \&quot;enumerate\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;makna\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;str\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;makna\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Mengembalikan representasi string untuk semua makna entri ini.\n\n        :returns: String representasi makna-makna\n        :rtype: str&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Mengembalikan&quot;,&quot;representasi&quot;,&quot;string&quot;,&quot;untuk&quot;,&quot;semua&quot;,&quot;makna&quot;,&quot;entri&quot;,&quot;ini&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Mengembalikan\&quot;,\n  \&quot;representasi\&quot;,\n  \&quot;string\&quot;,\n  \&quot;untuk\&quot;,\n  \&quot;semua\&quot;,\n  \&quot;makna\&quot;,\n  \&quot;entri\&quot;,\n  \&quot;ini\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;1a52ba8bcc6dc4c5c1215f9e00207aca264287d6&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L158-L170&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:37,&quot;string&quot;:&quot;37&quot;}}},{&quot;rowIdx&quot;:38,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;laymonage/kbbi-python&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;kbbi/kbbi.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Entri._nama&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _nama(self):\n        \&quot;\&quot;\&quot;Mengembalikan representasi string untuk nama entri ini.\n\n        :returns: String representasi nama entri\n        :rtype: str\n        \&quot;\&quot;\&quot;\n\n        hasil = self.nama\n        if self.nomor:\n            hasil += \&quot; [{}]\&quot;.format(self.nomor)\n        if self.kata_dasar:\n            hasil = \&quot; Â» \&quot;.join(self.kata_dasar) + \&quot; Â» \&quot; + hasil\n        return hasil&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _nama(self):\n        \&quot;\&quot;\&quot;Mengembalikan representasi string untuk nama entri ini.\n\n        :returns: String representasi nama entri\n        :rtype: str\n        \&quot;\&quot;\&quot;\n\n        hasil = self.nama\n        if self.nomor:\n            hasil += \&quot; [{}]\&quot;.format(self.nomor)\n        if self.kata_dasar:\n            hasil = \&quot; Â» \&quot;.join(self.kata_dasar) + \&quot; Â» \&quot; + hasil\n        return hasil&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;_nama&quot;,&quot;(&quot;,&quot;self&quot;,&quot;)&quot;,&quot;:&quot;,&quot;hasil&quot;,&quot;=&quot;,&quot;self&quot;,&quot;.&quot;,&quot;nama&quot;,&quot;if&quot;,&quot;self&quot;,&quot;.&quot;,&quot;nomor&quot;,&quot;:&quot;,&quot;hasil&quot;,&quot;+=&quot;,&quot;\&quot; [{}]\&quot;&quot;,&quot;.&quot;,&quot;format&quot;,&quot;(&quot;,&quot;self&quot;,&quot;.&quot;,&quot;nomor&quot;,&quot;)&quot;,&quot;if&quot;,&quot;self&quot;,&quot;.&quot;,&quot;kata_dasar&quot;,&quot;:&quot;,&quot;hasil&quot;,&quot;=&quot;,&quot;\&quot; Â» \&quot;.&quot;,&quot;j&quot;,&quot;oin(&quot;,&quot;s&quot;,&quot;elf.&quot;,&quot;k&quot;,&quot;ata_dasar)&quot;,&quot; &quot;,&quot; &quot;,&quot; Â» \&quot; +&quot;,&quot;h&quot;,&quot;sil&quot;,&quot;return&quot;,&quot;hasil&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;_nama\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;hasil\&quot;,\n  \&quot;=\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;nama\&quot;,\n  \&quot;if\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;nomor\&quot;,\n  \&quot;:\&quot;,\n  \&quot;hasil\&quot;,\n  \&quot;+=\&quot;,\n  \&quot;\\\&quot; [{}]\\\&quot;\&quot;,\n  \&quot;.\&quot;,\n  \&quot;format\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;nomor\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;kata_dasar\&quot;,\n  \&quot;:\&quot;,\n  \&quot;hasil\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot; Â» \\\&quot;.\&quot;,\n  \&quot;j\&quot;,\n  \&quot;oin(\&quot;,\n  \&quot;s\&quot;,\n  \&quot;elf.\&quot;,\n  \&quot;k\&quot;,\n  \&quot;ata_dasar)\&quot;,\n  \&quot; \&quot;,\n  \&quot; \&quot;,\n  \&quot; Â» \\\&quot; +\&quot;,\n  \&quot;h\&quot;,\n  \&quot;sil\&quot;,\n  \&quot;return\&quot;,\n  \&quot;hasil\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Mengembalikan representasi string untuk nama entri ini.\n\n        :returns: String representasi nama entri\n        :rtype: str&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Mengembalikan&quot;,&quot;representasi&quot;,&quot;string&quot;,&quot;untuk&quot;,&quot;nama&quot;,&quot;entri&quot;,&quot;ini&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Mengembalikan\&quot;,\n  \&quot;representasi\&quot;,\n  \&quot;string\&quot;,\n  \&quot;untuk\&quot;,\n  \&quot;nama\&quot;,\n  \&quot;entri\&quot;,\n  \&quot;ini\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;1a52ba8bcc6dc4c5c1215f9e00207aca264287d6&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L172-L184&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:38,&quot;string&quot;:&quot;38&quot;}}},{&quot;rowIdx&quot;:39,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;laymonage/kbbi-python&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;kbbi/kbbi.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Entri._varian&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _varian(self, varian):\n        \&quot;\&quot;\&quot;Mengembalikan representasi string untuk varian entri ini.\n        Dapat digunakan untuk \&quot;Varian\&quot; maupun \&quot;Bentuk tidak baku\&quot;.\n\n        :param varian: List bentuk tidak baku atau varian\n        :type varian: list\n        :returns: String representasi varian atau bentuk tidak baku\n        :rtype: str\n        \&quot;\&quot;\&quot;\n\n        if varian == self.bentuk_tidak_baku:\n            nama = \&quot;Bentuk tidak baku\&quot;\n        elif varian == self.varian:\n            nama = \&quot;Varian\&quot;\n        else:\n            return ''\n        return nama + ': ' + ', '.join(varian)&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _varian(self, varian):\n        \&quot;\&quot;\&quot;Mengembalikan representasi string untuk varian entri ini.\n        Dapat digunakan untuk \&quot;Varian\&quot; maupun \&quot;Bentuk tidak baku\&quot;.\n\n        :param varian: List bentuk tidak baku atau varian\n        :type varian: list\n        :returns: String representasi varian atau bentuk tidak baku\n        :rtype: str\n        \&quot;\&quot;\&quot;\n\n        if varian == self.bentuk_tidak_baku:\n            nama = \&quot;Bentuk tidak baku\&quot;\n        elif varian == self.varian:\n            nama = \&quot;Varian\&quot;\n        else:\n            return ''\n        return nama + ': ' + ', '.join(varian)&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;_varian&quot;,&quot;(&quot;,&quot;self&quot;,&quot;,&quot;,&quot;varian&quot;,&quot;)&quot;,&quot;:&quot;,&quot;if&quot;,&quot;varian&quot;,&quot;==&quot;,&quot;self&quot;,&quot;.&quot;,&quot;bentuk_tidak_baku&quot;,&quot;:&quot;,&quot;nama&quot;,&quot;=&quot;,&quot;\&quot;Bentuk tidak baku\&quot;&quot;,&quot;elif&quot;,&quot;varian&quot;,&quot;==&quot;,&quot;self&quot;,&quot;.&quot;,&quot;varian&quot;,&quot;:&quot;,&quot;nama&quot;,&quot;=&quot;,&quot;\&quot;Varian\&quot;&quot;,&quot;else&quot;,&quot;:&quot;,&quot;return&quot;,&quot;''&quot;,&quot;return&quot;,&quot;nama&quot;,&quot;+&quot;,&quot;': '&quot;,&quot;+&quot;,&quot;', '&quot;,&quot;.&quot;,&quot;join&quot;,&quot;(&quot;,&quot;varian&quot;,&quot;)&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;_varian\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;,\&quot;,\n  \&quot;varian\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;varian\&quot;,\n  \&quot;==\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;bentuk_tidak_baku\&quot;,\n  \&quot;:\&quot;,\n  \&quot;nama\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;Bentuk tidak baku\\\&quot;\&quot;,\n  \&quot;elif\&quot;,\n  \&quot;varian\&quot;,\n  \&quot;==\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;varian\&quot;,\n  \&quot;:\&quot;,\n  \&quot;nama\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;Varian\\\&quot;\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;''\&quot;,\n  \&quot;return\&quot;,\n  \&quot;nama\&quot;,\n  \&quot;+\&quot;,\n  \&quot;': '\&quot;,\n  \&quot;+\&quot;,\n  \&quot;', '\&quot;,\n  \&quot;.\&quot;,\n  \&quot;join\&quot;,\n  \&quot;(\&quot;,\n  \&quot;varian\&quot;,\n  \&quot;)\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Mengembalikan representasi string untuk varian entri ini.\n        Dapat digunakan untuk \&quot;Varian\&quot; maupun \&quot;Bentuk tidak baku\&quot;.\n\n        :param varian: List bentuk tidak baku atau varian\n        :type varian: list\n        :returns: String representasi varian atau bentuk tidak baku\n        :rtype: str&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Mengembalikan&quot;,&quot;representasi&quot;,&quot;string&quot;,&quot;untuk&quot;,&quot;varian&quot;,&quot;entri&quot;,&quot;ini&quot;,&quot;.&quot;,&quot;Dapat&quot;,&quot;digunakan&quot;,&quot;untuk&quot;,&quot;Varian&quot;,&quot;maupun&quot;,&quot;Bentuk&quot;,&quot;tidak&quot;,&quot;baku&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Mengembalikan\&quot;,\n  \&quot;representasi\&quot;,\n  \&quot;string\&quot;,\n  \&quot;untuk\&quot;,\n  \&quot;varian\&quot;,\n  \&quot;entri\&quot;,\n  \&quot;ini\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Dapat\&quot;,\n  \&quot;digunakan\&quot;,\n  \&quot;untuk\&quot;,\n  \&quot;Varian\&quot;,\n  \&quot;maupun\&quot;,\n  \&quot;Bentuk\&quot;,\n  \&quot;tidak\&quot;,\n  \&quot;baku\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;1a52ba8bcc6dc4c5c1215f9e00207aca264287d6&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L186-L202&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:39,&quot;string&quot;:&quot;39&quot;}}},{&quot;rowIdx&quot;:40,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;laymonage/kbbi-python&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;kbbi/kbbi.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Makna._init_kelas&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _init_kelas(self, makna_label):\n        \&quot;\&quot;\&quot;Memproses kelas kata yang ada dalam makna.\n\n        :param makna_label: BeautifulSoup untuk makna yang ingin diproses.\n        :type makna_label: BeautifulSoup\n        \&quot;\&quot;\&quot;\n\n        kelas = makna_label.find(color='red')\n        lain = makna_label.find(color='darkgreen')\n        info = makna_label.find(color='green')\n        if kelas:\n            kelas = kelas.find_all('span')\n        if lain:\n            self.kelas = {lain.text.strip(): lain['title'].strip()}\n            self.submakna = lain.next_sibling.strip()\n            self.submakna += ' ' + makna_label.find(color='grey').text.strip()\n        else:\n            self.kelas = {\n                k.text.strip(): k['title'].strip() for k in kelas\n            } if kelas else {}\n        self.info = info.text.strip() if info else ''&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _init_kelas(self, makna_label):\n        \&quot;\&quot;\&quot;Memproses kelas kata yang ada dalam makna.\n\n        :param makna_label: BeautifulSoup untuk makna yang ingin diproses.\n        :type makna_label: BeautifulSoup\n        \&quot;\&quot;\&quot;\n\n        kelas = makna_label.find(color='red')\n        lain = makna_label.find(color='darkgreen')\n        info = makna_label.find(color='green')\n        if kelas:\n            kelas = kelas.find_all('span')\n        if lain:\n            self.kelas = {lain.text.strip(): lain['title'].strip()}\n            self.submakna = lain.next_sibling.strip()\n            self.submakna += ' ' + makna_label.find(color='grey').text.strip()\n        else:\n            self.kelas = {\n                k.text.strip(): k['title'].strip() for k in kelas\n            } if kelas else {}\n        self.info = info.text.strip() if info else ''&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;_init_kelas&quot;,&quot;(&quot;,&quot;self&quot;,&quot;,&quot;,&quot;makna_label&quot;,&quot;)&quot;,&quot;:&quot;,&quot;kelas&quot;,&quot;=&quot;,&quot;makna_label&quot;,&quot;.&quot;,&quot;find&quot;,&quot;(&quot;,&quot;color&quot;,&quot;=&quot;,&quot;'red'&quot;,&quot;)&quot;,&quot;lain&quot;,&quot;=&quot;,&quot;makna_label&quot;,&quot;.&quot;,&quot;find&quot;,&quot;(&quot;,&quot;color&quot;,&quot;=&quot;,&quot;'darkgreen'&quot;,&quot;)&quot;,&quot;info&quot;,&quot;=&quot;,&quot;makna_label&quot;,&quot;.&quot;,&quot;find&quot;,&quot;(&quot;,&quot;color&quot;,&quot;=&quot;,&quot;'green'&quot;,&quot;)&quot;,&quot;if&quot;,&quot;kelas&quot;,&quot;:&quot;,&quot;kelas&quot;,&quot;=&quot;,&quot;kelas&quot;,&quot;.&quot;,&quot;find_all&quot;,&quot;(&quot;,&quot;'span'&quot;,&quot;)&quot;,&quot;if&quot;,&quot;lain&quot;,&quot;:&quot;,&quot;self&quot;,&quot;.&quot;,&quot;kelas&quot;,&quot;=&quot;,&quot;{&quot;,&quot;lain&quot;,&quot;.&quot;,&quot;text&quot;,&quot;.&quot;,&quot;strip&quot;,&quot;(&quot;,&quot;)&quot;,&quot;:&quot;,&quot;lain&quot;,&quot;[&quot;,&quot;'title'&quot;,&quot;]&quot;,&quot;.&quot;,&quot;strip&quot;,&quot;(&quot;,&quot;)&quot;,&quot;}&quot;,&quot;self&quot;,&quot;.&quot;,&quot;submakna&quot;,&quot;=&quot;,&quot;lain&quot;,&quot;.&quot;,&quot;next_sibling&quot;,&quot;.&quot;,&quot;strip&quot;,&quot;(&quot;,&quot;)&quot;,&quot;self&quot;,&quot;.&quot;,&quot;submakna&quot;,&quot;+=&quot;,&quot;' '&quot;,&quot;+&quot;,&quot;makna_label&quot;,&quot;.&quot;,&quot;find&quot;,&quot;(&quot;,&quot;color&quot;,&quot;=&quot;,&quot;'grey'&quot;,&quot;)&quot;,&quot;.&quot;,&quot;text&quot;,&quot;.&quot;,&quot;strip&quot;,&quot;(&quot;,&quot;)&quot;,&quot;else&quot;,&quot;:&quot;,&quot;self&quot;,&quot;.&quot;,&quot;kelas&quot;,&quot;=&quot;,&quot;{&quot;,&quot;k&quot;,&quot;.&quot;,&quot;text&quot;,&quot;.&quot;,&quot;strip&quot;,&quot;(&quot;,&quot;)&quot;,&quot;:&quot;,&quot;k&quot;,&quot;[&quot;,&quot;'title'&quot;,&quot;]&quot;,&quot;.&quot;,&quot;strip&quot;,&quot;(&quot;,&quot;)&quot;,&quot;for&quot;,&quot;k&quot;,&quot;in&quot;,&quot;kelas&quot;,&quot;}&quot;,&quot;if&quot;,&quot;kelas&quot;,&quot;else&quot;,&quot;{&quot;,&quot;}&quot;,&quot;self&quot;,&quot;.&quot;,&quot;info&quot;,&quot;=&quot;,&quot;info&quot;,&quot;.&quot;,&quot;text&quot;,&quot;.&quot;,&quot;strip&quot;,&quot;(&quot;,&quot;)&quot;,&quot;if&quot;,&quot;info&quot;,&quot;else&quot;,&quot;''&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;_init_kelas\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;,\&quot;,\n  \&quot;makna_label\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;kelas\&quot;,\n  \&quot;=\&quot;,\n  \&quot;makna_label\&quot;,\n  \&quot;.\&quot;,\n  \&quot;find\&quot;,\n  \&quot;(\&quot;,\n  \&quot;color\&quot;,\n  \&quot;=\&quot;,\n  \&quot;'red'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;lain\&quot;,\n  \&quot;=\&quot;,\n  \&quot;makna_label\&quot;,\n  \&quot;.\&quot;,\n  \&quot;find\&quot;,\n  \&quot;(\&quot;,\n  \&quot;color\&quot;,\n  \&quot;=\&quot;,\n  \&quot;'darkgreen'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;info\&quot;,\n  \&quot;=\&quot;,\n  \&quot;makna_label\&quot;,\n  \&quot;.\&quot;,\n  \&quot;find\&quot;,\n  \&quot;(\&quot;,\n  \&quot;color\&quot;,\n  \&quot;=\&quot;,\n  \&quot;'green'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;kelas\&quot;,\n  \&quot;:\&quot;,\n  \&quot;kelas\&quot;,\n  \&quot;=\&quot;,\n  \&quot;kelas\&quot;,\n  \&quot;.\&quot;,\n  \&quot;find_all\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'span'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;lain\&quot;,\n  \&quot;:\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;kelas\&quot;,\n  \&quot;=\&quot;,\n  \&quot;{\&quot;,\n  \&quot;lain\&quot;,\n  \&quot;.\&quot;,\n  \&quot;text\&quot;,\n  \&quot;.\&quot;,\n  \&quot;strip\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;lain\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'title'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;strip\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;}\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;submakna\&quot;,\n  \&quot;=\&quot;,\n  \&quot;lain\&quot;,\n  \&quot;.\&quot;,\n  \&quot;next_sibling\&quot;,\n  \&quot;.\&quot;,\n  \&quot;strip\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;submakna\&quot;,\n  \&quot;+=\&quot;,\n  \&quot;' '\&quot;,\n  \&quot;+\&quot;,\n  \&quot;makna_label\&quot;,\n  \&quot;.\&quot;,\n  \&quot;find\&quot;,\n  \&quot;(\&quot;,\n  \&quot;color\&quot;,\n  \&quot;=\&quot;,\n  \&quot;'grey'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;.\&quot;,\n  \&quot;text\&quot;,\n  \&quot;.\&quot;,\n  \&quot;strip\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;kelas\&quot;,\n  \&quot;=\&quot;,\n  \&quot;{\&quot;,\n  \&quot;k\&quot;,\n  \&quot;.\&quot;,\n  \&quot;text\&quot;,\n  \&quot;.\&quot;,\n  \&quot;strip\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;k\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'title'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;strip\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;k\&quot;,\n  \&quot;in\&quot;,\n  \&quot;kelas\&quot;,\n  \&quot;}\&quot;,\n  \&quot;if\&quot;,\n  \&quot;kelas\&quot;,\n  \&quot;else\&quot;,\n  \&quot;{\&quot;,\n  \&quot;}\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;info\&quot;,\n  \&quot;=\&quot;,\n  \&quot;info\&quot;,\n  \&quot;.\&quot;,\n  \&quot;text\&quot;,\n  \&quot;.\&quot;,\n  \&quot;strip\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;info\&quot;,\n  \&quot;else\&quot;,\n  \&quot;''\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Memproses kelas kata yang ada dalam makna.\n\n        :param makna_label: BeautifulSoup untuk makna yang ingin diproses.\n        :type makna_label: BeautifulSoup&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Memproses&quot;,&quot;kelas&quot;,&quot;kata&quot;,&quot;yang&quot;,&quot;ada&quot;,&quot;dalam&quot;,&quot;makna&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Memproses\&quot;,\n  \&quot;kelas\&quot;,\n  \&quot;kata\&quot;,\n  \&quot;yang\&quot;,\n  \&quot;ada\&quot;,\n  \&quot;dalam\&quot;,\n  \&quot;makna\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;1a52ba8bcc6dc4c5c1215f9e00207aca264287d6&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L239-L259&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:40,&quot;string&quot;:&quot;40&quot;}}},{&quot;rowIdx&quot;:41,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;laymonage/kbbi-python&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;kbbi/kbbi.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Makna._init_contoh&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _init_contoh(self, makna_label):\n        \&quot;\&quot;\&quot;Memproses contoh yang ada dalam makna.\n\n        :param makna_label: BeautifulSoup untuk makna yang ingin diproses.\n        :type makna_label: BeautifulSoup\n        \&quot;\&quot;\&quot;\n\n        indeks = makna_label.text.find(': ')\n        if indeks != -1:\n            contoh = makna_label.text[indeks + 2:].strip()\n            self.contoh = contoh.split('; ')\n        else:\n            self.contoh = []&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def _init_contoh(self, makna_label):\n        \&quot;\&quot;\&quot;Memproses contoh yang ada dalam makna.\n\n        :param makna_label: BeautifulSoup untuk makna yang ingin diproses.\n        :type makna_label: BeautifulSoup\n        \&quot;\&quot;\&quot;\n\n        indeks = makna_label.text.find(': ')\n        if indeks != -1:\n            contoh = makna_label.text[indeks + 2:].strip()\n            self.contoh = contoh.split('; ')\n        else:\n            self.contoh = []&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;_init_contoh&quot;,&quot;(&quot;,&quot;self&quot;,&quot;,&quot;,&quot;makna_label&quot;,&quot;)&quot;,&quot;:&quot;,&quot;indeks&quot;,&quot;=&quot;,&quot;makna_label&quot;,&quot;.&quot;,&quot;text&quot;,&quot;.&quot;,&quot;find&quot;,&quot;(&quot;,&quot;': '&quot;,&quot;)&quot;,&quot;if&quot;,&quot;indeks&quot;,&quot;!=&quot;,&quot;-&quot;,&quot;1&quot;,&quot;:&quot;,&quot;contoh&quot;,&quot;=&quot;,&quot;makna_label&quot;,&quot;.&quot;,&quot;text&quot;,&quot;[&quot;,&quot;indeks&quot;,&quot;+&quot;,&quot;2&quot;,&quot;:&quot;,&quot;]&quot;,&quot;.&quot;,&quot;strip&quot;,&quot;(&quot;,&quot;)&quot;,&quot;self&quot;,&quot;.&quot;,&quot;contoh&quot;,&quot;=&quot;,&quot;contoh&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;'; '&quot;,&quot;)&quot;,&quot;else&quot;,&quot;:&quot;,&quot;self&quot;,&quot;.&quot;,&quot;contoh&quot;,&quot;=&quot;,&quot;[&quot;,&quot;]&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;_init_contoh\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;,\&quot;,\n  \&quot;makna_label\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;indeks\&quot;,\n  \&quot;=\&quot;,\n  \&quot;makna_label\&quot;,\n  \&quot;.\&quot;,\n  \&quot;text\&quot;,\n  \&quot;.\&quot;,\n  \&quot;find\&quot;,\n  \&quot;(\&quot;,\n  \&quot;': '\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;indeks\&quot;,\n  \&quot;!=\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;:\&quot;,\n  \&quot;contoh\&quot;,\n  \&quot;=\&quot;,\n  \&quot;makna_label\&quot;,\n  \&quot;.\&quot;,\n  \&quot;text\&quot;,\n  \&quot;[\&quot;,\n  \&quot;indeks\&quot;,\n  \&quot;+\&quot;,\n  \&quot;2\&quot;,\n  \&quot;:\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;strip\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;contoh\&quot;,\n  \&quot;=\&quot;,\n  \&quot;contoh\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'; '\&quot;,\n  \&quot;)\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;contoh\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;]\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Memproses contoh yang ada dalam makna.\n\n        :param makna_label: BeautifulSoup untuk makna yang ingin diproses.\n        :type makna_label: BeautifulSoup&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Memproses&quot;,&quot;contoh&quot;,&quot;yang&quot;,&quot;ada&quot;,&quot;dalam&quot;,&quot;makna&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Memproses\&quot;,\n  \&quot;contoh\&quot;,\n  \&quot;yang\&quot;,\n  \&quot;ada\&quot;,\n  \&quot;dalam\&quot;,\n  \&quot;makna\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;1a52ba8bcc6dc4c5c1215f9e00207aca264287d6&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L261-L273&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:41,&quot;string&quot;:&quot;41&quot;}}},{&quot;rowIdx&quot;:42,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;laymonage/kbbi-python&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;kbbi/kbbi.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Makna.serialisasi&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def serialisasi(self):\n        \&quot;\&quot;\&quot;Mengembalikan hasil serialisasi objek Makna ini.\n\n        :returns: Dictionary hasil serialisasi\n        :rtype: dict\n        \&quot;\&quot;\&quot;\n\n        return {\n            \&quot;kelas\&quot;: self.kelas,\n            \&quot;submakna\&quot;: self.submakna,\n            \&quot;info\&quot;: self.info,\n            \&quot;contoh\&quot;: self.contoh\n        }&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def serialisasi(self):\n        \&quot;\&quot;\&quot;Mengembalikan hasil serialisasi objek Makna ini.\n\n        :returns: Dictionary hasil serialisasi\n        :rtype: dict\n        \&quot;\&quot;\&quot;\n\n        return {\n            \&quot;kelas\&quot;: self.kelas,\n            \&quot;submakna\&quot;: self.submakna,\n            \&quot;info\&quot;: self.info,\n            \&quot;contoh\&quot;: self.contoh\n        }&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;serialisasi&quot;,&quot;(&quot;,&quot;self&quot;,&quot;)&quot;,&quot;:&quot;,&quot;return&quot;,&quot;{&quot;,&quot;\&quot;kelas\&quot;&quot;,&quot;:&quot;,&quot;self&quot;,&quot;.&quot;,&quot;kelas&quot;,&quot;,&quot;,&quot;\&quot;submakna\&quot;&quot;,&quot;:&quot;,&quot;self&quot;,&quot;.&quot;,&quot;submakna&quot;,&quot;,&quot;,&quot;\&quot;info\&quot;&quot;,&quot;:&quot;,&quot;self&quot;,&quot;.&quot;,&quot;info&quot;,&quot;,&quot;,&quot;\&quot;contoh\&quot;&quot;,&quot;:&quot;,&quot;self&quot;,&quot;.&quot;,&quot;contoh&quot;,&quot;}&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;serialisasi\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;{\&quot;,\n  \&quot;\\\&quot;kelas\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;kelas\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;submakna\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;submakna\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;info\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;info\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;contoh\\\&quot;\&quot;,\n  \&quot;:\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;contoh\&quot;,\n  \&quot;}\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Mengembalikan hasil serialisasi objek Makna ini.\n\n        :returns: Dictionary hasil serialisasi\n        :rtype: dict&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Mengembalikan&quot;,&quot;hasil&quot;,&quot;serialisasi&quot;,&quot;objek&quot;,&quot;Makna&quot;,&quot;ini&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Mengembalikan\&quot;,\n  \&quot;hasil\&quot;,\n  \&quot;serialisasi\&quot;,\n  \&quot;objek\&quot;,\n  \&quot;Makna\&quot;,\n  \&quot;ini\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;1a52ba8bcc6dc4c5c1215f9e00207aca264287d6&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L275-L287&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:42,&quot;string&quot;:&quot;42&quot;}}},{&quot;rowIdx&quot;:43,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;mkouhei/bootstrap-py&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;bootstrap_py/docs.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;build_sphinx&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def build_sphinx(pkg_data, projectdir):\n    \&quot;\&quot;\&quot;Build sphinx documentation.\n\n    :rtype: int\n    :return: subprocess.call return code\n\n    :param `bootstrap_py.control.PackageData` pkg_data: package meta data\n    :param str projectdir: project root directory\n    \&quot;\&quot;\&quot;\n    try:\n        version, _minor_version = pkg_data.version.rsplit('.', 1)\n    except ValueError:\n        version = pkg_data.version\n    args = ' '.join(('sphinx-quickstart',\n                     '--sep',\n                     '-q',\n                     '-p \&quot;{name}\&quot;',\n                     '-a \&quot;{author}\&quot;',\n                     '-v \&quot;{version}\&quot;',\n                     '-r \&quot;{release}\&quot;',\n                     '-l en',\n                     '--suffix=.rst',\n                     '--master=index',\n                     '--ext-autodoc',\n                     '--ext-viewcode',\n                     '--makefile',\n                     '{projectdir}')).format(name=pkg_data.name,\n                                             author=pkg_data.author,\n                                             version=version,\n                                             release=pkg_data.version,\n                                             projectdir=projectdir)\n    if subprocess.call(shlex.split(args)) == 0:\n        _touch_gitkeep(projectdir)&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def build_sphinx(pkg_data, projectdir):\n    \&quot;\&quot;\&quot;Build sphinx documentation.\n\n    :rtype: int\n    :return: subprocess.call return code\n\n    :param `bootstrap_py.control.PackageData` pkg_data: package meta data\n    :param str projectdir: project root directory\n    \&quot;\&quot;\&quot;\n    try:\n        version, _minor_version = pkg_data.version.rsplit('.', 1)\n    except ValueError:\n        version = pkg_data.version\n    args = ' '.join(('sphinx-quickstart',\n                     '--sep',\n                     '-q',\n                     '-p \&quot;{name}\&quot;',\n                     '-a \&quot;{author}\&quot;',\n                     '-v \&quot;{version}\&quot;',\n                     '-r \&quot;{release}\&quot;',\n                     '-l en',\n                     '--suffix=.rst',\n                     '--master=index',\n                     '--ext-autodoc',\n                     '--ext-viewcode',\n                     '--makefile',\n                     '{projectdir}')).format(name=pkg_data.name,\n                                             author=pkg_data.author,\n                                             version=version,\n                                             release=pkg_data.version,\n                                             projectdir=projectdir)\n    if subprocess.call(shlex.split(args)) == 0:\n        _touch_gitkeep(projectdir)&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;build_sphinx&quot;,&quot;(&quot;,&quot;pkg_data&quot;,&quot;,&quot;,&quot;projectdir&quot;,&quot;)&quot;,&quot;:&quot;,&quot;try&quot;,&quot;:&quot;,&quot;version&quot;,&quot;,&quot;,&quot;_minor_version&quot;,&quot;=&quot;,&quot;pkg_data&quot;,&quot;.&quot;,&quot;version&quot;,&quot;.&quot;,&quot;rsplit&quot;,&quot;(&quot;,&quot;'.'&quot;,&quot;,&quot;,&quot;1&quot;,&quot;)&quot;,&quot;except&quot;,&quot;ValueError&quot;,&quot;:&quot;,&quot;version&quot;,&quot;=&quot;,&quot;pkg_data&quot;,&quot;.&quot;,&quot;version&quot;,&quot;args&quot;,&quot;=&quot;,&quot;' '&quot;,&quot;.&quot;,&quot;join&quot;,&quot;(&quot;,&quot;(&quot;,&quot;'sphinx-quickstart'&quot;,&quot;,&quot;,&quot;'--sep'&quot;,&quot;,&quot;,&quot;'-q'&quot;,&quot;,&quot;,&quot;'-p \&quot;{name}\&quot;'&quot;,&quot;,&quot;,&quot;'-a \&quot;{author}\&quot;'&quot;,&quot;,&quot;,&quot;'-v \&quot;{version}\&quot;'&quot;,&quot;,&quot;,&quot;'-r \&quot;{release}\&quot;'&quot;,&quot;,&quot;,&quot;'-l en'&quot;,&quot;,&quot;,&quot;'--suffix=.rst'&quot;,&quot;,&quot;,&quot;'--master=index'&quot;,&quot;,&quot;,&quot;'--ext-autodoc'&quot;,&quot;,&quot;,&quot;'--ext-viewcode'&quot;,&quot;,&quot;,&quot;'--makefile'&quot;,&quot;,&quot;,&quot;'{projectdir}'&quot;,&quot;)&quot;,&quot;)&quot;,&quot;.&quot;,&quot;format&quot;,&quot;(&quot;,&quot;name&quot;,&quot;=&quot;,&quot;pkg_data&quot;,&quot;.&quot;,&quot;name&quot;,&quot;,&quot;,&quot;author&quot;,&quot;=&quot;,&quot;pkg_data&quot;,&quot;.&quot;,&quot;author&quot;,&quot;,&quot;,&quot;version&quot;,&quot;=&quot;,&quot;version&quot;,&quot;,&quot;,&quot;release&quot;,&quot;=&quot;,&quot;pkg_data&quot;,&quot;.&quot;,&quot;version&quot;,&quot;,&quot;,&quot;projectdir&quot;,&quot;=&quot;,&quot;projectdir&quot;,&quot;)&quot;,&quot;if&quot;,&quot;subprocess&quot;,&quot;.&quot;,&quot;call&quot;,&quot;(&quot;,&quot;shlex&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;args&quot;,&quot;)&quot;,&quot;)&quot;,&quot;==&quot;,&quot;0&quot;,&quot;:&quot;,&quot;_touch_gitkeep&quot;,&quot;(&quot;,&quot;projectdir&quot;,&quot;)&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;build_sphinx\&quot;,\n  \&quot;(\&quot;,\n  \&quot;pkg_data\&quot;,\n  \&quot;,\&quot;,\n  \&quot;projectdir\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;try\&quot;,\n  \&quot;:\&quot;,\n  \&quot;version\&quot;,\n  \&quot;,\&quot;,\n  \&quot;_minor_version\&quot;,\n  \&quot;=\&quot;,\n  \&quot;pkg_data\&quot;,\n  \&quot;.\&quot;,\n  \&quot;version\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rsplit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'.'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;except\&quot;,\n  \&quot;ValueError\&quot;,\n  \&quot;:\&quot;,\n  \&quot;version\&quot;,\n  \&quot;=\&quot;,\n  \&quot;pkg_data\&quot;,\n  \&quot;.\&quot;,\n  \&quot;version\&quot;,\n  \&quot;args\&quot;,\n  \&quot;=\&quot;,\n  \&quot;' '\&quot;,\n  \&quot;.\&quot;,\n  \&quot;join\&quot;,\n  \&quot;(\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'sphinx-quickstart'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'--sep'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'-q'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'-p \\\&quot;{name}\\\&quot;'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'-a \\\&quot;{author}\\\&quot;'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'-v \\\&quot;{version}\\\&quot;'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'-r \\\&quot;{release}\\\&quot;'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'-l en'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'--suffix=.rst'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'--master=index'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'--ext-autodoc'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'--ext-viewcode'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'--makefile'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'{projectdir}'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;.\&quot;,\n  \&quot;format\&quot;,\n  \&quot;(\&quot;,\n  \&quot;name\&quot;,\n  \&quot;=\&quot;,\n  \&quot;pkg_data\&quot;,\n  \&quot;.\&quot;,\n  \&quot;name\&quot;,\n  \&quot;,\&quot;,\n  \&quot;author\&quot;,\n  \&quot;=\&quot;,\n  \&quot;pkg_data\&quot;,\n  \&quot;.\&quot;,\n  \&quot;author\&quot;,\n  \&quot;,\&quot;,\n  \&quot;version\&quot;,\n  \&quot;=\&quot;,\n  \&quot;version\&quot;,\n  \&quot;,\&quot;,\n  \&quot;release\&quot;,\n  \&quot;=\&quot;,\n  \&quot;pkg_data\&quot;,\n  \&quot;.\&quot;,\n  \&quot;version\&quot;,\n  \&quot;,\&quot;,\n  \&quot;projectdir\&quot;,\n  \&quot;=\&quot;,\n  \&quot;projectdir\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;subprocess\&quot;,\n  \&quot;.\&quot;,\n  \&quot;call\&quot;,\n  \&quot;(\&quot;,\n  \&quot;shlex\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;args\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;==\&quot;,\n  \&quot;0\&quot;,\n  \&quot;:\&quot;,\n  \&quot;_touch_gitkeep\&quot;,\n  \&quot;(\&quot;,\n  \&quot;projectdir\&quot;,\n  \&quot;)\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Build sphinx documentation.\n\n    :rtype: int\n    :return: subprocess.call return code\n\n    :param `bootstrap_py.control.PackageData` pkg_data: package meta data\n    :param str projectdir: project root directory&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Build&quot;,&quot;sphinx&quot;,&quot;documentation&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Build\&quot;,\n  \&quot;sphinx\&quot;,\n  \&quot;documentation\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;95d56ed98ef409fd9f019dc352fd1c3711533275&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/mkouhei/bootstrap-py/blob/95d56ed98ef409fd9f019dc352fd1c3711533275/bootstrap_py/docs.py#L8-L40&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:43,&quot;string&quot;:&quot;43&quot;}}},{&quot;rowIdx&quot;:44,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;christophertbrown/bioscripts&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ctbBio/crossmap.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;bowtiedb&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def bowtiedb(fa, keepDB):\n    \&quot;\&quot;\&quot;\n    make bowtie db\n    \&quot;\&quot;\&quot;\n    btdir = '%s/bt2' % (os.getcwd())\n    # make directory for\n    if not os.path.exists(btdir):\n        os.mkdir(btdir)\n    btdb = '%s/%s' % (btdir, fa.rsplit('/', 1)[-1])\n    if keepDB is True:\n        if os.path.exists('%s.1.bt2' % (btdb)):\n            return btdb\n    p = subprocess.Popen('bowtie2-build -q %s %s' \\\n        % (fa, btdb), shell = True)\n    p.communicate()\n    return btdb&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def bowtiedb(fa, keepDB):\n    \&quot;\&quot;\&quot;\n    make bowtie db\n    \&quot;\&quot;\&quot;\n    btdir = '%s/bt2' % (os.getcwd())\n    # make directory for\n    if not os.path.exists(btdir):\n        os.mkdir(btdir)\n    btdb = '%s/%s' % (btdir, fa.rsplit('/', 1)[-1])\n    if keepDB is True:\n        if os.path.exists('%s.1.bt2' % (btdb)):\n            return btdb\n    p = subprocess.Popen('bowtie2-build -q %s %s' \\\n        % (fa, btdb), shell = True)\n    p.communicate()\n    return btdb&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;bowtiedb&quot;,&quot;(&quot;,&quot;fa&quot;,&quot;,&quot;,&quot;keepDB&quot;,&quot;)&quot;,&quot;:&quot;,&quot;btdir&quot;,&quot;=&quot;,&quot;'%s/bt2'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;os&quot;,&quot;.&quot;,&quot;getcwd&quot;,&quot;(&quot;,&quot;)&quot;,&quot;)&quot;,&quot;# make directory for&quot;,&quot;if&quot;,&quot;not&quot;,&quot;os&quot;,&quot;.&quot;,&quot;path&quot;,&quot;.&quot;,&quot;exists&quot;,&quot;(&quot;,&quot;btdir&quot;,&quot;)&quot;,&quot;:&quot;,&quot;os&quot;,&quot;.&quot;,&quot;mkdir&quot;,&quot;(&quot;,&quot;btdir&quot;,&quot;)&quot;,&quot;btdb&quot;,&quot;=&quot;,&quot;'%s/%s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;btdir&quot;,&quot;,&quot;,&quot;fa&quot;,&quot;.&quot;,&quot;rsplit&quot;,&quot;(&quot;,&quot;'/'&quot;,&quot;,&quot;,&quot;1&quot;,&quot;)&quot;,&quot;[&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;,&quot;)&quot;,&quot;if&quot;,&quot;keepDB&quot;,&quot;is&quot;,&quot;True&quot;,&quot;:&quot;,&quot;if&quot;,&quot;os&quot;,&quot;.&quot;,&quot;path&quot;,&quot;.&quot;,&quot;exists&quot;,&quot;(&quot;,&quot;'%s.1.bt2'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;btdb&quot;,&quot;)&quot;,&quot;)&quot;,&quot;:&quot;,&quot;return&quot;,&quot;btdb&quot;,&quot;p&quot;,&quot;=&quot;,&quot;subprocess&quot;,&quot;.&quot;,&quot;Popen&quot;,&quot;(&quot;,&quot;'bowtie2-build -q %s %s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;fa&quot;,&quot;,&quot;,&quot;btdb&quot;,&quot;)&quot;,&quot;,&quot;,&quot;shell&quot;,&quot;=&quot;,&quot;True&quot;,&quot;)&quot;,&quot;p&quot;,&quot;.&quot;,&quot;communicate&quot;,&quot;(&quot;,&quot;)&quot;,&quot;return&quot;,&quot;btdb&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;bowtiedb\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fa\&quot;,\n  \&quot;,\&quot;,\n  \&quot;keepDB\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;btdir\&quot;,\n  \&quot;=\&quot;,\n  \&quot;'%s/bt2'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;getcwd\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;# make directory for\&quot;,\n  \&quot;if\&quot;,\n  \&quot;not\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;path\&quot;,\n  \&quot;.\&quot;,\n  \&quot;exists\&quot;,\n  \&quot;(\&quot;,\n  \&quot;btdir\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;mkdir\&quot;,\n  \&quot;(\&quot;,\n  \&quot;btdir\&quot;,\n  \&quot;)\&quot;,\n  \&quot;btdb\&quot;,\n  \&quot;=\&quot;,\n  \&quot;'%s/%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;btdir\&quot;,\n  \&quot;,\&quot;,\n  \&quot;fa\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rsplit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'/'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;keepDB\&quot;,\n  \&quot;is\&quot;,\n  \&quot;True\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;path\&quot;,\n  \&quot;.\&quot;,\n  \&quot;exists\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'%s.1.bt2'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;btdb\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;btdb\&quot;,\n  \&quot;p\&quot;,\n  \&quot;=\&quot;,\n  \&quot;subprocess\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Popen\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'bowtie2-build -q %s %s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fa\&quot;,\n  \&quot;,\&quot;,\n  \&quot;btdb\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;shell\&quot;,\n  \&quot;=\&quot;,\n  \&quot;True\&quot;,\n  \&quot;)\&quot;,\n  \&quot;p\&quot;,\n  \&quot;.\&quot;,\n  \&quot;communicate\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;btdb\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;make bowtie db&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;make&quot;,&quot;bowtie&quot;,&quot;db&quot;],&quot;string&quot;:&quot;[\n  \&quot;make\&quot;,\n  \&quot;bowtie\&quot;,\n  \&quot;db\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;83b2566b3a5745437ec651cd6cafddd056846240&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/crossmap.py#L16-L31&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:44,&quot;string&quot;:&quot;44&quot;}}},{&quot;rowIdx&quot;:45,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;christophertbrown/bioscripts&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ctbBio/crossmap.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;bowtie&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def bowtie(sam, btd, f, r, u, opt, no_shrink, threads):\n    \&quot;\&quot;\&quot;\n    generate bowtie2 command\n    \&quot;\&quot;\&quot;\n    bt2 = 'bowtie2 -x %s -p %s ' % (btd, threads)\n    if f is not False:\n        bt2 += '-1 %s -2 %s ' % (f, r)\n    if u is not False:\n        bt2 += '-U %s ' % (u)\n    bt2 += opt\n    if no_shrink is False:\n        if f is False:\n            bt2 += ' | shrinksam -u -k %s-shrunk.sam ' % (sam)\n        else:\n            bt2 += ' | shrinksam -k %s-shrunk.sam ' % (sam)\n    else:\n        bt2 += ' > %s.sam' % (sam)\n    return bt2&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def bowtie(sam, btd, f, r, u, opt, no_shrink, threads):\n    \&quot;\&quot;\&quot;\n    generate bowtie2 command\n    \&quot;\&quot;\&quot;\n    bt2 = 'bowtie2 -x %s -p %s ' % (btd, threads)\n    if f is not False:\n        bt2 += '-1 %s -2 %s ' % (f, r)\n    if u is not False:\n        bt2 += '-U %s ' % (u)\n    bt2 += opt\n    if no_shrink is False:\n        if f is False:\n            bt2 += ' | shrinksam -u -k %s-shrunk.sam ' % (sam)\n        else:\n            bt2 += ' | shrinksam -k %s-shrunk.sam ' % (sam)\n    else:\n        bt2 += ' > %s.sam' % (sam)\n    return bt2&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;bowtie&quot;,&quot;(&quot;,&quot;sam&quot;,&quot;,&quot;,&quot;btd&quot;,&quot;,&quot;,&quot;f&quot;,&quot;,&quot;,&quot;r&quot;,&quot;,&quot;,&quot;u&quot;,&quot;,&quot;,&quot;opt&quot;,&quot;,&quot;,&quot;no_shrink&quot;,&quot;,&quot;,&quot;threads&quot;,&quot;)&quot;,&quot;:&quot;,&quot;bt2&quot;,&quot;=&quot;,&quot;'bowtie2 -x %s -p %s '&quot;,&quot;%&quot;,&quot;(&quot;,&quot;btd&quot;,&quot;,&quot;,&quot;threads&quot;,&quot;)&quot;,&quot;if&quot;,&quot;f&quot;,&quot;is&quot;,&quot;not&quot;,&quot;False&quot;,&quot;:&quot;,&quot;bt2&quot;,&quot;+=&quot;,&quot;'-1 %s -2 %s '&quot;,&quot;%&quot;,&quot;(&quot;,&quot;f&quot;,&quot;,&quot;,&quot;r&quot;,&quot;)&quot;,&quot;if&quot;,&quot;u&quot;,&quot;is&quot;,&quot;not&quot;,&quot;False&quot;,&quot;:&quot;,&quot;bt2&quot;,&quot;+=&quot;,&quot;'-U %s '&quot;,&quot;%&quot;,&quot;(&quot;,&quot;u&quot;,&quot;)&quot;,&quot;bt2&quot;,&quot;+=&quot;,&quot;opt&quot;,&quot;if&quot;,&quot;no_shrink&quot;,&quot;is&quot;,&quot;False&quot;,&quot;:&quot;,&quot;if&quot;,&quot;f&quot;,&quot;is&quot;,&quot;False&quot;,&quot;:&quot;,&quot;bt2&quot;,&quot;+=&quot;,&quot;' | shrinksam -u -k %s-shrunk.sam '&quot;,&quot;%&quot;,&quot;(&quot;,&quot;sam&quot;,&quot;)&quot;,&quot;else&quot;,&quot;:&quot;,&quot;bt2&quot;,&quot;+=&quot;,&quot;' | shrinksam -k %s-shrunk.sam '&quot;,&quot;%&quot;,&quot;(&quot;,&quot;sam&quot;,&quot;)&quot;,&quot;else&quot;,&quot;:&quot;,&quot;bt2&quot;,&quot;+=&quot;,&quot;' > %s.sam'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;sam&quot;,&quot;)&quot;,&quot;return&quot;,&quot;bt2&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;bowtie\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;,\&quot;,\n  \&quot;btd\&quot;,\n  \&quot;,\&quot;,\n  \&quot;f\&quot;,\n  \&quot;,\&quot;,\n  \&quot;r\&quot;,\n  \&quot;,\&quot;,\n  \&quot;u\&quot;,\n  \&quot;,\&quot;,\n  \&quot;opt\&quot;,\n  \&quot;,\&quot;,\n  \&quot;no_shrink\&quot;,\n  \&quot;,\&quot;,\n  \&quot;threads\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;bt2\&quot;,\n  \&quot;=\&quot;,\n  \&quot;'bowtie2 -x %s -p %s '\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;btd\&quot;,\n  \&quot;,\&quot;,\n  \&quot;threads\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;f\&quot;,\n  \&quot;is\&quot;,\n  \&quot;not\&quot;,\n  \&quot;False\&quot;,\n  \&quot;:\&quot;,\n  \&quot;bt2\&quot;,\n  \&quot;+=\&quot;,\n  \&quot;'-1 %s -2 %s '\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;f\&quot;,\n  \&quot;,\&quot;,\n  \&quot;r\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;u\&quot;,\n  \&quot;is\&quot;,\n  \&quot;not\&quot;,\n  \&quot;False\&quot;,\n  \&quot;:\&quot;,\n  \&quot;bt2\&quot;,\n  \&quot;+=\&quot;,\n  \&quot;'-U %s '\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;u\&quot;,\n  \&quot;)\&quot;,\n  \&quot;bt2\&quot;,\n  \&quot;+=\&quot;,\n  \&quot;opt\&quot;,\n  \&quot;if\&quot;,\n  \&quot;no_shrink\&quot;,\n  \&quot;is\&quot;,\n  \&quot;False\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;f\&quot;,\n  \&quot;is\&quot;,\n  \&quot;False\&quot;,\n  \&quot;:\&quot;,\n  \&quot;bt2\&quot;,\n  \&quot;+=\&quot;,\n  \&quot;' | shrinksam -u -k %s-shrunk.sam '\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;)\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;bt2\&quot;,\n  \&quot;+=\&quot;,\n  \&quot;' | shrinksam -k %s-shrunk.sam '\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;)\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;bt2\&quot;,\n  \&quot;+=\&quot;,\n  \&quot;' > %s.sam'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;bt2\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;generate bowtie2 command&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;generate&quot;,&quot;bowtie2&quot;,&quot;command&quot;],&quot;string&quot;:&quot;[\n  \&quot;generate\&quot;,\n  \&quot;bowtie2\&quot;,\n  \&quot;command\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;83b2566b3a5745437ec651cd6cafddd056846240&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/crossmap.py#L33-L50&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:45,&quot;string&quot;:&quot;45&quot;}}},{&quot;rowIdx&quot;:46,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;christophertbrown/bioscripts&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ctbBio/crossmap.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;crossmap&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def crossmap(fas, reads, options, no_shrink, keepDB, threads, cluster, nodes):\n    \&quot;\&quot;\&quot;\n    map all read sets against all fasta files\n    \&quot;\&quot;\&quot;\n    if cluster is True:\n        threads = '48'\n    btc = []\n    for fa in fas:\n        btd = bowtiedb(fa, keepDB)\n        F, R, U = reads\n        if F is not False:\n            if U is False:\n                u = False\n            for i, f in enumerate(F):\n                r = R[i]\n                if U is not False:\n                    u = U[i]\n                sam = '%s/%s-vs-%s' % (os.getcwd(), \\\n                        fa.rsplit('/', 1)[-1], f.rsplit('/', 1)[-1].rsplit('.', 3)[0])\n                btc.append(bowtie(sam, btd, f, r, u, options, no_shrink, threads))\n        else:\n            f = False\n            r = False\n            for u in U:\n                sam = '%s/%s-vs-%s' % (os.getcwd(), \\\n                        fa.rsplit('/', 1)[-1], u.rsplit('/', 1)[-1].rsplit('.', 3)[0])\n                btc.append(bowtie(sam, btd, f, r, u, options, no_shrink, threads))\n    if cluster is False:\n        for i in btc:\n            p = subprocess.Popen(i, shell = True)\n            p.communicate()\n    else:\n        ID = ''.join(random.choice([str(i) for i in range(0, 9)]) for _ in range(5))\n        for node, commands in enumerate(chunks(btc, nodes), 1):\n            bs = open('%s/crossmap-qsub.%s.%s.sh' % (os.getcwd(), ID, node), 'w')\n            print('\\n'.join(commands), file=bs)\n            bs.close()\n            p = subprocess.Popen(\\\n                    'qsub -V -N crossmap %s' \\\n                        % (bs.name), \\\n                    shell = True)\n            p.communicate()&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def crossmap(fas, reads, options, no_shrink, keepDB, threads, cluster, nodes):\n    \&quot;\&quot;\&quot;\n    map all read sets against all fasta files\n    \&quot;\&quot;\&quot;\n    if cluster is True:\n        threads = '48'\n    btc = []\n    for fa in fas:\n        btd = bowtiedb(fa, keepDB)\n        F, R, U = reads\n        if F is not False:\n            if U is False:\n                u = False\n            for i, f in enumerate(F):\n                r = R[i]\n                if U is not False:\n                    u = U[i]\n                sam = '%s/%s-vs-%s' % (os.getcwd(), \\\n                        fa.rsplit('/', 1)[-1], f.rsplit('/', 1)[-1].rsplit('.', 3)[0])\n                btc.append(bowtie(sam, btd, f, r, u, options, no_shrink, threads))\n        else:\n            f = False\n            r = False\n            for u in U:\n                sam = '%s/%s-vs-%s' % (os.getcwd(), \\\n                        fa.rsplit('/', 1)[-1], u.rsplit('/', 1)[-1].rsplit('.', 3)[0])\n                btc.append(bowtie(sam, btd, f, r, u, options, no_shrink, threads))\n    if cluster is False:\n        for i in btc:\n            p = subprocess.Popen(i, shell = True)\n            p.communicate()\n    else:\n        ID = ''.join(random.choice([str(i) for i in range(0, 9)]) for _ in range(5))\n        for node, commands in enumerate(chunks(btc, nodes), 1):\n            bs = open('%s/crossmap-qsub.%s.%s.sh' % (os.getcwd(), ID, node), 'w')\n            print('\\n'.join(commands), file=bs)\n            bs.close()\n            p = subprocess.Popen(\\\n                    'qsub -V -N crossmap %s' \\\n                        % (bs.name), \\\n                    shell = True)\n            p.communicate()&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;crossmap&quot;,&quot;(&quot;,&quot;fas&quot;,&quot;,&quot;,&quot;reads&quot;,&quot;,&quot;,&quot;options&quot;,&quot;,&quot;,&quot;no_shrink&quot;,&quot;,&quot;,&quot;keepDB&quot;,&quot;,&quot;,&quot;threads&quot;,&quot;,&quot;,&quot;cluster&quot;,&quot;,&quot;,&quot;nodes&quot;,&quot;)&quot;,&quot;:&quot;,&quot;if&quot;,&quot;cluster&quot;,&quot;is&quot;,&quot;True&quot;,&quot;:&quot;,&quot;threads&quot;,&quot;=&quot;,&quot;'48'&quot;,&quot;btc&quot;,&quot;=&quot;,&quot;[&quot;,&quot;]&quot;,&quot;for&quot;,&quot;fa&quot;,&quot;in&quot;,&quot;fas&quot;,&quot;:&quot;,&quot;btd&quot;,&quot;=&quot;,&quot;bowtiedb&quot;,&quot;(&quot;,&quot;fa&quot;,&quot;,&quot;,&quot;keepDB&quot;,&quot;)&quot;,&quot;F&quot;,&quot;,&quot;,&quot;R&quot;,&quot;,&quot;,&quot;U&quot;,&quot;=&quot;,&quot;reads&quot;,&quot;if&quot;,&quot;F&quot;,&quot;is&quot;,&quot;not&quot;,&quot;False&quot;,&quot;:&quot;,&quot;if&quot;,&quot;U&quot;,&quot;is&quot;,&quot;False&quot;,&quot;:&quot;,&quot;u&quot;,&quot;=&quot;,&quot;False&quot;,&quot;for&quot;,&quot;i&quot;,&quot;,&quot;,&quot;f&quot;,&quot;in&quot;,&quot;enumerate&quot;,&quot;(&quot;,&quot;F&quot;,&quot;)&quot;,&quot;:&quot;,&quot;r&quot;,&quot;=&quot;,&quot;R&quot;,&quot;[&quot;,&quot;i&quot;,&quot;]&quot;,&quot;if&quot;,&quot;U&quot;,&quot;is&quot;,&quot;not&quot;,&quot;False&quot;,&quot;:&quot;,&quot;u&quot;,&quot;=&quot;,&quot;U&quot;,&quot;[&quot;,&quot;i&quot;,&quot;]&quot;,&quot;sam&quot;,&quot;=&quot;,&quot;'%s/%s-vs-%s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;os&quot;,&quot;.&quot;,&quot;getcwd&quot;,&quot;(&quot;,&quot;)&quot;,&quot;,&quot;,&quot;fa&quot;,&quot;.&quot;,&quot;rsplit&quot;,&quot;(&quot;,&quot;'/'&quot;,&quot;,&quot;,&quot;1&quot;,&quot;)&quot;,&quot;[&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;,&quot;,&quot;,&quot;f&quot;,&quot;.&quot;,&quot;rsplit&quot;,&quot;(&quot;,&quot;'/'&quot;,&quot;,&quot;,&quot;1&quot;,&quot;)&quot;,&quot;[&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;,&quot;.&quot;,&quot;rsplit&quot;,&quot;(&quot;,&quot;'.'&quot;,&quot;,&quot;,&quot;3&quot;,&quot;)&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;)&quot;,&quot;btc&quot;,&quot;.&quot;,&quot;append&quot;,&quot;(&quot;,&quot;bowtie&quot;,&quot;(&quot;,&quot;sam&quot;,&quot;,&quot;,&quot;btd&quot;,&quot;,&quot;,&quot;f&quot;,&quot;,&quot;,&quot;r&quot;,&quot;,&quot;,&quot;u&quot;,&quot;,&quot;,&quot;options&quot;,&quot;,&quot;,&quot;no_shrink&quot;,&quot;,&quot;,&quot;threads&quot;,&quot;)&quot;,&quot;)&quot;,&quot;else&quot;,&quot;:&quot;,&quot;f&quot;,&quot;=&quot;,&quot;False&quot;,&quot;r&quot;,&quot;=&quot;,&quot;False&quot;,&quot;for&quot;,&quot;u&quot;,&quot;in&quot;,&quot;U&quot;,&quot;:&quot;,&quot;sam&quot;,&quot;=&quot;,&quot;'%s/%s-vs-%s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;os&quot;,&quot;.&quot;,&quot;getcwd&quot;,&quot;(&quot;,&quot;)&quot;,&quot;,&quot;,&quot;fa&quot;,&quot;.&quot;,&quot;rsplit&quot;,&quot;(&quot;,&quot;'/'&quot;,&quot;,&quot;,&quot;1&quot;,&quot;)&quot;,&quot;[&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;,&quot;,&quot;,&quot;u&quot;,&quot;.&quot;,&quot;rsplit&quot;,&quot;(&quot;,&quot;'/'&quot;,&quot;,&quot;,&quot;1&quot;,&quot;)&quot;,&quot;[&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;,&quot;.&quot;,&quot;rsplit&quot;,&quot;(&quot;,&quot;'.'&quot;,&quot;,&quot;,&quot;3&quot;,&quot;)&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;)&quot;,&quot;btc&quot;,&quot;.&quot;,&quot;append&quot;,&quot;(&quot;,&quot;bowtie&quot;,&quot;(&quot;,&quot;sam&quot;,&quot;,&quot;,&quot;btd&quot;,&quot;,&quot;,&quot;f&quot;,&quot;,&quot;,&quot;r&quot;,&quot;,&quot;,&quot;u&quot;,&quot;,&quot;,&quot;options&quot;,&quot;,&quot;,&quot;no_shrink&quot;,&quot;,&quot;,&quot;threads&quot;,&quot;)&quot;,&quot;)&quot;,&quot;if&quot;,&quot;cluster&quot;,&quot;is&quot;,&quot;False&quot;,&quot;:&quot;,&quot;for&quot;,&quot;i&quot;,&quot;in&quot;,&quot;btc&quot;,&quot;:&quot;,&quot;p&quot;,&quot;=&quot;,&quot;subprocess&quot;,&quot;.&quot;,&quot;Popen&quot;,&quot;(&quot;,&quot;i&quot;,&quot;,&quot;,&quot;shell&quot;,&quot;=&quot;,&quot;True&quot;,&quot;)&quot;,&quot;p&quot;,&quot;.&quot;,&quot;communicate&quot;,&quot;(&quot;,&quot;)&quot;,&quot;else&quot;,&quot;:&quot;,&quot;ID&quot;,&quot;=&quot;,&quot;''&quot;,&quot;.&quot;,&quot;join&quot;,&quot;(&quot;,&quot;random&quot;,&quot;.&quot;,&quot;choice&quot;,&quot;(&quot;,&quot;[&quot;,&quot;str&quot;,&quot;(&quot;,&quot;i&quot;,&quot;)&quot;,&quot;for&quot;,&quot;i&quot;,&quot;in&quot;,&quot;range&quot;,&quot;(&quot;,&quot;0&quot;,&quot;,&quot;,&quot;9&quot;,&quot;)&quot;,&quot;]&quot;,&quot;)&quot;,&quot;for&quot;,&quot;_&quot;,&quot;in&quot;,&quot;range&quot;,&quot;(&quot;,&quot;5&quot;,&quot;)&quot;,&quot;)&quot;,&quot;for&quot;,&quot;node&quot;,&quot;,&quot;,&quot;commands&quot;,&quot;in&quot;,&quot;enumerate&quot;,&quot;(&quot;,&quot;chunks&quot;,&quot;(&quot;,&quot;btc&quot;,&quot;,&quot;,&quot;nodes&quot;,&quot;)&quot;,&quot;,&quot;,&quot;1&quot;,&quot;)&quot;,&quot;:&quot;,&quot;bs&quot;,&quot;=&quot;,&quot;open&quot;,&quot;(&quot;,&quot;'%s/crossmap-qsub.%s.%s.sh'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;os&quot;,&quot;.&quot;,&quot;getcwd&quot;,&quot;(&quot;,&quot;)&quot;,&quot;,&quot;,&quot;ID&quot;,&quot;,&quot;,&quot;node&quot;,&quot;)&quot;,&quot;,&quot;,&quot;'w'&quot;,&quot;)&quot;,&quot;print&quot;,&quot;(&quot;,&quot;'\\n'&quot;,&quot;.&quot;,&quot;join&quot;,&quot;(&quot;,&quot;commands&quot;,&quot;)&quot;,&quot;,&quot;,&quot;file&quot;,&quot;=&quot;,&quot;bs&quot;,&quot;)&quot;,&quot;bs&quot;,&quot;.&quot;,&quot;close&quot;,&quot;(&quot;,&quot;)&quot;,&quot;p&quot;,&quot;=&quot;,&quot;subprocess&quot;,&quot;.&quot;,&quot;Popen&quot;,&quot;(&quot;,&quot;'qsub -V -N crossmap %s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;bs&quot;,&quot;.&quot;,&quot;name&quot;,&quot;)&quot;,&quot;,&quot;,&quot;shell&quot;,&quot;=&quot;,&quot;True&quot;,&quot;)&quot;,&quot;p&quot;,&quot;.&quot;,&quot;communicate&quot;,&quot;(&quot;,&quot;)&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;crossmap\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fas\&quot;,\n  \&quot;,\&quot;,\n  \&quot;reads\&quot;,\n  \&quot;,\&quot;,\n  \&quot;options\&quot;,\n  \&quot;,\&quot;,\n  \&quot;no_shrink\&quot;,\n  \&quot;,\&quot;,\n  \&quot;keepDB\&quot;,\n  \&quot;,\&quot;,\n  \&quot;threads\&quot;,\n  \&quot;,\&quot;,\n  \&quot;cluster\&quot;,\n  \&quot;,\&quot;,\n  \&quot;nodes\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;cluster\&quot;,\n  \&quot;is\&quot;,\n  \&quot;True\&quot;,\n  \&quot;:\&quot;,\n  \&quot;threads\&quot;,\n  \&quot;=\&quot;,\n  \&quot;'48'\&quot;,\n  \&quot;btc\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;]\&quot;,\n  \&quot;for\&quot;,\n  \&quot;fa\&quot;,\n  \&quot;in\&quot;,\n  \&quot;fas\&quot;,\n  \&quot;:\&quot;,\n  \&quot;btd\&quot;,\n  \&quot;=\&quot;,\n  \&quot;bowtiedb\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fa\&quot;,\n  \&quot;,\&quot;,\n  \&quot;keepDB\&quot;,\n  \&quot;)\&quot;,\n  \&quot;F\&quot;,\n  \&quot;,\&quot;,\n  \&quot;R\&quot;,\n  \&quot;,\&quot;,\n  \&quot;U\&quot;,\n  \&quot;=\&quot;,\n  \&quot;reads\&quot;,\n  \&quot;if\&quot;,\n  \&quot;F\&quot;,\n  \&quot;is\&quot;,\n  \&quot;not\&quot;,\n  \&quot;False\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;U\&quot;,\n  \&quot;is\&quot;,\n  \&quot;False\&quot;,\n  \&quot;:\&quot;,\n  \&quot;u\&quot;,\n  \&quot;=\&quot;,\n  \&quot;False\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;,\&quot;,\n  \&quot;f\&quot;,\n  \&quot;in\&quot;,\n  \&quot;enumerate\&quot;,\n  \&quot;(\&quot;,\n  \&quot;F\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;r\&quot;,\n  \&quot;=\&quot;,\n  \&quot;R\&quot;,\n  \&quot;[\&quot;,\n  \&quot;i\&quot;,\n  \&quot;]\&quot;,\n  \&quot;if\&quot;,\n  \&quot;U\&quot;,\n  \&quot;is\&quot;,\n  \&quot;not\&quot;,\n  \&quot;False\&quot;,\n  \&quot;:\&quot;,\n  \&quot;u\&quot;,\n  \&quot;=\&quot;,\n  \&quot;U\&quot;,\n  \&quot;[\&quot;,\n  \&quot;i\&quot;,\n  \&quot;]\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;=\&quot;,\n  \&quot;'%s/%s-vs-%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;getcwd\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;fa\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rsplit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'/'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;f\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rsplit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'/'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rsplit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'.'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;3\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;btc\&quot;,\n  \&quot;.\&quot;,\n  \&quot;append\&quot;,\n  \&quot;(\&quot;,\n  \&quot;bowtie\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;,\&quot;,\n  \&quot;btd\&quot;,\n  \&quot;,\&quot;,\n  \&quot;f\&quot;,\n  \&quot;,\&quot;,\n  \&quot;r\&quot;,\n  \&quot;,\&quot;,\n  \&quot;u\&quot;,\n  \&quot;,\&quot;,\n  \&quot;options\&quot;,\n  \&quot;,\&quot;,\n  \&quot;no_shrink\&quot;,\n  \&quot;,\&quot;,\n  \&quot;threads\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;f\&quot;,\n  \&quot;=\&quot;,\n  \&quot;False\&quot;,\n  \&quot;r\&quot;,\n  \&quot;=\&quot;,\n  \&quot;False\&quot;,\n  \&quot;for\&quot;,\n  \&quot;u\&quot;,\n  \&quot;in\&quot;,\n  \&quot;U\&quot;,\n  \&quot;:\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;=\&quot;,\n  \&quot;'%s/%s-vs-%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;getcwd\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;fa\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rsplit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'/'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;u\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rsplit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'/'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rsplit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'.'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;3\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;btc\&quot;,\n  \&quot;.\&quot;,\n  \&quot;append\&quot;,\n  \&quot;(\&quot;,\n  \&quot;bowtie\&quot;,\n  \&quot;(\&quot;,\n  \&quot;sam\&quot;,\n  \&quot;,\&quot;,\n  \&quot;btd\&quot;,\n  \&quot;,\&quot;,\n  \&quot;f\&quot;,\n  \&quot;,\&quot;,\n  \&quot;r\&quot;,\n  \&quot;,\&quot;,\n  \&quot;u\&quot;,\n  \&quot;,\&quot;,\n  \&quot;options\&quot;,\n  \&quot;,\&quot;,\n  \&quot;no_shrink\&quot;,\n  \&quot;,\&quot;,\n  \&quot;threads\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;cluster\&quot;,\n  \&quot;is\&quot;,\n  \&quot;False\&quot;,\n  \&quot;:\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;in\&quot;,\n  \&quot;btc\&quot;,\n  \&quot;:\&quot;,\n  \&quot;p\&quot;,\n  \&quot;=\&quot;,\n  \&quot;subprocess\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Popen\&quot;,\n  \&quot;(\&quot;,\n  \&quot;i\&quot;,\n  \&quot;,\&quot;,\n  \&quot;shell\&quot;,\n  \&quot;=\&quot;,\n  \&quot;True\&quot;,\n  \&quot;)\&quot;,\n  \&quot;p\&quot;,\n  \&quot;.\&quot;,\n  \&quot;communicate\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;ID\&quot;,\n  \&quot;=\&quot;,\n  \&quot;''\&quot;,\n  \&quot;.\&quot;,\n  \&quot;join\&quot;,\n  \&quot;(\&quot;,\n  \&quot;random\&quot;,\n  \&quot;.\&quot;,\n  \&quot;choice\&quot;,\n  \&quot;(\&quot;,\n  \&quot;[\&quot;,\n  \&quot;str\&quot;,\n  \&quot;(\&quot;,\n  \&quot;i\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;in\&quot;,\n  \&quot;range\&quot;,\n  \&quot;(\&quot;,\n  \&quot;0\&quot;,\n  \&quot;,\&quot;,\n  \&quot;9\&quot;,\n  \&quot;)\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;_\&quot;,\n  \&quot;in\&quot;,\n  \&quot;range\&quot;,\n  \&quot;(\&quot;,\n  \&quot;5\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;node\&quot;,\n  \&quot;,\&quot;,\n  \&quot;commands\&quot;,\n  \&quot;in\&quot;,\n  \&quot;enumerate\&quot;,\n  \&quot;(\&quot;,\n  \&quot;chunks\&quot;,\n  \&quot;(\&quot;,\n  \&quot;btc\&quot;,\n  \&quot;,\&quot;,\n  \&quot;nodes\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;bs\&quot;,\n  \&quot;=\&quot;,\n  \&quot;open\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'%s/crossmap-qsub.%s.%s.sh'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;os\&quot;,\n  \&quot;.\&quot;,\n  \&quot;getcwd\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;ID\&quot;,\n  \&quot;,\&quot;,\n  \&quot;node\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'w'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;print\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'\\\\n'\&quot;,\n  \&quot;.\&quot;,\n  \&quot;join\&quot;,\n  \&quot;(\&quot;,\n  \&quot;commands\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;file\&quot;,\n  \&quot;=\&quot;,\n  \&quot;bs\&quot;,\n  \&quot;)\&quot;,\n  \&quot;bs\&quot;,\n  \&quot;.\&quot;,\n  \&quot;close\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;p\&quot;,\n  \&quot;=\&quot;,\n  \&quot;subprocess\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Popen\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'qsub -V -N crossmap %s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;bs\&quot;,\n  \&quot;.\&quot;,\n  \&quot;name\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;shell\&quot;,\n  \&quot;=\&quot;,\n  \&quot;True\&quot;,\n  \&quot;)\&quot;,\n  \&quot;p\&quot;,\n  \&quot;.\&quot;,\n  \&quot;communicate\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;map all read sets against all fasta files&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;map&quot;,&quot;all&quot;,&quot;read&quot;,&quot;sets&quot;,&quot;against&quot;,&quot;all&quot;,&quot;fasta&quot;,&quot;files&quot;],&quot;string&quot;:&quot;[\n  \&quot;map\&quot;,\n  \&quot;all\&quot;,\n  \&quot;read\&quot;,\n  \&quot;sets\&quot;,\n  \&quot;against\&quot;,\n  \&quot;all\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;files\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;83b2566b3a5745437ec651cd6cafddd056846240&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/crossmap.py#L55-L96&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:46,&quot;string&quot;:&quot;46&quot;}}},{&quot;rowIdx&quot;:47,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;disqus/nydus&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;nydus/db/base.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;BaseCluster.get_conn&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def get_conn(self, *args, **kwargs):\n        \&quot;\&quot;\&quot;\n        Returns a connection object from the router given ``args``.\n\n        Useful in cases where a connection cannot be automatically determined\n        during all steps of the process. An example of this would be\n        Redis pipelines.\n        \&quot;\&quot;\&quot;\n        connections = self.__connections_for('get_conn', args=args, kwargs=kwargs)\n\n        if len(connections) is 1:\n            return connections[0]\n        else:\n            return connections&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def get_conn(self, *args, **kwargs):\n        \&quot;\&quot;\&quot;\n        Returns a connection object from the router given ``args``.\n\n        Useful in cases where a connection cannot be automatically determined\n        during all steps of the process. An example of this would be\n        Redis pipelines.\n        \&quot;\&quot;\&quot;\n        connections = self.__connections_for('get_conn', args=args, kwargs=kwargs)\n\n        if len(connections) is 1:\n            return connections[0]\n        else:\n            return connections&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;get_conn&quot;,&quot;(&quot;,&quot;self&quot;,&quot;,&quot;,&quot;*&quot;,&quot;args&quot;,&quot;,&quot;,&quot;*&quot;,&quot;*&quot;,&quot;kwargs&quot;,&quot;)&quot;,&quot;:&quot;,&quot;connections&quot;,&quot;=&quot;,&quot;self&quot;,&quot;.&quot;,&quot;__connections_for&quot;,&quot;(&quot;,&quot;'get_conn'&quot;,&quot;,&quot;,&quot;args&quot;,&quot;=&quot;,&quot;args&quot;,&quot;,&quot;,&quot;kwargs&quot;,&quot;=&quot;,&quot;kwargs&quot;,&quot;)&quot;,&quot;if&quot;,&quot;len&quot;,&quot;(&quot;,&quot;connections&quot;,&quot;)&quot;,&quot;is&quot;,&quot;1&quot;,&quot;:&quot;,&quot;return&quot;,&quot;connections&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;else&quot;,&quot;:&quot;,&quot;return&quot;,&quot;connections&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;get_conn\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;,\&quot;,\n  \&quot;*\&quot;,\n  \&quot;args\&quot;,\n  \&quot;,\&quot;,\n  \&quot;*\&quot;,\n  \&quot;*\&quot;,\n  \&quot;kwargs\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;connections\&quot;,\n  \&quot;=\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;__connections_for\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'get_conn'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;args\&quot;,\n  \&quot;=\&quot;,\n  \&quot;args\&quot;,\n  \&quot;,\&quot;,\n  \&quot;kwargs\&quot;,\n  \&quot;=\&quot;,\n  \&quot;kwargs\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;len\&quot;,\n  \&quot;(\&quot;,\n  \&quot;connections\&quot;,\n  \&quot;)\&quot;,\n  \&quot;is\&quot;,\n  \&quot;1\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;connections\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;connections\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Returns a connection object from the router given ``args``.\n\n        Useful in cases where a connection cannot be automatically determined\n        during all steps of the process. An example of this would be\n        Redis pipelines.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Returns&quot;,&quot;a&quot;,&quot;connection&quot;,&quot;object&quot;,&quot;from&quot;,&quot;the&quot;,&quot;router&quot;,&quot;given&quot;,&quot;args&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Returns\&quot;,\n  \&quot;a\&quot;,\n  \&quot;connection\&quot;,\n  \&quot;object\&quot;,\n  \&quot;from\&quot;,\n  \&quot;the\&quot;,\n  \&quot;router\&quot;,\n  \&quot;given\&quot;,\n  \&quot;args\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;9b505840da47a34f758a830c3992fa5dcb7bb7ad&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/disqus/nydus/blob/9b505840da47a34f758a830c3992fa5dcb7bb7ad/nydus/db/base.py#L100-L113&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:47,&quot;string&quot;:&quot;47&quot;}}},{&quot;rowIdx&quot;:48,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;scottrice/pysteam&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;pysteam/_crc_algorithms.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Crc.__get_nondirect_init&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def __get_nondirect_init(self, init):\n        \&quot;\&quot;\&quot;\n        return the non-direct init if the direct algorithm has been selected.\n        \&quot;\&quot;\&quot;\n        crc = init\n        for i in range(self.Width):\n            bit = crc &amp; 0x01\n            if bit:\n                crc^= self.Poly\n            crc >>= 1\n            if bit:\n                crc |= self.MSB_Mask\n        return crc &amp; self.Mask&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def __get_nondirect_init(self, init):\n        \&quot;\&quot;\&quot;\n        return the non-direct init if the direct algorithm has been selected.\n        \&quot;\&quot;\&quot;\n        crc = init\n        for i in range(self.Width):\n            bit = crc &amp; 0x01\n            if bit:\n                crc^= self.Poly\n            crc >>= 1\n            if bit:\n                crc |= self.MSB_Mask\n        return crc &amp; self.Mask&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;__get_nondirect_init&quot;,&quot;(&quot;,&quot;self&quot;,&quot;,&quot;,&quot;init&quot;,&quot;)&quot;,&quot;:&quot;,&quot;crc&quot;,&quot;=&quot;,&quot;init&quot;,&quot;for&quot;,&quot;i&quot;,&quot;in&quot;,&quot;range&quot;,&quot;(&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Width&quot;,&quot;)&quot;,&quot;:&quot;,&quot;bit&quot;,&quot;=&quot;,&quot;crc&quot;,&quot;&amp;&quot;,&quot;0x01&quot;,&quot;if&quot;,&quot;bit&quot;,&quot;:&quot;,&quot;crc&quot;,&quot;^=&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Poly&quot;,&quot;crc&quot;,&quot;>>=&quot;,&quot;1&quot;,&quot;if&quot;,&quot;bit&quot;,&quot;:&quot;,&quot;crc&quot;,&quot;|=&quot;,&quot;self&quot;,&quot;.&quot;,&quot;MSB_Mask&quot;,&quot;return&quot;,&quot;crc&quot;,&quot;&amp;&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Mask&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;__get_nondirect_init\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;,\&quot;,\n  \&quot;init\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;crc\&quot;,\n  \&quot;=\&quot;,\n  \&quot;init\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;in\&quot;,\n  \&quot;range\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Width\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;bit\&quot;,\n  \&quot;=\&quot;,\n  \&quot;crc\&quot;,\n  \&quot;&amp;\&quot;,\n  \&quot;0x01\&quot;,\n  \&quot;if\&quot;,\n  \&quot;bit\&quot;,\n  \&quot;:\&quot;,\n  \&quot;crc\&quot;,\n  \&quot;^=\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Poly\&quot;,\n  \&quot;crc\&quot;,\n  \&quot;>>=\&quot;,\n  \&quot;1\&quot;,\n  \&quot;if\&quot;,\n  \&quot;bit\&quot;,\n  \&quot;:\&quot;,\n  \&quot;crc\&quot;,\n  \&quot;|=\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;MSB_Mask\&quot;,\n  \&quot;return\&quot;,\n  \&quot;crc\&quot;,\n  \&quot;&amp;\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Mask\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;return the non-direct init if the direct algorithm has been selected.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;return&quot;,&quot;the&quot;,&quot;non&quot;,&quot;-&quot;,&quot;direct&quot;,&quot;init&quot;,&quot;if&quot;,&quot;the&quot;,&quot;direct&quot;,&quot;algorithm&quot;,&quot;has&quot;,&quot;been&quot;,&quot;selected&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;return\&quot;,\n  \&quot;the\&quot;,\n  \&quot;non\&quot;,\n  \&quot;-\&quot;,\n  \&quot;direct\&quot;,\n  \&quot;init\&quot;,\n  \&quot;if\&quot;,\n  \&quot;the\&quot;,\n  \&quot;direct\&quot;,\n  \&quot;algorithm\&quot;,\n  \&quot;has\&quot;,\n  \&quot;been\&quot;,\n  \&quot;selected\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;1eb2254b5235a053a953e596fa7602d0b110245d&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/scottrice/pysteam/blob/1eb2254b5235a053a953e596fa7602d0b110245d/pysteam/_crc_algorithms.py#L98-L110&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:48,&quot;string&quot;:&quot;48&quot;}}},{&quot;rowIdx&quot;:49,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;scottrice/pysteam&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;pysteam/_crc_algorithms.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Crc.reflect&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def reflect(self, data, width):\n        \&quot;\&quot;\&quot;\n        reflect a data word, i.e. reverts the bit order.\n        \&quot;\&quot;\&quot;\n        x = data &amp; 0x01\n        for i in range(width - 1):\n            data >>= 1\n            x = (x << 1) | (data &amp; 0x01)\n        return x&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def reflect(self, data, width):\n        \&quot;\&quot;\&quot;\n        reflect a data word, i.e. reverts the bit order.\n        \&quot;\&quot;\&quot;\n        x = data &amp; 0x01\n        for i in range(width - 1):\n            data >>= 1\n            x = (x << 1) | (data &amp; 0x01)\n        return x&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;reflect&quot;,&quot;(&quot;,&quot;self&quot;,&quot;,&quot;,&quot;data&quot;,&quot;,&quot;,&quot;width&quot;,&quot;)&quot;,&quot;:&quot;,&quot;x&quot;,&quot;=&quot;,&quot;data&quot;,&quot;&amp;&quot;,&quot;0x01&quot;,&quot;for&quot;,&quot;i&quot;,&quot;in&quot;,&quot;range&quot;,&quot;(&quot;,&quot;width&quot;,&quot;-&quot;,&quot;1&quot;,&quot;)&quot;,&quot;:&quot;,&quot;data&quot;,&quot;>>=&quot;,&quot;1&quot;,&quot;x&quot;,&quot;=&quot;,&quot;(&quot;,&quot;x&quot;,&quot;<<&quot;,&quot;1&quot;,&quot;)&quot;,&quot;|&quot;,&quot;(&quot;,&quot;data&quot;,&quot;&amp;&quot;,&quot;0x01&quot;,&quot;)&quot;,&quot;return&quot;,&quot;x&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;reflect\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;,\&quot;,\n  \&quot;data\&quot;,\n  \&quot;,\&quot;,\n  \&quot;width\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;x\&quot;,\n  \&quot;=\&quot;,\n  \&quot;data\&quot;,\n  \&quot;&amp;\&quot;,\n  \&quot;0x01\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;in\&quot;,\n  \&quot;range\&quot;,\n  \&quot;(\&quot;,\n  \&quot;width\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;data\&quot;,\n  \&quot;>>=\&quot;,\n  \&quot;1\&quot;,\n  \&quot;x\&quot;,\n  \&quot;=\&quot;,\n  \&quot;(\&quot;,\n  \&quot;x\&quot;,\n  \&quot;<<\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;|\&quot;,\n  \&quot;(\&quot;,\n  \&quot;data\&quot;,\n  \&quot;&amp;\&quot;,\n  \&quot;0x01\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;x\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;reflect a data word, i.e. reverts the bit order.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;reflect&quot;,&quot;a&quot;,&quot;data&quot;,&quot;word&quot;,&quot;i&quot;,&quot;.&quot;,&quot;e&quot;,&quot;.&quot;,&quot;reverts&quot;,&quot;the&quot;,&quot;bit&quot;,&quot;order&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;reflect\&quot;,\n  \&quot;a\&quot;,\n  \&quot;data\&quot;,\n  \&quot;word\&quot;,\n  \&quot;i\&quot;,\n  \&quot;.\&quot;,\n  \&quot;e\&quot;,\n  \&quot;.\&quot;,\n  \&quot;reverts\&quot;,\n  \&quot;the\&quot;,\n  \&quot;bit\&quot;,\n  \&quot;order\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;1eb2254b5235a053a953e596fa7602d0b110245d&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/scottrice/pysteam/blob/1eb2254b5235a053a953e596fa7602d0b110245d/pysteam/_crc_algorithms.py#L115-L123&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:49,&quot;string&quot;:&quot;49&quot;}}},{&quot;rowIdx&quot;:50,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;scottrice/pysteam&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;pysteam/_crc_algorithms.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Crc.bit_by_bit&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def bit_by_bit(self, in_data):\n        \&quot;\&quot;\&quot;\n        Classic simple and slow CRC implementation.  This function iterates bit\n        by bit over the augmented input message and returns the calculated CRC\n        value at the end.\n        \&quot;\&quot;\&quot;\n        # If the input data is a string, convert to bytes.\n        if isinstance(in_data, str):\n            in_data = [ord(c) for c in in_data]\n\n        register = self.NonDirectInit\n        for octet in in_data:\n            if self.ReflectIn:\n                octet = self.reflect(octet, 8)\n            for i in range(8):\n                topbit = register &amp; self.MSB_Mask\n                register = ((register << 1) &amp; self.Mask) | ((octet >> (7 - i)) &amp; 0x01)\n                if topbit:\n                    register ^= self.Poly\n\n        for i in range(self.Width):\n            topbit = register &amp; self.MSB_Mask\n            register = ((register << 1) &amp; self.Mask)\n            if topbit:\n                register ^= self.Poly\n\n        if self.ReflectOut:\n            register = self.reflect(register, self.Width)\n        return register ^ self.XorOut&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def bit_by_bit(self, in_data):\n        \&quot;\&quot;\&quot;\n        Classic simple and slow CRC implementation.  This function iterates bit\n        by bit over the augmented input message and returns the calculated CRC\n        value at the end.\n        \&quot;\&quot;\&quot;\n        # If the input data is a string, convert to bytes.\n        if isinstance(in_data, str):\n            in_data = [ord(c) for c in in_data]\n\n        register = self.NonDirectInit\n        for octet in in_data:\n            if self.ReflectIn:\n                octet = self.reflect(octet, 8)\n            for i in range(8):\n                topbit = register &amp; self.MSB_Mask\n                register = ((register << 1) &amp; self.Mask) | ((octet >> (7 - i)) &amp; 0x01)\n                if topbit:\n                    register ^= self.Poly\n\n        for i in range(self.Width):\n            topbit = register &amp; self.MSB_Mask\n            register = ((register << 1) &amp; self.Mask)\n            if topbit:\n                register ^= self.Poly\n\n        if self.ReflectOut:\n            register = self.reflect(register, self.Width)\n        return register ^ self.XorOut&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;bit_by_bit&quot;,&quot;(&quot;,&quot;self&quot;,&quot;,&quot;,&quot;in_data&quot;,&quot;)&quot;,&quot;:&quot;,&quot;# If the input data is a string, convert to bytes.&quot;,&quot;if&quot;,&quot;isinstance&quot;,&quot;(&quot;,&quot;in_data&quot;,&quot;,&quot;,&quot;str&quot;,&quot;)&quot;,&quot;:&quot;,&quot;in_data&quot;,&quot;=&quot;,&quot;[&quot;,&quot;ord&quot;,&quot;(&quot;,&quot;c&quot;,&quot;)&quot;,&quot;for&quot;,&quot;c&quot;,&quot;in&quot;,&quot;in_data&quot;,&quot;]&quot;,&quot;register&quot;,&quot;=&quot;,&quot;self&quot;,&quot;.&quot;,&quot;NonDirectInit&quot;,&quot;for&quot;,&quot;octet&quot;,&quot;in&quot;,&quot;in_data&quot;,&quot;:&quot;,&quot;if&quot;,&quot;self&quot;,&quot;.&quot;,&quot;ReflectIn&quot;,&quot;:&quot;,&quot;octet&quot;,&quot;=&quot;,&quot;self&quot;,&quot;.&quot;,&quot;reflect&quot;,&quot;(&quot;,&quot;octet&quot;,&quot;,&quot;,&quot;8&quot;,&quot;)&quot;,&quot;for&quot;,&quot;i&quot;,&quot;in&quot;,&quot;range&quot;,&quot;(&quot;,&quot;8&quot;,&quot;)&quot;,&quot;:&quot;,&quot;topbit&quot;,&quot;=&quot;,&quot;register&quot;,&quot;&amp;&quot;,&quot;self&quot;,&quot;.&quot;,&quot;MSB_Mask&quot;,&quot;register&quot;,&quot;=&quot;,&quot;(&quot;,&quot;(&quot;,&quot;register&quot;,&quot;<<&quot;,&quot;1&quot;,&quot;)&quot;,&quot;&amp;&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Mask&quot;,&quot;)&quot;,&quot;|&quot;,&quot;(&quot;,&quot;(&quot;,&quot;octet&quot;,&quot;>>&quot;,&quot;(&quot;,&quot;7&quot;,&quot;-&quot;,&quot;i&quot;,&quot;)&quot;,&quot;)&quot;,&quot;&amp;&quot;,&quot;0x01&quot;,&quot;)&quot;,&quot;if&quot;,&quot;topbit&quot;,&quot;:&quot;,&quot;register&quot;,&quot;^=&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Poly&quot;,&quot;for&quot;,&quot;i&quot;,&quot;in&quot;,&quot;range&quot;,&quot;(&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Width&quot;,&quot;)&quot;,&quot;:&quot;,&quot;topbit&quot;,&quot;=&quot;,&quot;register&quot;,&quot;&amp;&quot;,&quot;self&quot;,&quot;.&quot;,&quot;MSB_Mask&quot;,&quot;register&quot;,&quot;=&quot;,&quot;(&quot;,&quot;(&quot;,&quot;register&quot;,&quot;<<&quot;,&quot;1&quot;,&quot;)&quot;,&quot;&amp;&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Mask&quot;,&quot;)&quot;,&quot;if&quot;,&quot;topbit&quot;,&quot;:&quot;,&quot;register&quot;,&quot;^=&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Poly&quot;,&quot;if&quot;,&quot;self&quot;,&quot;.&quot;,&quot;ReflectOut&quot;,&quot;:&quot;,&quot;register&quot;,&quot;=&quot;,&quot;self&quot;,&quot;.&quot;,&quot;reflect&quot;,&quot;(&quot;,&quot;register&quot;,&quot;,&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Width&quot;,&quot;)&quot;,&quot;return&quot;,&quot;register&quot;,&quot;^&quot;,&quot;self&quot;,&quot;.&quot;,&quot;XorOut&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;bit_by_bit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;,\&quot;,\n  \&quot;in_data\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;# If the input data is a string, convert to bytes.\&quot;,\n  \&quot;if\&quot;,\n  \&quot;isinstance\&quot;,\n  \&quot;(\&quot;,\n  \&quot;in_data\&quot;,\n  \&quot;,\&quot;,\n  \&quot;str\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;in_data\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;ord\&quot;,\n  \&quot;(\&quot;,\n  \&quot;c\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;c\&quot;,\n  \&quot;in\&quot;,\n  \&quot;in_data\&quot;,\n  \&quot;]\&quot;,\n  \&quot;register\&quot;,\n  \&quot;=\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;NonDirectInit\&quot;,\n  \&quot;for\&quot;,\n  \&quot;octet\&quot;,\n  \&quot;in\&quot;,\n  \&quot;in_data\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ReflectIn\&quot;,\n  \&quot;:\&quot;,\n  \&quot;octet\&quot;,\n  \&quot;=\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;reflect\&quot;,\n  \&quot;(\&quot;,\n  \&quot;octet\&quot;,\n  \&quot;,\&quot;,\n  \&quot;8\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;in\&quot;,\n  \&quot;range\&quot;,\n  \&quot;(\&quot;,\n  \&quot;8\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;topbit\&quot;,\n  \&quot;=\&quot;,\n  \&quot;register\&quot;,\n  \&quot;&amp;\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;MSB_Mask\&quot;,\n  \&quot;register\&quot;,\n  \&quot;=\&quot;,\n  \&quot;(\&quot;,\n  \&quot;(\&quot;,\n  \&quot;register\&quot;,\n  \&quot;<<\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;&amp;\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Mask\&quot;,\n  \&quot;)\&quot;,\n  \&quot;|\&quot;,\n  \&quot;(\&quot;,\n  \&quot;(\&quot;,\n  \&quot;octet\&quot;,\n  \&quot;>>\&quot;,\n  \&quot;(\&quot;,\n  \&quot;7\&quot;,\n  \&quot;-\&quot;,\n  \&quot;i\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;&amp;\&quot;,\n  \&quot;0x01\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;topbit\&quot;,\n  \&quot;:\&quot;,\n  \&quot;register\&quot;,\n  \&quot;^=\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Poly\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;in\&quot;,\n  \&quot;range\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Width\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;topbit\&quot;,\n  \&quot;=\&quot;,\n  \&quot;register\&quot;,\n  \&quot;&amp;\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;MSB_Mask\&quot;,\n  \&quot;register\&quot;,\n  \&quot;=\&quot;,\n  \&quot;(\&quot;,\n  \&quot;(\&quot;,\n  \&quot;register\&quot;,\n  \&quot;<<\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;&amp;\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Mask\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;topbit\&quot;,\n  \&quot;:\&quot;,\n  \&quot;register\&quot;,\n  \&quot;^=\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Poly\&quot;,\n  \&quot;if\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ReflectOut\&quot;,\n  \&quot;:\&quot;,\n  \&quot;register\&quot;,\n  \&quot;=\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;reflect\&quot;,\n  \&quot;(\&quot;,\n  \&quot;register\&quot;,\n  \&quot;,\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Width\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;register\&quot;,\n  \&quot;^\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;XorOut\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Classic simple and slow CRC implementation.  This function iterates bit\n        by bit over the augmented input message and returns the calculated CRC\n        value at the end.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Classic&quot;,&quot;simple&quot;,&quot;and&quot;,&quot;slow&quot;,&quot;CRC&quot;,&quot;implementation&quot;,&quot;.&quot;,&quot;This&quot;,&quot;function&quot;,&quot;iterates&quot;,&quot;bit&quot;,&quot;by&quot;,&quot;bit&quot;,&quot;over&quot;,&quot;the&quot;,&quot;augmented&quot;,&quot;input&quot;,&quot;message&quot;,&quot;and&quot;,&quot;returns&quot;,&quot;the&quot;,&quot;calculated&quot;,&quot;CRC&quot;,&quot;value&quot;,&quot;at&quot;,&quot;the&quot;,&quot;end&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Classic\&quot;,\n  \&quot;simple\&quot;,\n  \&quot;and\&quot;,\n  \&quot;slow\&quot;,\n  \&quot;CRC\&quot;,\n  \&quot;implementation\&quot;,\n  \&quot;.\&quot;,\n  \&quot;This\&quot;,\n  \&quot;function\&quot;,\n  \&quot;iterates\&quot;,\n  \&quot;bit\&quot;,\n  \&quot;by\&quot;,\n  \&quot;bit\&quot;,\n  \&quot;over\&quot;,\n  \&quot;the\&quot;,\n  \&quot;augmented\&quot;,\n  \&quot;input\&quot;,\n  \&quot;message\&quot;,\n  \&quot;and\&quot;,\n  \&quot;returns\&quot;,\n  \&quot;the\&quot;,\n  \&quot;calculated\&quot;,\n  \&quot;CRC\&quot;,\n  \&quot;value\&quot;,\n  \&quot;at\&quot;,\n  \&quot;the\&quot;,\n  \&quot;end\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;1eb2254b5235a053a953e596fa7602d0b110245d&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/scottrice/pysteam/blob/1eb2254b5235a053a953e596fa7602d0b110245d/pysteam/_crc_algorithms.py#L128-L156&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:50,&quot;string&quot;:&quot;50&quot;}}},{&quot;rowIdx&quot;:51,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;scottrice/pysteam&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;pysteam/_crc_algorithms.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Crc.gen_table&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def gen_table(self):\n        \&quot;\&quot;\&quot;\n        This function generates the CRC table used for the table_driven CRC\n        algorithm.  The Python version cannot handle tables of an index width\n        other than 8.  See the generated C code for tables with different sizes\n        instead.\n        \&quot;\&quot;\&quot;\n        table_length = 1 << self.TableIdxWidth\n        tbl = [0] * table_length\n        for i in range(table_length):\n            register = i\n            if self.ReflectIn:\n                register = self.reflect(register, self.TableIdxWidth)\n            register = register << (self.Width - self.TableIdxWidth + self.CrcShift)\n            for j in range(self.TableIdxWidth):\n                if register &amp; (self.MSB_Mask << self.CrcShift) != 0:\n                    register = (register << 1) ^ (self.Poly << self.CrcShift)\n                else:\n                    register = (register << 1)\n            if self.ReflectIn:\n                register = self.reflect(register >> self.CrcShift, self.Width) << self.CrcShift\n            tbl[i] = register &amp; (self.Mask << self.CrcShift)\n        return tbl&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def gen_table(self):\n        \&quot;\&quot;\&quot;\n        This function generates the CRC table used for the table_driven CRC\n        algorithm.  The Python version cannot handle tables of an index width\n        other than 8.  See the generated C code for tables with different sizes\n        instead.\n        \&quot;\&quot;\&quot;\n        table_length = 1 << self.TableIdxWidth\n        tbl = [0] * table_length\n        for i in range(table_length):\n            register = i\n            if self.ReflectIn:\n                register = self.reflect(register, self.TableIdxWidth)\n            register = register << (self.Width - self.TableIdxWidth + self.CrcShift)\n            for j in range(self.TableIdxWidth):\n                if register &amp; (self.MSB_Mask << self.CrcShift) != 0:\n                    register = (register << 1) ^ (self.Poly << self.CrcShift)\n                else:\n                    register = (register << 1)\n            if self.ReflectIn:\n                register = self.reflect(register >> self.CrcShift, self.Width) << self.CrcShift\n            tbl[i] = register &amp; (self.Mask << self.CrcShift)\n        return tbl&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;gen_table&quot;,&quot;(&quot;,&quot;self&quot;,&quot;)&quot;,&quot;:&quot;,&quot;table_length&quot;,&quot;=&quot;,&quot;1&quot;,&quot;<<&quot;,&quot;self&quot;,&quot;.&quot;,&quot;TableIdxWidth&quot;,&quot;tbl&quot;,&quot;=&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;*&quot;,&quot;table_length&quot;,&quot;for&quot;,&quot;i&quot;,&quot;in&quot;,&quot;range&quot;,&quot;(&quot;,&quot;table_length&quot;,&quot;)&quot;,&quot;:&quot;,&quot;register&quot;,&quot;=&quot;,&quot;i&quot;,&quot;if&quot;,&quot;self&quot;,&quot;.&quot;,&quot;ReflectIn&quot;,&quot;:&quot;,&quot;register&quot;,&quot;=&quot;,&quot;self&quot;,&quot;.&quot;,&quot;reflect&quot;,&quot;(&quot;,&quot;register&quot;,&quot;,&quot;,&quot;self&quot;,&quot;.&quot;,&quot;TableIdxWidth&quot;,&quot;)&quot;,&quot;register&quot;,&quot;=&quot;,&quot;register&quot;,&quot;<<&quot;,&quot;(&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Width&quot;,&quot;-&quot;,&quot;self&quot;,&quot;.&quot;,&quot;TableIdxWidth&quot;,&quot;+&quot;,&quot;self&quot;,&quot;.&quot;,&quot;CrcShift&quot;,&quot;)&quot;,&quot;for&quot;,&quot;j&quot;,&quot;in&quot;,&quot;range&quot;,&quot;(&quot;,&quot;self&quot;,&quot;.&quot;,&quot;TableIdxWidth&quot;,&quot;)&quot;,&quot;:&quot;,&quot;if&quot;,&quot;register&quot;,&quot;&amp;&quot;,&quot;(&quot;,&quot;self&quot;,&quot;.&quot;,&quot;MSB_Mask&quot;,&quot;<<&quot;,&quot;self&quot;,&quot;.&quot;,&quot;CrcShift&quot;,&quot;)&quot;,&quot;!=&quot;,&quot;0&quot;,&quot;:&quot;,&quot;register&quot;,&quot;=&quot;,&quot;(&quot;,&quot;register&quot;,&quot;<<&quot;,&quot;1&quot;,&quot;)&quot;,&quot;^&quot;,&quot;(&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Poly&quot;,&quot;<<&quot;,&quot;self&quot;,&quot;.&quot;,&quot;CrcShift&quot;,&quot;)&quot;,&quot;else&quot;,&quot;:&quot;,&quot;register&quot;,&quot;=&quot;,&quot;(&quot;,&quot;register&quot;,&quot;<<&quot;,&quot;1&quot;,&quot;)&quot;,&quot;if&quot;,&quot;self&quot;,&quot;.&quot;,&quot;ReflectIn&quot;,&quot;:&quot;,&quot;register&quot;,&quot;=&quot;,&quot;self&quot;,&quot;.&quot;,&quot;reflect&quot;,&quot;(&quot;,&quot;register&quot;,&quot;>>&quot;,&quot;self&quot;,&quot;.&quot;,&quot;CrcShift&quot;,&quot;,&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Width&quot;,&quot;)&quot;,&quot;<<&quot;,&quot;self&quot;,&quot;.&quot;,&quot;CrcShift&quot;,&quot;tbl&quot;,&quot;[&quot;,&quot;i&quot;,&quot;]&quot;,&quot;=&quot;,&quot;register&quot;,&quot;&amp;&quot;,&quot;(&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Mask&quot;,&quot;<<&quot;,&quot;self&quot;,&quot;.&quot;,&quot;CrcShift&quot;,&quot;)&quot;,&quot;return&quot;,&quot;tbl&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;gen_table\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;table_length\&quot;,\n  \&quot;=\&quot;,\n  \&quot;1\&quot;,\n  \&quot;<<\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;TableIdxWidth\&quot;,\n  \&quot;tbl\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;*\&quot;,\n  \&quot;table_length\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;in\&quot;,\n  \&quot;range\&quot;,\n  \&quot;(\&quot;,\n  \&quot;table_length\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;register\&quot;,\n  \&quot;=\&quot;,\n  \&quot;i\&quot;,\n  \&quot;if\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ReflectIn\&quot;,\n  \&quot;:\&quot;,\n  \&quot;register\&quot;,\n  \&quot;=\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;reflect\&quot;,\n  \&quot;(\&quot;,\n  \&quot;register\&quot;,\n  \&quot;,\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;TableIdxWidth\&quot;,\n  \&quot;)\&quot;,\n  \&quot;register\&quot;,\n  \&quot;=\&quot;,\n  \&quot;register\&quot;,\n  \&quot;<<\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Width\&quot;,\n  \&quot;-\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;TableIdxWidth\&quot;,\n  \&quot;+\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;CrcShift\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;j\&quot;,\n  \&quot;in\&quot;,\n  \&quot;range\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;TableIdxWidth\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;register\&quot;,\n  \&quot;&amp;\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;MSB_Mask\&quot;,\n  \&quot;<<\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;CrcShift\&quot;,\n  \&quot;)\&quot;,\n  \&quot;!=\&quot;,\n  \&quot;0\&quot;,\n  \&quot;:\&quot;,\n  \&quot;register\&quot;,\n  \&quot;=\&quot;,\n  \&quot;(\&quot;,\n  \&quot;register\&quot;,\n  \&quot;<<\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;^\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Poly\&quot;,\n  \&quot;<<\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;CrcShift\&quot;,\n  \&quot;)\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;register\&quot;,\n  \&quot;=\&quot;,\n  \&quot;(\&quot;,\n  \&quot;register\&quot;,\n  \&quot;<<\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ReflectIn\&quot;,\n  \&quot;:\&quot;,\n  \&quot;register\&quot;,\n  \&quot;=\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;reflect\&quot;,\n  \&quot;(\&quot;,\n  \&quot;register\&quot;,\n  \&quot;>>\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;CrcShift\&quot;,\n  \&quot;,\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Width\&quot;,\n  \&quot;)\&quot;,\n  \&quot;<<\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;CrcShift\&quot;,\n  \&quot;tbl\&quot;,\n  \&quot;[\&quot;,\n  \&quot;i\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;register\&quot;,\n  \&quot;&amp;\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Mask\&quot;,\n  \&quot;<<\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;CrcShift\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;tbl\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;This function generates the CRC table used for the table_driven CRC\n        algorithm.  The Python version cannot handle tables of an index width\n        other than 8.  See the generated C code for tables with different sizes\n        instead.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;This&quot;,&quot;function&quot;,&quot;generates&quot;,&quot;the&quot;,&quot;CRC&quot;,&quot;table&quot;,&quot;used&quot;,&quot;for&quot;,&quot;the&quot;,&quot;table_driven&quot;,&quot;CRC&quot;,&quot;algorithm&quot;,&quot;.&quot;,&quot;The&quot;,&quot;Python&quot;,&quot;version&quot;,&quot;cannot&quot;,&quot;handle&quot;,&quot;tables&quot;,&quot;of&quot;,&quot;an&quot;,&quot;index&quot;,&quot;width&quot;,&quot;other&quot;,&quot;than&quot;,&quot;8&quot;,&quot;.&quot;,&quot;See&quot;,&quot;the&quot;,&quot;generated&quot;,&quot;C&quot;,&quot;code&quot;,&quot;for&quot;,&quot;tables&quot;,&quot;with&quot;,&quot;different&quot;,&quot;sizes&quot;,&quot;instead&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;This\&quot;,\n  \&quot;function\&quot;,\n  \&quot;generates\&quot;,\n  \&quot;the\&quot;,\n  \&quot;CRC\&quot;,\n  \&quot;table\&quot;,\n  \&quot;used\&quot;,\n  \&quot;for\&quot;,\n  \&quot;the\&quot;,\n  \&quot;table_driven\&quot;,\n  \&quot;CRC\&quot;,\n  \&quot;algorithm\&quot;,\n  \&quot;.\&quot;,\n  \&quot;The\&quot;,\n  \&quot;Python\&quot;,\n  \&quot;version\&quot;,\n  \&quot;cannot\&quot;,\n  \&quot;handle\&quot;,\n  \&quot;tables\&quot;,\n  \&quot;of\&quot;,\n  \&quot;an\&quot;,\n  \&quot;index\&quot;,\n  \&quot;width\&quot;,\n  \&quot;other\&quot;,\n  \&quot;than\&quot;,\n  \&quot;8\&quot;,\n  \&quot;.\&quot;,\n  \&quot;See\&quot;,\n  \&quot;the\&quot;,\n  \&quot;generated\&quot;,\n  \&quot;C\&quot;,\n  \&quot;code\&quot;,\n  \&quot;for\&quot;,\n  \&quot;tables\&quot;,\n  \&quot;with\&quot;,\n  \&quot;different\&quot;,\n  \&quot;sizes\&quot;,\n  \&quot;instead\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;1eb2254b5235a053a953e596fa7602d0b110245d&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/scottrice/pysteam/blob/1eb2254b5235a053a953e596fa7602d0b110245d/pysteam/_crc_algorithms.py#L190-L212&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:51,&quot;string&quot;:&quot;51&quot;}}},{&quot;rowIdx&quot;:52,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;scottrice/pysteam&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;pysteam/_crc_algorithms.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Crc.table_driven&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def table_driven(self, in_data):\n        \&quot;\&quot;\&quot;\n        The Standard table_driven CRC algorithm.\n        \&quot;\&quot;\&quot;\n        # If the input data is a string, convert to bytes.\n        if isinstance(in_data, str):\n            in_data = [ord(c) for c in in_data]\n\n        tbl = self.gen_table()\n\n        register = self.DirectInit << self.CrcShift\n        if not self.ReflectIn:\n            for octet in in_data:\n                tblidx = ((register >> (self.Width - self.TableIdxWidth + self.CrcShift)) ^ octet) &amp; 0xff\n                register = ((register << (self.TableIdxWidth - self.CrcShift)) ^ tbl[tblidx]) &amp; (self.Mask << self.CrcShift)\n            register = register >> self.CrcShift\n        else:\n            register = self.reflect(register, self.Width + self.CrcShift) << self.CrcShift\n            for octet in in_data:\n                tblidx = ((register >> self.CrcShift) ^ octet) &amp; 0xff\n                register = ((register >> self.TableIdxWidth) ^ tbl[tblidx]) &amp; (self.Mask << self.CrcShift)\n            register = self.reflect(register, self.Width + self.CrcShift) &amp; self.Mask\n\n        if self.ReflectOut:\n            register = self.reflect(register, self.Width)\n        return register ^ self.XorOut&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def table_driven(self, in_data):\n        \&quot;\&quot;\&quot;\n        The Standard table_driven CRC algorithm.\n        \&quot;\&quot;\&quot;\n        # If the input data is a string, convert to bytes.\n        if isinstance(in_data, str):\n            in_data = [ord(c) for c in in_data]\n\n        tbl = self.gen_table()\n\n        register = self.DirectInit << self.CrcShift\n        if not self.ReflectIn:\n            for octet in in_data:\n                tblidx = ((register >> (self.Width - self.TableIdxWidth + self.CrcShift)) ^ octet) &amp; 0xff\n                register = ((register << (self.TableIdxWidth - self.CrcShift)) ^ tbl[tblidx]) &amp; (self.Mask << self.CrcShift)\n            register = register >> self.CrcShift\n        else:\n            register = self.reflect(register, self.Width + self.CrcShift) << self.CrcShift\n            for octet in in_data:\n                tblidx = ((register >> self.CrcShift) ^ octet) &amp; 0xff\n                register = ((register >> self.TableIdxWidth) ^ tbl[tblidx]) &amp; (self.Mask << self.CrcShift)\n            register = self.reflect(register, self.Width + self.CrcShift) &amp; self.Mask\n\n        if self.ReflectOut:\n            register = self.reflect(register, self.Width)\n        return register ^ self.XorOut&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;table_driven&quot;,&quot;(&quot;,&quot;self&quot;,&quot;,&quot;,&quot;in_data&quot;,&quot;)&quot;,&quot;:&quot;,&quot;# If the input data is a string, convert to bytes.&quot;,&quot;if&quot;,&quot;isinstance&quot;,&quot;(&quot;,&quot;in_data&quot;,&quot;,&quot;,&quot;str&quot;,&quot;)&quot;,&quot;:&quot;,&quot;in_data&quot;,&quot;=&quot;,&quot;[&quot;,&quot;ord&quot;,&quot;(&quot;,&quot;c&quot;,&quot;)&quot;,&quot;for&quot;,&quot;c&quot;,&quot;in&quot;,&quot;in_data&quot;,&quot;]&quot;,&quot;tbl&quot;,&quot;=&quot;,&quot;self&quot;,&quot;.&quot;,&quot;gen_table&quot;,&quot;(&quot;,&quot;)&quot;,&quot;register&quot;,&quot;=&quot;,&quot;self&quot;,&quot;.&quot;,&quot;DirectInit&quot;,&quot;<<&quot;,&quot;self&quot;,&quot;.&quot;,&quot;CrcShift&quot;,&quot;if&quot;,&quot;not&quot;,&quot;self&quot;,&quot;.&quot;,&quot;ReflectIn&quot;,&quot;:&quot;,&quot;for&quot;,&quot;octet&quot;,&quot;in&quot;,&quot;in_data&quot;,&quot;:&quot;,&quot;tblidx&quot;,&quot;=&quot;,&quot;(&quot;,&quot;(&quot;,&quot;register&quot;,&quot;>>&quot;,&quot;(&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Width&quot;,&quot;-&quot;,&quot;self&quot;,&quot;.&quot;,&quot;TableIdxWidth&quot;,&quot;+&quot;,&quot;self&quot;,&quot;.&quot;,&quot;CrcShift&quot;,&quot;)&quot;,&quot;)&quot;,&quot;^&quot;,&quot;octet&quot;,&quot;)&quot;,&quot;&amp;&quot;,&quot;0xff&quot;,&quot;register&quot;,&quot;=&quot;,&quot;(&quot;,&quot;(&quot;,&quot;register&quot;,&quot;<<&quot;,&quot;(&quot;,&quot;self&quot;,&quot;.&quot;,&quot;TableIdxWidth&quot;,&quot;-&quot;,&quot;self&quot;,&quot;.&quot;,&quot;CrcShift&quot;,&quot;)&quot;,&quot;)&quot;,&quot;^&quot;,&quot;tbl&quot;,&quot;[&quot;,&quot;tblidx&quot;,&quot;]&quot;,&quot;)&quot;,&quot;&amp;&quot;,&quot;(&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Mask&quot;,&quot;<<&quot;,&quot;self&quot;,&quot;.&quot;,&quot;CrcShift&quot;,&quot;)&quot;,&quot;register&quot;,&quot;=&quot;,&quot;register&quot;,&quot;>>&quot;,&quot;self&quot;,&quot;.&quot;,&quot;CrcShift&quot;,&quot;else&quot;,&quot;:&quot;,&quot;register&quot;,&quot;=&quot;,&quot;self&quot;,&quot;.&quot;,&quot;reflect&quot;,&quot;(&quot;,&quot;register&quot;,&quot;,&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Width&quot;,&quot;+&quot;,&quot;self&quot;,&quot;.&quot;,&quot;CrcShift&quot;,&quot;)&quot;,&quot;<<&quot;,&quot;self&quot;,&quot;.&quot;,&quot;CrcShift&quot;,&quot;for&quot;,&quot;octet&quot;,&quot;in&quot;,&quot;in_data&quot;,&quot;:&quot;,&quot;tblidx&quot;,&quot;=&quot;,&quot;(&quot;,&quot;(&quot;,&quot;register&quot;,&quot;>>&quot;,&quot;self&quot;,&quot;.&quot;,&quot;CrcShift&quot;,&quot;)&quot;,&quot;^&quot;,&quot;octet&quot;,&quot;)&quot;,&quot;&amp;&quot;,&quot;0xff&quot;,&quot;register&quot;,&quot;=&quot;,&quot;(&quot;,&quot;(&quot;,&quot;register&quot;,&quot;>>&quot;,&quot;self&quot;,&quot;.&quot;,&quot;TableIdxWidth&quot;,&quot;)&quot;,&quot;^&quot;,&quot;tbl&quot;,&quot;[&quot;,&quot;tblidx&quot;,&quot;]&quot;,&quot;)&quot;,&quot;&amp;&quot;,&quot;(&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Mask&quot;,&quot;<<&quot;,&quot;self&quot;,&quot;.&quot;,&quot;CrcShift&quot;,&quot;)&quot;,&quot;register&quot;,&quot;=&quot;,&quot;self&quot;,&quot;.&quot;,&quot;reflect&quot;,&quot;(&quot;,&quot;register&quot;,&quot;,&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Width&quot;,&quot;+&quot;,&quot;self&quot;,&quot;.&quot;,&quot;CrcShift&quot;,&quot;)&quot;,&quot;&amp;&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Mask&quot;,&quot;if&quot;,&quot;self&quot;,&quot;.&quot;,&quot;ReflectOut&quot;,&quot;:&quot;,&quot;register&quot;,&quot;=&quot;,&quot;self&quot;,&quot;.&quot;,&quot;reflect&quot;,&quot;(&quot;,&quot;register&quot;,&quot;,&quot;,&quot;self&quot;,&quot;.&quot;,&quot;Width&quot;,&quot;)&quot;,&quot;return&quot;,&quot;register&quot;,&quot;^&quot;,&quot;self&quot;,&quot;.&quot;,&quot;XorOut&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;table_driven\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;,\&quot;,\n  \&quot;in_data\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;# If the input data is a string, convert to bytes.\&quot;,\n  \&quot;if\&quot;,\n  \&quot;isinstance\&quot;,\n  \&quot;(\&quot;,\n  \&quot;in_data\&quot;,\n  \&quot;,\&quot;,\n  \&quot;str\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;in_data\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;ord\&quot;,\n  \&quot;(\&quot;,\n  \&quot;c\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;c\&quot;,\n  \&quot;in\&quot;,\n  \&quot;in_data\&quot;,\n  \&quot;]\&quot;,\n  \&quot;tbl\&quot;,\n  \&quot;=\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;gen_table\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;register\&quot;,\n  \&quot;=\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;DirectInit\&quot;,\n  \&quot;<<\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;CrcShift\&quot;,\n  \&quot;if\&quot;,\n  \&quot;not\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ReflectIn\&quot;,\n  \&quot;:\&quot;,\n  \&quot;for\&quot;,\n  \&quot;octet\&quot;,\n  \&quot;in\&quot;,\n  \&quot;in_data\&quot;,\n  \&quot;:\&quot;,\n  \&quot;tblidx\&quot;,\n  \&quot;=\&quot;,\n  \&quot;(\&quot;,\n  \&quot;(\&quot;,\n  \&quot;register\&quot;,\n  \&quot;>>\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Width\&quot;,\n  \&quot;-\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;TableIdxWidth\&quot;,\n  \&quot;+\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;CrcShift\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;^\&quot;,\n  \&quot;octet\&quot;,\n  \&quot;)\&quot;,\n  \&quot;&amp;\&quot;,\n  \&quot;0xff\&quot;,\n  \&quot;register\&quot;,\n  \&quot;=\&quot;,\n  \&quot;(\&quot;,\n  \&quot;(\&quot;,\n  \&quot;register\&quot;,\n  \&quot;<<\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;TableIdxWidth\&quot;,\n  \&quot;-\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;CrcShift\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;^\&quot;,\n  \&quot;tbl\&quot;,\n  \&quot;[\&quot;,\n  \&quot;tblidx\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;&amp;\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Mask\&quot;,\n  \&quot;<<\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;CrcShift\&quot;,\n  \&quot;)\&quot;,\n  \&quot;register\&quot;,\n  \&quot;=\&quot;,\n  \&quot;register\&quot;,\n  \&quot;>>\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;CrcShift\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;register\&quot;,\n  \&quot;=\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;reflect\&quot;,\n  \&quot;(\&quot;,\n  \&quot;register\&quot;,\n  \&quot;,\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Width\&quot;,\n  \&quot;+\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;CrcShift\&quot;,\n  \&quot;)\&quot;,\n  \&quot;<<\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;CrcShift\&quot;,\n  \&quot;for\&quot;,\n  \&quot;octet\&quot;,\n  \&quot;in\&quot;,\n  \&quot;in_data\&quot;,\n  \&quot;:\&quot;,\n  \&quot;tblidx\&quot;,\n  \&quot;=\&quot;,\n  \&quot;(\&quot;,\n  \&quot;(\&quot;,\n  \&quot;register\&quot;,\n  \&quot;>>\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;CrcShift\&quot;,\n  \&quot;)\&quot;,\n  \&quot;^\&quot;,\n  \&quot;octet\&quot;,\n  \&quot;)\&quot;,\n  \&quot;&amp;\&quot;,\n  \&quot;0xff\&quot;,\n  \&quot;register\&quot;,\n  \&quot;=\&quot;,\n  \&quot;(\&quot;,\n  \&quot;(\&quot;,\n  \&quot;register\&quot;,\n  \&quot;>>\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;TableIdxWidth\&quot;,\n  \&quot;)\&quot;,\n  \&quot;^\&quot;,\n  \&quot;tbl\&quot;,\n  \&quot;[\&quot;,\n  \&quot;tblidx\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;&amp;\&quot;,\n  \&quot;(\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Mask\&quot;,\n  \&quot;<<\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;CrcShift\&quot;,\n  \&quot;)\&quot;,\n  \&quot;register\&quot;,\n  \&quot;=\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;reflect\&quot;,\n  \&quot;(\&quot;,\n  \&quot;register\&quot;,\n  \&quot;,\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Width\&quot;,\n  \&quot;+\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;CrcShift\&quot;,\n  \&quot;)\&quot;,\n  \&quot;&amp;\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Mask\&quot;,\n  \&quot;if\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ReflectOut\&quot;,\n  \&quot;:\&quot;,\n  \&quot;register\&quot;,\n  \&quot;=\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;reflect\&quot;,\n  \&quot;(\&quot;,\n  \&quot;register\&quot;,\n  \&quot;,\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Width\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;register\&quot;,\n  \&quot;^\&quot;,\n  \&quot;self\&quot;,\n  \&quot;.\&quot;,\n  \&quot;XorOut\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;The Standard table_driven CRC algorithm.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;The&quot;,&quot;Standard&quot;,&quot;table_driven&quot;,&quot;CRC&quot;,&quot;algorithm&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;The\&quot;,\n  \&quot;Standard\&quot;,\n  \&quot;table_driven\&quot;,\n  \&quot;CRC\&quot;,\n  \&quot;algorithm\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;1eb2254b5235a053a953e596fa7602d0b110245d&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/scottrice/pysteam/blob/1eb2254b5235a053a953e596fa7602d0b110245d/pysteam/_crc_algorithms.py#L217-L242&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:52,&quot;string&quot;:&quot;52&quot;}}},{&quot;rowIdx&quot;:53,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;christophertbrown/bioscripts&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ctbBio/strip_masked.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;parse_masked&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def parse_masked(seq, min_len):\n    \&quot;\&quot;\&quot;\n    parse masked sequence into non-masked and masked regions\n    \&quot;\&quot;\&quot;\n    nm, masked = [], [[]]\n    prev = None\n    for base in seq[1]:\n        if base.isupper():\n            nm.append(base)\n            if masked != [[]] and len(masked[-1]) < min_len:\n                nm.extend(masked[-1])\n                del masked[-1]\n            prev = False\n        elif base.islower():\n            if prev is False:\n                masked.append([])\n            masked[-1].append(base)\n            prev = True\n    return nm, masked&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def parse_masked(seq, min_len):\n    \&quot;\&quot;\&quot;\n    parse masked sequence into non-masked and masked regions\n    \&quot;\&quot;\&quot;\n    nm, masked = [], [[]]\n    prev = None\n    for base in seq[1]:\n        if base.isupper():\n            nm.append(base)\n            if masked != [[]] and len(masked[-1]) < min_len:\n                nm.extend(masked[-1])\n                del masked[-1]\n            prev = False\n        elif base.islower():\n            if prev is False:\n                masked.append([])\n            masked[-1].append(base)\n            prev = True\n    return nm, masked&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;parse_masked&quot;,&quot;(&quot;,&quot;seq&quot;,&quot;,&quot;,&quot;min_len&quot;,&quot;)&quot;,&quot;:&quot;,&quot;nm&quot;,&quot;,&quot;,&quot;masked&quot;,&quot;=&quot;,&quot;[&quot;,&quot;]&quot;,&quot;,&quot;,&quot;[&quot;,&quot;[&quot;,&quot;]&quot;,&quot;]&quot;,&quot;prev&quot;,&quot;=&quot;,&quot;None&quot;,&quot;for&quot;,&quot;base&quot;,&quot;in&quot;,&quot;seq&quot;,&quot;[&quot;,&quot;1&quot;,&quot;]&quot;,&quot;:&quot;,&quot;if&quot;,&quot;base&quot;,&quot;.&quot;,&quot;isupper&quot;,&quot;(&quot;,&quot;)&quot;,&quot;:&quot;,&quot;nm&quot;,&quot;.&quot;,&quot;append&quot;,&quot;(&quot;,&quot;base&quot;,&quot;)&quot;,&quot;if&quot;,&quot;masked&quot;,&quot;!=&quot;,&quot;[&quot;,&quot;[&quot;,&quot;]&quot;,&quot;]&quot;,&quot;and&quot;,&quot;len&quot;,&quot;(&quot;,&quot;masked&quot;,&quot;[&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;,&quot;)&quot;,&quot;<&quot;,&quot;min_len&quot;,&quot;:&quot;,&quot;nm&quot;,&quot;.&quot;,&quot;extend&quot;,&quot;(&quot;,&quot;masked&quot;,&quot;[&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;,&quot;)&quot;,&quot;del&quot;,&quot;masked&quot;,&quot;[&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;,&quot;prev&quot;,&quot;=&quot;,&quot;False&quot;,&quot;elif&quot;,&quot;base&quot;,&quot;.&quot;,&quot;islower&quot;,&quot;(&quot;,&quot;)&quot;,&quot;:&quot;,&quot;if&quot;,&quot;prev&quot;,&quot;is&quot;,&quot;False&quot;,&quot;:&quot;,&quot;masked&quot;,&quot;.&quot;,&quot;append&quot;,&quot;(&quot;,&quot;[&quot;,&quot;]&quot;,&quot;)&quot;,&quot;masked&quot;,&quot;[&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;,&quot;.&quot;,&quot;append&quot;,&quot;(&quot;,&quot;base&quot;,&quot;)&quot;,&quot;prev&quot;,&quot;=&quot;,&quot;True&quot;,&quot;return&quot;,&quot;nm&quot;,&quot;,&quot;,&quot;masked&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;parse_masked\&quot;,\n  \&quot;(\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;,\&quot;,\n  \&quot;min_len\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;nm\&quot;,\n  \&quot;,\&quot;,\n  \&quot;masked\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;[\&quot;,\n  \&quot;[\&quot;,\n  \&quot;]\&quot;,\n  \&quot;]\&quot;,\n  \&quot;prev\&quot;,\n  \&quot;=\&quot;,\n  \&quot;None\&quot;,\n  \&quot;for\&quot;,\n  \&quot;base\&quot;,\n  \&quot;in\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;base\&quot;,\n  \&quot;.\&quot;,\n  \&quot;isupper\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;nm\&quot;,\n  \&quot;.\&quot;,\n  \&quot;append\&quot;,\n  \&quot;(\&quot;,\n  \&quot;base\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;masked\&quot;,\n  \&quot;!=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;[\&quot;,\n  \&quot;]\&quot;,\n  \&quot;]\&quot;,\n  \&quot;and\&quot;,\n  \&quot;len\&quot;,\n  \&quot;(\&quot;,\n  \&quot;masked\&quot;,\n  \&quot;[\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;<\&quot;,\n  \&quot;min_len\&quot;,\n  \&quot;:\&quot;,\n  \&quot;nm\&quot;,\n  \&quot;.\&quot;,\n  \&quot;extend\&quot;,\n  \&quot;(\&quot;,\n  \&quot;masked\&quot;,\n  \&quot;[\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;del\&quot;,\n  \&quot;masked\&quot;,\n  \&quot;[\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;prev\&quot;,\n  \&quot;=\&quot;,\n  \&quot;False\&quot;,\n  \&quot;elif\&quot;,\n  \&quot;base\&quot;,\n  \&quot;.\&quot;,\n  \&quot;islower\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;if\&quot;,\n  \&quot;prev\&quot;,\n  \&quot;is\&quot;,\n  \&quot;False\&quot;,\n  \&quot;:\&quot;,\n  \&quot;masked\&quot;,\n  \&quot;.\&quot;,\n  \&quot;append\&quot;,\n  \&quot;(\&quot;,\n  \&quot;[\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;masked\&quot;,\n  \&quot;[\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;append\&quot;,\n  \&quot;(\&quot;,\n  \&quot;base\&quot;,\n  \&quot;)\&quot;,\n  \&quot;prev\&quot;,\n  \&quot;=\&quot;,\n  \&quot;True\&quot;,\n  \&quot;return\&quot;,\n  \&quot;nm\&quot;,\n  \&quot;,\&quot;,\n  \&quot;masked\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;parse masked sequence into non-masked and masked regions&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;parse&quot;,&quot;masked&quot;,&quot;sequence&quot;,&quot;into&quot;,&quot;non&quot;,&quot;-&quot;,&quot;masked&quot;,&quot;and&quot;,&quot;masked&quot;,&quot;regions&quot;],&quot;string&quot;:&quot;[\n  \&quot;parse\&quot;,\n  \&quot;masked\&quot;,\n  \&quot;sequence\&quot;,\n  \&quot;into\&quot;,\n  \&quot;non\&quot;,\n  \&quot;-\&quot;,\n  \&quot;masked\&quot;,\n  \&quot;and\&quot;,\n  \&quot;masked\&quot;,\n  \&quot;regions\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;83b2566b3a5745437ec651cd6cafddd056846240&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/strip_masked.py#L13-L31&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:53,&quot;string&quot;:&quot;53&quot;}}},{&quot;rowIdx&quot;:54,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;christophertbrown/bioscripts&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ctbBio/strip_masked.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;strip_masked&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def strip_masked(fasta, min_len, print_masked):\n    \&quot;\&quot;\&quot;\n    remove masked regions from fasta file as long as\n    they are longer than min_len\n    \&quot;\&quot;\&quot;\n    for seq in parse_fasta(fasta):\n        nm, masked = parse_masked(seq, min_len)\n        nm = ['%s removed_masked >=%s' % (seq[0], min_len), ''.join(nm)]\n        yield [0, nm]\n        if print_masked is True:\n            for i, m in enumerate([i for i in masked if i != []], 1):\n                m = ['%s insertion:%s' % (seq[0], i), ''.join(m)]\n                yield [1, m]&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def strip_masked(fasta, min_len, print_masked):\n    \&quot;\&quot;\&quot;\n    remove masked regions from fasta file as long as\n    they are longer than min_len\n    \&quot;\&quot;\&quot;\n    for seq in parse_fasta(fasta):\n        nm, masked = parse_masked(seq, min_len)\n        nm = ['%s removed_masked >=%s' % (seq[0], min_len), ''.join(nm)]\n        yield [0, nm]\n        if print_masked is True:\n            for i, m in enumerate([i for i in masked if i != []], 1):\n                m = ['%s insertion:%s' % (seq[0], i), ''.join(m)]\n                yield [1, m]&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;strip_masked&quot;,&quot;(&quot;,&quot;fasta&quot;,&quot;,&quot;,&quot;min_len&quot;,&quot;,&quot;,&quot;print_masked&quot;,&quot;)&quot;,&quot;:&quot;,&quot;for&quot;,&quot;seq&quot;,&quot;in&quot;,&quot;parse_fasta&quot;,&quot;(&quot;,&quot;fasta&quot;,&quot;)&quot;,&quot;:&quot;,&quot;nm&quot;,&quot;,&quot;,&quot;masked&quot;,&quot;=&quot;,&quot;parse_masked&quot;,&quot;(&quot;,&quot;seq&quot;,&quot;,&quot;,&quot;min_len&quot;,&quot;)&quot;,&quot;nm&quot;,&quot;=&quot;,&quot;[&quot;,&quot;'%s removed_masked >=%s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;seq&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;,&quot;,&quot;min_len&quot;,&quot;)&quot;,&quot;,&quot;,&quot;''&quot;,&quot;.&quot;,&quot;join&quot;,&quot;(&quot;,&quot;nm&quot;,&quot;)&quot;,&quot;]&quot;,&quot;yield&quot;,&quot;[&quot;,&quot;0&quot;,&quot;,&quot;,&quot;nm&quot;,&quot;]&quot;,&quot;if&quot;,&quot;print_masked&quot;,&quot;is&quot;,&quot;True&quot;,&quot;:&quot;,&quot;for&quot;,&quot;i&quot;,&quot;,&quot;,&quot;m&quot;,&quot;in&quot;,&quot;enumerate&quot;,&quot;(&quot;,&quot;[&quot;,&quot;i&quot;,&quot;for&quot;,&quot;i&quot;,&quot;in&quot;,&quot;masked&quot;,&quot;if&quot;,&quot;i&quot;,&quot;!=&quot;,&quot;[&quot;,&quot;]&quot;,&quot;]&quot;,&quot;,&quot;,&quot;1&quot;,&quot;)&quot;,&quot;:&quot;,&quot;m&quot;,&quot;=&quot;,&quot;[&quot;,&quot;'%s insertion:%s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;seq&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;,&quot;,&quot;i&quot;,&quot;)&quot;,&quot;,&quot;,&quot;''&quot;,&quot;.&quot;,&quot;join&quot;,&quot;(&quot;,&quot;m&quot;,&quot;)&quot;,&quot;]&quot;,&quot;yield&quot;,&quot;[&quot;,&quot;1&quot;,&quot;,&quot;,&quot;m&quot;,&quot;]&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;strip_masked\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;,\&quot;,\n  \&quot;min_len\&quot;,\n  \&quot;,\&quot;,\n  \&quot;print_masked\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;for\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;in\&quot;,\n  \&quot;parse_fasta\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;nm\&quot;,\n  \&quot;,\&quot;,\n  \&quot;masked\&quot;,\n  \&quot;=\&quot;,\n  \&quot;parse_masked\&quot;,\n  \&quot;(\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;,\&quot;,\n  \&quot;min_len\&quot;,\n  \&quot;)\&quot;,\n  \&quot;nm\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'%s removed_masked >=%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;min_len\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;''\&quot;,\n  \&quot;.\&quot;,\n  \&quot;join\&quot;,\n  \&quot;(\&quot;,\n  \&quot;nm\&quot;,\n  \&quot;)\&quot;,\n  \&quot;]\&quot;,\n  \&quot;yield\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;,\&quot;,\n  \&quot;nm\&quot;,\n  \&quot;]\&quot;,\n  \&quot;if\&quot;,\n  \&quot;print_masked\&quot;,\n  \&quot;is\&quot;,\n  \&quot;True\&quot;,\n  \&quot;:\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;,\&quot;,\n  \&quot;m\&quot;,\n  \&quot;in\&quot;,\n  \&quot;enumerate\&quot;,\n  \&quot;(\&quot;,\n  \&quot;[\&quot;,\n  \&quot;i\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;in\&quot;,\n  \&quot;masked\&quot;,\n  \&quot;if\&quot;,\n  \&quot;i\&quot;,\n  \&quot;!=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;]\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;m\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'%s insertion:%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;seq\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;i\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;''\&quot;,\n  \&quot;.\&quot;,\n  \&quot;join\&quot;,\n  \&quot;(\&quot;,\n  \&quot;m\&quot;,\n  \&quot;)\&quot;,\n  \&quot;]\&quot;,\n  \&quot;yield\&quot;,\n  \&quot;[\&quot;,\n  \&quot;1\&quot;,\n  \&quot;,\&quot;,\n  \&quot;m\&quot;,\n  \&quot;]\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;remove masked regions from fasta file as long as\n    they are longer than min_len&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;remove&quot;,&quot;masked&quot;,&quot;regions&quot;,&quot;from&quot;,&quot;fasta&quot;,&quot;file&quot;,&quot;as&quot;,&quot;long&quot;,&quot;as&quot;,&quot;they&quot;,&quot;are&quot;,&quot;longer&quot;,&quot;than&quot;,&quot;min_len&quot;],&quot;string&quot;:&quot;[\n  \&quot;remove\&quot;,\n  \&quot;masked\&quot;,\n  \&quot;regions\&quot;,\n  \&quot;from\&quot;,\n  \&quot;fasta\&quot;,\n  \&quot;file\&quot;,\n  \&quot;as\&quot;,\n  \&quot;long\&quot;,\n  \&quot;as\&quot;,\n  \&quot;they\&quot;,\n  \&quot;are\&quot;,\n  \&quot;longer\&quot;,\n  \&quot;than\&quot;,\n  \&quot;min_len\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;83b2566b3a5745437ec651cd6cafddd056846240&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/strip_masked.py#L33-L45&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:54,&quot;string&quot;:&quot;54&quot;}}},{&quot;rowIdx&quot;:55,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;bin/network_plots_gephi.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;get_relative_abundance&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def get_relative_abundance(biomfile):\n    \&quot;\&quot;\&quot;\n    Return arcsine transformed relative abundance from a BIOM format file.\n\n    :type biomfile: BIOM format file\n    :param biomfile: BIOM format file used to obtain relative abundances for each OTU in\n                     a SampleID, which are used as node sizes in network plots.\n\n    :type return: Dictionary of dictionaries.\n    :return: Dictionary keyed on SampleID whose value is a dictionarykeyed on OTU Name\n             whose value is the arc sine tranfsormed relative abundance value for that\n             SampleID-OTU Name pair.\n    \&quot;\&quot;\&quot;\n    biomf = biom.load_table(biomfile)\n    norm_biomf = biomf.norm(inplace=False)\n    rel_abd = {}\n    for sid in norm_biomf.ids():\n        rel_abd[sid] = {}\n        for otuid in norm_biomf.ids(\&quot;observation\&quot;):\n            otuname = oc.otu_name(norm_biomf.metadata(otuid, axis=\&quot;observation\&quot;)[\&quot;taxonomy\&quot;])\n            otuname = \&quot; \&quot;.join(otuname.split(\&quot;_\&quot;))\n            abd = norm_biomf.get_value_by_ids(otuid, sid)\n            rel_abd[sid][otuname] = abd\n    ast_rel_abd = bc.arcsine_sqrt_transform(rel_abd)\n    return ast_rel_abd&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def get_relative_abundance(biomfile):\n    \&quot;\&quot;\&quot;\n    Return arcsine transformed relative abundance from a BIOM format file.\n\n    :type biomfile: BIOM format file\n    :param biomfile: BIOM format file used to obtain relative abundances for each OTU in\n                     a SampleID, which are used as node sizes in network plots.\n\n    :type return: Dictionary of dictionaries.\n    :return: Dictionary keyed on SampleID whose value is a dictionarykeyed on OTU Name\n             whose value is the arc sine tranfsormed relative abundance value for that\n             SampleID-OTU Name pair.\n    \&quot;\&quot;\&quot;\n    biomf = biom.load_table(biomfile)\n    norm_biomf = biomf.norm(inplace=False)\n    rel_abd = {}\n    for sid in norm_biomf.ids():\n        rel_abd[sid] = {}\n        for otuid in norm_biomf.ids(\&quot;observation\&quot;):\n            otuname = oc.otu_name(norm_biomf.metadata(otuid, axis=\&quot;observation\&quot;)[\&quot;taxonomy\&quot;])\n            otuname = \&quot; \&quot;.join(otuname.split(\&quot;_\&quot;))\n            abd = norm_biomf.get_value_by_ids(otuid, sid)\n            rel_abd[sid][otuname] = abd\n    ast_rel_abd = bc.arcsine_sqrt_transform(rel_abd)\n    return ast_rel_abd&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;get_relative_abundance&quot;,&quot;(&quot;,&quot;biomfile&quot;,&quot;)&quot;,&quot;:&quot;,&quot;biomf&quot;,&quot;=&quot;,&quot;biom&quot;,&quot;.&quot;,&quot;load_table&quot;,&quot;(&quot;,&quot;biomfile&quot;,&quot;)&quot;,&quot;norm_biomf&quot;,&quot;=&quot;,&quot;biomf&quot;,&quot;.&quot;,&quot;norm&quot;,&quot;(&quot;,&quot;inplace&quot;,&quot;=&quot;,&quot;False&quot;,&quot;)&quot;,&quot;rel_abd&quot;,&quot;=&quot;,&quot;{&quot;,&quot;}&quot;,&quot;for&quot;,&quot;sid&quot;,&quot;in&quot;,&quot;norm_biomf&quot;,&quot;.&quot;,&quot;ids&quot;,&quot;(&quot;,&quot;)&quot;,&quot;:&quot;,&quot;rel_abd&quot;,&quot;[&quot;,&quot;sid&quot;,&quot;]&quot;,&quot;=&quot;,&quot;{&quot;,&quot;}&quot;,&quot;for&quot;,&quot;otuid&quot;,&quot;in&quot;,&quot;norm_biomf&quot;,&quot;.&quot;,&quot;ids&quot;,&quot;(&quot;,&quot;\&quot;observation\&quot;&quot;,&quot;)&quot;,&quot;:&quot;,&quot;otuname&quot;,&quot;=&quot;,&quot;oc&quot;,&quot;.&quot;,&quot;otu_name&quot;,&quot;(&quot;,&quot;norm_biomf&quot;,&quot;.&quot;,&quot;metadata&quot;,&quot;(&quot;,&quot;otuid&quot;,&quot;,&quot;,&quot;axis&quot;,&quot;=&quot;,&quot;\&quot;observation\&quot;&quot;,&quot;)&quot;,&quot;[&quot;,&quot;\&quot;taxonomy\&quot;&quot;,&quot;]&quot;,&quot;)&quot;,&quot;otuname&quot;,&quot;=&quot;,&quot;\&quot; \&quot;&quot;,&quot;.&quot;,&quot;join&quot;,&quot;(&quot;,&quot;otuname&quot;,&quot;.&quot;,&quot;split&quot;,&quot;(&quot;,&quot;\&quot;_\&quot;&quot;,&quot;)&quot;,&quot;)&quot;,&quot;abd&quot;,&quot;=&quot;,&quot;norm_biomf&quot;,&quot;.&quot;,&quot;get_value_by_ids&quot;,&quot;(&quot;,&quot;otuid&quot;,&quot;,&quot;,&quot;sid&quot;,&quot;)&quot;,&quot;rel_abd&quot;,&quot;[&quot;,&quot;sid&quot;,&quot;]&quot;,&quot;[&quot;,&quot;otuname&quot;,&quot;]&quot;,&quot;=&quot;,&quot;abd&quot;,&quot;ast_rel_abd&quot;,&quot;=&quot;,&quot;bc&quot;,&quot;.&quot;,&quot;arcsine_sqrt_transform&quot;,&quot;(&quot;,&quot;rel_abd&quot;,&quot;)&quot;,&quot;return&quot;,&quot;ast_rel_abd&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;get_relative_abundance\&quot;,\n  \&quot;(\&quot;,\n  \&quot;biomfile\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;=\&quot;,\n  \&quot;biom\&quot;,\n  \&quot;.\&quot;,\n  \&quot;load_table\&quot;,\n  \&quot;(\&quot;,\n  \&quot;biomfile\&quot;,\n  \&quot;)\&quot;,\n  \&quot;norm_biomf\&quot;,\n  \&quot;=\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;.\&quot;,\n  \&quot;norm\&quot;,\n  \&quot;(\&quot;,\n  \&quot;inplace\&quot;,\n  \&quot;=\&quot;,\n  \&quot;False\&quot;,\n  \&quot;)\&quot;,\n  \&quot;rel_abd\&quot;,\n  \&quot;=\&quot;,\n  \&quot;{\&quot;,\n  \&quot;}\&quot;,\n  \&quot;for\&quot;,\n  \&quot;sid\&quot;,\n  \&quot;in\&quot;,\n  \&quot;norm_biomf\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ids\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;rel_abd\&quot;,\n  \&quot;[\&quot;,\n  \&quot;sid\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;{\&quot;,\n  \&quot;}\&quot;,\n  \&quot;for\&quot;,\n  \&quot;otuid\&quot;,\n  \&quot;in\&quot;,\n  \&quot;norm_biomf\&quot;,\n  \&quot;.\&quot;,\n  \&quot;ids\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;observation\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;otuname\&quot;,\n  \&quot;=\&quot;,\n  \&quot;oc\&quot;,\n  \&quot;.\&quot;,\n  \&quot;otu_name\&quot;,\n  \&quot;(\&quot;,\n  \&quot;norm_biomf\&quot;,\n  \&quot;.\&quot;,\n  \&quot;metadata\&quot;,\n  \&quot;(\&quot;,\n  \&quot;otuid\&quot;,\n  \&quot;,\&quot;,\n  \&quot;axis\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;observation\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;\\\&quot;taxonomy\\\&quot;\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;otuname\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot; \\\&quot;\&quot;,\n  \&quot;.\&quot;,\n  \&quot;join\&quot;,\n  \&quot;(\&quot;,\n  \&quot;otuname\&quot;,\n  \&quot;.\&quot;,\n  \&quot;split\&quot;,\n  \&quot;(\&quot;,\n  \&quot;\\\&quot;_\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;abd\&quot;,\n  \&quot;=\&quot;,\n  \&quot;norm_biomf\&quot;,\n  \&quot;.\&quot;,\n  \&quot;get_value_by_ids\&quot;,\n  \&quot;(\&quot;,\n  \&quot;otuid\&quot;,\n  \&quot;,\&quot;,\n  \&quot;sid\&quot;,\n  \&quot;)\&quot;,\n  \&quot;rel_abd\&quot;,\n  \&quot;[\&quot;,\n  \&quot;sid\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;otuname\&quot;,\n  \&quot;]\&quot;,\n  \&quot;=\&quot;,\n  \&quot;abd\&quot;,\n  \&quot;ast_rel_abd\&quot;,\n  \&quot;=\&quot;,\n  \&quot;bc\&quot;,\n  \&quot;.\&quot;,\n  \&quot;arcsine_sqrt_transform\&quot;,\n  \&quot;(\&quot;,\n  \&quot;rel_abd\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;ast_rel_abd\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Return arcsine transformed relative abundance from a BIOM format file.\n\n    :type biomfile: BIOM format file\n    :param biomfile: BIOM format file used to obtain relative abundances for each OTU in\n                     a SampleID, which are used as node sizes in network plots.\n\n    :type return: Dictionary of dictionaries.\n    :return: Dictionary keyed on SampleID whose value is a dictionarykeyed on OTU Name\n             whose value is the arc sine tranfsormed relative abundance value for that\n             SampleID-OTU Name pair.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Return&quot;,&quot;arcsine&quot;,&quot;transformed&quot;,&quot;relative&quot;,&quot;abundance&quot;,&quot;from&quot;,&quot;a&quot;,&quot;BIOM&quot;,&quot;format&quot;,&quot;file&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Return\&quot;,\n  \&quot;arcsine\&quot;,\n  \&quot;transformed\&quot;,\n  \&quot;relative\&quot;,\n  \&quot;abundance\&quot;,\n  \&quot;from\&quot;,\n  \&quot;a\&quot;,\n  \&quot;BIOM\&quot;,\n  \&quot;format\&quot;,\n  \&quot;file\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/bin/network_plots_gephi.py#L33-L57&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:55,&quot;string&quot;:&quot;55&quot;}}},{&quot;rowIdx&quot;:56,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;bin/iTol.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;find_otu&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def find_otu(otuid, tree):\n    \&quot;\&quot;\&quot;\n    Find an OTU ID in a Newick-format tree.\n    Return the starting position of the ID or None if not found.\n    \&quot;\&quot;\&quot;\n    for m in re.finditer(otuid, tree):\n        before, after = tree[m.start()-1], tree[m.start()+len(otuid)]\n        if before in [\&quot;(\&quot;, \&quot;,\&quot;, \&quot;)\&quot;] and after in [\&quot;:\&quot;, \&quot;;\&quot;]:\n            return m.start()\n    return None&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def find_otu(otuid, tree):\n    \&quot;\&quot;\&quot;\n    Find an OTU ID in a Newick-format tree.\n    Return the starting position of the ID or None if not found.\n    \&quot;\&quot;\&quot;\n    for m in re.finditer(otuid, tree):\n        before, after = tree[m.start()-1], tree[m.start()+len(otuid)]\n        if before in [\&quot;(\&quot;, \&quot;,\&quot;, \&quot;)\&quot;] and after in [\&quot;:\&quot;, \&quot;;\&quot;]:\n            return m.start()\n    return None&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;find_otu&quot;,&quot;(&quot;,&quot;otuid&quot;,&quot;,&quot;,&quot;tree&quot;,&quot;)&quot;,&quot;:&quot;,&quot;for&quot;,&quot;m&quot;,&quot;in&quot;,&quot;re&quot;,&quot;.&quot;,&quot;finditer&quot;,&quot;(&quot;,&quot;otuid&quot;,&quot;,&quot;,&quot;tree&quot;,&quot;)&quot;,&quot;:&quot;,&quot;before&quot;,&quot;,&quot;,&quot;after&quot;,&quot;=&quot;,&quot;tree&quot;,&quot;[&quot;,&quot;m&quot;,&quot;.&quot;,&quot;start&quot;,&quot;(&quot;,&quot;)&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;,&quot;,&quot;,&quot;tree&quot;,&quot;[&quot;,&quot;m&quot;,&quot;.&quot;,&quot;start&quot;,&quot;(&quot;,&quot;)&quot;,&quot;+&quot;,&quot;len&quot;,&quot;(&quot;,&quot;otuid&quot;,&quot;)&quot;,&quot;]&quot;,&quot;if&quot;,&quot;before&quot;,&quot;in&quot;,&quot;[&quot;,&quot;\&quot;(\&quot;&quot;,&quot;,&quot;,&quot;\&quot;,\&quot;&quot;,&quot;,&quot;,&quot;\&quot;)\&quot;&quot;,&quot;]&quot;,&quot;and&quot;,&quot;after&quot;,&quot;in&quot;,&quot;[&quot;,&quot;\&quot;:\&quot;&quot;,&quot;,&quot;,&quot;\&quot;;\&quot;&quot;,&quot;]&quot;,&quot;:&quot;,&quot;return&quot;,&quot;m&quot;,&quot;.&quot;,&quot;start&quot;,&quot;(&quot;,&quot;)&quot;,&quot;return&quot;,&quot;None&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;find_otu\&quot;,\n  \&quot;(\&quot;,\n  \&quot;otuid\&quot;,\n  \&quot;,\&quot;,\n  \&quot;tree\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;for\&quot;,\n  \&quot;m\&quot;,\n  \&quot;in\&quot;,\n  \&quot;re\&quot;,\n  \&quot;.\&quot;,\n  \&quot;finditer\&quot;,\n  \&quot;(\&quot;,\n  \&quot;otuid\&quot;,\n  \&quot;,\&quot;,\n  \&quot;tree\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;before\&quot;,\n  \&quot;,\&quot;,\n  \&quot;after\&quot;,\n  \&quot;=\&quot;,\n  \&quot;tree\&quot;,\n  \&quot;[\&quot;,\n  \&quot;m\&quot;,\n  \&quot;.\&quot;,\n  \&quot;start\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;tree\&quot;,\n  \&quot;[\&quot;,\n  \&quot;m\&quot;,\n  \&quot;.\&quot;,\n  \&quot;start\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;+\&quot;,\n  \&quot;len\&quot;,\n  \&quot;(\&quot;,\n  \&quot;otuid\&quot;,\n  \&quot;)\&quot;,\n  \&quot;]\&quot;,\n  \&quot;if\&quot;,\n  \&quot;before\&quot;,\n  \&quot;in\&quot;,\n  \&quot;[\&quot;,\n  \&quot;\\\&quot;(\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;,\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;)\\\&quot;\&quot;,\n  \&quot;]\&quot;,\n  \&quot;and\&quot;,\n  \&quot;after\&quot;,\n  \&quot;in\&quot;,\n  \&quot;[\&quot;,\n  \&quot;\\\&quot;:\\\&quot;\&quot;,\n  \&quot;,\&quot;,\n  \&quot;\\\&quot;;\\\&quot;\&quot;,\n  \&quot;]\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;m\&quot;,\n  \&quot;.\&quot;,\n  \&quot;start\&quot;,\n  \&quot;(\&quot;,\n  \&quot;)\&quot;,\n  \&quot;return\&quot;,\n  \&quot;None\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Find an OTU ID in a Newick-format tree.\n    Return the starting position of the ID or None if not found.&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Find&quot;,&quot;an&quot;,&quot;OTU&quot;,&quot;ID&quot;,&quot;in&quot;,&quot;a&quot;,&quot;Newick&quot;,&quot;-&quot;,&quot;format&quot;,&quot;tree&quot;,&quot;.&quot;,&quot;Return&quot;,&quot;the&quot;,&quot;starting&quot;,&quot;position&quot;,&quot;of&quot;,&quot;the&quot;,&quot;ID&quot;,&quot;or&quot;,&quot;None&quot;,&quot;if&quot;,&quot;not&quot;,&quot;found&quot;,&quot;.&quot;],&quot;string&quot;:&quot;[\n  \&quot;Find\&quot;,\n  \&quot;an\&quot;,\n  \&quot;OTU\&quot;,\n  \&quot;ID\&quot;,\n  \&quot;in\&quot;,\n  \&quot;a\&quot;,\n  \&quot;Newick\&quot;,\n  \&quot;-\&quot;,\n  \&quot;format\&quot;,\n  \&quot;tree\&quot;,\n  \&quot;.\&quot;,\n  \&quot;Return\&quot;,\n  \&quot;the\&quot;,\n  \&quot;starting\&quot;,\n  \&quot;position\&quot;,\n  \&quot;of\&quot;,\n  \&quot;the\&quot;,\n  \&quot;ID\&quot;,\n  \&quot;or\&quot;,\n  \&quot;None\&quot;,\n  \&quot;if\&quot;,\n  \&quot;not\&quot;,\n  \&quot;found\&quot;,\n  \&quot;.\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/bin/iTol.py#L17-L26&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:56,&quot;string&quot;:&quot;56&quot;}}},{&quot;rowIdx&quot;:57,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;smdabdoub/phylotoast&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;bin/iTol.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;newick_replace_otuids&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def newick_replace_otuids(tree, biomf):\n    \&quot;\&quot;\&quot;\n    Replace the OTU ids in the Newick phylogenetic tree format with truncated\n    OTU names\n    \&quot;\&quot;\&quot;\n    for val, id_, md in biomf.iter(axis=\&quot;observation\&quot;):\n        otu_loc = find_otu(id_, tree)\n        if otu_loc is not None:\n            tree = tree[:otu_loc] + \\\n                   oc.otu_name(md[\&quot;taxonomy\&quot;]) + \\\n                   tree[otu_loc + len(id_):]\n    return tree&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def newick_replace_otuids(tree, biomf):\n    \&quot;\&quot;\&quot;\n    Replace the OTU ids in the Newick phylogenetic tree format with truncated\n    OTU names\n    \&quot;\&quot;\&quot;\n    for val, id_, md in biomf.iter(axis=\&quot;observation\&quot;):\n        otu_loc = find_otu(id_, tree)\n        if otu_loc is not None:\n            tree = tree[:otu_loc] + \\\n                   oc.otu_name(md[\&quot;taxonomy\&quot;]) + \\\n                   tree[otu_loc + len(id_):]\n    return tree&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;newick_replace_otuids&quot;,&quot;(&quot;,&quot;tree&quot;,&quot;,&quot;,&quot;biomf&quot;,&quot;)&quot;,&quot;:&quot;,&quot;for&quot;,&quot;val&quot;,&quot;,&quot;,&quot;id_&quot;,&quot;,&quot;,&quot;md&quot;,&quot;in&quot;,&quot;biomf&quot;,&quot;.&quot;,&quot;iter&quot;,&quot;(&quot;,&quot;axis&quot;,&quot;=&quot;,&quot;\&quot;observation\&quot;&quot;,&quot;)&quot;,&quot;:&quot;,&quot;otu_loc&quot;,&quot;=&quot;,&quot;find_otu&quot;,&quot;(&quot;,&quot;id_&quot;,&quot;,&quot;,&quot;tree&quot;,&quot;)&quot;,&quot;if&quot;,&quot;otu_loc&quot;,&quot;is&quot;,&quot;not&quot;,&quot;None&quot;,&quot;:&quot;,&quot;tree&quot;,&quot;=&quot;,&quot;tree&quot;,&quot;[&quot;,&quot;:&quot;,&quot;otu_loc&quot;,&quot;]&quot;,&quot;+&quot;,&quot;oc&quot;,&quot;.&quot;,&quot;otu_name&quot;,&quot;(&quot;,&quot;md&quot;,&quot;[&quot;,&quot;\&quot;taxonomy\&quot;&quot;,&quot;]&quot;,&quot;)&quot;,&quot;+&quot;,&quot;tree&quot;,&quot;[&quot;,&quot;otu_loc&quot;,&quot;+&quot;,&quot;len&quot;,&quot;(&quot;,&quot;id_&quot;,&quot;)&quot;,&quot;:&quot;,&quot;]&quot;,&quot;return&quot;,&quot;tree&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;newick_replace_otuids\&quot;,\n  \&quot;(\&quot;,\n  \&quot;tree\&quot;,\n  \&quot;,\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;for\&quot;,\n  \&quot;val\&quot;,\n  \&quot;,\&quot;,\n  \&quot;id_\&quot;,\n  \&quot;,\&quot;,\n  \&quot;md\&quot;,\n  \&quot;in\&quot;,\n  \&quot;biomf\&quot;,\n  \&quot;.\&quot;,\n  \&quot;iter\&quot;,\n  \&quot;(\&quot;,\n  \&quot;axis\&quot;,\n  \&quot;=\&quot;,\n  \&quot;\\\&quot;observation\\\&quot;\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;otu_loc\&quot;,\n  \&quot;=\&quot;,\n  \&quot;find_otu\&quot;,\n  \&quot;(\&quot;,\n  \&quot;id_\&quot;,\n  \&quot;,\&quot;,\n  \&quot;tree\&quot;,\n  \&quot;)\&quot;,\n  \&quot;if\&quot;,\n  \&quot;otu_loc\&quot;,\n  \&quot;is\&quot;,\n  \&quot;not\&quot;,\n  \&quot;None\&quot;,\n  \&quot;:\&quot;,\n  \&quot;tree\&quot;,\n  \&quot;=\&quot;,\n  \&quot;tree\&quot;,\n  \&quot;[\&quot;,\n  \&quot;:\&quot;,\n  \&quot;otu_loc\&quot;,\n  \&quot;]\&quot;,\n  \&quot;+\&quot;,\n  \&quot;oc\&quot;,\n  \&quot;.\&quot;,\n  \&quot;otu_name\&quot;,\n  \&quot;(\&quot;,\n  \&quot;md\&quot;,\n  \&quot;[\&quot;,\n  \&quot;\\\&quot;taxonomy\\\&quot;\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;+\&quot;,\n  \&quot;tree\&quot;,\n  \&quot;[\&quot;,\n  \&quot;otu_loc\&quot;,\n  \&quot;+\&quot;,\n  \&quot;len\&quot;,\n  \&quot;(\&quot;,\n  \&quot;id_\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;]\&quot;,\n  \&quot;return\&quot;,\n  \&quot;tree\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;Replace the OTU ids in the Newick phylogenetic tree format with truncated\n    OTU names&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;Replace&quot;,&quot;the&quot;,&quot;OTU&quot;,&quot;ids&quot;,&quot;in&quot;,&quot;the&quot;,&quot;Newick&quot;,&quot;phylogenetic&quot;,&quot;tree&quot;,&quot;format&quot;,&quot;with&quot;,&quot;truncated&quot;,&quot;OTU&quot;,&quot;names&quot;],&quot;string&quot;:&quot;[\n  \&quot;Replace\&quot;,\n  \&quot;the\&quot;,\n  \&quot;OTU\&quot;,\n  \&quot;ids\&quot;,\n  \&quot;in\&quot;,\n  \&quot;the\&quot;,\n  \&quot;Newick\&quot;,\n  \&quot;phylogenetic\&quot;,\n  \&quot;tree\&quot;,\n  \&quot;format\&quot;,\n  \&quot;with\&quot;,\n  \&quot;truncated\&quot;,\n  \&quot;OTU\&quot;,\n  \&quot;names\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;0b74ef171e6a84761710548501dfac71285a58a3&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/bin/iTol.py#L29-L40&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:57,&quot;string&quot;:&quot;57&quot;}}},{&quot;rowIdx&quot;:58,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;christophertbrown/bioscripts&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ctbBio/cluster_ani.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;genome_info&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def genome_info(genome, info):\n    \&quot;\&quot;\&quot;\n    return genome info for choosing representative\n\n    if ggKbase table provided - choose rep based on SCGs and genome length\n        - priority for most SCGs - extra SCGs, then largest genome\n\n    otherwise, based on largest genome\n    \&quot;\&quot;\&quot;\n    try:\n        scg       = info['#SCGs']\n        dups      = info['#SCG duplicates']\n        length    = info['genome size (bp)']\n        return [scg - dups, length, genome]\n    except:\n        return [False, False, info['genome size (bp)'], genome]&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def genome_info(genome, info):\n    \&quot;\&quot;\&quot;\n    return genome info for choosing representative\n\n    if ggKbase table provided - choose rep based on SCGs and genome length\n        - priority for most SCGs - extra SCGs, then largest genome\n\n    otherwise, based on largest genome\n    \&quot;\&quot;\&quot;\n    try:\n        scg       = info['#SCGs']\n        dups      = info['#SCG duplicates']\n        length    = info['genome size (bp)']\n        return [scg - dups, length, genome]\n    except:\n        return [False, False, info['genome size (bp)'], genome]&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;genome_info&quot;,&quot;(&quot;,&quot;genome&quot;,&quot;,&quot;,&quot;info&quot;,&quot;)&quot;,&quot;:&quot;,&quot;try&quot;,&quot;:&quot;,&quot;scg&quot;,&quot;=&quot;,&quot;info&quot;,&quot;[&quot;,&quot;'#SCGs'&quot;,&quot;]&quot;,&quot;dups&quot;,&quot;=&quot;,&quot;info&quot;,&quot;[&quot;,&quot;'#SCG duplicates'&quot;,&quot;]&quot;,&quot;length&quot;,&quot;=&quot;,&quot;info&quot;,&quot;[&quot;,&quot;'genome size (bp)'&quot;,&quot;]&quot;,&quot;return&quot;,&quot;[&quot;,&quot;scg&quot;,&quot;-&quot;,&quot;dups&quot;,&quot;,&quot;,&quot;length&quot;,&quot;,&quot;,&quot;genome&quot;,&quot;]&quot;,&quot;except&quot;,&quot;:&quot;,&quot;return&quot;,&quot;[&quot;,&quot;False&quot;,&quot;,&quot;,&quot;False&quot;,&quot;,&quot;,&quot;info&quot;,&quot;[&quot;,&quot;'genome size (bp)'&quot;,&quot;]&quot;,&quot;,&quot;,&quot;genome&quot;,&quot;]&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;genome_info\&quot;,\n  \&quot;(\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;,\&quot;,\n  \&quot;info\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;try\&quot;,\n  \&quot;:\&quot;,\n  \&quot;scg\&quot;,\n  \&quot;=\&quot;,\n  \&quot;info\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'#SCGs'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;dups\&quot;,\n  \&quot;=\&quot;,\n  \&quot;info\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'#SCG duplicates'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;length\&quot;,\n  \&quot;=\&quot;,\n  \&quot;info\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'genome size (bp)'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;return\&quot;,\n  \&quot;[\&quot;,\n  \&quot;scg\&quot;,\n  \&quot;-\&quot;,\n  \&quot;dups\&quot;,\n  \&quot;,\&quot;,\n  \&quot;length\&quot;,\n  \&quot;,\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;]\&quot;,\n  \&quot;except\&quot;,\n  \&quot;:\&quot;,\n  \&quot;return\&quot;,\n  \&quot;[\&quot;,\n  \&quot;False\&quot;,\n  \&quot;,\&quot;,\n  \&quot;False\&quot;,\n  \&quot;,\&quot;,\n  \&quot;info\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'genome size (bp)'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;]\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;return genome info for choosing representative\n\n    if ggKbase table provided - choose rep based on SCGs and genome length\n        - priority for most SCGs - extra SCGs, then largest genome\n\n    otherwise, based on largest genome&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;return&quot;,&quot;genome&quot;,&quot;info&quot;,&quot;for&quot;,&quot;choosing&quot;,&quot;representative&quot;],&quot;string&quot;:&quot;[\n  \&quot;return\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;info\&quot;,\n  \&quot;for\&quot;,\n  \&quot;choosing\&quot;,\n  \&quot;representative\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;83b2566b3a5745437ec651cd6cafddd056846240&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/cluster_ani.py#L97-L112&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:58,&quot;string&quot;:&quot;58&quot;}}},{&quot;rowIdx&quot;:59,&quot;cells&quot;:{&quot;repo&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;christophertbrown/bioscripts&quot;},&quot;path&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;ctbBio/cluster_ani.py&quot;},&quot;func_name&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;print_clusters&quot;},&quot;original_string&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def print_clusters(fastas, info, ANI):\n    \&quot;\&quot;\&quot;\n    choose represenative genome and\n    print cluster information\n\n    *if ggKbase table is provided, use SCG info to choose best genome\n    \&quot;\&quot;\&quot;\n    header = ['#cluster', 'num. genomes', 'rep.', 'genome', '#SCGs', '#SCG duplicates', \\\n            'genome size (bp)', 'fragments', 'list']\n    yield header\n    in_cluster = []\n    for cluster_num, cluster in enumerate(connected_components(ANI)):\n        cluster = sorted([genome_info(genome, info[genome]) \\\n                            for genome in cluster], \\\n                            key = lambda x: x[0:], reverse = True)\n        rep = cluster[0][-1]\n        cluster = [i[-1] for i in cluster]\n        size = len(cluster)\n        for genome in cluster:\n            in_cluster.append(genome)\n            try:\n                stats = [size, rep, genome, \\\n                            info[genome]['#SCGs'], info[genome]['#SCG duplicates'], \\\n                            info[genome]['genome size (bp)'], info[genome]['# contigs'], cluster]\n            except:\n                stats = [size, rep, genome, \\\n                            'n/a', 'n/a', \\\n                            info[genome]['genome size (bp)'], info[genome]['# contigs'], cluster]\n            if rep == genome:\n                stats = ['*%s' % (cluster_num)] + stats\n            else:\n                stats = [cluster_num] + stats\n            yield stats\n    # print singletons\n    try:\n        start = cluster_num + 1\n    except:\n        start = 0\n    fastas = set([i.rsplit('.', 1)[0].rsplit('/', 1)[-1].rsplit('.contigs')[0] for i in fastas])\n    for cluster_num, genome in \\\n            enumerate(fastas.difference(set(in_cluster)), start):\n        try:\n            stats = ['*%s' % (cluster_num), 1, genome, genome, \\\n                        info[genome]['#SCGs'], info[genome]['#SCG duplicates'], \\\n                        info[genome]['genome size (bp)'], info[genome]['# contigs'], [genome]]\n        except:\n            stats = ['*%s' % (cluster_num), 1, genome, genome, \\\n                        'n/a', 'n/a', \\\n                        info[genome]['genome size (bp)'], info[genome]['# contigs'], [genome]]\n        yield stats&quot;},&quot;language&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;python&quot;},&quot;code&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;def print_clusters(fastas, info, ANI):\n    \&quot;\&quot;\&quot;\n    choose represenative genome and\n    print cluster information\n\n    *if ggKbase table is provided, use SCG info to choose best genome\n    \&quot;\&quot;\&quot;\n    header = ['#cluster', 'num. genomes', 'rep.', 'genome', '#SCGs', '#SCG duplicates', \\\n            'genome size (bp)', 'fragments', 'list']\n    yield header\n    in_cluster = []\n    for cluster_num, cluster in enumerate(connected_components(ANI)):\n        cluster = sorted([genome_info(genome, info[genome]) \\\n                            for genome in cluster], \\\n                            key = lambda x: x[0:], reverse = True)\n        rep = cluster[0][-1]\n        cluster = [i[-1] for i in cluster]\n        size = len(cluster)\n        for genome in cluster:\n            in_cluster.append(genome)\n            try:\n                stats = [size, rep, genome, \\\n                            info[genome]['#SCGs'], info[genome]['#SCG duplicates'], \\\n                            info[genome]['genome size (bp)'], info[genome]['# contigs'], cluster]\n            except:\n                stats = [size, rep, genome, \\\n                            'n/a', 'n/a', \\\n                            info[genome]['genome size (bp)'], info[genome]['# contigs'], cluster]\n            if rep == genome:\n                stats = ['*%s' % (cluster_num)] + stats\n            else:\n                stats = [cluster_num] + stats\n            yield stats\n    # print singletons\n    try:\n        start = cluster_num + 1\n    except:\n        start = 0\n    fastas = set([i.rsplit('.', 1)[0].rsplit('/', 1)[-1].rsplit('.contigs')[0] for i in fastas])\n    for cluster_num, genome in \\\n            enumerate(fastas.difference(set(in_cluster)), start):\n        try:\n            stats = ['*%s' % (cluster_num), 1, genome, genome, \\\n                        info[genome]['#SCGs'], info[genome]['#SCG duplicates'], \\\n                        info[genome]['genome size (bp)'], info[genome]['# contigs'], [genome]]\n        except:\n            stats = ['*%s' % (cluster_num), 1, genome, genome, \\\n                        'n/a', 'n/a', \\\n                        info[genome]['genome size (bp)'], info[genome]['# contigs'], [genome]]\n        yield stats&quot;},&quot;code_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;def&quot;,&quot;print_clusters&quot;,&quot;(&quot;,&quot;fastas&quot;,&quot;,&quot;,&quot;info&quot;,&quot;,&quot;,&quot;ANI&quot;,&quot;)&quot;,&quot;:&quot;,&quot;header&quot;,&quot;=&quot;,&quot;[&quot;,&quot;'#cluster'&quot;,&quot;,&quot;,&quot;'num. genomes'&quot;,&quot;,&quot;,&quot;'rep.'&quot;,&quot;,&quot;,&quot;'genome'&quot;,&quot;,&quot;,&quot;'#SCGs'&quot;,&quot;,&quot;,&quot;'#SCG duplicates'&quot;,&quot;,&quot;,&quot;'genome size (bp)'&quot;,&quot;,&quot;,&quot;'fragments'&quot;,&quot;,&quot;,&quot;'list'&quot;,&quot;]&quot;,&quot;yield&quot;,&quot;header&quot;,&quot;in_cluster&quot;,&quot;=&quot;,&quot;[&quot;,&quot;]&quot;,&quot;for&quot;,&quot;cluster_num&quot;,&quot;,&quot;,&quot;cluster&quot;,&quot;in&quot;,&quot;enumerate&quot;,&quot;(&quot;,&quot;connected_components&quot;,&quot;(&quot;,&quot;ANI&quot;,&quot;)&quot;,&quot;)&quot;,&quot;:&quot;,&quot;cluster&quot;,&quot;=&quot;,&quot;sorted&quot;,&quot;(&quot;,&quot;[&quot;,&quot;genome_info&quot;,&quot;(&quot;,&quot;genome&quot;,&quot;,&quot;,&quot;info&quot;,&quot;[&quot;,&quot;genome&quot;,&quot;]&quot;,&quot;)&quot;,&quot;for&quot;,&quot;genome&quot;,&quot;in&quot;,&quot;cluster&quot;,&quot;]&quot;,&quot;,&quot;,&quot;key&quot;,&quot;=&quot;,&quot;lambda&quot;,&quot;x&quot;,&quot;:&quot;,&quot;x&quot;,&quot;[&quot;,&quot;0&quot;,&quot;:&quot;,&quot;]&quot;,&quot;,&quot;,&quot;reverse&quot;,&quot;=&quot;,&quot;True&quot;,&quot;)&quot;,&quot;rep&quot;,&quot;=&quot;,&quot;cluster&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;[&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;,&quot;cluster&quot;,&quot;=&quot;,&quot;[&quot;,&quot;i&quot;,&quot;[&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;,&quot;for&quot;,&quot;i&quot;,&quot;in&quot;,&quot;cluster&quot;,&quot;]&quot;,&quot;size&quot;,&quot;=&quot;,&quot;len&quot;,&quot;(&quot;,&quot;cluster&quot;,&quot;)&quot;,&quot;for&quot;,&quot;genome&quot;,&quot;in&quot;,&quot;cluster&quot;,&quot;:&quot;,&quot;in_cluster&quot;,&quot;.&quot;,&quot;append&quot;,&quot;(&quot;,&quot;genome&quot;,&quot;)&quot;,&quot;try&quot;,&quot;:&quot;,&quot;stats&quot;,&quot;=&quot;,&quot;[&quot;,&quot;size&quot;,&quot;,&quot;,&quot;rep&quot;,&quot;,&quot;,&quot;genome&quot;,&quot;,&quot;,&quot;info&quot;,&quot;[&quot;,&quot;genome&quot;,&quot;]&quot;,&quot;[&quot;,&quot;'#SCGs'&quot;,&quot;]&quot;,&quot;,&quot;,&quot;info&quot;,&quot;[&quot;,&quot;genome&quot;,&quot;]&quot;,&quot;[&quot;,&quot;'#SCG duplicates'&quot;,&quot;]&quot;,&quot;,&quot;,&quot;info&quot;,&quot;[&quot;,&quot;genome&quot;,&quot;]&quot;,&quot;[&quot;,&quot;'genome size (bp)'&quot;,&quot;]&quot;,&quot;,&quot;,&quot;info&quot;,&quot;[&quot;,&quot;genome&quot;,&quot;]&quot;,&quot;[&quot;,&quot;'# contigs'&quot;,&quot;]&quot;,&quot;,&quot;,&quot;cluster&quot;,&quot;]&quot;,&quot;except&quot;,&quot;:&quot;,&quot;stats&quot;,&quot;=&quot;,&quot;[&quot;,&quot;size&quot;,&quot;,&quot;,&quot;rep&quot;,&quot;,&quot;,&quot;genome&quot;,&quot;,&quot;,&quot;'n/a'&quot;,&quot;,&quot;,&quot;'n/a'&quot;,&quot;,&quot;,&quot;info&quot;,&quot;[&quot;,&quot;genome&quot;,&quot;]&quot;,&quot;[&quot;,&quot;'genome size (bp)'&quot;,&quot;]&quot;,&quot;,&quot;,&quot;info&quot;,&quot;[&quot;,&quot;genome&quot;,&quot;]&quot;,&quot;[&quot;,&quot;'# contigs'&quot;,&quot;]&quot;,&quot;,&quot;,&quot;cluster&quot;,&quot;]&quot;,&quot;if&quot;,&quot;rep&quot;,&quot;==&quot;,&quot;genome&quot;,&quot;:&quot;,&quot;stats&quot;,&quot;=&quot;,&quot;[&quot;,&quot;'*%s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;cluster_num&quot;,&quot;)&quot;,&quot;]&quot;,&quot;+&quot;,&quot;stats&quot;,&quot;else&quot;,&quot;:&quot;,&quot;stats&quot;,&quot;=&quot;,&quot;[&quot;,&quot;cluster_num&quot;,&quot;]&quot;,&quot;+&quot;,&quot;stats&quot;,&quot;yield&quot;,&quot;stats&quot;,&quot;# print singletons&quot;,&quot;try&quot;,&quot;:&quot;,&quot;start&quot;,&quot;=&quot;,&quot;cluster_num&quot;,&quot;+&quot;,&quot;1&quot;,&quot;except&quot;,&quot;:&quot;,&quot;start&quot;,&quot;=&quot;,&quot;0&quot;,&quot;fastas&quot;,&quot;=&quot;,&quot;set&quot;,&quot;(&quot;,&quot;[&quot;,&quot;i&quot;,&quot;.&quot;,&quot;rsplit&quot;,&quot;(&quot;,&quot;'.'&quot;,&quot;,&quot;,&quot;1&quot;,&quot;)&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;.&quot;,&quot;rsplit&quot;,&quot;(&quot;,&quot;'/'&quot;,&quot;,&quot;,&quot;1&quot;,&quot;)&quot;,&quot;[&quot;,&quot;-&quot;,&quot;1&quot;,&quot;]&quot;,&quot;.&quot;,&quot;rsplit&quot;,&quot;(&quot;,&quot;'.contigs'&quot;,&quot;)&quot;,&quot;[&quot;,&quot;0&quot;,&quot;]&quot;,&quot;for&quot;,&quot;i&quot;,&quot;in&quot;,&quot;fastas&quot;,&quot;]&quot;,&quot;)&quot;,&quot;for&quot;,&quot;cluster_num&quot;,&quot;,&quot;,&quot;genome&quot;,&quot;in&quot;,&quot;enumerate&quot;,&quot;(&quot;,&quot;fastas&quot;,&quot;.&quot;,&quot;difference&quot;,&quot;(&quot;,&quot;set&quot;,&quot;(&quot;,&quot;in_cluster&quot;,&quot;)&quot;,&quot;)&quot;,&quot;,&quot;,&quot;start&quot;,&quot;)&quot;,&quot;:&quot;,&quot;try&quot;,&quot;:&quot;,&quot;stats&quot;,&quot;=&quot;,&quot;[&quot;,&quot;'*%s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;cluster_num&quot;,&quot;)&quot;,&quot;,&quot;,&quot;1&quot;,&quot;,&quot;,&quot;genome&quot;,&quot;,&quot;,&quot;genome&quot;,&quot;,&quot;,&quot;info&quot;,&quot;[&quot;,&quot;genome&quot;,&quot;]&quot;,&quot;[&quot;,&quot;'#SCGs'&quot;,&quot;]&quot;,&quot;,&quot;,&quot;info&quot;,&quot;[&quot;,&quot;genome&quot;,&quot;]&quot;,&quot;[&quot;,&quot;'#SCG duplicates'&quot;,&quot;]&quot;,&quot;,&quot;,&quot;info&quot;,&quot;[&quot;,&quot;genome&quot;,&quot;]&quot;,&quot;[&quot;,&quot;'genome size (bp)'&quot;,&quot;]&quot;,&quot;,&quot;,&quot;info&quot;,&quot;[&quot;,&quot;genome&quot;,&quot;]&quot;,&quot;[&quot;,&quot;'# contigs'&quot;,&quot;]&quot;,&quot;,&quot;,&quot;[&quot;,&quot;genome&quot;,&quot;]&quot;,&quot;]&quot;,&quot;except&quot;,&quot;:&quot;,&quot;stats&quot;,&quot;=&quot;,&quot;[&quot;,&quot;'*%s'&quot;,&quot;%&quot;,&quot;(&quot;,&quot;cluster_num&quot;,&quot;)&quot;,&quot;,&quot;,&quot;1&quot;,&quot;,&quot;,&quot;genome&quot;,&quot;,&quot;,&quot;genome&quot;,&quot;,&quot;,&quot;'n/a'&quot;,&quot;,&quot;,&quot;'n/a'&quot;,&quot;,&quot;,&quot;info&quot;,&quot;[&quot;,&quot;genome&quot;,&quot;]&quot;,&quot;[&quot;,&quot;'genome size (bp)'&quot;,&quot;]&quot;,&quot;,&quot;,&quot;info&quot;,&quot;[&quot;,&quot;genome&quot;,&quot;]&quot;,&quot;[&quot;,&quot;'# contigs'&quot;,&quot;]&quot;,&quot;,&quot;,&quot;[&quot;,&quot;genome&quot;,&quot;]&quot;,&quot;]&quot;,&quot;yield&quot;,&quot;stats&quot;],&quot;string&quot;:&quot;[\n  \&quot;def\&quot;,\n  \&quot;print_clusters\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fastas\&quot;,\n  \&quot;,\&quot;,\n  \&quot;info\&quot;,\n  \&quot;,\&quot;,\n  \&quot;ANI\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;header\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'#cluster'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'num. genomes'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'rep.'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'genome'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'#SCGs'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'#SCG duplicates'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'genome size (bp)'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'fragments'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'list'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;yield\&quot;,\n  \&quot;header\&quot;,\n  \&quot;in_cluster\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;]\&quot;,\n  \&quot;for\&quot;,\n  \&quot;cluster_num\&quot;,\n  \&quot;,\&quot;,\n  \&quot;cluster\&quot;,\n  \&quot;in\&quot;,\n  \&quot;enumerate\&quot;,\n  \&quot;(\&quot;,\n  \&quot;connected_components\&quot;,\n  \&quot;(\&quot;,\n  \&quot;ANI\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;cluster\&quot;,\n  \&quot;=\&quot;,\n  \&quot;sorted\&quot;,\n  \&quot;(\&quot;,\n  \&quot;[\&quot;,\n  \&quot;genome_info\&quot;,\n  \&quot;(\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;,\&quot;,\n  \&quot;info\&quot;,\n  \&quot;[\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;in\&quot;,\n  \&quot;cluster\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;key\&quot;,\n  \&quot;=\&quot;,\n  \&quot;lambda\&quot;,\n  \&quot;x\&quot;,\n  \&quot;:\&quot;,\n  \&quot;x\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;:\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;reverse\&quot;,\n  \&quot;=\&quot;,\n  \&quot;True\&quot;,\n  \&quot;)\&quot;,\n  \&quot;rep\&quot;,\n  \&quot;=\&quot;,\n  \&quot;cluster\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;cluster\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;i\&quot;,\n  \&quot;[\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;in\&quot;,\n  \&quot;cluster\&quot;,\n  \&quot;]\&quot;,\n  \&quot;size\&quot;,\n  \&quot;=\&quot;,\n  \&quot;len\&quot;,\n  \&quot;(\&quot;,\n  \&quot;cluster\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;in\&quot;,\n  \&quot;cluster\&quot;,\n  \&quot;:\&quot;,\n  \&quot;in_cluster\&quot;,\n  \&quot;.\&quot;,\n  \&quot;append\&quot;,\n  \&quot;(\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;)\&quot;,\n  \&quot;try\&quot;,\n  \&quot;:\&quot;,\n  \&quot;stats\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;size\&quot;,\n  \&quot;,\&quot;,\n  \&quot;rep\&quot;,\n  \&quot;,\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;,\&quot;,\n  \&quot;info\&quot;,\n  \&quot;[\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'#SCGs'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;info\&quot;,\n  \&quot;[\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'#SCG duplicates'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;info\&quot;,\n  \&quot;[\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'genome size (bp)'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;info\&quot;,\n  \&quot;[\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'# contigs'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;cluster\&quot;,\n  \&quot;]\&quot;,\n  \&quot;except\&quot;,\n  \&quot;:\&quot;,\n  \&quot;stats\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;size\&quot;,\n  \&quot;,\&quot;,\n  \&quot;rep\&quot;,\n  \&quot;,\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'n/a'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'n/a'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;info\&quot;,\n  \&quot;[\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'genome size (bp)'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;info\&quot;,\n  \&quot;[\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'# contigs'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;cluster\&quot;,\n  \&quot;]\&quot;,\n  \&quot;if\&quot;,\n  \&quot;rep\&quot;,\n  \&quot;==\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;:\&quot;,\n  \&quot;stats\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'*%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;cluster_num\&quot;,\n  \&quot;)\&quot;,\n  \&quot;]\&quot;,\n  \&quot;+\&quot;,\n  \&quot;stats\&quot;,\n  \&quot;else\&quot;,\n  \&quot;:\&quot;,\n  \&quot;stats\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;cluster_num\&quot;,\n  \&quot;]\&quot;,\n  \&quot;+\&quot;,\n  \&quot;stats\&quot;,\n  \&quot;yield\&quot;,\n  \&quot;stats\&quot;,\n  \&quot;# print singletons\&quot;,\n  \&quot;try\&quot;,\n  \&quot;:\&quot;,\n  \&quot;start\&quot;,\n  \&quot;=\&quot;,\n  \&quot;cluster_num\&quot;,\n  \&quot;+\&quot;,\n  \&quot;1\&quot;,\n  \&quot;except\&quot;,\n  \&quot;:\&quot;,\n  \&quot;start\&quot;,\n  \&quot;=\&quot;,\n  \&quot;0\&quot;,\n  \&quot;fastas\&quot;,\n  \&quot;=\&quot;,\n  \&quot;set\&quot;,\n  \&quot;(\&quot;,\n  \&quot;[\&quot;,\n  \&quot;i\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rsplit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'.'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rsplit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'/'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;-\&quot;,\n  \&quot;1\&quot;,\n  \&quot;]\&quot;,\n  \&quot;.\&quot;,\n  \&quot;rsplit\&quot;,\n  \&quot;(\&quot;,\n  \&quot;'.contigs'\&quot;,\n  \&quot;)\&quot;,\n  \&quot;[\&quot;,\n  \&quot;0\&quot;,\n  \&quot;]\&quot;,\n  \&quot;for\&quot;,\n  \&quot;i\&quot;,\n  \&quot;in\&quot;,\n  \&quot;fastas\&quot;,\n  \&quot;]\&quot;,\n  \&quot;)\&quot;,\n  \&quot;for\&quot;,\n  \&quot;cluster_num\&quot;,\n  \&quot;,\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;in\&quot;,\n  \&quot;enumerate\&quot;,\n  \&quot;(\&quot;,\n  \&quot;fastas\&quot;,\n  \&quot;.\&quot;,\n  \&quot;difference\&quot;,\n  \&quot;(\&quot;,\n  \&quot;set\&quot;,\n  \&quot;(\&quot;,\n  \&quot;in_cluster\&quot;,\n  \&quot;)\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;start\&quot;,\n  \&quot;)\&quot;,\n  \&quot;:\&quot;,\n  \&quot;try\&quot;,\n  \&quot;:\&quot;,\n  \&quot;stats\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'*%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;cluster_num\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;,\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;,\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;,\&quot;,\n  \&quot;info\&quot;,\n  \&quot;[\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'#SCGs'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;info\&quot;,\n  \&quot;[\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'#SCG duplicates'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;info\&quot;,\n  \&quot;[\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'genome size (bp)'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;info\&quot;,\n  \&quot;[\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'# contigs'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;[\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;]\&quot;,\n  \&quot;]\&quot;,\n  \&quot;except\&quot;,\n  \&quot;:\&quot;,\n  \&quot;stats\&quot;,\n  \&quot;=\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'*%s'\&quot;,\n  \&quot;%\&quot;,\n  \&quot;(\&quot;,\n  \&quot;cluster_num\&quot;,\n  \&quot;)\&quot;,\n  \&quot;,\&quot;,\n  \&quot;1\&quot;,\n  \&quot;,\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;,\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'n/a'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;'n/a'\&quot;,\n  \&quot;,\&quot;,\n  \&quot;info\&quot;,\n  \&quot;[\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'genome size (bp)'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;info\&quot;,\n  \&quot;[\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;]\&quot;,\n  \&quot;[\&quot;,\n  \&quot;'# contigs'\&quot;,\n  \&quot;]\&quot;,\n  \&quot;,\&quot;,\n  \&quot;[\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;]\&quot;,\n  \&quot;]\&quot;,\n  \&quot;yield\&quot;,\n  \&quot;stats\&quot;\n]&quot;},&quot;docstring&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;choose represenative genome and\n    print cluster information\n\n    *if ggKbase table is provided, use SCG info to choose best genome&quot;},&quot;docstring_tokens&quot;:{&quot;kind&quot;:&quot;list like&quot;,&quot;value&quot;:[&quot;choose&quot;,&quot;represenative&quot;,&quot;genome&quot;,&quot;and&quot;,&quot;print&quot;,&quot;cluster&quot;,&quot;information&quot;],&quot;string&quot;:&quot;[\n  \&quot;choose\&quot;,\n  \&quot;represenative\&quot;,\n  \&quot;genome\&quot;,\n  \&quot;and\&quot;,\n  \&quot;print\&quot;,\n  \&quot;cluster\&quot;,\n  \&quot;information\&quot;\n]&quot;},&quot;sha&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;83b2566b3a5745437ec651cd6cafddd056846240&quot;},&quot;url&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/cluster_ani.py#L114-L163&quot;},&quot;partition&quot;:{&quot;kind&quot;:&quot;string&quot;,&quot;value&quot;:&quot;train&quot;},&quot;idx&quot;:{&quot;kind&quot;:&quot;number&quot;,&quot;value&quot;:59,&quot;string&quot;:&quot;59&quot;}}}],&quot;truncated&quot;:true},&quot;error&quot;:{&quot;error&quot;:&quot;The dataset generation failed because of a cast error&quot;,&quot;cause_exception&quot;:&quot;DatasetGenerationCastError&quot;,&quot;cause_message&quot;:&quot;An error occurred while generating the dataset\n\nAll the data files must have the same columns, but at some point there are 9 new columns ({'parameters', 'function_tokens', 'nwo', 'docstring_summary', 'return_statement', 'score', 'function', 'identifier', 'argument_list'}) and 6 missing columns ({'original_string', 'partition', 'repo', 'code', 'code_tokens', 'func_name'}).\n\nThis happened while the json dataset builder was generating data using\n\nhf://datasets/semeru/Text-Code-CodeSearchNet-Python/valid.jsonl (at revision 8f65de8a4141f24c7a2099d21cb715cdb2b1b59a)\n\nPlease either edit the data files to have matching columns, or separate them into different configurations (see docs at https://hf.co/docs/hub/datasets-manual-configuration#multiple-configurations)&quot;,&quot;cause_traceback&quot;:[&quot;Traceback (most recent call last):\n&quot;,&quot;  File \&quot;/src/services/worker/.venv/lib/python3.9/site-packages/datasets/builder.py\&quot;, line 2011, in _prepare_split_single\n    writer.write_table(table)\n&quot;,&quot;  File \&quot;/src/services/worker/.venv/lib/python3.9/site-packages/datasets/arrow_writer.py\&quot;, line 585, in write_table\n    pa_table = table_cast(pa_table, self._schema)\n&quot;,&quot;  File \&quot;/src/services/worker/.venv/lib/python3.9/site-packages/datasets/table.py\&quot;, line 2302, in table_cast\n    return cast_table_to_schema(table, schema)\n&quot;,&quot;  File \&quot;/src/services/worker/.venv/lib/python3.9/site-packages/datasets/table.py\&quot;, line 2256, in cast_table_to_schema\n    raise CastError(\n&quot;,&quot;datasets.table.CastError: Couldn't cast\nurl: string\nsha: string\ndocstring_summary: string\nlanguage: string\nparameters: string\nreturn_statement: string\nargument_list: string\nfunction_tokens: list<item: string>\n  child 0, item: string\nfunction: string\npath: string\nidentifier: string\ndocstring: string\ndocstring_tokens: list<item: string>\n  child 0, item: string\nnwo: string\nscore: double\nidx: int64\nto\n{'repo': Value(dtype='string', id=None), 'path': Value(dtype='string', id=None), 'func_name': Value(dtype='string', id=None), 'original_string': Value(dtype='string', id=None), 'language': Value(dtype='string', id=None), 'code': Value(dtype='string', id=None), 'code_tokens': Sequence(feature=Value(dtype='string', id=None), length=-1, id=None), 'docstring': Value(dtype='string', id=None), 'docstring_tokens': Sequence(feature=Value(dtype='string', id=None), length=-1, id=None), 'sha': Value(dtype='string', id=None), 'url': Value(dtype='string', id=None), 'partition': Value(dtype='string', id=None), 'idx': Value(dtype='int64', id=None)}\nbecause column names don't match\n&quot;,&quot;\nDuring handling of the above exception, another exception occurred:\n\n&quot;,&quot;Traceback (most recent call last):\n&quot;,&quot;  File \&quot;/src/services/worker/src/worker/job_runners/config/parquet_and_info.py\&quot;, line 1321, in compute_config_parquet_and_info_response\n    parquet_operations = convert_to_parquet(builder)\n&quot;,&quot;  File \&quot;/src/services/worker/src/worker/job_runners/config/parquet_and_info.py\&quot;, line 935, in convert_to_parquet\n    builder.download_and_prepare(\n&quot;,&quot;  File \&quot;/src/services/worker/.venv/lib/python3.9/site-packages/datasets/builder.py\&quot;, line 1027, in download_and_prepare\n    self._download_and_prepare(\n&quot;,&quot;  File \&quot;/src/services/worker/.venv/lib/python3.9/site-packages/datasets/builder.py\&quot;, line 1122, in _download_and_prepare\n    self._prepare_split(split_generator, **prepare_split_kwargs)\n&quot;,&quot;  File \&quot;/src/services/worker/.venv/lib/python3.9/site-packages/datasets/builder.py\&quot;, line 1882, in _prepare_split\n    for job_id, done, content in self._prepare_split_single(\n&quot;,&quot;  File \&quot;/src/services/worker/.venv/lib/python3.9/site-packages/datasets/builder.py\&quot;, line 2013, in _prepare_split_single\n    raise DatasetGenerationCastError.from_cast_error(\n&quot;,&quot;datasets.exceptions.DatasetGenerationCastError: An error occurred while generating the dataset\n\nAll the data files must have the same columns, but at some point there are 9 new columns ({'parameters', 'function_tokens', 'nwo', 'docstring_summary', 'return_statement', 'score', 'function', 'identifier', 'argument_list'}) and 6 missing columns ({'original_string', 'partition', 'repo', 'code', 'code_tokens', 'func_name'}).\n\nThis happened while the json dataset builder was generating data using\n\nhf://datasets/semeru/Text-Code-CodeSearchNet-Python/valid.jsonl (at revision 8f65de8a4141f24c7a2099d21cb715cdb2b1b59a)\n\nPlease either edit the data files to have matching columns, or separate them into different configurations (see docs at https://hf.co/docs/hub/datasets-manual-configuration#multiple-configurations)\n&quot;],&quot;error_code&quot;:&quot;DatasetGenerationCastError&quot;}},&quot;jwt&quot;:&quot;eyJhbGciOiJFZERTQSJ9.eyJyZWFkIjp0cnVlLCJwZXJtaXNzaW9ucyI6eyJyZXBvLmNvbnRlbnQucmVhZCI6dHJ1ZX0sImlhdCI6MTc0MjkyMzA3NSwic3ViIjoiL2RhdGFzZXRzL3NlbWVydS9UZXh0LUNvZGUtQ29kZVNlYXJjaE5ldC1QeXRob24iLCJleHAiOjE3NDI5MjY2NzUsImlzcyI6Imh0dHBzOi8vaHVnZ2luZ2ZhY2UuY28ifQ.e1LqsISvqJrmZn0U8UPKAvn_sYv2f0nAmtwtwWtli0NhAjkvXHGnKzqp-hfhUZlScof814oiVH6_Y73tA3SNAQ&quot;,&quot;displayUrls&quot;:true},&quot;dataset&quot;:&quot;semeru/Text-Code-CodeSearchNet-Python&quot;,&quot;isGated&quot;:false,&quot;isPrivate&quot;:false,&quot;hasParquetFormat&quot;:false,&quot;author&quot;:{&quot;avatarUrl&quot;:&quot;https://cdn-avatars.huggingface.co/v1/production/uploads/1636726823137-5ebb6509a8e72729bee106a0.png&quot;,&quot;fullname&quot;:&quot;Semeru Lab&quot;,&quot;name&quot;:&quot;semeru&quot;,&quot;type&quot;:&quot;org&quot;,&quot;isHf&quot;:false,&quot;isHfAdmin&quot;:false,&quot;isMod&quot;:false,&quot;isEnterprise&quot;:false,&quot;followerCount&quot;:7},&quot;compact&quot;:true}"><div class="flex flex-col overflow-hidden shadow-xs mx-auto mb-10 rounded-lg border pt-2  px-2.5"><div class="mb-2 flex flex-wrap items-center gap-2"><div class="mr-auto flex items-center"><svg class="mr-1 flex-none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 12 12"><path d="M6 9.75828C4.86056 9.75828 3.81948 9.45144 2.87678 8.83776C1.93407 8.22373 1.22089 7.39474 0.737243 6.35077C0.712651 6.30901 0.696256 6.25673 0.688059 6.19393C0.679861 6.13146 0.675763 6.06681 0.675763 6C0.675763 5.93319 0.679861 5.86838 0.688059 5.80557C0.696256 5.7431 0.712651 5.69098 0.737243 5.64923C1.22089 4.60526 1.93407 3.77643 2.87678 3.16274C3.81948 2.54872 4.86056 2.24171 6 2.24171C7.13944 2.24171 8.18052 2.54872 9.12323 3.16274C10.0659 3.77643 10.7791 4.60526 11.2628 5.64923C11.2873 5.69098 11.3037 5.7431 11.3119 5.80557C11.3201 5.86838 11.3242 5.93319 11.3242 6C11.3242 6.06681 11.3201 6.13146 11.3119 6.19393C11.3037 6.25673 11.2873 6.30901 11.2628 6.35077C10.7791 7.39474 10.0659 8.22373 9.12323 8.83776C8.18052 9.45144 7.13944 9.75828 6 9.75828ZM6 8.75608C6.92631 8.75608 7.77688 8.50753 8.5517 8.01043C9.32619 7.51367 9.91838 6.84353 10.3282 6C9.91838 5.15647 9.32619 4.48616 8.5517 3.98907C7.77688 3.4923 6.92631 3.24392 6 3.24392C5.07369 3.24392 4.22312 3.4923 3.4483 3.98907C2.67381 4.48616 2.08162 5.15647 1.67175 6C2.08162 6.84353 2.67381 7.51367 3.4483 8.01043C4.22312 8.50753 5.07369 8.75608 6 8.75608Z" fill="currentColor"></path><path d="M7.80933 5.92992C7.80933 6.51092 7.53544 7.02796 7.10973 7.35894C6.80337 7.59714 6.41838 6.98403 6.00027 6.98403C5.58215 6.98403 5.19716 7.59714 4.8908 7.35894C4.46509 7.02796 4.1912 6.51092 4.1912 5.92992C4.1912 4.9308 5.00115 4.12086 6.00027 4.12086C6.99939 4.12086 7.80933 4.9308 7.80933 5.92992Z" fill="currentColor"></path></svg>
	<div class="whitespace-nowrap font-semibold">Dataset Preview</div>
	<div class="translate-y-px z-10"><svg class="mx-1.5 text-sm text-gray-400 hover:text-black dark:hover:text-white" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M17 22v-8h-4v2h2v6h-3v2h8v-2h-3z" fill="currentColor"></path><path d="M16 8a1.5 1.5 0 1 0 1.5 1.5A1.5 1.5 0 0 0 16 8z" fill="currentColor"></path><path d="M16 30a14 14 0 1 1 14-14a14 14 0 0 1-14 14zm0-26a12 12 0 1 0 12 12A12 12 0 0 0 16 4z" fill="currentColor"></path></svg>
	</div></div>
				
				<button class="btn shadow-xs flex cursor-pointer items-center rounded-sm border px-1 py-0.5 text-xs font-normal text-gray-700 hover:text-gray-800 hover:shadow-inner dark:hover:text-gray-200"><svg class="mr-1.5" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32" style="transform: rotate(360deg);"><path d="M31 16l-7 7l-1.41-1.41L28.17 16l-5.58-5.59L24 9l7 7z" fill="currentColor"></path><path d="M1 16l7-7l1.41 1.41L3.83 16l5.58 5.59L8 23l-7-7z" fill="currentColor"></path><path d="M12.419 25.484L17.639 6l1.932.518L14.35 26z" fill="currentColor"></path></svg>API</button>
					<button class="btn shadow-xs flex cursor-pointer items-center rounded-sm border px-1 py-0.5 text-xs font-normal text-gray-700 hover:text-gray-800 hover:shadow-inner dark:hover:text-gray-200"><svg class="mr-1.5" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" fill="currentColor" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 12 12"><path d="M9.80603 2.86737H3.56107C3.37704 2.86737 3.20055 2.94048 3.07042 3.0706C2.94029 3.20073 2.86719 3.37723 2.86719 3.56126V9.80622C2.86719 9.99025 2.94029 10.1667 3.07042 10.2969C3.20055 10.427 3.37704 10.5001 3.56107 10.5001H9.80603C9.99006 10.5001 10.1666 10.427 10.2967 10.2969C10.4268 10.1667 10.4999 9.99025 10.4999 9.80622V3.56126C10.4999 3.37723 10.4268 3.20073 10.2967 3.0706C10.1666 2.94048 9.99006 2.86737 9.80603 2.86737Z" fill="currentColor" fill-opacity="0.3"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M2.40942 1.66191C2.05175 1.66191 1.7618 1.95186 1.7618 2.30953V6.76191H1.43799V2.30953C1.43799 1.77303 1.87291 1.3381 2.40942 1.3381H6.45704V1.66191H2.40942Z" fill="currentColor"></path></svg>Embed</button>

					<button class="bg-linear-to-b shadow-xs flex items-center gap-1.5 rounded-full border from-white to-red-100/90 px-2 py-0.5 text-xs font-medium text-[#2D3648] transition-shadow hover:shadow-inner dark:from-gray-900 dark:to-red-800/30 dark:text-gray-100 dark:hover:shadow-inner dark:hover:shadow-red-800/30" ><svg class="h-3.5 w-3.5 text-red-500" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 12 12"><path fill="currentColor" d="M2.5 2h7a1 1 0 0 1 1 1v6a1 1 0 0 1-1 1h-7a1 1 0 0 1-1-1V3a1 1 0 0 1 1-1Zm0 2v2h3V4h-3Zm4 0v2h3V4h-3Zm-4 3v2h3V7h-3Zm4 0v2h3V7h-3Z"></path></svg>
						<span>Data Studio</span></button></div>
		<div class="flex flex-1 flex-col overflow-hidden -mx-2.5"><div class="flex flex-1 flex-col overflow-hidden"><div class="flex min-h-0 flex-1"><div class="flex flex-1 flex-col overflow-hidden"><div class="md:-mx-2.5 flex min-w-0 flex-wrap border-t"><div class="flex min-w-0 flex-1 flex-wrap"><div class="grid flex-1 grid-cols-1 overflow-hidden text-sm md:grid-cols-2 md:place-content-center sm:mx-2.5"><label class="relative block flex-1 px-3 py-2 hover:bg-gray-50 dark:border-gray-850 dark:hover:bg-gray-950 md:border-r md:border-r-0 hidden" title="default"><span class="text-gray-500">Subset (1)</span>
			<div class="flex items-center whitespace-nowrap"><span class="truncate">default</span>
				
				<svg class="ml-auto min-w-6 pl-2" width="1em" height="1em" viewBox="0 0 12 7" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M1 1L6 6L11 1" stroke="currentColor"></path></svg></div>
			<select class="absolute inset-0 z-10 w-full cursor-pointer border-0 bg-white text-base opacity-0"><optgroup label="Subset (1)"><option value="default" selected>default</option></optgroup></select></label>
		<label class="relative block flex-1 px-3 py-2 hover:bg-gray-50 dark:border-gray-850 dark:hover:bg-gray-900 md:border-r md:border-r" title="train"><div class="text-gray-500">Split (3)</div>
				<div class="flex items-center overflow-hidden whitespace-nowrap"><span class="truncate">train</span>
					
					<svg class="ml-auto min-w-6 pl-2" width="1em" height="1em" viewBox="0 0 12 7" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M1 1L6 6L11 1" stroke="currentColor"></path></svg></div>
				<select class="absolute inset-0 z-10 w-full cursor-pointer border-0 bg-white text-base opacity-0"><optgroup label="Split (3)"><option value="train" selected>train</option><option value="validation" >validation</option><option value="test" >test</option></optgroup></select></label></div></div>
								</div>

							<div class="flex min-h-0 flex-1 flex-col ">


<div class="flex flex-1 flex-col overflow-hidden  border-t"><div class="pt-2 px-2.5 py-1 font-mono text-xs text-gray-600"><details><summary class="mb-1 cursor-pointer hover:text-gray-800 dark:hover:text-gray-400">The full dataset viewer is not available (click to read why). Only showing a preview of the rows.</summary>
			<div class="mb-1 overflow-hidden overflow-x-auto rounded-sm bg-black text-white"><div class="bg-linear-to-r from-gray-800 to-black px-2 py-1 font-sans text-sm">The dataset generation failed because of a cast error</div>
				<pre class="p-4 text-left text-xs" translate="no">Error code:   DatasetGenerationCastError
Exception:    DatasetGenerationCastError
Message:      An error occurred while generating the dataset

All the data files must have the same columns, but at some point there are 9 new columns ({'parameters', 'function_tokens', 'nwo', 'docstring_summary', 'return_statement', 'score', 'function', 'identifier', 'argument_list'}) and 6 missing columns ({'original_string', 'partition', 'repo', 'code', 'code_tokens', 'func_name'}).

This happened while the json dataset builder was generating data using

hf://datasets/semeru/Text-Code-CodeSearchNet-Python/valid.jsonl (at revision 8f65de8a4141f24c7a2099d21cb715cdb2b1b59a)

Please either edit the data files to have matching columns, or separate them into different configurations (see docs at https://hf.co/docs/hub/datasets-manual-configuration#multiple-configurations)
Traceback:    Traceback (most recent call last):
                File "/src/services/worker/.venv/lib/python3.9/site-packages/datasets/builder.py", line 2011, in _prepare_split_single
                  writer.write_table(table)
                File "/src/services/worker/.venv/lib/python3.9/site-packages/datasets/arrow_writer.py", line 585, in write_table
                  pa_table = table_cast(pa_table, self._schema)
                File "/src/services/worker/.venv/lib/python3.9/site-packages/datasets/table.py", line 2302, in table_cast
                  return cast_table_to_schema(table, schema)
                File "/src/services/worker/.venv/lib/python3.9/site-packages/datasets/table.py", line 2256, in cast_table_to_schema
                  raise CastError(
              datasets.table.CastError: Couldn't cast
              url: string
              sha: string
              docstring_summary: string
              language: string
              parameters: string
              return_statement: string
              argument_list: string
              function_tokens: list&lt;item: string>
                child 0, item: string
              function: string
              path: string
              identifier: string
              docstring: string
              docstring_tokens: list&lt;item: string>
                child 0, item: string
              nwo: string
              score: double
              idx: int64
              to
              {'repo': Value(dtype='string', id=None), 'path': Value(dtype='string', id=None), 'func_name': Value(dtype='string', id=None), 'original_string': Value(dtype='string', id=None), 'language': Value(dtype='string', id=None), 'code': Value(dtype='string', id=None), 'code_tokens': Sequence(feature=Value(dtype='string', id=None), length=-1, id=None), 'docstring': Value(dtype='string', id=None), 'docstring_tokens': Sequence(feature=Value(dtype='string', id=None), length=-1, id=None), 'sha': Value(dtype='string', id=None), 'url': Value(dtype='string', id=None), 'partition': Value(dtype='string', id=None), 'idx': Value(dtype='int64', id=None)}
              because column names don't match
              
              During handling of the above exception, another exception occurred:
              
              Traceback (most recent call last):
                File "/src/services/worker/src/worker/job_runners/config/parquet_and_info.py", line 1321, in compute_config_parquet_and_info_response
                  parquet_operations = convert_to_parquet(builder)
                File "/src/services/worker/src/worker/job_runners/config/parquet_and_info.py", line 935, in convert_to_parquet
                  builder.download_and_prepare(
                File "/src/services/worker/.venv/lib/python3.9/site-packages/datasets/builder.py", line 1027, in download_and_prepare
                  self._download_and_prepare(
                File "/src/services/worker/.venv/lib/python3.9/site-packages/datasets/builder.py", line 1122, in _download_and_prepare
                  self._prepare_split(split_generator, **prepare_split_kwargs)
                File "/src/services/worker/.venv/lib/python3.9/site-packages/datasets/builder.py", line 1882, in _prepare_split
                  for job_id, done, content in self._prepare_split_single(
                File "/src/services/worker/.venv/lib/python3.9/site-packages/datasets/builder.py", line 2013, in _prepare_split_single
                  raise DatasetGenerationCastError.from_cast_error(
              datasets.exceptions.DatasetGenerationCastError: An error occurred while generating the dataset
              
              All the data files must have the same columns, but at some point there are 9 new columns ({'parameters', 'function_tokens', 'nwo', 'docstring_summary', 'return_statement', 'score', 'function', 'identifier', 'argument_list'}) and 6 missing columns ({'original_string', 'partition', 'repo', 'code', 'code_tokens', 'func_name'}).
              
              This happened while the json dataset builder was generating data using
              
              hf://datasets/semeru/Text-Code-CodeSearchNet-Python/valid.jsonl (at revision 8f65de8a4141f24c7a2099d21cb715cdb2b1b59a)
              
              Please either edit the data files to have matching columns, or separate them into different configurations (see docs at https://hf.co/docs/hub/datasets-manual-configuration#multiple-configurations)</pre></div>
			<p class="mb-1 font-sans text-sm">Need help to make the dataset viewer work? Make sure to review <a class="underline" href="https://huggingface.co/docs/hub/datasets-data-files-configuration">how to configure the dataset viewer</a>, and <a class="underline" href="/datasets/semeru/Text-Code-CodeSearchNet-Python/discussions/new?title=Dataset+Viewer+issue%3A+DatasetGenerationCastError&amp;description=The+dataset+viewer+is+not+working.%0A%0AError+details%3A%0A%0A%60%60%60%0AError+code%3A+++DatasetGenerationCastError%0AException%3A++++DatasetGenerationCastError%0AMessage%3A++++++An+error+occurred+while+generating+the+dataset%0A%0AAll+the+data+files+must+have+the+same+columns%2C+but+at+some+point+there+are+9+new+columns+%28%7B%27parameters%27%2C+%27function_tokens%27%2C+%27nwo%27%2C+%27docstring_summary%27%2C+%27return_statement%27%2C+%27score%27%2C+%27function%27%2C+%27identifier%27%2C+%27argument_list%27%7D%29+and+6+missing+columns+%28%7B%27original_string%27%2C+%27partition%27%2C+%27repo%27%2C+%27code%27%2C+%27code_tokens%27%2C+%27func_name%27%7D%29.%0A%0AThis+happened+while+the+json+dataset+builder+was+generating+data+using%0A%0Ahf%3A%2F%2Fdatasets%2Fsemeru%2FText-Code-CodeSearchNet-Python%2Fvalid.jsonl+%28at+revision+8f65de8a4141f24c7a2099d21cb715cdb2b1b59a%29%0A%0APlease+either+edit+the+data+files+to+have+matching+columns%2C+or+separate+them+into+different+configurations+%28see+docs+at+https%3A%2F%2Fhf.co%2Fdocs%2Fhub%2Fdatasets-manual-configuration%23multiple-configurations%29%0ATraceback%3A++++Traceback+%28most+recent+call+last%29%3A%0A++++++++++++++++File+%22%2Fsrc%2Fservices%2Fworker%2F.venv%2Flib%2Fpython3.9%2Fsite-packages%2Fdatasets%2Fbuilder.py%22%2C+line+2011%2C+in+_prepare_split_single%0A++++++++++++++++++writer.write_table%28table%29%0A++++++++++++++++File+%22%2Fsrc%2Fservices%2Fworker%2F.venv%2Flib%2Fpython3.9%2Fsite-packages%2Fdatasets%2Farrow_writer.py%22%2C+line+585%2C+in+write_table%0A++++++++++++++++++pa_table+%3D+table_cast%28pa_table%2C+self._schema%29%0A++++++++++++++++File+%22%2Fsrc%2Fservices%2Fworker%2F.venv%2Flib%2Fpython3.9%2Fsite-packages%2Fdatasets%2Ftable.py%22%2C+line+2302%2C+in+table_cast%0A++++++++++++++++++return+cast_table_to_schema%28table%2C+schema%29%0A++++++++++++++++File+%22%2Fsrc%2Fservices%2Fworker%2F.venv%2Flib%2Fpython3.9%2Fsite-packages%2Fdatasets%2Ftable.py%22%2C+line+2256%2C+in+cast_table_to_schema%0A++++++++++++++++++raise+CastError%28%0A++++++++++++++datasets.table.CastError%3A+Couldn%27t+cast%0A++++++++++++++url%3A+string%0A++++++++++++++sha%3A+string%0A++++++++++++++docstring_summary%3A+string%0A++++++++++++++language%3A+string%0A++++++++++++++parameters%3A+string%0A++++++++++++++return_statement%3A+string%0A++++++++++++++argument_list%3A+string%0A++++++++++++++function_tokens%3A+list%3Citem%3A+string%3E%0A++++++++++++++++child+0%2C+item%3A+string%0A++++++++++++++function%3A+string%0A++++++++++++++path%3A+string%0A++++++++++++++identifier%3A+string%0A++++++++++++++docstring%3A+string%0A++++++++++++++docstring_tokens%3A+list%3Citem%3A+string%3E%0A++++++++++++++++child+0%2C+item%3A+string%0A++++++++++++++nwo%3A+string%0A++++++++++++++score%3A+double%0A++++++++++++++idx%3A+int64%0A++++++++++++++to%0A++++++++++++++%7B%27repo%27%3A+Value%28dtype%3D%27string%27%2C+id%3DNone%29%2C+%27path%27%3A+Value%28dtype%3D%27string%27%2C+id%3DNone%29%2C+%27func_name%27%3A+Value%28dtype%3D%27string%27%2C+id%3DNone%29%2C+%27original_string%27%3A+Value%28dtype%3D%27string%27%2C+id%3DNone%29%2C+%27language%27%3A+Value%28dtype%3D%27string%27%2C+id%3DNone%29%2C+%27code%27%3A+Value%28dtype%3D%27string%27%2C+id%3DNone%29%2C+%27code_tokens%27%3A+Sequence%28feature%3DValue%28dtype%3D%27string%27%2C+id%3DNone%29%2C+length%3D-1%2C+id%3DNone%29%2C+%27docstring%27%3A+Value%28dtype%3D%27string%27%2C+id%3DNone%29%2C+%27docstring_tokens%27%3A+Sequence%28feature%3DValue%28dtype%3D%27string%27%2C+id%3DNone%29%2C+length%3D-1%2C+id%3DNone%29%2C+%27sha%27%3A+Value%28dtype%3D%27string%27%2C+id%3DNone%29%2C+%27url%27%3A+Value%28dtype%3D%27string%27%2C+id%3DNone%29%2C+%27partition%27%3A+Value%28dtype%3D%27string%27%2C+id%3DNone%29%2C+%27idx%27%3A+Value%28dtype%3D%27int64%27%2C+id%3DNone%29%7D%0A++++++++++++++because+column+names+don%27t+match%0A++++++++++++++%0A++++++++++++++During+handling+of+the+above+exception%2C+another+exception+occurred%3A%0A++++++++++++++%0A++++++++++++++Traceback+%28most+recent+call+last%29%3A%0A++++++++++++++++File+%22%2Fsrc%2Fservices%2Fworker%2Fsrc%2Fworker%2Fjob_runners%2Fconfig%2Fparquet_and_info.py%22%2C+line+1321%2C+in+compute_config_parquet_and_info_response%0A++++++++++++++++++parquet_operations+%3D+convert_to_parquet%28builder%29%0A++++++++++++++++File+%22%2Fsrc%2Fservices%2Fworker%2Fsrc%2Fworker%2Fjob_runners%2Fconfig%2Fparquet_and_info.py%22%2C+line+935%2C+in+convert_to_parquet%0A++++++++++++++++++builder.download_and_prepare%28%0A++++++++++++++++File+%22%2Fsrc%2Fservices%2Fworker%2F.venv%2Flib%2Fpython3.9%2Fsite-packages%2Fdatasets%2Fbuilder.py%22%2C+line+1027%2C+in+download_and_prepare%0A++++++++++++++++++self._download_and_prepare%28%0A++++++++++++++++File+%22%2Fsrc%2Fservices%2Fworker%2F.venv%2Flib%2Fpython3.9%2Fsite-packages%2Fdatasets%2Fbuilder.py%22%2C+line+1122%2C+in+_download_and_prepare%0A++++++++++++++++++self._prepare_split%28split_generator%2C+**prepare_split_kwargs%29%0A++++++++++++++++File+%22%2Fsrc%2Fservices%2Fworker%2F.venv%2Flib%2Fpython3.9%2Fsite-packages%2Fdatasets%2Fbuilder.py%22%2C+line+1882%2C+in+_prepare_split%0A++++++++++++++++++for+job_id%2C+done%2C+content+in+self._prepare_split_single%28%0A++++++++++++++++File+%22%2Fsrc%2Fservices%2Fworker%2F.venv%2Flib%2Fpython3.9%2Fsite-packages%2Fdatasets%2Fbuilder.py%22%2C+line+2013%2C+in+_prepare_split_single%0A++++++++++++++++++raise+DatasetGenerationCastError.from_cast_error%28%0A++++++++++++++datasets.exceptions.DatasetGenerationCastError%3A+An+error+occurred+while+generating+the+dataset%0A++++++++++++++%0A++++++++++++++All+the+data+files+must+have+the+same+columns%2C+but+at+some+point+there+are+9+new+columns+%28%7B%27parameters%27%2C+%27function_tokens%27%2C+%27nwo%27%2C+%27docstring_summary%27%2C+%27return_statement%27%2C+%27score%27%2C+%27function%27%2C+%27identifier%27%2C+%27argument_list%27%7D%29+and+6+missing+columns+%28%7B%27original_string%27%2C+%27partition%27%2C+%27repo%27%2C+%27code%27%2C+%27code_tokens%27%2C+%27func_name%27%7D%29.%0A++++++++++++++%0A++++++++++++++This+happened+while+the+json+dataset+builder+was+generating+data+using%0A++++++++++++++%0A++++++++++++++hf%3A%2F%2Fdatasets%2Fsemeru%2FText-Code-CodeSearchNet-Python%2Fvalid.jsonl+%28at+revision+8f65de8a4141f24c7a2099d21cb715cdb2b1b59a%29%0A++++++++++++++%0A++++++++++++++Please+either+edit+the+data+files+to+have+matching+columns%2C+or+separate+them+into+different+configurations+%28see+docs+at+https%3A%2F%2Fhf.co%2Fdocs%2Fhub%2Fdatasets-manual-configuration%23multiple-configurations%29%0A%60%60%60%0A%0A%0Acc+%40albertvillanova+%40lhoestq+%40severo.">open a discussion</a>
					for direct support.
				</p></details></div>
		

<div class="max-h-96 border-t relative overflow-auto"><table class="w-full table-auto rounded-lg font-mono text-xs text-gray-900"><thead class="shadow-xs sticky left-0 right-0 top-0 z-1 bg-white align-top"><tr class="space-y-54 h-full min-w-fit divide-x border-b text-left"><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">repo
				</div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span></div></div>

		<div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">path
				</div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span></div></div>

		<div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">func_name
				</div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span></div></div>

		<div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">original_string
				</div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span></div></div>

		<div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">language
				</div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span></div></div>

		<div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">code
				</div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span></div></div>

		<div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">code_tokens
				</div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>sequence</span></div></div>

		<div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">docstring
				</div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span></div></div>

		<div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">docstring_tokens
				</div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>sequence</span></div></div>

		<div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">sha
				</div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span></div></div>

		<div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">url
				</div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span></div></div>

		<div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">partition
				</div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>string</span></div></div>

		<div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th><th class="h-full max-w-sm p-2 text-left  relative w-auto"><div class="flex h-full flex-col flex-nowrap justify-between"><div><div class="flex items-center justify-between">idx
				</div>

			<div class="mb-2 whitespace-nowrap text-xs font-normal text-gray-500"><span>int64</span></div></div>

		<div></div></div>
	<div class="absolute right-0 top-0 z-10 h-full w-1 cursor-col-resize hover:bg-indigo-100 active:bg-indigo-500 dark:hover:bg-indigo-800 dark:active:bg-indigo-600/80"><div class="absolute right-0 top-0 h-full w-1"></div>
								</div>
						</th></tr></thead>
			<tbody class="h-16 overflow-scroll"><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="0"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">phylotoast/util.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">split_phylogeny</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def split_phylogeny(p, level="s"):
    """
    Return either the full or truncated version of a QIIME-formatted taxonomy string.

    :type p: str
    :param p: A QIIME-formatted taxonomy string: k__Foo; p__Bar; ...

    :type level: str
    :param level: The different level of identification are kingdom (k), phylum (p),
                  class (c),order (o), family (f), genus (g) and species (s). If level is
                  not provided, the default level of identification is species.

    :rtype: str
    :return: A QIIME-formatted taxonomy string up to the classification given
            by param level.
    """
    level = level+"__"
    result = p.split(level)
    return result[0]+level+result[1].split(";")[0]</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def split_phylogeny(p, level="s"):
    """
    Return either the full or truncated version of a QIIME-formatted taxonomy string.

    :type p: str
    :param p: A QIIME-formatted taxonomy string: k__Foo; p__Bar; ...

    :type level: str
    :param level: The different level of identification are kingdom (k), phylum (p),
                  class (c),order (o), family (f), genus (g) and species (s). If level is
                  not provided, the default level of identification is species.

    :rtype: str
    :return: A QIIME-formatted taxonomy string up to the classification given
            by param level.
    """
    level = level+"__"
    result = p.split(level)
    return result[0]+level+result[1].split(";")[0]</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "split_phylogeny",
  "(",
  "p",
  ",",
  "level",
  "=",
  "\"s\"",
  ")",
  ":",
  "level",
  "=",
  "level",
  "+",
  "\"__\"",
  "result",
  "=",
  "p",
  ".",
  "split",
  "(",
  "level",
  ")",
  "return",
  "result",
  "[",
  "0",
  "]",
  "+",
  "level",
  "+",
  "result",
  "[",
  "1",
  "]",
  ".",
  "split",
  "(",
  "\";\"",
  ")",
  "[",
  "0",
  "]"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Return either the full or truncated version of a QIIME-formatted taxonomy string.

    :type p: str
    :param p: A QIIME-formatted taxonomy string: k__Foo; p__Bar; ...

    :type level: str
    :param level: The different level of identification are kingdom (k), phylum (p),
                  class (c),order (o), family (f), genus (g) and species (s). If level is
                  not provided, the default level of identification is species.

    :rtype: str
    :return: A QIIME-formatted taxonomy string up to the classification given
            by param level.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Return",
  "either",
  "the",
  "full",
  "or",
  "truncated",
  "version",
  "of",
  "a",
  "QIIME",
  "-",
  "formatted",
  "taxonomy",
  "string",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/util.py#L159-L177</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">0</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="1"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">phylotoast/util.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ensure_dir</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def ensure_dir(d):
    """
    Check to make sure the supplied directory path does not exist, if so, create it. The
    method catches OSError exceptions and returns a descriptive message instead of
    re-raising the error.

    :type d: str
    :param d: It is the full path to a directory.

    :return: Does not return anything, but creates a directory path if it doesn't exist
             already.
    """
    if not os.path.exists(d):
        try:
            os.makedirs(d)
        except OSError as oe:
            # should not happen with os.makedirs
            # ENOENT: No such file or directory
            if os.errno == errno.ENOENT:
                msg = twdd("""One or more directories in the path ({}) do not exist. If
                           you are specifying a new directory for output, please ensure
                           all other directories in the path currently exist.""")
                return msg.format(d)
            else:
                msg = twdd("""An error occurred trying to create the output directory
                           ({}) with message: {}""")
                return msg.format(d, oe.strerror)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def ensure_dir(d):
    """
    Check to make sure the supplied directory path does not exist, if so, create it. The
    method catches OSError exceptions and returns a descriptive message instead of
    re-raising the error.

    :type d: str
    :param d: It is the full path to a directory.

    :return: Does not return anything, but creates a directory path if it doesn't exist
             already.
    """
    if not os.path.exists(d):
        try:
            os.makedirs(d)
        except OSError as oe:
            # should not happen with os.makedirs
            # ENOENT: No such file or directory
            if os.errno == errno.ENOENT:
                msg = twdd("""One or more directories in the path ({}) do not exist. If
                           you are specifying a new directory for output, please ensure
                           all other directories in the path currently exist.""")
                return msg.format(d)
            else:
                msg = twdd("""An error occurred trying to create the output directory
                           ({}) with message: {}""")
                return msg.format(d, oe.strerror)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "ensure_dir",
  "(",
  "d",
  ")",
  ":",
  "if",
  "not",
  "os",
  ".",
  "path",
  ".",
  "exists",
  "(",
  "d",
  ")",
  ":",
  "try",
  ":",
  "os",
  ".",
  "makedirs",
  "(",
  "d",
  ")",
  "except",
  "OSError",
  "as",
  "oe",
  ":",
  "# should not happen with os.makedirs",
  "# ENOENT: No such file or directory",
  "if",
  "os",
  ".",
  "errno",
  "==",
  "errno",
  ".",
  "ENOENT",
  ":",
  "msg",
  "=",
  "twdd",
  "(",
  "\"\"\"One or more directories in the path ({}) do not exist. If\n                           you are specifying a new directory for output, please ensure\n                           all other directories in the path currently exist.\"\"\"",
  ")",
  "return",
  "msg",
  ".",
  "format",
  "(",
  "d",
  ")",
  "else",
  ":",
  "msg",
  "=",
  "twdd",
  "(",
  "\"\"\"An error occurred trying to create the output directory\n                           ({}) with message: {}\"\"\"",
  ")",
  "return",
  "msg",
  ".",
  "format",
  "(",
  "d",
  ",",
  "oe",
  ".",
  "strerror",
  ")"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Check to make sure the supplied directory path does not exist, if so, create it. The
    method catches OSError exceptions and returns a descriptive message instead of
    re-raising the error.

    :type d: str
    :param d: It is the full path to a directory.

    :return: Does not return anything, but creates a directory path if it doesn't exist
             already.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Check",
  "to",
  "make",
  "sure",
  "the",
  "supplied",
  "directory",
  "path",
  "does",
  "not",
  "exist",
  "if",
  "so",
  "create",
  "it",
  ".",
  "The",
  "method",
  "catches",
  "OSError",
  "exceptions",
  "and",
  "returns",
  "a",
  "descriptive",
  "message",
  "instead",
  "of",
  "re",
  "-",
  "raising",
  "the",
  "error",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/util.py#L180-L206</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">1</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="2"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">phylotoast/util.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">file_handle</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def file_handle(fnh, mode="rU"):
    """
    Takes either a file path or an open file handle, checks validity and returns an open
    file handle or raises an appropriate Exception.

    :type fnh: str
    :param fnh: It is the full path to a file, or open file handle

    :type mode: str
    :param mode: The way in which this file will be used, for example to read or write or
                 both. By default, file will be opened in rU mode.

    :return: Returns an opened file for appropriate usage.
    """
    handle = None
    if isinstance(fnh, file):
        if fnh.closed:
            raise ValueError("Input file is closed.")
        handle = fnh
    elif isinstance(fnh, str):
        handle = open(fnh, mode)

    return handle</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def file_handle(fnh, mode="rU"):
    """
    Takes either a file path or an open file handle, checks validity and returns an open
    file handle or raises an appropriate Exception.

    :type fnh: str
    :param fnh: It is the full path to a file, or open file handle

    :type mode: str
    :param mode: The way in which this file will be used, for example to read or write or
                 both. By default, file will be opened in rU mode.

    :return: Returns an opened file for appropriate usage.
    """
    handle = None
    if isinstance(fnh, file):
        if fnh.closed:
            raise ValueError("Input file is closed.")
        handle = fnh
    elif isinstance(fnh, str):
        handle = open(fnh, mode)

    return handle</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "file_handle",
  "(",
  "fnh",
  ",",
  "mode",
  "=",
  "\"rU\"",
  ")",
  ":",
  "handle",
  "=",
  "None",
  "if",
  "isinstance",
  "(",
  "fnh",
  ",",
  "file",
  ")",
  ":",
  "if",
  "fnh",
  ".",
  "closed",
  ":",
  "raise",
  "ValueError",
  "(",
  "\"Input file is closed.\"",
  ")",
  "handle",
  "=",
  "fnh",
  "elif",
  "isinstance",
  "(",
  "fnh",
  ",",
  "str",
  ")",
  ":",
  "handle",
  "=",
  "open",
  "(",
  "fnh",
  ",",
  "mode",
  ")",
  "return",
  "handle"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Takes either a file path or an open file handle, checks validity and returns an open
    file handle or raises an appropriate Exception.

    :type fnh: str
    :param fnh: It is the full path to a file, or open file handle

    :type mode: str
    :param mode: The way in which this file will be used, for example to read or write or
                 both. By default, file will be opened in rU mode.

    :return: Returns an opened file for appropriate usage.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Takes",
  "either",
  "a",
  "file",
  "path",
  "or",
  "an",
  "open",
  "file",
  "handle",
  "checks",
  "validity",
  "and",
  "returns",
  "an",
  "open",
  "file",
  "handle",
  "or",
  "raises",
  "an",
  "appropriate",
  "Exception",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/util.py#L209-L231</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">2</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="3"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">phylotoast/util.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">gather_categories</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def gather_categories(imap, header, categories=None):
    """
    Find the user specified categories in the map and create a dictionary to contain the
    relevant data for each type within the categories. Multiple categories will have their
    types combined such that each possible combination will have its own entry in the
    dictionary.

    :type imap: dict
    :param imap: The input mapping file data keyed by SampleID
    :type header: list
    :param header: The header line from the input mapping file. This will be searched for
                   the user-specified categories
    :type categories: list
    :param categories: The list of user-specified category column name from mapping file
    :rtype: dict
    :return: A sorted dictionary keyed on the combinations of all the types found within
             the user-specified categories. Each entry will contain an empty DataCategory
             namedtuple. If no categories are specified, a single entry with the key
             'default' will be returned
    """
    # If no categories provided, return all SampleIDs
    if categories is None:
        return {"default": DataCategory(set(imap.keys()), {})}

    cat_ids = [header.index(cat)
               for cat in categories if cat in header and "=" not in cat]

    table = OrderedDict()
    conditions = defaultdict(set)
    for i, cat in enumerate(categories):
        if "=" in cat and cat.split("=")[0] in header:
            cat_name = header[header.index(cat.split("=")[0])]
            conditions[cat_name].add(cat.split("=")[1])

    # If invalid categories or conditions identified, return all SampleIDs
    if not cat_ids and not conditions:
        return {"default": DataCategory(set(imap.keys()), {})}

    #If only category column given, return column-wise SampleIDs
    if cat_ids and not conditions:
        for sid, row in imap.items():
            cat_name = "_".join([row[cid] for cid in cat_ids])
            if cat_name not in table:
                table[cat_name] = DataCategory(set(), {})
            table[cat_name].sids.add(sid)
        return table

    # Collect all condition names
    cond_ids = set()
    for k in conditions:
        try:
            cond_ids.add(header.index(k))
        except ValueError:
            continue
    idx_to_test = set(cat_ids).union(cond_ids)

    # If column name and condition given, return overlapping SampleIDs of column and
    # condition combinations
    for sid, row in imap.items():
        if all([row[header.index(c)] in conditions[c] for c in conditions]):
            key = "_".join([row[idx] for idx in idx_to_test])
            try:
                assert key in table.keys()
            except AssertionError:
                table[key] = DataCategory(set(), {})
            table[key].sids.add(sid)
    try:
        assert len(table) > 0
    except AssertionError:
        return {"default": DataCategory(set(imap.keys()), {})}
    else:
        return table</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def gather_categories(imap, header, categories=None):
    """
    Find the user specified categories in the map and create a dictionary to contain the
    relevant data for each type within the categories. Multiple categories will have their
    types combined such that each possible combination will have its own entry in the
    dictionary.

    :type imap: dict
    :param imap: The input mapping file data keyed by SampleID
    :type header: list
    :param header: The header line from the input mapping file. This will be searched for
                   the user-specified categories
    :type categories: list
    :param categories: The list of user-specified category column name from mapping file
    :rtype: dict
    :return: A sorted dictionary keyed on the combinations of all the types found within
             the user-specified categories. Each entry will contain an empty DataCategory
             namedtuple. If no categories are specified, a single entry with the key
             'default' will be returned
    """
    # If no categories provided, return all SampleIDs
    if categories is None:
        return {"default": DataCategory(set(imap.keys()), {})}

    cat_ids = [header.index(cat)
               for cat in categories if cat in header and "=" not in cat]

    table = OrderedDict()
    conditions = defaultdict(set)
    for i, cat in enumerate(categories):
        if "=" in cat and cat.split("=")[0] in header:
            cat_name = header[header.index(cat.split("=")[0])]
            conditions[cat_name].add(cat.split("=")[1])

    # If invalid categories or conditions identified, return all SampleIDs
    if not cat_ids and not conditions:
        return {"default": DataCategory(set(imap.keys()), {})}

    #If only category column given, return column-wise SampleIDs
    if cat_ids and not conditions:
        for sid, row in imap.items():
            cat_name = "_".join([row[cid] for cid in cat_ids])
            if cat_name not in table:
                table[cat_name] = DataCategory(set(), {})
            table[cat_name].sids.add(sid)
        return table

    # Collect all condition names
    cond_ids = set()
    for k in conditions:
        try:
            cond_ids.add(header.index(k))
        except ValueError:
            continue
    idx_to_test = set(cat_ids).union(cond_ids)

    # If column name and condition given, return overlapping SampleIDs of column and
    # condition combinations
    for sid, row in imap.items():
        if all([row[header.index(c)] in conditions[c] for c in conditions]):
            key = "_".join([row[idx] for idx in idx_to_test])
            try:
                assert key in table.keys()
            except AssertionError:
                table[key] = DataCategory(set(), {})
            table[key].sids.add(sid)
    try:
        assert len(table) > 0
    except AssertionError:
        return {"default": DataCategory(set(imap.keys()), {})}
    else:
        return table</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "gather_categories",
  "(",
  "imap",
  ",",
  "header",
  ",",
  "categories",
  "=",
  "None",
  ")",
  ":",
  "# If no categories provided, return all SampleIDs",
  "if",
  "categories",
  "is",
  "None",
  ":",
  "return",
  "{",
  "\"default\"",
  ":",
  "DataCategory",
  "(",
  "set",
  "(",
  "imap",
  ".",
  "keys",
  "(",
  ")",
  ")",
  ",",
  "{",
  "}",
  ")",
  "}",
  "cat_ids",
  "=",
  "[",
  "header",
  ".",
  "index",
  "(",
  "cat",
  ")",
  "for",
  "cat",
  "in",
  "categories",
  "if",
  "cat",
  "in",
  "header",
  "and",
  "\"=\"",
  "not",
  "in",
  "cat",
  "]",
  "table",
  "=",
  "OrderedDict",
  "(",
  ")",
  "conditions",
  "=",
  "defaultdict",
  "(",
  "set",
  ")",
  "for",
  "i",
  ",",
  "cat",
  "in",
  "enumerate",
  "(",
  "categories",
  ")",
  ":",
  "if",
  "\"=\"",
  "in",
  "cat",
  "and",
  "cat",
  ".",
  "split",
  "(",
  "\"=\"",
  ")",
  "[",
  "0",
  "]",
  "in",
  "header",
  ":",
  "cat_name",
  "=",
  "header",
  "[",
  "header",
  ".",
  "index",
  "(",
  "cat",
  ".",
  "split",
  "(",
  "\"=\"",
  ")",
  "[",
  "0",
  "]",
  ")",
  "]",
  "conditions",
  "[",
  "cat_name",
  "]",
  ".",
  "add",
  "(",
  "cat",
  ".",
  "split",
  "(",
  "\"=\"",
  ")",
  "[",
  "1",
  "]",
  ")",
  "# If invalid categories or conditions identified, return all SampleIDs",
  "if",
  "not",
  "cat_ids",
  "and",
  "not",
  "conditions",
  ":",
  "return",
  "{",
  "\"default\"",
  ":",
  "DataCategory",
  "(",
  "set",
  "(",
  "imap",
  ".",
  "keys",
  "(",
  ")",
  ")",
  ",",
  "{",
  "}",
  ")",
  "}",
  "#If only category column given, return column-wise SampleIDs",
  "if",
  "cat_ids",
  "and",
  "not",
  "conditions",
  ":",
  "for",
  "sid",
  ",",
  "row",
  "in",
  "imap",
  ".",
  "items",
  "(",
  ")",
  ":",
  "cat_name",
  "=",
  "\"_\"",
  ".",
  "join",
  "(",
  "[",
  "row",
  "[",
  "cid",
  "]",
  "for",
  "cid",
  "in",
  "cat_ids",
  "]",
  ")",
  "if",
  "cat_name",
  "not",
  "in",
  "table",
  ":",
  "table",
  "[",
  "cat_name",
  "]",
  "=",
  "DataCategory",
  "(",
  "set",
  "(",
  ")",
  ",",
  "{",
  "}",
  ")",
  "table",
  "[",
  "cat_name",
  "]",
  ".",
  "sids",
  ".",
  "add",
  "(",
  "sid",
  ")",
  "return",
  "table",
  "# Collect all condition names",
  "cond_ids",
  "=",
  "set",
  "(",
  ")",
  "for",
  "k",
  "in",
  "conditions",
  ":",
  "try",
  ":",
  "cond_ids",
  ".",
  "add",
  "(",
  "header",
  ".",
  "index",
  "(",
  "k",
  ")",
  ")",
  "except",
  "ValueError",
  ":",
  "continue",
  "idx_to_test",
  "=",
  "set",
  "(",
  "cat_ids",
  ")",
  ".",
  "union",
  "(",
  "cond_ids",
  ")",
  "# If column name and condition given, return overlapping SampleIDs of column and",
  "# condition combinations",
  "for",
  "sid",
  ",",
  "row",
  "in",
  "imap",
  ".",
  "items",
  "(",
  ")",
  ":",
  "if",
  "all",
  "(",
  "[",
  "row",
  "[",
  "header",
  ".",
  "index",
  "(",
  "c",
  ")",
  "]",
  "in",
  "conditions",
  "[",
  "c",
  "]",
  "for",
  "c",
  "in",
  "conditions",
  "]",
  ")",
  ":",
  "key",
  "=",
  "\"_\"",
  ".",
  "join",
  "(",
  "[",
  "row",
  "[",
  "idx",
  "]",
  "for",
  "idx",
  "in",
  "idx_to_test",
  "]",
  ")",
  "try",
  ":",
  "assert",
  "key",
  "in",
  "table",
  ".",
  "keys",
  "(",
  ")",
  "except",
  "AssertionError",
  ":",
  "table",
  "[",
  "key",
  "]",
  "=",
  "DataCategory",
  "(",
  "set",
  "(",
  ")",
  ",",
  "{",
  "}",
  ")",
  "table",
  "[",
  "key",
  "]",
  ".",
  "sids",
  ".",
  "add",
  "(",
  "sid",
  ")",
  "try",
  ":",
  "assert",
  "len",
  "(",
  "table",
  ")",
  ">",
  "0",
  "except",
  "AssertionError",
  ":",
  "return",
  "{",
  "\"default\"",
  ":",
  "DataCategory",
  "(",
  "set",
  "(",
  "imap",
  ".",
  "keys",
  "(",
  ")",
  ")",
  ",",
  "{",
  "}",
  ")",
  "}",
  "else",
  ":",
  "return",
  "table"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Find the user specified categories in the map and create a dictionary to contain the
    relevant data for each type within the categories. Multiple categories will have their
    types combined such that each possible combination will have its own entry in the
    dictionary.

    :type imap: dict
    :param imap: The input mapping file data keyed by SampleID
    :type header: list
    :param header: The header line from the input mapping file. This will be searched for
                   the user-specified categories
    :type categories: list
    :param categories: The list of user-specified category column name from mapping file
    :rtype: dict
    :return: A sorted dictionary keyed on the combinations of all the types found within
             the user-specified categories. Each entry will contain an empty DataCategory
             namedtuple. If no categories are specified, a single entry with the key
             'default' will be returned</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Find",
  "the",
  "user",
  "specified",
  "categories",
  "in",
  "the",
  "map",
  "and",
  "create",
  "a",
  "dictionary",
  "to",
  "contain",
  "the",
  "relevant",
  "data",
  "for",
  "each",
  "type",
  "within",
  "the",
  "categories",
  ".",
  "Multiple",
  "categories",
  "will",
  "have",
  "their",
  "types",
  "combined",
  "such",
  "that",
  "each",
  "possible",
  "combination",
  "will",
  "have",
  "its",
  "own",
  "entry",
  "in",
  "the",
  "dictionary",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/util.py#L238-L309</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">3</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="4"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">phylotoast/util.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">parse_unifrac</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def parse_unifrac(unifracFN):
    """
    Parses the unifrac results file into a dictionary

    :type unifracFN: str
    :param unifracFN: The path to the unifrac results file

    :rtype: dict
    :return: A dictionary with keys: 'pcd' (principle coordinates data) which is a
             dictionary of the data keyed by sample ID, 'eigvals' (eigenvalues), and
             'varexp' (variation explained)
    """
    with open(unifracFN, "rU") as uF:
        first = uF.next().split("\t")
        lines = [line.strip() for line in uF]

    unifrac = {"pcd": OrderedDict(), "eigvals": [], "varexp": []}
    if first[0] == "pc vector number":
        return parse_unifrac_v1_8(unifrac, lines)
    elif first[0] == "Eigvals":
        return parse_unifrac_v1_9(unifrac, lines)
    else:
        raise ValueError("File format not supported/recognized. Please check input "
                         "unifrac file.")</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def parse_unifrac(unifracFN):
    """
    Parses the unifrac results file into a dictionary

    :type unifracFN: str
    :param unifracFN: The path to the unifrac results file

    :rtype: dict
    :return: A dictionary with keys: 'pcd' (principle coordinates data) which is a
             dictionary of the data keyed by sample ID, 'eigvals' (eigenvalues), and
             'varexp' (variation explained)
    """
    with open(unifracFN, "rU") as uF:
        first = uF.next().split("\t")
        lines = [line.strip() for line in uF]

    unifrac = {"pcd": OrderedDict(), "eigvals": [], "varexp": []}
    if first[0] == "pc vector number":
        return parse_unifrac_v1_8(unifrac, lines)
    elif first[0] == "Eigvals":
        return parse_unifrac_v1_9(unifrac, lines)
    else:
        raise ValueError("File format not supported/recognized. Please check input "
                         "unifrac file.")</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "parse_unifrac",
  "(",
  "unifracFN",
  ")",
  ":",
  "with",
  "open",
  "(",
  "unifracFN",
  ",",
  "\"rU\"",
  ")",
  "as",
  "uF",
  ":",
  "first",
  "=",
  "uF",
  ".",
  "next",
  "(",
  ")",
  ".",
  "split",
  "(",
  "\"\\t\"",
  ")",
  "lines",
  "=",
  "[",
  "line",
  ".",
  "strip",
  "(",
  ")",
  "for",
  "line",
  "in",
  "uF",
  "]",
  "unifrac",
  "=",
  "{",
  "\"pcd\"",
  ":",
  "OrderedDict",
  "(",
  ")",
  ",",
  "\"eigvals\"",
  ":",
  "[",
  "]",
  ",",
  "\"varexp\"",
  ":",
  "[",
  "]",
  "}",
  "if",
  "first",
  "[",
  "0",
  "]",
  "==",
  "\"pc vector number\"",
  ":",
  "return",
  "parse_unifrac_v1_8",
  "(",
  "unifrac",
  ",",
  "lines",
  ")",
  "elif",
  "first",
  "[",
  "0",
  "]",
  "==",
  "\"Eigvals\"",
  ":",
  "return",
  "parse_unifrac_v1_9",
  "(",
  "unifrac",
  ",",
  "lines",
  ")",
  "else",
  ":",
  "raise",
  "ValueError",
  "(",
  "\"File format not supported/recognized. Please check input \"",
  "\"unifrac file.\"",
  ")"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Parses the unifrac results file into a dictionary

    :type unifracFN: str
    :param unifracFN: The path to the unifrac results file

    :rtype: dict
    :return: A dictionary with keys: 'pcd' (principle coordinates data) which is a
             dictionary of the data keyed by sample ID, 'eigvals' (eigenvalues), and
             'varexp' (variation explained)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Parses",
  "the",
  "unifrac",
  "results",
  "file",
  "into",
  "a",
  "dictionary"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/util.py#L311-L334</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">4</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="5"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">phylotoast/util.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">parse_unifrac_v1_8</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def parse_unifrac_v1_8(unifrac, file_data):
    """
    Function to parse data from older version of unifrac file obtained from Qiime version
    1.8 and earlier.

    :type unifrac: dict
    :param unifracFN: The path to the unifrac results file

    :type file_data: list
    :param file_data: Unifrac data lines after stripping whitespace characters.
    """
    for line in file_data:
        if line == "":
            break
        line = line.split("\t")
        unifrac["pcd"][line[0]] = [float(e) for e in line[1:]]

    unifrac["eigvals"] = [float(entry) for entry in file_data[-2].split("\t")[1:]]
    unifrac["varexp"] = [float(entry) for entry in file_data[-1].split("\t")[1:]]
    return unifrac</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def parse_unifrac_v1_8(unifrac, file_data):
    """
    Function to parse data from older version of unifrac file obtained from Qiime version
    1.8 and earlier.

    :type unifrac: dict
    :param unifracFN: The path to the unifrac results file

    :type file_data: list
    :param file_data: Unifrac data lines after stripping whitespace characters.
    """
    for line in file_data:
        if line == "":
            break
        line = line.split("\t")
        unifrac["pcd"][line[0]] = [float(e) for e in line[1:]]

    unifrac["eigvals"] = [float(entry) for entry in file_data[-2].split("\t")[1:]]
    unifrac["varexp"] = [float(entry) for entry in file_data[-1].split("\t")[1:]]
    return unifrac</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "parse_unifrac_v1_8",
  "(",
  "unifrac",
  ",",
  "file_data",
  ")",
  ":",
  "for",
  "line",
  "in",
  "file_data",
  ":",
  "if",
  "line",
  "==",
  "\"\"",
  ":",
  "break",
  "line",
  "=",
  "line",
  ".",
  "split",
  "(",
  "\"\\t\"",
  ")",
  "unifrac",
  "[",
  "\"pcd\"",
  "]",
  "[",
  "line",
  "[",
  "0",
  "]",
  "]",
  "=",
  "[",
  "float",
  "(",
  "e",
  ")",
  "for",
  "e",
  "in",
  "line",
  "[",
  "1",
  ":",
  "]",
  "]",
  "unifrac",
  "[",
  "\"eigvals\"",
  "]",
  "=",
  "[",
  "float",
  "(",
  "entry",
  ")",
  "for",
  "entry",
  "in",
  "file_data",
  "[",
  "-",
  "2",
  "]",
  ".",
  "split",
  "(",
  "\"\\t\"",
  ")",
  "[",
  "1",
  ":",
  "]",
  "]",
  "unifrac",
  "[",
  "\"varexp\"",
  "]",
  "=",
  "[",
  "float",
  "(",
  "entry",
  ")",
  "for",
  "entry",
  "in",
  "file_data",
  "[",
  "-",
  "1",
  "]",
  ".",
  "split",
  "(",
  "\"\\t\"",
  ")",
  "[",
  "1",
  ":",
  "]",
  "]",
  "return",
  "unifrac"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Function to parse data from older version of unifrac file obtained from Qiime version
    1.8 and earlier.

    :type unifrac: dict
    :param unifracFN: The path to the unifrac results file

    :type file_data: list
    :param file_data: Unifrac data lines after stripping whitespace characters.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Function",
  "to",
  "parse",
  "data",
  "from",
  "older",
  "version",
  "of",
  "unifrac",
  "file",
  "obtained",
  "from",
  "Qiime",
  "version",
  "1",
  ".",
  "8",
  "and",
  "earlier",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/util.py#L337-L356</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">5</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="6"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">phylotoast/util.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">parse_unifrac_v1_9</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def parse_unifrac_v1_9(unifrac, file_data):
    """
    Function to parse data from newer version of unifrac file obtained from Qiime version
    1.9 and later.

    :type unifracFN: str
    :param unifracFN: The path to the unifrac results file

    :type file_data: list
    :param file_data: Unifrac data lines after stripping whitespace characters.
    """
    unifrac["eigvals"] = [float(entry) for entry in file_data[0].split("\t")]
    unifrac["varexp"] = [float(entry)*100 for entry in file_data[3].split("\t")]

    for line in file_data[8:]:
        if line == "":
            break
        line = line.split("\t")
        unifrac["pcd"][line[0]] = [float(e) for e in line[1:]]
    return unifrac</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def parse_unifrac_v1_9(unifrac, file_data):
    """
    Function to parse data from newer version of unifrac file obtained from Qiime version
    1.9 and later.

    :type unifracFN: str
    :param unifracFN: The path to the unifrac results file

    :type file_data: list
    :param file_data: Unifrac data lines after stripping whitespace characters.
    """
    unifrac["eigvals"] = [float(entry) for entry in file_data[0].split("\t")]
    unifrac["varexp"] = [float(entry)*100 for entry in file_data[3].split("\t")]

    for line in file_data[8:]:
        if line == "":
            break
        line = line.split("\t")
        unifrac["pcd"][line[0]] = [float(e) for e in line[1:]]
    return unifrac</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "parse_unifrac_v1_9",
  "(",
  "unifrac",
  ",",
  "file_data",
  ")",
  ":",
  "unifrac",
  "[",
  "\"eigvals\"",
  "]",
  "=",
  "[",
  "float",
  "(",
  "entry",
  ")",
  "for",
  "entry",
  "in",
  "file_data",
  "[",
  "0",
  "]",
  ".",
  "split",
  "(",
  "\"\\t\"",
  ")",
  "]",
  "unifrac",
  "[",
  "\"varexp\"",
  "]",
  "=",
  "[",
  "float",
  "(",
  "entry",
  ")",
  "*",
  "100",
  "for",
  "entry",
  "in",
  "file_data",
  "[",
  "3",
  "]",
  ".",
  "split",
  "(",
  "\"\\t\"",
  ")",
  "]",
  "for",
  "line",
  "in",
  "file_data",
  "[",
  "8",
  ":",
  "]",
  ":",
  "if",
  "line",
  "==",
  "\"\"",
  ":",
  "break",
  "line",
  "=",
  "line",
  ".",
  "split",
  "(",
  "\"\\t\"",
  ")",
  "unifrac",
  "[",
  "\"pcd\"",
  "]",
  "[",
  "line",
  "[",
  "0",
  "]",
  "]",
  "=",
  "[",
  "float",
  "(",
  "e",
  ")",
  "for",
  "e",
  "in",
  "line",
  "[",
  "1",
  ":",
  "]",
  "]",
  "return",
  "unifrac"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Function to parse data from newer version of unifrac file obtained from Qiime version
    1.9 and later.

    :type unifracFN: str
    :param unifracFN: The path to the unifrac results file

    :type file_data: list
    :param file_data: Unifrac data lines after stripping whitespace characters.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Function",
  "to",
  "parse",
  "data",
  "from",
  "newer",
  "version",
  "of",
  "unifrac",
  "file",
  "obtained",
  "from",
  "Qiime",
  "version",
  "1",
  ".",
  "9",
  "and",
  "later",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/util.py#L359-L378</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">6</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="7"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">phylotoast/util.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">color_mapping</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def color_mapping(sample_map, header, group_column, color_column=None):
    """
    Determine color-category mapping. If color_column was specified, then map the category
    names to color values. Otherwise, use the palettable colors to automatically generate
    a set of colors for the group values.

    :type sample_map: dict
    :param unifracFN: Map associating each line of the mapping file with the appropriate
                      sample ID (each value of the map also contains the sample ID)

    :type header: tuple
    :param A tuple of header line for mapping file

    :type group_column: str
    :param group_column: String denoting the column name for sample groups.

    :type color_column: str
    :param color_column: String denoting the column name for sample colors.

    :type return: dict
    :param return: {SampleID: Color}
    """
    group_colors = OrderedDict()
    group_gather = gather_categories(sample_map, header, [group_column])

    if color_column is not None:
        color_gather = gather_categories(sample_map, header, [color_column])
        # match sample IDs between color_gather and group_gather
        for group in group_gather:
            for color in color_gather:
                # allow incomplete assignment of colors, if group sids overlap at
                # all with the color sids, consider it a match
                if group_gather[group].sids.intersection(color_gather[color].sids):
                    group_colors[group] = color
    else:
        bcolors = itertools.cycle(Set3_12.hex_colors)
        for group in group_gather:
            group_colors[group] = bcolors.next()

    return group_colors</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def color_mapping(sample_map, header, group_column, color_column=None):
    """
    Determine color-category mapping. If color_column was specified, then map the category
    names to color values. Otherwise, use the palettable colors to automatically generate
    a set of colors for the group values.

    :type sample_map: dict
    :param unifracFN: Map associating each line of the mapping file with the appropriate
                      sample ID (each value of the map also contains the sample ID)

    :type header: tuple
    :param A tuple of header line for mapping file

    :type group_column: str
    :param group_column: String denoting the column name for sample groups.

    :type color_column: str
    :param color_column: String denoting the column name for sample colors.

    :type return: dict
    :param return: {SampleID: Color}
    """
    group_colors = OrderedDict()
    group_gather = gather_categories(sample_map, header, [group_column])

    if color_column is not None:
        color_gather = gather_categories(sample_map, header, [color_column])
        # match sample IDs between color_gather and group_gather
        for group in group_gather:
            for color in color_gather:
                # allow incomplete assignment of colors, if group sids overlap at
                # all with the color sids, consider it a match
                if group_gather[group].sids.intersection(color_gather[color].sids):
                    group_colors[group] = color
    else:
        bcolors = itertools.cycle(Set3_12.hex_colors)
        for group in group_gather:
            group_colors[group] = bcolors.next()

    return group_colors</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "color_mapping",
  "(",
  "sample_map",
  ",",
  "header",
  ",",
  "group_column",
  ",",
  "color_column",
  "=",
  "None",
  ")",
  ":",
  "group_colors",
  "=",
  "OrderedDict",
  "(",
  ")",
  "group_gather",
  "=",
  "gather_categories",
  "(",
  "sample_map",
  ",",
  "header",
  ",",
  "[",
  "group_column",
  "]",
  ")",
  "if",
  "color_column",
  "is",
  "not",
  "None",
  ":",
  "color_gather",
  "=",
  "gather_categories",
  "(",
  "sample_map",
  ",",
  "header",
  ",",
  "[",
  "color_column",
  "]",
  ")",
  "# match sample IDs between color_gather and group_gather",
  "for",
  "group",
  "in",
  "group_gather",
  ":",
  "for",
  "color",
  "in",
  "color_gather",
  ":",
  "# allow incomplete assignment of colors, if group sids overlap at",
  "# all with the color sids, consider it a match",
  "if",
  "group_gather",
  "[",
  "group",
  "]",
  ".",
  "sids",
  ".",
  "intersection",
  "(",
  "color_gather",
  "[",
  "color",
  "]",
  ".",
  "sids",
  ")",
  ":",
  "group_colors",
  "[",
  "group",
  "]",
  "=",
  "color",
  "else",
  ":",
  "bcolors",
  "=",
  "itertools",
  ".",
  "cycle",
  "(",
  "Set3_12",
  ".",
  "hex_colors",
  ")",
  "for",
  "group",
  "in",
  "group_gather",
  ":",
  "group_colors",
  "[",
  "group",
  "]",
  "=",
  "bcolors",
  ".",
  "next",
  "(",
  ")",
  "return",
  "group_colors"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Determine color-category mapping. If color_column was specified, then map the category
    names to color values. Otherwise, use the palettable colors to automatically generate
    a set of colors for the group values.

    :type sample_map: dict
    :param unifracFN: Map associating each line of the mapping file with the appropriate
                      sample ID (each value of the map also contains the sample ID)

    :type header: tuple
    :param A tuple of header line for mapping file

    :type group_column: str
    :param group_column: String denoting the column name for sample groups.

    :type color_column: str
    :param color_column: String denoting the column name for sample colors.

    :type return: dict
    :param return: {SampleID: Color}</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Determine",
  "color",
  "-",
  "category",
  "mapping",
  ".",
  "If",
  "color_column",
  "was",
  "specified",
  "then",
  "map",
  "the",
  "category",
  "names",
  "to",
  "color",
  "values",
  ".",
  "Otherwise",
  "use",
  "the",
  "palettable",
  "colors",
  "to",
  "automatically",
  "generate",
  "a",
  "set",
  "of",
  "colors",
  "for",
  "the",
  "group",
  "values",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/util.py#L380-L419</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">7</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="8"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">christophertbrown/bioscripts</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ctbBio/shuffle_genome.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">rev_c</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def rev_c(read):
    """
    return reverse completment of read
    """
    rc = []
    rc_nucs = {'A':'T', 'T':'A', 'G':'C', 'C':'G', 'N':'N'}
    for base in read:
        rc.extend(rc_nucs[base.upper()])
    return rc[::-1]</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def rev_c(read):
    """
    return reverse completment of read
    """
    rc = []
    rc_nucs = {'A':'T', 'T':'A', 'G':'C', 'C':'G', 'N':'N'}
    for base in read:
        rc.extend(rc_nucs[base.upper()])
    return rc[::-1]</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "rev_c",
  "(",
  "read",
  ")",
  ":",
  "rc",
  "=",
  "[",
  "]",
  "rc_nucs",
  "=",
  "{",
  "'A'",
  ":",
  "'T'",
  ",",
  "'T'",
  ":",
  "'A'",
  ",",
  "'G'",
  ":",
  "'C'",
  ",",
  "'C'",
  ":",
  "'G'",
  ",",
  "'N'",
  ":",
  "'N'",
  "}",
  "for",
  "base",
  "in",
  "read",
  ":",
  "rc",
  ".",
  "extend",
  "(",
  "rc_nucs",
  "[",
  "base",
  ".",
  "upper",
  "(",
  ")",
  "]",
  ")",
  "return",
  "rc",
  "[",
  ":",
  ":",
  "-",
  "1",
  "]"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">return reverse completment of read</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "return",
  "reverse",
  "completment",
  "of",
  "read"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">83b2566b3a5745437ec651cd6cafddd056846240</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/shuffle_genome.py#L27-L35</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">8</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="9"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">christophertbrown/bioscripts</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ctbBio/shuffle_genome.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">shuffle_genome</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def shuffle_genome(genome, cat, fraction = float(100), plot = True, \
        alpha = 0.1, beta = 100000, \
        min_length = 1000, max_length = 200000):
    """
    randomly shuffle genome
    """
    header = '>randomized_%s' % (genome.name)
    sequence = list(''.join([i[1] for i in parse_fasta(genome)]))
    length = len(sequence)
    shuffled = []
    # break genome into pieces
    while sequence is not False:
        s = int(random.gammavariate(alpha, beta))
        if s &lt;= min_length or s >= max_length:
            continue
        if len(sequence) &lt; s:
            seq = sequence[0:]
        else:
            seq = sequence[0:s]
        sequence = sequence[s:]
#        if bool(random.getrandbits(1)) is True:
#            seq = rev_c(seq)
#            print('fragment length: %s reverse complement: True' % ('{:,}'.format(s)), file=sys.stderr)
#        else:
#            print('fragment length: %s reverse complement: False' % ('{:,}'.format(s)), file=sys.stderr)
        shuffled.append(''.join(seq))
        if sequence == []:
            break
    # shuffle pieces
    random.shuffle(shuffled)
    # subset fragments
    if fraction == float(100):
        subset = shuffled
    else:
        max_pieces = int(length * fraction/100)
        subset, total = [], 0
        for fragment in shuffled:
            length = len(fragment)
            if total + length &lt;= max_pieces:
                subset.append(fragment)
                total += length
            else:
                diff = max_pieces - total
                subset.append(fragment[0:diff])
                break
    # combine sequences, if requested
    if cat is True:
        yield [header, ''.join(subset)]
    else:
        for i, seq in enumerate(subset):
            yield ['%s fragment:%s' % (header, i), seq]</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def shuffle_genome(genome, cat, fraction = float(100), plot = True, \
        alpha = 0.1, beta = 100000, \
        min_length = 1000, max_length = 200000):
    """
    randomly shuffle genome
    """
    header = '>randomized_%s' % (genome.name)
    sequence = list(''.join([i[1] for i in parse_fasta(genome)]))
    length = len(sequence)
    shuffled = []
    # break genome into pieces
    while sequence is not False:
        s = int(random.gammavariate(alpha, beta))
        if s &lt;= min_length or s >= max_length:
            continue
        if len(sequence) &lt; s:
            seq = sequence[0:]
        else:
            seq = sequence[0:s]
        sequence = sequence[s:]
#        if bool(random.getrandbits(1)) is True:
#            seq = rev_c(seq)
#            print('fragment length: %s reverse complement: True' % ('{:,}'.format(s)), file=sys.stderr)
#        else:
#            print('fragment length: %s reverse complement: False' % ('{:,}'.format(s)), file=sys.stderr)
        shuffled.append(''.join(seq))
        if sequence == []:
            break
    # shuffle pieces
    random.shuffle(shuffled)
    # subset fragments
    if fraction == float(100):
        subset = shuffled
    else:
        max_pieces = int(length * fraction/100)
        subset, total = [], 0
        for fragment in shuffled:
            length = len(fragment)
            if total + length &lt;= max_pieces:
                subset.append(fragment)
                total += length
            else:
                diff = max_pieces - total
                subset.append(fragment[0:diff])
                break
    # combine sequences, if requested
    if cat is True:
        yield [header, ''.join(subset)]
    else:
        for i, seq in enumerate(subset):
            yield ['%s fragment:%s' % (header, i), seq]</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "shuffle_genome",
  "(",
  "genome",
  ",",
  "cat",
  ",",
  "fraction",
  "=",
  "float",
  "(",
  "100",
  ")",
  ",",
  "plot",
  "=",
  "True",
  ",",
  "alpha",
  "=",
  "0.1",
  ",",
  "beta",
  "=",
  "100000",
  ",",
  "min_length",
  "=",
  "1000",
  ",",
  "max_length",
  "=",
  "200000",
  ")",
  ":",
  "header",
  "=",
  "'>randomized_%s'",
  "%",
  "(",
  "genome",
  ".",
  "name",
  ")",
  "sequence",
  "=",
  "list",
  "(",
  "''",
  ".",
  "join",
  "(",
  "[",
  "i",
  "[",
  "1",
  "]",
  "for",
  "i",
  "in",
  "parse_fasta",
  "(",
  "genome",
  ")",
  "]",
  ")",
  ")",
  "length",
  "=",
  "len",
  "(",
  "sequence",
  ")",
  "shuffled",
  "=",
  "[",
  "]",
  "# break genome into pieces",
  "while",
  "sequence",
  "is",
  "not",
  "False",
  ":",
  "s",
  "=",
  "int",
  "(",
  "random",
  ".",
  "gammavariate",
  "(",
  "alpha",
  ",",
  "beta",
  ")",
  ")",
  "if",
  "s",
  "&lt;=",
  "min_length",
  "or",
  "s",
  ">=",
  "max_length",
  ":",
  "continue",
  "if",
  "len",
  "(",
  "sequence",
  ")",
  "&lt;",
  "s",
  ":",
  "seq",
  "=",
  "sequence",
  "[",
  "0",
  ":",
  "]",
  "else",
  ":",
  "seq",
  "=",
  "sequence",
  "[",
  "0",
  ":",
  "s",
  "]",
  "sequence",
  "=",
  "sequence",
  "[",
  "s",
  ":",
  "]",
  "#        if bool(random.getrandbits(1)) is True:",
  "#            seq = rev_c(seq)",
  "#            print('fragment length: %s reverse complement: True' % ('{:,}'.format(s)), file=sys.stderr)",
  "#        else:",
  "#            print('fragment length: %s reverse complement: False' % ('{:,}'.format(s)), file=sys.stderr)",
  "shuffled",
  ".",
  "append",
  "(",
  "''",
  ".",
  "join",
  "(",
  "seq",
  ")",
  ")",
  "if",
  "sequence",
  "==",
  "[",
  "]",
  ":",
  "break",
  "# shuffle pieces",
  "random",
  ".",
  "shuffle",
  "(",
  "shuffled",
  ")",
  "# subset fragments",
  "if",
  "fraction",
  "==",
  "float",
  "(",
  "100",
  ")",
  ":",
  "subset",
  "=",
  "shuffled",
  "else",
  ":",
  "max_pieces",
  "=",
  "int",
  "(",
  "length",
  "*",
  "fraction",
  "/",
  "100",
  ")",
  "subset",
  ",",
  "total",
  "=",
  "[",
  "]",
  ",",
  "0",
  "for",
  "fragment",
  "in",
  "shuffled",
  ":",
  "length",
  "=",
  "len",
  "(",
  "fragment",
  ")",
  "if",
  "total",
  "+",
  "length",
  "&lt;=",
  "max_pieces",
  ":",
  "subset",
  ".",
  "append",
  "(",
  "fragment",
  ")",
  "total",
  "+=",
  "length",
  "else",
  ":",
  "diff",
  "=",
  "max_pieces",
  "-",
  "total",
  "subset",
  ".",
  "append",
  "(",
  "fragment",
  "[",
  "0",
  ":",
  "diff",
  "]",
  ")",
  "break",
  "# combine sequences, if requested",
  "if",
  "cat",
  "is",
  "True",
  ":",
  "yield",
  "[",
  "header",
  ",",
  "''",
  ".",
  "join",
  "(",
  "subset",
  ")",
  "]",
  "else",
  ":",
  "for",
  "i",
  ",",
  "seq",
  "in",
  "enumerate",
  "(",
  "subset",
  ")",
  ":",
  "yield",
  "[",
  "'%s fragment:%s'",
  "%",
  "(",
  "header",
  ",",
  "i",
  ")",
  ",",
  "seq",
  "]"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">randomly shuffle genome</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "randomly",
  "shuffle",
  "genome"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">83b2566b3a5745437ec651cd6cafddd056846240</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/shuffle_genome.py#L37-L87</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">9</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="10"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">opengridcc/opengrid</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">opengrid/library/regression.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">MultiVarLinReg._prune</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _prune(self, fit, p_max):
        """
        If the fit contains statistically insignificant parameters, remove them.
        Returns a pruned fit where all parameters have p-values of the t-statistic below p_max

        Parameters
        ----------
        fit: fm.ols fit object
            Can contain insignificant parameters
        p_max : float
            Maximum allowed probability of the t-statistic

        Returns
        -------
        fit: fm.ols fit object
            Won't contain any insignificant parameters

        """

        def remove_from_model_desc(x, model_desc):
            """
            Return a model_desc without x
            """

            rhs_termlist = []
            for t in model_desc.rhs_termlist:
                if not t.factors:
                    # intercept, add anyway
                    rhs_termlist.append(t)
                elif not x == t.factors[0]._varname:
                    # this is not the term with x
                    rhs_termlist.append(t)

            md = ModelDesc(model_desc.lhs_termlist, rhs_termlist)
            return md

        corrected_model_desc = ModelDesc(fit.model.formula.lhs_termlist[:], fit.model.formula.rhs_termlist[:])
        pars_to_prune = fit.pvalues.where(fit.pvalues > p_max).dropna().index.tolist()
        try:
            pars_to_prune.remove('Intercept')
        except:
            pass
        while pars_to_prune:
            corrected_model_desc = remove_from_model_desc(pars_to_prune[0], corrected_model_desc)
            fit = fm.ols(corrected_model_desc, data=self.df).fit()
            pars_to_prune = fit.pvalues.where(fit.pvalues > p_max).dropna().index.tolist()
            try:
                pars_to_prune.remove('Intercept')
            except:
                pass
        return fit</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _prune(self, fit, p_max):
        """
        If the fit contains statistically insignificant parameters, remove them.
        Returns a pruned fit where all parameters have p-values of the t-statistic below p_max

        Parameters
        ----------
        fit: fm.ols fit object
            Can contain insignificant parameters
        p_max : float
            Maximum allowed probability of the t-statistic

        Returns
        -------
        fit: fm.ols fit object
            Won't contain any insignificant parameters

        """

        def remove_from_model_desc(x, model_desc):
            """
            Return a model_desc without x
            """

            rhs_termlist = []
            for t in model_desc.rhs_termlist:
                if not t.factors:
                    # intercept, add anyway
                    rhs_termlist.append(t)
                elif not x == t.factors[0]._varname:
                    # this is not the term with x
                    rhs_termlist.append(t)

            md = ModelDesc(model_desc.lhs_termlist, rhs_termlist)
            return md

        corrected_model_desc = ModelDesc(fit.model.formula.lhs_termlist[:], fit.model.formula.rhs_termlist[:])
        pars_to_prune = fit.pvalues.where(fit.pvalues > p_max).dropna().index.tolist()
        try:
            pars_to_prune.remove('Intercept')
        except:
            pass
        while pars_to_prune:
            corrected_model_desc = remove_from_model_desc(pars_to_prune[0], corrected_model_desc)
            fit = fm.ols(corrected_model_desc, data=self.df).fit()
            pars_to_prune = fit.pvalues.where(fit.pvalues > p_max).dropna().index.tolist()
            try:
                pars_to_prune.remove('Intercept')
            except:
                pass
        return fit</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "_prune",
  "(",
  "self",
  ",",
  "fit",
  ",",
  "p_max",
  ")",
  ":",
  "def",
  "remove_from_model_desc",
  "(",
  "x",
  ",",
  "model_desc",
  ")",
  ":",
  "\"\"\"\n            Return a model_desc without x\n            \"\"\"",
  "rhs_termlist",
  "=",
  "[",
  "]",
  "for",
  "t",
  "in",
  "model_desc",
  ".",
  "rhs_termlist",
  ":",
  "if",
  "not",
  "t",
  ".",
  "factors",
  ":",
  "# intercept, add anyway",
  "rhs_termlist",
  ".",
  "append",
  "(",
  "t",
  ")",
  "elif",
  "not",
  "x",
  "==",
  "t",
  ".",
  "factors",
  "[",
  "0",
  "]",
  ".",
  "_varname",
  ":",
  "# this is not the term with x",
  "rhs_termlist",
  ".",
  "append",
  "(",
  "t",
  ")",
  "md",
  "=",
  "ModelDesc",
  "(",
  "model_desc",
  ".",
  "lhs_termlist",
  ",",
  "rhs_termlist",
  ")",
  "return",
  "md",
  "corrected_model_desc",
  "=",
  "ModelDesc",
  "(",
  "fit",
  ".",
  "model",
  ".",
  "formula",
  ".",
  "lhs_termlist",
  "[",
  ":",
  "]",
  ",",
  "fit",
  ".",
  "model",
  ".",
  "formula",
  ".",
  "rhs_termlist",
  "[",
  ":",
  "]",
  ")",
  "pars_to_prune",
  "=",
  "fit",
  ".",
  "pvalues",
  ".",
  "where",
  "(",
  "fit",
  ".",
  "pvalues",
  ">",
  "p_max",
  ")",
  ".",
  "dropna",
  "(",
  ")",
  ".",
  "index",
  ".",
  "tolist",
  "(",
  ")",
  "try",
  ":",
  "pars_to_prune",
  ".",
  "remove",
  "(",
  "'Intercept'",
  ")",
  "except",
  ":",
  "pass",
  "while",
  "pars_to_prune",
  ":",
  "corrected_model_desc",
  "=",
  "remove_from_model_desc",
  "(",
  "pars_to_prune",
  "[",
  "0",
  "]",
  ",",
  "corrected_model_desc",
  ")",
  "fit",
  "=",
  "fm",
  ".",
  "ols",
  "(",
  "corrected_model_desc",
  ",",
  "data",
  "=",
  "self",
  ".",
  "df",
  ")",
  ".",
  "fit",
  "(",
  ")",
  "pars_to_prune",
  "=",
  "fit",
  ".",
  "pvalues",
  ".",
  "where",
  "(",
  "fit",
  ".",
  "pvalues",
  ">",
  "p_max",
  ")",
  ".",
  "dropna",
  "(",
  ")",
  ".",
  "index",
  ".",
  "tolist",
  "(",
  ")",
  "try",
  ":",
  "pars_to_prune",
  ".",
  "remove",
  "(",
  "'Intercept'",
  ")",
  "except",
  ":",
  "pass",
  "return",
  "fit"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">If the fit contains statistically insignificant parameters, remove them.
        Returns a pruned fit where all parameters have p-values of the t-statistic below p_max

        Parameters
        ----------
        fit: fm.ols fit object
            Can contain insignificant parameters
        p_max : float
            Maximum allowed probability of the t-statistic

        Returns
        -------
        fit: fm.ols fit object
            Won't contain any insignificant parameters</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "If",
  "the",
  "fit",
  "contains",
  "statistically",
  "insignificant",
  "parameters",
  "remove",
  "them",
  ".",
  "Returns",
  "a",
  "pruned",
  "fit",
  "where",
  "all",
  "parameters",
  "have",
  "p",
  "-",
  "values",
  "of",
  "the",
  "t",
  "-",
  "statistic",
  "below",
  "p_max"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">69b8da3c8fcea9300226c45ef0628cd6d4307651</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/opengridcc/opengrid/blob/69b8da3c8fcea9300226c45ef0628cd6d4307651/opengrid/library/regression.py#L222-L272</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">10</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="11"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">opengridcc/opengrid</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">opengrid/library/regression.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">MultiVarLinReg.find_best_rsquared</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def find_best_rsquared(list_of_fits):
        """Return the best fit, based on rsquared"""
        res = sorted(list_of_fits, key=lambda x: x.rsquared)
        return res[-1]</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def find_best_rsquared(list_of_fits):
        """Return the best fit, based on rsquared"""
        res = sorted(list_of_fits, key=lambda x: x.rsquared)
        return res[-1]</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "find_best_rsquared",
  "(",
  "list_of_fits",
  ")",
  ":",
  "res",
  "=",
  "sorted",
  "(",
  "list_of_fits",
  ",",
  "key",
  "=",
  "lambda",
  "x",
  ":",
  "x",
  ".",
  "rsquared",
  ")",
  "return",
  "res",
  "[",
  "-",
  "1",
  "]"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Return the best fit, based on rsquared</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Return",
  "the",
  "best",
  "fit",
  "based",
  "on",
  "rsquared"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">69b8da3c8fcea9300226c45ef0628cd6d4307651</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/opengridcc/opengrid/blob/69b8da3c8fcea9300226c45ef0628cd6d4307651/opengrid/library/regression.py#L275-L278</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">11</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="12"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">opengridcc/opengrid</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">opengrid/library/regression.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">MultiVarLinReg._predict</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _predict(self, fit, df):
        """
        Return a df with predictions and confidence interval

        Notes
        -----
        The df will contain the following columns:
        - 'predicted': the model output
        - 'interval_u', 'interval_l': upper and lower confidence bounds.

        The result will depend on the following attributes of self:
        confint : float (default=0.95)
            Confidence level for two-sided hypothesis
        allow_negative_predictions : bool (default=True)
            If False, correct negative predictions to zero (typically for energy consumption predictions)

        Parameters
        ----------
        fit : Statsmodels fit
        df : pandas DataFrame or None (default)
            If None, use self.df


        Returns
        -------
        df_res : pandas DataFrame
            Copy of df with additional columns 'predicted', 'interval_u' and 'interval_l'
        """

        # Add model results to data as column 'predictions'
        df_res = df.copy()
        if 'Intercept' in fit.model.exog_names:
            df_res['Intercept'] = 1.0
        df_res['predicted'] = fit.predict(df_res)
        if not self.allow_negative_predictions:
            df_res.loc[df_res['predicted'] &lt; 0, 'predicted'] = 0

        prstd, interval_l, interval_u = wls_prediction_std(fit,
                                                           df_res[fit.model.exog_names],
                                                           alpha=1 - self.confint)
        df_res['interval_l'] = interval_l
        df_res['interval_u'] = interval_u

        if 'Intercept' in df_res:
            df_res.drop(labels=['Intercept'], axis=1, inplace=True)

        return df_res</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _predict(self, fit, df):
        """
        Return a df with predictions and confidence interval

        Notes
        -----
        The df will contain the following columns:
        - 'predicted': the model output
        - 'interval_u', 'interval_l': upper and lower confidence bounds.

        The result will depend on the following attributes of self:
        confint : float (default=0.95)
            Confidence level for two-sided hypothesis
        allow_negative_predictions : bool (default=True)
            If False, correct negative predictions to zero (typically for energy consumption predictions)

        Parameters
        ----------
        fit : Statsmodels fit
        df : pandas DataFrame or None (default)
            If None, use self.df


        Returns
        -------
        df_res : pandas DataFrame
            Copy of df with additional columns 'predicted', 'interval_u' and 'interval_l'
        """

        # Add model results to data as column 'predictions'
        df_res = df.copy()
        if 'Intercept' in fit.model.exog_names:
            df_res['Intercept'] = 1.0
        df_res['predicted'] = fit.predict(df_res)
        if not self.allow_negative_predictions:
            df_res.loc[df_res['predicted'] &lt; 0, 'predicted'] = 0

        prstd, interval_l, interval_u = wls_prediction_std(fit,
                                                           df_res[fit.model.exog_names],
                                                           alpha=1 - self.confint)
        df_res['interval_l'] = interval_l
        df_res['interval_u'] = interval_u

        if 'Intercept' in df_res:
            df_res.drop(labels=['Intercept'], axis=1, inplace=True)

        return df_res</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "_predict",
  "(",
  "self",
  ",",
  "fit",
  ",",
  "df",
  ")",
  ":",
  "# Add model results to data as column 'predictions'",
  "df_res",
  "=",
  "df",
  ".",
  "copy",
  "(",
  ")",
  "if",
  "'Intercept'",
  "in",
  "fit",
  ".",
  "model",
  ".",
  "exog_names",
  ":",
  "df_res",
  "[",
  "'Intercept'",
  "]",
  "=",
  "1.0",
  "df_res",
  "[",
  "'predicted'",
  "]",
  "=",
  "fit",
  ".",
  "predict",
  "(",
  "df_res",
  ")",
  "if",
  "not",
  "self",
  ".",
  "allow_negative_predictions",
  ":",
  "df_res",
  ".",
  "loc",
  "[",
  "df_res",
  "[",
  "'predicted'",
  "]",
  "&lt;",
  "0",
  ",",
  "'predicted'",
  "]",
  "=",
  "0",
  "prstd",
  ",",
  "interval_l",
  ",",
  "interval_u",
  "=",
  "wls_prediction_std",
  "(",
  "fit",
  ",",
  "df_res",
  "[",
  "fit",
  ".",
  "model",
  ".",
  "exog_names",
  "]",
  ",",
  "alpha",
  "=",
  "1",
  "-",
  "self",
  ".",
  "confint",
  ")",
  "df_res",
  "[",
  "'interval_l'",
  "]",
  "=",
  "interval_l",
  "df_res",
  "[",
  "'interval_u'",
  "]",
  "=",
  "interval_u",
  "if",
  "'Intercept'",
  "in",
  "df_res",
  ":",
  "df_res",
  ".",
  "drop",
  "(",
  "labels",
  "=",
  "[",
  "'Intercept'",
  "]",
  ",",
  "axis",
  "=",
  "1",
  ",",
  "inplace",
  "=",
  "True",
  ")",
  "return",
  "df_res"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Return a df with predictions and confidence interval

        Notes
        -----
        The df will contain the following columns:
        - 'predicted': the model output
        - 'interval_u', 'interval_l': upper and lower confidence bounds.

        The result will depend on the following attributes of self:
        confint : float (default=0.95)
            Confidence level for two-sided hypothesis
        allow_negative_predictions : bool (default=True)
            If False, correct negative predictions to zero (typically for energy consumption predictions)

        Parameters
        ----------
        fit : Statsmodels fit
        df : pandas DataFrame or None (default)
            If None, use self.df


        Returns
        -------
        df_res : pandas DataFrame
            Copy of df with additional columns 'predicted', 'interval_u' and 'interval_l'</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Return",
  "a",
  "df",
  "with",
  "predictions",
  "and",
  "confidence",
  "interval"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">69b8da3c8fcea9300226c45ef0628cd6d4307651</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/opengridcc/opengrid/blob/69b8da3c8fcea9300226c45ef0628cd6d4307651/opengrid/library/regression.py#L292-L338</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">12</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="13"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">phylotoast/biom_calc.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">relative_abundance</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def relative_abundance(biomf, sampleIDs=None):
    """
    Calculate the relative abundance of each OTUID in a Sample.

    :type biomf: A BIOM file.
    :param biomf: OTU table format.

    :type sampleIDs: list
    :param sampleIDs: A list of sample id's from BIOM format OTU table.

    :rtype: dict
    :return: Returns a keyed on SampleIDs, and the values are dictionaries keyed on
             OTUID's and their values represent the relative abundance of that OTUID in
             that SampleID.
    """
    if sampleIDs is None:
        sampleIDs = biomf.ids()
    else:
        try:
            for sid in sampleIDs:
                assert sid in biomf.ids()
        except AssertionError:
            raise ValueError(
                "\nError while calculating relative abundances: The sampleIDs provided do"
                " not match the sampleIDs in biom file. Please double check the sampleIDs"
                " provided.\n")
    otuIDs = biomf.ids(axis="observation")
    norm_biomf = biomf.norm(inplace=False)

    return {sample: {otuID: norm_biomf.get_value_by_ids(otuID, sample)
                     for otuID in otuIDs} for sample in sampleIDs}</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def relative_abundance(biomf, sampleIDs=None):
    """
    Calculate the relative abundance of each OTUID in a Sample.

    :type biomf: A BIOM file.
    :param biomf: OTU table format.

    :type sampleIDs: list
    :param sampleIDs: A list of sample id's from BIOM format OTU table.

    :rtype: dict
    :return: Returns a keyed on SampleIDs, and the values are dictionaries keyed on
             OTUID's and their values represent the relative abundance of that OTUID in
             that SampleID.
    """
    if sampleIDs is None:
        sampleIDs = biomf.ids()
    else:
        try:
            for sid in sampleIDs:
                assert sid in biomf.ids()
        except AssertionError:
            raise ValueError(
                "\nError while calculating relative abundances: The sampleIDs provided do"
                " not match the sampleIDs in biom file. Please double check the sampleIDs"
                " provided.\n")
    otuIDs = biomf.ids(axis="observation")
    norm_biomf = biomf.norm(inplace=False)

    return {sample: {otuID: norm_biomf.get_value_by_ids(otuID, sample)
                     for otuID in otuIDs} for sample in sampleIDs}</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "relative_abundance",
  "(",
  "biomf",
  ",",
  "sampleIDs",
  "=",
  "None",
  ")",
  ":",
  "if",
  "sampleIDs",
  "is",
  "None",
  ":",
  "sampleIDs",
  "=",
  "biomf",
  ".",
  "ids",
  "(",
  ")",
  "else",
  ":",
  "try",
  ":",
  "for",
  "sid",
  "in",
  "sampleIDs",
  ":",
  "assert",
  "sid",
  "in",
  "biomf",
  ".",
  "ids",
  "(",
  ")",
  "except",
  "AssertionError",
  ":",
  "raise",
  "ValueError",
  "(",
  "\"\\nError while calculating relative abundances: The sampleIDs provided do\"",
  "\" not match the sampleIDs in biom file. Please double check the sampleIDs\"",
  "\" provided.\\n\"",
  ")",
  "otuIDs",
  "=",
  "biomf",
  ".",
  "ids",
  "(",
  "axis",
  "=",
  "\"observation\"",
  ")",
  "norm_biomf",
  "=",
  "biomf",
  ".",
  "norm",
  "(",
  "inplace",
  "=",
  "False",
  ")",
  "return",
  "{",
  "sample",
  ":",
  "{",
  "otuID",
  ":",
  "norm_biomf",
  ".",
  "get_value_by_ids",
  "(",
  "otuID",
  ",",
  "sample",
  ")",
  "for",
  "otuID",
  "in",
  "otuIDs",
  "}",
  "for",
  "sample",
  "in",
  "sampleIDs",
  "}"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Calculate the relative abundance of each OTUID in a Sample.

    :type biomf: A BIOM file.
    :param biomf: OTU table format.

    :type sampleIDs: list
    :param sampleIDs: A list of sample id's from BIOM format OTU table.

    :rtype: dict
    :return: Returns a keyed on SampleIDs, and the values are dictionaries keyed on
             OTUID's and their values represent the relative abundance of that OTUID in
             that SampleID.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Calculate",
  "the",
  "relative",
  "abundance",
  "of",
  "each",
  "OTUID",
  "in",
  "a",
  "Sample",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/biom_calc.py#L11-L41</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">13</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="14"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">phylotoast/biom_calc.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">mean_otu_pct_abundance</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def mean_otu_pct_abundance(ra, otuIDs):
    """
    Calculate the mean OTU abundance percentage.

    :type ra: Dict
    :param ra: 'ra' refers to a dictionary keyed on SampleIDs, and the values are
               dictionaries keyed on OTUID's and their values represent the relative
               abundance of that OTUID in that SampleID. 'ra' is the output of
               relative_abundance() function.

    :type otuIDs: List
    :param otuIDs: A list of OTUID's for which the percentage abundance needs to be
                   measured.

    :rtype: dict
    :return: A dictionary of OTUID and their percent relative abundance as key/value pair.
    """
    sids = ra.keys()
    otumeans = defaultdict(int)

    for oid in otuIDs:
        otumeans[oid] = sum([ra[sid][oid] for sid in sids
                             if oid in ra[sid]]) / len(sids) * 100
    return otumeans</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def mean_otu_pct_abundance(ra, otuIDs):
    """
    Calculate the mean OTU abundance percentage.

    :type ra: Dict
    :param ra: 'ra' refers to a dictionary keyed on SampleIDs, and the values are
               dictionaries keyed on OTUID's and their values represent the relative
               abundance of that OTUID in that SampleID. 'ra' is the output of
               relative_abundance() function.

    :type otuIDs: List
    :param otuIDs: A list of OTUID's for which the percentage abundance needs to be
                   measured.

    :rtype: dict
    :return: A dictionary of OTUID and their percent relative abundance as key/value pair.
    """
    sids = ra.keys()
    otumeans = defaultdict(int)

    for oid in otuIDs:
        otumeans[oid] = sum([ra[sid][oid] for sid in sids
                             if oid in ra[sid]]) / len(sids) * 100
    return otumeans</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "mean_otu_pct_abundance",
  "(",
  "ra",
  ",",
  "otuIDs",
  ")",
  ":",
  "sids",
  "=",
  "ra",
  ".",
  "keys",
  "(",
  ")",
  "otumeans",
  "=",
  "defaultdict",
  "(",
  "int",
  ")",
  "for",
  "oid",
  "in",
  "otuIDs",
  ":",
  "otumeans",
  "[",
  "oid",
  "]",
  "=",
  "sum",
  "(",
  "[",
  "ra",
  "[",
  "sid",
  "]",
  "[",
  "oid",
  "]",
  "for",
  "sid",
  "in",
  "sids",
  "if",
  "oid",
  "in",
  "ra",
  "[",
  "sid",
  "]",
  "]",
  ")",
  "/",
  "len",
  "(",
  "sids",
  ")",
  "*",
  "100",
  "return",
  "otumeans"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Calculate the mean OTU abundance percentage.

    :type ra: Dict
    :param ra: 'ra' refers to a dictionary keyed on SampleIDs, and the values are
               dictionaries keyed on OTUID's and their values represent the relative
               abundance of that OTUID in that SampleID. 'ra' is the output of
               relative_abundance() function.

    :type otuIDs: List
    :param otuIDs: A list of OTUID's for which the percentage abundance needs to be
                   measured.

    :rtype: dict
    :return: A dictionary of OTUID and their percent relative abundance as key/value pair.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Calculate",
  "the",
  "mean",
  "OTU",
  "abundance",
  "percentage",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/biom_calc.py#L44-L67</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">14</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="15"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">phylotoast/biom_calc.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">MRA</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def MRA(biomf, sampleIDs=None, transform=None):
    """
    Calculate the mean relative abundance percentage.

    :type biomf: A BIOM file.
    :param biomf: OTU table format.

    :type sampleIDs: list
    :param sampleIDs: A list of sample id's from BIOM format OTU table.

    :param transform: Mathematical function which is used to transform smax to another
                      format. By default, the function has been set to None.

    :rtype: dict
    :return: A dictionary keyed on OTUID's and their mean relative abundance for a given
             number of sampleIDs.
    """
    ra = relative_abundance(biomf, sampleIDs)
    if transform is not None:
        ra = {sample: {otuID: transform(abd) for otuID, abd in ra[sample].items()}
              for sample in ra.keys()}
    otuIDs = biomf.ids(axis="observation")
    return mean_otu_pct_abundance(ra, otuIDs)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def MRA(biomf, sampleIDs=None, transform=None):
    """
    Calculate the mean relative abundance percentage.

    :type biomf: A BIOM file.
    :param biomf: OTU table format.

    :type sampleIDs: list
    :param sampleIDs: A list of sample id's from BIOM format OTU table.

    :param transform: Mathematical function which is used to transform smax to another
                      format. By default, the function has been set to None.

    :rtype: dict
    :return: A dictionary keyed on OTUID's and their mean relative abundance for a given
             number of sampleIDs.
    """
    ra = relative_abundance(biomf, sampleIDs)
    if transform is not None:
        ra = {sample: {otuID: transform(abd) for otuID, abd in ra[sample].items()}
              for sample in ra.keys()}
    otuIDs = biomf.ids(axis="observation")
    return mean_otu_pct_abundance(ra, otuIDs)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "MRA",
  "(",
  "biomf",
  ",",
  "sampleIDs",
  "=",
  "None",
  ",",
  "transform",
  "=",
  "None",
  ")",
  ":",
  "ra",
  "=",
  "relative_abundance",
  "(",
  "biomf",
  ",",
  "sampleIDs",
  ")",
  "if",
  "transform",
  "is",
  "not",
  "None",
  ":",
  "ra",
  "=",
  "{",
  "sample",
  ":",
  "{",
  "otuID",
  ":",
  "transform",
  "(",
  "abd",
  ")",
  "for",
  "otuID",
  ",",
  "abd",
  "in",
  "ra",
  "[",
  "sample",
  "]",
  ".",
  "items",
  "(",
  ")",
  "}",
  "for",
  "sample",
  "in",
  "ra",
  ".",
  "keys",
  "(",
  ")",
  "}",
  "otuIDs",
  "=",
  "biomf",
  ".",
  "ids",
  "(",
  "axis",
  "=",
  "\"observation\"",
  ")",
  "return",
  "mean_otu_pct_abundance",
  "(",
  "ra",
  ",",
  "otuIDs",
  ")"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Calculate the mean relative abundance percentage.

    :type biomf: A BIOM file.
    :param biomf: OTU table format.

    :type sampleIDs: list
    :param sampleIDs: A list of sample id's from BIOM format OTU table.

    :param transform: Mathematical function which is used to transform smax to another
                      format. By default, the function has been set to None.

    :rtype: dict
    :return: A dictionary keyed on OTUID's and their mean relative abundance for a given
             number of sampleIDs.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Calculate",
  "the",
  "mean",
  "relative",
  "abundance",
  "percentage",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/biom_calc.py#L70-L92</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">15</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="16"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">phylotoast/biom_calc.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">raw_abundance</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def raw_abundance(biomf, sampleIDs=None, sample_abd=True):
    """
    Calculate the total number of sequences in each OTU or SampleID.

    :type biomf: A BIOM file.
    :param biomf: OTU table format.

    :type sampleIDs: List
    :param sampleIDs: A list of column id's from BIOM format OTU table. By default, the
                      list has been set to None.

    :type sample_abd: Boolean
    :param sample_abd: A boolean operator to provide output for OTUID's or SampleID's. By
                       default, the output will be provided for SampleID's.

    :rtype: dict
    :return: Returns a dictionary keyed on either OTUID's or SampleIDs and their
             respective abundance as values.
    """
    results = defaultdict(int)
    if sampleIDs is None:
        sampleIDs = biomf.ids()
    else:
        try:
            for sid in sampleIDs:
                assert sid in biomf.ids()
        except AssertionError:
            raise ValueError(
                "\nError while calculating raw total abundances: The sampleIDs provided "
                "do not match the sampleIDs in biom file. Please double check the "
                "sampleIDs provided.\n")
    otuIDs = biomf.ids(axis="observation")

    for sampleID in sampleIDs:
        for otuID in otuIDs:
            abd = biomf.get_value_by_ids(otuID, sampleID)
            if sample_abd:
                results[sampleID] += abd
            else:
                results[otuID] += abd
    return results</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def raw_abundance(biomf, sampleIDs=None, sample_abd=True):
    """
    Calculate the total number of sequences in each OTU or SampleID.

    :type biomf: A BIOM file.
    :param biomf: OTU table format.

    :type sampleIDs: List
    :param sampleIDs: A list of column id's from BIOM format OTU table. By default, the
                      list has been set to None.

    :type sample_abd: Boolean
    :param sample_abd: A boolean operator to provide output for OTUID's or SampleID's. By
                       default, the output will be provided for SampleID's.

    :rtype: dict
    :return: Returns a dictionary keyed on either OTUID's or SampleIDs and their
             respective abundance as values.
    """
    results = defaultdict(int)
    if sampleIDs is None:
        sampleIDs = biomf.ids()
    else:
        try:
            for sid in sampleIDs:
                assert sid in biomf.ids()
        except AssertionError:
            raise ValueError(
                "\nError while calculating raw total abundances: The sampleIDs provided "
                "do not match the sampleIDs in biom file. Please double check the "
                "sampleIDs provided.\n")
    otuIDs = biomf.ids(axis="observation")

    for sampleID in sampleIDs:
        for otuID in otuIDs:
            abd = biomf.get_value_by_ids(otuID, sampleID)
            if sample_abd:
                results[sampleID] += abd
            else:
                results[otuID] += abd
    return results</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "raw_abundance",
  "(",
  "biomf",
  ",",
  "sampleIDs",
  "=",
  "None",
  ",",
  "sample_abd",
  "=",
  "True",
  ")",
  ":",
  "results",
  "=",
  "defaultdict",
  "(",
  "int",
  ")",
  "if",
  "sampleIDs",
  "is",
  "None",
  ":",
  "sampleIDs",
  "=",
  "biomf",
  ".",
  "ids",
  "(",
  ")",
  "else",
  ":",
  "try",
  ":",
  "for",
  "sid",
  "in",
  "sampleIDs",
  ":",
  "assert",
  "sid",
  "in",
  "biomf",
  ".",
  "ids",
  "(",
  ")",
  "except",
  "AssertionError",
  ":",
  "raise",
  "ValueError",
  "(",
  "\"\\nError while calculating raw total abundances: The sampleIDs provided \"",
  "\"do not match the sampleIDs in biom file. Please double check the \"",
  "\"sampleIDs provided.\\n\"",
  ")",
  "otuIDs",
  "=",
  "biomf",
  ".",
  "ids",
  "(",
  "axis",
  "=",
  "\"observation\"",
  ")",
  "for",
  "sampleID",
  "in",
  "sampleIDs",
  ":",
  "for",
  "otuID",
  "in",
  "otuIDs",
  ":",
  "abd",
  "=",
  "biomf",
  ".",
  "get_value_by_ids",
  "(",
  "otuID",
  ",",
  "sampleID",
  ")",
  "if",
  "sample_abd",
  ":",
  "results",
  "[",
  "sampleID",
  "]",
  "+=",
  "abd",
  "else",
  ":",
  "results",
  "[",
  "otuID",
  "]",
  "+=",
  "abd",
  "return",
  "results"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Calculate the total number of sequences in each OTU or SampleID.

    :type biomf: A BIOM file.
    :param biomf: OTU table format.

    :type sampleIDs: List
    :param sampleIDs: A list of column id's from BIOM format OTU table. By default, the
                      list has been set to None.

    :type sample_abd: Boolean
    :param sample_abd: A boolean operator to provide output for OTUID's or SampleID's. By
                       default, the output will be provided for SampleID's.

    :rtype: dict
    :return: Returns a dictionary keyed on either OTUID's or SampleIDs and their
             respective abundance as values.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Calculate",
  "the",
  "total",
  "number",
  "of",
  "sequences",
  "in",
  "each",
  "OTU",
  "or",
  "SampleID",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/biom_calc.py#L95-L135</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">16</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="17"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">phylotoast/biom_calc.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">transform_raw_abundance</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def transform_raw_abundance(biomf, fn=math.log10, sampleIDs=None, sample_abd=True):
    """
    Function to transform the total abundance calculation for each sample ID to another
    format based on user given transformation function.

    :type biomf: A BIOM file.
    :param biomf: OTU table format.

    :param fn: Mathematical function which is used to transform smax to another format.
               By default, the function has been given as base 10 logarithm.

    :rtype: dict
    :return: Returns a dictionary similar to output of raw_abundance function but with
             the abundance values modified by the mathematical operation. By default, the
             operation performed on the abundances is base 10 logarithm.
    """
    totals = raw_abundance(biomf, sampleIDs, sample_abd)
    return {sid: fn(abd) for sid, abd in totals.items()}</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def transform_raw_abundance(biomf, fn=math.log10, sampleIDs=None, sample_abd=True):
    """
    Function to transform the total abundance calculation for each sample ID to another
    format based on user given transformation function.

    :type biomf: A BIOM file.
    :param biomf: OTU table format.

    :param fn: Mathematical function which is used to transform smax to another format.
               By default, the function has been given as base 10 logarithm.

    :rtype: dict
    :return: Returns a dictionary similar to output of raw_abundance function but with
             the abundance values modified by the mathematical operation. By default, the
             operation performed on the abundances is base 10 logarithm.
    """
    totals = raw_abundance(biomf, sampleIDs, sample_abd)
    return {sid: fn(abd) for sid, abd in totals.items()}</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "transform_raw_abundance",
  "(",
  "biomf",
  ",",
  "fn",
  "=",
  "math",
  ".",
  "log10",
  ",",
  "sampleIDs",
  "=",
  "None",
  ",",
  "sample_abd",
  "=",
  "True",
  ")",
  ":",
  "totals",
  "=",
  "raw_abundance",
  "(",
  "biomf",
  ",",
  "sampleIDs",
  ",",
  "sample_abd",
  ")",
  "return",
  "{",
  "sid",
  ":",
  "fn",
  "(",
  "abd",
  ")",
  "for",
  "sid",
  ",",
  "abd",
  "in",
  "totals",
  ".",
  "items",
  "(",
  ")",
  "}"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Function to transform the total abundance calculation for each sample ID to another
    format based on user given transformation function.

    :type biomf: A BIOM file.
    :param biomf: OTU table format.

    :param fn: Mathematical function which is used to transform smax to another format.
               By default, the function has been given as base 10 logarithm.

    :rtype: dict
    :return: Returns a dictionary similar to output of raw_abundance function but with
             the abundance values modified by the mathematical operation. By default, the
             operation performed on the abundances is base 10 logarithm.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Function",
  "to",
  "transform",
  "the",
  "total",
  "abundance",
  "calculation",
  "for",
  "each",
  "sample",
  "ID",
  "to",
  "another",
  "format",
  "based",
  "on",
  "user",
  "given",
  "transformation",
  "function",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/phylotoast/biom_calc.py#L138-L155</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">17</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="18"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">bin/diversity.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">print_MannWhitneyU</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def print_MannWhitneyU(div_calc):
    """
    Compute the Mann-Whitney U test for unequal group sample sizes.
    """
    try:
        x = div_calc.values()[0].values()
        y = div_calc.values()[1].values()
    except:
        return "Error setting up input arrays for Mann-Whitney U Test. Skipping "\
               "significance testing."
    T, p = stats.mannwhitneyu(x, y)
    print "\nMann-Whitney U test statistic:", T
    print "Two-tailed p-value: {}".format(2 * p)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def print_MannWhitneyU(div_calc):
    """
    Compute the Mann-Whitney U test for unequal group sample sizes.
    """
    try:
        x = div_calc.values()[0].values()
        y = div_calc.values()[1].values()
    except:
        return "Error setting up input arrays for Mann-Whitney U Test. Skipping "\
               "significance testing."
    T, p = stats.mannwhitneyu(x, y)
    print "\nMann-Whitney U test statistic:", T
    print "Two-tailed p-value: {}".format(2 * p)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "print_MannWhitneyU",
  "(",
  "div_calc",
  ")",
  ":",
  "try",
  ":",
  "x",
  "=",
  "div_calc",
  ".",
  "values",
  "(",
  ")",
  "[",
  "0",
  "]",
  ".",
  "values",
  "(",
  ")",
  "y",
  "=",
  "div_calc",
  ".",
  "values",
  "(",
  ")",
  "[",
  "1",
  "]",
  ".",
  "values",
  "(",
  ")",
  "except",
  ":",
  "return",
  "\"Error setting up input arrays for Mann-Whitney U Test. Skipping \"",
  "\"significance testing.\"",
  "T",
  ",",
  "p",
  "=",
  "stats",
  ".",
  "mannwhitneyu",
  "(",
  "x",
  ",",
  "y",
  ")",
  "print",
  "\"\\nMann-Whitney U test statistic:\"",
  ",",
  "T",
  "print",
  "\"Two-tailed p-value: {}\"",
  ".",
  "format",
  "(",
  "2",
  "*",
  "p",
  ")"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Compute the Mann-Whitney U test for unequal group sample sizes.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Compute",
  "the",
  "Mann",
  "-",
  "Whitney",
  "U",
  "test",
  "for",
  "unequal",
  "group",
  "sample",
  "sizes",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/bin/diversity.py#L54-L66</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">18</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="19"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">bin/diversity.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">print_KruskalWallisH</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def print_KruskalWallisH(div_calc):
    """
    Compute the Kruskal-Wallis H-test for independent samples. A typical rule is that
    each group must have at least 5 measurements.
    """
    calc = defaultdict(list)
    try:
        for k1, v1 in div_calc.iteritems():
            for k2, v2 in v1.iteritems():
                calc[k1].append(v2)
    except:
        return "Error setting up input arrays for Kruskal-Wallis H-Test. Skipping "\
               "significance testing."
    h, p = stats.kruskal(*calc.values())
    print "\nKruskal-Wallis H-test statistic for {} groups: {}".format(str(len(div_calc)), h)
    print "p-value: {}".format(p)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def print_KruskalWallisH(div_calc):
    """
    Compute the Kruskal-Wallis H-test for independent samples. A typical rule is that
    each group must have at least 5 measurements.
    """
    calc = defaultdict(list)
    try:
        for k1, v1 in div_calc.iteritems():
            for k2, v2 in v1.iteritems():
                calc[k1].append(v2)
    except:
        return "Error setting up input arrays for Kruskal-Wallis H-Test. Skipping "\
               "significance testing."
    h, p = stats.kruskal(*calc.values())
    print "\nKruskal-Wallis H-test statistic for {} groups: {}".format(str(len(div_calc)), h)
    print "p-value: {}".format(p)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "print_KruskalWallisH",
  "(",
  "div_calc",
  ")",
  ":",
  "calc",
  "=",
  "defaultdict",
  "(",
  "list",
  ")",
  "try",
  ":",
  "for",
  "k1",
  ",",
  "v1",
  "in",
  "div_calc",
  ".",
  "iteritems",
  "(",
  ")",
  ":",
  "for",
  "k2",
  ",",
  "v2",
  "in",
  "v1",
  ".",
  "iteritems",
  "(",
  ")",
  ":",
  "calc",
  "[",
  "k1",
  "]",
  ".",
  "append",
  "(",
  "v2",
  ")",
  "except",
  ":",
  "return",
  "\"Error setting up input arrays for Kruskal-Wallis H-Test. Skipping \"",
  "\"significance testing.\"",
  "h",
  ",",
  "p",
  "=",
  "stats",
  ".",
  "kruskal",
  "(",
  "*",
  "calc",
  ".",
  "values",
  "(",
  ")",
  ")",
  "print",
  "\"\\nKruskal-Wallis H-test statistic for {} groups: {}\"",
  ".",
  "format",
  "(",
  "str",
  "(",
  "len",
  "(",
  "div_calc",
  ")",
  ")",
  ",",
  "h",
  ")",
  "print",
  "\"p-value: {}\"",
  ".",
  "format",
  "(",
  "p",
  ")"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Compute the Kruskal-Wallis H-test for independent samples. A typical rule is that
    each group must have at least 5 measurements.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Compute",
  "the",
  "Kruskal",
  "-",
  "Wallis",
  "H",
  "-",
  "test",
  "for",
  "independent",
  "samples",
  ".",
  "A",
  "typical",
  "rule",
  "is",
  "that",
  "each",
  "group",
  "must",
  "have",
  "at",
  "least",
  "5",
  "measurements",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/bin/diversity.py#L69-L84</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">19</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="20"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">bin/diversity.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">handle_program_options</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def handle_program_options():
    """Parses the given options passed in at the command line."""
    parser = argparse.ArgumentParser(description="Calculate the alpha diversity\
                                     of a set of samples using one or more \
                                     metrics and output a kernal density \
                                     estimator-smoothed histogram of the \
                                     results.")
    parser.add_argument("-m", "--map_file",
                        help="QIIME mapping file.")
    parser.add_argument("-i", "--biom_fp",
                        help="Path to the BIOM table")
    parser.add_argument("-c", "--category",
                        help="Specific category from the mapping file.")
    parser.add_argument("-d", "--diversity", default=["shannon"], nargs="+",
                        help="The alpha diversity metric. Default \
                             value is 'shannon', which will calculate the Shannon\
                             entropy. Multiple metrics can be specified (space separated).\
                             The full list of metrics is available at:\
                             http://scikit-bio.org/docs/latest/generated/skbio.diversity.alpha.html.\
                             Beta diversity metrics will be supported in the future.")
    parser.add_argument("--x_label", default=[None], nargs="+",
                        help="The name of the diversity metric to be displayed on the\
                        plot as the X-axis label. If multiple metrics are specified,\
                        then multiple entries for the X-axis label should be given.")
    parser.add_argument("--color_by",
                        help="A column name in the mapping file containing\
                              hexadecimal (#FF0000) color values that will\
                              be used to color the groups. Each sample ID must\
                              have a color entry.")
    parser.add_argument("--plot_title", default="",
                        help="A descriptive title that will appear at the top \
                        of the output plot. Surround with quotes if there are\
                        spaces in the title.")
    parser.add_argument("-o", "--output_dir", default=".",
                        help="The directory plots will be saved to.")
    parser.add_argument("--image_type", default="png",
                        help="The type of image to save: png, svg, pdf, eps, etc...")
    parser.add_argument("--save_calculations",
                        help="Path and name of text file to store the calculated "
                        "diversity metrics.")
    parser.add_argument("--suppress_stats", action="store_true", help="Do not display "
                        "significance testing results which are shown by default.")
    parser.add_argument("--show_available_metrics", action="store_true",
                        help="Supply this parameter to see which alpha diversity metrics "
                             " are available for usage. No calculations will be performed"
                             " if this parameter is provided.")
    return parser.parse_args()</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def handle_program_options():
    """Parses the given options passed in at the command line."""
    parser = argparse.ArgumentParser(description="Calculate the alpha diversity\
                                     of a set of samples using one or more \
                                     metrics and output a kernal density \
                                     estimator-smoothed histogram of the \
                                     results.")
    parser.add_argument("-m", "--map_file",
                        help="QIIME mapping file.")
    parser.add_argument("-i", "--biom_fp",
                        help="Path to the BIOM table")
    parser.add_argument("-c", "--category",
                        help="Specific category from the mapping file.")
    parser.add_argument("-d", "--diversity", default=["shannon"], nargs="+",
                        help="The alpha diversity metric. Default \
                             value is 'shannon', which will calculate the Shannon\
                             entropy. Multiple metrics can be specified (space separated).\
                             The full list of metrics is available at:\
                             http://scikit-bio.org/docs/latest/generated/skbio.diversity.alpha.html.\
                             Beta diversity metrics will be supported in the future.")
    parser.add_argument("--x_label", default=[None], nargs="+",
                        help="The name of the diversity metric to be displayed on the\
                        plot as the X-axis label. If multiple metrics are specified,\
                        then multiple entries for the X-axis label should be given.")
    parser.add_argument("--color_by",
                        help="A column name in the mapping file containing\
                              hexadecimal (#FF0000) color values that will\
                              be used to color the groups. Each sample ID must\
                              have a color entry.")
    parser.add_argument("--plot_title", default="",
                        help="A descriptive title that will appear at the top \
                        of the output plot. Surround with quotes if there are\
                        spaces in the title.")
    parser.add_argument("-o", "--output_dir", default=".",
                        help="The directory plots will be saved to.")
    parser.add_argument("--image_type", default="png",
                        help="The type of image to save: png, svg, pdf, eps, etc...")
    parser.add_argument("--save_calculations",
                        help="Path and name of text file to store the calculated "
                        "diversity metrics.")
    parser.add_argument("--suppress_stats", action="store_true", help="Do not display "
                        "significance testing results which are shown by default.")
    parser.add_argument("--show_available_metrics", action="store_true",
                        help="Supply this parameter to see which alpha diversity metrics "
                             " are available for usage. No calculations will be performed"
                             " if this parameter is provided.")
    return parser.parse_args()</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "handle_program_options",
  "(",
  ")",
  ":",
  "parser",
  "=",
  "argparse",
  ".",
  "ArgumentParser",
  "(",
  "description",
  "=",
  "\"Calculate the alpha diversity\\\n                                     of a set of samples using one or more \\\n                                     metrics and output a kernal density \\\n                                     estimator-smoothed histogram of the \\\n                                     results.\"",
  ")",
  "parser",
  ".",
  "add_argument",
  "(",
  "\"-m\"",
  ",",
  "\"--map_file\"",
  ",",
  "help",
  "=",
  "\"QIIME mapping file.\"",
  ")",
  "parser",
  ".",
  "add_argument",
  "(",
  "\"-i\"",
  ",",
  "\"--biom_fp\"",
  ",",
  "help",
  "=",
  "\"Path to the BIOM table\"",
  ")",
  "parser",
  ".",
  "add_argument",
  "(",
  "\"-c\"",
  ",",
  "\"--category\"",
  ",",
  "help",
  "=",
  "\"Specific category from the mapping file.\"",
  ")",
  "parser",
  ".",
  "add_argument",
  "(",
  "\"-d\"",
  ",",
  "\"--diversity\"",
  ",",
  "default",
  "=",
  "[",
  "\"shannon\"",
  "]",
  ",",
  "nargs",
  "=",
  "\"+\"",
  ",",
  "help",
  "=",
  "\"The alpha diversity metric. Default \\\n                             value is 'shannon', which will calculate the Shannon\\\n                             entropy. Multiple metrics can be specified (space separated).\\\n                             The full list of metrics is available at:\\\n                             http://scikit-bio.org/docs/latest/generated/skbio.diversity.alpha.html.\\\n                             Beta diversity metrics will be supported in the future.\"",
  ")",
  "parser",
  ".",
  "add_argument",
  "(",
  "\"--x_label\"",
  ",",
  "default",
  "=",
  "[",
  "None",
  "]",
  ",",
  "nargs",
  "=",
  "\"+\"",
  ",",
  "help",
  "=",
  "\"The name of the diversity metric to be displayed on the\\\n                        plot as the X-axis label. If multiple metrics are specified,\\\n                        then multiple entries for the X-axis label should be given.\"",
  ")",
  "parser",
  ".",
  "add_argument",
  "(",
  "\"--color_by\"",
  ",",
  "help",
  "=",
  "\"A column name in the mapping file containing\\\n                              hexadecimal (#FF0000) color values that will\\\n                              be used to color the groups. Each sample ID must\\\n                              have a color entry.\"",
  ")",
  "parser",
  ".",
  "add_argument",
  "(",
  "\"--plot_title\"",
  ",",
  "default",
  "=",
  "\"\"",
  ",",
  "help",
  "=",
  "\"A descriptive title that will appear at the top \\\n                        of the output plot. Surround with quotes if there are\\\n                        spaces in the title.\"",
  ")",
  "parser",
  ".",
  "add_argument",
  "(",
  "\"-o\"",
  ",",
  "\"--output_dir\"",
  ",",
  "default",
  "=",
  "\".\"",
  ",",
  "help",
  "=",
  "\"The directory plots will be saved to.\"",
  ")",
  "parser",
  ".",
  "add_argument",
  "(",
  "\"--image_type\"",
  ",",
  "default",
  "=",
  "\"png\"",
  ",",
  "help",
  "=",
  "\"The type of image to save: png, svg, pdf, eps, etc...\"",
  ")",
  "parser",
  ".",
  "add_argument",
  "(",
  "\"--save_calculations\"",
  ",",
  "help",
  "=",
  "\"Path and name of text file to store the calculated \"",
  "\"diversity metrics.\"",
  ")",
  "parser",
  ".",
  "add_argument",
  "(",
  "\"--suppress_stats\"",
  ",",
  "action",
  "=",
  "\"store_true\"",
  ",",
  "help",
  "=",
  "\"Do not display \"",
  "\"significance testing results which are shown by default.\"",
  ")",
  "parser",
  ".",
  "add_argument",
  "(",
  "\"--show_available_metrics\"",
  ",",
  "action",
  "=",
  "\"store_true\"",
  ",",
  "help",
  "=",
  "\"Supply this parameter to see which alpha diversity metrics \"",
  "\" are available for usage. No calculations will be performed\"",
  "\" if this parameter is provided.\"",
  ")",
  "return",
  "parser",
  ".",
  "parse_args",
  "(",
  ")"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Parses the given options passed in at the command line.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Parses",
  "the",
  "given",
  "options",
  "passed",
  "in",
  "at",
  "the",
  "command",
  "line",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/bin/diversity.py#L122-L168</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">20</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="21"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">christophertbrown/bioscripts</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ctbBio/search.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">blastdb</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def blastdb(fasta, maxfile = 10000000):
    """
    make blast db
    """
    db = fasta.rsplit('.', 1)[0]
    type = check_type(fasta)
    if type == 'nucl':
        type = ['nhr', type]
    else:
        type = ['phr', type]
    if os.path.exists('%s.%s' % (db, type[0])) is False \
            and os.path.exists('%s.00.%s' % (db, type[0])) is False:
        print('# ... making blastdb for: %s' % (fasta), file=sys.stderr)
        os.system('makeblastdb \
                -in %s -out %s -dbtype %s -max_file_sz %s >> log.txt' \
                % (fasta, db, type[1], maxfile))
    else:
        print('# ... database found for: %s' % (fasta), file=sys.stderr)
    return db</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def blastdb(fasta, maxfile = 10000000):
    """
    make blast db
    """
    db = fasta.rsplit('.', 1)[0]
    type = check_type(fasta)
    if type == 'nucl':
        type = ['nhr', type]
    else:
        type = ['phr', type]
    if os.path.exists('%s.%s' % (db, type[0])) is False \
            and os.path.exists('%s.00.%s' % (db, type[0])) is False:
        print('# ... making blastdb for: %s' % (fasta), file=sys.stderr)
        os.system('makeblastdb \
                -in %s -out %s -dbtype %s -max_file_sz %s >> log.txt' \
                % (fasta, db, type[1], maxfile))
    else:
        print('# ... database found for: %s' % (fasta), file=sys.stderr)
    return db</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "blastdb",
  "(",
  "fasta",
  ",",
  "maxfile",
  "=",
  "10000000",
  ")",
  ":",
  "db",
  "=",
  "fasta",
  ".",
  "rsplit",
  "(",
  "'.'",
  ",",
  "1",
  ")",
  "[",
  "0",
  "]",
  "type",
  "=",
  "check_type",
  "(",
  "fasta",
  ")",
  "if",
  "type",
  "==",
  "'nucl'",
  ":",
  "type",
  "=",
  "[",
  "'nhr'",
  ",",
  "type",
  "]",
  "else",
  ":",
  "type",
  "=",
  "[",
  "'phr'",
  ",",
  "type",
  "]",
  "if",
  "os",
  ".",
  "path",
  ".",
  "exists",
  "(",
  "'%s.%s'",
  "%",
  "(",
  "db",
  ",",
  "type",
  "[",
  "0",
  "]",
  ")",
  ")",
  "is",
  "False",
  "and",
  "os",
  ".",
  "path",
  ".",
  "exists",
  "(",
  "'%s.00.%s'",
  "%",
  "(",
  "db",
  ",",
  "type",
  "[",
  "0",
  "]",
  ")",
  ")",
  "is",
  "False",
  ":",
  "print",
  "(",
  "'# ... making blastdb for: %s'",
  "%",
  "(",
  "fasta",
  ")",
  ",",
  "file",
  "=",
  "sys",
  ".",
  "stderr",
  ")",
  "os",
  ".",
  "system",
  "(",
  "'makeblastdb \\\n                -in %s -out %s -dbtype %s -max_file_sz %s >> log.txt'",
  "%",
  "(",
  "fasta",
  ",",
  "db",
  ",",
  "type",
  "[",
  "1",
  "]",
  ",",
  "maxfile",
  ")",
  ")",
  "else",
  ":",
  "print",
  "(",
  "'# ... database found for: %s'",
  "%",
  "(",
  "fasta",
  ")",
  ",",
  "file",
  "=",
  "sys",
  ".",
  "stderr",
  ")",
  "return",
  "db"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">make blast db</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "make",
  "blast",
  "db"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">83b2566b3a5745437ec651cd6cafddd056846240</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/search.py#L28-L46</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">21</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="22"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">christophertbrown/bioscripts</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ctbBio/search.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">usearchdb</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def usearchdb(fasta, alignment = 'local', usearch_loc = 'usearch'):
    """
    make usearch db
    """
    if '.udb' in fasta:
        print('# ... database found: %s' % (fasta), file=sys.stderr)
        return fasta
    type = check_type(fasta)
    db = '%s.%s.udb' % (fasta.rsplit('.', 1)[0], type)
    if os.path.exists(db) is False:
        print('# ... making usearch db for: %s' % (fasta), file=sys.stderr)
        if alignment == 'local':
            os.system('%s -makeudb_ublast %s -output %s >> log.txt' % (usearch_loc, fasta, db))
        elif alignment == 'global':
            os.system('%s -makeudb_usearch %s -output %s >> log.txt' % (usearch_loc, fasta, db))
    else:
        print('# ... database found for: %s' % (fasta), file=sys.stderr)
    return db</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def usearchdb(fasta, alignment = 'local', usearch_loc = 'usearch'):
    """
    make usearch db
    """
    if '.udb' in fasta:
        print('# ... database found: %s' % (fasta), file=sys.stderr)
        return fasta
    type = check_type(fasta)
    db = '%s.%s.udb' % (fasta.rsplit('.', 1)[0], type)
    if os.path.exists(db) is False:
        print('# ... making usearch db for: %s' % (fasta), file=sys.stderr)
        if alignment == 'local':
            os.system('%s -makeudb_ublast %s -output %s >> log.txt' % (usearch_loc, fasta, db))
        elif alignment == 'global':
            os.system('%s -makeudb_usearch %s -output %s >> log.txt' % (usearch_loc, fasta, db))
    else:
        print('# ... database found for: %s' % (fasta), file=sys.stderr)
    return db</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "usearchdb",
  "(",
  "fasta",
  ",",
  "alignment",
  "=",
  "'local'",
  ",",
  "usearch_loc",
  "=",
  "'usearch'",
  ")",
  ":",
  "if",
  "'.udb'",
  "in",
  "fasta",
  ":",
  "print",
  "(",
  "'# ... database found: %s'",
  "%",
  "(",
  "fasta",
  ")",
  ",",
  "file",
  "=",
  "sys",
  ".",
  "stderr",
  ")",
  "return",
  "fasta",
  "type",
  "=",
  "check_type",
  "(",
  "fasta",
  ")",
  "db",
  "=",
  "'%s.%s.udb'",
  "%",
  "(",
  "fasta",
  ".",
  "rsplit",
  "(",
  "'.'",
  ",",
  "1",
  ")",
  "[",
  "0",
  "]",
  ",",
  "type",
  ")",
  "if",
  "os",
  ".",
  "path",
  ".",
  "exists",
  "(",
  "db",
  ")",
  "is",
  "False",
  ":",
  "print",
  "(",
  "'# ... making usearch db for: %s'",
  "%",
  "(",
  "fasta",
  ")",
  ",",
  "file",
  "=",
  "sys",
  ".",
  "stderr",
  ")",
  "if",
  "alignment",
  "==",
  "'local'",
  ":",
  "os",
  ".",
  "system",
  "(",
  "'%s -makeudb_ublast %s -output %s >> log.txt'",
  "%",
  "(",
  "usearch_loc",
  ",",
  "fasta",
  ",",
  "db",
  ")",
  ")",
  "elif",
  "alignment",
  "==",
  "'global'",
  ":",
  "os",
  ".",
  "system",
  "(",
  "'%s -makeudb_usearch %s -output %s >> log.txt'",
  "%",
  "(",
  "usearch_loc",
  ",",
  "fasta",
  ",",
  "db",
  ")",
  ")",
  "else",
  ":",
  "print",
  "(",
  "'# ... database found for: %s'",
  "%",
  "(",
  "fasta",
  ")",
  ",",
  "file",
  "=",
  "sys",
  ".",
  "stderr",
  ")",
  "return",
  "db"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">make usearch db</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "make",
  "usearch",
  "db"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">83b2566b3a5745437ec651cd6cafddd056846240</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/search.py#L68-L85</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">22</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="23"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">mkouhei/bootstrap-py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">bootstrap_py/control.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">_pp</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _pp(dict_data):
    """Pretty print."""
    for key, val in dict_data.items():
        # pylint: disable=superfluous-parens
        print('{0:&lt;11}: {1}'.format(key, val))</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _pp(dict_data):
    """Pretty print."""
    for key, val in dict_data.items():
        # pylint: disable=superfluous-parens
        print('{0:&lt;11}: {1}'.format(key, val))</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "_pp",
  "(",
  "dict_data",
  ")",
  ":",
  "for",
  "key",
  ",",
  "val",
  "in",
  "dict_data",
  ".",
  "items",
  "(",
  ")",
  ":",
  "# pylint: disable=superfluous-parens",
  "print",
  "(",
  "'{0:&lt;11}: {1}'",
  ".",
  "format",
  "(",
  "key",
  ",",
  "val",
  ")",
  ")"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Pretty print.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Pretty",
  "print",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">95d56ed98ef409fd9f019dc352fd1c3711533275</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/mkouhei/bootstrap-py/blob/95d56ed98ef409fd9f019dc352fd1c3711533275/bootstrap_py/control.py#L11-L15</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">23</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="24"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">mkouhei/bootstrap-py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">bootstrap_py/control.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">print_licences</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def print_licences(params, metadata):
    """Print licenses.

    :param argparse.Namespace params: parameter
    :param bootstrap_py.classifier.Classifiers metadata: package metadata
    """
    if hasattr(params, 'licenses'):
        if params.licenses:
            _pp(metadata.licenses_desc())
        sys.exit(0)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def print_licences(params, metadata):
    """Print licenses.

    :param argparse.Namespace params: parameter
    :param bootstrap_py.classifier.Classifiers metadata: package metadata
    """
    if hasattr(params, 'licenses'):
        if params.licenses:
            _pp(metadata.licenses_desc())
        sys.exit(0)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "print_licences",
  "(",
  "params",
  ",",
  "metadata",
  ")",
  ":",
  "if",
  "hasattr",
  "(",
  "params",
  ",",
  "'licenses'",
  ")",
  ":",
  "if",
  "params",
  ".",
  "licenses",
  ":",
  "_pp",
  "(",
  "metadata",
  ".",
  "licenses_desc",
  "(",
  ")",
  ")",
  "sys",
  ".",
  "exit",
  "(",
  "0",
  ")"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Print licenses.

    :param argparse.Namespace params: parameter
    :param bootstrap_py.classifier.Classifiers metadata: package metadata</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Print",
  "licenses",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">95d56ed98ef409fd9f019dc352fd1c3711533275</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/mkouhei/bootstrap-py/blob/95d56ed98ef409fd9f019dc352fd1c3711533275/bootstrap_py/control.py#L27-L36</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">24</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="25"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">mkouhei/bootstrap-py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">bootstrap_py/control.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">check_repository_existence</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def check_repository_existence(params):
    """Check repository existence.

    :param argparse.Namespace params: parameters
    """
    repodir = os.path.join(params.outdir, params.name)
    if os.path.isdir(repodir):
        raise Conflict(
            'Package repository "{0}" has already exists.'.format(repodir))</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def check_repository_existence(params):
    """Check repository existence.

    :param argparse.Namespace params: parameters
    """
    repodir = os.path.join(params.outdir, params.name)
    if os.path.isdir(repodir):
        raise Conflict(
            'Package repository "{0}" has already exists.'.format(repodir))</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "check_repository_existence",
  "(",
  "params",
  ")",
  ":",
  "repodir",
  "=",
  "os",
  ".",
  "path",
  ".",
  "join",
  "(",
  "params",
  ".",
  "outdir",
  ",",
  "params",
  ".",
  "name",
  ")",
  "if",
  "os",
  ".",
  "path",
  ".",
  "isdir",
  "(",
  "repodir",
  ")",
  ":",
  "raise",
  "Conflict",
  "(",
  "'Package repository \"{0}\" has already exists.'",
  ".",
  "format",
  "(",
  "repodir",
  ")",
  ")"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Check repository existence.

    :param argparse.Namespace params: parameters</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Check",
  "repository",
  "existence",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">95d56ed98ef409fd9f019dc352fd1c3711533275</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/mkouhei/bootstrap-py/blob/95d56ed98ef409fd9f019dc352fd1c3711533275/bootstrap_py/control.py#L39-L47</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">25</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="26"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">mkouhei/bootstrap-py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">bootstrap_py/control.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">generate_package</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def generate_package(params):
    """Generate package repository.

    :param argparse.Namespace params: parameters
    """
    pkg_data = package.PackageData(params)
    pkg_tree = package.PackageTree(pkg_data)
    pkg_tree.generate()
    pkg_tree.move()
    VCS(os.path.join(pkg_tree.outdir, pkg_tree.name), pkg_tree.pkg_data)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def generate_package(params):
    """Generate package repository.

    :param argparse.Namespace params: parameters
    """
    pkg_data = package.PackageData(params)
    pkg_tree = package.PackageTree(pkg_data)
    pkg_tree.generate()
    pkg_tree.move()
    VCS(os.path.join(pkg_tree.outdir, pkg_tree.name), pkg_tree.pkg_data)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "generate_package",
  "(",
  "params",
  ")",
  ":",
  "pkg_data",
  "=",
  "package",
  ".",
  "PackageData",
  "(",
  "params",
  ")",
  "pkg_tree",
  "=",
  "package",
  ".",
  "PackageTree",
  "(",
  "pkg_data",
  ")",
  "pkg_tree",
  ".",
  "generate",
  "(",
  ")",
  "pkg_tree",
  ".",
  "move",
  "(",
  ")",
  "VCS",
  "(",
  "os",
  ".",
  "path",
  ".",
  "join",
  "(",
  "pkg_tree",
  ".",
  "outdir",
  ",",
  "pkg_tree",
  ".",
  "name",
  ")",
  ",",
  "pkg_tree",
  ".",
  "pkg_data",
  ")"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Generate package repository.

    :param argparse.Namespace params: parameters</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Generate",
  "package",
  "repository",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">95d56ed98ef409fd9f019dc352fd1c3711533275</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/mkouhei/bootstrap-py/blob/95d56ed98ef409fd9f019dc352fd1c3711533275/bootstrap_py/control.py#L59-L68</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">26</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="27"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">christophertbrown/bioscripts</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ctbBio/sam2fastq.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">print_single</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def print_single(line, rev):
    """
    print single reads to stderr
    """
    if rev is True:
        seq = rc(['', line[9]])[1]
        qual = line[10][::-1]
    else:
        seq = line[9]
        qual = line[10]
    fq = ['@%s' % line[0], seq, '+%s' % line[0], qual]
    print('\n'.join(fq), file = sys.stderr)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def print_single(line, rev):
    """
    print single reads to stderr
    """
    if rev is True:
        seq = rc(['', line[9]])[1]
        qual = line[10][::-1]
    else:
        seq = line[9]
        qual = line[10]
    fq = ['@%s' % line[0], seq, '+%s' % line[0], qual]
    print('\n'.join(fq), file = sys.stderr)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "print_single",
  "(",
  "line",
  ",",
  "rev",
  ")",
  ":",
  "if",
  "rev",
  "is",
  "True",
  ":",
  "seq",
  "=",
  "rc",
  "(",
  "[",
  "''",
  ",",
  "line",
  "[",
  "9",
  "]",
  "]",
  ")",
  "[",
  "1",
  "]",
  "qual",
  "=",
  "line",
  "[",
  "10",
  "]",
  "[",
  ":",
  ":",
  "-",
  "1",
  "]",
  "else",
  ":",
  "seq",
  "=",
  "line",
  "[",
  "9",
  "]",
  "qual",
  "=",
  "line",
  "[",
  "10",
  "]",
  "fq",
  "=",
  "[",
  "'@%s'",
  "%",
  "line",
  "[",
  "0",
  "]",
  ",",
  "seq",
  ",",
  "'+%s'",
  "%",
  "line",
  "[",
  "0",
  "]",
  ",",
  "qual",
  "]",
  "print",
  "(",
  "'\\n'",
  ".",
  "join",
  "(",
  "fq",
  ")",
  ",",
  "file",
  "=",
  "sys",
  ".",
  "stderr",
  ")"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">print single reads to stderr</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "print",
  "single",
  "reads",
  "to",
  "stderr"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">83b2566b3a5745437ec651cd6cafddd056846240</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/sam2fastq.py#L13-L24</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">27</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="28"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">christophertbrown/bioscripts</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ctbBio/sam2fastq.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">sam2fastq</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def sam2fastq(sam, singles = False, force = False):
    """
    convert sam to fastq
    """
    L, R = None, None
    for line in sam:
        if line.startswith('@') is True:
            continue
        line = line.strip().split()
        bit = [True if i == '1' else False \
                for i in bin(int(line[1])).split('b')[1][::-1]]
        while len(bit) &lt; 8:
            bit.append(False)
        pair, proper, na, nap, rev, mrev, left, right = bit
        # make sure read is paired
        if pair is False:
            if singles is True:
                print_single(line, rev)
            continue
        # check if sequence is reverse-complemented
        if rev is True:
            seq = rc(['', line[9]])[1]
            qual = line[10][::-1]
        else:
            seq = line[9]
            qual = line[10]
        # check if read is forward or reverse, return when both have been found
        if left is True:
            if L is not None and force is False:
                print('sam file is not sorted', file = sys.stderr)
                print('\te.g.: %s' % (line[0]), file = sys.stderr)
                exit()
            if L is not None:
                L = None
                continue
            L = ['@%s' % line[0], seq, '+%s' % line[0], qual]
            if R is not None:
                yield L
                yield R
                L, R = None, None
        if right is True:
            if R is not None and force is False:
                print('sam file is not sorted', file = sys.stderr)
                print('\te.g.: %s' % (line[0]), file = sys.stderr)
                exit()
            if R is not None:
                R = None
                continue
            R = ['@%s' % line[0], seq, '+%s' % line[0], qual]
            if L is not None:
                yield L
                yield R
                L, R = None, None</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def sam2fastq(sam, singles = False, force = False):
    """
    convert sam to fastq
    """
    L, R = None, None
    for line in sam:
        if line.startswith('@') is True:
            continue
        line = line.strip().split()
        bit = [True if i == '1' else False \
                for i in bin(int(line[1])).split('b')[1][::-1]]
        while len(bit) &lt; 8:
            bit.append(False)
        pair, proper, na, nap, rev, mrev, left, right = bit
        # make sure read is paired
        if pair is False:
            if singles is True:
                print_single(line, rev)
            continue
        # check if sequence is reverse-complemented
        if rev is True:
            seq = rc(['', line[9]])[1]
            qual = line[10][::-1]
        else:
            seq = line[9]
            qual = line[10]
        # check if read is forward or reverse, return when both have been found
        if left is True:
            if L is not None and force is False:
                print('sam file is not sorted', file = sys.stderr)
                print('\te.g.: %s' % (line[0]), file = sys.stderr)
                exit()
            if L is not None:
                L = None
                continue
            L = ['@%s' % line[0], seq, '+%s' % line[0], qual]
            if R is not None:
                yield L
                yield R
                L, R = None, None
        if right is True:
            if R is not None and force is False:
                print('sam file is not sorted', file = sys.stderr)
                print('\te.g.: %s' % (line[0]), file = sys.stderr)
                exit()
            if R is not None:
                R = None
                continue
            R = ['@%s' % line[0], seq, '+%s' % line[0], qual]
            if L is not None:
                yield L
                yield R
                L, R = None, None</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "sam2fastq",
  "(",
  "sam",
  ",",
  "singles",
  "=",
  "False",
  ",",
  "force",
  "=",
  "False",
  ")",
  ":",
  "L",
  ",",
  "R",
  "=",
  "None",
  ",",
  "None",
  "for",
  "line",
  "in",
  "sam",
  ":",
  "if",
  "line",
  ".",
  "startswith",
  "(",
  "'@'",
  ")",
  "is",
  "True",
  ":",
  "continue",
  "line",
  "=",
  "line",
  ".",
  "strip",
  "(",
  ")",
  ".",
  "split",
  "(",
  ")",
  "bit",
  "=",
  "[",
  "True",
  "if",
  "i",
  "==",
  "'1'",
  "else",
  "False",
  "for",
  "i",
  "in",
  "bin",
  "(",
  "int",
  "(",
  "line",
  "[",
  "1",
  "]",
  ")",
  ")",
  ".",
  "split",
  "(",
  "'b'",
  ")",
  "[",
  "1",
  "]",
  "[",
  ":",
  ":",
  "-",
  "1",
  "]",
  "]",
  "while",
  "len",
  "(",
  "bit",
  ")",
  "&lt;",
  "8",
  ":",
  "bit",
  ".",
  "append",
  "(",
  "False",
  ")",
  "pair",
  ",",
  "proper",
  ",",
  "na",
  ",",
  "nap",
  ",",
  "rev",
  ",",
  "mrev",
  ",",
  "left",
  ",",
  "right",
  "=",
  "bit",
  "# make sure read is paired",
  "if",
  "pair",
  "is",
  "False",
  ":",
  "if",
  "singles",
  "is",
  "True",
  ":",
  "print_single",
  "(",
  "line",
  ",",
  "rev",
  ")",
  "continue",
  "# check if sequence is reverse-complemented",
  "if",
  "rev",
  "is",
  "True",
  ":",
  "seq",
  "=",
  "rc",
  "(",
  "[",
  "''",
  ",",
  "line",
  "[",
  "9",
  "]",
  "]",
  ")",
  "[",
  "1",
  "]",
  "qual",
  "=",
  "line",
  "[",
  "10",
  "]",
  "[",
  ":",
  ":",
  "-",
  "1",
  "]",
  "else",
  ":",
  "seq",
  "=",
  "line",
  "[",
  "9",
  "]",
  "qual",
  "=",
  "line",
  "[",
  "10",
  "]",
  "# check if read is forward or reverse, return when both have been found",
  "if",
  "left",
  "is",
  "True",
  ":",
  "if",
  "L",
  "is",
  "not",
  "None",
  "and",
  "force",
  "is",
  "False",
  ":",
  "print",
  "(",
  "'sam file is not sorted'",
  ",",
  "file",
  "=",
  "sys",
  ".",
  "stderr",
  ")",
  "print",
  "(",
  "'\\te.g.: %s'",
  "%",
  "(",
  "line",
  "[",
  "0",
  "]",
  ")",
  ",",
  "file",
  "=",
  "sys",
  ".",
  "stderr",
  ")",
  "exit",
  "(",
  ")",
  "if",
  "L",
  "is",
  "not",
  "None",
  ":",
  "L",
  "=",
  "None",
  "continue",
  "L",
  "=",
  "[",
  "'@%s'",
  "%",
  "line",
  "[",
  "0",
  "]",
  ",",
  "seq",
  ",",
  "'+%s'",
  "%",
  "line",
  "[",
  "0",
  "]",
  ",",
  "qual",
  "]",
  "if",
  "R",
  "is",
  "not",
  "None",
  ":",
  "yield",
  "L",
  "yield",
  "R",
  "L",
  ",",
  "R",
  "=",
  "None",
  ",",
  "None",
  "if",
  "right",
  "is",
  "True",
  ":",
  "if",
  "R",
  "is",
  "not",
  "None",
  "and",
  "force",
  "is",
  "False",
  ":",
  "print",
  "(",
  "'sam file is not sorted'",
  ",",
  "file",
  "=",
  "sys",
  ".",
  "stderr",
  ")",
  "print",
  "(",
  "'\\te.g.: %s'",
  "%",
  "(",
  "line",
  "[",
  "0",
  "]",
  ")",
  ",",
  "file",
  "=",
  "sys",
  ".",
  "stderr",
  ")",
  "exit",
  "(",
  ")",
  "if",
  "R",
  "is",
  "not",
  "None",
  ":",
  "R",
  "=",
  "None",
  "continue",
  "R",
  "=",
  "[",
  "'@%s'",
  "%",
  "line",
  "[",
  "0",
  "]",
  ",",
  "seq",
  ",",
  "'+%s'",
  "%",
  "line",
  "[",
  "0",
  "]",
  ",",
  "qual",
  "]",
  "if",
  "L",
  "is",
  "not",
  "None",
  ":",
  "yield",
  "L",
  "yield",
  "R",
  "L",
  ",",
  "R",
  "=",
  "None",
  ",",
  "None"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">convert sam to fastq</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "convert",
  "sam",
  "to",
  "fastq"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">83b2566b3a5745437ec651cd6cafddd056846240</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/sam2fastq.py#L26-L78</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">28</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="29"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">christophertbrown/bioscripts</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ctbBio/subset_sam.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">sort_sam</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def sort_sam(sam, sort):
    """
    sort sam file
    """
    tempdir = '%s/' % (os.path.abspath(sam).rsplit('/', 1)[0])
    if sort is True:
        mapping = '%s.sorted.sam' % (sam.rsplit('.', 1)[0])
        if sam != '-':
            if os.path.exists(mapping) is False:
                os.system("\
                    sort -k1 --buffer-size=%sG -T %s -o %s %s\
                    " % (sbuffer, tempdir, mapping, sam)) 
        else:
            mapping = 'stdin-sam.sorted.sam'
            p = Popen("sort -k1 --buffer-size=%sG -T %s -o %s" \
                    % (sbuffer, tempdir, mapping), stdin = sys.stdin, shell = True) 
            p.communicate()
        mapping = open(mapping)
    else:
        if sam == '-':
            mapping = sys.stdin
        else:
            mapping = open(sam)
    return mapping</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def sort_sam(sam, sort):
    """
    sort sam file
    """
    tempdir = '%s/' % (os.path.abspath(sam).rsplit('/', 1)[0])
    if sort is True:
        mapping = '%s.sorted.sam' % (sam.rsplit('.', 1)[0])
        if sam != '-':
            if os.path.exists(mapping) is False:
                os.system("\
                    sort -k1 --buffer-size=%sG -T %s -o %s %s\
                    " % (sbuffer, tempdir, mapping, sam)) 
        else:
            mapping = 'stdin-sam.sorted.sam'
            p = Popen("sort -k1 --buffer-size=%sG -T %s -o %s" \
                    % (sbuffer, tempdir, mapping), stdin = sys.stdin, shell = True) 
            p.communicate()
        mapping = open(mapping)
    else:
        if sam == '-':
            mapping = sys.stdin
        else:
            mapping = open(sam)
    return mapping</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "sort_sam",
  "(",
  "sam",
  ",",
  "sort",
  ")",
  ":",
  "tempdir",
  "=",
  "'%s/'",
  "%",
  "(",
  "os",
  ".",
  "path",
  ".",
  "abspath",
  "(",
  "sam",
  ")",
  ".",
  "rsplit",
  "(",
  "'/'",
  ",",
  "1",
  ")",
  "[",
  "0",
  "]",
  ")",
  "if",
  "sort",
  "is",
  "True",
  ":",
  "mapping",
  "=",
  "'%s.sorted.sam'",
  "%",
  "(",
  "sam",
  ".",
  "rsplit",
  "(",
  "'.'",
  ",",
  "1",
  ")",
  "[",
  "0",
  "]",
  ")",
  "if",
  "sam",
  "!=",
  "'-'",
  ":",
  "if",
  "os",
  ".",
  "path",
  ".",
  "exists",
  "(",
  "mapping",
  ")",
  "is",
  "False",
  ":",
  "os",
  ".",
  "system",
  "(",
  "\"\\\n                    sort -k1 --buffer-size=%sG -T %s -o %s %s\\\n                    \"",
  "%",
  "(",
  "sbuffer",
  ",",
  "tempdir",
  ",",
  "mapping",
  ",",
  "sam",
  ")",
  ")",
  "else",
  ":",
  "mapping",
  "=",
  "'stdin-sam.sorted.sam'",
  "p",
  "=",
  "Popen",
  "(",
  "\"sort -k1 --buffer-size=%sG -T %s -o %s\"",
  "%",
  "(",
  "sbuffer",
  ",",
  "tempdir",
  ",",
  "mapping",
  ")",
  ",",
  "stdin",
  "=",
  "sys",
  ".",
  "stdin",
  ",",
  "shell",
  "=",
  "True",
  ")",
  "p",
  ".",
  "communicate",
  "(",
  ")",
  "mapping",
  "=",
  "open",
  "(",
  "mapping",
  ")",
  "else",
  ":",
  "if",
  "sam",
  "==",
  "'-'",
  ":",
  "mapping",
  "=",
  "sys",
  ".",
  "stdin",
  "else",
  ":",
  "mapping",
  "=",
  "open",
  "(",
  "sam",
  ")",
  "return",
  "mapping"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">sort sam file</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "sort",
  "sam",
  "file"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">83b2566b3a5745437ec651cd6cafddd056846240</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/subset_sam.py#L14-L37</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">29</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="30"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">christophertbrown/bioscripts</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ctbBio/subset_sam.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">sub_sam</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def sub_sam(sam, percent, sort = True, sbuffer = False):
    """
    randomly subset sam file
    """
    mapping = sort_sam(sam, sort)
    pool = [1 for i in range(0, percent)] + [0 for i in range(0, 100 - percent)]
    c = cycle([1, 2])
    for line in mapping:
        line = line.strip().split()
        if line[0].startswith('@'): # get the sam header
            yield line
            continue
        if int(line[1]) &lt;= 20: # is this from a single read?
            if random.choice(pool) == 1:
                yield line
        else:
            n = next(c)
            if n == 1:
                prev = line
            if n == 2 and random.choice(pool) == 1:
                yield prev
                yield line</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def sub_sam(sam, percent, sort = True, sbuffer = False):
    """
    randomly subset sam file
    """
    mapping = sort_sam(sam, sort)
    pool = [1 for i in range(0, percent)] + [0 for i in range(0, 100 - percent)]
    c = cycle([1, 2])
    for line in mapping:
        line = line.strip().split()
        if line[0].startswith('@'): # get the sam header
            yield line
            continue
        if int(line[1]) &lt;= 20: # is this from a single read?
            if random.choice(pool) == 1:
                yield line
        else:
            n = next(c)
            if n == 1:
                prev = line
            if n == 2 and random.choice(pool) == 1:
                yield prev
                yield line</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "sub_sam",
  "(",
  "sam",
  ",",
  "percent",
  ",",
  "sort",
  "=",
  "True",
  ",",
  "sbuffer",
  "=",
  "False",
  ")",
  ":",
  "mapping",
  "=",
  "sort_sam",
  "(",
  "sam",
  ",",
  "sort",
  ")",
  "pool",
  "=",
  "[",
  "1",
  "for",
  "i",
  "in",
  "range",
  "(",
  "0",
  ",",
  "percent",
  ")",
  "]",
  "+",
  "[",
  "0",
  "for",
  "i",
  "in",
  "range",
  "(",
  "0",
  ",",
  "100",
  "-",
  "percent",
  ")",
  "]",
  "c",
  "=",
  "cycle",
  "(",
  "[",
  "1",
  ",",
  "2",
  "]",
  ")",
  "for",
  "line",
  "in",
  "mapping",
  ":",
  "line",
  "=",
  "line",
  ".",
  "strip",
  "(",
  ")",
  ".",
  "split",
  "(",
  ")",
  "if",
  "line",
  "[",
  "0",
  "]",
  ".",
  "startswith",
  "(",
  "'@'",
  ")",
  ":",
  "# get the sam header",
  "yield",
  "line",
  "continue",
  "if",
  "int",
  "(",
  "line",
  "[",
  "1",
  "]",
  ")",
  "&lt;=",
  "20",
  ":",
  "# is this from a single read?",
  "if",
  "random",
  ".",
  "choice",
  "(",
  "pool",
  ")",
  "==",
  "1",
  ":",
  "yield",
  "line",
  "else",
  ":",
  "n",
  "=",
  "next",
  "(",
  "c",
  ")",
  "if",
  "n",
  "==",
  "1",
  ":",
  "prev",
  "=",
  "line",
  "if",
  "n",
  "==",
  "2",
  "and",
  "random",
  ".",
  "choice",
  "(",
  "pool",
  ")",
  "==",
  "1",
  ":",
  "yield",
  "prev",
  "yield",
  "line"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">randomly subset sam file</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "randomly",
  "subset",
  "sam",
  "file"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">83b2566b3a5745437ec651cd6cafddd056846240</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/subset_sam.py#L39-L60</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">30</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="31"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">christophertbrown/bioscripts</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ctbBio/fastq2fasta.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">fq2fa</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def fq2fa(fq):
    """
    convert fq to fa
    """
    c = cycle([1, 2, 3, 4])
    for line in fq:
        n = next(c)
        if n == 1:
            seq = ['>%s' % (line.strip().split('@', 1)[1])]
        if n == 2:
            seq.append(line.strip())
            yield seq</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def fq2fa(fq):
    """
    convert fq to fa
    """
    c = cycle([1, 2, 3, 4])
    for line in fq:
        n = next(c)
        if n == 1:
            seq = ['>%s' % (line.strip().split('@', 1)[1])]
        if n == 2:
            seq.append(line.strip())
            yield seq</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "fq2fa",
  "(",
  "fq",
  ")",
  ":",
  "c",
  "=",
  "cycle",
  "(",
  "[",
  "1",
  ",",
  "2",
  ",",
  "3",
  ",",
  "4",
  "]",
  ")",
  "for",
  "line",
  "in",
  "fq",
  ":",
  "n",
  "=",
  "next",
  "(",
  "c",
  ")",
  "if",
  "n",
  "==",
  "1",
  ":",
  "seq",
  "=",
  "[",
  "'>%s'",
  "%",
  "(",
  "line",
  ".",
  "strip",
  "(",
  ")",
  ".",
  "split",
  "(",
  "'@'",
  ",",
  "1",
  ")",
  "[",
  "1",
  "]",
  ")",
  "]",
  "if",
  "n",
  "==",
  "2",
  ":",
  "seq",
  ".",
  "append",
  "(",
  "line",
  ".",
  "strip",
  "(",
  ")",
  ")",
  "yield",
  "seq"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">convert fq to fa</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "convert",
  "fq",
  "to",
  "fa"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">83b2566b3a5745437ec651cd6cafddd056846240</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/fastq2fasta.py#L11-L22</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">31</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="32"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">elbow-jason/Uno-deprecated</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">uno/decorators.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">change_return_type</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def change_return_type(f):
    """
    Converts the returned value of wrapped function to the type of the
    first arg or to the type specified by a kwarg key return_type's value.
    """
    @wraps(f)
    def wrapper(*args, **kwargs):
        if kwargs.has_key('return_type'):
            return_type = kwargs['return_type']
            kwargs.pop('return_type')
            return return_type(f(*args, **kwargs))
        elif len(args) > 0:
            return_type = type(args[0])
            return return_type(f(*args, **kwargs))
        else:
            return f(*args, **kwargs)
    return wrapper</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def change_return_type(f):
    """
    Converts the returned value of wrapped function to the type of the
    first arg or to the type specified by a kwarg key return_type's value.
    """
    @wraps(f)
    def wrapper(*args, **kwargs):
        if kwargs.has_key('return_type'):
            return_type = kwargs['return_type']
            kwargs.pop('return_type')
            return return_type(f(*args, **kwargs))
        elif len(args) > 0:
            return_type = type(args[0])
            return return_type(f(*args, **kwargs))
        else:
            return f(*args, **kwargs)
    return wrapper</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "change_return_type",
  "(",
  "f",
  ")",
  ":",
  "@",
  "wraps",
  "(",
  "f",
  ")",
  "def",
  "wrapper",
  "(",
  "*",
  "args",
  ",",
  "*",
  "*",
  "kwargs",
  ")",
  ":",
  "if",
  "kwargs",
  ".",
  "has_key",
  "(",
  "'return_type'",
  ")",
  ":",
  "return_type",
  "=",
  "kwargs",
  "[",
  "'return_type'",
  "]",
  "kwargs",
  ".",
  "pop",
  "(",
  "'return_type'",
  ")",
  "return",
  "return_type",
  "(",
  "f",
  "(",
  "*",
  "args",
  ",",
  "*",
  "*",
  "kwargs",
  ")",
  ")",
  "elif",
  "len",
  "(",
  "args",
  ")",
  ">",
  "0",
  ":",
  "return_type",
  "=",
  "type",
  "(",
  "args",
  "[",
  "0",
  "]",
  ")",
  "return",
  "return_type",
  "(",
  "f",
  "(",
  "*",
  "args",
  ",",
  "*",
  "*",
  "kwargs",
  ")",
  ")",
  "else",
  ":",
  "return",
  "f",
  "(",
  "*",
  "args",
  ",",
  "*",
  "*",
  "kwargs",
  ")",
  "return",
  "wrapper"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Converts the returned value of wrapped function to the type of the
    first arg or to the type specified by a kwarg key return_type's value.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Converts",
  "the",
  "returned",
  "value",
  "of",
  "wrapped",
  "function",
  "to",
  "the",
  "type",
  "of",
  "the",
  "first",
  "arg",
  "or",
  "to",
  "the",
  "type",
  "specified",
  "by",
  "a",
  "kwarg",
  "key",
  "return_type",
  "s",
  "value",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">4ad07d7b84e5b6e3e2b2c89db69448906f24b4e4</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/elbow-jason/Uno-deprecated/blob/4ad07d7b84e5b6e3e2b2c89db69448906f24b4e4/uno/decorators.py#L11-L27</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">32</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="33"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">elbow-jason/Uno-deprecated</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">uno/decorators.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">convert_args_to_sets</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def convert_args_to_sets(f):
    """
    Converts all args to 'set' type via self.setify function.
    """
    @wraps(f)
    def wrapper(*args, **kwargs):
        args = (setify(x) for x in args)
        return f(*args, **kwargs)
    return wrapper</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def convert_args_to_sets(f):
    """
    Converts all args to 'set' type via self.setify function.
    """
    @wraps(f)
    def wrapper(*args, **kwargs):
        args = (setify(x) for x in args)
        return f(*args, **kwargs)
    return wrapper</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "convert_args_to_sets",
  "(",
  "f",
  ")",
  ":",
  "@",
  "wraps",
  "(",
  "f",
  ")",
  "def",
  "wrapper",
  "(",
  "*",
  "args",
  ",",
  "*",
  "*",
  "kwargs",
  ")",
  ":",
  "args",
  "=",
  "(",
  "setify",
  "(",
  "x",
  ")",
  "for",
  "x",
  "in",
  "args",
  ")",
  "return",
  "f",
  "(",
  "*",
  "args",
  ",",
  "*",
  "*",
  "kwargs",
  ")",
  "return",
  "wrapper"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Converts all args to 'set' type via self.setify function.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Converts",
  "all",
  "args",
  "to",
  "set",
  "type",
  "via",
  "self",
  ".",
  "setify",
  "function",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">4ad07d7b84e5b6e3e2b2c89db69448906f24b4e4</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/elbow-jason/Uno-deprecated/blob/4ad07d7b84e5b6e3e2b2c89db69448906f24b4e4/uno/decorators.py#L30-L38</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">33</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="34"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">laymonage/kbbi-python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">kbbi/kbbi.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">KBBI._init_entri</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _init_entri(self, laman):
        """Membuat objek-objek entri dari laman yang diambil.

        :param laman: Laman respons yang dikembalikan oleh KBBI daring.
        :type laman: Response
        """

        sup = BeautifulSoup(laman.text, 'html.parser')
        estr = ''
        for label in sup.find('hr').next_siblings:
            if label.name == 'hr':
                self.entri.append(Entri(estr))
                break
            if label.name == 'h2':
                if estr:
                    self.entri.append(Entri(estr))
                estr = ''
            estr += str(label).strip()</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _init_entri(self, laman):
        """Membuat objek-objek entri dari laman yang diambil.

        :param laman: Laman respons yang dikembalikan oleh KBBI daring.
        :type laman: Response
        """

        sup = BeautifulSoup(laman.text, 'html.parser')
        estr = ''
        for label in sup.find('hr').next_siblings:
            if label.name == 'hr':
                self.entri.append(Entri(estr))
                break
            if label.name == 'h2':
                if estr:
                    self.entri.append(Entri(estr))
                estr = ''
            estr += str(label).strip()</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "_init_entri",
  "(",
  "self",
  ",",
  "laman",
  ")",
  ":",
  "sup",
  "=",
  "BeautifulSoup",
  "(",
  "laman",
  ".",
  "text",
  ",",
  "'html.parser'",
  ")",
  "estr",
  "=",
  "''",
  "for",
  "label",
  "in",
  "sup",
  ".",
  "find",
  "(",
  "'hr'",
  ")",
  ".",
  "next_siblings",
  ":",
  "if",
  "label",
  ".",
  "name",
  "==",
  "'hr'",
  ":",
  "self",
  ".",
  "entri",
  ".",
  "append",
  "(",
  "Entri",
  "(",
  "estr",
  ")",
  ")",
  "break",
  "if",
  "label",
  ".",
  "name",
  "==",
  "'h2'",
  ":",
  "if",
  "estr",
  ":",
  "self",
  ".",
  "entri",
  ".",
  "append",
  "(",
  "Entri",
  "(",
  "estr",
  ")",
  ")",
  "estr",
  "=",
  "''",
  "estr",
  "+=",
  "str",
  "(",
  "label",
  ")",
  ".",
  "strip",
  "(",
  ")"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Membuat objek-objek entri dari laman yang diambil.

        :param laman: Laman respons yang dikembalikan oleh KBBI daring.
        :type laman: Response</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Membuat",
  "objek",
  "-",
  "objek",
  "entri",
  "dari",
  "laman",
  "yang",
  "diambil",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">1a52ba8bcc6dc4c5c1215f9e00207aca264287d6</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L46-L63</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">34</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="35"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">laymonage/kbbi-python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">kbbi/kbbi.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Entri._init_kata_dasar</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _init_kata_dasar(self, dasar):
        """Memproses kata dasar yang ada dalam nama entri.

        :param dasar: ResultSet untuk label HTML dengan class="rootword"
        :type dasar: ResultSet
        """

        for tiap in dasar:
            kata = tiap.find('a')
            dasar_no = kata.find('sup')
            kata = ambil_teks_dalam_label(kata)
            self.kata_dasar.append(
                kata + ' [{}]'.format(dasar_no.text.strip()) if dasar_no else kata
            )</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _init_kata_dasar(self, dasar):
        """Memproses kata dasar yang ada dalam nama entri.

        :param dasar: ResultSet untuk label HTML dengan class="rootword"
        :type dasar: ResultSet
        """

        for tiap in dasar:
            kata = tiap.find('a')
            dasar_no = kata.find('sup')
            kata = ambil_teks_dalam_label(kata)
            self.kata_dasar.append(
                kata + ' [{}]'.format(dasar_no.text.strip()) if dasar_no else kata
            )</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "_init_kata_dasar",
  "(",
  "self",
  ",",
  "dasar",
  ")",
  ":",
  "for",
  "tiap",
  "in",
  "dasar",
  ":",
  "kata",
  "=",
  "tiap",
  ".",
  "find",
  "(",
  "'a'",
  ")",
  "dasar_no",
  "=",
  "kata",
  ".",
  "find",
  "(",
  "'sup'",
  ")",
  "kata",
  "=",
  "ambil_teks_dalam_label",
  "(",
  "kata",
  ")",
  "self",
  ".",
  "kata_dasar",
  ".",
  "append",
  "(",
  "kata",
  "+",
  "' [{}]'",
  ".",
  "format",
  "(",
  "dasar_no",
  ".",
  "text",
  ".",
  "strip",
  "(",
  ")",
  ")",
  "if",
  "dasar_no",
  "else",
  "kata",
  ")"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Memproses kata dasar yang ada dalam nama entri.

        :param dasar: ResultSet untuk label HTML dengan class="rootword"
        :type dasar: ResultSet</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Memproses",
  "kata",
  "dasar",
  "yang",
  "ada",
  "dalam",
  "nama",
  "entri",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">1a52ba8bcc6dc4c5c1215f9e00207aca264287d6</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L126-L139</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">35</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="36"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">laymonage/kbbi-python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">kbbi/kbbi.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Entri.serialisasi</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def serialisasi(self):
        """Mengembalikan hasil serialisasi objek Entri ini.

        :returns: Dictionary hasil serialisasi
        :rtype: dict
        """

        return {
            "nama": self.nama,
            "nomor": self.nomor,
            "kata_dasar": self.kata_dasar,
            "pelafalan": self.pelafalan,
            "bentuk_tidak_baku": self.bentuk_tidak_baku,
            "varian": self.varian,
            "makna": [makna.serialisasi() for makna in self.makna]
        }</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def serialisasi(self):
        """Mengembalikan hasil serialisasi objek Entri ini.

        :returns: Dictionary hasil serialisasi
        :rtype: dict
        """

        return {
            "nama": self.nama,
            "nomor": self.nomor,
            "kata_dasar": self.kata_dasar,
            "pelafalan": self.pelafalan,
            "bentuk_tidak_baku": self.bentuk_tidak_baku,
            "varian": self.varian,
            "makna": [makna.serialisasi() for makna in self.makna]
        }</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "serialisasi",
  "(",
  "self",
  ")",
  ":",
  "return",
  "{",
  "\"nama\"",
  ":",
  "self",
  ".",
  "nama",
  ",",
  "\"nomor\"",
  ":",
  "self",
  ".",
  "nomor",
  ",",
  "\"kata_dasar\"",
  ":",
  "self",
  ".",
  "kata_dasar",
  ",",
  "\"pelafalan\"",
  ":",
  "self",
  ".",
  "pelafalan",
  ",",
  "\"bentuk_tidak_baku\"",
  ":",
  "self",
  ".",
  "bentuk_tidak_baku",
  ",",
  "\"varian\"",
  ":",
  "self",
  ".",
  "varian",
  ",",
  "\"makna\"",
  ":",
  "[",
  "makna",
  ".",
  "serialisasi",
  "(",
  ")",
  "for",
  "makna",
  "in",
  "self",
  ".",
  "makna",
  "]",
  "}"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Mengembalikan hasil serialisasi objek Entri ini.

        :returns: Dictionary hasil serialisasi
        :rtype: dict</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Mengembalikan",
  "hasil",
  "serialisasi",
  "objek",
  "Entri",
  "ini",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">1a52ba8bcc6dc4c5c1215f9e00207aca264287d6</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L141-L156</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">36</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="37"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">laymonage/kbbi-python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">kbbi/kbbi.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Entri._makna</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _makna(self):
        """Mengembalikan representasi string untuk semua makna entri ini.

        :returns: String representasi makna-makna
        :rtype: str
        """

        if len(self.makna) > 1:
            return '\n'.join(
                str(i) + ". " + str(makna)
                for i, makna in enumerate(self.makna, 1)
            )
        return str(self.makna[0])</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _makna(self):
        """Mengembalikan representasi string untuk semua makna entri ini.

        :returns: String representasi makna-makna
        :rtype: str
        """

        if len(self.makna) > 1:
            return '\n'.join(
                str(i) + ". " + str(makna)
                for i, makna in enumerate(self.makna, 1)
            )
        return str(self.makna[0])</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "_makna",
  "(",
  "self",
  ")",
  ":",
  "if",
  "len",
  "(",
  "self",
  ".",
  "makna",
  ")",
  ">",
  "1",
  ":",
  "return",
  "'\\n'",
  ".",
  "join",
  "(",
  "str",
  "(",
  "i",
  ")",
  "+",
  "\". \"",
  "+",
  "str",
  "(",
  "makna",
  ")",
  "for",
  "i",
  ",",
  "makna",
  "in",
  "enumerate",
  "(",
  "self",
  ".",
  "makna",
  ",",
  "1",
  ")",
  ")",
  "return",
  "str",
  "(",
  "self",
  ".",
  "makna",
  "[",
  "0",
  "]",
  ")"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Mengembalikan representasi string untuk semua makna entri ini.

        :returns: String representasi makna-makna
        :rtype: str</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Mengembalikan",
  "representasi",
  "string",
  "untuk",
  "semua",
  "makna",
  "entri",
  "ini",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">1a52ba8bcc6dc4c5c1215f9e00207aca264287d6</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L158-L170</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">37</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="38"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">laymonage/kbbi-python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">kbbi/kbbi.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Entri._nama</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _nama(self):
        """Mengembalikan representasi string untuk nama entri ini.

        :returns: String representasi nama entri
        :rtype: str
        """

        hasil = self.nama
        if self.nomor:
            hasil += " [{}]".format(self.nomor)
        if self.kata_dasar:
            hasil = " Â» ".join(self.kata_dasar) + " Â» " + hasil
        return hasil</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _nama(self):
        """Mengembalikan representasi string untuk nama entri ini.

        :returns: String representasi nama entri
        :rtype: str
        """

        hasil = self.nama
        if self.nomor:
            hasil += " [{}]".format(self.nomor)
        if self.kata_dasar:
            hasil = " Â» ".join(self.kata_dasar) + " Â» " + hasil
        return hasil</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "_nama",
  "(",
  "self",
  ")",
  ":",
  "hasil",
  "=",
  "self",
  ".",
  "nama",
  "if",
  "self",
  ".",
  "nomor",
  ":",
  "hasil",
  "+=",
  "\" [{}]\"",
  ".",
  "format",
  "(",
  "self",
  ".",
  "nomor",
  ")",
  "if",
  "self",
  ".",
  "kata_dasar",
  ":",
  "hasil",
  "=",
  "\" Â» \".",
  "j",
  "oin(",
  "s",
  "elf.",
  "k",
  "ata_dasar)",
  " ",
  " ",
  " Â» \" +",
  "h",
  "sil",
  "return",
  "hasil"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Mengembalikan representasi string untuk nama entri ini.

        :returns: String representasi nama entri
        :rtype: str</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Mengembalikan",
  "representasi",
  "string",
  "untuk",
  "nama",
  "entri",
  "ini",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">1a52ba8bcc6dc4c5c1215f9e00207aca264287d6</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L172-L184</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">38</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="39"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">laymonage/kbbi-python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">kbbi/kbbi.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Entri._varian</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _varian(self, varian):
        """Mengembalikan representasi string untuk varian entri ini.
        Dapat digunakan untuk "Varian" maupun "Bentuk tidak baku".

        :param varian: List bentuk tidak baku atau varian
        :type varian: list
        :returns: String representasi varian atau bentuk tidak baku
        :rtype: str
        """

        if varian == self.bentuk_tidak_baku:
            nama = "Bentuk tidak baku"
        elif varian == self.varian:
            nama = "Varian"
        else:
            return ''
        return nama + ': ' + ', '.join(varian)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _varian(self, varian):
        """Mengembalikan representasi string untuk varian entri ini.
        Dapat digunakan untuk "Varian" maupun "Bentuk tidak baku".

        :param varian: List bentuk tidak baku atau varian
        :type varian: list
        :returns: String representasi varian atau bentuk tidak baku
        :rtype: str
        """

        if varian == self.bentuk_tidak_baku:
            nama = "Bentuk tidak baku"
        elif varian == self.varian:
            nama = "Varian"
        else:
            return ''
        return nama + ': ' + ', '.join(varian)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "_varian",
  "(",
  "self",
  ",",
  "varian",
  ")",
  ":",
  "if",
  "varian",
  "==",
  "self",
  ".",
  "bentuk_tidak_baku",
  ":",
  "nama",
  "=",
  "\"Bentuk tidak baku\"",
  "elif",
  "varian",
  "==",
  "self",
  ".",
  "varian",
  ":",
  "nama",
  "=",
  "\"Varian\"",
  "else",
  ":",
  "return",
  "''",
  "return",
  "nama",
  "+",
  "': '",
  "+",
  "', '",
  ".",
  "join",
  "(",
  "varian",
  ")"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Mengembalikan representasi string untuk varian entri ini.
        Dapat digunakan untuk "Varian" maupun "Bentuk tidak baku".

        :param varian: List bentuk tidak baku atau varian
        :type varian: list
        :returns: String representasi varian atau bentuk tidak baku
        :rtype: str</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Mengembalikan",
  "representasi",
  "string",
  "untuk",
  "varian",
  "entri",
  "ini",
  ".",
  "Dapat",
  "digunakan",
  "untuk",
  "Varian",
  "maupun",
  "Bentuk",
  "tidak",
  "baku",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">1a52ba8bcc6dc4c5c1215f9e00207aca264287d6</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L186-L202</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">39</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="40"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">laymonage/kbbi-python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">kbbi/kbbi.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Makna._init_kelas</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _init_kelas(self, makna_label):
        """Memproses kelas kata yang ada dalam makna.

        :param makna_label: BeautifulSoup untuk makna yang ingin diproses.
        :type makna_label: BeautifulSoup
        """

        kelas = makna_label.find(color='red')
        lain = makna_label.find(color='darkgreen')
        info = makna_label.find(color='green')
        if kelas:
            kelas = kelas.find_all('span')
        if lain:
            self.kelas = {lain.text.strip(): lain['title'].strip()}
            self.submakna = lain.next_sibling.strip()
            self.submakna += ' ' + makna_label.find(color='grey').text.strip()
        else:
            self.kelas = {
                k.text.strip(): k['title'].strip() for k in kelas
            } if kelas else {}
        self.info = info.text.strip() if info else ''</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _init_kelas(self, makna_label):
        """Memproses kelas kata yang ada dalam makna.

        :param makna_label: BeautifulSoup untuk makna yang ingin diproses.
        :type makna_label: BeautifulSoup
        """

        kelas = makna_label.find(color='red')
        lain = makna_label.find(color='darkgreen')
        info = makna_label.find(color='green')
        if kelas:
            kelas = kelas.find_all('span')
        if lain:
            self.kelas = {lain.text.strip(): lain['title'].strip()}
            self.submakna = lain.next_sibling.strip()
            self.submakna += ' ' + makna_label.find(color='grey').text.strip()
        else:
            self.kelas = {
                k.text.strip(): k['title'].strip() for k in kelas
            } if kelas else {}
        self.info = info.text.strip() if info else ''</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "_init_kelas",
  "(",
  "self",
  ",",
  "makna_label",
  ")",
  ":",
  "kelas",
  "=",
  "makna_label",
  ".",
  "find",
  "(",
  "color",
  "=",
  "'red'",
  ")",
  "lain",
  "=",
  "makna_label",
  ".",
  "find",
  "(",
  "color",
  "=",
  "'darkgreen'",
  ")",
  "info",
  "=",
  "makna_label",
  ".",
  "find",
  "(",
  "color",
  "=",
  "'green'",
  ")",
  "if",
  "kelas",
  ":",
  "kelas",
  "=",
  "kelas",
  ".",
  "find_all",
  "(",
  "'span'",
  ")",
  "if",
  "lain",
  ":",
  "self",
  ".",
  "kelas",
  "=",
  "{",
  "lain",
  ".",
  "text",
  ".",
  "strip",
  "(",
  ")",
  ":",
  "lain",
  "[",
  "'title'",
  "]",
  ".",
  "strip",
  "(",
  ")",
  "}",
  "self",
  ".",
  "submakna",
  "=",
  "lain",
  ".",
  "next_sibling",
  ".",
  "strip",
  "(",
  ")",
  "self",
  ".",
  "submakna",
  "+=",
  "' '",
  "+",
  "makna_label",
  ".",
  "find",
  "(",
  "color",
  "=",
  "'grey'",
  ")",
  ".",
  "text",
  ".",
  "strip",
  "(",
  ")",
  "else",
  ":",
  "self",
  ".",
  "kelas",
  "=",
  "{",
  "k",
  ".",
  "text",
  ".",
  "strip",
  "(",
  ")",
  ":",
  "k",
  "[",
  "'title'",
  "]",
  ".",
  "strip",
  "(",
  ")",
  "for",
  "k",
  "in",
  "kelas",
  "}",
  "if",
  "kelas",
  "else",
  "{",
  "}",
  "self",
  ".",
  "info",
  "=",
  "info",
  ".",
  "text",
  ".",
  "strip",
  "(",
  ")",
  "if",
  "info",
  "else",
  "''"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Memproses kelas kata yang ada dalam makna.

        :param makna_label: BeautifulSoup untuk makna yang ingin diproses.
        :type makna_label: BeautifulSoup</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Memproses",
  "kelas",
  "kata",
  "yang",
  "ada",
  "dalam",
  "makna",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">1a52ba8bcc6dc4c5c1215f9e00207aca264287d6</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L239-L259</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">40</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="41"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">laymonage/kbbi-python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">kbbi/kbbi.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Makna._init_contoh</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _init_contoh(self, makna_label):
        """Memproses contoh yang ada dalam makna.

        :param makna_label: BeautifulSoup untuk makna yang ingin diproses.
        :type makna_label: BeautifulSoup
        """

        indeks = makna_label.text.find(': ')
        if indeks != -1:
            contoh = makna_label.text[indeks + 2:].strip()
            self.contoh = contoh.split('; ')
        else:
            self.contoh = []</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def _init_contoh(self, makna_label):
        """Memproses contoh yang ada dalam makna.

        :param makna_label: BeautifulSoup untuk makna yang ingin diproses.
        :type makna_label: BeautifulSoup
        """

        indeks = makna_label.text.find(': ')
        if indeks != -1:
            contoh = makna_label.text[indeks + 2:].strip()
            self.contoh = contoh.split('; ')
        else:
            self.contoh = []</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "_init_contoh",
  "(",
  "self",
  ",",
  "makna_label",
  ")",
  ":",
  "indeks",
  "=",
  "makna_label",
  ".",
  "text",
  ".",
  "find",
  "(",
  "': '",
  ")",
  "if",
  "indeks",
  "!=",
  "-",
  "1",
  ":",
  "contoh",
  "=",
  "makna_label",
  ".",
  "text",
  "[",
  "indeks",
  "+",
  "2",
  ":",
  "]",
  ".",
  "strip",
  "(",
  ")",
  "self",
  ".",
  "contoh",
  "=",
  "contoh",
  ".",
  "split",
  "(",
  "'; '",
  ")",
  "else",
  ":",
  "self",
  ".",
  "contoh",
  "=",
  "[",
  "]"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Memproses contoh yang ada dalam makna.

        :param makna_label: BeautifulSoup untuk makna yang ingin diproses.
        :type makna_label: BeautifulSoup</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Memproses",
  "contoh",
  "yang",
  "ada",
  "dalam",
  "makna",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">1a52ba8bcc6dc4c5c1215f9e00207aca264287d6</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L261-L273</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">41</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="42"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">laymonage/kbbi-python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">kbbi/kbbi.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Makna.serialisasi</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def serialisasi(self):
        """Mengembalikan hasil serialisasi objek Makna ini.

        :returns: Dictionary hasil serialisasi
        :rtype: dict
        """

        return {
            "kelas": self.kelas,
            "submakna": self.submakna,
            "info": self.info,
            "contoh": self.contoh
        }</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def serialisasi(self):
        """Mengembalikan hasil serialisasi objek Makna ini.

        :returns: Dictionary hasil serialisasi
        :rtype: dict
        """

        return {
            "kelas": self.kelas,
            "submakna": self.submakna,
            "info": self.info,
            "contoh": self.contoh
        }</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "serialisasi",
  "(",
  "self",
  ")",
  ":",
  "return",
  "{",
  "\"kelas\"",
  ":",
  "self",
  ".",
  "kelas",
  ",",
  "\"submakna\"",
  ":",
  "self",
  ".",
  "submakna",
  ",",
  "\"info\"",
  ":",
  "self",
  ".",
  "info",
  ",",
  "\"contoh\"",
  ":",
  "self",
  ".",
  "contoh",
  "}"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Mengembalikan hasil serialisasi objek Makna ini.

        :returns: Dictionary hasil serialisasi
        :rtype: dict</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Mengembalikan",
  "hasil",
  "serialisasi",
  "objek",
  "Makna",
  "ini",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">1a52ba8bcc6dc4c5c1215f9e00207aca264287d6</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/laymonage/kbbi-python/blob/1a52ba8bcc6dc4c5c1215f9e00207aca264287d6/kbbi/kbbi.py#L275-L287</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">42</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="43"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">mkouhei/bootstrap-py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">bootstrap_py/docs.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">build_sphinx</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def build_sphinx(pkg_data, projectdir):
    """Build sphinx documentation.

    :rtype: int
    :return: subprocess.call return code

    :param `bootstrap_py.control.PackageData` pkg_data: package meta data
    :param str projectdir: project root directory
    """
    try:
        version, _minor_version = pkg_data.version.rsplit('.', 1)
    except ValueError:
        version = pkg_data.version
    args = ' '.join(('sphinx-quickstart',
                     '--sep',
                     '-q',
                     '-p "{name}"',
                     '-a "{author}"',
                     '-v "{version}"',
                     '-r "{release}"',
                     '-l en',
                     '--suffix=.rst',
                     '--master=index',
                     '--ext-autodoc',
                     '--ext-viewcode',
                     '--makefile',
                     '{projectdir}')).format(name=pkg_data.name,
                                             author=pkg_data.author,
                                             version=version,
                                             release=pkg_data.version,
                                             projectdir=projectdir)
    if subprocess.call(shlex.split(args)) == 0:
        _touch_gitkeep(projectdir)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def build_sphinx(pkg_data, projectdir):
    """Build sphinx documentation.

    :rtype: int
    :return: subprocess.call return code

    :param `bootstrap_py.control.PackageData` pkg_data: package meta data
    :param str projectdir: project root directory
    """
    try:
        version, _minor_version = pkg_data.version.rsplit('.', 1)
    except ValueError:
        version = pkg_data.version
    args = ' '.join(('sphinx-quickstart',
                     '--sep',
                     '-q',
                     '-p "{name}"',
                     '-a "{author}"',
                     '-v "{version}"',
                     '-r "{release}"',
                     '-l en',
                     '--suffix=.rst',
                     '--master=index',
                     '--ext-autodoc',
                     '--ext-viewcode',
                     '--makefile',
                     '{projectdir}')).format(name=pkg_data.name,
                                             author=pkg_data.author,
                                             version=version,
                                             release=pkg_data.version,
                                             projectdir=projectdir)
    if subprocess.call(shlex.split(args)) == 0:
        _touch_gitkeep(projectdir)</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "build_sphinx",
  "(",
  "pkg_data",
  ",",
  "projectdir",
  ")",
  ":",
  "try",
  ":",
  "version",
  ",",
  "_minor_version",
  "=",
  "pkg_data",
  ".",
  "version",
  ".",
  "rsplit",
  "(",
  "'.'",
  ",",
  "1",
  ")",
  "except",
  "ValueError",
  ":",
  "version",
  "=",
  "pkg_data",
  ".",
  "version",
  "args",
  "=",
  "' '",
  ".",
  "join",
  "(",
  "(",
  "'sphinx-quickstart'",
  ",",
  "'--sep'",
  ",",
  "'-q'",
  ",",
  "'-p \"{name}\"'",
  ",",
  "'-a \"{author}\"'",
  ",",
  "'-v \"{version}\"'",
  ",",
  "'-r \"{release}\"'",
  ",",
  "'-l en'",
  ",",
  "'--suffix=.rst'",
  ",",
  "'--master=index'",
  ",",
  "'--ext-autodoc'",
  ",",
  "'--ext-viewcode'",
  ",",
  "'--makefile'",
  ",",
  "'{projectdir}'",
  ")",
  ")",
  ".",
  "format",
  "(",
  "name",
  "=",
  "pkg_data",
  ".",
  "name",
  ",",
  "author",
  "=",
  "pkg_data",
  ".",
  "author",
  ",",
  "version",
  "=",
  "version",
  ",",
  "release",
  "=",
  "pkg_data",
  ".",
  "version",
  ",",
  "projectdir",
  "=",
  "projectdir",
  ")",
  "if",
  "subprocess",
  ".",
  "call",
  "(",
  "shlex",
  ".",
  "split",
  "(",
  "args",
  ")",
  ")",
  "==",
  "0",
  ":",
  "_touch_gitkeep",
  "(",
  "projectdir",
  ")"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Build sphinx documentation.

    :rtype: int
    :return: subprocess.call return code

    :param `bootstrap_py.control.PackageData` pkg_data: package meta data
    :param str projectdir: project root directory</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Build",
  "sphinx",
  "documentation",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">95d56ed98ef409fd9f019dc352fd1c3711533275</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/mkouhei/bootstrap-py/blob/95d56ed98ef409fd9f019dc352fd1c3711533275/bootstrap_py/docs.py#L8-L40</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">43</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="44"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">christophertbrown/bioscripts</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ctbBio/crossmap.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">bowtiedb</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def bowtiedb(fa, keepDB):
    """
    make bowtie db
    """
    btdir = '%s/bt2' % (os.getcwd())
    # make directory for
    if not os.path.exists(btdir):
        os.mkdir(btdir)
    btdb = '%s/%s' % (btdir, fa.rsplit('/', 1)[-1])
    if keepDB is True:
        if os.path.exists('%s.1.bt2' % (btdb)):
            return btdb
    p = subprocess.Popen('bowtie2-build -q %s %s' \
        % (fa, btdb), shell = True)
    p.communicate()
    return btdb</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def bowtiedb(fa, keepDB):
    """
    make bowtie db
    """
    btdir = '%s/bt2' % (os.getcwd())
    # make directory for
    if not os.path.exists(btdir):
        os.mkdir(btdir)
    btdb = '%s/%s' % (btdir, fa.rsplit('/', 1)[-1])
    if keepDB is True:
        if os.path.exists('%s.1.bt2' % (btdb)):
            return btdb
    p = subprocess.Popen('bowtie2-build -q %s %s' \
        % (fa, btdb), shell = True)
    p.communicate()
    return btdb</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "bowtiedb",
  "(",
  "fa",
  ",",
  "keepDB",
  ")",
  ":",
  "btdir",
  "=",
  "'%s/bt2'",
  "%",
  "(",
  "os",
  ".",
  "getcwd",
  "(",
  ")",
  ")",
  "# make directory for",
  "if",
  "not",
  "os",
  ".",
  "path",
  ".",
  "exists",
  "(",
  "btdir",
  ")",
  ":",
  "os",
  ".",
  "mkdir",
  "(",
  "btdir",
  ")",
  "btdb",
  "=",
  "'%s/%s'",
  "%",
  "(",
  "btdir",
  ",",
  "fa",
  ".",
  "rsplit",
  "(",
  "'/'",
  ",",
  "1",
  ")",
  "[",
  "-",
  "1",
  "]",
  ")",
  "if",
  "keepDB",
  "is",
  "True",
  ":",
  "if",
  "os",
  ".",
  "path",
  ".",
  "exists",
  "(",
  "'%s.1.bt2'",
  "%",
  "(",
  "btdb",
  ")",
  ")",
  ":",
  "return",
  "btdb",
  "p",
  "=",
  "subprocess",
  ".",
  "Popen",
  "(",
  "'bowtie2-build -q %s %s'",
  "%",
  "(",
  "fa",
  ",",
  "btdb",
  ")",
  ",",
  "shell",
  "=",
  "True",
  ")",
  "p",
  ".",
  "communicate",
  "(",
  ")",
  "return",
  "btdb"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">make bowtie db</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "make",
  "bowtie",
  "db"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">83b2566b3a5745437ec651cd6cafddd056846240</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/crossmap.py#L16-L31</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">44</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="45"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">christophertbrown/bioscripts</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ctbBio/crossmap.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">bowtie</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def bowtie(sam, btd, f, r, u, opt, no_shrink, threads):
    """
    generate bowtie2 command
    """
    bt2 = 'bowtie2 -x %s -p %s ' % (btd, threads)
    if f is not False:
        bt2 += '-1 %s -2 %s ' % (f, r)
    if u is not False:
        bt2 += '-U %s ' % (u)
    bt2 += opt
    if no_shrink is False:
        if f is False:
            bt2 += ' | shrinksam -u -k %s-shrunk.sam ' % (sam)
        else:
            bt2 += ' | shrinksam -k %s-shrunk.sam ' % (sam)
    else:
        bt2 += ' > %s.sam' % (sam)
    return bt2</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def bowtie(sam, btd, f, r, u, opt, no_shrink, threads):
    """
    generate bowtie2 command
    """
    bt2 = 'bowtie2 -x %s -p %s ' % (btd, threads)
    if f is not False:
        bt2 += '-1 %s -2 %s ' % (f, r)
    if u is not False:
        bt2 += '-U %s ' % (u)
    bt2 += opt
    if no_shrink is False:
        if f is False:
            bt2 += ' | shrinksam -u -k %s-shrunk.sam ' % (sam)
        else:
            bt2 += ' | shrinksam -k %s-shrunk.sam ' % (sam)
    else:
        bt2 += ' > %s.sam' % (sam)
    return bt2</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "bowtie",
  "(",
  "sam",
  ",",
  "btd",
  ",",
  "f",
  ",",
  "r",
  ",",
  "u",
  ",",
  "opt",
  ",",
  "no_shrink",
  ",",
  "threads",
  ")",
  ":",
  "bt2",
  "=",
  "'bowtie2 -x %s -p %s '",
  "%",
  "(",
  "btd",
  ",",
  "threads",
  ")",
  "if",
  "f",
  "is",
  "not",
  "False",
  ":",
  "bt2",
  "+=",
  "'-1 %s -2 %s '",
  "%",
  "(",
  "f",
  ",",
  "r",
  ")",
  "if",
  "u",
  "is",
  "not",
  "False",
  ":",
  "bt2",
  "+=",
  "'-U %s '",
  "%",
  "(",
  "u",
  ")",
  "bt2",
  "+=",
  "opt",
  "if",
  "no_shrink",
  "is",
  "False",
  ":",
  "if",
  "f",
  "is",
  "False",
  ":",
  "bt2",
  "+=",
  "' | shrinksam -u -k %s-shrunk.sam '",
  "%",
  "(",
  "sam",
  ")",
  "else",
  ":",
  "bt2",
  "+=",
  "' | shrinksam -k %s-shrunk.sam '",
  "%",
  "(",
  "sam",
  ")",
  "else",
  ":",
  "bt2",
  "+=",
  "' > %s.sam'",
  "%",
  "(",
  "sam",
  ")",
  "return",
  "bt2"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">generate bowtie2 command</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "generate",
  "bowtie2",
  "command"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">83b2566b3a5745437ec651cd6cafddd056846240</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/crossmap.py#L33-L50</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">45</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="46"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">christophertbrown/bioscripts</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ctbBio/crossmap.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">crossmap</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def crossmap(fas, reads, options, no_shrink, keepDB, threads, cluster, nodes):
    """
    map all read sets against all fasta files
    """
    if cluster is True:
        threads = '48'
    btc = []
    for fa in fas:
        btd = bowtiedb(fa, keepDB)
        F, R, U = reads
        if F is not False:
            if U is False:
                u = False
            for i, f in enumerate(F):
                r = R[i]
                if U is not False:
                    u = U[i]
                sam = '%s/%s-vs-%s' % (os.getcwd(), \
                        fa.rsplit('/', 1)[-1], f.rsplit('/', 1)[-1].rsplit('.', 3)[0])
                btc.append(bowtie(sam, btd, f, r, u, options, no_shrink, threads))
        else:
            f = False
            r = False
            for u in U:
                sam = '%s/%s-vs-%s' % (os.getcwd(), \
                        fa.rsplit('/', 1)[-1], u.rsplit('/', 1)[-1].rsplit('.', 3)[0])
                btc.append(bowtie(sam, btd, f, r, u, options, no_shrink, threads))
    if cluster is False:
        for i in btc:
            p = subprocess.Popen(i, shell = True)
            p.communicate()
    else:
        ID = ''.join(random.choice([str(i) for i in range(0, 9)]) for _ in range(5))
        for node, commands in enumerate(chunks(btc, nodes), 1):
            bs = open('%s/crossmap-qsub.%s.%s.sh' % (os.getcwd(), ID, node), 'w')
            print('\n'.join(commands), file=bs)
            bs.close()
            p = subprocess.Popen(\
                    'qsub -V -N crossmap %s' \
                        % (bs.name), \
                    shell = True)
            p.communicate()</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def crossmap(fas, reads, options, no_shrink, keepDB, threads, cluster, nodes):
    """
    map all read sets against all fasta files
    """
    if cluster is True:
        threads = '48'
    btc = []
    for fa in fas:
        btd = bowtiedb(fa, keepDB)
        F, R, U = reads
        if F is not False:
            if U is False:
                u = False
            for i, f in enumerate(F):
                r = R[i]
                if U is not False:
                    u = U[i]
                sam = '%s/%s-vs-%s' % (os.getcwd(), \
                        fa.rsplit('/', 1)[-1], f.rsplit('/', 1)[-1].rsplit('.', 3)[0])
                btc.append(bowtie(sam, btd, f, r, u, options, no_shrink, threads))
        else:
            f = False
            r = False
            for u in U:
                sam = '%s/%s-vs-%s' % (os.getcwd(), \
                        fa.rsplit('/', 1)[-1], u.rsplit('/', 1)[-1].rsplit('.', 3)[0])
                btc.append(bowtie(sam, btd, f, r, u, options, no_shrink, threads))
    if cluster is False:
        for i in btc:
            p = subprocess.Popen(i, shell = True)
            p.communicate()
    else:
        ID = ''.join(random.choice([str(i) for i in range(0, 9)]) for _ in range(5))
        for node, commands in enumerate(chunks(btc, nodes), 1):
            bs = open('%s/crossmap-qsub.%s.%s.sh' % (os.getcwd(), ID, node), 'w')
            print('\n'.join(commands), file=bs)
            bs.close()
            p = subprocess.Popen(\
                    'qsub -V -N crossmap %s' \
                        % (bs.name), \
                    shell = True)
            p.communicate()</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "crossmap",
  "(",
  "fas",
  ",",
  "reads",
  ",",
  "options",
  ",",
  "no_shrink",
  ",",
  "keepDB",
  ",",
  "threads",
  ",",
  "cluster",
  ",",
  "nodes",
  ")",
  ":",
  "if",
  "cluster",
  "is",
  "True",
  ":",
  "threads",
  "=",
  "'48'",
  "btc",
  "=",
  "[",
  "]",
  "for",
  "fa",
  "in",
  "fas",
  ":",
  "btd",
  "=",
  "bowtiedb",
  "(",
  "fa",
  ",",
  "keepDB",
  ")",
  "F",
  ",",
  "R",
  ",",
  "U",
  "=",
  "reads",
  "if",
  "F",
  "is",
  "not",
  "False",
  ":",
  "if",
  "U",
  "is",
  "False",
  ":",
  "u",
  "=",
  "False",
  "for",
  "i",
  ",",
  "f",
  "in",
  "enumerate",
  "(",
  "F",
  ")",
  ":",
  "r",
  "=",
  "R",
  "[",
  "i",
  "]",
  "if",
  "U",
  "is",
  "not",
  "False",
  ":",
  "u",
  "=",
  "U",
  "[",
  "i",
  "]",
  "sam",
  "=",
  "'%s/%s-vs-%s'",
  "%",
  "(",
  "os",
  ".",
  "getcwd",
  "(",
  ")",
  ",",
  "fa",
  ".",
  "rsplit",
  "(",
  "'/'",
  ",",
  "1",
  ")",
  "[",
  "-",
  "1",
  "]",
  ",",
  "f",
  ".",
  "rsplit",
  "(",
  "'/'",
  ",",
  "1",
  ")",
  "[",
  "-",
  "1",
  "]",
  ".",
  "rsplit",
  "(",
  "'.'",
  ",",
  "3",
  ")",
  "[",
  "0",
  "]",
  ")",
  "btc",
  ".",
  "append",
  "(",
  "bowtie",
  "(",
  "sam",
  ",",
  "btd",
  ",",
  "f",
  ",",
  "r",
  ",",
  "u",
  ",",
  "options",
  ",",
  "no_shrink",
  ",",
  "threads",
  ")",
  ")",
  "else",
  ":",
  "f",
  "=",
  "False",
  "r",
  "=",
  "False",
  "for",
  "u",
  "in",
  "U",
  ":",
  "sam",
  "=",
  "'%s/%s-vs-%s'",
  "%",
  "(",
  "os",
  ".",
  "getcwd",
  "(",
  ")",
  ",",
  "fa",
  ".",
  "rsplit",
  "(",
  "'/'",
  ",",
  "1",
  ")",
  "[",
  "-",
  "1",
  "]",
  ",",
  "u",
  ".",
  "rsplit",
  "(",
  "'/'",
  ",",
  "1",
  ")",
  "[",
  "-",
  "1",
  "]",
  ".",
  "rsplit",
  "(",
  "'.'",
  ",",
  "3",
  ")",
  "[",
  "0",
  "]",
  ")",
  "btc",
  ".",
  "append",
  "(",
  "bowtie",
  "(",
  "sam",
  ",",
  "btd",
  ",",
  "f",
  ",",
  "r",
  ",",
  "u",
  ",",
  "options",
  ",",
  "no_shrink",
  ",",
  "threads",
  ")",
  ")",
  "if",
  "cluster",
  "is",
  "False",
  ":",
  "for",
  "i",
  "in",
  "btc",
  ":",
  "p",
  "=",
  "subprocess",
  ".",
  "Popen",
  "(",
  "i",
  ",",
  "shell",
  "=",
  "True",
  ")",
  "p",
  ".",
  "communicate",
  "(",
  ")",
  "else",
  ":",
  "ID",
  "=",
  "''",
  ".",
  "join",
  "(",
  "random",
  ".",
  "choice",
  "(",
  "[",
  "str",
  "(",
  "i",
  ")",
  "for",
  "i",
  "in",
  "range",
  "(",
  "0",
  ",",
  "9",
  ")",
  "]",
  ")",
  "for",
  "_",
  "in",
  "range",
  "(",
  "5",
  ")",
  ")",
  "for",
  "node",
  ",",
  "commands",
  "in",
  "enumerate",
  "(",
  "chunks",
  "(",
  "btc",
  ",",
  "nodes",
  ")",
  ",",
  "1",
  ")",
  ":",
  "bs",
  "=",
  "open",
  "(",
  "'%s/crossmap-qsub.%s.%s.sh'",
  "%",
  "(",
  "os",
  ".",
  "getcwd",
  "(",
  ")",
  ",",
  "ID",
  ",",
  "node",
  ")",
  ",",
  "'w'",
  ")",
  "print",
  "(",
  "'\\n'",
  ".",
  "join",
  "(",
  "commands",
  ")",
  ",",
  "file",
  "=",
  "bs",
  ")",
  "bs",
  ".",
  "close",
  "(",
  ")",
  "p",
  "=",
  "subprocess",
  ".",
  "Popen",
  "(",
  "'qsub -V -N crossmap %s'",
  "%",
  "(",
  "bs",
  ".",
  "name",
  ")",
  ",",
  "shell",
  "=",
  "True",
  ")",
  "p",
  ".",
  "communicate",
  "(",
  ")"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">map all read sets against all fasta files</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "map",
  "all",
  "read",
  "sets",
  "against",
  "all",
  "fasta",
  "files"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">83b2566b3a5745437ec651cd6cafddd056846240</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/crossmap.py#L55-L96</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">46</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="47"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">disqus/nydus</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">nydus/db/base.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">BaseCluster.get_conn</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def get_conn(self, *args, **kwargs):
        """
        Returns a connection object from the router given ``args``.

        Useful in cases where a connection cannot be automatically determined
        during all steps of the process. An example of this would be
        Redis pipelines.
        """
        connections = self.__connections_for('get_conn', args=args, kwargs=kwargs)

        if len(connections) is 1:
            return connections[0]
        else:
            return connections</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def get_conn(self, *args, **kwargs):
        """
        Returns a connection object from the router given ``args``.

        Useful in cases where a connection cannot be automatically determined
        during all steps of the process. An example of this would be
        Redis pipelines.
        """
        connections = self.__connections_for('get_conn', args=args, kwargs=kwargs)

        if len(connections) is 1:
            return connections[0]
        else:
            return connections</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "get_conn",
  "(",
  "self",
  ",",
  "*",
  "args",
  ",",
  "*",
  "*",
  "kwargs",
  ")",
  ":",
  "connections",
  "=",
  "self",
  ".",
  "__connections_for",
  "(",
  "'get_conn'",
  ",",
  "args",
  "=",
  "args",
  ",",
  "kwargs",
  "=",
  "kwargs",
  ")",
  "if",
  "len",
  "(",
  "connections",
  ")",
  "is",
  "1",
  ":",
  "return",
  "connections",
  "[",
  "0",
  "]",
  "else",
  ":",
  "return",
  "connections"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Returns a connection object from the router given ``args``.

        Useful in cases where a connection cannot be automatically determined
        during all steps of the process. An example of this would be
        Redis pipelines.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Returns",
  "a",
  "connection",
  "object",
  "from",
  "the",
  "router",
  "given",
  "args",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">9b505840da47a34f758a830c3992fa5dcb7bb7ad</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/disqus/nydus/blob/9b505840da47a34f758a830c3992fa5dcb7bb7ad/nydus/db/base.py#L100-L113</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">47</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="48"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">scottrice/pysteam</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">pysteam/_crc_algorithms.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Crc.__get_nondirect_init</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def __get_nondirect_init(self, init):
        """
        return the non-direct init if the direct algorithm has been selected.
        """
        crc = init
        for i in range(self.Width):
            bit = crc &amp; 0x01
            if bit:
                crc^= self.Poly
            crc >>= 1
            if bit:
                crc |= self.MSB_Mask
        return crc &amp; self.Mask</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def __get_nondirect_init(self, init):
        """
        return the non-direct init if the direct algorithm has been selected.
        """
        crc = init
        for i in range(self.Width):
            bit = crc &amp; 0x01
            if bit:
                crc^= self.Poly
            crc >>= 1
            if bit:
                crc |= self.MSB_Mask
        return crc &amp; self.Mask</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "__get_nondirect_init",
  "(",
  "self",
  ",",
  "init",
  ")",
  ":",
  "crc",
  "=",
  "init",
  "for",
  "i",
  "in",
  "range",
  "(",
  "self",
  ".",
  "Width",
  ")",
  ":",
  "bit",
  "=",
  "crc",
  "&amp;",
  "0x01",
  "if",
  "bit",
  ":",
  "crc",
  "^=",
  "self",
  ".",
  "Poly",
  "crc",
  ">>=",
  "1",
  "if",
  "bit",
  ":",
  "crc",
  "|=",
  "self",
  ".",
  "MSB_Mask",
  "return",
  "crc",
  "&amp;",
  "self",
  ".",
  "Mask"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">return the non-direct init if the direct algorithm has been selected.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "return",
  "the",
  "non",
  "-",
  "direct",
  "init",
  "if",
  "the",
  "direct",
  "algorithm",
  "has",
  "been",
  "selected",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">1eb2254b5235a053a953e596fa7602d0b110245d</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/scottrice/pysteam/blob/1eb2254b5235a053a953e596fa7602d0b110245d/pysteam/_crc_algorithms.py#L98-L110</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">48</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="49"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">scottrice/pysteam</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">pysteam/_crc_algorithms.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Crc.reflect</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def reflect(self, data, width):
        """
        reflect a data word, i.e. reverts the bit order.
        """
        x = data &amp; 0x01
        for i in range(width - 1):
            data >>= 1
            x = (x &lt;&lt; 1) | (data &amp; 0x01)
        return x</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def reflect(self, data, width):
        """
        reflect a data word, i.e. reverts the bit order.
        """
        x = data &amp; 0x01
        for i in range(width - 1):
            data >>= 1
            x = (x &lt;&lt; 1) | (data &amp; 0x01)
        return x</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "reflect",
  "(",
  "self",
  ",",
  "data",
  ",",
  "width",
  ")",
  ":",
  "x",
  "=",
  "data",
  "&amp;",
  "0x01",
  "for",
  "i",
  "in",
  "range",
  "(",
  "width",
  "-",
  "1",
  ")",
  ":",
  "data",
  ">>=",
  "1",
  "x",
  "=",
  "(",
  "x",
  "&lt;&lt;",
  "1",
  ")",
  "|",
  "(",
  "data",
  "&amp;",
  "0x01",
  ")",
  "return",
  "x"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">reflect a data word, i.e. reverts the bit order.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "reflect",
  "a",
  "data",
  "word",
  "i",
  ".",
  "e",
  ".",
  "reverts",
  "the",
  "bit",
  "order",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">1eb2254b5235a053a953e596fa7602d0b110245d</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/scottrice/pysteam/blob/1eb2254b5235a053a953e596fa7602d0b110245d/pysteam/_crc_algorithms.py#L115-L123</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">49</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="50"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">scottrice/pysteam</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">pysteam/_crc_algorithms.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Crc.bit_by_bit</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def bit_by_bit(self, in_data):
        """
        Classic simple and slow CRC implementation.  This function iterates bit
        by bit over the augmented input message and returns the calculated CRC
        value at the end.
        """
        # If the input data is a string, convert to bytes.
        if isinstance(in_data, str):
            in_data = [ord(c) for c in in_data]

        register = self.NonDirectInit
        for octet in in_data:
            if self.ReflectIn:
                octet = self.reflect(octet, 8)
            for i in range(8):
                topbit = register &amp; self.MSB_Mask
                register = ((register &lt;&lt; 1) &amp; self.Mask) | ((octet >> (7 - i)) &amp; 0x01)
                if topbit:
                    register ^= self.Poly

        for i in range(self.Width):
            topbit = register &amp; self.MSB_Mask
            register = ((register &lt;&lt; 1) &amp; self.Mask)
            if topbit:
                register ^= self.Poly

        if self.ReflectOut:
            register = self.reflect(register, self.Width)
        return register ^ self.XorOut</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def bit_by_bit(self, in_data):
        """
        Classic simple and slow CRC implementation.  This function iterates bit
        by bit over the augmented input message and returns the calculated CRC
        value at the end.
        """
        # If the input data is a string, convert to bytes.
        if isinstance(in_data, str):
            in_data = [ord(c) for c in in_data]

        register = self.NonDirectInit
        for octet in in_data:
            if self.ReflectIn:
                octet = self.reflect(octet, 8)
            for i in range(8):
                topbit = register &amp; self.MSB_Mask
                register = ((register &lt;&lt; 1) &amp; self.Mask) | ((octet >> (7 - i)) &amp; 0x01)
                if topbit:
                    register ^= self.Poly

        for i in range(self.Width):
            topbit = register &amp; self.MSB_Mask
            register = ((register &lt;&lt; 1) &amp; self.Mask)
            if topbit:
                register ^= self.Poly

        if self.ReflectOut:
            register = self.reflect(register, self.Width)
        return register ^ self.XorOut</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "bit_by_bit",
  "(",
  "self",
  ",",
  "in_data",
  ")",
  ":",
  "# If the input data is a string, convert to bytes.",
  "if",
  "isinstance",
  "(",
  "in_data",
  ",",
  "str",
  ")",
  ":",
  "in_data",
  "=",
  "[",
  "ord",
  "(",
  "c",
  ")",
  "for",
  "c",
  "in",
  "in_data",
  "]",
  "register",
  "=",
  "self",
  ".",
  "NonDirectInit",
  "for",
  "octet",
  "in",
  "in_data",
  ":",
  "if",
  "self",
  ".",
  "ReflectIn",
  ":",
  "octet",
  "=",
  "self",
  ".",
  "reflect",
  "(",
  "octet",
  ",",
  "8",
  ")",
  "for",
  "i",
  "in",
  "range",
  "(",
  "8",
  ")",
  ":",
  "topbit",
  "=",
  "register",
  "&amp;",
  "self",
  ".",
  "MSB_Mask",
  "register",
  "=",
  "(",
  "(",
  "register",
  "&lt;&lt;",
  "1",
  ")",
  "&amp;",
  "self",
  ".",
  "Mask",
  ")",
  "|",
  "(",
  "(",
  "octet",
  ">>",
  "(",
  "7",
  "-",
  "i",
  ")",
  ")",
  "&amp;",
  "0x01",
  ")",
  "if",
  "topbit",
  ":",
  "register",
  "^=",
  "self",
  ".",
  "Poly",
  "for",
  "i",
  "in",
  "range",
  "(",
  "self",
  ".",
  "Width",
  ")",
  ":",
  "topbit",
  "=",
  "register",
  "&amp;",
  "self",
  ".",
  "MSB_Mask",
  "register",
  "=",
  "(",
  "(",
  "register",
  "&lt;&lt;",
  "1",
  ")",
  "&amp;",
  "self",
  ".",
  "Mask",
  ")",
  "if",
  "topbit",
  ":",
  "register",
  "^=",
  "self",
  ".",
  "Poly",
  "if",
  "self",
  ".",
  "ReflectOut",
  ":",
  "register",
  "=",
  "self",
  ".",
  "reflect",
  "(",
  "register",
  ",",
  "self",
  ".",
  "Width",
  ")",
  "return",
  "register",
  "^",
  "self",
  ".",
  "XorOut"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Classic simple and slow CRC implementation.  This function iterates bit
        by bit over the augmented input message and returns the calculated CRC
        value at the end.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Classic",
  "simple",
  "and",
  "slow",
  "CRC",
  "implementation",
  ".",
  "This",
  "function",
  "iterates",
  "bit",
  "by",
  "bit",
  "over",
  "the",
  "augmented",
  "input",
  "message",
  "and",
  "returns",
  "the",
  "calculated",
  "CRC",
  "value",
  "at",
  "the",
  "end",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">1eb2254b5235a053a953e596fa7602d0b110245d</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/scottrice/pysteam/blob/1eb2254b5235a053a953e596fa7602d0b110245d/pysteam/_crc_algorithms.py#L128-L156</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">50</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="51"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">scottrice/pysteam</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">pysteam/_crc_algorithms.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Crc.gen_table</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def gen_table(self):
        """
        This function generates the CRC table used for the table_driven CRC
        algorithm.  The Python version cannot handle tables of an index width
        other than 8.  See the generated C code for tables with different sizes
        instead.
        """
        table_length = 1 &lt;&lt; self.TableIdxWidth
        tbl = [0] * table_length
        for i in range(table_length):
            register = i
            if self.ReflectIn:
                register = self.reflect(register, self.TableIdxWidth)
            register = register &lt;&lt; (self.Width - self.TableIdxWidth + self.CrcShift)
            for j in range(self.TableIdxWidth):
                if register &amp; (self.MSB_Mask &lt;&lt; self.CrcShift) != 0:
                    register = (register &lt;&lt; 1) ^ (self.Poly &lt;&lt; self.CrcShift)
                else:
                    register = (register &lt;&lt; 1)
            if self.ReflectIn:
                register = self.reflect(register >> self.CrcShift, self.Width) &lt;&lt; self.CrcShift
            tbl[i] = register &amp; (self.Mask &lt;&lt; self.CrcShift)
        return tbl</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def gen_table(self):
        """
        This function generates the CRC table used for the table_driven CRC
        algorithm.  The Python version cannot handle tables of an index width
        other than 8.  See the generated C code for tables with different sizes
        instead.
        """
        table_length = 1 &lt;&lt; self.TableIdxWidth
        tbl = [0] * table_length
        for i in range(table_length):
            register = i
            if self.ReflectIn:
                register = self.reflect(register, self.TableIdxWidth)
            register = register &lt;&lt; (self.Width - self.TableIdxWidth + self.CrcShift)
            for j in range(self.TableIdxWidth):
                if register &amp; (self.MSB_Mask &lt;&lt; self.CrcShift) != 0:
                    register = (register &lt;&lt; 1) ^ (self.Poly &lt;&lt; self.CrcShift)
                else:
                    register = (register &lt;&lt; 1)
            if self.ReflectIn:
                register = self.reflect(register >> self.CrcShift, self.Width) &lt;&lt; self.CrcShift
            tbl[i] = register &amp; (self.Mask &lt;&lt; self.CrcShift)
        return tbl</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "gen_table",
  "(",
  "self",
  ")",
  ":",
  "table_length",
  "=",
  "1",
  "&lt;&lt;",
  "self",
  ".",
  "TableIdxWidth",
  "tbl",
  "=",
  "[",
  "0",
  "]",
  "*",
  "table_length",
  "for",
  "i",
  "in",
  "range",
  "(",
  "table_length",
  ")",
  ":",
  "register",
  "=",
  "i",
  "if",
  "self",
  ".",
  "ReflectIn",
  ":",
  "register",
  "=",
  "self",
  ".",
  "reflect",
  "(",
  "register",
  ",",
  "self",
  ".",
  "TableIdxWidth",
  ")",
  "register",
  "=",
  "register",
  "&lt;&lt;",
  "(",
  "self",
  ".",
  "Width",
  "-",
  "self",
  ".",
  "TableIdxWidth",
  "+",
  "self",
  ".",
  "CrcShift",
  ")",
  "for",
  "j",
  "in",
  "range",
  "(",
  "self",
  ".",
  "TableIdxWidth",
  ")",
  ":",
  "if",
  "register",
  "&amp;",
  "(",
  "self",
  ".",
  "MSB_Mask",
  "&lt;&lt;",
  "self",
  ".",
  "CrcShift",
  ")",
  "!=",
  "0",
  ":",
  "register",
  "=",
  "(",
  "register",
  "&lt;&lt;",
  "1",
  ")",
  "^",
  "(",
  "self",
  ".",
  "Poly",
  "&lt;&lt;",
  "self",
  ".",
  "CrcShift",
  ")",
  "else",
  ":",
  "register",
  "=",
  "(",
  "register",
  "&lt;&lt;",
  "1",
  ")",
  "if",
  "self",
  ".",
  "ReflectIn",
  ":",
  "register",
  "=",
  "self",
  ".",
  "reflect",
  "(",
  "register",
  ">>",
  "self",
  ".",
  "CrcShift",
  ",",
  "self",
  ".",
  "Width",
  ")",
  "&lt;&lt;",
  "self",
  ".",
  "CrcShift",
  "tbl",
  "[",
  "i",
  "]",
  "=",
  "register",
  "&amp;",
  "(",
  "self",
  ".",
  "Mask",
  "&lt;&lt;",
  "self",
  ".",
  "CrcShift",
  ")",
  "return",
  "tbl"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">This function generates the CRC table used for the table_driven CRC
        algorithm.  The Python version cannot handle tables of an index width
        other than 8.  See the generated C code for tables with different sizes
        instead.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "This",
  "function",
  "generates",
  "the",
  "CRC",
  "table",
  "used",
  "for",
  "the",
  "table_driven",
  "CRC",
  "algorithm",
  ".",
  "The",
  "Python",
  "version",
  "cannot",
  "handle",
  "tables",
  "of",
  "an",
  "index",
  "width",
  "other",
  "than",
  "8",
  ".",
  "See",
  "the",
  "generated",
  "C",
  "code",
  "for",
  "tables",
  "with",
  "different",
  "sizes",
  "instead",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">1eb2254b5235a053a953e596fa7602d0b110245d</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/scottrice/pysteam/blob/1eb2254b5235a053a953e596fa7602d0b110245d/pysteam/_crc_algorithms.py#L190-L212</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">51</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="52"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">scottrice/pysteam</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">pysteam/_crc_algorithms.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Crc.table_driven</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def table_driven(self, in_data):
        """
        The Standard table_driven CRC algorithm.
        """
        # If the input data is a string, convert to bytes.
        if isinstance(in_data, str):
            in_data = [ord(c) for c in in_data]

        tbl = self.gen_table()

        register = self.DirectInit &lt;&lt; self.CrcShift
        if not self.ReflectIn:
            for octet in in_data:
                tblidx = ((register >> (self.Width - self.TableIdxWidth + self.CrcShift)) ^ octet) &amp; 0xff
                register = ((register &lt;&lt; (self.TableIdxWidth - self.CrcShift)) ^ tbl[tblidx]) &amp; (self.Mask &lt;&lt; self.CrcShift)
            register = register >> self.CrcShift
        else:
            register = self.reflect(register, self.Width + self.CrcShift) &lt;&lt; self.CrcShift
            for octet in in_data:
                tblidx = ((register >> self.CrcShift) ^ octet) &amp; 0xff
                register = ((register >> self.TableIdxWidth) ^ tbl[tblidx]) &amp; (self.Mask &lt;&lt; self.CrcShift)
            register = self.reflect(register, self.Width + self.CrcShift) &amp; self.Mask

        if self.ReflectOut:
            register = self.reflect(register, self.Width)
        return register ^ self.XorOut</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def table_driven(self, in_data):
        """
        The Standard table_driven CRC algorithm.
        """
        # If the input data is a string, convert to bytes.
        if isinstance(in_data, str):
            in_data = [ord(c) for c in in_data]

        tbl = self.gen_table()

        register = self.DirectInit &lt;&lt; self.CrcShift
        if not self.ReflectIn:
            for octet in in_data:
                tblidx = ((register >> (self.Width - self.TableIdxWidth + self.CrcShift)) ^ octet) &amp; 0xff
                register = ((register &lt;&lt; (self.TableIdxWidth - self.CrcShift)) ^ tbl[tblidx]) &amp; (self.Mask &lt;&lt; self.CrcShift)
            register = register >> self.CrcShift
        else:
            register = self.reflect(register, self.Width + self.CrcShift) &lt;&lt; self.CrcShift
            for octet in in_data:
                tblidx = ((register >> self.CrcShift) ^ octet) &amp; 0xff
                register = ((register >> self.TableIdxWidth) ^ tbl[tblidx]) &amp; (self.Mask &lt;&lt; self.CrcShift)
            register = self.reflect(register, self.Width + self.CrcShift) &amp; self.Mask

        if self.ReflectOut:
            register = self.reflect(register, self.Width)
        return register ^ self.XorOut</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "table_driven",
  "(",
  "self",
  ",",
  "in_data",
  ")",
  ":",
  "# If the input data is a string, convert to bytes.",
  "if",
  "isinstance",
  "(",
  "in_data",
  ",",
  "str",
  ")",
  ":",
  "in_data",
  "=",
  "[",
  "ord",
  "(",
  "c",
  ")",
  "for",
  "c",
  "in",
  "in_data",
  "]",
  "tbl",
  "=",
  "self",
  ".",
  "gen_table",
  "(",
  ")",
  "register",
  "=",
  "self",
  ".",
  "DirectInit",
  "&lt;&lt;",
  "self",
  ".",
  "CrcShift",
  "if",
  "not",
  "self",
  ".",
  "ReflectIn",
  ":",
  "for",
  "octet",
  "in",
  "in_data",
  ":",
  "tblidx",
  "=",
  "(",
  "(",
  "register",
  ">>",
  "(",
  "self",
  ".",
  "Width",
  "-",
  "self",
  ".",
  "TableIdxWidth",
  "+",
  "self",
  ".",
  "CrcShift",
  ")",
  ")",
  "^",
  "octet",
  ")",
  "&amp;",
  "0xff",
  "register",
  "=",
  "(",
  "(",
  "register",
  "&lt;&lt;",
  "(",
  "self",
  ".",
  "TableIdxWidth",
  "-",
  "self",
  ".",
  "CrcShift",
  ")",
  ")",
  "^",
  "tbl",
  "[",
  "tblidx",
  "]",
  ")",
  "&amp;",
  "(",
  "self",
  ".",
  "Mask",
  "&lt;&lt;",
  "self",
  ".",
  "CrcShift",
  ")",
  "register",
  "=",
  "register",
  ">>",
  "self",
  ".",
  "CrcShift",
  "else",
  ":",
  "register",
  "=",
  "self",
  ".",
  "reflect",
  "(",
  "register",
  ",",
  "self",
  ".",
  "Width",
  "+",
  "self",
  ".",
  "CrcShift",
  ")",
  "&lt;&lt;",
  "self",
  ".",
  "CrcShift",
  "for",
  "octet",
  "in",
  "in_data",
  ":",
  "tblidx",
  "=",
  "(",
  "(",
  "register",
  ">>",
  "self",
  ".",
  "CrcShift",
  ")",
  "^",
  "octet",
  ")",
  "&amp;",
  "0xff",
  "register",
  "=",
  "(",
  "(",
  "register",
  ">>",
  "self",
  ".",
  "TableIdxWidth",
  ")",
  "^",
  "tbl",
  "[",
  "tblidx",
  "]",
  ")",
  "&amp;",
  "(",
  "self",
  ".",
  "Mask",
  "&lt;&lt;",
  "self",
  ".",
  "CrcShift",
  ")",
  "register",
  "=",
  "self",
  ".",
  "reflect",
  "(",
  "register",
  ",",
  "self",
  ".",
  "Width",
  "+",
  "self",
  ".",
  "CrcShift",
  ")",
  "&amp;",
  "self",
  ".",
  "Mask",
  "if",
  "self",
  ".",
  "ReflectOut",
  ":",
  "register",
  "=",
  "self",
  ".",
  "reflect",
  "(",
  "register",
  ",",
  "self",
  ".",
  "Width",
  ")",
  "return",
  "register",
  "^",
  "self",
  ".",
  "XorOut"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">The Standard table_driven CRC algorithm.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "The",
  "Standard",
  "table_driven",
  "CRC",
  "algorithm",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">1eb2254b5235a053a953e596fa7602d0b110245d</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/scottrice/pysteam/blob/1eb2254b5235a053a953e596fa7602d0b110245d/pysteam/_crc_algorithms.py#L217-L242</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">52</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="53"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">christophertbrown/bioscripts</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ctbBio/strip_masked.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">parse_masked</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def parse_masked(seq, min_len):
    """
    parse masked sequence into non-masked and masked regions
    """
    nm, masked = [], [[]]
    prev = None
    for base in seq[1]:
        if base.isupper():
            nm.append(base)
            if masked != [[]] and len(masked[-1]) &lt; min_len:
                nm.extend(masked[-1])
                del masked[-1]
            prev = False
        elif base.islower():
            if prev is False:
                masked.append([])
            masked[-1].append(base)
            prev = True
    return nm, masked</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def parse_masked(seq, min_len):
    """
    parse masked sequence into non-masked and masked regions
    """
    nm, masked = [], [[]]
    prev = None
    for base in seq[1]:
        if base.isupper():
            nm.append(base)
            if masked != [[]] and len(masked[-1]) &lt; min_len:
                nm.extend(masked[-1])
                del masked[-1]
            prev = False
        elif base.islower():
            if prev is False:
                masked.append([])
            masked[-1].append(base)
            prev = True
    return nm, masked</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "parse_masked",
  "(",
  "seq",
  ",",
  "min_len",
  ")",
  ":",
  "nm",
  ",",
  "masked",
  "=",
  "[",
  "]",
  ",",
  "[",
  "[",
  "]",
  "]",
  "prev",
  "=",
  "None",
  "for",
  "base",
  "in",
  "seq",
  "[",
  "1",
  "]",
  ":",
  "if",
  "base",
  ".",
  "isupper",
  "(",
  ")",
  ":",
  "nm",
  ".",
  "append",
  "(",
  "base",
  ")",
  "if",
  "masked",
  "!=",
  "[",
  "[",
  "]",
  "]",
  "and",
  "len",
  "(",
  "masked",
  "[",
  "-",
  "1",
  "]",
  ")",
  "&lt;",
  "min_len",
  ":",
  "nm",
  ".",
  "extend",
  "(",
  "masked",
  "[",
  "-",
  "1",
  "]",
  ")",
  "del",
  "masked",
  "[",
  "-",
  "1",
  "]",
  "prev",
  "=",
  "False",
  "elif",
  "base",
  ".",
  "islower",
  "(",
  ")",
  ":",
  "if",
  "prev",
  "is",
  "False",
  ":",
  "masked",
  ".",
  "append",
  "(",
  "[",
  "]",
  ")",
  "masked",
  "[",
  "-",
  "1",
  "]",
  ".",
  "append",
  "(",
  "base",
  ")",
  "prev",
  "=",
  "True",
  "return",
  "nm",
  ",",
  "masked"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">parse masked sequence into non-masked and masked regions</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "parse",
  "masked",
  "sequence",
  "into",
  "non",
  "-",
  "masked",
  "and",
  "masked",
  "regions"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">83b2566b3a5745437ec651cd6cafddd056846240</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/strip_masked.py#L13-L31</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">53</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="54"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">christophertbrown/bioscripts</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ctbBio/strip_masked.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">strip_masked</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def strip_masked(fasta, min_len, print_masked):
    """
    remove masked regions from fasta file as long as
    they are longer than min_len
    """
    for seq in parse_fasta(fasta):
        nm, masked = parse_masked(seq, min_len)
        nm = ['%s removed_masked >=%s' % (seq[0], min_len), ''.join(nm)]
        yield [0, nm]
        if print_masked is True:
            for i, m in enumerate([i for i in masked if i != []], 1):
                m = ['%s insertion:%s' % (seq[0], i), ''.join(m)]
                yield [1, m]</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def strip_masked(fasta, min_len, print_masked):
    """
    remove masked regions from fasta file as long as
    they are longer than min_len
    """
    for seq in parse_fasta(fasta):
        nm, masked = parse_masked(seq, min_len)
        nm = ['%s removed_masked >=%s' % (seq[0], min_len), ''.join(nm)]
        yield [0, nm]
        if print_masked is True:
            for i, m in enumerate([i for i in masked if i != []], 1):
                m = ['%s insertion:%s' % (seq[0], i), ''.join(m)]
                yield [1, m]</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "strip_masked",
  "(",
  "fasta",
  ",",
  "min_len",
  ",",
  "print_masked",
  ")",
  ":",
  "for",
  "seq",
  "in",
  "parse_fasta",
  "(",
  "fasta",
  ")",
  ":",
  "nm",
  ",",
  "masked",
  "=",
  "parse_masked",
  "(",
  "seq",
  ",",
  "min_len",
  ")",
  "nm",
  "=",
  "[",
  "'%s removed_masked >=%s'",
  "%",
  "(",
  "seq",
  "[",
  "0",
  "]",
  ",",
  "min_len",
  ")",
  ",",
  "''",
  ".",
  "join",
  "(",
  "nm",
  ")",
  "]",
  "yield",
  "[",
  "0",
  ",",
  "nm",
  "]",
  "if",
  "print_masked",
  "is",
  "True",
  ":",
  "for",
  "i",
  ",",
  "m",
  "in",
  "enumerate",
  "(",
  "[",
  "i",
  "for",
  "i",
  "in",
  "masked",
  "if",
  "i",
  "!=",
  "[",
  "]",
  "]",
  ",",
  "1",
  ")",
  ":",
  "m",
  "=",
  "[",
  "'%s insertion:%s'",
  "%",
  "(",
  "seq",
  "[",
  "0",
  "]",
  ",",
  "i",
  ")",
  ",",
  "''",
  ".",
  "join",
  "(",
  "m",
  ")",
  "]",
  "yield",
  "[",
  "1",
  ",",
  "m",
  "]"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">remove masked regions from fasta file as long as
    they are longer than min_len</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "remove",
  "masked",
  "regions",
  "from",
  "fasta",
  "file",
  "as",
  "long",
  "as",
  "they",
  "are",
  "longer",
  "than",
  "min_len"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">83b2566b3a5745437ec651cd6cafddd056846240</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/strip_masked.py#L33-L45</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">54</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="55"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">bin/network_plots_gephi.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">get_relative_abundance</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def get_relative_abundance(biomfile):
    """
    Return arcsine transformed relative abundance from a BIOM format file.

    :type biomfile: BIOM format file
    :param biomfile: BIOM format file used to obtain relative abundances for each OTU in
                     a SampleID, which are used as node sizes in network plots.

    :type return: Dictionary of dictionaries.
    :return: Dictionary keyed on SampleID whose value is a dictionarykeyed on OTU Name
             whose value is the arc sine tranfsormed relative abundance value for that
             SampleID-OTU Name pair.
    """
    biomf = biom.load_table(biomfile)
    norm_biomf = biomf.norm(inplace=False)
    rel_abd = {}
    for sid in norm_biomf.ids():
        rel_abd[sid] = {}
        for otuid in norm_biomf.ids("observation"):
            otuname = oc.otu_name(norm_biomf.metadata(otuid, axis="observation")["taxonomy"])
            otuname = " ".join(otuname.split("_"))
            abd = norm_biomf.get_value_by_ids(otuid, sid)
            rel_abd[sid][otuname] = abd
    ast_rel_abd = bc.arcsine_sqrt_transform(rel_abd)
    return ast_rel_abd</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def get_relative_abundance(biomfile):
    """
    Return arcsine transformed relative abundance from a BIOM format file.

    :type biomfile: BIOM format file
    :param biomfile: BIOM format file used to obtain relative abundances for each OTU in
                     a SampleID, which are used as node sizes in network plots.

    :type return: Dictionary of dictionaries.
    :return: Dictionary keyed on SampleID whose value is a dictionarykeyed on OTU Name
             whose value is the arc sine tranfsormed relative abundance value for that
             SampleID-OTU Name pair.
    """
    biomf = biom.load_table(biomfile)
    norm_biomf = biomf.norm(inplace=False)
    rel_abd = {}
    for sid in norm_biomf.ids():
        rel_abd[sid] = {}
        for otuid in norm_biomf.ids("observation"):
            otuname = oc.otu_name(norm_biomf.metadata(otuid, axis="observation")["taxonomy"])
            otuname = " ".join(otuname.split("_"))
            abd = norm_biomf.get_value_by_ids(otuid, sid)
            rel_abd[sid][otuname] = abd
    ast_rel_abd = bc.arcsine_sqrt_transform(rel_abd)
    return ast_rel_abd</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "get_relative_abundance",
  "(",
  "biomfile",
  ")",
  ":",
  "biomf",
  "=",
  "biom",
  ".",
  "load_table",
  "(",
  "biomfile",
  ")",
  "norm_biomf",
  "=",
  "biomf",
  ".",
  "norm",
  "(",
  "inplace",
  "=",
  "False",
  ")",
  "rel_abd",
  "=",
  "{",
  "}",
  "for",
  "sid",
  "in",
  "norm_biomf",
  ".",
  "ids",
  "(",
  ")",
  ":",
  "rel_abd",
  "[",
  "sid",
  "]",
  "=",
  "{",
  "}",
  "for",
  "otuid",
  "in",
  "norm_biomf",
  ".",
  "ids",
  "(",
  "\"observation\"",
  ")",
  ":",
  "otuname",
  "=",
  "oc",
  ".",
  "otu_name",
  "(",
  "norm_biomf",
  ".",
  "metadata",
  "(",
  "otuid",
  ",",
  "axis",
  "=",
  "\"observation\"",
  ")",
  "[",
  "\"taxonomy\"",
  "]",
  ")",
  "otuname",
  "=",
  "\" \"",
  ".",
  "join",
  "(",
  "otuname",
  ".",
  "split",
  "(",
  "\"_\"",
  ")",
  ")",
  "abd",
  "=",
  "norm_biomf",
  ".",
  "get_value_by_ids",
  "(",
  "otuid",
  ",",
  "sid",
  ")",
  "rel_abd",
  "[",
  "sid",
  "]",
  "[",
  "otuname",
  "]",
  "=",
  "abd",
  "ast_rel_abd",
  "=",
  "bc",
  ".",
  "arcsine_sqrt_transform",
  "(",
  "rel_abd",
  ")",
  "return",
  "ast_rel_abd"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Return arcsine transformed relative abundance from a BIOM format file.

    :type biomfile: BIOM format file
    :param biomfile: BIOM format file used to obtain relative abundances for each OTU in
                     a SampleID, which are used as node sizes in network plots.

    :type return: Dictionary of dictionaries.
    :return: Dictionary keyed on SampleID whose value is a dictionarykeyed on OTU Name
             whose value is the arc sine tranfsormed relative abundance value for that
             SampleID-OTU Name pair.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Return",
  "arcsine",
  "transformed",
  "relative",
  "abundance",
  "from",
  "a",
  "BIOM",
  "format",
  "file",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/bin/network_plots_gephi.py#L33-L57</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">55</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="56"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">bin/iTol.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">find_otu</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def find_otu(otuid, tree):
    """
    Find an OTU ID in a Newick-format tree.
    Return the starting position of the ID or None if not found.
    """
    for m in re.finditer(otuid, tree):
        before, after = tree[m.start()-1], tree[m.start()+len(otuid)]
        if before in ["(", ",", ")"] and after in [":", ";"]:
            return m.start()
    return None</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def find_otu(otuid, tree):
    """
    Find an OTU ID in a Newick-format tree.
    Return the starting position of the ID or None if not found.
    """
    for m in re.finditer(otuid, tree):
        before, after = tree[m.start()-1], tree[m.start()+len(otuid)]
        if before in ["(", ",", ")"] and after in [":", ";"]:
            return m.start()
    return None</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "find_otu",
  "(",
  "otuid",
  ",",
  "tree",
  ")",
  ":",
  "for",
  "m",
  "in",
  "re",
  ".",
  "finditer",
  "(",
  "otuid",
  ",",
  "tree",
  ")",
  ":",
  "before",
  ",",
  "after",
  "=",
  "tree",
  "[",
  "m",
  ".",
  "start",
  "(",
  ")",
  "-",
  "1",
  "]",
  ",",
  "tree",
  "[",
  "m",
  ".",
  "start",
  "(",
  ")",
  "+",
  "len",
  "(",
  "otuid",
  ")",
  "]",
  "if",
  "before",
  "in",
  "[",
  "\"(\"",
  ",",
  "\",\"",
  ",",
  "\")\"",
  "]",
  "and",
  "after",
  "in",
  "[",
  "\":\"",
  ",",
  "\";\"",
  "]",
  ":",
  "return",
  "m",
  ".",
  "start",
  "(",
  ")",
  "return",
  "None"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Find an OTU ID in a Newick-format tree.
    Return the starting position of the ID or None if not found.</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Find",
  "an",
  "OTU",
  "ID",
  "in",
  "a",
  "Newick",
  "-",
  "format",
  "tree",
  ".",
  "Return",
  "the",
  "starting",
  "position",
  "of",
  "the",
  "ID",
  "or",
  "None",
  "if",
  "not",
  "found",
  "."
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/bin/iTol.py#L17-L26</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">56</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="57"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">smdabdoub/phylotoast</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">bin/iTol.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">newick_replace_otuids</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def newick_replace_otuids(tree, biomf):
    """
    Replace the OTU ids in the Newick phylogenetic tree format with truncated
    OTU names
    """
    for val, id_, md in biomf.iter(axis="observation"):
        otu_loc = find_otu(id_, tree)
        if otu_loc is not None:
            tree = tree[:otu_loc] + \
                   oc.otu_name(md["taxonomy"]) + \
                   tree[otu_loc + len(id_):]
    return tree</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def newick_replace_otuids(tree, biomf):
    """
    Replace the OTU ids in the Newick phylogenetic tree format with truncated
    OTU names
    """
    for val, id_, md in biomf.iter(axis="observation"):
        otu_loc = find_otu(id_, tree)
        if otu_loc is not None:
            tree = tree[:otu_loc] + \
                   oc.otu_name(md["taxonomy"]) + \
                   tree[otu_loc + len(id_):]
    return tree</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "newick_replace_otuids",
  "(",
  "tree",
  ",",
  "biomf",
  ")",
  ":",
  "for",
  "val",
  ",",
  "id_",
  ",",
  "md",
  "in",
  "biomf",
  ".",
  "iter",
  "(",
  "axis",
  "=",
  "\"observation\"",
  ")",
  ":",
  "otu_loc",
  "=",
  "find_otu",
  "(",
  "id_",
  ",",
  "tree",
  ")",
  "if",
  "otu_loc",
  "is",
  "not",
  "None",
  ":",
  "tree",
  "=",
  "tree",
  "[",
  ":",
  "otu_loc",
  "]",
  "+",
  "oc",
  ".",
  "otu_name",
  "(",
  "md",
  "[",
  "\"taxonomy\"",
  "]",
  ")",
  "+",
  "tree",
  "[",
  "otu_loc",
  "+",
  "len",
  "(",
  "id_",
  ")",
  ":",
  "]",
  "return",
  "tree"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">Replace the OTU ids in the Newick phylogenetic tree format with truncated
    OTU names</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "Replace",
  "the",
  "OTU",
  "ids",
  "in",
  "the",
  "Newick",
  "phylogenetic",
  "tree",
  "format",
  "with",
  "truncated",
  "OTU",
  "names"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">0b74ef171e6a84761710548501dfac71285a58a3</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/smdabdoub/phylotoast/blob/0b74ef171e6a84761710548501dfac71285a58a3/bin/iTol.py#L29-L40</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">57</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="58"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">christophertbrown/bioscripts</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ctbBio/cluster_ani.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">genome_info</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def genome_info(genome, info):
    """
    return genome info for choosing representative

    if ggKbase table provided - choose rep based on SCGs and genome length
        - priority for most SCGs - extra SCGs, then largest genome

    otherwise, based on largest genome
    """
    try:
        scg       = info['#SCGs']
        dups      = info['#SCG duplicates']
        length    = info['genome size (bp)']
        return [scg - dups, length, genome]
    except:
        return [False, False, info['genome size (bp)'], genome]</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def genome_info(genome, info):
    """
    return genome info for choosing representative

    if ggKbase table provided - choose rep based on SCGs and genome length
        - priority for most SCGs - extra SCGs, then largest genome

    otherwise, based on largest genome
    """
    try:
        scg       = info['#SCGs']
        dups      = info['#SCG duplicates']
        length    = info['genome size (bp)']
        return [scg - dups, length, genome]
    except:
        return [False, False, info['genome size (bp)'], genome]</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "genome_info",
  "(",
  "genome",
  ",",
  "info",
  ")",
  ":",
  "try",
  ":",
  "scg",
  "=",
  "info",
  "[",
  "'#SCGs'",
  "]",
  "dups",
  "=",
  "info",
  "[",
  "'#SCG duplicates'",
  "]",
  "length",
  "=",
  "info",
  "[",
  "'genome size (bp)'",
  "]",
  "return",
  "[",
  "scg",
  "-",
  "dups",
  ",",
  "length",
  ",",
  "genome",
  "]",
  "except",
  ":",
  "return",
  "[",
  "False",
  ",",
  "False",
  ",",
  "info",
  "[",
  "'genome size (bp)'",
  "]",
  ",",
  "genome",
  "]"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">return genome info for choosing representative

    if ggKbase table provided - choose rep based on SCGs and genome length
        - priority for most SCGs - extra SCGs, then largest genome

    otherwise, based on largest genome</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "return",
  "genome",
  "info",
  "for",
  "choosing",
  "representative"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">83b2566b3a5745437ec651cd6cafddd056846240</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/cluster_ani.py#L97-L112</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">58</div></div>
							</td>
					</tr><tr class="group cursor-pointer space-x-4 divide-x border-b outline-offset-[-2px] odd:bg-gray-50 hover:bg-gray-100 dark:odd:bg-gray-925 dark:hover:bg-gray-850  last:border-none" tabindex="0" data-row-idx="59"><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">christophertbrown/bioscripts</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">ctbBio/cluster_ani.py</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">print_clusters</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def print_clusters(fastas, info, ANI):
    """
    choose represenative genome and
    print cluster information

    *if ggKbase table is provided, use SCG info to choose best genome
    """
    header = ['#cluster', 'num. genomes', 'rep.', 'genome', '#SCGs', '#SCG duplicates', \
            'genome size (bp)', 'fragments', 'list']
    yield header
    in_cluster = []
    for cluster_num, cluster in enumerate(connected_components(ANI)):
        cluster = sorted([genome_info(genome, info[genome]) \
                            for genome in cluster], \
                            key = lambda x: x[0:], reverse = True)
        rep = cluster[0][-1]
        cluster = [i[-1] for i in cluster]
        size = len(cluster)
        for genome in cluster:
            in_cluster.append(genome)
            try:
                stats = [size, rep, genome, \
                            info[genome]['#SCGs'], info[genome]['#SCG duplicates'], \
                            info[genome]['genome size (bp)'], info[genome]['# contigs'], cluster]
            except:
                stats = [size, rep, genome, \
                            'n/a', 'n/a', \
                            info[genome]['genome size (bp)'], info[genome]['# contigs'], cluster]
            if rep == genome:
                stats = ['*%s' % (cluster_num)] + stats
            else:
                stats = [cluster_num] + stats
            yield stats
    # print singletons
    try:
        start = cluster_num + 1
    except:
        start = 0
    fastas = set([i.rsplit('.', 1)[0].rsplit('/', 1)[-1].rsplit('.contigs')[0] for i in fastas])
    for cluster_num, genome in \
            enumerate(fastas.difference(set(in_cluster)), start):
        try:
            stats = ['*%s' % (cluster_num), 1, genome, genome, \
                        info[genome]['#SCGs'], info[genome]['#SCG duplicates'], \
                        info[genome]['genome size (bp)'], info[genome]['# contigs'], [genome]]
        except:
            stats = ['*%s' % (cluster_num), 1, genome, genome, \
                        'n/a', 'n/a', \
                        info[genome]['genome size (bp)'], info[genome]['# contigs'], [genome]]
        yield stats</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">python</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">def print_clusters(fastas, info, ANI):
    """
    choose represenative genome and
    print cluster information

    *if ggKbase table is provided, use SCG info to choose best genome
    """
    header = ['#cluster', 'num. genomes', 'rep.', 'genome', '#SCGs', '#SCG duplicates', \
            'genome size (bp)', 'fragments', 'list']
    yield header
    in_cluster = []
    for cluster_num, cluster in enumerate(connected_components(ANI)):
        cluster = sorted([genome_info(genome, info[genome]) \
                            for genome in cluster], \
                            key = lambda x: x[0:], reverse = True)
        rep = cluster[0][-1]
        cluster = [i[-1] for i in cluster]
        size = len(cluster)
        for genome in cluster:
            in_cluster.append(genome)
            try:
                stats = [size, rep, genome, \
                            info[genome]['#SCGs'], info[genome]['#SCG duplicates'], \
                            info[genome]['genome size (bp)'], info[genome]['# contigs'], cluster]
            except:
                stats = [size, rep, genome, \
                            'n/a', 'n/a', \
                            info[genome]['genome size (bp)'], info[genome]['# contigs'], cluster]
            if rep == genome:
                stats = ['*%s' % (cluster_num)] + stats
            else:
                stats = [cluster_num] + stats
            yield stats
    # print singletons
    try:
        start = cluster_num + 1
    except:
        start = 0
    fastas = set([i.rsplit('.', 1)[0].rsplit('/', 1)[-1].rsplit('.contigs')[0] for i in fastas])
    for cluster_num, genome in \
            enumerate(fastas.difference(set(in_cluster)), start):
        try:
            stats = ['*%s' % (cluster_num), 1, genome, genome, \
                        info[genome]['#SCGs'], info[genome]['#SCG duplicates'], \
                        info[genome]['genome size (bp)'], info[genome]['# contigs'], [genome]]
        except:
            stats = ['*%s' % (cluster_num), 1, genome, genome, \
                        'n/a', 'n/a', \
                        info[genome]['genome size (bp)'], info[genome]['# contigs'], [genome]]
        yield stats</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "def",
  "print_clusters",
  "(",
  "fastas",
  ",",
  "info",
  ",",
  "ANI",
  ")",
  ":",
  "header",
  "=",
  "[",
  "'#cluster'",
  ",",
  "'num. genomes'",
  ",",
  "'rep.'",
  ",",
  "'genome'",
  ",",
  "'#SCGs'",
  ",",
  "'#SCG duplicates'",
  ",",
  "'genome size (bp)'",
  ",",
  "'fragments'",
  ",",
  "'list'",
  "]",
  "yield",
  "header",
  "in_cluster",
  "=",
  "[",
  "]",
  "for",
  "cluster_num",
  ",",
  "cluster",
  "in",
  "enumerate",
  "(",
  "connected_components",
  "(",
  "ANI",
  ")",
  ")",
  ":",
  "cluster",
  "=",
  "sorted",
  "(",
  "[",
  "genome_info",
  "(",
  "genome",
  ",",
  "info",
  "[",
  "genome",
  "]",
  ")",
  "for",
  "genome",
  "in",
  "cluster",
  "]",
  ",",
  "key",
  "=",
  "lambda",
  "x",
  ":",
  "x",
  "[",
  "0",
  ":",
  "]",
  ",",
  "reverse",
  "=",
  "True",
  ")",
  "rep",
  "=",
  "cluster",
  "[",
  "0",
  "]",
  "[",
  "-",
  "1",
  "]",
  "cluster",
  "=",
  "[",
  "i",
  "[",
  "-",
  "1",
  "]",
  "for",
  "i",
  "in",
  "cluster",
  "]",
  "size",
  "=",
  "len",
  "(",
  "cluster",
  ")",
  "for",
  "genome",
  "in",
  "cluster",
  ":",
  "in_cluster",
  ".",
  "append",
  "(",
  "genome",
  ")",
  "try",
  ":",
  "stats",
  "=",
  "[",
  "size",
  ",",
  "rep",
  ",",
  "genome",
  ",",
  "info",
  "[",
  "genome",
  "]",
  "[",
  "'#SCGs'",
  "]",
  ",",
  "info",
  "[",
  "genome",
  "]",
  "[",
  "'#SCG duplicates'",
  "]",
  ",",
  "info",
  "[",
  "genome",
  "]",
  "[",
  "'genome size (bp)'",
  "]",
  ",",
  "info",
  "[",
  "genome",
  "]",
  "[",
  "'# contigs'",
  "]",
  ",",
  "cluster",
  "]",
  "except",
  ":",
  "stats",
  "=",
  "[",
  "size",
  ",",
  "rep",
  ",",
  "genome",
  ",",
  "'n/a'",
  ",",
  "'n/a'",
  ",",
  "info",
  "[",
  "genome",
  "]",
  "[",
  "'genome size (bp)'",
  "]",
  ",",
  "info",
  "[",
  "genome",
  "]",
  "[",
  "'# contigs'",
  "]",
  ",",
  "cluster",
  "]",
  "if",
  "rep",
  "==",
  "genome",
  ":",
  "stats",
  "=",
  "[",
  "'*%s'",
  "%",
  "(",
  "cluster_num",
  ")",
  "]",
  "+",
  "stats",
  "else",
  ":",
  "stats",
  "=",
  "[",
  "cluster_num",
  "]",
  "+",
  "stats",
  "yield",
  "stats",
  "# print singletons",
  "try",
  ":",
  "start",
  "=",
  "cluster_num",
  "+",
  "1",
  "except",
  ":",
  "start",
  "=",
  "0",
  "fastas",
  "=",
  "set",
  "(",
  "[",
  "i",
  ".",
  "rsplit",
  "(",
  "'.'",
  ",",
  "1",
  ")",
  "[",
  "0",
  "]",
  ".",
  "rsplit",
  "(",
  "'/'",
  ",",
  "1",
  ")",
  "[",
  "-",
  "1",
  "]",
  ".",
  "rsplit",
  "(",
  "'.contigs'",
  ")",
  "[",
  "0",
  "]",
  "for",
  "i",
  "in",
  "fastas",
  "]",
  ")",
  "for",
  "cluster_num",
  ",",
  "genome",
  "in",
  "enumerate",
  "(",
  "fastas",
  ".",
  "difference",
  "(",
  "set",
  "(",
  "in_cluster",
  ")",
  ")",
  ",",
  "start",
  ")",
  ":",
  "try",
  ":",
  "stats",
  "=",
  "[",
  "'*%s'",
  "%",
  "(",
  "cluster_num",
  ")",
  ",",
  "1",
  ",",
  "genome",
  ",",
  "genome",
  ",",
  "info",
  "[",
  "genome",
  "]",
  "[",
  "'#SCGs'",
  "]",
  ",",
  "info",
  "[",
  "genome",
  "]",
  "[",
  "'#SCG duplicates'",
  "]",
  ",",
  "info",
  "[",
  "genome",
  "]",
  "[",
  "'genome size (bp)'",
  "]",
  ",",
  "info",
  "[",
  "genome",
  "]",
  "[",
  "'# contigs'",
  "]",
  ",",
  "[",
  "genome",
  "]",
  "]",
  "except",
  ":",
  "stats",
  "=",
  "[",
  "'*%s'",
  "%",
  "(",
  "cluster_num",
  ")",
  ",",
  "1",
  ",",
  "genome",
  ",",
  "genome",
  ",",
  "'n/a'",
  ",",
  "'n/a'",
  ",",
  "info",
  "[",
  "genome",
  "]",
  "[",
  "'genome size (bp)'",
  "]",
  ",",
  "info",
  "[",
  "genome",
  "]",
  "[",
  "'# contigs'",
  "]",
  ",",
  "[",
  "genome",
  "]",
  "]",
  "yield",
  "stats"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">choose represenative genome and
    print cluster information

    *if ggKbase table is provided, use SCG info to choose best genome</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-left" dir="auto">[
  "choose",
  "represenative",
  "genome",
  "and",
  "print",
  "cluster",
  "information"
]</div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">83b2566b3a5745437ec651cd6cafddd056846240</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">https://github.com/christophertbrown/bioscripts/blob/83b2566b3a5745437ec651cd6cafddd056846240/ctbBio/cluster_ani.py#L114-L163</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="" dir="auto"><span class="">train</span></div></div>
							</td><td class="min-w-fit max-w-sm break-words p-2 "><div class="line-clamp-2 "><div class="text-right" dir="auto">59</div></div>
							</td>
					</tr></tbody></table>
		<div class="bg-linear-to-b sticky left-0 border-t border-dashed border-gray-300 from-gray-100 to-white py-3 text-center font-mono text-xs dark:border-gray-700 dark:from-gray-950 dark:to-gray-900">End of preview. </div></div>




			</div>






</div></div></div></div></div></div></div>

				
					<div class="SVELTE_HYDRATER contents" data-target="RepoCodeCopy" data-props="{}"><div></div></div>
					

					<div class="SVELTE_HYDRATER contents" data-target="SideNavigation" data-props="{&quot;titleTree&quot;:[{&quot;id&quot;:&quot;task-definition&quot;,&quot;label&quot;:&quot;Task Definition&quot;,&quot;children&quot;:[],&quot;isValid&quot;:true,&quot;title&quot;:&quot;Task Definition&quot;},{&quot;id&quot;:&quot;dataset&quot;,&quot;label&quot;:&quot;Dataset&quot;,&quot;children&quot;:[{&quot;id&quot;:&quot;data-format&quot;,&quot;label&quot;:&quot;Data Format&quot;,&quot;children&quot;:[],&quot;isValid&quot;:true,&quot;title&quot;:&quot;Data Format&quot;},{&quot;id&quot;:&quot;data-statistics&quot;,&quot;label&quot;:&quot;Data Statistics&quot;,&quot;children&quot;:[],&quot;isValid&quot;:true,&quot;title&quot;:&quot;Data Statistics&quot;},{&quot;id&quot;:&quot;example&quot;,&quot;label&quot;:&quot;Example&quot;,&quot;children&quot;:[],&quot;isValid&quot;:true,&quot;title&quot;:&quot;Example&quot;},{&quot;id&quot;:&quot;input-predictions&quot;,&quot;label&quot;:&quot;Input Predictions&quot;,&quot;children&quot;:[],&quot;isValid&quot;:true,&quot;title&quot;:&quot;Input Predictions&quot;}],&quot;isValid&quot;:true,&quot;title&quot;:&quot;Dataset&quot;},{&quot;id&quot;:&quot;reference&quot;,&quot;label&quot;:&quot;Reference&quot;,&quot;children&quot;:[],&quot;isValid&quot;:true,&quot;title&quot;:&quot;Reference&quot;}],&quot;classNames&quot;:&quot;top-6&quot;}">

<div class="absolute -left-12 bottom-0 top-0 z-10 top-6"><div class="sticky top-4 flex"><div class="h-7 pt-[0.175rem]">
				<span class="peer" tabindex="0"><button class="select-none hover:cursor-pointer"><svg width="1em" height="1em" viewBox="0 0 10 10" class="text-lg opacity-80 hover:opacity-100" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" role="img" preserveAspectRatio="xMidYMid meet" fill="currentColor"><path fill-rule="evenodd" clip-rule="evenodd" d="M1.65039 2.9999C1.65039 2.8066 1.80709 2.6499 2.00039 2.6499H8.00039C8.19369 2.6499 8.35039 2.8066 8.35039 2.9999C8.35039 3.1932 8.19369 3.3499 8.00039 3.3499H2.00039C1.80709 3.3499 1.65039 3.1932 1.65039 2.9999ZM1.65039 4.9999C1.65039 4.8066 1.80709 4.6499 2.00039 4.6499H8.00039C8.19369 4.6499 8.35039 4.8066 8.35039 4.9999C8.35039 5.1932 8.19369 5.3499 8.00039 5.3499H2.00039C1.80709 5.3499 1.65039 5.1932 1.65039 4.9999ZM2.00039 6.6499C1.80709 6.6499 1.65039 6.8066 1.65039 6.9999C1.65039 7.1932 1.80709 7.3499 2.00039 7.3499H8.00039C8.19369 7.3499 8.35039 7.1932 8.35039 6.9999C8.35039 6.8066 8.19369 6.6499 8.00039 6.6499H2.00039Z"></path></svg></button></span>
				<div class="invisible w-0 -translate-x-24 -translate-y-6 overflow-hidden rounded-xl border bg-white transition-transform hover:visible hover:w-52 hover:translate-x-0 peer-focus-within:visible peer-focus-within:w-52 peer-focus-within:translate-x-0"><nav aria-label="Secondary" class="max-h-[550px] overflow-y-auto p-3"><ul><li class="mb-3 text-sm last:mb-0"><a class="mb-1 block break-words font-semibold text-gray-700 *:break-words hover:underline active:text-gray-900 dark:active:text-gray-200" href="#task-definition" title="Task Definition"><!-- HTML_TAG_START -->Task Definition<!-- HTML_TAG_END --></a>
									<ul class="pl-1"></ul>
								</li><li class="mb-3 text-sm last:mb-0"><a class="mb-1 block break-words font-semibold text-gray-700 *:break-words hover:underline active:text-gray-900 dark:active:text-gray-200" href="#dataset" title="Dataset"><!-- HTML_TAG_START -->Dataset<!-- HTML_TAG_END --></a>
									<ul class="pl-1"><li><a class="mb-0.5 block break-words hover:underline active:text-gray-700 dark:active:text-gray-300 text-gray-500" href="#data-format" title="Data Format"><!-- HTML_TAG_START -->Data Format<!-- HTML_TAG_END --></a>
												<ul class="pl-2"></ul>
											</li><li><a class="mb-0.5 block break-words hover:underline active:text-gray-700 dark:active:text-gray-300 text-gray-500" href="#data-statistics" title="Data Statistics"><!-- HTML_TAG_START -->Data Statistics<!-- HTML_TAG_END --></a>
												<ul class="pl-2"></ul>
											</li><li><a class="mb-0.5 block break-words hover:underline active:text-gray-700 dark:active:text-gray-300 text-gray-500" href="#example" title="Example"><!-- HTML_TAG_START -->Example<!-- HTML_TAG_END --></a>
												<ul class="pl-2"></ul>
											</li><li><a class="mb-0.5 block break-words hover:underline active:text-gray-700 dark:active:text-gray-300 text-gray-500" href="#input-predictions" title="Input Predictions"><!-- HTML_TAG_START -->Input Predictions<!-- HTML_TAG_END --></a>
												<ul class="pl-2"></ul>
											</li></ul>
								</li><li class="mb-3 text-sm last:mb-0"><a class="mb-1 block break-words font-semibold text-gray-700 *:break-words hover:underline active:text-gray-900 dark:active:text-gray-200" href="#reference" title="Reference"><!-- HTML_TAG_START -->Reference<!-- HTML_TAG_END --></a>
									<ul class="pl-1"></ul>
								</li></ul></nav></div></div></div></div></div>
					<div class="2xl:pr-6"><div class="prose pl-6 -ml-6 hf-sanitized hf-sanitized-QkfTho07TF4rkdwlZbvWZ">
	<!-- HTML_TAG_START --><h3 class="relative group flex items-center">
	<a rel="nofollow" href="#dataset-is-imported-from-codexglue-and-pre-processed-using-their-script" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full" id="dataset-is-imported-from-codexglue-and-pre-processed-using-their-script">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		Dataset is imported from CodeXGLUE and pre-processed using their script.
	</span>
</h3>
<h1 class="relative group flex items-center">
	<a rel="nofollow" href="#where-to-find-in-semeru" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full" id="where-to-find-in-semeru">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		Where to find in Semeru:
	</span>
</h1>
<p>The dataset can be found at /nfs/semeru/semeru_datasets/code_xglue/text-to-code/codesearchnet/python in Semeru</p>
<h1 class="relative group flex items-center">
	<a rel="nofollow" href="#codexglue----code-search-advtest" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full" id="codexglue----code-search-advtest">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		CodeXGLUE -- Code Search (AdvTest)
	</span>
</h1>
<h2 class="relative group flex items-center">
	<a rel="nofollow" href="#task-definition" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full" id="task-definition">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		Task Definition
	</span>
</h2>
<p>Given a natural language, the task is to search source code that matches the natural language. To test the generalization ability of a model,  function names and variables in test sets are replaced by special tokens.</p>
<h2 class="relative group flex items-center">
	<a rel="nofollow" href="#dataset" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		Dataset
	</span>
</h2>
<p>The dataset we use comes from <a rel="nofollow" href="https://arxiv.org/pdf/1909.09436.pdf">CodeSearchNet</a> and we filter the dataset as the following:</p>
<ul>
<li>Remove examples that codes cannot be parsed into an abstract syntax tree.</li>
<li>Remove examples that #tokens of documents is &lt; 3 or &gt;256</li>
<li>Remove examples that documents contain special tokens (e.g. &lt;img ...&gt; or https:...)</li>
<li>Remove examples that documents are not English.</li>
</ul>
<p>Besides, to test the generalization ability of a model,  function names and variables in test sets are replaced by special tokens.</p>
<h3 class="relative group flex items-center">
	<a rel="nofollow" href="#data-format" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full" id="data-format">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		Data Format
	</span>
</h3>
<p>After preprocessing dataset, you can obtain three .jsonl files, i.e. train.jsonl, valid.jsonl, test.jsonl</p>
<p>For each file, each line in the uncompressed file represents one function.  One row is illustrated below.</p>
<ul>
<li><strong>repo:</strong> the owner/repo</li>
<li><strong>path:</strong> the full path to the original file</li>
<li><strong>func_name:</strong> the function or method name</li>
<li><strong>original_string:</strong> the raw string before tokenization or parsing</li>
<li><strong>language:</strong> the programming language</li>
<li><strong>code/function:</strong> the part of the <code>original_string</code> that is code</li>
<li><strong>code_tokens/function_tokens:</strong> tokenized version of <code>code</code></li>
<li><strong>docstring:</strong> the top-level comment or docstring, if it exists in the original string</li>
<li><strong>docstring_tokens:</strong> tokenized version of <code>docstring</code></li>
<li><strong>url:</strong> the url for the example (identify natural language)</li>
<li><strong>idx</strong>: the index of code (identify code)</li>
</ul>
<h3 class="relative group flex items-center">
	<a rel="nofollow" href="#data-statistics" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full" id="data-statistics">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		Data Statistics
	</span>
</h3>
<p>Data statistics of the dataset are shown in the below table:</p>
<div class="max-w-full overflow-auto">
	<table>
		<thead><tr>
<th></th>
<th align="center">#Examples</th>
</tr>

		</thead><tbody><tr>
<td>Train</td>
<td align="center">251,820</td>
</tr>
<tr>
<td>Dev</td>
<td align="center">9,604</td>
</tr>
<tr>
<td>Test</td>
<td align="center">19,210</td>
</tr>
</tbody>
	</table>
</div>
<h3 class="relative group flex items-center">
	<a rel="nofollow" href="#example" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full" id="example">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		Example
	</span>
</h3>
<p>Given a text-code file evaluator/test.jsonl:</p>
<pre><code class="language-json"><span class="hljs-punctuation">{</span><span class="hljs-attr">"url"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"url0"</span><span class="hljs-punctuation">,</span> <span class="hljs-attr">"docstring"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"doc0"</span><span class="hljs-punctuation">,</span><span class="hljs-attr">"function"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"fun0"</span><span class="hljs-punctuation">,</span> <span class="hljs-attr">"idx"</span><span class="hljs-punctuation">:</span> <span class="hljs-number">10</span><span class="hljs-punctuation">}</span>
<span class="hljs-punctuation">{</span><span class="hljs-attr">"url"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"url1"</span><span class="hljs-punctuation">,</span> <span class="hljs-attr">"docstring"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"doc1"</span><span class="hljs-punctuation">,</span><span class="hljs-attr">"function"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"fun1"</span><span class="hljs-punctuation">,</span> <span class="hljs-attr">"idx"</span><span class="hljs-punctuation">:</span> <span class="hljs-number">11</span><span class="hljs-punctuation">}</span>
<span class="hljs-punctuation">{</span><span class="hljs-attr">"url"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"url2"</span><span class="hljs-punctuation">,</span> <span class="hljs-attr">"docstring"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"doc2"</span><span class="hljs-punctuation">,</span><span class="hljs-attr">"function"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"fun2"</span><span class="hljs-punctuation">,</span> <span class="hljs-attr">"idx"</span><span class="hljs-punctuation">:</span> <span class="hljs-number">12</span><span class="hljs-punctuation">}</span>
<span class="hljs-punctuation">{</span><span class="hljs-attr">"url"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"url3"</span><span class="hljs-punctuation">,</span> <span class="hljs-attr">"docstring"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"doc3"</span><span class="hljs-punctuation">,</span><span class="hljs-attr">"function"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"fun3"</span><span class="hljs-punctuation">,</span> <span class="hljs-attr">"idx"</span><span class="hljs-punctuation">:</span> <span class="hljs-number">13</span><span class="hljs-punctuation">}</span>
<span class="hljs-punctuation">{</span><span class="hljs-attr">"url"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"url4"</span><span class="hljs-punctuation">,</span> <span class="hljs-attr">"docstring"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"doc4"</span><span class="hljs-punctuation">,</span><span class="hljs-attr">"function"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"fun4"</span><span class="hljs-punctuation">,</span> <span class="hljs-attr">"idx"</span><span class="hljs-punctuation">:</span> <span class="hljs-number">14</span><span class="hljs-punctuation">}</span>
</code></pre>
<h3 class="relative group flex items-center">
	<a rel="nofollow" href="#input-predictions" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full" id="input-predictions">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		Input Predictions
	</span>
</h3>
<p>For each url for natural language, descending sort candidate codes and return their idx in order. For example:</p>
<pre><code class="language-json"><span class="hljs-punctuation">{</span><span class="hljs-attr">"url"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"url0"</span><span class="hljs-punctuation">,</span> <span class="hljs-attr">"answers"</span><span class="hljs-punctuation">:</span> <span class="hljs-punctuation">[</span><span class="hljs-number">10</span><span class="hljs-punctuation">,</span><span class="hljs-number">11</span><span class="hljs-punctuation">,</span><span class="hljs-number">12</span><span class="hljs-punctuation">,</span><span class="hljs-number">13</span><span class="hljs-punctuation">,</span><span class="hljs-number">14</span><span class="hljs-punctuation">]</span><span class="hljs-punctuation">}</span>
<span class="hljs-punctuation">{</span><span class="hljs-attr">"url"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"url1"</span><span class="hljs-punctuation">,</span> <span class="hljs-attr">"answers"</span><span class="hljs-punctuation">:</span> <span class="hljs-punctuation">[</span><span class="hljs-number">10</span><span class="hljs-punctuation">,</span><span class="hljs-number">12</span><span class="hljs-punctuation">,</span><span class="hljs-number">11</span><span class="hljs-punctuation">,</span><span class="hljs-number">13</span><span class="hljs-punctuation">,</span><span class="hljs-number">14</span><span class="hljs-punctuation">]</span><span class="hljs-punctuation">}</span>
<span class="hljs-punctuation">{</span><span class="hljs-attr">"url"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"url2"</span><span class="hljs-punctuation">,</span> <span class="hljs-attr">"answers"</span><span class="hljs-punctuation">:</span> <span class="hljs-punctuation">[</span><span class="hljs-number">13</span><span class="hljs-punctuation">,</span><span class="hljs-number">11</span><span class="hljs-punctuation">,</span><span class="hljs-number">12</span><span class="hljs-punctuation">,</span><span class="hljs-number">10</span><span class="hljs-punctuation">,</span><span class="hljs-number">14</span><span class="hljs-punctuation">]</span><span class="hljs-punctuation">}</span>
<span class="hljs-punctuation">{</span><span class="hljs-attr">"url"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"url3"</span><span class="hljs-punctuation">,</span> <span class="hljs-attr">"answers"</span><span class="hljs-punctuation">:</span> <span class="hljs-punctuation">[</span><span class="hljs-number">10</span><span class="hljs-punctuation">,</span><span class="hljs-number">14</span><span class="hljs-punctuation">,</span><span class="hljs-number">12</span><span class="hljs-punctuation">,</span><span class="hljs-number">13</span><span class="hljs-punctuation">,</span><span class="hljs-number">11</span><span class="hljs-punctuation">]</span><span class="hljs-punctuation">}</span>
<span class="hljs-punctuation">{</span><span class="hljs-attr">"url"</span><span class="hljs-punctuation">:</span> <span class="hljs-string">"url4"</span><span class="hljs-punctuation">,</span> <span class="hljs-attr">"answers"</span><span class="hljs-punctuation">:</span> <span class="hljs-punctuation">[</span><span class="hljs-number">10</span><span class="hljs-punctuation">,</span><span class="hljs-number">11</span><span class="hljs-punctuation">,</span><span class="hljs-number">12</span><span class="hljs-punctuation">,</span><span class="hljs-number">13</span><span class="hljs-punctuation">,</span><span class="hljs-number">14</span><span class="hljs-punctuation">]</span><span class="hljs-punctuation">}</span>
</code></pre>
<h2 class="relative group flex items-center">
	<a rel="nofollow" href="#reference" class="block pr-1.5 text-lg md:absolute md:p-1.5 md:opacity-0 md:group-hover:opacity-100 md:right-full" id="reference">
		<span class="header-link"><svg viewBox="0 0 256 256" preserveAspectRatio="xMidYMid meet" height="1em" width="1em" role="img" aria-hidden="true" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" class="text-gray-500 hover:text-black dark:hover:text-gray-200 w-4"><path fill="currentColor" d="M167.594 88.393a8.001 8.001 0 0 1 0 11.314l-67.882 67.882a8 8 0 1 1-11.314-11.315l67.882-67.881a8.003 8.003 0 0 1 11.314 0zm-28.287 84.86l-28.284 28.284a40 40 0 0 1-56.567-56.567l28.284-28.284a8 8 0 0 0-11.315-11.315l-28.284 28.284a56 56 0 0 0 79.196 79.197l28.285-28.285a8 8 0 1 0-11.315-11.314zM212.852 43.14a56.002 56.002 0 0 0-79.196 0l-28.284 28.284a8 8 0 1 0 11.314 11.314l28.284-28.284a40 40 0 0 1 56.568 56.567l-28.285 28.285a8 8 0 0 0 11.315 11.314l28.284-28.284a56.065 56.065 0 0 0 0-79.196z"></path></svg></span>
	</a>
	<span>
		Reference
	</span>
</h2>
<pre><code>@article{husain2019codesearchnet,
  title={Codesearchnet challenge: Evaluating the state of semantic code search},
  author={Husain, Hamel and Wu, Ho-Hsiang and Gazit, Tiferet and Allamanis, Miltiadis and Brockschmidt, Marc},
  journal={arXiv preprint arXiv:1909.09436},
  year={2019}
}</code></pre><!-- HTML_TAG_END --></div>
</div></section>
			
			<section class="pt-6 border-gray-100 md:pb-24 md:pl-6 md:w-64 lg:w-80 xl:w-96 flex-none order-first md:order-none md:border-l pt-3! md:pt-6!"><dl class="flex items-baseline justify-between"><dt class="text-sm text-gray-500">Downloads last month</dt><div class="mx-4 flex-1 border-b border-dotted"></div><dd class="font-semibold">56</dd></dl>
				
				<div class="divider-column-vertical"></div>
				<div class="grid grid-cols-2 gap-x-2 md:flex md:flex-row md:flex-wrap"><div class="SVELTE_HYDRATER contents" data-target="DatasetAndModelActionsDropdown" data-props="{&quot;classNames&quot;:&quot;order-last&quot;,&quot;discussionsDisabled&quot;:false,&quot;repo&quot;:{&quot;type&quot;:&quot;dataset&quot;,&quot;name&quot;:&quot;semeru/Text-Code-CodeSearchNet-Python&quot;},&quot;canWrite&quot;:false,&quot;canDisable&quot;:false,&quot;repoIsPrivate&quot;:false,&quot;repoIsGated&quot;:false,&quot;repoIsDisabled&quot;:false,&quot;repoIsAdminFlaggedNFAA&quot;:false,&quot;repoHasBlockedOids&quot;:false}"><div class="order-last"><div class="relative ">
	<button class="btn px-1.5 py-1.5 " type="button">
		
			<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" class="p-0.5" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><circle cx="16" cy="7" r="3" fill="currentColor"></circle><circle cx="16" cy="16" r="3" fill="currentColor"></circle><circle cx="16" cy="25" r="3" fill="currentColor"></circle></svg>
		
		</button>
	
	
	</div></div>













</div>
					<div class="SVELTE_HYDRATER contents" data-target="DatasetLibrary" data-props="{&quot;classNames&quot;:&quot;md:w-full xl:w-auto xl:flex-none&quot;,&quot;libraries&quot;:[]}">

</div>
					
					</div>
				
				
				
				
				

				
				<div class="divider-column-vertical md:hidden"></div></section></div></main>

	<footer class="b-12 mb-2 flex border-t border-gray-100 md:h-14"><nav class="container relative flex flex-col justify-between space-y-2 py-6 text-gray-500 max-md:*:self-start md:flex-row md:items-center md:space-y-0 md:py-0 md:text-sm"><div class="SVELTE_HYDRATER contents" data-target="ThemeSwitcher" data-props="{&quot;theme&quot;:&quot;system&quot;,&quot;isLoggedIn&quot;:false,&quot;menuClassNames&quot;:&quot;md:-top-24&quot;,&quot;classNames&quot;:&quot;max-md:mb-5 max-md:*:self-start&quot;}">
<div class="relative inline-block max-md:mb-5 max-md:*:self-start">
	<button class="rounded-full border border-gray-100 pl-2 py-1 pr-2.5  flex items-center text-sm text-gray-500 bg-white hover:bg-purple-50 hover:border-purple-200 dark:hover:bg-gray-800 dark:hover:border-gray-950 dark:border-gray-800 " type="button">
		<svg class="mr-1.5 text-gray-500" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" fill="currentColor" focusable="false" role="img" width="1em" height="1em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 32 32"><path d="M29 25H3a1 1 0 1 0 0 2h26a1 1 0 1 0 0-2Z" fill="currentColor"></path><path fill-rule="evenodd" clip-rule="evenodd" d="M6 22.5h20a2 2 0 0 0 2-2V7a2 2 0 0 0-2-2H6a2 2 0 0 0-2 2v13.5a2 2 0 0 0 2 2ZM7 7a1 1 0 0 0-1 1v11a1 1 0 0 0 1 1h18a1 1 0 0 0 1-1V8a1 1 0 0 0-1-1H7Z" fill="currentColor"></path><path d="M6 8a1 1 0 0 1 1-1h18a1 1 0 0 1 1 1v11a1 1 0 0 1-1 1H7a1 1 0 0 1-1-1V8Z" fill="currentColor" fill-opacity=".4"></path><path d="M29 25H3a1 1 0 1 0 0 2h26a1 1 0 1 0 0-2Z" fill="currentColor"></path></svg>
			System theme
		</button>
	
	
	</div></div>
		<div class="font-semibold text-black md:hidden">Company</div>
		<a class="hover:underline" href="/terms-of-service">TOS</a>
		<a class="hover:underline" href="/privacy">Privacy</a>
		<a class="hover:underline" href="/huggingface">About</a>
		<a class="hover:underline" href="https://apply.workable.com/huggingface/">Jobs</a>
		<a href="/" class="max-md:mb-4! max-md:mt-8! group flex-none max-md:order-last"><svg class="h-7 w-7 transition-transform group-hover:-translate-y-px" viewBox="0 0 95 88" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M47.2119 76.5C66.4037 76.5 81.9619 60.9419 81.9619 41.75C81.9619 22.5581 66.4037 7 47.2119 7C28.02 7 12.4619 22.5581 12.4619 41.75C12.4619 60.9419 28.02 76.5 47.2119 76.5Z" fill="#FFD21E"></path><path d="M81.9619 41.75C81.9619 22.5581 66.4037 7 47.2119 7C28.02 7 12.4619 22.5581 12.4619 41.75C12.4619 60.9419 28.02 76.5 47.2119 76.5C66.4037 76.5 81.9619 60.9419 81.9619 41.75ZM8.46185 41.75C8.46185 20.349 25.8108 3 47.2119 3C68.6129 3 85.9619 20.349 85.9619 41.75C85.9619 63.151 68.6129 80.5 47.2119 80.5C25.8108 80.5 8.46185 63.151 8.46185 41.75Z" fill="#FF9D0B"></path><path d="M58.5024 32.2915C59.7768 32.7415 60.2839 35.3615 61.5713 34.6769C64.0095 33.3805 64.9351 30.353 63.6387 27.9148C62.3423 25.4767 59.3148 24.5511 56.8766 25.8475C54.4384 27.1439 53.5128 30.1714 54.8092 32.6096C55.4211 33.7604 57.3632 31.8892 58.5024 32.2915Z" fill="#3A3B45"></path><path d="M34.9454 32.2915C33.671 32.7415 33.164 35.3615 31.8766 34.6769C29.4384 33.3805 28.5128 30.353 29.8092 27.9148C31.1056 25.4767 34.1331 24.5511 36.5713 25.8475C39.0095 27.1439 39.9351 30.1714 38.6387 32.6096C38.0268 33.7604 36.0846 31.8892 34.9454 32.2915Z" fill="#3A3B45"></path><path d="M46.9619 56.289C56.7903 56.289 59.9619 47.5261 59.9619 43.0262C59.9619 40.6875 58.3898 41.4236 55.8718 42.6702C53.5449 43.8222 50.4102 45.4101 46.9619 45.4101C39.7822 45.4101 33.9619 38.5263 33.9619 43.0262C33.9619 47.5261 37.1334 56.289 46.9619 56.289Z" fill="#3A3B45"></path><mask id="mask0" style="mask-type:alpha" maskUnits="userSpaceOnUse" x="33" y="41" width="27" height="16"><path d="M46.9619 56.289C56.7903 56.289 59.9619 47.5261 59.9619 43.0262C59.9619 40.6875 58.3898 41.4236 55.8718 42.6702C53.5449 43.8222 50.4102 45.4101 46.9619 45.4101C39.7822 45.4101 33.9619 38.5263 33.9619 43.0262C33.9619 47.5261 37.1334 56.289 46.9619 56.289Z" fill="white"></path></mask><g mask="url(#mask0)"><path d="M47.2119 66.5C52.0018 66.5 55.8848 62.617 55.8848 57.8271C55.8848 54.0962 53.5291 50.9156 50.224 49.6915C50.1023 49.6464 49.9794 49.604 49.8553 49.5643C49.0219 49.2979 48.1337 52.1623 47.2119 52.1623C46.3506 52.1623 45.5186 49.2797 44.7332 49.5135C41.151 50.5799 38.5389 53.8984 38.5389 57.8271C38.5389 62.617 42.4219 66.5 47.2119 66.5Z" fill="#F94040"></path></g><path d="M70.7119 37C72.5068 37 73.9619 35.5449 73.9619 33.75C73.9619 31.9551 72.5068 30.5 70.7119 30.5C68.9169 30.5 67.4619 31.9551 67.4619 33.75C67.4619 35.5449 68.9169 37 70.7119 37Z" fill="#FF9D0B"></path><path d="M24.2119 37C26.0068 37 27.4619 35.5449 27.4619 33.75C27.4619 31.9551 26.0068 30.5 24.2119 30.5C22.4169 30.5 20.9619 31.9551 20.9619 33.75C20.9619 35.5449 22.4169 37 24.2119 37Z" fill="#FF9D0B"></path><path class="origin-bottom-right transition-transform group-hover:-rotate-6" d="M17.5238 48C15.9048 48 14.4578 48.665 13.4488 49.871C12.8248 50.618 12.1728 51.822 12.1198 53.625C11.4408 53.43 10.7878 53.321 10.1778 53.321C8.6278 53.321 7.2278 53.915 6.2378 54.994C4.9658 56.379 4.4008 58.081 4.6468 59.784C4.7638 60.595 5.0348 61.322 5.4398 61.995C4.5858 62.686 3.9568 63.648 3.6528 64.805C3.4148 65.712 3.1708 67.601 4.4448 69.547C4.3638 69.674 4.2878 69.806 4.2168 69.941C3.4508 71.395 3.4018 73.038 4.0778 74.568C5.1028 76.887 7.6498 78.714 12.5958 80.675C15.6728 81.895 18.4878 82.675 18.5128 82.682C22.5808 83.737 26.2598 84.273 29.4448 84.273C35.2988 84.273 39.4898 82.48 41.9018 78.944C45.7838 73.25 45.2288 68.042 40.2058 63.022C37.4258 60.244 35.5778 56.148 35.1928 55.249C34.4168 52.587 32.3648 49.628 28.9538 49.628H28.9528C28.6658 49.628 28.3758 49.651 28.0898 49.696C26.5958 49.931 25.2898 50.791 24.3568 52.085C23.3498 50.833 22.3718 49.837 21.4868 49.275C20.1528 48.429 18.8198 48 17.5238 48ZM17.5238 52C18.0338 52 18.6568 52.217 19.3438 52.653C21.4768 54.006 25.5928 61.081 27.0998 63.833C27.6048 64.755 28.4678 65.145 29.2448 65.145C30.7868 65.145 31.9908 63.612 29.3858 61.664C25.4688 58.733 26.8428 53.942 28.7128 53.647C28.7948 53.634 28.8758 53.628 28.9538 53.628C30.6538 53.628 31.4038 56.558 31.4038 56.558C31.4038 56.558 33.6018 62.078 37.3778 65.851C41.1538 69.625 41.3488 72.654 38.5968 76.69C36.7198 79.442 33.1268 80.273 29.4448 80.273C25.6258 80.273 21.7108 79.379 19.5168 78.81C19.4088 78.782 6.0658 75.013 7.7558 71.805C8.0398 71.266 8.5078 71.05 9.0968 71.05C11.4768 71.05 15.8058 74.592 17.6668 74.592C18.0828 74.592 18.3758 74.415 18.4958 73.983C19.2888 71.138 6.4388 69.942 7.5218 65.821C7.7128 65.092 8.2308 64.796 8.9588 64.797C12.1038 64.797 19.1598 70.328 20.6388 70.328C20.7518 70.328 20.8328 70.295 20.8768 70.225C21.6178 69.029 21.2118 68.194 15.9888 65.033C10.7658 61.871 7.0998 59.969 9.1848 57.699C9.4248 57.437 9.7648 57.321 10.1778 57.321C13.3488 57.322 20.8408 64.14 20.8408 64.14C20.8408 64.14 22.8628 66.243 24.0858 66.243C24.3668 66.243 24.6058 66.132 24.7678 65.858C25.6348 64.396 16.7148 57.636 16.2118 54.847C15.8708 52.957 16.4508 52 17.5238 52Z" fill="#FF9D0B"></path><path class="origin-bottom-right transition-transform group-hover:-rotate-6" d="M38.5967 76.6898C41.3487 72.6538 41.1537 69.6248 37.3777 65.8508C33.6017 62.0778 31.4037 56.5578 31.4037 56.5578C31.4037 56.5578 30.5827 53.3518 28.7127 53.6468C26.8427 53.9418 25.4697 58.7328 29.3867 61.6638C33.3037 64.5938 28.6067 66.5848 27.0997 63.8328C25.5927 61.0808 21.4777 54.0058 19.3437 52.6528C17.2107 51.2998 15.7087 52.0578 16.2117 54.8468C16.7147 57.6358 25.6357 64.3958 24.7677 65.8588C23.8997 67.3208 20.8407 64.1398 20.8407 64.1398C20.8407 64.1398 11.2687 55.4288 9.18465 57.6988C7.10065 59.9688 10.7657 61.8708 15.9887 65.0328C21.2127 68.1938 21.6177 69.0288 20.8767 70.2248C20.1347 71.4208 8.60465 61.6998 7.52165 65.8208C6.43965 69.9418 19.2887 71.1378 18.4957 73.9828C17.7027 76.8288 9.44465 68.5978 7.75565 71.8048C6.06565 75.0128 19.4087 78.7818 19.5167 78.8098C23.8267 79.9278 34.7727 82.2968 38.5967 76.6898Z" fill="#FFD21E"></path><path class="origin-bottom-left transition-transform group-hover:rotate-6" d="M77.3999 48C79.0189 48 80.4659 48.665 81.4749 49.871C82.0989 50.618 82.7509 51.822 82.8039 53.625C83.4829 53.43 84.1359 53.321 84.7459 53.321C86.2959 53.321 87.6959 53.915 88.6859 54.994C89.9579 56.379 90.5229 58.081 90.2769 59.784C90.1599 60.595 89.8889 61.322 89.4839 61.995C90.3379 62.686 90.9669 63.648 91.2709 64.805C91.5089 65.712 91.7529 67.601 90.4789 69.547C90.5599 69.674 90.6359 69.806 90.7069 69.941C91.4729 71.395 91.5219 73.038 90.8459 74.568C89.8209 76.887 87.2739 78.714 82.3279 80.675C79.2509 81.895 76.4359 82.675 76.4109 82.682C72.3429 83.737 68.6639 84.273 65.4789 84.273C59.6249 84.273 55.4339 82.48 53.0219 78.944C49.1399 73.25 49.6949 68.042 54.7179 63.022C57.4979 60.244 59.3459 56.148 59.7309 55.249C60.5069 52.587 62.5589 49.628 65.9699 49.628H65.9709C66.2579 49.628 66.5479 49.651 66.8339 49.696C68.3279 49.931 69.6339 50.791 70.5669 52.085C71.5739 50.833 72.5519 49.837 73.4369 49.275C74.7709 48.429 76.1039 48 77.3999 48ZM77.3999 52C76.8899 52 76.2669 52.217 75.5799 52.653C73.4469 54.006 69.3309 61.081 67.8239 63.833C67.3189 64.755 66.4559 65.145 65.6789 65.145C64.1369 65.145 62.9329 63.612 65.5379 61.664C69.4549 58.733 68.0809 53.942 66.2109 53.647C66.1289 53.634 66.0479 53.628 65.9699 53.628C64.2699 53.628 63.5199 56.558 63.5199 56.558C63.5199 56.558 61.3219 62.078 57.5459 65.851C53.7699 69.625 53.5749 72.654 56.3269 76.69C58.2039 79.442 61.7969 80.273 65.4789 80.273C69.2979 80.273 73.2129 79.379 75.4069 78.81C75.5149 78.782 88.8579 75.013 87.1679 71.805C86.8839 71.266 86.4159 71.05 85.8269 71.05C83.4469 71.05 79.1179 74.592 77.2569 74.592C76.8409 74.592 76.5479 74.415 76.4279 73.983C75.6349 71.138 88.4849 69.942 87.4019 65.821C87.2109 65.092 86.6929 64.796 85.9649 64.797C82.8199 64.797 75.7639 70.328 74.2849 70.328C74.1719 70.328 74.0909 70.295 74.0469 70.225C73.3059 69.029 73.7119 68.194 78.9349 65.033C84.1579 61.871 87.8239 59.969 85.7389 57.699C85.4989 57.437 85.1589 57.321 84.7459 57.321C81.5749 57.322 74.0829 64.14 74.0829 64.14C74.0829 64.14 72.0609 66.243 70.8379 66.243C70.5569 66.243 70.3179 66.132 70.1559 65.858C69.2889 64.396 78.2089 57.636 78.7119 54.847C79.0529 52.957 78.4729 52 77.3999 52Z" fill="#FF9D0B"></path><path class="origin-bottom-left transition-transform group-hover:rotate-6" d="M56.3271 76.6898C53.5751 72.6538 53.7701 69.6248 57.5461 65.8508C61.3221 62.0778 63.5201 56.5578 63.5201 56.5578C63.5201 56.5578 64.3411 53.3518 66.2111 53.6468C68.0811 53.9418 69.4541 58.7328 65.5371 61.6638C61.6201 64.5938 66.3171 66.5848 67.8241 63.8328C69.3311 61.0808 73.4461 54.0058 75.5801 52.6528C77.7131 51.2998 79.2151 52.0578 78.7121 54.8468C78.2091 57.6358 69.2881 64.3958 70.1561 65.8588C71.0241 67.3208 74.0831 64.1398 74.0831 64.1398C74.0831 64.1398 83.6551 55.4288 85.7391 57.6988C87.8231 59.9688 84.1581 61.8708 78.9351 65.0328C73.7111 68.1938 73.3061 69.0288 74.0471 70.2248C74.7891 71.4208 86.3191 61.6998 87.4021 65.8208C88.4841 69.9418 75.6351 71.1378 76.4281 73.9828C77.2211 76.8288 85.4791 68.5978 87.1681 71.8048C88.8581 75.0128 75.5151 78.7818 75.4071 78.8098C71.0971 79.9278 60.1511 82.2968 56.3271 76.6898Z" fill="#FFD21E"></path></svg></a>
		<div class="max-md:mt-8! font-semibold text-black md:hidden">Website</div>

		<a class="hover:underline" href="/models">Models</a>
		<a class="hover:underline" href="/datasets">Datasets</a>
		<a class="hover:underline" href="/spaces">Spaces</a>
		<a class="hover:underline" href="/pricing">Pricing</a>
		<a class="hover:underline" href="/docs">Docs</a></nav></footer></div>

		<script>
			import("\/front\/build\/kube-68d7aa0\/index.js");
			window.moonSha = "kube-68d7aa0\/";
			window.__hf_deferred = {};
		</script>

		<!-- Stripe -->
		<script>
			if (["hf.co", "huggingface.co"].includes(window.location.hostname)) {
				const script = document.createElement("script");
				script.src = "https://js.stripe.com/v3/";
				script.async = true;
				document.head.appendChild(script);
			}
		</script>
	</body>
</html>
